<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>Human Genome Working Draft: Map Plots Index</TITLE>
	<LINK REL="STYLESHEET" HREF="http://genome.ucsc.edu/style/HGStyle.css">
	<BASE HREF="http://genome.ucsc.edu">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 >
<TH COLSPAN=2 ALIGN="left">	
    <IMG SRC="images/title.jpg">
	<!--<FONT FACE="Arial Black" SIZE=5 COLOR="#0033366">Human Genome Project Working Draft at UCSC</FONT>-->
</TH>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
   <CENTER>
	<TABLE BACKGROUND="images/hl_mid.jpg" BGCOLOR="253BDE" WIDTH=100% CELLSPACING=0 CELLPADDING=0 BORDER=0 HEIGHT="22"><TR><TD>
		<TABLE WIDTH=431 CELLSPACING=4 CELLPADDING=0 BORDER=0><TR>
		<TD WIDTH=1></TD>
		<TD WIDTH=75>
			<A HREF="index.html" onMouseOver="HOME.src='images/hl_home.jpg'" onMouseOut="HOME.src='images/h_home.jpg'">
			<IMG SRC="images/h_home.jpg" ALIGN="absmiddle" BORDER="0" NAME="HOME" ALT="Home" ></A></TD>

		<TD WIDTH=155>
			<A HREF="goldenPath/hgTracks.html" onMouseOver="BROWSER.src='images/hl_browser.jpg'" onMouseOut="BROWSER.src='images/h_browser.jpg'">
			<IMG SRC="images/h_browser.jpg" ALIGN="absmiddle" BORDER="0" NAME="BROWSER" ALT="&nbsp;&nbsp; Genome Browser"></A></TD>
			
		<TD WIDTH=100>
			<A HREF="cgi-bin/hgBlat?command=start" onMouseOver="SEARCH.src='images/hl_blat.jpg'" onMouseOut="SEARCH.src='images/h_blat.jpg'">
			<IMG SRC="images/h_blat.jpg" ALIGN="absmiddle" BORDER="0" NAME="SEARCH" ALT="&nbsp;&nbsp; BLAT Search"></A></TD>
			
		<TD WIDTH=100>
			<A HREF="FAQ.html" onMouseOver="GUIDE.src='images/hl_faq.jpg'" onMouseOut="GUIDE.src='images/h_faq.jpg'">
			<IMG SRC="images/h_faq.jpg" ALIGN="absmiddle" BORDER="0" NAME="GUIDE" ALT="&nbsp;&nbsp; FAQ"></A></TD>
		</TR></TABLE>
	 </TR></TABLE>
	 </CENTER>
    </TD>
</TR>

<!--LeftBar Links Are at bottom of page! (for viewing in LYNX! :) -->
<TR>
<!--Content Tables-->  
  <TD CELLPADDING=10>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="ffffff" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Comparison of the Working Draft with Other Maps </b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>

Here we show the positions of STS markers on the assembly of the working
draft genome sequence versus their positions in various maps.

<br><br><FONT COLOR="#006666" SIZE=4><b>
September 5th Freeze:</b></FONT>
<ul>
<li>
	<b><a href="http://genome.ucsc.edu/goldenPath/mapPlots/Sept5/RHgenetic/">Radiation
	Hybrid, Sex-averaged genetic, and YAC STS maps</a></b></li>
<li>
	<b><a href="http://genome.ucsc.edu/goldenPath/mapPlots/Sept5/cytogenetic/">Cytogenetic
	map</a></b></li>
</ul>

<br><FONT COLOR="#006666" SIZE=4><b>
October 7th Freeze:</b></FONT>
<ul>
<li>
	<b><a href="http://genome.ucsc.edu/goldenPath/mapPlots/Oct7/RHgenetic/">Radiation
	Hybrid, Sex-averaged genetic, and YAC STS maps</a></b></li>
<li>
	<b>Separate male and female genetic maps (<a href="http://genome.ucsc.edu/goldenPath/mapPlots/Oct7/male_female/index.html">Marshfield</a>)</b></li>
<li>
	<b><a href="http://genome.ucsc.edu/goldenPath/mapPlots/Oct7/cytogenetic/">Cytogenetic
	map</a></b></li>
</ul>

<br><FONT COLOR="#006666" SIZE=4><b>
December 12th Freeze:</b></FONT>
<ul>
<li>
	<b><a href="http://genome.ucsc.edu/goldenPath/mapPlots/Dec12/RHgenetic/">Radiation
	Hybrid, Sex-averaged genetic, and YAC STS maps</a></b></li>
<li>
	<b>Separate male and female genetic maps (<a href="http://genome.ucsc.edu/goldenPath/mapPlots/Dec12/male_female/index.html">Marshfield</a>)</b></li>
<li>
	<b><a href="http://genome.ucsc.edu/goldenPath/mapPlots/Dec12/cytogenetic/">Cytogenetic
	map</a></b></li>
</ul>

<br><FONT COLOR="#006666" SIZE=4><b>
April 1st Freeze:</b></FONT>
<ul>
<li>
	<b><a href="http://genome.ucsc.edu/goldenPath/mapPlots/Apr01/RHgenetic/">Radiation
	Hybrid, Sex-averaged genetic, and YAC STS maps</a></b></li>
<li>
	<b>Separate male and female genetic maps (<a href="http://genome.ucsc.edu/goldenPath/mapPlots/Apr01/male_female/index.html">Marshfield</a>)</b></li>
<li>
	<b><a href="http://genome.ucsc.edu/goldenPath/mapPlots/Apr01/cytogenetic/">Cytogenetic
	map</a></b></li>
</ul>
These plots are described more fully in "Initial sequencing and analysis
of the human genome," by the International Human Genome Sequencing Consortium,
<i>Nature</i>,
409: 860-921. Included are the <a href="ftp://ftp.genethon.fr/pub/Gmap/Mature-1995">Genethon
genetic map</a>, the <a href="http://research.marshfieldclinic.org/genetics/Map_Markers/maps/IndexMapFrames.html">Marshfield
genetic map</a>, the <a href="http://www.ncbi.nlm.nih.gov/genemap/">GeneMap99
radiation hybrid map</a>, the <a href="http://www-shgc.stanford.edu/Mapping/Marker/STSindex.html">G3
and TNG radiation hybrid maps</a>, the <a href="http://carbon.wi.mit.edu:8000/cgi-bin/contig/phys_map">Whitehead
YAC and RH maps</a>, and the <a href="http://fishfarm.fhcrc.org/bacresource/index.shtml">resource
of FISH-mapped BAC clones</a> (<a href="http://www.ncbi.nlm.nih.gov/genome/cyto/">NCBI
Clone Registry</a>). We thank the researchers who built these maps for
making their data available. The TNG map data is discussed in Olivier et
al., "A high resolution radiation hybrid map of the human genome draft
sequence," <i>Science</i>, to appear, and the cytogenetic data in "Integration
of cytogenetic landmarks into the draft sequence of the human genome,"
by the BAC Resource Consortium, <i>Nature</i>, Vol. 409. Data for sex-averaged
Genethon, sex-averaged Marshfield, GeneMap99 RH, G3 RH and Whitehead YAC
maps was first collected into a unified database by Greg Schuler at NCBI.
Prior to the December freeze, we also used his e-pcr program (very slightly
modified) to place the markers on the draft sequence, with the exception
of the FISH-mapped clones, which were placed by Wonhee Jang and Greg Schuler
at NCBI, with assistance from Arek Kasprzyk and Ewan Birney at EBI. Starting
with the December freeze, we began placing all markers and FISH-mapped
clones ourselves by matching the full STS sequence to the golden path whenever
possible, not just the primers. BAC ends and accession information were
also used to help place the FISH-mapped clones.
<h2>
User Guides:</h2>

<ul>
<li>
<b><a href="http://genome.ucsc.edu/goldenPath/mapPlots/RHguide.html">User
Guide for RH, genetic and YAC Map Comparison</a></b></li>

<li>
<b><a href="http://genome.ucsc.edu/goldenPath/mapPlots/cytoguide.html">User
Guide for Cytogenetic Map Comparison</a></b></li>
</ul>

</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>

</TD></TR></TABLE>
</BODY></HTML>
