<H2>Description</H2>
<P>
This track displays regions showing evidence for conservation with respect to mutations involving sequence insertions and deletions (indels).

These “indel-purified sequences” (IPSs) were obtained by comparing the predictions of a neutral model of indel evolution with data obtained from human, mouse and dog alignments.

The evidence for conservation is statistical, and each region is annotated with a posterior probability.  It may be interpreted as the probability that the segment shows the paucity of indels by selection, rather than by random chance.

Apart from the underlying alignment, this data is independent of the conservation of the nucleotide sequence itself.  Any inferred conservation of the sequence, e.g. as shown by phastCons, is therefore independent evidence for selection.  

It may happen that sequence is conserved with respect to indel mutations without concomitant evidence of conservation of the nucleotide sequence.  The opposite may also happen.


<H2>Display Conventions</H2>
The score (based on the false discovery rate (FDR)) is reflected in the 
bluescale density gradient coloring the track items. Lighter colours reflect a 
higher FDR.

<H2>Methods</H2>
<P>
In the absence of selection, indels have a certain predicted distribution over the genome.  The actual distribution shows an over-abundance of ungapped regions, due to selection purifying functional sequence from the deleterious effects of indels.  Neutrally evolving sequence, such as (by and large) ancestral repeats, show an exceedingly good fit to the neutral predictions.  This accurate fit allows the identification of a good proportion of conserved sequence at a relatively low false discovery rate (FDR).  For example, at an FDR of 10%, the predicted sensitivity is 75%.

Each identified indel-purified sequence (IPS) is annotated by two numbers: a false discovery rate (FDR), and a posterior probability (p).  

The FDR refers to a set of segments, not a given segment by itself.  In this case, it refers to the minimum FDR of all sets that include the segment of interest.  For example, a segment annotated with a 10% FDR belongs also belongs to a set with a 15% FDR, but not a set with a 5% FDR.

The posterior probability does refer to the single segment by itself.  It has a frequentist interpretation, namely, as the proportion of regions, annotated with the same posterior probability, that have been under purifying selection, rather than showing the observed lack of indels by random chance.

The data includes segments for a false-discovery rate of up to 50%.  The score directly reflects the FDR, through the following formula:

<BLOCKQUOTE><TT>
score = 90 / (FDR + 0.08)</TT></BLOCKQUOTE>

This maps FDR 1% (the most restrictive category) to 999, and FDR 10% to 500.

<H2>Verification</H2>
<P>
The neutral indel model was calibrated using ancestral repeats, against which it showed an excellent fit.  Among the identified IPSs at 10% FDR and predicted sensitivity of 75%, we found 75% of annotated protein-coding exons (weighted by length), and 75% of the 222 microRNAs that were annotated at the time.  Ancestral repeats were heavily depleted among the identified segments.

<H2>Credits</H2>
<P>
This data was generated by Gerton Lunter, MRC Functional Genetics Unit, 
University of Oxford, United Kingdom.
 
<H2>References</H2>
<P>
Lunter G, Ponting, CP, Hein J.  Genome-wide identification of human functional 
DNA using a neutral indel model. PLoS Comp Biol. 2006 Jan;2(1):e5.

The data may also be browsed here: http://wwwfgu.anat.ox.ac.uk/~gerton/IPS/


