<H2>Overview</H2>
<P>
This super track groups related tracks from Uppsala University, Sweden ChIP-chip
experiments.
These tracks contain ChIP-chip data for transcription factors and acetylated 
histone H3 (H3ac) and H4 (H4ac) in HepG2 cell line. </P>
<P>
<H3>Uppsala ChIP</H3>
<P>
This track displays the results of ENCODE region-wide localization for three
transcription factors (HNF-3b, HNF-4a and USF-1) and acetylated
histone H3 (H3ac). The heights of the peaks in the graphical display indicate
the ratio of enriched non-amplified DNA to input DNA. </P>
<P>
The raw data for this track are available at
<A HREF="http://www.ebi.ac.uk/arrayexpress" TARGET=_blank>EBI ArrayExpress</A> 
as experiment
<A HREF="http://www.ebi.ac.uk/arrayexpress/#ae-browse/q=E-MEXP-452"
TARGET=_BLANK>E-MEXP-452</A>.</P>
<P>
<H3>Uppsala ChIP Butyrate</H3>
<P>
ENCODE regions were investigated by ChIP-chip, analyzing both histone H3
acetylation (H3ac; H3 acetylated lysines 9 and14) and histone H4 acetylation
(H4ac; H4 acetylated lysined 5,8,12,16). This analysis was performed using
ChIP material obtained from cells that were either untreated or treated with
5mM Na-Butyrate for 12 hours. Na-Butyrate is a histone deacetylase inhibitor
(HDACi) that increases bulk levels of acetylated histones. Four tracks presented
in the genome browser represent the ChIP-chip signal obtained for either H3ac
or H4ac, using cells that were untreated or treated with butyrate: H3ac 0h,
H3ac 12h, H4ac 0h, H4ac 12h. Two additional tracks indicate those spots where
H3ac or H4ac levels are significantly changed by butyrate treatment.
</P>
<P>
The raw data for this track are available at
<A HREF="http://www.ebi.ac.uk/arrayexpress" TARGET=_blank>EBI ArrayExpress</A> 
as experiment
<A HREF="http://www.ebi.ac.uk/arrayexpress/#ae-browse/q=E-MEXP-693"
TARGET=_blank>E-MEXP-693</A>.</P>

<H2>Credits</H2>
<P>
These experiments were performed in the
<A HREF="http://www.genpat.uu.se/en/node1"
TARGET=_blank>Claes Wadelius lab</A>, Department of Genetics and Pathology,
Rudbeck Laboratory, Uppsala University. The statistical analysis was done at the
<A HREF="http://www.lcb.uu.se" TARGET=_blank>Linnaeus Centre for
Bioinformatics</A> at Uppsala University.  Microarrays were produced at the
<A HREF="http://www.sanger.ac.uk" TARGET=_blank>Sanger Institute</A>.
</P>

<H2>References</H2>
<P>
Ameur A, Yankovski V, Enroth S, Spjuth O, Komorowski J.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=16455749" 
TARGET=_blank>The LCB Data Warehouse</A>. 
<em>Bioinformatics</em>. 2006 Apr 15;22(8):1024-6. </P>
<P>
Rada-Iglesias A, Wallerman O, Koch C, Ameur A, Enroth S, Clelland G, Wester K,
Wilcox S, Dovey OM, Ellis PD <em>et al</em>.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=16221759" 
TARGET=_blank>Binding sites for metabolic disease related transcription factors 
inferred at base pair resolution by chromatin immunoprecipitation and genomic 
microarrays</A>. 
<em>Hum Mol Genet</em>. 2005 Nov 15;14(22):3435-47.</P>
<P>
Smyth GK
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=16646809" TARGET=_blank>
Linear models and empirical Bayes methods for assessing differential expression 
in microarray experiments</A>. 
<em>Stat Appl Genet Mol Biol.</em> 2004;3:Article3.</P>
<P>
Yang YH, Dudoit S, Luu P, Lin DM, Peng V, Ngai J, Speed TP.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Citation&list_uids=11842121" 
TARGET=_blank>Normalization for cDNA microarray data: a robust composite method 
addressing single and multiple slide systematic variation</A>. 
<em>Nucleic Acids Res</em>. 2002 Feb 15;30(4):e15. </P>
