<H2>Overview</H2>
This super-track groups related tracks of the ENCODE Extension data generated by Affymetrix.  There 
are two member tracks:
<UL>
<LI><B>Affymetrix ENCODE Extension Transcription Sites</B> the transcribed 
fragments (transfrags) based on the signal.</LI>
<LI><B>Affymetrix ENCODE Extension Transcription Signal</B> - RNA abundance signal.</LI>
</UL>

<H2>Methods</H2>
The data from replicate arrays were quantile-normalized (Bolstad et al., 2003) and all arrays were scaled to a median array intensity of 330. Using two different approaches: i) no sliding window ii)sliding 51 bp window centered on each probe, an estimate of RNA abundance (signal) was computed by calculating the median of all pairwise average PM-MM values, where PM is a perfect match and MM is a mismatch. Both Kapranov et al. (2002) and Cawley et al. (2004) are good references for the experimental methods; Cawley et al. also describes the analytical methods.

<H3>Verification</H3>
Single biological replicates were generated and hybridized to duplicate arrays (two technical replicates). Transcribed regions were generated from the composite signal track by merging genomic positions to which probes are mapped. This merging was based on a 5% false positive rate cutoff in negative bacterial controls, a maximum gap (MaxGap) of 25 base-pairs and minimum run (MinRun) of 25 base-pairs (see the Affy TransFrags track for the merged regions). 
 
<H2>Credits</H2>
These data were generated and analyzed by the collaboration of the following groups: the Tom Gingeras group at Affymetrix , Roderic Guigo group at Centre de Regulacio Genomica, , Alexandre Reymond group at  University of Lausanne and Stylianos Antonarakis group at University of Geneva. 
 
<H2>References</H2>
Please see the <a href="http://www.affymetrix.com/transcriptome">Affymetrix Transcriptome</a> site for a project overview and additional references to Affymetrix tiling array publications.
<P>
Bolstad, B. M., Irizarry, R. A., Astrand, M., and Speed, T. P. <a
href="http://bioinformatics.oupjournals.org/cgi/content/abstract/19/2/185">A
comparison of normalization methods for high density oligonucleotide
array data based on variance and bias</a>. Bioinformatics 19(2),
185-193 (2003).
<P>
Cawley, S., Bekiranov, S., Ng, H. H., Kapranov, P., Sekinger, E. A.,
Kampa, D., Piccolboni, A., Sementchenko, V., Cheng, J., Williams,
A. J., et al. <a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=14980218">Unbiased mapping of transcription factor binding sites along human chromosomes 21 and 22 points to widespread regulation of noncoding RNAs</a>. Cell 116(4), 499-509 (2004).
<P>
Kapranov, P., Cawley, S. E., Drenkow, J., Bekiranov, S., Strausberg,
R. L., Fodor, S. P., and Gingeras, T. R. <a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=11988577">Large-scale transcriptional activity in chromosomes 21 and 22</a>. Science 296(5569), 916-919 (2002).
