<H2>Overview</H2>
<P>
This super track groups related tracks from Stanford ChIP-chip and ChIP-seq 
experiments for multiple transcription
factors in multiple cell lines. Data are displayed as bound region and smoothed 
scores. </P>

<H3>Stanford ChIP, Stanford ChIP Score</H3>
<P>
These tracks display regions and smoothed (sliding-window mean) scores for
regions bound by Sp1 and Sp3 in the following
three cell lines, assayed by ChIP and microarray hybridization: </P>
<P>
<BLOCKQUOTE>
<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4>
  <TR><TH align=left>Cell Line</TH><TH align=left>Classification</TH><TH align=left>Isolated From</TH></TR>
  <TR><TD><A HREF="http://www.atcc.org/common/catalog/numSearch/numResults.cfm?atccNum=CCL-247" TARGET=_blank>HCT 116</A></TD><TD>colorectal carcinoma</TD><TD>colon</TD></TR>
  <TR><TD><A HREF="http://www.atcc.org/common/catalog/numSearch/numResults.cfm?atccNum=TIB-152" TARGET=_blank>Jurkat, Clone E6-1</A></TD><TD>acute T cell leukemia</TD><TD>T lymphocyte</TD></TR>
  <TR><TD><A HREF="http://www.atcc.org/common/catalog/numSearch/numResults.cfm?atccNum=CCL-243" TARGET=_blank>K-562</A></TD><TD>chronic myelogenous leukemia (CML)</TD><TD>bone marrow</TD></TR>
</TABLE>
</BLOCKQUOTE>
</P>

<H3>Stanford NRSF</H3>
<P>
This track shows 25mer sequence reads obtained via a ChIP-seq protocol
from an NRSF/REST-enriched ChIP sample and a companion control sample
of the same fixed chromatin without immuno-enrichment.   These sequence
reads were later analyzed using a peak locator (data to be available in the 
future) to identify NRSF-positive binding events.</P>
<P>
ChIP-seq is a new approach for investigating entire transcription factor-DNA 
interactomes. It is comprised of chromatin immunoprecipitation (ChIP) followed 
by single-molecule-based
sequencing (seq). ChIP-seq avoids the complications of array hybridization.</P>
<P>
Chromatin immunoprecipitation was applied to NRSF/REST loci identified by a
PSFM (position specific frequency matrix) computational screen.   NRSF/REST
is a zinc finger repressor that negatively regulates many neuronal genes in
stem and progenitor cells and in nonneuronal cell types such as the
Jurkat T cell line used for this track.</P>
<P>
NRSF (neuron-restrictive silencer factor), also known as REST (repressor 
element-1
silencing transcription factor), was chosen because prior studies provide
a large set of target genes.  In addition, the DNA motif bound by NRSF
(known as NRSE/RE1) is long (21bp) and well-specified, and there is a 
high-quality
monoclonal antibody that recognizes NRSF efficiently in ChIP experiments.</P>

<H2>Credits</H2>
<P>
The Sp1 and Sp3 data were generated in the
<A HREF="http://www-shgc.stanford.edu/myerslab/" TARGET="_blank">Richard M.
Myers lab</A> at Stanford University.</P>

<P>The NRSF data were generated by: <br>
<A HREF="http://www-shgc.stanford.edu/myerslab" TARGET=_blank>Myers Group</A>: 
David Johnson, Betsy Anton, Loan Nguyen, Cat Medina, Richard Myers.</P>
<P>
<A HREF="http://woldlab.caltech.edu/html/woldlab" TARGET=_blank>Wold Group</A>: 
Barbara Wold, Ali Mortazavi, Kenneth McCue.</P>
<P>
<A HREF="http://www.illumina.com" TARGET=_blank>Solexa/Illumina</A> 
Sequencing: Gary Schroth.</P>

<H2>References</H2>
<P>
Trinklein ND, Murray JI, Hartman SJ, Botstein D, Myers RM.
<A HREF="http://www.molbiolcell.org/cgi/content/full/15/3/1254"
TARGET=_blank>The role of heat shock transcription factor 1 in the
genome-wide regulation of the mammalian heat shock response</A>.
<em>Mol. Biol. Cell</em>. 2004 Mar;15(3):1254-61.</P> 
<P>
Fields S.
<A HREF="http://www.sciencemag.org/cgi/content/summary/316/5830/1441" 
TARGET=_blank>Site-seeing by sequencing</A>.
<em>Science</em>. 2007 Jun 8;316(5830):1441-2. </P>
<P>
Johnson DS, Mortazavi A, Myers RM, Wold B.
<A HREF="http://www.sciencemag.org/cgi/content/abstract/316/5830/1497" 
TARGET=_blank>Genome-wide mapping of in vivo protein-DNA interactions</A>.
<em>Science</em>. 2007 Jun 8;316(5830):1497-502. </P>
<P>
Mortazavi A, Thompson EC, Garcia ST, Myers RM, Wold B.
<A HREF="http://www.genome.org/cgi/content/abstract/16/10/1208" 
TARGET=_blank>Comparative genomics modeling of the NRSF/REST repressor network:
from single conserved sites to genome-wide repertoire</A>.
<em>Genome Res</em>. 2006 Oct;16(10):1208-21.</P>
<P>
Schoenherr CJ, Paquette AJ, Anderson DJ.
<A HREF="http://www.pnas.org/cgi/content/abstract/93/18/9881" 
TARGET=_blank>Identification of potential target genes for the 
neuron-restrictive silencer factor</A>.
<em>Proc Natl Acad Sci U S A</em>. 1996 Sep 3;93(18):9881-6. </P>
<P>
Mikkelsen TS, Ku M, Jaffe DB, Issac B, Lieberman E, Giannoukos G, Alvarez P, 
Brockman W, Kim TK, Koche RP <em> et al</em>.
<A HREF="http://www.nature.com/nature/journal/vaop/ncurrent/abs/nature06008.html" TARGET=_blank>
Genome-wide maps of chromatin state in pluripotent and lineage-committed cells.</A>
<em>Nature</em>. 2007 Jul 5;448(7149). </P>

