<H2>Description</H2>
<P>
This super track groups related tracks from Sanger ChIP-chip experiments.
These tracks contain ChIP-chip data for H3 and H4 histones
in multiple cell lines. Data are displayed as signals as well as hits and peak centers
identified by hidden Markov model (HMM) analysis.
</P>
<P>
1) Sanger ChIP <br>
2) Sanger ChIP Hits <br>
ENCODE region-wide location analysis of H3 and H4 histones
was conducted employing ChIP-chip using chromatin extracted from
<A HREF="http://ccr.coriell.org/nigms/nigms_cgi/display.cgi?GM06990"
	TARGET=_blank>GM06990</A> (lymphoblastoid),
   <A href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=789258&dopt=Abstract"
	TARGET=_blank>K562</A> (myeloid leukemia-derived),
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=13286432&query_hl=3" TARGET=_BLANK>
HeLaS3</A> (cervix carcinoma),
<A HREF="http://www.biotech.ist.unige.it/cldb/cl1670.html" TARGET=_BLANK>
HFL-1</A> (embryonic lung fibroblast),
<A HREF="http://www.biotech.ist.unige.it/cldb/cl3531.html" TARGET=_BLANK>
MOLT-4</A> (lymphoblastic leukemia), and
PTR8 cells.
Experiments were conducted with antibodies to the following histones:
<A HREF="http://www.abcam.com/index.html?datasheet=8895"
    TARGET=_blank>H3K4me1</A>,
<A HREF="http://www.abcam.com/index.html?datasheet=7766"
    TARGET=_blank>H3K4me2</A>,
<A HREF="http://www.abcam.com/index.html?datasheet=8580"
    TARGET=_blank>H3K4me3</A>,
<A HREF="http://www.upstate.com/browse/productdetail.asp?ProductId=07-523"
    TARGET=_blank>H3K9me3</A>,
<A HREF="http://www.upstate.com/browse/productdetail.asp?ProductId=05-851"
    TARGET=_blank>H3K27me3</A>,
<A HREF="http://www.abcam.com/index.html?datasheet=9050"
    TARGET=_blank>H3K36me3</A>,
<A HREF="http://www.abcam.com/index.html?datasheet=2621"
    TARGET=_blank>H3K79me3</A>,
<A HREF="http://www.upstate.com/browse/productdetail.q.productid.e.06-599"
    TARGET=_blank>H3ac</A>,
<A HREF="http://www.upstate.com/browse/productdetail.q.productid.e.06-866"
    TARGET=_blank>H4ac</A>, and
<A HREF="http://www.scbt.com/product.php?datasheet=15914"
    TARGET=_blank>CTCF</A>.
</P>

<P>
Histone methylation and acetylation serves as a stable genomic imprint
that regulates gene expression and other epigenetic
phenomena.  These histones are found in transcriptionally active domains
called euchromatin.</P>

<P>
The Hits track displays hit regions and peak centers for Sanger
ChIP-chip data, as identified by hidden Markov model (HMM) analysis.</P>


<H2>Credits</H2>
<P>
The data were generated by the
<A HREF="http://www.sanger.ac.uk/PostGenomics/encode/investigators.shtml"
TARGET=_blank>ENCODE investigators</A> at the
<A HREF="http://www.sanger.ac.uk/" TARGET=_blank>Wellcome Trust Sanger
Institute</a>, Hinxton, UK.</P>
<P>
Contacts:
<A HREF="mailto:&#105;&#100;l&#64;&#115;&#97;&#110;&#103;&#101;&#114;.a&#99;.&#117;k">Ian Dunham</A> and
<A HREF="mailto:&#99;&#109;k&#64;&#115;&#97;&#110;g&#101;&#114;.&#97;&#99;.u&#107;">Christoph Koch</A>. </P>
<P>
The HMM analysis was performed at the EBI by
<A HREF="mailto:&#102;l&#105;&#99;&#101;k&#64;&#101;&#98;&#105;.&#97;&#99;.u&#107;">Paul Flicek</A>.</P>
<P>
Raw data may be downloaded from the Sanger Institute website at
<A HREF="ftp://ftp.sanger.ac.uk/pub/encode" TARGET=_blank>ftp://ftp.sanger.ac.uk/pub/encode</A>.</P>

<H2>References</H2>
