<H2>Description</H2>
<P>
This super track groups related tracks from Affymetrix polyA+ RNA transcriptome analysis.
These tracks contain transcriptome data in multiple cell lines, with multiple time points
after drug treatment. Data are displayed
as signals (transcript abundance) and transfrags (sites of transcription).
</P>
<P>
1) Affy RNA Signal <br>
2) Affy Transfrag <br>
These tracks shows an estimate of RNA abundance (transcription) and the location of
sites showing transcriptionfor all ENCODE regions for several cell types.
Retinoic acid-stimulated HL-60 cells were harvested after 0, 2, 8, and 32 hours.
Purified cytosolic polyA+ RNA from unstimulated GM06990 and HeLa cells,
as well as purified polyA+ RNA from the RA-stimulated HL-60 samples,
was hybridized to Affymetrix ENCODE oligonucleotide
tiling arrays, which have 25-mer probes tiled every 22 bp on
average in the non-repetitive ENCODE regions.
Composite signals are shown in
separate subtracks for each cell type and for each of the four
timepoints for RA-stimulated HL-60.</P>
<P>
Data for all biological replicates can be downloaded from Affymetrix in
<A HREF="http://transcriptome.affymetrix.com/download/ENCODE/affy_oct1/RNA/wig/HeLa"
TARGET=_BLANK>wiggle</A>,
<A HREF="http://transcriptome.affymetrix.com/download/ENCODE/affy_oct1/RNA/cel_data/HeLa/"
TARGET=_BLANK>cel</A>, and
<A HREF="http://transcriptome.affymetrix.com/download/ENCODE/affy_oct1/RNA/soft/HeLa/"
TARGET=_BLANK>soft</A> formats.</P>


<H2>Credits</H2>
<P>
These data were generated and analyzed by the Gingeras/Struhl
collaboration with the Tom Gingeras group at
<A HREF="http://www.affymetrix.com/" TARGET=_blank>Affymetrix</A> and the
<A HREF="http://www.hms.harvard.edu/dms/bbs/fac/struhl.html"
TARGET=_blank>Kevin Struhl</A> group at Harvard Medical School.</P>


<H2>References</H2>
<P>
Please see the
<A HREF="http://www.affymetrix.com/transcriptome/index.affx"
TARGET=_blank>Affymetrix Transcriptome</A> site for a project overview and
additional references to Affymetrix tiling array publications.</P>

<P>
Bolstad BM, Irizarry RA, Astrand M, and Speed TP.
<A HREF="http://bioinformatics.oupjournals.org/cgi/content/abstract/19/2/185" TARGET=_blank>A comparison
of normalization methods for high density oligonucleotide array data based
on variance and bias</A>.
<I>Bioinformatics</I> <B>19</B>(2), 185-193 (2003).</P>

<P>
Cawley S, Bekiranov S, Ng HH, Kapranov P, Sekinger EA,
Kampa D, Piccolboni A, Sementchenko V, Cheng J,
Williams AJ <em>et al</em>.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=14980218" TARGET=_blank>Unbiased mapping of
transcription factor binding sites along human chromosomes 21 and 22 points
to widespread regulation of noncoding RNAs</A>.
<I>Cell</I> <B>116</B>(4), 499-509 (2004).</P>

<P>
Kapranov P, Cawley SE, Drenkow J, Bekiranov S, Strausberg RL, Fodor SP, Gingeras TR.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=11988577"
TARGET=_blank>Large-scale transcriptional activity in chromosomes 21 and
22</A>.
<em>Science</em> <B>296</B>(5569), 916-919 (2002).</P>

