<H2>Overview</H2>
<P>
This super track groups related tracks from Yale Transcript Map analysis.
These tracks contain transcriptome data of different
cell lines and biological samples as well as analysis of Transcriptionally
Active Regions (TARs).
</P>
<H3>Yale MAS RNA,  Yale MAS TAR</H3>
<P>
These tracks show the forward (+) and reverse (-) strand transcript map of
intensity scores (estimating RNA abundance) and transcriptionally-active regions
(TARs)/transcribed fragments
(transfrags) for human NB4 cell total RNA,
and human placental Poly(A)+ RNA, hybridized
to the Yale MAS (Maskless Array Synthesizer) ENCODE oligonucleotide
microarray, transcription mapping design #1. This array has 36-mer
oligonucleotide probes approximately every 36 bp (<i>i.e.</i>
end-to-end) covering all the non-repetitive DNA sequence of the ENCODE
regions ENm001-ENm012. See NCBI
<A HREF="http://www.ncbi.nlm.nih.gov/projects/geo/query/acc.cgi?acc=GPL2105"
TARGET=_blank>GEO GPL2105</A> for details of this array design. </P>
<P>
This transcript map is a combined signal from three biological replicates,
each with at least two technical replicates. Arrays were hybridized using
either the standard Nimblegen protocol or the protocol described in Bertone
<em>et al</em>. (2004). The label of each subtrack in this annotation
indicates the specific protocol used for that particular data set.</P>

<H3>Yale RNA, Yale TAR</H3>
<P>
These tracks show the transcript map of signal intensity (estimating RNA
abundance) and transcriptionally active regions
(TARs)/transcribed fragments (transfrags) for the following, hybridized to the 
Affymetrix ENCODE oligonucleotide microarray:
<UL>
<LI>human neutrophil (PMN) total RNA (10 biological samples from different 
individuals)
<LI>human placental Poly(A)+ RNA (3 biological replicates)
<LI>total RNA from human NB4 cells (4 biological replicates), each sample
divided into three parts and treated as follows:  untreated, treated with
retinoic acid (RA), and treated with 12-O-tetradecanoylphorbol-13 acetate (TPA)
(three out of the four original samples). Total RNA was extracted from each
treated sample and applied to arrays in duplicate (2 technical replicates).
</UL></P>
<P>
The human NB4 cell can be made to differentiate towards either monocytes (by
treatment with TPA) or neutrophils (by treatment with RA). See Kluger
<em>et al</em>., 2004 in the References section for more details about the
differentiation of hematopoietic cells.</P>
<P>
This array has 25-mer oligonucleotide probes tiled
approximately every 22 bp, covering all the non-repetitive DNA sequence
of the ENCODE regions. The transcript map is a combined signal for both
strands of DNA. This is derived from the number of different biological
samples indicated above, each with at least two technical replicates.</P>
<P>
See the following NCBI Gene Expression Omnibus (GEO) accessions for details
of experimental protocols:
<UL>
<LI> ENCODE Transcript Mapping for Human Neutrophil (PMN) Total RNA:
<A HREF="http://www.ncbi.nlm.nih.gov/projects/geo/query/acc.cgi?acc=GSE2678"
TARGET=_blank>GSE2678</A>
<LI> ENCODE Transcript Mapping for Human Placental Poly(A)+ RNA:
<A HREF="http://www.ncbi.nlm.nih.gov/projects/geo/query/acc.cgi?acc=GSE2671"
TARGET=_blank>GSE2671</A>
<LI> ENCODE Transcript Mapping for Total RNA from Human NB4 Cells untreated,
treated with RA, and treated with TPA:
<A HREF="http://www.ncbi.nlm.nih.gov/projects/geo/query/acc.cgi?acc=GSE2679"
TARGET=_blank>GSE2679</A>
</UL></P>

<H2>Credits</H2>
<H3>Yale MAS RNA, Yale MAS TAR</H3>
<P>
These data were generated and analyzed by the the labs of Michael Snyder,
Mark Gerstein and Sherman Weissman at Yale University.</P>

<H3>Yale RNA, Yale TAR</H3>
<P>
These data were generated and analyzed by the Yale/Affymetrix
collaboration between the labs of Michael Snyder, Mark Gerstein and
Sherman Weissman at Yale University and Tom Gingeras at Affymetrix.</P>

<H2>References</H2>
<P>
Bertone P, Stolc V, Royce TE, Rozowsky JS, Urban AE, Zhu X,
Rinn JL, Tongprasit W, Samanta M <em>et al</em>.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=15539566&query_hl=7"
TARGET=_blank>Global identification of human transcribed sequences with
genome tiling arrays</A>.
<em>Science</em>. 2004 Dec 24;306(5705):2242-6.</P>
<P>
Cheng J, Kapranov P, Drenkow J, Dike S, Brubaker S, Patel S,
Long J, Stern D, Tammana H <em>et al</em>.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=15790807&query_hl=5"
TARGET=_blank>Transcriptional maps of 10 human chromosomes at 5-nucleotide
resolution</A>.
<em>Science</em>. 2005 May 20;308(5725):1149-54.</P>
<P>
Kapranov P, Cawley SE, Drenkow J, Bekiranov S, Strausberg RL,
Fodor SP and Gingeras TR.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=11988577&query_hl=5"
TARGET=_blank>Large-scale transcriptional activity in chromosomes 21 and 22</A>.
<em>Science</em>. 2002 May 3;296(5569):916-9.</P>
<P>
Kluger Y, Tuck DP, Chang JT, Nakayama Y, Poddar R, Kohya N,
Lian Z, Ben Nasr A, Halaban HR <em>et al</em>.
<A HREF="http://www.pnas.org/cgi/content/full/101/17/6508"
TARGET=_blank>Lineage specificity of gene expression patterns</A>.
<em>Proc Natl Acad Sci U S A</em>. 2004 Apr 27;101(17):6508-13. </P>
<P>
Rinn JL, Euskirchen G, Bertone P, Martone R, Luscombe NM,
Hartman S, Harrison PM, Nelson FK, Miller P <em>et al.</em>
<A HREF="http://www.genesdev.org/cgi/content/full/17/4/529"
TARGET=_blank>The transcriptional activity of human Chromosome 22</A>.
<em>Genes Dev</em>. 2003 Feb 15;17(4):529-40. </P>
