<H2>Description</H2>
<P>
The UCSC Genes track shows gene predictions that are based on data
from RefSeq, Genbank, and UniProt. The gene predictions are a moderately
conservative set, requiring the support of one GenBank RNA sequence plus
at least one additional line of evidence. RefSeq RNAs are an exception to 
this, and require no additional evidence. The track includes both protein coding
and uncoding transcripts. Some of the uncoding transcripts may actually code
for protein, but the evidence for the associated protein is at best weak.
In general this gene set has about 10% more protein coding genes, about
5x as many uncoding genes, and about 2x as many splice varients as RefSeq.

<H2>Display Conventions and Configuration</H2>
<P>
This track in general follows the display conventions for
<A HREF="../goldenPath/help/hgTracksHelp.html#GeneDisplay">gene prediction
tracks</A>. For uncoding genes all of the exons are relatively thin, while
for coding genes the coding exons are relatively thick.
Black coloring indicates features that have corresponding entries
in the Protein Databank (PDB). Dark blue indicates that the transcript
has been reviewed or validated, either by the RefSeq or the SwissProt staff. 
Medium blue is used for other RefSeq transcripts.  Non-RefSeq transcripts are
colored light blue.
<P>
This track contains an optional codon coloring
feature that allows users to quickly validate and compare gene predictions.
To display codon colors, select the <em>genomic codons</em> option from the
<em>Color track by codons</em> pull-down menu. Click
<A HREF="../goldenPath/help/hgCodonColoring.html">here</A> for more
information about this feature. </P>

<H2>Methods</H2>
<P>
The UCSC Genes are built in a multistep pipeline.  The major steps are:
<UL>
<LI>Align RefSeq and GenBank RNA to genome with BLAT, keeping only the best 
alignments for each RNA, and discarding alignments of less than 98% identity.
<LI>Break up alignments at non-intronic gaps, and throw out small isolated fragments.
<LI>Create a splicing graph for each set of overlapping alignments. This graph has
an edge for each exon or intron, and a vertex for each splice site, start, and end.
Each RNA that contributes to an edge is kept as evidence for that edge.
<LI>Create a similar splicing graph in the mouse based on mouse RNA and ESTs. If
the mouse graph has an edge that is orthologous an edge in the human graph, add that
to the evidence for the human edge.
<LI>If an edge in the splicing graph is supported by two or more human ESTs, add that
as evidence for the edge.
<LI>Traverse the graph to generate all unique transcripts. The traversal is guided by
the initial RNAs to avoid a combinatorical explosion in alternative splicing. Only
transcripts that include an edge supported by one additional line of evidence past the
RNA are output. For single exon genes either two RNAs or two additonal lines of evidence
are required. A RefSeq RNA however is by itself enough.
<LI>Generate protein predictions. For non-RefSeq transcripts we use  the txCdsPredict program 
to determine if the transcript is protein coding and if so where the start and stop codons are. 
The txCdsPredict weighs the length of the protein, the presence of a Kozak consensus sequence at 
the start codon, and the length of the orthologous predicted protein in other species as positive evidence.
As negative evidence it considers nonsense-mediated-decay, and start codons in any frame
upstream of the predicted start codon.  For RefSeq transcripts we use the RefSeq protein 
prediction.
<LI>Find corresponding UniProt protein if any.
</UL>


<H2>Credits</H2>
<P>
The Known Genes track was produced at UCSC using a computational pipeline
developed largely by Jim Kent.  It is based on data in
<A HREF="http://www.ncbi.nlm.nih.gov/RefSeq/" TARGET=_blank>NCBI RefSeq</A>,
<A HREF="http://www.expasy.org/sprot/" TARGET=_blank>UniProt</A> 
(including TrEMBL and TrEMBL-NEW) and 
<A HREF="http://www.ncbi.nlm.nih.gov/Genbank/index.html"
TARGET=_blank>GenBank</A>. Our thanks to the people running these databases
and to the scientists worldwide who have made contributions to them. </P>

<H2>Data Use Restrictions</H2>
<P>
The UniProt data have the following terms of use, UniProt copyright(c) 2002 - 
2004 UniProt consortium:</P>
<P>
For non-commercial use, all databases and documents in the UniProt FTP
directory may be copied and redistributed freely, without advance
permission, provided that this copyright statement is reproduced with
each copy.</P>
<P>
For commercial use, all databases and documents in the UniProt FTP
directory except the files
<UL>
<LI>ftp://ftp.uniprot.org/pub/databases/uniprot/knowledgebase/uniprot_sprot.dat.gz
<LI>ftp://ftp.uniprot.org/pub/databases/uniprot/knowledgebase/uniprot_sprot.xml.gz
</UL>
may be copied and redistributed freely, without advance permission,
provided that this copyright statement is reproduced with each copy.
More information for commercial users can be found 
<A HREF="http://www.expasy.org/announce/sp_98.html" TARGET=_blank>here</A>.
<P>
From January 1, 2005, all databases and documents in the UniProt FTP
directory may be copied and redistributed freely by all entities,
without advance permission, provided that this copyright statement is
reproduced with each copy.</P>

<H2>References</H2>
<P>
Hsu, F., Kent, W.J., Clawson, H., Kuhn, R.M., Diekhans, M., and Haussler, D. 
<A HREF="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btl048?ijkey=qbTw0578H4cXuFw&keytype=ref" 
TARGET=_blank>The UCSC Known Genes</A>. 
<em>Bioinformatics</em> <B>22</B>(9), 1036-46 (2006).
<P>
Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Wheeler DL. (2004)
<A HREF="http://nar.oupjournals.org/cgi/content/full/32/suppl_1/D23" TARGET=_blank>GenBank: update</A>. 
<em>Nucleic Acids Res.</em> 32 Database issue:D23-6.
<P>
Kent, W.J.
<A HREF="http://www.genome.org/cgi/content/abstract/12/4/656"
TARGET=_blank>BLAT - the BLAST-like alignment tool</A>.
<em>Genome Res.</em> <B>12</B>(4), 656-664 (2002).</P>

