<H2>Description</H2>
<P>
To obtain a more "genome-centric" view of OMIM and to increase its usefulness
to the community, Omicia has mapped OMIM variants onto the human genome
sequence as nucleotide changes leading to protein changes.  
There are 2 subtracks.  One has alleles that were mapped in an automated 
manner, and the other is selected genes where the alleles were hand 
mapped.
The Omicia collection exceeds 2,000 genes with documented associations
predisposing to diverse disease phenotypes.
<H2>References</H2>
<P>
<H2>Methods</H2>
<UL>
<LI>
Automated mapping
<P>
Often this is
nontrivial since protein sequences derived from the reference human genome
differ from those in early publications, or alternatively spliced
mRNAs have been annotated onto the genome, leading Omicia to implement 
sophisticated "shifting algorithms" to resolve discrepancies. Omicia annotes
disease-causing intronic variants that may result in cryptic slice sites
leading to exon skipping, intron retention, and premature termination events,
as well as short deletions and insertions in OMIM.  
Mutations are scored by how well they overlap with data from other sources,
including dbSNP, the Human Genome Mutation Database (HGMD) and SNPs mapped
to OMIM by dbSNP.
<LI>
Hand mapping
<P>
Approximately 50 of the genes most accessed at UCSC have been chosen for
hand mapping of alleles.
</UL>
<H2>Credits</H2>
Thanks to Martin Reese, Donna Hendrix, and others? from 
<A HREF="http://www.omicia.com">Omicia</A> for providing these data.

