<H2>Disclaimer</H2>
<P>
The Omicia OMIM track is intended for research purposes.  The data are
obtained from OMIM and mapped by computational methods and/or human annotation,
and may contain errors.
<H2>Description</H2>
<P>
This track contains the OMIM allelic variants mapped to the human genome
sequence.  To obtain a more "genome-centric" view of OMIM and increase 
its usefulness to the community, Omicia has mapped OMIM variants as 
nucleotide changes leading to protein
changes.  Specifically, OMIM allelic variants with headings that refer to
amino acid mutations, e.g., ARG123CYS, and allelic variants that refer to 
mutations within or relative to introns, e.g., IVS1AS G-A -1, are mapped.
<P>
There are two subtracks:  one track, produced for all genes with mappable
allelic variant data within OMIM, where the data are mapped by computational
methods; a second track, determined for approximately 50 genes, where the 
data from the computational mapping method are reviewed and confirmed for 
accuracy, and for which
additional allelic variants that could not be mapped computationally are
mapped by hand.
<H2>Display Conventions</H2>
<P>
The track is color-coded by score.  (See Methods for details on scoring.)
<UL>
<LI>High confidence (score &gt; 250): green
<LI>Medium confidence (150 &gt; score &lt; 249): yellow
<LI>Low confidence (score &lt; 149): black
</UL>
Validation codes represent specific features of the score.
<UL>
<LI>d: overlap with dbSNP
<LI>D: overlap with a SNP within dbSNP that was annotated as this OMIM allele by
the NCBI.
<LI>o: overlap with other data sources
<LI>s: required a shifting for the allelic variants to align with the reference
protein sequence.
<LI>m: a "multi-path" mutation.  For some amino acid mutations, there can be
more than
one single mutation resulting in specified codon mutation.  An example is the
codon TAC for tyrosine can mutate by a single nucleotide change to a termination
codon either as TAG or TAA.
</UL>

<H2>Methods</H2>
<UL>
<LI>
Automated mapping
<P>
Often this is non-trivial since protein sequences derived from the reference
human genome differ from those in early publications.  Omicia has employed a
shifting algorithm to resolve discrepancies, whereby a best alignment is found
between the reference sequence and the data available directly from OMIM.  
Mutations are scored
by how well they overlap with data from other sources, namely dbSNP, the
Human Genome Mutation Database (HGMD), and SNPs mapped to OMIM by dbSNP, as well
as additional data sources when available.  Mapped variants that require
shifting have reduced scores based on the quality of the shift.
<LI>
Hand curation
<P>
Approximately 50 genes have been selected for hand curation of alleles.
Omicia selected these genes with the assistance of UCSC, and included 54 genes
frequently queried over a six month period spanning 2006-2007.  Additional 
genes were selected based on feedback from the community. The hand curation 
focused on confirming the accuracy of the computationally-mapped allelic 
variants and mapping the allelic
variants that could not be mapped by computational methods, including nucleotide
changes in non-coding regions, small deletions and insertions, and rearrangements.
</UL>
<H2>References</H2>
<P>
OMIM<BR>
Paper (submitted to )

<H2>Credits</H2>
Donna K. Hendrix, Andrew MacBride, Nomi L. Harris, Fidel Salas and Martin
Reese at <A HREF="http://www.omicia.com/" TARGET=_blank>Omicia</A>. <BR>
This project was funded by NHGRI 1R44 HG002993
