<H2>Description</H2>
<P>
This track shows measures of evolutionary conservation 
generated using the <em>phyloP</em> (Phylogenetic P-Values)
program from the  <A HREF="http://compgen.bscb.cornell.edu/phast">
PHAST package
</A>
developed by
<A HREF="http://compgen.bscb.cornell.edu/~acs/software.html" TARGET=_blank>
Adam Siepel</A> at Cornell University.
Two measurements are provided: 
conservation across 28 species, and an alternative measurement 
restricted to the placental mammal subset (17 species plus human) 
of the multiple alignment.
</P>
<P>
See the <em>Conservation</em> track description for information about
the multiple alignments used as the basis of these conservation measurements.
</P>

<H2>Display Conventions and Configuration</H2>
<P>
The track configuration options allow the user to display either
the vertebrate or placental mammal conservation scores, or both
simultaneously.
In full and pack display modes, conservation scores are displayed as a
<EM>wiggle track</EM> (histogram) in which the height reflects the 
size of the score. 
The conservation wiggles can be configured in a variety of ways to 
highlight different aspects of the displayed information. 
Click the <A HREF="../goldenPath/help/hgWiggleTrackHelp.html" 
TARGET=_blank>Graph configuration help</A> link for an explanation 
of the configuration options.</P>

<H2>Methods</H2>
<P> 
Conservation scoring was performed using the <em>phyloP</em> program
from the PHAST package.
PhyloP is a general method for computing p-values of conservation 
by comparing estimated numbers of substitutions along the 
branches of a phylogeny with the distribution expected 
under neutral evolution (Siepel, Pollard, and Haussler, 2006).  
Here it was used to produce separate scores at each 
base (--wig-scores option), considering all branches of
the phylogeny rather than a particular subtree or 
lineage (i.e., --subtree was not used).  
Alignment gaps were treated as missing data. 
</P>
<P>
PhyloP relies on a tree model containing the tree topology,
branch lengths representing evolutionary distance at neutrally
evolving sites, the background distribution of nucleotides, and a substitution
rate matrix.  The 
<A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg18/phastCons28way/28way.mod"
TARGET=_blank>vertebrate tree model</A> for this track was
generated using the phyloFit program from the PHAST package  
(REV model, EM algorithm, medium precision) using multiple alignments of 
4-fold degenerate sites extracted from the 28way alignment
(msa_view).  The 4d sites were derived from the 
<A HREF="http://genome.imim.es/gencode" TARGET=_blank>Oct 2005 Gencode 
Reference Gene set</A>,
which was filtered to select single-coverage long transcripts.  A second, 
<A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg18/phastCons28way/placental.mod" 
TARGET=_blank>mammalian tree model</A> including only placental mammals was used
to generate
the placental mammal conservation scoring.
</P>

<H2>Credits</H2>
<P> This track was created using phyloP, phyloFit, and other programs in PHAST by 
<A HREF="http://compgen.bscb.cornell.edu/~acs/software.html" 
TARGET=_blank>Adam Siepel's group</A> at Cornell University (original development
done at the Haussler lab at UCSC).
</P>
<P>The phylogenetic tree is based on Murphy <EM>et al</EM>. (2001) and general 
consensus in the vertebrate phylogeny community as of March 2007.
</P>

<H2>References</H2>
<P>
Siepel, A., Pollard, K.S. and Haussler,
D. <A href="http://compgen.bscb.cornell.edu/~acs/dless.pdf" TARGET=_blank>New methods for
detecting lineage-specific selection.</A> In Proc. 10th Int'l Conf. on
Research in Computational Molecular Biology (RECOMB '06) (2006).
</P>
<P>
Murphy WJ, Eizirik E, O'Brien SJ, Madsen O, Scally M, Douady CJ, Teeling E,
Ryder OA, Stanhope MJ, de Jong WW, Springer MS.
<A HREF="http://www.sciencemag.org/cgi/content/abstract/294/5550/2348"
TARGET=_blank>Resolution of the early placental mammal radiation using Bayesian phylogenetics</A>.
<EM>Science</EM>. 2001 Dec 14;294(5550):2348-51.
</P>
