<H2>General Description</H2>

<P>This track is produced as part of the ENCODE Transcriptome Project. 
It shows the expression of 5' cap analysis gene expression (CAGE) 
clusters that map to the genome on the "plus" or "minus" strand 
(<I>Plus-</I> and <I>Minus-</I> <I>Clusters</I> views) at specific locations.
Different 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=rnaExtract">RNA extracts</A> 
from different 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=localization">sub-cellular localizations</A> 
in different 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=cellType">cell lines</A>
are compared.
A cluster is defined as a region of overlapping read mappings. 
The expression is the number of reads making up the cluster divided by the 
total number of reads sequenced times 1 million.
</P>

<H2>Display Conventions and Configuration</H2>

<P>
This track is a multi-view composite track that contains multiple data types
(<EM>views</EM>).  For each view, there are multiple subtracks that
display individually on the browser.  Instructions for configuring multi-view
tracks are 
<A HREF="../../goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
</P>

<P>To show only selected views, uncheck the boxes next to the one that 
you wish to hide.</P>
 
<P>Color differences among the views are arbitrary. They provide a 
visual cue for 
distinguishing between the different cell types and compartments.
ENCODE tracks typically contain one or more of the following views:</P>

<DL>
<DT><I>Plus-</I> and <I>Minus-</I> <I>Clusters</I>
<DD>The clusters views show clusters of overlapping read mappings on the 
plus and minus genomic strands respectively.
<DT><I>Plus-</I> and <I>Minus-</I> <I>Raw Signal</I>
<DD>The Raw Signal views show the probe intensity based on all probes on the
plus and minus genomic strands respectively.
<DT><i>Alignments</i>
<DD>The alignments view shows the individual read mappings.
</DL>

<H2>Methods</H2>

<P>
Cells were grown according to the approved 
<A HREF="../../ENCODE/protocols/cell" TARGET=_BLANK>ENCODE cell culture protocols</A>.
RNA molecules longer than 200 nt 
and present in RNA population isolated from different subcellular compartments
were fractionated into polyA+ and polyA- 
fractions. 
The CAGE tags are sequenced from the 5' ends of cap trapped cDNAs produced using
RIKEN CAGE technology 
(Kodzius <EM>et al</EM>. 2006; Valen <EM>et al</EM>. 2009). 
To create the tag, a linker was attached to the 5'
end of polyA plus or minus reverse transcribed cDNA which were selected by cap
trapping (Carninci <EM>et al</EM>. 1996). The first 27 bp of
the cDNA were cleaved using class II restriction enzymes. And then linker was
attached to the 3' of restricted 27 bp cDNA. The result 96 bp were followed by
PCR amplification and sequencing. All tags were mapped to the human genome
(hg18) with the program nexalign (T. Lassmann manuscript in preparation). 
</P>

<P>We used an Illumina GA sequencer using the 36 nt reads. Please note 
that the CAGE tags are enzymatically cleaved about 27 nt from the ends of 
the mRNA, so the average length of the useful sequence is 27 (+/- few 
nucleotides, because the enzyme does not always cleave accurately). 
In any case we eliminate the linker sequences from the tags.
Note that these are short tags, so pair mate does not apply (only single 
reads).</P>

<P>Data for long poly(A)- RNA from the cytosol and nucleus of K562 cell lines, and 
from whole cell in prostate cells, were sequenced with ABI SOLiD technology. 
The remaining data was sequenced using Illumina/Solexa machines.</P>

<H2>Credits</H2>

<P>These data were generated and analyzed by Timo Lassmann, Phil Kapranov, 
Hazuki Takahashi, Yoshihide Hayashizaki, Tom Gingeras, and Piero Carninci.

<P>Contact: <A HREF="mailto:&#99;&#97;&#114;&#110;i&#110;&#99;i&#64;&#114;&#105;k&#101;&#110;.&#106;&#112;">
Piero Carninci</A>.
<!-- above address is carninci at riken.jp -->
</P>

<H2>References</H2>

<P>
Kodzius R, Kojima M, Nishiyori H, Nakamura M, Fukuda S, Tagami M, Sasaki D,
Imamura K, Kai C, Harbers M, <EM>et al</EM>. 
<A TARGET=_BLANK HREF="http://www.ncbi.nlm.nih.gov/pubmed/16489339">
CAGE: cap analysis of gene expression.</A>
<EM>Nat Methods</EM>. 2006 March 1; 3(3):211-222.
</P>

<P>
Valen E, Pascarella G, Chalk A, Maeda N, Kojima M, Kawazu C, Murata M, Nishiyori
H, Lazarevic D, Motti D, <EM>et al</EM>.
<A TARGET=_BLANK HREF="http://www.ncbi.nlm.nih.gov/pubmed/19074369">
Genome-wide detection and analysis of hippocampus core promoters using
DeepCAGE.</A>
<EM>Genome Res</EM>. 2009 February; 19(2):255-265.
</P>

<P>
Carninci P, Kvam C, Kitamura A, Ohsumi T, Okazaki Y, Itoh M, Kamiya M, 
Shibata K, Sasaki N, Izawa M, <EM>et al</EM>.
<A TARGET=_BLANK HREF="http://www.ncbi.nlm.nih.gov/pubmed/8938445">
High-efficiency full-length cDNA cloning by biotinylated CAP trapper.</A>
<EM>Genomics</EM>. 1996 November 1; 37(3):327-336.
</P>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior 
consent, submit publications that use an unpublished ENCODE dataset until 
nine months following the release of the dataset.  This date is listed in 
the <EM>Restricted Until</EM> column, above.  The full data release policy 
for ENCODE is available 
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P> 

