<H2>Description</H2>
<P>
These tracks display the level of sequence uniqueness of the reference hg18
genome.  They were generated using different window sizes and high signal
will be found in areas where the sequence is unique.

<!--
<H2> Display Conventions and Configuration </H2>
<P>
This track is a multi-view composite track that contains multiple data types
(<EM>views</EM>).  For each view, there are multiple subtracks that
display individually on the browser.  Instructions for configuring multi-view
tracks are <A HREF="../../goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
ENCODE tracks typically contain one or more of the following views:
<DL>
<DT><I>Peaks</I></DT><DD> Regions of signal enrichment based on processed data
(usually normalized data from pooled replicates). ENCODE Peaks tables contain
fields for statistical significance, including FDR
(<A HREF="../../goldenPath/help/qValue.html" TARGET=_BLANK>qValue</A>).</DD>
<DT><I>Signal</I></DT><DD>Density graph (wiggle) of signal enrichment based on
processed data.</DD>
<DT><I>Raw Signal</I></DT><DD>Density graph (wiggle) of signal from raw data
(single replicates, no control). For high-throughput sequencing data, this
reflects the aligned read density.</DD>
<DT><I>Alignments</I></DT><DD>Mappings of short reads to the genome.<DD>
</DL>
-->
<H2>Methods</H2>
<P>
The <em>Broad alignability</em> track displays whether a region is made up of
mostly unique or mostly non-unique sequence. To generate the track, every
36-mer in the genome was marker as "unique" if the most similar 36-mer
elsewhere  in the genome have at most 2 mismatches, and as "non-unique"
otherwise. Position X in the alignable track is marked by 1 if >50% of the
bases in [X-200,X+200] are "unique" and by 0 otherwise. Every point in the
alignable track has a corresponding position in  each of the ChIP signal
tracks.

<P>
The <em>Duke uniqueness</em> tracks display how unique is each sequence on the
positive strand starting at a particular base and of a particular length.
Thus, the 20bp track reflects the uniqueness of all 20 base sequences with
the score being assigned to the first base of the sequence. Scores are
normalized to between 0 and 1 with 1 representing a completely unique
sequence and 0 representing the sequence occurs >4 times in the genome
(excluding chrN_random and alternative haplotypes).  A score of 0.5
indicates the sequence occurs exactly twice, likewise 0.33 for three times
and 0.25 for four times.

<P>
The <em>UMass uniqueness</em> track displays a uniqueness signal for each
base which represents the sum of both plus and minus strand 15mer occurrences
of that particular 5'->3' (plus strand) sequence throughout the genome.  Scores
are normalized between 0 and 1 by calculating ( 1 / N ) where N is the number
of genome wide occurrences of the 15mer starting at position X.  A score of 1
represents a single genome wide occurrence of that 15mer.  A 0.5 would
represent either 2 plus strand occurrences or 1 plus and 1 minus strand
occurrence, and so on.  Ratios are rounded to 3 significant digits.  Therefore
a 0.000 would represent > 2000 occurrences.  A 0 is reserved for a given 15mer
that is either not assembled or contains at least one N at position X.

<H2>Credits</H2>
<P>
The <em>Broad alignability</em> track was created by the Broad Institute (contact:
<A HREF="mailto:&#110;&#115;h&#111;&#114;&#101;&#115;h&#64;&#98;r&#111;&#97;&#100;.&#109;it.&#101;&#100;&#117;"
title="&#110;&#115;h&#111;&#114;&#101;&#115;h&#64;&#98;r&#111;&#97;&#100;.&#109;it.&#101;&#100;&#117;" rel="nofollow" TARGET=_BLANK>
&#110;&#115;h&#111;&#114;&#101;&#115;h&#64;&#98;r&#111;&#97;&#100;.&#109;it.&#101;&#100;&#117;</a>).
Data generation and analysis was supported by funds from the NHGRI, the
Burroughs Wellcome Fund, Massachusetts General Hospital and the Broad Institute.
<P>
The <em>Duke uniqueness</em> tracks were created by Terry Furey (contact:
<A HREF="mailto:&#116;&#115;&#102;ur&#101;&#121;&#64;&#100;&#117;&#107;&#101;.e&#100;u"
title="&#116;&#115;&#102;ur&#101;&#121;&#64;&#100;&#117;&#107;&#101;.e&#100;u" rel="nofollow" TARGET=_BLANK>
&#116;&#115;&#102;ur&#101;&#121;&#64;&#100;&#117;&#107;&#101;.e&#100;u</A>)
<!-- above address is tsfurey at duke.edu -->
and Debbie Winter at Duke Univerisity's
<A HREF="http://www.genome.duke.edu/index.php" TARGET=_blank>Institute for Genome Sciences & Policy (IGSP)</A>;
and Stefan Graf at the
<A HREF="http://www.ebi.ac.uk/" TARGET=_blank>European Bioinformatics Insitute (EBI)</A>.
We thank NHGRI for ENCODE funding support.
<P>
The <em>UMass uniqueness</em> track was created by Bryan Lajoie (contact:
<A HREF="mailto:&#98;&#114;y&#97;&#110;.l&#97;&#106;&#111;&#105;&#101;&#64;&#117;&#109;&#97;s&#115;&#109;&#101;&#100;.&#101;&#100;&#117;"
title="&#98;&#114;y&#97;&#110;.l&#97;&#106;&#111;&#105;&#101;&#64;&#117;&#109;&#97;s&#115;&#109;&#101;&#100;.&#101;&#100;&#117;" rel="nofollow" TARGET=_BLANK>
&#98;&#114;y&#97;&#110;.l&#97;&#106;&#111;&#105;&#101;&#64;&#117;&#109;&#97;s&#115;&#109;&#101;&#100;.&#101;&#100;&#117;</A>)
<!-- above address is bryan.lajoie at umassmed.edu -->
in <A HREF="http://www.umassmed.edu/pgfe/faculty/Dekker.cfm?start=0&" traget=_BLANK>Job Dekker's Lab</a>
at the University of Massachusetts Medical School.  Funding Support: NIH grant
HG003143 to JD.  Keck Distinguished Young Scholar Award to JD.  This track was
generated as part of the ENCODE project funded by the NHGRI.
</P>

<!--
<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column on the track configuration page and
the download page.  The full data release policy for ENCODE is available
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P>
-->