<h2>Description</h2>
<p>This super-track displays related tracks of chromatin state maps
generated by the Broad/MGH ENCODE group using ChIP-Seq.
</p>

<p>The ChIP-Seq method involves crosslinking histones and other DNA associated proteins to genomic
DNA within cells using formaldehyde. The crosslinked chromatin is
subsequently extracted, mechanically sheared, and immunoprecipitated using specific
antibodies. After reversal of crosslinks, the immunoprecipitated DNA
is sequenced and mapped to the human reference genome. The
relative enrichment of each antibody-target (epitope) across the genome is inferred from
the density of mapped fragments.</p>
<p>For each <b>antibody</b> and <b>celltype</b>, two subtracks are provided:</p>

<p><b>Signal:</b> These subtracks represent the number of
sequenced and uniquely aligned ChIP fragments across the human genome
at 25bp resolution.</p>
<p><b>Peaks:</b> These subtracks represent discrete
intervals predicted to be significantly enriched with ChIP fragments
using a statistical model. Each interval is associated with an
enrichment ratio and a p-value.</p>
<p>This group of tracks also contains a single special track called
<b>Alignability</b>, which marks unique positions in the
reference genome as 1s and non-unique positions as 0.</p>

<h2>Methods</h2>

<h3>Growth protocol</h3>
<p>Tier 1 (<b>K562</b>, <b>GM12878</b>) and Tier 2 (<b>HepG2</b>,

<b>HeLa-S3</b>, <b>HUVEC</b>, <b>Keratinocyte</b>) cells were cultured
according to shared ENCODE protocols.
</p>

<h3>Extraction protocol</h3>
<p>Cells were fixed in 1% formaldehyde and
resuspended in lysis buffer. Chromatin was sheared to
200-700bp using a Diagenode Bioruptor. Solubilized
chromatin was immunoprecipitated with antibodies against
<b>H3K4me3</b> (Abcam 8580), <b>H3K4me2</b> (Abcam 7766),

</p><p><b>H3K4me1</b> (Abcam 8895), <b>H3K9ac</b> (Abcam 8896),
<b>H3K27ac</b> (Abcam 4729), <b>H3K27me3</b> (Millipore 07-449),
<b>H3K36me3</b> (Abcam 9050), <b>H3K9me1</b> (Abcam 8896),

</p><p><b>H4K20me1</b> (Abcam 9051) or <b>Pol2(b)</b>
(Covance MMS-128P). Antibody-chromatin complexes were pulled-down using protein
A-sepharose (or anti-IgM-conjugated agarose for RNA polymerase II),
washed and then eluted. After cross-link reversal and proteinase K
treatment, immunoprecipitated DNA was extracted with
phenol-chloroform, ethanol precipitated, treated
with RNAse and purified. One to ten nanograms of DNA were
end-repaired, adapter-ligated and sequenced by Illumina Genome
Analyzers as recommended by the manufacturer.
</p>

<h3>Data processing</h3>
<p>Sequence reads from each IP experiment were aligned to the human
reference genome (hg18) using MAQ with default parameters, except
C=10 (maximum number of alignments/read &lt;= 10).
Fragment densities were computed by counting the number of reads
overlapping each position in the genome (counted as 1 from the start
of each alignment to 200 bp downstream and by 0.25 from 200 to 300 bp,
and displayed at 25bp resolution). Discrete intervals of ChIP-Seq
fragment enrichment were identified using a scan statistics approach,
under the assumption of uniform background signal.
</p>

<h2>Credits</h2>

<p>The ChIP-Seq data were generated at the
 <a href="http://www.broad.mit.edu" title="http://www.broad.mit.edu" rel="nofollow">
 Broad Institute</a> and in the
 <a href="http://www.massgeneral.org/pathology/staff/bernstein.htm" title="http://www.massgeneral.org/pathology/staff/bernstein.htm" rel="nofollow">
 Bradley E. Bernstein lab</a> at the Massachusetts General Hospital/Harvard Medical School.
</p>
<p>Data generation and analysis was supported by funds from the NHGRI, the
Burroughs Wellcome Fund, Massachusetts General Hospital and the Broad Institute.
</p>

<h2>References</h2>
<p>Mikkelsen TS, Ku M, Jaffe DB, Issac B, Lieberman E <em> et al</em>.
 <a href="http://www.nature.com/nature/journal/v448/n7153/abs/nature06008.html" title="http://www.nature.com/nature/journal/v448/n7153/abs/nature06008.html" rel="nofollow">
 Genome-wide maps of chromatin state in pluripotent and lineage-committed cells</a>.
 <em>Nature</em>. 2007 Aug 2;448, 553-60. </p>

<p>Bernstein BE, Mikkelsen TS, Xie X, Kamal M, Huebert DJ <em> et al</em>.
 <a href="http://www.cell.com/content/article/abstract?uid=PIIS0092867406003801" title="http://www.cell.com/content/article/abstract?uid=PIIS0092867406003801" rel="nofollow">
 A bivalent chromatin structure marks key developmental genes in embryonic stem cells</a>.
 <em>Cell</em>. 2006 Apr 21;125, 315-26. </p>

<p>Bernstein BE, Kamal M, Lindblad-Toh K, Bekiranov S, Bailey DK <em> et al</em>.
 <a href="http://www.cell.com/content/article/abstract?uid=PIIS0092867405000395" title="http://www.cell.com/content/article/abstract?uid=PIIS0092867405000395" rel="nofollow">
 Genomic maps and comparative analysis of histone modifications in human and mouse</a>.
 <em>Cell</em>. 2005 Jan 28;120, 169-81.
</p>
