<H2> Description </H2>

<P>These tracks show probable binding sites of the specified transcription 
factors (TFs) in the given cell types as determined by chromatin 
immunoprecipitation followed by high throughput sequencing (ChIP-Seq). 
Included for each cell type is the input signal, which represents the control 
condition where no antibody targeting was performed.
For each experiment (cell type vs. antibody) this track shows one or
more of the following:
<UL>
<LI><I>Peaks</I> that represent sites with the greatest evidence of transcription factor binding.
<LI><I>Signal</I> of enrichment for TF binding derived from processed data (pooled replicates
normalized to control).
<LI><I>Raw Signal</I> of aligned read density for an individual replicate.
</UL>
The sequence reads, quality scores, and alignment coordinates are available from
the <I>Downloads</I> link, above.
</P>

<H2> Display Conventions and Configuration </H2>
<P>
This track is a multi-view composite track that contains multiple data types
(<EM>views</EM>).  For each view, there are multiple subtracks that
display individually on the browser.  Instructions for configuring multi-view
tracks are <A HREF="../../goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
</P>
ENCODE tracks typically contain one or more of the following views:
<DL>
<DT><I>Peaks</I></DT><DD> Regions of signal enrichment based on processed data (usually normalized data from pooled replicates). ENCODE Peaks tables contain fields for statistical significance, including FDR (qValue).</DD>
<DT><I>Signal</I></DT><DD>Density graph (wiggle) of signal enrichment based on processed data.</DD>
<DT><I>Raw Signal</I></DT><DD>Density graph (wiggle) of signal from raw data (single replicates, no control). For *-seq experiments, 
this reflects the aligned read density.</DD>
<DT><I>Alignments</I></DT><DD>Mappings of short reads to the genome.<DD>
</DL>

<H2> Methods </H2>
<P>
Cell preparations were similar to those previously published 
(Euskirchen <EM>et al.</EM>, 2007) with the exceptions that the cells 
were unstimulated and sodium orthovanadate was omitted from the buffers.
</P>

<H3> Extraction protocol </H3>
<p>All cells were grown either: (PolII and input control) at the National Cell Culture Center (Minneapolis, MN )
in Joklik's modified MEM supplemented 5% Newborn Calf Serum to a density of 5 x10^5 cells/ml.; or (Stat1 and
input control) in Minimum Essential Medium modified for Suspension (SMEM, Invitrogen), supplemented with glutamine,
10% FBS, and antibiotics (penicillin-streptomycin) at 37&deg;C in 5% CO<sub>2</sub>, to a density of 5 x10^5 cells/ml and
then stimulated at these same growth conditions for 30 min with 5 ng/ml human recombinant
IFN-&gamma; (R&amp;D Systems #285-IF).
</p>
<p>
All cells were then were fixed with 1% formaldehyde at room temperature for 10 min. Fixation was
terminated with 125 mM glycine, and cells were washed twice in cold 1x PBS, and then swelled on ice in
hypotonic lysis buffer (20 mM Hepes, pH 7.9, 10 mM KCl, 1 mM EDTA, pH 8, 10% glycerol, 1 mM DTT, 0.5 mM
PMSF, 0.1 mM sodium orthovanadate and Roche complete protease inhibitors). Cell lysates were homogenized
with 30 strokes in a Dounce homogenizer (Kontes, pestle B). Nuclear pellets were collected and lysed in
1x RIPA buffer (10 mM Tris-Cl, pH 8.0, 140 mM NaCl, 1% Triton X-100, 0.1% SDS, 1% deoxycholic acid, 0.5
mM PMSF, 1 mM DTT, 0.1 mM sodium orthovanadate and protease inhibitors). Chromatin was sheared with a
Branson 250 Sonifier (output setting 2, 100% duty cycle, seven 30-second pulses) to an average of 700
bp as verified on a 1.5% agarose gel.
</p><p>Clarified lysates for ChIP samples were agitated overnight at 4&deg;C with antibody. Following this
incubation, Protein A agarose beads (Upstate-Millipore #16-156) were added and allowed to bind for 60
min at 4&deg;C. Immunoprecipitates were washed three times in 1x RIPA, once in 1x PBS, and then eluted in
1xTE/1%SDS.  Clarified lysates for input DNA samples were stored overnight at 4&deg;C in parallel with ChIP
experiments.
</p>
<p>
For ChIP and input samples, crosslinks were reversed overnight at 65&deg;C. ChIP DNA was purified
by incubation with RNase A for 1 h at 37C, with 200 ug/ml Proteinase K for 2 h at 45&deg;C, phenol:chloroform:isoamyl
alcohol extraction, and precipitation with 0.1 volumes of 3M sodium acetate, 2 volumes of 100% ethanol
and 1.5 ul of pellet paint (Novagen).
</p>
<p>
ChIP-DNA sequencing libraries were prepared according to Illumina's instructions accompanying
the DNA Sample Kit (Part# 0801-0303).
</p>

<h3>Scoring of ChIP-Seq Data</h3>
<p>Only sequence reads (length ~30 bps) that align uniquely (using Eland) with at most two mismatches
against the hg18 build of the human genome are used. From the mapped tags a signal map of ChIP DNA
fragments (average fragment length ~ 200 bps) is constructed where the signal height is the number of
overlapping fragments at each nucleotide position in the genome. For each 1Mb segment of each chromosome
a peak height threshold is determined by requiring a false discovery rate &lt;= 0.05 when comparing the
number of peaks above threshold as compared the number obtained from multiple simulations of a random
null background with the same number of mapped reads (also accounting for the fraction of mapable bases
for sequence tags in that 1Mb segment). The number of mapped tags in a putative binding region is compared
against the normalized (normalized by correlating tag counts in genomic 10 Kb windows) number of mapped
tags in the same region from an input DNA control. Using a binomial test only regions that have a
p-value &lt;= 0.05 are considered to be significantly enriched compared to the input DNA control.
</p>

<H2> Credits </H2>
<P>These data were generated and analyzed by the labs of
<A HREF="http://www.yale.edu/snyder/" 
  class="external text" title="http://www.yale.edu/snyder/" 
  rel="nofollow" TARGET = _BLANK>Michael Snyder</a>,
<A HREF="http://bioinfo.mbb.yale.edu/" 
  class="external text" 
  title="http://bioinfo.mbb.yale.edu/" rel="nofollow" 
  TARGET = _BLANK>Mark Gerstein</a> and
<A HREF="http://info.med.yale.edu/bcmm/SMW/SMWhome2.html" 
  class="external text" title="http://info.med.yale.edu/bcmm/SMW/SMWhome2.html" 
  rel="nofollow" TARGET = _BLANK>Sherman Weissman</a> at Yale University;
<A HREF="http://www.genomecenter.ucdavis.edu/farnham/" 
  class="external text" title="http://www.genomecenter.ucdavis.edu/farnham/" 
  rel="nofollow" TARGET = _BLANK>Peggy Farnham</a> at UC Davis; and
<A HREF="http://struhl.med.harvard.edu/" 
  class="external text" title="http://struhl.med.harvard.edu/" 
  rel="nofollow" TARGET = _BLANK>Kevin Struhl</a> at Harvard.
&nbsp;&nbsp; Contact: <A HREF="mailto:datasubmission@gersteinlab.org" 
  title="datasubmission@gersteinlab.org" rel="nofollow" 
  TARGET = _BLANK>datasubmission@gersteinlab.org</a>.

</P>
<H2> References </H2>
<P>
Euskirchen G., Rozowsky J., Wei C., Zhang Z., Hartman S., Weissman S., 
Gerstein M., Ruan Y. and Snyder M.
<A HREF="http://www.genome.org/cgi/content/abstract/17/6/898"  
  CLASS="external text" title="http://www.genome.org/cgi/content/abstract/17/6/898" 
  rel="nofollow" TARGET = _BLANK>Mapping of transcription factor binding 
regions in mammalian cells by ChIP: comparison of array- and 
sequencing-based technologies.</A>
<EM>Genome Res.</EM> <B>17</B>:898-909 2007.
</P>

<P> 
Euskirchen G., Royce T.E., Bertone P., Martone R., Rinn J.L., Nelson F.K., 
Sayward F., Luscombe N.M., Miller P., Mark Gerstein, <EM>et al.</EM>
 <A HREF="http://mcb.asm.org/cgi/content/full/24/9/3804?view=long&amp;pmid=15082775" 
  class="external text" 
  title="http://mcb.asm.org/cgi/content/full/24/9/3804?view=long&amp;pmid=15082775" 
  rel="nofollow" TARGET = _BLANK>CREB binds to multiple loci on human chromosome 22.</a>
 <em>Mol Cell Biol.</em> <b>24</b>(9):3804-14 (2004).
</P><P>

Martone R., Euskirchen G., Bertone P., Hartman S., Royce T.E., Luscombe N.M., 
Rinn J.L., Nelson F.K., Miller P., Mark Gerstein, <EM>et al.</EM>
<A HREF="http://www.pnas.org/content/100/21/12247.full" 
  class="external text" title="http://www.pnas.org/content/100/21/12247.full" 
  rel="nofollow" TARGET = _BLANK> Distribution of NF-kappaB-binding sites across 
human chromosome 22.</A>
<EM>Proc Natl Acad Sci U S A.</EM> <B>100</B>(21):12247-52 (2003).
</P>

<P> 
Robertson, G., Hirst1, M., Bainbridge, M., Bilenky, M., Zhao, Y., Zeng, T., 
Euskirchen, G., Bernier, B., Varhol, R, Delaney, A., <EM>et al.</EM>
<A HREF="http://www.nature.com/doifinder/10.1038/nmeth1068" 
  class="external text" 
  title="http://www.nature.com/doifinder/10.1038/nmeth1068" rel="nofollow"
   TARGET = _BLANK> Genome-wide profiles of STAT1 DNA association using 
chromatin immunoprecipitation and massively parallel sequencing.</A>
<EM>Nat. Methods.</EM> <B>4</B>:651?657 (2007).
</P>

<P> 
Rozowsky J., Euskirchen G., Auerbach R., Zhang Z., Gibson T., Bjornson R., 
Carriero N., Snyder M., Gerstein M.B.
 PeakSeq: Systematic Scoring of ChIP-Seq Experiments Relative to Controls.
 <EM>Nature Biotech.</EM> (in Press).
</P>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior 
consent, submit publications that use an unpublished ENCODE dataset until 
nine months following the release of the dataset.  This date is listed in 
the <EM>Restricted Until</EM> column, above.  The full data release policy 
for ENCODE is available <A HREF="../ENCODE/terms.html">here</A>.</P> 

