<H2>Description</H2>

<P>This track shows the starts and ends of DNA determined by 
paired-end ditag (PET) sequencing. 
Sample genomic DNA was isolated, hydrosheared at a given size-range, then ligated with specific DNA linker sequence at both ends, followed by gel-selection of the desired size, e.g., 1 kb, 5 kb or 10 kb etc. respectively.  The DNA fragments modified with linker at both ends (e.g., 10 kb) were then circularized by ligation, followed by restriction digest with enzyme EcoP15I to generate DNA-PETs (25-bp tag from each end).  The PETs were then amplified (by PCR) and purified as complex DNA library for high throughput sequencing templates (SOLiD/Solexa platform).

See the Methods and References sections below for more details on PET sequencing
(Ng <EM>et al.</EM> 2006).
</P>

DNA-PETs were mapped onto reference genome.  The majority mapped on the same chromosome, in correct orientations and within expected distance span (e.g., ~10 kb span).  A small portion of misaligned PETs, called discordant PETs, mapped either too far from each other, had the wrong orientation or different chromosome indicating various genome structure variations between the sample and the reference genome eg. deletion, inversion, translocation, fusion etc.

<P>For different tissue source, growth condition and sub-compartment 
isolation, please refer to the general description for the ENCODE cell lines at ????.</P>

<P>In the graphical display, the ends are represented by blocks connected by a 
horizontal line. In full and packed display modes, the arrowheads on the 
horizontal line represent the strand, and an ID of the 
format <EM>XXXXX-N-M</EM> is shown to the left of each PET, where <EM>X</EM> 
is the unique ID for each PET, <EM>N</EM> indicates the number of mapping 
locations in the genome (1 for a single mapping location, 2 for two mapping 
locations, and so forth), and <EM>M</EM> is the number of PET sequences at 
this location. The total count of PET sequences mapped to the same locus but 
with slight nucleotide differences may reflect the expression level of the 
transcripts. PETs that mapped to multiple locations may represent low 
complexity or repetitive sequences.
</P>

<H2>Display Conventions and Configuration</H2>

<P></P>

<H2>Methods</H2>


<H2>Verification</H2>


<H2>Credits</H2>

<P>The GIS-PET libraries and sequence data for transcriptome analysis were produced at the 
<A HREF="http://www.gis.a-star.edu.sg/internet/site">Genome Institute of Singapore</A>. 
The data were mapped and analyzed by scientists from the Genome Institute of Singapore.
</P>

<H2>References</H2>

<P>Ng P, Tan JJ, Ooi HS, Lee YL, Chiu KP, Fullwood MJ, Srinivasan KG, Perbost C, 
Du L, Sung WK, et al., Multiplex sequencing of paired-end ditags (MS-PET): 
a strategy for the ultra-high-throughput analysis of transcriptomes and genomes. 
<EM>Nucleic Acids Res</EM>. 2006;34:e84.
</P>

