<H2>Description</H2>

<P>This track is produced as part of the ENCODE Transcriptome Project. 
It shows the starts and ends of mRNA transcripts determined by 
paired-end ditag (PET) sequencing using 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=rnaExtract">RNA extracts</A> 
from different 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=localization">sub-cellular
localizations</A> 
in different 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=cellType">cell lines</A>.
PETs are composed of 18 bases from both 
ends of a cDNA; 36 bp PETs were dimerized together and ligated by Illumina 
GA adaptor as sequencing templates. The diPET templates were amplified on 
the Illumina Cluster station and subjected to sequencing by synthesis. 
See the Methods and References sections below for more details on PET sequencing
(Ng <EM>et al.</EM> 2006).
</P>

<H2>Display Conventions and Configuration</H2>

<P>In the graphical display, the ends are represented by blocks connected by a 
horizontal line. In full and packed display modes, the arrowheads on the 
horizontal line represent the direction of transcription, and an ID of the 
format <EM>XXXXX-N-M</EM> is shown to the left of each PET, where <EM>X</EM> 
is the unique ID for each PET, <EM>N</EM> indicates the number of mapping 
locations in the genome (1 for a single mapping location, 2 for two mapping 
locations, and so forth), and <EM>M</EM> is the number of PET sequences at 
this location. The total count of PET sequences mapped to the same locus but 
with slight nucleotide differences may reflect the expression level of the 
transcripts. PETs that mapped to multiple locations may represent low 
complexity or repetitive sequences.
</P>

<P>To show only selected views, uncheck the boxes next to the one that 
you wish to hide.</P>
 
<P>Color differences among the views are arbitrary. They provide a 
visual cue for 
distinguishing between the different cell types and compartments.</P>

<DL>
<DT><i>Tags</i>
<DD>???.
<DT><i>Alignments</i>
<DD>The Alignments view shows alignment of individual PET sequences.
</DL>

<H2>Methods</H2>

<P>
Cells were grown according to the approved 
<A HREF="../../ENCODE/protocols/cell" TARGET=_BLANK>ENCODE cell culture
protocols</A>.
Two different protocols were used, one based on clone-free RNA-PET libraries
(unpublished) and the second with cloning-based libraries (Ng <i>et al.</i>
2005) described separately below.
</P>

<h3>Clone-free RNA-PET (50 bp reads, 25-25 bp tags)</h3>
<P>
<B>Method:</B>
The clone-free RNA-PET libraries were generated from polyA RNA samples,
constructed using a recent GIS-developed protocol (not published yet).  Total
RNA in good quality was used as starting material and purified through MACs
polyT column to obtain polyA.  Approximately 5-10 microgram enriched polyA RNA
was used for reverse transcription to convert polyA RNA to full length cDNA. The
obtained full length cDNA was modified and ligated with specific linker
sequences, followed by circularization through ligation to generate circular
cDNA molecules.  The 25bp tag from each end of the full length cDNA were
obtained by EcoP15I digestion. The resulting PETs was ligated with sequencing
adaptors at the both ends, then, amplified by PCR, further purified as complex
templates for paired end (PE) sequencing reads with Solexa or SOLiD platforms. 
</P>
<P>
<B>Data:</B>
The sequenced RNA-PETs are unified in 25/25bp length from each end of a cDNA.
The RNA-PETs were mapped onto reference genome allowing up to two mismatches.
Majority of PETs mapped on the known transcripts, or splice variants.  A small
portion of misaligned PETs, called discordant PETs, mapped either too far from
each other, had wrong orientation or in different chromosomes indicating exist
some transcription variations (e.g., caused by fusion, deletion, inversion,
translocation, or trans-splicing etc.). 
</P>

<h3>Cloning-based RNA-PET (34 bp reads, 18-16 bp tags)</h3>
<P>
<B>Method:</B>
The clone-based RNA-PET (GIS-PET) libraries were generated from polyA RNA
samples, constructed using the protocol described by Ng <i>et al.</i> (2005).
Total RNA in good quality was used as starting material and further purified
through MACs polyT column to enrich polyA and remove any contaminants (e.g.,
rRNA, tRNA, DNA, protein etc.).  Approximate 10 microgram polyA RNA was then
used for reverse transcription to convert polyA RNA to full length cDNA. The
obtained full length cDNA was modified with specific linker sequences, then,
ligated to a GIS-developed (pGIS4) vector to form a complex full length cDNA
library, which was made in E. coli. The plasmid DNA was then isolated from the
library, followed by MmeI digestion to generate 16/18bp ditags from each end of
the full length cDNA. The single ditags was then ligated further to form a diPET
structure, which is then ligated to Solexa adaptors for paired end (PE) high
throughput sequencing.   
</P>
<P>
<B>Data:</B>
The clone-based RNA-PETs are unified in 18/16bp length from 5' and 3'-end of the
cDNA. The redundant reads were filtered out initially and unique ones were
included for analysis. The RNA-PETs were mapped onto reference genome. Majority
of PETs mapped on the known transcripts or splice variants.  A small portion of
misaligned PETs, called discordant PETs, mapped either too far from each other,
had wrong orientation or in different chromosomes indicating exist some
transcription variations (caused by, e.g. fusion, deletion, inversion,
translocation, or trans-splicing etc.).
</P>

<P>PET sequences were mapped to the 
genome using the following specific criteria
(Ruan <EM>et al.</EM> 2007):

<UL>
<LI>A minimal continuous 16 bp match must exist for the 5' signature; the 3' 
signature must have a minimal continuous 14 bp match </LI> 
<LI>Both 5' and 3' signatures must be present on the same chromosome </LI>
<LI>Their 5' to 3' orientation must be correct (5' signature followed by 3' 
signature)</LI> 
<LI>The maximal genomic span of a PET genomic alignment must be less than one 
million bp</LI> 
</UL>

PETs mapping to 2-10 locations are also included and may represent duplicated
genes or pseudogenes in the genome.</P>

<H2>Verification</H2>

<P>To assess overall PET quality and mapping specificity, the top ten most 
abundant PET clusters that mapped to well-characterized known genes were 
examined. Over 99% of the PETs represented full-length transcripts, and the 
majority fell within ten bp of the known 5' and 3' boundaries of these 
transcripts. 
The PET mapping was further verified by confirming the existence of physical 
cDNA clones represented by the ditags. PCR primers were designed based on the 
PET sequences and amplified the corresponding cDNA inserts from the parental 
GIS flcDNA library for sequencing analysis. 
</P>

<H2>Credits</H2>

<P>The GIS-PET libraries and sequence data for transcriptome analysis were 
produced at the 
<A HREF="http://www.gis.a-star.edu.sg/internet/site">Genome Institute of
Singapore</A>. 
The data were mapped and analyzed by scientists from the Genome Institute of 
Singapore.
</P>

<H2>References</H2>

<P>Ng P, Tan JJ, Ooi HS, Lee YL, Chiu KP, Fullwood MJ, Srinivasan KG, Perbost C,
Du L, Sung WK, et al., Multiplex sequencing of paired-end ditags (MS-PET): 
a strategy for the ultra-high-throughput analysis of transcriptomes and genomes.
<EM>Nucleic Acids Res</EM>. 2006;34:e84.
</P>
<P>Ng P, Wei CL, Sung WK, Chiu KP, Lipovich L, Ang CC, Gupta S, Shahab A, 
Ridwan A, Wong CH, et al., Gene identification signature (GIS) analysis for 
transcriptome characterization and genome annotation. 
<EM>Nat Methods</EM>. 2005;2:105-111.
</P>
<P>Ruan Y, Ooi HS, Choo SW, Chiu KP, Zhao XD, Srinivasan KG, Yao F, Choo CY, 
Liu J, Ariyaratne P, et al., Fusion transcripts and transcribed retrotransposed 
loci discovered through comprehensive transcriptome analysis using Paired-End 
diTags (PETs). 
<EM>Genome Res</EM>. 2007;17:828-838.

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior 
consent, submit publications that use an unpublished ENCODE dataset until 
nine months following the release of the dataset.  This date is listed in 
the <EM>Restricted Until</EM> column, above.  The full data release policy 
for ENCODE is available 
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P> 
