<H2>General Description</H2>

<P>RNA-Seq is a method for mapping and quantifying the transcriptome of any organism that has a genomic DNA sequence assembly.
RNA-Seq is performed by reverse-transcribing an RNA sample into cDNA, followed by high throughput DNA sequencing, which 
was done here on an Illumina Genome Analyzer (GA2) (Mortazavi <EM>et al.</EM>, 2008).  
The transcriptome measurements shown on these tracks were performed on polyA selected RNA from total cellular RNA.  
Data have been produced in two formats: single reads, each of which comes from one end of a randomly primed cDNA molecule; 
and paired-end reads, which are obtained as pairs from both ends cDNAs resulting from random priming. 
The resulting sequence reads are then informatically mapped onto the genome sequence, which has been informatically
"expanded" to also contain known RNA splice junctions.  These mapped reads are then counted to determine their 
frequency of occurrence at known gene models.  Sequence reads that cluster at genome locations that lack an existing
transcript model are also identified informatically and they are quantified.  RNA-Seq is especially suited for giving 
information about RNA splicing patterns and for determining unequivocally the presence or absence of lower abundance class RNAs.  
As performed here, internal RNA standards are used to assist in quantification and to provide internal process controls.   
This RNA-Seq protocol does not specify the coding strand.  As a result, there will be ambiguity at loci where both strands are transcribed.  
The "randomly primed" reverse transcription is, apparently, not fully random.  
This is inferred from a sequence bias in the first residues of the read population, and this likely contributes to 
observed unevenness in sequence coverage across transcripts. </P>

<P>These tracks show 1x32 n.t. or 2x75 n.t. sequence reads of cDNA obtained from biological replicate samples 
(different culture plates) of the GM12878, K562 and HepG2 cell lines.  The 32 n.t. sequences were aligned to the
 human genome (hg18) and UCSC known-gene splice junctions using the bowtie software program (Langmead <EM>et al.</EM>, 2009). 
The 2x75 n.t. reads were mapped serially, first with bowtie against the genome and UCSC known-gene splice junctions. 
Bowtie-unmapped reads were then mapped using blat to find evidence of novel splicing, by requiring at least 10 bp 
on the short-side of the splice. </P>

<P>Reads that map uniquely in the genome are displayed in black.  Reads that map to low copy number multiple sites 
(2-10 occurrences in the genome) were allocated among the possible sites by ERANGE, as described previously 
(Mortazavi <EM>et al.</EM>, 2008).  These multi-reads are displayed in grey. In addition to tracks of 
individual reads, bedgraph files summarize the data in units of RPM (reads passing over a base 
normalized per million mapped reads in the measurement).  The RPM measure assists in visualizing 
the relative amount of a given transcript across multiple samples. </P>

<H2>Display Conventions and Configuration</H2>

<P>To show only selected subtracks, uncheck the boxes next to the tracks that you wish to hide.</P>
 
<P>Color differences among the subtracks are arbitrary. They provide a visual cue for 
distinguishing between the different cell types and compartments.</P>

<H2>Methods</H2>

<P> 
Cells were grown according to the approved
<A HREF="../../ENCODE/protocols/cell" TARGET=_BLANK>ENCODE cell culture protocols</A>.
The cells (either 2 X 10<SUP>7</SUP> or 4 X 10<SUP>7</SUP> cells &mdash; GM12878 and K562, 
and 8 X 10<SUP>7</SUP> cells HepG2) 
were lysed in either 4mls (GM12878 and K562) or 12 mls (HepG2) of RLT buffer (Qiagen RNEasy kit),
and processed on either 2 (GM12878 and K562) or 3 (HepG2) RNEasy midi columns according to the 
manufacturer's protocol, with the inclusion of the "on-column" DNAse digestion step to remove 
residual genomic DNA.</P>  

<P>75 &micro;gs of total RNA was selected twice with oligodT beads (Dynal) according to the manufacturer's 
protocol to isolate mRNA from each of the preparations.  100 ngs of mRNA was then processed according 
to the protocol in Mortazavi et al (2008), and prepared for sequencing on the Genome Analyzer flow cell
 according to the protocol for the ChIPSeq DNA genomic DNA kit (Illumina).  </P>

<P>Following alignment of the sequence reads to the genome assembly as described above, the sequence 
reads were further analyzed using the <a href="http://woldlab.caltech.edu/RNA-seq">ERANGE 3.0</a> software package,
 which quantifies the number of reads falling within the mapped boundaries of known transcripts from 
the Gencode annotations. ERANGE assigns both genomically unique reads and reads that occur in 2-10 
genomic locations for quantification.  ERANGE also contains a subroutine (RNAFAR) which allows the 
consolidation of reads that align close to, but outside the mapped borders of known transcripts, 
and the identification of novel transcribed regions of the genome using either a 20 kb radius for 
the 1x32 datasets or paired-end information for 2x75 datasets.</P>

<H2>Verification</H2>

<UL>
<LI>Known exon maps as displayed on the genome browser are confirmed by the alignment of sequence reads. </LI>
<LI>Known spliced exons are detected at the expected frequency for transcripts of given abundance.   </LI>
<LI>Linear range detection of spiked in RNA transcripts from Arabidopsis and phage lambda over 5 orders of magnitude.   </LI>
<LI>Endpoint RTPCR confirms presence of selected RNAFAR 3â€™UTR extensions.   </LI>
<LI>Correlation to published microarray data r = 0.62 </LI>
</UL>

<H2>Credits</H2>

<P>Wold Group: Ali Mortazavi, Brian Williams, Diane Trout, Brandon King, Ken McCue, Lorian Schaeffer.</P>

<P>Myers Group: Norma Neff, Florencia Pauli, Fan Zhang, Tim Reddy, Rami Rauch. </P> 

<P>Illumina gene expression group: Gary Schroth, Shujun Luo, Eric Vermaas.</P>

<H2>References</H2>

<P>Mortazavi A, Williams BA, McCue K, Schaeffer L, and Wold BJ (2008). 
<A HREF="http://www.ncbi.nlm.nih.gov/pubmed/18516045">Mapping and quantifying mammalian transcriptomes by RNA-Seq</A>
<EM>Nature Methods</EM>. 2008 Jul; 5(7):621-628.</P> 

<P>Langmead B, Trapnell C, Pop M, Salzberg SL.
<!-- A HREF="" --><U>Ultrafast and memory-efficient alignment of short DNA sequences to the human genome</U><!-- /A -->
<EM>Genome Biology</EM>. In press, 2009.</P>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column, above.  The full data release policy
for ENCODE is available
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P>

