<H2>Overview</H2>
<P>
At cell harvest, a subset of cells was stored at -20 degrees C in RNALater.
Total RNA from 5 X 10^6 cells were purified using Ribopure (Ambion) according
to vendor recommended protocols.  Total RNA quality was assessed on RNA 6000
Nano Chips (Agilent) using a Bioanalyzer (Agilent).  Approximately 3ug of
total RNA for each cell type was sent to the Center for Array Technology (CAT)
for labeling and hybridization to Affymetrix Human Exon 1.0 ST arrays.  Briefly,
a Whole Transcript Sense Target Labeling Assay (Affymetrix) was used by the CAT
to reduce the rRNA, perform in vitro transcription (IVT) and create labeled
sense strand DNA for hybridization to the arrays.  Hybridizations were carried
out according the manufacturer's protocol.  Intensity files from the scanned
exon arrays were sent to our lab for analysis at the exon level (Affymetrix
ExACT 1.2.1 software) and the gene level (Affymetrix Expression Console 1.1
software).  Samples were quantile normalized with PM-GCBG background correction
and PLIER (Probe Logarithmic Intensity Error) summarized. 
</P>

<H2>Methods</H2>
<P>
Cells were grown according to the approved ENCODE cell culture protocols. A
subset of the cells was stored frozen in RNALater.  Total RNA was labeled and
hybridized to Affymetrix Human Exon 1.0 ST arrays.  Exon and gene level
expression analysis were carried out using Affymetrix ExACT 1.2.1 and Affymetrix
Expression Console 1.1 software tools. Samples were quantile normalized for
background correction and Probe Logarithmic Intensity Error summarized. 
</P>

<H2>Verification</H2>

<P>
Data were verified by sequencing biological replicates displaying correlation
coefficient >0.9.
</P>

<H2>Credits</H2>

<P>These data were generated by the UW ENCODE group. </P>
<P>Contact: <A HREF="mailto:&#115;&#117;l&#108;&#64;&#117;.w&#97;&#115;h&#105;&#110;&#103;&#116;&#111;n.&#101;&#100;&#117;">Richard Sandstrom</A>
<!-- above address is sull at u.washington.edu -->
</P>
