<H2>Description</H2>

This track shows copy number variation
variation (CNV) in the ENCODE Tier 1 and Tier 2 human cell lines 
K562, GM12878, and HepG2 as determined by Illumina's Human 1M-Duo 
Infinium HD BeadChip assay and CNV analysis by circular binary segmentation (CBS).
CNV segments are colored by type:  red (amplification), blue (heterozygous deletion),
and green (homozygous deletion).
<P>
Two biological replicates are reported for each cell line.
Because biological replicates give very similar results, 
the replicates may later be averaged the to provide a single genotyping
dataset in order to apply these data to other ENCODE experiments.  
Possible uses of this data are for correction of copy number in 
peak-calling for interactome,
transcriptome, DNase hypersensitivity, and methylome determinations.

<H2>Methods</H2>

<H3>Isolation of genomic DNA and hybridization</H3>
Genomic DNA was extracted using the QIAGEN DNeasy Blood & Tissue Kit according
 to the instructions provided by the manufacturer.  
For each biological replicate of each cell line, DNA concentrations and
a level of quality were determined by UV absorbance.
Genotypes were determined from 400 nanograms of each sample at 1 million loci 
using Illumina Human 1M-Duo arrays and standard Illumina protocols.    

<H3>Processing and Analysis </H3>
Genotypes were ascertained from the 1M-Duo Arrays with BeadStudio using 
default settings and formatting with the A/B genotype designation for each SNP 
(see 1M-Duo manifest file for specific nucleotide).  

Copy Number Variation (CNV) analysis was performed using circular binary 
segmentation (DNAcopy) of the log R ratio values at each probe 
(Olshen et al., 2004).  
The parameters used were alpha=0.001, nperm=5000, sd.undo=1.  
Copy number segments are reported with the mean log R ratio for each 
chromosomal segment called by CBS.  
Log ratios of ~-0.2 to -1.5 can be considered heterozygous deletions,
 < -1.5 homozygous deletions, and > 0.2 amplifications.

The coordinates for the genotypes and copy number calls are from Human Genome Build 36.

<H2>Credits></H2>
Rebekka Sprouse (rsprouse@hudsonalpha.org),
Richard Myers (rmyers@hudsonalpha.org),
Devin Absher (dabsher@hudsonalpha.org), and
HudsonAlpha Institute.

<H2>References</H2>

Olshen, A. et al. (2004).  
Circular binary segmentation for the analysis of array-based DNA copy number data.
Biostatistics (Oxford, England) vol. 5 (4) pp. 557-572.
