<H2>Description</H2>
<P>
These tracks display evidence of open chromatin from the Duke/UNC/UT-Austin/EBI
ENCODE group. We identified open chromatin using two independent and complementary
methods: DNaseI hypersensitivity (HS) and Formaldehyde-Assisted Isolation of
Regulatory Elements (FAIRE), combined with chromatin immunoprecipitation (ChIP)
for select regulatory factors. Each method is verified by two detection platforms:
Illumina (formerly Solexa) sequencing by synthesis, and high-resolution 1% ENCODE
tiled microarrays supplied by NimbleGen.
</P><P>
<B>DNaseI HS data</B>: DNaseI is an enzyme that has long been used to map general
chromatin accessibility and the DNaseI "hyperaccessibility" or "hypersensitivity"
is a feature of active cis-regulatory sequences. The use of this method has led
to the discovery of functional regulatory elements that include enhancers, silencers,
insulators, promotors, locus control regions and novel elements. DNaseI
hypersensitivity signifies chromatin accessibility following binding of trans-acting
factors in place of a canonical nucleosome.
</P><P>
<B>FAIRE data</B>: FAIRE (Formaldehyde Assisted Isolation of Regulatory Elements)
is a method to isolate and identify nucleosome-depleted regions of the genome.
FAIRE was initially discovered in yeast, but subsequently was shown to also identify
active regulatory elements in human cells (Giresi <em>et al.</em>, 2006). Most
DNA fragments isolated by FAIRE are 200 bp in length.
</P><P>
<B>ChIP data</B>: ChIP (Chromatin Immunoprecipitation) is a method to identify the
specific location of proteins that are directly or indirectly bound to genomic DNA.
By identifying the binding location of sequence-specific transcription factors,
general transcription machinery components, and chromatin factors, ChIP can help
in the functional annotation of the open chromatin regions identified by DNaseI
HS mapping and FAIRE.
</P>

<H2> Display Conventions and Configuration </H2>
<P>
This track is a multi-view composite track that contains multiple data types
(<EM>views</EM>).  For each view, there are multiple subtracks that
display individually on the browser.  Instructions for configuring multi-view
tracks are <A HREF="../../goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
ENCODE tracks typically contain one or more of the following views:
<DL>
<DT><I>Peaks</I></DT><DD> Regions of signal enrichment based on processed data
(usually normalized data from pooled replicates). ENCODE Peaks tables contain
fields for statistical significance, including FDR
(<A HREF="../../goldenPath/help/qValue.html" TARGET=_BLANK>qValue</A>).</DD>
<DT><I>Signal</I></DT><DD>Density graph (wiggle) of combined signal which gives
a smoothed signal of the underlying data that makes it easy to identify larger
regions of enrichment.</DD>
<DT><I>Signal (Base Overlap)</I></DT><DD>Density graph (wiggle) similar to <I>Signal</I>
above but gives a better sense of the finer structure of the data, such as DNase
footprints and finer level fixing patterns for FAIRE.</DD>
<DT><I>Raw Signal</I></DT><DD>Density graph (wiggle) of signal from raw data
(single replicates, no control). For high-throughput sequencing data, this
reflects the aligned read density.</DD>
<DT><I>Alignments</I></DT><DD>Mappings of short reads to the genome.<DD>
<DT><I>Regions Excluded</I></DT><DD>Genomic regions which were excluded from
calling <I>Peaks</I> due to various reasons such as repetetive sequence or
genomic gaps.<DD>
</DL>

<H2>Methods</H2>
<P>
DNaseI hypersensitive sites were isolated using methods called DNase-seq or
DNase-chip (Boyle <em>et al.</em>, 2008, Crawford <em>et al.</em>, 2006).
Briefly, cells were lysed with NP40 and intact nuclei were digested with
optimal levels of DNaseI enzyme. DNaseI digested ends were captured from
three different DNase concentrations and material was sequenced using Illumina
(Solexa) sequencing. DNase-seq data was verified using material that was
hybridized to NimbleGen Human ENCODE tiled array (1% of the genome).
</P><P>
FAIRE was performed (Giresi <em>et al.</em>, 2007) by cross-linking proteins
to DNA using 1% formaldehyde solution, the complex was sheared using sonication,
and a phenol/chloroform extraction was performed to remove DNA fragments
crosslinked to protein. The DNA recovered in the aqueous phase was sequenced
using the Illumina sequencing and hybridized to NimbleGen Human ENCODE arrays
(1% of the genome).
</P><P>
To perform ChIP, proteins were cross-linked to DNA in vivo using 1% formaldehyde
solution (Bhinge <em>et al.</em>, 2007, ENCODE Project Consortium., 2007).
Crosslinked chromatin was sheared by sonication and immunoprecipitated using a
specific antibody against the protein of interest. After reversal of the
crosslinks, the immunoprecipitated DNA was used to identify the genomic
location of transcription factor binding. This was accomplished by Solexa
sequencing of the ends of the immunoprecipitated DNA (ChIP-seq) as well as
labeling and hybridization of the immunoprecipitated DNA to NimbleGen Human
ENCODE tiling arrays (1% of the genome) along with the input DNA as reference
(ChIP-chip).

<H3>High-Throughput Sequence Data Analysis:</H3>
ENCODE Array Data Analysis: Sequences from each experiment were aligned to the
genome using Maq (Li <em>et al.</em>, 2008) and those aligned to 4 or less
locations were retained. This digital signal was converted to a continuous
wiggle track using F-Seq that employs Parzen kernel density estimation to
create base pair scores (Boyle <em>et al.</em>, 2008b). Discrete DNase HS,
FAIRE, and ChIP sites were identified from DNase/FAIRE/ChIP-seq using F-Seq
by setting a Parzen cutoff based on ROC curve analysis using peaks and non-peaks
identified from DNase/FAIRE/ChIP-chip using 1% ENCODE NimbleGen arrays. Tiled
array peaks were identified using ChIPOTle (Buck <em>et al.</em>, 2005), and
non-peaks were identified using less stringent peak calling cutoffs.
</P>

<H2>Credits</H2>
<P>
These data and annotations were created by a collaboration of multiple institutions
(contact: <A HREF="mailto:&#116;&#115;&#102;ur&#101;&#121;&#64;&#100;&#117;&#107;&#101;.e&#100;u">
&#116;&#115;&#102;ur&#101;&#121;&#64;&#100;&#117;&#107;&#101;.e&#100;u</A>):
<!-- above address is tsfurey at duke.edu -->
</P><P>
Duke University's <A HREF="http://www.genome.duke.edu/index.php" TARGET=_blank>Institute for Genome Sciences & Policy (IGSP)</A>
Alan Boyle, Lingyun Song, <A HREF="http://www.genome.duke.edu/people/faculty/furey/" TARGET=_blank>Terry Furey</A>, and
<A HREF="http://www.genome.duke.edu/people/faculty/crawford/" TARGET=_blank>Greg Crawford</a>
</P><P>
University of North Carolina at Chapel Hill
Paul Giresi and <A HREF="http://www.bio.unc.edu/faculty/lieb/labpages/default.shtml" TARGET=_blank>Jason Lieb</a>
</P><P>
Universty of Texas at Austin
Zheng Liu, Ryan McDaniell, Bum-Kyu Lee, and <A HREF="http://microarray.icmb.utexas.edu/research.html" TARGET=_blank>Vishy Iyer</a>
</P><P>
<A HREF="http://www.ebi.ac.uk/" TARGET=_blank>European Bioinformatics Insitute</A>
Paul Flicek, Stefan Graf, Damian Keefe, and <A HREF="http://www.ebi.ac.uk/~birney/" TARGET=_blank>Ewan Birney</a>
</P><P>
We thank NHGRI for ENCODE funding support.

</P>
<H2>References</H2>

<P>
 <A HREF="http://www.nature.com/nature/journal/v447/n7146/abs/nature05874.html" TARGET=_blank>
 The ENCODE Project Consortium. Identification and analysis of functional elements in 1% of the human genome by the ENCODE pilot project.</A>
 <EM>Nature</EM> 2007;447:799-816
</P><P>

<P>
Bhinge, A.A., Kim, J., Euskirchen, G.M., Snyder, M., & Iyer, V.R.
 <A HREF="http://genome.cshlp.org/cgi/content/abstract/17/6/910" TARGET=_blank>
 Mapping the chromosomal targets of STAT1 by Sequence Tag Analysis of Genomic Enrichment (STAGE).</A>
 <EM>Genome Res.</EM> 2007;17:910-916.
</P><P>

Boyle AP, Davis S, Shulha HP, Meltzer P, Margulies EH, Weng Z, Furey TS, Crawford GE.
 <A HREF="http://www.cell.com/content/article/abstract?uid=PIIS0092867407016133" TARGET=_blank>
 High-resolution mapping and characterization of open chromatin across the genome.</A>
 <EM>Cell.</EM> 2008 Jan 25;132(2):311-22.
</P><P>

Boyle AP, Guinney J, Crawford GE, and Furey TS.
 <A HREF="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btn480" TARGET=_blank>
 F-Seq: a feature density estimator for high-throughput sequence tags.</A>
 <EM>Bioinformatics.</EM> 2008;24(21):2537-8.
</P><P>

Buck MJ, Nobel AB, Lieb JD,
 <A HREF="http://genomebiology.com/2005/6/11/R97" TARGET=_blank>
 ChIPOTle: a user-friendly tool for the analysis of ChIP-chip data.</A>
 <EM>Genome Biology.</EM> 2005.
</P><P>

Crawford GE, Davis S, Scacheri PC, Renaud G, Halawi MJ, Erdos MR, Green R, Meltzer PS, Wolfsberg TG, Collins FS.
 <A HREF="http://www.nature.com/nmeth/journal/v3/n7/abs/nmeth888.html" TARGET=_blank>
 DNase-chip: a high-resolution method to identify DNase I hypersensitive sites using tiled microarrays.</A>
 <EM>Nat Methods.</EM> 2006 Jul;3(7):503-9.
</P><P>

Crawford GE, Holt IE, Whittle J, Webb BD, Tai D, Davis S, Margulies EH, Chen Y, Bernat JA, Ginsburg D, Zhou D, Luo S, Vasicek TJ, Daly MJ, Wolfsberg TG, Collins FS.
 <A HREF="http://www.genome.org/cgi/content/abstract/16/1/123" TARGET=_blank>
 Genome-wide mapping of DNase hypersensitive sites using massively parallel signature sequencing (MPSS).</A>
 <EM>Genome Res.</EM> 2006 Jan;16(1):123-31.
</P><P>

Giresi PG, Kim J, McDaniel RM, Iyer VR, Lieb JD.
 <A HREF="http://www.genome.org/cgi/content/abstract/17/6/877" TARGET=_blank>
 FAIRE (Formaldehyde-Assisted Isolation of Regulatory Elements) isolated active regulatory elements in human chromatin.</A>
 <EM>Genome Res.</EM> 2007;17:877-885.
</P><P>

Li H, Ruan J, and Durbin R.
 <A HREF="http://genome.cshlp.org/content/early/2008/09/25/gr.078212.108.short" TARGET=_blank>
 Mapping short DNA sequencing reads and calling variants using mapping quality scores.</A>
 <EM>Genome Res.</EM> 2008;18(11):1851-8.
</P>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column on the track configuration page and
the download page.  The full data release policy for ENCODE is available
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P>
