<H2>Description</H2>

<P>This track is produced as part of the ENCODE Project.
Expression of different 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=ripTgtProtein">target RNA binding proteins (RBPs)</A>
is measured by RNA Immunoprecipitation and Affymetrix Exon Arrays 
(RIP-chip) using different
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=ripAntibody">RIP antibodies</A>
in different 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=cellType">cell lines</A>.
This initial version of the track contains a single view, 
<i>RBP Assoc RNA</i>, showing the genomic location of the transcripts associated with the array probes.
</P>

<P>In eukaryotic organisms gene regulatory networks require an additional 
level of coordination that links transcriptional and post-transcriptional 
processes. Messenger RNAs have traditionally been viewed as passive 
molecules in the pathway from transcription to translation. However, 
it is now clear that RNA-binding proteins play a major role in 
regulating multiple mRNAs in order to facilitate gene expression patterns. 
These tracks show the associated mRNAs that co-precipitated with the 
targeted RNA-binding proteins using RIP-Chip (Ribonomic) profiling
(Tenenbaum <EM>et al.</EM> 2002; Baroni <EM>et al.</EM> 2008;Penalva <EM>et al.</EM> 2004).
</P>

<H2>Display Conventions and Configuration</H2>

<P>The subtracks within this composite annotation track correspond to data 
from different cell types and tissues.  The graphical configuration options 
are shown at the top of the track description page, followed by a list of 
subtracks. To display only selected subtracks, uncheck the boxes next to 
the tracks you wish to hide. 
</P>

<DL>
<DT><i>RBP Assoc RNA</i>
<DD>The RBP Assoc RNA view shows the genomic extent of the transcripts associated with 
the Affymetrix Exon Array probes, shaded according to score.
</DL>

<H2>Methods</H2>

<P>
Cells were grown according to the approved 
<A HREF="../../ENCODE/protocols/cell" TARGET=_BLANK>ENCODE cell culture protocols</A>.
We used previously described methods for purifying endogenously formed RBP-mRNA 
complexes from cellular extracts and identifying the associated messages using 
genome-scale, microarray technology; a method called RNA-Binding protein 
Immunoprecipitation-Microarray (Chip) profiling or RIP-Chip 
(Tenenbaum <EM>et al.</EM> 2002; Baroni <EM>et al.</EM> 2008;Penalva <EM>et al.</EM> 2004). 
The use of the RIP-Chip methods has provided great insight into the 
infrastructure of coordinated eukaryotic post-transcriptional 
gene expression, which could not have been obtained using 
traditional RNA expression profiling approaches 
(Keene <EM>et al.</EM> 2002; George <EM>et al.</EM> 2006). 
Microarray analysis was performed using the Affymetrix Human Exon 1.0 ST Arrays. 
</P>

<H2>Processing</H2>

<P> CEL files for each sample were processed using the apt-probeset-summarize 
binary provided by Affymetrix. One qualitative (DABG) and two quantitative 
(RMA, PLIER) measures of expression were derived for each gene-level 
metaprobeset on the chips using the command line parameters given 
below (for a detailed explanation of the parameters please see 
Affymetrix documentation for apt-probeset-summarize 
<A TARGET=_BLANK HREF="http://www.affymetrix.com/support/developer/powertools/changelog/apt-probeset-summarize.html">here</A>):
</P>

<PRE>
apt-probeset-summarize -a rma-bg,pm-gcbg,med-polish -a pm-gcbg,plier -a pm-only,dabg.neglog10=true  \\
--precision 10 -p lib/HuEx-1_0-st-v2.r2.pgf -c lib/HuEx-1_0-st-v2.r2.clf -b lib/HuEx-1_0-st-v2.r2.antigenomic.bgp \\
-m lib/HuEx-1_0-st-v2.r2.dt1.hg18.full.mps -o gene/ --cel-files cel_list
</PRE>

<P>An R session was used to import the expression values and experiment 
descriptions so that the following steps could be performed.
<OL>
<LI> A conservative DABG p-value cutoff for presence was established at p &lt; 
1e-256 (neglog10 value = 256). Visual inspection of DABG score distribution 
across all HuEx10ST chips bear this out as a natural bipartite division point for the data.
</LI>
<LI> Probesets at or below the DABG p-val cutoff for all replicates were marked 
as present in the given condition &amp; cell-line.
</LI>
<LI> Lists of core probesets present in IP, in Total, and in Either are generated.
</LI>
<LI> Average log2 fold-changes for IP over total are taken for 
both the rma and plier expression scores.
</LI>
<LI> P-values for significant overexpression in IP replicates over total 
replicates are taken for rma and plier scores.
</LI>
<LI> The minimum of the rma &amp; plier fold-changes is used as the overall fold-change.
</LI>
<LI> The maximum of the rma &amp; plier p-values is used as the overall overexpression p-value.
</LI>
<LI> A default score of -10 is set and for all probesets present in IP, 
scores are replaced by -log10(max.p-value)*(min.log2.fold-change) [ranging from ~-.5 to ~15]
</LI>
<LI> Scores are scaled according to: x = 10*(x+10)+1 &mdash; now absent in 
IP probesets have a score of 1 and present in IP probesets range from ~100 to ~1000.
</LI>
<LI> A broadPeak bed file is created corresponding to the core gene-level 
probesets for each cell-line with the score field containing our scaled score, 
the signal field containing our minimum log2 fold-change, and the p-value 
field containing -log10 of our maximum p-value. The q-value is left at the default of -1.
</LI>
</OL>
</P>

<H2>Credits</H2>

<P>These data were produced and analyzed by the Tenenbaum/Penala collaboration 
with the Scott Tenenabum group at the University at Albany-SUNY, Gen*NY*Sis 
Center For Excellence in Cancer Genomics and the Luiz Penalva group at the 
Children's Cancer Research Institute - UTHSCSA.
</P>
<P>Contact: 
<A HREF="mailto:&#115;&#116;&#101;n&#101;&#110;&#98;&#97;u&#109;&#64;&#97;l&#98;&#97;&#110;&#121;.&#101;&#100;&#117;">
Scott Tenenbaum</A>
</P>
<!-- above address is stenenbaum at albany.edu -->

<H2>References</H2>

<P>Tenenbaum SA, Lager PJ, Carson CC, Keene JD. 
<A TARGET=_BLANK" HREF="http://www.ncbi.nlm.nih.gov/pubmed/12054896">
Ribonomics: identifying mRNA subsets in mRNP complexes using antibodies to RNA-binding proteins and genomic arrays. 
</A>
<EM>Methods</EM>. 2002 Feb;26(2):191-8.
</P>

<P>Baroni TE, Chittur SV, George AD, Tenenbaum SA.
<A TARGET=_BLANK" HREF="http://www.ncbi.nlm.nih.gov/pubmed/18369977">
Advances in RIP-chip analysis : RNA-binding protein immunoprecipitation-microarray profiling.
</A>
<EM>Methods Mol Biol</EM>. 2008;419:93-108.
</P>

<P>Penalva LO, Tenenbaum SA, Keene JD.
<A TARGET=_BLANK" HREF="http://www.ncbi.nlm.nih.gov/pubmed/14770002">
Gene expression analysis of messenger RNP complexes. 
</A>
<EM>Methods Mol Biol</EM>. 2004;257:125-34.
</P>

<P>Keene JD, Tenenbaum SA.
<A TARGET=_BLANK" HREF="http://www.ncbi.nlm.nih.gov/pubmed/12086614">
Eukaryotic mRNPs may represent posttranscriptional operons.
</A>
<EM>Mol Cell</EM>. 2002;9(6):1161-7.
</P>

<P>George AD, Tenenbaum SA.
<A TARGET=_BLANK" HREF="http://www.ncbi.nlm.nih.gov/pubmed/17114943">
MicroRNA modulation of RNA-binding protein regulatory elements.
</A>
<EM>RNA Biol</EM>. 2006;3(2):57-9. Epub 2006 Apr 1.
</P>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior 
consent, submit publications that use an unpublished ENCODE dataset until 
nine months following the release of the dataset.  This date is listed in 
the <EM>Restricted Until</EM> column, above.  The full data release policy 
for ENCODE is available 
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P> 


