<h1> Stanford/HudsonAlpha Methylation27 Docs </h1>
<a name="Description"></a><h2> Description </h2>
<p>These tracks display the methylation status of specific CpG dinucleotides 
in the given cell types as identified by Illumina® Infinium HumanMethylation27 
BeadArray platform. The Methylation27 platform uses bisulfite treated genomic 
DNA to assay the methylation status of specific CpG dinucleotides.  Genomic 
DNA treated with sodium bisulfite converts unmethylated cytosine of a CpG 
dinucleotides into uracil; methylated cytosines do not get converted.  After 
bisulfite treatment, the methylation status of a site is assayed by single 
base-pair extension with a Cy3 or Cy5 labeled nucleotide off of either 
methylated-type or unmethylated-type oligo beads. The oligo beads are 
hybridized to an array and scanned. A beta value is calculated for each CpG 
target – this value represents the intensity value from the methylated bead 
type divided by the sum of the intensity values from the methylated and 
unmethylated bead types for any given CpG target.  
</p>
<p>
This method was used to validate Methyl-seq data generated in collaboration 
with the laboratory of Dr. Julie Baker at Stanford University to study 
methylation and gene expression changes that occur in human embryonic stem 
cells before and after differentiation to definitive endoderm.  Based on 
the results of these experiments, cut-off beta values to call “methylated” 
and “unmethylated” were calculated. Any beta value equal to or greater than 
0.6 is considered fully methylated. Any beta value equal to or less than 0.2 
is considered to be fully unmethylated. Beta values between 0.2 and 0.6 are 
considered to be partially methylated.  Beta-values were quality filtered and 
spots that fell below the threshold p-value are displayed as “NA”.
</p>
<p>
The data set is in a tab delimited format with Illumina CpG Identifier, 
chromosome number, genomic coordinate, gene symbol for the nearest gene, 
and beta value for HepG2, GM12878, and K562 (2 replicates for each cell line).  
Detailed information for the CpG targets can be found on Illumina’s 
website: 
<a href="http://www.illumina.com/downloads/Illumina_HumanMethylation27_v1_Content.zip">
http://www.illumina.com/downloads/Illumina_HumanMethylation27_v1_Content.zip
</a>
</p>

<a name="Methods"></a><h2> Methods </h2>
<p><b> Sodium Bisulfite Conversion </b>
</p>
<p>Bisulfite conversion reaction was done using the Zymo Research® EZ-96 DNA 
MethylationTM Kit protocol posted at 
http://www.zymoresearch.com/protocols/dna/%5BD5003%5D%20EZ-96%20DNA%20methylation%20instruction%20v1.2.pdf. 
One step of the protocol was modified. During the incubation, we included 
a 30 sec 95oC denaturing step every hour to increase reaction efficiency as 
recommended by the Illumina® Infinium Human Methylation27 protocol. 
</p>
<p><b> Infinium HumanMethylation27 BeadArray </b>
</p>
<p>The bead arrays were ran according to the protocol provided by Illumina®. 
The protocol for this assay is posted on their website at 
<a href="http://www.illumina.com/pagesnrn.ilmn?ID=275">
http://www.illumina.com/pagesnrn.ilmn?ID=275</a>.
</p>
<br>
<b> Processing and analysis software used </b>
</p>
<p>The intensity data from the BeadArray was processed using Illumina’s 
BeadStudio software with the Methylation Module v3.2. The data was then 
quality-filtered using p-values.  
</p>
<a name="Credits"></a><h2> Credits </h2>
<p>
Dr. Richard M. Myers (rmyers@hudsonalpha.org) <br>
Mr. Yuya Kobayashi (yuyak@stanford.edu) <br>
Dr. Devin M. Absher (dabsher@hudsonalpha.org) <br>
Dr. Rebekka O. Sprouse (rsprouse@hudsonalpha.org) <br>
</p>
<a name="References"></a><h2> References </h2>
<p><a href="http://www.illumina.com/pages.ilmn?ID=243">Illumina website</a> 
</p>
<p>Brunner, A. L., Johnson, D. S., Kim, S. W., Valouev, A., Reddy, T. E., 
Neff, N. F., Anton, E., Medina, C., Nguyen. L., Chiao, E., Oyolu, C. B., 
Schroth, G. P., Absher, D., Baker, J. C. and Myers, R. M.  (2008).  
Distinct DNA methylation patterns characterize differentiated human embryonic 
stem cells and developing human fetal liver.  Submitted. 
</p>
