<H2>Description</H2>

<P> This track depicts high throughput sequencing of small RNAs (&lt;200 nt) 
from whole cell 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=rnaExtract">
RNA samples</A> from 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=localization">
tissues or sub cellular compartments</A> from 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=cellType">cell lines</A>
included in the ENCODE Transcriptome subproject. The overall 
goal of the ENCODE project is to identify and characterize all functional 
elements in the sequence of the human genome.
</P>

<H2>Display Conventions and Configuration</H2>

<P>To show only selected views, uncheck the boxes next to the one that 
you wish to hide.</P>
 
<P>Color differences among the views are arbitrary. They provide a 
visual cue for 
distinguishing between the different cell types and compartments.</P>

<DL>
<DT><i>Transfrags</i>
<DD>The Transfrags view includes all transfrags before filtering.
</DL>

<H2>Methods</H2>

<P>RNA libraries were generated by treatment of short RNA (&lt;200 nt) with 
Tobacco Acid Pyrophsophatase, followed by: 
1) C-tailing, 
2) 5&prime; ligation with adapter 
5&prime; rCrGrArCrUrGrGrArGrCrArCrGrArGrGrArCrArCrUrGrArCrArUrGrGrArCrUrGrArArGrGrArGrUrArGrArArA 3&prime;, or 
5&prime; rCrArCrGrArGrGrArCrArCrUrGrArCrArUrGrGrArCrUrGrArArGrGrArGrUrArGrArArA 3&prime;, 
3) reverse transcription with primer 
5&prime; TCGCGAGCGGCCGCGGGGGGGGGGGGGGG 3&prime;, and 
4) PCR with primers 
5&prime; AATGATACGGCGACCACCGACACGAGGACACTGACATGGACTGAAGGAGTAGAAA 3&prime; and
5&prime; CAAGCAGAAGACGGCATACGATCGCGAGCGGCCGCGGGGGG 3&prime;. 
Gel-purified libraries were sequenced with primer 
5&prime; CACGAGGACACTGACATGGACTGAAGGAGTAGAAA 3&prime; on Illumina GA1 or 
GA2 platform. 
</P>

<H2>Analysis</H2>

<P>Sequence reads underwent quality filtration using Illumina standard 
pipeline (Gerlad). 
Identical reads were collapsed while maintaining their multiplicity 
information. The 
3&prime; sequencing adaptor was removed from the reads using a custom clipper 
program, 
which aligned the adaptor sequence to the short-reads, allowing up to 2 
mismatches and 
no indels. The trimmed sequences were aligned to the human genome (NCBI build 
36, hg18) 
using Nexalign (Lassmann <EM>et al.</EM>, not published). The alignment 
parameters 
tuned to tolerate up to 2 mismatches and no indels, and to report up to 100 
hits of 
a multimappers. The best hit(s) were reported in a BED-like format. 
</P>

<H2>Verification</H2>

<P>Comparison of referential data generated from 8 individual sequencing lanes 
(Illumina technology). 
</P>

<H2>Credits</H2>

<P>AFFYMETRIX, Inc., Cold Spring Harbor Laboratories, and the Center for Genomic
Regulation (Barcelona), who are participants in the ENCODE Transcriptome Group.
</P>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior 
consent, submit publications that use an unpublished ENCODE dataset until 
nine months following the release of the dataset.  This date is listed in 
the <EM>Restricted Until</EM> column, above.  The full data release policy 
for ENCODE is available 
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P> 


