<H2>General Description</H2>

<P>This track is produced as part of the ENCODE Transcriptome Project. 
Transcription of different 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=rnaExtract">RNA extracts</A> 
from different 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=localization">sub-cellular localizations</A> 
in different 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=cellType">cell lines</A>
is compared in companion experiements ....

This tracks shows the expression of 5' cap analysis gene expression (CAGE) 
clusters that map to the genome on the "plus" or "minus" strand at specific 
locations. A cluster is defined as a region of overlapping read mappings. 
The expression is the number of reads making up the cluster divided by the 
total number of reads sequenced times 1 million.

The <i>Raw Signal</i> view shows an estimate of .....
and the <i>Transfrags</i> view shows the locations of sites corresponding to 
.....
</P>

<H2>Display Conventions and Configuration</H2>

<P>To show only selected views, uncheck the boxes next to the one that 
you wish to hide.</P>
 
<P>Color differences among the views are arbitrary. They provide a 
visual cue for 
distinguishing between the different cell types and compartments.</P>

<DL>
<DT><i>Transfrags</i>
<DD>The Transfrags view includes all transfrags before filtering.
<DT><i>Filtered Transfrags</i>
<DD>The Filtered Transfrags view excludes repeats and other known annotations 
including: 
tRNAs and rRNAs, mi/snoRNAs, things mapping to the mitochondrial or Y 
chromosomes, 
and many predicted snoRNAs and miRNAs.
<DT><i>Raw Signal</i>
<DD>The Raw Signal view shows the probe intensity based on all probes.
<DT><i>Filtered Raw Signal</i>
<DD>The Filtered Raw Signal view shows the probe intensity based on the 
filtered set of probes.
</DL>

<H2>Methods</H2>

<P>
Cells were grown according to the approved 
<A HREF="../../ENCODE/protocols/cell" TARGET=_BLANK>ENCODE cell culture protocols</A>.
RNA molecules longer than 200 nt 
and present in RNA population isolated from different subcellular compartments 
(such as cytosol, nucleus, polysomes and others) were fractionated into polyA+ 
and polyA- 
fractions. Each RNA fraction was converted into double-stranded cDNA using 
random hexamers, 
labeled and hybridized to a tiling 91-array set containing probes against the 
non-repetitive 
portion of the human genome tiled on average every 5 bp (center-to-center of 
each consecutive 25-mers). </P>

<P>
The CAGE tags are sequenced from the 5' ends of cap trapped cDNAs produced using
RIKEN CAGE technology 
(Kodzius <EM>et al</EM>. 2006; Valen <EM>et al</EM>. 2009). 
To create the tag, a linker was attached to the 5'
end of polyA plus or minus reverse transcribed cDNA which were selected by cap
trapping (Carninci et al, Genomics.1;37(3):327-36,1996 Nov ). The first 27 bp of
the cDNA were cleaved using class II restriction enzymes. And then linker was
attached to the 3' of restricted 27 bp cDNA. The result 96 bp were followed by
PCR amplification and sequencing. All tags were mapped to the human genome
(hg18) with the program nexalign (T. Lassmann manuscript in preparation). 
</P>

<H2>Verification</H2>

<P>The reproducibility of the labeling method was assessed separately. Three 
independent 
technical replicates were generated from the same RNA pool for each RNA 
preparation and 
hybridized to duplicate arrays (two technical replicates) that contain the 
ENCODE regions. 
Labeled RNA samples were then pooled and hybridized to the tiling 91-array 
set spanning 
the whole genome. Transcribed regions (transfrags; see the <i>Transfrags</i> 
view) were 
generated from the <i>Raw Signal</i> by merging genomic positions to which 
probes 
are mapped. This merging was based on a 5% false positive rate cutoff in 
negative 
bacterial controls, a maximum gap (MaxGap) of 40 base-pairs and minimum run 
(MinRun) 
of 40 base-pairs. </P>

<H2>Credits</H2>

<P>These data were generated and analyzed by Timo Lassmann, Phil Kapranov, 
Hazuki Takahashi, Yoshihide Hayashizaki, Tom Gingeras, and Piero Carninci.

<P>Contact: <A HREF="mailto:&#116;&#105;&#109;o&#108;&#97;&#115;s&#109;&#97;&#110;n&#64;&#103;m&#97;&#105;&#108;.&#99;&#111;m">
Timo Lassmann</A>.
<!-- above address is timolassmann at gmail.com -->
</P>

<H2>References</H2>

<P>
Kodzius R, Kojima M, Nishiyori H, Nakamura M, Fukuda S, Tagami M, Sasaki D,
Imamura K, Kai C, Harbers M, et al. 
<A TARGET=_BLANK HREF="http://www.ncbi.nlm.nih.gov/pubmed/16489339">
CAGE: cap analysis of gene expression.</A>
<EM>Nat Methods</EM>. 2006 March 1; 3(3):211-222.
</P>

<P>
Valen E, Pascarella G, Chalk A, Maeda N, Kojima M, Kawazu C, Murata M, Nishiyori
H, Lazarevic D, Motti D, et al.
<A TARGET=_BLANK HREF="http://www.ncbi.nlm.nih.gov/pubmed/19074369">
Genome-wide detection and analysis of hippocampus core promoters using
DeepCAGE.</A>
<EM>Genome Res</EM>. 2009 February; 19(2):255-65.
</P>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior 
consent, submit publications that use an unpublished ENCODE dataset until 
nine months following the release of the dataset.  This date is listed in 
the <EM>Restricted Until</EM> column, above.  The full data release policy 
for ENCODE is available 
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P> 

