<H2>Description</H2>

<P> This track depicts high throughput sequencing of small RNAs (&lt;200 nt) 
from whole cell 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=rnaExtract">
RNA samples</A> from 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=localization">
tissues or sub cellular compartments</A> from 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=cellType">cell lines</A>
included in the ENCODE Transcriptome subproject. The overall 
goal of the ENCODE project is to identify and characterize all functional 
elements in the sequence of the human genome.
RNA-Seq was performed by reverse-transcribing an RNA sample into cDNA,
followed by high throughput DNA sequencing of the cDNA, which was done here
on Helicos&#0153; Genetic Analysis System (Harris et al;
<a target=_blank href="http://www.helicosbio.com/">
http://www.helicosbio.com/</A>). 
</P>

<H2>Display Conventions and Configuration</H2>

<P>To show only selected views, uncheck the boxes next to the one that 
you wish to hide.</P>
 
<P>Color differences among the views are arbitrary. They provide a 
visual cue for 
distinguishing between the different cell types and compartments.</P>

<DL>
<DT><i>Alignments</i>
<DD>RNA-seq tag alignments.
<DT><i>Raw Signal</i>
<DD>Density graph (wiggle) of a number of reads overlapping a nucleotide
 in the genome.
</DL>

<H2>Methods</H2>
<P>
Cells were grown according to the approved ENCODE cell culture protocols. RNA
molecules longer than 200 nt and present in RNA population isolated from
different subcellular compartments (such as cytosol, nucleus, polysomes and
others) were fractionated into polyA+ and polyA- fractions. 
</P>
<P>
RNA was converted into first strand cDNA using a high excess of random hexamers
without prior fragmentation. Spurious second-strand cDNA synthesis could occur
under these conditions. The first strand cDNA molecules were tailed at the
3&prime; ends with polyA residues using terminal transferase and used 
directly for sequencing.
</P>
<P>
Filtered reads were aligned to the human genome using in-house and freely
available Helicos Alignment software indexDPgenomic
(<A target=_blank 
href="http://open.helicosbio.com/mwiki/index.php/Docs/Software/Bioinformatics#Executables"> 
http://open.helicosbio.com/mwiki/index.php/Docs/Software/Bioinformatics#Executables</A>
) with a minimum normalized alignment score of 4.5.
The normalized score was defined as following;
</P>
<PRE>
Score=(#matches*5-#mismatches*4)/length_read . 
</PRE>
<PRE>
For example, in the following alignment:

Tag Sequence CCTCCGTGTTGTTCCAGCC-CAGTGCTCGCAGG 
Ref Sequence C-TCCGTGTTGTTCCAGCCACAGTGCTCGCAGG
Length of alignment block: 33
Length of tag sequence: 32
Number of matches: 31
Number of errors: 2
Score: (31*5) - (2*4) = 155 - 8 = 147
Normalized score = 147/32 = 4.59375
</PRE>

<H2>Verification</H2>

<P>
Known exon maps as displayed on the genome browser are confirmed by the 
alignment of sequence reads.
</P>

<H2>Credits</H2>

<P>
Helicos BioSciences: Philipp Kapranov, Eldar Giladi, Steve Roels, Chris Hart,
Stan Letovsky,  Patrice Milos. 
Cold Spring Harbor Laboratory: Carrie Davis, Kim Bell, Huaien Wang, 
Tom Gingeras.
</P>
<P>
Contacts: 
<A HREF="mailto:&#112;&#107;ap&#114;&#97;&#110;&#111;&#118;&#64;&#104;&#101;&#108;&#105;&#99;os&#98;&#105;&#111;.&#99;&#111;m">
Philipp Kapranov
</A>
<!-- above address is pkapranov at helicosbio.com -->
; 
<A HREF="mailto:&#112;&#109;&#105;&#108;&#111;s&#64;&#104;e&#108;&#105;&#99;&#111;&#115;b&#105;&#111;.&#99;&#111;&#109;">
Patrice Milos
</A>
<!-- above address is pmilos at helicosbio.com -->
</P>

<H2>References</H2>

<P>Harris TD, Buzby PR, Babcock H, Beer E, Bowers J, Braslavsky I, Causey
M, Colonell J, Dimeo J, Efcavitch JW,Giladi E, Gill J, Healy J, Jarosz M, 
Lapen D, Moulton K, Quake SR, Steinmann K, Thayer E, Tyurina A, Ward R,
Weiss H, Xie Z.
<A TARGET=_BLANK HREF="http://www.ncbi.nlm.nih.gov/pubmed/18388294">
Single-molecule DNA sequencing of a viral genome
</A>
<EM>Science</EM>. 2008 Apr 4;320(5872):106-9</P>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior 
consent, submit publications that use an unpublished ENCODE dataset until 
nine months following the release of the dataset.  This date is listed in 
the <EM>Restricted Until</EM> column, above.  The full data release policy 
for ENCODE is available 
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P> 


