<H2>Description</H2>

<P>This track is produced as part of the ENCODE Transcriptome Project.
It shows high throughput sequencing of 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=rnaExtract">
RNA samples</A> from 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=localization">
tissues or sub cellular compartments</A> from 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=cellType">cell lines</A>
included in the ENCODE Transcriptome subproject. The overall 
goal of the ENCODE project is to identify and characterize all functional 
elements in the sequence of the human genome.
</P>

<H2>Display Conventions and Configuration</H2>

<P>To show only selected subtracks, uncheck the boxes next to the tracks that 
you wish to hide.</P>
 
<P>Color differences among the views are arbitrary. They provide a 
visual cue for 
distinguishing between the different cell types and compartments.</P>

<DL>
<DT><i>Alignments</i>
<DD>The Alignments view shows alignment of individual RNA sequences.
<DT><i>Plus Raw Signal</i>
<DD>The Plus Raw Signal view graphs the base-by-base density of tags 
on the + strand.
<DT><i>Minus Raw Signal</i>
<DD>The Minus Raw Signal view graphs the base-by-base density of tags 
on the - strand.
</DL>

<H2>Methods</H2>

<P>
The RNA-Seq data were generated from high quality polyA RNA and the RNA-Seq
libraries were constructed using SOLiD Whole Transcriptome (WT) protocol and
reagent kit.  Total RNA in good quality was used as starting materials and
purified twice through MACs polyT column aimed to enrich polyA and remove any
contaminants (e.g., rRNA, tRNA, DNA, protein etc.).  One microgram enriched
polyA RNA sample was then fragmented to small pieces and a gel-based selection
method was performed to collect fragmented random polyA at a size-range of
50-150 nt in length. The collected fragmental RNA was then hybridized and
ligated to a mix of adaptors provided from ABI, followed by reverse
transcription to generate corresponding cDNAs.  The resulting cDNA library was
further amplified by PCR and sequenced by SOLiD platform for single read at 35
bp length (new version in 50 bp length).    
Cells were grown according to the approved 
<A HREF="../../ENCODE/protocols/cell" TARGET=_BLANK>ENCODE cell culture
protocols</A>.
</P>

<P>
<B>Data:</B> The SOLiD-generated RNA-Seq reads was in 35bp length and an 
initial filtering
process was taken to remove any non-desirable contamination sequences, such as
rRNA, tRNA, and repeats etc. A read-split mapping approach was developed to map
the 35bp reads onto reference genome, NCBI Build36, hg18. Specifically, the 35bp
reads were divided into two parts (1st-25bp and 2nd-25bp with 10bp overlapping)
and mapped separately.  An extension mapping analysis was further performed to
generate a score counts from each read and use the score numbers as a gauge of
filtering reference (e.g., scoring &gt;26 was used). The reads with mapping
locations N&le;1 or N&gt;10 were excluded from further analysis.  As a unique
strand-specific feature from the SOLiD RNA-Seq, the data sets generated by 
SOLiD RNA-Seq were strand-specific, and mapped on exons with 
strand-specificity.
</P>

<P>
<B>Mapping parameters:</B>  Mapping was done using Applied Biosystems' SOLiD 
alignment
for whole transcriptome analysis pipeline.  Two mismatches were allowed in the
25bp color space seed sequence with progressive alignment performed to find the
full mapping location.  A score is computed for each mapping location and any
location that scored &le;26 was filtered.   
</P>



<H2>Verification</H2>


<H2>Credits</H2>

<P>The GIS RNA-seq libraries and sequence data for transcriptome analysis were
produced at the 
<A HREF="http://www.gis.a-star.edu.sg/internet/site">Genome Institute of
Singapore</A>. 
The data were mapped and analyzed by scientists from the Genome Institute of 
Singapore.
</P>

<H2>References</H2>


<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior 
consent, submit publications that use an unpublished ENCODE dataset until 
nine months following the release of the dataset.  This date is listed in 
the <EM>Restricted Until</EM> column, above.  The full data release policy 
for ENCODE is available 
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P> 
