<H2>Description</H2>

<P>In eukaryotic organisms gene regulatory networks require an additional level of coordination that links transcriptional and post-transcriptional processes. Messenger RNAs have traditionally been viewed as passive molecules in the pathway from transcription to translation. However, it is now clear that RNA-binding proteins play a major role in regulating multiple mRNAs in order to facilitate gene expression patterns. These tracks show the associated mRNAs that co-precipitated with the targeted RNA-binding proteins HuR, IMP-1 and PABP using RIP-Chip (Ribonomic) profiling
(Tenenbaum <EM>et al.</EM> 2002; Baroni <EM>et al.</EM> 2008;Penalva <EM>et al.</EM> 2004).
</P>

<H2>Methods</H2>

<P>We used previously described methods for purifying endogenously formed RBP-mRNA complexes from cellular extracts and identifying the associated messages using genome-scale, microarray technology; a method called RNA-Binding protein Immunoprecipitation-Microarray (Chip) profiling or RIP-Chip 
(Tenenbaum <EM>et al.</EM> 2002; Baroni <EM>et al.</EM> 2008;Penalva <EM>et al.</EM> 2004). 
The use of the RIP-Chip methods has provided great insight into the infrastructure of coordinated eukaryotic post-transcriptional gene expression, which could not have been obtained using traditional RNA expression profiling approaches 
(Keene <EM>et al.</EM> 2002; George <EM>et al.</EM> 2006). 
The data for these RIP-Chip profiling tracks are for HuR, IMP-1, PABP, and G10 as determined for K562 and GM12878 cell lines.  Microarray analysis was performed using the Affymetrix Human Exon 1.0 ST Arrays. 
</P>

<P> Targeted RNA-Binding Proteins</P>
<TABLE BORDER=1>
<THEAD>
<TR><TH>Protein</TH><TH>Description</TH></TR>
</THEAD>
<TBODY>
<TR><TD>HuR</TD><TD>a member of the embryonic lethal abnormal vision (ELAV) family of RBPs, which has been shown to preferentially interact with AU- rich elements (AREs) commonly found in many early response gene mRNAs and has been shown to shuttle its mRNA cargo between the nucleus and cytoplasm and effect mRNA stability
(Good <EM>et al.</EM> 1997; Good <EM>et al.</EM> 2000; Lopez de Silanes <EM>et al.</EM> 2003; Lopez de Silanes <EM>et al.</EM> 2004).
</TD></TR>
<TR><TD>IMP-1</TD><TD>a member of the Insulin-like mRNA binding protein family of RBPs which have been shown to be involved in mRNA localization and potentially, in storage 
(Nielsen <EM>et al.</EM> 2001; Yisraeli 2005; Oberman <EM>et al.</EM> 2007; Yaniv <EM>et al.</EM> 2003). 
</TD></TR>
<TR><TD>PABP</TD><TD>(poly-A binding protein), which is a well characterized RBP that binds to the poly-A tail present in most eukaryotic mRNAs resulting in enhanced stability and translation
(Penalva <EM>et al.</EM> 2004; Mangus <EM>et al.</EM> 2003; LinksKahvejian <EM>et al.</EM> 2001).
</TD></TR>
<TR><TD>G10</TD><TD>Xenopus G10 protein, potentially a transcription factor, here used as a negative control for the RNA-immunoprecipitation steps.
</TD></TR>
</TBODY>
</TABLE>
</P>

<H2>Processing</H2>

<P> CEL files for each sample were processed using the apt-probeset-summarize binary provided by Affymetrix. One qualitative (DABG) and two quantitative (RMA, PLIER) measures of expression were derived for each gene-level metaprobeset on the chips using the command line parameters given below (for a detailed explanation of the parameters please see Affymetrix documentation for apt-probeset-summarize <A HREF="http://www.affymetrix.com/support/developer/powertools/changelog/apt-probeset-summarize.html">here</A>):
</P>

<PRE>
apt-probeset-summarize -a rma-bg,pm-gcbg,med-polish -a pm-gcbg,plier -a pm-only,dabg.neglog10=true  --precision 10 -p lib/HuEx-1_0-st-v2.r2.pgf -c lib/HuEx-1_0-st-v2.r2.clf -b lib/HuEx-1_0-st-v2.r2.antigenomic.bgp -m lib/HuEx-1_0-st-v2.r2.dt1.hg18.full.mps -o gene/ --cel-files cel_list
</PRE>

<P>An R session was used to import the expression values and experiment descriptions so that the following steps could be performed.
<OL>
<LI> A conservative DABG p-value cutoff for presence was established at p &lt; 1e-256 (neglog10 value = 256). Visual inspection of DABG score distribution across all HuEx10ST chips bear this out as a natural bipartite division point for the data.
</LI>
<LI> Probesets at or below the DABG p-val cutoff for all replicates were marked as present in the given condition &amp; cell-line.
</LI>
<LI> Lists of core probesets present in IP, in Total, and in Either are generated.
</LI>
<LI> Average log2 fold-changes for IP over total are taken for both the rma and plier expression scores.
</LI>
<LI> P-values for significant overexpression in IP replicates over total replicates are taken for rma and plier scores.
</LI>
<LI> The minimum of the rma &amp; plier fold-changes is used as the overall fold-change.
</LI>
<LI> The maximum of the rma &amp; plier p-values is used as the overall overexpression p-value.
</LI>
<LI> A default score of -10 is set and for all probesets present in IP, scores are replaced by -log10(max.p-value)*(min.log2.fold-change) [ranging from ~-.5 to ~15]
</LI>
<LI> Scores are scaled according to: x = 10*(x+10)+1 &dash; now absent in IP probesets have a score of 1 and present in IP probesets range from ~100 to ~1000.
</LI>
<LI> A broadPeak bed file is created corresponding to the core gene-level probesets for each cell-line with the score field containing our scaled score, the signal field containing our minimum log2 fold-change, and the p-value field containing -log10 of our maximum p-value. The q-value is left at the default of -1.
</LI>
</OL>
</P>

<H2>Display Conventions and Configuration</H2>

<P>The subtracks within this composite annotation track correspond to data from different cell types and tissues.  The graphical configuration options are shown at the top of the track description page, followed by a list of subtracks. To display only selected subtracks, uncheck the boxes next to the tracks you wish to hide. For more information about the graphical configuration options, click the <A HREF="http://genome.ucsc.edu/goldenPath/help/hgWiggleTrackHelp.html">Graph configuration help</A> link.
</P>

<H2>Credits</H2>

<P>These data were produced and analyzed by the Tenenbaum/Penala collaboration with the Scott Tenenabum group at the University at Albany-SUNY, Gen*NY*Sis Center For Excellence in Cancer Genomics and the Luiz Penalva group at the Children's Cancer Research Institute - UTHSCSA.
</P>

<H2>References</H2>

<P>Tenenbaum SA, Lager PJ, Carson CC, Keene JD., Ribonomics: identifying mRNA subsets in mRNP complexes using antibodies to RNA-binding proteins and genomic arrays. 
<EM>Methods</EM>. 2002 Feb;26(2):191-8.
</P>
<P>Baroni TE, Chittur SV, George AD, Tenenbaum SA., Advances in RIP-chip analysis : RNA-binding protein immunoprecipitation-microarray profiling.
<EM>Methods Mol Biol</EM>. 2008;419:93-108.
</P>
<P>Penalva LO, Tenenbaum SA, Keene JD., Gene expression analysis of messenger RNP complexes. 
<EM>Methods Mol Biol</EM>. 2004;257:125-34.
</P>
<P>Keene JD, Tenenbaum SA., Eukaryotic mRNPs may represent posttranscriptional operons.
<EM>Mol Cell</EM>. 2002;9(6):1161-7.
</P>
<P>George AD, Tenenbaum SA., MicroRNA modulation of RNA-binding protein regulatory elements.
<EM>RNA Biol</EM>. 2006;3(2):57-9. Epub 2006 Apr 1.
</P>
<P>Good, P.J., The role of elav-like genes, a conserved family encoding RNA-binding proteins, in growth and development.
<EM>Semin Cell Dev Biol</EM>. 1997. 8(6): p. 577-84.
</P>
<P>Good, P.J., et al., A family of human RNA-binding proteins related to the Drosophila Bruno translational regulator.
<EM>J Biol Chem</EM>. 2000. 275(37): p. 28583-92.
</P>
<P>Lopez de Silanes, I., et al., Role of the RNA-binding protein HuR in colon carcinogenesis.
<EM>Oncogene</EM>. 2003. 22(46): p. 7146-54.
</P>
<P>Lopez de Silanes, I., et al., Identification of a target RNA motif for RNA-binding protein HuR.
<EM>Proc Natl Acad Sci U S A</EM>. 2004. 101(9): p. 2987-92.
</P>
<P>Nielsen FC, Nielsen J, Christiansen J., A family of IGF-II mRNA binding proteins (IMP) involved in RNA trafficking.
<EM>Scand J Clin Lab Invest Suppl.</EM>. 2001;234:93-9. 
</P>
<P>Yisraeli JK., VICKZ proteins: a multi-talented family of regulatory RNA-binding proteins. 
<EM>Biol Cell</EM>. 2005 Jan;97(1):87-96. 
</P>
<P>Oberman F, Rand K, Maizels Y, Rubinstein AM, Yisraeli JK., VICKZ proteins mediate cell migration via their RNA binding activity.
<EM>RNA</EM>. 2007 Sep;13(9):1558-69. Epub 2007 Jul 24.
</P>
<P>Yaniv, K., et al., The RNA-binding protein Vg1 RBP is required for cell migration during early neural development.
<EM>Development</EM>. 2003. 130(23): p. 5649-61.
</P>
<P>Penalva, L.O., S.A. Tenenbaum, and J.D. Keene, Gene Expression Analysis of Messenger RNP Complexes.
<EM>Methods Mol Biol</EM>. 2004. 257: p. 125-34.
</P>
<P>Mangus DA, Evans MC, Jacobson A., Poly(A)-binding proteins: multifunctional scaffolds for the post-transcriptional control of gene expression.
<EM>Genome Biol</EM>. 2003;4(7):223. Epub 2003 Jul 1.
</P>
<P>LinksKahvejian A, Roy G, Sonenberg N., The mRNA closed-loop model: the function of PABP and PABP-interacting proteins in mRNA translation.
<EM>Cold Spring Harb Symp Quant Biol</EM>. 2001;66:293-300.
</P>

