<h2>Description</h2>

<p>
The Gencode Genes track (freeze 2, <B>February 2009</B>) shows high-quality manual
annotations in the ENCODE regions generated by the
<a href="http://www.sanger.ac.uk/PostGenomics/encode/" target="_blank">GENCODE project</a>.
<br>

The annotations are divided into separate tracks based on
source/confidence.&nbsp;The Gencode project recommends that the
annotations from level 1 &amp; 2 be used as the reference gene annotation,
level 3 was added to fill gaps for methods that analyze the entire
genome and require a full set.
</p>

<ul>
  <li> Level 1: <b>validated</b> </li>
</ul>

<dl>
  <dd>At this time only pseudogene loci, that were predicted by the
analysis-pipelines from YALE, UCSC as well as by HAVANA manual annotation from
WTSI. </dd>
</dl>

<ul>
  <li> Level 2: <b>manual annotation</b> </li>
</ul>

<dl>
  <dd>HAVANA manual annotation from WTSI.<BR>
     The following regions are considered "fully annotated" and contain
level 2 annotation from HAVANA only, although they will still be updated:
     chromosomes 1, 2, 6, 9, 10, 13, 20, 21, 22, X, Y, ENCODE pilot
regions, chr11:2353995-3878750.</dd>
<!--  <dd>HAVANA manual annotation from WTSI. </dd> -->
</dl>

<ul>
  <li> Level 3: <b>automated annotation</b> </li>
</ul>

<dl>
  <dd>ENSEMBL loci in regions where no HAVANA annotation can be found. </dd>
</dl>

</p><p>
<B>NOTE</B>: The release cycles for Gencode, Havana and Ensembl differ. Users
are cautioned to compare release dates to determine which annotation is most
current.

</p><p>
The gene annotations are colored based on the HAVANA annotation type and
the confidence level. See the table below for the color key, as well as more
detail about the transcript and feature types.
</P>

<P>
<table border="1" bordercolor="#aaaaaa" cellpadding="4">
  <tbody>
    <tr>
      <th align="left">Class</th>
      <th align="left">Color</th>
      <th align="left">Description</th>
      <th align="left">Transcript Types (see <a href="http://vega.sanger.ac.uk/info/about/gene_and_transcript_types.html" target="_blank">Vega Transcript Types</a>)</th>
    </tr>
<tr><td>Validated_coding                <td><B><font color="#0000FF">Dark Blue    </font></B><td>Level 1 Validated: coding regions                 <td>protein_coding</tr>
<tr><td>Validated_processed             <td><B><font color="#3399FF">Light Blue   </font></B><td>Level 1 Validated: processed                      <td>processed_transcript</tr>
<tr><td>Validated_processed_pseudogene  <td><B><font color="#5522AA">Dark Purple  </font></B><td>Level 1 Validated: processed pseudogenes         <td>processed_pseudogene, transcribed_processed_pseudogene</tr>
<tr><td>Validated_unprocessed_pseudogene<td><B><font color="#6611FF">Medium Purple</font></B><td>Level 1 Validated: unprocessed pseudogenes       <td>unitary_processed_pseudogene, unprocessed_pseudogene</tr>
<tr><td>Validated_pseudogene            <td><B><font color="#9966FF">Light Purple </font></B><td>Level 1 Validated: pseudogenes                   <td>IG_pseudogene, pseudogene, transcribed_pseudogene</tr>
<tr><td>Havana_coding                   <td><B><font color="#993300">Dark Orange  </font></B><td>Level 2 Manual annotation: coding                <td>protein_coding, IG_gene</tr>
<tr><td>Havana_nonsense                 <td><B><font color="#DF7401">Medium Orange</font></B><td>Level 2 Manual annotation: nonsense              <td>nonsense_mediated_decay</tr>
<tr><td>Havana_non_coding               <td><B><font color="#FE9A2E">Light Orange </font></B><td>Level 2 Manual annotation: non-coding            <td>ambiguous_orf, antisense, non_coding, retained_intron, processed_transcript</tr>
<tr><td>Havana_processed_pseudogene     <td><B><font color="#990066">Dark Pink    </font></B><td>Level 2 Manual annotation: processed pseudogene  <td>processed_pseudogene, transcribed_processed_pseudogene, transcribed_pseudogene</tr>
<tr><td>Havana_unprocessed_pseudogene   <td><B><font color="#B404AE">Medium Pink  </font></B><td>Level 2 Manual annotation: unprocessed pseudogene<td>polymorphic_pseudogene, transcribed_unprocessed_pseudogene, unitary_pseudogene, unprocessed_pseudogene</tr>
<tr><td>Havana_pseudogene               <td><B><font color="#FF33FF">Light Pink   </font></B><td>Level 2 Manual annotation: pseudogene            <td>pseudogene, IG_pseudogene</tr>
<tr><td>Havana_TEC                      <td><B><font color="#666666">Grey         </font></B><td>Level 2 Manual annotation: TEC                   <td>TEC, artifact</tr>
<tr><td>Havana_polyA                    <td><B><font color="#000000">Black        </font></B><td>Level 2 Manual annotation: polyA                 <td>polyA_signal, polyA_site, pseudo_polyA</tr>
<tr><td>Ensembl_coding                  <td><B><font color="#DF0101">Dark Red     </font></B><td>Level 3 Automated annotation: Coding             <td>protein_coding, IG_protein, C_segment, J_segment, V_segment</tr>
<tr><td>Ensembl_RNA                     <td><B><font color="#FE2E2E">Light Red    </font></B><td>Level 3 Automated annotation: RNA transcripts    <td>miRNA, miRNA_pseudogene, misc_RNA, misc_RNA_pseudogene, Mt_rRNA, Mt_tRNA, Mt_tRNA_pseudogene, rRNA, rRNA_pseudogene, scRNA, scRNA_pseudogene, snoRNA, snoRNA_pseudogene, snRNA, snRNA_pseudogene, tRNAscan, tRNA_pseudogene</tr>
<tr><td>Ensembl_pseudogene              <td><B><font color="#990066">Dark Pink    </font></B><td>Level 3 Automated annotation: pseudogene         <td>pseudogene, retrotransposon</tr>
<!--<tr><td>Havana_pseudogene               <td><font color="#F781F3">Light Pink   </font><td>Level 2 Manual annotation: pseudogene            <td>pseudogene, IG_pseudogene</tr>-->
<!--<tr><td>Havana_non_coding               <td><font color="#F7BE81">Light Orange </font><td>Level 2 Manual annotation: non-coding            <td>ambiguous_orf, antisense, non_coding, retained_intron, processed_transcript</tr>-->
<!--    <tr>
<tr><td>Validated_coding                <td><font color="#04B404">Dark Green   </font><td>Level 1 Validated coding regions                 </tr>
<tr><td>Validated_processed             <td><font color="#ACFA58">Light Green  </font><td>Level 1 Validated processed                      </tr>
      <td>Known</td>
      <td><font color="#215b30">dark green</font></td>
      <td><b>Known protein-coding genes</b> (<em>i.e.</em>,
referenced in <a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=gene" target="_blank">Entrez Gene</a>)</td>   http://vega.sanger.ac.uk/Homo_sapiens/transview?transcript=
    </tr>
    <tr>
      <td>...</td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td>...</td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td></td>
      <td></td>
      <td></td>
    </tr> -->
  </tbody>
</table>
</P>

<H2> Display Conventions and Configuration </H2>
<P>
This track uses filtering by category to select subsets of transcripts and has
additional advanced features.  Help with these features can be found
<A HREF="../../goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.

<h2>Methods</h2>

<P>
We aim to annotate all evidence-based gene features at high accuracy on
the human reference sequence. This includes identifying all
protein-coding loci with associated alternative variants, non-coding
loci which have transcript evidence, and pseudogenes. We integrate
computational approaches (including comparative methods), manual
annotation and targeted experimental verification.
</P>
<P>For a detailed description of the methods and references used, see
Harrow <em>et al</em> (2006).
</P>

<h2>Verification</h2>

<P> See Harrow <em>et al.</em> (2006) for information on verification
techniques.
</P>

<h2>Credits</h2>

<P> This GENCODE release is the result of a collaborative effort among
the following laboratories: (contact: <A HREF="mailto:&#102;&#115;k&#64;&#115;an&#103;&#101;&#114;.&#97;&#99;.u&#107;">
Felix Kokocinski</A>)
</P>

<P>
<table style="text-align: left; width: 100%;" border="1" bordercolor="#aaaaaa" cellpadding="2" cellspacing="2">
  <tbody>
    <tr>
      <td style="vertical-align: top; text-align: left;"> <b>Lab/Institution</b><br>
      </td>
      <td style="vertical-align: top; text-align: left;"><b>Contributors</b><br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: left;"><a href="http://www.sanger.ac.uk/HGP/havana/havana.shtml" target="_blank">HAVANA annotation group</a>,
          Wellcome Trust Sanger Insitute (WTSI), Hinxton, UK</td>
      <td style="vertical-align: top;">Adam Frankish, James Gilbert, Jennifer Harrow,
Felix Kokocinski, Stephen Trevanion, Tim Hubbard (GENCODE Principal Investigator)<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: left;"><a href="http://genome.imim.es/" target="_blank">Genome Bioinformatics Lab (CRG)</a>,
          Barcelona, Spain</td>
      <td style="vertical-align: top;">Thomas Derrien, Tyler Alioto, Roderic Guig&oacute;</td>
    </tr>
    <tr>
      <td><a href="http://genome.ucsc.edu/" target="_blank">Genome Bioinformatics, University of California Santa Cruz</a> (UCSC), USA</td>
      <td>Rachel Harte, Mark Diekhans, Robert Baertsch, David Haussler</td>
    </tr>
    <tr>
      <td><a href="http://mblab.wustl.edu/" target="_blank">Comp. Genomics Lab, Washington University</a> St. Louis (WUSTL), USA</td>
      <td>Jeltje van Baren, Charlie Comstock, David Lu, Michael Brent</td>
    </tr>
    <tr>
      <td><a href="http://www.csail.mit.edu/" target="_blank">Computer Science and Artificial Intelligence Lab</a>,
          <a href="http://www.broad.mit.edu/" target="_blank">Broad Institute of MIT and Harvard</a>, USA</td>
      <td>Mike Lin, Manolis Kellis</td>
    </tr>
    <tr>
      <td><a href="http://bioinfo.mbb.yale.edu/" target="_blank">Bioinformatics, Yale University</a> (Yale), USA</td>
      <td>Philip Cayting, Mark Gerstein</td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: left;"><a href="http://www.unil.ch/cig" target="_blank">Center for Integrative Genomics</a>,
          University of Lausanne, Switzerland</td>
      <td style="vertical-align: top;"> Cedric Howald, Alexandre Reymond</td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: left;"><a href="http://www.ensembl.org/info/docs/genebuild/genome_annotation.html" target="_blank">ENSEMBL genebuild group</a>,
          Wellcome Trust Sanger Insitute (WTSI), Hinxton, UK</td>
      <td style="vertical-align: top;">Bronwen Aken, Julio Fernandez Banet, Stephen Searle<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: left;"><a href="http://www.cnio.es/ing/grupos/plantillas/presentacion.asp?pag=726" target="_blank">
Structural Computational Biology Group</a>, Centro Natcional de Investigaciones Oncologicas (CNIO), Madrid, Spain</td>
      <td>Manuel Rodr&iacute;guez Jos&eacute;, Jan-Jaap Wesselink, Michael Tress,&nbsp;Alfonso Valencia</td>
    </tr>
  </tbody>
</table>

</P>

<h2>References</h2>

<P> Harrow J, Denoeud F, Frankish A, Reymond A, Chen CK, Chrast J, Lagarde J, Gilbert JG, Storey R, Swarbreck D <em>et al</em>.
<a href="http://genomebiology.com/2006/7/S1/S4" target="_blank">GENCODE: producing a reference annotation
for ENCODE</a>. <em>Genome Biol</em>. 2006;7 Suppl
1:S4.1-9.</P>


<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column on the track configuration page and
the download page.  The full data release policy for ENCODE is available
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P>
