<H2>Description</H2>

This track shows copy number variation (CNV) in the ENCODE Tier 1 and Tier 2 
human cell lines GM12878, HepG2, and K562 as determined by Illumina's Human 
1M-Duo Infinium HD BeadChip assay and CNV analysis by circular binary segmentation (CBS).
<P>
CNV segments are colored by type: 
<ul>
<li><font color="#e600e6">magenta</font> = amplified  </li>
<li><font color="#00cd00">green</font> = homozygous deletion</li>
<li><font color="#0000CD">blue</font> = heterozygous deletion  </li>
<li><font color="#000000">black</font> = normal</li>
</ul>
</p>
<P>
Two biological replicates were generated for each cell line.
Because biological replicates gave very similar results, 
the replicates were averaged to provide a single genotyping
dataset in order to apply these data to other ENCODE experiments.  
Possible uses of this data are for correction of copy number in 
peak-calling for interactome,
transcriptome, DNase hypersensitivity, and methylome determinations.

<H2>Methods</H2>

<H3>Isolation of genomic DNA and hybridization</H3>
Genomic DNA was extracted using the QIAGEN DNeasy Blood & Tissue Kit according
 to the instructions provided by the manufacturer.  
For each biological replicate of each cell line, DNA concentrations and
a level of quality were determined by UV absorbance.
Genotypes were determined from 400 nanograms of each sample at 1 million loci 
using Illumina Human 1M-Duo arrays and standard Illumina protocols.    

<H3>Processing and Analysis </H3>
Genotypes were ascertained from the 1M-Duo Arrays with BeadStudio using 
default settings and formatting with the A/B genotype designation for each SNP 
(see 1M-Duo manifest file for specific nucleotide).  

Copy Number Variation (CNV) analysis was performed using circular binary 
segmentation (DNAcopy) of the log R ratio values at each probe 
(Olshen et al., 2004).  
The parameters used were alpha=0.001, nperm=5000, sd.undo=1.  
Copy number segments are reported with the mean log R ratio for each 
chromosomal segment called by CBS.  
Log ratios of ~-0.2 to -1.5 can be considered heterozygous deletions,
 < -1.5 homozygous deletions, and > 0.2 amplifications.

The coordinates for the genotypes and copy number calls are from Human Genome Build 36.

<H2>Credits</H2>
Tim Reddy, Rebekka Sprouse, Richard Myers, Devin Absher from HudsonAlpha Institute.
<P>
Contact:
<A HREF="mailto:&#116;&#114;&#101;d&#100;y&#64;&#104;&#117;&#100;&#115;&#111;n&#97;lp&#104;&#97;.
&#111;&#114;&#103;">
&#116;&#114;&#101;d&#100;y&#64;&#104;&#117;&#100;&#115;&#111;n&#97;lp&#104;&#97;.
&#111;&#114;&#103;</A>
<!-- above address is treddy at hudsonalpha.org -->

<H2>References</H2>

Olshen, A. et al. (2004).  
<a href="http://www.ncbi.nlm.nih.gov/pubmed/15475419" TARGET=_BLANK>
Circular binary segmentation for the analysis of array-based DNA copy number 
data</a>.<EM>Biostatistics</EM>. 2004 Oct;5(4)557-572.

<h2> Data Release Policy </h2>

<p>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column on the track configuration page and
the download page.  The full data release policy for ENCODE is available
<a href="../ENCODE/terms.html" TARGET=_BLANK>here</a>.</p>

