<h1>HudsonAlpha Methyl-seq Docs</h1>
<a name="Description"></a><h2>Description</h2>
<p>
Methyl-seq is a method developed in the Myers laboratory to measure the
methylation status at CpGs throughout the genome. It combines DNA digestion
by a methyl-sensitive enzyme HpaII and its methyl-insensitive isoschizomer
MspI with the Illumina DNA sequencing platform.  The method was first applied
in a collaboration with the laboratory of Dr. Julie Baker at Stanford
University to study methylation and gene expression changes that occur in
human embryonic stem cells before and after differentiation to definitive
endoderm.  A paper describing the results as well as the method has been
submitted for publication [1].
</p>
<p>
Methyl-seq assays more than 250,000 methyl-sensitive restriction enzyme
cleavage sites, representing more than 90,000 genomic regions.  These regions
include 35,528 annotated CpG islands, while the remaining 55,084 non-CpG
island regions are distributed across the genome in promoters, genes, and
intergenic regions. Sequence tags present in MspI libraries but not in HpaII
libraries are derived from methylated regions.  Conversely, sequence tags that
occur in HpaII libraries come from at least partially unmethylated regions.
</p>
<a name="Methods"></a><h2>Methods</h2>
<h3>Isolation of genomic DNA</h3>
<p>
Genomic DNA is isolated from biological replicates of each cell line by using
the QIAGEN DNeasy Blood & Tissue Kit according to the instructions provided by
the manufacturer. DNA concentrations and a level of quality of each
preparation is determined by UV absorbance.
</p>
<h3>HpaII and MspI digestions</h3>
<p>
For the MspI library, 1 &mu;g genomic DNA is digested in a 100 &mu;l reaction
with 1X NEB Buffer2 and 20 units MspI restriction enzyme and incubated for 18 hr
at 37&deg;C.  For the HpaII library, 1&mu;g genomic DNA is digested in a
100 &mu;l reaction with 1X NEB Buffer1 and 20 units HpaII restriction enzyme
and incubated for 18 hr at 37&deg;C.
</p>
<h3>DNA library construction and sequencing</h3>
<p>
We generate high-throughput sequencing libraries from DNA fragments of the
HpaII or MspI digested genomic DNA according to the protocol posted at the
website: <a
href='http://myers.hudsonalpha.org/content/protocols.html'>http://myers.hudsonalpha.org/content/protocols.html</a>.
We recently modified our approach by removing the first PCR amplification
step, just prior to the gel electrophoresis size-selection step, which we
found reduces a fragment-size bias in the sequencing libraries.  We sequence
these libraries with an Illumina Genome Analyzer (GA2) according to the
manufacturer's recommendations.
</p>
<h3>Data analysis</h3> 
<p>
Reads that align to the genome and contain the CGG
HpaII-cut signature on their 5' end are used for analysis.  Aligned sequence
reads are mapped to CCGG sites predicted <emph>in silico</emph>. Sites with four or more
MspI tags occurring in either the forward or reverse direction are retained in
the analysis.  These "assayable" sites are then grouped with neighboring sites
that are within 35-75 bp.  Thus, a "region" can be comprised of between 2 and
18 digestion sites that are each within 35-75 bp of each another site.
Methylation and non-methylation calls are made by using HpaII tag data from
all assayable cut sites.  The larger of either the forward read count or
reverse read count for each site is averaged across each region.  Regions that
have an average of 0 or 1 read per cut site are called methylated, and regions
with more than one sequence read per site are called unmethylated
</p>
<a name="Credits"></a><h2>Credits</h2>
<a name="References"></a><h2>References</h2>
<p>
<a name='1'>1</a>.
Brunner AL, Johnson DS, Kim SW, Valouev A, Reddy TE, Neff NF, Anton E, Medina C, Nguyen L, 
Chiao E, Oyolu CB, Schroth GP, Absher D, Baker JC, Myers RM.
Distinct DNA methylation patterns characterize differentiated human embryonic stem cells and developing human fetal liver. <b>Submitted</b>
</p>
