<H2>Description</H2>
<P>
These tracks display the level of sequence uniquness of the reference hg18
genome.  They were generated using different window sizes and high signal
will be found in areas where the sequence is not unique.

<!--
<H2> Display Conventions and Configuration </H2>
<P>
This track is a multi-view composite track that contains multiple data types
(<EM>views</EM>).  For each view, there are multiple subtracks that
display individually on the browser.  Instructions for configuring multi-view
tracks are <A HREF="../../goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
ENCODE tracks typically contain one or more of the following views:
<DL>
<DT><I>Peaks</I></DT><DD> Regions of signal enrichment based on processed data
(usually normalized data from pooled replicates). ENCODE Peaks tables contain
fields for statistical significance, including FDR
(<A HREF="../../goldenPath/help/qValue.html" TARGET=_BLANK>qValue</A>).</DD>
<DT><I>Signal</I></DT><DD>Density graph (wiggle) of signal enrichment based on
processed data.</DD>
<DT><I>Raw Signal</I></DT><DD>Density graph (wiggle) of signal from raw data
(single replicates, no control). For high-throughput sequencing data, this
reflects the aligned read density.</DD>
<DT><I>Alignments</I></DT><DD>Mappings of short reads to the genome.<DD>
</DL>
-->
<H2>Methods</H2>
<P>
The Duke uniqueness tracks were generated by self-alignment of a sliding
window of fixed size across the entire genome.  The signal is essentially the
count of alignments of one window to the rest of the genome, normalized to
between 0 and 1, with 0 representing unique sequence.  The window is shifted by
1bp so that each base has a signal that represents the degree to which the
sequence surrounding that base is unique.  Thus, if the window size is 20bp
then a 0 signal at a single base means that the sequence of 39 bases which has
that one base at its center is unique within the entire genome.  A signal of 1
means that there are at least <em>???40???</em> regions within the genome which align to the
region containing the one base.

<H2>Credits</H2>
<P>
These uniqueness tracks were created by a Terry Furey
(contact: <A HREF="mailto:&#116;&#115;&#102;ur&#101;&#121;&#64;&#100;&#117;&#107;&#101;.e&#100;u">
&#116;&#115;&#102;ur&#101;&#121;&#64;&#100;&#117;&#107;&#101;.e&#100;u</A>):
<!-- above address is tsfurey at duke.edu -->
</P><P>
Duke University's <A HREF="http://www.genome.duke.edu/index.php" TARGET=_blank>Institute for Genome Sciences & Policy (IGSP)</A>
Alan Boyle, Lingyun Song, <A HREF="http://www.genome.duke.edu/people/faculty/furey/" TARGET=_blank>Terry Furey</A>, and
<A HREF="http://www.genome.duke.edu/people/faculty/crawford/" TARGET=_blank>Greg Crawford</a>
</P><P>
We thank NHGRI for ENCODE funding support.

</P>
<H2>References</H2>

<P>
 <A HREF="http://www.nature.com/nature/journal/v447/n7146/abs/nature05874.html" TARGET=_blank>
 The ENCODE Project Consortium. Identification and analysis of functional elements in 1% of the human genome by the ENCODE pilot project.</A>
 <EM>Nature</EM> 2007;447:799-816
</P><P>

<P>
Bhinge, A.A., Kim, J., Euskirchen, G.M., Snyder, M., & Iyer, V.R.
 <A HREF="http://genome.cshlp.org/cgi/content/abstract/17/6/910" TARGET=_blank>
 Mapping the chromosomal targets of STAT1 by Sequence Tag Analysis of Genomic Enrichment (STAGE).</A>
 <EM>Genome Res.</EM> 2007;17:910-916.
</P><P>

Boyle AP, Davis S, Shulha HP, Meltzer P, Margulies EH, Weng Z, Furey TS, Crawford GE.
 <A HREF="http://www.cell.com/content/article/abstract?uid=PIIS0092867407016133" TARGET=_blank>
 High-resolution mapping and characterization of open chromatin across the genome.</A>
 <EM>Cell.</EM> 2008 Jan 25;132(2):311-22.
</P><P>

Boyle AP, Guinney J, Crawford GE, and Furey TS.
 <A HREF="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btn480" TARGET=_blank>
 F-Seq: a feature density estimator for high-throughput sequence tags.</A>
 <EM>Bioinformatics.</EM> 2008;24(21):2537-8.
</P><P>

Li H, Ruan J, and Durbin R.
 <A HREF="http://genome.cshlp.org/content/early/2008/09/25/gr.078212.108.short" TARGET=_blank>
 Mapping short DNA sequencing reads and calling variants using mapping quality scores.</A>
 <EM>Genome Res.</EM> 2008;18(11):1851-8.
</P>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column on the track configuration page and
the download page.  The full data release policy for ENCODE is available
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P>
