<h2>Description</h2>
<p>This track displays related tracks of chromatin state maps
generated by the Broad/MGH ENCODE group using ChIP-Seq.
</p>

<p>The ChIP-Seq method involves crosslinking histones and other DNA associated
proteins to genomic DNA within cells using formaldehyde. The crosslinked
chromatin is subsequently extracted, mechanically sheared, and
immunoprecipitated using specific antibodies. After reversal of crosslinks,
the immunoprecipitated DNA is sequenced and mapped to the human reference
genome. The relative enrichment of each antibody-target (epitope) across the
genome is inferred from the density of mapped fragments.</p>

<H2> Display Conventions and Configuration </H2>
<P>
This track is a multi-view composite track that contains multiple data types
(<EM>views</EM>).  For each view, there are multiple subtracks that
display individually on the browser.  Instructions for configuring multi-view
tracks are <A HREF="../../goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
ENCODE tracks typically contain one or more of the following views:
<DL>
<DT><I>Peaks</I></DT><DD> Regions of signal enrichment based on processed data
(usually normalized data from pooled replicates). ENCODE Peaks tables contain
fields for statistical significance, including FDR
(<A HREF="../../goldenPath/help/qValue.html" TARGET=_BLANK>qValue</A>).</DD>
<DT><I>Signal</I></DT><DD>Density graph (wiggle) of signal enrichment based on
processed data.</DD>
<DT><I>Raw Signal</I></DT><DD>Density graph (wiggle) of signal from raw data
(single replicates, no control). For high-throughput sequencing data,
this reflects the aligned read density.</DD>
<DT><I>Alignments</I></DT><DD>Mappings of short reads to the genome.<DD>
</DL>

<!--

<p><b>Signal:</b> These subtracks represent the number of
sequenced and uniquely aligned ChIP fragments across the human genome
at 25bp resolution.</p>
<p><b>Peaks:</b> These subtracks represent discrete
intervals predicted to be significantly enriched with ChIP fragments
using a statistical model. Each interval is associated with an
enrichment ratio and a p-value.</p>
<p>This group of tracks also contains a single special track called
<b>Alignability</b>, which marks unique positions in the
reference genome as 1s and non-unique positions as 0.</p>
-->

<h2>Methods</h2>

<P>
Cells were grown according to the approved
<A HREF="../../ENCODE/protocols/cell" TARGET=_BLANK>ENCODE cell culture protocols</A>.

<p>Cells were fixed in 1% formaldehyde and
resuspended in lysis buffer. Chromatin was sheared to
200-700bp using a Diagenode Bioruptor. Solubilized
chromatin was immunoprecipitated with antibodies against
each of the histone antibodies listed above.
Antibody-chromatin complexes were pulled-down using protein
A-sepharose (or anti-IgM-conjugated agarose for RNA polymerase II),
washed and then eluted. After cross-link reversal and proteinase K
treatment, immunoprecipitated DNA was extracted with
phenol-chloroform, ethanol precipitated, treated
with RNAse and purified. One to ten nanograms of DNA were
end-repaired, adapter-ligated and sequenced by Illumina Genome
Analyzers as recommended by the manufacturer.
</p>

<p>Sequence reads from each IP experiment were aligned to the human
reference genome (hg18) using MAQ with default parameters, except
C=10 (maximum number of alignments/read &lt;= 10).
Fragment densities were computed by counting the number of reads
overlapping each position in the genome (counted as 1 from the start
of each alignment to 200 bp downstream and by 0.25 from 200 to 300 bp,
and displayed at 25bp resolution). Discrete intervals of ChIP-Seq
fragment enrichment were identified using a scan statistics approach,
under the assumption of uniform background signal.
</p>

<h2>Credits</h2>

<p>The ChIP-Seq data were generated at the
 <a href="http://www.broad.mit.edu" title="http://www.broad.mit.edu" rel="nofollow" TARGET = _BLANK>
 Broad Institute</a> and in the
 <a href="http://www.broad.mit.edu/about/associate-members/brad-bernstein" title="http://www.broad.mit.edu/about/associate-members/brad-bernstein" rel="nofollow" TARGET = _BLANK>
 Bradley E. Bernstein lab</a> at the Massachusetts General Hospital/Harvard Medical School.
</p>
<p>Data generation and analysis was supported by funds from the NHGRI, the
Burroughs Wellcome Fund, Massachusetts General Hospital and the Broad Institute.
</p>

<h2>References</h2>
<p>Mikkelsen TS, Ku M, Jaffe DB, Issac B, Lieberman E <em> et al</em>.
 <a href="http://www.nature.com/nature/journal/v448/n7153/abs/nature06008.html" title="http://www.nature.com/nature/journal/v448/n7153/abs/nature06008.html" rel="nofollow" TARGET = _BLANK>
 Genome-wide maps of chromatin state in pluripotent and lineage-committed cells</a>.
 <em>Nature</em>. 2007 Aug 2;448, 553-60. </p>

<p>Bernstein BE, Mikkelsen TS, Xie X, Kamal M, Huebert DJ <em> et al</em>.
 <a href="http://www.cell.com/content/article/abstract?uid=PIIS0092867406003801" title="http://www.cell.com/content/article/abstract?uid=PIIS0092867406003801" rel="nofollow" TARGET = _BLANK>
 A bivalent chromatin structure marks key developmental genes in embryonic stem cells</a>.
 <em>Cell</em>. 2006 Apr 21;125, 315-26. </p>

<p>Bernstein BE, Kamal M, Lindblad-Toh K, Bekiranov S, Bailey DK <em> et al</em>.
 <a href="http://www.cell.com/content/article/abstract?uid=PIIS0092867405000395" title="http://www.cell.com/content/article/abstract?uid=PIIS0092867405000395" rel="nofollow" TARGET = _BLANK>
 Genomic maps and comparative analysis of histone modifications in human and mouse</a>.
 <em>Cell</em>. 2005 Jan 28;120, 169-81.
</p>
