<H2>Description</H2>

<P>This track is produced as part of the ENCODE Project.
This track shows DNaseI sensitivity measured genome-wide in different
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=cellType">cell lines</A>.
using the Digital DNaseI methodology (see below), and DNaseI hypersensitive 
sites.  
DNaseI has long been used to map general chromatin accessibility and 
DNaseI hypersensitivity is a universal feature of active 
<EM>cis</EM>-regulatory sequences. The use of this method has led 
to the discovery of functional regulatory elements that include enhancers, 
insulators, promotors, locus control regions and novel elements.
For each experiment (cell type) this track shows DNaseI hypersensitive zones 
(<I>HotSpots</I>),
hypersensitive sites (<I>Peaks</I>) based on the sequencing tag density 
(<I>Signal</I>).
</P>

<H2>Display Conventions and Configuration</H2>

<P>DNaseI sensitivity is shown as the absolute density of <EM>in vivo</EM> 
cleavage sites across the genome mapped using the Digital DNasI methodology 
(see below).  Data have been normalized to 25 million reads per cell type.  
DNaseI hypersensitive sites (DHSs) at False Discovery Rate (FDR) 0.5% are shown 
within each density track as colored red subregions.</P>
 
<P>The subtracks within this composite annotation track correspond to data from 
different cell types and tissues.  The graphical configuration options are 
shown at the top of the track description page, followed by a list of 
subtracks. To 
display only selected subtracks, uncheck the boxes next to the tracks you 
wish to 
hide. For more information about the graphical configuration options, click 
the <A TARGET=_BLANK HREF="http://genome.ucsc.edu/goldenPath/help/hgWiggleTrackHelp.html">Graph configuration help</A> link.</P>

<DL>
<DT><I>HotSpots</I>
<DD> DNaseI hypersensitive zones identified using the HotSpot algorithm.
<DT><I>Peaks</I>
<DD> DNaseI hypersensitive sites (DHSs) identified as signal peaks within 
FDR 0.5% hypersensitive zones.
<DT><I>Signal</I>
<DD>Density graph (wiggle) of signal enrichment based on aligned read density.
</DL>

<H2>Methods</H2>

<P>
Cells were grown according to the approved
<A TARGET=_BLANK HREF="../../ENCODE/protocols/cell" TARGET=_BLANK>ENCODE cell culture protocols</A>.
Digital DNaseI was performed by performing DNaseI digestion of intact 
nuclei, isolating DNaseI 'double-hit' fragments as described in 
Sabo <EM>et al.</EM> (2006), and direct sequencing of fragment 
ends (which correspond to <EM>in vivo</EM> DNaseI cleavage sites) 
using the Solexa platform (27 bp reads).  Uniquely mapping high-quality reads 
were mapped to the genome.  DNaseI sensitivity is directly reflected in 
raw tag density (<I>Signal</I>), which is shown in the track as density of tags 
mapping within a 150 bp sliding window (at a 20 bp step across the genome).  
DNaseI hypersensitive zones (<I>HotSpots</I>) were identified using the 
HotSpot algorithm 
described in Sabo <EM>et al.</EM> (2004).  0.5% false discovery 
rate thresholds (FDR 0.005) were computed for each cell type by 
applying the HotSpot algorithm to an equivalent number of random 
uniquely mapping 27mers.  DNaseI hypersensitive sites (DHSs or <I>Peaks</I>) 
were identified as signal peaks within FDR 0.5% hypersensitive 
zones using a peak-finding algorithm.</P>

<H2>Verification</H2>

<P>Data were verified by sequencing biological replicates displaying 
correlation coefficient >0.9.  Results are extensively validated by 
conventional DNaseI hypersensitivity assays using end-labeling/Southern 
blotting method. </P>

<H2>Credits</H2>

<P>These data were generated by the UW ENCODE group. </P>
<P>Contact: <A HREF="mailto:&#115;&#117;l&#108;&#64;&#117;.w&#97;&#115;h&#105;&#110;&#103;&#116;&#111;n.&#101;&#100;&#117;">
Richard Sandstrom</A>
<!-- above address is sull at u.washington.edu -->
</P>

<H2>References</H2>

<P>Sabo PJ, Kuehn MS, Thurman R, Johnson BE, Johnson EM, Cao H, Yu M, 
Rosenzweig E, Goldy J, Haydock A, Weaver M, Shafer A, Lee K, Neri F, 
Humbert R, Singer MA, Richmond TA, Dorschner MO, McArthur M, Hawrylycz M, 
Green RD, Navas PA, Noble WS, Stamatoyannopoulos JA.
<A TARGET=_BLANK HREF="http://www.nature.com/nmeth/journal/v3/n7/abs/nmeth890.html">Genome-scale mapping of DNase I sensitivity in vivo using tiling DNA microarrays</A>
<EM>Nature Methods</EM>. 2006;3:511-518.</P>

<P>Sabo PJ, Hawrylycz M, Wallace JC, Humbert R, Yu M, Shafer A, Kawamoto J, 
Hall R, Mack J, Dorschner MO, McArthur M, Stamatoyannopoulos JA.
<A TARGET=_BLANK HREF="http://www.pnas.org/content/101/48/16837">Discovery of functional noncoding elements by digital analysis of chromatin structure.</A>
<EM>PNAS</EM>. 2004;101:16837-16842.</P>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column, above.  The full data release policy
for ENCODE is available
<A TARGET=_BLANK HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P>

