<H2>Description</H2>

<P>This track shows the starts and ends of mRNA transcripts determined by paired-end ditag (PET) sequencing. PETs are composed of 18 bases from both ends of a cDNA; 36 bp PETs were dimerized together and ligated by Illumina GA adaptor as sequencing templates. The diPET templates were amplified on the Illumina Cluster station and subjected to sequencing by synthesis. See the Methods and References sections below for more details on PET sequencing.</P>

<P>The PET sequences in this track are full-length transcripts derived from 4 different polyA+ RNA sources:</P>
<UL>
<LI>The prostate tissues (RNA prepared by Ambion with catalog number ???)</LI>
<LI>K562 cell line cytoplasmic fraction (K562-Cyt)</LI>
<LI>K562 cell line nucleus fraction (K562-Nuc)</LI>
<LI>K562 cell line polysome fraction (K562-Poly)</LI>
</UL>

<P>For different tissue source, growth condition and sub-compartment isolation, please refer to the general description for the ENCODE cell lines at ????.</P>

<P>In the graphical display, the ends are represented by blocks connected by a horizontal line. In full and packed display modes, the arrowheads on the horizontal line represent the direction of transcription, and an ID of the format <EM>XXXXX-N-M</EM> is shown to the left of each PET, where <EM>X</EM> is the unique ID for each PET, <EM>N</EM> indicates the number of mapping locations in the genome (1 for a single mapping location, 2 for two mapping locations, and so forth), and <EM>M</EM> is the number of PET sequences at this location. The total count of PET sequences mapped to the same locus but with slight nucleotide differences may reflect the expression level of the transcripts. PETs that mapped to multiple locations may represent low complexity or repetitive sequences.</P>

<H2>Display Conventions and Configuration</H2>

<P></P>

<H2>Methods</H2>

<P>PolyA+ RNA was isolated from the cells. A full-length cDNA library was constructed and converted into a PET library for Gene Identification Signature analysis. Generation of PET sequences involved cloning of cDNA sequences into the plasmid vector pGIS3. pGIS3 contains two <EM>MmeI</EM> recognition sites that flank the cloning site, which were used to produce a 36 bp PET. Each 36 bp PET sequence contains 18 bp from each of the 5' and 3' ends of the original full-length cDNA clone. The 18 bp 3' signature contains 16 bp 3'-specific nucleotides and an AA residual of the polyA tail to indicate the sequence orientation. PET sequences were mapped to the genome using the following specific criteria:</P>

<UL>
<LI>A minimal continuous 16 bp match must exist for the 5' signature; the 3' signature must have a minimal continuous 14 bp match </LI> 
<LI>Both 5' and 3' signatures must be present on the same chromosome </LI>
<LI>Their 5' to 3' orientation must be correct (5' signature followed by 3' signature)</LI> 
<LI>The maximal genomic span of a PET genomic alignment must be less than one million bp</LI> 
</UL>

<P>PETs mapping to 2-10 locations are also included and may represent duplicated genes or pseudogenes in the genome.</P>

<H2>Verification</H2>

<P>To assess overall PET quality and mapping specificity, the top ten most abundant PET clusters that mapped to well-characterized known genes were examined. Over 99% of the PETs represented full-length transcripts, and the majority fell within ten bp of the known 5' and 3' boundaries of these transcripts. The PET mapping was further verified by confirming the existence of physical cDNA clones represented by the ditags. PCR primers were designed based on the PET sequences and amplified the corresponding cDNA inserts from the parental GIS flcDNA library for sequencing analysis. </P>

<H2>Credits</H2>

<P>The GIS-PET libraries and sequence data for transcriptome analysis were produced at the <A HREF="http://www.gis.a-star.edu.sg/internet/site">Genome Institute of Singapore</A>. The data were mapped and analyzed by scientists from the Genome Institute of Singapore.</P>

<H2>References</H2>
