<H2>Description</H2>

<P> This track depicts high throughput sequencing of small RNAs (&lt;200 nt) 
from whole cell RNA samples from tissues or sub cellular compartments from 
cell lines included in the ENCODE Transcriptome subproject. The overall 
goal of the ENCODE project is to identify and characterize all functional 
elements in the sequence of the human genome.
</P>

<H2>Methods</H2>

<P>RNA libraries were generated by treatment of short RNA (&lt;200 nt) with 
Tobacco Acid Pyrophsophatase, followed by: 
1) C-tailing, 
2) 5&prime; ligation with adapter 
5&prime; rCrGrArCrUrGrGrArGrCrArCrGrArGrGrArCrArCrUrGrArCrArUrGrGrArCrUrGrArArGrGrArGrUrArGrArArA 3&prime;, or 
5&prime; rCrArCrGrArGrGrArCrArCrUrGrArCrArUrGrGrArCrUrGrArArGrGrArGrUrArGrArArA 3&prime;, 
3) reverse transcription with primer 5&prime; TCGCGAGCGGCCGCGGGGGGGGGGGGGGG 3&prime;, and 
4) PCR with primers 5&prime; AATGATACGGCGACCACCGACACGAGGACACTGACATGGACTGAAGGAGTAGAAA 3&prime; and
 5&prime; CAAGCAGAAGACGGCATACGATCGCGAGCGGCCGCGGGGGG 3&prime;. 
Gel-purified libraries were sequenced with primer 
5&prime; CACGAGGACACTGACATGGACTGAAGGAGTAGAAA 3&prime; on Illumina GA1 or GA2 platform. 
</P>

<H2>Analysis</H2>

<P>Sequence reads underwent quality filtration using Illumina standard pipeline (Gerlad). 
Identical reads were collapsed while maintaining their multiplicity information. The 
3&prime; sequencing adaptor was removed from the reads using a custom clipper program, 
which aligned the adaptor sequence to the short-reads, allowing up to 2 mismatches and 
no indels. The trimmed sequences were aligned to the human genome (NCBI build 36, hg18) 
using Nexalign (Lassmann <EM>et al.</EM>, not published). The alignment parameters 
tuned to tolerate up to 2 mismatches and no indels, and to report up to 100 hits of 
a multimappers. The best hit/s were reported in a bed-like format. 
</P>

<H2>Verification</H2>

<P>Comparison of referential data generated from 8 individual sequencing lanes (Illumina technology). 
</P>

<H2>Credits</H2>

<P>AFFYMETRIX, Inc., Cold Spring Harbor Laboratories, and the Center for Genomic 
Regulation (Barcelona), who are participants in the ENCODE Transcriptome Group.
</P>

