<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=315>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Caenorhabditis+briggsae"
		TARGET=_blank>
    <IMG SRC="/images/Caenorhabditis_briggsae.jpg" WIDTH=300 HEIGHT=238 ALT="C. briggsae"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Caenorhabditis briggsae</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of
      <A HREF="http://www.life.umd.edu/biology/faculty/haag/Haag_Lab_Members_Page.html" 
      TARGET=_blank>Eric Haag,</A> All Rights Reserved.
    </FONT>
  </TD></TR>
</TABLE>

<P>
The Jan. 2007 <em>Caenorhabditis briggsae</em> draft assembly 
(Washington University School of Medicine GSC and Sanger Institute cb3) was produced by the *** sequencing center.

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
C. briggsae genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>chrZ</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all of chromosome Z</TD>
    </TR>
    <TR><TD VALIGN=Top>chr3:1-1000000</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first million bases of chr 3</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr3:1000000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chr 3 that spans 2000 bases, starting with position 1000000 </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>BX950328</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession
		number BX950328</TD>
    </TR>
    <TR><TD VALIGN=Top>BU335953</TD>
        <TD WIDTH=14>&nbsp;</TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession
		BU335953 on chr Z</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>pseudogene mRNA</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes, but not cDNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>homeobox caudal</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top>zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top>zhang</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Zhang</TD>
    </TR>
    <TR><TD VALIGN=Top>Taguchi,T.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author T. Taguchi</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Taguchi T</em>, internally it
	uses the format <em>Taguchi,T.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
This version was assembled from 2.05 million whole genome shotgun
(WGS) reads, of which 88.2% were in read pairs. The sequence reads
were generated at Washington University School of Medicine's Genome
Sequencing Center and the Sanger Centre (Hinxton, UK).  The Phusion
(Mullikin, J) assembler was used to assemble the WGS reads into
contigs based on overlap information, and then into supercontigs using
read pair information to cross gaps. These supercontigs were assembled
into mapped ultracontigs based on FPC fingerprint mapping (generated
at Washington University School of Medicine's Genome Sequencing
Center), with material from previously finished clones used to bridge
gaps. Using the sequence data to choose primers along major
ultracontigs, a genetic map (Ray Miller, Washington University School
of Medicine) was constructed which yielded 6 chromosomes.  The genetic
mapping data along with data from 1:1 elegans:briggsae orthologs were
used to refine ultracontigs and to order and orient the sequences
along the C. briggsae chromosomes.
</P>
<P>
The six chromosomes are named: chrI, chrII, chrIII, chrIV, chrV and chrX,
containing 91,234,787 bp of ordered and oriented sequence.  
An additional 9,932,049 bp of sequence is localized to a
specific chromosome but not ordered and displayed on the artifical
chr*_random chromosomes.  7,311,690 bp of sequence is unordered and
not localized, displayed on the chrUn artifical chromosome.
</P>
<P>
The sequencing centers
estimate that this whole genome shotgun assembly achieved 98% coverage
of the genome.
</P>
<P>
The briggsae 
mitochondrial sequence is also available as the virtual chromosome 
&quot;chrM&quot;. 
</P>


<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/cb3/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#c_briggsae">Downloads</A> page. 

  *** Developer -- remove this if no conditions for use:
These data have
<A HREF="../goldenPath/credits.html#c_briggsae_use">specific 
conditions for use</A>. 

The C. briggsae browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#c_briggsae_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
