<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=315>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Caenorhabditis+briggsae"
		TARGET=_blank>
    <IMG SRC="/images/Caenorhabditis_briggsae.jpg" WIDTH=300 HEIGHT=238 ALT="C. briggsae"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Caenorhabditis briggsae</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of
      <A HREF="http://www.life.umd.edu/biology/faculty/haag/Haag_Lab_Members_Page.html" 
      TARGET=_blank>Eric Haag,</A> All Rights Reserved.
    </FONT>
  </TD></TR>
</TABLE>

<P>
The Jan. 2007 <em>Caenorhabditis briggsae</em> draft assembly 
produced by the
<A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Caenorhabditis+briggsae"
	TARGET=_blank>
Washington University School of Medicine GSC</A> with the following
<A HREF="http://genome.wustl.edu/data.cgi" TARGET=_blank>data usage policy.</A>

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
C. briggsae genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
       	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><BR></TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>chrI</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays all of chromosome I</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>chrI:1-1000000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays the first million bases of chrI</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chrV:1000000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chrVn that spans 2000 bases, starting at position 1000000 </TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>R02870</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of EST associated with GenBank accession R02870</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>U48289</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays the region associated with the mRNA accession U48289</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>protein phosphatase PP2C </TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists only records pertaining to protein phophatase PP2C</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>Kimble,J.</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author J. Kimble</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries.
	Although GenBank requires the search format <em>Kimble J</em>,
	internally it uses the format <em>Kimble,J.</em>.</TD>
    </TR>
</TABLE>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>

<HR>
<H3>Assembly details</H3>
<P>
See also:
<A HREF="ftp://genome.wustl.edu/pub/organism/Invertebrates/Caenorhabditis_briggsae/assembly/Caenorhabditis_briggsae-1.0/ASSEMBLY"
	TARGET=_blank>genome.wustl.edu</A> assembly description.
</P>
<P>
The six chromosomes are named: chrI, chrII, chrIII, chrIV, chrV and chrX,
containing 91,234,787 bp of ordered and oriented sequence.  
An additional 9,932,049 bp of sequence is localized to a
specific chromosome but not ordered and displayed on the artifical
chr*_random chromosomes.  7,311,690 bp of sequence is unordered and
not localized, displayed on the chrUn artifical chromosome.
</P>
<P>
The sequencing centers
estimate that this whole genome shotgun assembly achieved 98% coverage
of the genome.
</P>
<P>
The briggsae mitochondrial sequence, length 14,420 bp, is also available
as the virtual chromosome &quot;chrM&quot;.  Genbank accession:
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/viewer.fcgi?db=nucleotide&val=95102164"
	TARGET=_blank>AC186293.1</A>
</P>

<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/cb3/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#c_briggsae">Downloads</A> page. 

These data have
<A HREF="http://genome.wustl.edu/data.cgi" TARGET=_blank>specific conditions for use</A>. 

The C. briggsae browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#c_briggsae_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
