<H2>Description</H2>
<P>
This track shows chimp/$organism genomic
alignments using
a gap scoring system that allows longer gaps than traditional
affine gap scoring systems.  It can also tolerate gaps
in both chimp and $organism simultaneously.  These &quot;double-sided&quot;
gaps can be caused by local inversions and overlapping deletions
in both species. 
<P>
The graphical display shows boxes joined together by either single or 
double lines. The boxes represent aligning regions. Single lines 
indicate gaps that are largely due to missing sequence in the 
non-$organism species. Double lines represent gaps in the chain where 
sequence are present in both species but do not align.
In cases where multiple chains exist over a particular portion of the 
$organism genome, chains with single-lined gaps are frequently the result 
of processed pseudogenes, while chains with double-lined gaps are more 
often due to paralogs and unprocessed pseudogenes. In the fuller display
modes, the individual feature names indicate the scaffold name, strand, and 
location (in thousands) of the match for each matching alignment.
<P>
The chimp sequence used in these alignments is from 
the 13 Nov. 2003 Arachne assembly.</P>

<H2>Methods</H2>
<P>
Transposons inserted since the chimp/$organism
split were removed, and the resulting abbreviated genomes were
aligned with blastz. The transposons were then put back into the
alignments. 
<P>
The resulting alignments were converted into axt format,
and these axts were fed into the axtChain program. AxtChain organizes all 
the alignments between a single chimp scaffold and a single $organism 
chromosome into a group and makes a kd-tree out of all the gapless subsections
(blocks) of the alignments. The maximally-scoring chains of these
blocks were found by running a dynamic program over the kd-tree.  Chains
scoring below a certain threshold were discarded. 
<P>
To place additional chimp scaffolds that weren't initially aligned by blastz, 
an additional DNA blat of the unmasked sequence was performed. The resulting
blat alignments were chained using axtChain, and then merged with the 
blastz-based chains produced in the previous step. These chains are
displayed in this track.  
</P>

<H2>Credits</H2>
The chimp sequence used in this track was obtained from the 13 Nov. 2003 Arachne 
assembly. We'd like to thank the National Human Genome Research Institute 
(NHGRI), the Eli & Edythe L. Broad Institute at MIT/Harvard, and Washington University 
School of Medicine for providing this sequence.</P>
<P>
Blastz was developed at <A HREF="http://bio.cse.psu.edu">Pennsylvania State University</A> by 
Scott Schwartz, Zheng Zhang, and Webb Miller with advice from
Ross Hardison.</P>
<P>
Lineage-specific repeats were identified by Arian Smit and his
program <A HREF="http://repeatmasker.genome.washington.edu/cgi-bin/RepeatMasker">RepeatMasker</A>.</P>
<P>
The axtChain program was developed at the University of California
at Santa Cruz by Jim Kent with advice from Webb Miller and David Haussler.</P>
<P>
The browser display and database storage of the chains were made
by Robert Baertsch and Jim Kent.</P>

<H2>References</H2>
<P>
<A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>Human-Mouse Alignments with BLASTZ</A>. 
Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison R, Haussler H, 
and Miller W.  <em>Genome Research</em> 2003 Jan;13(1):103-7.</P>
<P>
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" TARGET=_blank>Scoring pairwise genomic sequence alignments</A>. 
Chiaromonte F, Yap VB, Miller W. <em>Pac Symp Biocomput</em> 2002;:115-26.</P>

