<H2>Description</H2>
<P>
This annotation shows regions detected as putative copy number polymorphisms
(CNP) and sites of detected intermediate-sized structural variation (ISV). 
The CNPs and ISVs were determined by four different methods, displayed in 
individual subtracks within the annotation:</P>
<UL>
<LI>
<B>BAC microarray analysis (Iafrate):</B> 255 putative CNP regions detected by BAC 
microarray analysis in a population of 55 individuals, 16 of which had 
previously-characterized chromosomal abnormalities.
<LI>
<B>BAC microarray analysis (Sharp):</B> 160 putative CNP regions detected by BAC
microarray analysis in a population of 47 individuals comprised of 8 
Chinese, 4 Japanese, 10 Czech, 2 Druze, 7 Biaka, 9 Mbuti, and 7 Amerindians. 
<LI>
<B>Representational oligonucleotide microarray analysis (ROMA) (Sebat):</B> 81 putative
CNP regions detected by ROMA in a population of 20 normal individuals.
<LI>
<B>Fosmids mapping:</B> 285 ISV sites detected by mapping paired-end sequences 
from a human fosmid DNA library against the July 2003 human genome assembly 
(NCBI Build 34).
</UL></P>

<H2>Display Conventions and Configuration</H2>
<P>
CNP and ISV regions are indicated by solid blocks that are color-coded to 
indicated the type of variation detected:
<UL>
<LI>
<B><FONT COLOR="green">Green</FONT>:</B> gain (duplications)
<LI>
<B><FONT COLOR="red">Red</FONT>:</B> loss (deletions)
<LI>
<B><FONT COLOR="blue">Blue</FONT>:</B> gain and loss (both deletion and duplication)
<LI>
<B>Black:</B> inversion
</UL></P>
<P>
<B>Sharp subtrack: </B>On the details pages for elements in this subtrack, 
the &quot;Disease Percent&quot; value reflects the percent of the BAC that lies 
within a &quot;rearrangement hotspot&quot;, as defined in Sharp <em>et al</em>. 
(2005) (the rationale used to choose BACs for the array construction). A 
rearrangement hotspot is defined by the presence of flanking intrachromosomal 
duplications &gt;10 kb in length with &gt;95% similarity and separated by 
50 kb - 10 Mb of intervening sequence.</P>
<P>
<B>Fosmids subtrack: </B>Items are labeled using the following naming convention:
<UL>
<LI><B>First letter:</B> rearrangement type (<B>D</B>=deletion, <B>I</B>=insertion, 
<B>V</B>=inversion).
<LI><B>Second letter:</B> association with repeat or duplication
(<B>R</B>=human-specific repeat, <B>D</B>=duplication, <B>N</B>=neither 
(unique)).
<LI><B>Third letter:</B> second haplotype support (<B>N</B>=variant site lacking 
support from the human genome reference, <B>S</B>=variant site with support 
from the human genome reference). 
</UL></P>

<H2>Methods</H2>

<H3>Sharp BAC microarray analysis</H3>
<P>
All hybridizations were all performed in duplicate incorporating a dye-reversal 
using a custom array consisting of 2194 end-sequence or FISH-confirmed BACs, 
targeted to regions of the genome flanked by segmental duplications. 
The false positive rate was estimated at ~3 clones per 4,000 tested.</P>

<H3>Iafrate BAC microarray analysis</H3>
<P>
All hybridizations were performed in duplicate incorporating a dye-reversal 
using proprietary 1 Mb GenomeChip V1.2 Human BAC Arrays consisting of 2,632 BAC 
clones (Spectral Genomics, Houston, TX). The false positive rate was estimated 
at ~1 clone per 5,264 tested. </P>
<P>
Further information is available from the 
<A HREF="http://projects.tcag.ca/variation/" TARGET=_blank>Database of Genome 
Variants</A> website.</P>

<H3>Sebat ROMA</H3>
<P>
Following digestion with BglII or HindIII, genomic DNA was hybridized to a 
custom array consisting of 85,000 oligonucleotide probes. The probes were 
selected to be free of common repeats and have unique homology within the human 
genome. The average resolution of the array was ~35kb; however, only intervals 
in which three consecutive probes showed concordant signal were scored as 
CNPs. All hybridizations were performed in duplicate incorporating a 
dye-reversal, with the false positive rate estimated to be ~6%.</P>
<P>
Note that CNP intervals, as detailed by Sebat <em>et al</em>. (2004), were 
converted from the April 2003 human genome assembly (NCBI Build 33) to the 
July 2003 assembly (NCBI Build 34) using the UCSC liftOver tool.</P>

<H3>Fosmids mapping</H3>
<P>
Paired-end sequences from a human fosmid DNA library were mapped to the July 
2003 human genome assembly (NCBI Build 34). The average resolution of this 
technique was ~8kb, and included 56 sites of inversion not detectable by 
the array-based approaches.  However, because of the physical constraints of 
fosmid insert size, this technique was unable to detect insertions greater than 
40 kb in size.</P>

<H2>References</H2>
<P>
Iafrate, J.A., Feuk, L., Rivera, M.N., Listewnik, M.L., Donahoe, P.K., Qi, Y., 
Scherer, S.W. and Lee, C.  
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=15286789&query_hl=1"
TARGET=_blank>Detection of large-scale variation in the human genome</A>.
<em>Nature Genet</em> <B>36</B>(9), 949-51 (2004). </P>
<P>
Sebat, J., Lakshmi, B., Troge, J., Alexander, J., Young, J., Lundin, P., 
Maner, S., Massa, H., Walker, M., Chi, M. <em>et al</em>.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=pubmed&amp;dopt=Abstract&amp;list_uids=15273396&amp;query_hl=3"
TARGET=_blank>Large-scale copy number polymorphism in the human genome</A>.
<em>Science</em> <B>305</B>(5683), 525-8 (2004).</P>
<P>
Sharp, A.J., Locke, D.P., McGrath, S.D., Cheng, Z., Bailey, J.A., Samonte, R.V.,
Pertz, L.M., Clark, R.A., Schwartz, S., Segraves, R. <em>et al</em>.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=15918152&query_hl=3"
TARGET=_blank>Segmental duplications and copy number variation in the human 
genome</A>. 
<em>Am J Hum Genet</em> <B>77</B>(1), 78-88 (2005).</P>
<P>
Tuzun, E., Sharp, A.J., Bailey, J.A., Kaul, R., Morrison, V.A., Pertz, L.M., 
Haugen, E., Hayden, H., Albertson, D.  Pinkel, D. <em>et al</em>.
<A HREF="http://www.nature.com/ng/journal/v37/n7/abs/ng1562.html;jsessionid=3B6D5A840A23E79306831482B898025D"
TARGET=_blank>Fine-scale structural variation of the human genome</A>. 
<em>Nature Genet</em> <B>37</B>(7), 727-32 (2005). </P>
