<H2>Display Colors</H2>

The color of the elements in this display indicates the type of variation that was detected.
<table>
<TR> <TD><font color="green"> Green &nbsp; </font></TD><TD> Gain: Duplications </TD></TR>
<TR> <TD><font color="red">   Red          </font></TD><TD> Loss: Deletions    </TD></TR>
<TR> <TD><font color="blue">  Blue         </font></TD><TD> Gain and Loss: Both deletion and duplication </TD></TR>
</table>


<H2>Iafrate Description</H2>
<P>
This track shows 255 regions detected as putative copy number polymorphisms by BAC microarray analysis 
in a population of 55 individuals, 16 of which had previously characterized chromosome abnormalities.

<H2>Iafrate Methods</H2>
<P>
Hybridizations were all performed in duplicate incorporating a dye-reversal using proprietary 1Mb 
GenomeChip V1.2 Human BAC Arrays consisting of 2,632 BAC clones (Spectral Genomics, Houston, TX). 
The false positive rate was estimated as ~1 clone per 5,264 tested. 
<P>
Further information is available at 
<a href="http://projects.tcag.ca/variation/">http://projects.tcag.ca/variation</a>.

<H2>Iafrate References</H2>
<P>
Iafrate JA, Feuk L, Rivera MN, Listewnik ML, Donahoe PK, Qi Y, Scherer SW, Lee C (2004) 
<a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=15286789&query_hl=1 ">
Detection of large-scale variation in the human genome</a>.
<I>Nature Genet</I> 36:949-951


<H2>Sebat Description</H2>
<P>
This track shows 81 regions detected as putative copy number polymorphisms by representational 
oligonucleotide microarray analysis (ROMA) in a population of 20 normal individuals. 

<H2>Sebat Methods</H2>
<P>

Following digestion of with BglII or HindIII, genomic DNA was hybridized to a 
custom array consisting of 85,000 oligonucleotide probes, 
probes were selected to be free of common repeats and have unique homology within the human genome. 

The average resolution of the array is ~35kb, however only intervals in which 3 consecutive probes 
showed concordant signal were scored as CNPs. 

All hybridizations were performed in duplicate incorporating a dye-reversal, 
with the false positive rate estimated to be ~6%.
<br>
Note that CNP intervals as detailed by Sebat et al. (2004) were converted from the 
April 2003 (build33) into the July 2003 (build34) assembly using liftOver.

<H2>Sebat References</H2>
<P>
Sebat J, Lakshmi B, Troge J, Alexander J,
Young J, Lundin P, Maner S, Massa H, Walker M, Chi M, Navin N, Lucito R, Healy
J, Hicks J, Ye K, Reiner A, Gilliam TC, Trask B, Patterson N, Zetterberg A, Wigler
M (2004) <a
href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=pubmed&amp;dopt=Abstract&amp;list_uids=15273396&amp;query_hl=3">
Large-Scale Copy Number Polymorphism in the Human Genome</a>.
<i>Science</i> <b>305</b>:525-528



<H2>Sharp Description</H2>
<P>
This track shows 160 regions detected as putative copy number 
polymorphisms by BAC microarray analysis in a population of 47 individuals, 
comprising 8 Chinese, 4 Japanese, 10 Czech, 2 Druze, 7 Biaka, 9 Mbuti, 
and 7 Amerindians. 

<H2>Sharp Methods</H2>
<P>
Hybridizations were all performed in duplicate incorporating a dye-reversal 
using a custom array consisting of 2194 end-sequence or FISH confirmed BACs, 
targeted to regions of the genome flanked by segmental duplications. 
The false positive rate was estimated as ~3 clones per 4,000 tested.

<H2>Sharp References</H2>
<P>
Sharp AJ, Locke DP, McGrath SD, Cheng Z, Bailey JA, Samonte RV, 
Pertz LM, Clark RA, Schwartz S, Segraves R, Oseroff VV, Albertson DG, 
Pinkel D, Eichler EE (in press) 
Segmental duplications and copy number variation in the human genome. 
<I>Am J Hum Genet</I>
