<H2>Description</H2>
<P>
This track displays DNaseI-hypersensitive sites in CD4+ T-cells
before and after activation.  DNaseI-hypersensitive sites are
associated with gene regulatory regions, particularly for upregulated
genes.  CD4+ T-cells, also known as helper or inducer T cells, are
involved in generating an immune response.  CD4+ T-cells are also
one of the primary targets of the HIV virus.</P>
<P>
The top subtrack corresponds to unactivated T cells; the bottom
subtrack shows activated T cells.  Within the subtracks, the gray and black 
lines represent probable hypersensitive sites. The darker the lines, the more 
likely the site is to be hypersensitive.</P>

<H2>Methods</H2>
<P>
DNaseI-hypersensitive sites were cloned from primary human CD4+ T-cells and
sequenced using massively parallel signature sequencing 
(Brenner <em>et al</em>., 2000; Crawford and Collins, unpublished data). 
Only those clusters of multiple DNaseI library sequences that map within 500 
bases of each other are displayed.
Each cluster has a unique identifier, visible when the track is displayed
in full or packed mode. The last digit of each identifier represents the 
number of sequences that map within that particular cluster.  </P>

<H2>Verification</H2>
<P>
Real-time PCR assay was used to verify valid
DNaseI-hypersensitive sites.  Approximately 50% of
clusters of two sequences are valid.  These clusters are shown
in light gray. 80% of clusters of three sequences are valid, indicated by
dark gray. 100% of clusters of four or more
sequences are valid, shown in black.</P>
<P>
This data set includes confirmed elements for 35 of the 44
ENCODE regions.
It is estimated that these data identify
10-20% of all hypersensitive sites within CD4+ T cells. Further
sequencing will be required to identify additional sites.</P>

<H2>Credits</H2>
<P>
These data were produced at the 
<A HREF="http://www.genome.gov/10000351" TARGET=_blank>Collins Lab</A> 
at NHGRI.  Thanks to Gregory E. Crawford and Francis S. Collins for supplying 
the information for this track.</P>

<H2>References</H2>
<P>
Brenner, S. <em>et al</em>.
<A HREF="http://www.nature.com/cgi-taf/DynaPage.taf?file=/nbt/journal/v18/n6/abs/nbt0600_630.html&dynoptions=doi1097883799" 
TARGET=_blank>Gene expression analysis by massively parallel signature 
sequencing (MPSS) on microbead arrays</A>.
<em>Nat. Biotechnol.</em> <B>18</B>(6), 630-4 (2000).</P>
<P>
Crawford, G.E. <em>et al</em>.
<A HREF="http://www.pubmedcentral.nih.gov/articlerender.fcgi?tool=pubmed&pubmedid=14732688" 
TARGET=_blank>Identifying gene regulatory elements by genome-wide recovery of 
DNase hypersensitive sites</A>.
<em>Proc. Natl. Acad. Sci. USA</em> <B>101</B>(4), 992-7 (2004).</P>
<P>
McArthur, M., Gerum, S., and Stamatoyannopoulos, G.
<A HREF="http://www.sciencedirect.com/science?_ob=ArticleURL&_udi=B6WK7-457CXYS-4M&_coverDate=10%2F12%2F2001&_alid=210456595&_rdoc=1&_fmt=&_orig=search&_qd=1&_cdi=6899&_sort=d&view=c&_acct=C000050221&_version=1&_urlVersion=0&_userid=10&md5=5a2dd2b59e368e190942669df47d72c8"
TARGET=_blank>Quantification of DNaseI-sensitivity by real-time PCR: quantitative analysis of DNaseI-hypersensitivity of the mouse beta-globin LCR</A>.
<em>J. Mol. Biol.</em> <B>313</B>(1), 27-34 (2001).
