<H2>Description</H2>
<P>
This track displays multiple sequence alignments 
for 23 vertebrates in
the ENCODE regions, based on comparative sequence
data generated for the ENCODE project.
The alignments in this track were generated using the
Threaded Blockset Aligner (TBA).
<P>
The Genome Browser companion tracks, TBA Conservation and TBA Elements, display 
conservation scoring and conserved elements for these alignments based on 
various conservation methods.</P>

<H2>Display Conventions and Configuration</H2>
<P>
In full display mode, this track shows pairwise alignments
of each species aligned to the human genome.  
The pairwise alignments are shown in dense display mode using a grayscale
density gradient.  The checkboxes in the track configuration section allow
the exclusion of species from the pairwise display.
<P>
When zoomed-in to the base-display level, the track shows the base
composition of each alignment. The numbers and symbols on the &quot;human
gap&quot; line indicate the lengths of gaps in the human sequence at those
alignment positions relative to the longest non-human sequence. If the gap
size is greater than 9, a non-numeric symbol is shown; if the 
gap size is a multiple of 3, a &quot;*&quot; is displayed, 
otherwise &quot;+&quot; is shown.
To view detailed information about the
alignments at a specific position, zoom in the display to 30,000 or fewer 
bases, then click on the alignment.  </P>

<H2>Methods</H2>
<P>
The TBA was used to align sequences in the
<A HREF="http://genome.ucsc.edu/ENCODE/newsarch.html#052405b"
TARGET=_blank>May 2005 ENCODE sequence data freeze</A>.
The multiple alignments were seeded from a series of 
combinatorial pairwise blastz alignments (not referenced to any one species). 
The specific combinations were determined by the
<A HREF= "ftp://kronos.nhgri.nih.gov/pub/outgoing/elliott/encode_tba/MAY-2005/phylo/tree.nh"
TARGET=_blank>species guide tree</A>. Additionally, a 
<A HREF="ftp://kronos.nhgri.nih.gov/pub/outgoing/elliott/tba/aux/blastz.specs"
TARGET=_blank>blastz.specs file</A>
was used to fine-tune the blastz parameters, based on the evolutionary
distance of the species being compared.
The resulting multiple alignments were projected onto the human reference
sequence. </P>

<H2>Credits</H2>
<P>
The multiple alignments for this track were created by
Elliott Margulies of the Green Lab at NHGRI. 
The alignments were generated using the programs
Blastz and TBA from Scott Schwartz and Webb Miller of the
<A HREF="http://bio.cse.psu.edu/" TARGET=_blank>Penn State Bioinformatics 
Group</A>.

<H2>References</H2>
<P>
Blanchette, M., Kent, W.J., Reimer, C., Elnitski, L., Smit, A.,
Roskin, K., Baertsch, R., Rosenbloom, K.R., Clawson, H. <em>et al.</em>
<A HREF="http://dx.doi.org/10.1101/gr.1933104"
TARGET=_blank>Aligning Multiple Genomic Sequencs With the Threaded Blockset 
Aligner</A>.
<em>Genome Res</em> <B>14</B>, 708-15 (2004).</P>
<P>
Chiaromonte, F., Yap, V.B., and Miller, W. 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" 
TARGET=_blank>Scoring pairwise genomic sequence alignments</A>.
<em>Pac Symp Biocomput</em> 2002, 115-26 (2002).</P>
<P>
Schwartz, S., Kent, W.J., Smit, A., Zhang, Z., Baertsch, R., Hardison, R., 
Haussler, D. and Miller, W.
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103" 
TARGET=_blank>Human-Mouse Alignments with BLASTZ</A>.
<em>Genome Res</em> <B>13</B>(1):103-7 (2003).</P>
<P>
