<H2>Description</H2>
<P>
This track displays multiple sequence alignments and
evolutionary conservation  for 23 vertebrates in
the ENCODE regions, based on comparative sequence
data generated for the ENCODE project.
The alignments in this track were generated using the
Threaded Blockset Aligner (TBA), and the conservation scoring
was produced by the phastCons program.
<P>
Companion tracks in the browser display conservation scoring
and conserved elements for these alignments based on phastCons and 
binCons (a binomial-based conservation method).
<P>
In full display mode, this track shows the overall conservation score 
across all species as well as pairwise alignments
of each species aligned to the
human genome.  The pairwise alignments are
shown in dense display mode using a grayscale
density gradient.  The checkboxes in the track configuration section allow
the exclusion of species from the pairwise display; however, this does not
remove them from the conservation score display.  </P>
<P>
When zoomed-in to the base-display level, the track shows the base
composition of each alignment. The numbers and symbols on the &quot;human
gap&quot; line indicate the lengths of gaps in the human sequence at those
alignment positions relative to the longest non-human sequence. If the gap
size is greater than 9, a non-numeric symbol is shown; if the 
gap size is a multiple of 3, a &quot;*&quot; is displayed, 
otherwise &quot;+&quot; is shown.
symbol is displayed. To view detailed information about the
alignments at a specific position, zoom in the display to 30,000 or fewer bases,
then click on the alignment.  </P>
<P>
This track may be configured in a variety of ways to highlight different aspects
of the displayed information. Click the &quot;Graph configuration help&quot; 
link for an explanation of the configuration options.

<H2>Methods</H2>
<P>
TBA was used to align sequences in the
<A TARGET=_BLANK HREF="http://genome.ucsc.edu/ENCODE/newsarch.html#052405b">
May-2005 ENCODE sequence data freeze</A>
The multiple alignments were seeded from a series of 
combinatorial pair-wise blastz alignments (not referenced to any one species),
the specific combinations were determined by the
<A TARGET=_BLANK HREF= "ftp://kronos.nhgri.nih.gov/pub/outgoing/elliott/encode_tba/MAY-2005/phylo/tree.nh">
species guide tree </A>
Additionally, a 
<A TARGET=_BLANK HREF="ftp://kronos.nhgri.nih.gov/pub/outgoing/elliott/tba/aux/blastz.specs">
blastz-specs file</A>
was used to fine-tune the blastz parameters, based on the evolutionary
distance of the species being compared.
<P>
The resulting multiple alignments were projected onto the human reference
sequence. 
<P>
The phastCons program was used to produce base-by-base conservation
scores for this track.  More detail on phastCons can be
found in the TBA Conservation track description.

<H2>Credits</H2>
<P>
The multiple alignments for this track were created by
Elliott Margulies of the Green Lab at NHGRI.  He also produced 
the phastCons scoring, with assistance from Adam Siepel of UCSC.
The following programs were used to generate this track:
<UL>
<LI>
Blastz and TBA from Scott Schartz and Webb Miller of the
<A TARGET=_BLANK HREF="http://bio.cse.psu.edu/" TARGET=_blank>Penn State Bioinformatics 
Group</A>.
<LI>
Tree estimation (phyloFit) and conservation-scoring (phastCons)
software by Adam Siepel at UCSC.
</UL>
</P>

<H2>References</H2>
<P>
M. Blanchette, W. J. Kent, C. Reimer, L. Elnitski, A. Smit,
K. Roskin, R. Baertsch, K. R. Rosenbloom, H. Clawson, E. D. Green,
D. Haussler, and W. Miller.
<A HREF="http://dx.doi.org/10.1101/gr.1933104">
Aligning Multiple Genomic Sequencs With the Threaded Blockset Aligner</A>
<P>
A. Siepel, G. Bejerano, J. S. Pedersen, A. Hinrichs, M. Hou,
K. Rosenbloom, H. Clawson, J. Spieth, L. W. Hillier, S. Richards,
G. M. Weinstock, R. K. Wilson, R. A. Gibbs, W. J. Kent, W.
Miller,
and D. Haussler. 
Evolutionarily conserved elements in vertebrate,
insect, worm, and yeast genomes.  Genome Res (in press).
<P>
Chiaromonte F, Yap VB, and Miller W (2002).
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" 
TARGET=_blank>Scoring pairwise genomic sequence alignments</A>.
<em>Pac Symp Biocomput</em> 2002;:115-26.</P>
<P>
Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison R, Haussler D, and
Miller W. (2003).
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103" 
TARGET=_blank>Human-Mouse Alignments with BLASTZ</A>.
<em>Genome Res.</em> 13(1):103-7.</P>

