<html>
<H2>Description</H2>
<P>
This track shows updated versions of gene predictions submitted for the 
<A TARGET=_BLANK HREF="http://genome.imim.es/gencode/workshop2005.html">EGASP/2005</A>
ENCODE Gene Prediction Workshop.
The following gene predictions are included:
<UL>
<LI>
<A TARGET=_BLANK href="http://cbcb.umd.edu/software/jigsaw">JIGSAW</a>
</LI>
<LI>
<A TARGET=_BLANK href="http://augustus.gobics.de/"> AUGUSTUS</A>
</LI>
</UL>
The original EGASP submissions are displayed in the companion tracks, 
<EM>EGASP Full</EM> and <EM>EGASP Partial</EM>.
<H2>Methods</H2>
<P>
<H3>Jigsaw</H3>
JIGSAW is a gene prediction program which calls genes based on 
target genomic sequence and output from a gene structure annotation database.
Data available for download from UCSC's annotation database is 
used as input and includes the following tracks of evidence:
Known Genes, Ensembl, RefSeq, Geneid, Genscan, SGP, Twinscan, Human mRNAs,
TIGR Gene Index, UniGene, 'Most Conserved' Elements and Non-human RefSeq Genes.
<a href="http://cbcb.umd.edu/software/GlimmerHMM">GlimmerHMM</a> and <a href="http://genezilla.org"> Genezilla</a>, two open source <i>ab initio</i> gene finding programs
are also used.
<H3>Augustus</H3>
is a generalized Hidden Markov Model (GHMM) that
models coding and non-coding sequence, slice
sites, the branch point region, the tranlation start and end
and the lengths of exons and introns. This version has
been trained on a training set with 1284 human genes.
There are four sets of predictions in this track: ab initio,
predictions augmented using EST and protein evidence,
predictions augmented using mouse homology,
and predictions with both additional information sources used.

<H2>Credits</H2>
<P>
AUGUSTUS was written by Mario Stanke at the
Department of Bioinformatics of the University of Göttingen in Germany.
<P>

<H2>References</H2>
</html>
