<H2>Description</H2>
<P>
Genomic Evolutionary Rate Profiling 
<A HREF="http://mendel.stanford.edu/sidowlab/downloads/ABC_GERP/abcgerp.html" 
TARGET=_blank>(GERP)</A> constrained elements exhibit significant evidence of 
the effects of purifying selection.  
Elements are scored according to the inferred intensity of purifying selection 
and are measured as &quot;rejected substitutions&quot; (RSs). RSs capture the 
magnitude of difference between the number of &quot;observed&quot; substitutions
(estimated using maximum likelihood) and the number that would be 
&quot;expected&quot; under a neutral model 
of evolution.   The RS is displayed as part of the item name, and items with
higher RSs are displayed in a darker shade of blue.  The score shown on
the details page is generally not as accurate as the RS count that is part of
the item name.
<P>
Analyses are based on alignments of human ENCODE genomic DNA to many 
other mammals using the 
<A HREF="http://lagan.stanford.edu/lagan_web/index.shtml" 
TARGET=_blank>LAGAN</A> alignment package.</P>

<H2>Methods</H2>
<P>
The GERP method relies on maximal likelihood to estimate the
difference between observed and expected substitution rates.
The expected and observed rates are both calculated on a phylogenic 
tree using the same fixed topology.  The branch lengths of the 
expected tree are based on the average substitutions at neutral sites.  
The branch lengths of the observed tree, which is calculated separately for 
each human base, are based on the substitutions seen 
at the column of the multiple alignment at that base.  Species that 
have gaps at a particular column are not considered in the estimation for that
column.</P>
<P>
&quot;Constrained elements&quot; are identified as those groups
of consecutive human bases that have an observed rate of evolution that is
smaller than the expected rate.  These groups of columns are merged if they
are less than a few nucleotides apart and are scored according to the sum of 
the site-by-site difference between observed and expected rates (RS).</P>

<H2>Verification</H2>
<P>
Permutations of the actual alignments were analyzed, and &quot;constrained 
elements&quot identified in these permuted alignments were treated as 
&quot;false positives&quot;.  Subsequently, an RS threshold was picked such 
that the total length of &quot;false positive&quot;  constrained elements 
(identified in the permuted alignments) was less than 5% of the length of 
constrained elements identified in the actual alignment.  
Thus, all annotated constrained elements are significant at better
than 95% confidence, and the total fraction of the ENCODE regions 
annotated as constrained is 5-7%.</P>

<H2>Credits</H2>
<P>
GERP was developed primarily by Greg Cooper in the lab of 
<A HREF="http://mendel.stanford.edu/sidowlab/" TARGET=_blank>Arend Sidow</A> 
at <A HREF="http://www.stanford.edu" TARGET=_blank>Stanford University</A> 
(Depts of Pathology and Genetics), in close collaboration
with Eric Stone (Biostatistics, NC State), and George Asimenos and 
Eugene Davydov in the lab of 
<A HREF="http://ai.stanford.edu/~serafim/" TARGET=_blank>Serafim Batzoglou</A> 
(Dept. of Computer Science, Stanford).</P>

<H2>References</H2>
<P>
Cooper, G.M., Stone, E.A., Asimenos, G., NISC Comparative Sequencing Program, 
Green, E.D., Batzoglou, S. and Sidow, A.  
Distribution and intensity of constraint in mammalian genomic sequence.   
<em>Genome Res.</em>. In press (2005).</P>
