<html>
<h2>DESCRIPTION</h2>

Genome-wide location analysis of histone H3 monomethyl lysine 4 (H3K4me1)
was conducted employing ChIP/chip using chromatin extracted from GM06990 (lymphoblastoid) cells.
Histone methylation serves as a stable genomic imprint that regulates gene expression and other epigenetic phenomena.
Methylation of lysine four on histone H3 is found in transcriptionally active domains called euchromatin.


<p>
Cell line: <a TARGET=_BLANK HREF="http://ccr.coriell.org/nigms/nigms_cgi/display.cgi?GM06990">GM06990</a>
<br>
Antibody: <a TARGET=_BLANK HREF="http://www.abcam.com/index.html?datasheet=8580">ab8580</a>
<br>
Epitope: H3K4me1
<br>
DNA microarray: <a TARGET=_BLANK HREF="ftp://ftp.sanger.ac.uk/pub/encode/microarrays/ENCODE3.1.1.hg16.adf">ENCODE3.1.1</a>

<h2>METHODS</h2>

Chromatin from the cell line was cross-linked with 1% formaldehyde, precipitated with 
antibody binding to histone H3 monomethyl lysine 4, sheared and hybridized to a DNA array.  
DNA was not amplified prior to hybridization.
                                   
The raw and transformed data files reflect fold enrichment over background, averaged over
six replicates.

<h2>VERIFICATION</h2>
                      
There are six replicates total: two technical replicates (immunoprecipitations) for each of the 
three biological replicates (cell cultures).

Raw and transformed (averaged) data can be downloaded from the Wellcome Trust Sanger Institute 
<a TARGET=_BLANK HREF="ftp://ftp.sanger.ac.uk/pub/encode/H3K4me1_GM06990_1">ftp</a> site.

<h2>CREDITS</h2>

The data for this track were generated by the 
<a TARGET=_BLANK HREF="http://www.sanger.ac.uk/PostGenomics/encode/investigators.shtml">ENCODE investigators</a> at the 

<a TARGET=_BLANK HREF="http://www.sanger.ac.uk/">Wellcome Trust Sanger Institute</a>, Hinxton, UK.
</html>
