<H2>Description</H2>
<P>
Genomic Evolutionary Rate Profiling 
(<A HREf="http://mendel.stanford.edu/sidowlab/downloads/ABC_GERP/abcgerp.html" 
TARGET=_blank>GERP</A>) 
scores describe nucleotide substitution rates observed in alignments 
containing all human bases in the ENCODE portion of the genome.  
Scores are estimated on a column-by-column basis using multiple sequence
alignments of mammalian genomic DNA generated by the 
<A HREF="http://lagan.stanford.edu/lagan_web/index.shtml" 
TARGET=_blank>LAGAN</A> alignment package.  </P>
<P>
The scores range from 0 to 3, with higher scores corresponding to human
bases in alignment columns with higher degrees of similarity, <em>i.e.</em>
bases that have evolved slowly, some of which have been under purifying 
selection. The opposite holds true for swiftly evolving (low similarity) 
columns. </P>

<H2>Display Conventions and Configuration</H2>
<P>
This annotation follows the display conventions for composite 
tracks. The subtracks within this annotation may be configured in a variety of 
ways to highlight different aspects of the displayed data. The graphical 
configuration options are shown at the top of the track description page, 
followed by a list of subtracks. To display only selected subtracks, uncheck 
the boxes next to the tracks you wish to hide. For more information about the 
graphical configuration options, click the 
<A HREF="../goldenPath/help/hgWiggleTrackHelp.html">Graph configuration 
help</A> link.</P>

<H2>Methods</H2>
<P>
The GERP score is the expected substitution rate divided by
the observed substitution rate at a particular human base.
Scores greater than 3 are clipped to 3.  The expected and observed rates are 
both calculated on a phylogenic tree using the same fixed topology.  
The branch lengths of the expected tree are based on the average 
substitutions at neutral sites.  The branch lengths of the observed
tree, which is calculated separately for each human base, are based on the 
substitutions seen at the column of the 
multiple alignment at that base.  Species that have gaps at
a particular column are not considered in the scoring for that
column.</P>

<H2>Verification</H2>
<P>
Scores are deterministic, given a maximum-likelihood model of 
nucleotide substitution, species topology, neutral tree, and alignment.</P>

<H2>Credits</H2>
<P>
GERP was developed primarily by Greg Cooper in the lab of 
<A HREF="http://mendel.stanford.edu/sidowlab/" TARGET=_blank>Arend Sidow</A> 
at <A HREF="http://www.stanford.edu" TARGET=_blank>Stanford University</A> 
(Depts of Pathology and Genetics), in close collaboration with 
Eric Stone (Biostatistics, NC State), and George Asimenos and 
Eugene Davydov in the lab of 
<A HREF="http://ai.stanford.edu/~serafim/" TARGET=_blank>Serafim Batzoglou</A> 
(Dept. of Computer Science, Stanford).</P>

<H2>References</H2>
<P>
Cooper, G.M., Stone, E.A., Asimenos, G., NISC Comparative Sequencing Program, 
Green, E.D., Batzoglou, S. and Sidow, A.  
Distribution and intensity of constraint in mammalian genomic sequence.   
<em>Genome Res.</em>. In press (2005).</P>
