<H2>Description</H2>
<P>
This track shows an estimate of RNA abundance (transcription) for
all ENCODE regions.  Retinoic acid-stimulated HL-60 cells were
harvested after 0, 2, 8, and 32 hours.  Purified polyA+ RNA from
these samples was hybridized to Affymetrix ENCODE oligonucleotide
tiling arrays.  These arrays have 25-mer probes tiled every 21 bp on
average in the non-repetitive ENCODE regions.
</P>
<H2>Display Conventions and Configuration</H2>
<P>
The subtracks within this annotation 
may be configured in a variety of ways to highlight different aspects of the 
displayed data. The graphical configuration options for the signal subtracks 
are shown at the top of the track description page, followed by a list of 
subtracks.  To display only selected subtracks, uncheck the boxes next to 
the tracks you wish to hide. 
Subtracks for the sites are hidden by default but can be enabled.  
Color differences in the subtracks are arbitrary. They provide a
visual cue for distinguishing between the different timepoints.
For more information about the graphical configuration options, click the 
<A HREF="../goldenPath/help/hgWiggleTrackHelp.html" TARGET=_blank>Graph
configuration help</A> link.</P>

<H2>Methods</H2>
<P>
The data from replicate arrays were quantile-normalized (Bolstad
<em>et al</em>., 2003) and all arrays were scaled to a median array intensity
of 22.  Within a sliding 101 bp window centered on each probe, 
an estimate of RNA abundance (signal) was found by calculating the median 
of all pairwise average PM-MM (where PM is perfect match and MM is 
mismatch) values. 
(Both Kaproanov <em>et al</em>., 2002 and Cawley 
<em>et al</em>., 2004 are good references for experimental methods; 
Cawley <em>et al</em>., 2004 also describes analytical methods.)</P>

<H2>Verification</H2>
<P>
Three independent biological replicates were generated and
hybridized to duplicate arrays (two technical replicates).
Transcribed regions were generated from this signal track based on a
5% false positive rate cutoff in negative bacterial controls.  These
sites were then ranked individually in each of the replicates,
based on their signal as computed from the respective replicates.
All pairwise differences in these ranks were computed among
biological replicates, and replicates were accepted when the sum of
rank differences was relatively low (there are currently no hard
cutoff criteria).</P>

<H2>Credits</H2>
<P>
These data were generated and analyzed by the Gingeras/Struhl
collaboration with the Tom Gingeras group at 
<A HREF="http://www.affymetrix.com/" TARGET=_BLANK>Affymetrix</A> and the 
<A HREF="http://www.hms.harvard.edu/dms/bbs/fac/struhl.html"
TARGET=_blank>Kevin Struhl</A> group at Harvard Medical School.</P>

<H2>References</H2>
<P>
Please see the 
<A HREF="http://www.affymetrix.com/transcriptome/index.affx" 
TARGET=_blank>Affymetrix Transcriptome</A> site for a project overview and
additional references to Affymetrix tiling array publications.</P>
<P>
Bolstad, B. M., Irizarry, R. A., Astrand, M., and Speed, T. P. 
<A HREF="http://bioinformatics.oupjournals.org/cgi/content/abstract/19/2/185" 
TARGET=_blank>A comparison of normalization methods for high density 
oligonucleotide array data based on variance and bias</A>. 
<em>Bioinformatics</em> <B>19</B>(2), 185-193 (2003).</P>
<P>
Cawley, S., Bekiranov, S., Ng, H. H., Kapranov, P., Sekinger,
E. A., Kampa, D., Piccolboni, A., Sementchenko, V., Cheng, J.,
Williams, A. J., <em>et al</em>. 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=14980218" 
TARGET=_blank>Unbiased mapping of transcription factor binding sites along 
human chromosomes 21 and 22 points to widespread regulation of noncoding 
RNAs</A>. 
<em>Cell</em> <B>116</B>(4), 499-509 (2004).</P>
<P>
Kapranov, P., Cawley, S. E., Drenkow, J., Bekiranov, S., Strausberg,
R.  L., Fodor, S. P., and Gingeras, T. R. 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=11988577" 
TARGET=_blank>Large-scale transcriptional activity in chromosomes 21 and 
22</A>. 
<em>Science</em> <B>296</B>(5569), 916-919 (2002).</P>
