<!-- Use this simple HTML template to organize the
     description text that accompanies your data.-->

<H2>Description</H2>
<P>
This track shows the locations of Transcriptionally Active Regions 
(TARs) / Transcribed Fragments (transfrags) for Human Neutrophil (PMN) 
Total RNA hybridized to the Affymetrix ENCODE oligonucleotide 
microarray. This array has 25 mer oligonucleotide probes tiled 
approximately every 21 bp covering all the non-repetitive DNA sequence 
of the ENCODE regions. The transcript map is a combined signal for both 
strands of DNA. This is derived from 10 different biological samples 
(each sample was extracted from a different individual), each with at 
least 2 technical replicates. See NCBI GEO <a 
href="http://www.ncbi.nlm.nih.gov/projects/geo/query/acc.cgi?acc=GSE2678
">GSE2678</a> for details of experimental protocols.

<H2>Methods</H2>
<P>
The data from technical replicates were median scaled and quantile 
normalized to each other. Using a 101 bp sliding window centered on 
each oligonucleotide probe a signal map (estimating RNA abundance) is 
generated by computing the pseudomedian signal of all PM-MM pairs 
(median of pairwise PM-MM averages) within the window (including 
replicates). Biological replicate signal maps are combined by quantile 
normalizing them between replicates and computing the median signal at 
each oligonucleotide probe location. Ten independant biological 
replicates were generated and each was hybridized to at least two 
different arrays (technical replicates). Transcribed regions 
(TARs/transfrags) are then identified using a signal theshold of 90 
percentile of signal intensities as well as a maximum gap of 50 bp and 
a minimum run of 50 bp (between oligonucleotide positions).

<H2>Verification</H2>
<P>
Transcribed regions (TARs/transfrags) as determined by individual 
biological samples are compared to ensure significant overlap.

<H2>Credits</H2>
<P>
This data was generated and analyzed by the Yale/Affymetrix 
collaboration between the labs of Michael Snyder, Mark Gerstein and 
Sherman Weissman at Yale and Tom Gingeras at Affymetrix.

<H2>References</H2>
<P>

Kapranov P, Cawley SE, Drenkow J, Bekiranov S, Strausberg RL, Fodor SP, 
Gingeras TR,
<a href 
="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&d
opt=Abstract&list_uids=11988577&query_hl=5">
Large-scale transcriptional activity in chromosomes 21 and 22
</a>, 
<i>Science</i>. 2002 May 3;296(5569):916-9.

<BR><BR>
Rinn JL, Euskirchen G, Bertone P, Martone R, Luscombe NM, Hartman S, 
Harrison PM, Nelson FK, Miller P, Gerstein M, Weissman S, Snyder M, 
<a 
href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubm
ed&dopt=Abstract&list_uids=12600945&query_hl=1">
The transcriptional activity of human Chromosome 22
</a>, 
<i>Genes Dev</i>, 2003 Feb 15;17(4):529-40.

<BR><BR>
Bertone P, Stolc V, Royce TE, Rozowsky JS, Urban AE, Zhu X, Rinn JL, 
Tongprasit W, Samanta M, Weissman S, Gerstein M, Snyder M,
<a href = 
"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&do
pt=Abstract&list_uids=15539566&query_hl=7">
Global identification of human transcribed sequences with genome tiling 
arrays
</a>, 
<i>Science</i>. 2004 Dec 24;306(5705):2242-6. Epub 2004 Nov 11.

<BR><BR>
Cheng J, Kapranov P, Drenkow J, Dike S, Brubaker S, Patel S, Long J, 
Stern D, Tammana H, Helt G, Sementchenko V, Piccolboni A, Bekiranov S, 
Bailey DK, Ganesh M, Ghosh S, Bell I, Gerhard DS, Gingeras TR,
<a href = 
"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&do
pt=Abstract&list_uids=15790807&query_hl=5">
Transcriptional maps of 10 human chromosomes at 5-nucleotide resolution
</a>, 
<i>Science</i>. 2005 May 20;308(5725):1149-54. Epub 2005 Mar 24.

