<H2>Description</H2>
<P>
Genomic Evolutionary Rate Profiling 
<a href=http://mendel.stanford.edu/sidowlab/downloads/ABC_GERP/abcgerp.html>(GERP)</a> 
constrained element annotations describe regions of the human genome
(throughout the ENCODE target regions) that exhibit significant evidence of
the effects of purifying selection.  
Elements are scored according to the
inferred intensity of purifying selection, and are measured as 
"rejected substitutions" (RS).  RS capture the magnitude of difference
between the number of "observed" substitutions (estimated using maximum
likelihood) and the number that would be "expected" under a neutral model 
of evolution.  
<P>
Analyses are based on alignments of human ENCODE genomic DNA to many 
other mammals using the 
<a href=http://lagan.stanford.edu/lagan_web/index.shtml>LAGAN</a> 
alignment package.
<H2>Methods</H2>
<P>
Alignments, generated target by target for all ENCODE regions, are 
first compressed so that the human sequence is ungapped.  
Maximum likelihood rate estimation is used to estimate an "observed" rate
of evolution for each column, given a fixed species topology relating
the sequences, where gapped species are eliminated from consideration.  
A neutral tree that describes the average rates of evolution (in terms 
of nucleotide substitutions affecting neutral sites) over all the branches
of the species topology relating the analyzed sequence is used to generate 
an estimate of the "expected" rate of evolution for each column, where gapped
species are eliminated.  
<P>
"Constrained elements" are identified as those groups
of consecutive human bases that have an observed rate of evolution that is
smaller than the expected rate.  These groups of columns are merged if they
are less than a few nucleotides apart, and scored according to the sum of 
the site-by-site difference between observed and expected rates (RS).

<H2>Verification</H2>
<P>
We analyze permutations of the actual alignments, and treat 
"constrained elements" identified in these permuted alignments as 
"false positives".  We subsequently pick an RS threshold such that the 
total length of "false positive" (that is, identified in the permuted
alignments) constrained elements is less than 5% of the length of constrained
elements identified in the actual alignment.  
Thus, all annotated constrained elements are significant at better
than 95% confidence, and the total fraction of the ENCODE regions 
annotated as constrained is 5-7%.

<H2>Credits</H2>
<P>
GERP was primarily developed by Greg Cooper in the lab of 
<a href=http://mendel.stanford.edu/sidowlab/>Arend Sidow</a> 
at <a href=http://www.stanford.edu>Stanford University</a> 
(Depts of Pathology and Genetics), in close collaboration
with Eric Stone (Biostatistics, NC State), and George Asimenos and 
Eugene Davydov in the lab of 
<a href=http://ai.stanford.edu/~serafim/>Serafim Batzoglou</a> (Dept of Computer Science, Stanford).

<H2>References</H2>
<P>
Cooper, G.M., Stone, E.A., Asimenos, G., NISC Comparative Sequencing Program, Green, E.D., Batzoglou, S., and Sidow, A.  2005.  
Distribution and intensity of constraint in mammalian genomic sequence.  <a href=http://www.genome.org>Genome Research</a>. In press.
