<H2>Description</H2>
<P>
This track displays
a set of human-centric multiple alignments of the sequencies included
in the <A
HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg16/encode/alignments/MAY-2005/">
May 2005 ENCODE freeze</A>, generated by MLAGAN.

<H2>Methods</H2>
<P>
The alignments were created in the following way: First, the sequence of
each other species is &quot;rearranged&quot; so that it is orthologously collinear with
respect to the human sequence. This is done with a modified version of
SuperMap, based on Shuffle-LAGAN. Then, a multiple global alignment is created
for every region using MLAGAN. Finally, the alignments are refined using
MUSCLE, which realigns some parts in a limited-size window, in an
iterative fashion.

<H2>Credits</H2>
<P>
George Asimenos from Stanford's ENCODE group has generated the alignments.<BR>
Mike Brudno is the author of Shuffle-LAGAN, SuperMap and MLAGAN.<BR>
Bob Edgar is the author of MUSCLE.

<H2>References</H2>
<P>
Michael Brudno, Chuong Do, Gregory Cooper, Michael F. Kim, Eugene Davydov,
Eric D. Green, Arend Sidow and Serafim Batzoglou: &quot;LAGAN and Multi-LAGAN:
efficient tools for large-scale multiple alignment of genomic DNA&quot;,
Genome Research 2003 Apr;13(4):721-31.<BR><BR>
Michael Brudno, Sanket Malde, Alexander Poliakov, Chuong Do, Olivier Courone,
Inna Dubchak, and Serafim Batzoglou: &quot;Glocal alignment: finding
rearrangements during alignment&quot;. Special Issue on the Proceedings of the ISMB
2003, Bioinformatics 19: 54i-62i, 2003.<BR><BR>
Edgar, Robert C. (2004): &quot;MUSCLE: multiple sequence alignment with high
accuracy and high throughput&quot;, Nucleic Acids Research 32(5), 1792-97.
