<H2>Description</H2>
<P>
This track shows deletion/insertion polymorphisms (DIPs) at their genomic 
locations with the variant alleles displayed as detected from the identified 
trace names. The quality score reflects the minimum PHRED quality value over 
the entire range of the DIP within the trace.</P>
<P>
The location of the DIP is its 3' position within the trace. The alleles are 
reported relative to the &quot;+&quot; strand of the reference sequence; 
however, the trace may actually align to the &quot;-&quot; strand. 
When viewing the chromatogram using the URL provided, 
if the trace aligned to the &quot;-&quot; strand, the DIP bases in the trace 
will be the reverse compliment of the variant allele given.</P>

<H2>Methods</H2>
<P>
All human trace data from NCBI's trace archive were aligned to hg17 with 
ssahaSNP, followed by ssahaDIP post-processing to detect deletion/insertion 
polymorphisms. DIPs within ENCODE regions were extracted, and 40 of the 44 
regions were translated to hg16 coordinates.</P>

<H2>Verification</H2>
<P>
For verification, 500k traces from the mouse whole genome shotgun (WGS) 
<A HREF="ftp://ftp.ncbi.nih.gov/pub/TraceDB/mus_musculus/fasta.mus_musculus.045.gz"
TARGET=_blank>sequencing effort</A> were compared to mm6 using ssahaSNP and 
ssahaDIP. Because mm6 and these traces are from the same mouse strain, 
C57BL/6J, the DIP rate should be very low. Applying a quality threshold of 
Q23, the detected DIP rate was one DIP per 140k Neighborhood Quality Standard 
(NQS) bases.  This level was ten-fold lower than the SNP rate for the same 
data set using ssahaSNP, which has been validated as having a 5% FP rate. 
The detected DIP rate for human traces against hg17 is one DIP per 12k NQS 
bases, indicating a FP rate of 12k/140k, or about 8%.</P>
<P>
Further validation experiments are in progress.</P>

<H2>Credits</H2>
<P>
All analyses were performed by Jim Mullikin using ssahaSNP and ssahaDIP. 
The Trace data were contributed to the trace archive by many sequencing 
centers.</P>

<H2>References</H2>
<P>
Ning, Z., Cox, A.J. and Mullikin, J.C. 
<A HREF="http://www.genome.org/cgi/content/short/11/10/1725"
TARGET=_blank>SSAHA: A fast search method for large DNA databases</A>.
<em>Genome Res.</em> <B>11</B>(10), 1725-9 (2001). </P> 
<P>
The International SNP Map Working Group. 
<A HREF="http://www.nature.com/nature/journal/v409/n6822/abs/409928a0_fs.html"
TARGET=_blank>A map of human genome sequence variation containing 
1.4 million single nucleotide polymorphisms</A>.
<em>Nature</em> <B>409</B>(6822), 928-33 (2001).</P>
