<h2>DESCRIPTION</h2>

Genome-wide location analysis of H3 and H4 histones
was conducted employing ChIP/chip using chromatin extracted from GM06990
(lymphoblastoid) cells.  Histone methylation serves as a stable genomic
imprint that regulates gene expression and other epigenetic phenomena.
These histones are found in transcriptionally active domains called euchromatin.

<P>
<TABLE BORDER="0" BGCOLOR="888888" CELLSPACING="0" CELLPADDING="1"><tr><td>
<TABLE BORDER=1 BGCOLOR="FFFEE8">
<TR><TH>Track</TH><TH>Cell<BR>Line</TH><TH>Antibody</TH><TH>Epitope</TH>
    <TH>Data&nbsp;FTP<BR>access</TH><TH>Histone</TH></TR>
<TR><TD>1</TD><TD>
    <a TARGET=_BLANK
	HREF="http://ccr.coriell.org/nigms/nigms_cgi/display.cgi?GM06990">
	GM06990</a></TD><TD>
    <a TARGET=_BLANK
	HREF="http://www.abcam.com/index.html?datasheet=8580">
	ab8580</a></TD><TD>
    H3K4me1</TD><TD>
    <a TARGET=_BLANK
	HREF="ftp://ftp.sanger.ac.uk/pub/encode/H3K4me1_GM06990_1">
	H3K4me1_GM06990_1</a></TD><TD>
    H3 monomethyl lysine 4</TD>
    </TR>
<TR><TD>2</TD><TD>
    <a TARGET=_BLANK
	HREF="http://ccr.coriell.org/nigms/nigms_cgi/display.cgi?GM06990">
	GM06990</a></TD><TD>
    <a TARGET=_BLANK
	HREF="http://www.abcam.com/index.html?datasheet=7766">
	ab7766</a></TD><TD>
    H3K4me2</TD><TD>
    <a TARGET=_BLANK
	HREF="ftp://ftp.sanger.ac.uk/pub/encode/H3K4me2_GM06990_1">
	H3K4me2_GM06990_1</a></TD><TD>
    H3 dimethyl lysine 4</TD>
    </TR>
<TR><TD>3</TD><TD>
    <a TARGET=_BLANK
	HREF="http://ccr.coriell.org/nigms/nigms_cgi/display.cgi?GM06990">
	GM06990</a></TD><TD>
    <a TARGET=_BLANK
	HREF="http://www.abcam.com/index.html?datasheet=8580">
	ab8580</a></TD><TD>
    H3K4me3</TD><TD>
    <a TARGET=_BLANK
	HREF="ftp://ftp.sanger.ac.uk/pub/encode/H3K4me3_GM06990_2">
	H3K4me3_GM06990_2</a></TD><TD>
    H3 trimethyl lysine 4</TD>
    </TR>
<TR><TD>4</TD><TD>
    <a TARGET=_BLANK
	HREF="http://ccr.coriell.org/nigms/nigms_cgi/display.cgi?GM06990">
	GM06990</a></TD><TD>
    <a TARGET=_BLANK
	HREF="http://www.upstate.com/browse/productdetail.q.productid.e.06-599">
	06-599</a></TD><TD>
    H3ac</TD><TD>
    <a TARGET=_BLANK
	HREF="ftp://ftp.sanger.ac.uk/pub/encode/H3ac_GM06990_1">
	H3ac_GM06990_1</a></TD><TD>
    H3 acetylated lysines 9 and 14</TD>
    </TR>
<TR><TD>5</TD><TD>
    <a TARGET=_BLANK
	HREF="http://ccr.coriell.org/nigms/nigms_cgi/display.cgi?GM06990">
	GM06990</a></TD><TD>
    <a TARGET=_BLANK
	HREF="http://www.upstate.com/browse/productdetail.q.productid.e.06-866">
	06-866</a></TD><TD>
    H4ac</TD><TD>
    <a TARGET=_BLANK
	HREF="ftp://ftp.sanger.ac.uk/pub/encode/H4ac_GM06990_1">
	H4ac_GM06990_1</a></TD><TD>
    H4 acetylated lysines 5, 8, 12, 16</TD>
    </TR>
</TABLE>
</TD></TR></TABLE>
</P>

<P>
DNA microarray: <a TARGET=_BLANK
    HREF="ftp://ftp.sanger.ac.uk/pub/encode/microarrays/ENCODE3.1.1.hg16.adf">
	ENCODE3.1.1</a>
</P>

<h2>METHODS</h2>

Chromatin from the cell line was cross-linked with 1% formaldehyde,
precipitated with antibody binding to the histone, sheared and hybridized to
a DNA array.  DNA was not amplified prior to hybridization.

<P>
The raw and transformed data files reflect fold enrichment over background,
averaged over six replicates.

<h2>VERIFICATION</h2>
                      
There are six replicates total: two technical replicates (immunoprecipitations)
for each of the three biological replicates (cell cultures).
 
<P>
Raw and transformed (averaged) data can be downloaded from the Wellcome
Trust Sanger Institute FTP site as indicated in the table above.

<h2>CREDITS</h2>

The data for this track were generated by the 
<a TARGET=_BLANK
HREF="http://www.sanger.ac.uk/PostGenomics/encode/investigators.shtml">
	ENCODE investigators</a> at the 

<a TARGET=_BLANK
	HREF="http://www.sanger.ac.uk/">Wellcome Trust Sanger Institute</a>,
	Hinxton, UK.
