<H2>Description</H2>
This track displays activity levels of predicted promoters in the ENCODE regions, 
based on luciferase reporter assays.  Activity level is indicated by color, ranging 
from black (no activity) to red.  Experiments were performed on 14 cell lines:
<P>
<!--
<TABLE BORDER=2 CELLSPACING=2 CELLPADDING=2>
-->
<BLOCKQUOTE>
  <TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4>
  <TR><TD>Snu-182</TD><TD>hepatocellular carcinoma</TD><TD>liver</TR>
  <TR><TD>AGS</TD><TD>gastric adenocarcinoma </TD><TD>stomach</TR>
  <TR><TD>U87 MG</TD><TD>glioblastoma-astrocytoma </TD><TD>brain</TR>
  <TR><TD>CRL1690</TD><TD>glioblastoma</TD><TD>brain</TR>
  <TR><TD>G-402</TD><TD>renal leiomyoblastoma</TD><TD>kidney</TR>
  <TR><TD>HCT116</TD><TD>colorectal carcinoma</TD><TD>colon</TR>
  <TR><TD>HMCB </TD><TD>melanoma </TD><TD>skin</TR>
  <TR><TD>HT1080</TD><TD>fibrosarcoma</TD><TD>connective tissue</TR>
  <TR><TD>Hela </TD><TD>adenocarcinoma </TD><TD>cervix</TR>
  <TR><TD>HepG2</TD><TD>  hepatocellular carcinoma </TD><TD>liver</TR>
  <TR><TD>JEG-3</TD><TD> choriocarcinoma</TD><TD>placenta</TR>
  <TR><TD>MG-63</TD><TD> osteosarcoma </TD><TD>bone</TR>
  <TR><TD>MRC-5</TD><TD> fibroblast </TD><TD>lung</TR>
  <TR><TD>PANC-1</TD><TD>epitheliod carcinoma </TD><TD>pancreas</TR>
  </TABLE>
</BLOCKQUOTE>
  
<H2>Methods</H2>
<P>
Promoters in the ENCODE region were predicted using a variation on methods 
previously described (Trinklein et al., 2003, Trinklein et al., 2004). Any human
cDNAs in the database with at least one base pair of exon overlap were merged.
This generated the gene models. The transcription start sites were predicted
by starting with the 5' end of each gene model as well as other 5' ends 
that were at least 500 base pairs away from the 5' end. Promoters were 
identified as the approximately 600 base pairs upstream and 100 base pairs 
downstream of each transcription start site. 
<P>
Primer3 was used to pick primers yielding approximately 500 base pair fragments 
containing the transcription start site.  Each fragment of DNA represented in 
this track was cloned into a luciferase reporter vector (pGL3-Basic, Promega) 
using the BD Clontech Infusion Cloning System. The Dual Luciferase 
system (Promega) was used to co-transfect the experimental DNA along with a control plasmid 
expressing Renilla in 96-well format into one of the fourteen cell types using 
FuGENE Transfection Reagent (Roche).  Each transfection was done in duplicate. 
<P>
Values here represent averages of the Luciferase-to-Renilla ratio. To determine 
fold increase, 96 random genomic fragments were cloned and tested, using the
average values to determine a threshold (3 standard deviations) for functional 
promoters. 
<P>
Any promoters that did not meet the threshold for function have a value of zero; 
those that are positive have been scaled to show graded color on the browser.

<H2>Verification</H2>
<P>
Data were verified by repeating the preparation and measurement of 48 random 
fragments. No significant variation between the two preparations
was detected.

<H2>Credits</H2>
<P>
This work was done in collaboration at the 
<A TARGET=_BLANK HREF="http://www-shgc.stanford.edu/myerslab/" TARGET=_BLANK> Myers Lab</A> at 
Stanford. The following people contributed: 
Sara J. Hartman,
Nathan D. Trinklein,
Elizabeth D. Anton,
Loan Nguyen,
Simone S. Marticke, and
Richard M. Myers.

<H2>References</H2>
<P>
Trinklein, N.D., Aldred, S.J., Saldanha, A.J., and Myers, R.M. 
<A HREF="http://www.genome.org/cgi/content/short/13/2/308" 
TARGET=_BLANK>Identification and functional analysis of human transcriptional 
promoters</A>. <em>Genome Res.</em> 2003 Feb;<B>13</B>(2):308-12 (2003).
<P>
Trinklein, N.D., Aldred, S.F., Hartman, S.J., Schroeder, D.I., Otillar, R.P., 
and Myers, R.M. 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=14707170&dopt=Abstract" 
TARGET=_BLANK>An abundance of bidirectional promoters in the human 
genome</A>. <em>Genome Res.</em> 2004 Jan;<B>14</B>(1):62-6 (2004).
