<H2>Description</H2>
<P>
Genome-wide location analysis of histone H3 trimethyl lysine 4 (H3K4me3)
was conducted employing ChIP/chip using chromatin extracted from GM06990 
(lymphoblastoid) cells.  Histone methylation serves as a stable genomic 
imprint that regulates gene expression and other epigenetic phenomena.
Methylation of lysine four on histone H3 is found in transcriptionally 
active domains called euchromatin.</P>
<P>
Cell line: <A HREF="http://ccr.coriell.org/nigms/nigms_cgi/display.cgi?GM06990" 
TARGET=_blank>GM06990K</A>
<BR>
Antibody: <A HREF="http://www.abcam.com/index.html?datasheet=8580"
TARGET=_blank>ab8580</A>
<BR>
Epitope: H3K4me3
<BR>
DNA microarray: <A HREF="ftp://ftp.sanger.ac.uk/pub/encode/microarrays/ENCODE2.1.1.adf" 
TARGET=_blank>ENCODE2.1.1</A>

<H2>Methods</H2>
<P>
Chromatin from the cell line was cross-linked with 1% formaldehyde, 
precipitated with antibody binding to histone H3 trimethyl lysine 4, sheared 
and hybridized to a DNA array. DNA was not amplified prior to hybridization.
<P>                                
The raw and transformed data files reflect fold enrichment over background, 
averaged over six replicates.</P>

<H2>Verification</H2>
<P>                   
There are six replicates: two technical replicates (chromatin preparations) 
for each of the three biological replicates (cell cultures).</P>
<P>
Raw and transformed (averaged) data can be downloaded from the Wellcome Trust 
Sanger Institute 
<A HREF="ftp://ftp.sanger.ac.uk/pub/encode/H3K4me3_GM06990_1" 
TARGET=_blank>ftp</A> site.

<H2>Credits</H2>
<P>
The data for this track were generated by the 
<A HREF="http://www.sanger.ac.uk/PostGenomics/encode/investigators.shtml" 
TARGET=_blank>ENCODE investigators</A> at the 
<A HREF="http://www.sanger.ac.uk/" TARGET=_blank>Wellcome Trust Sanger 
Institute</A>, Hinxton, UK.</P>
