<H2>Description</H2>
<P>
This track displays associations between gene expression
data from the 60 unrelated CEPH individuals of the 
<A HREF="http://www.hapmap.org/" TARGET=_BLANK>International 
HapMap Project</A> with SNPs
genotyped by HapMap.  The expression data were generated with the
Illumina platfrom at the 
<A HREF="http://www.sanger.ac.uk/" TARGET=_BLANK>Wellcome Trust 
Sanger Institute</A>.  

<H2>Display Conventions and Configuration</H2>
<P>
In the graphical display, an association is displayed as a block 
drawn at the location of the associated SNP.  In pack or full modes, 
the name of the associated gene is drawn to the left of the block.  
The shading of the block indicates the strength of the association:
light gray indicates a (-log10) P-value close to 0, and black indicates a 
P-value of 2 or more.  
</P>

<H2>Methods</H2>
<P>
An association analysis was performed for each ENCODE RefSeq gene with the 
genotypes of SNPs in the same ENCODE region (cis).  Expression values were
initially log2 transformed and subsequently normalized with quantile
normalization to ensure homogeneous levels between arrays. 
Analysis Of Variance (ANOVA) was then performed with 1 or 2 degress of 
freedom (depending on whether only 2 or all 3 genotypes in the population 
were available), using 
the genotype as a categorical variable and the normalized/transformed
expression values as the response.  The values presented here are the
-log10 P-value.


<H2>Verification</H2>
<P>
There were 6 technical replicates for each sample and the average values
from the 6 replicates were used for the ANOVA. 


<H2>Credits</H2>
<P>
The following people contributed to this analysis:
Barbara Stranger (Wellcome Trust Sanger Institute),
Matthew Forrest (Wellcome Trust Sanger Institute),
Simon Tavare (Cambridge University),
Panos Deloukas (Wellcome Trust Sanger Institute), and
Manolis Dermitzakis (Wellcome Trust Sanger Institute).

