<H2> Description </H2>
<P>
This track shows the location of known and predicted non-protein coding RNA genes and
pseudogenes that fall within the encode regions.  It contains all information in Sean
Eddy's RNA gene track for these regions combined with computational predictions generated 
by Jakob Skou Pedersen's EvoFold algorithm.  In addition to the fields contained in the RNA gene
track, this track also includes encode-related fields describing overlap with transcribed 
regions and repeats.</P>


<B>Feature types include:</B>
<table>
<TR><TD>  tRNA:         </TD><TD>Transfer RNA (or pseudogene) </TD></TR>
<TR><TD>  rRNA:         </TD><TD>Ribosomal RNA (or pseudogene) </TD></TR>
<TR><TD>  scRNA:        </TD><TD>Small cytoplasmic RNA (or pseudogene) </TD></TR>
<TR><TD>  snRNA:        </TD><TD>Small nuclear RNA (or pseudogene) </TD></TR>
<TR><TD>  snoRNA:       </TD><TD>Small nucleolar RNA (or pseudogene) </TD></TR>
<TR><TD>  miRNA:        </TD><TD>MicroRNA (or pseudogene)     </TD></TR>
<TR><TD>  misc_RNA:     </TD><TD>Miscellaneous other RNA, such as Xist (or pseudogene) </TD></TR>
</table>

<P>
<B>Color legend:</B>
<table>
<TR> <TD> <FONT COLOR="000000"><B>BLACK</B>:       </FONT> </TD> <TD> Region is Repeatmasked.</TD> </TR>
<TR> <TD> <FONT COLOR="79AA3D"><B>GREEN</B>:       </FONT> </TD> <TD> Region is transcribed.</TD> </TR>
<TR> <TD> <FONT COLOR="FF0000"><B>RED</B>:         </FONT> </TD> <TD> Region is from the RNA Gene track, and is not transcribed.</TD> </TR>
<TR> <TD> <FONT COLOR="0000FF"><B>BLUE</B>:        </FONT> </TD> <TD> Region is an EvoFold prediction, and is not transcribed.</TD> </TR>
</table>


<H2> Methods </H2>
<P>
The RNA Gene track was supplemented with Jakob Skou Pedersen's EvoFold predictions and filtered down
to entries that lie within the Encode regions.  Regions that are at least 10 percent 
repeatmasked are flagged because no transcriptional data is available for them.  A region is
considered transcribed if at least 10 percent overlaps with any Affymetrix transfrag (derived 
from 6 microarray experiments) or Yale TAR (derived from 15 microarray experiments.)  Each
array is listed from which the overlapped transfrags and TARs were derived.
<P>
EvoFold is a comparative method which exploits the evolutionary signal
of genomic multiple-sequence alignments for identifying conserved
functional RNA structures. The method makes use of phylogenetic
Stochastic Context-free Grammars (phylo-SCFGs), which are combined
probabilistic models of RNA secondary structure and primary sequence
evolution. The predictions consist both of a specific RNA secondary
structure and an overall score. The overall score is essentially a
log-odd score phylo-SCFG modeling the constrained evolution of
stem-pairing regions and one which only models unpaired regions.</P>
<P>
Two sets of EvoFold predictions are included on this track.  The first,
labelled EvoFold, contains predictions based on the conserved elements of an 
8-way vertebrate alignment of the human, chimpanzee, mouse, rat, dog, chicken, zebrafish, and Fugu assemblies.
The second set of predictions, TBA23_EvoFold, were based on the conserved elements of the 23-way 
TBA alignments present in the ENCODE regions. When a pair of these predictions overlap, 
only the EvoFold prediction is shown.</P>

<H2>EvoFold References</H2>
<P> Pedersen, J.S., Bejerano, G., and Haussler, D. Identification and
    Classification of Conserved RNA Secondary Structures in the Human
    Genome. <em>in preparation </em>.

<P>Knudsen, B. and J.J. Hein.  <A
HREF="http://nar.oxfordjournals.org/cgi/content/full/31/13/3423"
TARGET=_blank> Using stochastic context free grammars and molecular
evolution to predict RNA secondary
structure</A>. <em>Bioinformatics</em> <B>15</B>, 446-54 (1999).</P>
</P>

<H2>Credits</H2>
<P>
These data were kindly provided by Sean Eddy at Washington University,
Jakob Skou Pedersen at UC Santa Cruz, and The Encode
Consortium.<BR>This track was created by Matt Weirauch.</P>
