<H2>Description</H2>
<P>
Genomic Evolutionary Rate Profiling 
<a href="http://mendel.stanford.edu/sidowlab/downloads/ABC_GERP/abcgerp.html" TARGET=_blank>(GERP)</a> 
scores describe nucleotide substitution rates observed in alignments 
containing all human bases in the ENCODE portion of the genome.  
Scores are estimated on a column-by-column basis using multiple sequence
alignments of mammalian genomic DNA generated by the 
<a href="http://lagan.stanford.edu/lagan_web/index.shtml" TARGET=_blank>LAGAN</a> alignment
package.  
<P>
The range of scores is from 0 to 3, with higher scores corresponding to human
bases in alignment columns with higher degrees of similarity (bases
that have evolved slowly, some of which have been under purifying selection),
while the opposite holds true for swiftly evolving (low similarity) columns. 
<H2>Methods</H2>
<P>
The GERP score is the expected substitution rate divided by
the observed substitution rate at a particular human base.
Scores over 3 are clipped to 3.  The expected and observed are 
both calculated on phylogenic tree using the same, fixed topology.  
The branch lengths of the expected tree are based on the average 
substitutions at neutral sites.  The observed tree is calculated
separately for each human base.  The branch lengths of the observed
tree is based on the substitutions seen at the column of the 
multiple alignment at that base.  Species which have gaps at
a particular column are not considered in the scoring for that
column.
<H2>Verification</H2>
<P>
Scores are deterministic given a maximum-likelihood model of 
nucleotide substitution, species topology, neutral tree, and alignment.

<H2>Credits</H2>
<P>
GERP was primarily developed by Greg Cooper in the lab of 
<a href="http://mendel.stanford.edu/sidowlab/" TARGET=_blank>Arend Sidow</a> 
at <a href="http://www.stanford.edu" TARGET=_blank>Stanford University</a> 
(Depts of Pathology and Genetics), in close collaboration with 
Eric Stone (Biostatistics, NC State), and George Asimenos and 
Eugene Davydov in the lab of Serafim Batzoglou (Dept of Computer 
Science, Stanford).

<H2>References</H2>
<P>
Cooper, G.M., Stone, E.A., Asimenos, G., NISC Comparative Sequencing Program, Green, E.D., Batzoglou, S., and Sidow, A.  2005.  
Distribution and intensity of constraint in mammalian genomic sequence.   <a href="http://www.genome.org" TARGET=_blank>Genome Research</a>. In press.
