<html><head></head><body><h2>Description</h2>
<p>
A ChIP-PET based sequencing approach that combines chromatin
immunoprecipitation (ChIP) with the paired-end di-tag (PET) sequencing
strategy was adopted to survey p53 TFBS on human genome. This track
shows PET sequences derived from 65,572 individual p53 ChIP fragments
of 5FU stimulated HCT116 cells. The actual PET sequences were
represented by blocks and each ChIP fragment was defined by two
boundary blocks connected by arrowed thin line. The PET singletons from
non-specific ChIP fragments were discretely located along the genome as
background. The overlapping PET clusters (PET fragment overlaps with
each other) that originated from ChIP enrichment process therefore
define the genomic loci which are potential p53 binding sites. </p><h2>Methods</h2>
<p> The ChIP enriched DNA fragments were cloned into a plasmid vector
which were then converted into paired-end ditags (PET) for
concatenation, cloning, and sequencing. The PET sequences containing
18bp from 5' end and 18bp from 3' end of original ChIP fragments were
extracted from raw sequence reads and mapped to the genome
independently for demarcating the ChIP DNA fragments. The specific
mapping criteria are that both the 5' and 3' signatures must be present
on the same chromosome, correct orientation, minimal 17bp match and
within 4 Kb of genomic distance. </p><h2>Verification</h2>
<p>
40 genomic locations identified by PET-3+ clusters were randomly
selected and subjected for quantitative real-time PCR analyses. The
relative enrichment of candidate regions compared to control GST ChIP
DNA was determined and all 40 regions (100%) were confirmed significant
enrichment of p53 ChIP, indicating that all these regions are true p53
bound targets. </p><h2>Credits</h2>
<p>
The p53 ChIP-PET library and sequence data were produced at Genome
Institute of Singapore. The data was mapped and analyzed by scientists
from Genome Institute and Bioinformatics Institute of Singapore, and
Boston University.
</p><h2>References</h2>
<p>
P. Ng et al., Nature Methods 2, 105 (2005).

</p></body></html>