<H2>Description</H2>

<P>Transcription factor binding sites (TFBS) were surveyed by adopting
a ChIP-PET based sequencing approach that combines chromatin
immunoprecipitation (ChIP) with paired-end di-tag (PET)
sequencing. This track currently shows PET sequences derived from
65,572 individual p53 ChIP fragments of 5FU-stimulated HCT116
cells.</P>

<P>In the display, the actual PET sequences are represented by
blocks. Each ChIP fragment is defined by two boundary blocks connected
by a thin arrowed line. The PET singletons from non-specific ChIP
fragments are displayed as background. Overlapping PET clusters (PET
fragments that overlap one another) originating from the ChIP
enrichment process define the genomic loci that are potential binding
sites.</P>

<H2>Methods</H2>

<P>The ChIP-enriched DNA fragments <B>[add more detail on the ChIP
enrichment before cloning]</B> were cloned into a plasmid vector and
converted <B>[what does 'converted into ditags' mean?  expand on this
a bit more]</B> into paired-end ditags (PET) for concatenation,
cloning, and sequencing. PET sequences containing 18 bp from the 5'
end and 18 bp from the 3' end of the original ChIP fragments were
extracted from raw sequence reads and mapped to the genome
independently for demarcating the ChIP DNA fragments. The following
specific mapping criteria were used:
<UL>
<LI>Both 5' and 3' signatures must be present on the same chromosome
<LI>The genomic distance between the tags ends must be less than 4 Kb
<LI>The orientations of the two tags must be consistent
<LI>At least 17 bp of each tag must align <B>[describe size of tags -- 18 or 20 bp]</B>
</UL></P>

<H2>Verification</H2>

<P>40 genomic locations identified by PET-3+ clusters <B>[define
PET-3+]</B> were randomly selected and subjected to quantitative
real-time PCR analyses. The relative enrichment of candidate regions
compared to control glutathione-S-transferase (GST) ChIP DNA was
determined and all 40 regions (100%) were confirmed to have
significant enrichment of p53 ChIP, indicating that all these regions
are true p53-bound targets. <B>[It seems like this isn't enough
verification -- there's not enough to get a sense of the FP rate, and
doesn't address the FN rate at all.]</B>
</P>

<H2>Credits</H2>

<P>
The p53 ChIP-PET library and sequence data were produced at the <A
HREF="http://www.gis.a-star.edu.sg/homepage/default.jsp"
TARGET=_BLANK>Genome Institute of Singapore </A>.  The data were
mapped and analyzed by scientists from the Genome Institute and
Bioinformatics Institute of Singapore and from Boston University. </P>

<H2>References</H2>

<P>Ng, P. <em>et al</em>. <B>[Expand this??]</B>
<A HREF="http://www.nature.com/nmeth/journal/v2/n2/abs/nmeth733.html"
TARGET=_blank>Gene identification signature (GIS) analysis for
transcriptome characterization and genome annotation</A>.  
<em>Nature Methods</em> <B>2</B>:105-111 (2005). </P>
