<H2>Description</H2>
<P>
This track displays different measurements of conservation based on 
the Threaded Blockset Aligner (TBA) multiple sequence alignments of 
ENCODE regions displayed in the TBA Alignments track. Two programs -- binCons
(binomial conservation method) and phastCons (phylogenetic hidden-Markov
model method) -- were used to generate the conservation scoring
used to create this track.  A third related track,
TBA Elements, shows multi-species conserved elements (MCSs) based on
the conservation measurements displayed in this track.</P>

<H2>Display Conventions and Configuration</H2>
<P>
The subtracks within this composite annotation track, which
show data from the binCons and phastCons programs, may be configured in a 
variety of ways to highlight different aspects of the 
displayed data. The graphical configuration options 
are shown at the top of the track description page, followed by a list of 
the subtracks.  A subtrack may be hidden from view by checking the box to the
left of the track name in the list.  For more information about the 
graphical configuration options, click the 
<A HREF="../goldenPath/help/hgWiggleTrackHelp.html" TARGET=_blank>Graph
configuration help</A> link.</P>

<H2>Methods</H2>
<P>
The methods used to create the TBA alignments in the ENCODE
region are described in the TBA Alignments track description.</P>

<H3>Binomial-based Conservation Score (binCons)</H3>
<P>
The binCons score is based on the cumulative binomial probability of 
detecting the observed number of identical bases (or greater) in 
sliding 25 bp windows (moving one bp at a time) between the 
reference sequence and each other species, given the neutral rate
at four-fold degenerate sites. Neutral rates are calculated
separately at each targeted region. For targets with no gene annotations,
the average percent identity across all alignable sequence was instead used
to weight the individual species binomial scores (this latter
weighting scheme was found to closely match 4D weights).</P>
<P>
The negative log of these P-values was then averaged across all 
human-referenced pairwise combinations, and the highest scoring overlapping 
25 bp window for each base was the resulting score. This track shows the 
plotting of a ranked percentile score normalized between 0 and 1 across all 
ENCODE regions, such that the top 5% most conserved sequence across all ENCODE
regions have a score of 0.95 or greater (top 10% have a score of 0.9 or 
greater, and so on).</P>
<P>
More details on binCons can be found in Margulies <em>et. al</em>. (2003)
cited below.</P>

<H3>Phylogenetic HMM-based Conservation Score (phastCons)</H3>
<P>
The phastCons program predicts conserved elements and produces base-by-base
conservation scores using a two-state phylogenetic hidden Markov model.
The model consists of a state for conserved regions and a 
state for nonconserved regions, each of which is associated with a 
phylogenetic model.  These two models are identical
except that the branch lengths of the conserved phylogeny are 
multiplied by a scaling parameter rho (0 &lt; rho &lt; 1).</P>
<P>
For determining the conservation for the ENCODE TBA
alignments,  the nonconserved model was estimated 
from four-fold degenerate coding sites within the ENCODE regions using 
the program phyloFit. The parameter rho was then estimated by 
maximum likelihood, conditional on the nonconserved model, using the EM 
algorithm implemented in phastCons.  Parameter estimation was based on 
a single large alignment, constructed by concatenating the 
alignments for all conserved regions.</P>
<P>
PhastCons was run with the options <TT>--expected-lengths 15</TT> and
<TT>--target-coverage 0.05</TT> to obtain the desired level of 
&quot;smoothing&quot; and a final coverage by conserved elements of 5%.</P>
<P>
The conservation score at each base is the posterior probability that the
base was generated by the conserved state of the phylo-HMM.  It can
be interpreted as the probability that the base is in a conserved
element, given the assumptions of the model and the estimated parameters.</P>
<P>
More details on phastCons can be found in Siepel <em>et. al.</em> (2004)
cited below.</P>

<H2>Credits</H2>
<P>
The data for this track were generated by Elliott Margulies of 
the Green lab at NHGRI, with assistance from Adam Siepel of UCSC.</P>

<H2>References</H2>
<P>
<H3>BinCons References:</H3>
<P>
Margulies, E.H., Blanchette, M., NISC Comparative Sequencing Program, 
Haussler, D. and Green, E.D. 
<A HREF="http://dx.doi.org/10.1101/gr.1602203"
TARGET=_blank>Identification and characterization of multi-species conserved 
sequences</A>. 
<em>Genome Res.</em> <B>13</B>, 2507-18 (2003). </P>
<P>
<H3>PhastCons References:</H3>
<P>
Felsenstein, J. and Churchill, G.A. 
A hidden Markov model approach to variation among sites in rate of evolution.
<em>Mol Biol Evol</em> <B>13</B>, 93-104 (1996). </P>
<P>
Siepel, A. and Haussler, D. 
<A HREF="http://www.cse.ucsc.edu/~acs/phyhmm.pdf"
TARGET=_blank>Combining phylogenetic and hidden Markov models in biosequence 
analysis</A>. 
In <em>Proceedings of the Seventh Annual International Conference on 
Computational Molecular Biology (RECOMB 2003)</em>. pp. 277-286 (2003).</P>
<P>
Siepel, A. and Haussler, D. 
Phylogenetic hidden Markov models.
In <em>Statistical Methods in Molecular Evolution</em> (ed. Nielsen, R.). 
<em>In press</em> (Springer, 2004).  </P>
