<html>
<H2>Description</H2>
<P>
This track shows updated versions of gene predictions submitted for the 
<A TARGET=_BLANK HREF="http://genome.imim.es/gencode/workshop2005.html">EGASP/2005</A>
ENCODE Gene Prediction Workshop.
The following gene predictions are included:
<UL>
<LI>
<A TARGET=_BLANK href="http://cbcb.umd.edu/software/jigsaw">JIGSAW</a>
</LI>
<LI>
<A TARGET=_BLANK href="http://augustus.gobics.de/"> AUGUSTUS</A>
</LI>
<LI>Exogean</LI>
<LI><A TARGET=BLANK href="http://genome.imim.es/software/geneid/index.html">geneid</A>-u12 </LI>
<LI><a target=_BLANK href="http://genome.imim.es/software/sgp2/index.html">sgp2</a>-u12</LI>
<LI><A TARGET=BLANK HREF="http://www.pseudogenes.org">Yale Pseudogenes</A></LI>
</UL>
The original EGASP submissions are displayed in the companion tracks, 
<EM>EGASP Full</EM> and <EM>EGASP Partial</EM>.
<H2>Methods</H2>
<P>
<H3>Jigsaw</H3>
JIGSAW is a gene prediction program which calls genes based on 
target genomic sequence and output from a gene structure annotation database.
Data available for download from UCSC's annotation database is 
used as input and includes the following tracks of evidence:
Known Genes, Ensembl, RefSeq, Geneid, Genscan, SGP, Twinscan, Human mRNAs,
TIGR Gene Index, UniGene, 'Most Conserved' Elements and Non-human RefSeq Genes.
<a href="http://cbcb.umd.edu/software/GlimmerHMM">GlimmerHMM</a> and <a href="http://genezilla.org"> Genezilla</a>, two open source <i>ab initio</i> gene finding programs
are also used.
<H3>Augustus</H3>
Augustus is a generalized Hidden Markov Model (GHMM) that
models coding and non-coding sequence, slice
sites, the branch point region, the tranlation start and end
and the lengths of exons and introns. This version has
been trained on a training set with 1284 human genes.
There are four sets of predictions in this track: ab initio,
EST and protein-based, mouse homology-based, and those using
EST/protein and mouse homology evidence as additional input to Augustus
for the predictions.
<P>
The EST and protein evidence was generated by aligning sequences from the dbEST and the nr databases to the ENCODE region using wublastn and wublastx.
The resulting alignments were used to generate hints about putative splice 
sites, exons, coding regions, introns, the translation start and the 
translation stop. 
<P>
The mouse homology evidence was generated by aligning pairs of human and
mouse genomic sequences using the program <a href="http://dialign.gobics.de/chaos-dialign-submission">DIALIGN</a>.
Regions conserved at the peptide level were used to generate hints about coding regions.
<P>

<H3>Exogean</H3>
Exogean produces alternative transcripts by combining mRNA and cross-species sequence alignement using heuristic rules.
Exogean implements a generic framework based on directed acyclic coloured multigraphs (DACMs).
In Exogean, DACMs nodes represent biological objects (mRNA or protein HSPs/transcripts) and
multiple edges between nodes represent known relations between these objects derived from human expertise.
Exogean DACMs are succesively built and reduced leading to more and more complexe objects. This process
enables the production of alternative transcripts from initial HSPs.
<H3>Geneid-u12</H3>
The <em>geneid</em> program predicts genes in anonymous genomic sequences 
designed with a hierarchical structure.
In the first step, splice sites, start and stop codons are predicted and scored along the sequence using Position Weight Arrays (PWAs).
Next, exons are built from the sites. Exons are scored as the sum of the scores of the defining sites, plus the the log-likelihood ratio of a Markov Model for coding DNA.
Finally, from the set of predicted exons, the gene structure is assembled, maximizing the sum of the scores of the assembled exons.
The modified version of <em>geneid</em> used to generate the predictions in this track uses models for U12-dependent splice signals, in addition to U2 splice signals.
<P>
All <em>geneid</em> genes are displayed in the GeneID subtrack. Only U12 introns
and their flanking exons are displayed in the GeneID U12 subtrack.
Exons flanking predicted U12-dependent introns are assigned an a type
attribute that reflects their splice sites.  This is displayed on
the details page of the GeneID U12 subtrack as the "Alternate Name" of the 
item composed of the intron plus flanking exons.
<H3>Sgp2-u12</H3>
To predict genes in a genomic query, <em>sgp2</em> combines <em>geneid</em> predictions with <em>tblastx</em> comparisons of the genomic query against other genomic sequences.
This modified version of <em>sgp2</em> uses models for U12-dependent splice signals, in addition to U2 splice signals. 
The reference genomic sequence for this data set is the OCT 2004 release of mouse sequence syntenic to ENCODE regions.
<P>
The SGP2 and SGP2U12 tracks follow the same display conventions as the 
GeneID and GeneID U12 subtracks described above.
<H3>Yale Pseudogenes</H3>
For this analysis, pseudogenes were defined as genomic sequences similar 
to known human genes and with various disablements (premature stop codons or
frameshifts) in their "putative" protein coding regions.
<P>
The protein sequences of known human genes (as annotated by ENSEMBL) were used
to search for similar nogenic sequences in ENCODE regions.  The matching
sequences were assessed as disabled copies of genes based on the occurrences of
premature stop codons or frameshifts.  The intron-exon structure of the
functional gene was further used to infer whether a pseudogene was a duplicated
or processed one. A duplicated pseudogene keeps the intron-exon structure of its
parent functional gene. Small pseudogene sequences were labeled as fragments or
other types.  
<P>
All pseudogenes in this track were manually curated.
In the browser, the track details page shows the pseudogene type.
<H2>Credits</H2>
<P>
Jigsaw was developed at The Institute for Genomic Research 
by Jonathan Allen and Steven Salzberg,
with computational genefinder contributions from Mihaela Pertea and William Majoros.
<P>
AUGUSTUS was written by Mario Stanke at the
Department of Bioinformatics of the University of Göttingen in Germany.
<P>
Exogean was developed by Sarah Djebali and Hugues Roest Crollius from the
Dyogen Lab, Ecole Normale Supérieure (Paris, France) and Franck Delaplace
from the LAMI, (Evry, France).
<P>
The <em>geneid-u12</em> and <em>sgp2-u12</em> programs were developed at the
<a href="http://www.imim.es/grib/eng/default.htm">Grup de Recerca en Informàtica
Biomèdica (GRIB)</a> at the Institut Municipal d'Investigació Mèdica (IMIM) in Barcelona.
The version of <em>geneid</em> on which GeneIDu12 is based (geneid_v1.2) was written by Enrique Blanco and Roderic Guigó.
The parameter files were constructed by Genis Parra and Francisco Camara.
Additional contributions were made by Josep F.Abril, Moises Burset and Xavier Messeguer.
Modifications to <em>geneid</em> which allow for the prediction of U12-dependent splice sites and incorporation
of U12 introns into gene models were made by Tyler Alioto.
<P>
The Yale Pseudogenes were generated by the pseudogene annotation group of 
Mark Gerstein at Yale University.

<H2>References</H2>
<H3>Jigsaw</H3>
J. E. Allen, S. L. Salzberg
JIGSAW: integration of multiple sources of evidence for gene prediction
2005 under submission.
<P>
J. E. Allen, M. Pertea and S.L. Salzberg
Computational gene prediction using multiple sources of evidence. 
Genome Research, 14(1), 2004.
<H3>Augustus</H3>
Mario Stanke (2004)<br>
"<a href=http://webdoc.sub.gwdg.de/diss/2004/stanke/index.html>Gene Prediction with a Hidden Markov Model</a>",<br>
<i>Ph.D. thesis</i>, Universit&auml;t G&ouml;ttingen, Germany</a>
<P>
Mario Stanke and Stephan Waack (2003)<br>
"<a href=http://bioinformatics.oupjournals.org/cgi/content/abstract/19/suppl_2/ii215>Gene 
prediction with a hidden Markov model and a new intron submodel</a>",<br>
<i>Bioinformatics</i>, Vol. 19, Suppl. 2, pages ii215-ii225
<P>
Mario Stanke, Rasmus Steinkamp, Stephan Waack and Burkhard Morgenstern (2004)<br>
"<a href=http://nar.oupjournals.org/cgi/content/full/32/suppl_2/W309>AUGUSTUS: a 
web server for gene finding in eukaryotes</a>",<br>
<i>Nucleic Acids Research</i>, Vol. 32, W309-W312


<H3>Geneid</H3>
E. Blanco, G. Parra and R. Guigó, "Using geneid to Identify Genes.", In A. Baxevanis, editor: Current Protocols in Bioinformatics. Unit 4.3. John Wiley & Sons Inc., New York (2002)
<P>
G. Parra, E. Blanco, and R. Guigó, "Geneid in Drosophila", Genome Research 10(4):511-515 (2000).
<P>
R. Guigó, "Assembling genes from predicted exons in linear time with dynamic programming", Journal of Computational Biology, 5:681-702 (1998).
<P>
R. Guigó, S. Knudsen, N. Drake, and T. F. Smith,"Prediction of gene structure", Journal of Molecular Biology, 226:141-157 (1992).
<H3>Sgp2</H3>
G. Parra, P. Agarwal, J.F. Abril, T. Wiehe, J.W. Fickett and R. Guigó. "Comparative gene prediction in human and mouse." Genome Research 13(1):108-117 (2003)
<P>
R. Guigó, E.T. Dermitzakis, P. Agarwal, C.P. Ponting, G. Parra, A. Reymond, J.F. Abril, E. Keibler, R. Lyle, C. Ucla, S.E. Antonarakis and M.R. Brent. "Comparison of mouse and human genomes followed by
experimental verification yields an estimated 1,019 additional genes." Proc. Nat. Acad. Sci. 100(3):1140-1145 (2003)

</html>
