<H2>Description</H2>
<P>
This track shows clusters of RNA transcription for all ENCODE
regions.  Retinoic acid-stimulated HL-60 cells were harvested after
0, 2, 8, and 32 hours.  Purified polyA+ RNA from these samples was
hybridized to Affymetrix ENCODE oligonucleotide tiling arrays.
These arrays have 25-mer probes tiled every 21 bp on average in the
non-repetitive ENCODE regions.

<H2>Methods</H2>
<P>
The data from replicate arrays were quantile-normalized (Bolstad
<I>et al</I>.) and all arrays were scaled to a median array intensity
of 25. Within a sliding 101 bp window centered on each probe, an estimate 
of RNA abundance (signal) was found by calculating the median of all pairwise 
average PM-MM (where PM is perfect match and MM is mismatch) values.
Transcribed regions were generated by applying a
cutoff to the abundance estimate or 'signal' associated with a 5%
false positive rate in negative bacterial controls tiled on the
array and joining probes within 50bp of probes which pass the
threshold.  (Both Kaproanov <I>et al</I>. and Kampa <I>et al</I>. are
good references for experimental methods; Kampa <I>et
al</I>. describes analytical methods.)

<H2>Verification</H2>
<P>
Three independent biological replicates were generated and
hybridized to duplicate arrays (two technical replicates).
Transcribed regions were generated from this signal track based on a
5% false positive rate cutoff in negative bacterial controls.  These
sites were then ranked, individually in each of the replicates,
based on their signal as computed from the respective replicates.
All pairwise differences in these ranks were computed among
biological replicates and replicates were accepted when the sum of
rank differences was relatively low (there is currently no hard
cutoff criteria).

<H2>Credits</H2>
<P>
These data were generated and analyzed by the Gingeras/Struhl
collaboration with Tom Gingeras' group at Affymetrix and <A
HREF="http://www.hms.harvard.edu/dms/bbs/fac/struhl.html"
TARGET=_BLANK> Kevin Struhl</A>'s group at Harvard Medical School.

<H2>References</H2>
<P>
Please see the <A TARGET=_BLANK
HREF="http://www.affymetrix.com/transcriptome/index.affx">
Affymetrix Transcriptome</A> site for a project overview and
additional references to Affymetrix tiling array publications.
<P>
Bolstad, B. M., Irizarry, R. A., Astrand, M., and Speed,
T. P. A comparison of normalization methods for high density
oligonucleotide array data based on variance and
bias. <I>Bioinformatics</I> <B>19</B>, 185-193 (2003).
<P>
Kampa, D., Cheng, J., Kapranov, P., Yamanaka, M., Brubaker, S.,
Cawley, S., Drenkow, J., Piccolboni, A., Bekiranov, S., Helt, G., et
al. Novel RNAs identified from an in-depth analysis of the
transcriptome of human chromosomes 21 and 22. <I>Genome Res</I> <B>14</B>,
331-342 (2004).
<P>
Kapranov, P., Cawley, S. E., Drenkow, J., Bekiranov, S., Strausberg,
R.  L., Fodor, S. P., and Gingeras, T. R. Large-scale
transcriptional activity in chromosomes 21 and 22. <I>Science</I> 
<B>296</B>, 916-919 (2002).
