<H2>Description</H2>
<P>
This track displays different measurements of conservation based on 
the TBA multiple sequence alignments of the ENCODE regions that are
displayed in the TBA Alignments track.  The programs binCons
(binomial conservation method) and phastCons (phylogenetic hidden-markov
model method) were used to generate conservation scoring
used to create this track.  A third related track,
TBA Elements, shows conserved elements (MCS's) based on
the conservation measurements displayed in this track.

<H2>Methods</H2>
<P>
The methods used to create the TBA alignments in the ENCODE
region are described in the TBA Alignments track description.
<P>
<em>Binomial-based Conservation Score (binCons)</em>
<P>
The binCons score is based on the cumulative binomial probability of 
detecting the observed number of identical bases (or greater) in 
sliding 25 bp windows (moving 1 bp at a time) between the 
reference sequence and each other species, given the neutral rate
at four-fold degenerate sites. Neutral rates are calculated
separately at each targeted region. For targets with no gene annotations,
the average percent identity across all alignable sequence was instead used
to weight the individual species binomial scores (we have found this latter
weighting scheme to closely match 4D weights).
<P>
The negative log of these p-values was then averaged across all human-
referenced pair-wise combinations and the highest scoring overlapping 25 bp
window for each base is the resulting score. Plotted here is a ranked
percentile score normalized between 0 and 1 across all ENCODE regions, such
that the top 5% most conserved sequence across all ENCODE regions have a
score >= 0.95 (top 10% have a score of 0.9 or greater, and so on).
<P>
More details on binCons can be found in Margulies, et. al.,
cited below.
<P>
<em>Phylogenetic HMM-based Conservation Score (phastCons)</em>
<P>
The phastCons program predicts conserved elements and produces base-by-base
conservation scores using a two-state phylogenetic hidden Markov model,
The model consists of a state for conserved regions and a 
state for nonconserved regions, each which is associated with a 
phylogenetic model.  These two models are identical
except that the branch lengths of the conserved phylogeny are 
multiplied by a scaling parameter rho (0 < rho < 1).  
<P>
For determining the conservation for the ENCODE TBA
alignments,  the nonconserved model was estimated 
from fourfold degenerate coding sites within the ENCODE regions using 
the program phyloFit, and then the parameter rho was estimated by 
maximum likelihood, conditional on the nonconserved model, using the EM 
algorithm implemented in phastCons.  Parameter estimation was based on 
a single large alignment, constructed by concatenating together the 
alignments for all conserved regions.
<P>
PhastCons was run with the options --expected-lengths 15 and
--target-coverage 0.05, to obtain the desired level of "smoothing"
and a final coverage by conserved elements of 5%.
<P>
The conservation score at each base is the posterior probability that that
base was generated by the conserved state of the phylo-HMM.  It can
be interpreted as the probability that the base is in a conserved
element, given the assumptions of the model and the estimated parameters.
<P>
More details on phastCons can be found in Siepel, et. al.,
cited below.
<H2>Credits</H2>
<P>
The data for this track were generated by Elliott Margulies of 
the Green lab at NHGRI, with assistance from Adam Siepel of UCSC.
<H2>References</H2>
<P>
<em>BinCons References</em>
<P>
<A TARGET=_BLANK HREF=http://dx.doi.org/10.1101/gr.1602203>
Margulies et al. (2003)</A>
<P>
<em>PhastCons References</em>
<P>
Siepel A and Haussler D (2003).
<a href="http://www.cse.ucsc.edu/~acs/phyhmm.pdf">Combining
phylogenetic and hidden Markov models in biosequence analysis</a>.
In <em> Proceedings of the Seventh Annual 
International Conference on Computational Molecular Biology (RECOMB
        2003)</em>, pp. 277-286. </P>
<P>
Siepel A and Haussler D (2004). Phylogenetic hidden Markov models.
In R. Nielsen, ed.,  Statistical Methods in Molecular Evolution, Springer
(in press). </P>
<P>
Felsenstein J and Churchill GA (1996).  A hidden Markov model approach to
variation among sites in rate of evolution.
<em>Mol Biol Evol</em> 13:93-104. </P>

