<H2>Description</H2>
<P>
Genomic Evolutionary Rate Profiling 
<a href=http://mendel.stanford.edu/sidowlab/downloads/ABC_GERP/abcgerp.html>(GERP)</a> 
scores describe nucleotide substitution rates observed in alignments 
containing all human bases in the ENCODE portion of the genome.  
Scores are estimated on a column-by-column basis using multiple sequence
alignments of mammalian genomic DNA generated by the 
<a href=http://lagan.stanford.edu/lagan_web/index.shtml>LAGAN</a> alignment
package.  
<P>
The range of scores is from 0 to 3, with higher scores corresponding to human
bases in alignment columns with higher degrees of similarity (bases
that have evolved slowly, some of which have been under purifying selection),
while the opposite holds true for swiftly evolving (low similarity) columns. 
<H2>Methods</H2>
<P>
Alignments, generated target by target for all ENCODE regions, 
are first compressed so that the human sequence is ungapped.  
Maximum likelihood rate estimation is used to estimate an "observed" 
rate of evolution for each column, given a fixed species topology relating
the sequences, where gapped species are eliminated from consideration.  
A neutral tree that describes the average rates of evolution (in terms
of nucleotide substitutions affecting neutral sites) over all the branches
of the species topology relating the analyzed sequence is used to generate 
an estimate of the "expected" rate of evolution for each column, 
where gapped species are eliminated.  
The scores plotted reflect the inverse of the ratio of "observed" 
to "expected", so higher scores (capped so that columns of identical
residues are scored at 3) show high levels of similarity.  
<H2>Verification</H2>
<P>
Scores are deterministic given a maximum-likelihood model of 
nucleotide substitution, species topology, neutral tree, and alignment.

<H2>Credits</H2>
<P>
GERP was primarily developed by Greg Cooper in the lab of 
<a href=http://mendel.stanford.edu/sidowlab/>Arend Sidow</a> 
at <a href=http://www.stanford.edu>Stanford University</a> 
(Depts of Pathology and Genetics), in close collaboration with 
Eric Stone (Biostatistics, NC State), and George Asimenos and 
Eugene Davydov in the lab of Serafim Batzoglou (Dept of Computer 
Science, Stanford).

<H2>References</H2>
<P>
Cooper, G.M., Stone, E.A., Asimenos, G., NISC Comparative Sequencing Program, Green, E.D., Batzoglou, S., and Sidow, A.  2005.  
Distribution and intensity of constraint in mammalian genomic sequence.   <a href=http://www.genome.org>Genome Research</a>. In press.
