<H2>Description</H2>
<A HREF="http://www.sanger.ac.uk/Software/analysis/eponine/" TARGET=_BLANK>
Eponine</A> is a probabilistic method for detecting transcription start
sites (TSS) in mammalian genomic sequence, with good specificity and
excellent positional accuracy.

<H2>Methods</H2>
Eponine models consist of a set of DNA weight matrices recognizing
specific sequence motifs. Each of these is associated with a position
distribution relative to the transcription start site. The model
currently in use on this server is shown below:

<P>
Eponine has been tested by comparing the output with annotated mRNAs
from human chromosome 22. From this work, we estimate that using the
default threshold (0.999) it detects >50% of transcription start
sites, with around 70% specificity. However, it does not always
predict the direction of transcription correctly -- an effect which
seems to be common among computational TSS finders.
</P>

<P>
Thomas A. Down and Tim J. P. Hubbard. 2002.
Computational Detection and Location of Transcription Start Sites 
in Mammalian Genomic DNA
<A HREF="http://www.genome.org/cgi/content/abstract/12/3/458?maxtoshow=&HITS=10&hits=10&RESULTFORMAT=&searchid=1015426262427_886&stored_search=&FIRSTINDEX=0&volume=12&firstpage=458&journalcode=genome" TARGET=_BLANK>
Genome Res. 12:458-461</A>.
</P>

<H2>Credits</H2>
Thanks to Thomas Down at the 
<A HREF="http://www.sanger.ac.uk/" TARGET=_BLANK>Sanger Institute</A> 
for providing the 
<A HREF="http://www.sanger.ac.uk/Software/analysis/eponine/" TARGET=_BLANK>
Eponine</A> program, which was run at UCSC to produce 
this track.

