<H2> Description </H2>
<P>
This track shows RNA secondary structure predictions made with the
EvoFold program. The specific secondary structure predictions are
shown in parenthesis format at the base-level, with position-specific
confidence scores indicated by the use of shades of gray. When zoomed
out, stem-pairing regions will be shown in a darker shade than
un-paired regions. </P>

<P> The predicted secondary structures (SS anno) are shown together
with a multiple alignment on the details page. Substitutions, relative
to the human sequence, are color coded according to their
compatibility with the predicted secondary structure (see color
legend). Each prediction is assigned an overall score and a sequence
of position specific scores. The overall score measures evidence for
any functional RNA structures in the given region, while the position
specific scores (0-9) measure the confidence of the base specific
annotations. Positions which are base pairing are annotated with the
same pair symbol.</P>

<H2> Methods </H2>
<P>
EvoFold is a comparative method which exploits the evolutionary signal
of genomic multiple-sequence alignments for identifying conserved
functional RNA structures. The method makes use of phylogenetic
Stochastic Context-free Grammars (phylo-SCFGs), which are combined
probabilistic models of RNA secondary structure and primary sequence
evolution. The predictions consist both of a specific RNA secondary
structure and an overall score. The overall score is essentially a
log-odd score phylo-SCFG modeling the constrained evolution of
stem-pairing regions and one which only models unpaired regions.</P>

<P>
The predictions for this track were based on an 8-way vertebrate
alignment (containing: human, chimpanzee, mouse, rat, dog, chicken,
zebra-fish, and puffer-fish). These predictions were lifted from the
May 2004 assembly to the July 2003 assembly, and are therefore shown
with a different alignment above.
</P>

<H2>References</H2>
<P> Pedersen, J.S., Bejerano, G., and Haussler, D. Identification and
    Classification of Conserved RNA Secondary Structures in the Human
    Genome. <em>(in preparation)</em>.

<P> Siepel, A., Bejerano, G., Pedersen, J.S., Hinrichs, A., Hou, M.,
Rosenbloom, K., Clawson, H., Spieth, J., Hillier, L. W., Richards, S.,
Weinstock, G. M., Wilson, R. K., Gibbs, R. A., Kent, W. J., Miller,
W., and Haussler, D. Evolutionarily conserved elements in vertebrate,
insect, worm, and yeast genomes. <I>Genome Res</I> <em>(in press)</em>.</P>

<P>Knudsen, B. and J.J. Hein.  <A
HREF="http://nar.oxfordjournals.org/cgi/content/full/31/13/3423"
TARGET=_blank> Using stochastic context free grammars and molecular
evolution to predict RNA secondary
structure</A>. <em>Bioinformatics</em> <B>15</B>, 446-54 (1999).</P>
</P>

<P> Pedersen, J.S., I.M. Meyer, R. Forsberg, P. Simmonds and
J. Hein. <A
HREF="http://nar.oxfordjournals.org/cgi/content/full/32/16/4925"
TARGET=_blank>A comparative method for finding and folding RNA
secondary structures within protein-coding regions</A>. <em> Nucleic
acids research </em>, <B>32</B>, 4925-36 (2004).</P>

<H2>Credits</H2> 
The EvoFold program and browser track were developed by <A
HREF="mailto:jsp@soe.ucsc.edu">Jakob Skou Pedersen</A> of the UCSC
Genome Bioinformatics Group.
