<H2>Description</H2>
<P>
The EXONIPHY program identifies evolutionarily conserved protein-coding exons in
multiple, aligned sequences using a phylogenetic hidden Markov model
(phylo-HMM) -- a kind of statistical model that simultaneously describes exon
structure and exon evolution.  This track shows EXONIPHY predictions
for the human Jul. 2003 (hg16), mouse Feb. 2003 (mm3), and rat Jun. 2003
(rn3) genomes, as aligned by the MULTIZ program.
</P>
<H2>Methods</H2>
<P>
EXONIPHY is described in Siepel A & Haussler D (2004), "<a
href="http://www.cse.ucsc.edu/~acs/recomb2004.pdf">Computational
identification of evolutionarily conserved exons</a>," <em>RECOMB '04.</em>
MULTIZ is described in Blanchette M <em>et al.</em> (2004), "<a
href="http://www.genome.org/cgi/content/abstract/14/4/708">Aligning
multiple genomic sequences with the threaded blockset aligner</a>,"
<em>Genome Res.</em> 14:708-175.
