<H2>Description</H2> <P>
This track shows a measure of evolutionary conservation in human, mouse, rat, and
chicken based on a phylogenetic hidden Markov model (phylo-HMM).
The multiz alignments of the human Jul. 2003 (hg16), mouse Feb. 2003 (mm3), rat Jun. 2003 (rn3), and chicken Feb. 2004 (galGal2) assemblies were used to generate
the annotation.</P>

<P>
In &quot;full&quot; visibility mode, this track displays pairwise alignments 
of mouse, rat, and chicken, each aligned to the human genome.  The pairwise alignments are
displayed in the standard UCSC browser &quot;dense&quot; mode using a greyscale 
density gradient.  The checkboxes in the track configuration section allow
the exclusion of species from the pairwise display; however, this does not
remove them from the conservation score display.
</P>

<P>
When zoomed-in to the base display level, the track shows the base 
composition of each alignment. To view detailed information about the 
alignments at a specific position, zoom in the display to 30,000 or fewer bases, 
then click on the alignment.
</P>

<H2>Methods</H2>

<P> Best-in-genome blastz pairwise alignments of human-mouse and
human-rat were multiply aligned using a program called humor
(HUman-MOuse-Rat), which is a special variant of the <em>Multiz</em>
program.  Multiz was used to align the humor results with
best-in-genome blastz human-chicken alignments.  The resulting
human-mouse-rat-chicken multiple alignments were then assigned 
conservation scores by phylo-HMM.</P>

  <P>A phylo-HMM is a probabilistic model that describes both the process
    of DNA substitution at each site in a genome, and the way this process
    changes from one site to the next (Felsenstein and Churchill 1996,
    Yang 1995, Siepel and Haussler 2003).  A phylo-HMM can be thought
    of as a machine that generates a multiple alignment, in the same way
    that an ordinary hidden Markov model (HMM) generates an individual
    sequence.  While the states of an ordinary HMM are associated with
    simple multinomial probability distributions, however, the states of a
    phylo-HMM are associated with more complex distributions defined by
    probabilistic phylogenetic models.  These distributions can capture
    differences in the rates and patterns of nucleotide
    substitution observed in different types of genomic regions (e.g., coding
    or noncoding regions, conserved or nonconserved regions).

    <P> <img src="/images/fels-church.jpg" border="1"
    align="right"> To compute a conservation score, we use a
    <em>k</em>-state phylo-HMM, whose <em>k</em> associated phylogenetic
    models differ only in overall evolutionary rate (Felsenstein and
    Churchill 1996, Yang 1995). In the image at right, k = 3 states, 
    S<SUB>1</SUB>, S<SUB>2</SUB>, and S<SUB>3</SUB>, but in practice we 
    use k = 10.  
    Using a separate "rate constant," <EM>r<SUB>i</SUB></EM>, for each state i, 
    which is multiplied by all branch lengths in the globally estimated model,
    a phylogenetic model is estimated globally, using the discrete gamma model
    for rate variation (Yang 1994), then a scaled version of the estimated model
    is associated with each state in a phylo-HMM.  The
    transition probabilities between states allow for autocorrelation of
    substitution rates, i.e., for adjacent sites to tend to exhibit similar
    overall substitution rates.  A single parameter, lambda, describes the
    degree of autocorrelation and defines all transition probabilities (see
    picture).  Here, we have estimated the rate constants from the data,
    similarly to Yang (1995) (see Siepel and Haussler 2003), but have
    allowed lambda to be treated as a tuning parameter.  For the
    conservation score, we use the posterior probability that each site was
    "generated" by the state having the smallest rate constant.  Because of
    the way the rate categories are defined, the plotted values can be
    thought of as approximately representing the posterior probability that
    each site is among the 10% most conserved sites in the data set
    (allowing for autocorrelation of substitution rates).  </P>

    <P> In this case, the general reversible (REV) substitution model was
    used in parameter estimation, and lambda was set to 0.9.  Alignment
    gaps were treated as missing data, which sometimes has the effect of
    producing undesirably high posterior probabilities in gappy regions of
    the alignment.  We are looking at several possible ways of improving
    the handling of alignment gaps.</P>


<H2>Using the Filter</H2>
<P>
The track filter can be used to configure display attributes of
the track's graph. 
<UL>
<LI><B>Pairwise alignments</B>: The checked boxes indicate the pairwise 
alignments displayed in the track. To exclude one or more species, uncheck the
appropriate boxes.
<LI><B>Type of graph</B>: The default &quot;Bar&quot; setting depicts the graph
data using color-filled bars. To view the data as a series of points or lines, 
select the &quot;Points&quot; setting. 
<LI><B>y = 0.0 line</B>: Select the &quot;ON&quot; setting to display a
line marking the 0.0 position on the graph (default is &quot;OFF&quot;).
<LI><B>Track height</B>: To change the default display height of the graph
in pixels, type in a value from within the indicated range.
<LI><B>Vertical viewing range</B>: The min and max values specify the 
vertical portion of the graph that is displayed (default
range is 0-0.01). These numbers can be used to set data threshold indicators. 
For example, setting the min to &quot;0.005&quot; restricts the display
to data equal to or greater than that value. 
<LI><B>Data view scaling</B>: The default &quot;Use vertical viewing range
setting&quot; option displays the data using the parameters specified in the Vertical
viewing range setting. To configure the graph to automatically scale to a
range defined by the minimum and maximum data points in the current 
view, select the &quot;Auto-scale to data view&quot; option. 
<LI><B>Windowing function</B>: When a  view is too large to show individual
data values, the values must be combined to produce a plot point.
This option specifies the combining function to be used: &quot;Maximim&quot; - 
the maximum value of all the points to be combined, &quot;Mean&quot; - 
the mean, or &quot;Minimum&quot; - the minimum of all the
points to be combined (default is &quot;Mean&quot;).
<LI><B>Smoothing window</B>: When set to a numerical value, this option 
determines the size, in pixels, of a smoothing window
to be passed over the plot to smooth the edges of the bars or lines.  
This is equivalent to a trend line calculation on the graph.
The default setting is &quot;OFF&quot;.
<LI><B>Draw indicator line at y =</B>: When set to &quot;ON&quot;, the track
displays an arbitrary line on the graph at the specified numerical value 
(defaults are &quot;0&quot; and &quot;OFF&quot;). This line can be used to 
mark a significant threshold on the graph.
</UL></P>
<P>
When you have finished configuring the filter, click the <em>Submit</em> button.
</P>

<H2>Credits</H2>
<P>
This track was created at UCSC using the following programs:
<UL>
<LI>
Blastz and multiz from Scott Schartz and Webb Miller of the 
<A HREF="http://bio.cse.psu.edu/" TARGET=_blank>Penn State Bioinformatics 
Group</A>. 
<LI>
AxtBest, axtChain, chainNet, netSyntenic, and netClass 
developed by Jim Kent at UCSC. 
<LI>Tree estimation and phylo-HMM software by Adam Siepel at UCSC.
<LI>"Wiggle track" plotting software was developed by Hiram Clawson at UCSC.
</UL>
</P>

<H2>References</H2>

<H3>Phylo-HMM</H3>
<P>
Felsenstein J and Churchill GA (1996).  A hidden Markov model approach to
variation among sites in rate of evolution.
<em>Mol Biol Evol</em> 13:93-104. </P>
<P>
Siepel A and Haussler D (2003).  
<a href="http://www.cse.ucsc.edu/~acs/phyhmm.pdf">Combining
phylogenetic and hidden Markov models in biosequence analysis</a>.
In <em> Proceedings of the Seventh Annual 
International Conference on Computational Molecular Biology (RECOMB
2003)</em>, pp. 277-286. </P>
<P>
Siepel A and Haussler D.  Phylogenetic hidden Markov models.
Submitted book chapter. </P>
<P>
Yang Z (1994).  Maximum likelihood phylogenetic estimation from DNA sequences
with variable rates over sites: approximate methods,
<em>J Mol Evol</em> 39:306-314.</P>
<P>
Yang Z (1995).  A space-time process model for the evolution of DNA
sequences.  <em>Genetics</em>, 139:993-1005. </P>

<H3>Chain/Net</H3>
<P>
Kent WJ, Baertsch R, Hinrichs A, Miller W, and Haussler D (2003). 
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1" 
TARGET=_blank>Evolution's cauldron: 
Duplication, deletion, and rearrangement in the mouse and human genomes</A>. 
<I>Proc Natl Acad Sci USA</I> 100(20):11484-11489 Sep 30 2003.

<H3>Multiz</H3>
<P>
Blanchette M, Kent WJ, Riemer C, Elnitski L, Smit AFA, Roskin KM,
Baertsch R, Rosenbloom K, Clawson H, Green ED, Haussler D, Miller W.
Aligning Multiple Genomic Sequences with the Threaded Blockset Aligner.
<em>Submitted for publication</em>.</P>

<H3>Blastz</H3>
<P>
Chiaromonte F, Yap VB, and Miller W (2002). 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" 
TARGET=_blank>Scoring pairwise genomic sequence alignments</A>. 
<em>Pac Symp Biocomput</em> 2002;:115-26.</P>
<P>
Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison R, Haussler D, and 
Miller W. (2003). 
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103" 
TARGET=_blank>Human-Mouse Alignments with BLASTZ</A>. 
<em>Genome Research</em> 13(1):103-7.</P>
