<H2>Description</H2> <P>
This track shows a measure of evolutionary conservation in human, mouse, and
chicken that is based on a phylogenetic hidden Markov model (phylo-HMM).
The alignments used were the multiz alignments of the human July 2003 (hg16), mouse Feb. 2003 (mm3), rat Jun. 2003 (rn3), and chicken Feb. 2004 (galGal2) assemblies.</P>

<P>
In "full" visibility mode, this track displays pairwise alignments for each 
of mouse, rat, and chicken, aligned to human.  The pairwise alignments are
displayed in the standard UCSC browser "dense" mode -- as a greyscale 
density gradient.  The checkboxes in the track configuration allow
excluding species from the pairwise display, however this does not
remove them from the conservation score display.
</P>

<P>
When zoomed in to base level, the track shows
the multiple alignments at the base level.  With 30,0000 or fewer bases
in the window, clicking on the track display will display the
alignment for all sequence in the window.
</P>

<H2>Methods</H2>

<P> Best-in-genome blastz pairwise alignments of human-mouse and
human-rat were multiply aligned using a program called humor
(HUman-MOuse-Rat), which is a special variant of the <em>Multiz</em>
program.  Multiz was used to align the humor results with
best-in-genome blastz human-chicken alignments.  The resulting
human-mouse-rat-chicken multiple alignments were then assigned 
conservation scores by phylo-HMM.</P>

    <P>A phylo-HMM is a generative probabilistic model that describes both
    the process of DNA substitution along the branches of a phylogeny at
    each site in an alignment, and changes in the mode of this process from
    one site to the next (Felsenstein and Churchill [1996], Yang [1995],
    Siepel and Haussler [2003], Siepel and Haussler [submitted]).  Each of
    the states of a phylo-HMM may be associated with a different
    phylogenetic model, describing both the overall <em>rates</em> of
    substitution along the branches of the phylogeny, and the 
    <em>pattern</em> of substitution (relative rates for all pairs of
    nucleotides).</P>

    <P> In this case, the pattern of substitution is assumed to be the same
    for all states, but the rates are allowed to differ.  The model is
    similar to the one proposed by Felsenstein and Churchill (1996).  There
    are <em>k</em> states corresponding to <em>k</em> "rate categories,"
    each of which is defined by a scaling constant that is applied
      uniformly to the branches of the tree.  The transition
    probabilities between states allow for autocorrelation of substitution
    rates---i.e., for adjacent sites to tend to exhibit similar overall
    substitution rates.  A single parameter lambda describes the degree of
    autocorrelation and defines all transition probabilities.
    Here, we have estimated the rate constants from the data, similarly to
    Yang (1995) (see Siepel and Haussler [2003]), but have allowed
    lambda to be treated as a tuning parameter. </P>

    <P> A phylogenetic model was fitted by maximum likelihood to the
    human/mouse/rat alignments using the REV substitution model and the
    discrete gamma model for rate variation (Yang [1994]; <em>k</em>=10 rate
    categories were used).  All sites in the alignments were included in
    the analysis and alignment gaps were treated as missing data.  Branch
    lengths were estimated as follows, in units of expected substitutions
    per site:
      <pre>
      (human:0.193,(mouse:0.076,rat:0.083):0.193)
     </pre>
      (As shown, the tree is arbitrarily rooted at the midpoint of the
      branch between human and the mouse/rat ancestor.)  The shape
      parameter alpha for the discrete gamma model was estimated to have
      value 4.4. Next, a 10-state phylo-HMM was derived from the estimated
      phylogenetic model (with lambda = 0.9), and used to compute the
      posterior probability that each site was "generated" by the state
      having the smallest rate constant.  It is these posterior
      probabilities that are plotted for the track.  Because of the way the
      rate categories are defined, the plotted values can be thought of as
      representing the posterior probability that each site is among the
      10% most conserved sites in the data set, allowing for
      autocorrelation of substitution rates.  </P>

    <P>Currently, gaps are treated as missing data by the phylo-HMM as well
    as in the estimation of the phylogenetic model, which sometimes leads
    to undesirably high posterior probabilities in gappy regions of the
    alignment.  We are looking at several possible ways of improving the
    handling of alignment gaps.</P>



<H2>Credits</H2>
<P>
This track was created at UCSC, using the blastz and multiz programs from 
Scott Schartz and Webb Miller of the 
<A HREF="http://bio.cse.psu.edu/" TARGET=_blank>
 Penn State Bioinformatics Group</A>; 
the axtBest, axtChain, chainNet, netSyntenic, and netClass programs 
developed by Jim Kent at UCSC; 
and the phylo-HMM package from Adam Siepel at UCSC.
</P>

<H2>References</H2>

<H3>Phylo-HMM</H3>
<P> J. Felsenstein and G. A. Churchill.  A hidden Markov model approach to
      variation among sites in rate of evolution.
      <em>Mol. Biol. Evol.</em> 13:93-104, 1996. </P>

<P> A. Siepel and D. Haussler.  <a
        href="http://www.cse.ucsc.edu/~acs/phyhmm.pdf">Combining
        phylogenetic and hidden Markov models in biosequence analysis</a>.
      In <em> Proceedings of the Seventh Annual 
        International Conference on Computational Molecular Biology (RECOMB
        2003)</em>, pages 277-286, 2003. </P>

<P> A. Siepel and D. Haussler.  Phylogenetic hidden Markov models.
      Submitted book chapter. </P>

<P> Z. Yang.  A space-time process model for the evolution of DNA
      sequences.  <em>Genetics</em>, 139:993-1005, 1995. </P>

<P> Z. Yang.  Maximum likelihood phylogenetic estimation from DNA sequences
    with variable rates over sites: approximate methods,
      <em>J. Mol. Evol.</em> 39:306-314, 1994.</P>

<H3>Chain/Net</H3>
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1" TARGET=_blank>Evolution's cauldron: 
Duplication, deletion, and rearrangement in the mouse and human genomes</A>. 
Kent, W.J., Baertsch, R., Hinrichs, A., Miller, W. and Haussler, D. (2003). 
<I>P Natl Acad Sci USA</I> 100(20):11484-11489 Sep 30 2003.

<H3>Multiz</H3>
<P> Aligning Multiple Genomic Sequences with the Threaded Blockset Aligner.
Blanchette M, Kent WJ, Riemer C, Elnitski L, Smit AFA, Roskin KM,
Baertsch R, Rosenbloom K, Clawson H, Green ED, Haussler D, Miller W.
Submitted for publication.</P>

<H3>Blastz</H3>
<P><A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>Human-Mouse Alignments with BLASTZ</A>. Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison R, Haussler D, and Miller W.  <em>Genome Research</em> 2003 Jan;13(1):103-7.</P>

<P><A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" TARGET=_blank>Scoring pairwise genomic sequence alignments</A>. Chiaromonte F, Yap VB, Miller W. <em>Pac Symp Biocomput</em> 2002;:115-26.</P>

