<H2>Description</H2>
<P>
This track shows the "reciprocal best" $organism/$o_organism alignment net.
It is useful for finding orthologous regions and for studying genome
rearrangement.  
<P>
In the graphical display, the boxes represent ungapped 
alignments, while the lines represent gaps.  
In full display mode, the top-level (Level 1)
chains are the largest, highest-scoring chains that
span this region.  In many cases, gaps exist in the
top-level chains.  When possible, these are filled in by
other chains displayed at Level 2.  The gaps in 
Level 2 chains may be filled by Level 3 chains and so
forth.
Clicking on a box displays detailed information about the chain
as a whole, while clicking on a line shows information
on the gap.  The detailed information is useful in determining
the cause of the gap or, for lower-level chains, the genomic
rearrangement. 
<P>
Individual track features are categorized as one of four types
(other than gap):
<UL>
<LI><B>Top</B> - the best, longest match. Displayed on Level 1.
<LI><B>Syn</B> - aligns to the same chromosome as the gap in the level
    above it.
<LI><B>Inv</B> - aligns to the same chromosome as the gap above it, but in the
    opposite orientation.
<LI><B>NonSyn</B> - matches a different chromosome from the gap in the level 
above it.
</UL></P>
<H2>Methods</H2>
These alignments were generated using <A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" 
TARGET=_blank>blastz</A> and blat alignments of $o_organism
genomic sequence from the 13 Nov. 2003 Arachne $o_organism draft assembly.
The initial alignments were generated using blastz on 
repeatmasked sequence using the following $organism/$o_date scoring matrix:
<PRE>
     A    C    G    T
A   100 -300 -150 -300
C  -300  100 -300 -150
G  -150 -300  100 -300
T  -300 -150 -300  100

K = 4500, L = 3000,  Y=3400, H=2000
</PRE>
</P>
<P>
The resulting alignments were processed by the axtChain program.
AxtChain organizes all the alignments between a single $o_organism chromosome
and a single $organism chromosome into a group and makes a kd-tree out
of all the gapless subsections (blocks) of the alignments.
The maximally-scoring chains of these blocks were found by running a
dynamic program over the kd-tree.  Chains scoring below a certain
threshold were discarded.
<P>
To place additional $o_organism scaffolds that weren't initially aligned by blastz,
a DNA blat of the unmasked sequence was performed. The resulting
blat alignments were also chained, and then merged with the
blastz-based chains produced in the previous step to produce "all chains".
<P>
These chains were sorted such that the highest scoring chain in the
genome was first. The program chainNet was then used to place
each chain one at a time, trimming it as necessary to
fit into a section not already covered by 
a higher-scoring chain. During this process, a
natural hierarchy emerges in which chains that fill gaps
in a previous chain are considered underneath the
previous chain. The program netSyntenic was used to fill in
information about the relationship between upper-
and lower-level chains, including whether a lower-level
chain is syntenic with the higher-level chain,  whether
it is inverted with respect to the higher-level chain,
and so forth.
</P>
Due to the draft nature of this initial genome assembly,
this net track (and the companion chain track) was generated using
a "reciprocal best" strategy.  This strategy attempts to minimize
paralog fill-in for missing orthologous $o_organism sequence by filtering
from the $organism net all sequences not found in the $o_organism side of the
net.  
After generating the $organism alignment net, 
the subset of chains in the $o_organism reference net was extracted
and used for an additional netting step, which was then filtered
for non-syntenic sequences smaller than 50 bases.

<H2>Credits</H2>
The $o_organism sequence used in this track was obtained from the 13 Nov. 2003 
Arachne assembly. We'd like to thank the National Human Genome Research Institute 
(NHGRI), the Eli & Edythe L. Broad Institute at MIT/Harvard, and Washington University 
School of Medicine for providing this sequence.</P>
<P>
The chaining and netting programs were
developed at the University of California
at Santa Cruz by Jim Kent.
For more information, see 
Kent WJ, Baertsch R, Hinrichs A, Miller W, and Haussler D (2003). 
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1" 
TARGET=_blank>Evolution's cauldron: 
Duplication, deletion, and rearrangement in the mouse and human genomes</A>. 
<I>Proc Natl Acad Sci USA</I> 100(20):11484-11489 Sep 30 2003.
<P>
The browser display and database storage of the nets were made
by Robert Baertsch and Jim Kent.</P>

