<H2>Description</H2> <P>
This track shows a measure of evolutionary conservation in human, rat, and
mouse that is based on a phylogenetic hidden Markov model (phylo-HMM).
The alignments used were the multiz alignments of the mouse Feb. 2003 (mm3),
rat Jun. 2003 (rn3), and human Jul. 2003 (hg16) assemblies.</P>

<H2>Methods</H2>

<P>A phylo-HMM is a generative probabilistic model that describes both the
    process of DNA substitution along the branches of a phylogeny at any given
    site in an alignment, and the potential for the mode of this process
    to change from one site to the next.  Each of the states of a phylo-HMM
    may be associated with a different phylogenetic model, describing
    both a set of overall *rates* of substitution along the branches of the
    phylogeny, and a particular *pattern* of substitution (relative rates
    for all pairs of nucleotides)</P>

    <P> In this case, the pattern of substitution is assumed to be the same for all
      states, but the overall rates are different.
      Essentially the same model is 
      assumed as was described by Felsenstein and Churchill (1996).  There are
      <em>k</em> states corresponding to <em>k</em> "rate categories," each
      of which is 
      defined by a single rate constant.  An autocorrelation parameter
      (lambda) defines the transition probabilities between states.  We have
      estimated the rate constants from the data, as was done by Yang (1995), but
      have allowed the autocorrelation parameter lambda to be treated as a
      tuning parameter, which we adjust to get the desired level of
      "smoothing" in our plots. </P>

(more to write) k=10 states, post. prob., REV model, gaps missing data, show tree


<H2>Credits</H2>
<P>
This track was created by Adam Siepel, with suggestions from
David Haussler and others.  The display technology ("wiggle track") was
developed by Hiram Clawson.<P>

<H2>References</H2>

<P> J. Felsenstein and G. A. Churchill.  A hidden Markov model approach to
      variation among sites in rate of evolution.
      <em>Mol. Biol. Evol.</em> 13:93-104, 1996. </P>

<P> A. Siepel and D. Haussler.  <a
        href="http://www.cse.ucsc.edu/~acs/phyhmm.pdf">Combining
        phylogenetic and hidden Markov models in biosequence analysis</a>.
      In <em> Proceedings of the Seventh Annual 
        International Conference on Computational Molecular Biology (RECOMB
        2003)</em>, pages 277-286, 2003. </P>

<P> A. Siepel and D. Haussler.  Phylogenetic hidden Markov models.
      Submitted book chapter. </P>

<P> Z. Yang.  A space-time process model for the evolution of DNA
      sequences.  <em>Genetics</em>, 139:993-1005, 1995. </P>

