<H2>Description</H2>

<p>
This track displays the 3-way Regulatory Potential Score, computed 
from alignments of human (hm16, Oct 03), mouse (mm3, Feb 03) and rat
(rn3, Jun 03).
</p><p>
RP scores compare frequencies of short alignment patterns between 
regulatory elements and neutral DNA. 
</p><p>
The default display range for the track goes from 0 to 0.1. Score values 
at or below 0 indicate resemblance with alignment patterns typical of 
neutral DNA, and are truncated and displayed at 0. Score values at or above 
0.1 indicate very marked resemblance with alignment patterns typical of 
regulatory elemenents, and are truncated and displayed at 0.1. Absence of a 
score value at a given location indicates lack of 3-way alignment.
</p><p>
Preliminary results from a calibration study investigating sensitivity 
and specificity of the 3-way RP score on the hemoglobin beta gene 
cluster suggest the use of a threshold ~ 0.0002 for identifying new 
putative regulatory elements.
</p><p>
(A 2-way RP score, computed from alignments of human and mouse only
is also available on this browser).
</p>

<H2>Methods</H2>
<p>
The comparison employs log-ratios of transitions probabilities from two 
Markov models. Training the score entails selecting appropriate alphabet
(alignment column symbols) and oder (length of the patterns=order+1) for 
the Markov models, and estimating their transition probabilities, based 
on alignment data from known regulatory elements and ancestral repeats.
The 3-way RP score uses a 10-symbol alphabet, and order 2.
</p><p>
In the track, score values are displayed using a system of overlapping 
windows of size 100bp along aligned portions of the human sequence 
(log-ratios are added over positions in a window, and the sum normalized 
for length). 
</p>

<H2>Credits</H2>
<p>
Work on RP scores is performed by members of the
<A HREF="http://www.bx.psu.edu/" TARGET=_blank>
Comparative Genomics and
Bioinformatics Center at Penn State University. </A>
More information on this research and the collection of known regulatory 
elements employed in training the score can be found at our site. 
</p><p>
Mouse and Rat sequence data were provided by the Mouse and Rat Sequencing 
Consortia, and alignment data created in collaboration with the UCSC Genome 
Bioinformatics group.
</p>

<H2>References</H2>
<p>
Kolbe D., Taylor J., Elnitski L., Eswara P., Li J., Miller W., Hardison R.C. 
and Chiaromonte F. (2003) Regulatory potential scores from genome-wide 3-way 
alignments of human, mouse and rat. Submitted for publication.
</p><p>
Blanchette M., Kent W.J., Riemer C., Elnitski L., Smit A.F.A., Roskin K.M., 
Baertsch R., Rosenbloom K., Clawson H., Green E.D., Haussler D. and 
Miller W. (2003) Aligning Multiple Genomic Sequences with the Threaded 
Blockset Aligner. Submitted for publication.
</p>
