<H2>Description</H2>
<P>
This track displays the 2-way regulatory potential (RP) score computed 
from alignments of human (hg16, Jul. '03) and mouse (mm4, Oct. '03).
RP scores compare frequencies of short alignment patterns between 
regulatory elements and neutral DNA. 
<P>
Score values 
at or below 0 indicate resemblance to alignment patterns typical of 
neutral DNA. Score values at or above 
0.1 indicate very marked resemblance to alignment patterns typical of 
regulatory elements. Absence of a 
score value at a given location indicates lack of alignment.
<P>
Preliminary results from a calibration study investigating sensitivity 
and specificity of the 2-way RP score on the hemoglobin beta gene 
cluster suggest the use of a threshold just above 0 for identifying new 
putative regulatory elements.
<P>
(A 3-way RP score, computed from alignments of human, mouse and rat
is also available on this browser).
</P>

<H2>Methods</H2>
<P>
The comparison employs log-ratios of transitions probabilities from two 
Markov models. Training the score entails selecting appropriate alphabet
(alignment column symbols) and order (length of the patterns = order + 1) for 
the Markov models, and estimating their transition probabilities, based on 
alignment data from known regulatory elements and ancestral repeats.
The 2-way RP score uses a 5-symbol alphabet and order 5.
<P>
In the track, score values are displayed using a system of overlapping 
windows of size 100 bp along aligned portions of the human sequence 
Log-ratios are added over positions in a window, and the sum normalized 
for length. 
</P>

<H2>Using the Filter</H2>
<P>The track filter can be used to configure display attributes of
the track's graph. 
<UL>
<LI><B>Type of graph</B>: The default &quot;Bar&quot; setting depicts the graph
data using color-filled bars. To view the data as a series of points or lines, 
select the &quot;Points&quot; setting. 
<LI><B>y = 0.0 line</B>: Select the &quot;ON&quot; setting to display a
line marking the 0.0 position on the graph (default is &quot;OFF&quot;).
<LI><B>Track height</B>: To change the default display height of the graph
in pixels, type in a value from within the indicated range.
<LI><B>Vertical viewing range</B>: The min and max values specify the 
vertical portion of the graph that is displayed (default
range is 0-0.01). These numbers can be used to set data threshold indicators. 
For example, setting the min to &quot;0.005&quot; restricts the display
to data equal to or greater than that value. 
<LI><B>Data view scaling</B>: The default &quot;Use vertical viewing range
setting&quot; option displays the data using the parameters specified in the Vertical
viewing range setting. To configure the graph to automatically scale to a
range defined by the minimum and maximum data points in the current 
view, select the &quot;Auto-scale to data view&quot; option. 
<LI><B>Windowing function</B>: When a  view is too large to show individual
data values, the values must be combined to produce a plot point.
This option specifies the combining function to be used: &quot;Maximim&quot; - 
the maximum value of all the points to be combined, &quot;Mean&quot; - 
the mean, or &quot;Minimum&quot; - the minimum of all the
points to be combined (default is &quot;Mean&quot;).
<LI><B>Smoothing window</B>: When set to a numerical value, this option 
determines the size, in pixels, of a smoothing window
to be passed over the plot to smooth the edges of the bars or lines.  
This is equivalent to a trend line calculation on the graph.
The default setting is &quot;OFF&quot;.
<LI><B>Draw indicator line at y =</B>: When set to &quot;ON&quot;, the track
displays an arbitrary line on the graph at the specified numerical value 
(defaults are &quot;0&quot; and &quot;OFF&quot;). This line can be used to 
mark a significant threshold on the graph.
</UL></P>
<P>
When you have finished configuring the filter, click the <em>Submit</em> button.
</P>

<H2>Credits</H2>
<P>
Work on RP scores is performed by members of the
<A HREF="http://www.bx.psu.edu/" TARGET=_blank>
Comparative Genomics and
Bioinformatics Center at Penn State University</A>.
More information on this research and the collection of known regulatory 
elements used in training the score can be found at this site. 
<P>
Mouse sequence data were provided by the Mouse Sequencing Consortium. The 
alignment data were created in collaboration with the UCSC Genome Bioinformatics 
group.
</P>

<H2>References</H2>
<P>
Elnitski L., Hardison R., Li J., Yang S., Kolbe D., Eswara P., O'Connor M., 
Schwartz S., Miller W. and Chiaromonte F. (2003) 
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/64" TARGET=_blank>Distinguishing 
regulatory DNA from neutral sites</A>. <em>Genome Research</em>, 13(1):64-72.
</P>
Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison R, Haussler D, and 
Miller W. (2003) 
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103" TARGET=_blank>Human-Mouse 
Alignments with BLASTZ</A>. <em>Genome Research</em>, 13(1):103-7.</P>

