<H2>Description</H2>
<P>
This track displays the 2-way regulatory potential (RP) score computed 
from alignments of human (hg16, Jul. '03) and mouse (mm4, Oct. '03).
RP scores compare frequencies of short alignment patterns between 
regulatory elements and neutral DNA. 
<P>
Score values 
at or below 0 indicate resemblance to alignment patterns typical of 
neutral DNA. Score values at or above 
0.1 indicate very marked resemblance to alignment patterns typical of 
regulatory elements. Absence of a 
score value at a given location indicates lack of alignment.
<P>
Preliminary results from a calibration study investigating sensitivity 
and specificity of the 2-way RP score on the hemoglobin beta gene 
cluster suggest the use of a threshold just above 0 for identifying new 
putative regulatory elements.
<P>
(A 3-way RP score computed from alignments of human, mouse and rat
is also available on this browser).
</P>
<P>
This track may be configured in a variety of ways to highlight different aspects 
of the displayed information. Click the &quot;Graph configuration help&quot; link
for an explanation of the configuration options.

<H2>Methods</H2>
<P>
The comparison employs log-ratios of transitions probabilities from two 
Markov models. Training the score entails selecting appropriate alphabet
(alignment column symbols) and order (length of the patterns = order + 1) for 
the Markov models, and estimating their transition probabilities, based on 
alignment data from known regulatory elements and ancestral repeats.
The 2-way RP score uses a 5-symbol alphabet and order 5.
<P>
In the track, score values are displayed using a system of overlapping 
windows of size 100 bp along aligned portions of the human sequence 
Log-ratios are added over positions in a window, and the sum normalized 
for length. 
</P>

<H2>Credits</H2>
<P>
Work on RP scores is performed by members of the
<A HREF="http://www.bx.psu.edu/" TARGET=_blank>
Comparative Genomics and
Bioinformatics Center at Penn State University</A>.
More information on this research and the collection of known regulatory 
elements used in training the score can be found at this site. 
<P>
Mouse sequence data were provided by the Mouse Sequencing Consortium. The 
alignment data were created in collaboration with the UCSC Genome Bioinformatics 
group.
</P>

<H2>References</H2>
<P>
Elnitski L, Hardison R, Li J, Yang S, Kolbe D, Eswara P, O'Connor M, 
Schwartz S, Miller W and Chiaromonte F. (2003). 
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/64" TARGET=_blank>Distinguishing regulatory DNA from neutral sites</A>. <em>Genome Res.</em> 13(1):64-72.
</P>
Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison R, Haussler D, and 
Miller W. (2003). 
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103" TARGET=_blank>Human-Mouse 
Alignments with BLASTZ</A>. <em>Genome Res.</em> 13(1):103-7.</P>
