<H2>Description</H2>
<P>
This track displays the 3-way regulatory potential (RP) score, computed 
from alignments of human (hg16, Jul. '03), mouse (mm3, Feb. '03) and rat 
(rn3, Jun. '03). RP scores compare frequencies of short alignment patterns 
between regulatory elements and neutral DNA. 
<P>
Score values 
at or below 0 indicate resemblance to alignment patterns typical of 
neutral DNA. Score values at or above 
0.1 indicate very marked resemblance to alignment patterns typical of 
regulatory elements. Absence of a 
score value at a given location indicates lack of 3-way alignment.
<P>
Preliminary results from a calibration study investigating sensitivity 
and specificity of the 3-way RP score on the hemoglobin beta gene 
cluster suggest the use of a threshold ~ 0.0002 for identifying new 
putative regulatory elements.
<P>
(A 2-way RP score computed from alignments of human and mouse only 
is also available on this browser).
<P>
This track may be configured in a variety of ways to highlight different aspects 
of the displayed information. Click the &quot;Graph configuration help&quot; link
for an explanation of the configuration options.

<H2>Methods</H2>
<P>
The comparison employs log-ratios of transitions probabilities from two 
Markov models. Training the score entails selecting appropriate alphabet
(alignment column symbols) and order (length of the patterns = order + 1) for 
the Markov models, and estimating their transition probabilities, based on 
alignment data from known regulatory elements and ancestral repeats.
The 3-way RP score uses a 10-symbol alphabet and order 2.
<P>
In the track, score values are displayed using a system of overlapping 
windows of size 100 bp along aligned portions of the human sequence. 
Log-ratios are added over positions in a window, and the sum is normalized 
for length. 
</P>

<H2>Credits</H2>
<P>
Work on RP scores is performed by members of the
<A HREF="http://www.bx.psu.edu/" TARGET=_blank>
Comparative Genomics and
Bioinformatics Center at Penn State University</A>.
More information on this research and the collection of known regulatory 
elements used in training the score can be found at this site. 
<P>
Mouse and rat sequence data were provided by the Mouse and Rat Sequencing 
Consortia. The alignment data were created in collaboration with the UCSC 
Genome Bioinformatics group.
</P>

<H2>References</H2>
<P>
<P>
Blanchette M, Kent WJ, Riemer C, Elnitski L, Smit AFA, Roskin KM, Baertsch R,
Rosenbloom K, Clawson H, Green ED, Haussler D, Miller W. (2004). 
<A HREF="http://www.genome.org/cgi/content/abstract/14/4/708" TARGET=_blank>Aligning Multiple Genomic Sequences with the Threaded Blockset Aligner</A>.
<em>Genome Res.</em> 14(4):708-15.
<P>
Kolbe D, Taylor J, Elnitski L, Eswara P, Li J, Miller W, Hardison RC 
and Chiaromonte F. (2004). 
<A HREF="http://www.genome.org/cgi/content/full/14/4/700" TARGET=_blank>Regulatory potential scores from genome-wide three-way alignments of human, mouse, and rat</A>.
<em>Genome Res.</em> 14(4):700-707. 
</P>
