<H2>Description</H2>
<P>
This track displays the 3-way regulatory potential (RP) score, computed 
from alignments of human (hg16, Jul. '03), mouse (mm3, Feb. '03) and rat 
(rn3, Jun. '03). RP scores compare frequencies of short alignment patterns 
between regulatory elements and neutral DNA. 
<P>
The default display range for the track is 0 to 0.1. Score values 
at or below 0 indicate resemblance to alignment patterns typical of 
neutral DNA. Score values at or above 
0.1 indicate very marked resemblance to alignment patterns typical of 
regulatory elemenents. Absence of a 
score value at a given location indicates lack of 3-way alignment.
<P>
Preliminary results from a calibration study investigating sensitivity 
and specificity of the 3-way RP score on the hemoglobin beta gene 
cluster suggest the use of a threshold ~ 0.0002 for identifying new 
putative regulatory elements.
<P>
(A 2-way RP score computed from alignments of human and mouse only 
is also available on this browser).

<H2>Methods</H2>
<P>
The comparison employs log-ratios of transitions probabilities from two 
Markov models. Training the score entails selecting appropriate alphabet
(alignment column symbols) and order (length of the patterns=order+1) for 
the Markov models, and estimating their transition probabilities, based on 
alignment data from known regulatory elements and ancestral repeats.
The 3-way RP score uses a 10-symbol alphabet and order 2.
<P>
In the track, score values are displayed using a system of overlapping 
windows of size 100bp along aligned portions of the human sequence. 
Log-ratios are added over positions in a window, and the sum is normalized 
for length. 
</P>

<H2>Using the Filter</H2>
<P>The track filter can be used to configure display attributes of the track's graph. 
<UL>
<LI><B>Type of graph</B>: The default &quot;Bar&quot; setting depicts the graph data using filled-in 
bars. To switch to a view in which the data are represented by points or lines, select the 
&quot;Points&quot; setting. 
<LI><B>Show y=-0.0 line</B>: Select the &quot;ON&quot; setting to display a line marking the 0.0
position on the graph (default is &quot;OFF&quot;).
<LI><B>Track height</B>: To change the default display height of the graph in pixels, type in a different
value between 11 and 128 (the default display height is 128 pixels).
<LI><B>Vertical viewing range</B>: By default, the viewing range is set to the values of the minimum and 
maximum data points viewable at this position. Type in a different value to reduce the range of data
values displayed. 
<LI><B>Data view scaling</B>: The default &quot;Use Vertical Viewing Range Setting" displays the
data using the parameters specified in the Vertical viewing range setting. To configure the graph 
to automatically scale to a range defined by the minimum and maximum data points in the current 
view, select the &quot;Auto-Scale to data view&quot; option. 
</UL></P>
<P>
When you have finished configuring the filter, click the <em>Submit</em> button.
</P>

<H2>Credits</H2>
<P>
Work on RP scores is performed by members of the
<A HREF="http://www.bx.psu.edu/" TARGET=_blank>
Comparative Genomics and
Bioinformatics Center at Penn State University</A>.
More information on this research and the collection of known regulatory 
elements used in training the score can be found at this site. 
<P>
Mouse and rat sequence data were provided by the Mouse and Rat Sequencing 
Consortia. The alignment data were created in collaboration with the UCSC 
Genome Bioinformatics group.
</P>

<H2>References</H2>
<P>
Kolbe D., Taylor J., Elnitski L., Eswara P., Li J., Miller W., Hardison R.C. 
and Chiaromonte F. (2003) Regulatory potential scores from genome-wide 3-way 
alignments of human, mouse and rat. <em>Submitted for publication</em>.
</P>
<P>
Blanchette M., Kent W.J., Riemer C., Elnitski L., Smit A.F.A., Roskin K.M., 
Baertsch R., Rosenbloom K., Clawson H., Green E.D., Haussler D. and 
Miller W. (2003) Aligning Multiple Genomic Sequences with the Threaded 
Blockset Aligner. <em>Submitted for publication</em>.
</P>
