<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=273>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Callithrix%20jacchus"
    TARGET=_blank>
    <IMG SRC="/images/Callithrix_jacchus.jpg" WIDTH=258 HEIGHT=200 ALT="Marmoset"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Callithrix jacchus</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of NHGRI
      (<A HREF="http://www.genome.gov/pressDisplay.cfm?photoID=82" 
      TARGET=_blank>press photos</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The Jun. 2007 <em>Callithrix jacchus</em> draft assembly 
(WUSTL 2.0.2) was produced by the
    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Callithrix%20jacchus"
    TARGET=_blank>WUSTL Genome sequencing center</A>.

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
Marmoset genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>Contig12371</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all of contig 12371</TD>
    </TR>
    <TR><TD VALIGN=Top>Contig3759:130,210-140,209</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays ten thousand bases of contig 3759</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Contig2815:171328+1500</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of Contig 2815 that spans 1500 bases, starting with position 171328 </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>EF215092</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession
		number EF215092</TD>
    </TR>
    <TR><TD VALIGN=Top>EH380513</TD>
        <TD WIDTH=14>&nbsp;</TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession
		EH380513</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>mRNA helicase</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>lists mRNA helicase</TD>
    </TR>
    <TR><TD VALIGN=Top>homeobox caudal</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top>zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top>misra</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Misra</TD>
    </TR>
    <TR><TD VALIGN=Top>Taguchi,T.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author T. Taguchi</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Taguchi T</em>, internally it
	uses the format <em>Taguchi,T.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
This is a 6X coverage assembly.  There are 49,724 supercontigs with a total of
3,029,401,840 bases.  The largest contig is Contig1, with a size of 2,551,648 bases.
Half of the sequence is contained in the 2,807 largest contigs of size
303,219 bases (N50) or larger.  The average contig size is 60,924 bases,
the median contig size is 8,789 bases.  There are 154,810 gaps of size
100 bases or more for a total of 98,876,036 bases.  The largest gap,
3,000 bases, occurs 1,043 times.  See also:
<A HREF="http://genome.wustl.edu/pub/organism/Primates/Callithrix_jacchus/assembly/Callithrix_jacchus-2.0.2/ASSEMBLY"
TARGET=_blank>WUSTL Callithrix jacchus</A> assembly details.
</P>

<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/calJac1/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#marmoset">Downloads</A> page. 

These data have
<A HREF="../goldenPath/credits.html#marmoset_use">specific 
conditions for use</A>. 
</P>
<P>
It is requested that users of this marmoset sequence assembly acknowledge
the Washington University School of Medicine, Genome Sequencing Center
in any publications that result from use of this sequence assembly.
</P>
<P>
The Marmoset browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#marmoset_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
