<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=185>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://www.genome.gov/pressDisplay.cfm?photoID=42" TARGET=_blank>
    <IMG SRC="/images/Felis_catus.jpg" WIDTH=170 HEIGHT=143 ALT="F. catus"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Felis catus</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
    (NHGRI <A HREF="http://www.genome.gov/pressDisplay.cfm?photoID=42" 
	        TARGET=_blank>Press Photos</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The March 2006 <em>Felis catus</em> draft assembly 
(Broad Release 3) was produced by the <A HREF="http://www.broad.mit.edu/"
TARGET=_blank>Broad Institute</A> of MIT/Harvard and <A
HREF="http://www.agencourt.com/" TARGET=_blank>Agencourt Bioscience</A>. 

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
<EM>F. catus</EM> genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>scaffold_217569</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all of scaffold 217569</TD>
    </TR>
    <TR><TD VALIGN=Top>scaffold_217569:1-1000000</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first million bases of scaffold 217569</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>scaffold_217569:1000000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of scaffold 217569 that spans 2000 bases, starting with position 1000000 </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>AB030650</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession
		number AB030650</TD>
    </TR>
    <TR><TD VALIGN=Top>DN828316</TD>
        <TD WIDTH=14>&nbsp;</TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession
		DN828316</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>allergen</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists allergen-related genes and mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>pseudogene mRNA</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes, but not cDNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>homeobox caudal</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top>zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top>benson</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Benson</TD>
    </TR>
    <TR><TD VALIGN=Top>Ohno,K.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author K. Ohno</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Ohno K</em>, internally it
	uses the format <em>Ohno,K.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
The genome has been sequenced to 2X coverage and consists of 217,790
scaffolds. The total contig length for this assembly is 1,582,533,823 bases,
spanning 3,967,163,286 bases (with 60.1% in gaps) in 149,283 supercontigs.
</P>

<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/felCat3/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#f_catus">Downloads</A> page. 

These data have
<A HREF="../goldenPath/credits.html#f_catus_use">specific 
conditions for use</A>. 

The <EM>F. catus</EM> browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#f_catus_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
