<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
</head>
<body>
The <span style="font-style: italic;">Aeropyrum pernix</span>
genome sequence is 1,669,695 bp long and contains approximately 2,694
ORFs.&nbsp; <span style="font-style: italic;">A.
pernix</span> is an aerobic hyperthermophilic archaeon that grows optimally at around 95C.&nbsp; The sequence
was completed in Apr 1999 by the National Institute of Technology and Evaluation in Tokyo, and was published
in <a
 href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=10382966">Complete genome sequence of 
an aerobic hyper-thermophilic crenarchaeon, Aeropyrum pernix K1</a>.&nbsp; Kawarabayasi Y, Hino Y, Horikawa H, Yamazaki S, Haikawa Y, Jin-no 
K, Takahashi M, Sekine M, Baba S, Ankai A, Kosugi H, Hosoyama A, Fukui S, Nagai Y, Nishijima K, Nakazawa H, Takamiya M, Masuda S, Funahashi T, 
Tanaka T, Kudoh Y, Yamazaki J, Kushida N, Oguchi A, Kikuchi H, et al.&nbsp; DNA Res. 1999, 6(2):83-101, 145-52.
<h3>Sample position queries</h3>
<p>
A genome position can be specified by chromosomal coordinate range, COG
ID, or keywords from the GenBank or TIGR description of an gene. The
following list shows examples of valid position queries for this
genome.&nbsp;
</p>
<p>
<table cellspacing="0" cellpadding="0" border="0">
  <tbody>
    <tr>
      <td nowrap="nowrap" valign="top"><b>Request:</b><br>
      </td>
      <td colspan="2" valign="top"><b>&nbsp;&nbsp; Genome Browser
Response:</b><br>
      </td>
    </tr>
    <tr>
      <td valign="top"><br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">chr</td>
      <td width="14"><br>
      </td>
      <td valign="top">Displays all of chromosome (the entire genome).<br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">chr:1-10000</td>
      <td width="14"><br>
      </td>
      <td valign="top">Displays first ten thousand bases on the
chromosome.<br>
      </td>
    </tr>
<!-- 
    <TR><TD VALIGN=Top NOWRAP>ctgchr7_ctg</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of the clone contig ctgchr7_ctg 
                       (set "Map contigs" track to "dense" and refresh to see contigs)</TD>
    </TR> 
-->
    <tr>
      <td nowrap="nowrap" valign="top">NADH</td>
      <td width="14"><br>
      </td>
      <td valign="top">Lists genes with "NADH" in the description.<br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">APE0772</td>
      <td width="14"><br>
      </td>
      <td valign="top">Display chromosome at position of gene <a
 href="http://www.ncbi.nlm.nih.gov/entrez/viewer.fcgi?db=protein&val=5104435">APE0772</a>.<br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">APE077</td>
      <td width="14"><br>
      </td>
      <td valign="top">Lists genes with names beginning with "APE077".<br>
      </td>
    </tr>
    <tr>
    </tr>
  </tbody>
</table>
</p>
</p>
<P>
<HR>
<H3>Credits</H3>
The Archaeal Genome Browsers at UCSC were developed by members of the
<a href="http://lowelab.ucsc.edu/">Lowe Lab</a> (Kevin Schneider,
Katherine Pollard, Andy Pohl, Todd Lowe) with significant support from
the <a href="http://genome.ucsc.edu/staff.html">UCSC Human Genome
Browser group</a> (especially Robert Baertsch & Hiram Clawson).
The Archaeal Browsers are run by a slightly modified
version of the <a href="http://genome.ucsc.edu/">UCSC Human Genome
Browser</a> system.  All queries, bug reports, content corrections, suggested improvements,
and new track data submissions should be sent to Todd Lowe (lowe
@soe.ucsc.edu).

<P>If you use the browser for your research and wish to cite it, a
paper describing the work has been submitted to the 2006 <I>Nucleic
Acids Research</I>
Database Issue (KL Schneider, KS Pollard, A Pohl, TM Lowe,
submitted).  Citations will help us obtain funding to continue
development of this community resource.
</body>
</html>
