<H2>Description</H2>
<P>
This track shows the draft assembly of the <em>$Organism</em> genome.  
The Release 1.0 <em>D. simulans</em> assembly, produced by the 
<A HREF="http://genome.wustl.edu/" 
TARGET=_blank>Genome Sequencing Center</A> at the Washington University 
(WUSTL) School of Medicine in St. Louis, represents a mosaic of
several different <em>D. simulans</em> lines. The assembly process
began with an ~4X Whole Genome Shotgun assembly of the the D. simulans
w501 line.  The w501 contigs were initially anchored, ordered and
oriented by alignment with the D. melanogaster genome. The assembly
was then examined for places where the w501 assembly suggested
inversions with respect to the melanogaster assembly. One major
inversion was found, confirming the already documented inversion found
by 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=5729&dopt=Citation"
TARGET=_BLANK>Lemeunier and Ashburner (1976)</A>.  Six other
D. simulans lines (c167.4, md106ts, md199s, nc48s, sim4, and sim6)
were assembled with ~1X coverage.  Using the 4X WGS assembly of the
simulans w501 genome as a scaffold, contigs and unplaced reads from
the 1X assemblies of the other individual D. simulans lines were used
to cover gaps in the w501 assembly where possible.  Thus the resulting
assembly is a mosaic containing the w501 contigs as the primary
scaffolding with contigs and unplaced reads from the other lines
filling gaps in the w501 assembly.
</P>
<P>In dense mode, this track depicts the path through the clones 
(aka the golden path) used to create the assembled sequence. 
Clone boundaries are distinguished by the use of alternating gold and brown 
coloration. Where gaps
exist in the path, spaces are shown between the gold and brown
blocks.  If the relative order and orientation of the contigs
between the two blocks is known, a line is drawn to bridge the
blocks.</P>
<P>
All components within this track are of fragment type &quot;W&quot;: 
Whole Genome Shotgun contig. 

