<H2>Description</H2>
<P>
The UCSC Known Genes track shows known protein-coding genes based on
protein data from UniProt (SWISS-PROT and TrEMBL) and mRNA data from
the NCBI reference sequences collection (RefSeq) and GenBank. 
Each Known Gene is represented by an mRNA and a protein.
<H2>Display Conventions and Configuration</H2>
<P>
This track follows the display conventions for 
<A HREF="../goldenPath/help/hgTracksHelp.html#GeneDisplay">gene prediction 
tracks</A>. Black coloring indicates features that have corresponding entries 
in the Protein Databank (PDB). Blue indicates features associated with
mRNAs from NCBI RefSeq or (dark blue) items having associated proteins in 
the UniProt database. The variation in blue shading of RefSeq items
corresponds to the level of review the RefSeq record has undergone: 
predicted (light), provisional (medium), or reviewed (dark). </P>
<P>
This track contains an optional codon coloring 
feature that allows users to quickly validate and compare gene predictions.
To display codon colors, select the <em>genomic codons</em> option from the
<em>Color track by codons</em> pull-down menu. Click 
<A HREF="../goldenPath/help/hgCodonColoring.html">here</A> for more 
information about this feature. </P>

<H2>Methods</H2>
<P>This release of UCSC Known Genes was built by a new process, KG II, 
as described below.
<P>
UniProt protein sequences (including alternative splicing isoforms) 
and mRNA sequences from RefSeq and GenBank 
were aligned against the base genome using BLAT. 
RefSeq alignments having a base identity level within 0.1% of the best 
and at least 96% base identity with the genomic sequence were kept.  
GenBank mRNA alignments having a base identity level within 0.2% of 
the best and at least 97% base identity with the genomic sequence were kept.  
Protein alignments having a base identity level within 0.2% of the best and 
at least 80% base identity with the genomic sequence were kept.
<P>Then the genomic mRNA and protein alignments were compared, 
and protein-mRNA pairings were determined from their overlaps. 
mRNA CDS data were obtained from RefSeq and GenBank data 
and supplemented by CDS structures derived from UCSC protein-mRNA alignments using tblastn. 
The initial set of UCSC Known Genes candidates consists of 
all protein-mRNA pairs with valid mRNA CDS structures.  
A gene-check program (similar to the one used for the Consensus CDS (CCDS) project) 
is used to remove questionable candidates, such as those with in-frame stop codons, bad frame, 
missing start or stop codons, etc.
<P>From each group of gene candidates that share the same CDS structure, 
the protein-mRNA pair having the best ranking and protein-mRNA alignment score 
is selected as a UCSC Known Gene.  
The ranking of a gene candidate depends on its gene-check quality measures.  
When all else is equal, 
a preference is given to RefSeq mRNAs and next to MGC mRNAs.  
Similarly a preference is given to gene candidates represented by Swiss-Prot proteins.  
The protein-mRNA alignment score is calculated based on protein to 
mRNA alignment using TBLASTN, plus weighted sub-scores according 
to the date and length of the mRNA. 
<H2>Credits</H2>
<P>
The UCSC Known Genes track was produced using protein data from 
<A HREF="http://www.expasy.org/sprot/" TARGET=_blank>UniProt</A> and mRNA 
data from NCBI 
<A HREF="http://www.ncbi.nlm.nih.gov/RefSeq/" TARGET=_blank>RefSeq</A>
and <A HREF="http://www.ncbi.nlm.nih.gov/Genbank/index.html" 
TARGET=_blank>GenBank</A>.</P>

<H2>Data Use Restrictions</H2>
<P>
The UniProt entries in this annotation track are copyrighted. They
are produced through a collaboration between the Swiss Institute of
Bioinformatics and the EMBL Outstation - the European Bioinformatics
Institute. There are no restrictions on their use by non-profit
institutions as long as their content is in no way modified and this
statement is not removed. Usage by and for commercial entities requires
a license agreement (see <A HREF="http://www.isb-sib.ch/announce/"
TARGET="_blank">http://www.isb-sib.ch/announce/</A> or send an email to
<A HREF="mailto:license@isb-sib.ch">license@isb-sib.ch</A>).  </P>

<H2>References</H2>
<P>
Hsu, F., Kent, W.J., Clawson, H., Kuhn, R.M., Diekhans, M., and Haussler, D. 
<A HREF="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btl048?ijkey=qbTw0578H4cXuFw&keytype=ref" 
TARGET=_blank>The UCSC Known Genes</A>. 
<em>Bioinformatics</em> <B>22</B>(9), 1036-46 (2006).
<P>
Benson, D.A., Karsch-Mizrachi, I., Lipman, D.J., Ostell, J., and 
Wheeler, D.L. 
<A HREF="http://nar.oupjournals.org/cgi/content/full/32/suppl_1/D23" 
TARGET=_blank>GenBank: update</A>. <em>Nucleic Acids Res.</em> <B>32</B>,
D23-6 (2004).</P>
<P>
Kent, W.J.
<A HREF="http://www.genome.org/cgi/content/abstract/12/4/656" 
TARGET=_blank>BLAT - the BLAST-like alignment tool</A>.
<em>Genome Res.</em> <B>12</B>(4), 656-664 (2002).</P>

