<H2>Description</H2>
<P>
MicroRNA target sites in 3' UTRs as predicted by PicTar, based on the
RefSeq annotation of 3' UTRs.
</P>

<H2>Methods</H2>
<P>
The original PicTar algorithm was published in Krek <EM>et al.</EM>,
Nature Genetics, 2005. Displayed are predictions based on mouse as the
reference species using the improved PicTar method as published in
Lall <EM>et al.</EM>, Current Biology 2006.  Briefly, PicTar is a Hidden
Markov Model
that assigns probabilities to 3' UTR subsequences to be a binding site for a
microRNA, considers all possible ways the 3' UTR could be bound by microRNAs,
and then uses a maximum likelihood method to compute the optimal likelihood
under which the 3' UTR could be explained by microRNAs and background.
The score is this likelihood divided by background, <EM>i.e.</EM>, the local
base composition of each 3' UTR is taken into account. To fit the score
track conventions of the UCSC browser (integers), we scaled all scores by
the maximum score of all microRNA 3'-UTR scores observed. Note that the PicTar
algorithm
scores any 3' UTR that has at least one aligned conserved predicted binding
site for a microRNA, but then incorporates all possible binding sites into
the score, even if they appear to be non-conserved. Because the score for
a 3' UTR is a "phylo" average over all orthologous 3' UTRs used, "scattered"
sites that appear in many species may boost the score, and individual sites
shown in the display may not be aligned and conserved in all species under
consideration.
</P>

<P>
Two levels of conservation can be chosen: conservation among 
mouse/rat/rabbit/human/chimp/macaque/dog, and conservation among 
mouse/rat/rabbit/human/chimp/macaque/dog/cow/armadillo/elephant/tenrec/opossum/c
hicken. The latter settings have improved quality, but lower sensitivity.
</P>

<P>
To produce the phylogenetic scoring as describe in Krek et al., Nature 
Genetics, 2005, the following phylogenetic groupings where used:
<TABLE BORDER=0>
<TR><TD ALIGN=LEFT>(i)</TD><TD ALIGN=LEFT>mouse/rat/rabbit</TD></TR>
<TR><TD ALIGN=LEFT>(ii)</TD><TD ALIGN=LEFT>human/chimp/macaque</TD></TR>
<TR><TD ALIGN=LEFT>(iii)</TD><TD ALIGN=LEFT>dog/cow</TD></TR>
<TR><TD ALIGN=LEFT>(iv)</TD><TD ALIGN=LEFT>armadillo</TD></TR>
<TR><TD ALIGN=LEFT>(v)</TD><TD ALIGN=LEFT>elephant/tenrec</TD></TR>
<TR><TD ALIGN=LEFT>(vi)</TD><TD ALIGN=LEFT>opossum</TD></TR>
<TR><TD ALIGN=LEFT>(vii)</TD><TD ALIGN=LEFT>chicken</TD></TR>
<TR><TD ALIGN=LEFT>(viii)</TD><TD ALIGN=LEFT>xenopus</TD></TR>
<TR><TD ALIGN=LEFT>(ix)</TD><TD ALIGN=LEFT>tetraodon/fugu/zebrafish</TD></TR>
</TABLE>
</P>
<P>
mammalian score =( 1/3 *((i) + (ii) + (iii)) + (iv) + (v) + (vi) )<BR>
final score =1/2*(1/2*( 1/2*(mammalian score + (vii)) + (viii)) + (ix))
</P>

<H2>References</H2>
<P>
Krek, A., Grün, D., Poy, M.N., Wolf, R,, Rosenberg, L., Epstein E.J., 
MacMenamin, P., da Piedade, I., Gunsalus, K.C., Stoffel, M., and Rajewsky, N.,
<A HREF="http://www.nature.com/ng/journal/v37/n5/abs/ng1536.html" TARGET=_blank>
Combinatorial microRNA target predictions</A>,
<EM>Nature Genetics</EM>, <B>37</B>,495-500 (2005). 
</P>
<P>
Lall, S., Grün, D., Krek, A., Chen, K., Wang, Y-L.,  Dewey, C.N., Sood, P.,
Colombo, T., Bray, N., MacMenamin, P., Kao, H-L., Gunsalus, K.C., Pachter, L.,
Piano, F. and Rajewsky, N.,
<A HREF="http://www.current-biology.com/content/article/abstract?uid=PIIS0960982206010591" TARGET=_blank>
A genome-wide map of conserved microRNA targets in <EM>C. elegans</EM></A>,
<EM>Current Biology</EM>, <B>16</B>,460-471 (2006).
</P>

<H2>Credits</H2>
<P>
Thanks to the Dominic Grün, Yi-Lu Wang, and Nikolaus Rajewsky for providing
this annotation.  More detailed information about individual
predictions, including links to other databases, can be found at the
<A HREF="http://pictar.bio.nyu.edu/" TARGET=_blank>PicTar Website,</A>
a project of the
<A HREF="http://www.nyu.edu/fas/dept/biology/faculty/rajewsky/index.html" TARGET=_blank>Rajewsky lab</A> at
<A HREF="http://www.nyu.edu/fas/dept/biology/" TARGET=_blank>
NYU's Center for Comparative Functional Genomics.</A>
</P>
