<H2>Description</H2>
<P>
This track shows alignments between $Organism Expressed
Sequence Tags (ESTs) in 
<A HREF="http://www.ncbi.nlm.nih.gov/Genbank/index.html" 
TARGET=_blank>GenBank</A> and the genome.</P> 

<P>Expressed sequence tags are single-read (typically
approximately 500 base) sequences which usually
represent fragments of transcribed genes.  Aligning 
regions (usually exons) are shown as black boxes 
connected by lines for gaps (usually spliced out introns). 
In full display mode, arrows on the introns
indicate the direction of transcription, which is
determined by looking at the splice sites. 
</P>

<P>Strand information provided for ESTs (+/-) indicates the
direction of the match between the EST and the matching
genomic sequence.  It bears no relationship to the direction
of transcription of the RNA with which it might be associated.

<H2>Method</H2>
<P>To make an EST, RNA is isolated from cells and reverse
transcribed into cDNA.  Typically, the cDNA is cloned
into a plasmid vector, and a read taken from the 5'
and/or 3' primer.  For most - but not all - ESTs, the
reverse transcription is primed by an oligo-dT, which
hybridizes with the poly-A tail of mature mRNA. The
reverse transcriptase may or may not make it to the 5'
end of the mRNA,  which may or may not be degraded.</P>

<P>In general, the 3' ESTs mark the end of transcription
reasonably well, but the 5' ESTs may end at any point
within the transcript.  Some of the newer cap-selected
libraries are starting to hit transcription start
reasonably well. Before the cap-selection techniques
emerged, some projects used random rather than poly-A
priming in an attempt to get sequence distant from the
3' end.  These projects were successful at this, but as
a side effect also deposited sequences from unprocessed
mRNA and perhaps even genomic sequences into the EST databases.
(Even outside of the random-primed projects, there is a
degree of non-mRNA contamination.) Because of this, a
single unspliced EST should be viewed with considerable
skepticism.  </P>

<P>To generate this track, $Organism ESTs from GenBank are aligned 
against the genome using the 
<A HREF="http://www.genome.org/cgi/content/abstract/12/4/656" TARGET=_blank>blat</A> 
program. Note that the maximum intron length
allowed by blat is 500,000 bases, which may eliminate some ESTs with very 
long introns that might otherwise align. When a single 
EST aligns in multiple places, the alignment having the 
highest base identity is found.  Only alignments that have 
a base identity level within 1% of the best are kept. 
Alignments must also have at least 93% base identity to be kept.</P>

<H2>Using the Filter</H2>
<P>The track filter can be used to change the color or include/exclude a subset of 
individual items within a track. This is helpful when many items are shown in the 
track display, especially when only some are relevant to the current task. To use the
filter:
<OL>
<LI>Enter a value in one or more of the text boxes to filter the EST display. For
example, to apply the filter to all ESTs expressed in the liver, type &quot;liver&quot; in the 
tissue box. For a list of permissible filter values, consult the non-positional table in
the Table Browser that corresponds to the factor on which you wish to filter. For
example, the non-positional table &quot;tissue&quot; contains all of the types of tissues
that can be entered into the tissue text box. Wildcards can also be used in the
filter.
<LI>If filtering on more than one value, choose the desired combination
logic. If &quot;and&quot; is selected, only ESTs that match all of the filter criteria will
be highlighted. If &quot;or&quot; is selected, ESTs that match any one of the filter criteria
will be highlighted.
<LI>Choose the color or display characteristic that should be used to highlight or
include/exclude the filtered items. If &quot;exclude&quot; is chosen, the browser will not 
display ESTs that match the filter criteria. If &quot;include&quot; is selected, the browser 
will display only those ESTs that match the filter criteria.
</OL></P>
<P>
When you have finished configuring the filter, click the <em>Submit</em> button.</P>

<H2>Credits</H2>
<P>
The $Organism EST track is produced at UCSC from EST sequence data
submitted to the international public sequence databases by 
scientists worldwide.

<H2>References</H2>
<P>

Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Wheeler DL. (2004)
<A HREF="http://nar.oupjournals.org/cgi/content/full/32/suppl_1/D23" TARGET=_blank>GenBank: update</A>. <em>Nucleic Acids Res.</em> 32 Database issue:D23-6.


