<H2>Description</H2>
<P>
The Protein Multiple Sequence Alignment (MSA) track shows MSA result of HIV-1 gp120 protein sequences.
The protein sequences shown are translated from the corresponding sequences in the DNA Multiple Sequence Alignment (MSA) track.
<P>
Only the sequences of the subjects selected by the user are shown.  If a user has not selected specific subjects, a default set (first 50 subjects in the database) will be shown.
</P>
<H2>Methods</H2>
<P>
HIV-1 gp120 nucleotide sequences were translated into amino acids using AlignmentHelper1.2 (http://inbio.byu.edu/faculty/dam83/cdm) and the resulting peptide sequences were then aligned in MAFFT v5.7 (Katoh et al. 2005). 
<P>
We applied two local (L-INS-i and E-INS-i) and one global (G-INS-i) iterative refinement algorithms. MAFFT also incorporates a BLOSUM62 scoring matrix (an empirical matrix calculated from large datasets representing extreme protein family diversity) and assumes a fixed frequency for each AA for building MSAs; however, the estimated relative rates of changes in this matrix may be too general to fit datasets of specific gene families or high substitution rates, such as gp120. Hence, here we performed multiple MAFFT alignments using both the BLOSUM62 and gp120-specific scorings matrixes and/or fixed and variable AA frequencies estimated for the data at hand. Gene-specific empirical matrices and AA frequencies based on the VAX004 and LAGB-VAX004 datasets were generated via the program MATRIXGEN (http://matrixgen.sourceforge.net). 
<P>The quality of the MSA was then compared using the maximum likelihood (ML) tree building approach in PhyML (Guindon and Gascuel 2003) under the GTR+G+I model. ML trees showing the highest likelihood scores were indicative of better alignments. G-INS-i + BLOSUM62 + AA specific frequencies generated the trees with the best likelihood scores. 
<P>Uncertainty in our MSAs was assessed using GBlocks v0.91b (Castresana 2000). Ambiguous blocks were selected and removed according to the following settings: minimum number of sequences for a conserved position = 517, minimum number of sequences for a flank position = 877, maximum number of contiguous non-conserved positions = 8, minimum length of a block = 10, and allowed gap positions = all. These settings generated a conserved MSA of 1,401 sites.

<H2>Credits</H2>
<P>
The Multiple Sequence Alignment (MSA) of VAX004 HIV1 sequences was performed by Keith A. Crandall at 
<A HREF="http://www.genoma-llc.com" TARGET=_BLANK>Genoma LLC</A>.
</P>

<H2>References</H2>
<P>
Castresana J. <A HREF="http://mbe.oxfordjournals.org/cgi/content/abstract/17/4/540" TARGET=_blank>Selection of conserved blocks from multiple alignments for their use in phylogenetic analysis</A>. Molecular Biology and Evolution. 2000 April;17(4):540-52.
</P>
<P>
Guindon S, Gascuel O. A simple, fast, and accurate algorithm to estimate large phylogenies by maximum likelihood. Systematic Biology. 2003;52(5):696-704.
</P>
<P>
Katoh K, Kuma K, Toh H, Miyata T. <A HREF="http://nar.oxfordjournals.org/cgi/content/abstract/33/2/511" TARGET=_blank>MAFFT version 5: improvement in accuracy of multiple sequence alignment</A>. Nucleic Acids Research. 2005;33(2):511-18.
</P>
