<H2>Description</H2>
<P>
The VAX004 DNA Sequence Conservation track shows conservation based the MSA (Multiple Sequence Alignment) result of VAX004 HIV-1 gp120 nucleotide sequences.
</P>

<H2>Methods</H2>
<P>
<h3>Conservation</h3>
At each genomic position of the MSA, the counts of different bases (A, T, C, G, "-") were tallied and the frequency distribution calculated.  The frequency (fraction) of the most frequent base at each genomic position is used as the conservation measure.  The result is compiled into a wig format file and loaded to construct this track.
<P>
<h3>MSA method</h3>
We applied two local (L-INS-i and E-INS-i) and one global (G-INS-i) iterative refinement algorithms. MAFFT also incorporates a BLOSUM62 scoring matrix (an empirical matrix calculated from large datasets representing extreme protein family diversity) and assumes a fixed frequency for each AA for building MSAs; however, the estimated relative rates of changes in this matrix may be too general to fit datasets of specific gene families or high substitution rates, such as gp120. Hence, here we performed multiple MAFFT alignments using both the BLOSUM62 and gp120-specific scorings matrixes and/or fixed and variable AA frequencies estimated for the data at hand. Gene-specific empirical matrices and AA frequencies based on the VAX004 and LAGB-VAX004 datasets were generated via the program MATRIXGEN (http://matrixgen.sourceforge.net). 
<P>The quality of the MSA was then compared using the maximum likelihood (ML) tree building approach in PhyML (Guindon and Gascuel 2003) under the GTR+G+I model. ML trees showing the highest likelihood scores were indicative of better alignments. G-INS-i + BLOSUM62 + AA specific frequencies generated the trees with the best likelihood scores. 
<P>Uncertainty in our MSAs was assessed using GBlocks v0.91b (Castresana 2000). Ambiguous blocks were selected and removed according to the following settings: minimum number of sequences for a conserved position = 517, minimum number of sequences for a flank position = 877, maximum number of contiguous non-conserved positions = 8, minimum length of a block = 10, and allowed gap positions = all. These settings generated a conserved MSA of 1,401 sites.

<H2>Credits</H2>
<P>
The Multiple Sequence Alignment (MSA) of VAX004 HIV1 sequences was performed by Keith A. Crandall at 
<A HREF="http://www.genoma-llc.com" TARGET=_BLANK>Genoma LLC</A>.
</P>

<H2>References</H2>
<P>
Castresana J. <A HREF="http://mbe.oxfordjournals.org/cgi/content/abstract/17/4/540" TARGET=_blank>Selection of conserved blocks from multiple alignments for their use in phylogenetic analysis</A>. 
<EM>Molecular Biology and Evolution</EM>. 2000 April;17(4):540-52.
</P>
<P>
Guindon S, Gascuel O. A simple, fast, and accurate algorithm to estimate large 
phylogenies by maximum likelihood. <EM>Systematic Biology</EM>. 2003;52(5):696-704.
</P>
<P>
Katoh K, Kuma K, Toh H, Miyata T. 
<A HREF="http://nar.oxfordjournals.org/cgi/content/abstract/33/2/511" TARGET=_blank>MAFFT version 5: improvement in accuracy of multiple sequence alignment</A>
. <EM>Nucleic Acids Research</EM>. 2005;33(2):511-18.
</P>
