<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Bioinformatics: Credits</TITLE>
	<LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="/images/title.jpg"></TH></TR>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
	<table bgcolor="000000" cellpadding="1" cellspacing="1" width="100%" height="27"><tr bgcolor="2636D1"><td valign="middle">
	<table BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24">
			<TR><TD VALIGN="middle"><font color="#89A1DE">&nbsp; 
			&nbsp;<a href="/index.html" class="topbar">
				Home</a> &nbsp; - &nbsp;
			<a href="/cgi-bin/hgGateway" class="topbar">
				Genome Browser</a> &nbsp; - &nbsp; 
			<A HREF="/cgi-bin/hgBlat?command=start" class="topbar">
				Blat Search</a> &nbsp; - &nbsp; 
			<A HREF="/FAQ.html" class="topbar">
				FAQ</a> &nbsp; - &nbsp; 
			<A HREF="/goldenPath/help/hgTracksHelp.html" class="topbar">
				User Guide</a> &nbsp;</font></TD>
	</TR></TABLE>
	</TD></TR></TABLE>
</TD></TR>

<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=2 CELLPADDING=10>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Credits - UCSC Human Genome Project </FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
		<TABLE BORDER=0 HEIGHT=25 WIDTH=435 ALIGN="right" CELLPADDING=2 CELLSPACING=2 MARGINS=2><TR><TD><IMG SRC="../images/DOT.gif" WIDTH=10 HEIGHT=1></TD><TD>
		<TABLE BORDERCOLOR="999999" BORDER=1 CELLPADDING=0 CELLSPACING=0 MARGINS=0 HEIGHT=20 ALIGN="right"><TR><TD>
		<TABLE BORDERCOLOR="333333" BORDER=1 CELLPADDING=3 CELLSPACING=3 MARGINS=0 HEIGHT=20>
		<TR>
			<TD ROWSPAN=2 BGCOLOR="CECE9C" WIDTH="10%">
				<A HREF="http://www.hhmi.org">
				<IMG SRC="../images/HHMI.gif" BORDER=0></A></TD>
			<TD BGCOLOR="002D59" HEIGHT="38">	
				<A HREF="http://www.ucop.edu/california-institutes">
				<IMG SRC="../images/CISI.gif" BORDER=0></A>
			</TD><TR></TD><TD BGCOLOR="D9E4F8" >			
				<A HREF="http://www.nhgri.nih.gov">
				<IMG SRC="../images/NHGRI.gif" BORDER=0></A></TD>
		</TR></TABLE>			
		</TD></TR></TABLE>		
		</TD></TR></TABLE>	
		<P CLASS="leftjustify">
			<P>The UCSC Human Genome Project is conducted in collaboration with the 
		<A HREF="http://www.nhgri.nih.gov/HGP">International Human Genome Project</A>. We have worked with many collaborators at other institutions
		to produce and annotate this working draft of the human genome, as described in our 
		<a href="pubs.html">recent publications</a>.
		The UCSC Human Genome Project is made possible by a grant from the 
		<A HREF="http://www.nhgri.nih.gov">National Human Genome Research Institute</a>,
		directed by Francis Collins, and generous support from the 
		<A HREF="http://www.hhmi.org">Howard Hughes Medical Institute</a> and the 
		<A HREF="http://www.ucop.edu/california-institutes">California Institutes for 
			Science and Innovation</a>.</P>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<BR>
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Sources of the human data </FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
		The human genome sequence data we use has been generated by
		laboratories involved with <a href="labs.html">the Human Genome Sequencing Consortium</a>. 
		Periodic freezes of this data are generated by 
		<A HREF="http://www.ncbi.nlm.nih.gov/">NCBI</A>.
		EST, mRNA, BACend and other types of data used for the assembly and annotation are taken
		from <A HREF="http://www.ncbi.nlm.nih.gov/Ftp/index.html">GenBank</A>
		/<A HREF=" http://www.ebi.ac.uk/embl/">EMBL</A>
		/<A HREF="http://www.ddbj.nig.ac.jp/">DDBJ</A>.
		Clone maps were created by the individual sequencing centers. 
		
	<BR><BR>
	The assembly of the finished chromosomes is taken directly from files edited by 
	Greg Schuler at NCBI. The original files are produced by the institutions and 
	people listed in the table below.
	<BR><BR>
	<!--finished chromosome table-->
	<TABLE BORDER=1 BORDERCOLOR="#333333" CELLPADDING=0 WIDTH="100%"><TR><TD>
	<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
	<TR><TD COLSPAN=4><b> Finished Chromosomes </b></TD></TR>
	<TR><TD nowrap><font color="#006666"><b> Chr. # </b></font></TD>
		<TD nowrap><font color="#006666"><b> Sequencing Centers </b></font></TD>
		<TD nowrap><font color="#006666"><b> AGP File</b></font></TD></TR>
	
	<TR valign="top"><TD align="center">6</TD>
		<TD><ul>
			<li><A HREF="http://www.sanger.ac.uk/"> The Sanger Centre</A>, Cambridge, UK.
			</TD> <!--</ul> left out to remove <br> effect -->
		<TD><a href="mailto:jrh@@sanger.ac.uk">Jane Rogers</a></TD></TR>

	<TR valign="top"><TD align="center">7</TD>
		<TD><ul>
			<li><A HREF="http://genome.wustl.edu/projects/human/">Genome Sequencing Center</A>, 
				Washington University School of Medicine, St Louis, USA
			</TD> <!--</ul> left out to remove <br> effect -->
		<TD nowrap>
			<a href="mailto:bwaterst@watson.wustl.edu">Bob Waterston</a> <br>
			<a href="mailto:rwilson@watson.wustl.edu">Rick Wilson</a></TD></TR>

	<TR valign="top"><TD align="center">13</TD>
		<TD><ul>
			<li><A HREF="http://www.sanger.ac.uk/"> The Sanger Centre</A>, Cambridge, UK.
			</TD> <!--</ul> left out to remove <br> effect -->
		<TD><a href="mailto:jrh@@sanger.ac.uk">Jane Rogers</a></TD></TR>

	<TR valign="top"><TD align="center">20</TD>
		<TD><ul>
			<li><A HREF="http://www.sanger.ac.uk/"> The Sanger Centre</A>, Cambridge, UK.
			</TD> <!--</ul> left out to remove <br> effect -->
		<TD><a href="mailto:jgrg@sanger.ac.uk">James Gilbert</a></TD></TR>
	
	<TR valign="top"><TD align="center">21</TD>
		<TD><ul>
			<li><A HREF="http://www-eri.uchsc.edu/chromosome21/consortium/">Chromosome 21 Consortium</A>
			<li><A HREF="http://chr21.rz-berlin.mpg.de/">Max Plank Institute</A>, Berlin, Germany  
			<li><A HREF="http://hgp.gsc.riken.go.jp/chr21/index.html">RIKEN</A>, Japan
			</TD> <!--</ul> left out to remove <br> effect -->
		<TD nowrap>
			<a href="mailto:taylor@gsc.riken.go.jp">Todd Taylor</a> <br>
			</TD></TR>
	
	
	<TR valign="top"><TD align="center">22</TD>
		<TD><ul>
			<li><A HREF="http://www.sanger.ac.uk/"> 
			The Sanger Centre</A>, Cambridge, UK.
			<li><A HREF="http://www.dmb.med.keio.ac.jp/">
			Keio University School of Medicine</A>, Tokyo, Japan
			<li><A HREF="http://dna1.chem.ou.edu/index.html">
			The University of Oklahoma</A>, USA.
			<li><A HREF="http://genome.wustl.edu/projects/human/">
			Genome Sequencing Center</A>, Washington University 
			School of Medicine, St Louis, USA
			</TD> <!--</ul> left out to remove <br> effect -->
		<TD nowrap>
			<a href="mailto:birney@ebi.ac.uk">Ewan Birney</a> <br>
			<a href="mailto:dmb@sanger.ac.uk">David Beare</a> <br>
			<a href="mailto:jgrg@sanger.ac.uk">James Gilbert</a> <br>
			</TD></TR>

	<TR valign="top"><TD align="center">Y</TD>
		<TD><ul>
			<li><A HREF="http://genome.wustl.edu/projects/human/">Genome Sequencing Center</A>, 
				Washington University School of Medicine, St Louis, USA
			</TD> <!--</ul> left out to remove <br> effect -->
		<TD nowrap><a href="mailto:rwilson@watson.wustl.edu">Rick Wilson</a></TD></TR>
	</TABLE>
	</TD></TR></TABLE>
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<BR>

<A NAME="mouse_credits"></A>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><a NAME="mouserestrict"></a><b>&nbsp; Credits - UCSC Mouse Genome Project</FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
		<P>The UCSC Mouse Genome Project is conducted in collaboration with the
		Mouse Sequencing Consortium and the Mouse Genome Sequencing
		Consortium.
		<P>The first 3X coverage of the mouse genome was produced by the Mouse 
		Sequencing Consortium, with the following acknowledgements: 
		<UL>
		<LI><B>Project Oversight</B> - Arthur Holden and Francis Collins<br>
	
		<LI><B>Funding</B> - <A HREF="http://www.gsk.com/"
		TARGET=_blank>GlaxoSmithKline</A>, the <A
		HREF="http://www.mgri.org/home.html" TARGET=_blank>Merck Genome Research 
		Institute</A>, <A HREF="http://www.affymetrix.com/"
		TARGET=_blank>Affymetrix, Inc.</A>, the <A
		HREF="http://www.wellcome.ac.uk/" TARGET=_blank>Wellcome Trust</A>, and six of the 
		National Institutes, including the <A
		HREF="http://www.nci.nih.gov/" TARGET=_blank>National Cancer
		Institute</A>, the <A HREF="http://www.nhgri.nih.gov/"
		TARGET=_blank> National Human Genome Research Institute</A>, 
		the <A HREF="http://www.nidcd.nih.gov/" TARGET=_blank>National 
		Institute on 
		Deafness and Other Communication Disorders</A>, the <A
		HREF="http://www.niddk.nih.gov/" TARGET=_blank>National Institute of 
		Diabetes and Digestive and Kidney Disease</A>, the <A
		HREF="http://www.ninds.nih.gov/" TARGET=_blank>National 
		Institute of Neurological Disorders and Stroke</A>, and
		the <A
		HREF="http://www.nimh.nih.gov/home.cfm" TARGET=_blank>National 
		Institute of Mental Health</A>.<br>

	  	<LI><B>Sequencing</B> - 
		<A HREF="http://www-genome.wi.mit.edu"
		TARGET=_blank>Whitehead Institute for Biomedical
		Research</A> in Cambridge, Mass., 
		<A HREF="http://medicine.wustl.edu/"
		TARGET=_blank>Washington University School of
		Medicine</A> in St. Louis, and the <A HREF="http://www.sanger.ac.uk"
		TARGET=_blank>Sanger Centre</A> in the U.K. 
		</UL></P>
			
		<P>The remaining portion of the coverage, the map, and
		the assembly were produced by the Mouse Genome Sequencing
		Consortium, with these acknowledgements: 
		<UL>  
	    	<LI><B>Assembly</B> - 
		<A HREF="http://www-genome.wi.mit.edu" TARGET = _blank>Whitehead's  
		Arachne Group</A>; 
		<A HREF="http://www.sanger.ac.uk"
		TARGET=_blank>Sanger Centre</A> group: Jim Mullikin, Zemin Ning, Tim
		Hubbard <br>

	     	<LI><B>Mapping</B> - <A HREF="http://www.sanger.ac.uk"
		TARGET=_blank>The Wellcome Trust Sanger Institute</A>,
		the <A HREF="http://www.genome.wustl.edu/projects/mouse/"
		TARGET=_blank>Genome Sequencing
		Center</A> at Washington University School of
		Medicine, 
		<A HREF="http://www-genome.wi.mit.edu"
		TARGET=_blank>Whitehead Institute for Biomedical
		Research</A> in Cambridge, Mass., 
		<A HREF="http://www.bcgsc.bc.ca"
		TARGET=_blank>British Columbia 
		Genome Sequence Centre</A>, <A HREF="http://www.tigr.org/"
		TARGET=_blank>The Institute for Genome Research</A>, <A
		HREF="http://www.childrenshospitaloakland.org/chori.html"
		TARGET=_blank>Children's 
		Hospital Oakland Research Institute</A>. Mapping details
		can be found at
		<A HREF="http://www.ensembl.org/Mus_musculus/assembly.html"
		TARGET=_blank>http://www.ensembl.org/Mus_musculus/assembly.html</A>.<br>
		            
	  	<LI><B>Computational Leadership for the Mouse Genome Sequencing
		Consortium</B> - Ewan Birney at <A
		HREF="http://www.ebi.ac.uk/" TARGET=_blank>EBI</A><br>

	  	<LI><B>Testing</B> - Deanna Church at <A
		HREF="http://www.ncbi.nlm.nih.gov/" TARGET=_blank>NCBI</A><br>

	  	<LI><B>Annotations</B> - <A HREF="http://www.jax.org/"
		TARGET=_blank>Jackson Labs</A>, <A
		HREF="http://www.riken.go.jp/eng/"
		TARGET=_blank>Riken</A>, <A
		HREF="http://www.ncbi.nlm.nih.gov/" TARGET=_blank>NCBI</A>, <A
		HREF="http://www.ensembl.org/" TARGET=_blank>Ensembl</A>, <A
		HREF="http://www.psu.edu/" TARGET=_blank>Penn
		State</A>, <A HREF="http://www-hgc.lbl.gov/"
		TARGET=_blank>UC Berkeley/LBNL</A>, <A
		HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/"
		TARGET=_blank>UCSC</A>
		</UL> <P>
						  
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

<BR>

<A NAME="mouse_use"></A>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><a NAME="mouserestrict"></a><b>&nbsp; Sources of the mouse data -
	Conditions for use</FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
		<P>Mouse genome sequence information is released weekly into a
		public repository maintained by EBI and NCBI. This data is  made 
		available before scientific publication with the following understanding:
		<OL><LI>The data may be freely downloaded, used in analyses, and
		repackaged in databases.
		<LI>Users are free to use the data in scientific papers analyzing
		particular genes and regions if the providers of this
		data (the Mouse Sequencing Consortium) are properly acknowledged.
		<LI>The Centers producing the data reserve the right to publish the
		initial large-scale analyses of the dataset, including large-scale
		identification of regions of evolutionary conservation and
		large-scale genomic assembly. Large-scale refers to regions with
		size on the
		order of a chromosome (that is, 30 Mb or more). 
		<LI>Any redistribution of the data should carry this notice.
		</OL></P>
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<BR>

<A NAME="rat_credits"></A>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><a NAME="mouserestrict"></a><b>&nbsp; Credits - UCSC Rat Genome Project</FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
		<P>The UCSC Rat Genome Project is conducted in collaboration 
		with the Rat Genome Sequencing Consortium.
		<P>
		The Rat genome sequence was produced with the following 
		acknowledgements:
		<UL>
		<LI><B>Project Coordination</B> - <A HREF="http://www.hgsc.bcm.tmc.edu/" TARGET=_blank>Baylor College of Medicine Human Genome Sequencing Center</A>, Richard Gibbs, director<br>
		<LI><B>Funding</B> - The <A HREF="http://www.nhgri.nih.gov/" TARGET=_blank>National Human Genome Research Institute</A> 
		and the <A HREF="http://www.nhlbi.nih.gov/" TARGET=_blank>National Heart, Lung and Blood Institute</A> of the National 
		Institutes of Health.<br>

	  	<LI><B>Sequencing</B> - Baylor College of Medicine Human Genome Sequencing Center, <A HREF="http://www.celera.com/" TARGET=_blank>Celera Genomics Group of Applera Corporation</A>, 
		<A HREF="http://www.genomecorp.com/" TARGET=_blank>Genome Therapeutics Corporation</A>, <A HREF="http://www.utah.edu/" TARGET=_blank>The University of Utah</A><br>

	    	<LI><B>Assembly</B> - Baylor College of Medicine Human Genome 
		Sequencing Center ATLAS group:  Rui Chen, James Durbin, Paul 
		Havlak<br>

	     	<LI><B>Fingerprint Mapping</B> - <A HREF="http://www.bcgsc.bc.ca/" TARGET=_blank>The British Columbia Cancer 
		Agency Genome Sciences Center</A>: Marco Marra, Jackie Schein<br>
		            
	     	<LI><B>BAC End Sequencing</B> - <A HREF="http://www.tigr.org/tdb/bac_ends/rat/bac_end_intro.shtml" TARGET=_blank>The Institute for Genomic 
		Research</A>: Shaying Zhao<br>

	     	<LI><B>BAC Library Construction and Distribution</B> - <A HREF="http://www.chori.org/bacpac/" TARGET=_blank>Children's Hospital Oakland Research Institute</A>:  Pieter deJong<br>

	     	<LI><B><A HREF="http://rgd.mcw.edu" TARGET=_blank>Rat Genome Database</B></A> - Medical College of Wisconsin
		</UL> <P>
						  
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<BR>

<A NAME="rat_use"></A>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><a NAME="mouserestrict"></a><b>&nbsp; Sources of the rat data -
	Conditions for use</FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
		<P>This data is  made 
		available before scientific publication with the following understanding:
		<OL><LI>The data may be freely downloaded, used in analyses, and
		repackaged in databases.
		<LI>Users are free to use the data in scientific papers analyzing
		particular genes and regions if the providers of this
		data (the Rat Genome Sequencing Consortium) are properly acknowledged.
		<LI>The Centers producing the data reserve the right to publish the
		initial large-scale analyses of the dataset, including large-scale
		identification of regions of evolutionary conservation and
		large-scale genomic assembly. Large-scale refers to regions with
		size on the
		order of a chromosome (that is, 30 Mb or more). 
		<LI>Any redistribution of the data should carry this notice.
		</OL></P>
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<BR>
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Contributions of the UCSC team </FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD COLSPAN=2>
	<ul>
		<li><a href="http://www.cse.ucsc.edu/~kent"><b>Jim Kent</b></a>, 
			Co-director of the project, Research
			Scientist:
			Design and implementation of the 
			<a href="hgTracks.html">Genome Browser</a> and the 
			<a
			href="../cgi-bin/hgBlat?command=start">BLAT</a> 
			sequence search tool. Author 
			of the <a href="algo.html">GigAssembler</a> 
			program.  Implemented many tracks, including the FPC
			contigs, assembly, gap, coverage, known genes, 
			full mRNA, spliced ESTs, human ESTs, 
			Genscan genes, non-human mRNA and EST, and mouse BLAT 
			browser tracks.
		<li><a href="http://www.cse.ucsc.edu/~booch"><b>Terry Furey</b></a>, 
			PostDoc: Clone map analysis, STS browser tracks and STS 
			analysis (genetic, RH, FISH and YAC maps), BACend, cytogenetic band 
			and GC-content browser tracks.
		<li><a href="http://www.cse.ucsc.edu/~sugnet"><b>Chuck Sugnet</b></a>, 
			Graduate Student: Integration of functional genomics data, lifting of 
			annotation from one freeze to another, SAGE/UNIGENE track, general 
			browser support
		<li><b>Mark Diekhans</b>, Graduate Student and Staff: Mouse homology reference 
			sequence and phylogenetic analysis
		<li><a href="http://people.ucsc.edu/~daryl"><b>Daryl Thomas</b></a>, 
			Graduate student: haplotype track, in collaboration with Ryan Weber and Chuck Sugnet
		<li><a href="http://www.cse.ucsc.edu/~krish"><b>Krishna Roskin</b></a>, 
			Graduate student: Browser database support, text-based browser, and
			phylogenetic analysis
		<li><b>Yontao Lu</b>, Graduate student: Help with genetic maps
		<li><A HREF="http://www.soe.ucsc.edu/~baertsch/"><B>Robert
		Baertsch</B></A>, 
		Graduate student: Synteny, pseudogenes, orthology, browser support
		<li><a href="http://www.cse.ucsc.edu/~acs"><b>Adam Siepel</b></a>, 
			Graduate student: General Browser support
		<li><a href="http://64.175.27.98/"><b>Ryan Weber</b></a>, 
			Undergraduate student: General browser and map support
		<li><b>Fan Hsu</b>, Research Scientist: Director of
			Proteomics.  Mouse Genome Browser, proteomics initiative
		<li><a href ="http://www.cse.ucsc.edu/~matt"><b>Matt Schwartz</b></a>,
			Staff: Genome Browser engineering
		<li><b>Angie Hinrichs</b>, Staff: Genome Browser engineering
		<li><b>Heather Trumbower</b>, Staff: Genome Browser QA
		<li><b>Donna Karolchik</b>, Staff: Project manager, documentation, QA
		<li><b>Paul Tatarsky</b>, Staff: Hardware, networking, security, mirror 
			sites, general advice
		<li><b>Patrick Gavin</b>, Staff: Enhancing and running scripts to assemble 
			the genome, testing, compressing the genome to fit on a CD, design of 
			the kilocluster, general system support 
		<li><a href="http://www.cse.ucsc.edu/~jgarcia"><b>Jorge Garcia</b></a>, 
			Staff: Co-design of the kilocluster. Chief hardware bug finder, 
			codine guru. Web, mail, laptop and general system support.
		<li><b>Ann Pace</b>, Co-director of the 
			<a href="http://www.cse.ucsc.edu/centers/cbe">UCSC Center for 
			Biomolecular Science and Engineering</a>: Grants, budget, space and 
			other administrative support
		<li><b>Katy Elliot</b>, Staff: Administrative Support
		<li><b>Mary McMurtry</b>, Administrative Support
		<li><a href="http://www.cse.ucsc.edu/~haussler"><b>David Haussler</b></a>, 
			Professor: Project Director 
	</ul>
</TD></TR><TR><TD WIDTH=10></TD><TD>
We gratefully acknowledge the help of former UCSC genome team members Scot Kennedy, Nick Littlestone, Aaron Tomb,
Degui Zhi, Maria Guarino, Matt Xia, Wei Wu, Sanja Rogic, Jenny Draper, and Trevor Bruen.<br>

</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<BR>
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Other scientists whose work is featured at this site </FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD COLSPAN=2>
</TD></TR><TR><TD COLSPAN=2>
<ul>

<li><b>Jennifer Ashurst, James Gilbert, and all the annotators</b> at the 
	<a href="http://www.sanger.ac.uk/HGP/">Sanger Centre</A>: 
	Sanger chromosome 20 gene annotations track
<li><b>Ewan Birney, Guy Slater, Imre Vastrik, Arek Kasprzyk, Michele Clamp</b> 
	at <A HREF="http://www.ensembl.org/">Ensembl</A>:
	gene predictions, data exchange formats, TPF clone map coordiation, 
	integration with the Wash. U. fingerprint map,  
	exonerate mouse homology track, early help placing FISH-mapped clones. 
<li><b>Carol Bult</B> at <A HREF="http://www.jax.org/">Jackson Labs</A>: 
	mouse ortholog links from known genes track
<li><b>Deanna Church</b> 
    at <A HREF="http://www.ncbi.nlm.nih.gov/index.html">NCBI</A>:
	Mouse synteny track, database of whole genome shotgun mouse reads
	and assemblies used for mouse tracks	
<li><b>Ian Dunham</b> 
	at the <a href="http://www.sanger.ac.uk/HGP/">Sanger Centre </a>,  
	<b>Bill Grundy</b> and <b>Victoria Haghighi</b> at
	<a href="http://www.columbia.edu">Columbia University</a>: experimentally 
	verified gene annotations for chromosome 22 (Sanger 22 track).
<li><A href="http://www.genetics.wustl.edu/eddy/"><b>Sean Eddy</b></A> 
	at Wash. U. and <b>Todd Lowe</b> (now at UCSC): the RNA genes track.
<li><b>Evan Eichler</b> and <b>Jeff Bailey</b>
	at <A HREF="http://www.cwru.edu/">Case Western Reserve University</A>: 
	the segmental duplications track.
<li><b>Richard Gibbs, John Bouck, Kim Worley and James Durbin</b> 
    at <a href="http://www.hgsc.bcm.tmc.edu/">Baylor College of Medicine</a>:
	CpG island track, simple repeats track.
<li><b>David Hinds, Kelly Frazer, and David Cox</b> 
	at <a href="http://www.perlegen.com">Perlegen Sciences</a>: STS markers for the         
	<A href="http://www-shgc.stanford.edu/Mapping/Marker/STSindex.html">
	TNG radiation hybrid map</A> and haplotype blocks for common SNPs on chr. 21.
<li><b>David Kulp, Alan Williams</b>, and <b>Ray Wheeler</b> 
	at <A HREF = "http://www.affymetrix.com">Affymetrix</A>:
	Genie gene prediction track for the browser, design and analysis support in 
	the early stages of the project, including software for the synthetic 
	chromosome 22 data we used to test our assembly methods.
<li><b>Eric Lander, Mike Zody, Chad Nusbaum</b> 
	at the <a href="http://www-genome.wi.mit.edu/">Whitehead Institute for
	Biomedical Research</a>: plasmid end reads used starting with the Sept. 
	2000 freeze. Special thanks to Eric for inviting us to join the Hardcore 
	Analysis Group for the draft sequence project, and for his direction of that 
	group. 
<li><b>Ming Li</b> and colleagues at 
	<a href="http://www.bioinformaticssolutions.com/">Bioinformatics Solutions, Inc.</a>:
	Pattern Hunter mouse homology track.
<li><b>Victor Pollara</b> at <a href="http://www.viaken.com/">Viaken</a> and <b>Arian Smit</b> at the 
	<a href="http://www.systemsbiology.org">Intsitute for Systems Biology</a>:
	repeat identification and masking.
<li><b>Thomas Pringle</b> at 
	the <A HREF="http://www.mad-cow.org/00/sci_archive_frame.html">
	Sperling Biomedical Foundation</A>: user's guides, email moderator, web links, 
	testing, myriad ideas for browser.
<li><b>Greg Schuler</b> at 
	<A HREF="http://www.ncbi.nlm.nih.gov/index.html">NCBI</A>: GenBank freezes of 
	human clones used in the assembly, 'NT contigs' of multiple finished clones,
	STS maps (in earlier freezes), specially formated versions of the finished 
	chromosomes.
<li><b>Victor Solovyev</b> at 
	<A HREF="http://www.softberry.com/">Softberry Inc.</A>:
	Fgenesh++ gene predictions and TSSW promoters tracks.
<li><a href="http://stein.cshl.org/~lstein/"><b>Lincoln Stein </b></a> 
	at Cold Spring Harbor Laboratory, <b>Steve Sherry</b> at 
	<A HREF="http://www.ncbi.nlm.nih.gov/index.html">NCBI</A>, and the 
	SNP Consortium: SNP tracks.
<li><b>John Sulston, Jane Rogers, Jim Mullikin, and Richard Durbin</b> at the 
	<a href="http://www.sanger.ac.uk/HGP/">Sanger Centre </a>:  
	plasmid end reads starting with October 2000 freeze, help with gene 
	predictions, processing of whole genome shotgun mouse reads.
<li><b>Danielle and Jean Thierry-Mieg </b> at 
	<a href="http://www.ncbi.nlm.nih.gov/index.html"> NCBI </a>:  
	Acembly genes track.
<li><b>Barbara Trask</b> at the 
	<a href="http://www.fhcrc.org/">Fred Hutchison Cancer Center</a>, 
	<b>Wonhee Jang</b> at
	<A HREF="http://www.ncbi.nlm.nih.gov/index.html">NCBI</A>, and the 
	BAC Resource Consortium:<A HREF="http://www.ncbi.nlm.nih.gov/genome/cyto/">
	Cytogenetic markers</A> (FISH-mapped clones).
<li><b>Robert Waterston, John McPhearson, Asif Chinwalla, Shiaw-Pyng Yang, John Wallis</b> and 
	<b>LaDeana Hillier</b> at Washington University in St. Louis: 
	<A HREF="http://genome.wustl.edu/gsc/human/Mapping"> fingerprint-based map</A> 
	of BAC clones from the human genome and a corresponding layout of sequenced 
	clones, plasmid end reads starting with October 2000 freeze, BACend analysis,
	CpG islands track.
<li><b>Lukas Wagner</b> at 
	<A HREF="http://www.ncbi.nlm.nih.gov/index.html">NCBI</A>: 
	3' EST track in the July 2000 browser.
<li><b>Jean Weissenbach</b> and <b>Olivier Jaillon</b> at 
	<A href="http://kicy.genoscope.cns.fr/cgi-bin/exofish_kicy.cgi">Genoscope
	</A>: Exofish track.

</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<BR>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Acknowledgements of early support  </FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>

<P>During the initial stage of this project, from January 2000 to June 2001,
we received no direct federal funding to develop the working draft other than travel 
support. However, we gratefully acknowledge the support for our personnel provided 
by our basic bioinformatics research grants, especially DOE grant
DE-FG03-99ER62849 (to develop genefinding methods), but also NSF grant DBI-9809007, 
UC BIOTECH grant 99-11, a grant from the Sloan Foundation, the David and Lucille 
Packard Foundation, NIH grant GM-52848, and the Zahler grant that provided support 
for Jim Kent in the early stages of this work. 
We are particularly thankful for the risks that Patrick Mantey, Dean of Engineering, 
James Gill, Vice Chancellor for Research, John Simpson, UCSC Provost, and 
M.R.C. Greenwood, Chancellor, were willing to take 
in advancing us the money in February, 2000 needed to build the original 100-node 
LINUX cluster used in this work, and in the case of Dean Mantey, additional 
personnel support. Phokion Kolaitis, Chair of Computer Science, contributed as 
well. We thank <A HREF="http://www.compaq.com/">Compaq</A> for providing an 
additional 100-cpu computational cluster in Cambridge, MA for use
by the public human genome effort. We thank <A HREF="http://www.ilog.com/">ILOG,
Inc.</A> for providing discounted licenses to its CPLEX linear programming software, 
which was essential for some of our early experiments.</P>
<P>
Beginning in December, 2000, we started to utilize some support from 
<A HREF="http://www.hhmi.org">HHMI</a> for equipment and personnel. In the summer 
of 2001, HHMI provided the primary funding for a new cluster of 1000 CPUs running 
LINUX purchased from <a href="http://www.rackable.com">Rackable</a>, with additional 
support from <A HREF="http://www.ucop.edu/california-institutes">CISI</A> and 
<A HREF="http://www.nhgri.nih.gov">NHGRI</A>. Dubbed the 
UCSC KiloCluster, this equipment provides the computational backbone
for the project.   
</P>

</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>

</TD></TR></TABLE>
</BODY></HTML>


