<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>Human Genome Working Draft</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 >
<TH COLSPAN=2 ALIGN="left">	
    <IMG SRC="images/title.jpg">
	<!--<FONT FACE="Arial Black" SIZE=5 COLOR="#0033366">Human Genome Project Working Draft at UCSC</FONT>-->
</TH>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
   <CENTER>
	<TABLE BACKGROUND="images/hl_mid.jpg" BGCOLOR="253BDE" WIDTH=100% CELLSPACING=0 CELLPADDING=0 BORDER=0 HEIGHT="22"><TR><TD>
		<TABLE WIDTH=500 CELLSPACING=4 CELLPADDING=0 BORDER=0><TR><TD WIDTH=3></TD>
		<TD WIDTH=90>
			<A HREF="goldenPath/hgTracks.html" onMouseOver="BROWSER.src='images/hl_browser.jpg'" onMouseOut="BROWSER.src='images/h_browser.jpg'">
			<IMG SRC="images/h_browser.jpg" ALIGN="absmiddle" BORDER="0" NAME="BROWSER" ALT="&nbsp;&nbsp; Genome Browser"></A></TD>
			
		<TD WIDTH=90>
			<A HREF="../cgi-bin/hgBlat?command=start" onMouseOver="SEARCH.src='images/hl_blat.jpg'" onMouseOut="SEARCH.src='images/h_blat.jpg'">
			<IMG SRC="images/h_blat.jpg" ALIGN="absmiddle" BORDER="0" NAME="SEARCH" ALT="&nbsp;&nbsp; Sequence Search"></A></TD>
		
		<TD WIDTH=50>
			<A HREF="#Downloading" onMouseOver="HOME.src='images/hl_downloads.jpg'" onMouseOut="HOME.src='images/h_downloads.jpg'">
			<IMG SRC="images/h_downloads.jpg" ALIGN="absmiddle" BORDER="0" NAME="HOME" ALT="Home" ></A></TD>

		<TD WIDTH=90>
			<A HREF="FAQ.html" onMouseOver="FAQ.src='images/hl_faq.jpg'" onMouseOut="FAQ.src='images/h_faq.jpg'">
			<IMG SRC="images/h_faq.jpg" ALIGN="absmiddle" BORDER="0" NAME="FAQ" ALT="&nbsp;&nbsp; FAQ"></A></TD>
	 </TR></TABLE>
 </TR></TABLE>
	 </CENTER>
    </TD>
</TR>

<!--LeftBar Links Are at bottom of page! (for viewing in LYNX! :) -->
<TR>
<!--Content Tables-->  
  <TD CELLPADDING=10>	
  
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="ffffff" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<A NAME="About"></A><FONT SIZE="4"><b>&nbsp; Instructions </FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
<pre>
General instructions for credits.html page:
++++++++++++++++++++++++++++++++++++++++++

1. Add the logos for NHGRI, HHMI and CISI to the top of this page.
2. Where I put <> around something in the revised text below, please make it a link.
3. make proper html lists and headers in the revised text below.
4. Make a new page for the <>laboratories involved the human genome 
   sequencing consortium<> using this table from the bottom of the old 
   other.resources.html file:
</pre>
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<br>
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="ffffff" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Credits </FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
<P>
The UCSC Human Genome Project is conducted 
collaboration with the <A HREF="http://www.nhgri.nih.gov/HGP">International Human Genome Project</A>,
directed by Francis Collins. We have worked with many collaborators at other institutions
to produce and annotate this working draft of the human genome, as described in our <recent
publications>.
The UCSC Human Genome Project is made possible by a grant from the <National Human Genome
Research Institute> and generous support from the <Howard Hughes Medical Institute> and
the <California Institutes for Science Innovation>.<br>

<br><FONT SIZE=4 COLOR="#006666"> Sources of the data: </FONT><br><br>
The human genome sequence data we use has been generated by the <laboratories
involved the human genome sequencing consortium>. 
EST, mRNA, BACend and other types of data used for the assembly and annotation are taken
from <A HREF="http://www.ncbi.nlm.nih.gov/Ftp/index.html">GenBank.</A><br> 

<br><FONT SIZE=4 COLOR="#006666">
The institutions that form the Human Genome Sequencing Consortium include:
</FONT>
<ol>
  <li><a href="http://www.hgsc.bcm.tmc.edu/">Baylor College of Medicine,
Houston, Texas, USA</a></li>
  <li><a href="http://hgc.igtp.ac.cn/">Beijing Human Genome Center, Institute
of Genetics, Chinese Academy of Sciences, Beijing, China</a></li>
  <li><a href="http://nucleus.cshl.org/genseq/lita_annenberg_hazen_genome_cent.htm">Cold Spring Harbor Laboratory,
Lita Annenberg Hazen Genome Center, Cold Spring Harbor, NY, USA</a></li>
  <li><a href="http://genome.gbf.de/">Gesellschaft f&uuml;r Biotechnologische
Forschung mbH, Braunschweig, Germany</a></li>
  <li><a href="http://www.genoscope.fr/">Genoscope, Evry, France</a></li>
  <li><a href="http://www.genomecorp.com/">Genome Therapeutics Corporation,
Waltham, MA, USA</a></li>
  <li><a href="http://genome.imb-jena.de/">Institute for Molecular
Biotechnology, Jena, Germany</a></li>
  <li><a href="http://www.jgi.doe.gov/">Joint Genome Institute, U.S.
Department of Energy, Walnut Creek, CA, USA</a></li>
  <li><a href="http://www-alis.tokyo.jst.go.jp/HGS/">Keio University, Tokyo,
Japan</a></li>
  <li><a href="http://seq.mpimg-berlin-dahlem.mpg.de/">Max Planck Institute
for Molecular Genetics, Berlin, Germany</a></li>
  <li><a href="http://hgp.gsc.riken.go.jp/">RIKEN Genomic Sciences Center,
Saitama, Japan</a></li>
  <li><a href="http://www.sanger.ac.uk/HGP/">The Sanger Centre, Hinxton,
U.K.</a></li>
  <li><a href="http://www-sequence.stanford.edu/">Stanford Genome Technology Center,
Palo Alto, CA, USA</a></li>
  <li><a href="http://shgc.stanford.edu/">Stanford Human Genome Center,
Palo Alto, CA, USA</a></li>
  <li><a href="http://www.genome.ou.edu/">University of
Oklahoma's Advanced Center for Genome Technology, OK, USA</a></li>
  <li><a href="http://www3.utsouthwestern.edu/index.htm">University of
Texas Southwestern Medical Center at Dallas, TX, USA</a></li>
  <li><a href="http://www.genome.washington.edu/UWGC/">University of
Washington Genome Center, Seattle, WA, USA</a></li>
  <li><a href="http://www.systemsbiology.org/">
Multimegabase Sequencing Center, Institute for Systems Biology, Seattle, WA,USA</a></li>
  <li><a href="http://www-genome.wi.mit.edu/">Whitehead Institute for
Biomedical Research, MIT, Cambridge, MA, USA</a></li>
  <li><a href="http://genome.wustl.edu/gsc/">Washington University Genome
Sequencing Center, St. Louis, MO, USA</a></li>
</ol>
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<BR>
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="ffffff" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Contributions of the UCSC team </FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD COLSPAN=2>
<ul>
<li><Jim Kent>, Co-director of the project, Graduate student: Design and implementation of the <GigAssembler> 
algorithm used to assemble the working draft sequence, design and implementation of the the <genome browser>, 
the <BLAT> sequence search tool, and the comparative genomics software. FPC contigs, assembly, gap,
coverage, known genes, full mRNA, spliced ESTs, human ESTs, and mouse BLAT browser tracks.

<li><Terry Furey>, Graduate student: Clone map analysis, STS browser tracks and STS analysis 
(genetic, RH, FISH and YAC maps), BACend, cytogenetic band and GC-content browser tracks.

<li><Chuck Sugnet>, Graduate Student: Integration of functional genomics data, lifting of annotation from 
one freeze to another, UNIGENE track, general browser support

<li><Sanja Rogic>, Graduate student: E-PCR mapping of STSs, cDNA analysis, pair-EST track

<li><Wei Wu>, Graduate student: Mirror sites, help with genetic maps

<li><Matt Xia>, Graduate student: Computation of the statistics for the working draft, scripts for phylogenetic data 
analysis, help with the mirror sites, assembly testing

<li><Mark Diekhans>, Graduate student: Mouse homology reference sequence and phylogenetic analysis

<li><Krishna Roskin>, Undergraduate student: Browser database support, text-based browser

<li><Jenny Draper>, Undergraduate student: Webmaster

<li><Paul Tatarsky>, Staff: Hardware, networking, security, mirror sites, general advice

<li><Patrick Gavin>, Staff: Enhancing and running scripts to assemble the genome, testing, compressing the genome
to fit on a CD, design of the kilocluster, general system support

<li><Jorge Garcia>, Staff: Co-design of the kilocluster. Web, mail, laptop and general system sypport

<li><Ann Pace>, Co-director of the <UCSC Center for Biomolecular Science and Engineering>: Grants, budget,
space and other administrative support

<li><Katy Elliot>, Staff: Administrative Support

<li><David Haussler>, Professor: Project Director 
</ul>
</TD></TR><TR><TD WIDTH=10></TD><TD>
We gratefully acknowledge the help of former UCSC genome team members Scot Kennedy, Nick Littlestone, Aaron Tomb
Degui Zhi and Maria Guarino.<br>

<br><FONT SIZE=4 COLOR="#006666">
Other contributors to the International Genome Project whose work is featured at this site:  
</FONT><br><br>
</TD></TR><TR><TD COLSPAN=2>
<ul>
<li>Ewan Birney, Guy Slater, Imre Vastrik, Arek Kasprzyk,
	Michele Clamp at <A HREF="http://www.ensembl.org/">Ensembl</A>:
	gene predictions, data exchange formats, TPF clone map coordiation, 
	integration with the Wash. U. fingerprint map,  
	exonerate mouse homology track, early help placing FISH-mapped clones. 
<li>Deanna Church at
	<A HREF="http://www.ncbi.nlm.nih.gov/index.html">NCBI</A>:
	Mouse synteny track.
<li>David Cox at <Perlegen>: STS markers for the         
	<A href="http://www-shgc.stanford.edu/Mapping/Marker/STSindex.html">
	TNG radiation hybrid map</A>.
<li>Ian Dunham at the
	<a href="http://www.sanger.ac.uk/HGP/">Sanger Centre </a>,  
	Bill Grundy and Victoria Haghighi at
	<Columbia University>: experimentally verified gene annotations
	for chromosome 22 (Sanger 22 track).
<li><A href="http://www.genetics.wustl.edu/eddy/">Sean Eddy</A> at Wash. U. and Todd Lowe
	(now at UCSC): the RNA genes track.
<li>Evan Eichler and Jeff Bailey
	at <A HREF="http://www.cwru.edu/">Case Western Reserve University</A>: the segmental duplications
	track in the July 2000 browser.
<li>Richard Gibbs, John Bouck, Kim Worley and James Durbin at  
	<a href="http://www.hgsc.bcm.tmc.edu/">Baylor College of Medicine</a>:
	CpG island track, simple repeats track.
<li>David Kulp, Alan Williams, and Ray Wheeler at 
	<A HREF = "http://www.affymetrix.com">Affymetrix</A>:
	Genie gene prediction track for the browser,
	design and analysis support in the early stages of the project, 
	including software for the synthetic chromosome 22 data we used to test our assembly methods.
<li>Eric Lander, Mike Zody, Chad Nusbaum at the <a href="http://www-genome.wi.mit.edu/">Whitehead Institute for
	Biomedical Research</a>: plasmid end reads used starting with the Sept. 2000 freeze. Special thanks
	to Eric for inviting us to join the Hardcore Analysis Group for the draft sequence project, and
	for his direction of that group. 
<li>Victor Pollara at <Viaken> and Arian Smit at the <Intsitute for Systems Biology>:
	repeat identification and masking.
<li>Thomas Pringle at the <A HREF="http://www.mad-cow.org/00/sci_archive_frame.html">
	Sperling Biomedical Foundation</A>: user's guides, email hosting, web links, testing,
	myriad ideas for browser.
<li>Greg Schuler at <A HREF="http://www.ncbi.nlm.nih.gov/index.html">NCBI</A>: freezes consisting of a 
	set of processed, managed DNA sequences from this layout, split at
	the appropriate point for working draft clones and with obvious
	contaminants removed and ends identified where possible, STS maps (in earlier freezes),
	specially formated versions of the finished chromosomes
	21 and 22. 
<li>Victor Solovyev at 
	<A HREF="http://www.softberry.com/">Softberry Inc.</A>:
	Fgenesh++ gene predictions and TSSW promoters tracks.
<li><a href="http://stein.cshl.org/~lstein/">
	Lincoln Stein </a> at Cold Spring Harbor Laboratory,
	and the <a href="http://snp.cshl.org/">SNP consortium</a>:
	SNP tracks.
<li>John Sulston, Jane Rogers, Jim Mullikin, and Richard Durbin at the 
	<a href="http://www.sanger.ac.uk/HGP/">Sanger Centre </a>:  
	plasmid end reads starting with October 2000 freeze, help with gene predictions, processing of whole genome 
	shotgun mouse reads.
<li>Barbara Trask at the <Fred Hutchison Cancer Center>, Wonhee Jang at
	<A HREF="http://www.ncbi.nlm.nih.gov/index.html">NCBI</A>, 
	and the BAC Resource Consortium:
	<A HREF="http://www.ncbi.nlm.nih.gov/genome/cyto/">
	Cytogenetic markers</A> (FISH-mapped clones).
<li>Robert Waterston, John McPhearson, Asif Chinwalla, Shiaw-Pyng Yang, John Wallis and 
	LaDeana Hillier at Washington University in St. Louis: <A HREF="http://genome.wustl.edu/gsc/human/Mapping">
	fingerprint-based map</A> of BAC clones from the human genome
	and a corresponding layout of sequenced clones, plasmid end reads starting with October 2000 freeze, BACend analysis,
	CpG islands track.
<li>Lukas Wagner at 
	<A HREF="http://www.ncbi.nlm.nih.gov/index.html">NCBI</A>: 
	3' EST track in the July 2000 browser.
<li>Jean Weissenbach and Olivier Jaillon at 
	<A href="http://kicy.genoscope.cns.fr/cgi-bin/exofish_kicy.cgi">Genoscope
	</A>: Exofish track.

</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<BR>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="ffffff" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Acknowledgements of Early Support  </FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>

<P>During the initial stage of this project, from January 2000 to June 2001,
we received no direct federal funding to develop the working draft other than travel support.
However, we gratefully acknowledge the support for our personnel provided by
our basic bioinformatics research grants, especially DOE grant
DE-FG03-99ER62849 (to develop genefinding methods), but also NSF
grant DBI-9809007, UC BIOTECH grant 99-11, a grant from the Sloan
Foundation, the David and Lucille Packard Foundation, NIH grant
GM-52848, and the Zahler grant that provided support for Jim Kent in the early stages
of this work. 
We are particularly thankful for the
risks that Patrick Mantey, Dean of Engineering, John Simpson, UCSC Provost, and
M.R.C. Greenwood, Chancellor, were willing to take in advancing us the money in February, 2000
needed to build the original 100-node LINUX cluster used in this work, and
in the case of Dean Mantey, additional personnel support. Phokion Kolaitis,
Chair of Computer Science, contributed as well. We thank <A
HREF="http://www.compaq.com/">Compaq</A> for
providing an additional 100-cpu computational cluster in Cambridge, MA for use
by the public human genome effort. We thank <A HREF="http://www.ilog.com/">ILOG,
Inc.</A> for providing discounted licenses to its CPLEX linear programming software, which 
was essential for some of our early experiments.</P>

<P>
Beginning in December, 2000, we started to utilize some support from <HHMI> for
equipment and personnel. In the summer of 2001, HHMI provided the primary funding
for a new cluster of 1000 CPUs running LINUX purchased from <Rackable>, with additional support 
from <CISI> and <NHGRI>. Dubbed the <UCSC kilocluster>, this equipment provides the computational backbone
for the project.   
</P>

</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>


</TD></TR></TABLE>

</BODY>

</HTML>


