<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>Human Genome Working Draft: Credits</TITLE>
	<LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0><TR><TH COLSPAN=2 ALIGN="left">	<IMG SRC="../images/title.jpg"></TH></TR>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
   <CENTER>
	<TABLE BGCOLOR="2636D1" WIDTH=100% CELLSPACING=0 CELLPADDING=0 BORDERCOLOR="000000" BORDER=1 HEIGHT="22"><TR><TD>
		<TABLE WIDTH=431 CELLSPACING=4 CELLPADDING=0 BORDER=0><TR>
		<TD WIDTH=1></TD>
		<TD WIDTH=75>
			<A HREF="../index.html" onMouseOver="HOME.src='../images/hl_home.jpg'" onMouseOut="HOME.src='../images/h_home.jpg'">
			<IMG SRC="../images/h_home.jpg" ALIGN="absmiddle" BORDER="0" NAME="HOME" ALT="Home" ></A></TD>

		<TD WIDTH=155>
			<A HREF="hgTracks.html" onMouseOver="BROWSER.src='../images/hl_browser.jpg'" onMouseOut="BROWSER.src='../images/h_browser.jpg'">
			<IMG SRC="../images/h_browser.jpg" ALIGN="absmiddle" BORDER="0" NAME="BROWSER" ALT="&nbsp;&nbsp; Genome Browser"></A></TD>
			
		<TD WIDTH=100>
			<A HREF="../cgi-bin/hgBlat?command=start" onMouseOver="SEARCH.src='../images/hl_blat.jpg'" onMouseOut="SEARCH.src='../images/h_blat.jpg'">
			<IMG SRC="../images/h_blat.jpg" ALIGN="absmiddle" BORDER="0" NAME="SEARCH" ALT="&nbsp;&nbsp; BLAT Search"></A></TD>
			
		<TD WIDTH=100>
			<A HREF="../FAQ.html" onMouseOver="GUIDE.src='../images/hl_faq.jpg'" onMouseOut="GUIDE.src='../images/h_faq.jpg'">
			<IMG SRC="../images/h_faq.jpg" ALIGN="absmiddle" BORDER="0" NAME="GUIDE" ALT="&nbsp;&nbsp; FAQ"></A></TD>
		</TR></TABLE>
	 </TR></TABLE>
	 </CENTER>
    </TD>
</TR>

<!--Content Tables--> 
<TR> 
  <TD CELLPADDING=10>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Credits </FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
			<P>The UCSC Human Genome Project is conducted in collaboration with the 
		<A HREF="http://www.nhgri.nih.gov/HGP">International Human Genome Project</A>. We have worked with many collaborators at other institutions
		to produce and annotate this working draft of the human genome, as described in our 
		<a href="pubs.html">recent publications</a>.
		The UCSC Human Genome Project is made possible by a grant from the 
		<A HREF="http://www.nhgri.nih.gov">National Human Genome Research Institute</a>,
		directed by Francis Collins, and generous support from the 
		<A HREF="http://www.hhmi.org">Howard Hughes Medical Institute</a> and the 
		<A HREF="http://www.ucop.edu/california-institutes">California Institutes for 
			Science and Innovation</a>.<br>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<BR>
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Sources of the data </FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
		The human genome sequence data we use has been generated by <a href="labs.html">the 
		laboratories involved the human genome sequencing consortium</a>. 
		Periodic freezes of this data are generated by 
		<A HREF="http://www.ncbi.nlm.nih.gov/">NCBI</A>.
		EST, mRNA, BACend and other types of data used for the assembly and annotation are taken
		from <A HREF="http://www.ncbi.nlm.nih.gov/Ftp/index.html">GenBank</A>
		/<A HREF=" http://www.ebi.ac.uk/embl/">EMBL</A>
		/<A HREF="http://www.ddbj.nig.ac.jp/">DDBJ</A>.<br> 
		Clone maps were created both by individual sequencing centers and by 
		<a href="http://genome.wustl.edu/gsc/human/Mapping/"> Washington University</a>, 
		and merged by the team at <a href="http://assembly.ensembl.org/">Ensembl</a>.

</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<BR>
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Contributions of the UCSC team </FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD COLSPAN=2>
	<ul>
		<li><a href="http://www.cse.ucsc.edu/~kent"><b>Jim Kent</b></a>, 
			Co-director of the project, Graduate student: Design and implementation 
			of the <a href="algo.html">GigAssembler</a> 
			algorithm used to assemble the working draft sequence, design and 
			implementation of the the 
			<a href="hgTracks.html">genome browser</a>, the 
			<a href="../cgi-bin/hgBlat?command=start">BLAT</a> sequence search tool, 
			and the comparative genomics software. FPC contigs, assembly, gap,
			coverage, known genes, full mRNA, spliced ESTs, human ESTs, and mouse 
			BLAT browser tracks.
		<li><a href="http://www.cse.ucsc.edu/~booch"><b>Terry Furey</b></a>, 
			Graduate student: Clone map analysis, STS browser tracks and STS 
			analysis (genetic, RH, FISH and YAC maps), BACend, cytogenetic band 
			and GC-content browser tracks.
		<li><a href="http://www.cse.ucsc.edu/~sugnet"><b>Chuck Sugnet</b></a>, 
			Graduate Student: Integration of functional genomics data, lifting of 
			annotation from one freeze to another, UNIGENE track, general browser 
			support
		<li><a href="http://www.cse.ucsc.edu/~rogic"><b>Sanja Rogic</b></a>, 
			Graduate student: E-PCR mapping of STSs, cDNA analysis, pair-EST track
		<li><a href="http://www.cse.ucsc.edu/~weiwu"><b>Wei Wu</b></a>, 
			Graduate student: Mirror sites, help with genetic maps
		<li><a href="http://www.cse.ucsc.edu/~xia"><b>Matt Xia</b></a>, 
			Graduate student: Computation of the statistics for the working draft, 
			scripts for phylogenetic data analysis, help with the mirror sites, 
			assembly testing
		<li><b>Mark Diekhans</b>, Graduate student: Mouse homology reference 
			sequence and phylogenetic analysis
		<li><b>Krishna Roskin</b>, Undergraduate student: Browser database support, 
			text-based browser
		<li><a href="http://people.ucsc.edu/~learithe"><b>Jenny Draper</b></a>, 
			Undergraduate student: Webmaster
		<li><b>Paul Tatarsky</b>, Staff: Hardware, networking, security, mirror 
			sites, general advice
		<li><b>Patrick Gavin</b>, Staff: Enhancing and running scripts to assemble 
			the genome, testing, compressing the genome to fit on a CD, design of 
			the kilocluster, general system support
		<li><a href="http://www.cse.ucsc.edu/~jgarcia"><b>Jorge Garcia</b></a>, 
			Staff: Co-design of the kilocluster. Web, mail, laptop and general 
			system sypport
		<li><b>Ann Pace</b>, Co-director of the 
			<a href="http://www.cse.ucsc.edu/centers/cbe">UCSC Center for 
			Biomolecular Science and Engineering</a>: Grants, budget, space and 
			other administrative support
		<li><b>Katy Elliot</b>, Staff: Administrative Support
		<li><a href="http://www.cse.ucsc.edu/~haussler"><b>David Haussler</b></a>, 
			Professor: Project Director 
	</ul>
</TD></TR><TR><TD WIDTH=10></TD><TD>
We gratefully acknowledge the help of former UCSC genome team members Scot Kennedy, Nick Littlestone, Aaron Tomb,
Degui Zhi, and Maria Guarino.<br>

</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<BR>
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Other Scientists whose work is featured at this site </FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD COLSPAN=2>
</TD></TR><TR><TD COLSPAN=2>
<ul>
<li><b>Ewan Birney, Guy Slater, Imre Vastrik, Arek Kasprzyk, Michele Clamp</b> 
	at <A HREF="http://www.ensembl.org/">Ensembl</A>:
	gene predictions, data exchange formats, TPF clone map coordiation, 
	integration with the Wash. U. fingerprint map,  
	exonerate mouse homology track, early help placing FISH-mapped clones. 
<li><b>Deanna Church</b> 
    at <A HREF="http://www.ncbi.nlm.nih.gov/index.html">NCBI</A>:
	Mouse synteny track.
<li><b>David Cox</b> 
	at <a href="http://www.perlegen.com">Perlegen</a>: STS markers for the         
	<A href="http://www-shgc.stanford.edu/Mapping/Marker/STSindex.html">
	TNG radiation hybrid map</A>.
<li><b>Ian Dunham</b> 
	at the <a href="http://www.sanger.ac.uk/HGP/">Sanger Centre </a>,  
	<b>Bill Grundy</b> and <b>Victoria Haghighi</b> at
	<a href="http://www.columbia.edu">Columbia University</a>: experimentally 
	verified gene annotations for chromosome 22 (Sanger 22 track).
<li><A href="http://www.genetics.wustl.edu/eddy/"><b>Sean Eddy</b></A> 
	at Wash. U. and <b>Todd Lowe</b> (now at UCSC): the RNA genes track.
<li><b>Evan Eichler</b> and <b>Jeff Bailey</b>
	at <A HREF="http://www.cwru.edu/">Case Western Reserve University</A>: 
	the segmental duplications track in the July 2000 browser.
<li><b>Richard Gibbs, John Bouck, Kim Worley and James Durbin</b> 
    at <a href="http://www.hgsc.bcm.tmc.edu/">Baylor College of Medicine</a>:
	CpG island track, simple repeats track.
<li><b>David Kulp, Alan Williams</b>, and <b>Ray Wheeler</b> 
	at <A HREF = "http://www.affymetrix.com">Affymetrix</A>:
	Genie gene prediction track for the browser, design and analysis support in 
	the early stages of the project, including software for the synthetic 
	chromosome 22 data we used to test our assembly methods.
<li><b>Eric Lander, Mike Zody, Chad Nusbaum</b> 
	at the <a href="http://www-genome.wi.mit.edu/">Whitehead Institute for
	Biomedical Research</a>: plasmid end reads used starting with the Sept. 
	2000 freeze. Special thanks to Eric for inviting us to join the Hardcore 
	Analysis Group for the draft sequence project, and for his direction of that 
	group. 
<li><b>Victor Pollara</b> at <a href="http://www.viaken.com/">Viaken</a> and <b>Arian Smit</b> at the 
	<a href="http://www.systemsbiology.org">Intsitute for Systems Biology</a>:
	repeat identification and masking.
<li><b>Thomas Pringle</b> at 
	the <A HREF="http://www.mad-cow.org/00/sci_archive_frame.html">
	Sperling Biomedical Foundation</A>: user's guides, email moderator, web links, 
	testing, myriad ideas for browser.
<li><b>Greg Schuler</b> at 
	<A HREF="http://www.ncbi.nlm.nih.gov/index.html">NCBI</A>: GenBank freezes of 
	human clones used in the assembly, 'NT contigs' of multiple finished clones,
	STS maps (in earlier freezes), specially formated versions of the finished 
	chromosomes 21 and 22. 
<li><b>Victor Solovyev</b> at 
	<A HREF="http://www.softberry.com/">Softberry Inc.</A>:
	Fgenesh++ gene predictions and TSSW promoters tracks.
<li><a href="http://stein.cshl.org/~lstein/"><b>Lincoln Stein </b></a> 
	at Cold Spring Harbor Laboratory, <b>Steve Sherry</b> at 
	<A HREF="http://www.ncbi.nlm.nih.gov/index.html">NCBI</A>, and the 
	SNP Consortium: SNP tracks.
<li><b>John Sulston, Jane Rogers, Jim Mullikin, and Richard Durbin</b> at the 
	<a href="http://www.sanger.ac.uk/HGP/">Sanger Centre </a>:  
	plasmid end reads starting with October 2000 freeze, help with gene 
	predictions, processing of whole genome shotgun mouse reads.
<li><b>Barbara Trask</b> at the 
	<a href="http://www.fhcrc.org/">Fred Hutchison Cancer Center</a>, 
	<b>Wonhee Jang</b> at
	<A HREF="http://www.ncbi.nlm.nih.gov/index.html">NCBI</A>, and the 
	BAC Resource Consortium:<A HREF="http://www.ncbi.nlm.nih.gov/genome/cyto/">
	Cytogenetic markers</A> (FISH-mapped clones).
<li><b>Robert Waterston, John McPhearson, Asif Chinwalla, Shiaw-Pyng Yang, John Wallis</b> and 
	<b>LaDeana Hillier</b> at Washington University in St. Louis: 
	<A HREF="http://genome.wustl.edu/gsc/human/Mapping"> fingerprint-based map</A> 
	of BAC clones from the human genome and a corresponding layout of sequenced 
	clones, plasmid end reads starting with October 2000 freeze, BACend analysis,
	CpG islands track.
<li><b>Lukas Wagner</b> at 
	<A HREF="http://www.ncbi.nlm.nih.gov/index.html">NCBI</A>: 
	3' EST track in the July 2000 browser.
<li><b>Jean Weissenbach</b> and <b>Olivier Jaillon</b> at 
	<A href="http://kicy.genoscope.cns.fr/cgi-bin/exofish_kicy.cgi">Genoscope
	</A>: Exofish track.

</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<BR>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Acknowledgements of Early Support  </FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>

<P>During the initial stage of this project, from January 2000 to June 2001,
we received no direct federal funding to develop the working draft other than travel 
support. However, we gratefully acknowledge the support for our personnel provided 
by our basic bioinformatics research grants, especially DOE grant
DE-FG03-99ER62849 (to develop genefinding methods), but also NSF grant DBI-9809007, 
UC BIOTECH grant 99-11, a grant from the Sloan Foundation, the David and Lucille 
Packard Foundation, NIH grant GM-52848, and the Zahler grant that provided support 
for Jim Kent in the early stages of this work. 
We are particularly thankful for the risks that Patrick Mantey, Dean of Engineering, 
John Simpson, UCSC Provost, and M.R.C. Greenwood, Chancellor, were willing to take 
in advancing us the money in February, 2000 needed to build the original 100-node 
LINUX cluster used in this work, and in the case of Dean Mantey, additional 
personnel support. Phokion Kolaitis, Chair of Computer Science, contributed as 
well. We thank <A HREF="http://www.compaq.com/">Compaq</A> for providing an 
additional 100-cpu computational cluster in Cambridge, MA for use
by the public human genome effort. We thank <A HREF="http://www.ilog.com/">ILOG,
Inc.</A> for providing discounted licenses to its CPLEX linear programming software, 
which was essential for some of our early experiments.</P>
<P>
Beginning in December, 2000, we started to utilize some support from 
<A HREF="http://www.hhmi.org">HHMI</a> for equipment and personnel. In the summer 
of 2001, HHMI provided the primary funding for a new cluster of 1000 CPUs running 
LINUX purchased from <a href="http://www.rackable.com">Rackable</a>, with additional 
support from <A HREF="http://www.ucop.edu/california-institutes">CISI</A> and 
<A HREF="http://www.nhgri.nih.gov">NHGRI</A>. Dubbed the 
UCSC KiloCluster, this equipment provides the computational backbone
for the project.   
</P>

</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>

</TD></TR></TABLE>
</BODY></HTML>


