<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>Human Genome Working Draft: Credits</TITLE>
	<LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 >
<TH COLSPAN=2 ALIGN="left">	
    <IMG SRC="../images/title.jpg">
	<!--<FONT FACE="Arial Black" SIZE=5 COLOR="#0033366">Human Genome Project Working Draft at UCSC</FONT>-->
</TH>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
   <CENTER>
	<TABLE BACKGROUND="../images/hl_mid.jpg" BGCOLOR="253BDE" WIDTH=100% CELLSPACING=0 CELLPADDING=0 BORDER=0 HEIGHT="22"><TR><TD>
		<TABLE WIDTH=431 CELLSPACING=4 CELLPADDING=0 BORDER=0><TR>
		<TD WIDTH=1></TD>
		<TD WIDTH=75>
			<A HREF="../index.html" onMouseOver="HOME.src='../images/hl_home.jpg'" onMouseOut="HOME.src='../images/h_home.jpg'">
			<IMG SRC="../images/h_home.jpg" ALIGN="absmiddle" BORDER="0" NAME="HOME" ALT="Home" ></A></TD>

		<TD WIDTH=155>
			<A HREF="hgTracks.html
" onMouseOver="BROWSER.src='../images/hl_browser.jpg'" onMouseOut="BROWSER.src='../images/h_browser.jpg'">
			<IMG SRC="../images/h_browser.jpg" ALIGN="absmiddle" BORDER="0" NAME="BROWSER" ALT="&nbsp;&nbsp; Genome Browser"></A></TD>
			
		<TD WIDTH=100>
			<A HREF="../cgi-bin/hgBlat?command=start" onMouseOver="SEARCH.src='../images/hl_blat.jpg'" onMouseOut="SEARCH.src='../images/h_blat.jpg'">
			<IMG SRC="../images/h_blat.jpg" ALIGN="absmiddle" BORDER="0" NAME="SEARCH" ALT="&nbsp;&nbsp; BLAT Search"></A></TD>
			
		<TD WIDTH=100>
			<A HREF="../FAQ.html" onMouseOver="GUIDE.src='../images/hl_faq.jpg'" onMouseOut="GUIDE.src='../images/h_faq.jpg'">
			<IMG SRC="../images/h_faq.jpg" ALIGN="absmiddle" BORDER="0" NAME="GUIDE" ALT="&nbsp;&nbsp; FAQ"></A></TD>
		</TR></TABLE>
	 </TR></TABLE>
	 </CENTER>
    </TD>
</TR>

<!--Content Tables-->  
  <TD CELLPADDING=10 ROWSPAN=2 VALIGN="top">
  	<IMG SRC="../images/DOT.gif" HEIGHT=5 WIDTH=1 ALT=""><BR>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="ffffff" BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
	<FONT SIZE="4"><B> &nbsp;Credits </B></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>

<P>The software to construct the working draft presented here was designed and implemented by Jim Kent, who is a
graduate student in Biology at UCSC, working with Prof. Alan Zahler on alternative splicing. Jim also
developed the genome browser. Tracks for the genome browser and other data have been provided by 
<A HREF="http://genome-test.cse.ucsc.edu/goldenPath/other.resources.html">many people</A>.
The user's guide for the browser was contributed by
Thomas Pringle of the <A
HREF="http://www.mad-cow.org/00/sci_archive_frame.html">
Sperling Biomedical Foundation</A>, Eugene, OR.</P>

<P>
Scot Kennedy, Terry Furey and Sanja Rogic, graduate students,
computed the statistics on the working draft, provided STS, GC-content and
cytogenetic band tracks for the browser, along with other browser enhancements,
and did the comparison
to the radiation hybrid, genetic, YAC, and cytogenetic maps.
Patrick Gavin, Paul Tatarsky and Jorge Garcia, our systems engineers, 
also contributed significantly to this project.</P>
Nick Littlestone, a visiting scientist at the 
<A HREF="http://www.cse.ucsc.edu/centers/cbe/"> 
UCSC Center for Biomolecular Science and Engineering (CBSE),</A> 
and Aaron Tomb, an undergraduate, helped in the early stages.
In January 2001, Degui Zhi took over the computation of the statistics for the working draft.
</P>

<P>We benefited significantly from our collaboration with <A
HREF="http://www.neomorphic.com">Neomorphic, Inc.</A>, a Berkeley genomics company, now Affymetrix Berkeley.
David Kulp, VP for Bioinformatics, provided design and analysis support in the early stages
of the project, and directed Ray Wheeler and Alan Williams at Neomorphic in the development of supporting tools, 
including software for the synthetic chromosome 22 data we used to test our methods.
More recently, they have provided a set of gene predictions that we display on our browser.</P>

<P>Ann Pace, Assistant Director of CBSE and Maria Guarino, Administrative Assistant, provided administrative support.</P>

<P>David Haussler, CBSE Director, Howard Hughes Medical Institute Investigator, and Professor of Computer Science, directed the overall project, in conjunction with Eric
Lander, director of the Genome Center at MIT's Whitehead Institute, Bob Waterston, director of the Genome Sequencing
Center at Washington University in St. Louis, Ewan Birney, director of the Ensembl project at the European Bioinformatics
Institute, Greg Schuler, scientist at the National Center for Biotechnology Information, and other
members of the human genome analysis working group. This project is part
of the larger public human genome effort, directed by Francis Collins.</P>

<P>We received no direct federal funding to develop the working draft other than travel support.
However, we gratefully acknowledge the support for our personnel provided by
our basic bioinformatics research grants, especially DOE grant
DE-FG03-99ER62849 (to develop genefinding methods), but also NSF
grant DBI-9809007, UC BIOTECH grant 99-11, a grant from the Sloan
Foundation, the David and Lucille Packard Foundation, NIH grant
GM-52848, and the Zahler grant that provided support for Jim Kent in the early stages
of this work. Beginning in December, 2000, we started to utilize some support from HHMI for
equipment and personnel. We are particularly thankful for the
risks that Patrick Mantey, Dean of Engineering, John Simpson, UCSC Provost, and
M.R.C. Greenwood, Chancellor, were willing to take in advancing us the money in February, 2000
needed to build the 100-node UCSC computational cluster used in this work, and
in the case of Dean Mantey, additional personnel support. Phokion Kolaitis,
Chair of Computer Science, contributed as well. We thank <A
HREF="http://www.compaq.com/">Compaq</A> for
providing an additional 100-cpu computational cluster in Cambridge, MA for use
by the public human genome effort. We thank <A HREF="http://www.ilog.com/">ILOG,
Inc.</A> for providing discounted licenses to its CPLEX linear programming software, which was essential for some of our
early experiments.</P></BODY>


	<BR></TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>	

</TD></TR></TABLE>
</BODY></HTML>




