<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>Human Genome Working Draft: Data Organization and Format</TITLE>
	<LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 >
<TH COLSPAN=2 ALIGN="left">	
    <IMG SRC="../images/title.jpg">
	<!--<FONT FACE="Arial Black" SIZE=5 COLOR="#0033366">Human Genome Project Working Draft at UCSC</FONT>-->
</TH>
<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
   <CENTER>
	<TABLE BGCOLOR="2636D1" WIDTH=100% CELLSPACING=0 CELLPADDING=0 BORDERCOLOR="000000" BORDER=1 HEIGHT="22"><TR><TD>
		<TABLE WIDTH=431 CELLSPACING=4 CELLPADDING=0 BORDER=0><TR>
		<TD WIDTH=1></TD>
		<TD WIDTH=75>
			<A HREF="../index.html" onMouseOver="HOME.src='../images/hl_home.jpg'" onMouseOut="HOME.src='../images/h_home.jpg'">
			<IMG SRC="../images/h_home.jpg" ALIGN="absmiddle" BORDER="0" NAME="HOME" ALT="Home" ></A></TD>

		<TD WIDTH=155>
			<A HREF="hgTracks.html
" onMouseOver="BROWSER.src='../images/hl_browser.jpg'" onMouseOut="BROWSER.src='../images/h_browser.jpg'">
			<IMG SRC="../images/h_browser.jpg" ALIGN="absmiddle" BORDER="0" NAME="BROWSER" ALT="&nbsp;&nbsp; Genome Browser"></A></TD>
			
		<TD WIDTH=100>
			<A HREF="../cgi-bin/hgBlat?command=start" onMouseOver="SEARCH.src='../images/hl_blat.jpg'" onMouseOut="SEARCH.src='../images/h_blat.jpg'">
			<IMG SRC="../images/h_blat.jpg" ALIGN="absmiddle" BORDER="0" NAME="SEARCH" ALT="&nbsp;&nbsp; BLAT Search"></A></TD>
			
		<TD WIDTH=100>
			<A HREF="../FAQ.html" onMouseOver="GUIDE.src='../images/hl_faq.jpg'" onMouseOut="GUIDE.src='../images/h_faq.jpg'">
			<IMG SRC="../images/h_faq.jpg" ALIGN="absmiddle" BORDER="0" NAME="GUIDE" ALT="&nbsp;&nbsp; FAQ"></A></TD>
		</TR></TABLE>
	 </TR></TABLE>
	 </CENTER>
    </TD>
</TR>

<!--Content Tables-->  
  <TD CELLPADDING=10 ROWSPAN=2 VALIGN="top">
  	<IMG SRC="../images/DOT.gif" HEIGHT=8 WIDTH=1 ALT=""><BR>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
	<FONT SIZE="4"><B> &nbsp;Data Organization and Format </B></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=15>&nbsp;&nbsp;</TD><TD>

<P>The data for the working draft is organized hierarchically by
chromosome and by the sequenced-clone layouts within each chromosome, as defined by the <A
HREF="http://genome.wustl.edu/gsc/human/Mapping">genome layout</A>.
At the top level there are 26 folders; 22 of these are for the numbered chromosomes (autosomes), folders X and Y
are for the  sex chromosomes, UL is for clone layouts that cannot be placed confidently on a chromosome, and NA is
for individual clones that cannot be placed. For simplicity, all clones in NA are treated as one "virtual clone
layout". Each of the chromosomal folders contains a separate clone layout folder for each of the clone layouts for that chromosome, and analogously for the UL and NA folders. </P>
<P>There are two primary files in each clone layout folder; these have suffixes .fa and .agp respectively. The .fa
files gives the working draft sequence for the clone layout. The format is Fasta format, e.g.</P>
<PRE>&gt;ctg124
GAATTCTTTCCCCATTTCTTGTTGTTGTCAGGTTTGTCAAAGATCAGATG
GTTGCAGATGTGTGGTATTATTTCTGAGGGCTCTGTTCCGTTCCATTGGT
CTATATTTCTGTTTTGGTACCAGTACCATGCTGTTTTGGTTACTGTAGCT
...</PRE>
<P>The .agp files is a kind of index that tells how the .fa file is built. It looks like</P>
<PRE>
18/ctg124       1       29863   1       P       AC024252.3      49331   79193   -
18/ctg124       29864   31063   2       P       AL353626.1      110043  111242  -
18/ctg124       31064   31514   3       P       AC024252.3      3453    3903    +
18/ctg124       31515   37083   4       P       AL353626.1      104105  109673  -
18/ctg124       37084   37183   5       N       100             fragment yes
18/ctg124       37184   50074   6       P       AC024252.3      17081   29971   +
...</PRE>
<P>Each line represents either an actual sequence record or a gap. If the line represents an actual sequence record then it has the form</P>
<P>&lt;chromosome/ctg&gt; &lt;start-in-ctg&gt; &lt;end-in-ctg&gt; &lt;number&gt; &lt;type&gt; &lt;accession&gt;.&lt;version&gt; &lt;start&gt; &lt;end&gt; &lt;orientation&gt;</P>
<P>and if it represents a gap it has the form</P>
<P>&lt;chromosome/ctg&gt; &lt;start-in-ctg&gt; &lt;end-in-ctg&gt; &lt;number&gt; N &lt;number-of-Ns&gt; &lt;kind&gt; &lt;bridged?&gt;</P>
<P>The positions &lt;start-in-ctg&gt; and &lt;end-in-ctg&gt; are the start and end positions for where the sequence
is to be put in the .fa file. For a sequence record, the positions &lt;start&gt; and &lt;end&gt; are the start and
end positions of where the sequence came from in the GenBank record &lt;accession&gt;.&lt;version&gt;. The field
&lt;orientation&gt; tells whether or not the sequence must be reverse complemented before it is inserted into its
place in the .fa file. For example, the records above mean that to build the .fa file for clone layout ctg124 from chromosome 18 you take</P>
<PRE>
AC024252 version 3, residues 49331 to 79193, reverse complemented, followed by 
AL353626 version 1, residues 110043 to 111242, reverse complemented, followed by
AC024252 version 3, residues 3453 to 3903, followed by
AL353626 version 1, residues 104105 to 109673, reverse complemented, followed by
a gap of 100 Ns, followed by
AC024252 version 3, residues 17081 to 29971, followed by
...</PRE>
<P>The joins perfectly abutt.  In a sequence record, &lt;type&gt; can be </P>
<P>     F - Finished, D - Draft, P - PreDraft, O - Other sequence </P>
<P>and in a gap record it is always N.  The &lt;number&gt; field just sequentially numbers the records.</P>
<P>In a gap record, &lt;number-of-Ns&gt; is the size of the gap and &lt;kind&gt; is</P>
<UL>
<LI>    fragment - a gap between two sequence contigs (also called a "sequence gap") 
<LI>    split_finished - a special sized gap between two finished sequence contigs 
<LI>    clone - a gap between two clones that do not overlap
<LI>    contig - a gap between clone layouts in the genome layout (also called a "layout gap")
<LI>    centromere - a gap inserted for the centromere 
<LI>    short_arm - a gap inserted at the start of an acrocentric chromosome 
<LI>    heterochromatin - a gap inserted for an especially large region of heterochromatin (may include
the centromere as well.) 
</UL>
<P>&lt;bridged?&gt; is "yes" if there is a cDNA or BACend pair or plasmid end pair that spans the gap, else it is "no".  # means a comment</P>
<P>Currently all fragment gaps are 100 Ns, all bridged clone gaps are 50,000 Ns, all unbridged clone gaps are
100,000 Ns, and all contig gaps are 200,000 Ns. </P>

<P>We provide three ways you can download these .fa and .agp files:  
<OL>
<LI> full data set: the entire hierarchy in a zipped format.
<LI> by chromosome: one zipped file for each chromosome containing all the sequence ordered along that
chromosome.
<LI> by individual clone layout: separate files, not zipped, for each clone layout.
</OL>
<BR></TD><TD WIDTH=15>&nbsp;&nbsp;</TD></TR></TABLE>
	</TD></TR></TABLE>
		
</TD></TR></TABLE>
</BODY></HTML>






