<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>Human Genome Working Draft: Genome Browser Database</TITLE>
	<LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0>
<TH COLSPAN=2 ALIGN="left">	
    <IMG SRC="../images/title.jpg">
	<!--<FONT FACE="Arial Black" SIZE=5 COLOR="#0033366">Human Genome Project Working Draft at UCSC</FONT>-->
</TH>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
   <CENTER>
	<TABLE BACKGROUND="../images/hl_mid.jpg" BGCOLOR="253BDE" WIDTH=100% CELLSPACING=0 CELLPADDING=0 BORDER=0 HEIGHT="22"><TR><TD>
		<TABLE WIDTH=431 CELLSPACING=4 CELLPADDING=0 BORDER=0><TR>
		<TD WIDTH=1></TD>
		<TD WIDTH=75>
			<A HREF="../index.html" onMouseOver="HOME.src='../images/hl_home.jpg'" onMouseOut="HOME.src='../images/h_home.jpg'">
			<IMG SRC="../images/h_home.jpg" ALIGN="absmiddle" BORDER="0" NAME="HOME" ALT="Home" ></A></TD>

		<TD WIDTH=155>
			<A HREF="hgTracks.html" onMouseOver="BROWSER.src='../images/hl_browser.jpg'" onMouseOut="BROWSER.src='../images/h_browser.jpg'">
			<IMG SRC="../images/h_browser.jpg" ALIGN="absmiddle" BORDER="0" NAME="BROWSER" ALT="&nbsp;&nbsp; Genome Browser"></A></TD>
			
		<TD WIDTH=100>
			<A HREF="../cgi-bin/hgBlat?command=start" onMouseOver="SEARCH.src='../images/hl_blat.jpg'" onMouseOut="SEARCH.src='../images/h_blat.jpg'">
			<IMG SRC="../images/h_blat.jpg" ALIGN="absmiddle" BORDER="0" NAME="SEARCH" ALT="&nbsp;&nbsp; BLAT Search"></A></TD>
			
		<TD WIDTH=100>
			<A HREF="../FAQ.html" onMouseOver="GUIDE.src='../images/hl_faq.jpg'" onMouseOut="GUIDE.src='../images/h_faq.jpg'">
			<IMG SRC="../images/h_faq.jpg" ALIGN="absmiddle" BORDER="0" NAME="GUIDE" ALT="&nbsp;&nbsp; FAQ"></A></TD>
		</TR></TABLE>
	 </TR></TABLE>
	 </CENTER>
    </TD>
</TR>

<!--Content Tables-->  
  <TD CELLPADDING=10 ROWSPAN=2 VALIGN="top">
  	<IMG SRC="../images/DOT.gif" HEIGHT=5 WIDTH=1 ALT=""><BR>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="ffffff" BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
	<FONT SIZE="4"><B> &nbsp;The Genome Browser Database </B></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
		This page describes the format of the genome annotation databases that underlie the 
		<A href="hgTracks.html">Human Genome Browser</A>. 
		This data was contributed by many researchers, as described in 
		<A href="other.resources.html">Data Sources</A>. 
		Please acknowledge the contributor(s) of the data you use. All of the tables are 
		freely usable for any purpose except as indicated in the README.txt file in the 
		database directory. 
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<BR>
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="ffffff" BACKGROUND="../images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><b>&nbsp; Table Descriptions</FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
<P>The data displayed in the <A href="http://genome.ucsc.edu/goldenPath/hgTracks.html">
Human Genome Browser</A> 
is stored in a <A href="http://www.mysql.com/">MySQL</A> database. The tables in 
the database can be grouped into three categories. The first are ones for which 
the data has been split into separate tables for each chromosome. The second 
group has tables that contain information for all chromosomes. The third group 
has tables that contain information specifically related to mRNA sequences.</P>
<P>All of the coordinates in these tables are of "half-open zero based." This 
means that the first 100 bases of a chromosome are represented as 0-100. The 
second 100 bases are represented as 100-200, and so forth.</P>
<P>The database currently includes the following tables for each chromosome 
(chrN_[table] where N is replaced by the chromosome name). In many cases there 
is an additional table for the unordered part (random) of the chromosome 
(chrN_random_[table]). Finally, in addition to the standard chromosome names 1, 
..., 22, X, and Y, there are two special chromosome names, UL and NA, 
representing fingerprint clone contigs that could not be placed confidently on a 
chromosome, and individual clones that could not be placed, respectively. The 
table names are hyperlinked to an AutoSql format description of their 
fields.</P>
<UL>
  <LI><A 
  href="#PslInfo">chrN_mrna</A> 
  - mRNA alignments 
  <LI><A 
  href="#PslInfo">chrN_est</A> 
  - EST alignments 
  <LI><A 
  href="#PslInfo">chrN_intronEst</A> 
  - EST alignments in introns 
  <LI><A 
  href="#PslInfo">chrN_blatMouse</A> 
  - Mouse whole genome shotgun alignments 
  <LI><A 
  href="#GapInfo">chrN_gap</A> 
  - gap locations and types 
  <LI><A 
  href="#FragInfo">chrN_gl</A> 
  - golden path positions of all clone fragments 
  <LI><A 
  href="#GpInfo">chrN_gold</A> 
  - instructions on creating golden path from fragments 
  <LI><A 
  href="#RepeatInfo">chrN_rmsk</A> 
  - repeating elements created using RepeatMasker </LI>
</UL>

<P>The following tables contain data that is placed on the golden path, but is 
stored in a single table rather than one table per chromosome:</P>
<UL>
  <LI><A 
  href="#PslInfo">all_est</A> 
  - EST alignments (concatenation of chr*_est) 
  <LI><A 
  href="#PslInfo">all_mrna</A> 
  - mRNA alignments (concatenation of chr*_mrna) 
  <LI><A 
  href="#ChromInfo">chromInfo</A> 
  - describes name and size of each chromosome as well as an external file 
  containing the chromosome 
  <LI><A 
  href="#CloneInfo">clonePos</A> 
  - positions of all clones in golden path. Also HTG phase, 
  finished/draft/predraft stage, and sequence size. 
  <LI><A 
  href="#CpG Islands">cpgIsland</A> 
  - CpG islands contributed by Baylor University and Washington University 
  <LI><A 
  href="#ContigInfo">ctgPos</A> 
  - the name, size, and golden path position of all FPC contigs. 
  <LI><A 
  href="#ChromosomeBand">cytoBand</A> 
  - Terry Furey's mapping of the 800 band resolution cytogenetic/chromosome 
  bands to the golden path using FISH clone information from the BAC Resource 
  Consortium. 
  <LI><A 
  href="#Exofish">exoFish</A> 
  - Genescope Exofish Tetraodon evolutionarily conserved regions. 
  <LI><A 
  href="#ExoMouse">exoMouse</A> 
  - Exonerate mouse/human evolutionarily conserved regions. 
  <LI><A 
  href="#GCpercent">gcPercent</A> 
  - average number of G and C per 1000 bases in non-overlapping 20Kb windows 
  <LI><A 
  href="#GenePredictions">genieAlt</A> 
  - Genie gene predictions - all constrained by one degree or another by EST or 
  mRNA. 
  <LI><A 
  href="#PepPredictions">genieAltPep</A> 
  - translations of Genie gene predictions into its corresponding amino acid 
  sequence 
  <LI><A 
  href="#GenePredictions">genieKnown</A> 
  - known genes processed a little by Genie to allow sequencing error. (Oct. 
  2000 and earlier known gene track). 
  <LI><A 
  href="#PepPredictions">genieKnownPep</A> 
  - translations of known genes into its corresponding amino acid sequence (Oct. 
  2000 and earlier known gene track). 
  <LI><A 
  href="#OtherKnown">knownInfo</A> 
  - other information about known genes (Oct. 2000 and earlier known gene 
  track). 
  <LI><A 
  href="#MoreKnown">knownMore</A> 
  - information about links for known genes to other databases such as to 
  GenBank, OMIN, HUGO, RefSeq, GBD. (Oct. 2000 and earlier known gene track). 
  <LI><A 
  href="#GenePredictions">refKnown</A> 
  - known genes as derived from alignments of RefSeq mRNA (Dec 2000 and later 
  known gene track). 
  <LI><A 
  href="#PepPredictions">refPep</A> 
  - RefSeq translated protiens (Dec. 2000 and later) 
  <LI><A 
  href="#MrnaPredictions">refMrna</A> 
  - RefSeq mRNA (Dec. 2000 and later) 
  <LI><A 
  href="#RefLink">refLink</A> 
  - Relates RefSeq mRNA accession to LocusLink, HUGO, etc. (Dec. 2000 and later) 

  <LI><A 
  href="#MouseSyn">mouseSyn</A> 
  - synteny between mouse and human chromosomes 
  <LI><A 
  href="#RNAgenes">rnaGene</A> 
  - functional RNA genes 
  <LI><A 
  href="#SimpleRepeats">simpleRepeats</A> 
  - simple tandem repeats generated with the TRF program 
  <LI><A 
  href="#SNPsites">snpNih</A> 
  - locations of overlap SNPs as determined by the SNP Consortium 
  <LI><A 
  href="#SNPsites">snpTsc</A> 
  - locations of random SNPs as determined by the SNP Consortium 
  <LI><A 
  href="#STSsites">stsMarker</A> 
  - STS marker mappings from various maps and FISH clone mappings from the BAC 
  Resource Consortium </LI>
</UL>

<P>The following tables contain additional information about mRNA sequences: 
<UL>
  <LI><A 
  href="#MrnaInfo">mrna</A> 
  - very relational mRNA and EST information table. Consists mostly of the mRNA 
  accession and pointers into the tables below. 
  <LI><A 
  href="#MrnaInfo">author</A> 
  - list of authors 
  <LI><A 
  href="#MrnaInfo">cds</A> 
  - coding sequence in Genbank format 
  <LI><A 
  href="#MrnaInfo">cell</A> 
  - cell types 
  <LI><A 
  href="#MrnaInfo">center</A> 
  - sequencing center 
  <LI><A 
  href="#MrnaInfo">cytoMap</A> 
  - position on a cytological map 
  <LI><A 
  href="#MrnaInfo">description</A> 
  - a short (one line) description 
  <LI><A 
  href="#MrnaInfo">development</A> 
  - stage of development of organism 
  <LI><A 
  href="#MrnaInfo">geneName</A> 
  - name of gene 
  <LI><A 
  href="#MrnaInfo">keyword</A> 
  - associated keywords in Genbank 
  <LI><A 
  href="#MrnaInfo">library</A> 
  - clone library (mostly for ESTs) 
  <LI><A 
  href="#MrnaInfo">mrnaClone</A> 
  - clone name (mostly for ESTs) 
  <LI><A 
  href="#MrnaInfo">organism</A> 
  - name of organism 
  <LI><A 
  href="#MrnaInfo">productName</A> 
  - protein product name 
  <LI><A 
  href="#MrnaInfo">sex</A> 
  - sex of organism 
  <LI><A 
  href="#MrnaInfo">source</A> 
  - source field in Genbank, often fairly redundant with other fields 
  <LI><A 
  href="#MrnaInfo">tissue</A> 
  - tissue isolated from </LI></UL>
<P>The last three tables are primarily for internal use:</P>
<UL>
  <LI><A 
  href="#seqInfo">seq</A> 
  - describes where mRNA sequence lives. Includes reference to an external file. 

  <LI><A 
  href="#extFileInfo">extFile</A> 
  - references to external files, again only used by RNA. 
  <LI><A 
  href="#HistoryInfo">history</A> 
  - modification history of database, only used by RNA programs </LI>
</UL>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<BR>

	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="ffffff" BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
	<FONT SIZE="4"><B> &nbsp;<A name=TableFormats></A>Table Formats </B></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10>&nbsp;</TD><TD>

<P>This section describes the format of each table field by field in AutoSql 
format.</P>
<H3><A name=ChromInfo></A>Chromosome Information</H3>
<PRE>table chromInfo
"Chromosome names and sizes"
    (
    string chrom;       "Chromosome name"
    uint   size;        "Chromosome size"
    string fileName;    "Chromosome file (raw one byte per base)"
    )
</PRE>
<H3><A name=ChromosomeBand></A>Chromosome Band Info</H3>
<PRE>table cytoBand
"Describes the positions of cytogenetic bands within a chromosome"
   (
   string chrom;        "Human chromosome number"
   uint   chromStart;   "Start position in genoSeq"
   uint   chromEnd;     "End position in genoSeq"
   string name;         "Name of cytogenetic band"
   string gieStain;     "Giesma stain results"
   )
</PRE>
<H3><A name=FragInfo></A>Clone Fragment Positions</H3>
<PRE>table chrN_gl
"Fragment positions in golden path"
    (
    string  frag;       "Fragment name"
    uint    start;      "Start position in golden path"
    uint    end;        "End position in golden path"
    char[1] strand;     "+ or - for strand"
    )
</PRE>
<H3><A name=CloneInfo></A>Clone Information</H3>
<PRE>table clonePos
"A clone's position and other info."
    (
    string  name;       "Name of clone including version"
    uint    seqSize;    "base count not including gaps"
    ubyte   phase;      "htg phase"
    string  chrom;      "Chromosome name"
    uint    chromStart; "Start in chromosome"
    uint    chromEnd;   "End in chromosome"
    char[1] stage;      "F/D/P for finished/draft/predraft"
    string  faFile;     "File with sequence."
    )
</PRE>
<H3><A name=ContigInfo></A>Contig Information</H3>
<PRE>table ctgPos
"Where a contig is inside of a chromosome."
    (
    string contig;      "Name of contig"
    uint   size;        "Size of contig"
    string chrom;       "Chromosome name"
    uint   chromStart;  "Start in chromosome"
    uint   chromEnd;    "End in chromosome"
    )
</PRE>
<H3><A name="CpG Islands"></A>CpG Islands</H3>
This table contains CpG islands of 50 bases or more which are at least 50% 
G/C.
<PRE>table cpgIsland
"Describes the CpG Islands"
   (
   string chrom;        "Human chromosome or FPC contig"
   uint   chromStart;   "Start position in chromosome"
   uint   chromEnd;     "End position in chromosome"
   string name;         "CpG Island"
   uint   minLength;    "Minimum Island Length"
   uint   cpgNum;       "Number of CpGs in island"
   uint   gcNum;        "Number of C and G in island"
   uint   perCpg;       "Percentage of island that is CpG"
   uint   perGc;        "Percentage of island that is C or G"
   )
</PRE>
<H3><A name=Exofish></A>Exofish Information</H3>
<PRE>table exoFish
"An evolutionarily conserved region (ecore) with Tetroadon"
   (
   string chrom;        "Human chromosome or FPC contig"
   uint   chromStart;   "Start position in chromosome"
   uint   chromEnd;     "End position in chromosome"
   string name;         "Just a dummy dot"
   uint   score;        "Score from 0 to 1000"
   )
</PRE>
<H3><A name=ExoMouse></A>Exonerate Mouse Homologies</H3>
<PRE>table exoMouse
"A rough alignment - not detailed"
   (
   string chrom;        "Human chromosome or FPC contig"
   uint   chromStart;   "Start position in chromosome"
   uint   chromEnd;     "End position in chromosome"
   string name;         "Name of other sequence"
   uint   score;        "Score from 0 to 1000"
   char[1] strand;      " or -"
   uint otherStart;     "Start in other sequence"
   uint otherEnd;       "End in other sequence"
   )
</PRE>
<H3><A name=GapInfo></A>Gap Positions</H3>
<PRE>table chrN_gap
"Gaps in golden path"
    (
    string chrom;       "which chromosome"
    uint   chromStart;  "start position in chromosome"
    uint   chromEnd;    "end position in chromosome"
    int    ix;          "ix of this fragment (useless)"
    char[1] n;          "always 'N'"
    uint   size;        "size of gap"
    string type;        "contig, clone, fragment, etc."
    string bridge;      "yes, no, mrna, bacEndPair, etc."
    )
</PRE>
<H3><A name=GCpercent></A>GC Content</H3>
<PRE>table gcPercent
"Displays GC content in 20Kb non-overlapping blocks for genome"
   (
   string chrom;        "Human chromosome number"
   uint   chromStart;   "Start position in chromosome"
   uint   chromEnd;     "End position in chromosome"
   string name;         "Constant string GC"
   uint   gcPpt;        ":Average number of G and C per 1000 bases"
   )                                                                            
</PRE>
<H3><A name=GenePredictions></A>Gene Predictions and Known Genes</H3>
The following definition is used for the refGene, ensGene, genieAlt and similar tables. In alternative splicing situations
each transcript has a row in this table.
<PRE>table genePred
"A gene prediction."
    (
    string  name;       "Name of gene"
    string  chrom;      "Chromosome name"
    char[1] strand;     "+ or - for strand"
    uint    txStart;    "Transcription start position"
    uint    txEnd;      "Transcription end position"
    uint    cdsStart;   "Coding region start"
    uint    cdsEnd;     "Coding region end"
    uint    exonCount;  "Number of exons"
    uint[exonCount] exonStarts; "Exon start positions"
    uint[exonCount] exonEnds;   "Exon end positions"
    )
</PRE>

<H3><A name=PepPredictions></A>Gene Predictions and Known Genes Amino Acid Translations</H3>
<PRE>table pepPred
"A predicted peptide - linked to a predicted gene."
    (
    string  name;       "Name of gene - same as in genePred"
    lstring seq;        "Peptide sequence"
    )
</PRE>

<H3><A name=MrnaPredictions></A>mRNA sequences</H3>
<PRE>table mrnaPred
"A mRNA - linked to a gene."
    (
    string  name;       "Name of gene - same as in genePred"
    lstring seq;        "mRNA sequence"
    )
</PRE>


<H3><A name=RefLink></A>Link between RefSeq mRNAs and HUGO, LocusLink etc.</H3>
First used Dec. 2000
<PRE>table refLink
"Link together a refseq mRNA and other stuff"
    (
    string name;        "Name displayed in UI"
    string product;     "Name of protein product"
    string mrnaAcc;     "mRNA accession"
    string protAcc;     "protein accession"
    uint geneId;        "pointer to geneName table"
    uint prodId;        "pointer to prodName table"
    uint locusLinkId;   "Locus Link ID"
    uint omimId;        "OMIM ID"
    )
</PRE>


<H3><A name=MoreKnown></A>Known Gene IDs in Other Databases</H3>
Last used Oct. 2000
<PRE>table knownMore
"Lots of auxiliary info about a known gene"
    (
    string name;          "The name displayed in the browser: 
	                       OMIM, gbGeneName, or transId"
    string transId;       "Transcript id. Genie generated ID."
    string geneId;        "Gene (not transcript) Genie ID"
    uint   gbGeneName;    "Connect to geneName table. Genbank gene name"
    uint   gbProductName; "Connects to productName table. Genbank product name"
    string gbProteinAcc;  "Genbank accession of protein"
    string gbNgi;         "Genbank gi of nucleotide seq."
    string gbPgi;         "Genbank gi of protein seq."
    uint   omimId;        "OMIM ID or 0 if none"
    string omimName;      "OMIM primary name"
    uint   hugoId;        "HUGO Nomeclature Committee ID or 0 if none"
    string hugoSymbol;    "HUGO short name"
    string hugoName;      "HUGO descriptive name"
    string hugoMap;       "HUGO Map position"
    uint   pmId1;         "I have no idea - grabbed from a HUGO nomeids.txt"
    uint   pmId2;         "Likewise, I have no idea"
    string refSeqAcc;     "Accession of RefSeq mRNA"
    string aliases;       "Aliases if any.  Comma and space separated list"
    uint   locusLinkId;   "Locus link ID"
    string gdbId;         "NCBI GDB database ID"
    )                                                                              
</PRE>
<H3><A name=OtherKnown></A>Other Known Gene Information</H3>
Last used Oct. 2000
<PRE>table knownInfo
"Auxiliary info about a known gene"
    (
    string name;         "connects with genieKnown-&gt;name"
    string transId;      "Transcript id. Genie generated ID."
    string geneId;       "Gene (not transcript) id"
    uint   geneName;     "Connect to geneName table"
    uint   productName;  "Connects to productName table"
    string proteinId;    "Genbank accession of protein?"
    string ngi;          "Genbank gi of nucleotide seq."
    string pgi;          "Genbank gi of protein seq."
    )                                                                           
</PRE>

<H3><A name=GpInfo></A>Golden Path Construction</H3>
<PRE>table chrN_gold
"How to get through chromosome based on fragments"
    (
    string  chrom;      "which chromosome"
    uint    chromStart; "start position in chromosome"
    uint    chromEnd;   "end position in chromosome"
    int     ix;         "ix of this fragment (useless)"
    char[1] type;       "(P)redraft, (D)raft, (F)inished or (O)ther"
    string  frag;       "which fragment"
    uint    fragStart;  "start position in frag"
    uint    fragEnd;    "end position in frag"
    char[1] strand;     "+ or - (orientation of fragment)"
    )
</PRE>

<H3><A name=MouseSyn></A>Mouse Synteny</H3>
<PRE>table mouseSyn
"Synteny between mouse and human chromosomes."
    (
    string  chrom;       "Name of chromosome"
    uint    chromStart;  "Start in chromosome"
    uint    chromEnd;    "End in chromosome"
    string  name;        "Name of mouse chromosome"
    int     segment;     "Number of segment"
    )
</PRE>

<H3><A name=PslInfo></A>mRNA/EST/Mouse Alignments</H3>
The following definition is used for the chrN_mrna, chrN_est, chrN_intronEst, 
all_mrna and all_est tables.
<PRE>table psl
"Summary info about a patSpace alignment"
    (
    uint    matches;      "Number of bases that match that aren't repeats"
    uint    misMatches;   "Number of bases that don't match"
    uint    repMatches;   "Number of bases that match but are part of repeats"
    uint    nCount;       "Number of 'N' bases"
    uint    qNumInsert;   "Number of inserts in query"
    int     qBaseInsert;  "Number of bases inserted in query"
    uint    tNumInsert;   "Number of inserts in target"
    int     tBaseInsert;  "Number of bases inserted in target"
    char[2] strand;       "+ or - for query strand. 
	                       For mouse second +/- for genomic strand"
    string  qName;        "Query sequence name"
    uint    qSize;        "Query sequence size"
    uint    qStart;       "Alignment start position in query"
    uint    qEnd;         "Alignment end position in query"
    string  tName;        "Target sequence name"
    uint    tSize;        "Target sequence size"
    uint    tStart;       "Alignment start position in target"
    uint    tEnd;         "Alignment end position in target"
    uint    blockCount;   "Number of blocks in alignment"
    uint[blockCount] blockSizes;  "Size of each block"
    uint[blockCount] qStarts;     "Start of each block in query."
    uint[blockCount] tStarts;     "Start of each block in target."
    )
</PRE>
There is a little gotcha in this table format. It has to 
do with how coordinates are handled on the negative strand.
In the qStart/qEnd fields the coordinates are where it
matches from the point of view of the forward strand 
(even when the match is on the reverse strand). However
on the qStarts[] list, the coordinates are reversed. 
Here's an example of a 30-mer that has 2 blocks that
align on the minus strand and 2 blocks on the plus
strand (this sort of stuff happens in real life in
response to assembly errors sometimes). 
<PRE>
0         1         2         3 tens position in query
0123456789012345678901234567890 ones position in query
            ++++          +++++ plus strand alignment on query
    --------    ----------      minus strand alignment on query
	
Plus strand:
     qStart 12 qEnd 31 blockSizes 4,5 qStarts 12,26
Minus strand:
     qStart 4 qEnd 26 blockSizes 10,8 qStarts 5,19 
</PRE>
Essentially the minus strand blockSizes and qStarts are 
what you would get if you reverse complemented the query.
However the qStart and qEnd are non-reversed. To get
from one to the other:
<PRE>
     qStart = qSize - revQEnd
     qEnd = qSize - revQStart
</PRE>

<H3><A name=RepeatInfo></A>Repeating Elements</H3>
These tables were created from RepeatMasker .out files.
<PRE>table chrN_rmsk
"RepeatMasker .out record"
    (
    uint    swScore;    "Smith Waterman alignment score"
    uint    milliDiv;   "Base mismatches in parts per thousand"
    uint    milliDel;   "Bases deleted in parts per thousand"
    uint    milliIns;   "Bases inserted in parts per thousand"
    string  genoName;   "Genomic sequence name"
    uint    genoStart;  "Start in genomic sequence"
    uint    genoEnd;    "End in genomic sequence"
    int     genoLeft;   "Size left in genomic sequence"
    char[1] strand;     "Relative orientation + or -"
    string  repName;    "Name of repeat"
    string  repClass;   "Class of repeat"
    string  repFamily;  "Family of repeat"
    int     repStart;   "Start in repeat sequence"
    uint    repEnd;     "End in repeat sequence"
    int     repLeft;    "Size left in repeat sequence"
    char[1] id;         "'*' or ' '.  I don't know what this means"
    )
</PRE>

<H3><A name=RNAgenes></A>RNA Gene Positions</H3>
<PRE>table rnaGene
"Describes functional RNA genes."
    (
    string  chrom;        "Chromosome gene is on"
    uint    chromStart;   "Start position in chromosome"
    uint    chromEnd;     "End position in chromosome"
    string  name;         "Name of gene"
    uint    score;        "Score from 0 to 1000"
    char[1] strand;       "Strand: + or -"
    string  source;       "Source as in Sean Eddy's files."
    string  type;         "Type - snRNA, rRNA, tRNA, etc."
    float   fullScore;    "Score as in Sean Eddy's files."
    ubyte   isPsuedo;     "TRUE(1) if psuedo, FALSE(0) otherwise"
    )                                                                           
</PRE>

<H3><A name=SimpleRepeats></A>Simple Repeats</H3>
This table was generated by the trf program.  It does a more thorough job 
of finding simple tandem repeats than RepeatMaskerdoes.
<PRE>table simpleRepeat
"Describes the Simple Tandem Repeats"
   (
   string chrom;          "Human chromosome or FPC contig"
   uint   chromStart;     "Start position in chromosome"
   uint   chromEnd;       "End position in chromosome"
   string name;           "Simple Repeats tag name"
   uint   period;         "Length of repeat unit"
   float  copyNum;        "Mean number of copies of repeat"
   uint   consensusSize;  "Length of consensus sequence"
   uint   perMatch;       "Percentage Match"
   uint   perIndel;       "Percentage Indel"
   uint   score;          "Score between  and .  Best is ."
   uint   A;              "Number of A's in repeat unit"
   uint   C;              "Number of C's in repeat unit"
   uint   T;              "Number of T's in repeat unit"
   uint   G;              "Number of G's in repeat unit"
   float  entropy;        "Entropy"
   string sequence;       "Sequence of repeat unit element"
   )
</PRE>

<H3><A name=SNPsites></A>SNP Positions</H3>
This table is used for both the snpNih and snpTsc tables.
<PRE>table snp
"Single nucleotide polymorphisms"
   (
   string chrom;          "Human chromosome or FPC contig"
   uint   chromStart;     "Start position in chromosome"
   uint   chromEnd;       "End position in chromosome"
   string name;           "Name of SNP"
   )                                                                           
</PRE>

<H3><A name=STSsites></A>STS and FISH clone Positions</H3>
<PRE>table stsMarker
"STS marker and its position on golden path and various maps"
   (
   string chrom;            "Chromosome or 'unknown'"
   int    chromStart;       "Start position in chrom - negative 1 if unpositioned"
   uint   chromEnd;         "End position in chrom"
   string name;             "Name of STS marker"
   uint   score;            "Score of a marker - depends on how many contigs it hits"
   uint   identNo;          "Identification number of STS"
   string ctgAcc;           "Contig accession number"
   string otherAcc;         "Accession number of other contigs that the marker hits" 
   string genethonChrom;    "Chromosome from Genethon map or 0 if none"
   float  genethonPos;      "Position on Genethon map"
   string marshfieldChrom;  "Chromosome from Marshfield map or 0 if none"
   float  marshfieldPos;    "Position on Marshfield map"
   string gm99Gb4Chrom;     "Chromosome from GeneMap99 map or 0 if none"
   float  gm99Gb4Pos;       "Position on GeneMap99 map"
   string shgcG3Chrom;      "Chromosome from G3 map or 0 if none"
   float  shgcG3Pos;        "Position on G3 map"
   string wiYacChrom;       "Chromosome from Whitehead YAC map or 0 if none"
   float  wiYacPos;         "Position on Whitehead YAC map"
   string shgcTngChrom;     "Chromosome from TNG map or 0 if none"
   float  shgcTngPos;       "Position on TNG map"
   string fishChrom;        "Chromosome from FISH map or 0 if none"
   string beginBand;        "Beginning of range of bands on FISH map"
   string endBand;          "End of range of bands on FISH map"
   string lab;              "Laboratory that placed the FISH clone"
   )                                                                                                      
</PRE>

	<BR></TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>

	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="ffffff" BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
	<FONT SIZE="4"><B><A name=MrnaInfo></A>&nbsp; mRNA and EST Data</B></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10>&nbsp;</TD><TD>

This is a family of tables. The main table, <TT>mrna</TT>, consists of minor mRNA and EST identification
information, and then ids pointing to the corresponding tables which contain other information.  For example,
to determine the organism, the 'organism' field in the <TT>mrna</TT> table contains an id. Finding this id
in the corresponding <TT>organism</TT> table will give the name of the organism.


<H4>Main mRNA/EST Information Table</H4>
<PRE>table mrna
"Basic information about mRNA and ESTs with links to other tables"
   (
   uint   id               "Id, same as seq ID"
   string acc              "Genbank accession"
   enum   type             "Full length ('mRNA') or EST ('EST')"
   enum   direction        "Read direction ('5','3','0')"
   uint   source           "Ref in source table"
   uint   organism         "Ref in organism table"
   uint   library          "Ref in library table"
   uint   mrnaClone        "Ref in clone table"
   uint   sex              "Ref in sex table"
   uint   tissue           "Ref in tissue table"
   uint   development      "Ref in development table"
   uint   cell             "Ref in cell table"
   uint   cds              "Ref in CDS table"
   uint   keyword          "Ref in key table"
   uint   description      "Ref in description table"
   uint   geneName         "Ref in geneName table"
   uint   productName      "Ref in productName table"
   uint   author           "Ref in author table"
   )
</PRE>

<H4>mRNA/EST Characteristic Tables</H4>
This table is used for the author, cds, cell, center, cytoMap, description, development, geneName,
keyword, library, mrnaClone, organism, productName, sex, source, and tissue tables.  Each has a 
reference in the above mrna table.
<PRE>table mrnaCharateristic
"Characteristics of mRNA and EST referenced in the mrna table"
   (
   int    id               "Unique id of characteristic in each table"
   string name             "Characteristic"
   )
</PRE>

	<BR></TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>

	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="ffffff" BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
	<FONT SIZE="4"><B>&nbsp; Internal Use Tables</B></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10>&nbsp;</TD><TD>

<H4><A name=extFileInfo></A>Reference to an External File</H4>
<PRE>table extFile 
"This keeps track of external files and directories"
   (
   uint   id           "Unique ID for each file"
   string name         "Symbolic name of file"
   string path         "Full path. Ends in '/' if a dir"
   uint   size         "Size of file (checked)"
   )
</PRE>

<H4><A name=seqInfo></A>Reference to an External Sequence in an FA File</H4>
<PRE>table seq
"This keeps track of a sequence"
  (
  uint   id           "Unique ID for each external sequence"
  string acc          "GenBank accession number"
  uint   size         "Size of sequence in bases"
  date   gb_date      "GenBank last modified date"
  uint   extFile      "File it is in"
  uint   file_offset  "Offset in file"
  uint   file_size    "Size in file."
  )
</PRE>

<H4><A name=HistoryInfo></A>History Table - Tracks Modification History of mRNA Table and Linked Tables</H4>
<PRE>table history
  (
  int       ix        "Update number"
  uint      startId   "Start this session's ids"
  uint      endId     "First id for next session"
  string    who       "User who updated"
  string    what      "What they did"
  timestamp when      "When they did it"
  )
</PRE>

	<BR></TD><TD WIDTH=7></TD></TR></TABLE>
	</TD></TR></TABLE>
			
</TD></TR></TABLE>
</BODY></HTML>