<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>Human Genome Working Draft: News Archives</TITLE>
	<LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 >
<TH COLSPAN=2 ALIGN="left">	
    <IMG SRC="../images/title.jpg">
	<!--<FONT FACE="Arial Black" SIZE=5 COLOR="#0033366">Human Genome Project Working Draft at UCSC</FONT>-->
</TH>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
   <CENTER>
	<TABLE BACKGROUND="../images/hl_mid.jpg" BGCOLOR="253BDE" WIDTH=100% CELLSPACING=0 CELLPADDING=0 BORDER=0 HEIGHT="22"><TR><TD>
		<TABLE WIDTH=431 CELLSPACING=4 CELLPADDING=0 BORDER=0><TR>
		<TD WIDTH=1></TD>
		<TD WIDTH=75>
			<A HREF="../index.html" onMouseOver="HOME.src='../images/hl_home.jpg'" onMouseOut="HOME.src='../images/h_home.jpg'">
			<IMG SRC="../images/h_home.jpg" ALIGN="absmiddle" BORDER="0" NAME="HOME" ALT="Home" ></A></TD>

		<TD WIDTH=155>
			<A HREF="hgTracks.html" onMouseOver="BROWSER.src='../images/hl_browser.jpg'" onMouseOut="BROWSER.src='../images/h_browser.jpg'">
			<IMG SRC="../images/h_browser.jpg" ALIGN="absmiddle" BORDER="0" NAME="BROWSER" ALT="&nbsp;&nbsp; Genome Browser"></A></TD>
			
		<TD WIDTH=100>
			<A HREF="../cgi-bin/hgBlat?command=start" onMouseOver="SEARCH.src='../images/hl_blat.jpg'" onMouseOut="SEARCH.src='../images/h_blat.jpg'">
			<IMG SRC="../images/h_blat.jpg" ALIGN="absmiddle" BORDER="0" NAME="SEARCH" ALT="&nbsp;&nbsp; BLAT Search"></A></TD>
			
		<TD WIDTH=100>
			<A HREF="../FAQ.html" onMouseOver="GUIDE.src='../images/hl_faq.jpg'" onMouseOut="GUIDE.src='../images/h_faq.jpg'">
			<IMG SRC="../images/h_faq.jpg" ALIGN="absmiddle" BORDER="0" NAME="GUIDE" ALT="&nbsp;&nbsp; FAQ"></A></TD>
		</TR></TABLE>
	 </TR></TABLE>
	 </CENTER>
    </TD>
</TR>

<TR>

<!--Content Tables-->  
  <TD CELLPADDING=10 ROWSPAN=2 VALIGN="top">
  	<IMG SRC="../images/DOT.gif" HEIGHT=9 WIDTH=1 ALT=""><BR>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="ffffff" BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
	<FONT SIZE="4"><B> &nbsp;News Archives </B></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>

		
<FONT SIZE=4 COLOR="#006666">Archived on 8/23/01:</FONT>	
		<P>Tracks continue to be added to the 
	<A HREF="goldenPath/apr2001Tracks.html">April 2001 browser</A>.
	Our old friend the Exofish track is back.  The blat mouse homology
	track is now up as well, computed at somewhat more sensitive settings
	than it was in the 
	<A HREF="goldenPath/decTracks.html">December 2000 browser.</A></P>
		<P>We've recently received some significant funding from NHGRI to
	maintain and extend this site.  This has allowed us among other things
	to hire an artist, Jenny Draper, who is responsible for the new look.</P>

   </TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	
<BR>
	
<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="ffffff" BACKGROUND="../images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4" ><a NAME="#BROWSER"></a><b>&nbsp; Freeze-Specific News Archives</FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=7></TH><TR><TD WIDTH=10></TD><TD>
		
<FONT SIZE=4 COLOR="#006666">April 1, 2001 Freeze</FONT>
<P>July 13, 2001: fixed bug where some UTRs were
mis-annotated in the known genes on the minus strand.</P>

<BR><FONT SIZE=4 COLOR="#006666">Dec. 12, 2000 Freeze</FONT>
<P>July 13, 2001: fixed bug where some UTRs were
mis-annotated in the known genes on the minus strand.</P>
<P>April 5, 2001: chromosome level files (but not contig level files)
updated to fix bug where some of the centromeres were misplaced</P>
<P>April 1, 2001: chromosome Y updated to fix a bug that put a large
gap between each clone.  This bug was limited to the Y chromosome.</P>

<BR><FONT SIZE=4 COLOR="#006666">Oct. 7, 2000 Freeze</FONT>
<P>April 7, 2001: Affymetrix gene predictions updated and available
for bulk download.</P>
<P>January 9, 2001: all files were updated after a bug that had caused
some finished clones to be flipped in the assembly was caught and fixed.
Our apologies for any inconvenience this has caused.

<P>November 10, 2000: the sequence (.fa) files for two contigs:
X/ctg18523/ctg18523.fa and 7/ctg15082/ctg15082.fa were updated.
These files had null (zero valued) characters that have been
replaced with N characters.  These characters were a result of
a mismatch between clone sizes in the map and in finished NT
contigs.  The sequence for chromosomes X and 7, which contain
these contigs, has also been updated.</P>

<BR><FONT SIZE=4 COLOR="#006666">Sept. 5, 2000 Freeze</FONT>
<P>October 11, 2000: chr21.agp and chr22.fa were updated.  chr21.agp
was a version which went with the UCSC draft assembly rather than the
Sanger/NCBI final assembly of this chromosome.  chr21.agp and chr21.fa
are now in sync.   chr22.fa and chr22.agp were also previously out of
sync.  chr22.fa was obtained from Sanger while chr22.agp had been obtained
by NCBI.  With this update they are consistent, both NCBI versions.
Apologies for any rework this causes you. </P>

<P>October 9, 2000: chr21_random.* and chr22_random.* were removed from
the zip-files in the September 5th freeze.  These files were relics that
should not have been included in the first place.  The files
chr9_random.agp, chr10_random.agp, chr11_random.agp, chr12_random.agp,
chr13_random.agp and chr14_random.agp were updated.  There was a bug
where the initial field of the initial lines in these files was
"(null)" rather than "chrN_random" as it should have been. </P>

<BR><FONT SIZE=4 COLOR="#006666">July 17, 2000 Freeze</FONT>
<P>September 22, 2000: The zip-files chromFa.zip and chromAgp.zip under the
<A HREF="http://genome.cse.ucsc.edu/goldenPath/17july2000/bigZips/">July 17th full data set </A>
were updated to fix some duplications of clone contigs that occurred in the chrN_random.agp and chrN_random.fa
files contained within these zip-files. These "_random" files are files that contain clone contigs that were mapped to a particular chromosome, but
could not be placed at a specific position within that chromosome. They correspond to the "RANDOM" sections of the WashU map. 
None of the regular chrN.agp or chrN.fa files were affected by this update, nor was any of the information in the 
contigAgp.zip or contigFa.zip files changed. For convenience, we include two new files, chromRandAgp.zip and
chromRandFa.zip, for users who would like to download only the data that has changed. These zips consist of
the updated chrN_random.agp and chrN_random.fa files, respectively.

<P>September 4, 2000: The files chromFa.zip and contigFa.zip under the
<A HREF="http://genome.cse.ucsc.edu/goldenPath/17july2000/bigZips/">July 17th full data set </A> and the files under 
<A HREF="http://genome.cse.ucsc.edu/goldenPath/17july2000/pieces/">
July 17th data by individual clone contig </A>
were updated to fix some incorrect (null(0))
characters that needed to be replaced by 'n' characters in some
Fasta files. The following contig Fasta files on chromosomes 1,8,11,12,16,
17 and 19 were affected:

<PRE>
1/ctg14250/ctg14250.fa
8/ctg17325/ctg17325.fa
8/ctg16307/ctg16307.fa
8/ctg25150/ctg25150.fa
8/ctg15216/ctg15216.fa
11/ctg15548/ctg15548.fa
12/ctg15285/ctg15285.fa
16/ctg16864/ctg16864.fa
16/ctg28077/ctg28077.fa
16/ctg13970/ctg13970.fa
16/ctg25401/ctg25401.fa
16/ctg13336/ctg13336.fa
16/ctg14937/ctg14937.fa
16/ctg16860/ctg16860.fa
16/ctg17794/ctg17794.fa
17/ctg14047/ctg14047.fa
19/ctg16564/ctg16564.fa
</PRE>

We apologize for any inconvenience this may have caused. 

	<BR><BR></TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
		
</TD></TR></TABLE>
</BODY></HTML>







