<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Bioinformatics: News Arcive</TITLE>
	<LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="/images/title.jpg"></TH></TR>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
	<table bgcolor="000000" cellpadding="1" cellspacing="1" width="100%" height="27"><tr bgcolor="2636D1"><td valign="middle">
	<table BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24">
			<TR><TD VALIGN="middle"><font color="#89A1DE">&nbsp; 
			&nbsp;<a href="/index.html" class="topbar">
				Home</a> &nbsp; - &nbsp;
			<a href="/cgi-bin/hgGateway" class="topbar">
				Genome Browser</a> &nbsp; - &nbsp; 
			<A HREF="/cgi-bin/hgBlat?command=start" class="topbar">
				Blat Search</a> &nbsp; - &nbsp; 
			<A HREF="/FAQ.html" class="topbar">
				FAQ</a> &nbsp; - &nbsp; 
			<A HREF="/goldenPath/help/hgTracksHelp.html" class="topbar">
				User Guide</a> &nbsp;</font></TD>
	</TR></TABLE>
	</TD></TR></TABLE>
</TD></TR>

<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=2 CELLPADDING=10>	
	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
	<FONT SIZE="4"><B> &nbsp;News Archives </B></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>

<FONT SIZE=4 COLOR="#006666">Posted on Oct. 18, 2002</FONT>
	<P>
	We've rolled out a new version of the Genome Browser - v.16. In 
	addition to several bug fixes, this release contains some interesting 
	new features. 
	<P>
	The Table Browser has undergone major enhancements. Users 
	can now restrict their queries by specifying a value or range for any of 
	the fields in a table, and by selecting which fields should be displayed in 
	the output. The Table Browser also provides the ability to do a free-form SQL 
	query on a table and supports several new output formats. 
	<P>	
	We've extended the capabilities of the DNA retrieval functionality
	in the Genome Browser and the Table Browser. The new 
	mechanism offers the user several options for configuring the amount and 
	type of sequence region that is retrieved, and options for formatting the 
	sequence output. The retrieval options vary based on the type of table 
	selected.
	<P>	
	The Genome Browser's gene prediction tracks now offer a Comparative 
	Sequence link in addition to the predicted protein, mRNA sequence, and 
	genomic sequence links. The Comparative Sequence feature displays 
	annotated codons and translated protein for the region in alignment to 
	another species. 
	<P>	
	We've recently added a few new tracks/tables to the hg12 and mm2 Browsers. On the 
	latest human assembly (hg12), we now have Chimp Blat and Chimp BAC tracks 
	provided by Ingo Ebersberger, Joshua Bacher, and Svante P‰‰bo at the Max
	Planck Institute for Evolutionary Anthropology. Jim Kent at UCSC has generated a 
	new annotation - Gene Boundaries - that shows the boundaries of genes 
	and the direction of transcription as deduced from clustering spliced 
	ESTs and mRNAs against the genome. Daryl Thomas of UCSC has also added SNP tracks to both 
	hg12 and the latest mouse assembly (mm2), based on data from the SNP
	Consortium and NIH.
	<P>	
	We encourage you to experiment with these new features. Comments, 
	questions, and suggestions are always welcome at 
	<A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A>. 

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Oct. 17, 2002</FONT>
	<P>
	We have found errors with the RepeatMasker track on the 
	Feb. '02 mouse assembly (mm2). This problem affects the RepeatMasker track 
	and RepeatMasked DNA obtained via the browser's DNA links. It does 
	<I>not</I> affect data downloaded from the browser's downloads page or ftp 
	site. We have replaced the erroneous data set with a corrected version. If you have questions about how this change may affect your project, email <A HREF=mailto:genome@soe.ucsc.edu>genome@soe.ucsc.edu</A>.

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Oct. 4, 2002</FONT>
	<P>
	Please note that the coordinate range for a portion of the Build 30 chr22 in 
	the UCSC Genome 
	Browser differs from that of NCBI and the Sanger Centre. The latter version 
	contains a 100K bp gap 
	inserted into chr22 just after the centromere. This
	modification was made after UCSC released the Build 30 assembly in the
	Genome Browser. All of the chr22 annotations displayed by the UCSC
	browser are correctly positioned relative to one another. However the
	coordinates of all features past the centromere will be 100K less than those 
	of NCBI and Sanger. The Ensembl Genome Browser is consistent with the
	UCSC browser.

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Sep. 19, 2002</FONT>
	<P>
	We're pleased to announce the release of the Mouse Cons (Human/Mouse
	Evolutionary Conservation Score) annotation track for the <A
	HREF="http://genome.cse.ucsc.edu/cgi-bin/hgTracks?db=hg12">June 2002
	human genome assembly</A>. This track allows a user to interactively explore 
	conservation between the human and mouse genomes and identify highly conserved 
	regions. Highest levels of conservation are typically seen over coding exons. 
	High levels of conservation are also frequently associated with noncoding RNA,
	promoters, other regulatory elements, and pseudogenes.

	<P>The Mouse Cons annotation is displayed using a new type of Genome
	Browser graphical track that plots a continuous function along a chromosome. The
	conservation levels are calculated over 50bp  windows in the human genome
	that have at least 15 bp aligned to mouse. The score for a window reflects
	the probability that the level of observed conservation in that 50bp region
	would occur by chance under neutral evolution. This information is given on
	a logarithmic scale and displays in the track as "mountain ranges". Details
	pages associated with the individual peaks in the track provide access to
	the base level alignments for the whole region and for the individual 50bp
	windows.

	<P>The team that produced this track includes Ryan Weber, Krishna Roskin,
	Mark Diekhans, Jim Kent, Scott Schwartz, and Webb Miller.
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Sep. 12, 2002</FONT>
	<P>
	<I>Nature Genetics</I> has just published <A
	HREF="http://www.nature.com/cgi-taf/DynaPage.taf?file=/ng/journal/v32/n1s/index.html"
	TARGET=_blank>User's Guide to the Human Genome</A>, a
	hands-on tutorial for using genome browsers as web tools for browsing and
	analyzing data from the Human Genome Project and other sequencing efforts. The 
	3 browsers featured in the tutorial include the UCSC Genome Browser, NCBI's 
	Map Viewer, and the Ensembl Genome Browser. The guide is organized around a 
	collection of step-by-step solutions to 13 typical research questions,
	and serves as a nice supplement to the documentation materials
	available on the UCSC Genome Browser web site. The guide is accessible as a 
	link off the <I>Nature Genetics</I> home page at 
	<A HREF="http://www.nature.com/ng/" TARGET=_blank>http://www.nature.com.ng</A>. 
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Sep. 12, 2002</FONT>
	<P>
	We've corrected a problem with the Human June 2002 (hg12) cytoBand annotation 
	track that affected chromosome Y. The clones on this chromosome were
	erroneously pushed from the q-arm onto the p-arm, creating some confusion. The
	currently available version of the cytoBand data on this website contains this 
	correction.

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Aug. 15, 2002</FONT>
	<P>
	We've just released several new annotation tracks/tables for the human genome. 
	On the June 2002 assembly, we've added Gene Bounds, 
	UniGene, CpG Islands, Nonhuman mRNA & EST, SNPs, NC160, and GNF Ratio.
	On the April 2002 assembly we've added a Fgenesh++ Genes track, and have also
	updated the RepeatMasker track. The Dec. 2001 release now includes a Sanger 22 
	track. 

	<P>On the mouse genome, we have 2 new tracks for the Feb. 2002 assembly:
	TIGR Gene Index and RNA Genes.
<HR>
	
<FONT SIZE=4 COLOR="#006666">Posted on Aug. 6, 2002</FONT>
	<P>
	We have fixed an error with six of the chr3 contigs in the bigZips/contigAgp.zip 
	file. The following .agp files were corrected: NT_005684.agp, NT_005663.agp, 
	NT_022554.agp, NT_022459.agp, NT_006031.agp, and NT_022419.agp. The
	chr3.agp file in bigZips/chromAgp.zip was also modified. This change does not 
	affect the .gl files, the .fa files, the 
	lift files, or the annotations. Alignments made on the previous version
	of chr3 are still good. Updated versions of the contigAgp.zip and
	chromAgp.zip files were
	uploaded to our site today. You can download the new versions via ftp at <A 
	HREF="ftp://genome.cse.ucsc.edu/goldenPath/28jun2002" TARGET=_blank>
        ftp://genome.cse.ucsc.edu/goldenPath/28jun2002</A> or via the Downloads
	link in the left sidebar.
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Aug. 2, 2002</FONT>
	<P>The problems with the June 2002 Build 30 (hg12) RepeatMasker track have been 
	resolved. The new RepeatMasker track, along with regenerated Fish Blat
	and Genscan tracks, are now available in the Browser and 
	through our
	Downloads link. We've also added a few new annotation tracks for the
	June 2002 release, and will be adding more over the next 2 weeks.

	<P>The latest Genome Browser has 2 new features. We've added filter
	functionality to the Table Browser, accessible via the Filter Fields
	button on the Table Browser main page. Also, some of the Dec. 2001 human genome tracks (eg. RefSeq Genes) now have a
	Comparative Sequence link from the details page that shows annotated
	codons and translated protein with alignment to the mouse genome.
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on July 27, 2002</FONT>
	<P>
	We've experienced some RepeatMasker problems on Build 30 and are
	rerunning it. This will directly affect the RepeatMasker track and the
	masking of the fasta files. The Fish Blat and Genscan tracks may also
	change slightly once we're redone this. The EST, mRNA, and RefSeq tracks
	should not be affected. We will also post a new RepeatMasker track for
	Build 29 (see news item below) as soon as the Build 30 tracks are
	completed. We apologize for any rework this may cause.
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on July 26, 2002</FONT>
	<P>
	Bulk downloads of the June 2002 Build 30 human genome assembly (hg12) are now
	available from
	<A HREF="ftp://genome.cse.ucsc.edu/goldenPath/28jun2002" TARGET=_blank>
	ftp://genome.cse.ucsc.edu/goldenPath/28jun2002</A>. You can also 
	access the data via the Downloads link in the left sidebar. This initial
	release of the annotation database download contains a limited set of
	tables. Additional files will be available for download next week.
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on July 24, 2002</FONT>

	<P>
	The BLAT server and the coordinates conversion feature for human genome 
	assembly Build 30 (hg12) are now functional.
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on July 23, 2002</FONT>
	<P>We're pleased to announce the pre-release of a browser for 
	human genome assembly Build 30 from NCBI (UCSC version hg12). This assembly was 
	produced at NCBI based on sequence information submitted into GenBank as of
	June 28, 2002. Build 30 release notes and statistics will soon be
	available from the <A
	HREF="http://www.ncbi.nlm.nih.gov/genome/guide/human/" TARGET=_blank> NCBI web 
	site</A>.  
	<P>
	Build 30 is an excellent high-quality assembly. It contains nearly 87%
 	finished sequence, and 94%-97% coverage. The sequence coverage of this build
	is much higher than in previous releases, and there is a high level of
	correspondence between the sequence and the map. Currently, the human genome
	project appears to be on track to achieve the goal of finishing at least 95%
	of the human genome (using Bermuda standards) by April 2003.
	<P>
	UCSC has generated a set of high-level comparisons of the Build 30 draft
	sequence against various types of information (STS maps, BAC end pairs, and
	clone overlaps), accessible from the <A 
	HREF="http://hgwdev.cse.ucsc.edu/goldenPath/chromReports/24Jun2002/"
	TARGET=_blank>Chromosome Reports</A> and <A 
	HREF="http://hgwdev.cse.ucsc.edu/goldenPath/mapPlots/" TARGET=_blank>Genome 
	Map Plots</A> links in the "Technical Information about the Assembled 
	Sequence" section below.
	<P>
	A Blat server for Build 30 is not yet available, but should be
	accessible from this site later this week. Data for the mitochondrial genome 
	and several more annotation tracks will be posted for this release as they 
	become available.  Bulk downloads of the hg12 data should be available from 
	this site in a few days. 
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on July 9, 2002</FONT>
	<P>
	We've found some problems with the repeat-masking of the Build 29 (hg11)
	human sequence. We're in the process of replacing the RepeatMasker
	track, but do not plan to redo the other tracks due to the imminent
	release of Build 30. Because of this, we advise that you do not use the
	cross-species tracks for statistical purposes.

<hr>

<FONT SIZE=4 COLOR="#006666">Posted on July 1, 2002</FONT>
	<P>
	The UCSC Genome Bioinformatics home page is sporting an updated interface to 
	accommodate the growing number of organisms supported by the UCSC Genome 
	Browser, BLAT, and Table Browser. The list of assembly versions accessible 
	through each of these tools can now be found on the tool's Gateway page. To 
	reach the Gateway page, choose an organism from the dropdown list on the left 
	sidebar of this page, then click the Browser, BLAT, or Tables link. New 
	organisms will be added to the list in the months ahead.

	<P>The UCSC site continues to provide a variety of bulk downloads of a genome 
	assemblies and annotations. The list of downloadable data has been removed 
	from the home page, but is readily available through the Downloads link on 
	the left sidebar.  The downloads list can also be accessed directly at 
	<A HREF="downloads.html">http://genome.ucsc.edu/downloads.html</A> or through 
	our ftp site at 
	<A HREF="ftp://genome.cse.ucsc.edu/goldenPath/" TARGET=_blank>ftp://genome.cse.ucsc.edu/goldenPath/</A>.

	<P>Several new annotation tracks have been added to our site in the past 
	couple weeks.  The Feb. 2002 mouse assembly now has tracks for BAC End pairs, 
	Fgenesh++ gene predictions, and AltGenie gene predictions based on Affymetrix's 
	Genie gene-finding software. New to the Apr. 2002 human assembly is the 
	GenMapDB Clones track, which shows placements of BAC clones from the GenMapDB 
	database based on BAC end sequencing information and confirmed using STS 
	markers by Vivian Cheung's lab at U. Penn.  We've also changed the Known Genes 
	track name to RefSeq Genes in all assemblies.

	<P>This release also includes an updated 
	<A HREF="http://genome.ucsc.edu/goldenPath/help/hgTracksHelp.html" TARGET=_blank>User's Guide</A>
	and more detailed documentation on creating & using 
	<A HREF="http://genome.ucsc.edu/goldenPath/help/customTrack.html" TARGET=_blank>custom annotation tracks</A>.

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on May 24, 2002</FONT>
	<P>
	Bulk downloads of the April 2002 hg11 human genome assembly (NCBI Build 29) 
	are now available from <A
	HREF="ftp://genome.cse.ucsc.edu/goldenPath/05apr2002"
	TARGET=_blank>ftp://genome.cse.ucsc.edu/goldenPath/05apr2002</A>. You
	can also access the data from the Downloads link in the left sidebar.

<hr>
<FONT SIZE=4 COLOR="#006666">Posted on May 22, 2002</FONT>
	<P>
	We've just released a browser and BLAT server on 
	the latest Build 29 human genome assembly from NCBI (UCSC 
	version hg11). This assembly is based 
	on sequence information submitted into GenBank as of Apr. 5 2002. As with the 
	Dec. 2001 (hg10) release, this assembly was produced at 
	NCBI rather than at UCSC.  Consult NCBI's 
	<A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/human/release_notes.html"
	TARGET=_blank>Build 29 release notes</A> and 
	<A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/human/HsStats.html"
	TARGET=_blank>statistics</A> for more information about this release.
	This assembly contains nearly 75% finished sequence. 
	Currently, the human genome project appears to be on track to achieve the goal 
	of finishing at least 95% of the human genome (using Bermuda standards) by April 2003.
	
	<P>Although the NCBI human genome assembly has been steadily
	improving over the past year, mapping problems still exist in
	the current release. Most are small, relatively local rearrangements. 
	Larger scale problems include a rearrangement in the p-arm of Chr16 and
	several discrepancies in Chr17.  Researchers - especially positional 
	cloners - are strongly encouraged to use the tools provided 
	(<A HREF="http://genome.ucsc.edu/goldenPath/mapPlots/"
	TARGET=_blank>comparison plots</A>, <A
	HREF="http://genome.ucsc.edu/goldenPath/chromReports/"
	TARGET=_blank>chromosome reports</A>) to evaluate the
	accuracy of the assembly in specific regions of interest.

	<P>Bulk downloads of the hg11 data should be available from this site
	in approximately one week. New annotation tracks will be posted as soon
	as they become available.
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on Apr 24, 2002</FONT>
	<P>Bulk downloads of the February 2002 mouse genome assembly are now available 
	from <A HREF="ftp://genome.cse.ucsc.edu/goldenPath/mmFeb2002" 
	TARGET=_blank>ftp://genome.cse.ucsc.edu/goldenPath/mmFeb2002</A>.  You can 
	also access the data from the Downloading the Mouse Genome section below.<br></P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on Apr 19, 2002</FONT>
        <P>The February 2002 mouse genome assembly is now available in the browser and 
	for BLAT searching.  This assembly was produced at the Whitehead Institute 
	using their Arachne software.  We'd like to thank them and the Mouse Genome 
	Sequencing Consortium for providing this assembly, which has 
	<A HREF="goldenPath/credits.html#mouserestrict">specific conditions for use</A>.
	Bulk downloads of the data should be available in approximately one week.  
	Coordination with mouse genome data access at 
	<A HREF="http://www.ensembl.org/Mus_musculus/">Ensembl</A> and 
	<A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/M_musculus.html">NCBI</A> is 
	in progress.  We'd also like to acknowledge the UCSC team that produced this 
	release: Jim Kent, Terry Furey, Matt Schwartz, Fan Hsu, Yontao Lu, 
	Donna Karolchik, Chuck Sugnet, and Ryan Weber.<br></P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on Apr 9, 2002</FONT>
        <P>Bulk downloads of the November 2001 mouse genome assembly are now available 
	from <A HREF="ftp://genome.cse.ucsc.edu/goldenPath/mmNov2001"
        TARGET=_blank>ftp://genome.cse.ucsc.edu/goldenPath/mmNov2001</A>.
	You can also access the data from the Downloading the Mouse Genome section 
	below.<br></P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on Apr 2, 2002</FONT>
	<P>An updated version of the UCSC Genome Browser (v.11) is now available.
	Along with the v.11 browser, we've released several new annotation tracks on 
	the latest human and mouse assemblies. The new Human Dec. 2001 tracks include:
	Mouse Synteny, Ensembl, Genscan, CpG Islands, Mouse Blat, Fish Blat, 
	Unigene/SAGE, NCI60 Microarray, GNF Affymetrix Microarray, Rosetta
	Microarray, and SNPs. An STS Markers track has been added to the Mouse
	Nov. 2001 browser.<br></P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on Mar 14, 2002</FONT>
	<P>The November 2001 mouse genome assembly is now available
	for viewing in the browser and for BLAT searching. This assembly
	was produced at the Sanger Center using the Phusion software developed
	by Jim Mullikin and Zemin Ning, and was tied to the 
	<A HREF="http://www.ensembl.org/Mus_musculus/assembly.html"> mouse 
	fingerprint map</A> by Tim Hubbard. We'd like to
	thank them and the Mouse Genome Sequencing Consortium for providing this
	assembly, which has <A HREF="/goldenPath/credits.html#mouserestrict">specific 
	conditions for use</A>. Bulk downloads of the data
	will be available in approximately one week. Coordination with
	mouse genome data access at 
	<A HREF="http://www.ensembl.org/Mus_musculus/">Ensembl</A> and 
	<A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/M_musculus.html">NCBI</A> 
	is in progress.  We'd also like to acknowledge the UCSC team that produced 
	this release: Jim Kent, Terry Furey, Matt Schwartz, Fan Hsu, Yontao Lu, 
	and Donna Karolchik.<br></P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on Feb 16, 2002</FONT>
	<P>A new assembly based on sequence submitted as of Dec. 22 in Genbank (Build 28) is
	now available in the browser and for BLAT search.  
	This assembly was produced at NCBI rather than UCSC, primarily by
	Richa Agarwala, Greg Schuler, and Paul Kitts.  The NCBI assembly has
	been steadily improving over the past year.  Currently it shows slightly
	better local order and orientation compared to the UCSC assembly on the
	same sequence, but somewhat worse tracking of the chromosome level maps.
	The NCBI assembly has the advantage that it can be generated significantly
	faster than the UCSC assembly.  With the human genome sequencing now
	in the end game - over two thirds of the human clones are now finished -
	we feel it more productive to focus worldwide annotation efforts on a
	single assembly rather than continue producing competing assemblies.
	We're working with NCBI to improve their map tracking. <br></P>

<hr>

<FONT SIZE=4 COLOR="#006666">Posted on Feb 4, 2002</FONT>
	<P>Chromosome Reports detailing correspondence with STS map, overlap, 
	and BAC end sequence information are available under the "Technical 
	Information About the Assembled Sequence" section below.  This also 
	gives information about the clone map on which the assembled sequence 
	is based.<br></P>		
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on Dec 18, 2001</FONT>
        <P>There are some major enhancements to the browser. The complete user interface settings
		including track controls, labels, and position are now saved from session to session.
		You can configure the browser once to your liking and it will stay that way. This feature
		will only work if cookies are enabled in your browser. If you
		want to restore the default settings use the reset all button under the main graphic.
		Also under the main graphic are new controls that move just the start or just the end
		of the genome window. These are useful for getting exactly the right view without having
		to do arithmetic on the position.  These controls by default will move two guideline units
		at a time, but you can specify other increments.  There's a new page associated with each
		track. This page is accessible by clicking on the mini-buttons to the left of the track 
		in the main graphic, or by clicking on the new hyperlink associated with the track in the
		track controls section under the graphic.  These pages contain a description of the track
		and in many cases new controls.  The mRNA and EST associated controls let you color or
		filter the display according to tissue, author, organism, and so forth.</P>
		<P>As with any new enhancement there are likely to be a few new bugs too.
		Many of these have been spotted and fixed already.  Please let us know (genome@cse.ucsc.edu) 
		if you find a problem that persists more than a day or two.  It's always helpful to include 
		the freeze and genomic position with a problem report.<br></P>
<hr>
<FONT SIZE=4 COLOR="#006666">Posted on Nov 30, 2001</FONT>
        <P>There is now a link from the known genes details page to the Jackson Lab's MGI Mouse Ortholog 
		when the ortholog is known.  Thanks to Carol Bult for her help setting up this link.<br></P>
<hr>
<FONT SIZE=4 COLOR="#006666">Posted on Nov 29, 2001</FONT>
        <P>A duplications track is now available in the August browser.  This track 
		shows duplicate blocks of sequence larger than 1000 bases.  The track is hidden
		by default.  To open it look for 'Duplications'  in the third row of track controls 
		under the main graphic window, and change the setting to 'dense'.  Thanks to Evan Eichler 
		and Jeff Bailey for this track.<br></P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Nov 28, 2001</FONT>
        <P>Sanger curated gene annotations are now available on chromosome 20.  Thanks to 
		Jennifer Ashurst, James Gilbert, and all the annotators at the Sanger Institute.<br></P>
<hr>		
<FONT SIZE=4 COLOR="#006666">Posted on Nov 27, 2001</FONT>
        <P>A new track has been added to the August freeze browser showing
		haplotype blocks derived from common SNPs on Chromosome 21 by Perlegen,
		as described in "<A HREF="http://www.sciencemag.org/cgi/content/abstract/294/5547/1719">Common 
		High-Resolution Haplotypes</A>." Patil, N. et. al. Science 294:1719-1723 (2001).<br></P>
<hr>		
<FONT SIZE=4 COLOR="#006666">Posted on Nov 19, 2001</FONT>
        <P>The SNP and Mouse Blat tracks are now available
		for August.  The Mouse Blat track uses a partial assembly of
		the public whole genome shotgun data courtesy of Whitehead's 
		Arachne program.  <br></P>
<hr>		
<FONT SIZE=4 COLOR="#006666">Posted on Nov. 8, 2001</FONT>
        <P>The detail web pages for each of the tracks have been updated to reflect
        the overall look and design of this site.  You will now see the familiar
        blue navigation bar with links to the Browser, BLAT, Downloads, and the  
        FAQ page from each of the track detail pages.<br></P>
<hr>	        
<FONT SIZE=4 COLOR="#006666">Posted on Nov. 8, 2001</FONT>
        <P>The STS Markers track has been updated on the April and August browsers to
        now include much more information on the detail page including links to
        UniSTS and details on the alignments of the markers to the draft sequence.
        In addition, all known aliases of the markers can be entered in the
        "position" window, and the corresponding merker will be found and
        displayed if its location has been determined.<br></P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Nov. 8, 2001</FONT>
        <P>A new FISH Clones track has been added to the April and August browsers.
        Previously, this information has been included in the STS Markers track.
        Now, this has been broken out into a separate track with additional
        information provided on the detail page not previously shown.<br></P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Nov. 6, 2001</FONT>
        <P>The fgenesh++ gene prediction and the cross-species
        mRNA tracks are now available in the August browser.<br></P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Oct 31, 2001</FONT>
	<P>The 'DNA' button at the top of the browser has been significantly 
	upgraded.   By default it now returns DNA that has repeating elements 
	in lower case and other DNA in upper case.  There is also an option to 
	color the DNA output with various tracks.  You can have the case and font 
	features such as underline, bold, and italic follow tracks too.<br></P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Oct 29, 2001</FONT>
	<P>There is now a TIGR Gene Index track
	in the April 2001 freeze browser.  The TIGR Gene Index is based
	on alignments of assembled ESTs from a number of species.
	Be sure to click into the track and follow the outside links to
	the TIGR site, which contains a wealth of information on the genes.<br></P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Oct 29, 2001</FONT>
	<P>The Acembly track on the August 2001 freeze
	has been updated to include predictions based on human ESTs 
	and Genbank  mRNAs as well as RefSeq human mRNAs.  Protein 
	predictions are now also available in the details page for this
	track.  The outside link for this track is also very informative.<br></P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Oct 27, 2001</FONT>
        <P>You can now share your custom tracks with the
        community. The easiest way to do this is to construct a link from your own
        web pages to the browser.  Here is an example of a URL for
        such a link:
        <pre>   http://genome.ucsc.edu/cgi-bin/hgTracks?
        position=chr22:1-20000&db=hg8&
        hgt.customText=http://genome-test.cse.ucsc.edu/test.bed</pre>

        The position variable tells the browser which part of the genome to
        display.  The db variable refers to the freeze number.
        'hg8' corresponds to the August 2001 freeze.  The customText
        variable should refer to a URL containing plain text in one of the
        formats described in <a href="/goldenPath/help/customTrack.html">
        http://genome.cse.ucsc.edu/goldenPath/help/customTrack.html</a>.
        <br><br>
        Note that generally we only keep the last three versions of the
        genome online (hg6, hg7, and hg8).  You'll have to update
        your link and track about every 4 months as a result.
        <br><br>
        Please send in the URLs of tracks you'd like to share to
        <a href="mailto:genome@cse.ucsc.edu"> genome@cse.ucsc.edu</a>,
        along with a brief description of
        the track and the genome version it is tied to.  We'll create
        an index page of these here.
    <br><br>
        You can also access an external custom track by including
        the URL of the track data (on a separate line starting with
        http://) in the custom track box at the bottom of the browser gateway.</P>

<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Oct. 15, 2001</FONT>
        <P>Fresh tab-delimited files from the browser database
        are now available.  They will now be updated automatically
        every Sunday evening.   The table browser queries the database
        directly, so it is always up to the minute.</P>

<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Oct. 12, 2001</FONT>
	<P>Several new sets of gene predictions came in this week.
	We now have fgenesh++ predictions for the April freeze,
	and Genscan and Acembly predictions for the August freeze.</P>

<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Oct. 11, 2001</FONT>
	<P>Why struggle with massive genomic file downloads when the UCSC 
	<A HREF="http://genome.cse.ucsc.edu/goldenPath/hgText.html">Table Browser</A> 
	lets you select exactly the track data desired via a convenient web 
	interface? Major new improvements by Krish Roskin have empowered and 
	simplified this feature available now for the three most recent assemblies.</P>

<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Oct.8, 2001</FONT> 
	<P>The October 2000 assembly has been moved to the 
	<A HREF="http://genome-archive.cse.ucsc.edu">Archives</A>
	to make room for the August assembly.</P>
	
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Oct.5, 2001</FONT> 
	<P>A revised August 2001 freeze assembly is now up.  The problems
	with flipped contigs of finished clones and high levels of
	sequence duplication are fixed.   You can now download this
	assembly <A HREF="ftp://genome.ucsc.edu/goldenPath/06aug2001/bigZips">in bulk</A> as 
	well as browse it.   Chromosome by chromosome and annotation
	database files will be following over the next day or two.</P>

<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Oct.2, 2001</FONT>
	<P>You can now <A HREF=/cgi-bin/hgCoordConv>convert
	coordinates</A> between different versions of the draft
	using a new program, hgCoordConv, by Chuck Sugnet.
 	hgCoordConv attempts to cut out sequences of the original 
	draft and align them to the new draft.  When aligning the 
	sequences to the new draft hgCoordConv makes sure that the
	sequences are in the same order, orientation, and have the 
	correct distances between them. </P>
	
	<P>Chuck has also implemented a SAGE/Unigene track in the browser. 
	This track displays data from the <A HREF= http://www.ncbi.nlm.nih.gov/SAGE/>
	SAGEMap project at NCBI </A>.   UniGene cluster sequences are
	displayed in the browser and colored according to their average SAGE
	counts over a series of experiments. Selecting one of the UniGene
	representative sequences displays the SAGE results for UniGene 
	sequences.</P>
	
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Sept. 21, 2001:</FONT>
	<P>Some systematic problems were found in the clone order on the
	preliminary August 2001 freeze assembly. The sequencing center at
	Washington University, EBI's Ensembl group, and our group here at UCSC
	are currently working together to revise the merged BAC clone maps and
	the assembly process to fix these problems. We hope to update the August
	browser with a revised assembly soon. Then, after further testing via
	the browser, we will release the assembled August freeze genome sequence
	itself. </P>
	
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on 9/11/01:</FONT>	
	<P>A preliminary assembly of the August 2001 freeze is now
	available in the genome browser. Due to significant progress 
	by the mapping and finishing groups of the international 
	public consortium, this assembly is a major improvement over 
	the April 2001 freeze assembly.	Imre Vastrik, Ewan Birney and 
	colleagues at Ensembl have computed a merge of BAC clone maps
	provided by the individual sequencing centers with
	fingerprint-based maps prepared at Washington University. These
	merged maps were used for the first time in this August 
	assembly.</P>

	<P>The August assembly has successfully passed our internal 
	quality control tests. We will release the sequence and
	annotations in bulk downloadable form in a week or so, after 
	the external testers have had a chance to further verify
	it. Meanwhile if you notice any systematic problems please let 
	us know at genome@cse.ucsc.edu. Though the state of the working
	draft has improved considerably, remember that where you see 
	solid marks in the 'gap' track, the relative order and 
	orientation of flanking contigs is still uncertain. In some 
	cases of complex repeat structure it is also possible that
	the assembly may be incorrect even in the absence of gaps.
	Also, sometimes ambiguities in the data cause a BAC clone to be 
	split, with parts of it placed at opposite ends of a run of other clones.	
	Localized errors of this type should be corrected by additional 
	finishing efforts at the individual sequencing centers and 
	should not be reported to UCSC. However, please report any
	large-scale or systematic problems you detect with this assembly	
	that could have been caused by our data processing.</P>

	<P>The tracks available on the August 2001 browser are quite
	limited at the moment. More tracks will show up over time.</P>

<hr>
<FONT SIZE=4 COLOR="#006666">Archived on 9/11/01:</FONT>	
                <P>The April 2001 assembly is now the default for the browser.
        The SNP and Ensembl gene tracks have come in for this version.
        There is also a new track depicting non-human vertebrate mRNA
        alignments.</P>
                <P>The Sept. 2000 and July 2000 versions of the genome are        
        now only available on our archive site.  Please see the link
        in the blue box to the left for more details.</P>
                <P>The August 6 freeze is progressing through the pipeline.
        We've recently received an updated accession map from Wash U.
        Ensembl will shortly be integrating this with chromosome
        specific maps from the sequencing centers.  We are still
        on track for an early September next release.  </P>
<hr>
<FONT SIZE=4 COLOR="#006666">Archived on 8/28/01:</FONT>	
		<P> Meanwhile we've been continuing work on the genome browser.  
  	It's now possible  to upload your own annotations to be displayed 
  	alongside the  built-in tracks.  Please scroll to the bottom of the 
 	browser gateway pages for further information.  The browser has 
  	also been sped  up, particularly on the larger chromosomes by 
  	using a 'binning'  technique suggested by Lincoln Stein and Richard Durbin.</P>

<hr>
<FONT SIZE=4 COLOR="#006666">Archived on 8/23/01:</FONT>	
		<P>Tracks continue to be added to the 
	<A HREF="/goldenPath/apr2001Tracks.html">April 2001 browser</A>.
	Our old friend the Exofish track is back.  The blat mouse homology
	track is now up as well, computed at somewhat more sensitive settings
	than it was in the 
	<A HREF="/goldenPath/decTracks.html">December 2000 browser.</A></P>
		<P>We've recently received some significant funding from NHGRI to
	maintain and extend this site.  This has allowed us among other things
	to hire an artist, Jenny Draper, who is responsible for the new look.</P>

   </TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	
<BR>
	
<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4" ><a NAME="#BROWSER"></a><b>&nbsp; Freeze-Specific News Archives</FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=7></TH><TR><TD WIDTH=10></TD><TD>
		
<FONT SIZE=4 COLOR="#006666">April 1, 2001 Freeze</FONT>
<P>July 13, 2001: fixed bug where some UTRs were
mis-annotated in the known genes on the minus strand.</P>

<hr><FONT SIZE=4 COLOR="#006666">Dec. 12, 2000 Freeze</FONT>
<P>July 13, 2001: fixed bug where some UTRs were
mis-annotated in the known genes on the minus strand.</P>
<P>April 5, 2001: chromosome level files (but not contig level files)
updated to fix bug where some of the centromeres were misplaced</P>
<P>April 1, 2001: chromosome Y updated to fix a bug that put a large
gap between each clone.  This bug was limited to the Y chromosome.</P>

<hr><FONT SIZE=4 COLOR="#006666">Oct. 7, 2000 Freeze</FONT>
<P>April 7, 2001: Affymetrix gene predictions updated and available
for bulk download.</P>
<P>January 9, 2001: all files were updated after a bug that had caused
some finished clones to be flipped in the assembly was caught and fixed.
Our apologies for any inconvenience this has caused.

<P>November 10, 2000: the sequence (.fa) files for two contigs:
X/ctg18523/ctg18523.fa and 7/ctg15082/ctg15082.fa were updated.
These files had null (zero valued) characters that have been
replaced with N characters.  These characters were a result of
a mismatch between clone sizes in the map and in finished NT
contigs.  The sequence for chromosomes X and 7, which contain
these contigs, has also been updated.</P>

<hr><FONT SIZE=4 COLOR="#006666">Sept. 5, 2000 Freeze</FONT>
<P>October 11, 2000: chr21.agp and chr22.fa were updated.  chr21.agp
was a version which went with the UCSC draft assembly rather than the
Sanger/NCBI final assembly of this chromosome.  chr21.agp and chr21.fa
are now in sync.   chr22.fa and chr22.agp were also previously out of
sync.  chr22.fa was obtained from Sanger while chr22.agp had been obtained
by NCBI.  With this update they are consistent, both NCBI versions.
Apologies for any rework this causes you. </P>

<P>October 9, 2000: chr21_random.* and chr22_random.* were removed from
the zip-files in the September 5th freeze.  These files were relics that
should not have been included in the first place.  The files
chr9_random.agp, chr10_random.agp, chr11_random.agp, chr12_random.agp,
chr13_random.agp and chr14_random.agp were updated.  There was a bug
where the initial field of the initial lines in these files was
"(null)" rather than "chrN_random" as it should have been. </P>

<hr><FONT SIZE=4 COLOR="#006666">July 17, 2000 Freeze</FONT>
<P>September 22, 2000: The zip-files chromFa.zip and chromAgp.zip under the
<A HREF="http://genome.cse.ucsc.edu/goldenPath/17july2000/bigZips/">July 17th full data set </A>
were updated to fix some duplications of clone contigs that occurred in the chrN_random.agp and chrN_random.fa
files contained within these zip-files. These "_random" files are files that contain clone contigs that were mapped to a particular chromosome, but
could not be placed at a specific position within that chromosome. They correspond to the "RANDOM" sections of the WashU map. 
None of the regular chrN.agp or chrN.fa files were affected by this update, nor was any of the information in the 
contigAgp.zip or contigFa.zip files changed. For convenience, we include two new files, chromRandAgp.zip and
chromRandFa.zip, for users who would like to download only the data that has changed. These zips consist of
the updated chrN_random.agp and chrN_random.fa files, respectively.

<P>September 4, 2000: The files chromFa.zip and contigFa.zip under the
<A HREF="http://genome.cse.ucsc.edu/goldenPath/17july2000/bigZips/">July 17th full data set </A> and the files under 
<A HREF="http://genome.cse.ucsc.edu/goldenPath/17july2000/pieces/">
July 17th data by individual clone contig </A>
were updated to fix some incorrect (null(0))
characters that needed to be replaced by 'n' characters in some
Fasta files. The following contig Fasta files on chromosomes 1,8,11,12,16,
17 and 19 were affected:

<PRE>
1/ctg14250/ctg14250.fa
8/ctg17325/ctg17325.fa
8/ctg16307/ctg16307.fa
8/ctg25150/ctg25150.fa
8/ctg15216/ctg15216.fa
11/ctg15548/ctg15548.fa
12/ctg15285/ctg15285.fa
16/ctg16864/ctg16864.fa
16/ctg28077/ctg28077.fa
16/ctg13970/ctg13970.fa
16/ctg25401/ctg25401.fa
16/ctg13336/ctg13336.fa
16/ctg14937/ctg14937.fa
16/ctg16860/ctg16860.fa
16/ctg17794/ctg17794.fa
17/ctg14047/ctg14047.fa
19/ctg16564/ctg16564.fa
</PRE>

We apologize for any inconvenience this may have caused. 

	<BR><BR></TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
		
</TD></TR></TABLE>
</BODY></HTML>







