<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>

	
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META http-equiv="Content-Script-Type" content="text/javascript">
	<TITLE>
Track and Table Releases	</TITLE>
	<LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">

<TR><TH COLSPAN=1 ALIGN="left"><IMG SRC="/images/title.jpg"></TH></TR>


<!--HOTLINKS BAR---------------------------------------------------------->
<TR><TD COLSPAN=3 HEIGHT=40 >
<table bgcolor="000000" cellpadding="1" cellspacing="1" width="100%%" height="27">
<tr bgcolor="2636D1"><td valign="middle">
	<table BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24"><TR>
	 	<TD VALIGN="middle"><font color="#89A1DE">&nbsp;

&nbsp;<A HREF="/index.html" class="topbar">
           Home</A> &nbsp; - &nbsp;
       <A HREF="/cgi-bin/hgGateway" class="topbar">
           Genome Browser</A> &nbsp; - &nbsp;
       <A HREF="/cgi-bin/hgNear" class="topbar">
           Gene Sorter</A> &nbsp; - &nbsp;
       <A HREF="/cgi-bin/hgBlat?command=start&" class="topbar">           Blat</A> &nbsp; - &nbsp;
       <A HREF="/cgi-bin/hgText" class="topbar">
           Table Browser</A> &nbsp; - &nbsp;
       <A HREF="/FAQ.html" class="topbar">
           FAQ</A> &nbsp; - &nbsp;
       <A HREF="/goldenPath/help/hgTracksHelp.html" class="topbar">
           Help</A> &nbsp;</font></TD>
       </TR></TABLE>
</TD></TR></TABLE>
</TD></TR>	


<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=3>	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp;
Track and Table Releases</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
	

 This page contains track and table release information for the following genome assemblies:<br>
<ul>
<li><a CLASS="toc" HREF="#hg16">Human July 2003 (hg16)</a></li><li><a CLASS="toc" HREF="#hg15">Human April 2003 (hg15)</a></li><li><a CLASS="toc" HREF="#hg13">Human Nov. 2002 (hg13)</a></li><li><a CLASS="toc" HREF="#panTro1">Chimp Nov. 2003 (panTro1)</a></li><li><a CLASS="toc" HREF="#mm4">Mouse Oct. 2003 (mm4)</a></li><li><a CLASS="toc" HREF="#mm3">Mouse Feb. 2003 (mm3)</a></li><li><a CLASS="toc" HREF="#rn3">Rat Jun 2003 (rn3)</a></li><li><a CLASS="toc" HREF="#rn2">Rat Jan. 2003 (rn2)</a></li><li><a CLASS="toc" HREF="#galGal2">Chicken Feb. 2004 (galGal2)</a></li><li><a CLASS="toc" HREF="#fr1">Fugu Aug. 2002 (fr1)</a></li><li><a CLASS="toc" HREF="#dm1">D. melanogaster Jan. 2003 (dm1)</a></li><li><a CLASS="toc" HREF="#ce2">C. elegans March 2004 (ce2)</a></li><li><a CLASS="toc" HREF="#ce1">C. elegans May 2003 (ce1)</a></li><li><a CLASS="toc" HREF="#cb1">C. briggsae July 2002 (cb1)</a></li><li><a CLASS="toc" HREF="#sacCer1">S. cerevisiae Oct. 2003 (sacCer1)</a></li><li><a CLASS="toc" HREF="#sc1">SARS Apr. 2003 (sc1)</a></li><li><a CLASS="toc" HREF="#zoo1">NISC (Zoo) Jun. 2002 (zoo1)</a></li></ul>
<p>
 For more information about the tracks and tables listed on this page, refer to the <a href=/goldenPath/gbdDescriptions.html>Description of the annotation database</a> and the <a href=/goldenPath/help/hgTracksHelp.html#IndivTracks>User's Guide</a>.<p>
<em>Last updated 03 Jun 2004. <a HREF="mailto:genome@soe.ucsc.edu">Inquiries and feedback welcome</a>.</em>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=hg16></A>Human July 2003 (hg16, NCBI Build 34)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
GC% Win20K (dropped - replaced by 5-base window GC Percent track)</td>
<td>01 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
Alt Splicing (data update)</td>
<td>27 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Chain, Fugu Net</td>
<td>24 May 2004</td>
</tr>
<TR valign=top><TD align=left>
stsInfo2 table</td>
<td>24 May 2004</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>21 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Chimp Recip Chain, Chimp Recip Net (updated to panTro1 assembly)
</td>
<td>21 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Chimp Chain (updated to panTro1 assembly)</td>
<td>21 May 2004</td>
</tr>
<TR valign=top><TD align=left>
RNA Genes</td>
<td>19 May 2004</td>
</tr>
<TR valign=top><TD align=left>
GC Percent (updated to 5-base window track)</td>
<td>19 May 2004</td>
</tr>
<TR valign=top><TD align=left>
TFBS Conserved</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA (updated)</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Ecores</td>
<td>03 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Microsatellites</td>
<td>03 May 2004</td>
</tr>
<TR valign=top><TD align=left>
jaxOrtholog table</td>
<td>30 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Chicken Chain, Chicken Net (using galGal2)</td>
<td>29 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Ecores</td>
<td>29 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Acembly</td>
<td>20 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
SNPs (updated to dbSNP Build 120)</td>
<td>20 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
RNA Genes</td>
<td>13 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Hu/Mouse/Rat/Chick removed (replaced by Hu/Chp/Mo/Ra/Chk)</td>
<td>12 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Hu/Chp/Mo/Ra/Chk</td>
<td>12 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (updated to use proteins040315)</td>
<td>09 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix U133 (updated)</td>
<td>09 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
GNF Atlas 2</td>
<td>09 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix GNF1H</td>
<td>09 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
ECgene Genes (updated to include new peptide table)</td>
<td>05 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Human/Mouse/Rat removed (replaced by Hu/Mouse/Rat/Chick)</td>
<td>29 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Hu/Mouse/Rat/Chick (using mm3, rn3, galGal2) - replaces Human/Mouse/Rat track</td>
<td>19 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
ECgene Genes (updated)</td>
<td>19 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA</td>
<td>19 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
SGP (updated to use mm4)</td>
<td>05 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Map Contigs (updated)</td>
<td>02 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Perlegen (updated)</td>
<td>02 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Fgenesh++ Genes</td>
<td>13 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
affyU133, affyU95 tables</td>
<td>10 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (updated)</td>
<td>06 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
2X Reg Potential, 3X Reg Potential</td>
<td>05 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes (update)</td>
<td>04 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
affyTranscriptome table</td>
<td>29 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Mm3 Chain, Mm3 Net removed (replaced by mm4 versions)</td>
<td>20 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
chr4_random_gap, chr4_random_gold, chr8_random_gap, chr8_random_gold, chrX_random_gap, chrX_random_gold, chrUn_random_gap, chrUn_random_gold tables (updated)</td>
<td>16 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Chimp Diff</td>
<td>16 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Affy10K</td>
<td>14 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Mm4 Net (updates Mouse Net using mm4)</td>
<td>12 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Mm4 Chain (updates Mouse Chain using mm4)</td>
<td>09 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Mm4 Tight (updates Mouse Tight using mm4)</td>
<td>08 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
ECgene Genes</td>
<td>22 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
vsPt0/axtBest downloads (chimp reciprocal best)</td>
<td>22 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
Alt-Splicing</td>
<td>18 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
Twinscan Genes</td>
<td>18 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
Chimp All Chains, Chimp Net, Chimp sequence and chain downloads</td>
<td>10 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs (updated)</td>
<td>19 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>18 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>18 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Chain Self</td>
<td>15 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
UniGene_2, sage</td>
<td>12 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Gene Bounds</td>
<td>12 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Chain Self</td>
<td>11 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Vega Genes, Vega Pseudogenes</td>
<td>07 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
NCI60</td>
<td>24 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (add UTR 3' and 5' folding data)</td>
<td>24 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Haplotype Blocks, Perlegen (chr 21 and 22 only)</td>
<td>24 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Human/Mouse/Rat downloads</td>
<td>24 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Synteny</td>
<td>21 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Genbank daily updates enabled</td>
<td>14 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Affy 120K SNPs</td>
<td>13 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Random SNPs, Overlap SNPs (dbSnp 117)</td>
<td>10 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Human/Mouse/Rat (using mm3 and rn3)</td>
<td>09 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (update)</td>
<td>08 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Fugu Blat (v. 3.0)</td>
<td>02 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
affyRatio table</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
vsMm3, vsRn3 data downloads</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>18 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Geneid</td>
<td>18 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain (using mm3)</td>
<td>10 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Net (using mm3) (contains fixes to chr1 data)</td>
<td>10 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Tight (using mm3) (contains fixes to chr1 data)</td>
<td>10 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
First EF (contains fixes to chrX and chrY)</td>
<td>09 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Net (using mm3)</td>
<td>02 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Rat Net (using rn3)</td>
<td>02 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Rat Chain</td>
<td>29 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain</td>
<td>29 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Tight Mouse</td>
<td>29 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Self Chain</td>
<td>29 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes</td>
<td>26 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
STS Markers (updated)</td>
<td>11 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial hg16 release (using NCBI Build 34): Chromosome Band, FISH Clones, Assembly, Gap, STS Markers, Map Contigs, Recomb Rate, GC Percent, Coverage, BAC End Pairs, Fosmid End Pairs, RefSeq Genes, MGC Genes, Genscan Genes, Human mRNAs, Spliced ESTs, Human ESTs, NonHuman mRNAs, NonHuman ESTs, CpG Islands, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosomes, database</td>
<td>08 Aug 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=hg15></A>Human April 2003 (hg15, NCBI Build 33)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
RNA Genes</td>
<td>19 May 2004</td>
</tr>
<TR valign=top><TD align=left>
TFBS Conserved</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
GNF Ratio (updated)</td>
<td>14 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
GNF Ratio (updated)</td>
<td>05 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes (updated)</td>
<td>05 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (updated to use proteins040315)</td>
<td>05 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes (update)</td>
<td>30 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
affyTranscriptome table</td>
<td>29 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Affy10K</td>
<td>14 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Random SNPs, Overlap SNPs (dbSnp 112)</td>
<td>31 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Affy 120K SNPs</td>
<td>13 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
KnownGenes (extended)</td>
<td>08 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
affyRatio table</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
ensPep table</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Segmental Dups</td>
<td>18 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
First EF (contains fixes to chrX and chrY)</td>
<td>09 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
First EF</td>
<td>27 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Downloads: Mouse (mm3)/Rat (rn2)/Human multiple alignments</td>
<td>22 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
NonHuman ESTs</td>
<td>25 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>15 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Cons (using mm3)</td>
<td>09 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Best Mouse (using mm3)</td>
<td>09 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
bigZips (upstream files updated)</td>
<td>09 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Blastz axtTight mouse alignments (using mm3)</td>
<td>25 Jun 2003</td>
</tr>
<TR valign=top><TD align=left>
Acembly</td>
<td>05 Jun 2003</td>
</tr>
<TR valign=top><TD align=left>
AxtInfo (updated to use axtNetMm3)</td>
<td>30 May 2003</td>
</tr>
<TR valign=top><TD align=left>
UniGene (v. Hs 160)</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Tight Rat (using rn2)</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Gene Bounds</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Rat Synteny (using rn2)</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Human/Mouse/Rat (using mm3, rn2)</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
MGC Genes</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Fosmid End Pairs</td>
<td>20 May 2003</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes (using mm3)</td>
<td>20 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Haplotype Blocks (chr 21 and 22 only)</td>
<td>20 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Overlap and Random SNPs (dbSnp 114)</td>
<td>15 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Blastz axtBest mouse alignments (using mm3)</td>
<td>13 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Human/Mouse Chain and Net (using mm3)</td>
<td>09 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>08 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Twinscan (using mm2)</td>
<td>08 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Recomb Rate</td>
<td>01 May 2003</td>
</tr>
<TR valign=top><TD align=left>
geneBands (table only)</td>
<td>30 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
NCI60</td>
<td>30 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
GNF Ratio</td>
<td>30 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
GC Pecent (data reloaded)</td>
<td>29 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Chromosome Band</td>
<td>28 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Fgenesh++ Genes</td>
<td>28 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
FISH Clones</td>
<td>28 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs</td>
<td>25 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Coverage</td>
<td>25 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
liftAll.zip (correction for NT_008183 - table only)</td>
<td>16 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial hg15 release (using NCBI Build 33): Assembly, Gap, STS Markers, Map Contigs, GC Percent, Known Genes, RefSeq Genes, Genscan Genes, Human mRNAs, Spliced ESTs, Human ESTs, NonHuman mRNAs, CpG Islands, Fugu Blat (v 3.0), Tight Mouse (mm3), RepeatMasker, Simple Repeats, Mouse Synteny (mm3)</td>
<td>14 Apr 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=hg13></A>Human Nov. 2002 (hg13, NCBI Build 31)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
RNA Genes</td>
<td>19 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Affy 120K SNPs</td>
<td>13 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Vega Genes, Vega Pseudogenes</td>
<td>09 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Segmental Dups</td>
<td>24 Jun 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Synteny (using mm3)</td>
<td>06 Jun 2003</td>
</tr>
<TR valign=top><TD align=left>
Fgenesh++ Genes (new data)</td>
<td>30 May 2003</td>
</tr>
<TR valign=top><TD align=left>
RNA Genes</td>
<td>28 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Transcriptome (chr21 and 22)</td>
<td>28 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Haplotype Blocks (chr21 and 22)</td>
<td>20 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Random SNPs, Overlap SNPs (updated to dbSnp 114)</td>
<td>15 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Human/Mouse Chain and Net (using mm3)</td>
<td>02 May 2003</td>
</tr>
<TR valign=top><TD align=left>
SLAM Mouse (using mm2)</td>
<td>28 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
SLAM Rat (using rn1)</td>
<td>28 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Rat Synteny (using rn1)</td>
<td>07 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Superfamily</td>
<td>30 Mar 2003</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>28 Mar 2003</td>
</tr>
<TR valign=top><TD align=left>
Random SNPs, Overlap SNPs (dbSnp 112)</td>
<td>27 Mar 2003</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>14 Mar 2003</td>
</tr>
<TR valign=top><TD align=left>
RepeatMasker (updated)</td>
<td>13 Mar 2003</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs</td>
<td>27 Feb 2003</td>
</tr>
<TR valign=top><TD align=left>
Acembly Genes (IDs changed)</td>
<td>27 Feb 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Cons (using mm2)</td>
<td>21 Feb 2003</td>
</tr>
<TR valign=top><TD align=left>
UniGene (v. Hs 159)</td>
<td>20 Feb 2003</td>
</tr>
<TR valign=top><TD align=left>
Coverage</td>
<td>10 Feb 2003</td>
</tr>
<TR valign=top><TD align=left>
Fosmid End Pairs</td>
<td>10 Feb 2003</td>
</tr>
<TR valign=top><TD align=left>
Blastz human alignments downloads (using hg13)</td>
<td>07 Feb 2003</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes (using mm2)</td>
<td>06 Feb 2003</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>06 Feb 2003</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>22 Jan 2003</td>
</tr>
<TR valign=top><TD align=left>
GenMapDB Clones</td>
<td>22 Jan 2003</td>
</tr>
<TR valign=top><TD align=left>
Recomb Rate</td>
<td>20 Jan 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial hg13 Release (using NCBI Build 31): Assembly, Gap, STS Markers, Map Contigs, GC Percent, FISH Clones, Chromosome Band, Known Genes, RefSeq Genes, Genscan Genes, Human mRNAs, Spliced ESTs, Human ESTs, NonHuman mRNAs, Gene Bounds, CpG Islands, Fugu Blat (v. 3.0), Tight Mouse (mm3), Best Mouse (mm3), RepeatMasker, Simple Repeats</td>
<td>23 Dec 2002</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=panTro1></A>Chimp Nov. 2003 (panTro1, CGSC Build 1 Version 1 Arachne assembly)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
BAC End Pairs</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Blat</td>
<td>14 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Genscan Genes</td>
<td>10 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Contigs</td>
<td>28 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>27 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Deletions (updated)</td>
<td>14 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes (chr1 updated)</td>
<td>07 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>24 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Best, vsHg15 downloads</td>
<td>02 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial panTro1 Release (using the Build 1 Version 1 Arachne assembly dated 13 Nov. 2003): Scaffolds, Gap, GC Percent, Quality Scores, Human RefSeq, Chimp mRNAs, Spliced ESTs, Chimp ESTs, NonChimp mRNAs, NonChimp ESTs, CpG Islands, Human Chain, Human Net, Human Deletions, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosome, database.</td>
<td>23 Feb 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=mm4></A>Mouse Oct. 2003 (mm4, NCBI Build 32)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Gene Sorter: MOE430 ID</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter: Max Rinn Sex</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Rinn Sex Exp</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter: Locus Link</td>
<td>26 May 2004</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>21 May 2004</td>
</tr>
<TR valign=top><TD align=left>
GC Percent (updated to 5-base window track)</td>
<td>19 May 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA (updated)</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix MOE430</td>
<td>28 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix GNF1M</td>
<td>27 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (updated to proteins040314)</td>
<td>16 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Superfamily</td>
<td>16 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Chromosome Band</td>
<td>13 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
STS Markers</td>
<td>19 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA</td>
<td>19 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
GNF U74A, GNF U74B, GNF U74C, Affymetrix U74</td>
<td>08 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Overlap SNPs (using dbSnp 118)</td>
<td>25 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes</td>
<td>09 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>08 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes</td>
<td>08 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial mm4 Release (using NCBI Build 32): Assembly, Gap, Map Contigs, BAC End Pairs, GC Percent, Known Genes, RefSeq Genes, MGC Genes, Genscan Genes, Mouse mRNAs, Spliced ESTs, Mouse ESTs, NonMouse mRNAs, NonMouse ESTs, CpG Islands, Human Chain, Human Net, Human Tight, Rat Chain, Rat Net, Rat Tight, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosome, database.</td>
<td>21 Nov 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=mm3></A>Mouse Feb. 2003 (mm3, NCBI Build 30)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Geneid</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Ecores</td>
<td>04 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Ecores</td>
<td>04 May 2004</td>
</tr>
<TR valign=top><TD align=left>
GNF U74A, GNF U74B, GNF U74C, Affymetrix U74</td>
<td>08 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Recomb Rate</td>
<td>02 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Chromosome Band</td>
<td>23 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
ECgene Genes</td>
<td>22 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>18 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Rat Synteny (updated)</td>
<td>27 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
knownGenes (updated)</td>
<td>20 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter (updated)</td>
<td>20 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Genbank daily updates enabled</td>
<td>14 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter</td>
<td>10 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
proteins092903 database (used by Known Genes)</td>
<td>09 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
axtBest/chr2.axt.gz (was missing from axtBest data set)</td>
<td>02 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
ensemblGenes (proteins added)</td>
<td>01 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Human Best (updated to use hg15)</td>
<td>25 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Superfamily</td>
<td>25 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Bac End Pairs (added data for chrX)</td>
<td>25 Jun 2003</td>
</tr>
<TR valign=top><TD align=left>
Human Cons (using hg15)</td>
<td>24 Jun 2003</td>
</tr>
<TR valign=top><TD align=left>
Human Chain/Net (updated from hg13 to hg15)</td>
<td>17 Jun 2003</td>
</tr>
<TR valign=top><TD align=left>
Ensembl EST Genes</td>
<td>13 Jun 2003</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>13 Jun 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse ESTs, Spliced ESTs, Gene Bounds (additions for random chromosomes)</td>
<td>30 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>28 May 2003</td>
</tr>
<TR valign=top><TD align=left>
MGC Genes</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Human Synteny (using hg15)</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands (random data added)</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Rat Synteny (using rn1)</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Twinscan (using hg13)</td>
<td>22 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes</td>
<td>16 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Blastz human alignments downloads (using hg15)</td>
<td>14 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Blastz rat alignments downloads (using rn2)</td>
<td>13 May 2003</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes (using hg13)</td>
<td>08 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Human/Mouse Chain and Net (using hg15)</td>
<td>02 May 2003</td>
</tr>
<TR valign=top><TD align=left>
STS Markers</td>
<td>29 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
BAC Ends Pairs</td>
<td>29 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Fugu Blat</td>
<td>29 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Simple Repeats (data added for randoms)</td>
<td>17 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Fgenesh++ Genes</td>
<td>17 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Best Human, Tight Human (using hg15)</td>
<td>02 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Best Rat, Tight Rat (using rn2)</td>
<td>02 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
NonMouse mRNAs</td>
<td>01 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
AxtInfo (tables)</td>
<td>31 Mar 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial mm3 Release (using NCBI Build 30): Assembly, Gap, Map Contigs, GC Percent, RefSeq Genes, Genscan Genes, Gene Bounds, Mouse mRNAs, Spliced ESTs, Mouse ESTs, CpG Islands, RepeatMasker, Simple Repeats, Overlap SNPs (dbSnp 111)</td>
<td>07 Mar 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=rn3></A>Rat Jun 2003 (rn3, Baylor HGSC v. 3.1)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
RGD SSLP</td>
<td>01 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
RGD ESTs</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
RGD Genes</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
RGD QTL</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
GNF Atlas 2</td>
<td>27 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix RG-U34A</td>
<td>24 May 2004</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>21 May 2004</td>
</tr>
<TR valign=top><TD align=left>
GC Percent (updated to 5-base window track)</td>
<td>19 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Proteome Browser</td>
<td>07 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Superfamily</td>
<td>06 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Ecores</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Ecores</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA (updated)</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Superfamily</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA (updated)</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Ecores, Tetraodon Ecores</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>03 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix RAE230</td>
<td>15 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (updated)</td>
<td>02 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Overlap SNPs (dbSnp 118)</td>
<td>25 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
Chromosome Band</td>
<td>23 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
ECgene Genes</td>
<td>22 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>18 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Genbank daily updates enabled</td>
<td>14 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain, Mouse Net (using mm3)</td>
<td>18 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (extended)</td>
<td>18 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Rat/Mouse/Human (using mm3 and hg16)</td>
<td>05 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Human Chain, Human Net (using hg16)</td>
<td>04 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Human Tight (using hg16)</td>
<td>03 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes</td>
<td>05 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes</td>
<td>30 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Fgenesh++ Genes</td>
<td>18 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial rn3 Release (using Baylor HGSC v. 3.1): Assembly, Gap, Bactigs, GC Percent, Bac End Pairs, RefSeq Genes, STS Markers, Geneid Genes, Genscan Genes, Rat mRNAs, Spliced ESTs, Rat ESTs, NonRat mRNAs, NonRat ESTs, CpG Islands, RepeatMasker, Simple Repeats, bigZips downloads, chromosome downloads</td>
<td>14 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Twinscan (using hg15)</td>
<td>14 Jul 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=rn2></A>Rat Jan. 2003 (rn2, Baylor HGSC v. 2.1)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Geneid</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes (update)</td>
<td>28 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes</td>
<td>30 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Fugu Blat (v. 3.0)</td>
<td>29 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>28 May 2003</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes</td>
<td>20 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Fgenesh++ Genes</td>
<td>17 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs</td>
<td>13 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Gene Bounds</td>
<td>13 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Twinscan (using hg13)</td>
<td>02 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial rn2 Release (using Baylor HGSC v. 2.1): Assembly, Gap, Bactigs, GC Percent, RefSeq Genes, STS Markers, Genscan Genes, Rat mRNAs, Spliced ESTs, Rat ESTs, NonRat mRNAs, CpG Islands, RepeatMasker, Simple Repeats, bigZips downloads, chromosome downloads</td>
<td>14 Mar 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=galGal2></A>Chicken Feb. 2004 (galGal2, Chicken Genome Sequencing Consortium Feb. 2004 release)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
RNA Genes</td>
<td>30 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
CpG Island (updated)</td>
<td>06 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes (updated)</td>
<td>17 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs, Genscan Genes, Geneid Genes, UniGene, CpG Island, Fugu Blat, GC Percent, Quality Scores, Human Chain, Human Net, Human Synteny, Self Chain, Self Net</td>
<td>10 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial galGal2 Release (using the Feb. 2004 assembly): Supercontigs, Assembly, Gap, Chicken mRNAs, Spliced ESTs, Chicken ESTs, NonChicken mRNAs, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosome, database.</td>
<td>01 Mar 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=fr1></A>Fugu Aug. 2002 (fr1, JGI V3.0)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Twinscan</td>
<td>29 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial fr1 Release (using JGI V3.0): Scaffolds, Gap, GC Percent, Ensembl Genes, Genscan Genes, Fugu mRNAs, Fugu ESTs, Non-Fugu mRNAs, CpG Islands, Human Blat, RepeatMasker, Simple Repeats. Downloads: bigZips, database</td>
<td>23 Sep 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=dm1></A>D. melanogaster Jan. 2003 (dm1, BDGP V3.1)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
miRNA (updated)</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Anopheles Ecores</td>
<td>17 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial dm1 Release (using BDGP V3.1): Gap, GC Percent, RefSeq Genes, BDGP Genes, BDGP Non-coding, Genscan Genes, Fruitfly mRNAs, Spliced ESTs, Fruitfly ESTs, Fugu Blat, D. pseud. Chain, D. pseud. Net, D. pseud Aligns, RepeatMasker, Simple Repeats. Downloads: bigZips, database</td>
<td>30 Oct 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=ce2></A>C. elegans March 2004 (ce2, WormBase v. WS120)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
WormBase Genes</td>
<td>26 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Self Chain</td>
<td>20 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial ce2 Release (using WormBase v. WS120): Assembly, Gap, GC Percent, WormBase Genes, WormBase Genefinder, RefSeq Genes, Twinscan, miRNA, Worm mRNAs, Spliced ESTs, Worm ESTs, Gene Bounds, Fugu Blat, Briggsae Waba, Briggsae Blastz, Briggsae Chain, Briggsae Net, Briggsae Aligns, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosome, annotations.</td>
<td>11 May 2004</td>
</tr>
<TR valign=top><TD align=left>
New worm asembly</td>
<td>07 May 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=ce1></A>C. elegans May 2003 (ce1, WormBase v. WS100)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
RefSeq Genes</td>
<td>19 May 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA (updated)</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>23 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA</td>
<td>19 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Sanger Links (data update)</td>
<td>28 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter</td>
<td>03 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
sangerPep table</td>
<td>23 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Fgenesh++ Genes</td>
<td>27 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial ce1 Release (using WormBase v. WS100): Assembly, Gap, GC Percent, RefSeq Genes, Sanger Genes, Worm mRNAs, Worm ESTs, Fugu Blat, Briggsae Waba, Briggsae Chain, Briggsae Blastz, RepeatMasker, Simple Repeats, Self Chain, chromosome downloads</td>
<td>24 Jun 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=cb1></A>C. briggsae July 2002 (cb1, WormBase v. cb25.agp8)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Elegans Chain (updated to ce2)</td>
<td>25 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Elegans Blastz (updated to ce2)</td>
<td>25 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial ce1 Release (using WormBase v. cb25.agp8): Assembly, Gap, GC Percent, Worm mRNAs, Worm ESTs, Elegans Chain, Elegans Blastz, RepeatMasker, Simple Repeats, data set downloads</td>
<td>24 Jun 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=sacCer1></A>S. cerevisiae Oct. 2003 (sacCer1, SGD sequence data as of 1 Oct. 2003)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Initial sacCer1 Release (based on sequence in SGD as of 1 Oct 2003): WashU Clones, GC Percent, SGD Genes, SGD Other, Yeast ESTs, Spliced ESTs, Yeast ESTs, Multiz Yeast, Simple Repeats. Downloads: bigZips, database.</td>
<td>14 Jan 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=sc1></A>SARS Apr. 2003 (sc1, GenBank sequence as of 14 Apr 2003)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
bigZips downloads</td>
<td>05 Jun 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial sc1 Release (based on sequence deposited into GenBank as of 14 Apr 2003): GC Percent, GenBank Protein Annotations, Fgenesv+ Genes, Viral mRNAs, Viral mRNAs Chained, Other SARS, Coronavirus Proteins</td>
<td>23 May 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=zoo1></A>NISC (Zoo) Jun. 2002 (zoo1, Comparative Sequencing Program Target 1)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Downloads: Annotation database (by organism)</td>
<td>23 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial NISC Target 1 Release: Assembly, Gap, GC Samples, Gene Annotations, Genscan Genes, mRNAs, NonOrganism mRNAs, NonOrganism ESTs, Tetraodon Blat, ZooSeq, RepeatMasker, Simple Repeats. Human only: RefSeq Genes, ESTs, Spliced ESTs, DNase HyperSens, MCS (Binomial), Binomial Cons Score, MCS (ParsPVal), Parsimony P-Values, MCS InterMerged, MCS Inter</td>
<td>15 Aug 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
<!-- END SECOND SECTION ---------------------------------------------------------->

</TD></TR></TABLE>
</BODY></HTML>

