<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>

	
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META http-equiv="Content-Script-Type" content="text/javascript">
	<TITLE>
Track and Table Releases	</TITLE>
	<LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">

<TR><TH COLSPAN=1 ALIGN="left">
<IMG SRC="/images/title.jpg">
</TH></TR>


<!--HOTLINKS BAR---------------------------------------------------------->
<TR><TD COLSPAN=3 HEIGHT=40 >
<table bgcolor="000000" cellpadding="1" cellspacing="1" width="100%%" height="27">
<tr bgcolor="2636D1"><td valign="middle">
	<table BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24"><TR>
	 	<TD VALIGN="middle"><font color="#89A1DE">&nbsp;

&nbsp;<A HREF="/index.html" class="topbar">
           Home</A> &nbsp; - &nbsp;
       <A HREF="/cgi-bin/hgGateway" class="topbar">
           Genomes</A> &nbsp; - &nbsp;
       <A HREF="/cgi-bin/hgNear" class="topbar">
           Gene Sorter</A> &nbsp; - &nbsp;
       <A HREF="/cgi-bin/hgBlat?command=start&" class="topbar">           Blat</A> &nbsp; - &nbsp;
       <A HREF="/cgi-bin/hgPcr" class="topbar">
           PCR</A> &nbsp; - &nbsp;
       <A HREF="/cgi-bin/hgTables" class="topbar">
           Tables</A> &nbsp; - &nbsp;
       <A HREF="/FAQ.html" class="topbar">
           FAQ</A> &nbsp; - &nbsp;

       <A HREF="/goldenPath/help/hgTracksHelp.html"
       class="topbar">
           Help</A> 
&nbsp;</font></TD>
       </TR></TABLE>
</TD></TR></TABLE>
</TD></TR>	


<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=3>	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp;
Track and Table Releases</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
	

 This page contains track and table release information for the following genome assemblies:<br>
<ul>
<li><a CLASS="toc" HREF="#recent" ><b>10 Latest Changes (all assemblies)</b></a></li><li><a CLASS="toc" HREF="#hg17">Human May 2004 (hg17)</a></li>
<li><a CLASS="toc" HREF="#hg16">Human July 2003 (hg16)</a></li>
<li><a CLASS="toc" HREF="#hg15">Human April 2003 (hg15)</a></li>
<li><a CLASS="toc" HREF="#panTro1">Chimp Nov. 2003 (panTro1)</a></li>
<li><a CLASS="toc" HREF="#canFam1">Dog Jul. 2004 (canFam1)</a></li>
<li><a CLASS="toc" HREF="#bosTau1">Cow Sep. 2004 (bosTau1)</a></li>
<li><a CLASS="toc" HREF="#mm5">Mouse May 2004 (mm5)</a></li>
<li><a CLASS="toc" HREF="#mm4">Mouse Oct. 2003 (mm4)</a></li>
<li><a CLASS="toc" HREF="#mm3">Mouse Feb. 2003 (mm3)</a></li>
<li><a CLASS="toc" HREF="#rn3">Rat Jun 2003 (rn3)</a></li>
<li><a CLASS="toc" HREF="#rn2">Rat Jan. 2003 (rn2)</a></li>
<li><a CLASS="toc" HREF="#monDom1">Opossum Oct 2004 (monDom1)</a></li>
<li><a CLASS="toc" HREF="#galGal2">Chicken Feb. 2004 (galGal2)</a></li>
<li><a CLASS="toc" HREF="#xenTro1">X. tropicalis Oct. 2004 (xenTro1)</a></li>
<li><a CLASS="toc" HREF="#danRer2">Zebrafish June 2004 (danRer2)</a></li>
<li><a CLASS="toc" HREF="#danRer1">Zebrafish Nov. 2003 (danRer1)</a></li>
<li><a CLASS="toc" HREF="#tetNig1">Tetraodon Feb. 2004 (tetNig1)</a></li>
<li><a CLASS="toc" HREF="#fr1">Fugu Aug. 2002 (fr1)</a></li>
<li><a CLASS="toc" HREF="#ci1">C. intestinalis Dec. 2002 (ci1)</a></li>
<li><a CLASS="toc" HREF="#dm2">D. melanogaster Apr. 2004 (dm2)</a></li>
<li><a CLASS="toc" HREF="#dm1">D. melanogaster Jan. 2003 (dm1)</a></li>
<li><a CLASS="toc" HREF="#droYak1">D. yakuba Apr. 2004 (droYak1)</a></li>
<li><a CLASS="toc" HREF="#droAna1">D. ananassae Jul. 2004 (droAna1)</a></li>
<li><a CLASS="toc" HREF="#dp2">D. pseudoobscura Aug. 2003 (dp2)</a></li>
<li><a CLASS="toc" HREF="#droVir1">D. virilis Jul. 2004 (droVir1)</a></li>
<li><a CLASS="toc" HREF="#droMoj1">D. mojavensis Aug. 2004 (droMoj1)</a></li>
<li><a CLASS="toc" HREF="#apiMel2">A. mellifera Jan. 2005 (apiMel2)</a></li>
<li><a CLASS="toc" HREF="#apiMel1">A. mellifera Jul. 2004 (apiMel1)</a></li>
<li><a CLASS="toc" HREF="#anoGam1">A. gambiae Feb. 2003 (anoGam1)</a></li>
<li><a CLASS="toc" HREF="#ce2">C. elegans March 2004 (ce2)</a></li>
<li><a CLASS="toc" HREF="#ce1">C. elegans May 2003 (ce1)</a></li>
<li><a CLASS="toc" HREF="#cb1">C. briggsae July 2002 (cb1)</a></li>
<li><a CLASS="toc" HREF="#sacCer1">S. cerevisiae Oct. 2003 (sacCer1)</a></li>
<li><a CLASS="toc" HREF="#sc1">SARS Apr. 2003 (sc1)</a></li>
<li><a CLASS="toc" HREF="#zoo1">NISC (Zoo) Jun. 2002 (zoo1)</a></li>
</ul>
<p>
 For more information about the tracks and tables listed on this page, refer to the <a href=/goldenPath/gbdDescriptions.html>Description of the annotation database</a> and the <a href=/goldenPath/help/hgTracksHelp.html#IndivTracks>User's Guide</a>.<p>
<em>Last updated 09 Mar 2005. <a HREF="mailto:genome@soe.ucsc.edu">Inquiries and feedback welcome</a>.</em>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=recent></A> 10 Latest Changes (all assemblies)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR>
<TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
<TD nowrap><FONT color="#006666"><B>Assembly</B></FONT></TD>
<TD nowrap><FONT color="#006666"><B>Release Date</B></FONT></TD>
</TR>
<TR valign=top><TD align=left>
Map Contigs (non-C57BL contigs removed)</td>
<td>mm5</td>
<td>04 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
FlyReg (Drosophila DNase I Footprint Database)</td>
<td>dm2</td>
<td>04 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
CCDS genes (enabled robust details page)</td>
<td>hg17</td>
<td>04 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
Non-Human RefSeqs</td>
<td>hg17</td>
<td>04 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins (data update)</td>
<td>xenTro1</td>
<td>03 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes (update ensGene table to correct chrMT -> chrM)</td>
<td>mm5</td>
<td>01 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
D. melanogaster Proteins</td>
<td>apiMel2</td>
<td>28 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Assembly Scaffolds</td>
<td>apiMel2</td>
<td>28 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
FlyBase Genes, FB Noncoding (updated FlyBase annotations from version 4.0 to 4.1)</td>
<td>dm2</td>
<td>28 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
STS Markers (updated mapping)</td>
<td>hg17</td>
<td>28 Feb 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=hg17></A>Human May 2004 (hg17, NCBI Build 35)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
CCDS genes (enabled robust details page)</td>
<td>04 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
Non-Human RefSeqs</td>
<td>04 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
STS Markers (updated mapping)</td>
<td>28 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Yale Pseudo (Processed Pseudogenes from yale)</td>
<td>25 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Cow Chain/Net</td>
<td>18 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Ecores</td>
<td>15 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
RGD QTL (data update)</td>
<td>14 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Bounds
(Gene Boundaries as Defined by RNA and Spliced EST Clusters)</td>
<td>10 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Acembly (AceView Gene Models)</td>
<td>07 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
TFBS Conserved (transcr. factor binding sites)</td>
<td>04 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
SNPs (dbSNP build 124)</td>
<td>01 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
SNP track enhancement to support exceptions from dbSnp. (dbSnp build 123)</td>
<td>20 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>10 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
sno/miRNA - C/D and H/ACA Box snoRNAs, scaRNAs, and microRNAs from Weber and Griffiths-Jones (replaces the miRNA track)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish (danRer2) Chain and Net</td>
<td>06 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter extension (rankProp, psi-blast)</td>
<td>23 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Superfamily</td>
<td>21 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Proteome Browser (data update)</td>
<td>18 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (data update)</td>
<td>18 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter (data update)</td>
<td>18 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA data correction for hsa-mir-30a</td>
<td>15 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
SNPs (track based on 'snp' table, not 'snpMap' table)</td>
<td>05 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>24 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
X. tropicalis Net</td>
<td>22 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
X. tropicalis Chain</td>
<td>22 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
ECgene new track, EST clustering gene prediction</td>
<td>22 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
WSSD Duplication (from Xinwei She and Evan Eichler)</td>
<td>29 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Segmental Dups (from Xinwei She and Evan Eichler)</td>
<td>29 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
UniGene version 171 added to Human browser</td>
<td>20 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Affy U133Plus2</td>
<td>13 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
RDG QTL</td>
<td>13 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (add 3', 5' UTR mRNA secondary structure predictions)</td>
<td>11 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (add BioCyc pathways links)</td>
<td>11 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes</td>
<td>01 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation (data update), Most Conserved added</td>
<td>21 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Self Chain</td>
<td>21 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter Expression Distance (expanded data set)</td>
<td>16 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Regions</td>
<td>15 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Chain, Net</td>
<td>13 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
MicroRNAs from the miRNA Registry</td>
<td>09 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan gene predictions</td>
<td>08 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Retroposed Genes</td>
<td>01 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
fishClones</td>
<td>13 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
H-Inv (Human gene database, with full-length cDNA clones)</td>
<td>13 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter Expression Distance</td>
<td>11 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter Gene Maps</td>
<td>11 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter Fundamentals</td>
<td>11 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid</td>
<td>10 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial hg17 release (using NCBI Build 35): Chromosome Band, STS Markers, FISH Clones, Recomb Rate, Map Contigs, Assembly, Gap, Coverage, BAC End Pairs, Fosmid End Pairs, GC Percent, Known Genes, Proteome Browser, RefSeq Genes, MGC Genes, Genscan Genes, Human mRNAs, Spliced ESTs, Human ESTs, NonHuman mRNAs, NonHuman ESTs, CpG Islands, Fugu Blat, 
Conservation, Fugu Chain, Fugu Net, Zebrafish Chain, Zebrafish Net, Chicken Chain, Chicken Net, Rat Chain, Rat Net, Mouse Chain, Mouse Net, Dog Chain, Dog Net, Chimp Chain, Chimp Net, RepeatMasker, Simple Repeats.

 Downloads: bigZips, chromosomes, database</td>
<td>23 Jul 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=hg16></A>Human July 2003 (hg16, NCBI Build 34)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Yale Pseudo (Processed Pseudogenes from yale)</td>
<td>25 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
RNA/Affy Sites (Affymetrix PolyA+ RNA from retinoic acid-stimulated HL60 cells at 4 timepoints, in ENCODE regions)
</td>
<td>31 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
RNA/Affy Signal (Affymetrix PolyA+ RNA abundance from retinoic acid-stimulated HL60 cells at 4 timepoints, in ENCODE regions)
</td>
<td>31 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIP/Affy Pol2 Sites (Affymetrix ChIP/Chip of Pol2 from retinoic acid-stimulated HL60 cells at 4 timepoints, in ENCODE regions)
</td>
<td>31 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIP/Affy Pol2 Pval (Affymetrix ChIP/Chip enrichment of Pol2 from retinoic acid-stimulated HL60 cells at 4 timepoints, in ENCODE regions)
</td>
<td>31 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIP/chip data in ENCODE regions</td>
<td>05 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
DNA Rep/UVA (ENCODE DNA Replication data from Dutta Lab, University of VIrginia)</td>
<td>03 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
Promoter/Stanford (Promoters in ENCODE regions, from Myers lab at Stanford)</td>
<td>02 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
DNaseI-HS/NHGRI (ENCODE DNaseI Hypersensitive sites, from Collins lab, NHGRI)</td>
<td>01 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
WSSD Overlay (from Xinwei She and Evan Eichler)</td>
<td>29 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
WSSD Coverage (from Xinwei She and Evan Eichler)</td>
<td>29 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
WSSD Duplication (from Xinwei She and Evan Eichler)</td>
<td>29 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Segmental Dups (from Xinwei She and Evan Eichler)</td>
<td>29 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Perlegen Haplotypes (8 records changed to correct block ordering)</td>
<td>11 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl EST Genes</td>
<td>10 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Vega Gene: data update (extending 3' UTR for 398 annotations).</td>
<td>23 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan: data update (additional exons added for 27 annotations).</td>
<td>20 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes: data update (64 annotations changed by 1 or 2 bases).</td>
<td>20 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Exoniphy: Conserved Exon Predictions</td>
<td>14 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Retroposed Genes</td>
<td>19 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
H-Inv (Full-length cDNA from the H-Invitational Human Gene Annotation Database)</td>
<td>08 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
ensTranscript and ensGeneXref tables dropped (replaced by
ensemblXref3)  </td>
<td>03 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Gene (updated to 34d), Superfamily (updated)</td>
<td>03 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
GC% Win20K (dropped - replaced by 5-base window GC Percent track)</td>
<td>01 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
Alt-Splicing (data update)</td>
<td>27 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Chain, Fugu Net (using fr1)</td>
<td>24 May 2004</td>
</tr>
<TR valign=top><TD align=left>
STS Markers (updated)</td>
<td>24 May 2004</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>21 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Chimp Recip Chain, Chimp Recip Net (updated to panTro1 assembly)
</td>
<td>21 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Chimp Chain (updated to panTro1 assembly)</td>
<td>21 May 2004</td>
</tr>
<TR valign=top><TD align=left>
RNA Genes</td>
<td>19 May 2004</td>
</tr>
<TR valign=top><TD align=left>
GC Percent (updated to 5-base window track)</td>
<td>19 May 2004</td>
</tr>
<TR valign=top><TD align=left>
TFBS Conserved</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA (updated)</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Ecores</td>
<td>03 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Microsatellites</td>
<td>03 May 2004</td>
</tr>
<TR valign=top><TD align=left>
jaxOrtholog table</td>
<td>30 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Chicken Chain, Chicken Net (using galGal2)</td>
<td>29 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Ecores</td>
<td>29 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Acembly</td>
<td>20 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
SNPs (updated to dbSNP Build 120)</td>
<td>20 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
RNA Genes</td>
<td>13 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Hu/Mouse/Rat/Chick removed (replaced by Hu/Chp/Mo/Ra/Chk)</td>
<td>12 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Hu/Chp/Mo/Ra/Chk</td>
<td>12 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (updated to use proteins040315)</td>
<td>09 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix U133 (updated)</td>
<td>09 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
GNF Atlas 2</td>
<td>09 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix GNF1H</td>
<td>09 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
ECgene Genes (updated to include new peptide table)</td>
<td>05 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Human/Mouse/Rat removed (replaced by Hu/Mouse/Rat/Chick)</td>
<td>29 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Hu/Mouse/Rat/Chick (using mm3, rn3, galGal2) - replaces Human/Mouse/Rat track</td>
<td>19 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
ECgene Genes (updated)</td>
<td>19 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA</td>
<td>19 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
SGP (updated to use mm4)</td>
<td>05 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Map Contigs (updated)</td>
<td>02 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Perlegen (updated)</td>
<td>02 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Fgenesh++ Genes</td>
<td>13 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
affyU133, affyU95 tables</td>
<td>10 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (updated)</td>
<td>06 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
2X Reg Potential, 3X Reg Potential</td>
<td>05 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes (update)</td>
<td>04 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
Mm3 Chain, Mm3 Net removed (replaced by mm4 versions)</td>
<td>20 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
chr4_random_gap, chr4_random_gold, chr8_random_gap, chr8_random_gold, chrX_random_gap, chrX_random_gold, chrUn_random_gap, chrUn_random_gold tables (updated)</td>
<td>16 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Chimp Diff</td>
<td>16 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Affy10K</td>
<td>14 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Mm4 Net (updates Mouse Net using mm4)</td>
<td>12 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Mm4 Chain (updates Mouse Chain using mm4)</td>
<td>09 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Mm4 Tight (updates Mouse Tight using mm4)</td>
<td>08 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
ECgene Genes</td>
<td>22 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
vsPt0/axtBest downloads (chimp reciprocal best)</td>
<td>22 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
Alt-Splicing</td>
<td>18 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
Twinscan Genes</td>
<td>18 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
Chimp All Chains, Chimp Net, Chimp sequence and chain downloads</td>
<td>10 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs (updated)</td>
<td>19 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>18 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>18 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Chain Self</td>
<td>15 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
UniGene_2, sage</td>
<td>12 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Gene Bounds</td>
<td>12 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Chain Self</td>
<td>11 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Vega Genes, Vega Pseudogenes</td>
<td>07 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
NCI60</td>
<td>24 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (add UTR 3' and 5' folding data)</td>
<td>24 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Haplotype Blocks, Perlegen (chr 21 and 22 only)</td>
<td>24 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Human/Mouse/Rat downloads</td>
<td>24 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Synteny</td>
<td>21 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Genbank daily updates enabled</td>
<td>14 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Affy 120K SNPs</td>
<td>13 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Random SNPs, Overlap SNPs (dbSnp 117)</td>
<td>10 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Human/Mouse/Rat (using mm3 and rn3)</td>
<td>09 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (update)</td>
<td>08 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Fugu Blat (v. 3.0)</td>
<td>02 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
affyRatio table</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
vsMm3, vsRn3 data downloads</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>18 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Geneid</td>
<td>18 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain (using mm3)</td>
<td>10 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Net (using mm3) (contains fixes to chr1 data)</td>
<td>10 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Tight (using mm3) (contains fixes to chr1 data)</td>
<td>10 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
First EF (contains fixes to chrX and chrY)</td>
<td>09 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Net (using mm3)</td>
<td>02 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Rat Net (using rn3)</td>
<td>02 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Rat Chain</td>
<td>29 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain</td>
<td>29 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Tight Mouse</td>
<td>29 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Self Chain</td>
<td>29 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes</td>
<td>26 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
STS Markers (updated)</td>
<td>11 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial hg16 release (using NCBI Build 34): Chromosome Band, FISH Clones, Assembly, Gap, STS Markers, Map Contigs, Recomb Rate, GC Percent, Coverage, BAC End Pairs, Fosmid End Pairs, RefSeq Genes, MGC Genes, Genscan Genes, Human mRNAs, Spliced ESTs, Human ESTs, NonHuman mRNAs, NonHuman ESTs, CpG Islands, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosomes, database</td>
<td>08 Aug 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=hg15></A>Human April 2003 (hg15, NCBI Build 33)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Transcription and Transfrags</td>
<td>13 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Retroposed Genes</td>
<td>22 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
RNA Genes</td>
<td>19 May 2004</td>
</tr>
<TR valign=top><TD align=left>
TFBS Conserved</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
GNF Ratio (updated)</td>
<td>14 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
GNF Ratio (updated)</td>
<td>05 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes (updated)</td>
<td>05 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (updated to use proteins040315)</td>
<td>05 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes (update)</td>
<td>30 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Affy10K</td>
<td>14 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Random SNPs, Overlap SNPs (dbSnp 112)</td>
<td>31 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Affy 120K SNPs</td>
<td>13 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
KnownGenes (extended)</td>
<td>08 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
affyRatio table</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
ensPep table</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Segmental Dups</td>
<td>18 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
First EF (contains fixes to chrX and chrY)</td>
<td>09 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
First EF</td>
<td>27 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Downloads: Mouse (mm3)/Rat (rn2)/Human multiple alignments</td>
<td>22 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
NonHuman ESTs</td>
<td>25 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>15 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Cons (using mm3)</td>
<td>09 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Best Mouse (using mm3)</td>
<td>09 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
bigZips (upstream files updated)</td>
<td>09 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Blastz axtTight mouse alignments (using mm3)</td>
<td>25 Jun 2003</td>
</tr>
<TR valign=top><TD align=left>
Acembly</td>
<td>05 Jun 2003</td>
</tr>
<TR valign=top><TD align=left>
AxtInfo (updated to use axtNetMm3)</td>
<td>30 May 2003</td>
</tr>
<TR valign=top><TD align=left>
UniGene (v. Hs 160)</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Tight Rat (using rn2)</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Gene Bounds</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Rat Synteny (using rn2)</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Human/Mouse/Rat (using mm3, rn2)</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
MGC Genes</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Fosmid End Pairs</td>
<td>20 May 2003</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes (using mm3)</td>
<td>20 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Haplotype Blocks (chr 21 and 22 only)</td>
<td>20 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Overlap and Random SNPs (dbSnp 114)</td>
<td>15 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Blastz axtBest mouse alignments (using mm3)</td>
<td>13 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Human/Mouse Chain and Net (using mm3)</td>
<td>09 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>08 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Twinscan (using mm2)</td>
<td>08 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Recomb Rate</td>
<td>01 May 2003</td>
</tr>
<TR valign=top><TD align=left>
geneBands (table only)</td>
<td>30 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
NCI60</td>
<td>30 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
GNF Ratio</td>
<td>30 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
GC Percent (data reloaded)</td>
<td>29 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Chromosome Band</td>
<td>28 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Fgenesh++ Genes</td>
<td>28 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
FISH Clones</td>
<td>28 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs</td>
<td>25 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Coverage</td>
<td>25 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
liftAll.zip (correction for NT_008183 - table only)</td>
<td>16 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial hg15 release (using NCBI Build 33): Assembly, Gap, STS Markers, Map Contigs, GC Percent, Known Genes, RefSeq Genes, Genscan Genes, Human mRNAs, Spliced ESTs, Human ESTs, NonHuman mRNAs, CpG Islands, Fugu Blat (v 3.0), Tight Mouse (mm3), RepeatMasker, Simple Repeats, Mouse Synteny (mm3)</td>
<td>14 Apr 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=panTro1></A>Chimp Nov. 2003 (panTro1, CGSC Build 1 Version 1 Arachne assembly)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>22 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Blat</td>
<td>14 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Genscan Genes</td>
<td>10 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Contigs</td>
<td>28 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>27 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Deletions (updated)</td>
<td>14 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes (chr1 updated)</td>
<td>07 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>24 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Best, vsHg15 downloads</td>
<td>02 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial panTro1 Release (using the Build 1 Version 1 Arachne assembly dated 13 Nov. 2003): Scaffolds, Gap, GC Percent, Quality Scores, Human RefSeq, Chimp mRNAs, Spliced ESTs, Chimp ESTs, NonChimp mRNAs, NonChimp ESTs, CpG Islands, Human Chain, Human Net, Human Deletions, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosome, database.</td>
<td>23 Feb 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=canFam1></A>Dog Jul. 2004 (canFam1, Broad Institute v. 1.0)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Genscan (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>15 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation (data update)</td>
<td>06 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Chain, Human Net (chained and netted blastz alignments to May 2004 Human)</td>
<td>30 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation (dog-human-mouse multiz alignments, scored by phastCons)</td>
<td>29 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain, Mouse Net (chained and netted blastz alignments to May 2004 Mouse)</td>
<td>29 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu BLAT (translated BLAT of Fugu scaffolds to Dog genome)</td>
<td>26 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial canFam1 release (using Broad Institute WGS v1.0): Assembly, Gap, GC Percent, Genscan Genes, Dog mRNAs, Spliced ESTs, Dog ESTs, NonDog mRNAs, RepeatMasker, Simple Repeats

 Downloads: bigZips, chromosomes, database</td>
<td>13 Jul 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=bosTau1></A>Cow Sep. 2004 (bosTau1, Baylor v1.0)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Initial bosTau1 release (Baylor ): Assembly, Gap, GC Percent, Genscan, Cow mRNAs, Spliced ESTs, Cow ESTs, CpG Islands, Human Proteins, Human Chain/Net, RepeatMasker, Simple Repeats.
Downloads: bigZips, database, comparatives.</td>
<td>22 Feb 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=mm5></A>Mouse May 2004 (mm5, NCBI Build 33)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Map Contigs (non-C57BL contigs removed)</td>
<td>04 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes (update ensGene table to correct chrMT -> chrM)</td>
<td>01 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>10 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Genscan (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish (danRer2) Chain and Net</td>
<td>06 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Superfamily</td>
<td>22 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>21 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
ECgene new track, EST clustering gene prediction</td>
<td>22 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
NIA Gene</td>
<td>17 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
MicroRNAs - miRNA Registry</td>
<td>04 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes</td>
<td>04 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Most Conserved (data update)</td>
<td>23 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Cytoband (updated data from NCBI)</td>
<td>16 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Chain, Net</td>
<td>13 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Blat</td>
<td>09 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
STS Markers</td>
<td>26 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
	</td>
<td>20 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Genscan</td>
<td>11 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid</td>
<td>11 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation (Rat/Human/Dog/Chicken Multiz Alignments and PhyloHMM Conservation)</td>
<td>08 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Dog Chain and Net</td>
<td>08 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Rat Chain and Net</td>
<td>08 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Chicken Chain and Net</td>
<td>08 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Chain and Net</td>
<td>08 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter Expression Distance</td>
<td>04 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter Gene Maps</td>
<td>04 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter Orthologs</td>
<td>04 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter Fundamentals</td>
<td>04 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs</td>
<td>30 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Expression Tracks (mm5)</td>
<td>29 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial mm5 release (using NCBI Build 33): Assembly, Gap, GC Percent, Known Genes, Proteome Browser, RefSeq Genes, MGC Genes, Mouse mRNAs, Spliced ESTs, Mouse ESTs, NonMouse mRNAs, NonMouse ESTs, CpG Islands, RepeatMasker, Simple Repeats.  Downloads: bigZips, chromosomes, database.</td>
<td>16 Jul 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=mm4></A>Mouse Oct. 2003 (mm4, NCBI Build 32)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
NIAGene, NIA Mouse Gene Index Version 3 track for mm4.
</td>
<td>18 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
SNPs corrections for incorrect alignments</td>
<td>27 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
STS Markers RH values corrected</td>
<td>19 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter: MOE430 ID</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter: Max Rinn Sex</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Rinn Sex Exp</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter: Locus Link</td>
<td>26 May 2004</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>21 May 2004</td>
</tr>
<TR valign=top><TD align=left>
GC Percent (updated to 5-base window track)</td>
<td>19 May 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA (updated)</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix MOE430</td>
<td>28 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix GNF1M</td>
<td>27 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (updated to proteins040314)</td>
<td>16 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Superfamily</td>
<td>16 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Chromosome Band</td>
<td>13 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
STS Markers</td>
<td>19 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA</td>
<td>19 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
GNF U74A, GNF U74B, GNF U74C, Affymetrix U74</td>
<td>08 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Overlap SNPs (using dbSnp 118)</td>
<td>25 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes</td>
<td>09 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>08 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes</td>
<td>08 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial mm4 Release (using NCBI Build 32): Assembly, Gap, Map Contigs, BAC End Pairs, GC Percent, Known Genes, RefSeq Genes, MGC Genes, Genscan Genes, Mouse mRNAs, Spliced ESTs, Mouse ESTs, NonMouse mRNAs, NonMouse ESTs, CpG Islands, Human Chain, Human Net, Human Tight, Rat Chain, Rat Net, Rat Tight, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosome, database.</td>
<td>21 Nov 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=mm3></A>Mouse Feb. 2003 (mm3, NCBI Build 30)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Geneid</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Ecores</td>
<td>04 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Ecores</td>
<td>04 May 2004</td>
</tr>
<TR valign=top><TD align=left>
GNF U74A, GNF U74B, GNF U74C, Affymetrix U74</td>
<td>08 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Recomb Rate</td>
<td>02 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Chromosome Band</td>
<td>23 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
ECgene Genes</td>
<td>22 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>18 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Rat Synteny (updated)</td>
<td>27 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
knownGenes (updated)</td>
<td>20 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter (updated)</td>
<td>20 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Genbank daily updates enabled</td>
<td>14 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter</td>
<td>10 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
proteins092903 database (used by Known Genes)</td>
<td>09 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
axtBest/chr2.axt.gz (was missing from axtBest data set)</td>
<td>02 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
ensemblGenes (proteins added)</td>
<td>01 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Human Best (updated to use hg15)</td>
<td>25 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Superfamily</td>
<td>25 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Bac End Pairs (added data for chrX)</td>
<td>25 Jun 2003</td>
</tr>
<TR valign=top><TD align=left>
Human Cons (using hg15)</td>
<td>24 Jun 2003</td>
</tr>
<TR valign=top><TD align=left>
Human Chain/Net (updated from hg13 to hg15)</td>
<td>17 Jun 2003</td>
</tr>
<TR valign=top><TD align=left>
Ensembl EST Genes</td>
<td>13 Jun 2003</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>13 Jun 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse ESTs, Spliced ESTs, Gene Bounds (additions for random chromosomes)</td>
<td>30 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>28 May 2003</td>
</tr>
<TR valign=top><TD align=left>
MGC Genes</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Human Synteny (using hg15)</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands (random data added)</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Rat Synteny (using rn1)</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Twinscan (using hg13)</td>
<td>22 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes</td>
<td>16 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Blastz human alignments downloads (using hg15)</td>
<td>14 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Blastz rat alignments downloads (using rn2)</td>
<td>13 May 2003</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes (using hg13)</td>
<td>08 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Human/Mouse Chain and Net (using hg15)</td>
<td>02 May 2003</td>
</tr>
<TR valign=top><TD align=left>
STS Markers</td>
<td>29 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
BAC Ends Pairs</td>
<td>29 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Fugu Blat</td>
<td>29 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Simple Repeats (data added for randoms)</td>
<td>17 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Fgenesh++ Genes</td>
<td>17 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Best Human, Tight Human (using hg15)</td>
<td>02 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Best Rat, Tight Rat (using rn2)</td>
<td>02 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
NonMouse mRNAs</td>
<td>01 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
AxtInfo (tables)</td>
<td>31 Mar 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial mm3 Release (using NCBI Build 30): Assembly, Gap, Map Contigs, GC Percent, RefSeq Genes, Genscan Genes, Gene Bounds, Mouse mRNAs, Spliced ESTs, Mouse ESTs, CpG Islands, RepeatMasker, Simple Repeats, Overlap SNPs (dbSnp 111)</td>
<td>07 Mar 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=rn3></A>Rat Jun 2003 (rn3, Baylor HGSC v. 3.1)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>10 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
vsHg17 and vsMm5 pairwise alignments</td>
<td>01 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Cytoband Ideo</td>
<td>30 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
MGC Genes</td>
<td>24 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
ECgene track update, EST clustering gene predictions</td>
<td>22 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
Assembly/gold track update to fix fragStart off by 1</td>
<td>19 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain and Net (update to mm5)</td>
<td>13 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Chain, Human Net (update  hg15 --> hg17)</td>
<td>30 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Rat/Mouse/Human (updated)</td>
<td>20 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter for rn3</td>
<td>24 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
RGD SSLP</td>
<td>01 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
RGD ESTs</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
RGD Genes</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
RGD QTL</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
GNF Atlas 2</td>
<td>27 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix RG-U34A</td>
<td>24 May 2004</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>21 May 2004</td>
</tr>
<TR valign=top><TD align=left>
GC Percent (updated to 5-base window track)</td>
<td>19 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Proteome Browser</td>
<td>07 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Superfamily</td>
<td>06 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Ecores</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Ecores</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA (updated)</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Superfamily</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA (updated)</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Ecores, Tetraodon Ecores</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>03 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix RAE230</td>
<td>15 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (updated)</td>
<td>02 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Overlap SNPs (dbSnp 118)</td>
<td>25 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
Chromosome Band</td>
<td>23 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
ECgene Genes</td>
<td>22 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>18 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Genbank daily updates enabled</td>
<td>14 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain, Mouse Net (using mm3)</td>
<td>18 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (extended)</td>
<td>18 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Rat/Mouse/Human (using mm3 and hg16)</td>
<td>05 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Human Chain, Human Net (using hg16)</td>
<td>04 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Human Tight (using hg16)</td>
<td>03 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes</td>
<td>05 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes</td>
<td>30 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Fgenesh++ Genes</td>
<td>18 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial rn3 Release (using Baylor HGSC v. 3.1): Assembly, Gap, Bactigs, GC Percent, Bac End Pairs, RefSeq Genes, STS Markers, Geneid Genes, Genscan Genes, Rat mRNAs, Spliced ESTs, Rat ESTs, NonRat mRNAs, NonRat ESTs, CpG Islands, RepeatMasker, Simple Repeats, bigZips downloads, chromosome downloads</td>
<td>14 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Twinscan (using hg15)</td>
<td>14 Jul 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=rn2></A>Rat Jan. 2003 (rn2, Baylor HGSC v. 2.1)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Geneid</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter: rn2 added</td>
<td>01 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes (update)</td>
<td>28 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes</td>
<td>30 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Fugu Blat (v. 3.0)</td>
<td>29 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>28 May 2003</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes</td>
<td>20 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Fgenesh++ Genes</td>
<td>17 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs</td>
<td>13 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Gene Bounds</td>
<td>13 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Twinscan (using hg13)</td>
<td>02 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial rn2 Release (using Baylor HGSC v. 2.1): Assembly, Gap, Bactigs, GC Percent, RefSeq Genes, STS Markers, Genscan Genes, Rat mRNAs, Spliced ESTs, Rat ESTs, NonRat mRNAs, CpG Islands, RepeatMasker, Simple Repeats, bigZips downloads, chromosome downloads</td>
<td>14 Mar 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=monDom1></A>Opossum Oct 2004 (monDom1, Broad Inst. Prelim Oct04)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Initial monDom1 release (using Broad Inst. Prelim Oct04): Assembly, Gap, GC Percent, Quality Scores, Oppossum mRNAs, Non-Opposum mRNAs, Conservation, Most Conserved, Human Proteins, Human Chain/Net, Mouse Chain/Net, Chicken Chain/Net, Zebrafish Chain/Net, CpG Islands, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosomes, database.</td>
<td>18 Jan 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=galGal2></A>Chicken Feb. 2004 (galGal2, Chicken Genome Sequencing Consortium Feb. 2004 release)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Human Chain, Human Net (updated to May 2004 assembly)</td>
<td>24 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Updated SGP Genes.</td>
<td>29 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
cpgIslands (data update) replace WUSTL cpgIslands with calculations by Andy Law, Roslin Institute</td>
<td>29 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins</td>
<td>14 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Ecores</td>
<td>20 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Contamination (alignments of Maize mitochondrion, E. Coli and vector sequence to WGS assembly)</td>
<td>03 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
BGI SNPs, BGI Genes, BGI Coverage (Beijing Genomics Institute SNPs, gene annotations and read coverage locations) </td>
<td>02 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes</td>
<td>21 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>18 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>17 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
RNA Genes</td>
<td>30 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
CpG Island (updated)</td>
<td>06 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes (updated)</td>
<td>17 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs, Genscan Genes, Geneid Genes, UniGene, CpG Island, Fugu Blat, GC Percent, Quality Scores, Human Chain, Human Net, Human Synteny, Self Chain, Self Net</td>
<td>10 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial galGal2 Release (using the Feb. 2004 assembly): Supercontigs, Assembly, Gap, Chicken mRNAs, Spliced ESTs, Chicken ESTs, NonChicken mRNAs, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosome, database.</td>
<td>01 Mar 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=xenTro1></A>X. tropicalis Oct. 2004 (xenTro1, JGI v3.0)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Human Proteins (data update)</td>
<td>03 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
MGC Genes</td>
<td>14 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Genscan (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Frog Browser Initial Release (using JGI version 3.0): Assembly, Gap, GC Percent, Mapped Human, Genscan, Frog mRNA, Spliced ESTs, Frog ESTs, CpG Islands, Conservation, Most Conserved, Fugu Blat, Zebrafish Chain/Net, Mouse Chain/Net, Human Chain/Net, Chicken Chain/Net, RepeatMasker, Simple Repeats</td>
<td>17 Nov 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=danRer2></A>Zebrafish June 2004 (danRer2, Sanger Centre, Danio rerio Sequencing Project Zv4)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Human Proteins</td>
<td>11 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>10 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Zfish WZ EST Clusters</td>
<td>06 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Fugu (fr1) Chain and Net</td>
<td>06 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Mouse (mm5) Chain and Net</td>
<td>06 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Human (hg17) Chain and Net</td>
<td>06 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Initial danRer2 release (using Zv4): Contigs, Scaffolds, Gap, GC Percent, Radiation Hybrid Map, RefSeq Genes, ZGC Genes, Ensembl Genes, Zebrafish mRNAs, Spliced ESTs, Zebrafish ESTs, Non-Zebrafish mRNAs, Affy Zebrafish, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosomes, database.</td>
<td>23 Dec 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=danRer1></A>Zebrafish Nov. 2003 (danRer1, Sanger Centre, Danio rerio Sequencing Project Zv3)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
ZGC Genes</td>
<td>11 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
UniGene</td>
<td>04 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins (data update)</td>
<td>17 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Chain, Human Net</td>
<td>19 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs and BAC End Singles</td>
<td>13 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial danRer1 release (using Zv3): Contigs, Scaffolds, Gap, GC Percent, RefSeq Genes, Ensembl Genes, Zebrafish mRNAs, Spliced ESTs, Zebrafish ESTs, Zebrafish WZ EST clusters, NonZebrafish mRNAs, Affy Zebrafish, Fugu Chain, Fugu Net, Human Proteins, RepeatMasker, Simple Repeats. 

 Downloads: bigZips, chromosomes, database, vsFr1, vsHg17</td>
<td>28 Jul 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=tetNig1></A>Tetraodon Feb. 2004 (tetNig1, Genoscope and Broad Institute, V7)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Human Ecores</td>
<td>15 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Self Chain, Self Net</td>
<td>26 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Cytokine Genes (removed extensions)</td>
<td>10 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
HOX Genes (removed extensions)</td>
<td>10 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Genoscope GAZE (removed extensions)</td>
<td>10 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Genscan (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>29 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Cytokine Genes</td>
<td>17 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
HOX Genes</td>
<td>15 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Genoscope GAZE gene predictions</td>
<td>15 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain, Net (mm5)</td>
<td>13 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial tetNig1 release (Genoscope v6): Scaffolds, Gap, GC Percent, Short Match, Genscan, mRNAs, Spliced ESTs, ESTs, Xeno ESTs, Human Chain/Net, Fugu Chain/Net, RepeatMasker, Simple Repeats.
Downloads: bigZips, chromosomes, database, comparatives.</td>
<td>10 Sep 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=fr1></A>Fugu Aug. 2002 (fr1, JGI V3.0)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Zebrafish (danRer2) Chain and Net</td>
<td>06 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins</td>
<td>14 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Ecores, Mouse Ecores, Rat Ecores</td>
<td>07 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>29 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial fr1 Release (using JGI V3.0): Scaffolds, Gap, GC Percent, Ensembl Genes, Genscan Genes, Fugu mRNAs, Fugu ESTs, Non-Fugu mRNAs, CpG Islands, Human Blat, RepeatMasker, Simple Repeats. Downloads: bigZips, database</td>
<td>23 Sep 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=ci1></A>C. intestinalis Dec. 2002 (ci1, JGI v1.0)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Human Proteins (data update)</td>
<td>17 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial ci1 release (using JGI v1.0): Gap, GC Percent, Non-Ciona RefSeq Genes, SNAP Genes, JGI Genes, Ciona mRNAs, Spliced ESTs, Ciona ESTs, NonCiona mRNAs, C. savignyi Blat, Human Proteins, RepeatMasker, Simple Repeats

Downloads: bigZips, database</td>
<td>15 Jul 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=dm2></A>D. melanogaster Apr. 2004 (dm2, BDGP v. 4 / DHGP v. 3.2)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
FlyReg (Drosophila DNase I Footprint Database)</td>
<td>04 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
FlyBase Genes, FB Noncoding (updated FlyBase annotations from version 4.0 to 4.1)</td>
<td>28 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Initial D. melanogaster 2 release (BDGP Rel.4 euchromatic, DHGP Rel. 3.2 heterochromatic): Gap, GC Percent, Genscan, RefSeq Genes, Flybase, FB-Noncoding, Conservation, Most-Conserved,  mRNAs, xenoMrnas, Spliced ESTs, ESTs, {anoGam1, apiMel1, dp2, droAna1, droMoj1, droVir1, droYak1} Chain/Net, RepeatMasker, Simple Repeats.
Downloads: bigZips, database, comparatives.</td>
<td>14 Feb 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=dm1></A>D. melanogaster Jan. 2003 (dm1, BDGP v. 3)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
FlyBase Non-coding (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
FlyBase Genes (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
FlyReg</td>
<td>23 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>21 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation (chrUn-based dp1 replaced with scaffold-based dp2 for D. pseudoobscura sequence; rescored with latest phastCons)</td>
<td>01 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Incremental update to BDGP Insertions.</td>
<td>27 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins</td>
<td>14 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
D. pseudoobscura Chain, Net (updated to dp2)</td>
<td>01 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
BDGP Insertions (P element insertion locations from the BDGP Gene Disruption Project)</td>
<td>27 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
D. yakuba Chain/Net</td>
<td>12 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Anopheles Chain and Net</td>
<td>12 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation (D. yakuba, D. pseudoob., A. gambiae)</td>
<td>11 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter</td>
<td>01 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
BDGP Non-coding</td>
<td>30 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
Chromosome Bands (Cytology) from Flybase</td>
<td>30 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
BDGP Genes</td>
<td>30 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA (updated)</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Anopheles Ecores</td>
<td>17 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial dm1 Release (using BDGP V3.1): Gap, GC Percent, RefSeq Genes, BDGP Genes, BDGP Non-coding, Genscan Genes, Fruitfly mRNAs, Spliced ESTs, Fruitfly ESTs, Fugu Blat, D. pseud. Chain, D. pseud. Net, D. pseud Aligns, RepeatMasker, Simple Repeats. Downloads: bigZips, database</td>
<td>30 Oct 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=droYak1></A>D. yakuba Apr. 2004 (droYak1, WUSTL version 1.0)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Fly (dm1) Proteins (data update)</td>
<td>16 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
D. mel. Proteins (chained tblastn alignments of D. melanogaster proteins)</td>
<td>21 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
GC Percent</td>
<td>26 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial D. yakuba Release (version 1) - Assembly, Gap, Genscan Genes, D. yakuba mRNAs, Spliced ESTs, D. yakuba ESTs, Non-D. yakuba mRNAs, D. mel. Chain, D. mel. Net, RepeatMasker, Simple Repeats</td>
<td>23 Jul 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=droAna1></A>D. ananassae Jul. 2004 (droAna1, TIGR 15 July 2004)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Genscan (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
D. ananassae Browser Initial Release (TIGR 15 July 2004): Gap, GC Percent, Non-D.anan. Refseq, Genscan, D.anan. mRNA, Non D.anan. mRNAs, D. mel. proteins, D. mel. Chain/Net, RepeatMasker, Simple Repeats.  Downloads bigZips, database, vsDm1.</td>
<td>15 Dec 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=dp2></A>D. pseudoobscura Aug. 2003 (dp2, Baylor HGSC Freeze 1)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Genscan (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Initial dp2 release (Baylor freeze 1 August 2003): GC Percent, Xeno RefSeq, Twinscan Genes, Genscan Genes, mRNAs, Xeno mRNAs, D. melanogaster, proteins, D. mel. Chain, D. mel. Net, RepeatMasker, Simple Repeats.
Downloads: bigZips, chromosomes, database</td>
<td>01 Sep 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=droVir1></A>D. virilis Jul. 2004 (droVir1, Agencourt 12 July 2004)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Genscan (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
D. virilis Browser Initial Release (Agencourt 12 July 2004): Gap, GC Percent, Non-D. vir. Refseq, Genscan, D. vir. mRNA, Non-D. vir. mRNA, D. mel. proteins, D. mel. Chain/Net, RepeatMasker, Simple Repeats</td>
<td>15 Dec 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=droMoj1></A>D. mojavensis Aug. 2004 (droMoj1, Agencourt 11 Aug 2004)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Genscan (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
D. mojavensis Browser Initial Release (Agencourt 11 Aug 2004): Gap, GC Percent, Non-D.moj. Refseq, Genscan, D.moj. mRNA, Non D.moj. mRNAs, D. mel. proteins, D. mel. Chain/Net, RepeatMasker, Simple Repeats</td>
<td>15 Dec 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=apiMel2></A>A. mellifera Jan. 2005 (apiMel2, Baylor HGSC Amel_2.0)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
D. melanogaster Proteins</td>
<td>28 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Assembly Scaffolds</td>
<td>28 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
A. mellifera (Honeybee) Browser Jan. 2005 Release (apiMel2, Baylor HGSC Amel_2.0): Assembly, Gap, GC Percent, Genscan,  A. mellifera mRNAs, Non-A. mellifera mRNAs, A. mellifera Spliced ESTs, A. mellifera ESTs, D. melanogaser Chain/Net, RepeatMasker, Simple Repeats. Downloads: bigZips, database, vsDm2.</td>
<td>23 Feb 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=apiMel1></A>A. mellifera Jul. 2004 (apiMel1, Baylor HGSC Amel_1.2)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Genscan (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Honeybee Browser Initial Release (Baylor HGSC Amel_1.2): Assembly, Gap, GC Percent, Genscan, Honeybee mRNA, Spliced ESTs, Honeybee ESTs, Mapped D.melanogaster Proteins, D. melanogaser Chain/Net, RepeatMasker, Simple Repeats</td>
<td>22 Dec 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=anoGam1></A>A. gambiae Feb. 2003 (anoGam1, IAGEC v.MOZ2)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Genscan (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
GC Percent</td>
<td>22 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Fly (dm1) Proteins (data update)</td>
<td>16 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
D. mel. Proteins (chained tblastn alignments of D. melanogaster proteins)</td>
<td>21 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial anoGam1 release (MOZ2 draft genome sequence Feb. 2003): Assembly, Gap, Ensembl Genes, Genscan Genes, A. gambiae mRNAs, Spliced ESTs, A. gambiae ESTs, Non-A. gambiae mRNAs, Non-A.gambiae ESTs, D. mel Chain, D. mel. Net, RepeatMasker, Simple Repeats

 Downloads: bigZips, chromosomes, database</td>
<td>10 Aug 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=ce2></A>C. elegans March 2004 (ce2, WormBase v. WS120)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Conservation, Most Conserved</td>
<td>17 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins</td>
<td>14 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter</td>
<td>04 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
WormBase Genes</td>
<td>26 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Self Chain</td>
<td>20 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial ce2 Release (using WormBase v. WS120): Assembly, Gap, GC Percent, WormBase Genes, WormBase Genefinder, RefSeq Genes, Twinscan, miRNA, Worm mRNAs, Spliced ESTs, Worm ESTs, Gene Bounds, Fugu Blat, Briggsae Waba, Briggsae Blastz, Briggsae Chain, Briggsae Net, Briggsae Aligns, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosome, annotations.</td>
<td>11 May 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=ce1></A>C. elegans May 2003 (ce1, WormBase v. WS100)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
RefSeq Genes</td>
<td>19 May 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA (updated)</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>23 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA</td>
<td>19 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Sanger Links (data update)</td>
<td>28 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter</td>
<td>03 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
sangerPep table</td>
<td>23 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Fgenesh++ Genes</td>
<td>27 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial ce1 Release (using WormBase v. WS100): Assembly, Gap, GC Percent, RefSeq Genes, Sanger Genes, Worm mRNAs, Worm ESTs, Fugu Blat, Briggsae Waba, Briggsae Chain, Briggsae Blastz, RepeatMasker, Simple Repeats, Self Chain, chromosome downloads.</td>
<td>24 Jun 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=cb1></A>C. briggsae July 2002 (cb1, WormBase v. cb25.agp8)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Twinscan</td>
<td>03 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Elegans Chain (updated to ce2)</td>
<td>25 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Elegans Blastz (updated to ce2)</td>
<td>25 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial cb1 Release (using WormBase v. cb25.agp8): Assembly, Gap, GC Percent, Worm mRNAs, Worm ESTs, Elegans Chain, Elegans Blastz, RepeatMasker, Simple Repeats, data set downloads.</td>
<td>24 Jun 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=sacCer1></A>S. cerevisiae Oct. 2003 (sacCer1, SGD sequence data as of 1 Oct. 2003)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Known Genes -- UCSC SAM-T02 protein structure analysis and prediction.</td>
<td>10 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Regulatory Code and Reg. CHIP/CHIP</td>
<td>17 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation (data update), Most Conserved (data update)</td>
<td>28 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter: Regulatory Module column</td>
<td>24 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Regulatory Module (first public release of Eran Segal's regulatory module track.)</td>
<td>24 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins</td>
<td>14 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation, Most Conserved</td>
<td>10 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial sacCer1 Release (based on sequence in SGD as of 1 Oct 2003): WashU Clones, GC Percent, SGD Genes, SGD Other, Yeast ESTs, Spliced ESTs, Yeast ESTs, Multiz Yeast, Simple Repeats. Downloads: bigZips, database.</td>
<td>14 Jan 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=sc1></A>SARS Apr. 2003 (sc1, GenBank sequence as of 14 Apr 2003)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
bigZips downloads</td>
<td>05 Jun 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial sc1 Release (based on sequence deposited into GenBank as of 14 Apr 2003): GC Percent, GenBank Protein Annotations, Fgenesv+ Genes, Viral mRNAs, Viral mRNAs Chained, Other SARS, Coronavirus Proteins</td>
<td>23 May 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=zoo1></A>NISC (Zoo) Jun. 2002 (zoo1, Comparative Sequencing Program Target 1)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Downloads: Annotation database (by organism)</td>
<td>23 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial NISC Target 1 Release: Assembly, Gap, GC Samples, Gene Annotations, Genscan Genes, mRNAs, NonOrganism mRNAs, NonOrganism ESTs, Tetraodon Blat, ZooSeq, RepeatMasker, Simple Repeats. Human only: RefSeq Genes, ESTs, Spliced ESTs, DNase HyperSens, MCS (Binomial), Binomial Cons Score, MCS (ParsPVal), Parsimony P-Values, MCS InterMerged, MCS Inter</td>
<td>15 Aug 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
<!-- END SECOND SECTION ---------------------------------------------------------->

</TD></TR></TABLE>
</BODY></HTML>

