<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
    <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
    <META name="description" content="UCSC Genome Browser Release Log">
    <META name="keywords" content="genome, genome browser, genome assembly, UCSC, bioinformatics, release, table, track, annotation">
    <LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
    <TITLE>UCSC Genome Bioinformatics: Release Log</TITLE>
</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="/images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>
</TABLE>

<!--Hotlinks bar-------------------------------------------------->
<TABLE BORDER=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><TH></TR>
<TR><TD COLSPAN=2 HEIGHT=40>
    <TABLE bgcolor="000000" cellpadding="1" cellspacing="1" width="100%" height="27"><TR bgcolor="2636D1"><TD valign="middle">
        <TABLE BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24">
            <TR><TD VALIGN="middle"><font color="#89A1DE">&nbsp;
                &nbsp;<A HREF="index.html" class="topbar">
                        Home</a> &nbsp; - &nbsp;
                <A HREF="/cgi-bin/hgGateway" class="topbar">
                        Genome Browser</a> &nbsp; - &nbsp;
                <A HREF="/cgi-bin/hgBlat?command=start" class="topbar">
                        Blat Search</a> &nbsp; - &nbsp;
                <A HREF="/cgi-bin/hgText?command=start" class="topbar">
                        Table Browser</a> &nbsp; - &nbsp;
                <A HREF="../FAQ.html" class="topbar">
                        FAQ</a> &nbsp; - &nbsp;
                <A HREF="help/hgTracksHelp.html" class="topbar">
                        User Guide</a> &nbsp;</font></TD>
        </TR></TABLE>
    </TD></TR></TABLE>
</TD></TR></TABLE>

<!--Menu------------------------------------------------------->
<TABLE BORDER=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><TH></TR>
<TR><TD CELLPADDING=10>	
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
                <TR><TD> <FONT SIZE="4"><b>&nbsp;   Track and Table Releases</b></FONT>
	        </TD></TR></TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	        <TR><TD WIDTH=10>&nbsp;</TD><TD>

 	        This page contains track and table release information for 
		the following genome assemblies: 
	        <UL>
	        <LI><A CLASS="toc" HREF="#hg16">Human Jul. 2003 (hg16)</A></LI>
	        <LI><A CLASS="toc" HREF="#hg15">Human Apr. 2003 (hg15)</A></LI>
	        <LI><A CLASS="toc" HREF="#hg13">Human Nov. 2002 (hg13)</A></LI>
	        <LI><A CLASS="toc" HREF="#mm3">Mouse Feb. 2003 (mm3)</A></LI>
	        <LI><A CLASS="toc" HREF="#rn3">Rat Jun. 2003 (rn3)</A></LI>
	        <LI><A CLASS="toc" HREF="#rn2">Rat Jan. 2003 (rn2)</A></LI>
	        <LI><A CLASS="toc" HREF="#ce1">C. elegans May 2003 (ce1)</A></LI>
	        <LI><A CLASS="toc" HREF="#cb1">C. briggsae Jul. 2002 (cb1)</A></LI>
	        <LI><A CLASS="toc" HREF="#sc1">SARS Apr. 2003 (sc1)</A></LI>
	        </UL>
		<P>
	        For more information about the tracks and tables listed on this 
		page, refer to the <A HREF="gbdDescriptions.html">Description 
	        of the annotation database</A> and the <A 
	        HREF="help/hgTracksHelp.html#IndivTracks">User's Guide</A>.
            </TD><TD WIDTH=15></TD></TR></TABLE>
        <br></TD></TR></TABLE>
    </TD></TR></TABLE>
</TD></TR></TABLE>

<A NAME="hg16"></A>
<!--hg16 Table------------------------------------------------------->
<TABLE BORDER=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><TH></TR>
<TR><TD CELLPADDING=10>
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>
            <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
                <TR><TD> <FONT SIZE="4"><b>&nbsp;   Human Jul. 2003 (hg16, NCBI Build 34)</B></FONT>
                </TD></TR></TABLE>
    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
        <TR><TD WIDTH=10></TD><TD>
	<!--track table-->
	    <TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
	    <TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
	        <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
            </TD></TR>
            <TR valign="top"><TD align="left">
                STS Markers (updated)</TD>
                <TD>11 Aug 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                Initial hg16 release (using NCBI Build 34): Chromosome Band,
		FISH Clones, Assembly, Gap, 
		STS Markers, Map Contigs, Recomb Rate,
                GC Percent, Coverage, BAC End Pairs, Fosmid End Pairs,
		RefSeq Genes, MGC Genes, Genscan Genes, Human mRNAs, Spliced ESTs,
                Human ESTs, NonHuman mRNAs, NonHuman ESTs, CpG Islands,
                RepeatMasker, Simple Repeats. Downloads: bigZips, chromosomes, 
		database</TD>
                <TD>8 Aug 2003</TD>
            </TR>
	    </TABLE>
    </TD><TD WIDTH=15></TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
</TD></TR></TABLE>

<A NAME="hg15"></A>
<!--hg15 Table------------------------------------------------------->
<TABLE BORDER=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><TH></TR>
<TR><TD CELLPADDING=10>
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>
            <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
                <TR><TD> <FONT SIZE="4"><b>&nbsp;   Human Apr. 2003 (hg15, NCBI Build 33)</B></FONT>
                </TD></TR></TABLE>
    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
        <TR><TD WIDTH=10></TD><TD>
	<!--track table-->
	    <TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
	    <TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
	        <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
            </TD></TR>

            <TR valign="top"><TD align="left">
                NonHuman ESTs</TD>
                <TD>25 Jul 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                Ensembl Genes</TD>
                <TD>15 Jul 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                bigZips (upstream files updated)</TD>
                <TD>9 Jul 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                Best Mouse (using mm3)</TD>
                <TD>9 Jul 2003</TD>
            </TR>
	    <TR valign="top"><TD align="left">
                Mouse Cons (using mm3)</TD>
                <TD>9 Jul 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                Blastz axtTight mouse alignments (using mm3)</TD>
                <TD>25 Jun 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                Acembly</TD>
                <TD>5 Jun 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                AxtInfo (updated to use axtNetMm3)</TD>
                <TD>30 May 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                MGC Genes</TD>
                <TD>27 May 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                Human/Mouse/Rat (using mm3, rn2)</TD>
                <TD>27 May 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                Rat Synteny (using rn2)</TD>
                <TD>27 May 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                Gene Bounds</TD>
                <TD>27 May 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                Tight Rat (using rn2)</TD>
                <TD>27 May 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                UniGene (v. Hs 160)</TD>
                <TD>27 May 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                Haplotype Blocks (chr 21 and 22 only)</TD>
                <TD>20 May 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                SGP Genes (using mm3)</TD>
                <TD>20 May 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                Fosmid End Pairs</TD>
                <TD>20 May 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                Overlap and Random SNPs (dbSnp 114)</TD>
                <TD>15 May 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                Blastz axtBest mouse alignments (using mm3)</TD>
                <TD>13 May 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                Human/Mouse Chain and Net (using mm3)</TD>
                <TD>9 May 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                Twinscan (using mm2)</TD>
                <TD>8 May 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                Geneid Genes</TD>
                <TD>8 May 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                Recomb Rate</TD>
                <TD>1 May 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                GNF Ratio</TD>
                <TD>30 Apr 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                NCI60 </TD>
                <TD>30 Apr 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                geneBands (table only)</TD>
                <TD>30 Apr 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                GC Pecent (data reloaded) </TD>
                <TD>29 Apr 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                FISH Clones</TD>
                <TD>28 Apr 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                Fgenesh++ Genes </TD>
                <TD>28 Apr 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                Chromosome Band</TD>
                <TD>28 Apr 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                Coverage</TD>
                <TD>25 Apr 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                BAC End Pairs</TD>
                <TD>25 Apr 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                liftAll.zip (correction for NT_008183 - table only)</TD>
                <TD>16 Apr 2003</TD>
            </TR>
            <TR valign="top"><TD align="left">
                Initial hg15 release (using NCBI Build 33): Assembly, Gap, STS Markers, Map Contigs,
                GC Percent, Known Genes, RefSeq Genes,
                Genscan Genes, Human mRNAs, Spliced ESTs,
                Human ESTs, NonHuman mRNAs, CpG Islands,
                Fugu Blat (v 3.0), Tight Mouse (mm3), RepeatMasker,
                Simple Repeats, Mouse Synteny (mm3)</TD>
                <TD>14 Apr 2003</TD>
            </TR>
	    </TABLE>
    </TD><TD WIDTH=15></TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
</TD></TR></TABLE>

<A NAME="hg13"></A>
<!--hg13 Table------------------------------------------------------->
<TABLE BORDER=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><TH></TR>
<TR><TD CELLPADDING=10>
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>
    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
        <TR><TD>
	<FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; Human Nov. 2002 (hg13, NCBI Build 31)</B></FONT>
    </TD></TR></TABLE>
    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
        <TR><TD WIDTH=10></TD><TD>
	<!--track table-->
	    <TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
	    <TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
	        <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
            </TD></TR>
	    <TR valign="top"><TD align="left">
                Segmental Dups</TD>
		<TD>24 Jun 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Mouse Synteny (using mm3)</TD>
		<TD>6 Jun 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Fgenesh++ Genes (new data)</TD>
		<TD>30 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Transcriptome (chr21 and 22)</TD>
		<TD>28 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                RNA Genes</TD>
		<TD>28 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Haplotype Blocks (chr21 and 22)</TD>
		<TD>20 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Random SNPs, Overlap SNPs (updated to dbSnp 114)</TD>
		<TD>15 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Human/Mouse Chain and Net (using mm3)</TD>
		<TD>2 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                SLAM Rat (using rn1)</TD>
		<TD>28 Apr 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                SLAM Mouse (using mm2)</TD>
		<TD>28 Apr 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Rat Synteny (using rn1)</TD>
		<TD>7 Apr 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Superfamily</TD>
		<TD>30 Mar 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                TIGR Gene Index</TD>
		<TD>28 Mar 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Random SNPs, Overlap SNPs</TD>
		<TD>27 Mar 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Ensembl Genes </TD>
		<TD>14 Mar 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                RepeatMasker (updated)</TD>
		<TD>13 Mar 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Acembly Genes (IDs changed)</TD>
		<TD>27 Feb 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                BAC End Pairs</TD>
		<TD>27 Feb 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Mouse Cons (using mm2) </TD>
		<TD>21 Feb 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                UniGene (v. Hs 159)</TD>
		<TD>20 Feb 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Fosmid End Pairs</TD>
		<TD>10 Feb 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Coverage</TD>
		<TD>10 Feb 2003</TD> 
	    </TR>
	    <TR valign="top"><TD align="left">
                Blastz human alignments downloads (using hg13)</TD>
		<TD>7 Feb 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Geneid Genes</TD>
		<TD>6 Feb 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                SGP Genes (using mm2)</TD>
		<TD>6 Feb 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                GenMapDB Clones</TD>
		<TD>22 Jan 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Twinscan</TD>
		<TD>22 Jan 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Recomb Rate</TD>
		<TD>20 Jan 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Initial hg13 Release (using NCBI Build 31): Assembly, Gap, STS Markers, Map Contigs,
                GC Percent, FISH Clones, Chromosome Band, Known Genes, RefSeq Genes, 
                Genscan Genes, Human mRNAs, Spliced ESTs, 
                Human ESTs, NonHuman mRNAs, Gene Bounds, CpG Islands,
                Fugu Blat (v. 3.0), Tight Mouse (mm3), Best Mouse (mm3), 
		RepeatMasker, Simple Repeats</TD>
                <TD>23 Dec 2002</TD>
	    </TR>
	    </TABLE>
    </TD><TD WIDTH=15></TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
</TD></TR></TABLE>

<A NAME="mm3"></A>
<!--mm3 Table------------------------------------------------------->
<TABLE BORDER=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><TH></TR>
<TR><TD CELLPADDING=10>
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>
    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
        <TR><TD>
	<FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; Mouse Feb. 2003 (mm3, NCBI Build 30)</B></FONT>
    </TD></TR></TABLE>
    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
        <TR><TD WIDTH=10></TD><TD>
	<!--track table-->
	    <TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
	    <TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
	        <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
            </TD></TR>
	    <TR valign="top"><TD align="left">
                Superfamily</TD>
		<TD>25 Jul 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Bac End Pairs (added data for chrX)</TD>
		<TD>25 Jun 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Human Cons (using hg15)</TD>
		<TD>24 Jun 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Human Chain/Net (updated from hg13 to hg15)</TD>
		<TD>17 Jun 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Ensembl Genes</TD>
		<TD>13 Jun 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Ensembl EST Genes </TD>
		<TD>13 Jun 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Mouse ESTs, Spliced ESTs, Gene Bounds (additions for 
		random chromosomes)</TD>
		<TD>30 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Geneid Genes</TD>
		<TD>28 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Rat Synteny (using rn1)</TD>
		<TD>27 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                CpG Islands (random data added)</TD>
		<TD>27 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Human Synteny (using hg15)</TD>
		<TD>27 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                MGC Genes </TD>
		<TD>27 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Twinscan (using hg13)</TD>
		<TD>22 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Known Genes</TD>
		<TD>16 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Blastz human alignments downloads (using hg15)</TD>
		<TD>14 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Blastz rat alignments downloads (using rn2)</TD>
		<TD>13 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                SGP Genes (using hg13)</TD>
		<TD>8 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Human/Mouse Chain and Net (using hg15)</TD>
		<TD>2 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Fugu Blat</TD>
		<TD>29 Apr 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                BAC Ends Pairs</TD>
		<TD>29 Apr 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                STS Markers</TD>
		<TD>29 Apr 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Fgenesh++ Genes</TD>
		<TD>17 Apr 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Simple Repeats (data added for randoms)</TD>
		<TD>17 Apr 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Best Rat, Tight Rat (using rn2)</TD>
		<TD>2 Apr 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Best Human, Tight Human (using hg15)</TD>
		<TD>2 Apr 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                NonMouse mRNAs</TD>
		<TD>1 Apr 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                AxtInfo (tables)</TD>
		<TD>31 Mar 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Initial mm3 Release (using NCBI Build 30): Assembly, Gap, Map Contigs,
                GC Percent, RefSeq Genes, 
                Genscan Genes, Gene Bounds, Mouse mRNAs, Spliced ESTs, 
                Mouse ESTs, CpG Islands,
                RepeatMasker, Simple Repeats, Overlap SNPs (dbSnp 111)</TD>
                <TD>7 Mar 2003</TD>
	    </TR>
	    </TABLE>
    </TD><TD WIDTH=15></TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
</TD></TR></TABLE>

<A NAME="rn3"></A>
<!--rn3 Table------------------------------------------------------->
<TABLE BORDER=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><TH></TR>
<TR><TD CELLPADDING=10>
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>
    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
        <TR><TD>
	<FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; Rat Jun. 2003 (rn3, Baylor HGSC v. 3.1)</B></FONT>
    </TD></TR></TABLE>
    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
        <TR><TD WIDTH=10></TD><TD>
	<!--track table-->
	    <TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
	    <TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
	        <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
            </TD></TR>
	    <TR valign="top"><TD align="left">
                SGP Genes </TD>
		<TD>5 Aug 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Known Genes </TD>
		<TD>30 Jul 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Fgenesh++ Genes</TD>
		<TD>18 Jul 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Twinscan (using hg15)</TD>
		<TD>14 Jul 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Initial rn3 Release (using Baylor HGSC v. 3.1): Assembly, Gap, Bactigs,
                GC Percent, Bac End Pairs, RefSeq Genes, STS Markers,
                Geneid Genes, 
		Genscan Genes, Rat mRNAs, Spliced ESTs, Rat ESTs, NonRat mRNAs, 
		NonRat ESTs, CpG Islands, RepeatMasker, Simple Repeats, 
		bigZips downloads, chromosome downloads</TD>
                <TD>14 Jul 2003</TD>
	    </TR>
	    </TABLE>
    </TD><TD WIDTH=15></TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
</TD></TR></TABLE>

<A NAME="rn2"></A>
<!--rn2 Table------------------------------------------------------->
<TABLE BORDER=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><TH></TR>
<TR><TD CELLPADDING=10>
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>
    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
        <TR><TD>
	<FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; Rat Jan. 2003 (rn2, Baylor HGSC v. 2.1)</B></FONT>
    </TD></TR></TABLE>
    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
        <TR><TD WIDTH=10></TD><TD>
	<!--track table-->
	    <TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
	    <TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
	        <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
            </TD></TR>
	    <TR valign="top"><TD align="left">
                Known Genes</TD>
		<TD>30 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Fugu Blat (v. 3.0)</TD>
		<TD>29 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Geneid Genes</TD>
		<TD>28 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                SGP Genes </TD>
		<TD>20 May 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Fgenesh++ Genes </TD>
		<TD>17 Apr 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Gene Bounds</TD>
		<TD>13 Apr 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                BAC End Pairs</TD>
		<TD>13 Apr 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Twinscan (using hg13)</TD>
		<TD>2 Apr 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Initial rn2 Release (using Baylor HGSC v. 2.1): Assembly, Gap, Bactigs,
                GC Percent, RefSeq Genes, STS Markers,
                Genscan Genes, Rat mRNAs, Spliced ESTs, 
                Rat ESTs, NonRat mRNAs, CpG Islands,
                RepeatMasker, Simple Repeats, bigZips downloads, chromosome
		downloads</TD>
                <TD>14 Mar 2003</TD>
	    </TR>
	    </TABLE>
    </TD><TD WIDTH=15></TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
</TD></TR></TABLE>

<A NAME="ce1"></A>
<!--ce1 Table------------------------------------------------------->
<TABLE BORDER=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><TH></TR>
<TR><TD CELLPADDING=10>
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>
    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
        <TR><TD>
	<FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; C. elegans May 2003 (ce1, WormBase v. WS100)</B></FONT>
    </TD></TR></TABLE>
    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
        <TR><TD WIDTH=10></TD><TD>
	<!--track table-->
	    <TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
	    <TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
	        <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
            </TD></TR>
	    <TR valign="top"><TD align="left">
                Initial ce1 Release (using WormBase v. WS100): Assembly, Gap, 
                GC Percent, RefSeq Genes, Sanger Genes,
                Worm mRNAs, Worm ESTs, 
                Fugu Blat, Briggsae Waba, Briggsae Chain, Briggsae Blastz,
                RepeatMasker, Simple Repeats, Self Chain, chromosome
		downloads </TD>
                <TD>24 Jun 2003</TD>
	    </TR>
	    </TABLE>
    </TD><TD WIDTH=15></TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
</TD></TR></TABLE>

<A NAME="cb1"></A>
<!--cb1 Table------------------------------------------------------->
<TABLE BORDER=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><TH></TR>
<TR><TD CELLPADDING=10>
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>
    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
        <TR><TD>
	<FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; C. briggsae Jul. 2002 (cb1, WormBase v. cb25.agp8)</B></FONT>
    </TD></TR></TABLE>
    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
        <TR><TD WIDTH=10></TD><TD>
	<!--track table-->
	    <TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
	    <TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
	        <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
            </TD></TR>
	    <TR valign="top"><TD align="left">
                Initial ce1 Release (using WormBase v. cb25.agp8): Assembly, Gap, 
                GC Percent, Worm mRNAs, Worm ESTs, 
                Elegans Chain, Elegans Blastz, 
                RepeatMasker, Simple Repeats, data set downloads </TD>
                <TD>24 Jun 2003</TD>
	    </TR>
	    </TABLE>
    </TD><TD WIDTH=15></TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
</TD></TR></TABLE>

<A NAME="sc1"></A>
<!--SARS Table------------------------------------------------------->
<TABLE BORDER=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><TH></TR>
<TR><TD CELLPADDING=10>
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>
    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
        <TR><TD>
	<FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; SARS Apr. 2003 (sc1, GenBank sequence as of 14 Apr 2003)</B></FONT>
    </TD></TR></TABLE>
    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
        <TR><TD WIDTH=10></TD><TD>
	<!--track table-->
	    <TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
	    <TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
	        <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
            </TD></TR>
	    <TR valign="top"><TD align="left">
                bigZips downloads </TD>
                <TD>5 Jun 2003</TD>
	    </TR>
	    <TR valign="top"><TD align="left">
                Initial sc1 Release (based on sequence deposited into GenBank as of 14 Apr 2003): GC Percent, GenBank Protein Annotations,
                Fgenesv+ Genes, Viral mRNAs, Viral mRNAs Chained, 
                Other SARS, Coronavirus Proteins</TD>
                <TD>23 May 2003</TD>
	    </TR>
	    </TABLE>
    </TD><TD WIDTH=15></TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
</TD></TR></TABLE>

</BODY></HTML>
