<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
</head>
<body>
The <I>Haloarcula marismortui ATCC 43049</I> genome is 4.27 Million bp long and contains approximately 4301
predicted genes. H. marismortui is a halophilic archaeon found in the dead sea, where it optimally grows at around 4.5 molar salt.<BR><BR>
The sequence was released Nov. 2004 by Institute for Systems Biology, and was described
in <I>Genome Res</I> <B>14</B>:2221-34 (2004) Baliga NS, Bonneau R, Facciotti MT, Pan M, Glusman G, <I>et al.</I> &nbsp;"<a
 href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=15520287">Genome sequence of Haloarcula marismortui: a halophilic archaeon from the Dead Sea.
</a>"
<blockquote><B>Abstract:</B> We report the complete sequence of the 4,274,642-bp genome of Haloarcula marismortui, a halophilic archaeal isolate from the Dead Sea. The genome is organized into nine circular replicons of varying G+C compositions ranging from 54% to 62%. Comparison of the genome architectures of Halobacterium sp. NRC-1 and H. marismortui suggests a common ancestor for the two organisms and a genome of significantly reduced size in the former. Both of these halophilic archaea use the same strategy of high surface negative charge of folded proteins as means to circumvent the salting-out phenomenon in a hypersaline cytoplasm. A multitiered annotation approach, including primary sequence similarities, protein family signatures, structure prediction, and a protein function association network, has assigned p... [Click above reference link for full abstract]
</blockquote>
<P>
<h3>Sample position queries</h3>
<p>
A genome position can be specified by chromosomal coordinate range, COG
ID, or keywords from the GenBank or TIGR description of a gene. 
The available chromosome/plasmid names are:<BR><BR> <table border="2" cellspacing=2 cellpadding=2>
<tr><th>Browser Chrom/Plasmid Name</th><th>Length (bp)</th><th>Gene Count</ht><th>NCBI RefSeq Accession</th></tr>
<tr><td>plasmid_pNG100</td><td align=right>33303</td><td align=right>36</td><td align=center>NC_006389</td></tr>
<tr><td>plasmid_pNG200</td><td align=right>33452</td><td align=right>42</td><td align=center>NC_006390</td></tr>
<tr><td>plasmid_pNG300</td><td align=right>39521</td><td align=right>40</td><td align=center>NC_006391</td></tr>
<tr><td>plasmid_pNG400</td><td align=right>50060</td><td align=right>51</td><td align=center>NC_006392</td></tr>
<tr><td>plasmid_pNG500</td><td align=right>132678</td><td align=right>131</td><td align=center>NC_006393</td></tr>
<tr><td>plasmid_pNG600</td><td align=right>155300</td><td align=right>167</td><td align=center>NC_006394</td></tr>
<tr><td>plasmid_pNG700</td><td align=right>410554</td><td align=right>363</td><td align=center>NC_006395</td></tr>
<tr><td>chr_I</td><td align=right>3131724</td><td align=right>3186</td><td align=center>NC_006396</td></tr>
<tr><td>chr_II</td><td align=right>288050</td><td align=right>285</td><td align=center>NC_006397</td></tr>
</table>

</p>
<p>
<p>
The following list shows examples of valid position queries for this
genome:&nbsp;
</p>
<p>
<table cellspacing="0" cellpadding="0" border="0">
  <tbody>
    <tr>
      <td nowrap="nowrap" valign="top"><b>Request:</b>
      </td>
      <td colspan="2" valign="top"><b>Genome Browser
Response:</b><br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">chr_I</td>
      <td><br>
      </td>
      <td valign="top">Displays the entire 'chr_I' sequence<br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">chr_I:1-10000</td>
      <td><br>
      </td>
      <td valign="top">Displays first ten thousand bases of the 'chr_I' sequence<br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">oxidoreductase&nbsp;&nbsp;&nbsp;&nbsp;</td>
      <td><br>
      </td>
      <td valign="top">Lists all genes with "oxidoreductase" in the name or description<br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">pNG1011</td>
      <td><br>
      </td>
      <td valign="top">Display genome at position of gene pNG1011<br>
      </td>
    </tr>
  </tbody>
</table>
</p>
<P>
<HR>
<H3>Credits</H3>
The Archaeal Genome Browsers at UCSC were developed by members of the
<a href="http://lowelab.ucsc.edu/">Lowe Lab</a> (Kevin Schneider,
Katherine Pollard, Andy Pohl, Todd Lowe) and Robert Baertch, with significant support from
the <a href="http://genome.ucsc.edu/staff.html">UCSC Human Genome
Browser group</a>.
The Archaeal Browsers are run by a slightly modified
version of the <a href="http://genome.ucsc.edu/">UCSC Human Genome
Browser</a> system.  All queries, bug reports, content corrections, suggested improvements,
and new track data submissions should be sent to Todd Lowe (lowe
@soe.ucsc.edu).

<P>If you use the browser for your research and wish to cite it, a
paper describing the work has been accepted to the 2006 <I>Nucleic
Acids Research</I>
<a href="http://www3.oup.co.uk/nar/database/a/">Database Issue</a> (KL Schneider, KS Pollard, R Baertsch, A Pohl, TM Lowe). 
Citations will help us obtain funding to continue
development of this community resource.

</body>
</html>
