<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
</head>
<body>
The <span style="font-style: italic;">Haloarcula marismortui</span>
has 2 chromosomes and 7 plasmids.  The total sequences are 4,274,642 bp long and contain approximately 4,242
predicted proteins.&nbsp; <span style="font-style: italic;">H.
marisnortui</span> is a halophilic archaeon found in the dead sea, where it optimally grows at around 
4.5 molar salt.&nbsp; The sequences
were completed in Nov 2004 by the Institute for Systems Biology in Seattle, and was published
in <a
 href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=15520287">Genome sequence of 
Haloarcula marismortui: a halophilic archaeon from the Dead Sea.</a>.&nbsp; Baliga NS, Bonneau R, Facciotti MT, Pan M, Glusman G, Deutsch EW, 
Shannon P, Chiu Y, Weng RS, Gan RR, Hung P, Date SV, Marcotte E, Hood L, Ng WV.&nbsp; Genome Res. 2004;14(11):2221-34.
<h3>Sample position queries</h3>
<p>
A genome position can be specified by chromosomal coordinate range, COG
ID, or keywords from the GenBank or TIGR description of an gene. 
This archae has multiply chromosomes possible.  
</p>
<p>
The available chromosome/plasmid names are
chr1, chr2, plasmid1, plasmid2, plasmid3, plasmid4, plasmid5, plasmid6, plasmid7.
</p>
<p>
The following list shows examples of valid position queries for this
genome.&nbsp;
</p>
<p>
<table cellspacing="0" cellpadding="0" border="0">
  <tbody>
    <tr>
      <td nowrap="nowrap" valign="top"><b>Request:</b><br>
      </td>
      <td colspan="2" valign="top"><b>&nbsp;&nbsp; Genome Browser
Response:</b><br>
      </td>
    </tr>
    <tr>
      <td valign="top"><br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">chr2</td>
      <td width="14"><br>
      </td>
      <td valign="top">Displays all of chromosome 2.<br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">plasmid1</td>
      <td width="14"><br>
      </td>
      <td valign="top">Displays all of plasmid 1.<br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">chr1:1-10000</td>
      <td width="14"><br>
      </td>
      <td valign="top">Displays first ten thousand bases on the
chromosome.<br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">NADH</td>
      <td width="14"><br>
      </td>
      <td valign="top">Lists genes with "NADH" in the description.<br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">rrnAC0862</td>
      <td width="14"><br>
      </td>
      <td valign="top">Display chromosome at position of gene <a
 href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?dopt=GenPept&cmd=Retrieve&db=protein&list_uids=55377699">rrnAC0862</a>.<br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">rrnAC086</td>
      <td width="14"><br>
      </td>
      <td valign="top">Lists genes with names beginning with "rrnAC086".<br>
      </td>
    </tr>
    <tr>
    </tr>
  </tbody>
</table>
</p>
<P>
<HR>
<H3>Credits</H3>
The Archaeal Genome Browsers at UCSC were developed by members of the
<a href="http://lowelab.ucsc.edu/">Lowe Lab</a> (Kevin Schneider,
Katherine Pollard, Andy Pohl, Todd Lowe) with significant support from
the <a href="http://genome.ucsc.edu/staff.html">UCSC Human Genome
Browser group</a> (especially Robert Baertsch & Hiram Clawson).
The Archaeal Browsers are run by a slightly modified
version of the <a href="http://genome.ucsc.edu/">UCSC Human Genome
Browser</a> system.  All queries, bug reports, content corrections, suggested improvements,
and new track data submissions should be sent to Todd Lowe (lowe
@soe.ucsc.edu).

<P>If you use the browser for your research and wish to cite it, a
paper describing the work has been submitted to the 2006 <I>Nucleic
Acids Research</I>
Database Issue (KL Schneider, KS Pollard, A Pohl, TM Lowe,
submitted).  Citations will help us obtain funding to continue
development of this community resource.

</body>
</html>
