# hapmapAlleles.sql was originally generated by the autoSql program, which also 
# generated hapmapAlleles.c and hapmapAlleles.h.  This creates the database representation of
# an object which can be loaded and saved from RAM in a fairly 
# automatic way.

#HapMap allele counts
CREATE TABLE hapmapAlleles (
    bin smallint(5) unsigned not null,			# For browser speed up
    chrom varchar(15) not null,				# Chromosome
    chromStart int(10) unsigned not null,		# Start position in chrom (0 based)
    chromEnd int(10) unsigned not null,			# End position in chrom (1 based)
    name varchar(15) not null,				# Reference SNP identifier from dbSnp
    score smallint(5) unsigned not null,		# Minor allele frequency (0-500) 
    strand enum ('+', '-', '?') default '?' not null,	# Which genomic strand contains the observed alleles
    observed varchar(255) not null,			# Observed string from genotype file
    allele1 enum('A', 'C', 'G', 'T')  not null,		# This allele has been observed
    homoCount1 smallint(5) unsigned not null,		# Count of individuals who are homozygous for allele1
    allele2 enum("C", "G", "T", "none") not null,	# This allele may not have been observed
    homoCount2 smallint(5) unsigned not null,		# Count of individuals who are homozygous for allele2
    heteroCount smallint(5) unsigned not null		# Count of individuals who are heterozygous
);
