# hapmapSnps.sql was originally generated by the autoSql program, which also 
# generated hapmapSnps.c and hapmapSnps.h.  This creates the database representation of
# an object which can be loaded and saved from RAM in a fairly 
# automatic way.

#HapMap primate alleles and population specific derived allele frequencies
CREATE TABLE hapmapSnps (
    bin        smallint(5) unsigned NOT NULL,
    chrom      enum( 'unknown', 'chr1', 'chr2', 'chr3', 'chr4', 'chr5', 
		'chr6', 'chr7', 'chr8', 'chr9', 'chr10', 'chr11', 'chr12', 
		'chr13', 'chr14', 'chr15', 'chr16', 'chr17', 'chr18', 
		'chr19', 'chr20', 'chr21', 'chr22', 'chrM', 'chrX', 'chrY'
               ) DEFAULT 'unknown' NOT NULL, # Chromosome
    chromStart int unsigned not null,	# Start position in chrom
    chromEnd   int unsigned not null,	# End position in chrom
    name       varchar(255) not null,	# Reference SNP identifier or Affy SNP name
    score      int unsigned not null,	# Not used
    strand     enum('?','+','-') DEFAULT '?' NOT NULL,	# Which DNA strand contains the observed alleles
    hReference char(1)      not null,	# Human reference base
    hOther     char(1)      not null,	# Human other base
    cBase      char(1)      not null,	# chimp base
    rBase      char(1)      not null,	# rhesus base
    cQual      int unsigned not null,	# chimp quality score
    rQual      int unsigned not null,	# rhesus quality score
    ceu        float        not null,	# Derived allele frequency for the CEU population
    chb        float        not null,	# Derived allele frequency for the CHB population
    jpt        float        not null,	# Derived allele frequency for the JPT population
    jptchb     float        not null,	# Derived allele frequency for the JPT+CHB population
    yri        float        not null,	# Derived allele frequency for the YRI population
              #Indices
    INDEX      chrom      (chrom,bin),
    INDEX      name       (name)
);
