<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: bigBed Track Format</TITLE>
	<LINK REL="STYLESHEET" HREF="../../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../../images/title.jpg"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../../inc/topbar.html"--></TD></TR>
<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=2 CELLPADDING=10>	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><b>&nbsp;  bigBed Track Format</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
		    <P>
The bigBed format stores annotation items that can either be simple, or a
linked collection of exons, much as 
<A HREF="customTrack.html#BED">bed</A> files do.  
BigBed files are created initially from bed type files, 
using the program <TT>bedToBigBed</TT>.   The
resulting bigBed files are in an indexed binary format.  The main advantage of
the bigBed files is that only the portions of the files needed to display a
particular region are transferred to UCSC, so for large data sets bigBed is
considerably faster than regular bed files.  The bigBed file remains on 
your web accessible server (http, https, or ftp), not on the UCSC server. 
Only the portion that is needed
for the chromosomal position you are currently viewing is locally cached as a 
&quot;sparse file&quot;.</P>

<P>
See 
<A HREF="http://genomewiki.ucsc.edu/index.php/Selecting_a_graphing_track_data_format"
TARGET=_blank>this page</A> for help in selecting a graphing track data format
that is most approriate for the type of data you have.
</P>

<P>Please note that the <TT>bedToBigBed</TT> utility uses a lot of memory;
somewhere on the order of 5 times more memory than the uncompressed
bed input file.  We recommend that you monitor your memory usage by 
using the <TT>top</TT> command while you are running the program.</P>

<P>
To create a bigBed track, follow these steps:<BR>

<OL>
 <LI> Create a bed format file following the directions 
  <A HREF="customTrack.html#BED">here</A>.  
  Note that when converting a bed file to a bigBed file, you are limited to 
  one track of data in your input file; you must create a separate bed file
  for each data track. Note that this is the file that is referred to as
  <TT>input.bed</TT> in <EM>step 5</EM> below.
 <LI> Remove any existing 'track' or 'browser' lines from your bed file
  so that it contains only data.
 <LI> Download the <TT>bedToBigBed</TT> program from the 
  <A HREF="http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/">directory</A> 
  of binary utilities.
 <LI> Use the <TT>fetchChromSizes</TT> script from the same 
  <A HREF="http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/">directory</A> 
  to create the chrom.sizes file for the UCSC database you are working with 
  (e.g. hg19).  Note that this is the file that is referred to as
  <TT>chrom.sizes</TT> in <EM>step 5</EM> below.
 <LI> Create the bigBed file from your bed file using the <TT>bedToBigBed</TT>
  utility like so: 
  <TT><B>bedToBigBed</B> input.bed chrom.sizes myBigBed.bb</TT><BR>
  Note that you can also use a gzipped input file.
 <LI> Move the newly created bigBed file (<TT>myBigBed.bb</TT>) to a http, 
  https, or ftp location.
 <LI> Construct a <A HREF="hgTracksHelp.html#CustomTracks">custom track</A> 
  using a single track line. Note that any of the track attributes listed
  <A HREF="customTrack.html#TRACK">here</A> are applicable 
  to tracks of type bigBed. 
  The most basic version of the &quot;track&quot; line will look something 
  like this:<BR>
  <TT>track type=bigBed name="My Big Bed" description="A Graph of 
  Data from My Lab" dataUrl=http://myorg.edu/mylab/myBigBed.bb</TT>
 <LI> Paste this custom track line into the text box in the 
  <A HREF="../../cgi-bin/hgCustom">custom track</A> management page.
</OL>

<H3>Example One</H3>
<P>In this example, you will use an existing bigBed file to create a bigBed 
custom track.  A bigBed file that contains data on chromosome 21 on the hg19 
assembly has been placed on our http server.  
You can create a custom track using this bigBed file by constructing a 
&quot;track&quot; line that references this file like so:</P>

<TT>track type=bigBed name=&quot;bigBed Example One&quot; 
description=&quot;A bigBed file&quot; 
dataUrl=http://genome.ucsc.edu/goldenPath/help/bigBedExample.bb</TT>

<P>Include the following &quot;browser&quot; line to ensure that the custom
track opens at the correct position:</P>

<TT>browser position chr21:33,031,597-33,041,570</TT>

<P>Paste the &quot;browser&quot; line and &quot;track&quot; line into the 
<A HREF="../../cgi-bin/hgCustom">custom track</A> management page for the
human assembly hg19 (Feb. 2009), then
press the <EM>submit</EM> button.
On the following page, press the <EM>chr21</EM> link in the custom track
listing to view the bigBed track in the Genome Browser.</P>

<H3>Example Two</H3>
<P>In this example, you will create your own bigBed file from an existing 
bed file.

<UL>
 <LI> Save this <A HREF="bedExample.txt.gz">bed file</A> 
  to your machine 
  (this satisfies <EM>steps 1</EM> and <EM>2</EM> above).
 <LI> Save this <A HREF="hg19.chrom.sizes">text file</A> to your machine.
  It contains the chrom.sizes for the human (hg19) assembly 
  (this satisfies <EM>step 4</EM> above).
 <LI> Download the <TT>bedToBigBed</TT> utility (see <EM>step 3</EM>).
 <LI> Run the utility to create the bigBed output file 
  (see <EM>step 5</EM>):<BR>
  <TT><B>bedToBigBed</B> bedExample.txt.gz hg19.chrom.sizes myBigBed.bb</TT>
 <LI> Place the bigBed file you just created (<TT>myBigBed.bb</TT>) on a 
  web-accessible server (see <EM>step 6</EM>).
 <LI> Construct a &quot;track&quot; line that points to your bigBed file
  (see <EM>step 7</EM>).
 <LI> Create the custom track on the human assembly hg19 (Feb. 2009), and 
  view it in the genome browser (see <EM>step 8</EM>). Note that the original
  bed file contains data on only chromsome 21.
</UL>

<H3>Example Three</H3>
<P>In this example, you will create your own bigBed file from a fully-featured
existing bed file that contains all standard bed fields including the color
field.  BigBed files can store extra fields past the 
<A HREF="customTrack.html#BED">predefined bed fields</A>.  
If you add extra fields to your bigBed file (or if you use the
color field), you should include
a <TT>.as</TT> (AutoSQL) format file describing the fields.  See
<A HREF="http://www.linuxjournal.com/article/5949" TARGET=_blank>this paper</A>
for information on AutoSQL.  There are several sample <TT>.as</TT> files 
<A HREF="http://genome-test.cse.ucsc.edu/~kent/src/unzipped/hg/lib/">here</A>.
</P>

<UL>
 <LI> Save this <A HREF="bedExample2.bed.gz">bed file</A> 
  to your machine 
  (this satisfies <EM>steps 1</EM> and <EM>2</EM> above).
 <LI> Save this <A HREF="hg18.chrom.sizes">text file</A> to your machine.
  It contains the chrom.sizes for the human (hg18) assembly 
  (this satisfies <EM>step 4</EM> above).
 <LI> Save this <A HREF="bedExample2.as">.as file</A> to your machine.  
  The <TT>.as</TT> file contains a description of the fields in the bed file.  
  This is required when the bed file contains a field for color.
 <LI> Download the <TT>bedToBigBed</TT> utility (see <EM>step 3</EM>).
 <LI> Run the utility to create the bigBed output file 
  (see <EM>step 5</EM>):<BR>
  <TT><B>bedToBigBed</B> -as=bedExample2.as bedExample2.bed.gz 
   hg18.chrom.sizes myBigBed2.bb</TT>
 <LI> Place the bigBed file you just created (<TT>myBigBed2.bb</TT>) on a 
  web-accessible server (see <EM>step 6</EM>).
 <LI> Construct a &quot;track&quot; line that points to your bigBed file
  (see <EM>step 7</EM>).  Because this bigBed file includes a field for color,
  you must include the <A HREF="customTrack.html#BED"><TT>itemRgb</TT></A> 
  attribute in the &quot;track&quot; line. It will look somewhat similar to 
  this (note that you must insert the correct URL to your bigBed file):<BR>
   <TT>track type=bigBed name="bigBed Example Three" 
   description="A bigBed File with Color" itemRgb="On" 
   dataUrl=http://yourWebAddress/bigBedExample2.bb</TT>
 <LI> Create the custom track on the human assembly hg18 (Mar. 2006), and 
  view it in the genome browser (see <EM>step 8</EM>). Note that the original
  bed file contains data only on chromsome 7.
</UL>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	</TD></TR></TABLE>
</TD></TR></TABLE>
</BODY></HTML>
