<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: bigMaf Track Format</TITLE>
	<LINK REL="STYLESHEET" HREF="../../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></a>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../../images/title.jpg"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../../cgi-bin/hgMenubar"--></TD></TR>
<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=2 CELLPADDING=10>	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><b>&nbsp;  bigMaf Track Format</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
<p>
The bigMaf format stores multiple alignments in a format compatible with
<a href="../../FAQ/FAQformat.html#format5">MAF</a> files, that are then compressed and 
indexed as bigBeds are. bigMaf files are created using the program <tt>bedToBigBed</tt> with a
special AutoSQL file that defines the fields of the bigMaf. The
resulting bigMaf files are in an indexed binary format. The main advantage of
the bigMaf files is that only portions of the files needed to display a
particular region are transferred to UCSC. So for large data sets, bigMaf is
considerably faster than regular Maf files.  The bigMaf file remains on 
your web accessible server (http, https, or ftp), not on the UCSC server. 
Only the portion that is needed
for the chromosomal position you are currently viewing is locally cached as a 
&quot;sparse file&quot;.
</p>

<h4><a name=bigMaf></a>Big MAF</h4>
<p>
The following AutoSQL definition is used for bigMaf multiple alignment files. 
This is the <a href="examples/bigMaf.as"><tt>bigMaf.as</tt></a>
file defined by the <tt>-as</tt> option when using <tt>bedToBigBed</tt>.
<!--Click this <a href="examples/bigMaf.txt"><tt>bed3+1</tt></a> file for
an example of bigMaf input.  -->
</p>
<pre>
table bedMaf
"Bed3 with MAF block"
    (
    string chrom;      "Reference sequence chromosome or scaffold"
    uint   chromStart; "Start position in chromosome"
    uint   chromEnd;   "End position in chromosome"
    lstring mafBlock;   "MAF block"
    )
</pre>

<p>Note that the <tt>bedToBigBed</tt> utility uses a substantial amount of 
memory; somewhere on the order of 1.25 times more RAM than the 
uncompressed BED input file.</p>

<p>
To create a bigMaf track, follow these steps:<br>
<pre>
This needs to be incorporated below:
awk -f mafToBigMaf.awk chr22_KI270731v1_random.maf | sed '/^$/d' | sed 's/hg38.//' &gt; bigMaf.txt
hgLoadMafSummary -test hg38 bigMafSummary stdin
hgLoadMafSummary -minSeqSize=1 -test hg38 bigMafSummary chr22_KI270731v1_random.maf
cut -f 2- bigMafSummary.tab | sort -k1,1 -k2,2n &gt; bigMafSummary.bed
bedToBigBed -type=bed3+4 -as=mafSummary.as  -tab bigMafSummary.bed chrom.sizes bigMafSummary.bb
chr22_KI270731v1_random:11,822-12,023
genePredSingleCover chr22_KI270731v1_random.gp single.gp
genePredToMafFrames hg38 chr22_KI270731v1_random.maf bigMafFrames.txt hg38 single.gp
</pre>

<ol>
   <li>Create a <tt>bed3+1</tt> bigMaf format file that has the first three fields
       described by a normal BED file as described <a href="../../FAQ/FAQformat.html#format1">here</a>.
       (You can also read about MAF <a href="../../FAQ/FAQformat.html#format5">here</a>.)  
   <ul>
      <li>Your bigMaf file must have the extra field described in the AutoSQL file above:
          <tt>mafBlock</tt> 
      <li>Your bigMaf file must be sorted by chrom then chromStart. You can use
          the UNIX sort command to do this: 
          <tt>sort -k1,1 -k2,2n unsorted.bed &gt; input.bed</tt>
   </ul>
   <li>Download the <tt>bedToBigBed</tt> program from the 
       <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</a> 
       of binary utilities.
   <li>Use the <tt>fetchChromSizes</tt> script from the same 
       <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</a> 
       to create a chrom.sizes file for the UCSC database you are working with 
       (e.g. hg38). Alternatively, you can download the chrom.sizes file for
       any assembly hosted at UCSC from our <a href="http://hgdownload.soe.ucsc.edu/downloads.html">
       downloads</a> page (click on "Full data set" for any assembly). For example, for the hg38
       database, the hg38.chrom.sizes are located at
       <a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.chrom.sizes"
       target="_blank">http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.chrom.sizes</a>.
   <li>Create the bigBed file from your sorted bigMaf input file using the <tt>bedToBigBed</tt>
       utility like so: 
       <tt><b>bedToBigBed</b> -type=bed3+1 -as=bigMaf.as  -tab bigMaf.txt chrom.sizes  bigMaf.bb </tt><br>
   <li>Move the newly created bigBed file (<tt>bigMaf.bb</tt>) to an http, 
       https, or ftp location.
   <li>Construct a <a href="hgTracksHelp.html#CustomTracks">custom track</a> 
       using a single 
       <a href="hgTracksHelp.html#TRACK">track line</a>. 
       Note that any of the track attributes listed
       <a href="customTrack.html#TRACK">here</a> are applicable 
       to tracks of type bigBed. 
       The most basic version of the &quot;track&quot; line will look something 
       like this:<br>
       <tt>track type=bigMaf name="My Big MAF" description="A Multiple Alignment" bigDataUrl=http://myorg.edu/mylab/myBigGenePred.bb</tt>
   <li>Paste this custom track line into the text box in the 
       <a href="../../cgi-bin/hgCustom">custom track</a> management page.
</ol>

The <tt>bedToBigBed</tt> program can also be run with several additional options.
A full list of the available options can be seen by running
<tt>bedToBigBed</tt> with no arguments to display the usage message.

<h3>Example One</h3>
<p>
<pre>
awk -f mafToBigMaf.awk chr22_KI270731v1_random.maf | sed '/^$/d' | sed 's/hg38.//'  &gt; bigMaf.txt
<tt><B>bedToBigBed</B> -type=bed3+1 -as=bigMaf.as  -tab bigMaf.txt chrom.sizes  bigMaf.bb </tt><br>
<tt><B>bedToBigBed</B> -type=bed4+7 -as=mafFrames.as  -tab bigMafFrames.txt chrom.sizes  bigMafFrames.bb </tt><br>
track type=bigMaf name="bigMaf Example One" bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bigMaf.bb
frames=http://genome.ucsc.edu/goldenPath/help/examples/bigMafFrames.bb summary=http://genome.ucsc.edu/goldenPath/help/examples/bigMafSummary.bb
</pre>
</p>

<p>
In this example, you will use an existing bigMaf file to create a bigMaf 
custom track.  A bigMaf file that contains data on the hg38 
assembly has been placed on our http server.  
You can create a custom track using this bigMaf file by constructing a 
&quot;track&quot; line that references this file like so:
</p>

<tt>track type=bigMaf name=&quot;bigMaf Example One&quot; 
description=&quot;A bigMaf file&quot; 
bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bigMaf.bb</tt>

<p>
Paste the above &quot;track&quot; line into the 
<a href="../../cgi-bin/hgCustom?db=hg38">custom track</a> management page for the
human assembly hg38 (Dec. 2013), then press the <em>submit</em> button.

<p>
Custom tracks can also be loaded via one URL line.  The below link loads the same
bigMaf track, but includes parameters on the URL line:
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg38&hgct_customText=track%20type=bigMaf%20name=Example%20bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bigMaf.bb"
target="_blank"><tt>http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg38&hgct_customText=track%20type=bigMaf%20name=Example%20bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bigMaf.bb</tt></a>
</p>
<p>
With this example bigMaf loaded, click into a gene from the track.  Note
that the details page has a &quot;Links to sequence:&quot; section that includes
&quot;Translated Protein&quot;, &quot;Predicted mRNA&quot;, and &quot;Genomic
Sequence&quot; links. Click the &quot;Go to ... track controls&quot; link.
There change the &quot;Color track by codons:&quot; option from &quot;OFF&quot;
too &quot;genomic codons&quot; and be sure &quot;Display mode:&quot; is &quot;full&quot;
then click &quot;Submit&quot;.  Then zoom to a region where amino acids display,
such as <tt>chr9:133,255,650-133,255,700</tt> and see how bigMaf allows
the display of codons. Click back into the track controls page and click the box next
to &quot;Show codon numbering&quot;. Return to the browser to see amino acid
numbering.
</p>

<p>
You can also add a parameter in the custom track line,
<tt>baseColorDefault=genomicCodons</tt>, to set the display of codons:
<p>
<tt>browser position chr10:67,884,600-67,884,900<br>
track type=bigMaf baseColorDefault=genomicCodons name="bigMaf Example Two" description="A bigMaf file" visibility=pack bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bigMaf.bb
</tt>
<p>
Paste the above into the hg38 <a href="../../cgi-bin/hgCustom?db=hg38">custom track</a>
page to see an example of bigMaf amino acid display around the beginning of the gene SIRT1
on chromosome 10.</p>
</p>

<h3>Example Two</h3>
<p>In this example, you will create your own bigMaf file from an existing 
bigMaf input file.
<ul>
   <li>Save this <tt>bed12+8</tt> <a href="examples/bigMaf.txt">bigMaf.txt</a>
       example input file to your machine (<em>satisfies above step 1</em>).
   <li>Download the <tt>bedToBigBed</tt> <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">utility</a>
       (<em>step 2</em>).
   <li>Save this <tt>hg38.chrom.sizes</tt> <a href="hg38.chrom.sizes">text file</a> to your machine.
       It contains the chrom.sizes for the human (hg38) assembly 
       (<em>step 3</em>).
<li>Save this <tt>bigMaf.as</tt> <a href="examples/bigMaf.as"> text file</a> to your machine.
   <li>Run the utility to create the bigBed output file 
       (<em>step 4</em>):<br>
       <tt><B>bedToBigBed</B> -type=bed12+8 -tab -as=bigMaf.as bigMaf.txt hg38.chrom.sizes bigMaf.bb</tt>
   <li>Place the bigBed file you just created (<tt>bigMaf.bb</tt>) on a 
       web-accessible server (<em>step 5</em>).
   <li>Construct a &quot;track&quot; line that points to your bigMaf file
       (see <em>step 6</em>).
   <li>Create the custom track on the human assembly hg38 (Dec. 2013), and 
       view it in the genome browser (see <em>step 7</em>).
</ul>
Note the above description in <b>Example One</b> on how to view genomic codons,
including numbering. 

<h3>Sharing Your Data with Others</h3>
<p>
If you would like to share your bigMaf data track with a colleague, learn
how to create a URL by looking at <b><em>Example 11</em></b> on
<a href="customTrack.html#SHARE">this</a> page.
</p>

<h3>Extracting Data from bigBed Format</h3>
<p>
Since the bigMaf files are an extension of bigBed files, which are indexed binary files, 
they can be difficult to
extract data from.  We have developed the following 
programs, all of which are available from the 
<a href="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</a> of binary
utilities.<br>
<ul>
   <li><tt>bigBedToBed</tt> &mdash; this program converts a bigBed file 
       to ASCII BED format.
   <li><tt>bigBedSummary</tt> &mdash; this program extracts summary information
       from a bigBed file.
   <li><tt>bigBedInfo</tt> &mdash; this program prints out information about a 
       bigBed file.
</ul>
As with all UCSC Genome Browser programs, simply type the program name
at the command line with no parameters to see the usage statement.
</p>

<h3>Troubleshooting</h3>
<p>If you encounter an error when you run the <tt>bedToBigBed</tt> program,
it may be because your input bigMaf file has data off the end of a chromosome.
In this case, use the <tt>bedClip</tt> program 
<a href="http://hgdownload.cse.ucsc.edu/admin/exe/">here</a> before the
<tt>bedToBigBed</tt> program.  It will remove the row(s) in your input BED
file that are off the end of a chromosome.
</p>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	</TD></TR></TABLE>
</TD></TR></TABLE>
</BODY></HTML>
