<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:st1="urn:schemas-microsoft-com:office:smarttags"
xmlns="http://www.w3.org/TR/REC-html40">

<head>
<meta http-equiv=Content-Type content="text/html; charset=windows-1252">
<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 11">
<meta name=Originator content="Microsoft Word 11">
<link rel=File-List href="blatSpecs_files/filelist.xml">
<title>BLAT Program Specifications</title>
<o:SmartTagType namespaceuri="urn:schemas-microsoft-com:office:smarttags"
 name="City"/>
<o:SmartTagType namespaceuri="urn:schemas-microsoft-com:office:smarttags"
 name="PlaceName"/>
<o:SmartTagType namespaceuri="urn:schemas-microsoft-com:office:smarttags"
 name="PlaceType"/>
<o:SmartTagType namespaceuri="urn:schemas-microsoft-com:office:smarttags"
 name="place"/>
<!--[if gte mso 9]><xml>
 <o:DocumentProperties>
  <o:Author>Jim Kent</o:Author>
  <o:LastAuthor>Soe</o:LastAuthor>
  <o:Revision>2</o:Revision>
  <o:TotalTime>96</o:TotalTime>
  <o:LastPrinted>2003-06-11T22:46:00Z</o:LastPrinted>
  <o:Created>2005-08-11T23:07:00Z</o:Created>
  <o:LastSaved>2005-08-11T23:07:00Z</o:LastSaved>
  <o:Pages>1</o:Pages>
  <o:Words>4035</o:Words>
  <o:Characters>23002</o:Characters>
  <o:Company> </o:Company>
  <o:Lines>191</o:Lines>
  <o:Paragraphs>53</o:Paragraphs>
  <o:CharactersWithSpaces>26984</o:CharactersWithSpaces>
  <o:Version>11.6408</o:Version>
 </o:DocumentProperties>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <w:WordDocument>
  <w:DisplayHorizontalDrawingGridEvery>0</w:DisplayHorizontalDrawingGridEvery>
  <w:DisplayVerticalDrawingGridEvery>0</w:DisplayVerticalDrawingGridEvery>
  <w:UseMarginsForDrawingGridOrigin/>
  <w:ValidateAgainstSchemas/>
  <w:SaveIfXMLInvalid>false</w:SaveIfXMLInvalid>
  <w:IgnoreMixedContent>false</w:IgnoreMixedContent>
  <w:AlwaysShowPlaceholderText>false</w:AlwaysShowPlaceholderText>
  <w:Compatibility>
   <w:FootnoteLayoutLikeWW8/>
   <w:ShapeLayoutLikeWW8/>
   <w:AlignTablesRowByRow/>
   <w:ForgetLastTabAlignment/>
   <w:LayoutRawTableWidth/>
   <w:LayoutTableRowsApart/>
   <w:UseWord97LineBreakingRules/>
   <w:SelectEntireFieldWithStartOrEnd/>
   <w:UseWord2002TableStyleRules/>
  </w:Compatibility>
  <w:BrowserLevel>MicrosoftInternetExplorer4</w:BrowserLevel>
 </w:WordDocument>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <w:LatentStyles DefLockedState="false" LatentStyleCount="156">
 </w:LatentStyles>
</xml><![endif]--><!--[if !mso]><object
 classid="clsid:38481807-CA0E-42D2-BF39-B33AF135CC4D" id=ieooui></object>
<style>
st1\:*{behavior:url(#ieooui) }
</style>
<![endif]-->
<style>
<!--
 /* Font Definitions */
 @font-face
	{font-family:Courier;
	panose-1:2 7 4 9 2 2 5 2 4 4;
	mso-font-alt:"Courier New";
	mso-font-charset:0;
	mso-generic-font-family:modern;
	mso-font-format:other;
	mso-font-pitch:fixed;
	mso-font-signature:3 0 0 0 1 0;}
@font-face
	{font-family:Wingdings;
	panose-1:5 0 0 0 0 0 0 0 0 0;
	mso-font-charset:2;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:0 268435456 0 0 -2147483648 0;}
@font-face
	{font-family:Tahoma;
	panose-1:2 11 6 4 3 5 4 4 2 4;
	mso-font-charset:0;
	mso-generic-font-family:swiss;
	mso-font-pitch:variable;
	mso-font-signature:1627421319 -2147483648 8 0 66047 0;}
@font-face
	{font-family:"Fang Song";
	mso-font-charset:80;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:16777216 3592 268435456 0 262144 0;}
 /* Style Definitions */
 p.MsoNormal, li.MsoNormal, div.MsoNormal
	{mso-style-parent:"";
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman";
	mso-fareast-font-family:"Times New Roman";}
h2
	{mso-style-next:Normal;
	margin-top:12.0pt;
	margin-right:0in;
	margin-bottom:3.0pt;
	margin-left:0in;
	mso-pagination:widow-orphan;
	page-break-after:avoid;
	mso-outline-level:2;
	font-size:12.0pt;
	mso-bidi-font-size:10.0pt;
	font-family:Arial;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-font-weight:normal;
	font-style:italic;
	mso-bidi-font-style:normal;}
h3
	{mso-style-next:Normal;
	margin-top:12.0pt;
	margin-right:0in;
	margin-bottom:3.0pt;
	margin-left:0in;
	mso-pagination:widow-orphan;
	page-break-after:avoid;
	mso-outline-level:3;
	font-size:12.0pt;
	mso-bidi-font-size:10.0pt;
	font-family:Arial;
	mso-bidi-font-family:"Times New Roman";
	font-weight:normal;}
p.MsoBodyTextIndent, li.MsoBodyTextIndent, div.MsoBodyTextIndent
	{margin-top:0in;
	margin-right:0in;
	margin-bottom:0in;
	margin-left:193.7pt;
	margin-bottom:.0001pt;
	text-indent:-193.7pt;
	mso-pagination:widow-orphan;
	tab-stops:193.5pt;
	font-size:10.0pt;
	font-family:"Courier New";
	mso-fareast-font-family:"Times New Roman";
	mso-bidi-font-family:"Times New Roman";}
a:link, span.MsoHyperlink
	{color:blue;
	text-decoration:underline;
	text-underline:single;}
a:visited, span.MsoHyperlinkFollowed
	{color:purple;
	text-decoration:underline;
	text-underline:single;}
p.MsoDocumentMap, li.MsoDocumentMap, div.MsoDocumentMap
	{mso-style-noshow:yes;
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	background:navy;
	font-size:10.0pt;
	font-family:Tahoma;
	mso-fareast-font-family:"Times New Roman";
	mso-bidi-font-family:"Times New Roman";}
@page Section1
	{size:8.5in 11.0in;
	margin:1.0in 1.25in .8in 1.25in;
	mso-header-margin:.5in;
	mso-footer-margin:.5in;
	mso-paper-source:0;}
div.Section1
	{page:Section1;}
 /* List Definitions */
 @list l0
	{mso-list-id:125394729;
	mso-list-type:hybrid;
	mso-list-template-ids:-770682052;}
@list l0:level1
	{mso-level-text:"%1\)";
	mso-level-tab-stop:.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1
	{mso-list-id:348071437;
	mso-list-type:hybrid;
	mso-list-template-ids:-458559594;}
@list l1:level1
	{mso-level-text:"%1\)";
	mso-level-tab-stop:.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2
	{mso-list-id:1009257236;
	mso-list-type:hybrid;
	mso-list-template-ids:1662435710;}
@list l2:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:.5in;
	mso-level-number-position:left;
	text-indent:-.25in;
	font-family:Symbol;}
@list l3
	{mso-list-id:1239092154;
	mso-list-type:hybrid;
	mso-list-template-ids:50606982;}
@list l3:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:.5in;
	mso-level-number-position:left;
	text-indent:-.25in;
	font-family:Symbol;}
@list l4
	{mso-list-id:1292908230;
	mso-list-type:hybrid;
	mso-list-template-ids:724436240;}
@list l4:level1
	{mso-level-text:"%1\)";
	mso-level-tab-stop:.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l5
	{mso-list-id:1919442949;
	mso-list-type:hybrid;
	mso-list-template-ids:-915084300;}
@list l5:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:.5in;
	mso-level-number-position:left;
	text-indent:-.25in;
	font-family:Symbol;}
ol
	{margin-bottom:0in;}
ul
	{margin-bottom:0in;}
-->
</style>
<!--[if gte mso 10]>
<style>
 /* Style Definitions */
 table.MsoNormalTable
	{mso-style-name:"Table Normal";
	mso-tstyle-rowband-size:0;
	mso-tstyle-colband-size:0;
	mso-style-noshow:yes;
	mso-style-parent:"";
	mso-padding-alt:0in 5.4pt 0in 5.4pt;
	mso-para-margin:0in;
	mso-para-margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman";
	mso-ansi-language:#0400;
	mso-fareast-language:#0400;
	mso-bidi-language:#0400;}
</style>
<![endif]-->
</head>

<body lang=EN-US link=blue vlink=purple style='tab-interval:.5in'>

<div class=Section1>

<h2><span style='font-size:16.0pt;mso-bidi-font-size:10.0pt;font-weight:normal;
font-style:normal'>BLAT Suite Program Specifications and User Guide<o:p></o:p></span></h2>

<h3><span style='font-size:16.0pt;mso-bidi-font-size:10.0pt'>General:<o:p></o:p></span></h3>

<p class=MsoNormal>Blat produces two major classes of alignments: at the DNA
level between two sequences that are of 95% or greater identity, but which may
include large inserts, and at the protein or translated DNA level between
sequences that are of 80% or greater identity and may also include large
inserts.<span style='mso-spacerun:yes'>  </span>The output of BLAT is
flexible.<span style='mso-spacerun:yes'>  </span>By default it is a simple
tab-delimited file which describes the alignment, but which does not include
the sequence of the alignment itself.<span style='mso-spacerun:yes'> 
</span>Optionally it can produce BLAST and WU-BLAST compatable output as well
as a number of other formats.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>The main programs in the blat suite are:</p>

<ul style='margin-top:0in' type=disc>
 <li class=MsoNormal style='mso-list:l5 level1 lfo4;tab-stops:list .5in'>gfServer
     – a server that maintains an index of the genome in memory and uses the
     index to quickly find regions with high levels of sequence similarity to a
     query sequence.</li>
 <li class=MsoNormal style='mso-list:l5 level1 lfo4;tab-stops:list .5in'>gfClient
     – a program that queries gfServer over the network, and then does a
     detailed alignment of the query sequence with regions found by gfServer.</li>
 <li class=MsoNormal style='mso-list:l5 level1 lfo4;tab-stops:list .5in'>blat
     –combines client and server into a single program, first building the
     index, then using the index, and then exiting.<span
     style='mso-spacerun:yes'>  </span></li>
 <li class=MsoNormal style='mso-list:l5 level1 lfo4;tab-stops:list .5in'>webBlat
     – a web based version of gfClient that presents the alignments in an
     interactive fashion.</li>
</ul>

<p class=MsoNormal>Building an index of the genome typically takes 10 or 15
minutes. <span style='mso-spacerun:yes'> </span>Typically for interactive
applications one uses gfServer to build a whole genome index.<span
style='mso-spacerun:yes'>  </span>At that point gfClient or webBlat can align a
single query within few seconds.<span style='mso-spacerun:yes'>  </span>If one
is aligning a lot of sequences in a batch mode then blat can be more efficient,
particularly if run on a cluster of computers.<span style='mso-spacerun:yes'> 
</span>Each blat run is typically done against a single chromosome, but with a
large number of query sequences.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>Other programs in the blat suite are:</p>

<ul style='margin-top:0in' type=disc>
 <li class=MsoNormal style='mso-list:l2 level1 lfo5;tab-stops:list .5in'>pslSort
     – combines and sorts the output of multiple blat runs. <span
     style='mso-spacerun:yes'> </span>(The blat default output format is .psl).
     </li>
 <li class=MsoNormal style='mso-list:l2 level1 lfo5;tab-stops:list .5in'>pslReps
     – selects the best alignments for a particular query sequence, using a
     ‘near best in genome’ approach.</li>
 <li class=MsoNormal style='mso-list:l2 level1 lfo5;tab-stops:list .5in'>pslPretty
     – converts alignments from the psl format, which is tab-delimited format
     and does not include the bases themselves, to a more readable alignment
     format. </li>
 <li class=MsoNormal style='mso-list:l2 level1 lfo5;tab-stops:list .5in'>faToTwoBit
     – convert Fasta format sequence files to a dense randomly-accessable<span
     style='mso-spacerun:yes'>  </span>.2bit format that gfClient can use.</li>
 <li class=MsoNormal style='mso-list:l2 level1 lfo5;tab-stops:list .5in'>twoBitToFa
     – convert from the .2bit format back to fasta</li>
 <li class=MsoNormal style='mso-list:l2 level1 lfo5;tab-stops:list .5in'>faToNib
     – convert from Fasta to a somewhat less dense randomly accessible format
     that predates .2bit.<span style='mso-spacerun:yes'>  </span>Note each .nib
     file can only contain a single sequence.</li>
 <li class=MsoNormal style='mso-list:l2 level1 lfo5;tab-stops:list .5in'>nibFrag
     – convert portions of a nib file back to fasta.</li>
</ul>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>In addition you may be interested in the following programs
which are not part of the BLAT suite:</p>

<ul style='margin-top:0in' type=disc>
 <li class=MsoNormal style='mso-list:l3 level1 lfo6;tab-stops:list .5in'>In
     Silico PCR –<span style='mso-spacerun:yes'>  </span>given two primers
     quickly find the sequence between them.<span style='mso-spacerun:yes'> 
     </span>Available from Kent Informatics. This includes webPCR, an interface
     similar to webBlat.</li>
 <li class=MsoNormal style='mso-list:l3 level1 lfo6;tab-stops:list .5in'>The
     Genome Browser – display annotations as a series of tracks on top of the
     genome.<span style='mso-spacerun:yes'>  </span>Available from the <st1:PlaceType
     w:st="on">University</st1:PlaceType> of <st1:PlaceName w:st="on">California</st1:PlaceName>
     <st1:City w:st="on"><st1:place w:st="on">Santa Cruz</st1:place></st1:City>.<span
     style='mso-spacerun:yes'>  </span>See <a
     href="http://genome.ucsc.edu/license/">http://genome.ucsc.edu/license/</a>.</li>
</ul>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<h3><span style='font-size:16.0pt;mso-bidi-font-size:10.0pt'>Running<span
style='mso-spacerun:yes'>  </span>the<span style='mso-spacerun:yes'> 
</span>Programs:<o:p></o:p></span></h3>

<p class=MsoNormal>The command line options of each of the programs is described
below. Similar summaries of usage are printed when a command is run with no
arguments.<span style='mso-spacerun:yes'>  </span>See the next section for info
on installing webBlat.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>blat<o:p></o:p></span></b></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>blat
- Standalone BLAT sequence search command line tool<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>blat database query [-ooc=11.ooc] output.psl<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>where:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>database is either a .fa file, a .nib file,
or a list of .fa or .nib<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>files, query is similarly a .fa, .nib, or
list of .fa or .nib files<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-ooc=11.ooc tells the program to load
over-occurring 11-mers from<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>and external file.<span
style='mso-spacerun:yes'>  </span>This will increase the speed<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>by a factor of 40 in many cases,
but is not required<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>output.psl is where to put the output.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>options:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-t=type<span style='mso-spacerun:yes'>    
</span>Database type.<span style='mso-spacerun:yes'>  </span>Type is one of:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span>dna - DNA sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                </span><span
style='mso-spacerun:yes'> </span>prot - protein sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span>dnax - DNA sequence translated
in six frames to protein<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>The default is dna<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-q=type<span style='mso-spacerun:yes'>    
</span>Query type.<span style='mso-spacerun:yes'>  </span>Type is one of:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span>dna - DNA sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span>rna - RNA sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>        </span><span
style='mso-spacerun:yes'>         </span>prot - protein sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span>dnax - DNA sequence translated
in six frames to protein<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span>rnax - DNA sequence translated
in three frames to protein<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>The default is dna<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-prot<span style='mso-spacerun:yes'>      
</span>Synonymous with -d=prot -q=prot<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-ooc=N.ooc<span style='mso-spacerun:yes'> 
</span>Use overused tile file N.ooc.<span style='mso-spacerun:yes'>  </span>N
should correspond to<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>the tileSize<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-tileSize=N sets the size of match that
triggers an alignment.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>Usually between 8 and 12<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>Default is 11 for DNA and 5 for protein.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-oneOff=N<span style='mso-spacerun:yes'>  
</span>If set to 1 this allows one mismatch in tile and still<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>triggers an alignments.<span
style='mso-spacerun:yes'>  </span>Default is 0.<o:p></o:p></span></p>

<span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-fareast-font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman";
mso-ansi-language:EN-US;mso-fareast-language:EN-US;mso-bidi-language:AR-SA'><br
clear=all style='page-break-before:always'>
</span>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-minMatch=N sets the number of tile
matches.<span style='mso-spacerun:yes'>  </span>Usually set from 2 to 4<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>Default is 2 for nucleotide, 1
for protein.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-minScore=N sets minimum score.<span
style='mso-spacerun:yes'>  </span>This is twice the matches minus the<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>mismatches minus some sort of
gap penalty.<span style='mso-spacerun:yes'>  </span>Default is 30<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-minIdentity=N Sets minimum sequence
identity (in percent).<span style='mso-spacerun:yes'>  </span>Default is<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>90 for nucleotide searches, 25
for protein or translated<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>protein searches.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-maxGap=N<span style='mso-spacerun:yes'>  
</span>sets the size of maximum gap between tiles in a clump.<span
style='mso-spacerun:yes'>  </span>Usually<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>set from 0 to 3.<span
style='mso-spacerun:yes'>  </span>Default is 2. Only relevent for minMatch &gt;
1.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-noHead<span style='mso-spacerun:yes'>    
</span>suppress .psl header (so it's just a tab-separated file)<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-makeOoc=N.ooc Make overused tile file<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-repMatch=N sets the number of repetitions
of a tile allowed before<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>it is marked as overused.<span
style='mso-spacerun:yes'>  </span>Typically this is 256 for tileSize<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>             </span><span
style='mso-spacerun:yes'>  </span>12, 1024 for tile size 11, 4096 for tile size
10.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>Default is 1024.<span
style='mso-spacerun:yes'>  </span>Typically only comes into play with makeOoc<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-mask=type<span style='mso-spacerun:yes'> 
</span>Mask out repeats.<span style='mso-spacerun:yes'>  </span>Alignments
won't be started in masked region<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>but may extend through it in
nucleotide searches.<span style='mso-spacerun:yes'>  </span>Masked areas<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>are ignored entirely in protein
or translated searches. Types are<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span>lower - mask out lower cased
sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span>upper - mask out upper cased
sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span>out<span
style='mso-spacerun:yes'>   </span>- mask according to database.out
RepeatMasker .out file<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span>file.out - mask database
according to RepeatMasker file.out<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-qMask=type Mask out repeats in query
sequence.<span style='mso-spacerun:yes'>  </span>Similar to -mask above but<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>for query rather than target
sequence.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'> </span><span style='mso-spacerun:yes'> 
</span>-minRepDivergence=NN - minimum percent divergence of repeats to allow<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>them to be unmasked.<span
style='mso-spacerun:yes'>  </span>Default is 15.<span
style='mso-spacerun:yes'>  </span>Only relevant for<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>masking using RepeatMasker .out
files.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-dots=N<span style='mso-spacerun:yes'>    
</span>Output dot every N sequences to show program's progress<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-trimT<span style='mso-spacerun:yes'>     
</span>Trim leading poly-T<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-noTrimA<span style='mso-spacerun:yes'>   
</span>Don't trim trailing poly-A<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-trimHardA<span style='mso-spacerun:yes'> 
</span>Remove poly-A tail from qSize as well as alignments in psl output<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-out=type<span style='mso-spacerun:yes'>  
</span>Controls output file format.<span style='mso-spacerun:yes'>  </span>Type
is one of:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span>psl - Default.<span
style='mso-spacerun:yes'>  </span>Tab separated format without actual sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span>pslx - Tab separated format
with sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span>axt - blastz-associated axt
format<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span>maf - multiz-associated maf
format<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span>wublast - similar to wublast
format<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span>blast - similar to NCBI
blast format<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-fine<span style='mso-spacerun:yes'>      
</span>For high quality mRNAs look harder for small initial and<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>terminal exons.<span
style='mso-spacerun:yes'>  </span>Not recommended for ESTs<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>Here are some blat settings for
common usage scenarios:</p>

<p class=MsoNormal style='margin-right:-.25in'><o:p>&nbsp;</o:p></p>

<p class=MsoNormal style='margin-right:-.25in'>1) Mapping ESTs to the genome
within the same species</p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-ooc=11.ooc<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>2) Mapping full length mRNAs to
the genome in the same species</p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-ooc=11.ooc -fine -q=rna<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>3) Mapping ESTs to the genome
across species</p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-q=dnax -t=dnax<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>4) Mapping mRNA to the genome
across species</p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-q=rnax -t=dnax<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>5) Mapping proteins to the
genome</p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-q=prot -t=dnax<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>6) Mapping DNA to DNA in the
same species</p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-ooc=11.ooc -fastMap<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>7) Mapping DNA from one species
to another species</p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-q=dnax -t=dnax<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>When mapping DNA from one species to
another the<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>query side of the alignment should be cut
up into chunks<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>of 25kb or less for best performance.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<b style='mso-bidi-font-weight:normal'><span style='font-size:12.0pt;
mso-bidi-font-size:10.0pt;font-family:"Times New Roman";mso-fareast-font-family:
"Times New Roman";mso-ansi-language:EN-US;mso-fareast-language:EN-US;
mso-bidi-language:AR-SA'><br clear=all style='page-break-before:always'>
</span></b>

<p class=MsoNormal style='margin-right:-.25in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>gfServer<o:p></o:p></span></b></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>gfServer -
Make a server to quickly find where DNA occurs in genome.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To set up a
server:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>gfServer start host port file(s).nib<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To remove a
server:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>gfServer stop host port<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To query a
server with DNA sequence:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>gfServer query host port probe.fa<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To query a
server with protein sequence:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>gfServer protQuery host port probe.fa<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To query a
server with translated dna sequence:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>gfServer transQuery host port probe.fa<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To process
one probe fa file against a .nib format genome (not starting server):<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>gfServer direct probe.fa file(s).nib<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To figure out
usage level<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>gfServer status host port<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To get input
file list<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>gfServer files host port<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>Options:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-tileSize=N size of n-mers to index.<span
style='mso-spacerun:yes'>  </span>Default is 11 for nucleotides, 4 for<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>proteins (or translated
nucleotides).<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-minMatch=N Number of n-mer matches that
trigger detailed alignment<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>Default is 2 for nucleotides, 3
for protiens.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-maxGap=N<span style='mso-spacerun:yes'>  
</span>Number of insertions or deletions allowed between n-mers.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>Default is 2 for nucleotides, 0
for protiens.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-trans<span style='mso-spacerun:yes'>     
</span>Translate database to protein in 6 frames.<span
style='mso-spacerun:yes'>  </span>Note: it is best<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>to run this on RepeatMasked data
in this case.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-log=logFile keep a log file that records
server requests.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-seqLog<span style='mso-spacerun:yes'>   
</span>Include sequences in log file<o:p></o:p></span></p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>gfClient<o:p></o:p></span></b></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>gfClient - A
client for the genomic finding program <o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>gfClient host port nibDir in.fa out.psl<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>where<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>host is the name of the machine running the
gfServer<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>port is the same as you started the gfServer
with<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>nibDir is the path of the nib files relative
to the current dir<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>       </span>(note these are needed by the client as
well as the server)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>in.fa a fasta format file.<span
style='mso-spacerun:yes'>  </span>May contain multiple records<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>out.psl where to put the output<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>options:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-t=type<span style='mso-spacerun:yes'>    
</span>Database type.<span style='mso-spacerun:yes'>  </span>Type is one of:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span>dna - DNA sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span>prot - protein sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>          </span><span
style='mso-spacerun:yes'>       </span>dnax - DNA sequence translated in six
frames to protein<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>The default is dna<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-q=type<span style='mso-spacerun:yes'>    
</span>Query type.<span style='mso-spacerun:yes'>  </span>Type is one of:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span>dna - DNA sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span>rna - RNA sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span>prot - protein sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>  </span><span
style='mso-spacerun:yes'>               </span>dnax - DNA sequence translated
in six frames to protein<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span>rnax - DNA sequence translated
in three frames to protein<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-dots=N<span style='mso-spacerun:yes'>  
</span>Output a dot every N query sequences<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-nohead<span style='mso-spacerun:yes'>  
</span>Suppresses psl five line header<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-minScore=N sets minimum score.<span
style='mso-spacerun:yes'>  </span>This is twice the matches minus the <o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>mismatches minus some sort of
gap penalty.<span style='mso-spacerun:yes'>  </span>Default is 30<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-minIdentity=N Sets minimum sequence
identity (in percent).<span style='mso-spacerun:yes'>  </span>Default is<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>90 for nucleotide searches, 25
for protein or translated<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>protein searches.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-out=type<span style='mso-spacerun:yes'>  
</span>Controls output file format.<span style='mso-spacerun:yes'>  </span>Type
is one of:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span>psl - Default.<span
style='mso-spacerun:yes'>  </span>Tab separated format without actual sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span>pslx - Tab separated format
with sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span>axt - blastz-associated axt
format<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span>maf - multiz-associated maf
format<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span>wublast - similar to wublast
format<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span>blast - similar to NCBI
blast format<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-maxIntron=N<span style='mso-spacerun:yes'> 
</span>Sets maximum intron size. Default is 750000<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'>webBlat<o:p></o:p></span></b></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>webBlat
generally is not run from the command line.<span style='mso-spacerun:yes'> 
</span>See ‘Setting Up webBlat instructions below` for information on this
program.<o:p></o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'>faToTwoBit<o:p></o:p></span></b></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>faToTwoBit -
Convert DNA from fasta to 2bit format<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>faToTwoBit in.fa [in2.fa in3.fa ...]
out.2bit<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>options:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-noMask - Ignore lower-case masking in fa
file.<o:p></o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'>twoBitToFa<o:p></o:p></span></b></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>twoBitToFa -
Convert all or part of .2bit file to fasta<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>twoBitToFa input.2bit output.fa<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>options:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-seq=name - restrict this to just one
sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-start=X<span style='mso-spacerun:yes'> 
</span>- start at given position in sequence (zero-based)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-end=X - end at given position in sequence
(non-inclusive)</span><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'><o:p></o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'>faToNib<o:p></o:p></span></b></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>faToNib -
Convert from .fa to .nib format<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>faToNib in.fa out.nib<o:p></o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>nibFrag<o:p></o:p></span></b></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>nibFrag -
Extract part of a nib file as .fa<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>nibFrag file.nib start end strand out.fa<o:p></o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>pslPretty<o:p></o:p></span></b></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>pslPretty -
Convert PSL to human readable output<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>pslPretty in.psl target.lst query.lst
pretty.out<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>options:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-axt - save in Scott Schwartz's axt format<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-dot=N Put out a dot every N records<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-long - Don't abbreviate long inserts<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>It's a really good idea if the
psl file is sorted by target if it contains multiple targets.<span
style='mso-spacerun:yes'>  </span>Otherwise this will be very very slow.<span
style='mso-spacerun:yes'>   </span>The target and query lists can either be fasta
files, nib files, or a list of fasta and/or nib files one per line.<span
style='mso-spacerun:yes'>  </span>Currently this only handles nucleotide based
psl files.</p>

<p class=MsoNormal style='margin-right:-.25in'><o:p>&nbsp;</o:p></p>

<p class=MsoNormal style='margin-right:-.25in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>pslSort<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>pslSort - merge and sort
psCluster .psl output files<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>  </span>pslSort dirs[1|2] outFile tempDir inDir(s)<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>This will sort all of the
.psl files in the directories<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>inDirs in two stages -
first into temporary files in tempDir<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>and second into
outFile.<span style='mso-spacerun:yes'>  </span>The device on tempDir needs to
have<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>enough space (typically
15-20 gigabytes if processing whole genome)<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>  </span>pslSort g2g[1|2] outFile tempDir inDir(s)<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>This will sort a genome to
genome alignment, reflecting the<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>alignments across the
diagonal.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>Adding 1 or 2 after the
dirs or g2g will limit the program to<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>only the first or second
pass repectively of the sort<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>Options:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-verbose=N Set verbosity level, higher for
more output. Default 1<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>Note for huge files pslSort
will run out of memory.<span style='mso-spacerun:yes'>  </span>The unix sort
command<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>  </span>sort –k 10 *.psl &gt; sorted.psl<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>may be preferable in these
situations, though the psl header lines should<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>be removed or avoided with
the –noHead option to blat.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;font-family:
"Courier New";mso-fareast-font-family:"Fang Song";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal style='margin-right:-.25in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>pslReps</span></b><span
style='font-size:9.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-fareast-font-family:"Fang Song";mso-bidi-font-family:"Times New Roman"'><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>pslReps - analyse repeats
and generate genome wide best<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>alignments from a sorted
set of local alignments<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>pslReps in.psl out.psl out.psr<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>where in.psl is an
alignment file generated by psLayout and<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>sorted by pslSort, out.psl
is the best alignment output<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>and out.psr contains repeat
info<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>options:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-nohead don't add PSL header<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-ignoreSize Will not weigh in favor of
larger alignments so much<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-noIntrons Will not penalize for not having
introns when calculating<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>       </span><span
style='mso-spacerun:yes'>       </span>size factor<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-singleHit<span style='mso-spacerun:yes'> 
</span>Takes single best hit, not splitting into parts<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-minCover=0.N minimum coverage to
output.<span style='mso-spacerun:yes'>  </span>Default is 0.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-ignoreNs Ignore 'N's when calculating
minCover.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-minAli=0.N minimum alignment ratio<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>default is 0.93<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-nearTop=0.N how much can deviate from top
and be taken<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>default is 0.01<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-minNearTopSize=N<span
style='mso-spacerun:yes'>  </span>Minimum size of alignment that is near top<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>for aligmnent to be kept.<span
style='mso-spacerun:yes'>  </span>Default 30.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-coverQSizes=file Tab-separate file with
effective query sizes.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                     </span>When used with -minCover,
this allows polyAs<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                     </span>to be excluded from the
coverage calculation<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><o:p>&nbsp;</o:p></p>

<h3><span style='font-size:16.0pt;mso-bidi-font-size:10.0pt'>Setting Up webBlat<o:p></o:p></span></h3>

<p class=MsoNormal>INSTALLING WEBBLAT</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>Installing A Web-Based Blat Server involves four major
steps:</p>

<p class=MsoNormal><span style='mso-spacerun:yes'> </span>1) Creating sequence
databases.</p>

<p class=MsoNormal><span style='mso-spacerun:yes'> </span>2) Running the
gfServer program to create in-memory indexes of the databases.</p>

<p class=MsoNormal><span style='mso-spacerun:yes'> </span>3) Editing the
webBlat.cfg file to tell it what machine and port the gfServer(s)</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>    </span>are running on,
and optionally customizing the webBlat appearance to users.</p>

<p class=MsoNormal><span style='mso-spacerun:yes'> </span>4) Copying the
webBlat executable and webBlat.cfg to a directory where the web server can</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>    </span>execute webBlat as
a CGI.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>CREATING SEQUENCE DATABASES</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>You create databases with the program faToTwoBit. Typically
you'll create</p>

<p class=MsoNormal>a separate database for each genome you are indexing.<span
style='mso-spacerun:yes'>  </span>Each database can</p>

<p class=MsoNormal>contain up to four billion bases of sequence in an unlimited
number of</p>

<p class=MsoNormal>records.<span style='mso-spacerun:yes'>  </span>The
databases for webPcr and webBlat are identical.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>The input to faToTwoBit is one or more fasta format files
each of which</p>

<p class=MsoNormal>can contain multiple records.<span
style='mso-spacerun:yes'>  </span>If the sequence contains repeat sequences,</p>

<p class=MsoNormal>as is the case with vertebrates and many plants, the repeat
sequences can</p>

<p class=MsoNormal>be represented in lower case and the other sequence in upper
case.<span style='mso-spacerun:yes'>  </span>The gfServer</p>

<p class=MsoNormal>program can be configured to ignore the repeat
sequences.<span style='mso-spacerun:yes'>  </span>The output of</p>

<p class=MsoNormal>faToTwoBit is a file which is designed for fast random
access and efficient</p>

<p class=MsoNormal>storage.<span style='mso-spacerun:yes'>  </span>The output
files store four bases per byte.<span style='mso-spacerun:yes'>  </span>They
use a small amount</p>

<p class=MsoNormal>of additional space to store the case of the DNA and to keep
track of runs of</p>

<p class=MsoNormal>N's in the input.<span style='mso-spacerun:yes'> 
</span>Non-N ambiguity codes such as Y and U in the input sequence</p>

<p class=MsoNormal>will be converted to N.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>Here's how a typical installation might create a mouse and a
human genome database:</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>    </span>cd /data/genomes</p>

<p class=MsoNormal><span style='mso-spacerun:yes'> </span><span
style='mso-spacerun:yes'>   </span>mkdir twoBit</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>    </span>faToTwoBit
human/hg16/*.fa twoBit/hg16.2bit</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>    </span>faToTwoBit
mouse/mm4/*.fa twoBit/mm4.2bit</p>

<p class=MsoNormal>There's no need to put all of the databases in the same
directory, but it can</p>

<p class=MsoNormal>simplify bookkeeping.<span style='mso-spacerun:yes'> 
</span></p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>The databases can also be in the .nib format which was used
with blat and</p>

<p class=MsoNormal>gfClient/gfServer until recently.<span
style='mso-spacerun:yes'>  </span>The .nib format only packed 2 bases per</p>

<p class=MsoNormal>byte, and could only handle one record per nib file.<span
style='mso-spacerun:yes'>  </span>Recent versions of blat</p>

<p class=MsoNormal>and related programs can use .2bit files as well.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>CREATING IN-MEMORY INDICES WITH GFSERVER</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>The gfServer program creates an in-memory index of a
nucleotide sequence database.<span style='mso-spacerun:yes'>  </span></p>

<p class=MsoNormal>The index can either be for translated or untranslated
searches.<span style='mso-spacerun:yes'>  </span>Translated indexes</p>

<p class=MsoNormal>enable protein-based blat queries and use approximately two
bytes per unmasked base </p>

<p class=MsoNormal>in the database.<span style='mso-spacerun:yes'> 
</span>Untranslated indexes are used nucleotide-based blat queries as well</p>

<p class=MsoNormal>as for In-silico PCR.<span style='mso-spacerun:yes'> 
</span>An index for normal blat uses approximately 1/4 byte per base.<span
style='mso-spacerun:yes'>  </span>For</p>

<p class=MsoNormal>blat on smaller (primer-sized) queries or for In-silico PCR
a more thorough index that</p>

<p class=MsoNormal>requires 1/2 byte per base is recommended.<span
style='mso-spacerun:yes'>  </span>The gfServer is memory intensive but
typically</p>

<p class=MsoNormal>doesn not require a lot of CPU power.<span
style='mso-spacerun:yes'>  </span>Memory permitting multiple gfServers can be</p>

<p class=MsoNormal>run on the same machine.<span style='mso-spacerun:yes'> 
</span></p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>A typical installation might go:</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>    </span>ssh bigRamMachine</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>    </span>cd
/data/genomes/twoBit</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>    </span>gfServer start
bigRamMachine 17779 hg16.2bit &amp;</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>    </span>gfServer -trans
-mask start bigRamMachine 17778 hg16.2bit &amp;</p>

<p class=MsoNormal>the -trans flag makes a translated index.<span
style='mso-spacerun:yes'>   </span>It will take approximately</p>

<p class=MsoNormal>15 minutes to build an untranslated index, and 45 minutes to
build a</p>

<p class=MsoNormal>translate index.<span style='mso-spacerun:yes'>  </span>To
build an untranslated index to be shared with </p>

<p class=MsoNormal>In-silico PCR do</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>    </span>gfServer
-stepSize=5 bigRamMachine 17779 hg16.2bit &amp;</p>

<p class=MsoNormal>This index will be slightly more sensitive, noticeably so for
small query sequences,</p>

<p class=MsoNormal>with blat.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>EDITING THE WEBBLAT.CFG FILE</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>The webBlat.cfg file tells the webBlat program where to look
for gfServers and</p>

<p class=MsoNormal>for sequence.<span style='mso-spacerun:yes'>  </span>The
basic format of the .cfg file is line oriented with the</p>

<p class=MsoNormal>first word of the line being a command.<span
style='mso-spacerun:yes'>  </span>Blank lines and lines starting with #</p>

<p class=MsoNormal>are ignored.<span style='mso-spacerun:yes'>  </span>The
webBlat.cfg and webPcr.cfg files are similar. The webBlat.cfg</p>

<p class=MsoNormal>commands are:</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>   </span>gfServer<span
style='mso-spacerun:yes'>  </span>-<span style='mso-spacerun:yes'> 
</span>defines host and port a (untranslated) gfServer is running on, the </p>

<p class=MsoNormal><span style='mso-spacerun:yes'>   </span><span
style='mso-tab-count:2'>                             </span>associated sequence
directory, and the name of the database to display in</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>                </span>the
webPcr web page.</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>   </span>gfServerTrans -
defines location of a translated server.</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>   </span>background -
defines the background image if any to display on web page</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>   </span>company<span
style='mso-spacerun:yes'>    </span>- defines company name to display on web
page</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>   </span>tempDir<span
style='mso-spacerun:yes'>    </span>- where to put temporary files.<span
style='mso-spacerun:yes'>  </span>This path is relative to where the web</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>                </span>server
executes CGI scripts.<span style='mso-spacerun:yes'>  </span>It is good to
remove files that haven't</p>

<p class=MsoNormal><span style='mso-tab-count:2'>                                </span>been
accessed for 24 hours from this directory periodically, </p>

<p class=MsoNormal><span style='mso-tab-count:2'>                                </span>via
a cron job or similar mechanism.</p>

<p class=MsoNormal>The background and company commands are optional.<span
style='mso-spacerun:yes'>  </span>The webBlat.cfg file must</p>

<p class=MsoNormal>have at least one valid gfServer or gfServerTrans line, and
a tempDir line.</p>

<p class=MsoNormal>.<span style='mso-spacerun:yes'>  </span>Here is a
webBlat.cfg file that you</p>

<p class=MsoNormal>might find at a typical installation:</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>company Awesome Research Amalgamated</p>

<p class=MsoNormal>background /images/dnaPaper.jpg</p>

<p class=MsoNormal>gfServer bigRamMachine 17778 /data/genomes/2bit/hg16.2bit
Human Genome</p>

<p class=MsoNormal>gfServer bigRamMachine 17779 /data/genomes/2bit/hg16.2bit
Human Genome</p>

<p class=MsoNormal>gfServer mouseServer 17780 /data/genomes/2bit/mm4.2bit Mouse
Genome</p>

<p class=MsoNormal>gfServer mouseServer 17781 /data/genomes/2bit/mm4.2bit Mouse
Genome</p>

<p class=MsoNormal>tempDir ../trash</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>PUTTING WEBBLAT WHERE THE WEB SERVER CAN EXECUTE IT</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>The details of this step vary highly from web server to web
server.<span style='mso-spacerun:yes'>  </span>On</p>

<p class=MsoNormal>a typical Apache installation it might be:</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>     </span>ssh webServer</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>     </span>cd kent/webBlat</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>     </span>cp webBlat
webBlat.cfg /usr/local/apache/cgi-bin</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>     </span>mkdir
/usr/local/apache/trash</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>     </span>chmod 777
/usr/local/apache/trash</p>

<p class=MsoNormal>assuming that you've put the executable and config file in
kent/webBlat.</p>

<p class=MsoNormal>The program will create some files in the trash
directory.<span style='mso-spacerun:yes'>  </span>It is good to </p>

<p class=MsoNormal>periodically clean out old files from this directory.<span
style='mso-spacerun:yes'>   </span>On Mac OS-X instead</p>

<p class=MsoNormal>you might do:</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>     </span>cp webBlat
webBlat.cfg /Library/WebServer/CGI-Executables</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>     </span>mkdir
/Library/WebServer/trash</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>     </span>chmod 777
/Library/WebServer/trash</p>

<p class=MsoNormal>Unless you are administering your own computer you will
likely need to ask your </p>

<p class=MsoNormal>local system administrators for help with this part of the
webBlat installation.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<h3><span style='font-size:16.0pt;mso-bidi-font-size:10.0pt'>File Formats<o:p></o:p></span></h3>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>.psl files<o:p></o:p></span></b></p>

<p class=MsoNormal>A .psl file describes a series of alignments in a dense
easily parsed text format.<span style='mso-spacerun:yes'>  </span>It begins
with a five line header which describes each field.<span
style='mso-spacerun:yes'>  </span>Following this is one line for each alignment
with a tab between each field.<span style='mso-spacerun:yes'>  </span>The
fields are describe below in<span style='mso-spacerun:yes'>  </span>a format
suitable for many relational databases.</p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>matches int unsigned ,<span
style='mso-spacerun:yes'>   </span><span style='mso-tab-count:1'>        </span>#
Number of bases that match that aren't repeats<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>misMatches int unsigned ,<span
style='mso-tab-count:1'>        </span># Number of bases that don't match<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>repMatches int unsigned , <span
style='mso-tab-count:1'>       </span># Number of bases that match but are part
of repeats<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>nCount int unsigned ,<span
style='mso-spacerun:yes'>       </span><span style='mso-tab-count:1'>     </span>#
Number of 'N' bases<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>qNumInsert int unsigned , <span
style='mso-tab-count:1'>       </span># Number of inserts in query<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>qBaseInsert int unsigned , <span
style='mso-tab-count:1'>      </span># Number of bases inserted in query<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>tNumInsert int unsigned
,<span style='mso-spacerun:yes'>  </span><span style='mso-tab-count:1'>      </span>#
Number of inserts in target<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>tBaseInsert int unsigned
,<span style='mso-spacerun:yes'>  </span><span style='mso-tab-count:1'>     </span>#
Number of bases inserted in target<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>strand char(2) ,<span
style='mso-spacerun:yes'>            </span><span style='mso-tab-count:1'>     </span>#
+ or - for query strand, optionally followed by + or – for target strand<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>qName varchar(255) ,<span
style='mso-spacerun:yes'>       </span><span style='mso-tab-count:1'>      </span>#
Query sequence name<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>qSize int unsigned ,<span
style='mso-spacerun:yes'>        </span><span style='mso-tab-count:1'>     </span>#
Query sequence size<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>qStart int unsigned ,<span
style='mso-spacerun:yes'>     </span><span style='mso-tab-count:1'>       </span>#
Alignment start position in query<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>qEnd int unsigned ,<span
style='mso-spacerun:yes'>         </span><span style='mso-tab-count:1'>     </span>#
Alignment end position in query<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>tName varchar(255) ,<span
style='mso-spacerun:yes'>       </span><span style='mso-tab-count:1'>      </span>#
Target sequence name<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>tSize int unsigned ,<span
style='mso-spacerun:yes'>        </span><span style='mso-tab-count:1'>     </span>#
Target sequence size<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>tStart int unsigned ,<span
style='mso-spacerun:yes'>       </span><span style='mso-tab-count:1'>     </span>#
Alignment start position in target<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>tEnd int unsigned ,<span
style='mso-spacerun:yes'>         </span><span style='mso-tab-count:1'>     </span>#
Alignment end position in target<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>blockCount int unsigned ,<span
style='mso-tab-count:1'>        </span># Number of blocks in alignment<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>blockSizes longblob ,<span
style='mso-spacerun:yes'>    </span><span style='mso-tab-count:1'>        </span>#
Size of each block in a comma separated list<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>qStarts longblob ,<span
style='mso-spacerun:yes'>  </span><span style='mso-tab-count:1'>             </span>#
Start of each block in query in a comma separated list<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>tStarts longblob ,<span
style='mso-spacerun:yes'>  </span><span style='mso-tab-count:1'>             </span>#
Start of each block in target in a comma separated list<o:p></o:p></span></p>

<p class=MsoNormal>Currently the program does not distinguish between matches
and repMatches.<span style='mso-spacerun:yes'>  </span>repMatches is always
zero.</p>

<p class=MsoNormal style='margin-right:-.25in'>There is a little gotcha in the
.psl format. It has to do with how coordinates are handled on the negative
strand. In the qStart/qEnd fields the coordinates are where it matches from the
point of view of the forward strand (even when the match is on the reverse
strand). However on the qStarts[] list, the coordinates are reversed. </p>

<p class=MsoNormal style='margin-right:-.25in'><o:p>&nbsp;</o:p></p>

<p class=MsoNormal style='margin-right:-.25in'>Here's an example of a 30-mer
that has 2 blocks that align on the minus strand and 2 blocks on the plus strand
(this sort of stuff happens in real life in response to assembly errors
sometimes). </p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>0<span
style='mso-spacerun:yes'>         </span>1<span
style='mso-spacerun:yes'>         </span>2<span
style='mso-spacerun:yes'>         </span>3 tens position in query<br>
0123456789012345678901234567890 ones position in query<br>
<span style='mso-spacerun:yes'>            </span>++++<span
style='mso-spacerun:yes'>          </span>+++++ plus strand alignment on query<br>
<span style='mso-spacerun:yes'>    </span>--------<span
style='mso-spacerun:yes'>    </span>----------<span
style='mso-spacerun:yes'>      </span>minus strand alignment on query <o:p></o:p></span></p>

<p class=MsoNormal>Plus strand:<br>
<span style='font-family:Courier'><span style='mso-spacerun:yes'>    
</span>qStart 12 qEnd 31 blockSizes 4,5 qStarts 12,26<br>
</span>Minus strand:<br>
<span style='font-family:Courier'><span style='mso-spacerun:yes'>    
</span>qStart 4 qEnd 26 blockSizes 10,8 qStarts 5,19 <o:p></o:p></span></p>

<p class=MsoNormal>Essentially the minus strand blockSizes and qStarts are what
you would get if you reverse complemented the query.However the qStart and qEnd
are non-reversed. To get from one to the other:<br>
<span style='font-family:Courier'><span style='mso-spacerun:yes'>    
</span>qStart = qSize - revQEnd<br>
<span style='mso-spacerun:yes'>     </span>qEnd = qSize - revQStart<br
style='mso-special-character:line-break'>
<![if !supportLineBreakNewLine]><br style='mso-special-character:line-break'>
<![endif]><o:p></o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>.2bit files<o:p></o:p></span></b></p>

<p class=MsoNormal>A .2bit file can store multiple DNA sequence (up to 4 gig
total) in a compact randomly accessible format.<span style='mso-spacerun:yes'> 
</span>The two<span style='mso-spacerun:yes'>  </span>bit files contain masking
information as well as the DNA itself.<span style='mso-spacerun:yes'>  
</span>The file begins with a 16 byte header containing the following fields:</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l1 level1 lfo1;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>1)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]>signature
– the number 0x1A412743 in the architecture of the machine that created the
file.</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l1 level1 lfo1;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>2)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]>version
– zero for now. Readers should abort if they see a version number higher than
0.</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l1 level1 lfo1;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>3)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]>sequenceCount
– the number of sequences in the file</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l1 level1 lfo1;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>4)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]>reserved
– always zero for now.</p>

<p class=MsoNormal>All fields are 32 bits unless noted. If the signature value
is not as given, the reader program should byte swap the signature and see if
the swapped version matches.<span style='mso-spacerun:yes'>  </span>If so all
multiple-byte entities in the file will need to be byte-swapped.<span
style='mso-spacerun:yes'>  </span>This enables these binary files to be used
unchanged on different architectures.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>The header is followed by a file index.<span
style='mso-spacerun:yes'>  </span>There is one entry in the index for each
sequence.<span style='mso-spacerun:yes'>   </span>Each index entry contains
three fields:</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l0 level1 lfo2;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>1)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]>nameSize
– a byte containing the length of the name field</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l0 level1 lfo2;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>2)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]>name
– this contains the sequence name itself, and is variable length depending on
nameSize.</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l0 level1 lfo2;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>3)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]>offset
– 32 bit offset of the sequence data relative to the start of the file</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>The index is followed by the sequence records.<span
style='mso-spacerun:yes'>  </span>These contain 9 fields:</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l4 level1 lfo3;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>1)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]>dnaSize
– number of bases of DNA in the sequence.</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l4 level1 lfo3;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>2)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]>nBlockCount
– the number of blocks of N’s in the file (representing unknown sequence).</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l4 level1 lfo3;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>3)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]>nBlockStarts
– a starting position for each block of N’s</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l4 level1 lfo3;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>4)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]>nBlockSizes
– the size of each block of N’s</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l4 level1 lfo3;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>5)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]>maskBlockCount
– the number of masked (lower case) blocks</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l4 level1 lfo3;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>6)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]>maskBlockStarts
– starting position for each masked block</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l4 level1 lfo3;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>7)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]>maskBlockSizes
– the size of each masked block</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l4 level1 lfo3;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>8)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]>packedDna
– the dna packed to two bits per base as so: 00 – T,<span
style='mso-spacerun:yes'>  </span>01 – C,<span style='mso-spacerun:yes'> 
</span>10 – A, 11 – G.<span style='mso-spacerun:yes'>  </span>The first base is
in the most significant 2 bits byte, and the last base in the least significant
2 bits, so that the sequence TCAG would be represented as 00011011.<span
style='mso-spacerun:yes'>  </span>The packedDna field will be padded with 0
bits as necessary<span style='mso-spacerun:yes'>  </span>so that it takes an even
multiple of 32 bit in the file, as this improves i/o performance on some machines.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>.nib files<o:p></o:p></span></b></p>

<p class=MsoNormal>A .nib file describes a DNA sequence packing two bases into
each byte. Each nib file contains only a single sequence.<span
style='mso-spacerun:yes'>  </span>A nib file begins with a 32 bit signature
which is 0x6BE93D3A in the architecture of the machine that created the file,
and possibly a byte-swapped version of the same number on another machine.<span
style='mso-spacerun:yes'>  </span>This is followed by a 32 bit number in the
same format which describes the number of bases in the file.<span
style='mso-spacerun:yes'>   </span>This is followed by the bases themselves
packed two bases to the byte.<span style='mso-spacerun:yes'>  </span>The first
base is packed in the high order 4 bits (nibble), the second base in the low
order four bits.<span style='mso-spacerun:yes'>  </span>In C code:</p>

<p class=MsoNormal style='margin-left:.5in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>byte
= (base1&lt;&lt;4) + base2<o:p></o:p></span></p>

<p class=MsoNormal>The numerical values for the bases are:</p>

<p class=MsoNormal style='margin-left:.5in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>0
– T,<span style='mso-spacerun:yes'>  </span>1 – C,<span
style='mso-spacerun:yes'>  </span>2 – A,<span style='mso-spacerun:yes'> 
</span>3 – G,<span style='mso-spacerun:yes'>  </span>4 – N (unknown)<o:p></o:p></span></p>

<p class=MsoNormal>The most significant bit in a nibble is set if the base is
masked. <span style='font-size:16.0pt;mso-bidi-font-size:10.0pt'><o:p></o:p></span></p>

<h3><span style='font-size:16.0pt;mso-bidi-font-size:10.0pt'>Limits<o:p></o:p></span></h3>

<p class=MsoNormal>The gfServer program requires approximately 1 byte for every
3 bases in the genome it is indexing in DNA mode, and 1.5 bytes for each
unmasked base in translated mode. The blat program requires approximately two
bytes for each base in the genome in DNA mode, and three bytes for each base in
translated mode. The other programs use relatively little memory.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

</div>

</body>

</html>
