<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns="http://www.w3.org/TR/REC-html40">

<head>
<meta http-equiv=Content-Type content="text/html; charset=windows-1252">
<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 9">
<meta name=Originator content="Microsoft Word 9">
<link rel=File-List href="./blatSpec_files/filelist.xml">
<title>BLAT Program Specifications</title>
<!--[if gte mso 9]><xml>
 <o:DocumentProperties>
  <o:Author>Jim Kent</o:Author>
  <o:LastAuthor>Donna Karolchik</o:LastAuthor>
  <o:Revision>2</o:Revision>
  <o:TotalTime>1</o:TotalTime>
  <o:LastPrinted>2003-06-11T23:46:00Z</o:LastPrinted>
  <o:Created>2004-03-11T17:21:00Z</o:Created>
  <o:LastSaved>2004-03-11T17:21:00Z</o:LastSaved>
  <o:Pages>5</o:Pages>
  <o:Words>2055</o:Words>
  <o:Characters>11719</o:Characters>
  <o:Company> </o:Company>
  <o:Lines>97</o:Lines>
  <o:Paragraphs>23</o:Paragraphs>
  <o:CharactersWithSpaces>14391</o:CharactersWithSpaces>
  <o:Version>9.4402</o:Version>
 </o:DocumentProperties>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <w:WordDocument>
  <w:DisplayHorizontalDrawingGridEvery>0</w:DisplayHorizontalDrawingGridEvery>
  <w:DisplayVerticalDrawingGridEvery>0</w:DisplayVerticalDrawingGridEvery>
  <w:UseMarginsForDrawingGridOrigin/>
  <w:Compatibility>
   <w:FootnoteLayoutLikeWW8/>
   <w:ShapeLayoutLikeWW8/>
   <w:AlignTablesRowByRow/>
   <w:ForgetLastTabAlignment/>
   <w:LayoutRawTableWidth/>
   <w:LayoutTableRowsApart/>
  </w:Compatibility>
 </w:WordDocument>
</xml><![endif]-->
<style>
<!--
 /* Font Definitions */
@font-face
	{font-family:Courier;
	panose-1:0 0 0 0 0 0 0 0 0 0;
	mso-font-charset:0;
	mso-generic-font-family:modern;
	mso-font-format:other;
	mso-font-pitch:fixed;
	mso-font-signature:3 0 0 0 1 0;}
@font-face
	{font-family:Tahoma;
	panose-1:2 11 6 4 3 5 4 4 2 4;
	mso-font-charset:0;
	mso-generic-font-family:swiss;
	mso-font-pitch:variable;
	mso-font-signature:553679495 -2147483648 8 0 66047 0;}
 /* Style Definitions */
p.MsoNormal, li.MsoNormal, div.MsoNormal
	{mso-style-parent:"";
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman";
	mso-fareast-font-family:"Times New Roman";}
h2
	{mso-style-next:Normal;
	margin-top:12.0pt;
	margin-right:0in;
	margin-bottom:3.0pt;
	margin-left:0in;
	mso-pagination:widow-orphan;
	page-break-after:avoid;
	mso-outline-level:2;
	font-size:12.0pt;
	mso-bidi-font-size:10.0pt;
	font-family:Arial;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-font-weight:normal;
	font-style:italic;
	mso-bidi-font-style:normal;}
h3
	{mso-style-next:Normal;
	margin-top:12.0pt;
	margin-right:0in;
	margin-bottom:3.0pt;
	margin-left:0in;
	mso-pagination:widow-orphan;
	page-break-after:avoid;
	mso-outline-level:3;
	font-size:12.0pt;
	mso-bidi-font-size:10.0pt;
	font-family:Arial;
	mso-bidi-font-family:"Times New Roman";
	font-weight:normal;}
p.MsoBodyTextIndent, li.MsoBodyTextIndent, div.MsoBodyTextIndent
	{margin-top:0in;
	margin-right:0in;
	margin-bottom:0in;
	margin-left:193.7pt;
	margin-bottom:.0001pt;
	text-indent:-193.7pt;
	mso-pagination:widow-orphan;
	tab-stops:193.5pt;
	font-size:10.0pt;
	font-family:"Courier New";
	mso-fareast-font-family:"Times New Roman";
	mso-bidi-font-family:"Times New Roman";}
p.MsoDocumentMap, li.MsoDocumentMap, div.MsoDocumentMap
	{margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	background:navy;
	font-size:10.0pt;
	font-family:Tahoma;
	mso-fareast-font-family:"Times New Roman";
	mso-bidi-font-family:"Times New Roman";}
@page Section1
	{size:8.5in 11.0in;
	margin:1.0in 1.25in .8in 1.25in;
	mso-header-margin:.5in;
	mso-footer-margin:.5in;
	mso-paper-source:0;}
div.Section1
	{page:Section1;}
-->
</style>
</head>

<body lang=EN-US style='tab-interval:.5in'>

<div class=Section1>

<h2>BLAT Program Specifications</h2>

<h3>General:</h3>

<p class=MsoNormal>Blat produces two major classes of alignments: at the DNA
level between two sequences that are of 95% or greater identity, but which may
include large inserts, and at the protein or translated DNA level between
sequences that are of 80% or greater identity and may also include large
inserts.<span style="mso-spacerun: yes">  </span>The output of BLAT is
flexible.<span style="mso-spacerun: yes">  </span>By default it is a simple
tab-delimited file which describes the alignment, but which does not include
the sequence of the alignment itself.<span style="mso-spacerun: yes"> 
</span>Optionally it can produce BLAST and WU-BLAST compatable output as well
as a number of other formats.</p>

<p class=MsoNormal><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></p>

<p class=MsoNormal>There are three main programs in the BLAT suite:<span
style="mso-spacerun: yes">  </span>a stand-alone program called ‘blat’.<span
style="mso-spacerun: yes">  </span>A server which maintains an index of a
genome in memory called ‘gfServer’, and a client which can query the index over
the network called ‘gfClient’.<span style="mso-spacerun: yes">  </span>Since it
takes some time (10 to 25 minutes) to index an entire genome, the
gfServer/gfClient model is best suited for situations where interactive users
wish to quickly locate a few sequences in the genome.<span style="mso-spacerun:
yes">  </span>The genome index does take memory – close to a gigabyte for a
nucleotide based index and over two gigabytes for a translated protein index.
The stand-alone program is most suited for large batch alignments, which can be
spread across many machines.<span style="mso-spacerun: yes">  </span>There are
two additional programs – faToNib which converts a .fa nucleotide file to a
denser format which is suitable for random access, and nibFrag which can
convert all or a portion of a .nib file back to .fa format.</p>

<h3>Command Line:</h3>

<p class=MsoNormal>The command line options of each of the programs is
described below. Similar summaries of usage are printed when a command is run
with no arguments.</p>

<p class=MsoNormal><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>blat<o:p></o:p></span></b></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>blat
- Standalone BLAT sequence search command line tool<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>blat database query [-ooc=11.ooc]
output.psl<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>where:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>database is either a .fa file, a .nib file,
or a list of .fa or .nib<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>files, query is similarly a .fa, .nib, or
list of .fa or .nib files<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-ooc=11.ooc tells the program to load
over-occurring 11-mers from<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>and external file.<span
style="mso-spacerun: yes">  </span>This will increase the speed<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>by a factor of 40 in many
cases, but is not required<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>output.psl is where to put the output.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>options:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-t=type<span style="mso-spacerun: yes">    
</span>Database type.<span style="mso-spacerun: yes">  </span>Type is one of:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                 </span>dna - DNA sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">             </span><span style="mso-spacerun:
yes">    </span>prot - protein sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                 </span>dnax - DNA sequence
translated in six frames to protein<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>The default is dna<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-q=type<span style="mso-spacerun: yes">    
</span>Query type.<span style="mso-spacerun: yes">  </span>Type is one of:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                 </span></span><span lang=ES
style='font-size:9.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman";mso-ansi-language:ES'>dna - DNA sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span lang=ES style='font-size:
9.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:
"Times New Roman";mso-ansi-language:ES'><span style="mso-spacerun:
yes">                 </span>rna - RNA sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span lang=ES style='font-size:
9.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:
"Times New Roman";mso-ansi-language:ES'><span style="mso-spacerun: yes">    
</span><span style="mso-spacerun: yes">            </span></span><span
style='font-size:9.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'>prot - protein sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                 </span>dnax - DNA sequence
translated in six frames to protein<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                 </span>rnax - DNA sequence
translated in three frames to protein<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>The default is dna<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-prot<span style="mso-spacerun: yes">      
</span>Synonymous with -d=prot -q=prot<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-ooc=N.ooc<span style="mso-spacerun: yes"> 
</span>Use overused tile file N.ooc.<span style="mso-spacerun: yes">  </span>N
should correspond to<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>the tileSize<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-tileSize=N sets the size of match that
triggers an alignment.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>Usually between 8 and 12<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>Default is 11 for DNA and 5 for
protein.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-oneOff=N<span style="mso-spacerun: yes">  
</span>If set to 1 this allows one mismatch in tile and still<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>triggers an alignments.<span
style="mso-spacerun: yes">  </span>Default is 0.<o:p></o:p></span></p>

<span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-fareast-font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman";
mso-ansi-language:EN-US;mso-fareast-language:EN-US;mso-bidi-language:AR-SA'><br
clear=all style='page-break-before:always'>
</span>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-minMatch=N sets the number of tile
matches.<span style="mso-spacerun: yes">  </span>Usually set from 2 to 4<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>Default is 2 for nucleotide, 1 for
protein.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-minScore=N sets minimum score.<span
style="mso-spacerun: yes">  </span>This is twice the matches minus the<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>mismatches minus some sort of
gap penalty.<span style="mso-spacerun: yes">  </span>Default is 30<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-minIdentity=N Sets minimum sequence
identity (in percent).<span style="mso-spacerun: yes">  </span>Default is<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>90 for nucleotide searches, 25
for protein or translated<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>protein searches.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-maxGap=N<span style="mso-spacerun: yes">  
</span>sets the size of maximum gap between tiles in a clump.<span
style="mso-spacerun: yes">  </span>Usually<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>set from 0 to 3.<span
style="mso-spacerun: yes">  </span>Default is 2. Only relevent for minMatch
&gt; 1.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-noHead<span style="mso-spacerun: yes">    
</span>suppress .psl header (so it's just a tab-separated file)<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-makeOoc=N.ooc Make overused tile file<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-repMatch=N sets the number of repetitions
of a tile allowed before<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>it is marked as overused.<span
style="mso-spacerun: yes">  </span>Typically this is 256 for tileSize<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">          </span><span style="mso-spacerun:
yes">     </span>12, 1024 for tile size 11, 4096 for tile size 10.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>Default is 1024.<span
style="mso-spacerun: yes">  </span>Typically only comes into play with makeOoc<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-mask=type<span style="mso-spacerun: yes"> 
</span>Mask out repeats.<span style="mso-spacerun: yes">  </span>Alignments
won't be started in masked region<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>but may extend through it in nucleotide
searches.<span style="mso-spacerun: yes">  </span>Masked areas<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>are ignored entirely in protein
or translated searches. Types are<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                 </span>lower - mask out lower cased
sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                 </span>upper - mask out upper cased
sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                 </span>out<span style="mso-spacerun:
yes">   </span>- mask according to database.out RepeatMasker .out file<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                 </span>file.out - mask database
according to RepeatMasker file.out<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-qMask=type Mask out repeats in query
sequence.<span style="mso-spacerun: yes">  </span>Similar to -mask above but<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>for query rather than target
sequence.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-minRepDivergence=NN - minimum percent
divergence of repeats to allow<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>them to be unmasked.<span
style="mso-spacerun: yes">  </span>Default is 15.<span style="mso-spacerun:
yes">  </span>Only relevant for<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>masking using RepeatMasker .out
files.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-dots=N<span style="mso-spacerun: yes">    
</span>Output dot every N sequences to show program's progress<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-trimT<span style="mso-spacerun: yes">     
</span>Trim leading poly-T<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-noTrimA<span style="mso-spacerun: yes">   
</span>Don't trim trailing poly-A<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-trimHardA<span style="mso-spacerun: yes"> 
</span>Remove poly-A tail from qSize as well as alignments in psl output<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-out=type<span style="mso-spacerun: yes">  
</span>Controls output file format.<span style="mso-spacerun: yes"> 
</span>Type is one of:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                   </span>psl - Default.<span
style="mso-spacerun: yes">  </span>Tab separated format without actual sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                   </span>pslx - Tab separated format
with sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                   </span>axt - blastz-associated axt
format<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                   </span>maf - multiz-associated maf
format<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                   </span>wublast - similar to
wublast format<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                   </span>blast - similar to NCBI
blast format<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-fine<span style="mso-spacerun: yes">      
</span>For high quality mRNAs look harder for small initial and<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>terminal exons.<span
style="mso-spacerun: yes">  </span>Not recommended for ESTs<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>Here are some blat settings for
common usage scenarios:</p>

<p class=MsoNormal style='margin-right:-.25in'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></p>

<p class=MsoNormal style='margin-right:-.25in'>1) Mapping ESTs to the genome
within the same species</p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">    </span>-ooc=11.ooc<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>2) Mapping full length mRNAs to
the genome in the same species</p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">    </span>-ooc=11.ooc -fine -q=rna<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>3) Mapping ESTs to the genome
across species</p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">    </span>-q=dnax -t=dnax<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>4) Mapping mRNA to the genome
across species</p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">    </span>-q=rnax -t=dnax<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>5) Mapping proteins to the
genome</p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">    </span>-q=prot -t=dnax<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>6) Mapping DNA to DNA in the
same species</p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">    </span>-ooc=11.ooc -fastMap<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>7) Mapping DNA from one species
to another species</p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">    </span>-q=dnax -t=dnax<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">    </span>When mapping DNA from one species to
another the<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">    </span>query side of the alignment should be cut
up into chunks<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">    </span>of 25kb or less for best performance.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p>

<b style='mso-bidi-font-weight:normal'><span style='font-size:12.0pt;
mso-bidi-font-size:10.0pt;font-family:"Times New Roman";mso-fareast-font-family:
"Times New Roman";mso-ansi-language:EN-US;mso-fareast-language:EN-US;
mso-bidi-language:AR-SA'><br clear=all style='page-break-before:always'>
</span></b>

<p class=MsoNormal style='margin-right:-.25in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>gfServer<o:p></o:p></span></b></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>gfServer -
Make a server to quickly find where DNA occurs in genome.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To set up a
server:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>gfServer start host port file(s).nib<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To remove a
server:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>gfServer stop host port<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To query a
server with DNA sequence:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>gfServer query host port probe.fa<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To query a
server with protein sequence:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>gfServer protQuery host port probe.fa<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To query a
server with translated dna sequence:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>gfServer transQuery host port probe.fa<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To process
one probe fa file against a .nib format genome (not starting server):<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>gfServer direct probe.fa file(s).nib<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To figure out
usage level<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>gfServer status host port<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To get input
file list<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>gfServer files host port<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>Options:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-tileSize=N size of n-mers to index.<span
style="mso-spacerun: yes">  </span>Default is 11 for nucleotides, 4 for<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>proteins (or translated
nucleotides).<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-minMatch=N Number of n-mer matches that
trigger detailed alignment<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">            </span><span style="mso-spacerun:
yes">   </span>Default is 2 for nucleotides, 3 for protiens.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-maxGap=N<span style="mso-spacerun: yes">  
</span>Number of insertions or deletions allowed between n-mers.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>Default is 2 for nucleotides, 0
for protiens.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-trans<span style="mso-spacerun: yes">     
</span>Translate database to protein in 6 frames.<span style="mso-spacerun:
yes">  </span>Note: it is best<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>to run this on RepeatMasked
data in this case.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-log=logFile keep a log file that records
server requests.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-seqLog<span style="mso-spacerun: yes">   
</span>Include sequences in log file<o:p></o:p></span></p>

<p class=MsoNormal><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>gfClient<o:p></o:p></span></b></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>gfClient - A
client for the genomic finding program <o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>gfClient host port nibDir in.fa out.psl<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>where<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>host is the name of the machine running the
gfServer<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>port is the same as you started the
gfServer with<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>nibDir is the path of the nib files
relative to the current dir<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">       </span>(note these are needed by the client as
well as the server)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>in.fa a fasta format file.<span
style="mso-spacerun: yes">  </span>May contain multiple records<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>out.psl where to put the output<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>options:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-t=type<span style="mso-spacerun: yes">    
</span>Database type.<span style="mso-spacerun: yes">  </span>Type is one of:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                 </span>dna - DNA sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                 </span>prot - protein sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">       </span><span style="mso-spacerun:
yes">          </span>dnax - DNA sequence translated in six frames to protein<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">               </span>The default is dna<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-q=type<span style="mso-spacerun: yes">    
</span>Query type.<span style="mso-spacerun: yes">  </span>Type is one of:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                 </span>dna - DNA sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                 </span>rna - RNA sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                 </span>prot - protein sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                 </span>dnax - DNA sequence
translated in six frames to protein<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                 </span>rnax - DNA sequence
translated in three frames to protein<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-dots=N<span style="mso-spacerun: yes">  
</span>Output a dot every N query sequences<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-nohead<span style="mso-spacerun: yes">  
</span>Suppresses psl five line header<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-out=type<span style="mso-spacerun: yes">  
</span>Controls output file format.<span style="mso-spacerun: yes"> 
</span>Type is one of:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                   </span>psl - Default.<span
style="mso-spacerun: yes">  </span>Tab separated format without actual sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                   </span>pslx - Tab separated format
with sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                   </span>axt - blastz-associated axt
format<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">            </span><span style="mso-spacerun:
yes">       </span>maf - multiz-associated maf format<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                   </span>wublast - similar to
wublast format<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">                   </span>blast - similar to NCBI
blast format<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'>faToNib<o:p></o:p></span></b></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>faToNib -
Convert from .fa to .nib format<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>faToNib in.fa out.nib<o:p></o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>nibFrag<o:p></o:p></span></b></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>nibFrag - Extract
part of a nib file as .fa<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>nibFrag file.nib start end strand out.fa<o:p></o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>pslPretty<o:p></o:p></span></b></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>pslPretty -
Convert PSL to human readable output<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>pslPretty in.psl target.lst query.lst
pretty.out<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>options:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-axt - save in Scott Schwartz's axt format<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-dot=N Put out a dot every N records<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">   </span>-long - Don't abbreviate long inserts<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>It's a really good idea if the
psl file is sorted by target if it contains multiple targets.<span
style="mso-spacerun: yes">  </span>Otherwise this will be very very slow.<span
style="mso-spacerun: yes">   </span>The target and query lists can either be
fasta files, nib files, or a list of fasta and/or nib files one per line.<span
style="mso-spacerun: yes">  </span>Currently this only handles nucleotide based
psl files.</p>

<p class=MsoNormal style='margin-right:-.25in'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></p>

<h3>File Formats</h3>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>.nib files<o:p></o:p></span></b></p>

<p class=MsoNormal>A .nib file describes a DNA sequence packing two bases into
each byte.<span style="mso-spacerun: yes">  </span>A nib file begins with a 32
bit signature which is 0x6BE93D3A in the archetecture of the machine that
created the file, and possibly a byte-swapped version of the same number on
another machine.<span style="mso-spacerun: yes">  </span>This is followed by a
32 bit number in the same format which describes the number of bases in the
file.<span style="mso-spacerun: yes">   </span>This is followed by the bases
themselves packed two bases to the byte.<span style="mso-spacerun: yes"> 
</span>The first base is packed in the high order 4 bits, the second base in
the low order four bits.<span style="mso-spacerun: yes">  </span>In C code:</p>

<p class=MsoNormal style='margin-left:.5in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>byte
= (base1&lt;&lt;4) + base2<o:p></o:p></span></p>

<p class=MsoNormal>The numerical values for the bases are:</p>

<p class=MsoNormal style='margin-left:.5in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>0
– T,<span style="mso-spacerun: yes">  </span>1 – C,<span style="mso-spacerun:
yes">  </span>2 – A,<span style="mso-spacerun: yes">  </span>3 – G,<span
style="mso-spacerun: yes">  </span>4 – N (unknown), 5-15 – unused<o:p></o:p></span></p>

<p class=MsoNormal><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>.psl files<o:p></o:p></span></b></p>

<p class=MsoNormal>A .psl file describes a series of alignments in a dense
easily parsed text format.<span style="mso-spacerun: yes">  </span>It begins
with a five line header which describes each field.<span style="mso-spacerun:
yes">  </span>Following this is one line for each alignment with a tab between
each field.<span style="mso-spacerun: yes">  </span>The fields are describe
below in<span style="mso-spacerun: yes">  </span>a format suitable for many
relational databases.</p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style="mso-spacerun: yes">    </span>matches int unsigned ,<span
style="mso-spacerun: yes">   </span><span style='mso-tab-count:1'>    </span>#
Number of bases that match that aren't repeats<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style="mso-spacerun: yes">    </span>misMatches int unsigned ,<span
style='mso-tab-count:1'>    </span># Number of bases that don't match<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style="mso-spacerun: yes">    </span>repMatches int unsigned , <span
style='mso-tab-count:1'>    </span># Number of bases that match but are part of
repeats<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style="mso-spacerun: yes">    </span>nCount int unsigned ,<span
style="mso-spacerun: yes">       </span><span style='mso-tab-count:1'>    </span>#
Number of 'N' bases<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style="mso-spacerun: yes">    </span>qNumInsert int unsigned , <span
style='mso-tab-count:1'>    </span># Number of inserts in query<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style="mso-spacerun: yes">    </span>qBaseInsert int unsigned , <span
style='mso-tab-count:1'>    </span># Number of bases inserted in query<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style="mso-spacerun: yes">    </span>tNumInsert int unsigned
,<span style="mso-spacerun: yes">  </span><span style='mso-tab-count:1'>    </span>#
Number of inserts in target<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style="mso-spacerun: yes">    </span>tBaseInsert int unsigned
,<span style="mso-spacerun: yes">  </span><span style='mso-tab-count:1'>    </span>#
Number of bases inserted in target<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style="mso-spacerun: yes">    </span>strand char(2) ,<span
style="mso-spacerun: yes">            </span><span style='mso-tab-count:1'>    </span>#
+ or - for query strand, optionally followed by + or – for target strand<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style="mso-spacerun: yes">    </span>qName varchar(255) ,<span
style="mso-spacerun: yes">       </span><span style='mso-tab-count:1'>    </span>#
Query sequence name<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style="mso-spacerun: yes">    </span>qSize int unsigned ,<span
style="mso-spacerun: yes">        </span><span style='mso-tab-count:1'>    </span>#
Query sequence size<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style="mso-spacerun: yes">    </span>qStart int unsigned ,<span
style="mso-spacerun: yes">     </span><span style='mso-tab-count:1'>    </span>#
Alignment start position in query<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style="mso-spacerun: yes">    </span>qEnd int unsigned ,<span
style="mso-spacerun: yes">         </span><span style='mso-tab-count:1'>    </span>#
Alignment end position in query<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style="mso-spacerun: yes">    </span>tName varchar(255) ,<span
style="mso-spacerun: yes">       </span><span style='mso-tab-count:1'>    </span>#
Target sequence name<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style="mso-spacerun: yes">    </span>tSize int unsigned ,<span
style="mso-spacerun: yes">        </span><span style='mso-tab-count:1'>    </span>#
Target sequence size<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style="mso-spacerun: yes">    </span>tStart int unsigned ,<span
style="mso-spacerun: yes">       </span><span style='mso-tab-count:1'>    </span>#
Alignment start position in target<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style="mso-spacerun: yes">    </span>tEnd int unsigned ,<span
style="mso-spacerun: yes">         </span><span style='mso-tab-count:1'>    </span>#
Alignment end position in target<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style="mso-spacerun: yes">    </span>blockCount int unsigned ,<span
style='mso-tab-count:1'>    </span># Number of blocks in alignment<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style="mso-spacerun: yes">    </span>blockSizes longblob ,<span
style="mso-spacerun: yes">    </span><span style='mso-tab-count:1'>    </span>#
Size of each block in a comma separated list<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style="mso-spacerun: yes">    </span>qStarts longblob ,<span
style="mso-spacerun: yes">  </span><span style='mso-tab-count:1'>    </span>#
Start of each block in query in a comma separated list<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style="mso-spacerun: yes">    </span>tStarts longblob ,<span
style="mso-spacerun: yes">  </span><span style='mso-tab-count:1'>    </span>#
Start of each block in target in a comma separated list<o:p></o:p></span></p>

<p class=MsoNormal>Currently the program does not distinguish between matches
and repMatches.<span style="mso-spacerun: yes">  </span>repMatches is always
zero.</p>

<p class=MsoNormal style='margin-right:-.25in'>There is a little gotcha in the
.psl format. It has to do with how coordinates are handled on the negative
strand. In the qStart/qEnd fields the coordinates are where it matches from the
point of view of the forward strand (even when the match is on the reverse
strand). However on the qStarts[] list, the coordinates are reversed. </p>

<p class=MsoNormal style='margin-right:-.25in'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></p>

<p class=MsoNormal style='margin-right:-.25in'>Here's an example of a 30-mer
that has 2 blocks that align on the minus strand and 2 blocks on the plus
strand (this sort of stuff happens in real life in response to assembly errors
sometimes). </p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>0<span
style="mso-spacerun: yes">         </span>1<span style="mso-spacerun:
yes">         </span>2<span style="mso-spacerun: yes">         </span>3 tens
position in query<br>
0123456789012345678901234567890 ones position in query<br>
<span style="mso-spacerun: yes">            </span>++++<span
style="mso-spacerun: yes">          </span>+++++ plus strand alignment on query<br>
<span style="mso-spacerun: yes">    </span>--------<span style="mso-spacerun:
yes">    </span>----------<span style="mso-spacerun: yes">      </span>minus
strand alignment on query <o:p></o:p></span></p>

<p class=MsoNormal>Plus strand:<br>
<span style='font-family:Courier'><span style="mso-spacerun: yes">    
</span>qStart 12 qEnd 31 blockSizes 4,5 qStarts 12,26<br>
</span>Minus strand:<br>
<span style='font-family:Courier'><span style="mso-spacerun: yes">    
</span>qStart 4 qEnd 26 blockSizes 10,8 qStarts 5,19 <o:p></o:p></span></p>

<p class=MsoNormal>Essentially the minus strand blockSizes and qStarts are what
you would get if you reverse complemented the query.However the qStart and qEnd
are non-reversed. To get from one to the other:<br>
<span style='font-family:Courier'><span style="mso-spacerun: yes">    
</span>qStart = qSize - revQEnd<br>
<span style="mso-spacerun: yes">     </span>qEnd = qSize - revQStart<br
style='mso-special-character:line-break'>
<![if !supportLineBreakNewLine]><br style='mso-special-character:line-break'>
<![endif]></span></p>

<h3>Limits</h3>

<p class=MsoNormal>The gfServer program requires approximately 1 byte for every
3 bases in the genome it is indexing in DNA mode, and 1.5 bytes for each
unmasked base in translated mode. The blat program requires approximately two
bytes for each base in the genome in DNA mode, and three bytes for each base in
translated mode. The other programs use relatively little memory.</p>

<p class=MsoNormal><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></p>

<p class=MsoNormal><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></p>

</div>

</body>

</html>
