<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: Custom Tracks</TITLE>
	<LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="/images/title.jpg"></TH></TR>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
	<table bgcolor="000000" cellpadding="1" cellspacing="1" width="100%" height="27"><tr bgcolor="2636D1"><td valign="middle">
	<table BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24">
			<TR><TD VALIGN="middle"><font color="#89A1DE">&nbsp;&nbsp;
		<A HREF="/index.html" class="topbar">
			Home</A> &nbsp; - &nbsp;
		<A HREF="/cgi-bin/hgGateway" class="topbar">
			Genome Browser</A> &nbsp; - &nbsp; 
		<A HREF="/cgi-bin/hgNear" class="topbar">
			Gene Sorter</A> &nbsp; - &nbsp; 
		<A HREF="/cgi-bin/hgBlat?command=start" class="topbar">
			Blat </A> &nbsp; - &nbsp; 
		<A HREF="/cgi-bin/hgTables?command=start" class="topbar">
			Table Browser</A> &nbsp; - &nbsp; 
		<A HREF="/FAQ/" class="topbar">
			FAQ</A> &nbsp; - &nbsp; 
		<A HREF="/goldenPath/help/hgTracksHelp.html" class="topbar">
			Help</A> &nbsp;</font></TD>
	</TR></TABLE>
	</TD></TR></TABLE>
</TD></TR>

<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=2 CELLPADDING=10>	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><b>&nbsp;  Displaying Your Own Annotations in the Genome Browser</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>

<P>The Genome Browser provides dozens of aligned annotation tracks 
that have been computed at UCSC or have been provided by
outside collaborators. In addition to these standard tracks, it is 
also possible for users to upload their own annotations for
temporary display in the Genome Browser. These custom annotation 
tracks are viewable only on the machine from which they were uploaded and 
are kept only for 8 hours after the last time they were accessed. Users 
can optionally make custom annotations viewable by others as well.

<P>Genome Browser annotation tracks are based on files in line-oriented format. 
Each line in the file defines a display characteristic for the annotation 
track or defines a data item within the track. 
Annotation files contain 3 types of lines: browser lines, track lines, 
and data lines. Empty lines and lines starting with # in the 
annotation file are ignored. 

<P>To construct an annotation file and display it in the Genome 
Browser, follow these steps:

<P><B>Step 1. Format the data set</B><BR>
Formulate your data set as a tab-separated file using one of the formats 
supported by the Genome Browser. Annotation data can be in 
standard <A HREF="#GFF">GFF</A> format or in a format designed specifically 
for the Human Genome Project or UCSC Genome Browser, including 
<A HREF="#GTF">GTF</A>, <A HREF="#PSL">PSL</A>, <A HREF="#BED">BED</A>, 
<A HREF="wiggle.html">WIG</A>, or <A HREF="maf.html">MAF</A>. 
GFF and GTF files <I>must</I> be tab-delimited rather than
space-delimited to display correctly. You may include more than one data
set in your annotation file. However, all of the data lines for a given 
annotation track must be in the same format.

<P><B>Step 2. Define the Genome Browser display characteristics</B><BR>
Add one or more optional <A HREF="#BROWSER">browser lines</A> to the 
beginning of your formatted data file to configure the overall 
display of the Genome Browser when it initially displays your annotation 
data. Browser lines allow you to configure such things 
as the genome position that the Genome Browser will initially open to, 
the width of the display, and the configuration of the other
annotation tracks that are shown (or hidden) in the initial display. 
NOTE: If the browser position is not explicitly set in the annotation 
file, the initial display will default to the position setting most 
recently used by the user, which may not be an
appropriate position for viewing the annotation track.

<P><B>Step 3. Define the annotation track display characteristics</B><BR>
Following the browser lines - and immediately preceding the formatted 
data - add a <A HREF="#TRACK">track line</A> to 
define the display attributes for your annotation data set. Track lines 
allow you to define annotation track characteristics 
such as the name, description, colors, initial display mode, use score,
etc. If you have included more than one data set in your annotation
file, insert a track line at the beginning of each new set of data.

<P><B><I>Example:</I></B><BR>
Here is an example of an annotation file that defines 2 separate
annotation tracks in BED format. The first track displays blue one-base
tick marks every 10000 bases at the beginning of chr 22. 
The second track displays red 100-base features alternating with 
blank space in the same region of chr 22.
<PRE><TT>browser position chr22:1-20001
track name=spacer description="My blue ticks" color=0,0,255
chr22   0        1
chr22   10000 10001
chr22   20000 20001
track name=even description="My red 100 bases tracks" color=255,0,0
chr22   0       100 first
chr22   200     300 second
chr22   400     500 third</TT></PRE>

<P><B><I>Example:</I></B><BR>
This example shows an annotation file containing one data set in BED
format. The track displays paired features with a thick end and thin
end, and hatch marks indicating the direction of transcription. The track
labels display in green (0,128,0), and the gray level of the each feature
reflects the score value of that line. NOTE: The track name line in this example
has been split over 2 lines for documentation purposes. 
If you paste this example into the Genome Browser, you must remove the line 
break to display the track successfully. 
Click <A HREF ="pairedReads.txt">
here</A> for a copy of this example that can be pasted into the browser without editing. 
<PRE><TT>browser position chr22:1000-10000
browser hide all
track name=pairedReads description="Clone Paired Reads" visibility=2
color=0,128,0 useScore=1
chr22 1000 5000 cloneA 960 + 1000 5000 0 2 567,488, 0,3512
chr22 2000 6000 cloneB 200 - 2000 6000 0 2 433,399, 0,3601
</TT></PRE>

<P><B>Step 4. View your annotation track in the Genome Browser</B><BR>
To view your annotation data in the Genome Browser, open the 
Genome Browser home page <A HREF="http://genome.ucsc.edu/"
TARGET=_blank>(http://genome.ucsc.edu/)</a> and click the Genome 
Browser link in the top menu bar. On the Genome Browser Gateway page that
displays, select the genome and assembly on which your
annotation data is based, then click the Add Your Own Tracks button. Upload 
your annotation file by entering the name of your file in the Annotation File 
box or by pasting the contents of your file into the large edit
box. Click the Submit button to
display the Genome Browser track window with your annotation. If you
encounter difficulties in displaying your annotation, read the section 
<A HREF="#TROUBLE">Troubleshooting Annotation Display Problems</A>.</P> 
<P>
To upload a custom annotation track from another machine or web site, paste the
URL of the track into the large edit box. Custom tracks can be displayed in
conjunction with ordinary BLAT tracks.</P>

<P><B>Step 5. (Optional) Add details pages for individual track features</B><BR>
After you've constructed your track and have successfully displayed it in the
Genome Browser, you may wish to customize the details pages for individual track
features.  The Genome Browser automatically creates a default details page for each
feature in the track containing the feature's name, position information, and a
link to the corresponding DNA sequence. To view the
details page for a feature in your custom annotation track (in full, pack, or 
squish display
mode), click on the item's label in the annotation track window. 
<P>
You can add a link from a details page to an external web page containing
additional information about the feature by using the track line <I>url</I> attribute. 
In the annotation file, set the <I>url</I> attribute in the track line to point
to a publicly available page on a web server. The <I>url</I> attribute 
substitutes each occurrence of '$$' in the URL string with the name defined by the 
<I>name</I> attribute. You can take advantage of this feature to provide 
individualized information for each feature in your track by creating HTML anchors 
that correspond to the feature names in your web page.

<P><B><I>Example:</I></B><BR>
Here is an example of a file in which the <I>url</I> attribute has been set to
point to the file <A HREF="http://genome.ucsc.edu/goldenPath/help/clones.html"
TARGET=_blank>http://genome.ucsc.edu/goldenPath/help/clones.html</A>. The '#$$'
appended to the end of the file name in the example points to the HTML NAME tag 
within the file that matches the name of the feature (cloneA, cloneB, etc.). 
NOTE: The track line in this example has been split over 2 lines for 
documentation purposes. If you paste this example into
the browser, you must remove the line break to display the track successfully.
Click <A HREF ="clones.txt">
here</A> for a copy of this example that can be pasted into browser without editing. 
<BR>
<PRE><TT>browser position chr22:1000-10000
browser hide all
track name=clones description="Clones" visibility=2
color=0,128,0 useScore=1 
url="http://genome.ucsc.edu/goldenPath/help/clones.html#$$"
chr22 1000 5000 cloneA 960 
chr22 2000 6000 cloneB 200 
chr22 5000 9000 cloneC 700 
chr22 6000 10000 cloneD 600
chr22 11000 15000 cloneE 300
chr22 13000 17000 cloneF 100 
</TT></PRE>
<P>
<P><B>Step 6. (Optional) Share your annotation track with others</B><BR>
The previous steps showed you how to upload annotation data for your own use 
on your own machine. However, many users would like to 
share their annotation data with members of their research group on 
different machines or with colleagues at other sites. To learn how 
to make your Genome Browser annotation track viewable by others, read 
the section <A HREF="#SHARE">Sharing Your Annotation Track with
Others</A>.</P>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><a name="BROWSER"></a><b>&nbsp;  Browser Lines   </b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>Browser lines configure the overall display of the Genome Browser
window when your annotation file is uploaded. Each line defines one display 
attribute. Browser lines consist of the format:
<P>
<tt><ALIGN="center">browser attribute_name attribute_value(s)</tt>

<P>For example, if the browser line <tt>browser
position chr22:1-20000</tt> is included in the annotation file, the Genome
Browser window will initially display the first 20000 bases of chr 22. 

<P>The following browser line attribute name/value options are available: 
<UL>
   <LI><B>position <<I>position</I>></B> - Determines the part of the genome
   that the Genome Browser will initially open to, in chromosome:start-end format.  
   <LI><B>pix <<I>width</I>></B> - Sets the Genome Browser window to the 
	specified width in pixels.
   <LI><B>hide all</B> - Hides all annotation tracks except for custom ones.
   <LI><B>hide <<I>track_name(s)</I>></B> - Hides the listed tracks.
   Tracks must be referenced by their <A HREF="../symNames.html". 
   TARGET=_blank>symbolic names</A>. Multiple track names should be space-separated. 
   <LI><B>dense all</B> - Displays all tracks in dense mode.
   <LI><B>dense <<I>track_name(s)</I>></B> - Displays the specified tracks in
   dense mode. Symbolic names must be used. Multiple track names should be
   space-separated. 
   <LI><B>pack all</B> - Displays all tracks in pack mode.
   <LI><B>pack <<I>track_name(s)</I>></B> - Displays the specified tracks in
   pack mode. Symbolic names must be used. Multiple track names should be 
   space-separated. 
   <LI><B>squish all</B> - Displays all tracks in squish mode.
   <LI><B>squish <<I>track_name(s)</I>></B> - Displays the specified tracks in
   squish mode. Symbolic names must be used. Multiple track names should be 
   space-separated. 
   <LI><B>full all</B> - Displays all tracks in full mode.
   <LI><B>full <<I>track_name(s)</I>></B> - Displays the specified tracks in
   full mode. Symbolic names must be used. Multiple track names should be 
   space-separated. 
</UL>
<P>
Note that the Genome Browser will open to the range defined in the
Gateway page <I>position</I> box or the position saved as the default unless the browser
line position attribute is defined in the annotation file. Although this
attribute is optional, it's recommended that you set this value in your
annotation file to ensure that the track will appear in the display range when
it is uploaded into the Genome Browser.
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><a name="TRACK"></a><b>&nbsp; Track Lines</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
	
<P>Track lines define the display attributes for all lines in an
annotation data set. If more than one data set is included in the
annotation file, each group of data must be preceded by a track line
that describes the display characteristics for that set of data.
A track line begins with the word <tt>track</tt>, followed by one or
more <tt>attribute=value</tt> pairs.  Unlike browser lines - in which each
attribute is defined on a separate line - all of the track attributes for a
given set of data are listed on one line <I>with no line
breaks</I>. The inadvertent insertion of a line break into a track line
will generate an error when you attempt to upload the annotation track
into the Genome Browser.

<P>The following track line attribute=value pairs are defined in the
Genome Browser:</P>

<UL>
	<LI><B>name=<<I>track_label</I>></B> - Defines the track label
	that will be displayed to the left of the track in the Genome
	Browser window, and also the label of the track control at the
	bottom of the screen. The name can consist of up to 15
	characters, and must be enclosed in quotes if the text contains
	spaces. The default value is <I>User Track</I>.
	<LI><B>description=<<I>center_label</I>></B> - Defines the center label of the track
	in the Genome Browser window. The description can consist of up
	to 60 characters, and must be enclosed in quotes if the text contains
	spaces. The default
	value is <I>User Supplied Track</I>.
	<LI><B>visibility=<<I>display_mode</I>></B> - Defines the
	initial display mode of the annotation track. Values for
	<I>display_mode</I> include: 0 - hide, 1 - dense, 2 - full, 3 - pack, 
	and 4 - squish. The default is 1.
	<LI><B>color=<<I>RRR,GGG,BBB</I>></B> - Defines the main color
	for the annotation track.  The track color consists of three comma-separated
	RGB values from 0-255.  The default value is 0,0,0 (black).
	<LI><B>altColor=<<I>RRR,GGG,BBB</I>></B> - Defines the
	secondary color for the track.  The alternate color consists of
	three comma-separated RGB values from 0-255. The default is a
	lighter shade of whatever the <I>color</I> attribute is set to.
	<LI><B>useScore=<<I>use_score</I>></B> - If this attribute is
	present and is set to 1, the <I>score</I> field in each of the track's data 
	lines will be used to determine the level of shading in which 
	the data is displayed. The track will display in shades of gray unless
	the <I>color</I> attribute is set to 100,50,0 (shades of brown) or
	0,60,120 (shades of blue). The default setting for <I>useScore</I> is 0.
	<LI><B>priority=<<I>priority</I>></B> - Defines the display
	position of the track relative to other tracks in the Genome
	Browser window. 
	<LI><B>offset=<<I>offset</I>></B> - Defines a number to be added
	to all coordinates in the annotation track. The default is 0.
	<LI><B>url=<<I>external_url</I>></B> - Defines a URL for an
	external link associated with this track. This URL will be used
	in the details page for the track. Any '$$' in this string this will be substituted
	with the item name. There is no default for this attribute.
</UL>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>
	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><a name="BED"></a><b>&nbsp; BED Lines </b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>

<P>BED format provides a flexible way to define the data lines that
are displayed in an annotation track. BED lines have three required fields and 
nine additional optional fields. The number of fields per line must be consistent
throughout any single set of data in an annotation track.  

<P>The first three required BED fields are:</P>
<OL>
	<LI><B>chrom</B> - The name of the chromosome (e.g. chr3, chrY,
	chr2_random) or contig (e.g. ctgY1).
	<LI><B>chromStart</B> - The starting position of the feature in the
	chromosome or contig. The first base in a chromosome is numbered 0.
	<LI><B>chromEnd</B> - The ending position of the feature in the
	chromosome or contig. The <I>chromEnd</I> base is not included in the
	display of the feature. For example, the first 100 bases of a
	chromosome are defined as <I>chromStart=0, chromEnd=100</I>, and span
	the bases numbered 0-99.
</OL>

<P>The 9 additional optional BED fields are:</P>
<OL START=4>
	<LI><B>name</B> - Defines the name of the BED line.  This label is
	displayed to the left of the BED line in the Genome Browser
	window when the track is open to full display mode or directly to the
	left of the item in pack mode. 
	<LI><B>score</B> - A score between 0 and 1000. If the track line 
	<I>useScore</I> attribute is set to 1 for this annotation data set, the
	<I>score</I> value will determine the level of gray in which
	this feature is displayed (higher numbers = darker gray).
	<LI><B>strand</B> - Defines the strand - either '+' or '-'.
	<LI><B>thickStart</B> - The starting position at which the feature
	is drawn thickly (for example, the start codon in gene
	displays).
	<LI><B>thickEnd</B> - The ending position at which the feature is
	drawn thickly (for example, the stop codon in gene displays).
	<LI><B>reserved</B> - This should always be set to zero.
	<LI><B>blockCount</B> - The number of blocks (exons) in the
	BED line.
	<LI><B>blockSizes</B> - A comma-separated list of the block
	sizes. The number of items in this list should correspond to
	<I>blockCount</I>.
	<LI><B>blockStarts</B> - A comma-separated list of block starts.
	All of the <I>blockStart</I> positions should be calculated relative to
	<I>chromStart</I>. The number of items in
	this list should correspond to <I>blockCount</I>.
</OL>

<P><B><I>Example:</I></B><BR>
Here's an example of an annotation track that uses a complete BED definition:</P>
<PRE><TT>track name=pairedReads description=&quot;Clone Paired Reads&quot; useScore=1
chr22 1000 5000 cloneA 960 + 1000 5000 0 2 567,488, 0,3512
chr22 2000 6000 cloneB 900 - 2000 6000 0 2 433,399, 0,3601</TT></PRE>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>
	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><a name="PSL"></a><b>&nbsp; PSL Lines </b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>

<P>PSL lines represent alignments, and are typically taken from files generated by BLAT 
or psLayout. See the 
<a
href="http://genome.ucsc.edu/goldenPath/help/hgTracksHelp.html#BLATAlign"
TARGET=_blank>BLAT 
documentation</a> for more details. All of the following fields are
required on each data line within a PSL file:
<OL>
<LI><B>matches</B> - Number of bases that match that aren't repeats
<LI><B>misMatches</B> - Number of bases that don't match
<LI><B>repMatches</B> - Number of bases that match but are part of repeats
<LI><B>nCount</B> - Number of 'N' bases
<LI><B>qNumInsert</B> - Number of inserts in query
<LI><B>qBaseInsert</B> - Number of bases inserted in query
<LI><B>tNumInsert</B> - Number of inserts in target
<LI><B>tBaseInsert</B> - Number of bases inserted in target
<LI><B>strand</B> - '+' or '-' for query strand. In mouse, second '+'or '-' is for genomic strand
<LI><B>qName</B> - Query sequence name
<LI><B>qSize</B> - Query sequence size
<LI><B>qStart</B> - Alignment start position in query
<LI><B>qEnd</B> - Alignment end position in query
<LI><B>tName</B> - Target sequence name
<LI><B>tSize</B> - Target sequence size
<LI><B>tStart</B> - Alignment start position in target
<LI><B>tEnd</B> - Alignment end position in target
<LI><B>blockCount</B> - Number of blocks in the alignment
<LI><B>blockSizes</B> - Comma-separated list of sizes of each block
<LI><B>qStarts</B> - Comma-separated list of starting positions of each block in query
<LI><B>tStarts</B> - Comma-separated list of starting positions of each block in target
</OL>
<P><B><I>Example:</I></B><BR>
Here is an example of an annotation track in PSL format. Note that line
breaks have been inserted into the PSL lines in this example for
documentation display purposes.
 Click <A HREF ="fishBlats.txt">
here</A> for a copy of this example that can be pasted into the browser without editing. 

<PRE><TT>track name=fishBlats description=&quot;Fish BLAT&quot; useScore=1
59 9 0 0 1 823 1 96 +- FS_CONTIG_48080_1 1955 171 1062 chr22 
    47748585 13073589 13073753 2 48,20,  171,1042,  34674832,34674976,
59 7 0 0 1 55 1 55 +- FS_CONTIG_26780_1 2825 2456 2577 chr22 
    47748585 13073626 13073747 2 21,45,  2456,2532,  34674838,34674914,
59 7 0 0 1 55 1 55 -+ FS_CONTIG_26780_1 2825 2455 2676 chr22 
    47748585 13073727 13073848 2 45,21,  249,349,  13073727,13073827,
</TT></PRE>

<P>Be aware that the coordinates for a negative strand in a PSL 
line are handled in a special way. In the 
<I>qStart</I> and <I>qEnd</I> fields, the coordinates indicate 
the position where the query matches from 
the point of view of the forward strand, even when the match is on the reverse strand.
However, in the <I>qStarts</I> list, the coordinates are reversed.</P> 
<P>
<B><I>Example:</I></B><BR>
Here is a 30-mer containing 2 blocks that align on the minus strand and 
2 blocks that align on the plus strand (this sometimes can happen in response to 
assembly errors): <BR><BR>
<TABLE CELLPADDING=0 CELLSPACING=0 MARGINS=0 BORDER=0>
<TR><TD><PRE>0         1         2         3 tens position in query   </TD></TR>
<TR><TD><PRE>0123456789012345678901234567890 ones position in query   </TD></TR>
<TR><TD><PRE>            ++++          +++++ plus strand alignment on query   </TD></TR>
<TR><TD><PRE>    --------    ----------      minus strand alignment on query   </TD></TR>
<TR><TD></TD></TD>	
<TR><TD><PRE></TD></TD>	
<TR><TD><PRE>Plus strand:   </TD></TR>
<TR><TD><PRE>     qStart=12 </TD></TR>
<TR><TD><PRE>     qEnd=31 </TD></TR>
<TR><TD><PRE>     blockSizes=4,5 </TD></TR>
<TR><TD><PRE>     qStarts=12,26   </TD></TR>
<TR><TD><PRE>                      </TD></TR>
<TR><TD><PRE>Minus strand:   </TD></TR>
<TR><TD><PRE>     qStart=4 </TD></TR>
<TR><TD><PRE>     qEnd=26 </TD></TR>
<TR><TD><PRE>     blockSizes=10,8 </TD></TR>
<TR><TD><PRE>     qStarts=5,19    </TD></TR>
</TABLE><BR>
Essentially, the minus strand <I>blockSizes</I> and <I>qStarts</I> are 
what you would get if you reverse-complemented the query.
However, the <I>qStart</I> and <I>qEnd</I> are not reversed. To convert one to the other:
<PRE>
     qStart = qSize - revQEnd
     qEnd = qSize - revQStart
</PRE>


	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><a name="GFF"></a><b>&nbsp; GFF Lines </b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>

<P>GFF (General Feature Format) lines are based on the GFF standard file format.  GFF 
lines have nine required fields that <I>must</I> be tab-separated. If the fields are 
separated by spaces instead of tabs, the track will not display correctly. For more
information on GFF format, refer to <A
HREF="http://www.sanger.ac.uk/Software/formats/GFF"
TARGET=_blank>http://www.sanger.ac.uk/Software/formats/GFF</A>. </P>  
<P>
Here is a brief description of the GFF fields:</P>

<OL>
	<LI><B>seqname</B> - The name of the sequence.  Must be a chromosome or a contig.
	<LI><B>source</B> - The program that generated this feature.
	<LI><B>feature</B> - The name of this type of feature. Some examples of
	standard feature types are "CDS", "start_codon", "stop_codon", and
	"exon".  
	<LI><B>start</B> - The starting position of the feature in the sequence.  
	The first base is numbered 1.
	<LI><B>end</B> - The ending position of the feature (inclusive).
	<LI><B>score</B> - A score between 0 and 1000. If the track line 
	<I>useScore</I> attribute is set to 1 for this annotation data set, the
	<I>score</I> value will determine the level of gray in which
	this feature is displayed (higher numbers = darker gray). If there is no
	score value, enter ".".
	<LI><B>strand</B> - Valid entries include '+', '-', or '.' (for don't know/don't care).
	<LI><B>frame</B> - If the feature is a coding exon, <I>frame</I> should 
	be a number between 0-2 that represents the reading frame of the
	first base.  If the feature is not a coding exon, the value should be '.'.
	<LI><B>group</B> - All lines with the same group are linked together into a single item.
</OL>

<P><B><I>Example:</I></B><BR>
Here's an example of a GFF-based track. 
Click <A HREF ="regulatory.txt">
here</A> for a copy of this example that can be pasted into the browser without editing. 
NOTE: Paste operations on some operating systems will replace tabs with spaces, which
will result in an error when the GFF track is uploaded. You can circumvent this 
problem by pasting the URL of the above example 
(http://genome.ucsc.edu/goldenPath/help/regulatory.txt) instead of the text itself
into the custom annotation track text box. 

<PRE><TT>track name=regulatory description=&quot;TeleGene(tm) Regulatory Regions&quot;
chr22  TeleGene	enhancer  1000000  1001000  500	+  .  touch1
chr22  TeleGene	promoter  1010000  1010100  900	+  .  touch1
chr22  TeleGene	promoter  1020000  1020000  800	-  .  touch2</TT></PRE>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><a name="GTF"></a><b>&nbsp;  GTF Lines </b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>

<P>GTF (Gene Transfer Format) is a refinement to GFF that tightens the specification.  
The first eight GTF fields are the same as GFF. The <I>group</I> field has been 
expanded into an <I>attribute</I> field that includes a list of
semicolon-separated attribute/value pairs. For more information on this format,
see <A HREF="http://genes.cs.wustl.edu/GTF2.html" 
TARGET=_blank>http://genes.cs.wustl.edu/GTF2.html</A>.

<P>Some examples of entries for the <I>attribute</I> field include:
<UL>
<LI><B>gene_id <I>value</I></B> - A globally unique identifier for the genomic
source of the sequence.
<LI><B>transcript_id <I>value</I></B> - A globally unique identifier for the
predicted transcript.
</UL>
<P><B><I>Example:</I></B><BR>
Here is an example of the ninth field in a GTF data line:
<P>
<tt>gene_id Em:U62317.C22.6.mRNA; transcript_id Em:U62317.C22.6.mRNA;
exon_number 1</tt>

<P>The Genome Browser groups together GTF lines that have the same
<I>transcript_id</I> value.  It only looks at <I>EXON</I> and <I>CDS</I>
type features.</P>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
	<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><a name="SHARE"></a><b>&nbsp;  Sharing Your Annotation Track with Others</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>

To make your Genome Browser annotation track viewable by people on 
other machines or at other sites, follow the steps below. (Note that 
some of the URL examples in this section have been broken up into 2
lines for documentation display purposes).<BR><BR>

<P><B>Step 1.</B>
Put your formatted annotation file on your web site. Be sure that 
the file permissions allow it to be read by others.</P>

<P><B>Step 2.</B>
Construct a URL that will link this annotation file to the Genome 
Browser. The URL must contain 3 pieces of information specific
to your annotation data: 
<UL>
<LI>The genome freeze on which your annotation data is based. This
information is of the form <tt>db=database_name</tt>, where
<I>database_name</I> is to the UCSC code for the genome freeze. For a list
of these codes, see the Genome Browser <A
HREF="http://genome.ucsc.edu/FAQ.html#124" TARGET=_blank>FAQ</A>.
Examples of this include: db=hg16 (Human July 2003 freeze), db=mm3 
(Mouse Feb 2003 freeze).
<LI>The genome position that the Genome Browser should initially 
open to. This information is of the form
<tt>position=chr_position</tt>, where <I>chr_position</I> is a 
chromosome number, with or without a set of 
coordinates. Examples of this include: position=chr22, position=chr22:15916196-31832390.
<LI>The URL of the annotation file on your web site. This information is
of the form <tt>hgt.customText=URL</tt>, where
<I>URL</I> points to the annotation file on your website. An example of an
annotation file URL is <font color="000099"><A
HREF="http://genome.ucsc.edu/goldenPath/help/test.bed"
TARGET=_blank>http://genome.ucsc.edu/goldenPath/help/test.bed</A></font>.  
</UL>

<P>Combine the above pieces of information into a URL of the following
format (the information specific to your annotation file is highlighted):
<BR><BR>
<tt>http://genome.ucsc.edu/cgi-bin/hgTracks?db=<font color="000099"><I>database_name</I></font>&amp;<BR>position=<font
color="000099"><I>chr_position</I></font>&amp;hgt.customText=<font color="000099"><I>URL</I></font></tt>.

<P><B><I>Example:</I></B><BR>
The following URL will open up the Genome Browser 
window to display chr 22 of the July 2003 freeze of the 
human genome (hg16), and will display the annotation track 
pointed to by the URL
http://genome.ucsc.edu/goldenPath/help/test.bed:<BR><BR>

<font color="000099"><A HREF="
http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg16&amp;position=chr22&amp;hgt.customText=http://genome.ucsc.edu/goldenPath/help/test.bed"
TARGET=_BLANK><tt>http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg16&amp;position=chr22&amp;<br>hgt.customText=http://genome.ucsc.edu/goldenPath/help/test.bed</tt></A></font><BR><BR>

<P><B>Step 3.</B> Provide the URL to others. To upload a custom annotation
pointed to by a URL into the Genome Browser, paste the URL into the large text edit 
box in the Add Your Own Tracks section of the Genome Browser Gateway page, then click 
the Submit button. 
<P>If you'd like to share your annotation track with a broader 
audience, send the URL for your track - along with a description of 
the format, methods, and data used - to the UCSC Genome mailing list 
<A HREF="mailto:&#103;&#101;&#110;&#111;&#109;e&#64;&#115;&#111;&#101;.&#117;&#99;sc.e&#100;&#117;">&#103;&#101;&#110;&#111;&#109;e&#64;&#115;&#111;&#101;.&#117;&#99;sc.e&#100;&#117;</A>.
<!-- above address is genome at soe.ucsc.edu -->
</P>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
	<BR>


  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><a name="TROUBLE"></a><b>&nbsp;  Troubleshooting Annotation Display Problems</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>

<P>Occasionally users encounter problems when uploading 
annotation files to the Genome Browser. In most cases, these problems
are caused by errors in the format of the annotation file and 
can be tracked down using the information displayed in the error
message. This section contains suggestions for resolving common 
display problems. If you are still unable to successfully
display your annotation data after reading this section, contact
the genome mailing list at 
<A HREF="mailto:&#103;&#101;&#110;&#111;&#109;e&#64;&#115;&#111;&#101;.&#117;&#99;sc.e&#100;&#117;">&#103;&#101;&#110;&#111;&#109;e&#64;&#115;&#111;&#101;.&#117;&#99;sc.e&#100;&#117;</A>
<!-- above address is genome at soe.ucsc.edu -->
for further assistance.

<P><B><I>Problem:</I></B> When I try one of your examples by cutting and pasting
it into the Genome Browser, I get an error message.<BR> 
<B><I>Solution:</I></B> Check that none of the browser lines, track lines, or data 
lines in your annotation file contains a line break. If the example contains GFF
or GTF data lines, check that all the fields are tab-separated rather than
space-separated.

<P><B><I>Problem:</I></B> When I click the Submit button, I get 
the error message &quot;line 1 of custom input:&quot;.<BR>
<B><I>Solution:</I></B> Check that none of the browser lines, 
track lines, or data lines in your annotation file contains a line break. A 
common source for this problem is the track line: all of the attribute
pairs must on the same line and must not be separated by
a line break. If you are uploading your annotation file by pasting it
into the text box on the Genome Browser Gateway page, check that the cut and paste operation
did not inadvertently insert unwanted line feeds into the longer lines.

<P><B><I>Problem:</I></B> When I click the Submit button, I get the
error message &quot;line # of custom input: missing = in var/val pair&quot;.<BR>
<B><I>Solution:</I></B> Check for incorrect syntax in the track 
lines in the annotation file. Be sure that each track line
attribute pair consists of the format <I>attribute=attribute name</I>.

<P><B><I>Problem:</I></B> When I click the Submit button, I get the
error message &quot;line # of custom input: BED chromStarts[i] must be in ascending order&quot;.<BR>
<B><I>Solution:</I></B> This is most likely caused by a logical conflict in the 
Genome Browse software. It accepts custom GFF tracks that have multiple &quot;exons&quot; at
the same position, but not BED tracks. Because the browser translates GFF tracks 
to BED format before storing the custom track data, GFF tracks with multiple 
exons will case an error when the BED is read back in. To work around this 
problem, remove duplicate lines in the GFF track. 

<P><B><I>Problem:</I></B> When I click the Submit button, the Genome Browser
track window displays OK, but my track isn't visible.<BR>
<B><I>Solution:</I></B> Check the browser and track lines in your annotation
file to make sure that you haven't accidentally set
the display mode for the track to <I>hide</I>. If you are using the
Annotation File box on the Genome Browser Gateway page to 
upload the track, check that you've
entered the correct file name. If neither of these are the cause 
of the problem, try resetting the Genome Browser to clear any 
settings that may be preventing the annotation to display. To 
reset the Genome Browser, click on &quot;Click here to reset&quot; on the Gateway
page. If the annotation track still doesn't display, you may 
need to clear the cookies in your Internet browser as well
(refer to your Internet browser's documentation for further information).

<P><B><I>Problem:</I></B> I've gotten my annotation track to display, 
but now I can't make it go away! How do I remove an annotation
track from my Genome Browser display?<BR>
<B><I>Solution:</I></B> Reset the Genome Browser by clicking the 
&quot;reset all&quot; button on the Genome Browser tracks window or by clicking on 
&quot;Click here to reset&quot; on the Gateway page. This should reset 
your Genome Browser display to its default settings.

</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

</TD></TR></TABLE>
</BODY></HTML>



