<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>

<HEAD>
	<META NAME="GENERATOR" Content="Visual Page 1.0 for Windows">
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>untitled</TITLE>
</HEAD>

<BODY>

<H1>Formats for Add Your Own Track</H1>
<P>You can temporarily add your own tracks to the genome browser by 
uploading or pasting in text in the formats
described in this document.    Here is a simple example:</P>
<PRE><TT>
browser position chr22:1-20001
track name=spacer description=&quot;Blue ticks every 10000 bases&quot; color=0,0,255
chr22	0	 1
chr22	10000 10001
chr22	20000 20001
track name=even description=&quot;This track paints 100 bases red, skips 100&quot; color=255,0,0
chr22	0	100 first
chr22	200	300 second
chr22	400	500 third</TT></PRE>
<P>The format is line oriented.  A line can describe a track, or an item in a track. Empty lines and lines starting with # are ignored.  Track lines always
begin with the word 'track'.  Item lines can be in any one of four formats: BED, PSL,  GFF or GTF.  All the lines
for a given track must be in the same format. </P>
<H2>Browser Lines</H2>
<P>Lines that begin with the word 'browser' control the overall browser
display.  The second word in the line specifies what to control and may
be one of the following:</P>
<UL>
   <LI>position - Describes the part of the genome the browser will display in 
       chromosome:start-end format.  
   <LI>hide all - Hides all tracks except for custom ones.
   <LI>hide trackName(s) - this hides tracks based on their symbolic names.  You can find the symbolic names of tracks by viewing the genome browser as source under your web browser.
   <LI>dense all - Shows all tracks in dense mode.
   <LI>dense trackName(s) - makes particular tracks dense.
   <LI>full all - Shows all tracks in full mode.
   <LI>full trackName(s) - makes particular tracks full.
</UL>
<H2>Track Lines</H2>
<P>Track lines are used to define attributes that are used for all lines of a track.  Track lines begin with the
word 'track' and are followed by a number of attribute=value settings.   Here are the currently defined attributes:</P>

<UL>
	<LI>name - This will be displayed in the left column in the browser, and also in the track controls below. Can
	be any text up to 15 letters.  Default value is  &quot;User Track.&quot;
	<LI>description - This will be displayed as a center label in the browser. Can be any text up to 60 letters. Default
	value is &quot;User Supplied Track.&quot;
	<LI>visibility - This controls how the track will initially be displayed.  Allowable values are 0 for hide, 1 for
	dense, and 2 for full. Default is 1.
	<LI>color - the main color for the track.  This should be three comma separated RGB values from 0-255.  Default
	is 0,0,0 (black)
	<LI>altColor - secondary color for the track.  Default is a lighter shade of whatever color is set to.
	<LI>useScore - If this attribute is present and set to one then the 'score' field will be reflected as the level
	of gray in the display.
	<LI>offset - A number to be added to all coordinates in track.
	<LI>url - A URL for an external link associated with this track.  Any '$$' in this string this will be substituted
	with the item name. No default.
</UL>

<H2>BED Lines</H2>
<P>BED lines are a flexible way of defining things to display in the browser.  BED lines must have at least three
fields, and optionally can have up to twelve fields.  The number of fields must be consistent throughout any single
track.  The first three, required fields are:</P>

<OL>
	<LI>chrom - Either a chromosome name (such as chr3, chrY, chr2_random), or a contig name (such as ctgY1).
	<LI>chromStart - Start of feature.  First base in chromosome is numbered 0.
	<LI>chromEnd - End of feature.  The end is not included.  (So the first 100 bases are chromStart=0, chromEnd=100.)
</OL>

<P>In addition BED lines can have up to 9 additional fields:</P>

<OL>
	<LI>name - Name of item.  Will be displayed in left column of browser when track is fully open.
	<LI>score - A score between 0 and 1000.
	<LI>strand - Either '+' or '-'.
	<LI>thickStart - start of where feature is drawn thickly.  (Start codon in gene displays.)
	<LI>thickEnd - end of where feature is drawn thickly. (Stop codon in gene displays.)
	<LI>reserved - Currently should always be set to zero.
	<LI>blockCount - The number of blocks (exons) in this item.
	<LI>blockSizes - A comma separated list of block sizes.
	<LI>blockStarts - A comma separated list of block starts.  These are all relative to chromStart.
</OL>

<P>Here's an example of a track which uses a complete BED definition:</P>
<PRE><TT>track name=pairedReads description=&quot;Reads taken from each side of a clone&quot; useScore=1
chr22 1000 5000 cloneA 960 + 1000 5000 0 2 567,488, 0,4512
chr22 2000 6000 cloneB 900 - 2000 6000 0 2 433,399, 0,5601</TT></PRE>
<H2>PSL Lines</H2>
<P>PSL lines represent alignments, and are typically taken from files generated by BLAT or psLayout.  Please see
the BLAT documentation for more details.</P>
<H2>GFF Lines</H2>
<P>GFF lines are based on the GFF standard file format.  GFF lines have 9 fields which must be tab separated. 
(Space separated won't do.)  Here is a brief description of the fields:</P>

<OL>
	<LI>seqname - The name of the sequence.  Must be a chromosome or a contig.
	<LI>source - The program that generated this feature.
	<LI>feature - The name of this type of feature. 'Exon' is frequently used here.
	<LI>start - The start of the feature in the sequence.  The first base is numbered 1.
	<LI>end - The end of the feature (inclusive).
	<LI>score - A score.  This should be between 0 and 1000 if you want to display it as a gray level.
	<LI>strand - Either '+' or '-' or '.' (for don't know/don't care).
	<LI>frame - If this is a coding exon this should be a number between 1-3 representing the reading frame of the
	first base.  Otherwise it's '.'.
	<LI>group - All lines with the same group are linked together into a single item.<A HREF="http://www.sanger.ac.uk/Software/formats/GFF/GFF_Spec.shtml"></A>
</OL>

<P>Here's an example of a GFF based track:</P>
<PRE><TT>track name=regulatory description=&quot;Regulatory Regions from TeleGene(tm)&quot; color=150,0,180
chr22	TeleGene	enhancer	1000000	1001000	500	+	.	touch1
chr22	TeleGene	promoter	1010000	1010100	900	+	.	touch1
chr22	TeleGene	promoter	1020000	1020000	800	-	.	touch2</TT></PRE>
<P>
<H2>GTF Lines</H2>
<P>GTF is a recent refinement to GFF which tightens the spec in many ways.  The first eight fields are the same
as GFF.  The 9th field is expanded to include a list of semicolon separated attribute/value pairs such as:</P>

<P>gene_id Em:U62317.C22.6.mRNA ; transcript_id Em:U62317.C22.6.mRNA ; exon_number 1</P>

<P>The browser will group together GTF lines that have the same transcript_id.  It only looks at 'EXON'' and 'CDS'
type features.

</BODY>

</HTML>
