<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>Human Genome Working Draft: Custom Track Formats</TITLE>
	<LINK REL="STYLESHEET" HREF="http://genome.ucsc.edu/style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 WIDTH="100%">
<TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="/images/title.jpg"></TH></TR>


<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
   <CENTER>
	<TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
	<TABLE WIDTH="100%" BGCOLOR="#2636D1" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
		<TABLE WIDTH=431 CELLSPACING=4 CELLPADDING=0 BORDER=0><TR>
		<TD WIDTH=1></TD>
		<TD WIDTH=75>
			<A HREF="/index.html" onMouseOver="HOME.src='/images/hl_home.jpg'" onMouseOut="HOME.src='/images/h_home.jpg'">
			<IMG SRC="/images/h_home.jpg" ALIGN="absmiddle" BORDER="0" NAME="HOME" ALT="Home" ></A></TD>
		<TD WIDTH=90 ALIGN="center">
			<A HREF="/goldenPath/hgTracks.html" onMouseOver="BROWSER.src='/images/hl_browser.jpg'" onMouseOut="BROWSER.src='/images/h_browser.jpg'">
			<IMG SRC="/images/h_browser.jpg" ALIGN="absmiddle" BORDER="0" NAME="BROWSER" ALT="&nbsp;&nbsp; Genome Browser"></A></TD>		
		<TD WIDTH=90 ALIGN="center">
			<A HREF="/cgi-bin/hgBlat?command=start" onMouseOver="SEARCH.src='/images/hl_blat.jpg'" onMouseOut="SEARCH.src='/images/h_blat.jpg'">
			<IMG SRC="/images/h_blat.jpg" ALIGN="absmiddle" BORDER="0" NAME="SEARCH" ALT="&nbsp;&nbsp; BLAT Search"></A></TD>	
		<TD WIDTH=30 ALIGN="center">
			<A HREF="/FAQ.html"
			onMouseOver="FAQ.src='/images/hl_faq.jpg'" onMouseOut="FAQ.src='/images/h_faq.jpg'">
			<IMG SRC="/images/h_faq.jpg" ALIGN="absmiddle" BORDER="0"
			NAME="FAQ" ALT="&nbsp;&nbsp; FAQ"></A></TD>
				<TD WIDTH=100 ALIGN="center">
			<A HREF="/goldenPath/help/hgTracksHelp.html" onMouseOver="GUIDE.src='/images/hl_guide.jpg'" onMouseOut="GUIDE.src='/images/h_guide.jpg'">
			<IMG SRC="/images/h_guide.jpg" ALIGN="absmiddle"
			BORDER="0" NAME="GUIDE" ALT="&nbsp;&nbsp; GUIDE"></A></TD>
		</TR></TABLE>
		</TR></TABLE>
	 </TD></TR></TABLE>
	 </TD></TR></TABLE>
	 </CENTER>
</TD></TR>

<!--Content Tables-->  
<TR><TD CELLPADDING=10>	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><b>&nbsp;  Adding Your Own Track  </b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>

<P>You can temporarily add your own tracks to the genome browser by uploading
or pasting in text in the formats described in this document. Here is a
simple example:</P>

<P><B>Step 1.</B><BR>
Formulate the data set as a tab-separated file using one of the
formats listed in the sections below.
Here is an example of a file that defines one track showing blue
1-base tick marks every 10000 bases at the begining of chromosome
22 and another track showing red 100-base features alternating with 
blank space in part of that same region:
<PRE><TT>browser position chr22:1-20001
track name=spacer description="My blue ticks" color=0,0,255
chr22   0        1
chr22   10000 10001
chr22   20000 20001
track name=even description="My red 100 bases tracks" color=255,0,0
chr22   0       100 first
chr22   200     300 second
chr22   400     500 third</TT></PRE>

The format is line- oriented. A line can describe a track, or an item in a
track. Empty lines and lines starting
with # are ignored. Track lines always begin with the word 'track'. Item
lines can be in any one of four
formats:
	<A HREF="#BED">BED</A>, 
	<A HREF="#PSL">PSL</A>,  
	<A HREF="#GFF">GFF</A> or 
	<A HREF="#GTF">GTF</A>.  
All the lines for a given track must be in
the same format. For syntax information, refer to the sections below.</P>

<P><B>Step 2.</B><BR>
Point your browser at 
<a href="http://genome.ucsc.edu/cgi-bin/hgGateway?db=hg10">http://genome.ucsc.edu/cgi-bin/hgGateway?db=hg10</a>,
scroll down to the "Add Your Own Tracks" section, 
and upload your tab-separated file to see how it looks in the context of the
other data tracks on the browser.</P>

<P><B>Step 3.</B><BR>
When you are ready to share your track with the world, or with the
other members of your research group, put your tab-separated file on a web
site. You can then construct a URL that will link your file to the browser.
<BR><BR>
For example, if the URL for your track file is
<A
HREF="http://genome-test.cse.ucsc.edu/test.bed"><tt>http://genome-test.cse.ucsc.edu/test.bed</tt></A>,
you can use the URL<BR>
<font
color="000099"><A HREF="
http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg10&amp;<br>hgt.customText=http://genome-test.cse.ucsc.edu/test.bed"><tt>http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg10&amp;<br>hgt.customText=http://genome-test.cse.ucsc.edu/test.bed</tt></A></font>
<BR>to share your track with others, either by mailing it to colleagues or putting 
it on your web page. Please send the URL for your track along with a description of 
the methods, data used and format to genome@cse.ucsc.edu.
<BR><BR>
In this URL example, "db=hg10" specifies the Dec. 2001 freeze. 
The hgt.customText variable tells the browser where to find your track file.
</P>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><a name="BROWSER"></a><b>&nbsp;  Browser Lines   </b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>Lines in the file such as 'browser position chr22:1-20001' that begin with the 
   word 'browser' control the overall browser display. The second word in the 
   line specifies what to control and may be one of the following:
<UL>
   <LI>position - Describes the part of the genome the browser will initially display in 
       chromosome:start-end format.  
   <LI>pix - Sets the display to width pixels.
   <LI>hide all - Hides all tracks except for custom ones.
   <LI>hide trackName(s) - this hides tracks based on their symbolic names.  You can find the symbolic names of tracks by viewing the genome browser as source under your web browser.
   <LI>dense all - Shows all tracks in dense mode.
   <LI>dense trackName(s) - makes particular tracks dense.
   <LI>full all - Shows all tracks in full mode.
   <LI>full trackName(s) - makes particular tracks full.
</UL>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><a name="TRACK"></a><b>&nbsp; Track Lines</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
	
<P>Track lines are used to define attributes that are used for all lines of a track.  Track lines begin with the
word 'track' and are followed by a number of attribute=value settings.   Here are the currently defined attributes:</P>

<UL>
	<LI>name - This will be displayed in the left column in the browser, and also in the track controls below. Can
	be any text up to 15 letters.  Default value is  &quot;User Track.&quot;
	<LI>description - This will be displayed as a center label in the browser. Can be any text up to 60 letters. Default
	value is &quot;User Supplied Track.&quot;
	<LI>visibility - This controls how the track will initially be displayed.  Allowable values are 0 for hide, 1 for
	dense, and 2 for full. Default is 1.
	<LI>color - the main color for the track.  This should be three comma separated RGB values from 0-255.  Default
	is 0,0,0 (black)
	<LI>altColor - secondary color for the track.  Default is a lighter shade of whatever color is set to.
	<LI>useScore - If this attribute is present and set to one then the 'score' field will be reflected as the level
	of gray in the display.
	<LI>offset - A number to be added to all coordinates in track.
	<LI>url - A URL for an external link associated with this track.  Any '$$' in this string this will be substituted
	with the item name. No default.
</UL>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>
	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><a name="BED"></a><b>&nbsp; Bed Lines </b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>

<P>BED lines are a flexible way of defining things to display in the browser.  BED lines must have at least three
fields, and optionally can have up to twelve fields.  The number of fields must be consistent throughout any single
track.  The first three, required fields are:</P>

<OL>
	<LI>chrom - Either a chromosome name (such as chr3, chrY, chr2_random), or a contig name (such as ctgY1).
	<LI>chromStart - Start of feature.  First base in chromosome is numbered 0.
	<LI>chromEnd - End of feature.  The end is not included.  (So the first 100 bases are chromStart=0, chromEnd=100.)
</OL>

<P>In addition BED lines can have up to 9 additional fields:</P>

<OL>
	<LI>name - Name of item.  Will be displayed in left column of browser when track is fully open.
	<LI>score - A score between 0 and 1000.
	<LI>strand - Either '+' or '-'.
	<LI>thickStart - start of where feature is drawn thickly.  (Start codon in gene displays.)
	<LI>thickEnd - end of where feature is drawn thickly. (Stop codon in gene displays.)
	<LI>reserved - Currently should always be set to zero.
	<LI>blockCount - The number of blocks (exons) in this item.
	<LI>blockSizes - A comma separated list of block sizes.
	<LI>blockStarts - A comma separated list of block starts.  These are all relative to chromStart.
</OL>

<P>Here's an example of a track which uses a complete BED definition:</P>
<PRE><TT>track name=pairedReads description=&quot;Clone Paired Reads&quot; useScore=1
chr22 1000 5000 cloneA 960 + 1000 5000 0 2 567,488, 0,4512
chr22 2000 6000 cloneB 900 - 2000 6000 0 2 433,399, 0,5601</TT></PRE>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>
	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><a name="PSL"></a><b>&nbsp; PSL Lines </b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>

<P>PSL lines represent alignments and are typically taken from files generated by BLAT 
or psLayout. See the 
<a href="http://genome.cse.ucsc.edu/goldenPath/help/hgTracksHelp.html#BLATAlign">BLAT 
documentation</a> for more details. PSL lines contain 22 fields: 
<OL>
<LI>bin - Field to speed indexing added to many tables August 2001
<LI>matches - Number of bases that match that aren't repeats
<LI>misMatches - Number of bases that don't match
<LI>repMatches - Number of bases that match but are part of repeats
<LI>nCount - Number of 'N' bases
<LI>qNumInsert - Number of inserts in query
<LI>qBaseInsert - Number of bases inserted in query
<LI>tNumInsert - Number of inserts in target
<LI>tBaseInsert - Number of bases inserted in target
<LI>strand - '+' or '-' for query strand. In mouse, second '+'or '-' is for genomic strand
<LI>qName - Query sequence name
<LI>qSize - Query sequence size
<LI>qStart - Alignment start position in query
<LI>qEnd - Alignment end position in query
<LI>tName - Target sequence name
<LI>tSize - Target sequence size
<LI>tStart - Alignment start position in target
<LI>tEnd - Alignment end position in target
<LI>blockCount - Number of blocks in the alignment
<LI>blockSizes - Comma-separated list of sizes of each block
<LI>qStarts - Comma-separated list of starting positions of each block in query
<LI>tStarts - Comma-separated list of starting positions of each block in target
</OL>
<P>
Caution should be taken when examining coordinates for a negative strand. In the 
qStart/qEnd fields, the coordinates indicate the position where the query matches from 
the point of view of the forward strand, even when the match is on the reverse strand.
However, in the qStarts list, the coordinates are reversed.</P> 
<P>
For example, here is a 30-mer containing 2 blocks that align on the minus strand and 
2 blocks that align on the plus strand (this sometimes can happen in response to 
assembly errors): <BR><BR>
<TABLE CELLPADDING=0 CELLSPACING=0 MARGINS=0 BORDER=0>
<TR><TD><PRE>0         1         2         3 tens position in query   </TD></TR>
<TR><TD><PRE>0123456789012345678901234567890 ones position in query   </TD></TR>
<TR><TD><PRE>            ++++          +++++ plus strand alignment on query   </TD></TR>
<TR><TD><PRE>    --------    ----------      minus strand alignment on query   </TD></TR>
<TR><TD></TD></TD>	
<TR><TD><PRE></TD></TD>	
<TR><TD><PRE>Plus strand:   </TD></TR>
<TR><TD><PRE>     qStart=12 </TD></TR>
<TR><TD><PRE>     qEnd=31 </TD></TR>
<TR><TD><PRE>     blockSizes=4,5 </TD></TR>
<TR><TD><PRE>     qStarts=12,26   </TD></TR>
<TR><TD><PRE>                      </TD></TR>
<TR><TD><PRE>Minus strand:   </TD></TR>
<TR><TD><PRE>     qStart=4 </TD></TR>
<TR><TD><PRE>     qEnd=26 </TD></TR>
<TR><TD><PRE>     blockSizes=10,8 </TD></TR>
<TR><TD><PRE>     qStarts=5,19    </TD></TR>
</TABLE><BR>
Essentially, the minus strand blockSizes and qStarts are 
what you would get if you reverse-complemented the query.
However, the qStart and qEnd are not reversed. To get
from one to the other:
<PRE>
     qStart = qSize - revQEnd
     qEnd = qSize - revQStart
</PRE>


	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><a name="GFF"></a><b>&nbsp; GFF Lines </b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>

<P>GFF lines are based on the GFF standard file format.  GFF lines have 9 fields which must be tab separated. 
(Space separated won't do.)  Here is a brief description of the fields:</P>

<OL>
	<LI>seqname - The name of the sequence.  Must be a chromosome or a contig.
	<LI>source - The program that generated this feature.
	<LI>feature - The name of this type of feature. 'Exon' is frequently used here.
	<LI>start - The start of the feature in the sequence.  The first base is numbered 1.
	<LI>end - The end of the feature (inclusive).
	<LI>score - A score.  This should be between 0 and 1000 if you want to display it as a gray level.
	<LI>strand - Either '+' or '-' or '.' (for don't know/don't care).
	<LI>frame - If this is a coding exon this should be a number between 1-3 representing the reading frame of the
	first base.  Otherwise it's '.'.
	<LI>group - All lines with the same group are linked together into a single item.<A HREF="http://www.sanger.ac.uk/Software/formats/GFF/GFF_Spec.html"></A>
</OL>

<P>Here's an example of a GFF based track:</P>
<PRE><TT>track name=regulatory description=&quot;TeleGene(tm) Regulatory Regions&quot;
chr22  TeleGene	enhancer  1000000  1001000  500	+  .  touch1
chr22  TeleGene	promoter  1010000  1010100  900	+  .  touch1
chr22  TeleGene	promoter  1020000  1020000  800	-  .  touch2</TT></PRE>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><a name="GTF"></a><b>&nbsp;  GTF Lines </b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>

<P>GTF is a recent refinement to GFF which tightens the spec in many ways.  The first eight fields are the same
as GFF.  The 9th field is expanded to include a list of semicolon separated attribute/value pairs such as:</P>

<P>gene_id Em:U62317.C22.6.mRNA ; transcript_id Em:U62317.C22.6.mRNA ; exon_number 1</P>

<P>The browser will group together GTF lines that have the same transcript_id.  It only looks at 'EXON'' and 'CDS'
type features.</P>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
</TD></TR></TABLE>
</BODY></HTML>



