<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
        <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
        <TITLE>UCSC Genome Browser: Genome Allese Guide</TITLE>
        <LINK REL="STYLESHEET" HREF="../../style/HGStyle.css">
        <script type="text/javascript" src="/js/jquery.js"></script>
        <script type="text/javascript" src="/js/staticDoc.js"></script>
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../../images/title.jpg"></TH></TR>
    <TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../../inc/help.topbar.html"--></TD></TR>

<TR><TD>    
<!-- - - - - - - Genome Haplotype Alleles Section - - - - - - -->
<!-- - - IMPORTANT:  Page details are expected to change with each 1000 Genomes release - - - -->
<A NAME="Topic5"></A>

        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
                <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                    <TR><TD>
                        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
                            <TR><TD><FONT SIZE="4"><A
                            NAME="TOC"></A><B>&nbsp; Gene Haplotype Alleles</B></FONT>
                            </TD></TR>
                        </TABLE>
                        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                            <TR><TH HEIGHT=10></TH></TR>
                            <TR><TD WIDTH=10>&nbsp;</TD><TD>

<div id='haplotypes'> <!-- Allows using same CSS as the haplotypes aplet itself  -->
<P>
With the availability <A HREF='http://www.1000genomes.org/' TARGET=_BLANK>1000 Genomes</A> Phase 1
data, there are currently 1092 sets of variation in humans phased across chromosomes.  Using 
these data, we can now see the common <em class='textInfo'>haplotypes</em> (a.k.a alleles), 
or sets of variations occurring on the same chromosome, falling within the (exon) boundaries of 
protein coding genes as they are seen in healthy adults.  
While many individual variants have been studied, the effect of one SNP may not be fully 
appreciated until it is seen in the context of co-segregating variants falling within the 
same gene. For any 
given gene there can be no more than 2184 distinct haplotypes given the data, 
though in most cases there are dramatically fewer (average 4).  Many haplotypes will have 
little or no phenotypic effect, being defined by synonymous SNPs or resulting in amino acid 
substitutions of little consequence to the overall efficacy of a protein.  It is expected, 
however, that many haplotypes will result in significant phenotypic variation within humans.  
Haplotypes associated with disease may show evidence of negative selection, just as haplotypes 
associated with the immune system may show striking evidence for positive selection.  Occasionally 
the skew in a haplotype's distribution among populations will give hints to its anthropological 
history.   However, the vast majority of common haplotypes are represented in all major 
population groups, which may speak both to the age of individual haplotypes and the ongoing 
mixing of the human gene pool.  
</P> 

<H2>How to use the &quot;Gene Haplotype Alleles&quot; section</H2>

<P> 
Click on any protein-coding gene in the UCSC Genes track and scroll to the <B>Common Gene 
Haplotype Alleles</B> section. (The feature is currently implemented only on GRCh37/hg19 
protein-coding genes.)  There will be a table of haplotypes for the protein-coding 
portion of the gene.  Each row in the table represents a unique gene haplotype as found in 
the 1000 Genomes Phase 1 project data.  The table is sortable on any column by clicking on 
the column headers.
</P> 

<dl>
<dt><B class='textOpt'>Haplotype Frequency</b></dt>
<dd>Haplotype frequencies are based upon all releveant chromosomes in the data set.  The total
    number is almost always 2,184 (1,092 for Y and location-dependent for X).  Hover over
    the frequency calculations to show the number of a particular haplotype in the dataset 
    (<em>e.g.</em>, <span title='N=1327 of 2184'>"N=1327 of 2184"</span>).  
    If appropriate, the homozygous frequency will also be shown and will reflect the number 
    of individuals in the dataset.
    <BR><BR></dd>

<dt><B class='textOpt'>Variants</b></dt>
<dd>By default, only non-synonymous, <em class='textInfo'>common variants</em> are displayed. 
    Common variants occur in at least 1&#37; of 1000 Genomes subject chromosomes. Pressing 
    <input type='button' style='font-size:.9em;' value='Include all variants' 
    title='Include or Restrict to non-synonymous, common variants with a frequency of at least 1%' 
    onclick='return false;'> 
    will replace the list of haplotypes by the list of <em>all</em> haplotypes found in 
    1000 Genomes subjects, though many of these haplotypes may have little or no phenotypic
    effect.  To return to haplotypes defined by common and non-synonmous variation, press
    <input type='button' style='font-size:.9em;' value='Common variants only' 
    title='Include or Restrict to non-synonymous, common variants with a frequency of at least 1%' 
    onclick='return false;'>.
    <BR><BR>
    
    By default, all variants are displayed as their predicted effect on amino acid sequence.  
    Press <input type='button' style='font-size:.9em;' value='Display as DNA bases'
    title='Display variants and sequence as amino acids or DNA bases' onclick='return false;'>
    to view the variants as DNA bases instead.
    Return to the protein sequence view by pressing 
    <input type='button' style='font-size:.9em;' value='Display as amino acids' 
    title='Display variants and sequence as amino acids or DNA bases' onclick='return false;'>.
    The <em class='textInfo'>reference variant</em> is shown at the top of each variant column.  
    This is the value found in the GRCh37/hg19 reference genome.  In most cases it is a single 
    letter (AA code/DNA base).  In the case of an insertion with respect to the reference genome, 
    the reference value will be seen as '-' .  Large deletions will be represented by the first 
    two sequence letters followed by &quot;+++&quot;.  Note that hovering the pointer over any 
    of these variants will show a more complete description (example: 
    <span title='AA:16 A|T chr9:136137554 SNP: G|A (0.995|0.005) rs55917063'>
    'AA:16 A|T chr9:136137554 SNP: G|A (0.995|0.005) rs55917063'</span>).
    The description includes the <span title='AA:16 A|T'>AA residue number and variants 
    (AA view only)</span>, <span title='chr9:136137554'>genome location</span>,
    <span title='SNP: G|A (0.995|0.005)'>base level variants and frequencies</span>, 
    and the <span title='rs55917063'>dbSNP variant name</span> if one is known.  Clicking on any 
    <em class='textInfo'>non-reference</em> variant shown in the variants section of the table 
    will link to the full details of that variant as seen in the 1000 Genomes phase 1 track.    
    <BR><BR></dd>

<dt><B class='textOpt'>Full sequence</b></dt>
<dd>Press <input type='button' style='font-size:.9em;' value='Show full sequence'
    title='Show/Hide predicted full sequence of each gene haplotype' onclick='return false;'>
    to see the predicted effects of variation on gene sequence for each of the haplotypes.
    If variants are currently displayed as DNA bases, then the predicted DNA coding (exons only) 
    sequence is shown.  If variants are displayed as amino acids, the predicted protein sequence
    is shown and you may optionally press 
    <input type='button' style='font-size:.9em;' value='Show DNA triplet code'
    title='Show/Hide DNA sequence above amino acid sequence' onclick='return false;'>
    to simultaneously show the DNA sequence above the protein sequence for easy comparison.
    Showing protein sequence with the DNA triplets is the easiest way to visualize the 
    synonymous variants.  Press 
    <input type='button' style='font-size:.9em;' value='Hide DNA triplet code'
    title='Show/Hide DNA sequence above amino acid sequence' onclick='return false;'>
    to revert to the simplified protein sequences view and press
    <input type='button' style='font-size:.9em;' value='Hide full sequence'
    title='Show/Hide predicted full sequence of each gene haplotype' onclick='return false;'>
    to simplify the haplotype table further.  
    <BR><BR>
   
    When the full sequence is displayed, columns with variants are highlighted by 
    <span style='background-color:#CCFFAA;'>green vertical lines</span>. The effects of variants are 
    highlighted by <B class='textAlert'>bold red letters</B>.  Synonymous changes are only 
    evident when DNA bases are displayed. Each haplotype allele sequence is generated from 
    GRCh37/hg19 reference DNA, with variants spliced in, then translated into amino acids.  
    Note that sorting on a variant column will highlight that one variant with a 
    <span style='background-color:#CCFFFF;'>blue vertical line</span> 
    in blue in the full sequence.  This can be used to quickly locate one variant out of many.
    In AA view, the AA residue is shown when hovering over any part of the sequence.
    <BR><BR></dd>

<dt><B class='textOpt'>Rare haplotypes</b></dt>
<dd>By default, only common haplotype alleles are displayed. Common haplotypes occur in at 
    least 1&#37; of 1000 Genomes subject chromosomes.  Press
    <input type='button' style='font-size:.9em;' value='Show rare haplotypes' 
    title='Show/Hide rare haplotypes that occur with a frequency of less than 1%' 
    onclick='return false;'>     
    to see all haplotypes and then press 
    <input type='button' style='font-size:.9em;' value='Hide rare haplotypes' 
    title='Show/Hide rare haplotypes that occur with a frequency of less than 1%' 
    onclick='return false;'>     
    to hide the rare ones again.  
    <BR><BR></dd>

<dt><B class='textOpt'>Populations</b></dt>
<dd>Each haplotype is found in one or more subjects participating in phase 1 of the 
    1000 Genomes project.  The distribution of the haplotypes across different population 
    groups can be examined by pressing 
    <input type='button' style='font-size:.9em;' value='Show populations'
    title='Show/Hide haplotype distribution across population groups' onclick='return false;'>. 
    Press
    <input type='button' style='font-size:.9em;' value='1000 Genome groups' 
    title='Show haplotype distribution across 1000 Genome population groups' 
    onclick='return false;'> 
    to display the distribution across a broader spectrum of groups 
    (<A HREF='#population'>defined below</A>).  
    Press 
    <input type='button' style='font-size:.9em;' value='Major groups' 
    title='Show haplotype distribution across major population groups' 
    onclick='return false;'> 
    to return to the major grouping and 
    <input type='button' style='font-size:.9em;' value='Hide populations' 
    title='Show/Hide haplotype distribution across population groups' onclick='return false;'> 
    to hide these data again.
    <BR><BR></dd>

<dt><B class='textOpt'>Scoring</b></dt>
<dd>By default, scoring is hidden. Three types of scores are provided to help users find 
    haplotype alleles that occur more or less frequently than expected or that have unusual 
    distributions in populations. 
    <A HREF='#scoring'>See definitions below</A>.  
    To see the calculated scores, press
    <input type='button' style='font-size:.9em;' value='Show scoring' 
    title='Show/Hide all haploptye scores' onclick='return false;'>.  
    <BR><BR></dd>
</dl>

<h2 id='population'>Population Group definitions</h2>

<h3>Major Groups</h3>
<div class='textOpt'>Includes only major groups for which there are data in phase 1 of the 
1000 Genomes project.</div>
<table>
     <tr><td>AFR</td><td>African          </td><td>246 individuals</td></tr>
     <tr><td>AMR</td><td>Ad Mixed American</td><td>181 individuals</td></tr>
     <tr><td>ASN</td><td>East Asian       </td><td>286 individuals</td></tr>
     <tr><td>EUR</td><td>European         </td><td>379 individuals</td></tr>
<!-- <tr><td>SAN</td><td>South Asian      </td><td>0 individuals</td></tr>  NOT YET -->
</table>

<h3>1000 Genomes Groups</h3>
<div class='textOpt'>Includes only 1000 Genomes groups for which there are data in phase 1 
of the project.</div>
<table>
    
     <tr><td></td><td>African:</td><td></td></tr>
     <tr><td>ASW</td><td>African Ancestry in Southwest US       </td><td>61 individuals</td></tr>
<!-- <tr><td>ESN</td><td>Esan in Nigeria                        </td><td> 0 individuals</td></tr> -->
     <tr><td>LWK</td><td>Luhya in Webuye, Kenya                 </td><td>97 individuals</td></tr>
<!-- <tr><td>GWD</td><td>Gambian in Western Division, The Gambia</td><td> 0 individuals</td></tr> -->
<!-- <tr><td>MSL</td><td>Mende in Sierra Leone                  </td><td> 0 individuals</td></tr> -->
     <tr><td>YRI</td><td>Yoruba in Ibadan, Nigeria              </td><td>88 individuals</td></tr>

     <tr><td></td><td><BR>Ad Mixed American:</td><td></td></tr>
<!-- <tr><td>ACB</td><td>African Caribbean in Barbados              </td><td> 0 individuals</td></tr> -->
     <tr><td>CLM</td><td>Colombian in Medellin, Colombia            </td><td>60 individuals</td></tr>
     <tr><td>MXL</td><td>Mexican Ancestry in Los Angeles, California</td><td>66 individuals</td></tr>
<!-- <tr><td>PEL</td><td>Peruvian in Lima, Peru                     </td><td> 0 individuals</td></tr> -->
     <tr><td>PUR</td><td>Puerto Rican in Puerto Rico                </td><td>55 individuals</td></tr> 

     <tr><td></td><td><BR>East Asian:</td><td></td></tr>
<!-- <tr><td>CDX</td><td>Chinese Dai in Xishuangbanna, China       </td><td>  0 individuals</td></tr> -->
     <tr><td>CHB</td><td>Han Chinese in Beijing, China             </td><td> 97 individuals</td></tr>
<!-- <tr><td>CHD</td><td>Chinese in Denver, Colorado (pilot 3 only)</td><td>  0 individuals</td></tr> -->
     <tr><td>CHS</td><td>Han Chinese South                         </td><td>100 individuals</td></tr>
     <tr><td>JPT</td><td>Japanese in Tokyo, Japan                  </td><td> 89 individuals</td></tr>
<!-- <tr><td>KHV</td><td>Kinh in Ho Chi Minh City, Vietnam         </td><td>  0 individuals</td></tr> -->

     <tr><td></td><td><BR>Europeans:</td><td></td></tr>
     <tr><td>CEU</td><td>Utah residents with Northern and Western European ancestry</td><td>85 individuals</td></tr>
     <tr><td>FIN</td><td>Finnish in Finland                          </td><td>93 individuals</td></tr>
     <tr><td>GBR</td><td>British in England and Scotland             </td><td>89 individuals</td></tr>
     <tr><td>IBS</td><td>Iberian populations in Spain                </td><td>14 individuals</td></tr>
     <tr><td>TSI</td><td>Toscani in Italia                           </td><td>98 individuals</td></tr>
    
<!-- <tr><td></td><td><BR>South Asian:</td><td></td></tr> -->
<!-- <tr><td>BEB</td><td>Belgali in Bangladesh        </td><td>0 individuals</td></tr> -->
<!-- <tr><td>GIH</td><td>Gujarati Indian in Houston,TX</td><td>0 individuals</td></tr> -->
<!-- <tr><td>ITU</td><td>Indian Telugu in the UK      </td><td>0 individuals</td></tr> -->
<!-- <tr><td>PJL</td><td>Punjabi in Lahore,Pakistan   </td><td>0 individuals</td></tr> -->
<!-- <tr><td>STU</td><td>Sri Lankan Tamil in the UK   </td><td>0 individuals</td></tr> -->
</table> 

<h2 id='scoring'>Scoring definitions</h2>
<dl>
<dt>Hap score</dt>
<dd>The &quot;haplotype score&quot; is the normalized (-log10) probability of finding exactly 
    N subject chromosomes with this haplotype, given the frequencies of individual variants. 
    The score is normalized by multiplying by the total number of variants. Normalization allows 
    comparing the scores between genes with many variants and those with few. The score will 
    be positive if the haplotype is more frequent than expected by chance and negative if 
    less frequent.  For haplotyes made from common, non-synonymous variants, haplotype scores 
    above 606 are seen in only 2&#37; of genes.  Likewise, a score of less than -199 is only seen 
    in 2&#37; of genes.</dd> 

<dt>Hom score</dt>
<dd>The &quot;homozygous score&quot; is the (-log10) probability of finding exactly N individuals 
    with this haplotype on both chromosomes, given the <em>actual</em> frequency of the haplotype 
    in subject chromosomes. The score will be positive if the haplotype is found homozygous in 
    more individuals than expected and negative when found in fewer individuals than expected.
    Negative values might suggest that the haplotype is deleterious when homozygous.  For 
    haplotyes made from common, non-synonymous variants, homozygous scores above 92 are seen 
    in only 2&#37; of genes.  Likewise, a score of less than -15 is only seen in 2&#37; 
    of genes.</dd> 

<dt>Pop score</dt>
<dd>(only visible when population distributions are displayed) The &quot;population score&quot; 
    is the variance between population groups divided N, the number occurrences of the haplotype. 
    The most frequently occurring haplotypes will potentially have larger scores, but if N is 
    small, a skew in population distribution is not unexpected.  Polulation scores are always 
    greater than or equal to zero.  For haplotyes made from common, non-synonymous variants, 
    population scores above 32.6 are seen in only 2&#37; of genes and scores above 37 are 
    seen in only 1&#37; of genes.</dd>
</dl>

<div id='notes' class='redBox' style:'margin:0px;'>
<span class='textInfo'>Notes:</span>
<ol style='margin:0px;'>
<li id='NoteAntisense'>
    If the gene is on the negative ('-' or &quot;antisense&quot;) strand, all variants and 
    sequences will be presented with respect to the negative strand. Note that individual 
    variants displayed in the 1000 Genomes phase 1 variations track are all shown as they 
    appear on the positive ('+' or "sense") strand. </li>
<li id='noteHapId'>Hovering within the table column with the header "Reference Varaints:" will 
    give an <span title='uc004cda.1:jfa'>ID</span> for each non-reference haplotype.  This 
    <span title='uc004cda.1:jfa'>ID</span> is unique across all haplotypes across the entire 
    genome given the set of variants used.  That is, haploptyes produced from common variants 
    will not have the same <span title='uc004cda.1:jfa'>ID</span> when seen with all 
    variants.  The <span title=':jfa'>ID suffixes</span> of a set of haplotypes for a single 
    gene model will reflect the 
    <span title='uc004cda.1:jfa includes all the variants of uc004cda.1:jf'>relationships</span> 
    of the sets variants.</li>
<li id='noteVarsIncluded'>
    Haplotype and homozygous frequency calculations depend upon which variants are included.</li>
<li id='noteNoSplice'>
    Only variants occurring within coding exons are currently included in haplotypes. 
    Variants occurring within intron splice junctions are not included.</li>
<li id='noteTooManyHaps'>
    Some large gene models cover many variants and therefore have a very large number of distinct 
    haplotypes represented in the 1000 Genomes project data.  If this is the case, only the 100 
    most frequently occurring haploptyes, will be shown in the table, though the true number will 
    be noted. </li>
<li id='noteTooManyVars'>
    If the number of variants covered exceeds 200, then the haplotype table will not be 
    displayed and the reason so noted. </li>
<li id='noteRefHap'>
    The <em class='textInfo'>reference haplotype</em> (made of entirely reference variants) 
    may not be represented in the 1000 Genomes data. If there is, it will be marked as 
    &quot;reference&quot; in the table of haplotypes. </li>
<li id='noteScoring'>
    Scores alone cannot be used to draw definitive conclusions about any haplotype.</li>
<li id='noteVcf'>
    The haplotypes displayed are not pregenerated but are derived from 1000 Genomes VCF files 
    and other Genome Browser dataset at the time they are requested.  Consequently, scoring is
    calculated in the context of a single gene model and the variants used to derive haplotypes.
    </li>
</ol> 
</div>


</div> <!-- Allows using same CSS as the haplotypes aplet itself  -->

                                </TD>
                                <TD WIDTH=15></TD>
                            </TR>
                        </TABLE>
                    <BR></TD></TR>
                </TABLE>
            </TD></TR>
        </TABLE>
    </TABLE>
</BODY></HTML>

