<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: User's Guide</TITLE>
	<LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="/images/title.jpg"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include file="../../inc/topbar.html"--></TD></TR>
<!-----------mRNA Codon Coloring Help Page------------------------------>
<A NAME="Topic5"></A><BR>

	<!--outer table is for border purposes-->
	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
	    <TR><TD>	
    		<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
		    <TR><TD>	
			<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
			    <TR><TD><FONT SIZE="4"><A
                            NAME="TOC"></A><B>&nbsp; Coloring mRNAs by codon</B></FONT>
			    </TD></TR>
			</TABLE>
			<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
			    <TR><TH HEIGHT=10></TH></TR>
			    <TR><TD WIDTH=10>&nbsp;</TD><TD>
				<P>
                                The Genome Browser's codon coloring feature allows
			 	users to quickly validate and compare mRNAs. 
				To turn on codon coloring, select the
				desired option from the <em>Color 
				track by codons</em> pull-down menu. 
                                All mRNA, blastz mRNA, and xeno mRNA tracks may 
				be colored by codon. 
                                <a href="hgCodonColoring.html">Gene prediction 
				tracks</a> can also be colored, 
                                but only by genomic codons.
                                <P>
				<B>Coloring Options</B><BR>
                                <em>genomic codons</em> - Each mRNA is colored 
				according to its alignment with the target genome.				  The codons and the frame are based on the target
                                sequence only and may differ from the NCBI record.
				<P>
                                <em>mRNA codons</em> - Each mRNA is colored 
				according to the query sequence. The frame 
				information takes into account 
				&quot;hidden gaps&quot; in the query sequence 
				that aren't shown in the browser. The codons
				are colored and labeled by the mRNA sequence 
				rather than the genomic sequence to which it 
				aligns.
				<P>
                                <em>different mRNA codons</em> - Similar to the 
				<em>mRNA codons</em> option, the frame and
                                color are based on the mRNA itself. However, only 
				those mRNA codons that differ between the mRNA and
				the target genomic sequence are labeled.
				<P>
                                <em>mRNA bases</em> - The frame and coloring are 
				based on the mRNA, but the mRNAs bases are
				shown rather than the amino acids.
				<P>
                                <em>different mRNA bases</em> - Same as the 
				previous option, but only those mRNA bases that 
				differ between the genome and the mRNA are printed.
                                <P> 
				<B>How are the mRNAs colored?</B><BR>
                                The CDS for each mRNA is obtained from GenBank.
                                Starting at the <em>cdsStart</em> position of a
				prediction, each codon is colored 
                                and labeled (when zoomed-in closely enough) until 
				the <em>cdsEnd</em> position.
                               	Those on the negative strand are labeled and 
				colored similarly, except in <em>mRNA bases</em> 
				and <em>different mRNA bases</em> mode where the 
				bases are always shown with respect to the 
				positive strand. 
                                <P>
				Start codons (the first methionine) are indicated 
				in green, stop codons are shown in red, and 
				insertions are colored orange. All 
				other codons are shown in the original track 
				color, with codon boundaries indicated by 
				alternating lighter and darker shades. 

                                <P>
				Codons are labeled by the amino acid they 
				represent or the bases comprising the codon,
				depending on the display mode. Codons on the
				boundaries of a gap are labeled 
				at both ends of 
				the gap; the cyan color of the label text 
				indicates a partial codon.
                                Insertions are also labeled in cyan to designate 
				a partial codon. Stop codons are labeled with '*';
			 	errors are represented by 'X' (e.g. a truncated 
				last codon).
                                <P>
                                <B>When are the mRNAs colored?</B><BR>
                                To view codon coloring, the option must be is 
				turned on, the track must be in squish, pack, or
				full display mode, and the browser must be 
				zoomed-in to a level at which there is at least 
				one pixel per codon. The zoom-level requirement
				varies with the image width of the browser.
				At the default <em>image width</em> value of 620 
				pixels, the annotation tracks window must be
				zoomed-in to at least 1400 bp or fewer to view the
				codon coloring. To view the labeling, the
				tracks image must be zoomed to within 3 times 
				the base level. A gene prediction with an 
				invalid CDS (such as 0,0) will not be colored,
                                except in <em>mRNA bases</em> or <em>different 
				mRNA bases</em> display mode where labeling 
				occurs only at the base level. 
                                An mRNA with an incomplete GenBank CDS at either 
				end, an invalid GenBank CDS (such as 0,0), or no 
				CDS will not be colored or labeled.
			        </TD>
			        <TD WIDTH=15></TD>
			    </TR>
			</TABLE>
		    <BR></TD></TR>
		</TABLE>
	    </TD></TR>
	</TABLE>
    </TABLE>
</BODY></HTML>
