<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>

<HEAD>
  <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
  <META name="description" content="UCSC Genome Graphs">
  <META name="keywords" content="genome, genome data, table browser, genome browser, genome assembly, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA">
  <TITLE>UCSC Genome Browser: Genome Graphs User's Guide</TITLE>
  <LINK REL="STYLESHEET" HREF="../../style/HGStyle.css">
</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
  <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../../images/title.jpg"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../../inc/topbar.html"--></TD></TR>
<!--Content Tables-->  
  <TR><TD CELLPADDING=10 ROWSPAN=2 VALIGN="top">
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>	
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
          <TR><TD>	
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	        <FONT SIZE="4"><B>&nbsp;Genome Graphs User's Guide </B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
   	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
              <A NAME="TOC"></A>
	        <H3> Table of Contents: </H3>
	 	<UL><B>
  		<LI><A CLASS="toc" HREF="#Introduction">Introduction</A>
        	<LI><A CLASS="toc" HREF="#FormatUpload">Formatting &amp; Uploading Data</A>
	   	  <UL>
                  <LI><A CLASS="toc" HREF="#Format">Formatting Data</A>
                  <LI><A CLASS="toc" HREF="#Upload">Uploading Data</A>
		  </UL>
		<LI><A CLASS="toc" HREF="#Sample">Quick Start</A>
        	<LI><A CLASS="toc" HREF="#Displaying">Displaying Data in Genome Graphs</A>
	   	  <UL>
                  <LI><A CLASS="toc" HREF="#Configure">Configuring the Display</A>
                  <LI><A CLASS="toc" HREF="#SigThreshold">Setting a Significance Threshold</A>
                  <LI><A CLASS="toc" HREF="#DataRegion">Setting a Data Region</A>
		  </UL>
        	<LI><A CLASS="toc" HREF="#ViewInGB">Viewing Data in the Genome Browser</A>
        	<LI><A CLASS="toc" HREF="#ViewInGS">Viewing Data in the Gene Sorter</A>
        	<LI><A CLASS="toc" HREF="#DataSets">Deleting Data</A>
        	<LI><A CLASS="toc" HREF="#Correlate">Correlating Data</A>
		</B></UL>
		<P>

		<A HREF="mailto:&#103;&#101;&#110;&#111;&#109;e&#64;&#115;&#111;&#101;.&#117;&#99;sc.e&#100;&#117;">Questions 
                and feedback on this User's Guide are welcome.</A>

<!---  enable this after there are a few Genome Graphs questions in the archive
 
                &nbsp;User questions and answers are archived in the 
                <A HREF = "https://www.soe.ucsc.edu/pipermail/genome"
                TARGET = _BLANK> public archive</A>.
              
                <FORM name="googleForm2" method="GET" action="http://www.google.com/search" onSubmit="document.googleForm2.q.value=document.googleForm2.qq.value+'   site:www.cse.ucsc.edu/pipermail/genome';">
	        Search the Genome mailing list archives: &nbsp;
	        <INPUT TYPE="hidden" NAME="q" VALUE="">
                <INPUT TYPE="hidden" NAME="num" VALUE="10">
                <INPUT TYPE="hidden" NAME="filter" VALUE="0">
                <INPUT TYPE=text NAME=qq size=30 MAXLENGTH=255 VALUE="Genome Graphs">
                <INPUT TYPE="submit" VALUE="Submit">
                </FORM>
-->

	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
  <P>

<!------Introduction-------------------------->
<A NAME="Introduction"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
          <TR><TD>
 	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	    	<FONT SIZE="4"><B>&nbsp; Introduction </B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
		<P>
		Genome Graphs is a tool for displaying genome-wide data sets 
		such as the results of genome-wide SNP association studies, 
		linkage studies and homozygosity mapping.</P>
		<P>
		Using the Genome Graphs tool, you can:
                <UL>
                <LI> upload several sets of genome-wide data and display
                     them simultaneously</LI>
                <LI> click on an area of interest and go directly to the
                     genome browser at that position</LI>
                <LI> set a significance threshold for your data and view
                     only regions that meet that threshold</LI>
                <LI> view the genes that exist in areas where your data
                     meet your significance threshold</LI>
                </UL>

		<P>
		To return to Genome Graphs from any other location on the 
		Genome Browser website, use your browser's Back button, or 
		press <EM>Home</EM> on the blue navigation bar, then 
		press the <EM>Genome Graphs</EM> link.
		</P>

		<P>
		This User's Guide is aimed at both the novice Genome Graphs user
		as well the advanced user. If you are new to the Genome Graphs 
		tool, read the <A HREF="#Sample">Quick Start</A> section to 
		learn about the basics using some sample data. Advanced users 
		may want to proceed directly to the section that addresses 
		a particular area of functionality in detail.</P>
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>

<!-------Format & Upload-------------------------->
<A NAME="FormatUpload"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
          <TR><TD>
 	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	    	<FONT SIZE="4"><B>&nbsp; Formatting &amp; Uploading Data </B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
		<P>
		</P>

<!-------Formatting Data------------------------->
<A NAME="Format"></A>
	        <B><FONT SIZE="3" COLOR="#0000FF">Formatting Data</FONT></B>
		<P>
		Genome Graphs allows you to upload data from files that reside 
		on your computer. Several file formats are accepted by the 
		program.  For all formats there is a single line for each marker. 
		Each line starts with information on the marker, and ends with 
		the numerical values associated with that marker. The markers 
		can be of one of the following types:
		</P>

		<table>
		<tbody><tr><td valign="top" width="20">&mdash;</td><td><b>chromosome 
		base: </b> e.g. <TT>chr1 130000</TT> (Note that the first base 
		in a chromosome is considered position 0.) 
		</td></tr>

		<tr><td valign="top" width="20">&mdash;</td><td><b>STS Marker: </b>
		e.g. <TT>RH75228</TT>
		</td></tr>

		<tr><td valign="top" width="20">&mdash;</td><td><b>dbSNP rsID: </b>
		e.g. <TT>rs12345</TT>
		</td></tr>

		<tr><td valign="top" width="20">&mdash;</td><td><b>Affymetrix 500k 
		Gene Chip: </b>e.g. <TT>SNP_A-1780270</TT>
		</td></tr>

		<tr><td valign="top" width="20">&mdash;</td><td><b>Illumina HumanHap300 
		Bead Chip: </b>e.g. <TT>rs12345</TT>
		</td></tr>
		</table>

		<P>
		The marker-value pairs in each line of the file can be separated 
		with a single space, a tab, or a comma. The file can 
		contain multiple values for each marker.  In that case, a 
		separate graph will be created for each value column in the 
		input file.
		</P>

		<P>
		For example, <B>chromosome base</B> markers with only one 
		value associated with the marker would be entered like this:
		</P>

		<TT>chrX 100000 1.23</TT>

		<P>
		<B>dbSNP rsID</B> markers with two values associated 
		with the marker would be entered like this:
		</P>

		<TT>rs10218492 0.384 0.882</TT>

		<P>
		The Genome Graph program will map the marker IDs to the genome. 
		In cases where the marker maps to more than one location in 
		the genome, the value(s) in your input file will be associated 
		with each location.
		</P>

		<P>
		Note that markers can only be mapped to assemblies for which 
		there already exists a track of the type that contains your 
		marker type. You can not, for example, use dbSNP rsID markers 
		for the cow genome, as it does not have a SNP track.
		</P>
	        <P>

<!-------Uploading Data---------------------------->
<A NAME="Upload"></A>
	        <HR>
	        <B><FONT SIZE="3" COLOR="#0000FF">Uploading Data</FONT></B>
	        <P>
		Once you have created your input file, you must upload it to 
		Genome Graphs. From the main Genome Graphs page, choose your 
		clade, genome, and assembly to which your data pertains.
		If you are unsure of the UCSC assembly name, you can check 
		<A HREF="../../FAQ/FAQreleases#release1">this</A> page. 
		Now, press the upload button to go to the upload page.
		</P>
		<P>
		To upload a file in any of the supported formats, locate the 
		file on your computer using the controls next to <B>file 
		name</B>, and then submit. The other controls on this form are 
		optional, though filling them out will sometimes enhance the 
		display. In general the controls that default to &qout;best 
		guess&quot; can be left alone, since the guess is almost always 
		correct. 
		The controls for display min and max values and connecting 
		lines can be set later via the configuration page as well. 
		Here is a description of each control.
		</P>
<A NAME="Controls"></A>

		<UL>
		<LI><B>name of data set:</B> Displayed in graph drop-down in 
		Genome Graphs and as the track name in Genome Browser. Only 
		the 
		first 16 characters are visible in some contexts. For data 
		sets with multiple graphs, this is the first part of the 
		name, shared with all members of the data set.</LI>

		<LI><B>description:</B>  A short sentence describing the 
		data set. Displayed in the Genome Graphs and Genome Browser 
		configuration pages, and as the center label in the Genome 
		Browser.</LI>

		<LI><B>file format:</B> Controls whether the upload file is 
		a tab-separated, comma-separated, or space separated table.
		</LI>

		<LI><B>markers are:</B> Describes how to map the data to 
		chromosomes. The choices are that either the first column 
		of the 
		file is an ID of some sort, or the first column is a 
		chromosome 
		and the next a base. The IDs can be SNP rs numbers, STS marker 
		names or ID's from any of the supported genotyping platforms.
                </LI>

		<LI><B>column labels:</B> Controls whether the first row of 
		the 
		upload file is interpreted as labels or data. If the 
		first row contains text in the numerical fields, or if the 
		mapping fields are empty, it is interpreted by &quot;best 
		guess&quot; as labels. This is generally correct, but you can 
		override this interpretation by explicitly setting the control.
		</LI>

		<LI><B>display min value/max value:</B> Set the range of the 
		data set that will be plotted. If left blank, the range will 
		be taken from the min/max values in the data set itself. For 
		all data sets to share the same scale, you will usually need 
                to set this.</LI>

		<LI><B>label values:</B> A comma-separated list of numbers 
		for the vertical axis. If left blank, the axis will be 
		labeled at the 1/3 and 2/3 points of your data range.</LI>

		<LI><B>draw connecting lines:</B> Lines are drawn connecting 
                data points that are separated by this number of bases or 
		fewer.</LI>

		<LI><B>file name:</B> The controls here let you select 
		which file on your computer to upload.</LI>
		<UL>


		</P>
		<P>
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>


<!-------Quick Start------------------------->
<A NAME="Sample"></A>
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
<TR><TD>
 <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
 <TR><TD>
  <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
  <TR><TD>
  <FONT SIZE="4"><B>&nbsp; Quick Start </B></FONT>
  </TD></TR>
  </TABLE>
   <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
   <TR><TH HEIGHT=10></TH></TR>
   <TR><TD WIDTH=10>&nbsp;</TD><TD>
   <P>
   The Genome Graphs tool does not come pre-loaded with any sample data.  Use 
   the examples in this section of the User's Guide to get a feel for how the 
   tool works. Refer to other <A HREF="#TOC">sections</A> 
   in this User's Guide for details and instructions for more advanced features.
   </P>

<!-------Example #1---------------------------->
<A NAME="example1"></A>
 <HR>
 <B><FONT SIZE="3" COLOR="#0000FF">Example #1 &mdash; SNPs on chr22</FONT></B>
   <P>
   Follow these steps to display in Genome Graphs all of the highest quality 
   SNPs on chromosome 22 for the hg18 assembly whose predicted functional 
   role is &quot;coding non-synonymous&quot; 
   (where there is a change in the peptide for the allele with respect to 
   the reference assembly). Note that there are no SNPs on the p-arm of 
   chromosome 22. 
   </P>
   <P>
   This data set is <A HREF="#Format">formatted</A> in the <TT>marker value</TT> 
   style. The <TT>markers</TT> are <B>dbSNP rsIDs</B>. The associated 
   <TT>value</TT> is <B>1</B> if the SNP is on the positive strand, and 
   <B>-1</B> if the SNP is on the negative strand. Here are the first 
   ten rows of the data file:<BR><BR>

<TT><PRE>
rs2154787       1
rs12160578      1
rs8140723       1
rs8141435       1
rs2186477       1
rs2508089       1
rs7292455       1
rs7292597       1
rs11705636      1
rs2261629       1
</TT></PRE>

   Note that if you simply cut and paste these ten rows into Genome Graphs, 
   it may not work well as these have been formatted in HTML to display 
   properly on this web page. Click <A HREF="hgGenome_example1.txt">here</A> 
   for a copy of the entire data set for this example that can be uploaded 
   into Genome Graphs without editing.
   </P>

   <P><B>Step 1. Upload the data into the Genome Graphs tool</B><BR>
   Copy the entire sample <A HREF="hgGenome_example1.txt">data set</A> 
   into a text editor and save the file to your computer. This data 
   set is associated with the most recent human assembly: hg18 
   (Mar. 2006). Be sure to 
   configure the Genome Graphs tool to use the hg18 assembly 
   like so:<BR><BR>

<PRE>
<B>clade:</B>		Vertebrate
<B>genome:</B>		Human
<B>assembly:</B>	Mar. 2006
</PRE>
    
   <A HREF="#Upload">Upload</A> the file into the Genome Graphs tool. 
   You can configure each <A HREF="#Controls">control</A> on the upload 
   page, or just leave them set to their default values.<BR>
   The upload process may take some time, as the program is actually 
   mapping each rsID in the input file to its location(s) in the genome.
   </P>

   <P><B>Step 2. Display the graph in Genome Graphs</B><BR>
   Now that your input file has been uploaded to the server, you will want 
   to display it in the Genome Graphs tool. To display your uploaded data, 
   simply choose the graph name from the 
   <B>graph</B> drop-down list, then choose the color in which you 
   would like it to be displayed from the <B>in</B> drop-down list. 
   Your graph will be displayed directly above the chromosomes 
   in Genome Graphs. You should see the data plotted directly 
   above chromosome 22.
   </P>

   <P><B>Step 3. View the graph in the Genome Browser</B><BR>
   From the Genome Graphs display, press anywhere on the graph 
   or on chromosome 22 to open the
   Genome Browser for hg18 centered at that location on chr22. 
   The graph will be drawn as a track near the top of the Genome
   Browser display. 
   </P>
   <P>


<!-------Example #2--------------------------***
<A NAME="example2"></A>
 <HR>
 <B><FONT SIZE="3" COLOR="#0000FF">Example #2</FONT></B>
   <P>
   </P>
   <P>
   </P>
   <P>

*** ADD EXAMPLE #2 WHEN READY -->

   <P>
   </TD><TD WIDTH=15></TD></TR>
   </TABLE>
   <BR></TD></TR>
 </TABLE>
 </TD></TR>
</TABLE>
<P>


<!-------Displaying Data in Genome Graphs-------------------------->
<A NAME="Displaying"></A>
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
<TR><TD>
 <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
 <TR><TD>
  <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
  <TR><TD>
  <FONT SIZE="4"><B>&nbsp; Displaying Data in Genome Graphs </B></FONT>
  </TD></TR>
  </TABLE>
   <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
   <TR><TH HEIGHT=10></TH></TR>
   <TR><TD WIDTH=10>&nbsp;</TD><TD>
   <P>
   Once you have uploaded your data, you will want to display it in the 
   Genome Graphs tool. To display your uploaded data, simply choose the 
   graph name from the 
   <B>graph</B> drop-down list, then choose the color in which you 
   would like it to be displayed from the <B>in</B> drop-down list. 
   Your graph will be displayed directly above the chromosomes 
   in Genome Graphs. Read on to learn how to 
   <A HREF="#Configure">customize</A> the display.
   </P>
   <P>

<!-------Configuring the Display------------------------->
<A NAME="Configure"></A>
                <HR>
                <B><FONT SIZE="3" COLOR="#0000FF">Configuring the Display</FONT></B>
                <P>
		<B>Configuring the graphs display</B><BR>
		To go to the configuration page, press the <EM>configure</EM> button
		on the main Genome Graphs page.  This is the page from which you can
		configure many overall aspects of the Genome Graphs display.
		Individual graphs can also be configured (see the next section
		for help on that).
		</P>

		<P>
		On this page you will find the following controls:
		<UL>
		<LI><B>image width</B> - controls the overall width of the graphs
			display on the main Genome Graphs page.  The default is
			620 pixels.</LI>
		<LI><B>graph height</B> - controls the height of the graph(s) in
			the space above each chromosome.  The default is
			27 pixels.</LI>
		<LI><B>graphs per line</B> - controls how many graphs are displayed
			on each line in the space above each chromosome. For example,
			if you set this value to two, the display will superimpose
			two graphs on top of each other on one line.  The axis label
			for the first graph will appear on the left side of the display
			and the axis for the second graph on the right side.</LI>
		<LI><B>lines of graphs</B> - controls how many sets of graphs will
			appear above each chromosome.  For example, if you set this
			value to 2, the display will make room for two lines of
			graphs (each at the <B>graph height</B> above) in the space
			above each chromosome.
		<LI><B>chromosome layout</B> - controls how the chromosomes are laid
			out in the Genome Graphs display. You can choose to view one
			or two chromosomes on each horizontal line in the display. 
			Alternatively, you can set up the display such that all of the
			chromosomes appear in one long line. If you choose this layout,
			you may want to adjust the width of the image (<B>image width</B> 
			above).</LI>
		<LI><B>numerical labels</B> - check this box if you would like to see
			axis labels to the right/left of the display. If you did not specify
			<B>label values</B> when you uploaded your file, the numerical
			labels will default to 1/3 and 2/3 of the max and min values in
			your data input file.</LI>
		<LI><B>highlight missing</B> - check this box if you would like to
			see the areas in your graph where there is no data. Note that if
			you are displaying more than one graph, this attribute only
			pertains to the first graph.</LI>
		<LI><B>region padding</B> - controls the size of the data 
                       regions. The data points in your graphs which exceed 
		       the significance threshold are padded by this number
		       of bases on either side.  
		       The default places 25,000 bases on each side.</LI>
		</UL>
		</P>

		<P>
		When you have completed configuring the display, press the 
		<B>submit</B>
		button to return to the Genome Graphs display.
		</P>
	
		<P>
		<B>Configuring individual graphs</B><BR>
		</P>

		<P>
		Near the bottom of the Configuration page, you will see a list of
		the graphs that you have uploaded. Click on the hyperlinked graph
		name to configure that graph. This configuration pertains to 
		the Genome Graphs view. 
		</P>

		<P>
		You can set the range of the display 
		by editing the <B>display min/max value</B> values. This will
		restrict the Genome Graphs display for this graph to that
		data range. The axis will be labeled at 1/3 and 2/3 of the 
		data range that you set.
		</P>

		<P>
		If your data is sparse, you may want to draw lines between
		your data points. You can configure that by editing the
		<B>draw connecting lines between markers separated by
		up to ... bases</B> value.  The default value is 25,000,000
		bases.
		</P>
		
		<P>
		When you have completed configuring the display, press the
		<B>submit</B>
		button twice to return to the Genome Graphs display.
		</P>
                <P>

<!-------Setting a Significance Threshold------------------------->
<A NAME="SigThreshold"></A>
                <HR>
                <B><FONT SIZE="3" COLOR="#0000FF">Setting a Significance Threshold</FONT></B>
                <P>
		Most genome-wide data has some amount of noise and is only interesting when 
		the data values are above a certain value.  You can set this value using the 
		significance threshold input box.  Enter a decimal number in this input box
		and press Enter.  The display will now have a light gray line across the 
		graph at this data value.  If you have more than one graph displayed, the 
		significance threshold only pertains to the graphs that contain the significance
		threshold in the displayed data range.
		</P>

		<P>
		The significance threshold works in concert with the <B>browse
		regions</B> 
		and <B>sort genes</B> buttons; it will affect the regions 
		that are 
		displayed once you press either of these two buttons.
		</P>

		<P>
		To open the Genome Browser with a view of all of the regions 
		in your graph 
		that include data points that pass the significance threshold,
		press the 
		<B>browse regions</B> button. This will open the Genome 
		Browser with a 
		navigation pane on the left side of the screen. This pane 
		will contain 
		links to all regions which pass your significance threshold.
		</P>

		<P>
		To view a list of genes which are in regions that pass the 
		significance 
		threshold, press the <B>sort genes</B> button.  This will 
		open the 
		Gene Sorter with only the genes that are in significant 
		locations 
		with respect to your data.
		</P>

		<P>
		If you would rather view all of your regions without restricting the 
		output to only those regions that pass the significance threshold, simply 
		delete any values from the significance threshold input box and press 
		Enter before pressing <B>browse regions</B>.
		</P>
                <P>

<!-------Setting a Data Region------------------------->
<A NAME="DataRegion"></A>
                <HR>
                <B><FONT SIZE="3" COLOR="#0000FF">Setting a Data Region</FONT></B>
                <P>
		The data region is the span of bases that will be added to either side
		of the data points in your graphs which exceed the significance 
		threshold. Set the data region by editing the <B>region 
		padding</B> value on the configuration page. 
		The combination of setting the data region and the significance 
		threshold will affect two things:
		<UL>
		<LI>the regions displayed in the 
			<A HREF="#ViewInGB">Genome Browser</A> 
			after you press the <B>browse regions</B> button,</LI>
		<LI>the genes displayed in the 
			<A HREF="#ViewInGS">Gene Sorter</A> 
			after you press the <B>sort genes</B> button.</LI>
		</UL> 
                </P>

		<P>
		For example, take a data set that contains the following data:
		<P>
		<TT>chr2 400100000 2.3<BR>
		    chr2 400100500 4.5<BR>
		    chr2 400101000 1.2<BR></TT>
		<P>If you set the significance threshold at 4.0, one data 
		point
		in the data set passes that threshold. If you then set the data 
		range to 200, then the one significant data point will be 
		padded on each side by 200 base pairs.  In that case, the 
		only resulting significant data region will be 
		chr2:400,100,300-400,100,700.<BR><BR>
		If instead you set the data range
		to 2,000, then the one significant data point will be
		padded on each side by 2,000 base pairs.  In that case, the 
		resulting significant data region will be 
		chr2:400,098,500-400,102,500.
		</P>

		<P>

	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>

<!-------Displaying Data in the Genome Browser-------------------------->
<A NAME="ViewInGB"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
          <TR><TD>
 	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	    	<FONT SIZE="4"><B>&nbsp; Viewing Data in the Genome Browser </B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
		<P>
		To view your graphs in the Genome Browser, press the 
		<B>browse regions</B> button. This will open the Genome
		Browser with your graph(s) displayed as track(s). You can
		configure and <A HREF="hgTracksHelp.html#FineTuning">edit your track</A> 
		as you can any other track in the Genome Browser. 
		In addition to the Genome Browser, you will also see a pane 
		on the left-hand side, which contains links to all of the 
		<A HREF="#DataRegion">significant regions</A> in your data.
		Please note that if you are displaying more than one graph in 
		Genome Graphs, the significant regions are based only on the 
		first graph in the display list. 
		</P>
		<P>
		You can also navigate to the Genome Browser by clicking directly 
		on a graph or chromosome in Genome Graphs. The Genome Browser 
		will open with a 1,000,000 bp window centered on the location 
		on which you clicked.
		</P>
		<P>

	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>


<!-------Displaying Data in the Gene Sorter-------------------------->
<A NAME="ViewInGS"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
          <TR><TD>
 	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	    	<FONT SIZE="4"><B>&nbsp; Viewing Data in the Gene Sorter </B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
		<P>
		To view the set of genes that are in 
		<A HREF="#DataRegion">significant regions</A> in your data, 
		press the <B>sort genes</B> button. This will open the Gene
		Sorter with a filter to include only genes that are located 
		in regions in your input data that are above the significance 
		threshold. Please note that if you are displaying more than 
		one graph in Genome Graphs, the significant genes are based 
		only on the first graph in the display list.
		</P>
		<P>
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>

<!-------Deleting Data-------------------------->
<A NAME="DataSets"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
          <TR><TD>
 	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	    	<FONT SIZE="4"><B>&nbsp; Deleting Data </B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
		<P>
		There are several ways to delete your data once it has 
		been uploaded. If you are viewing your data as a track in 
		the Genome Browser, you can click on the mini-button 
		or track control for the track and delete the track 
		using the <B>Remove custom track</B> button. You can also
		choose to reset your cart which will reset the browser 
		interface settings to their defaults, as well as delete 
		all custom tracks and data.  Do this by visiting the 
		gateway page and pressing the hyper link: &quot;Click 
		here to reset&quot;.
		</P>

		<P>
		Your data will be saved on our server for at least 48 hours 
		from the time you last access it. If you do not access it  
		for more than 48 hours, it will automatically be deleted 
		from our server. If you want to view it again, simply 
		upload it again.
		</P>
		<P>
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>



<!-------Correlating Data-------------------------->
<A NAME="Correlate"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
          <TR><TD>
 	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	    	<FONT SIZE="4"><B>&nbsp; Correlating Data Sets </B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
		<P>
		To calculate how well correlated with one another your data 
		sets are, press the <B>correlate</B> button. This will 
		calculate and display the correlation coefficient (<B>R</B>) 
		among each of your data sets. <B>R</B>, also known as Pearson's 
		correlation coefficient, is a measure of the extent that two 
		graphs move together. The value of <B>R</B> ranges between 
		-1 and 1. A positive <B>R</B> indicates that the graphs 
		tend to move in the same direction, while a negative <B>R</B> 
		indicates that they tend to move in opposite directions.
		<B>R-Squared</B> (which is indeed just <B>R*R</B>) measures 
		how much of the variation in one graph can be explained by 
		a linear dependence on the other graph. <B>R-Squared</B> 
		ranges between 0 when the two graphs are independent to 1 
		when the graphs are completely dependent.
		</P>
		<P>
		To return to the Genome Graphs, press the <B>return to 
		graphs</B> button.
		</P>
		<P>
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>



<!------End of Document--------------------->
  </TD></TR>
</TABLE>
</BODY></HTML>

