<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>

<HEAD>
  <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
  <META name="description" content="UCSC Genome Graphs">
  <META name="keywords" content="genome, genome data, table browser, genome browser, genome assembly, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA">
  <TITLE>UCSC Genome Browser: Genome Graphs User's Guide</TITLE>
  <LINK REL="STYLESHEET" HREF="../../style/HGStyle.css">
</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
  <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../../images/title.jpg"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../../inc/topbar.html"--></TD></TR>
<!--Content Tables-->  
  <TR><TD CELLPADDING=10 ROWSPAN=2 VALIGN="top">
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>	
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
          <TR><TD>	
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	        <FONT SIZE="4"><B>&nbsp;Genome Graphs User's Guide </B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
   	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
	        <H3> Table of Contents: </H3>
	 	<UL><B>
  		<LI><A CLASS="toc" HREF="#Introduction">Introduction</A>
        	<LI><A CLASS="toc" HREF="#FormatUpload">Formatting &amp; Uploading Data</A>
	   	  <UL>
                  <LI><A CLASS="toc" HREF="#Format">Formatting Data</A>
                  <LI><A CLASS="toc" HREF="#Upload">Uploading Data</A>
		  </UL>
        	<LI><A CLASS="toc" HREF="#Sample">Quick Start</A>
        	<LI><A CLASS="toc" HREF="#Displaying">Displaying Data in Genome Graphs</A>
	   	  <UL>
                  <LI><A CLASS="toc" HREF="#Configure">Configuring the Display</A>
                  <LI><A CLASS="toc" HREF="#SigThreshold">Setting a Significance Threshold</A>
                  <LI><A CLASS="toc" HREF="#DataRegion">Setting a Data Region</A>
		  </UL>
        	<LI><A CLASS="toc" HREF="#ViewInGB">Viewing Data in the Genome Browser</A>
        	<LI><A CLASS="toc" HREF="#ViewInGS">Viewing Data in the Gene Sorter</A>
        	<LI><A CLASS="toc" HREF="#DataSets">Deleting Data</A>
        	<LI><A CLASS="toc" HREF="#Correlate">Correlating Data</A>
		</B></UL>
		<P>
		The UCSC Genome Graphs User's Guide was last updated on 2 
		February 2007. 
		<A HREF="mailto:&#103;&#101;&#110;&#111;&#109;e&#64;&#115;&#111;&#101;.&#117;&#99;sc.e&#100;&#117;">Questions and feedback are welcome</A>.
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
  <P>

<!------Introduction-------------------------->
<A NAME="Introduction"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
          <TR><TD>
 	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	    	<FONT SIZE="4"><B>&nbsp; Introduction </B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
		<P>
		Genome Graphs is a tool for displaying genome-wide data sets 
		such as the results of genome-wide SNP association studies, 
		linkage studies and homozygosity mapping.</P>
		<P>
		Using the Genome Graphs tool, you can:
                <UL>
                <LI> upload several sets of genome-wide data and display
                     them simultaneously</LI>
                <LI> click on an area of interest and go directly to the
                     genome browser at that position</LI>
                <LI> set a significance threshold for your data and view
                     only regions that meet that threshold</LI>
                <LI> view the genes that exist in areas where your data
                     meet your significance threshold</LI>
                </UL>
		<P>
		This User's Guide is aimed at both the novice Genome Graphs user
		as well the advanced user. If you are new to the Genome Graphs 
		tool, read the <A HREF="#Sample">Quick Start</A> section to 
		learn about the basics using some sample data. Advanced users 
		may want to proceed directly to the section that addresses 
		a particular area of functionality in detail.</P>
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>

<!-------Format & Upload-------------------------->
<A NAME="FormatUpload"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
          <TR><TD>
 	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	    	<FONT SIZE="4"><B>&nbsp; Formatting &amp; Uploading Data </B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
		<P>
		</P>

<!-------Formatting Data------------------------->
<A NAME="Format"></A>
	        <HR>
	        <B><FONT SIZE="3" COLOR="#0000FF">Formatting Data</FONT></B>
		<P>
		Genome Graphs allows you to upload data from files that reside 
		on your computer. Several file formats are accepted by the 
		program.  For all formats there is a single line for each marker. 
		Each line starts with information on the marker, and ends with 
		the numerical values associated with that marker. The markers 
		can be of one of the following types:
		</P>

		<table>
		<tbody><tr><td valign="top" width="20">&mdash;</td><td><b>chromosome 
		base: </b> e.g. <TT>chr1 130000</TT> (Note that the first base 
		in a chromosome is considered position 0.) 
		</td></tr>

		<tr><td valign="top" width="20">&mdash;</td><td><b>STS Marker: </b>
		e.g. <TT>RH75228</TT>
		</td></tr>

		<tr><td valign="top" width="20">&mdash;</td><td><b>dbSNP rsID: </b>
		e.g. <TT>rs12345</TT>
		</td></tr>

		<tr><td valign="top" width="20">&mdash;</td><td><b>Affymetrix 500k 
		Gene Chip: </b>e.g. <TT>SNP_A-1780270</TT>
		</td></tr>

		<tr><td valign="top" width="20">&mdash;</td><td><b>Illumina HumanHap300 
		Bead Chip: </b>e.g. <TT>***</TT>
		</td></tr>
		</table>

		<P>
		The marker-value pairs in each line of the file can be separated 
		with either a single space, a tab, or a comma. The file can 
		contain multiple values for each marker.  In that case, a 
		separate graph will be created for each value column in the 
		input file.
		</P>

		<P>
		For example, <B>chromosome base</B> markers with only one 
		value associated with the marker would be entered like this:
		</P>

		<TT>chrX 100000 1.23</TT>

		<P>
		<B>dbSNP rsID</B> markers with two values associated 
		with the marker would be entered like this:
		</P>

		<TT>rs10218492 0.384 0.882</TT>

		<P>
		The Genome Graph program will map the marker IDs to the genome. 
		In cases where the marker maps to more than one location in 
		the genome, the value(s) in your input file will be associated 
		with each location.
		</P>

		<P>
		Note that markers can only be mapped to assemblies for which 
		there already exists a track of the type that contains your 
		marker type. You can not, for example, use dbSNP rsID markers 
		for the cow genome, as it does not have a SNP track.
		</P>
	        <P>

<!-------Uploading Data---------------------------->
<A NAME="Upload"></A>
	        <HR>
	        <B><FONT SIZE="3" COLOR="#0000FF">Uploading Data</FONT></B>
	        <P>
		Once you have created your input file, you must upload it to 
		Genome Graphs. From the main Genome Graphs page, choose your 
		clade, genome, and assembly to which your data pertains.
		If you are unsure of the UCSC assembly name, you can check 
		<A HREF="../../FAQ/FAQreleases#release1">this</A> page. 
		Now, press the upload button to go to the upload page.
		</P>

		<P>
		To upload a file in any of the supported formats, locate the 
		file on your computer using the controls next to <B>File 
		name:</B>, and then submit. The other controls on this form are 
		optional, though filling them out will sometimes enhance the 
		display. In general the controls that default to best guess 
		can be left alone, since the guess is almost always correct. 
		The controls for display min and max values and connecting 
		lines can be set later via the configuration page as well. 
		Here is a description of each control.
		</P>

		<UL>
		<LI><B>Name of data set</B>: Displayed in graph drop-down in 
		Genome Graphs and as trackname in Genome Browser. Only the 
		first 16 characters are visible in some contexts. For data 
		sets with multiple graphs, this is the first part of the 
		name, shared with all members of the data set.</LI>

		<LI><B>Description</B>: A short sentence describing the 
		data set. Displayed in the Genome Graphs and Genome Browser 
		configuration pages, and as the center label in the Genome 
		Browser.</LI>

		<LI><B>File format</B>: Controls whether the upload file is 
		a tab-separated, comma-separated, or space separated table.</LI>

		<LI><B>Markers are</B>: Describes how to map the data to 
		chromosomes. The choices are that either the first column of the 
		file is an ID of some sort, or the first column is a chromosome 
		and the next a base. The IDs can be SNP rs numbers, STS marker 
		names or ID's from any of the supported genotyping platforms.</LI>

		<LI><B>Column labels</B>: Controls whether the first row of the 
		upload file is interpreted as labels or data. If the 
		first row contains text in the numerical fields, or if the 
		mapping fields are empty, it is interpreted by &quot;best 
		guess&quot; as labels. This is generally correct, but you can 
		override this interpretation by explicitly setting the control.
		</LI>

		<LI><B>Display min value/max value</B>: Set the range of the 
		data set that will be plotted. If left blank, the range will 
		be taken from the min/max values in the data set itself. For 
		all data sets to share the same scale usually you'll need to 
		set this.</LI>

		<LI><B>Label values</B>: A comma-separated list of numbers 
		for the vertical axis. If left blank, the axis will be 
		labeled at the 1/3 and 2/3 point.</LI>

		<LI><B>Draw connecting lines</B>: Lines connecting data 
		points separated by no more than this number of bases are 
		drawn.</LI>

		<LI><B>File name</B>: The controls here let you select 
		which file on your computer to upload.</LI>
		<UL>


		</P>
		<P>
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>


<!-------Quick Start-------------------------->
<A NAME="Sample"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
          <TR><TD>
 	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	    	<FONT SIZE="4"><B>&nbsp; Quick Start </B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
		<P>
		[Use sample data to get a feel for what Genome Graphs is capable of.]</P>
		<P>
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>


<!-------Displaying Data in Genome Graphs-------------------------->
<A NAME="Displaying"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
          <TR><TD>
 	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	    	<FONT SIZE="4"><B>&nbsp; Displaying Data in Genome Graphs </B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
		<P>
		[How to display data in Genome Graphs.  Including configuring.]</P>
		<P>

<!-------Configuring the Display------------------------->
<A NAME="Configure"></A>
                <HR>
                <B><FONT SIZE="3" COLOR="#0000FF">Configuring the Display</FONT></B>
                <P>
                [how to configure the Genome Graphs display (configure button).  And
		how to configure individual graphs.]</P>
                <P>

<!-------Setting a Significance Threshold------------------------->
<A NAME="SigThreshold"></A>
                <HR>
                <B><FONT SIZE="3" COLOR="#0000FF">Setting a Significance Threshold</FONT></B>
                <P>
                [how to set the sig. thresh. and what it means (significance threshold box)]</P>
                <P>

<!-------Setting a Data Region------------------------->
<A NAME="DataRegion"></A>
                <HR>
                <B><FONT SIZE="3" COLOR="#0000FF">Setting a Data Region</FONT></B>
                <P>
                [how to set the data region and what it means (region boxES)]</P>
                <P>

	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>

<!-------Displaying Data in the Genome Browser-------------------------->
<A NAME="ViewInGB"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
          <TR><TD>
 	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	    	<FONT SIZE="4"><B>&nbsp; Viewing Data in the Genome Browser </B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
		<P>
		[How to view your Graph data in the Genome Browser.  Including
		moving around with the frames and navigating back to hgGenome.]</P>
		<P>
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>


<!-------Displaying Data in the Gene Sorter-------------------------->
<A NAME="ViewInGS"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
          <TR><TD>
 	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	    	<FONT SIZE="4"><B>&nbsp; Viewing Data in the Gene Sorter </B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
		<P>
		[How to view your Graph data in the Gene Sorter.]</P>
		<P>
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>

<!-------Deleting Data-------------------------->
<A NAME="DataSets"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
          <TR><TD>
 	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	    	<FONT SIZE="4"><B>&nbsp; Deleting Data </B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
		<P>
		[How to delete your data sets.  Including life-span of your data
		on the server.]</P>
		<P>
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>



<!-------Correlating Data-------------------------->
<A NAME="Correlate"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
          <TR><TD>
 	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	    	<FONT SIZE="4"><B>&nbsp; Correlating Data Sets </B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
		<P>
		[How to correlate your data sets.]</P>
		<P>
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>



<!------End of Document--------------------->
  </TD></TR>
</TABLE>
</BODY></HTML>

