<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
	"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
	<title>Data Integrator User's Guide - UCSC Genome Browser</title>
	<meta http-equiv="content-type" content="text/html; charset=utf-8">
	<link rel="stylesheet" href="/style/HGStyle.css">
</head>
<body class="oldStaticPage">

<div class="header">
	<img src="../../images/title.jpg" alt="UCSC Genome Bioinformatics">
</div><!-- .header -->

<div class="nav">
<!--#include virtual="../../inc/topbar.html"-->
</div><!-- .nav -->

<div class="content">
<div class="section">
	
	<h1>Introduction</h1>
	<div class="sectionContent">
	<p>The Data Integrator is a fast and powerful graphical interface that can
	combine and export data from multiple tracks simultaneously. Like the
	Genome Browser and Table Browser, the it can combine
	data from the browser database, user custom tracks and hub tracks. 
	</div><!-- .sectionContent --> 
</div><!-- .section -->

<div class="section">
	<h1>Getting Started</h1>
	<div class="sectionContent">
	<p>The Data Integrator allows for the intersection an output of data from
	up to five tracks. Data is output based on its overlap with items in the 
	first select track. These tracks can be those provided by UCSC or user
	data that has been uploaded through custom tracks or track hubs. It allows
	for the customization of the output columns where you can select columns
	from each track to include in the output. The output can be view in the
	web browser window or downloaded to a local file, optionally compressed
	with gzip.
	<p>
	<b><i>Simple position-based query</i></b>
	<p>Follow these instructions to perform a basic query that intersects two
	or more tracks:
	<p>
	<b>Step 1. Pick a genome assembly</b><br>
	Specify the genome assembly from which you'd like to retrieve the
	data by choosing the appropriate organism in the <tt><b>genome</b></tt>
	list, then selecting the assembly version from the
	<tt><b>assembly</b></tt> list. Note that the <tt><b>assembly</b></tt> list
	refreshes each time a different option is selected in the
	<tt><b>genome</b></tt> list. Assemblies are typically named after the first three
	characters of an organism's genus and species names. Assemblies that are
	uploaded as part of an <a href="hgTrackHubHelp.html#Assembly">assembly hub</a>
	will be available under the hub's name in the <tt><b>group</b></tt> list.
	<p>
	<b>Step 2. Pick a genomic region</b><br>
	By default, the Data Integrator region is set to <tt><b>genome</b></tt>,
	which will display all the data records in the selected table.
		<ul>
		<li>
		To restrict the data to a specific position
		range, select "position or search term" from the 
		<tt><b>region to annoatate</b></tt> menu
		and type the position into the box. Some
		examples of specific positions include a chromosome name
		(<em>chrX</em>), a coordinate range within a chromosome
		(<em>chrX:100000-400000</em>), or a scaffold name.
		<li>
		To look up the position range of a genomic element -- such
		as a gene name, an accession ID, an STS marker, etc. -- or
		keywords from the GenBank description of an mRNA,
		select "position or search term" from the 
		<tt><b>region to annoatate</b></tt> menu and then type
		your search term into the box.
		<li>
		To select multiple genomic regions, select "define regions" from 
		<tt><b>region to annoatate</b></tt> menu and then
		entering up to 1,000 regions in a 3- or 4-field
		<a href="../../FAQ/FAQformat.html#format1">BED</a>
		file format.
		</ul>
	<p>
	<b>Step 3. Configure your data sources</b><br>
	The <tt><b>track group</b></tt> list shows all the annotation track groups
	available in the selected genome assembly. The names correspond
	to the groupings displayed at the bottom of the Genome Browser
	annotation tracks page. When a group is selected from the list,
	the <tt><b>track</b></tt> list automatically updates to show all the
	annotation tracks available within that group.
		<ul>
		<li>
		To examine all the tracks available within a certain group (e.g.
		all gene prediction tracks), select the group name from the
		<tt><b>track group</b></tt> list, then browse the entries in the
		<tt><b>track</b></tt> list.
		<li>
		Custom annotation tracks created during the current session
		are listed under the <em>Custom Tracks</em> group.
		<li>
		Each connected track hub will be listed as it's own group under the
		<tt><b>track group</b></tt> list.
		</ul>
	<p>
	<b>Step 4. Rearrange your data sources (Optional)</b><br>
	You can rearrage you data sources by clicking on a track name
	and dragging it into the desired order. Items are output based on their overlap
	with the first listed track. The order of the tracks will also determine
	the order of the fields in the output. Fields from tracks higher in the list
	will appear first in the output.
	<p>
	<b>Step 4. Choose output fields (Optional)</b><br>
	Click <tt><b>Choose fields</b></tt> button to bring up a checklist of all
	fields from all selected data sources. By default every field is selected.
	You can quickly select or deselect all of the fields by clicking 
	<tt><b>Set all</b></tt> or <tt><b>Clear all</b></tt> buttons below the track
	name. You can choose the columns you want in your output by checking the box
	next to that field.
	<p>
	<b>Step 5. Display the output</b><br>
	Click the <tt><b>Get Output</b></tt> button to display the
	results of the query in the web browser window. The results will be displayed
	as tab-delimited text. Download results as a file by checking the box next to
	<tt><b>Send output to file</b></tt> and entering a file name into the box.
	The output file can also be compressed by gzip.

</div><!-- .content -->
</body>
</html>
