<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Gene Family Browser: User's Guide</TITLE>
	<LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="/images/title.jpg"></TH></TR>

<!----Hotlinks Bar-------------------------------------->
    <TR><TD COLSPAN=2 HEIGHT=40>
	<TABLE BGCOLOR="000000" CELLPADDING="1" CELLSPACING="1" WIDTH="100%" HEIGHT="27">
	    <TR BGCOLOR="2636D1"><TD VALIGN="middle">
	        <TABLE BORDER=0 CELLSPACING=0 CELLPADDING=0 BGCOLOR="2636D1" HEIGHT="24">
		    <TR><TD VALIGN="middle"><FONT COLOR="#89A1DE">&nbsp; &nbsp;
			<A HREF="/index.html" class="topbar">
				Home</A> &nbsp; - &nbsp;
			<A HREF="/cgi-bin/hgGateway" class="topbar">
				Genome Browser</A> &nbsp; - &nbsp; 
			<A HREF="/cgi-bin/hgNear" class="topbar">
				Family Browser</A> &nbsp; - &nbsp; 
			<A HREF="/cgi-bin/hgBlat?command=start" class="topbar">
				Blat </A> &nbsp; - &nbsp; 
			<A HREF="/cgi-bin/hgText?command=start" class="topbar">
				Table Browser</A> &nbsp; - &nbsp; 
			<A HREF="/FAQ/" class="topbar">
				FAQ</A> &nbsp; - &nbsp; 
			<A HREF="/goldenPath/help/hgTracksHelp.html" class="topbar">
				Help</A> &nbsp;</font></TD>
		    </TR>
	        </TABLE>
	    </TD></TR>
	</TABLE>
    </TD></TR>

<!-----Content Tables------------------------------------>  
    <TD CELLPADDING=10 ROWSPAN=2 VALIGN="top">
    <TR><TD CELLPADDING=10>	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
	    <TR><TD>	
    		<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
		    <TR><TD>	
			<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
			    <TR><TD>
			    <FONT SIZE="4"><B>&nbsp;UCSC Gene Family Browser User's Guide </B></FONT> 
			    </TD></TR>
			</TABLE>
			<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
			    <TR><TH HEIGHT=10></TH></TR>
			    <TR><TD WIDTH=10>&nbsp;</TD><TD> 
				<B><FONT COLOR="#006666" SIZE=4> Table of Contents: </FONT></B>
				<TABLE BORDER=0 NOWRAP CELLPADDING=0 CELLSPACING=0 MARGINS=0>
			 	    <TR><TH HEIGHT=10></TH></TR>
				    <TR><TD>
				        <UL><B>
				  	    <LI><A HREF="#Intro">Introduction</A>
  					    <LI><A HREF="#Start">Getting Started</A>
					    <LI><A HREF="#Table">Understanding the Browser display</A>
					    <LI><A HREF="#Configure">Configuring the Browser display</A>
					    <LI><A HREF="#Filter">Filtering the gene display</A>
					    <LI><A HREF="#Download">Displaying sequence and text-based output</A>
				  	</B></UL></TD>
				    </TR>
				</TABLE>
				This user's guide was last updated on 26 Sep 2003. 
				<A HREF="mailto:genome@soe.ucsc.edu">Questions and feedback 
				are welcome</A>.
				</TD>
				<TD WIDTH=15></TD>
			    </TR>
			</TABLE>
	 	    <BR></TD></TR>
		</TABLE>
	    </TD></TR>
	</TABLE>

<!-----------Introduction-------------------------------------------->
<A NAME="Intro"></A><BR>

	<!--outer table is for border purposes-->
	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
	    <TR><TD>	
    		<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
		    <TR><TD>	
			<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
			    <TR><TD><FONT SIZE="4"><B><A NAME="TOC"></A><B>&nbsp; Introduction</B></FONT>
			    </TD></TR>
			</TABLE>
			<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
			    <TR><TH HEIGHT=10></TH></TR>
			    <TR><TD WIDTH=10>&nbsp;</TD><TD>
				<P>
				Genes function and evolve together. To understand a gene,
				you often need to understand an entire gene family.
				Many such families are already known and described, such as the 
				HOX family that mediates many aspects of limb and brain development 
				and the Cytochrome P450 family that is central to the metabolism of many 
				medications.
				<P>
				One easy way to identify well-known relatives of a gene 
				is by looking for genes with name similarity, because biologists 
				tend to use similar names for similar genes. However, scientists
				only partly understand the function of perhaps one-third of
				the genes of the genome; therefore, other techniques for grouping genes
				into families are necessary.  
				<P>
				The UCSC Gene Family Browser is an excellent resource for 
				exploring gene families and the relationships among genes. This 
				tool displays a table of genes within 
				a selected genome that are related to one another. Several 
				different relationships may be explored: protein-level homology, 
				similarity of gene expression profiles, or genomic proximity. The 
				Browser supports searches on a variety of terms and phrases, 
				including the gene name, the SwissProt protein name, a GenBank 
				accession, or a word or phrase present in a gene's description. The 
				gene family display is highly configurable, allowing the user to 
				control the order and number of columns, the number of rows, and 
				the genes displayed. The tool provides several output formats, 
				including a simple tab-delimited format that may be imported into a 
				spreadsheet or a relational database.
				<P>
				An important use of the Browser is to gather 
				together a collection of genes that share similar properties for
				statistical analysis.  For instance, one might want to examine
				promotor regions of genes that share a similar expression
				pattern  or look for protein sequence motifs in genes that
				share similar GO annotations.  
				<P>
				One of the most powerful features of the Browser is its 
				filtering capabilities. The filter enables the user to 
				quickly select an interesting subset of the 25,000 genes in 
				the genome based on a variety of detailed and flexible selection criteria. 
				For example, the filter 
				may be used to select all human genes over-expressed in the cerebellum 
				that have GO-annotated G-protein coupled receptor activity. 
				<P>
				The Gene Family Browser was designed and implemented by Jim Kent, 
				Fan Hsu, David Haussler, and the UCSC Genome Bioinformatics Group. 
				This work is supported by a grant from the National Human Genome 
				Research Institute and by the Howard Hughes Medical Institute.
			        </TD>
			        <TD WIDTH=15></TD>
			    </TR>
			</TABLE>
		    <BR></TD></TR>
		</TABLE>
	    </TD></TR>
	</TABLE>

<!-----------Getting Started----------------------------------------->
<A NAME="Start"></A><BR>

	<!--outer table is for border purposes-->
	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
	    <TR><TD>	
    		<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
		    <TR><TD>	
			<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
			    <TR><TD><FONT SIZE="4"><B><A NAME="TOC"></A><B>&nbsp; Getting Started</B></FONT>
			    </TD></TR>
			</TABLE>
			<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
			    <TR><TH HEIGHT=10></TH></TR>
			    <TR><TD WIDTH=10>&nbsp;</TD><TD>
				<P>
				To begin using the Gene Family Browser, you will first
				have to select a genomic region and the type of gene relationship
				you wish to display.  
				You may also want to change some of the Browser's configuration
				settings to tailor the display to your research needs. These
				configuration options are described in <A HREF="#Configure">Configuring
				the Browser display</A>.
				<P>
				<B>Starting the Gene Family Browser</B>
				<OL>
				<LI>Open the Browser
				<A HREF="http://genome.ucsc.edu/cgi-bin/hgNear" TARGET=_BLANK">home 
				page</A>. 
				<LI>Specify the genome and assembly you wish to view by selecting 
				the appropriate options from the <em>genome</em> and 
				<em>assembly</em> pull-down menus.
				<LI>Type a term or phrase into the <em>search</em> text box to
				determine which genes will be displayed in the browser. Valid search
				terms (with human genome examples) include:
				    <UL>
				    <LI>a gene name (HOXA9)
				    <LI>a SwissProt protein name (HXA9) 
				    <LI>a word or phrase that occurs in the description of a gene (MAP kinase)
				    <LI>a GenBank mRNA accession (U14680)
				    </UL>
				<LI>Choose the gene relationship that you would like to examine by 
				selecting an option from the <em>sort by</em> pull-down menu. Genes will
				be sorted in order of proximity to the chosen gene, based on one of the
				folowing criteria:
				    <UL>
<!--Commented out til implemented
				    <LI>Expression (UCLA) - similarity in expression, based on UCLA data
				    <LI>Expression (GNF) - similarity in expression, based on GNF data
-->
				    <LI>Protein Homology - similarity in protein homology
				    <LI>Gene Distance - absolute distance (left or right) on the chromosome 
					from the selected gene
				    <LI>Name Similarity - similarity to name of selected gene
				    <LI>GO Similarity - number of Gene Ontology (GO) terms shared with 
					selected gene 
				    <LI>Name - list sorted by gene name
				    <LI>Chromosome - list sorted by chromosomal location
				    </UL>
				<LI>Choose the number of items to display from the <em>display</em> 
				pull-down menu (the default is 50).
				<LI>Press the Go! button to display your search results.
				</UL>
			        </TD>
			        <TD WIDTH=15></TD>
			    </TR>
			</TABLE>
		    <BR></TD></TR>
		</TABLE>
	    </TD></TR>
	</TABLE>

<!-----------Understanding the Browser display----------------------->
<A NAME="Table"></A><BR>       
                                
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>            
                <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                    <TR><TD>    
                        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%> 
                            <TR><TD><FONT SIZE="4"><B><A NAME="TOC"></A><B>&nbsp; Understanding 
                                the Browser display</B></FONT>
                            </TD></TR>
                        </TABLE>
                        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                            <TR><TH HEIGHT=10></TH></TR>
                            <TR><TD WIDTH=10>&nbsp;</TD><TD>
                                <P>
                                The main page of the Gene Family Browser displays a table
				containing rows of genes and associated attributes. In most 
				cases, the currently-selected gene is shown at the top of 
				the list, highlighted in light green. The remaining genes
				are ordered relative to the selected gene based on the sort
				criteria specified in the <em>sort by</em> menu. For example,
				in a table sorted by gene distance, the genes are listed in
				order of greater to lesser proximity to the selected gene.
				<P>				
				The initial Browser display contains only a default subset of
				the columns available. The set of columns may be expanded,
				reduced, and rearranged by using the Browser's 
				<A HREF="#Configure">configuration utility</A>. To view 
				information about the data shown in the column, click on
				the column's label.
				<P>
				<B>Changing the selected gene</B><BR>
				To select a different gene in the table, click on the name of
				the gene. The Browser will move the gene to the top of the 
				list and highlight it. The remaining genes will be reordered
				relative to the new selection.
                              	<P>  
				<B># column</B><BR>
                              	This column displays the position of each gene in the table,
				which is sometimes useful when examining data in tables with
				many rows. Clicking on the gene number selects it and moves
				it to the top of the list.
				<P>
    				<B>Name column</B><BR>
				This column displays the name of the gene. When possible, 
				the HUGO name is shown. If the gene does not yet have a HUGO 
				name, the GenBank accession number of the associated RefSeq 
				or mRNA record is shown instead. Clicking on the gene name
				selects it and moves it to the top of the list.
                              	<P>  
				<B>Swiss-Prot column</B><BR>
				This column displays the Swiss-Prot protein name of each gene, 
				if it is available. Otherwise, it shows the primary 
				accession number. Clicking on a protein name or accession 
				displays the corresponding Swiss-Prot NiceProt view of the
				protein. 
                              	<P>  
				<B>RefSeq column</B><BR>
				This column displays the NCBI RefSeq accession associated with
				the gene, if available. RefSeq genes are a non-redundant set of
				high-quality mRNA sequences. Clicking on the accession number
				displays the NCBI LocusLink record for a RefSeq accession. If
				LocusLink shows a link to the Online Mendelian Inheritance in
				Man (OMIM) -- represented by an orange box with an "O"
				on it -- an OMIM record is available for the gene. OMIM is 
				often an excellent source of human-curated information about
				human genes. 
                              	<P>  
				<B>GenBank column</B><BR>
				This column shows the GenBank RefSeq or mRNA accession number
				associated with the gene. Clicking on the accession number 
				displays the GenBank record associated with it. 
                              	<P>  
				<B>Ensembl column</B><BR>
				This column shows the Ensembl transcript ID associated with 
				the gene. Ensembl is an automatic gene prediction pipeline and
				a major genome database and web site run by the Sanger
				Institute and the European Bioinformatics Institute (EMBL-EBI).
				It is especially effective at mapping genes that are known 
				in one organism to another organism. Compared to other gene
				predictions, those of Ensembl tend to have high specificity 
				but low sensitivity to genes not already associated with
				characterized mRNA or protein sequence. Clicking on an 
				Ensembl ID displays the Ensembl GeneView page for the gene. 
                              	<P>  
				<B>WormBase column (<em>C. elegans</em>)</B><BR>
				This column shows the ORF name associated with each gene.  Clicking on an 
				ORF name displays the associated WormBase record. 
                              	<P>  
				<B>U74 ID column (Mouse)</B><BR>
				This column shows the Affymetrix ID from the U74 series of
				chips (A, B, and C) that best corresponds to each gene. For
				more information about the selection criteria used for this 
				column, click on the U74 ID column's label.
                              	<P>  
				<B>U133 ID column (Human)</B><BR>
				This column shows the Affymetrix ID from the HG-U133 chip that best 
				corresponds to each gene. For more information about the selection
				criteria used for this column, click on the U133 ID column's label.
                              	<P>  
				<B>U95 ID column (Human)</B><BR>
				This column shows the Affymetrix ID from the HG-U95 chip that best 
				corresponds to each gene. For more information about the selection
				criteria used for this column, click on the U95 ID column's label.
                              	<P>  
				<B>Expression ratio columns</B><BR>
				These columns show the ratio of 		
				expression of the gene in selected tissues to the expression 		
				of the gene overall. 
				<P>
				By default, a gene that is more highly
				expressed in a tissue is colored red, and less expressed 
				genes are shown in green. This coloring is standard, but is 
				the opposite of what a naive user might expect: in this 
				case, red means go, and green means stop! White is used to 
				show instances where no data is available, and black denotes
				cases in which the gene expression level in the tissue is the 
				same as it is overall.
				<P>
				Color-blind users 
				may wish to switch the coloring from red/green to blue/green with
				the <A HREF="#Configure">configuration utility</A>.
                              	<P>  
				<B>Bits column</B><BR>
				This column shows the bit score measure of protein similarity
				between a gene and the selected gene. The greater the 
				similarity between two proteins, the higher the bit score. For
				more information about how the bit score was calculated, click
				on the Bits column label.
                              	<P>  
				<B>E-Value column</B><BR>
				This column shows the Blastp e-value (expectation value) 
				between each gene and the selected gene. The greater the 
				similarity of two proteins, the lower the e-value is. 
				Identical long proteins have an e-value of 0. 
				Formally, an e-value is the number of other known genes that 
				are expected to have at least this level of homology by chance.
				An e-value of less than 0.1 can be safely interpreted as the
				probability that a match this good would occur merely by
				chance. For more information about how the e-value was 
				calculated, click on the E-Value column label.
                             	<P>  
				<B>% ID column</B><BR>
				This column shows the percentage identity at the protein 
				level between the gene and the selected gene. For more 
				information about how the % ID was 
				calculated, click on the % ID column label.
                              	<P>  
				<B>Genome Position column</B><BR>
				This column shows the chromosomal location of each gene in 
				the genome. Clicking on a chromosomal position displays the 
				gene at that location in the UCSC Genome Browser. 
                              	<P>  
				<B>Coding SNPs column</B><BR>
				This column shows the Simple Nucleotide Polymorphisms (SNPs) 
				located in the coding region of each gene. Clicking on a SNP 
				ID displays the SNP record associated with the gene in 
				NCBI's dbSNP. 
                              	<P>  
				<B>Human column</B><BR>
				This column shows that best Blastp match to the known genes
				protein set from the UCSC Human Genome Browser database. 
				Clicking the accession number displays the gene in the UCSC 
				Human Genome Browser. 
                              	<P>  
				<B>Mouse column</B><BR>
				This column shows the accession number of the best Blastp match to the 
				known genes protein set in the UCSC Mouse Genome Browser database. Clicking
				on the accession number displays the gene in the UCSC Mouse Genome Browser. 
                              	<P>  
				<B>Zebrafish column</B><BR>
				This column shows the Ensembl peptide ID of the best Blastp 
				match to the Ensembl gene predictions on the zebrafish genome. 
				Clicking on the Ensembl peptide ID displays it in Ensembl 
				Zebrafish Protein View. 
                              	<P>  
				<B>Drosophila column</B><BR>
				This column shows the FlyBase ID of the best Blastp match to 
				the FlyBase protein set. Clicking on the FlyBase ID displays
				the corresponding FlyBase report. 
                              	<P>  
				<B><em>C. elegans</em> column</B><BR>
				This column shows the best Blastp match to the WormBase 
				protein set. Clicking on the ID number displays the 
				corresponding WormBase database record. 
                              	<P>  
				<B>Yeast column</B><BR>
				This column shows the best Blastp match to the Saccharomyces 
				Genome Database (SGD) protein set. Clicking on the yeast ID 
				displays the corresponding SGD record. 
                              	<P>  
				<B>Pfam Domains column</B><BR>
				This column shows a list of Protein Family (Pfam) domains 
				contained in the gene product. Clicking on a domain displays
				the associated Pfam record. 
                              	<P>  
				<B>Gene Ontology column</B><BR>
				This column shows the Gene Ontology (GO) terms associated with 
				the gene. GO terms are words from a controlled vocabulary 
				assigned to a gene by human curators. Clicking on a GO term 
				displays the associated Gene Ontology Consortium database 
				record. 
                              	<P>  
				<B>Description column</B><BR>
				This column shows a brief description of each gene taken from 
				its mRNA record. Clicking on a description displays a page 
				showing additional information and links for the gene. 
				</TD>
                                <TD WIDTH=15></TD>
                            </TR>
                        </TABLE>
                    <BR></TD></TR>
                </TABLE>        
            </TD></TR>          
        </TABLE>           

<!-----------Configuring the Browser display------------------------->
<A NAME="Configure"></A><BR>

	<!--outer table is for border purposes-->
	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
	    <TR><TD>	
    		<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
		    <TR><TD>	
			<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
			    <TR><TD><FONT SIZE="4"><B><A NAME="TOC"></A><B>&nbsp; Configuring the 
				Browser display</B></FONT>
			    </TD></TR>
			</TABLE>
			<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
			    <TR><TH HEIGHT=10></TH></TR>
			    <TR><TD WIDTH=10>&nbsp;</TD><TD>
				<P>
				The Gene Family Browser is highly configurable, allowing you
				to fine-tune the display to show just the genes and data
				columns in which you're interested in an order that best suits your research
				needs. Most of the configuration is controlled through settings on
				the Configuration page, accessed via the <em>configure</em> button at
				the top of the Browser page.
				<P>
				<B>Changing the number of rows displayed</B><BR>
				To increase or decrease the number of rows shown in the table, pick a
				new value from the <em>display</em> pull-down menu, then click the Go! 
				button.
				<P>
                                <B>Changing the number of columns displayed</B><BR>
                                By default, the Browser shows only a small subset of the
				table columns available for the genome. You can view the full list of columns,
				or add or remove columns from your display, on the Configuration page.
				<P>The configuration table shows all the
				columns available for the currently-selected genome, listed in left-to-right
				display order. To add or remove a column from the Browser display, click the
				<em>On</em> checkbox to toggle the setting (a check indicates that the column
				is displayed). To quickly change the <em>On</em> settings of all columns,
				click the Hide All or Show All button at the top of the page. Click the 
				Submit button to display the changes in the Browser.
				<P>
				<B>Changing the column positions</B><BR>
				In addition to adding or removing columns, it is 
				also possible to move the columns to the left or right within the Browser
				table. The order of the column names in the configuration table indicates the
				current relative position of the columns in the Browser display from left to
				right. To shift a column one position to the left, click the <em>up</em> arrow in the
				item's <em>Position</em> column. Similarly, click the <em>down</em> arrow to shift a 
				column to the right. When you have finished making changes, click the Submit
				button.
				<P>
				<B>Changing the expression ratio colors</B><BR>
				By default, the gene expression ratios are shown using a red/green color 
				scheme,
				where red indicates a gene that is more highly expressed and green corresponds
				to less expression. Color-blind users may find it helpful to switch the 
				coloring from red/green to blue/green. To do so, select the <em>blue high/green
				low</em> option from the Expression ratio colors pull-down menu on the 
				Configuration page, then click the Submit button.
				<P>
				<B>Changing the brightness of expression ratio colors</B><BR>
				To increase or decrease the brightness of the colors in an expression ratio 
				column, edit the <em>Brightness</em> value for the corresponding entry in the 
				configuration table. Values greater than 1.0 increase the brightness, while 
				those less than 1.0 dim the color. Click the Submit button to display the 
				new values.
				<P>
				<B>Displaying splicing variants</B><BR>
				By default, the Browser shows only one splicing variant: the one that produces
				the largest protein. To show all splicing variants, click the <em>Show all 
				splicing variants</em> checkbox. Note that in most cases, the column values
				(and sometimes the names) will be identical across variants.
				<P>
				<B>Restoring the default settings</B><BR>
				At any time during your Browser session, you can restore the Browser table
				to its default layout by clicking the Default button on the Configuration page,
				then clicking Submit.
				<P>
				<B>Saving a configuration for future use</B><BR>
				The Browser configuration utility allows you to store multiple configurations
				for use in future sessions. This feature is particularly useful if you require
				different layouts for different research uses.
				<P>
				To save the current configuration of the Browser layout, click the Save button
				on the Configuration page. Type in a name for the configuration in the text
				box at the top of the page, then click Save. 
				<P>
				<B>Loading a previously-saved configuration</B><BR>
				Once you have saved a configuration, you can load it back into your Browser
				in a future session. To load a configuration, click the Load button on the
				Configuration page. The Browser will display a list of the names of your saved 
				configurations. Click on a name to highlight it, then click Load to 
				reconfigure your Browser based on the saved settings.
				<P>
				<B>Viewing a list of saved configurations</B><BR>
				To display a list of configurations that you have saved, click the Save button
				on the Configuration page. If you have any saved configurations, the Browser 
				will display an Existing Setups list that shows the configuration names. To
				permanently remove a configuration from the list, click on the name to 
				highlight it, then click the Delete Existing Setup button.
			        </TD>
			        <TD WIDTH=15></TD>
			    </TR>
			</TABLE>
		    <BR></TD></TR>
		</TABLE>
	    </TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>

<!-----------Filtering the gene display------------------------------>
<A NAME="Filter"></A><BR>

	<!--outer table is for border purposes-->
	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
	    <TR><TD>	
    		<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
		    <TR><TD>	
			<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
			    <TR><TD><FONT SIZE="4"><B><A NAME="TOC"></A><B>&nbsp; Filtering the gene display</B></FONT>
			    </TD></TR>
			</TABLE>
			<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
			    <TR><TH HEIGHT=10></TH></TR>
			    <TR><TD WIDTH=10>&nbsp;</TD><TD>
				<P>
				The Gene Family Browser's gene filtering capabilities provide a versatile
				way to fine-tune the Browser display to show just the genes in which you are
				interested. Filters are applied to individual gene fields, and may be
				combined to increase the specifity of the search. To access the Filter page, 
				click the <em>filter</em> button at the top of the Browser page. 
				<P>
				At any time during the filter setup process, you can click the List Names
				button on the Filter page to view a list of genes
				that will be returned when the current filter settings are applied to the
				genome. You may find this list helpful in fine-tuning the filter. 
				<P>
				<B>Filtering based on matching one or more terms</B><BR>
				Filters based on names, IDs, or other words restrict the display 
				to only those genes that match one or more terms typed into the 
				<em>search</em> text box. 
				Examples of values that can be filtered on this basis include the gene
				name, RefSeq accession number, gene description, coding SNPs, and GO terms.
			 	<P>
				This search supports wildcard matching on &quot;*&quot; and &quot;?&quot;.
				Multiple terms must be separated by a space or tab. For example,
				the search criteria "HOXA9 FOX*" on the gene name field returns the gene 
				named HOXA9 and any gene whose name begins with the letters &quot;FOX&quot;. 
				When searching on fields that consist of values containing more than one 
				word (GO terms, coding SNPs, Pfam domains, and gene descriptions), the 
				multi-word
				elements must be enclosed in single quotes. For instance, a search on the
				description phrase <em>forkhead box protein</em> should be entered as
				'forkhead box protein'.
				Use the 
				<em>any</em> and <em>all</em> options to determine whether the search should 
				return any gene that matches any term (<em>any</em>) or only those genes that 
				match all terms (<em>all</em>).
				<P>
				To facilitate searching on multiple terms, the Browser provides
				the option to paste in or upload a list of search terms. To paste in a list
				of terms, click the filter's Paste List button, then paste or type the 
				terms into the text box. Terms must be separated by a space, a tab, or be 
				entered on separate lines, and may not include wildcards. When you have 
				completed the list, click the Submit button to return to the main Filter page.
				The file upload utility - accessed via the Upload List button - has a similiar
				functionality.
				<P>
				<B>Filtering based on numerical ranges</B><BR>
				Several of the gene fields can be filtered by specifying a numerical range
				within which the value must fall. Examples of
				fields in this category include expression ratios, Blastp data, and genome
				position. To use this type of filter, enter the minimum and maximum
				values delimiting the range in which you are interested. In some cases, the
				range of valid values is indicated in the filter box. 
				<P>
				The genome position filter requires the name of a chromosome (in the 
				format <em>chrN</em>) in addition
				to the chromosomal start and end positions. To list all genes on a chromosome,
				enter only the chromosome name.
				<P>
				Expression ratio filters include <em>any</em> and <em>all</em> options to 
				determine whether the search should 
				return a gene if any of the tissue expression values meet the minimum and
				maximum criteria (<em>any</em>) or only if all tissue expression values meet 
				the search criteria (<em>all</em>).
				<P>
				<B>Saving filter settings</B><BR>
				The Browser provides a mechanism for saving filter settings for use in future
				sessions. To preserve the current filter configuration, click the Save Filter
				button on the Filter page. Type in a name for the filter, then click Save
				to save the filter and return to the Filter page.
				<P>
				<B>Loading a saved filter</B><BR>
				Once a filter configuration has been saved, you can retrieve it in later
				sessions by loading it back into your Browser. To load the saved filter
				settings, click the Load Filter button on the Filter page. Click on the
				name of the filter you wish to load, then click the Load button. Click
				the Submit button on the Filter page to apply the filter settings to the
				Browser.
				<P>
				<B>Viewing a list of saved filters</B><BR>
				To display a list of filter settings that you have saved, click the Save button
				on the Filter page. If you have any saved filters, the Browser 
				will display an Existing Setups list that shows the filter names. To
				permanently remove a filter from the list, click on the name to 
				highlight it, then click the Delete Existing Setup button.
			        </TD>
			        <TD WIDTH=15></TD>
				
			        </TD>
			        <TD WIDTH=15></TD>
			    </TR>
			</TABLE>
		    <BR></TD></TR>
		</TABLE>
	    </TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>

<!-----------Displaying sequence and text-based output--------------->
<A NAME="Download"></A><BR>

	<!--outer table is for border purposes-->
	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
	    <TR><TD>	
    		<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
		    <TR><TD>	
			<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
			    <TR><TD><FONT SIZE="4"><B><A NAME="TOC"></A><B>&nbsp; Displaying sequence and 
				text-based output</B></FONT>
			    </TD></TR>
			</TABLE>
			<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
			    <TR><TH HEIGHT=10></TH></TR>
			    <TR><TD WIDTH=10>&nbsp;</TD><TD>
				<P>
				The Brower's graphical presentation of data facilitates the visual
				observation of relationships and patterns among the genes in the
				display. However, it is often useful to convert the data to a text-based
				format that can be easily saved to a file or loaded into another program,
				database, or spreadsheet for further analysis. The Browser provides
				a mechanism for saving the current display in a tab-delimited text file or
				showing a text-based view of the sequence underlying the current display.
				<P>
				<B>Creating text-based output</B><BR>
				To output the current Browser table as text, click the text button at the
				top of the Browser page. The Browser will display each row of table data 
				on a separate tab-delimited line.
				<P>
				<B>Viewing the underlying sequence</B><BR>
				To display the protein, mRNA, or genomic sequence underlying the current
				Browser table, click the sequence button at the top of the Browser
				page. On the Get Sequence page, select the desired sequence configuration 
				settings that you'd like, then click the Get Sequence button. The Browser will 
				display a text-based list of FASTA format records for each gene displayed
				in the table. The FASTA records may be cut and pasted into
				Blat for further study.
			        </TD>
			        <TD WIDTH=15></TD>
			    </TR>
			</TABLE>
		    <BR></TD></TR>
		</TABLE>
	    </TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>

</BODY></HTML>
