<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>Human Genome Working Draft: User Guide</TITLE>
	<LINK REL="STYLESHEET" HREF="http://genome.ucsc.edu/style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 >
<TH COLSPAN=2 ALIGN="left">	
    <IMG SRC="/images/title.jpg">
</TH>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
   <CENTER>
	<TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
	<TABLE WIDTH="100%" BGCOLOR="#2636D1" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
		<TABLE WIDTH=431 CELLSPACING=4 CELLPADDING=0 BORDER=0><TR>
		<TD WIDTH=1></TD>
		<TD WIDTH=75>
			<A HREF="/index.html" onMouseOver="HOME.src='/images/hl_home.jpg'" onMouseOut="HOME.src='/images/h_home.jpg'">
			<IMG SRC="/images/h_home.jpg" ALIGN="absmiddle" BORDER="0" NAME="HOME" ALT="Home" ></A></TD>
		<TD WIDTH=155>
			<A HREF="/goldenPath/hgTracks.html" onMouseOver="BROWSER.src='/images/hl_browser.jpg'" onMouseOut="BROWSER.src='/images/h_browser.jpg'">
			<IMG SRC="/images/h_browser.jpg" ALIGN="absmiddle" BORDER="0" NAME="BROWSER" ALT="&nbsp;&nbsp; Genome Browser"></A></TD>		
		<TD WIDTH=100>
			<A HREF="/cgi-bin/hgBlat?command=start" onMouseOver="SEARCH.src='/images/hl_blat.jpg'" onMouseOut="SEARCH.src='/images/h_blat.jpg'">
			<IMG SRC="/images/h_blat.jpg" ALIGN="absmiddle" BORDER="0" NAME="SEARCH" ALT="&nbsp;&nbsp; BLAT Search"></A></TD>	
		<TD WIDTH=100>
			<A HREF="/FAQ.html" onMouseOver="GUIDE.src='/images/hl_faq.jpg'" onMouseOut="GUIDE.src='/images/h_faq.jpg'">
			<IMG SRC="/images/h_faq.jpg" ALIGN="absmiddle" BORDER="0" NAME="GUIDE" ALT="&nbsp;&nbsp; FAQ"></A></TD>
		</TR></TABLE>
	 </TD></TR></TABLE>
	 </TD></TR></TABLE>
	 </CENTER>
</TD></TR>

<TR>
<!--Content Tables-->  
<TD CELLPADDING=10 ROWSPAN=2 VALIGN="top">
<TR><TD CELLPADDING=10>	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><b>&nbsp;Human Genome Browser User Guide </B></FONT> 
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
			
<CENTER><TABLE BGCOLOR="ffffdd" BORDERCOLOR="cccc99" BORDER=1 CELLPADDING=3><TR><TD>	
	<CENTER><FONT SIZE="2">Web tool created by <A
	HREF="http://www.cse.ucsc.edu/~kent">Jim Kent</A>, <A
	HREF="http://www.cse.ucsc.edu/~sugnet">Charles Sugnet</A>, <A
	HREF="http://www.cse.ucsc.edu/~booch">Terry
	Furey</A>, <BR>and <A HREF="http://www.cse.ucsc.edu/~haussler">David
	Haussler</A> of UCSC,<BR> in collaboration with the <BR>
	<A HREF="http://www.nature.com/genomics/human/papers/409860a0_fs_1.html">genome analysis group of the International Human Genome Project</A><BR>
	and the <A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/">UCSC Human Genome Project</A>.<BR>
	Individual tracks contributed by various <A HREF="http://genome.cse.ucsc.edu/goldenPath/credits.html">individuals</A> and <A HREF="http://genome.cse.ucsc.edu/goldenPath/other.resources.html">institutions</A>.<BR>
	Last updated 14 Mar 02; questions and suggestions welcome.</FONT></CENTER>
</TD></TR></TABLE></CENTER>
<BR>
<B><FONT COLOR="#006666" SIZE=4> Table of Contents: </FONT></B>
<table border=0 nowrap cellpadding=0 cellspacing=0 margins=0><tr><th height=10></th></tr>
<tr><td><ul><b>
  		<li><A CLASS="toc" HREF="#What">
			What does it do?</A>
  		<li><A CLASS="toc" HREF="#GetStarted">
			Getting started: browser gateways</A>
  		<li><A CLASS="toc" HREF="#FineTuning">
			Fine-tuning the Display</A>
  		<li><A CLASS="toc" HREF="#IndivTracks">
			Details on individual tracks </A>
	</b></ul></td><td><ul><b>
  		<li><A CLASS="toc" HREF="#BLATAlign">
			BLAT alignments </A>
  		<li><A CLASS="toc" HREF="#TableBrowser">
			Table Browser </A>
		<li><A CLASS="toc" HREF="#TrackFormatDNA">
			Track-formatted DNA </A>
		<li><A CLASS="toc" HREF="#CustomTracks">
			Custom tracks </A>
	</b></ul></td></tr></table>
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="What"></A><BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b><a NAME="TOC"></a><b>&nbsp; What does it do?  </b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
<p>
As the human and mouse genome sequences near completion and research re-focuses on their analysis, the issue of effective sequence display becomes critical -- it is not helpful to have 3 billion letters of genomic DNA shown as plain text. As an alternative, the UCSC  Genome Browser provides a rapid and reliable display of any requested portion of the genome at any scale aligned in register with dozens of annotation tracks (known genes, predicted genes, ESTs, mRNAs, CpG islands, assembly gaps and coverage, chromosomal bands, mouse homologies, and more.). 
<p>
The genome browser stacks experimental and predicted features as tracks beneath genome coordinate position, encouraging rapid visual correlation of different types of information. The visitor can look at a whole chromosome to get a feeling for gene density, open a particular cytogenetic band to see posiltionally mapped disease gene candidatew, or zoom into a particular gene for its spliced ESTs and possible alternative splicing. The browser itself does not draw conclusions; rather, it collates all relevent information in one location, leaving exploration and interpretation to the visitor.
<P>
To control information overload, tracks need not be displayed in full -- options allow hidden, flattened, or filtered display according to user criteria.  An individual track will not display more than 300 items, reverting to dense mode despite a controller set at full. Zooming and scrolling controls also help narrow or broaden the chromosomal range displayed to just the desired region. 
<P>
A blue navigation bar link at the top allows convenient capture of raw genomic DNA sequence for the coordinate range displayed in the browser window. This DNA can encode track features via elaborate text formating options. Other links tie the genome browser to the BLAT alignment tool, provide table browser access to the underlying relational database, convert coordinates across different assembly dates, and open the window at the complementary <A HREF="http://www.ensembl.org/">Ensembl</A> annotation.
<p>
Clicking on an individual item in a track opens its details page with summaries of feature properties and links to off-site repositories such as PubMed, GenBank, LocusLink, and OMIM. Position, cytoband, strand, data source, and any encoded protein, mRNA, genomic sequence and alignment are furnished as appropriate to the nature of the track.
<P>
The browser data represents an immense <A HREF="http://genome.cse.ucsc.edu/goldenPath/credits.html">collaborative effort</A> involving thousands of people from the international biomedical research community. UCSC itself does no sequencing and creates only about half of the tracks in-house; its other major roles are building genome assemblies, creating the genome browser work environment, and serving it all online. Sequence data, annotation tracks, and even software are in the public domain (with rare caveats) and available to anyone from the <A HREF="http://genome.cse.ucsc.edu/#Downloading">download</A> page. 
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="GetStarted"></A><BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Getting started: browser gateways</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
<p>
To get oriented, it is best to jump to a gene or region of the genome that you are already familiar with. To do this, choose an assembly date from the <A HREF="http://genome.cse.ucsc.edu/">main page</A>, locate the position text box in the gateway page, and enter a gene symbol, mRNA or EST accession number, chromosome band, descriptive term likely to occur in GenBank text, or a specific chromosomal range (eg, chr20:1-25000), or simply accept the default. Proceed to the active track browser window and <B>experiment for 15 minutes</B> with the scroll adjustments, window width, track controllers, and auxillary links.
<p>
The genome browser sometimes returns several matches in response to a position search instead of an active window. Clicking on any item opens the track window to the indicated segment of the genome. Base position numbering begins at the telomere of short p arm at the far left. A track may be hidden, projected, displayed selectively according to a filter, or displayed in full according to how its controller is set.
<P>
Once the browser is correctly displaying the region of interest, save the coordinate range (or bookmark the page) if you plan to return to that view or share it, remembering that individual preference settings may vary.  Coordinates of a given feature on an unfinished chromosome will change with each genome assembly because of filled gaps, fewer artefactual duplications, and corrected strand orientations. Coordinates thus become obsolete; it is usually best to work with the most recent assembly even though a full set of tracks might not yet be ready. Note however the browser offers a tool that can correctly<A HREF="http://genome.cse.ucsc.edu/cgi-bin/hgCoordConv?origDb=hg8"> convert coordinates</A> between different assembly dates.
<p>
The keyword gateway doesn't alway find its query. It is not a sitewide search engine but instead derives primarily from GenBank mRNA records whose text annotations can include gene names, gene symbols, journal title words, author names, and RefSeq mRNAs (NM series). Queries such as post-freeze GenBank entries, withdrawn gene names, abandoned synonyms, and genomic intervals (eg, NT contigs) will not work. UniGene and NP protein accession numbers will shortly be accepted as queries. LocusLink and OMIM accessions are difficult to support being plain 4 to 6 digit numbers lacking diagnostic letters; however <A HREF="http://www3.ncbi.nlm.nih.gov/omim/">OMIM</A> links to <A HREF="http://www.ncbi.nlm.nih.gov/LocusLink/">LocusLink</A> which now carries map links to the UCSC browser. Additional text can be found in detail pages for individual features and in searchable non-positional tables such as refLink (see below).
<P>
 If you have genomic, mRNA, or protein sequence but no name or no idea where it maps in the genome, the <A HREF="http://genome.cse.ucsc.edu/cgi-bin/hgBlat?db=hg8">BLAT gateway</A> will rapidly locate it by homology alignment, provided the region has been sequenced. Close members of the gene family will be found at the same time, as well as assembly duplication artefacts. A whole set of query sequences can be looked up at once when provided in fasta format. In BLAT output, first look at details of the alignment and then procede into the browser itself if warranted by match quality. When too many hits occur, try resubmitting the query sequence after filtering in slow mode at <A HREF="http://ftp.genome.washington.edu/cgi-bin/RepeatMasker">RepeatMasker</A>.
<P>
The <A HREF="http://genome.cse.ucsc.edu/goldenPath/hgText.html">table browser</A>, a friendly portal to the underlying open source <A HREF="http://www.MySQL.com">MySQL</A> relational database driving the genome browser, allows genomic data to be viewed as columns of text rather than as graphical tracks. Select a positional table for any track and enter a chromosomal range (enter "genome" to get all). Choose fields to limit the columns displayed; get DNA to obtain fasta sequences. Positional tables have chrom, chromStart, and chromEnd fields which coalesce to a genome browser gateway position (eg chrX:12345-67890). Non-positional tables offer various kinds of accessory information needed to build the detail pages. The genome and table browsers are necessarily in synchrony with each other being just different views of the same fundamental MySQL database they reference. Technical explanations of individual fields are provided <A HREF="http://genome.cse.ucsc.edu/goldenPath/gbdDescriptions.html">onsite</A> but opening up that column usually offers adequate clues.
<P>
External gateways can also provide direct links into the UCSC genome browser, for example <A HREF="http://www.ncbi.nlm.nih.gov/LocusLink/">LocusLink</A>, <A HREF="http://www.ncbi.nlm.nih.gov/IEB/Research/Acembly/index.html">AceView</A>, <A HREF="http://www.ensembl.org/">Ensembl</A>, <A HREF="http://stash.mrc-lmb.cam.ac.uk/SUPERFAMILY/">SuperFamily</A>, <A HREF="http://www.genelynx.org/">GeneLynx</A>, and <A HREF="http://bioinformatics.weizmann.ac.il/cards/">GeneCards</A>. Journal articles can also link to the browser and provide custom tracks; be sure to use the assembly date appropriate to the provided coordinates.
<P>
To insure uninterupted browser services for your research during UCSC server maintainence and power outages, bookmark a geographically convenient <A HREF="http://genome.cse.ucsc.edu/mirror.html">mirror</A> site that replicates the UCSC genome browser.
<p>
Please bear in mind that the genome browser cannot outperform the underlying quality of the draft genome -- assembly errors and sequence gaps may still occur well into 2003 due to intrinsically difficult-to-sequence regions. Artefactual duplications arise as unavoidable compromises during a build, causing misleading matches in genome coordinates found by alignment. Post-freeze GenBank submissions lead to gradual track obsolescence (which can be addressed by supplemental <A HREF="http://www.ncbi.nlm.nih.gov/blast/index.nojs.cgi">Blast</A> searches). The genome browser does the best it can to operate under this environment.
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

	<A NAME="FineTuning"></A><BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Fine-tuning the track display: </b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
<p>
The browser track window opens to default values that may need adjustment. First set the pixel width to take advantage of your monitor. For example, setting a 19" monitor to 1100 pixels makes more of the genome visible without redraws. The maximum width supported is 5000 pixels but it is best not to exceed your monitor width initially. Adjust the track controllers according to your interest in individual tracks. The browser can remember many user preferences from session to session (however it never monitors visitor activities or records submitted data). Use the reset button periodically to restore default settings.
<p>
Tracks offer considerable control over what is displayed: 
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>If "<B>hide</B>" is chosen, the track will not be displayed at all. Someone not interested in SNPs or mouse synteny might want these tracks hidden to reduce track clutter and improve speed. Some tracks may pertain to only one chromosome (eg, the Sanger annotation of chromosome 20); no controller will display when other chromosomes are in the browser.</TD></TR>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>If "<B>dense</B>" is chosen, the track will display all relevent features condensed (projected) to a single line. To get a feel for known gene density without excessive detail, open a whole chromosome and set the known gene track to dense. </TD></TR>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>If "<B>full</B>" is chosen, the track will display a separate line for each constituent feature. Use this option sparingly as hundreds of ESTs might align with a given gene. When the number of requested tracks exceeds 300, the browser defaults to dense mode. Narrowing the chromosomal range or using a track filter (eg, restricting the EST track to a specified tissue or library) reduces the number of items displayed, re-enabling full display.</TD></TR>
</TABLE> 
<P>
Track filters are accessed from linked text above each controller underneath
the main graphic or from the mini-buttons to the left of the track in the 
main graphic.  Depending on the track, this new feature allows users to color a 
subset of individual items in a track and display them at
the top of the stack. This is helpful when too many items occur, especially
when only some are relevent to the current task.  For instance to hilight ESTs
expressed in the liver, put 'liver' into tissue control of the EST filters page.
Consult entries in the <a href="http://genome.cse.ucsc.edu/goldenPath/hgText.html">
table browser</a> for permissable filter word values.
For example the non-positional table "library" shows "NIH_MGC_67 " is an 
option for an EST filter.  You can use wildcards in filters as well.  "NIH_MGC*"
would hilight all ESTs generated by the Mammalian Gene Collection project.
In the filter screen there is also a description of the track, the method 
used in making it, and credit its source.  Filters, like all user interface
settings, are now persistent from session to session on the same web
browser.  Either delete all the text in the filter controls or use the 'reset all' 
button to remove filters.


<p>
Usually the amount of flanking region needs adjustment. Zoom in and out using the 1.5, 3 and 10-fold buttons available at the top of the browser, or scroll sideways by 10%, 50%, or 95% of the displayed size (given in base pairs). It is also possible to hold one edge fixed and adjust the other by any multiple of the vertical blue line interval. Another option is to manually adjust position range numbers and refresh the display with the jump button.
<p>
The browser graphic responds to mouse clicks on tracks and labels:
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>If the base position track is clicked, the display zooms in 3-fold, re-centered on the coordinate of the click. </TD></TR>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>If the click is on the optional explanatory label in the center of the display or left panel, the track toggles between full and dense mode. </TD></TR>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>If the track is already in full mode, clicking on a specific feature opens a details window with name, sequence, and off-site links. </TD></TR>
</TABLE>
<p>
At 1 pixel per base pair scale, the browser display of genes will accurately provide exon and intron width and direction of transcription by hatches for multi-exon features (consult the details page for strand of single exon genes). At larger scale, thin exons would soon disappear unless favored; this comes at the expense of intron width and overall display fidelity -- an exon may falsely appear to fall within a RepeatMasker feature at some scales. The browser seeks to maintain a useful view even at scales where a whole chromosome is shown in a 610 pixel window.
<p>
The genome browser can open from a lab web site or journal article link to preferred defaults to provide a consistent visitor experience. This also applies to custom tracks. A simpler static alternative is to save the browser window as a gif graphic. This can be done as a screen shot to embed chromosomal range, size, and scale. Browser text in the left panel is already rendered as image, meaning it cannot be copied into a text editor. However text can be extracted from "source" html or more conveniently from the table browser.
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

	<A NAME="IndivTracks"></A><BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><b>&nbsp; Details on individual tracks: </b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
<B>Remember:</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<A HREF="http://genome.cse.ucsc.edu/goldenPath/credits.html">Credit</A> goes to many individuals and institutions for generously contributing tracks.</TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	A track does not appear if its display mode is set to hide. </TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	If not interested in technical assembly details, hide the associated tracks.
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	A track set to full mode will display as dense if total tracks exceed 300.</TD></TR>  
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Clicking on central explanatory text toggles between full and dense display.</TD></TR> 
			<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Clicking a track feature or name in full display brings up a details page. </TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Only the three most recent assemblies are fully active; older assemblies are <A HREF="http://genome-archive.cse.ucsc.edu/">archived</A>.</TD></TR>  
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Not all tracks appear in all assemblies. Only basic tracks appear at first in a new assembly.</TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Tracks can be viewed equivalently in columns as text tables. Individual field names (columns) are explained <A HREF="http://genome.cse.ucsc.edu/goldenPath/gbdDescriptions.html">elsewhere</A>.</TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	The tracks briefly described below are those found in recent assemblies. Further information can often be found on the track filter pages. </TD></TR> 
</TABLE>
<P>
<B>Base Position</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Shows chromosomal coordinates as numbered from the telomere of the short p arm. Coordinates change in new assemblies.
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Clicking here zooms in 3x centered about click. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	On or off, no dense mode</TD></TR>
</TD></TR></TABLE></P>

<P><B>Chromosome Bands</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Light and dark blocks show traditional cytological bands seen with Giemsa staining 
	<A HREF="http://www.ncbi.nlm.nih.gov:80/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11237021&dopt=Abstract">
	as mapped</A> by FISH of large-insert clones with sequence tags, STS markers, and BAC 
	end sequence locations giving a resolution of 862 bands genomewide. The uncertainty of 
	mapping to band staining is still a megabase or more. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Clicking on a band in full display brings up details, a discussion of mapping methods used, and 
	lab credits. The corresponding table, cytoBand, in the table browser provides the complete data set.
</TD></TR></TABLE></P>

<P><B>FISH Clones</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Shows location of fluorescent in situ hybridization (FISH) mapped clones along the 
	draft assembly sequence used for cytogenetic map. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Clicking on a specific marker in full display brings up information on clone placement, 
	clone registry, its STS markers, and BAC end sequences.</TD></TR>
</TABLE></P>

<P><B>STS Markers</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Shows location of sequence tagged site (STS) markers on genetic maps 
	(Genethon and Marshfield) in blue and radiation hybrid maps (Stanford, Whitehead 
	RH, and GeneMap99) in black. FISH-mapped clones first received their own track 
	in the August 01 assembly.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Clicking on a specific marker in full display 
	brings up synonyms, primers, YAC map position and uniSTS link.</TD></TR>
</TABLE></P>

<P><B>GC Percent</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Darker shades of gray show higher GC content, as figured from chromosome 
	partition into 20,000 bp blocks. Both Celera and public consortium reported 
	isochore concept in conflict with human genome sequence data. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Clicking in full mode provides minimal additional information.</TD></TR>
</TD></TR></TABLE></P>

<P><B>Map Contigs</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Starting with the 6 August 01 assembly, this track uses tiling path fingerprint 
	(TPF) maps curated by sequencing centers responsible for individual chromosomes to 
	merge with the fingerprint contig (FPC) map used in earlier assemblies (see 
	<A HREF="http://www.genome.wustl.edu/gsc/">WU GSC</A>) to determine the UCSC assembly. 
	In the event of conflict, the TPF map is used.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Clicking in full mode brings up a brief description of the extent and number of 
	clones in contig.</TD></TR>
</TD></TR></TABLE></P>

<P><B>Assembly</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	An individual box shows the assembly extractable from a single clone fragment. 
	Gaps in the path appear as white space between the boxes at an adequate zoom level, 
	with a thin horizontal line connecting boxes bridged over a gap.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
	Clicking in full mode brings up a brief description of how clone fragment 
	parts were used in golden path.</TD></TR>
</TD></TR></TABLE></P>

<P><B>Gap</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Shows locations of gaps in the assembly in black. Small gaps can artefactually 
	coalesce in the graphic at coarse scales. Gaps bridged by mRNAs, ESTs or paired BAC 
	end reads appear horizontally split. Strand misorientation errors can occur adjacent 
	to unbridged gaps. Gaps are assigned somewhat arbitrary sequence sizes (run's of N's 
	in genomic DNA); uncertainty is less with bridged gaps. A gap within contig fragments 
	of a clone is typically smaller than a gap between draft clones. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
	In full mode, clicking brings up a specific description of the gap. Unfinished 
	clones may have gaps of uncertain size between fragments.</TD></TR>
</TD></TR></TABLE></P>

<P><B>Coverage</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	In dense display, increasing level of gray indicates improved coverage:<BR>
	<BLOCKQUOTE>
		White: no coverage (gap) <BR>
		Light Gray: predraft (less than 4x shotgun) <BR>
		Medium Gray: draft (at least 4x shotgun) <BR>
		Dark Gray: covered by multiple draft clones <BR>
		Black: covered by a finished clone
	</BLOCKQUOTE>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
	In full display, each clone coverage is represented as a double gray box. 
	The fragments inside a clone are shown as black boxes; their position corresponds 
	to assembly placement including strand assignment.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
	Clicking in full mode brings up clone hyperlink to Genbank record and fasta 
	format sequence (one per fragment). The associated table is called clonePos.</TD></TR>
</TD></TR></TABLE></P>

<P><B>BAC End Pairs</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Shows genome sequence in bacterial artificial chromosomes (BACs) sequenced a 
	short way in from both ends to place full insert on the assembly. A valid BAC 
	end pair has 50-600Kb of separation and opposite orientations. Strand assignment 
	follows <a href="http://www.ncbi.nlm.nih.gov/genome/clone/">Clone Registry</a> 
	storage convention. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
	Clicking in full display brings up a wealth of detail including any alignment 
	ambiguity. </TD></TR>
</TD></TR></TABLE></P>

<P><B>Duplications </B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Shows genomic 
	<a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11381028&dopt=Abstract">
	duplications</a> greater than 1,000 bp if not overly repeat-masked: 
	as red if 99%+ identity (likely unmerged assembly overlap artefact), as yellow 
	for 98-99% identity (origin uncertain), and as gray for 90-98% identity (below 
	draft sequencing error, often genuine pericentromeric or subtelomeric duplication). 
	Duplications are more common in unordered and unassigned contigs (ie, difficult 
	to assign to their proper position). </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
	In the left panel of the browser, duplications are named by the chromosome and start 
	of the other member of the duplication pair. Clicking on an individual duplication 
	gives details of both locations and match quality parameters. </TD></TR>
</TD></TR></TABLE></P>

<P><B>Known Genes </B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Shows known protein coding genes taken from mRNA reference sequences compiled at 
	LocusLink (interim FLJ-prefixed entries await a permanent gene name and symbol). 
	Coding exons are displayed taller than 5' and 3' untranslated regions (UTR); 
	connecting introns are one-pixel lines with hatch marks indicating direction of 
	transcription. Non-coding RNA genes have their own track in some assemblies. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
	Clicking in full display brings up the corresponding protein, mRNA, and genomic 
	sequence and links to outside databases. </TD></TR>
</TD></TR></TABLE></P>

<P><B>Acembly Genes</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	A predicted gene track showing gene models reconstructed solely from mRNA and EST data. 
	The same track display conventions as known genes are used. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
	Clicking in full display brings up basic track information plus a link to an extensive 
	text and graphical annotation of supporting data at NCBI's 
	<A HREF="http://www.ncbi.nlm.nih.gov/IEB/Research/Acembly/index.html">AceView</A></TD></TR>
</TD></TR></TABLE></P>

<P><B>Ensembl Genes</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	A predicted gene track from <A HREF="http://www.ensembl.org">Ensembl</A> 
	based on Blast matches to various EMBL databases of GenScan predictions. Track 
	display conventions are the same as for known genes. 
<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
Clicking in full display brings up basic track information plus a link to detailed supporting evidence.</TD></TR>
</TD></TR></TABLE></P>

<P><B>Fgenesh++ Genes</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	A predicted gene track from <A HREF="http://www.softberry.com">Softberry</A> 
	based on hidden Markov models (HMM) and protein similarity, with less emphasis on 
	cDNA/EST data . The same track display conventions as Known Genes are used.  </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
	Clicking in full display brings up basic track information plus a link to the 
	best matching protein at GenBank. </TD></TR>
</TD></TR></TABLE></P>

<P><B>Genscan Genes</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	An empirical predicted gene track from <A HREF="http://genes.mit.edu/GENSCAN.html">
	GenScan</A> based on transcriptional, translational and donor and acceptor splicing signals 
	plus and length and compositional distributions of exons, introns and intergenic regions; 
	does not use similarity to known proteins like 
	<A HREF="http://www.ncbi.nlm.nih.gov:80/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11337476&dopt=Abstract">
	GenomeScan</A>. Track display conventions are those of known genes. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
	Clicking in full display brings up the final prediction without GenScan program 
	details on exon phases and coordinates -- the BLAT details page can provide these. </TD></TR>
</TD></TR></TABLE></P>

<P><B>Tigr Gene Index</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	The <A HREF="http://www.tigr.org/tdb/hgi/index.html ">TIGR Gene Index</A> is 
	based mainly on assembled EST sequences from public databases and TIGR. Entries are 
	actual assemblies providing a consensus sequence, not simply clusters of overlapping 
	sequences, but don't generally predict whole genes. Gene indexes for mouse, rat, cow, 
	and pig also contribute to the track. </TD></TR>
</TD></TR></TABLE></P>

<P><B>Rosetta</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	This innovative track displays microarray tissue 
	<A HREF="http://www.ncbi.nlm.nih.gov:80/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11237012&dopt=Abstract">
	expression data</A> provided by <A HREF="http://www.rii.com">Rosetta Inpharmatics</A>. 
	It helps catalogue genes and transcriptome, complements computational gene predictions, 
	and seeks to define gene boundaries via co-regulated expression of nearby exons. The data 
	is stratified by various tissue and disease-specific conditions of gene expression. 
	Confirmed and predicted exons are placed in separate tracks. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking in full mode on a Rosetta exon brings up an averaged display that can be 
	expanded via checkboxes to a graph of actual intesities for an individual experiment.</TD></TR>
</TD></TR></TABLE></P>

<P><B>UniGene</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
 	This track shows <A HREF="http://www.ncbi.nlm.nih.gov/UniGene/">UniGene</A> 
 	clusters of ESTs aligned to the genome using BLAT. This data imperfectly indicates 
 	exon/intron boundaries and to a lesser extent genes. The Serial Analysis of Gene 
 	Expression <A HREF="http://www.ncbi.nlm.nih.gov/SAGE/ ">(SAGE)</A> project at NCBI, 
 	a quantative measurement of gene expression in various tissues and disease states, 
 	mapped small 10bp bar-coded expression markers onto these UniGene clusters in the 
 	SageMap project. The genome browser uses this mapping to provide access to the SAGE 
 	data associated with the UniGene track in the assembled and annotated genome. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking on a UniGene feature calls up an elaborate display of tabular and 
	graphical data from the SAGE project relevent to that cluster.</TD></TR>
</TD></TR></TABLE></P>

<P><B>Human mRNAs</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
 	GenBank mRNA are aligned to the genome, typically giving a set of blocks 
	(exons) connected by one pixel lines (spliced-out introns). In full display, 
	hatches on the introns indicate the direction of transcription. Some mRNAs align 
	to multiple locations with varying degrees of match quality; only near-perfect 
	are retained.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking in full display mode brings up a summary of the Genbank entry on 
	that mRNA, plus convenient hyperlinks to major databases, along with base-by-base 
	alignments. </TD></TR>
</TD></TR></TABLE></P>

<P><B>Spliced ESTs</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	This track shows a highly relevent subset of human ESTs, those that are
	spliced. Quality control has excluded  some ESTs that seem to be spliced
	but whose exons and introns are more likely assembly or mismatch artefacts.
	Requiring introns eliminates more than 99% of the contamination of dbEST by
	genomic sequence. This track may also suggest alternate splices. The GT-AG
	splice donor/acceptor rule has not been applied.  Single-exon genes and 3'
	UTR ESTs are located in the all-inclusive EST track.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	In dense mode, the degree of gray in the track 
	indicates the extent of spliced ESTs projecting to the region. In full mode only 
	300 tracks can be displayed in the browser, so zoom in to a smaller region to see 
	individual ESTs. Blue and red spliced ESTs indicate 5' and 3' read directions, 
	respectively. Black indicates no read direction was available in the GenBank entry. 
	Hatch marks indicate direction of transcription according to the entry; these should 
	be verified in the details page alignment on the basis of the GT-AG splice 
	donor/acceptor rule. Clicking on an individual mRNA provides sequence, accession number, 
	GenBank highlights, and Medline link.</TD></TR>
</TD></TR></TABLE></P>

<P><B>Human ESTs </B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	This track shows all human expressed sequence tags (ESTs) at GenBank aligned to the 
	genome assembly, spliced and unspliced. It may be necessary to filter or zoom in to 
	open this track in full mode because the browser will display in dense mode whenever 
	the number of total tracks exceeds 300. Color conventions and hatching are the same 
	as for the spliced EST track. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking in full display provides sequence, accession number, GenBank highlights, 
	Medline link. </TD></TR>
</TD></TR></TABLE></P>

<P><B>CpG Islands  </B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	This track shows CpG islands, areas rich in the dinucleotide CG often found around the 
	promoter/ first exon of housekeeping genes (where they are protected from methylation 
	and subsequent deamidation of the C). CpG islands are defined here as 200 or more bases 
	with 50% or higher GC content having CpG dinucleotides significantly disproportionate 
	to this composition. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Darker green shows CpG islands exceeding 400 occurences. Clicking in full mode brings 
	detailed information and sequence for individual CpG islands. </TD></TR>
</TD></TR></TABLE></P>

<P><B>Nonhuman mRNAs </B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	This track places GenBank mRNAs from other species onto the genome using BLAT, 
	with increasing level of gray indicating better quality of match. For a distant vertebrate, alignment 
	requires a fairly slowly evolving gene. The matches are often plausible ortholog 
	candidates. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking in full display brings up a summary of the Genbank entry and links to other 
	major databases. Clicking on an mRNA/genomic alignment brings up a base-by-base 
	alignment. Some mRNAs align to multiple places but only the near-best are retained. </TD></TR>
</TD></TR></TABLE></P>

<P><B>Nonhuman ESTs </B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	This track shows all alignable nonhuman expressed sequence tags (ESTs) at GenBank, 
	with level of gray indicating quality of match. Matches help validate human exon 
	identification. It may be necessary to filter or zoom in to open this track in full 
	mode as the number of total tracks can exceed 3. Other conventions and hatching 
	are the same as for the human EST track.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking in full display brings up a summary of the Genbank entry plus hyperlink to 
	that full record and launches of PubMed searches by gene name, product, and author. 
	The mRNA/genomic alignment offers a base-by-base alignment. Some ESTs align to 
	multiple places in the genome but only the near-best are retained. </TD></TR>
</TD></TR></TABLE></P>

<P><B>Mouse BLAT </B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	This track displays alignments of mouse genome to human. The mouse genomic 
	DNA is from the Arachne assembly of mouse whole genome shotgun reads available 
	at the <A HREF="http://www.ncbi.nlm.nih.gov/Traces/">Trace Repository</A>. 
	The alignments were done with BLAT in translated protein mode requiring 2 nearby 
	4-mer matches to trigger a detailed alignment. The human genome was masked with 
	<A HREF="http://ftp.genome.washington.edu/cgi-bin/RepeatMasker">RepeatMasker</A> 
	and <A HREF="http://c3.biomath.mssm.edu/trf.html">Tandem Repeat Finder</A> prior 
	to running BLAT. Regions with more than 300 aligned reads were assumed novel repeat 
	elements and filtered out. Alignments of greater than 97% identity may represent 
	mouse contamination of human or vice versa.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking calls up details on mouse sequence fragment and its alignment details. </TD></TR>
</TD></TR></TABLE></P>

<P><B>Mouse Synteny </B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	This track shows the syntenic chromosomal region in mouse if known. NCBI 
	<A HREF="http://www.ncbi.nlm.nih.gov/Homology/">made</A> by seeking patches 
	of homology of mapped adjacent mouse genes to the draft human assembly. Mouse 
	is provided at the level of chromosome (not cytoband); the proteins used are a 
	subset of the displayed human orthologs.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking in full mode calls up details of the correspondence. This track 
will be greatly improved with the availability of a mouse genome assembly.</TD></TR>
</TD></TR></TABLE></P>

<P><B>Exofish ecores</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	The track displays conserved regions, generally protein-coding homologs, 
	from a genomics project of long random reads in the 
	<A HREF="http://www.genoscope.cns.fr/exofish">pufferfish</A>, Tetraodon 
	nigroviridis. Regions of greater homology are displayed in a darker blue. 
	Ecore is an acronym for <B>e</B>volutionary <B>co</B>nserved <b>re</b>gion; the data was 
	<A HREF="http://www.ncbi.nlm.nih.gov:80/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=10835645&dopt=Abstract">
	used</A> to estimate the number of human genes. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking in full mode brings up information on a particular ecore, access 
	the relevent Genoscope page, and the relevent DNA.</TD></TR>
</TD></TR></TABLE></P>

<P><B>RNA Genes</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	This track shows human non-protein-coding RNA genes and RNA pseudogenes. This 
	includes tRNAs, rRNAs, SRPs, methylation guide snoRNAs, U2-U6 RNA-like, and 
	similar elements both known and predicted (by tRNAscanSE etc). The density of 
	these elements can range from 1-4 per million bp in some chromosomes. (Protein 
	coding genes are displayed elsewhere, in the Known Genes track.)</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking in full mode brings up feature classification, method of prediction,
	 location, and reliability score. </TD></TR>
</TD></TR></TABLE></P>

<P><B>Overlap SNPs</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	This track shows single nucleotide polymorphisms found on overlapping contigs 
	that are useful markers in mapping genes and diseases using data provided by the 
	<A HREF="http://snp.cshl.org/">SNP Consortium</A>.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking in full mode brings up a location description and 
<A HREF="http://www.ncbi.nlm.nih.gov/SNP/">dbSNP</A> link.</TD></TR>
</TD></TR></TABLE></P>

<P><B>Random SNPs</B><BR> 
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	 This track displays single nucleotide polymorphisms found by random 
 	sequencing that are useful markers in mapping genes and complex traits 
 	using data provided by the <A HREF="http://snp.cshl.org/">SNP 
 	Consortium</A>.</TR></TD>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
 	Clicking in full mode brings up a location description 
 	and <A HREF="http://www.ncbi.nlm.nih.gov/SNP/">dbSNP</A> link.</TD></TR>
</TD></TR></TABLE></P>

<P><B>RepeatMasker</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	The track shows dispersed repeats as determined by 
	<A HREF="http://ftp.genome.washington.edu/cgi-bin/RepeatMasker">RepeatMasker</A>. 
	These elements include SINE, LINE, LTR, DNA, simple, low complexity, 
	micro-satellite, tRNA, and other repeat families. Darker gray indicates a more 
	recent age (as dated by alignment to a canonical repeat).</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	In dense mode for large chromosomal regions, the black boxes representing repeats 
	may merge into misleading single blocks in the display. Full mode shows 9 tracks 
	for the major repeat classes.</TD></TR>
</TD></TR></TABLE></P>

 <P><B>Simple Repeats</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
 	This track displays simple tandem repeats (possibly imperfect) located by 
 	<A HREF="http://c3.biomath.mssm.edu/trf.html">Tandem Repeat Finder </A> which 
 	is specialized to this purpose. These can occur within coding regions of genes, 
 	may be quite polymorphic, and repeat expansions are sometimes associated with 
 	specific diseases.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	 Clicking in full mode brings up the simple repeat sequence, copy number, 
 	and quality of match statistics.</TD></TR>
</TD></TR></TABLE></P>

 <P><B>NCI60</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
 	This track displays cDNA expression data for a micro-array of 8,000 genes for 60 cell lines used in an <A HREF="http://genome-www.stanford.edu/nci60/">NCI cancer drug screen</A>.  To understand the track, experiment with it set to full mode but other tracks (except known genes) hidden, while reading its<A HREF="http://genome.cse.ucsc.edu/cgi-bin/hgTrackUi?hgsid=1017057&c=chr3&g=nci60"> filter page</A>. Each column of colored boxes represents variation in transcript levels for a given cDNA across array experiments; each row represents the measured transcript levels for all genes in a single sample. The saturation of a color corresponds to the magnitude of transcript variation. Black  indicates an undetectable change in expression; gray shows missing data. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	 Clicking on an individual track item in full mode open a rich details page. The exon probe and experiment selected are highlighted in blue under item name. Use filter page options to limit tissue types displayed and change the color scheme of the display. </TD></TR>
</TD></TR></TABLE></P>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

	<A NAME="BLATAlign"></A><BR>	

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; BLAT Alignments </b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
To locate a nucleotide or protein within the human genome, paste in a sequence or fasta formatted list (header lines start with > and contain a unique name), select the preferred assembly date, sort order, and output option. Numbers, spaces, and extraneous characters are ignored. Alternatively, upload an unformatted plain text file containing the sequences from your desktop using the file browse button.
<P>
A single DNA sequence over 20,000 bp will be rejected, but with multiple sequences 50,000 bp is acceptible (eg, 25 sequences of length 2,000). For protein, these limits are 4,000 and 12,500 amino acids, respectively.  
<P>
For a DNA query, BLAT quickly finds sequences with 95% or greater similarity over 40 bp or more. BLAT may not locate more diverged or shorter genomic sequences, though perfect sequence matches of 33 bp are found and sometimes as few as 22 bp. Large inserts (eg, introns) do not throw off alignment. A primate query will generally work but more remote tetrapod DNA will often be too distant.
<P>
For a protein query, BLAT rapidly locates genomic sequences with 80% or greater similarity over 20 amino acids or more. This means gene family members that arose within the last 350 million years can generally be detected. More diverged sequences can be aligned to human genome using NCBI's Blast and psi-Blast followed by BLAT of the resulting match into the UCSC genome assembly.
<P>
A single query, whether DNA or protein, can generate multiple hits. This happens when the genome contains multiple copies, paralogs, pseudogenes, statistical coincidences, artefactual assembly duplications, or when the query itself contains repeats or common retrotransposons. When several nearby BLAT matches occur on a single chromosome, a simple trick can quickly adjust the genome browser view to display them all: open the match having the smallest initial coordinate, then paste in the most distant coordinate from the BLAT output page and refresh the browser view. 
<P>
Common uses of BLAT are finding genomic coordinates of a mRNA or protein within the current assembly, determining exon structure of a gene, displaying a coding region within a full length gene, isolating an EST of special interest as its own track, searching for gene family members, and finding human homologs of a query from another species. Set the sort order option at the time of query submission to reflect your research interests. The highest scoring output line is the closest in sequence to the query. A BLAT track can be added at any time from within a genome browser view or used simultaneously with custom tracks. 
<P>
To make a custom track directly from BLAT, select the<A HREF="http://genome.cse.ucsc.edu/goldenPath/help/customTrack.html"><!--CHUCK: this link doesn't actually have any information about the psl format--> psl format</A> output option. The custom track can be launched by pasting it into the bottom of the main browser gateway page.
<P>
Ordinary BLAT output is a flat database file that summarizes any alignments found and provides further links to sequence alignment details and the appropriate genome browser view. The details link gives the letter-by-letter alignment to genomic. The score is figured as matches - mismatches in the final alignment of query to genome. Details of BLAT's internal scoring schemes and its overall n-mer alignment seed strategy will be described shortly in a paper by WJ Kent.
<p>
BLAT is not BLAST. With DNA, BLAT uses an index of the entire genome kept in memory. The index consists of all non-overlapping 11-mers except for those heavily associated with repeats. The index occupies a gigabyte of RAM. Once index matches are found, corresponding regions of genomic sequence are then loaded into memory for a detailed alignment. Thus the genome itself need not be kept in memory. With protein queries, BLAT works similarly but with 4-mers of genome translated in all frames. The protein index for a mammalian genome requires 2.5 gigabytes of RAM and a separate server. 
<P>
The BLAT servers are <A HREF="http://genome.cse.ucsc.edu/mirror.html">mirrored</A> at several institutions to ensure continuous availability. For large batch jobs and internal parameter changes, it is best to install command line BLAT on your own Linux server. Sources and executables are free for academic, personal, and non-profit purposes. Non-exclusive commercial licenses are also available; contact <A HREF="http://www.cse.ucsc.edu/~kent">Jim Kent</A>. 
<P>
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="TableBrowser"></A><BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; Getting started on the Table Browser  </b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
The table browser allows convenient and precise access to the primary database text tables containing the human genome sequence and associated annotation tracks. By specifying chromosomal range, table type, and desired columns, the exact data set of interest can be displayed without information overload. This is equivalent to but far more convenient than downloading and manipulating the whole human genome file with its multi-gigabyte data tracks (though that option always remains <A HREF="http://genome.cse.ucsc.edu/index.html#Downloading">available</A>.)
<P>
After each round of genome assembly, features such as mRNAs are located within the genome by BLAT alignment. This process generates <B>positional</B> stop-start coordinates and other descriptive data which are then stored within MySQL relational database tables. It is these tables that drive the graphical tracks in the Genome Browser, meaning that the two views of the data, tabular or graphic, are always in agreement (a week's lag can occur in downloadable files). Chromosomal coordinates usually change with each build because of newly filled gaps or assembly procedure refinements.
<P>
Other data is inherently <B>non-positional</B> (not tied to genome coordinates), such as submitting author of a GenBank EST, tissue of origin , or link to a RefSeq. These data types are accessed separately.
<P>
Familiarize yourself with the table browser tool by a <B>few minutes of exploration</B>. Starting at the top of the form, use the default position (or enter a cytological band, eg 20p12, or type in a keyword from a GenBank entry to look up). Notice that if you look up a gene, the tool can return a choice of several mRNAs. Clicking on one automatically enters the chromosomal range in the text box.Next select a positional table from the menu, for example, chrN_mrna which holds mRNA alignment data broken down by individual chromosome. Now choose the fields you wish to view, or bypass this step to see all, or view a fasta list of relevent sequences without any table attributes other than chromosome, freeze date, and start-stop genomic coordinates. If the chromosomal range is too narrow, it may happen that no mRNA occurs there (meaning no data will be returned).
<P>
Visitors not already familiar with what the table offers should select "Choose fields". Some tables offer many columns of data (explained <A HREF="http://genome.cse.ucsc.edu/goldenPath/gbdDescriptions.html">elsewhere</A>) of which only a few might be relevent. The fewer fields selected, the simpler the next stage of data return. UCSC plans to add record filtering shortly. This will reduce the size of huge tables by allowing both row and column selection.
<P>
Limited boolean logic processing (eg, joins of two tracks) may be offered in the future. For now, post-processing of table browser output can be done in the visitor's spreadsheet or database. These steps can produce output in the format needed for custom tracks. 
 <P>
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="TrackFormatDNA"></A><BR>	

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; Track-formatted DNA </b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
		<P>
		This aspect of the browser allows features to be displayed via text formatting of the DNA underlying a browser window. It is accessed from the navigation bar at the top of the main browser or from appropriate detail pages. Formatting can range from simply putting exons in upper case to elaborately mark-up of sequence according to multiple track data -- DNA covered by various tracks can be highlighted by case, underlining, bold, italic, and color.  
		<P>
		Color values range from 0 (darkest) to 255 (lightest) and are additive (red and green make yellow etc.). The examples below show a few ways to highlight individual tracks and their interplay. Keep it simple at first -- it is easy to make pretty but completely cryptic displays. 
		<P>
		<TABLE>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>To put exons from the Known Genes track in upper case red text, check the appropriate box in the Toggle Case column and set the color to pure red, RGB (255,0,0). Upon submitting, any known gene within the designated chromosomal interval will appear in red capital letters. 
		</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>To see the overlap between Known Genes and Genscan predictions, try setting Known Genes to red (255,0,0) and Genscan to green (0,255,0). Places where the Known Genes and Genscan overlap will be painted yellow (255,255,0). 
		</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>To get a level-of-coverage effect for tracks like Spliced Ests with many overlapping features in full mode, initially select a darker color such as deep green, RGB (0,64,0). Nucleotides covered by a single EST will appear dark green, while regions covered with more ESTs get progressively brighter -- saturating at 4 ESTs.
		</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>To use a track to mask unwanted features, for example set the RepeatMasker track to RGB (255,255,255) to white-out Genscan predictions of LINEs that are not mainstream host genes. Similarly, masking with Known Genes will show what is new in the gene prediction sector. 
		</TABLE>
		<P>
		The default line width of 60 characters is conventional; set this higher to see more DNA at once with fewer line breaks  if you plan on locating specific sequences with your web browser search tool. 
		<P>
		Be careful about requesting complex formatting for very large chromosomal regions. After all the html tags are added to the output page, the file size may exceed size limits that your browser, clipboard, and other software can safely display. The tool will format about 10 Mbp.</P>
		<P>
		Now copying and pasting the web page output to a text editor such as Word will retain upper case but lose colors and other formatting. That's still useful because other web tools such as <A HREF="http://www.ncbi.nlm.nih.gov/blast/index.nojs.cgi">Blast</A> can be set to ignore lower case. To fully capture formatting such as color and underlining, view the output as "source" html in your web browser, copy the output page into an html editor, and display on a web page. It is also possible via selective deletion to process formatted text into a Blat query that will result in a custom track.
		<P>
		The items in the track column usable for masking are auto-generated and so correctly correspond to browser tracks currently offered for particular assembly dates. Formatting is available on all actively supported assemblies. 
		<P>
		Further features under consideration include a menu of pre-cooked palettes that do common tasks effectively (to avoid re-entry of a successful but complex coding scheme) and a legend gif on the output page displaying track formattings chosen.
		<P>
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="CustomTracks"></A><BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; Custom Tracks </b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
		<P>
		Custom tracks are a very important research feature of the browser. Because space on the main genome browser is very tight, some excellent potential genomewide tracks cannot be offered. Researchers also develop track data for a portion of a chromosome that might not be of general interest or that cannot be shared until journal publication.
		<P>
		Custom tracks work very similarly to ordinary tracks, except that they never become part of the MySQL genome database. They have their own controller and persist as long as their launching window is kept open<!--CHUCK:this is WRONG they are kept up to 8 hours from the last access-->, even if some portion of the genome is displayed where they are not represented.  Custom tracks are normally aligned under genomic sequence but they can also be completely unrelated to human data and run under strings of millions of N's that occur in some assemblies.
		<P>
		Custom tracks are made in one of four <A HREF="customTrack.html">formats</A>  adhered to fairly strictly. The easiest way to make correctly formatted custom tracks is to collect Blat psl output.  Advanced users can make custom tracks from the table browser and track-formatted DNA. To make a custom track from scratch, begin with the examples and experiment with new lines and altered values in a spreadsheet or text editor. 
		<P>
		A custom track can be launched by pasting or uploading it into the bottom of the main browser gateway page.  An external custom track (a  correctly formatted plain text web page) can be included by adding a separate line starting with http://   Custom tracks can appear in conjunction with ordinary Blat tracks.
		<P>
		After getting your data to display within the UCSC genome browser, you may wish to create your own detail pages from individual track items.  This can be done using an attribute of the Track Lines formatting instructions: "url - A URL for an external link associated with this track. Any '$$' in this string this will be substituted with the item name."  This creates a chimeric browser, with browser display at UCSC but details at the visitor's site (or some other database).  All the detail pages can reside in a single document if the name denotes an html anchor (eg, #name).
		<P>
		To share your custom tracks with the biomedical research community, construct a link from your own web pages to the browser based on this <!--CHUCK:the example doesn't seem to work, there is no test.bed--><A HREF="http://genome.ucsc.edu/cgi-bin/hgTracks?position=chr22:1-20000&db=hg8&hgt.customText=http://genome-test.cse.ucsc.edu/test.bed">example</A>. The position variable tells the browser which part of the genome to display, here chr22:1-20000. The db variable provides the freeze number with  'hg8' corresponding to the  August 2001 assembly (only the last 3 are supported). The customText variable refers to any web URL, in the example a UCSC <A HREF="http://genome-test.cse.ucsc.edu/test.bed">page</A> containing a custom track as plain text in one of the <A HREF="customTrack.html">allowed formats</A>. The visitor normally would change this link to their own track data site.
		<P>
<tt>http://genome.ucsc.edu/cgi-bin/hgTracks?position=chr22:1-20000<br>&amp;db=hg8&amp;hgt.customText=http://genome.cse.ucsc.edu/test.bed</tt>
		<P>
		Please <A HREF="mailto:genome@cse.ucsc.edu">send in</A> links to tracks you'd like to share, along with a one paragraph description of the track and the genome version it is tied to. UCSC plans to create a gallery of examples of supplemental and interesting custom tracks. 
 		<P>
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

</TD></TR></TABLE>
</BODY></HTML>
