<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>Human Genome Working Draft: User Guide</TITLE>
	<LINK REL="STYLESHEET" HREF="http://genome.ucsc.edu/style/HGStyle.css">
	<BASE HREF="http://www.cse.ucsc.edu/~learithe/browser/">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 >
<TH COLSPAN=2 ALIGN="left">	
    <IMG SRC="images/title.jpg">
</TH>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
   <CENTER>
	<TABLE BACKGROUND="images/hl_mid.jpg" BGCOLOR="253BDE" WIDTH=100% CELLSPACING=0 CELLPADDING=0 BORDER=0 HEIGHT="22"><TR><TD>
		<TABLE WIDTH=431 CELLSPACING=4 CELLPADDING=0 BORDER=0><TR>
		<TD WIDTH=1></TD>
		<TD WIDTH=75>
			<A HREF="index.html" onMouseOver="HOME.src='images/hl_home.jpg'" onMouseOut="HOME.src='images/h_home.jpg'">
			<IMG SRC="images/h_home.jpg" ALIGN="absmiddle" BORDER="0" NAME="HOME" ALT="Home" ></A></TD>

		<TD WIDTH=155>
			<A HREF="goldenPath/hgTracks.html" onMouseOver="BROWSER.src='images/hl_browser.jpg'" onMouseOut="BROWSER.src='images/h_browser.jpg'">
			<IMG SRC="images/h_browser.jpg" ALIGN="absmiddle" BORDER="0" NAME="BROWSER" ALT="&nbsp;&nbsp; Genome Browser"></A></TD>
			
		<TD WIDTH=100>
			<A HREF="cgi-bin/hgBlat?command=start" onMouseOver="SEARCH.src='images/hl_blat.jpg'" onMouseOut="SEARCH.src='images/h_blat.jpg'">
			<IMG SRC="images/h_blat.jpg" ALIGN="absmiddle" BORDER="0" NAME="SEARCH" ALT="&nbsp;&nbsp; BLAT Search"></A></TD>
			
		<TD WIDTH=100>
			<A HREF="FAQ.html" onMouseOver="GUIDE.src='images/hl_faq.jpg'" onMouseOut="GUIDE.src='images/h_faq.jpg'">
			<IMG SRC="images/h_faq.jpg" ALIGN="absmiddle" BORDER="0" NAME="GUIDE" ALT="&nbsp;&nbsp; FAQ"></A></TD>
		</TR></TABLE>
	 </TR></TABLE>
	 </CENTER>
    </TD>
</TR>

<TR>
<!--Content Tables-->  
<TD CELLPADDING=10 ROWSPAN=2 VALIGN="top">
	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="ffffff" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	<FONT SIZE="4"><B> &nbsp;Human Genome Browser User Guide </B></FONT> </TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
	
<CENTER><TABLE BGCOLOR="ffffdd" BORDERCOLOR="cccc99" BORDER=1 CELLPADDING=3><TR><TD>	
<CENTER><FONT SIZE="2">Web tool created by <A HREF="http://www.cse.ucsc.edu/~kent">Jim Kent</A> of the <A HREF="http://bionet.ucsc.edu/people/zahler/">Zahler lab</A> at <A HREF="http://www.biology.ucsc.edu/mcd">MCD Biology at UCSC</A><BR>
in collaboration with the <BR>
<A HREF="http://www.nature.com/genomics/human/papers/409860a0_fs_1.html">genome analysis group of the International Human Genome Project</A><BR>
and the <A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/">UCSC Human Genome Project</A>.<BR>
Individual tracks contributed by various <A HREF="http://genome.cse.ucsc.edu/goldenPath/credits.html">individuals</A> and <A HREF="http://genome.cse.ucsc.edu/goldenPath/other.resources.html">institutions</A>.<BR>
User Guide last updated 27 Dec 00; comments and suggestions welcome.</FONT></CENTER>
</TD></TR></TABLE></CENTER>

<BR>
<B><FONT COLOR="#006666" SIZE=4> Table of Contents: </FONT></B>
<table border=0 nowrap cellpadding=0 cellspacing=0 margins=0><th height=10></th>
<tr><td valign="top"><ul><b>
  		<li><A CLASS="toc" HREF="goldenPath/help/hgTracksHelp2.html#What">
			What does it do?</A>
  		<li><A CLASS="toc" HREF="goldenPath/help/hgTracksHelp2.html#GetStarted">
			Getting Started</A>
  		<li><A CLASS="toc" HREF="goldenPath/help/hgTracksHelp2.html#FineTuning">
			Fine-tuning the Display</A>
	</b></ul></td><td><ul><b>
  		<li><A CLASS="toc" HREF="goldenPath/help/hgTracksHelp2.html#UnderstandDisplay">
			Understand the display </A>
  		<li><A CLASS="toc" HREF="goldenPath/help/hgTracksHelp2.html#IndivTracks">
			Details on individual tracks </A>
  		<li><A CLASS="toc" HREF="goldenPath/help/hgTracksHelp2.html#OlderTracks">
			Details on older tracks </A>
	</b></ul></td></tr></table>
</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>

	<A NAME="What"></A><BR>
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="ffffff" BACKGROUND="images/hr.gif" WIDTH=100% BORDER=0><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; What does it do?  </FONT>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
<P>
The sequence of the human genome is too big to see at all at once; few people
want to look at raw DNA sequence anyway. The alternative is the <A
HREF="http://genome.cse.ucsc.edu/hgTracks.html">Human Genome
Browser</A> for a quick display of any requested portion of the genome at any
scale, along with more than two dozen tracks of information (genes, ESTs, CpG islands, assembly gaps, chromosomal band, ...) associated with the complete human genome sequence.
<P>
The genome browser stacks features in tracks registered with genome coordinate position, encouraging rapid visual correlation of different types of information. Look at a whole chromosome to get a feeling for gene density, open a cytogenetic band to see its predicted genes, or zoom into a particular gene and its ESTs with introns for details of alternative splicing. 
<P>
Clicking on a displayed feature opens a second window providing protein sequence, coordinates and accession numbers, as appropriate. Clicking in the corner of the display calls up raw DNA sequence corresponding to the display window boundaries. This look-up feature is far more convenient than manual retrieval of a precise coordinate range from GenBank entries.
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<A NAME="GetStarted"></A><BR>
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="ffffff" BACKGROUND="images/hr.gif" WIDTH=100% BORDER=0><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Getting started:  </FONT>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
<P>
To get oriented, it is best to jump to a region of the genome that you are
already familiar with. In the genome position window, enter a gene symbol,
contig, EST or STS marker, cytogenetically mapped clone (e.g. RP11-36J16), GenBank mRNA accession, chromosome band, concept
(iron response element, vitamin D receptor, zinc finger, homeobox, Huntington
disease, tRNA, etc.), or a specific chromosomal range (chr20:514273-514309). 
<P>
The keyword database was built primarily from GenBank mRNA records, which can include gene names, gene symbols, annotation, article titles and authors. Some types of accession numbers, such as NT_xxxxxx, UniGene, and LocusLink do not work. However, synonomous <A HREF="http://www.gene.ucl.ac.uk/nomenclature/">HGNC</A> gene names, <A HREF="http://www.ncbi.nlm.nih.gov/LocusLink/build.html">RefSeqs</A> (NP_xxxxxx and NM_xxxxxx), and <A HREF="http://www3.ncbi.nlm.nih.gov/omim/">OMIM</A> (6 digits, same as <A HREF="http://www.ncbi.nlm.nih.gov/LocusLink/">LocusLink</A>) accessions will soon be recognized. 
<P>
 Frequent assembly updates change genome coordinates, requiring revision of thousands of gene coordinates: only non-manual methods are feasible. This limits keyword lookup possibilities until the human genome sequence is finalized.
<P>
If you have a DNA or protein sequence, but have no idea where it maps to in the genome, simply blast the NCBI <A HREF="http://www.ncbi.nlm.nih.gov/genome/seq/page.cgi?F=HsBlast.html&&ORG=Hs">finished</A>, <A HREF="http://www.ncbi.nlm.nih.gov/blast/blast.cgi?Jform=1">unfinished </A> (htgs), or dbEST human databases to recover a recognized accession number. The Genome Browser may offer its own direct-to-genome blast service in the near future. 
<P>
The Genome Browser usually returns a short list of matches in response to a text search. Clicking on any item opens the track window to the appropriate segment of the genome. You may wish to zoom out to display more flanking sequence or scroll over at the same scale (telomere of short p arm is to the left). Then add a bookmark to return easily later. 
<P>
Keep in mind that the Genome Browser can do no better than the quality of the draft genome; many assembly errors and gaps<A HREF="http://genome.cse.ucsc.edu/goldenPath/limits.html">still exist</A>. Duplications, both legitimate and assembly error, mean any genome coordinates found by blast are subject to ambiguity, pseudo-matches, and outright error. The Genome Browser does the best it can <A HREF="http://genome.cse.ucsc.edu/goldenPath/wdaccu.html">under the circumstances</A>.
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<A NAME="FineTuning"></A><BR>
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="ffffff" BACKGROUND="images/hr.gif" WIDTH=100% BORDER=0><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Fine-tuning the track display: </FONT>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
<P>
The track window opens to default values that may need adjustment. First set the pixel width to take maximum advantage of your monitor. For example, setting a 19" monitor to 1100 pixels means more of the genome is visible without redrawing. The maximum virtual width is 5000 pixels, but stick to your monitor width at first.
<P>
Tracks have controls with three display choices:
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>If "<B>hide</B>" is chosen, the track will not be displayed at all. Someone not interested in sequencing gaps, SNPs, or mouse synteny would want these tracks hidden.</TD></TR>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>If "<B>dense</B>" is chosen, the track will display all relevent features condensed to a single line. For example, to show where predicted genes exist without giving excessive detail, set the Genie track to dense.</TD></TR>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>If "<B>full</B>" is chosen, the track will be displayed in great detail, with a separate line for each constituent. There might be hundreds of ESTs relevent to genes displayed in the track window, so use sparingly. This option tops out at 300 items -- image height becomes overwhelming. </TD></TR>
</TABLE> 
<P>
Take some time to play around with the track settings to see the effects on the
display when a half dozen genes are showing. Controls to pan left and right,
and to zoom in and out 1.5, 3 and 10-fold are available at the top of the
browser.
After determining your preferences, save track settings to a bookmark so you won't have to reset defaults each time you return to the site. (Using the jump button to move to a different location retains settings, so even if the project has changed the settings bookmark is still helpful.) 
<P>
As an alternative to bookmarks, set up web links on your lab web site to open the Genome Browser with preferred defaults to frequently visited regions. 
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<A NAME="UnderstandDisplay"></A><BR>
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="ffffff" BACKGROUND="images/hr.gif" WIDTH=100% BORDER=0><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Understanding the display window: </FONT>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
<P>
The window displaying the tracks is fully clickable. Three things can happen: 
<TABLE>
<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>a window opens with name, sequence, and off-site links for the feature clicked upon. </TD></TR> 
<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>optional explanatory label in the center of the display toggles its track between full and dense. </TD></TR> 
<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>the display zooms 3-fold, re-centering on where the base position track was clicked.</TD></TR>
</TABLE> 
<P>
At 1 pixel per base scale, genes accurately display exons, UTR, introns, and direction of transcription as expected. At coarser scales, exons would soon disappear unless favoritism is shown; this comes at the expense of introns and overall display accuracy.
<P>
 After getting everything perfect, the display graphic can be saved as a gif image to your local computer. Another option is a screen shot if settings are to be captured too. The gif can be used for a lab web site (or printed) or it may be further processed in a photo editor; for example, the display is quite dramatic with its colors inverted. The scale (base pairs per pixel of width) can be calculated from data shown but will be an embedded feature shortly.
<P>
 The text itself is rendered as image, meaning it cannot be copied to a text editor. However, by viewing as source html (a feature of all web browsers), a long list of genes, mRNAs or ESTs can quickly be recovered as ordinary text. It will shortly be possible to directly query the underlying database to extract all manner of information without this html step.
<P>
 Example: hide all tracks but set Known Genes to full. Then choose a small chromosome, refresh, and view html. This gives an ordered list of all well-established genes (names and start-stop coordinates) on the chosen chromosome beginning with the p arm telomere. 
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<A NAME="IndivTracks"></A><BR>
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="ffffff" BACKGROUND="images/hr.gif" WIDTH=100% BORDER=0><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4"><b>&nbsp; Details on individual tracks: </FONT>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
<B>Remember:</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	A track does not appear if display mode is set to hide.</TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Clicking a track in dense display refreshes window to full display.</TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Clicking a track feature or its name in full display brings up detailed sequence information.</TD></TR>  
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Clicking on central explanatory text toggles full and dense display.</TD></TR> 
</TABLE>
<P>
<B>Base Position</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Shows chromosomal coordinates, numbered from the telomere of the short arm. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Clicking zooms in 3x centered about the click.</TD></TR>
</TD></TR></TABLE></P>

<P><B>Chromosome Band</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Light and dark blocks show traditional cytological bands seen with Giemsa
	staining. The correspondence between bands and sequence coordinate became more 
	accurate beginning with the 5 Sept 2000 assembly but is still subject to a few 
	megabases of uncertainty due to the vagaries of FISH mapping and band staining. 
	Track supplied by Terry Furey at UCSC. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Clicking in full display brings up detailed bibiographic information on how 
	cytological bands were molecularly mapped.
</TD></TR></TABLE></P>

<P><B>STS Markers</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Shows location of markers from genetic, RH, YAC, and FISH maps. The many
	sources of this data are described in 
	<a href="http://genome.ucsc.edu/goldenPath/other.resources.html">DataSources</A>. 
	Track supplied by Terry Furey and Sanja Rogic at UCSC.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Clicking on a specific marker in full display brings up information on a particular 
	STS and hyperlinks to STS data as a whole.</TD></TR>
</TABLE></P>

<P><B>Mouse Synteny</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Shows syntenic chromosomal region in mouse if known. Supplied by Deanna
	Church at NCBI.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Clicking calls up details on position and hyperlink to NCBI human/mouse 
	homology area.</TD></TR>
</TD></TR></TABLE></P>
	
<P><B>GC Percent</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Darker shades of gray correspond to higher percentages of GC figured for to a
	window of 20 kbp. Track supplied by Terry Furey at UCSC.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Clicking in full mode does not provide further information.</TD></TR>
</TD></TR></TABLE></P>

<P><B>FPC Contigs</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Large dark red blocks correspond to fingerprint map contigs at the Washington 
	University <A HREF="http://genome.wustl.edu/gsc/">WUGSC</A>.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Clicking in full mode brings up a very brief description of the extent and 
	number of clones in contig.</TD></TR>
</TD></TR></TABLE></P>

<P><B>Assembly</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	An individual box shows the assembly extracted from a single clone fragment. 
	Gaps in the path appear as white space between the boxes at an adequate zoom 
	level, with a thin horizontal line connecting boxes bridged over a gap.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
	Clicking in full mode brings up a brief description of clone fragment parts used.</TD></TR>
</TD></TR></TABLE></P>

<P><B>Gap</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Shows locations of gaps in the assembly with black boxes. At zoom levels with 
	much of a chromosome in view, small gaps can artefactually coalesce in the 
	graphic. Gaps spanned by mRNA and paired reads have a white horizontal line 
	through the black box to indicate bridging. Contig orientation errors can occur 
	around unbridged gaps.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
	In full mode, clicking may bring up a specific description of the gap. </TD></TR>
</TD></TR></TABLE></P>

<P><B>Coverage</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	In dense display, the level of gray gives level of coverage:<BR>
	<BLOCKQUOTE>
	White/Clear: no coverage (gap)<BR>
	Light Gray:  predraft (less than 4x shotgun)<BR>
	Medium Gray: draft (at least 4x shotgun)<BR>
	Dark Gray:   multiple draft, covered by more than one draft clone<BR>
	Finished:    covered by a finished clone</BLOCKQUOTE></TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
	In full display, each clone is represented as a gray box. The fragments inside 
	a clone are shown as black boxes. The position of fragments corresponds to 
	placement in the assembly.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
	Clicking in full mode brings up clone hyperlinks to Genbank record and fasta 
	format sequences (one per fragment). </TD></TR>
</TD></TR></TABLE></P>

<P><B>Known Genes</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Shows known protein coding genes from LocusLink. Track data supplied by David
	Kulp at <A HREF="http://www.affymetrix.com">Affymetrix</A>. Blue boxes representing 
	exons are connected by thin lines representing introns. In the full view, the 
	arrows on the introns indicate direction of transcription. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
	Clicking in full display brings up the corresponding protein sequence
	and link to <A HREF="http://bioinfo.weizmann.ac.il/cards/">GeneCards</A>.</TD></TR>
</TD></TR></TABLE></P>

<P><B>Genie</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Gene predictions based on ESTs, mRNA, codon usage, and splice site statistics
	by the proprietary program Genie, supplied by David Kulp at Affymetrix. Purple 
	boxes representing exons are connected by lines representing introns. In the full 
	view, the arrows on the introns indicate direction of transcription. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
	Clicking in full mode does not bring up a protein translation at this time. 
	Viewing html allows start and stop positions to be recovered; opening Genome 
	Browser to these coordinates allows recovery of DNA for the predicted gene.</TD></TR>
</TD></TR></TABLE></P>

<P><B>Full mRNAs</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Shows aligning regions (exons usually) as black boxes connected by lines for gaps 
	(spliced out introns usually). In full display, arrows on the introns indicate the 
	direction of transcription.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking in dense display brings up a good summary of the Genbank entry on that 
	mRNA, plus hyperlink to that full record and launches of PubMed searches by gene 
	name, product, and author.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking on an mRNA/genomic alignment brings up a base-by-base alignment.</TD></TR>
</TD></TR></TABLE></P>

<P><B>Spliced ESTs</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Shows spliced human ESTs (introns are the gaps after alignment with genomic 
	sequence). Requiring introns eliminates more than 99% of the contamination of 
	dbEST by genomic sequence. (Missing here, but found in the Human EST track, are
	authentic 3' ESTs and single-exon genes.) This track may suggest alternative 
	splicings.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	In dense mode, the level of gray in this track indicates the number of aligning 
	ESTs for the region. Genome Browser cannot display in full mode if more than 300 
	ESTs align in the window, so it may be necessary to zoom in to see individual 
	ESTs. Direction arrows on the "introns" are unreliable (3' and 5' read info is 
	missing at dbEST), though the sense strand can often be inferred from the GT-AG 
	splice donor/acceptor rule. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking in full display provides sequence, accession number, GenBank highlights, 
	and Medline link. </TD></TR>
</TD></TR></TABLE></P>

<P><B>Human ESTs</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	In dense mode the level of gray in this track represents the number of ESTs that 
	align at that region. The program will always display in dense mode if there are 
	more than 300 ESTs aligning in the window. It may be necessary to zoom in to 
	open up this track. Beware that the direction arrows on the 'introns' are 
	unreliable in this track because the EST database is missing info on whether 
	many ESTs are 3' or 5' reads.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking in full display provides sequence, accession number, GenBank highlights, 
	and Medline link. </TD></TR>
</TD></TR></TABLE></P>

<P><B>CpG Islands</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Green boxes display CpG islands, taken here as 50+ bases with 50% or higher
	GC content, with CpG dinucleotides disproportionate to composition. CpG tracks 
	are provided by Kim Worley and John Bouck at the Baylor College of Medicine
	and also by LaDeana Hillier at Washington University Genome Sequencing Center.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking in full mode brings detailed information and sequence for individual 
	CpG islands.</TD></TR>
</TD></TR></TABLE></P>

<P><B>Exofish ecores</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Displays conserved regions (generally protein-coding) from a <A HREF=
	"http://www.genoscope.cns.fr/externe/tetraodon/">genomics project</A>
	(long random reads) in the pufferfish, Tetraodon nigroviridis. Regions of
	greater homology are displayed in a darker gray. Ecore is an acronym for
	<U><B>e</B></U>volutionary <U><B>co</B></U>nserved <u><b>re</b></u>gion. Track is provided by
	Olivier Jaillon at Genoscope.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking in full mode brings up information on a particular ecore and background 
	notes on the track as a whole.</TD></TR>
</TD></TR></TABLE></P>

<P><B>RNA Genes</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	This track shows where non-protein coding RNA genes and pseudogenes are
	located. This includes tRNAs, rRNAs, SRPs, C/D box methylation guide snoRNAs,
	U6 etc. RNA-like, HBII, and hVS-like elements, as determined by Sean Eddy at
	Washington University. The density of these elements can range from 10-40 per
	ten million bp in some chromosomes. They are not displayed in the Known Genes track.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking in full mode brings up data on type of feature and score.</TD></TR>
</TD></TR></TABLE></P>

<P><B>Overlap SNPs</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	This track shows single nucleotide polymorphisms found on overlapping contigs
	that are useful markers in mapping genes and diseases. Track is provided by
	Lincoln Stein at Cold Spring Harbor, in conjunction with the SNP Consortium.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking in full mode brings up SNP report with dbSNP link.</TD></TR>
</TD></TR></TABLE></P>

<P><B>Random SNPs</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	This track shows single nucleotide polymorphisms found by random sequencing that 
	are useful markers in mapping genes and diseases.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Track is provided by Lincoln Stein at Cold Spring Harbor, in conjunction with 
	the SNP Consortium.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking in full mode brings up SNP report with dbSNP link.</TD></TR>
</TD></TR></TABLE></P>

<P><B>RepeatMasker</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>
	Full display opens 9 tracks - one each for SINE, LINE, LTR, DNA, simple, low 
	complexity, micro-satellite, tRNA, and other repeat families, as determined by. 
	Arian Smit's <A HREF="http://ftp.genome.washington.edu/cgi-bin/RepeatMasker">
	RepeatMasker</A>. Each individual repeat is represented by a gray box, with darker 
	gray indicating more recent age (dating by alignment to a canonical repeat). </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	In dense mode for large chromosomal regions, the black boxes representing repeats 
	may merge into single blocks in the display.</TD></TR>
</TD></TR></TABLE></P>

 <P><B>Simple Repeats</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
	This track contains not only simple repeats of the RepeatMasker track, but
	also imperfect tandem repeats of longer period. Data provided by Kim Worley and
	John Bouck at the Baylor College of Medicine.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking in full mode brings up detailed simple repeat sequence and score.</TD></TR>
</TD></TR></TABLE></P>
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	
	
	<A NAME="OlderTracks"></A><BR>
	
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="ffffff" BACKGROUND="images/hr.gif" WIDTH=100% BORDER=0><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4"><b>&nbsp; Details on tracks available for older assemblies only: </FONT>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
<P><B>Isochores</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
	This track is only available in the July 2000 browser. Dark blocks correspond
	to GC-rich sections and light blocks to AT-rich sections. These isoforms were
	defined with a HMM with 100,000 bp windows. Theoretically there might be an
	inverse relationship between the grayscale value of the blocks in this track
	and that of the chromosome band track; in practice the relationship is not very
	apparent. Track supplied by Terry Furey at UCSC. </TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	This brings up a more detailed description of the item and the track.</TD></TR>
</TD></TR></TABLE></P>
 
<P><B>Ensembl Genes</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
	This track contains gene predictions from <A HREF=
	"http://www.ensembl.org/perl/mapview?chr=20">Ensembl</A>. Maroon boxes representing 
	exons are connected by lines representing introns. In the full view, the arrows on 
	the introns indicate direction of transcription. Data provided by Ewan Birney.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking in full display brings up a protein translation.</TD></TR>
</TD></TR></TABLE></P>

<P><B>Pufferfish Homologies</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD> 
	This track is only available in the July 2000 freeze browser. Blue boxes show 
	regions with significant homology to the pufferfish genome (Tetraodon nigroviridu). 
	About a third of the compact Tetraodon genome has been sequenced by single-read 
	shotgun. Homology strength is shown by darkness of blue (strong homology states 
	are often coding); gaps between boxes inside of a track represent long inserts.</TD></TR>
		<TR><TD WIDTH="10" VALIGN="top">-</TD><TD>  
	Clicking on an individual line brings up a base-by-base alignment of Tetraodon 
	vs. human with the Tetraodon sequence on top. Underneath the alignments are a row 
	of symbols with the following significance:
	<PRE> 
	H - high degree of homology 
	L: low degree of homology 
	1: probable coding region 
	2: probable coding region 
	3: probable coding region (wobble codon) 
	Q: Long insert in query (Tetraodon) 
	T: Long insert in target (human) 
	</PRE></TD></TR>
</TD></TR></TABLE>

</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>

</TD></TR></TABLE>
</BODY></HTML>
