<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: Multiple Alignment Format (MAF)</TITLE>
	<LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="/images/title.jpg"></TH></TR>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
	<table bgcolor="000000" cellpadding="1" cellspacing="1" width="100%" height="27"><tr bgcolor="2636D1"><td valign="middle">
	<table BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24">
			<TR><TD VALIGN="middle"><font color="#89A1DE">&nbsp;&nbsp;
		<A HREF="/index.html" class="topbar">
			Home</A> &nbsp; - &nbsp;
		<A HREF="/cgi-bin/hgGateway" class="topbar">
			Genome Browser</A> &nbsp; - &nbsp; 
		<A HREF="/cgi-bin/hgNear" class="topbar">
			Gene Sorter</A> &nbsp; - &nbsp; 
		<A HREF="/cgi-bin/hgBlat?command=start" class="topbar">
			Blat </A> &nbsp; - &nbsp; 
		<A HREF="/cgi-bin/hgTables?command=start" class="topbar">
			Table Browser</A> &nbsp; - &nbsp; 
		<A HREF="/FAQ/" class="topbar">
			FAQ</A> &nbsp; - &nbsp; 
		<A HREF="/goldenPath/help/hgTracksHelp.html" class="topbar">
			Help</A> &nbsp;</font></TD>
	</TR></TABLE>
	</TD></TR></TABLE>
</TD></TR>

<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=2 CELLPADDING=10>	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><b>&nbsp;  Multiple Alignment Format (MAF)</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
		    The multiple alignment format stores a series of
		    multiple alignments in a format that is easy to
		    parse and relatively easy to read.  This format
		    stores multiple alignments at the DNA level
		    between entire genomes. The previously existing
		    formats are suitable for multiple alignments of
		    single proteins or regions of DNA without
		    rearrangements, and would require considerable
		    extension to cope with genomic issues such as
		    forward and reverse strand directions, multiple
		    pieces to the alignment, and so forth.  

		    <P><B>General Structure</B><P> The .maf format is
		    line-oriented. Each multiple alignment ends with a
		    blank line.  Each sequence in an alignment is on a
		    single line, which can get quite long, but there is no 
		    length limit. Words in a
		    line are delimited by any white space. Lines
		    starting with # are considered to be comments.
		    Lines starting with ## can be ignored by most
		    programs, but contain meta-data of one form or
		    another.
		    <P>
		    The file is broken into paragraphs that terminate
		    in a blank line.  Within a paragraph, the first
		    word of a line indicates its type.  Each multiple
		    alignment is in a separate paragraph that begins
		    with an 'a' line and contains an 's' line for each
		    sequence in the multiple alignment.  For now,
		    parsers should ignore other types of paragraphs
		    and other types of lines within an alignment
		    paragraph.

		    <P><B>The First Line -- Header</B><P> 
		    <FONT color="#0000FF"><PRE><TT> ##maf version=1 scoring=tba.v8</TT></PRE></FONT>
		    The first line of a .maf file begins with ##maf.
		    This word is followed by white-space-separated
		    variable=value pairs.  There must *not* be white
		    space surrounding the =.  The currently defined
		    variables are: 
		    <UL>
		    <LI>version -- Required.  Currently set
		                   to one.  </LI>
		    <LI>scoring -- Optional. A name for the scoring
		                   scheme used for the alignments.
		                   The current scoring schemes are:
		        <UL><LI>bit -- roughly corresponds to blast bit
		    		   values -- roughly 2 points per
		    		   aligning base minus penalties for
		    		   mismatches and inserts.  </LI>
		            <LI>blastz -- blastz scoring scheme -- roughly
		                   100 points per aligning base.</LI>
		            <LI>probability -- some score normalized
		                   between 0 and 1.  </LI> 
		        </UL>

		    <LI>program -- Optional. Name of the program
		                   generating the alignment.</LI>
                    </UL>
                    Parsers ignore variables they do not understand.
		    <P><B>The Second Line -- Parameters for creating the alignment</B><P> 
		    <FONT color="#0000FF"><PRE><TT> # tba.v8 (((human chimp) baboon) (mouse rat))</TT></PRE></FONT>
		    The second line displays the parameters that were
		    used to run the alignment program.

		    <P><B>Lines starting with 'a' -- parameters for a new alignment block</B><P> 
		    <FONT color="#0000FF"><PRE><TT> a score=23262.0</TT></PRE></FONT>
		    Each alignment begins with an 'a' line.  The 'a'
		    is followed by name=value pairs.  There are no
		    required name=value pairs.  The currently defined
		    variables are: <UL>

		    <LI> score -- Optional. Floating point score. If
		                 this is present, it is good practice
		                 to also define scoring in the first
		                 line.  </LI>

		    <LI> pass --  Optional.  Positive integer value.
		                 For programs that do multiple pass
		                 alignments such as blastz, this shows
		                 which pass this alignment came from.
		                 Typically, pass 1 will find the
		                 strongest alignments genome-wide, and
		                 pass 2 will find weaker alignments
		                 between two first-pass alignments.</LI> 
		     </UL>

		    <P><B>Lines starting with 's' -- a sequence within an alignment block</B><P> 
<FONT color="#0000FF"><PRE><TT> s hg16.chr7    27707221 13 + 158545518 gcagctgaaaaca
 s panTro1.chr6 28869787 13 + 161576975 gcagctgaaaaca
 s baboon         249182 13 +   4622798 gcagctgaaaaca
 s mm4.chr6     53310102 13 + 151104725 ACAGCTGAAAATA</TT></PRE></FONT>
		    The 's' lines together with the 'a' lines define a
		    multiple alignment.  The 's' lines have the
		    following fields which are defined by position
		    rather than name=value pairs.<UL>

		    <LI> src -- The name of one of the source sequences
		               for the alignment.  For sequences that
		               are resident in a browser assembly, the
		               form 'database.chromosome' allows
		               automatic creation of links to other
		               assemblies.  Non-browser sequences are
		               typically reference by the species name
		               alone.</LI>

		    <LI> start -- The start of the aligning region in
		               the source sequence.  This is a
		               zero-based number. If the strand field
		               is '-' then this is the start
		               relative to the reverse-complemented
		               source sequence.  </LI>
		    <LI> size -- The size of the aligning region in the
		               source sequence.  This number is equal
				       tosthe number of non-dash characters in
		               the alignment text field below.  </LI>
		    <LI> strand -- Either '+' or '-'.  If '-', then the
		               alignment is to the
		               reverse-complemented source.  </LI>
		    <LI> srcSize -- The size of the entire source
		               sequence, not just the parts involved
		               in the alignment.  </LI>
		    <LI> text -- The nucleotides (or amino acids) in
		               the alignment and any insertions
		               (dashes) as well.  </LI>
		    </UL>

		    <P><B>A Simple Example</B><P> 
		    Here is a simple example of a three alignment
		    blocks derived from five starting sequences.
		    Repeats are shown as lowercase, and each block may
		    have a subset of the input sequences.  All
		    sequence columns and rows must contain at least one nucleotide
		    (no full columns or rows of insertions).
<FONT color="#0000FF"><PRE><TT> ##maf version=1 scoring=tba.v8
 # tba.v8 (((human chimp) baboon) (mouse rat))
 # multiz.v7
 # maf_project.v5 _tba_right.maf3 mouse _tba_C
 # single_cov2.v4 single_cov2 /dev/stdin

 a score=23262.0
 s hg16.chr7    27578828 38 + 158545518 AAA-GGGAATGTTAACCAAATGA---ATTGTCTCTTACGGTG
 s panTro1.chr6 28741140 38 + 161576975 AAA-GGGAATGTTAACCAAATGA---ATTGTCTCTTACGGTG
 s baboon         116834 38 +   4622798 AAA-GGGAATGTTAACCAAATGA---GTTGTCTCTTATGGTG
 s mm4.chr6     53215344 38 + 151104725 -AATGGGAATGTTAAGCAAACGA---ATTGTCTCTCAGTGTG
 s rn3.chr4     81344243 40 + 187371129 -AA-GGGGATGCTAAGCCAATGAGTTGTTGTCTCTCAATGTG

 a score=5062.0
 s hg16.chr7    27699739 6 + 158545518 TAAAGA
 s panTro1.chr6 28862317 6 + 161576975 TAAAGA
 s baboon         241163 6 +   4622798 TAAAGA
 s mm4.chr6     53303881 6 + 151104725 TAAAGA
 s rn3.chr4     81444246 6 + 187371129 taagga

 a score=6636.0
 s hg16.chr7    27707221 13 + 158545518 gcagctgaaaaca
 s panTro1.chr6 28869787 13 + 161576975 gcagctgaaaaca
 s baboon         249182 13 +   4622798 gcagctgaaaaca
 s mm4.chr6     53310102 13 + 151104725 ACAGCTGAAAATA</TT></PRE></FONT>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

</TD></TR></TABLE>
</BODY></HTML>



