<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: Multiple Alignment Format (MAF)</TITLE>
	<LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
  <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="/images/title.jpg"></TH></TR>

<!--HOTLINKS BAR-->
  <TR><TD COLSPAN=2 HEIGHT=40>
    <TABLE bgcolor="000000" cellpadding="1" cellspacing="1" width="100%" height="27">
      <TR bgcolor="2636D1"><TD valign="middle">
	<TABLE BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24">
	  <TR><TD VALIGN="middle"><font color="#89A1DE">&nbsp;&nbsp;
		<A HREF="/index.html" class="topbar">
			Home</A> &nbsp; - &nbsp;
		<A HREF="/cgi-bin/hgGateway" class="topbar">
			Genome Browser</A> &nbsp; - &nbsp; 
		<A HREF="/cgi-bin/hgNear" class="topbar">
			Gene Sorter</A> &nbsp; - &nbsp; 
		<A HREF="/cgi-bin/hgBlat?command=start" class="topbar">
			Blat </A> &nbsp; - &nbsp; 
		<A HREF="/cgi-bin/hgTables?command=start" class="topbar">
			Table Browser</A> &nbsp; - &nbsp; 
		<A HREF="/FAQ/" class="topbar">
			FAQ</A> &nbsp; - &nbsp; 
		<A HREF="/goldenPath/help/hgTracksHelp.html" class="topbar">
			Help</A> &nbsp;</font></TD>
	  </TR>
 	</TABLE>
      </TD></TR>
    </TABLE>
  </TD></TR>

<!--Content Tables------------------------------------------------------->
  <TR><TD COLSPAN=2 CELLPADDING=10>	
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>	
    	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	  <TR><TD>	
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	      <TR><TD>
		<FONT SIZE="4"><B>&nbsp;  Multiple Alignment Format (MAF)</B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD>
	      <TD>
		<P>
	   	The multiple alignment format (MAF) stores a series of
		multiple alignments in an ASCII text format that is easy to
		parse and relatively easy to read.  MAF was created to 
		store multiple alignments at the DNA level between entire 
		genomes. It circumvents some of the problems inherent in other
		multiple alignment formats, which are suitable for 
		alignments of single proteins or regions of DNA 
		without rearrangements, but are unable to handle genomic 
		issues such as forward and reverse strand directions or 
		multi-part alignments. 
		<P>
		<B>General Structure</B>
   		<P> 
		MAF files (*.maf) are line-oriented. A file consists
		of a header line and a parameter line, followed by a series 
		of alignment blocks (or paragraphs), 
		each of which contains one set of multiple alignments. Each
		block is separated from the next by a blank line. 
		<P>
		Blocks consist of two parts: an alignment line showing
		the alignment parameters for the block, followed by 
		a set of sequence lines showing each sequence in the multiple 
		alignment at that location. In the current
		implementation, parsers should ignore other types of 
		blocks and other types of lines within an alignment
		block, as well as extra blank lines between blocks.
		<P>
		Lines starting with &quot;#&quot; are considered to 
		be comments. Lines starting with &quot;##&quot; 
		contain meta-data of one form or another and can be ignored
		by most programs.
		<P>
		<B><em>Example:</em></B><BR>
		Here is a simple example
		of a file containing 2 alignments of 3 sequences each, 
		from human, horse, and fugu:
		<FONT color="#0000FF"><PRE><TT> ##maf version=1 scoring=probability
 #mblastz 8.91 02-Jan-2005
 
 a score=0.128
 s human_hoxa 100  9 + 100257 ACA-TTACT
 s horse_hoxa 120 10 -  98892 ACAATTGCT
 s fugu_hoxa   88  8  + 90788 ACA--TGCT
 
 a score=0.071
 s human_unc 9077 8 + 10998 ACAGTATT
 s horse_unc 4555 6 -  5099 ACA--ATT
 s fugu_unc  4000 4 +  4038 AC----TT
		    </TT></PRE></FONT>

		<P><B>1. Header line</B><P> 
		<FONT color="#0000FF"><PRE><TT> ##maf version=1 scoring=probability</TT></PRE></FONT>
		The first line of a .maf file begins with <TT><B>##maf</B></TT>,
		followed by one or more <TT><B>variable=value</B></TT> pairs 
		separated by white space.  There must be <em>no</em> white
		space on either side of the &quot;=&quot;. Defined
		variables include: 
		<UL>
		<LI><B><em>version</em></B> -- Required. Default = 1.
		<LI><B><em>scoring</em></B> -- (Optional) The scoring 
		scheme used for the alignments. The current scoring 
 		schemes are:
		  <UL>
		  <LI>
		  <em>bit</em> -- roughly corresponds to blast bit values:
		  approximately 2 points per aligning base minus penalties for
		  mismatches and inserts.  
		  <LI>
		  <em>blastz</em> -- blastz scoring scheme: roughly 100 points per 
		  aligning base.
		  <LI><em>probability</em> -- some score normalized between 0 and 1.
        	  </UL>
		<LI><B><em>program</em></B> -- (Optional) Name of the program 
		generating the alignment.
                </UL>
                Parsers ignore variables they do not understand.
		<P><B>2. Parameter line</B><P> 
		<FONT color="#0000FF"><PRE><TT>  #mblastz 8.91 02-Jan-2005</TT></PRE></FONT>
		One or more <em>parameter lines</em> follow the header. These
		lines begin with &quot;#&quot; and display the parameters 
		used to run the alignment program.

		<P><B>3. Alignment lines</B><P> 
		<FONT color="#0000FF"><PRE><TT>  a score=0.128</TT></PRE></FONT>
		<em>Alignment lines</em> contain the parameters used for that
		particular alignment block. Each alignment block must begin with an 
		alignment line. Each alignment line begins with 
		&quot;a&quot;</B></TT>, 
		followed by optional <TT><B>name=value</B></TT> 
		pairs.  The currently defined <TT><B>name</B></TT> variables 
		are: 
		<UL>
		<LI>
		<em><B>score</em></B> -- (Optional) Floating point score. If
                included, the <em>scoring</em> attribute in the header line 
		should also be defined.  
		<LI>
		<em><B>pass</em></B> --  (Optional). Positive integer value.
	        For programs that do multiple pass alignments, such as blastz,
		<em>pass</em> indicates which pass the alignment came from.
	        Typically, pass 1 will find the strongest alignments 
		genome-wide, and pass 2 will find weaker alignments
	        between two first-pass alignments.</LI> 
	     	</UL>

	   	<P><B>4. Sequence lines</B><P>
		<FONT color="#0000FF"><PRE><TT>  s human_hoxa 100  9 + 100257 ACA-TTACT
  s horse_hoxa 120 10 -  98892 ACAATTGCT
  s fugu_hoxa   88  8  + 90788 ACA--TGCT</TT></PRE></FONT>

	    	<em>Sequence lines</em> define the sequence of each species
		within an alignment block. Not all species
		will have a sequence line in every alignment block. 
		Each sequence in the alignment is entered on 
		a separate line, one per species, which can get quite long. 
		All sequences within an alignment block must be of the same 
		length. Alignment gaps are padded with &quot;-&quot;. 
		Words within a line are delimited by white space. 
		<P>
		Each sequence line begins with &quot;s&quot; and contains the 
		following required fields:
		<UL>
	    	<LI>
		<B><em>src</em></B> -- Name of one of the source sequences
		included in the alignment. For sequences that are resident 
		in a 
		browser assembly, the form <em>database.chromosome</em> allows
	        automatic creation of links to other assemblies.  Non-browser 
		sequences are typically referenced by the species name alone.
		Species names must not contain spaces: concatenate multi-word 
		names or replace the space with an underscore.
	    	<LI>
		<B><em>start</em></B> -- Start of the aligning region in
		the source sequence, using zero-based position coordinates.  
		If the strand value is &quot;-&quot;, this field defines the 
		start relative to the reverse-complemented source sequence.  
		<LI> 
		<B><em>size</em></B> -- Size of the aligning region in 
		the source sequence.  This is equal to the number of 
		non-dash characters in the <TT><B>text</B></TT> field 
		(see below).
		<LI>
		<B><em>strand</em></B> -- &quot;+&quot; or &quot;-&quot;.  
		If the value is &quot;-&quot;, the sequence aligns to the 
		reverse-complemented source.
		<LI> 
		<B><em>srcSize</em></B> -- Size of the entire source
		sequence, not just the portions involved in the alignment.  
		<LI> 
		<B><em>text</em></B> -- Nucleotides (or amino acids) in
		the alignment and any insertions (indicated by &quot;-&quot;).
		</UL>
		<P><B><em>Example:</em></B><BR> 
		Here is an example of three alignment
		blocks derived from five starting sequences.
		Repeats are shown in lower case, and each block may
		have a subset of the input sequences.  Each
		column and row in the block of sequence lines must contain 
		at least one nucleotide; no columns or rows may consist
		entirely of &quot;-&quot;s.
	    <FONT color="#0000FF"><PRE><TT> ##maf version=1 scoring=tba.v8
 # tba.v8 (((human chimp) baboon) (mouse rat))
 # multiz.v7
 # maf_project.v5 _tba_right.maf3 mouse _tba_C
 # single_cov2.v4 single_cov2 /dev/stdin

 a score=23262.0
 s hg16.chr7    27578828 38 + 158545518 AAA-GGGAATGTTAACCAAATGA---ATTGTCTCTTACGGTG
 s panTro1.chr6 28741140 38 + 161576975 AAA-GGGAATGTTAACCAAATGA---ATTGTCTCTTACGGTG
 s baboon         116834 38 +   4622798 AAA-GGGAATGTTAACCAAATGA---GTTGTCTCTTATGGTG
 s mm4.chr6     53215344 38 + 151104725 -AATGGGAATGTTAAGCAAACGA---ATTGTCTCTCAGTGTG
 s rn3.chr4     81344243 40 + 187371129 -AA-GGGGATGCTAAGCCAATGAGTTGTTGTCTCTCAATGTG

 a score=5062.0
 s hg16.chr7    27699739 6 + 158545518 TAAAGA
 s panTro1.chr6 28862317 6 + 161576975 TAAAGA
 s baboon         241163 6 +   4622798 TAAAGA
 s mm4.chr6     53303881 6 + 151104725 TAAAGA
 s rn3.chr4     81444246 6 + 187371129 taagga

 a score=6636.0
 s hg16.chr7    27707221 13 + 158545518 gcagctgaaaaca
 s panTro1.chr6 28869787 13 + 161576975 gcagctgaaaaca
 s baboon         249182 13 +   4622798 gcagctgaaaaca
 s mm4.chr6     53310102 13 + 151104725 ACAGCTGAAAATA</TT></PRE></FONT>

	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE><BR>
	  </TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
  </TD></TR>
</TABLE>
</BODY></HTML>
