<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
	"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
	<title>Multi-Region Display Help Guide - UCSC Genome Browser</title>
        <meta name="description" content="Multi-Region Display">
        <meta name="keywords" content="genome, genome data, data integrator, table browser, genome browser, genome assembly, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, multi region">
	<meta http-equiv="content-type" content="text/html; charset=utf-8">
	<link rel="stylesheet" href="/style/HGStyle.css">
</head>
<body class="oldStaticPage">

<div class="header">
	<img src="../../images/title.jpg" alt="UCSC Genome Bioinformatics">
</div><!-- .header -->

<div class="nav">
<!--#include virtual="../../cgi-bin/hgMenubar"-->
</div><!-- .nav -->

<div class="content">

<div class="section">
<h1>Multi Region Display Help</h1>
<div class="sectionContent">
<p><b>Table of Contents:</b>
<td>
  <ul><b>
    <li><a class="toc" href="#Intro">Introduction</a>
    <li><a class="toc" href="#GetStarted">Getting started with the multi-region view</a>
    <li><a class="toc" href="#About">About the multi-region view</a>
    <ul>
       <li><a class="toc" href="#ExonOnly">Exon only</a>
       <li><a class="toc" href="#GeneOnly">Gene only</a>
       <li><a class="toc" href="#CustomRegions">Custom Regions</a>
       <li><a class="toc" href="#Haplotype">Haplotype view</a>    
    </ul>
    <li><a class="toc" href="#Examples">Examples</a>
  </b></ul>
</td>
                <P>
                <FORM name="googleForm1" method="GET" action="http://www.google.com/search" onSubmit="document.googleForm1.q.value=document.googleForm1.qq.value+'   site:genome.ucsc.edu/goldenPath/help';">          
                Search the Genome Browser help pages: &nbsp;
                <input type="hidden" name="q" value="">
                <input type="hidden" name="num" value="10">
                <input type="hidden" name="filter" value="0">
                <INPUT TYPE=text name=qq size=30 maxlength=255 value="">
                <input TYPE="submit" VALUE="Submit">
                </FORM>
                </p><p>
                <A HREF="../../contacts.html">Questions and feedback are welcome</A>.

</div><!-- .sectionContent -->
</div><!-- .section -->


<a name="Intro"></a>	
<div class="section">
	<h1>Introduction</h1>
	<div class="sectionContent">
	   <p>
              The multi-region display allows users the ability to &quot;slice&quot; up
              their normal track viewing experience into a variety of different modes:
              exon, gene, or user defined BED coordinates, and visualize 
              the track annotations only in those regions, effectively removing
              the intergenic, intron or otherwise unwanted regions from the viewing window.
        
           <p>
              In addition, for the human assemblies hg17, hg18, hg19 and hg38, the multi-region view
              also supports embedding a haplotype chromosome on top of it's
              reference sequence in order to visualize the differences between the haplotype
              sequence and the reference sequence.
           
           <p>
             Using the multi-region display, one can:<br>
             <ul>
                <li>Show all the SNP's that are within exons using the exon only view. 
                    ### make examples section same as these examples? or different? 
                <li>Show only the genes on a chromosome with the gene only view.
                <li>Display the same chromosomal position on either side of a gene in order 
                    to model a repeat region using a BED file with custom regions.
                <li>See how an alternate haplotype fits onto it's chromosome with the 
                    alternate haplotype view.
             </ul>
        
	</div><!-- .sectionContent --> 
</div><!-- .section -->

<a name="GetStarted"></a>
<div class="section">
	<h1>Getting Started with the multi-region view</h1>
	<div class="sectionContent">
	<p>
        Use the following instructions to display mutliple regions in the genome browser.
        
	<p>
	<b>Step 1. Pick a genome assembly and region</b><br>
           <ul>
              <li>Select the genome assembly you would like to view just as when normally
                  viewing an assembly, by using the top menu's Genomes link. If you would like
                  to view custom regions for your assembly hub, navigate to it's browser
                  display as normal.

              <li>Once an assembly has been selected, you may search for a gene of interest 
                  or select a region to view using the Genome Browser search bar if desired,
                  just as normal.
           </ul>	
        <p>
	<b>Step 2. Select the multi-region view</b><br>
	   <ul>
              <li>Click <button type="button">multi-region</button> while on the default display
                  page to bring up the multi-region configuration window.
                  <p>
                  <img style="border: 1px solid black; display: block;"
                       src="../../images/MultiRegionButton.png">
              <li>By default the <b>Show single-chromosome view</b> will be selected. This is the
                  standard Genome Browser display, and can be used to return to a standard 
                  display.
           </ul>
        <p>
	<b>Step 3. Configure the multi-region view</b>
           <table cellpadding=1>
              <tr><td>
	         <ul>
                    <li>Now we can select which multi-region view we would like to display:<br>
                    <ul>
                       <li>For exons only select  <b>Show exons</b> 
                       <li>For genes only select <b>genes using ...</b>
                       <li>For custom regions select <b>Custom regions from BED URL</b>
                       <li>For a haplotype chromosome select <b>Show one alternate haplotype...</b>
                    </ul>
                    <li>Select the <b>Highlight alternating regions in multi-region view</b> box if
                        you would like every other &quot;slice&quot;to be highlighted in blue. 
                    <li>Click submit to be taken back to the Genome Browser display.
                 </ul>
              </td>
              <td>
                 <img style="border: 1px solid black; display: block;"
                       src="../../images/MultiRegionConfigurationWindow.png">
              </td></tr>
           </table> 
        <p style="text-align: center">The result of selecting <b>Show exons</b> and <b>Highlight
                                      alternating regions</b>
        <img style="margin: auto; border: 1px solid black; display: block;"
             src="../../images/MultiRegionHighlighted.png">
        <p>
        After entering multi-region mode, the usual browser display configurations will still 
        work, such as zooming in and out, reversing and resizing the view, highlighting and 
        zooming to highlighted selection, and track search.
        
        <p>
        ###### NEED TO REWORD ######<br>
        Searching and zooming while using custom BED regions are a little different than while in
        exon or gene only and haplotype mode. Searching should work normally when in gene only, 
        exon only, and haplotype mode, however in custom regions mode, when a term or position 
        is entered into the search box from the tracks display page, the browser first searches 
        for the term within the multi-region view, and if found, the browser will zoom to that 
        feature. If the term is not found, then the browser will search the default search set 
        and redirect if found. 
         
    </div><!-- .sectionContent --> 
</div><!-- .section -->

<a name="About"></a>
<div class="section">
        <h1>About the multi-region view</h1>
        <div class="sectionContent">

        <a name="ExonOnly"></a>
        <p>
        <b>Exon Only View</b><br>
        The exon only view removes any bases that fall within an intron from the display, thus
        removing those annotations as well. The intergenic sequence is also removed, thus the 
        browser display will only consist of annotations and bases that fall within an exon or 
        padding region.

        <p>
        The padding field defines the number of intronic or intergenic bases to buffer
        each &quot;slice&quot; of the view.
        <br>### Screen here of end of an exon, padding bases and next exon ### 
                  <img style="border: 1px solid black; display: block;"
                       src="../../images/MultiRegionExonPadding.png">

        <p>
        The exon only view uses various gene tracks to compute where to &quot;slice&quot; the
        display, but only uses the highest <em>priority</em> track per organism. Thus if your
        organism has multiple gene tracks available, only one gene track will be used to
        pull exon positions from. If you are noticing intron regions in your display while you
        are in exon only mode, be sure that you have the correct gene track turned on. The 
        track used will be listed next to <b>genes</b> option in the multi-region configuration
        window, as both the exon and gene only view use the same track. 
        
        <br>#### Also need to mention jumping of position if no annotations in view ####
        <p>
        
        <p>
        If the gene track used to compute the slicing is a 
        <a href="">super track link help page here</a> 
        or contains other transcripts, then only the longest exons will be used to determine
        slices, regardless of whether or not they are in the same transcript. The below screenshot
        shows the exon only view for the SOD1 gene in human. The 3rd &quot;exon&quot; from the 
        left falls within the intron of the actual SOD1 gene, but is shown as an exon because it 
        is part of the GENCODE v22 super-track.
        <br>### Screenshot of SOD1 hg38 in exon only mode ###
                  <img style="border: 1px solid black; display: block;"
                       src="../../images/MultiRegionSOD1.png">
        
        <a name="GeneOnly"></a>
        <p>
        <b>Gene Only View</b><br>
        The gene only view is similiar to the exon only view, however it removes all 
        intergenic bases in a given range, thus only displaying annotations that fall within the 
        coordinates of a gene. 
 
        <p>
        As with exon only mode, gene only mode uses the padding field to set the number
        of intergenic bases to buffer each gene. 

        <p>
        Again similiar to exon only mode, gene only mode also uses the longest transcript from 
        its reference gene track as the region to display?

        <a name="CustomRegions"></a>
        <p>
        <b>Custom Regions</b></br>
        <p>
        The custom regions from BED URL mode allows the browser to display only those coordinates
        specified in a BED file, and thus allows for user-specific modeling of various regions. 

        <p>
        ### Write about comments here ###<br>
        options are #database #shortDesc and #padding<br>
        only grabs positions from BED3 or BED12, all other bed options ignored 
        <p>
        For instance, one may create a BED file containg different regions from across mutliple
        chromosomes, like <a class="toc" href="#Example3">Example 3</a> below, and you can now
        explore 3 different chromosomes at one time.
                  <img style="border: 1px solid black; display: block;"
                       src="">

        <p>
        Or, if a repeat region was clogging up your normal browser display, you could
        specify a BED file containing the upstream portion you are interested in, one unit
        of the repeating element, and then downstream region, to gain a better understanding
        of what is around the repeat. This is only possible if the BED file has the correct
        coordinates in the right order, as the display only depends on the order of regions
        specified in the file, not the coordinates relative genomic position to each other.

        <p>
        It is important to note that the BED files supported can be of any type, although
        only information from BED 3 or BED 12 files are relevant to the display. Be careful
        about the order of the regions specified, as the &quot;slices&quot; displayed are in
        order of the BED file, and not in order of their genomic position. Check out 
        <a class="toc" href="#Example4">Example 4</a> for an illustrative example of how 
        important this ordering is.
                  <img style="border: 1px solid black; display: block;"
                       src="">

        <p>
        As mentioned in the <a class="toc" href="#GetStarted">Getting Started</a> section, 
        searching, scrolling and zooming while viewing regions from a custom BED URL may be
        a little counterintuitive. While viewing multiple custom regions, your view is 
        limited to only the regions you have specified, so although you can still zoom in
        down to single base resolution you can only zoom out as far as the intial view, and 
        the up and downstream scrolling is also limited to the range specified by the BED file.
        

        <a name="Haplotype"></a>
        <p>
        <b>Haplotype</b><br>
	<p>
        The human hg38, hg19, hg18 and hg17 assemblies all possess an extra feature while
        using the multi-region view: showing an alternate haplotype on it's chromosome.
        <p>
        For a list of possible haplotypes, check the goldenPath/hg$/bigZips/hg$.chrom.sizes file 
        <br># Picture of link here #
	<img style="border: 1px solid black; display: block;"
	   src="">
        Any haplotype of the form <em>chr_name_alt</em> can be placed on it's chromosome,
        unlike the random and unplaced sequences, which are currently not supported by
        the multi-region view.
        
        <br># Example of haplotype view #
	<img style="border: 1px solid black; display: block;"
	src="">

        <p><b>Notes about haplotype display</b>
        <ul>
           <li>Only one haplotype can be displayed at a time. Thus you cannot display
               multiple haplotypes on the same chromsome to see how they compare. 
        </ul>
        <p>
        Be careful zooming out too far while using the haplotype view. You may zoom out off the
        end of a chromsome and onto the next one. Theoretically you could display all 25
        chromosomes this way, although it will be slow.

	</div><!-- .sectionContent --> 
</div><!-- .section -->


<a name="Examples"></a>
<div class="section">
<h1>Examples</h1>       
        <div class="sectionContent">
        <p>
        <a name="Example1"></a>
        <b>Gene only example</b><br>

        <p>
        <a name="Example2"></a>
        <b>Exon only example</b><br>
        
        <p>
        <a name="Example3"></a>
        <b>BED3 example</b><br>
           # show multiple chromosomes
        <p><b>Displaying regions from multiple chromsomes</b></br>
        Multi-region view also supports BED files specifying regions from across multiple
        chromosomes, allowing  display of annotations from different genomic regions in the same
        browser window. The following <a href="../../~chmalee/multRegion2.txt">file</a> 
        specifies regions from across three different chromosomes:
        
           <pre>
     chr9 133255000 133276861
     chr21 31659622 31668931
     chr13 69700594 69708459
           </pre>
        
        Again the multi-region view will display the &quot;slices&quot; according to the order
        specified in the BED file, and allows the same region to be displayed twice.

        <ul><img style="border: 1px solid black; display: block;"
             src="">
 
        <p>
        <a name="Example4"></a>
        <b>BED12 example</b><br>
           # be sure to emphasize how regions show in order specified
             not genomic order <br>
           # probably should be from same chromosome since bed3
                  example is from mult chromosomes
        <p><b>Displaying regions from the same chromsome</b></br>
        The following BED12 <a href="../../~chmalee/bed12formultregion.txt">example</a> specifies
        a 3 block region on chromsome 22 between bases 1000 and 5000, and a second 3 block 
        region between bases 2000 and 6000:<br>
        <pre>
     chr22 1000 5000 cloneA 960 + 1000 5000 0 3 567,488,900, 0,2512,3100
     chr22 2000 6000 cloneB 900 + 2000 6000 0 3 433,399,800, 0,2601,3200
        </pre>
        <p>
        When loaded as a custom track, we can see that there are overlapping regions, as well
        as a block between positions 2001 and 2433.
        <img style="border: 1px solid black; display: block;"
             src="../../images/hgIntegratorHgTracksExample.png">
        <p>
        Now examine this region under the multi-region view, with the above BED12 positions
        loaded as a custom BED <a href="../../~chmalee/bed12formultregion.txt">URL</a>.
        <img style="border: 1px solid black; display: block;"
             src="../../images/hgIntegratorHgTracksExample.png">

        <p>
        The positions display in the order specified by the BED file, not by genomic position.
        Notice how the <em># need to update later #</em> block is in the fourth region, not the second
        region, as in the custom track example.  
        <p>
        For an example BED12 URL, click <a href="../../~chmalee/bed12formultregion.txt">here</a>.

        <p>
        <a name="Example5"></a>
        <b>Haplotyple example</b><br>
        <img style="border: 1px solid black; display: block;"
             src="../../images/MultiRegionHaplotypeView.png">

	</div><!-- .sectionContent --> 
</div><!-- .section -->

<!--start removed sections
<div class="section">
<h1>Display a Custom Region with a BED File</h1>
	<div class="sectionContent">
	<p>
        One can now display multiple regions specified via 
        <a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format1">BED file</a> in 
        the Genome Browser display with the multi-region tool. These regions may be on one 
        chromosome or across multiple chromosomes.

        <p>
        To display multiple custom regions in the browser, click 
        <button type="button">multi-region</button> while on the track display
        page to bring up the multi-region configuration window. Select 
        <b>Custom regions from BED URL:</b> and paste in the URL to your BED
        file, then click <button type="button">submit</button>. 

        <p>
        The browser will now display the region(s) you have specified as &quot;slices&quot;,
        <em>with the regions displayed in the order of the BED file</em>. Note that you may 
        specify the same region multiple times in the BED file and it will be displayed
        as a separate region each time, making the modeling and comparison of the same region
        against many regions possible in one browser display.
 
        <p><b>Displaying regions from the same chromsome</b></br>
        The following BED12 <a href="../../~chmalee/bed12formultregion.txt">example</a> specifies
        a 3 block region on chromsome 22 between bases 1000 and 5000, and a second 3 block 
        region between bases 2000 and 6000:<br>
        <pre>
     chr22 1000 5000 cloneA 960 + 1000 5000 0 3 567,488,900, 0,2512,3100
     chr22 2000 6000 cloneB 900 + 2000 6000 0 3 433,399,800, 0,2601,3200
        </pre>
        <p>
        When loaded as a custom track, we can see that there are overlapping regions, as well
        as a block between positions 2001 and 2433.
        <img style="border: 1px solid black; display: block;"
             src="../../images/hgIntegratorHgTracksExample.png">
        <p>
        Now examine this region under the multi-region view, with the above BED12 positions
        loaded as a custom BED <a href="../../~chmalee/bed12formultregion.txt">URL</a>.
        <img style="border: 1px solid black; display: block;"
             src="../../images/hgIntegratorHgTracksExample.png">

        <p>
        The positions display in the order specified by the BED file, not by genomic position.
        Notice how the <em># need to update later #</em> block is in the fourth region, not the second
        region, as in the custom track example.  
        <p>
        For an example BED12 URL, click <a href="../../~chmalee/bed12formultregion.txt">here</a>.
        
        <p><b>Displaying regions from multiple chromsomes</b></br>
        Multi-region view also supports BED files specifying regions from across multiple
        chromosomes, allowing  display of annotations from different genomic regions in the same
        browser window. The following <a href="../../~chmalee/multRegion2.txt">file</a> 
        specifies regions from across three different chromosomes:
        
           <pre>
     chr9 133255000 133276861
     chr21 31659622 31668931
     chr13 69700594 69708459
           </pre>
        
        Again the multi-region view will display the &quot;slices&quot; according to the order
        specified in the BED file, and allows the same region to be displayed twice.

        <ul><img style="border: 1px solid black; display: block;"
             src="">

	</div>< .sectionContent --> 
</div><!-- .section -->

</body>
</html>
