<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
	"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
	<title>Multi-Region Display Help Guide - UCSC Genome Browser</title>
        <meta name="description" content="Multi-Region Display">
        <meta name="keywords" content="genome, genome data, data integrator, table browser, genome browser, genome assembly, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, multi region">
	<meta http-equiv="content-type" content="text/html; charset=utf-8">
	<link rel="stylesheet" href="/style/HGStyle.css">
</head>
<body class="oldStaticPage">

<div class="header">
	<img src="../../images/title.jpg" alt="UCSC Genome Bioinformatics">
</div><!-- .header -->

<div class="nav">
<!--#include virtual="../../cgi-bin/hgMenubar"-->
</div><!-- .nav -->

<div class="content">

<div class="section">
<h1>Multi Region Display Help</h1>
<div class="sectionContent">
<p><b>Table of Contents:</b>
<td>
  <ul><b>
    <li><a class="toc" href="#Intro">Introduction</a>
    <li><a class="toc" href="#GetStarted">Getting started with the multi-region view</a>
    <li><a class="toc" href="#About">About the multi-region modes</a>
    <ul>
       <li><a class="toc" href="#ExonOnly">Exon only</a>
       <li><a class="toc" href="#GeneOnly">Gene only</a>
       <li><a class="toc" href="#CustomRegions">Custom Regions</a>
       <li><a class="toc" href="#Haplotype">Haplotype view</a>    
    </ul>
    <li><a class="toc" href="#Examples">Examples</a>
  </b></ul>
</td>
                <P>
                <FORM name="googleForm1" method="GET" action="http://www.google.com/search" onSubmit="document.googleForm1.q.value=document.googleForm1.qq.value+'   site:genome.ucsc.edu/goldenPath/help';">          
                Search the Genome Browser help pages: &nbsp;
                <input type="hidden" name="q" value="">
                <input type="hidden" name="num" value="10">
                <input type="hidden" name="filter" value="0">
                <INPUT TYPE=text name=qq size=30 maxlength=255 value="">
                <input TYPE="submit" VALUE="Submit">
                </FORM>
                </p><p>
                <A HREF="../../contacts.html">Questions and feedback are welcome</A>.

</div><!-- .sectionContent -->
</div><!-- .section -->


<a name="Intro"></a>	
<div class="section">
	<h1>Introduction</h1>
	<div class="sectionContent">
	   <p>
              The multi-region display allows users the ability to &quot;slice&quot; up
              their normal track viewing experience into a variety of different modes:
              exon, gene, or user defined BED coordinates, and visualize 
              the track annotations only in those regions, effectively removing
              the intergenic, intron or otherwise unwanted regions from the viewing window.
        
           <p>
              In addition, for the human assemblies hg17, hg18, hg19 and hg38, the multi-region view
              also supports embedding a haplotype chromosome on top of it's
              reference sequence in order to visualize the differences between the haplotype
              sequence and the reference sequence.
           
           <p>
             Using the multi-region display, one can:<br>
             <ul>
                <li>Show all the SNP's that are within exons using the exon only view. 
                <li>Show only the genes on a chromosome with the gene only view.
                <li>Display the same chromosomal position on either side of a gene in order 
                    to model a repeat region using a BED file with custom regions.
                <li>See how an alternate haplotype fits onto it's chromosome with the 
                    alternate haplotype view.
             </ul>
        
	</div><!-- .sectionContent --> 
</div><!-- .section -->

<a name="GetStarted"></a>
<div class="section">
	<h1>Getting Started with the multi-region view</h1>
	<div class="sectionContent">
	<p>
        Use the following instructions to display mutliple regions in the genome browser.
        
	<p>
	<b>Step 1. Pick a genome assembly and region</b><br>
           <ul>
              <li>Select the genome assembly you would like to view just as when normally
                  viewing an assembly, by using the top menu's Genomes link. If you would like
                  to view custom regions for your assembly hub, navigate to it's browser
                  display as normal.

              <li>Once an assembly has been selected, you may search for a gene of interest 
                  or select a region to view using the Genome Browser search bar if desired,
                  just as normal.
           </ul>	
        <p>
	<b>Step 2. Select the multi-region view</b><br>
	   <ul>
              <li>Click <button type="button">multi-region</button> while on the default display
                  page to bring up the multi-region configuration window:
                  <p>
                  <img style="border: 1px solid black; display: block;"
                       src="../../images/MultiRegionButton.png">
              
           </ul>
        <p>
	<b>Step 3. Configure the multi-region view</b>
           <ul>
              <li>By default the <b>Show single-chromosome view</b> will be selected. This 
                  is the standard Genome Browser display, and can be used to return to a 
                  standard display.<br>
              <table cellpadding=15>
              <tr><td>
                 <li>Now we can select which multi-region view we would like to display:<br>
                    <ul>
                       <li>For exons only select  <b>Show exons</b> 
                       <li>For genes only select <b>genes using ...</b>
                       <li>For custom regions select <b>Custom regions from BED URL</b>
                       <li>For a haplotype chromosome select <b>Show one alternate haplotype...</b>
                    </ul>
                 <li>Select the <b>Highlight alternating regions in multi-region view</b> box if
                     you would like every other &quot;slice&quot;to be highlighted in blue. 
              </td>
              <td>
                 <img style="border: 1px solid black; display: block;"
                       src="../../images/MultiRegionConfigurationWindow.png">
              </td></tr>
              </table> 
              <li>Click submit to be taken back to the Genome Browser display:
           </ul>
        <img style="margin: auto; border: 1px solid black; display: block;"
             src="../../images/MultiRegionMouseHighlighted.png">
        <p style="text-align: center">An example <b>exon only</b> view.
        <p>
        After entering multi-region mode, the usual browser display configurations will still 
        work, such as zooming in and out, reversing and resizing the view, highlighting and 
        zooming to the highlighted selection, and track search.
        
        <p>
        <b>Searching and zooming</b><br>
        Searching and zooming while using custom BED regions are a little different than while in
        exon or gene only and haplotype mode. Searching should work normally when in gene only, 
        exon only, and haplotype mode, however in custom regions mode, the process is slightly
        different, see the <a href="#CustomRegions">About Custom Regions</a> section for more
        information. 
        <p>
        Additionally, while viewing custom regions, the browser can only zoom out as far as the
        BED defined coordinates, although single base resolution is still available. Again, for
        more information see the <a href="#CustomRegions">About Custom Regions</a> section. 
         
    </div><!-- .sectionContent --> 
</div><!-- .section -->

<a name="About"></a>
<div class="section">
        <h1>About the multi-region modes</h1>
        <div class="sectionContent">

        <a name="ExonOnly"></a>
        <p>
        <b>Exon Only Mode</b><br>
        The exon only view removes any bases that fall within an intron from the display, thus
        removing those annotations as well. Intergenic sequence is also removed, resulting in a 
        browser display consisting of annotations and bases that only fall within an exon or 
        padding region.
 
        <p>
        The exon only view is built by gathering all the exons present in the gene track listed
        next to the <b>genes using</b> option in the multi-track configuration window, adding the 
        appropriate number of intron or intergenic bases specified by the padding field, and 
        building a &quot;virtual&quot; chromosome. These form the slices of the exon only mode.

        <p>
        Visual spacing between exons is controlled by the padding field. The number of bases pads 
        each side of an exon, in effect doubling the number of padded bases specified, as shown 
        below, where padding is specified as 6:
        <p>
        <img style="margin:auto; border: 1px solid black; display: block;"
             src="../../images/MultiRegionExonPadding.png">
        
        <p>
        If the gene track used to compute the slicing is a 
        <a href="./trackDb/trackDbDoc.html#superTrack">super track</a> 
        or contains other transcripts, only the longest exon will be used to determine
        slicing, regardless of whether or not they are in the same transcript. See 
        <a href="#Example1">Example 1</a> for more infromation. 
        
        <a name="GeneOnly"></a>
        <p>
        <b>Gene Only Mode</b><br>
        The gene only view is similiar to the exon only view, however it removes all 
        intergenic bases in a given range, thus only displaying annotations that fall within the 
        coordinates of a gene.  
 
        <p>
        As with exon only mode, gene only mode uses the padding field to set the number
        of intergenic bases to buffer each gene. On tightly packed chromosomes or regions
        where there is little intergenic space, be careful the padding is not set too high,
        or multiple genes may display as a single gene, since overlapping transcripts are 
        considered as one region.
 
        <p>
        For an example of how both overlapping transcripts and intergenic padding affects
        the gene only view, see <a href="#Example2">Example 2</a>.
        
        <a name="CustomRegions"></a>
        <p>
        <b>Custom Regions</b></br>
        <p>
        The custom regions from BED URL mode allows the browser to display only those coordinates
        specified in a BED file, and thus allows for user-specific modeling of various regions. 

        <p>
        Like BED custom tracks, some comments are supported in the BED files used by multi-region
        mode. The supported comments are: 
        <table>
        <tr><td>
           <ul style="list-style:none;">
              <li>database
              <li>shortDesc
              <li>padding
           </ul>
        </td>
        <td>
           <ul style="list-style:none;">
              <li>The database comment ensures that if this BED file is loaded on a database other
                  than the one specified, the custom regions mode will not work.
              <li>shortDesc specifies the content in the parentheses here:
                  "customUrl" becomes your comment.
              <li>The padding field changes the padding similiar to exon and gene only mode's
                  padding field.
           </ul>
        </td></tr>
        </table>
        <br>
        To use one of these comments, just include the comment at the top of your BED file, 
        preceded by a &quot;#&quot;, like so:<br>
        <pre>
      #database hg38
      #shortDesc Example of a short description
      chr1 pos1 pos2
      chr2 pos1 pos2
        </pre>
        <p>

        <p>
        It is important to note that the BED files supported can be of any type, although
        only information from BED 3 or BED 12 files are relevant to the display. Be careful
        about the order of the regions specified, as the &quot;slices&quot; displayed are in
        order of the BED file, and not in order of their genomic position. Check out 
        <a class="toc" href="#Example3">Example 3</a> for an illustrative example of how 
        important this ordering is.

        <p>
        Searching and zooming while using custom BED regions are a little different than while in
        exon or gene only and haplotype mode. Searching should work normally when in gene only, 
        exon only, and haplotype mode, however in custom regions mode, when a term or position 
        is entered into the search box from the track display page, the browser first searches 
        for the term within the multi-region view, and, if found, the browser will zoom to that 
        feature. 
        <table>
        <tr><td>
              If the term is not found in the BED defined regions but is found elsewhere,
              this message will ask if you would like to return to a default view at that location.
              Click ok to return to your custom region.
        </td>
        <td> 
           <img style="margin: auto; border: 1px solid black; display: block;"
             src="../../images/MultiRegionBEDSearchRedirectMessage.png">
        </td></tr>
        </table>
        <p>
        If the term is not found at all in our databases, then the normal search error message will 
        trigger.
        <p>
        As mentioned in the <a class="toc" href="#GetStarted">Getting Started</a> section, 
        searching, scrolling and zooming while viewing regions from a custom BED URL may be
        a little counterintuitive. While viewing multiple custom regions, your view is 
        limited to only the regions you have specified, so although you can still zoom in
        down to single base resolution you can only zoom out as far as the BED defined regions, and 
        the up and downstream scrolling is also limited to the range specified in the BED file.
        

        <a name="Haplotype"></a>
        <p>
        <b>Haplotype</b><br>
	
        <p>
        The human hg38, hg19, hg18 and hg17 assemblies all possess an extra feature while
        using the multi-region view: showing one alternate haplotype on it's chromosome.
        <p>
        #### How to structure the bigZips link to make this text work? ####<br>
        For a list of possible haplotypes, check the goldenPath/hg$/bigZips/hg$.chrom.sizes file,
        where hg$ is one of either hg38, hg19, hg18 or hg17. Any haplotype of the form 
        <em>chr_name_alt</em> can be placed on it's chromosome, unlike the random and unplaced 
        sequences, which are currently not supported by the multi-region view.
        
        <p>
        For an example of how what the haplotype mode looks like in the browser, see 
        <a href="#Example5">example 5</a> below.

	</div><!-- .sectionContent --> 
</div><!-- .section -->


<a name="Examples"></a>
<div class="section">
<h1>Examples</h1>       
        <div class="sectionContent">
        <p>
        <a name="Example1"></a>
        <b>Exon only example</b><br>
        The below screenshot
        shows the exon only view for the SOD1 gene in human assembly hg38. The 3rd &quot;exon&quot; 
        from the left falls within the intron of the actual SOD1 gene, but is shown as an exon 
        because it is part of the GENCODE v22 super-track.
        
        <p>
        <img style="margin:auto; border: 1px solid black; display: block;"
             src="../../images/MultiRegionSOD1.png">

        <p>
        <a name="Example2"></a>
        <b>Gene only example</b><br>
        
        <p>
        Here is the hg38 mitochondrial chromosome, shown in gene only mode with padding set to
        10:
        
        <p>
        <img style="margin:auto; border: 1px solid black; display: block;"
             src="../../images/MultiRegionMitochondrialPacked.png">

        <p>
        Even though many genes are present on this chromosome, they are almost all overlapping
        with each other, preventing the gene only view from definining separate regions. 
        
        <p>
        Now compare to the hg38 mitochondrial chromosome with padding set to 0: 
        
        <p>
        <img style="margin:auto; border: 1px solid black; display: block;"
             src="../../images/MultiRegionMitochondrialUnpacked.png">
        
        <p>
        With more alternating regions, we can see that not all the genes overlapped with each
        other, they simply had no intergenic space betweem them with which to draw slices.

        <p>
        <a name="Example3"></a>
        <b>Custom Regions example</b><br>
        Here is an example <a href="../../~chmalee/mm10multiRegionexample.txt">BED file</a> for 
        mm10 that contains a list of exon positions for the Pax9 gene. If you load this file as 
        a custom track <em>without</em> turning on multi-region, you will notice that the blocks 
        display in order of genomic position, not in the order they are specified in the file:
        
        <p>
        <img style="margin: auto; border: 1px solid black; display: block;"
             src="../../images/MultiRegionBEDFileInOrder.png">
        
        <p>
        However, if after loading the custom track you turn on multi-region mode and use the file
        as a list of custom regions, you will see that now all of a sudden the blocks are out of
        order, and the corresponding exons of the Pax9 gene have been split up and are out of 
        order as well.
        
        <p>
        <img style="margin: auto; border: 1px solid black; display: block;"
             src="../../images/MultiRegionBEDFileOutOrder.png">
 
        <p>
        <a name="Example4"></a>
        <b>BED12 example</b><br>
        The following example <a href="../../~chmalee/bed12formultregion.txt">BED file</a> 
        utilizes a BED12 file to show two regions from two different chromosomes, chromosome
        22 and chromosom 21.

        #### Picture coming soon ####

        <p>
        <a name="Example5"></a>
        <b>Haplotyple example</b><br>
        <img style="border: 1px solid black; display: block;"
             src="../../images/MultiRegionHaplotypeView.png">

	</div><!-- .sectionContent --> 
</div><!-- .section -->

<!--start removed sections
<div class="section">
<h1>Display a Custom Region with a BED File</h1>
	<div class="sectionContent">
	<p>
        One can now display multiple regions specified via 
        <a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format1">BED file</a> in 
        the Genome Browser display with the multi-region tool. These regions may be on one 
        chromosome or across multiple chromosomes.

        <p>
        To display multiple custom regions in the browser, click 
        <button type="button">multi-region</button> while on the track display
        page to bring up the multi-region configuration window. Select 
        <b>Custom regions from BED URL:</b> and paste in the URL to your BED
        file, then click <button type="button">submit</button>. 

        <p>
        The browser will now display the region(s) you have specified as &quot;slices&quot;,
        <em>with the regions displayed in the order of the BED file</em>. Note that you may 
        specify the same region multiple times in the BED file and it will be displayed
        as a separate region each time, making the modeling and comparison of the same region
        against many regions possible in one browser display.
 
        <p><b>Displaying regions from the same chromsome</b></br>
        The following BED12 <a href="../../~chmalee/bed12formultregion.txt">example</a> specifies
        a 3 block region on chromsome 22 between bases 1000 and 5000, and a second 3 block 
        region between bases 2000 and 6000:<br>
        <pre>
     chr22 1000 5000 cloneA 960 + 1000 5000 0 3 567,488,900, 0,2512,3100
     chr22 2000 6000 cloneB 900 + 2000 6000 0 3 433,399,800, 0,2601,3200
        </pre>
        <p>
        When loaded as a custom track, we can see that there are overlapping regions, as well
        as a block between positions 2001 and 2433.
        <img style="border: 1px solid black; display: block;"
             src="../../images/hgIntegratorHgTracksExample.png">
        <p>
        Now examine this region under the multi-region view, with the above BED12 positions
        loaded as a custom BED <a href="../../~chmalee/bed12formultregion.txt">URL</a>.
        <img style="border: 1px solid black; display: block;"
             src="../../images/hgIntegratorHgTracksExample.png">

        <p>
        The positions display in the order specified by the BED file, not by genomic position.
        Notice how the <em># need to update later #</em> block is in the fourth region, not the second
        region, as in the custom track example.  
        <p>
        For an example BED12 URL, click <a href="../../~chmalee/bed12formultregion.txt">here</a>.
        
        <p><b>Displaying regions from multiple chromsomes</b></br>
        Multi-region view also supports BED files specifying regions from across multiple
        chromosomes, allowing  display of annotations from different genomic regions in the same
        browser window. The following <a href="../../~chmalee/multRegion2.txt">file</a> 
        specifies regions from across three different chromosomes:
        
           <pre>
     chr9 133255000 133276861
     chr21 31659622 31668931
     chr13 69700594 69708459
           </pre>
        
        Again the multi-region view will display the &quot;slices&quot; according to the order
        specified in the BED file, and allows the same region to be displayed twice.

        <ul><img style="border: 1px solid black; display: block;"
             src="">

	</div>< .sectionContent --> 
</div><!-- .section -->

</body>
</html>
