<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
	"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
	<title>Multi-Region Display Help Guide - UCSC Genome Browser</title>
        <meta name="description" content="Multi-Region Display">
        <meta name="keywords" content="genome, genome data, data integrator, table browser, genome browser, genome assembly, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, multi region">
	<meta http-equiv="content-type" content="text/html; charset=utf-8">
	<link rel="stylesheet" href="/style/HGStyle.css">
</head>
<body class="oldStaticPage">

<div class="header">
	<img src="../../images/title.jpg" alt="UCSC Genome Bioinformatics">
</div><!-- .header -->

<div class="nav">
<!--#include virtual="../../cgi-bin/hgMenubar"-->
</div><!-- .nav -->

<div class="content">

<div class="section">
<h1>Multi Region Display Help</h1>
<div class="sectionContent">
<p><b>Table of Contents:</b>
<td>
  <ul><b>
    <li><a class="toc" href="#Intro">Introduction</a>
    <li><a class="toc" href="#About">About the multi-region view</a>
    <li><a class="toc" href="#GetStarted">Getting started with the multi-region view</a>
    <ul>
       <li><a class="toc" href="#ExonOnly">Exon only</a>
       <li><a class="toc" href="#GeneOnly">Gene only</a>
       <li><a class="toc" href="#CustomRegions">Custom Regions</a>
       <li><a class="toc" href="#Haplotype">Haplotype view</a>    
    </ul>
    <li><a class="toc" href="#Examples">Examples</a>
  </b></ul>
</td>
                <P>
                <FORM name="googleForm1" method="GET" action="http://www.google.com/search" onSubmit="document.googleForm1.q.value=document.googleForm1.qq.value+'   site:genome.ucsc.edu/goldenPath/help';">          
                Search the Genome Browser help pages: &nbsp;
                <input type="hidden" name="q" value="">
                <input type="hidden" name="num" value="10">
                <input type="hidden" name="filter" value="0">
                <INPUT TYPE=text name=qq size=30 maxlength=255 value="">
                <input TYPE="submit" VALUE="Submit">
                </FORM>
                </p><p>
                <A HREF="../../contacts.html">Questions and feedback are welcome</A>.

</div><!-- .sectionContent -->
</div><!-- .section -->


<a name="Intro"></a>	
<div class="section">
	<h1>Introduction</h1>
	<div class="sectionContent">
	   <p>
              The multi-region display allows users the ability to &quot;slice&quot; up
              their normal track viewing experience into a variety of different modes:
              exon, gene, or user defined BED coordinates, and visualize 
              the track annotations only in those regions, effectively removing
              the intergenic, intron or otherwise unwanted regions from the viewing window.
        
           <p>
              In addition, for the human assemblies hg17, hg18, hg19 and hg38, the multi-region view
              also supports embedding a haplotype chromosome on top of it's
              reference sequence in order to visualize the differences between the haplotype
              sequence and the reference sequence.
           
           <p>
             Using the multi-region display, one can:<br>
             <ul>
                <li>Show all the SNP's that are within exons using the exon only view.
                <li>Show only the genes on a chromosome with the gene only view.
                <li>Display the same chromosomal position on either side of a gene in order 
                    to model a repeat region using a BED file with custom regions.
                <li>See how an alternate haplotype fits onto it's chromosome with the 
                    alternate haplotype view.
             </ul>
        
	</div><!-- .sectionContent --> 
</div><!-- .section -->

<a name="GetStarted"></a>
<div class="section">
	<h1>Getting Started with the multi-region view</h1>
	<div class="sectionContent">
	<p>
        Use the following instructions to display mutliple regions in the genome browser.
        
	<p>
	<b>Step 1. Pick a genome assembly and region</b><br>
           <ul>
              <li>Select the genome assembly you would like to view just as when normally
                  viewing an assembly, by using the top menu's Genomes link. If you would like
                  view custom regions for your assembly hub, navigate to it's browser
                  display as normal.

              <li>Once an assembly has been selected, you may search for a gene of interest 
                  or select a region to view using the Genome Browser search bar if desired,
                  just as normal.
           </ul>	
        <p>
	<b>Step 2. Select the multi-region view</b><br>
	   <ul>
              <li>Click <button type="button">multi-region</button> while on the track display
                  page to bring up the multi-region configuration window. <br># image needed here
                  for  where to click 
                  <img style="border: 1px solid black; display: block;"
                       src="">
              <li>By default the <b>Show single-chromosome view</b> will be selected. This is the
                  standard Genome Browser display, and can be used to return to a standard 
                  display.
           </ul>
        <p>
	<b>Step 3. Configure the multi-region view</b><br>
	   <ul>
              # Perhaps show the configuration window to the side of this text?
              <li>Now we can select which multi-region view we would like to display:<br>
                 <ul>
                    <li>For exon only select  <b>Show exons</b> 
                    <li>For gene only select <b>genes using ...</b>
                    <li>For custom regions select <b>Custom regions from BED URL</b>
                    <li>For haplotype select <b>Show one alternate haplotype...</b>
                 </ul>
              <li>Select the <b>Highlight alternating regions in multi-region view</b> box if
                  you would like every other exon or gene to be highlighted in blue. <br># image of
                  highlighted browser here.
                  <img style="border: 1px solid black; display: block;"
                       src="">
              <li>Click submit to be taken back to the Genome Browser display.
           </ul>
        <p>
        After entering multi-region mode, the usual browser display configurations will still 
        work, such as zooming in and out, reversing and resizing the view, highlighting and 
        zooming to highlighted selection, and track search.
 
    </div><!-- .sectionContent --> 
</div><!-- .section -->

<a name="About"></a>
<div class="section">
        <h1>About the multi-region view</h1>
        <div class="sectionContent">

        <a name="ExonOnly"></a>
        <p>
        <h4>Exon Only View</h4><br>
           The exon only view removes the intronic bases, and thus only displays annotations
           that fall within an exon.

        <p>depending on whether you would like to view intronic regions (genes view)
           or non intronic (exons view). Note that both views remove all intergenic regions, 
           except for the number of bases noted in the <b>padding</b> field, pads each exon 
           or gene with the specified number of bases on either side of the item. Thus if 
           the exon only view is selected the padding bases will be intron or intergenic 
           while if gene only view is selected the padding bases will be all intergenic.
        
        <a name="GeneOnly"></a>
        <p>
        <b>Gene Only View</b><br>
           The gene only view allows one to remove all intergenic bases in a given range, thus
           only displaying annotations that fall within the range of a gene. 

        <b>Important Notes about the exon and gene only view</b>
           <ul>
              <li>Exon and gene slicing is computed relative to the "most annotated" gene track for
                  that assembly, and includes all exons and genes in every subtrack if the 
                  gene track used is a super-track or composite track.
              <br># Session or screenshot example<br>
              <li>For the gene only option, the longest transcript in a given region is used as the 
                  reference for where to slice. This may be confusing when genes are nested inside
                  other genes:
                  # Screenshot example of nested gene
                  <img style="border: 1px solid black; display: block;"
                       src="">
                  
              <li>Exon and gene only views are unavailable with assembly hubs. This is because
                  assembly hubs do not come with a default gene track, however viewing custom
                  regions should still work.
           </ul>
        <p>  
        While in exon or gene only mode, the number of bases used for padding comes from the
        intergenic sequence(gene only) or intron and intergenic sequence (exon only). This may
        lead to confusion when there are fewer intergenic or intronic bases between each exon
        or gene than the padding dictates. For example, on tightly packed chromosomes like 
        the mitochondiral chromosome, there may actually be zero bases in between elements,
        and thus the multi-region display may behave oddly in these types of situtations.
        
        <a name="CustomRegions"></a>
        <p>
        <b>Custom Regions</b></br>
        <p>
        Searching and zooming while using BED URLs are a little different than exon and gene
        only mode. When a term or position is entered into the search box from the tracks
        display page, the browser first searches for the term within the multi-region view,
        and if found, the browser will zoom to that feature. If the term is not found, 
        then the browser will search the default search set and redirect if found.
        
        <a name="Haplotype"></a>
        <p>
        <b>Haplotype</b><br>
	<p>
        The human hg38, hg19, hg18 and hg17 assemblies all possess an extra feature while
        using the multi-region view: showing an alternate haplotype on it's chromosome.
        <p>
        For a list of possible haplotypes, click on the <b>sequences</b> link while
        on the human gateway page:
        <br># Picture of link here #
	<img style="border: 1px solid black; display: block;"
	   src="">
        Any haplotype of the form <em>chr_name_alt</em> can be placed on it's chromosome,
        unlike the random and unplaced sequences, which are currently not supported by
        the multi-region view.
        
        <br># Example of haplotype view #
	<img style="border: 1px solid black; display: block;"
	src="">

        <p><b>Notes about haplotype display</b>
        <ul>
           <li>Only one haplotype can be displayed at a time. Thus you cannot display
               multiple haplotypes on the same chromsome to see how they compare. 
        </ul>
        <p>
        Be careful zooming out too far while using the haplotype view. You may zoom out off the
        end of a chromsome and onto the next one. Theoretically you could display all 25
        chromosomes this way, although it will be slow.

	</div><!-- .sectionContent --> 
</div><!-- .section -->


<a name="Examples"></a>
<div class="section">
<h1>Examples</h1>       
        <div class="sectionContent">
        <p>
        <b>Gene only example</b><br>

        <p>
        <b>Exon only example</b><br>
        
        <p>
        <b>BED3 example</b><br>
           # show multiple chromosomes
 
        <p>
        <b>BED12 example</b><br>
           # be sure to emphasize how regions show in order specified
             not genomic order <br>
           # probably should be from same chromosome since bed3
                  example is from mult chromosomes
        <p>
        <b>Haplotyple example</b><br>

	</div><!-- .sectionContent --> 
</div><!-- .section -->

<!--start removed sections
<div class="section">
<h1>Display a Custom Region with a BED File</h1>
	<div class="sectionContent">
	<p>
        One can now display multiple regions specified via 
        <a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format1">BED file</a> in 
        the Genome Browser display with the multi-region tool. These regions may be on one 
        chromosome or across multiple chromosomes.

        <p>
        To display multiple custom regions in the browser, click 
        <button type="button">multi-region</button> while on the track display
        page to bring up the multi-region configuration window. Select 
        <b>Custom regions from BED URL:</b> and paste in the URL to your BED
        file, then click <button type="button">submit</button>. 

        <p>
        The browser will now display the region(s) you have specified as &quot;slices&quot;,
        <em>with the regions displayed in the order of the BED file</em>. Note that you may 
        specify the same region multiple times in the BED file and it will be displayed
        as a separate region each time, making the modeling and comparison of the same region
        against many regions possible in one browser display.
 
        <p><b>Displaying regions from the same chromsome</b></br>
        The following BED12 <a href="../../~chmalee/bed12formultregion.txt">example</a> specifies
        a 3 block region on chromsome 22 between bases 1000 and 5000, and a second 3 block 
        region between bases 2000 and 6000:<br>
        <pre>
     chr22 1000 5000 cloneA 960 + 1000 5000 0 3 567,488,900, 0,2512,3100
     chr22 2000 6000 cloneB 900 + 2000 6000 0 3 433,399,800, 0,2601,3200
        </pre>
        <p>
        When loaded as a custom track, we can see that there are overlapping regions, as well
        as a block between positions 2001 and 2433.
        <img style="border: 1px solid black; display: block;"
             src="../../images/hgIntegratorHgTracksExample.png">
        <p>
        Now examine this region under the multi-region view, with the above BED12 positions
        loaded as a custom BED <a href="../../~chmalee/bed12formultregion.txt">URL</a>.
        <img style="border: 1px solid black; display: block;"
             src="../../images/hgIntegratorHgTracksExample.png">

        <p>
        The positions display in the order specified by the BED file, not by genomic position.
        Notice how the <em># need to update later #</em> block is in the fourth region, not the second
        region, as in the custom track example.  
        <p>
        For an example BED12 URL, click <a href="../../~chmalee/bed12formultregion.txt">here</a>.
        
        <p><b>Displaying regions from multiple chromsomes</b></br>
        Multi-region view also supports BED files specifying regions from across multiple
        chromosomes, allowing  display of annotations from different genomic regions in the same
        browser window. The following <a href="../../~chmalee/multRegion2.txt">file</a> 
        specifies regions from across three different chromosomes:
        
           <pre>
     chr9 133255000 133276861
     chr21 31659622 31668931
     chr13 69700594 69708459
           </pre>
        
        Again the multi-region view will display the &quot;slices&quot; according to the order
        specified in the BED file, and allows the same region to be displayed twice.

        <ul><img style="border: 1px solid black; display: block;"
             src="">

	</div>< .sectionContent --> 
</div><!-- .section -->

</body>
</html>
