<P>
The <em> Tetraodon nigroviridis</em> V7 (February 2004) assembly was provided 
<A HREF="http://www.genoscope.cns.fr/" TARGET=_blank>Genoscope</A>, Evry, France
and it results from a collaboration between Genoscope and the 
<A HREF="http://www.broad.mit.edu/annotation/tetraodon/background.html" TARGET=_blank>Eli and Edythe L. Broad Institute</A> of MIT and Harvard.  The sequence 
data was generated by both institutes and the assembly was performed using the
Arachne program.
</P>
<P> The project was supported by the Consortium National de Recherche en 
Genomique and the National Human Genome Research Institute (NHGRI).

<H3>Sample position queries</P></H3>
<P>
A genome position can be specified by the accession number of
an mRNA or EST, a chromosome range, or keywords from 
the GenBank description of an mRNA. The following list provides 
examples of various types of position queries for the Tetraodon genome. Note 
that some position queries (e.g. "huntington") may return matches to the mRNA 
records of other species. In these cases, the mRNAs are mapped to their 
homologs in Tetraodon.
See the 
<A HREF="http://genome.cse.ucsc.edu/goldenPath/help/hgTracksHelp.html" TARGET=_blank>User Guide</A> for more help. 
<P>

<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
<TR><TD VALIGN=Top NOWRAP><B>Request:</B><br></TD>
	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><br></TD></TR>
	
<TR><TD VALIGN=Top><br></TD></TR>
	
<TR><TD VALIGN=Top NOWRAP>chr1</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays all of chromosome 1</TD></TR>
<TR><TD VALIGN=Top NOWRAP>chr1:1-200000</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays first two hundred thousand bases of chromosome 1</TD></TR>
<TR><TD VALIGN=Top><br></TD></TR>

<TR><TD VALIGN=Top NOWRAP>CR731623</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays region containing Tetraodon mRNA with GenBank accession number CR731623</TD></TR>

<TR><TD VALIGN=Top NOWRAP>BU572495</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays region containing Tetraodon EST with GenBank accession BU572495
</TD></TR>

<TR><TD VALIGN=Top><br></TD></TR>

<TR><TD VALIGN=Top NOWRAP>IL11</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs related to IL11 (interleukin-11)</TD></TR> 
</TD></TR>


<TR><TD VALIGN=Top NOWRAP>pseudogene mRNA</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists transcribed pseudogenes but not cDNAs, in GenBank</TD></TR>
<TR><TD VALIGN=Top NOWRAP>T-cell receptor</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs for T-cell receptor genes in GenBank</TD></TR>
<TR><TD VALIGN=Top NOWRAP>kinase</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists many kinase mRNAs</TD></TR
<TR><TD VALIGN=Top NOWRAP>tyrosine kinase</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists only tyrosine kinases</TD></TR>
<TR><TD VALIGN=Top NOWRAP>breast cancer</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs associated with breast cancer</TD></TR>

<TR><TD VALIGN=Top><br></TD></TR>

<TR><TD VALIGN=Top NOWRAP>abkevich</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Abkevich</TD></TR>
<TR><TD VALIGN=Top NOWRAP>Jaillon, O.</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs deposited by co-author O. Jaillon</TD></TR>

<TR><TD VALIGN=Top><br></TD></TR>
	
<TR><TD COLSPAN="3" > Use this last format for entry authors -- even though GenBank searches require Jaillon O format, GenBank entries themselves use 
Jaillon, O. internally.
</TABLE>
<P>
<HR>
<H3>Assembly details</H3>
This assembly was constructed with using Whole Genome Shotgun (WGS) resulting 
in a sequence coverage of about 7.9X. This assembly contains 45,609 contigs and 25,773 scaffolds generated by the Arachne program, and covers >90% of the 
genome. Additional linking data were used to build ultracontigs and to organize the assembly into chromosomes.  The size of the Tetraodon genome is estimated,
by Genoscope, to be about 385 Mb.
<P>
Downloads of the Tetraodon data and annotations may be obtained from the UCSC
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/tetNig1/" TARGET=_blank>FTP 
site</A> or <A HREF=http://hgdownload.cse.ucsc.edu/downloads.html#Tetraodon>
Downloads page</A>. 
</P>
<P>
These data have been freely provided by Genoscope before publication for use
in the UCSC Genome Browser with the following understanding:
These data are made available before scientific publication with the following understanding
<OL>
<li>The data may be freely downloaded, used in analyses, and repackaged in 
databases. </li>
<li>Users are free to use the data in scientific papers analyzing particular 
genes and regions, provided that the Genoscope is properly acknowledged.</li>
<li>Genoscope reserves the right to publish the initial large-scale analyses of the dataset-including large-scale identification of regions of evolutionary 
conservation and large-scale genomic assembly. Large-scale refers to regions 
with size on the order of a Tetraodon chromosome (that is, 5 Mb or more).</li>
</OL>
</P>

