<P>
The <em> Tetraodon nigroviridis</em> V7 (May 2002) whole genome shotgun assembly was provided by 
<A HREF="http://www.genoscope.cns.fr/" TARGET=_blank>Genoscope</A>, Evry, France
as part of the Genoscope and Center for Genome Research at the
<A HREF="http://www.broad.mit.edu/annotation/tetraodon/background.html" TARGET=_blank>Eli and Edythe L. Broad Institute</A> collaboration.  The part of the 
project at the Center for Genome Research is funded by the National Human
Genome Research Institute (NHGRI). 

<H3>Sample position queries</P></H3>
<P>
A genome position can be specified by the accession number of
an mRNA or EST, a chromosome range, or keywords from 
the GenBank description of an mRNA. The following list provides 
examples of various types of position queries for the Fugu genome. Note that
some position queries (e.g. "huntington") may return matches to the mRNA records
of other species. In these cases, the mRNAs are mapped to their homologs in Fugu.
See the 
<A HREF="http://genome.cse.ucsc.edu/goldenPath/help/hgTracksHelp.html" TARGET=_blank>User Guide</A> for more help. 
<P>

<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
<TR><TD VALIGN=Top NOWRAP><B>Request:</B><br></TD>
	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><br></TD></TR>
	
<TR><TD VALIGN=Top><br></TD></TR>
	
<TR><TD VALIGN=Top NOWRAP>chr1</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays all of chromosome 1</TD></TR>
<TR><TD VALIGN=Top NOWRAP>chr1:1-200000</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays first two hundred thousand bases of chromosome 1</TD></TR>
<TR><TD VALIGN=Top><br></TD></TR>

<TR><TD VALIGN=Top NOWRAP>AF083811</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays region containing Tetraodon mRNA with GenBank accession number AJ310911</TD></TR>

<TR><TD VALIGN=Top NOWRAP>BU572495</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays region containing Tetraodon EST with GenBank accession BU572495
</TD></TR>

<TR><TD VALIGN=Top><br></TD></TR>

<TR><TD VALIGN=Top NOWRAP>p53</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs related to the p53 tumor suppressor</TD></TR> 
</TD></TR>


<TR><TD VALIGN=Top NOWRAP>pseudogene mRNA</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists transcribed pseudogenes but not cDNAs, in GenBank</TD></TR>
<TR><TD VALIGN=Top NOWRAP>homeobox caudal</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes in GenBank</TD></TR>
<TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD></TR>
<TR><TD VALIGN=Top NOWRAP>kruppel zinc finger</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD></TR>
<TR><TD VALIGN=Top NOWRAP>huntington</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs associated with Huntington's disease</TD></TR>

<TR><TD VALIGN=Top><br></TD></TR>

<TR><TD VALIGN=Top NOWRAP>zahler</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Zahler</TD></TR>
<TR><TD VALIGN=Top NOWRAP>Evans,J.E.</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs deposited by co-author J.E.  Evans</TD></TR>

<TR><TD VALIGN=Top><br></TD></TR>
	
<TR><TD COLSPAN="3" > Use this last format for entry authors -- even though GenBank searches require Evans JE format, GenBank entries themselves use Evans,J.E.  internally.
</TABLE>
<P>
<HR>
<H3>Assembly details</H3>
This assembly was constructed with using Whole Genome Shotgun (WGS) resulting 
in a sequence coverage of about 7.9X. WGC libraries were obtained from 
Genoscope.  A plasmid library was prepared by random shearing and size selction of DNA resulting in an average insert size of 4 kb.  Genoscope has also 
generated reads that are from plasmid or BAC end sequences.  This assembly 
contains 49,509 contigs totaling 319 million base pairs.  Coverage is 
estimated to be 83% in current contigs.  
<P>
Downloads of the Tetraodon data and annotations may be obtained from the UCSC
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/tetNig1/" TARGET=_blank>FTP 
site</A> or <A HREF=http://hgdownload.cse.ucsc.edu/downloads.html#Tetraodon>
Downloads page</A>. 
</P>
<P>
These data have been freely provided by Genoscope before publication for use
in the UCSC Genome Browser with the following understanding:
These data are made available before scientific publication with the following understanding
<OL>
<li>The data may be freely downloaded, used in analyses, and repackaged in 
databases. </li>
<li>Users are free to use the data in scientific papers analyzing particular 
genes and regions, provided that the Genoscope is properly acknowledged.</li>
<li>Genoscope reserves the right to publish the initial large-scale analyses of the dataset-including large-scale identification of regions of evolutionary 
conservation and large-scale genomic assembly. Large-scale refers to regions 
with size on the order of a Tetraodon chromosome (that is, 5 Mb or more).</li>
</OL>
</P>

