<!DOCTYPE html PUBLIC "-//IETF//DTD HTML 2.0//EN">

<HTML>
  <HEAD>
    <TITLE>Genbank configuration file</TITLE>
  </HEAD>

  <BODY bgcolor="white">
    <H1>Genbank configuration file</H1>
    The file <CODE><EM>$gbRoot</EM>/etc/genbank.conf</CODE>
    contains options control the alignment and loading of
    GenBank and RefSeq data.  The file is in the same format as
    <CODE>.hg.conf</CODE>:
    <UL>
    <LI> blank lines or lines where the first non-blank character is a &quot;#&quot; are ignored.
    <LI> records are lines of:<BR>
         <CODE><EM>name</EM>=<EM>value</EM></CODE>
    </UL>
    <H3>Global configuration parameters</H3>
    <UL>
    <LI> <CODE>cluster.iservers</CODE> - space-separated list of iservers to use
         to serve sequence fasta files.  Empty <CODE>no</CODE> or empty
         for a global filesystem.
    <LI> <CODE>cluster.rootDir</CODE> - root directory on cluster filesystems,
         either iserver or local file systems.
    <LI> <CODE>cluster.paraHub</CODE> - Host where parasol hub is running.
    </UL>
    <H3>per-database configuration parameters</H3>
    Per-database configuration parameters start with the database name,
    represented here by <CODE><EM>$db</EM></CODE>.  Default values are set with a
    database name of <CODE>default</CODE>

    <UL>
     <li> <code><em>$db</em>.genome</code> - glob pattern for genome
          sequences, which can be NIB or FASTA files.
     <li> <code><em>$db</em>.lift</code> - path to lift file for genome,
          or <CODE>no</CODE> if none.

     <li> <code><em>$db</em>.genbank.mrna.native.pslReps</code> - Arguments
           for <CODE>pslReps</CODE> for GenBank native mRNAs, or <CODE>no</CODE> to skip pslReps.
     <li> <code><em>$db</em>.genbank.mrna.xeno.pslReps</code>
     <li> <code><em>$db</em>.genbank.est.native.pslReps</code>
     <li> <code><em>$db</em>.genbank.est.xeno.pslReps</code>

     <li> <code><em>$db</em>.refseq.mrna.name.pslReps</code> - Arguments
           for <CODE>pslReps</CODE> for RefSeq native mRNAs, or <CODE>no</CODE> to skip pslReps.
     <li> <code><em>$db</em>.refseq.mrna.xeno.pslReps</code>

     <li> <code><em>$db</em>.mgcTables</code> - indicates which MGC tables to
          create.  <CODE></CODE>
     <li> <code><em>$db</em>.perChromTables</code> - Set to <CODE>no</CODE>
          if per-chromosome alignment tables should not be created.
     <li> <code><em>$db</em>.genbank.mrna.native.load</code> - Load native
          GenBank mMRNAs, <CODE>yes</CODE> or <CODE>no</CODE>.
          Also for type of <CODE>est</CODE> and organism category
          of <CODE>xeno</CODE>.
     <li> <code><em>$db</em>.refseq.mrna.native.load</code> - Load native
          GenBank mMRNAs, <CODE>yes</CODE> or <CODE>no</CODE>.
          Also for organism category of <CODE>xeno</CODE>.
     <li> <code><em>$db</em>.genbank.mrna.native.description</code>
           - Load description table for native mRNAs, <CODE>yes</CODE> or <CODE>no</CODE>.
          Also for type of <CODE>est</CODE> and organism category
          of <CODE>xeno</CODE>.
     <li> <code><em>$db</em>.refseq.mrna.native.description</code>
           - Load description table for native mRNAs, <CODE>yes</CODE> or <CODE>no</CODE>.
          Also for type of <CODE>est</CODE> and organism category
          of <CODE>xeno</CODE>.
     <li> <code><em>$db</em>.iservers</code> - List of iservers to use for
          alignment.  Normally the <CODE>default</CODE> entry is used,
    </UL>
  </BODY>
</HTML>

