<H2>Description</H2>
<P>
This track depicts the conservation score resulting from an analysis of 
the 12 species represented in the 
<A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>blastz</A> 
alignments. </P>

<H2>Methods</H2>
<P>
Conservation scores 
were generated by weighing the relative contribution of each species 
sequence to the overall conservation score, based on how diverged each 
species was from the reference sequence. This weighting scheme was 
determined by calculating the neutral substitution rates at 4-fold 
degenerate positions. </P>
<P>
Additional information about this conservation score will be 
available in the paper Margulies, E. et al. "Multi-Species Sequencing of Targeted Genomic 
Regions: Comparative Studies of Genome Evolution and Function". Submitted to 
<em>Genome Res</em>.</P>

<H2>Credits</H2>
<P>
This track was generated by <A HREF="mailto:elliott@nhgri.nih.gov">Elliott 
Margulies</A> at NHGRI. <A HREF="mailto:weber@soe.ucsc.edu">Ryan Weber</A> of 
the UCSC Genome Bioinformatics Group 
provided the computational support for the track display.</P>
