<H2>Description</H2>
<P>
This track depicts the conservation score resulting from an analysis of 
the species represented in the 
<A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>blastz</A> 
alignments shown in the ZooSeq track. </P>

<H2>Methods</H2>
<P>
Conservation scores 
were generated by weighing the relative contribution of each species 
sequence to the overall conservation score, based on how diverged each 
species was from the reference sequence. This weighting scheme was 
determined by calculating the neutral substitution rates at 4-fold 
degenerate positions. </P>
<P>
Additional information about this conservation score can be found
in the paper Margulies, E. et al. (2003) 
<A HREF="http://www.genome.org/cgi/content/abstract/13/12/2507?etoc" 
TARGET=_blank>"Identification and Characterization 
of Multi-Species Conserved Sequences"</A>. <em>Genome Res.</em> 
13:2507-2518. 

<H2>Credits</H2>
<P>
This track was generated by 
<A HREF="mailto:&#101;l&#108;&#105;o&#116;&#116;&#64;&#110;hg&#114;&#105;.&#110;&#105;h.&#103;&#111;&#118;">Elliott Margulies</A> at NHGRI. 
<A HREF="mailto:&#119;e&#98;&#101;&#114;&#64;&#115;&#111;&#101;.u&#99;&#115;&#99;.ed&#117;">Ryan Weber</A> of 
the UCSC Genome Bioinformatics Group 
provided the computational support for the track display.</P>
