<H2>Description</H2>
<P>This track displays the level conservation of the Zoo sequences at any 
given position. More precisely, it computes the p-value of observing a certain 
parsimony score for a given column of the multiple alignment, under the null 
model of neutral evolution.</P>

<H2>Methods</H2>
<P>
Adam Siepel inferred the phylogenetic tree relating the Zoo
species, as well as the length of each branch of the tree. This
tree was used to measure the surprise of the conservation observed.
<P>
For each column of the multiple alignment produced by Webb Miller, 
the parsimony score for the columm was computed using Adam's phylogenetic
tree. The p-value was then computed for that parsimony score under a null
model of neutral evolution (see below for a description of the null
model). Species with gaps in the column were ignored. A column received a
low p-value if the parsimony score for the set of non-gap species was
surprisingly low, given the divergence of these species. For column i, let
Xi be the negative log of the p-value for that column.
<P>
For position i of the alignment, the score plotted on the track is the
average of the Xj's in a window of 10bp around position i.
</P>

<H2>Credits</H2>
<P>This track was generated by 
<A HREF="mailto:blanchem@soe.ucsc.edu">Mathieu Blanchette</A>, a former
post-doctoral fellow with the UCSC Genome Bioinformatics Group. Many thanks to 
Adam Siepel for the phylogenetic inferrences and to Webb Miller for providing 
the multiple alignment of the Zoo sequences.</P>
