<H2>Description</H2>
<P>
N-SCAN gene predictions.
</P>

<H2>Methods</H2>
<H3>N-SCAN</H3>
<P>
N-SCAN combines biological-signal modeling in the target genome sequence along
with information from a multiple-genome alignment to generate de novo gene
predictions. It extends the TWINSCAN target-informant genome pair to allow for
an arbitrary number of informant sequences as well as richer models of
sequence evolution. It models the phylogenetic relationships between the
aligned genome sequences, context-dependent substitution rates, insertions,
and deletions.
</P>
<P>
Mouse N-SCAN uses rhesus (rheMac2) as the informant.
</P>

<H2>References</H2>
<P>

Marijke J. van Baren and Michael R. Brent
<A HREF="http://www.genome.org/cgi/reprint/16/5/678.pdf">
Iterative gene prediction and pseudogene removal improves genome annotation</A>
<EM>Genome Research</EM> 16:678-685 (2006).</P>

<P>
Gross, S.S. and M.R. Brent, 
<A HREF="http://www.stanford.edu/~ssgross/RECOMB-2005.pdf">
Using Multiple Alignments To Improve Gene Prediction</A>, in
<EM>9th Annual International Conference, RECOMB 2005</EM>,
S. Miyano, et al., Editors. 2005, Springer: Boston. p. 374-388.
</P>
<P>
Korf, I., Flicek, P., Duan, D. and Brent, M.R. 
<A HREF="http://bioinformatics.oupjournals.org/cgi/reprint/17/suppl_1/S140.pdf"
TARGET=_blank>Integrating genomic homology into gene structure prediction</A>.
<em>Bioinformatics</em> <B>17</B>, S140-148 (2001).</P>

<H2>Credits</H2>
<P>
Thanks to <A HREF="http://genes.cs.wustl.edu/BrentLab/MB-Lab-index.html" 
TARGET=_blank>Michael Brent's Computational Genomics Group</A> at Washington 
University St. Louis for providing this data.
</P>
