<H2>Description</H2> <P>
This track displays multiz multiple alignments of seven species
of the genus Saccharomyces.  The graphic display shows the alignment
projected onto S. cerevisiae.  Click on the alignment to see the full
alignment including sequence not in S. cerevisiae.

The genomes were downloaded from:<BR>
<UL>
<LI><I>S. cerevisiae</I> -  ftp://genome-ftp.stanford.edu/pub/yeast/data_download/sequence/genomic_sequence
<LI><I>S. paradoxus</I> - http://www.broad.mit.edu/ftp/pub/annotation/fungi/comp_yeasts/S1a.Assembly/Spar_contigs.fasta
<LI><I>S. mikatae</I> - http://www-genome.wi.mit.edu/ftp/pub/annotation/fungi/comp_yeasts/S1a.Assembly/Smik_contigs.fasta
<LI><I>S. kudriavzevii</I> - http://www.genetics.wustl.edu/saccharomycesgenomes/Contigs/YM6553.fsa.gz
<LI><I>S. bayanus</I> - http://www-genome.wi.mit.edu/ftp/pub/annotation/fungi/comp_yeasts/S1a.Assembly/Sbay_contigs.fasta
<LI><I>S. castelli</I> - http://www.genetics.wustl.edu/saccharomycesgenomes/Contigs/YM476.fsa.gz
<LI><I>S. kluyveri</I> - http://www.genetics.wustl.edu/saccharomycesgenomes/Contigs/YM479.fsa.gz
</UL>

<H2>Methods</H2>
Multiz is a multiple alignment program that takes
<A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>blastz</A> 
"Best-in-Genome" alignments (axtBest) as input.  Multiz and blastz were run
with was run with the scoring matrix:
<PRE>
        A     C     G     T
  A    91  -114   -31  -123
  C  -114   100  -125   -31
  G   -31  -125   100  -114
  T  -123   -31  -114    91

  O = 400, E = 30, K = 3000, L = 3000, M = 50
</PRE> 
The overall score is the sum of the score over all pairs.  </P>

<H2>Credits</H2>
<P>
The <em>Multiz</em> program  created by Webb Miller of the 
<A HREF="http://bio.cse.psu.edu/" TARGET=_blank> Penn State Bioinformatics Group</A>.   
Jim Kent at UCSC wrote
<A HREF="http://www.genome.org/cgi/content/full/13/1/103" TARGET=_blank>axtBest</A>
and the scripts to run multiz genome-wide and to display
the alignments in this browser.  
Sequence data was provided by Washington University and 
the Whitehead Institute. </P>

<H2>References</H2>

<P> Aligning Multiple Genomic Sequences with the Threaded Blockset Aligner.
Blanchette M, Kent WJ, Riemer C, Elnitski L, Smit AFA, Roskin KM, Baertsch R,
Rosenbloom K, Clawson H, Green ED, Haussler D, Miller W.  Submitted for publication.</P>

<P><A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>Human-Mouse Alignments with BLASTZ</A>. Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison R, Haussler H, and Miller W.  <em>Genome Research</em> 2003 Jan;13(1):103-7.</P>

<P><A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" TARGET=_blank>Scoring pairwise genomic sequence alignments</A>. Chiaromonte F, Yap VB, Miller W. <em>Pac Symp Biocomput</em> 2002;:115-26.</P>


