<H2>Description</H2> <P>
This track displays <tt>multiz</tt> multiple alignments of seven species
of the genus Saccharomyces.  The graphic display shows the alignment
projected onto <em>S. cerevisiae</em>. Click on the alignment to see the full
alignment including sequence not in <em>S. cerevisiae</em>.
<P>
The genomes were downloaded from:<BR>
<UL>
<LI><I>S. cerevisiae</I> - <A HREF="ftp://genome-ftp.stanford.edu/pub/yeast/data_download/sequence/genomic_sequence" TARGET=_blank>ftp://genome-ftp.stanford.edu/pub/yeast/data_download/sequence/genomic_sequence</A>
<LI><I>S. paradoxus</I> - <A HREF="http://www.broad.mit.edu/ftp/pub/annotation/fungi/comp_yeasts/S1a.Assembly/Spar_contigs.fasta" TARGET=_blank>http://www.broad.mit.edu/ftp/pub/annotation/fungi/comp_yeasts/S1a.Assembly/Spar_contigs.fasta</A>
<LI><I>S. mikatae</I> - <A HREF="http://www-genome.wi.mit.edu/ftp/pub/annotation/fungi/comp_yeasts/S1a.Assembly/Smik_contigs.fasta" TARGET=_blank>http://www-genome.wi.mit.edu/ftp/pub/annotation/fungi/comp_yeasts/S1a.Assembly/Smik_contigs.fasta</A>
<LI><I>S. kudriavzevii</I> - <A HREF="http://www.genetics.wustl.edu/saccharomycesgenomes/Contigs/YM6553.fsa.gz" TARGET=_blank>http://www.genetics.wustl.edu/saccharomycesgenomes/Contigs/YM6553.fsa.gz</A>
<LI><I>S. bayanus</I> - <A HREF="http://www-genome.wi.mit.edu/ftp/pub/annotation/fungi/comp_yeasts/S1a.Assembly/Sbay_contigs.fasta" TARGET=_blank>http://www-genome.wi.mit.edu/ftp/pub/annotation/fungi/comp_yeasts/S1a.Assembly/Sbay_contigs.fasta</A>
<LI><I>S. castelli</I> - <A HREF="http://www.genetics.wustl.edu/saccharomycesgenomes/Contigs/YM476.fsa.gz" TARGET=_blank>http://www.genetics.wustl.edu/saccharomycesgenomes/Contigs/YM476.fsa.gz</A>
<LI><I>S. kluyveri</I> - <A HREF="http://www.genetics.wustl.edu/saccharomycesgenomes/Contigs/YM479.fsa.gz" TARGET=_blank>http://www.genetics.wustl.edu/saccharomycesgenomes/Contigs/YM479.fsa.gz</A>
</UL>

<H2>Methods</H2>
<tt>Multiz</tt> is a multiple alignment program that takes
<A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank><tt>blastz</tt></A> 
"best-in-genome" alignments (<tt>axtBest</tt>) as input.  <tt>Multiz</tt> and <tt>blastz</tt> were run
with the scoring matrix:
<PRE>
        A     C     G     T
  A    91  -114   -31  -123
  C  -114   100  -125   -31
  G   -31  -125   100  -114
  T  -123   -31  -114    91

  O = 400, E = 30, K = 3000, L = 3000, M = 50
</PRE> 
The overall score is the sum of the score over all pairs.  </P>

<H2>Credits</H2>
<P>
The <tt>multiz</tt> program was created by Webb Miller of the 
<A HREF="http://bio.cse.psu.edu/" TARGET=_blank>Penn State Bioinformatics Group</A>.   
Jim Kent at UCSC wrote <tt>axtBest</tt> and the scripts to run <tt>multiz</tt> genome-wide and to display
the alignments in this browser.  
Sequence data were provided by <A HREF="http://genome.wustl.edu/" TARGET=_blank>Washington University</A> 
and the <A HREF="http://www.wi.mit.edu/" TARGET=_blank>Eli & Edythe L. Broad Institute</A> at 
MIT/Harvard. </P>

<H2>References</H2>

<P> Aligning Multiple Genomic Sequences with the Threaded Blockset Aligner.
Blanchette M, Kent WJ, Riemer C, Elnitski L, Smit AFA, Roskin KM, Baertsch R,
Rosenbloom K, Clawson H, Green ED, Haussler D, Miller W.  Submitted for publication.</P>

<P><A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>Human-Mouse Alignments with BLASTZ</A>. Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison R, Haussler D, and Miller W.  <em>Genome Research</em> 2003 Jan;13(1):103-7.</P>

<P><A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" TARGET=_blank>Scoring pairwise genomic sequence alignments</A>. Chiaromonte F, Yap VB, Miller W. <em>Pac Symp Biocomput</em> 2002;:115-26.</P>


