<h2>Description</h2>
<p>
Allele-Specific Expression (ASE) is a measurement of the difference of gene expression 
between the two alleles of a gene in a heterozygous individual.  
This difference (also called `allelic imbalance') can be due to many biological processes, 
including presence of cis-regulatory genetic variants, imprinting, and nonsense-mediated decay.</p>   
<p>
This track hub shows summary tracks of ASE identified from transcriptome and
genotype data collected in 53 tissues by the 
<a target="_blank" href="https://commonfund.nih.gov/GTEx/index">Genotype-Tissue Expression (GTEx) project</a>, and analyzed by the GTEx Analysis Group
(<a target="_blank" href="http://tllab.org">Lappalainen Lab</a> at the New York Genome Center).</p>

<p>
The GTEx project was initiated by the NIH as a sample and data resource for studies 
on the relationship between genetic variation and gene expression in multiple human tissues.
The data presented here is from the GTEx midpoint milestone data release (V6, October 2015).
For additional information, see the 
<a target="_blank" 
href="http://genome-test.soe.ucsc.edu/gbdb/hubs/gtexAnalysis/gtexAnalysisHub.html">
<!-- Replace with below when releasing to hgdownload -->
<!-- href="http://hgdownload.soe.ucsc.edu/hubs/gtexAnalysis/gtexAnalysisHub.html">-->
GTEx Analysis hub description</a>.</p>

<p>
For detailed analysis of ASE patterns in GTEx, the full ASE data is available via dbGap.

<h2>Display Conventions</h2>
<p>
There is a summary track of ASE sites for each of the 53 tissues assayed by the GTEx Consortium.
Each site is identified by the SNP ID, or if none assigned, then by chromosomal start position.
Sites are colored based on the median ASE for the site across all samples of the tissue;
shaded from gray (low) 
to the GTEx convention tissue color (high) with an intermediate color representing
moderate ASE level.</p>
<p>
Cross-tissue summaries are provided by the ASE Sites track which shows the median allelic 
imbalance at each SNP across all tissues with ASE data, and a Density graph, 
which shows the median allelic imbalance across tissues in a sliding window.

<h2>Methods</h2>
<p>
Allelic expression tracks were generated on a per tissue basis. Allelic counts were generated 
at heterozygous variants called from Illumina OMNI 2.5 Array genotyping with imputation using
the GATK ASEReadCounter tool. Only uniquely aligned reads with a base quality of at least 10
at heterozygous sites were used.</p>
<p>
The allelic imbalance value per site (ranged between 0 and 0.5) is the median allelic imbalance
 across individuals with sufficient allelic coverage, calculated as 
|0.5-REF_COUNT/(REF_COUNT+ALT_COUNT)|. Sites with low mappability (ENCODE 50 bp mappability 
score less than 1) or that showed mapping bias in simulations (Panousis et al., 2014) 
were removed.</p>
<p>
The cross-tissue density graph was generated by a sliding window algorithm (window size 10 variants, step of 2 variants).  Only those variants within 10kb of the first variant in the window 
are included.

<h2>Credits</h2>
<p>
Data and suggestions for track design were provided by the Lappalainen lab at the New York Genome Center, part of the GTEx Analysis Working Group. </p>

<h2>Contacts</h2>
<p>
Stephane Castel (scastel@nygenome.org), Lappalainen Lab, NY Genome Center.</p>

<h2>References</h2>
<p>
Castel SE, Levy-Moonshine A, Mohammadi P, Banks E, Lappalainen T.
<a href="http://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0762-6" target="_blank">
Tools and best practices for data processing in allelic expression analysis</a>.
<em>Genome Biol</em>. 2015 Sep 17;16:195.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/26381377" target="_blank">26381377</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4574606/" target="_blank">PMC4574606</a></p>

