<H2>Description and Methods </H2>
<P>
Excerpted from the <A HREF="http://vega.sanger.ac.uk/" TARGET=_BLANK>Vega</A> 
main page:</P>
<P>
"The 
<A HREF="http://vega.sanger.ac.uk/" TARGET=_BLANK>
Vertebrate Genome Annotation (VEGA)</A> 
 is designed to be a central repository for high quality, frequently updated  
manual annotation of different vertebrate finished genome sequence.
Vega attempts to present consistent high-quality curation of the published 
chromosome sequences."</P>
<P>
"Finished genomic sequence is analysed on a clone by clone basis using
a combination of similarity searches against DNA and protein databases
as well as a series of <em>ab initio</em> gene predictions (GENSCAN, Fgenes).
The annotation is based on supporting evidence only."</P>
<P>
NOTE: <A HREF="http://vega.sanger.ac.uk/Danio_rerio/index.html" 
_TARGET=blank>VEGA</A> annotations are shown only on chromosomes 1-25 in this
assembly. For zebrafish, the VEGA annotations consist of 
<A HREF="http://www.sanger.ac.uk/Projects/D_rerio/clones.shtml" 
TARGET=_BLANK>BAC clones</A>
that were sequenced for the zebrafish clone mapping and sequencing
project. It is based on the tiling path information from the 
<A HREF="http://www.sanger.ac.uk/Projects/D_rerio/WebFPC/zebrafish/small.shtml"
 TARGET=_BLANK>fpc database</A>.

<H2> Credits </H2>
<P>
Thanks to Kerstin Howe, Mario Caccamo and Ian Sealy at the
<A HREF="http://www.sanger.ac.uk/" TARGET=_BLANK>
Wellcome Trust Sanger Institute</A> 
for providing the GFF files and additional annotations for the Vega genes. 
Vega gene annotations are 
generated by manual annotation by the 
<A HREF="http://www.sanger.ac.uk/Projects/D_rerio/" TARGET=_BLANK>
Zebrafish sequence analysis group</A> and 
the <A HREF="http://www.sanger.ac.uk/HGP/havana/" TARGET=_blank>HAVANA</A> 
group at the Sanger Institute and <A HREF="http://www.zfin.org/" 
TARGET=_BLANK>ZFIN</A>.
<P>
