<H2>Display</H2>
<P>This track displays repeats from <i>phusion.3 mouse</i> aligned to 
the <i>human</i> sequence. Each block in the alignment is displayed
as a colored block on the track with a line connecting all the blocks.
The color of each alignment indicates the percent identity of aligned
residues over all blocks of the alignment. 50% identity and below
is lightly colored and the color gets linearly darker as the percent identity
approaches 100%.

<H2>Methods</H2>
<P>The sequences were aligned with blastz (discontiguous exact seeds,
   ungapped extension, local alignments via dynamic programming) and
   postprocessed for single coverage.
   
<H2>Data</H2>
<h4>human sequence from:</h4>
   <a href="http://genome-test.cse.ucsc.edu/gs.8/oo.33/chromFa.zip">
   http://genome-test.cse.ucsc.edu/gs.8/oo.33/chromFa.zip</a>
<h4>mouse sequence from phusion.3:</h4>
   <a
   href="ftp://ftp.ncbi.nlm.nih.gov/pub/TraceDB/mus_musculus/ClipReads/Assemblies/Sanger_Oct15/">
   ftp://ftp.ncbi.nlm.nih.gov/pub/TraceDB/mus_musculus/ClipReads/Assemblies/Sanger_Oct15/</a>
<h4>repeats from:</h4>
   <a href="http://genome.ucsc.edu/goldenPath/06aug2001/database/">
   http://genome.ucsc.edu/goldenPath/06aug2001/database/</a><br>
    (chrN_rmsk.txt.gz for chromosome N)
<br>
<hr>
<br>
Alignments contributed by: Scott Schwartz<br>
See <a href="http://bio.cse.psu.edu/genome/hummus/2001-12-16/aar/README">http://bio.cse.psu.edu/genome/hummus/2001-12-16/aar/README</a>
