<H2>Description</H2>
<P>This track displays <EM>blastz</EM> alignments of the Feb. 2002 mouse draft assembly to the human genome filtered so that only the best alignment for any given region of the human genome is displayed.
<br>
<H2>Methods</H2>
For <em>blastz</em> use 12 of 19 seeds and then score using 
<br>
<table>
<tr><td>&nbsp;</td><td>A</td><td>C</td><td>G</td><td>T</td></tr>
<tr><td>A</td><td>&nbsp;&nbsp;91</td><td>-114</td><td>&nbsp;-31</td><td>-123</td></tr>
<tr><td>C</td><td>-114</td><td>&nbsp;100</td><td>-125</td><td>&nbsp;-31</td></tr>
<tr><td>G</td><td>&nbsp;-31</td><td>-125</td><td>&nbsp;100</td><td>-114</td></tr>
<tr><td>T</td><td>-123</td><td>&nbsp;-31</td><td>-114</td><td>&nbsp;91</td></tr>
</table>
<br>
  O = 400, E = 30, K = 3000, L = 3000, M = 50

However, we now do a second pass at reduced stringency (7mer seeds and
MSP threshold of K=2200) to attempt to fill in gaps (up to about 10Kbp).
Also, lineage specific repeats are abridged during this alignment.



<br>
<H2>Credits</H2>
Alignments contributed by Scott Schwartz, Penn State Bioinformatics Group.
Best in genome filtering by UCSC's axtBest program.
