<H2>Description</H2>
<P>Reference alignment of mouse genome vs. the human genome
constructed from the <EM>Mouse Translated Blat Alignments</EM>
track.
</P>
<H2>Methods</H2>
<P>The track is constructed by selecting the
best-scoring BLAT alignment for each aligned position in the genome.
Mouse insertions are included in the reference alignment when 
they follow the last base of a BLAT block, although this process
is not entirely perfected.
</P>
<H2>Data Release Policy</H2>
<P>This data is made available before scientific publication with the 
following understanding:
<OL>
<LI>The data may be freely downloaded, used in analyses, and 
repackaged in databases.
<LI>Users are free to use the data in scientific papers analyzing 
particular genes and regions if the providers of this data 
(the Mouse Sequencing Consortium) are properly acknowledged.
<LI>The Centers producing the data reserve the right to publish the initial 
large-scale analyses of the dataset-including large-scale identification 
of regions of evolutionary conservation and large-scale 
genomic assembly. Large-scale refers to regions with size on the order of 
a chromosome (that is, 30 Mb or more). 
<LI>Any redistribution of the data should carry this notice.</OL>
</P>
