<H2>Description</H2>
<P>The Spliced EST track shows Expressed Sequence Tags 
(ESTs) from Genbank that show signs of splicing when
aligned against the genome.  Requiring splicing reduces
the level of contamination in the EST databases drastically
at the expense of eliminating many genuine 3' ESTs.
Please see also the $Organism EST track.</P>

<P>Expressed sequence tags are single read (typically approximately
500 base) sequences which usually represent fragments of transcribed
genes.  Aligning regions (exons usually) are shown as black boxes
connected by lines for gap (spliced out introns usually). In full
display, arrows on the introns indicate the direction of
transcription. For the Dec 22, 2001 assembly and on the direction is
determined by using the GT-AG splice donor/acceptor rule, otherwise
direction is determined by the Genbank record.</P>

<H2>Method</H2>
<P>To make an EST, RNA is isolated from cells and reverse
transcribed into cDNA.  The cDNA is typically cloned
into a plasmid vector, and a read taken from the 5'
and/or 3' primer.  For most but not all ESTs the
reverse transcription is primed by an oligo-dT, which
hybridizes with the poly-A tail of mature mRNA. The
reverse transcriptase may or may not make it to the 5'
end of the mRNA,  which may or may not be degraded.
In general the 3' ESTs mark the end of transcription
reasonably well, but the 5' ESTs may end at any point
within the transcript.  Some of the newer cap-selected
libraries are starting to hit transcription start
reasonably well however. Before the cap-selection techniques
emerged, some projects used random rather than poly-A
priming in an attempt to get sequence distant from the
3' end.  These projects were successful at this, but as
a side effect also deposited sequences from unprocessed
mRNA and perhaps even genomic sequences into the EST databases.
(Even outside of the random-primed projects there is a
degree of non-mRNA contamination.) Because of this a
single unspliced EST should be viewed with considerable
skepticism.  However since the $organism 3' UTRs are quite
long requiring splicing does eliminate many genuine 3'
ESTs.</P>

<P>To generate this track $organism ESTs from Genbank are aligned 
against the genome using the BLAT program. When a single 
EST aligns in multiple places the alignment having the 
highest base identity is found.  Only alignments that have 
a base identity level within 1% of the best are kept. 
Alignments must also have at least 93% base identity to be kept.</P>

<H2>Using the Filter</H2>
<P>The track filter can be used to color, include, or exclude a subset of individual 
items within a track. This is helpful when many items are shown in the track
display, especially when only some are relevant to the current task. To use the
filter:
<OL>
<LI>Enter a value in one or more of the text boxes to filter the EST display. For
example, to apply the filter to all ESTs expressed in the liver, type "liver" in the 
tissue box. For a list of permissible filter values, consult the non-positional table in
the Table Browser that corresponds to the factor on which you wish to filter. For
example, the non-positional table "tissue" contains all of the types of tissues
that can be entered into the tissue text box. Wildcards can also be used in the
filter.
<LI>If filtering on more than one value, choose the desired combination
logic. If "and" is selected, only ESTs that match all of the filter criteria will
be highlighted. If "or" is selected, ESTs that match any 1 of the filter criteria
will be highlighted.
<LI>Choose the color or display characteristic that will be used to highlight or
include/exclude the filtered items. If "exclude" is chosen, the browser will not 
display ESTs that match the filter criteria. If "include" is selected, the browser 
will display only those ESTs that match the filter criteria.
<LI>Click the Submit button.
</OL></P>

<H2>Credits</H2>
The Spliced EST track is produced at UCSC from EST sequence data
submitted to the international public sequence databases by 
scientists worldwide.
