<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
</head>
<body>
The <span style="font-style: italic;">Pyrococcus abyssi GE5</span>
genome sequence is 1,765,118 bp long and contains approximately 1896
protein-coding genes, plus a 3,444 bp plasmid.&nbsp; <span style="font-style: italic;">P.
abyssi</span> is a hyperthermophilic archaeon isolated from samples taken close to a hot spring situated 3500 meters deep in the south-east Pacific.  The optimal growth conditions are 103 C at 200 atmospheres pressure.
&nbsp; 
<P>
The sequence was release in July 1999 by Genoscope, a collaboration among four teams led by
Daniel Prieur (CNRS, Roscoff, France), Patrick Forterre (Universiti d'Orsay, France), Jean-Claude Thierry (IGBMC, Strasbourg, France), and Jacques Dietrich (IFREMER, Brest,France).
The original source for the sequence and re-annotation efforts by the authors are located at the <a
 href="http://www.genoscope.cns.fr/Pab">Genoscope <I>P. abyssi</I> web site</a>.
<h3>Sample position queries</h3>
<p>
A genome position can be specified by chromosomal coordinate range, COG
ID, or keywords from the GenBank or TIGR description of an gene. The
following list shows examples of valid position queries for this
genome.&nbsp;
</p>
<p>
<table cellspacing="0" cellpadding="0" border="0">
  <tbody>
    <tr>
      <td nowrap="nowrap" valign="top"><b>Request:</b><br>
      </td>
      <td colspan="2" valign="top"><b>&nbsp;&nbsp; Genome Browser
Response:</b><br>
      </td>
    </tr>
    <tr>
      <td valign="top"><br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">chr</td>
      <td width="14"><br>
      </td>
      <td valign="top">Displays all of chromosome (the entire genome).<br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">chr:1-10000</td>
      <td width="14"><br>
      </td>
      <td valign="top">Displays first ten thousand bases on the
chromosome.<br>
      </td>
    </tr>
<!-- 
    <TR><TD VALIGN=Top NOWRAP>ctgchr7_ctg</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of the clone contig ctgchr7_ctg 
                       (set "Map contigs" track to "dense" and refresh to see contigs)</TD>
    </TR> 
-->
    <tr>
      <td nowrap="nowrap" valign="top">NADH</td>
      <td width="14"><br>
      </td>
      <td valign="top">Lists genes with "NADH" in the description.<br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">PAB1011</td>
      <td width="14"><br>
      </td>
      <td valign="top">Display chromosome at position of gene <a
 href="http://www.ncbi.nlm.nih.gov/entrez/viewer.fcgi?db=protein&val=5458949">PAB1011</a>.<br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">PAB101</td>
      <td width="14"><br>
      </td>
      <td valign="top">Lists genes with names beginning with "PAB101".<br>
      </td>
    </tr>
    <tr>
    </tr>
  </tbody>
</table>
</p>
<P>
<HR>
<H3>Credits</H3>
The Archaeal Genome Browsers at UCSC were developed by members of the
<a href="http://lowelab.ucsc.edu/">Lowe Lab</a> (Kevin Schneider,
Katherine Pollard, Andy Pohl, Todd Lowe) with significant support from
the <a href="http://genome.ucsc.edu/staff.html">UCSC Human Genome
Browser group</a> (especially Robert Baertsch & Hiram Clawson).
The Archaeal Browsers are run by a slightly modified
version of the <a href="http://genome.ucsc.edu/">UCSC Human Genome
Browser</a> system.  All queries, bug reports, content corrections, suggested improvements,
and new track data submissions should be sent to Todd Lowe (lowe
@soe.ucsc.edu).

<P>If you use the browser for your research and wish to cite it, a
paper describing the work has been submitted to the 2006 <I>Nucleic
Acids Research</I>
Database Issue (KL Schneider, KS Pollard, A Pohl, TM Lowe,
submitted).  Citations will help us obtain funding to continue
development of this community resource.
</body>
</html>
