<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
</head>
<body>
The <I>Thermoplasma volcanium GSS1</I> genome is 1.58 Million bp long and contains approximately 1548
predicted genes. <I>T. volcanium</I> is a thermophilic, acidophilic mycoplasma that grows optimally at 60C and pH 2.0. This species is a facultative aerobe.  The thermoplasamas are euryarchaea, distantly related to sulfolobes which are crenarchaea, yet both inhabit similar thermal acidic environments.<BR><BR>
The sequence was released Dec 2004 by National Institute of Advanced Industrial Science and Technology, Japan, and was described
in <I>Proc Natl Acad Sci U S A</I> <B>97</B>:14257-62 (2000) Kawashima T, Amano N, Koike H, Makino S, Higuchi S, <I>et al.</I> &nbsp;"<a
 href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=11121031">Archaeal adaptation to higher temperatures revealed by genomic sequence of Thermoplasma volcanium.
</a>"
<blockquote><B>Abstract:</B> The complete genomic sequence of the archaeon Thermoplasma volcanium, possessing optimum growth temperature (OGT) of 60 degrees C, is reported. By systematically comparing this genomic sequence with the other known genomic sequences of archaea, all possessing higher OGT, a number of strong correlations have been identified between characteristics of genomic organization and the OGT. With increasing OGT, in the genomic DNA, frequency of clustering purines and pyrimidines into separate dinucleotides rises (e.g., by often forming AA and TT, whereas avoiding TA and AT). Proteins coded in a genome are divided into two distinct subpopulations possessing isoelectric points in different ranges (i.e., acidic and basic), and with increasing OGT the size of the basic subpopulation becomes larger. At ... [Click above reference link for full abstract]
</blockquote>
<P>
<h3>Sample position queries</h3>
<p>
A genome position can be specified by chromosomal coordinate range, COG
ID, or keywords from the GenBank or TIGR description of a gene. 
The available chromosome/plasmid names are:<BR><BR>
<table border="2" cellspacing=2 cellpadding=2>
<tr><th>Browser Chrom/Plasmid Name</th><th>Length (bp)</th><th>Gene Count</ht><th>NCBI RefSeq Accession</th></tr>
<tr><td>chr</td><td align=right>1584804</td><td align=right>1548</td><td align=center>NC_002689</td></tr>
</table>

<p>
The following list shows examples of valid position queries for this
genome:&nbsp;
</p>
<p>
<table cellspacing="1" cellpadding="1" border="0">
<tr><th align=left>Request:</th><th align=left>Genome Browser Response:</th></tr>
<tr><td>chr</td><td>Displays the entire sequence "chr" in the browser window</td></tr>
<tr><td>chr:1-10000</td><td>Displays first ten thousand bases of the sequence "chr"</td></tr>
<tr><td>transporter&nbsp;&nbsp;&nbsp;&nbsp;</td><td>Lists all genes with "transporter" in the name or description</td></tr>
<tr><td>TVN0010</td><td>Display genome at position of gene TVN0010</td></tr>
</tbody>
</table>
<P>
<HR>
<H3>Credits</H3>
The Archaeal Genome Browsers at UCSC were developed by members of the
<a href="http://lowelab.ucsc.edu/">Lowe Lab</a> (Kevin Schneider,
Katherine Pollard, Andy Pohl, Todd Lowe) and Robert Baertsch, with significant support from
the <a href="http://genome.ucsc.edu/staff.html">UCSC Human Genome
Browser group</a>.
The Archaeal Browsers are run by a slightly modified
version of the <a href="http://genome.ucsc.edu/">UCSC Human Genome
Browser</a> system.  All queries, bug reports, content corrections, suggested improvements,
and new track data submissions should be sent to Todd Lowe (lowe
@soe.ucsc.edu).

<P>If you use the browser for your research and wish to cite it, a
paper describing the work has been accepted to the 2006 <I>Nucleic
Acids Research</I>
<a href="http://www3.oup.co.uk/nar/database/a/">Database Issue</a> (KL Schneider, KS Pollard, R Baertsch, A Pohl, TM Lowe). 
Citations will help us obtain funding to continue
development of this community resource.

</body>
</html>
