<H2>Description</H2>

<P>
This track shows the single nucleotide polymorphisms (SNPs) and/or
insertions/deletions (indels) determined by the 
<A HREF="http://www.genomics.org.cn/bgi_new/index.htm" TARGET=_BLANK>
Beijing Genomics Institute</A> (BGI).  
Genomic sequence reads from 3 alternate strains of chicken (Broiler,
Layer and Silkie) were used to determine the SNPs relative to the
reference assembly for Red Jungle Fowl (RJF). SNPs and indels are
associated with genes (annotations by BGI until Ensembl annotations
are ready), and information on the association can be displayed along
with a link to the details page for the gene.
</P>
<P>
SNP names are of the format 
snp.<EM>Supercontig</EM>.<EM>Contig</EM>.<EM>Position</EM>.<EM>Type</EM>.<EM>Strain</EM>:
</P>
<UL>
<LI><EM>Supercontig</EM>: Supercontig number from assembly</LI>
<LI><EM>Contig</EM>: Contig number (within supercontig) from assembly</LI>
<LI><EM>Position</EM>: Starting offset within 
    Contig.<EM>Supercontig</EM>.<EM>Contig</EM></LI>
<LI><EM>Type</EM>: S=Substitution, I=Insertion or D=Deletion</LI>
<LI><EM>Strain</EM>: 1=Broiler, 2=Layer, 3=Silkie</LI>
</UL>

<H2>Methods</H2>

<P>
Sequence reads from the Broiler, Layer and Silkie strains were mapped
to the reference Chicken assembly with blast.  Then cross_match was
used for the detailed base-by-base comparison.  The Phred quality
scores (-10 * log_10 estimated error probability) are given for the 
SNP base(s) in the assembly and the (single read) SNP read.  
</P>


<H2>Credits</H2>

<P>Thanks to the Beijing Genomics Institute for providing these data.
</P>

<H2>Terms of Use</H2>

<OL>
<LI>
Users are free to use this data in scientific papers analyzing
particular genes, if the Beijing Genomics Institute (BGI) is
acknowledged.
</LI>
<LI>
BGI and its collaborators reserve the right to publish the initial
analyses of this data, including but not restricted to the large-scale
identification of functional polymorphisms, evolutionary patterns and
signs of selection, correlations with known QTLs, utility of data for
genetic mapping purposes, etc.
</LI>
<LI>
Any redistribution of this data should carry this notice.
</LI>
</OL>

