<P>
The February 2004 <em>Gallus gallus</em> draft sequence 
was produced by the Genome Sequencing Center at the Washington University 
School of Medicine in St. Louis.
The genome has been sequenced to 6.63X coverage.
Approximately 88% of the sequence has been anchored to chromosomes, 
which include autosomes 1-24, 26-28, and 32, and sex chromosomes W and Z. 
(In contrast to mammals, the female chicken is heterogametic (ZW) and the 
male is homogametic (ZZ).) 
The remaining unanchored contigs have been concatenated into the virtual 
chromosome &quot;chrUn&quot;, separated by gaps of 10,000 bp. The chicken 
mitochondrial sequence is also available as the virtual chromosome 
&quot;chrM&quot;. 
<P>
WARNING: This release of the G. gallus chrW contains some sequence
that is not specific to chromosome W.  A large portion of the
chromosome was assigned to W based on the presence of W-specific
repeats which, as it has now been shown, are not W-specific.  Thus, the
only portions of chrW which should currently be considered specific to
W are:
<UL>
               <LI>chrW:1-195,831</LI>
               <LI>chrW:4,895,452-4,916,845</LI>
               <LI>chrW_random (all)</LI>
</UL>
<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/galGal2/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#chicken">Downloads</A> page. These data
have <A HREF="/goldenPath/credits.html#chicken_use">specific conditions for use</A>. 
The chicken browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#chicken_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic supercontig, an mRNA or EST, a chromosomal coordinate range, or keywords from 
the GenBank description of an mRNA. The following list shows 
examples of valid position queries for the chicken genome.
See the
<A HREF="http://genome.cse.ucsc.edu/goldenPath/help/hgTracksHelp.html" TARGET=_blank>
User's Guide</A> for more information.
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
       	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><br></TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>chrZ</TD>
        <TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays all of chromosome Z</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr3:1-1000000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays first million bases of chr 3</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Contig44</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of assembly for the supercontig with ID contig44</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Contig44.48</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of assembly for the contig with ID contig44.48</TD>
    </TR>
    <TR><TD VALIGN=Top><br></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>BX950328</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession number BX950328</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>BU335953</TD>
       	<TD WIDTH=14></TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession BU335953 on chr Z</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr13_3.145</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of genome containing Genscan gene prediction chr13_3.145</TD>
    </TR> 
    <TR><TD VALIGN=Top NOWRAP>CWNT-8C</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of genome with gene symbol CWNT-8C</TD>
    </TR>
    <TR>
	<TD VALIGN=Top NOWRAP>NM_204735</TD>
	<TD WIDTH=14</TD>
	<TD VALIGN=Top>Displays the region of genome with RefSeq identifier NM_204735</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>374180</TD>
       	<TD WIDTH=14></TD>
	<TD VALIGN=Top NOWRAP>Displays the region of genome with LocusLink identifier 374180</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>CWNT-8C</TD>
       	<TD WIDTH=14></TD>
       	<TD>Displays region of genome with gene symbol CWNT-8C</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>ENSGALT00000023622</TD>
       	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays the region of genome with Ensembl identifier 
	ENSGALT00000023622. NOTE: Due to space constraints, the Genome Browser 
	displays abridged Ensembl identifiers, but the full ID name must be used in 
	searches. For example, the ENSGALT00000023622 identifier displays as 
	ENSGAL23622 in the Genome Browser.</TD>
    </TR>
    <TR><TD VALIGN=Top><br></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>pseudogene mRNA</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes, but not cDNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>homeobox caudal</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>kruppel zinc finger</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>zhang</TD>
       	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Zhang</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Taguchi,T.</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author T. Taguchi</TD>
    </TR>
    <TR><TD VALIGN=Top><br></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Taguchi T</em>, internally it uses 
			the format <em>Taguchi,T.</em>.</TD>
    </TR>
</TABLE>
