<!-- Display image at top right of page -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=137>
  <TR><TD ALIGN=RIGHT><A HREF="http://www.genome.gov/11007323" TARGET=_blank>
    <IMG SRC="/images/Canis_familiaris.jpg" WIDTH=122 HEIGHT=160 ALT="Dog"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1> <em>Canis familiaris</em><BR>
    </FONT>
    <FONT SIZE=-2> (NHGRI <A HREF="http://www.genome.gov/10005141" 
      TARGET=_blank>Press Photos</A>) 
    </FONT>
  </TD></TR>
</TABLE>

<P>
The May 2005 dog (<em>Canis familiaris</em>) whole genome shotgun
(WGS) assembly v2.0 was sequenced and assembled by the
<A HREF="http://www.broad.mit.edu/seq/" TARGET=_blank>
Broad Institute</A> of MIT/Harvard and 
<A HREF="http://www.agencourt.com/" TARGET=_blank>Agencourt Bioscience</A>. 

<H3>Sample Position Queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic clone, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA.
The following list provides examples 
of various types of position queries for the dog genome. See the 
<A HREF="../goldenPath/help/hgTracksHelp.html" 
TARGET=_blank>User's Guide</A> for more information. 
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top><B>Genome Browser Response:</B><BR></TD>
    </TR>
	
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>chr16</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays all of chromosome 16</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>chr16:1-5000000</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays first 5 million bases of chr 16</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>AY572227</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays location of mRNA with GenBank accession AY572227</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>BM538765</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays location of EST with GenBank accession BM538765</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr6_8.29</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays location of Genscan gene prediction with identifier chr6_8.29</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>pseudogene mRNA</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists transcribed pseudogenes but not cDNAs</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists zinc finger mRNAs</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>breast cancer</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs associated with breast cancer susceptibility proteins</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>evans</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Evans</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Murphy,K.E.</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by co-author K.E. Murphy</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Murphy KE</em>, internally
	it uses the format <em>Murphy,K.E.</em>.</TD>
    </TR>
</TABLE>

<H3>Assembly Details</H3>
<P>
The dog genome contains approximately 2.5 billion base pairs. 
This sequence is based on 7.6X coverage of the dog genome, assuming a WGS
assembly size of 2.4 Gb. 
The boxer breed was selected for the initial sequencing effort based on the
lower variation rate in its genome relative to other breeds. For more
information about this assembly, see the Broad Institute
<A HREF="http://www.broad.mit.edu/mammals/dog/" TARGET=_blank>Dog Genome
Sequencing Project</A> web page.
</P>
<P>
The canFam2 sequence and annotation data can be downloaded from the 
UCSC Genome Browser 
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/canFam2/">FTP server</A>
or 
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#dog">downloads</A> 
page. These data have <A HREF="../goldenPath/credits.html#dog_use">specific 
conditions for use</A>.
</P>
<P>
Many thanks to the Broad Institute of MIT/Harvard, NHGRI, and the many
institutions who contributed to the sequencing and mapping effort for this
release. The canFam2 annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#dog_credits">credits</A> page
for a detailed list of the organizations and individuals who contributed 
to this release. 
</P>
