<H2>Description</H2>
<P>
This track displays <A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>blastz</A> alignments of the Nov. 2002 rat assembly to the mouse genome.
</P>
<H2>Methods</H2>
<P>
For blastz, we use 12 of 19 seeds and then score using: 
<PRE>
      A     C     G     T
A    91  -114   -31  -123
C  -114   100  -125   -31
G   -31  -125   100  -114
T  -123   -31  -114    91

O = 400, E = 30, K = 3000, L = 3000, M = 50
</PRE>
</P><P>
A second pass is done at reduced stringency (7mer seeds and
MSP threshold of K=2200) to attempt to fill in gaps of up to about 10Kbp.
Lineage specific repeats are abridged during this alignment.
</P>
<H2>Using the Filter</H2>
<P>The track filter can be used to turn on the chromosome color track or to 
filter the display output by chromosome.
<UL>
<LI><B>Color track</B>: To display the chromosome color track, click the <em>on</em>
button next to &quot;Color track based on chromosome&quot;. When the color track is 
activated, each of the items within the annotation track will be colored to 
show the chromosome in the genome assembly from which the alignment 
originated.
<LI><B>Chromosome filter</B>: To display only alignments from a specific 
chromomsome in the aligning assembly, enter the chromosome number (in the form
chrN) in the text box to the right of &quot;Filter by chromosome&quot;. For example, to
display alignments from chromosome 6, type &quot;chr6&quot;.
</UL>
When you have finished configuring the filter, click the <em>Submit button</em>.
</P>
<H2>Credits</H2>
The alignments are contributed by Scott Schwartz of the 
<A HREF="http://bio.cse.psu.edu/" TARGET=_blank>
Penn State Bioinformatics Group</A>. Mouse sequence data is provided by the 
Mouse Genome Sequencing Consortium. 
