<H2>Description</H2>
This track shows <em>D. pseudoobscura</em>/<em>$organism</em> genomic 
alignments using
a gap scoring system that allows longer gaps than traditional
affine gap scoring systems.  It can also tolerate gaps
in both organisms simultaneously.  These &quot;double-sided&quot;
gaps can be caused by local inversions and overlapping deletions
in both species. The <em>D. pseudoobscura</em> sequence is from the 
<A HREF="http://www.hgsc.bcm.tmc.edu/projects/drosophila/" TARGET=_BLANK>
Baylor HGSC August 2003 Freeze 1 assembly</A>.
<P>
The chain track displays boxes joined together by either single or 
double lines. The boxes represent aligning regions. 
Single lines indicate gaps that are largely due to a deletion in the genome of 
<em>D. pseudoobscura</em> or an insertion in the <em>$organism</em> assembly.
Double lines represent more complex gaps that involve substantial
sequence in both species. This may result from inversions, overlapping
deletions, an abundance of local mutation, or an unsequenced gap in one species.
In cases where there are multiple 
chains over a particular portion of the <em>$organism</em> genome, chains with 
single-lined gaps are often due to processed pseudogenes, while chains 
with double-lined gaps are more often due to paralogs and unprocessed 
pseudogenes.
<P>
The display indicates the chromosome, strand, and location of the 
match for each matching alignment (in thousands). 

<H2>Methods</H2>
The <em>D. pseudoobscura</em>/<em>$organism</em> 
genomes were aligned with 
<A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>blastz</A>.  
and converted into axt format with <em>lavToAxt</em>.
The resulting axts were fed into <em>axtChain</em>.  AxtChain organizes all 
the alignments between a single <em>D. pseudoobscura</em> and a single 
<em>$organism</em> chromosome into a group, and makes a kd-tree out of 
all the gapless subsections
(blocks) of the alignments.  Next, maximally scoring chains of these
blocks are found by running a dynamic program over the kd-tree.  Chains
scoring below a threshold are discarded, and the remaining chains are
displayed here.

<H2>Credits</H2>
<P>Blastz was developed at 
<A HREF="http://www.bx.psu.edu" TARGET=_blank>Pennsylvania State University</A>
by Scott Schwartz, Zheng Zhang, and Webb Miller with advice from
Ross Hardison.</P>

<P>The axtChain program was developed at the University of California
at Santa Cruz by Jim Kent with advice from Webb Miller and David Haussler.</P>

<P>The browser display and database storage of the chains were made
by Robert Baertsch and Jim Kent.</P>

<H2>References</H2>
<P>
Chiaromonte F, Yap VB, Miller W (2002). 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" TARGET=_blank>Scoring pairwise genomic sequence alignments</A>. 
<em>Pac Symp Biocomput</em> 2002;:115-26.</P>
<P>
Kent WJ, Baertsch R, Hinrichs A, Miller W, and Haussler D (2003).
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1" 
TARGET=_blank>Evolution's cauldron: 
Duplication, deletion, and rearrangement in the mouse and human genomes</A>. 
<I>Proc Natl Acad Sci USA</I> 100(20):11484-11489 Sep 30 2003.
<P>
Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison R, Haussler D, and 
Miller W (2003). 
<A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>Human-Mouse 
Alignments with BLASTZ</A>. <em>Genome Res.</em> 13(1):103-7.</P>
