<H2>Description</H2>
<P>
This track shows <em>A. gambiae</em>/<em>D. melanogaster</em> genomic 
alignments using a gap scoring system that allows longer gaps than 
traditional affine gap scoring systems. It can also tolerate gaps
in both organisms simultaneously. These &quot;double-sided&quot;
gaps can be caused by local inversions and overlapping deletions
in both species. The <em>A. gambiae</em> sequence is from the 
<A HREF="http://www.ensembl.org/Anopheles_gambiae/" TARGET=_BLANK>
MOZ2 assembly</A>.</P>
<P>
The chain track displays boxes joined together by either single or 
double lines. The boxes represent aligning regions.  Single lines indicate 
gaps that are largely due to a deletion in the <em>$o_Organism</em> assembly
or an insertion in the <em>$Organism</em> assembly.
Double lines represent more complex gaps that involve substantial
sequence in both species. This may result from inversions, overlapping
deletions, an abundance of local mutation, or an unsequenced gap in one 
species.  In cases where there are multiple chains over a particular portion
of the <em>$Organism</em> genome, chains with 
single-lined gaps are often due to processed pseudogenes, while chains 
with double-lined gaps are more often due to paralogs and unprocessed 
pseudogenes.</P>
<P>
The display indicates the chromosome, strand, and location of the 
match for each matching alignment (in thousands). </P>

<H2>Methods</H2>
<P>
The <em>A. gambiae</em>/<em>D. melanogaster</em> genomes were aligned with 
blastz and converted into axt format with <em>lavToAxt</em>.
The resulting axts were fed into <em>axtChain</em>.  AxtChain organizes all 
the alignments between a single <em>A. gambiae</em> and a single 
<em>D. melanogaster</em> chromosome into a group and creates a kd-tree out 
of all the gapless subsections (blocks) of the alignments.  Next, maximally 
scoring chains of these blocks were found by running a dynamic program over 
the kd-tree. Chains scoring below a threshold were discarded, and the 
remaining chains are displayed here.</P>

<H2>Credits</H2>
<P>
Blastz was developed at <A HREF="http://bio.cse.psu.edu" 
TARGET=_blank>Pennsylvania State University</A> by 
Scott Schwartz, Zheng Zhang, and Webb Miller with advice from
Ross Hardison.</P>
<P>
Lineage-specific repeats were identified by Arian Smit and his program 
<A HREF="http://www.repeatmasker.org" TARGET=_blank>RepeatMasker</A>.</P>
<P>
The axtChain program was developed at the University of California at 
Santa Cruz by Jim Kent with advice from Webb Miller and David Haussler.</P>
<P>
The browser display and database storage of the chains were generated
by Robert Baertsch and Jim Kent.</P>

<H2>References</H2>
<P>
Chiaromonte, F., Yap, V.B., Miller, W. 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" 
TARGET=_blank>Scoring pairwise genomic sequence alignments</A>. 
<em>Pac Symp Biocomput 2002</em>, 115-26 (2002).</P>
<P>
Kent, W.J., Baertsch, R., Hinrichs, A., Miller, W., and Haussler, D.
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1" 
TARGET=_blank>Evolution's cauldron: Duplication, deletion, and rearrangement
in the mouse and human genomes</A>.
<em>Proc Natl Acad Sci USA</em> <B>100</B>(20), 11484-11489 (2003).</P>
<P>
Schwartz, S., Kent, W.J., Smit, A., Zhang, Z., Baertsch, R., Hardison, R., 
Haussler, D., and Miller, W.
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103" 
TARGET=_blank>Human-Mouse Alignments with BLASTZ</A>. 
<em>Genome Res.</em> <B>13</B>(1), 103-7 (2003).</P>
