<H2>Description</H2> <P>
This track shows a measure of evolutionary conservation in 
three Drosophila species and mosquito (A. gambiae), 
based on a phylogenetic hidden Markov model (phastCons).
Multiz alignments of the following assemblies were used to generate this
annotation: 
<em>$organism</em> $date ($db), 
<em>D. yakuba</em> Apr. 2004 (droYak1),
<em>D. pseudoobscura</em> Aug. 2003 (dp2),
and <EM>A. gambiae</EM> Sep. 2003 (anoGam1).  
<P>
In full display mode, this track shows the overall conservation score 
across all species as well as pairwise alignments of 
<em>D. yakuba</em>, <em>D. pseudoobscura</em>, and <em>A. gambiae</em>,
each aligned to the <EM>$organism </EM> genome.  The pairwise alignments are
shown in dense display mode using a grayscale density gradient.  
The checkboxes in the track configuration section allow
the exclusion of species from the pairwise display; however, this does not
remove them from the conservation score display.  </P>
<P>
When zoomed-in to the base-display level, the track shows the base 
composition of each alignment. The numbers and symbols on the &quot;$organism 
gap&quot; line indicate the lengths of gaps in the <em>$organism</em>
sequence at those alignment positions, relative to the longest non-<EM>$organism</EM>
sequence at that spot. If the gap size is greater than
9, the &quot;+&quot; symbol is displayed. To view detailed information
about the alignments at a specific position, zoom in the display to
30,000 or fewer bases, then click on the alignment.  </P>
<P>
This track may be configured in a variety of ways to highlight
different aspects of the displayed information. Click the &quot;Graph
configuration help&quot; link for an explanation of the configuration
options.</P>

<H2>Methods</H2>
<P> 
Best-in-genome blastz pairwise alignments 
were multiply aligned using multiz, beginning with 
<em>$organism</em>-<em>D. yakuba</em> alignments
and subsequently adding in <em>D. pseudoobscura</em> and <em>A. gambiae</em>.  
The resulting multiple alignments were then assigned 
conservation scores by phastCons.</P>
<P>
<P>PhastCons constructs a two-state phylo-HMM with a state for conserved
regions and a state for non-conserved regions.  The two states share a
single phylogenetic model, except that the branch lengths of the tree
associated with the conserved state are multiplied by a constant scaling
factor &rho (0 &le &rho &le 1).  The phylogenetic model and the scaling
factor &rho are estimated from the data by maximum likelihood using an EM
algorithm.  The state-transition probabilities of the phylo-HMM may also be
estimated by maximum likelihood, but we find that more biologically useful
results are obtained by treating them as tuning parameters.  In this case,
the transition probabilities were set such that the "coverage" of the track
(portion of sites in conserved elements) is approximately 30% and the prior
expected length of a conserved element is 25 bp.  To ensure that the scores
in different regions of the genome are directly comparable, we first obtain
global estimates of all parameters, by averaging estimates obtained from
different regions of the genome, and then, in a second pass, we produce
predictions across the genome using the same parameters everywhere.  
The value plotted at each site is the posterior probability that the
corresponding alignment column was "generated" by the conserved state of
the phylo-HMM.  These scores reflect the phylogeny (including branch
lengths) of the species in question, a continuous-time Markov model of the
nucleotide substitution process, and a tendency for conservation levels to
be autocorrelated along the genome (i.e., to be similar at adjacent sites).  
The general reversible (REV) substitution model was used, and variation of
substitution rates within each major category of sites (conserved and
non-conserved) was allowed for using Yang's (1994) discrete gamma model.
Note that, unlike many conservation-scoring programs, phastCons does not
rely on a sliding window of fixed size, so short highly-conserved regions
and long moderately conserved regions can both obtain high scores.</P>
<P> 
PhastCons currently treats alignment gaps as missing data, which
sometimes has the effect of producing undesirably high conservation scores
in gappy regions of the alignment.  We are looking at several possible ways
of improving the handling of alignment gaps.</P>

<H2>Credits</H2>
<P>
This track was created at UCSC using the following programs:
<UL>
<LI>
Blastz and multiz from Scott Schwartz and Webb Miller of the 
<A HREF="http://www.bx.psu.edu/" TARGET=_blank>Penn State Bioinformatics 
Group</A>. 
<LI>
AxtBest, axtChain, chainNet, netSyntenic, and netClass 
developed by Jim Kent at UCSC. 
<LI>Tree estimation (phyloFit) and conservation-scoring (phastCons)
software by Adam Siepel at UCSC. 
<LI>"Wiggle track" plotting software by Hiram Clawson at UCSC.
</UL>
</P>

<H2>References</H2>

<H3>Phylo-HMM</H3>
<P>
Felsenstein J and Churchill GA (1996).  A hidden Markov model approach to
variation among sites in rate of evolution.
<em>Mol Biol Evol</em> 13:93-104. </P>
<P>
Siepel A and Haussler D (2003).  
<a href="http://www.cse.ucsc.edu/~acs/phyhmm.pdf">Combining
phylogenetic and hidden Markov models in biosequence analysis</a>.
In <em> Proceedings of the Seventh Annual 
International Conference on Computational Molecular Biology (RECOMB
2003)</em>, pp. 277-286. </P>
<P>
Siepel A and Haussler D (2004). Phylogenetic hidden Markov models.
In R. Nielsen, ed.,  Statistical Methods in Molecular Evolution, Springer  
(in press). </P>
<P>
Yang Z (1994).  Maximum likelihood phylogenetic estimation from DNA sequences
with variable rates over sites: approximate methods,
<em>J Mol Evol</em> 39:306-314.</P>
<P>
Yang Z (1995).  A space-time process model for the evolution of DNA
sequences.  <em>Genetics</em>, 139:993-1005. </P>

<H3>Chain/Net</H3>
<P>
Kent WJ, Baertsch R, Hinrichs A, Miller W, and Haussler D (2003). 
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1" 
TARGET=_blank>Evolution's cauldron: 
Duplication, deletion, and rearrangement in the mouse and human genomes</A>. 
<I>Proc Natl Acad Sci USA</I> 100(20):11484-11489 Sep 30 2003.

<H3>Multiz</H3>
<P>
Blanchette M, Kent WJ, Riemer C, Elnitski L, Smit AFA, Roskin KM, Baertsch R,
Rosenbloom K, Clawson H, Green ED, Haussler D, Miller W. (2004). 
<A HREF="http://www.genome.org/cgi/content/abstract/14/4/708" TARGET=_blank>Aligning Multiple Genomic Sequences with the Threaded Blockset Aligner</A>.
<em>Genome Res.</em> 14(4):708-15.

<H3>Blastz</H3>
<P>
Chiaromonte F, Yap VB, and Miller W (2002). 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" 
TARGET=_blank>Scoring pairwise genomic sequence alignments</A>. 
<em>Pac Symp Biocomput</em> 2002;:115-26.</P>
<P>
Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison R, Haussler D, and 
Miller W. (2003). 
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103" 
TARGET=_blank>Human-Mouse Alignments with BLASTZ</A>. 
<em>Genome Res.</em> 13(1):103-7.</P>
