<H2>Description</H2> 
<P> 
This track shows predictions of conserved elements produced by the phastCons
program.  They are based on a phylogenetic hidden Markov model
(phylo-HMM), a type of probabilistic model that describes both the process
of DNA substitution at each site in a genome and the way this process
changes from one site to the next. </P>

<H2>Methods</H2>
<P> 
Best-in-genome blastz pairwise alignments 
were multiply aligned using multiz, beginning with 
<em>$organism</em>-<em>D. yakuba</em> alignments
and subsequently adding in <em>D. pseudoobscura</em> and <em>A. gambiae</em>.  
Predictions of conserved elements were then obtained by running phastCons
on these alignments with the --viterbi option.</P>

<P>PhastCons constructs a two-state phylo-HMM with a state for conserved
regions and a state for non-conserved regions.  The two states share a
single phylogenetic model, except that the branch lengths of the tree
associated with the conserved state are multiplied by a constant scaling
factor &rho (0 &le &rho &le 1).  The phylogenetic model and the scaling
factor &rho are estimated from the data by maximum likelihood using an EM
algorithm.  The state-transition probabilities of the phylo-HMM may also be
estimated by maximum likelihood, but we find that more biologically useful
results are obtained by treating them as tuning parameters.  In this case,
the transition probabilities were set such that the "coverage" of the track
(portion of sites in conserved elements) is approximately 30% and the prior
expected length of a conserved element is 25 bp.  To ensure that the scores
in different regions of the genome are directly comparable, we first obtain
global estimates of all parameters, by averaging estimates obtained from
different regions of the genome, and then, in a second pass, we produce
predictions across the genome using the same parameters everywhere.  The
predicted elements are segments of the alignment that are likely to have
been "generated" by the conserved state of the phylo-HMM--i.e., maximal
segments in which the maximum-likelihood (Viterbi) path completely resides
in the conserved state.  </P>

<P>Each predicted conserved element is assigned a log-odds score equal to
its log probability under the conserved model minus its log probability
under the non-conserved model.  The "score" field associated with this
track contains transformed log-odds scores, taking values between 0 and
1000.  (The scores are transformed using a monotonic function of the form a
* log(x) + b.)  The transformed scores determine how darkly each element is
shaded in the browser, with higher scores resulting in darker shading and
lower scores resulting lighter shading.  The raw log odds scores are
retained in the "name" field and can be seen on the details page or in the
browser when the track's display mode is set to "pack" or "full".</P>

<H2>Credits</H2>
<P>
This track was created at UCSC using the following programs:
<UL>
<LI>
Blastz and multiz from Scott Schwartz and Webb Miller of the 
<A HREF="http://www.bx.psu.edu/" TARGET=_blank>Penn State Bioinformatics 
Group</A>. 
<LI>
AxtBest, axtChain, chainNet, netSyntenic, and netClass 
developed by Jim Kent at UCSC. 
<LI>Tree estimation (phyloFit) and conservation-scoring (phastCons)
software by Adam Siepel at UCSC. 
</UL>
</P>

<H2>References</H2>

<H3>Phylo-HMMs</H3>
<P>
Felsenstein J and Churchill GA (1996).  A hidden Markov model approach to
variation among sites in rate of evolution.
<em>Mol Biol Evol</em> 13:93-104. </P>
<P>
Siepel A and Haussler D (2003).  
<a href="http://www.cse.ucsc.edu/~acs/phyhmm.pdf">Combining
phylogenetic and hidden Markov models in biosequence analysis</a>.
In <em> Proceedings of the Seventh Annual 
International Conference on Computational Molecular Biology (RECOMB
2003)</em>, pp. 277-286. </P>
<P>
Siepel A and Haussler D (2004). Phylogenetic hidden Markov models.
In R. Nielsen, ed.,  Statistical Methods in Molecular Evolution, Springer  
(in press). </P>
<P>
Yang Z (1995).  A space-time process model for the evolution of DNA
sequences.  <em>Genetics</em>, 139:993-1005. </P>

<H3>Chain/Net</H3>
<P>
Kent WJ, Baertsch R, Hinrichs A, Miller W, and Haussler D (2003). 
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1" 
TARGET=_blank>Evolution's cauldron: 
Duplication, deletion, and rearrangement in the mouse and human genomes</A>. 
<I>Proc Natl Acad Sci USA</I> 100(20):11484-11489 Sep 30 2003.

<H3>Multiz</H3>
<P>
Blanchette M, Kent WJ, Riemer C, Elnitski L, Smit AFA, Roskin KM, Baertsch R,
Rosenbloom K, Clawson H, Green ED, Haussler D, Miller W. (2004). 
<A HREF="http://www.genome.org/cgi/content/abstract/14/4/708" TARGET=_blank>Aligning Multiple Genomic Sequences with the Threaded Blockset Aligner</A>.
<em>Genome Res.</em> 14(4):708-15.

<H3>Blastz</H3>
<P>
Chiaromonte F, Yap VB, and Miller W (2002). 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" 
TARGET=_blank>Scoring pairwise genomic sequence alignments</A>. 
<em>Pac Symp Biocomput</em> 2002;:115-26.</P>
<P>
Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison R, Haussler D, and 
Miller W. (2003). 
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103" 
TARGET=_blank>Human-Mouse Alignments with BLASTZ</A>. 
<em>Genome Res.</em> 13(1):103-7.</P>
