<H3>About the <em>Caenorhabditis elegans</em> assembly</H3>
<P>
This genome assembly (UCSC version ce1)
is based on sequence version WS100 deposited into 
<A HREF="http://www.wormbase.org/about/about_Celegans.html" TARGET=_blank>WormBase</A> 
as of 02 May 2003. <em>C. elegans</em> is a major model 
organism used for biomedical research. 
It is the first multicellular animal to have a fully 
sequenced genome, a joint collaboration between the 
<A HREF="http://genome.wustl.edu/" TARGET=_blank>Genome Sequencing Center</A> 
at Washington University in St. Louis and the 
<A HREF="http://www.sanger.ac.uk/" TARGET=_blank>Sanger Institute</A>. 
For more information, see the special issue of <em>Science</em> 
&quot;<A HREF="http://www.sciencemag.org/content/vol282/issue5396/" 
TARGET=_blank>C. elegans: Sequence to Biology</A>&quot;.
<P>
The ce1 sequence and annotation data may be downloaded from the UCSC Genome 
Browser <A HREF="ftp://genome.ucsc.edu/goldenPath/ceMay2003/">FTP server</A> or
<A HREF="/downloads.html#c_elegans">Downloads</A> web page.
The ce1 annotation tracks were generated by UCSC and 
collaborators worldwide. See the <A HREF="/goldenPath/credits.html#ce_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a 
sequenced genomic clone, an mRNA, or EST, or a gene identifier, 
chromosomal coordinate range, or keywords from 
the GenBank description of an mRNA. The following list shows 
examples of valid position queries for the <em>C. elegans</em> genome. 
See the
<A HREF="http://genome.cse.ucsc.edu/goldenPath/help/hgTracksHelp.html" TARGET=_blank>
User's Guide</A> for more information.
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
       	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><BR></TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>chrII</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays all of chromosome II (valid chr entries include I, II, III, IV, V, M, X)</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>chrI:1-100000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays the first 100,000 bases of chrI</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>U40798.1</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays clone fragment U40798.1</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>mec-8</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of genome associated with RefSeq gene mec-8</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>F46A9.4</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays the region associated with the Sanger predicted gene F46A9.4</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>NM_060107</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of genome associated with RefSeq identifier NM_060107</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>BJ111711</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of EST associated with GenBank accession BJ111711</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>ring zinc finger</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists only RING zinc fingers</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>Jones,D.</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author D. Jones</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Jones D</em>, internally it uses 
			the format <em>Jones,D.</em>.</TD>
    </TR>
</TABLE>

