<H2>Description</H2>
<P>
This track shows syntenous (corresponding) regions between human and mouse chromosomes. 
<H2>Methods</H2>
<P>
We passed a 100k non-overlapping window over the genome and - using the blastz best in mouse 
genome alignments - looked for high-scoring regions with at least 40% of the bases aligning 
with the same region in mouse. 100k segments were joined together if they agreed in direction and
were within 500kb of each other in the human genome and within 4mb of each other in the mouse. 
Gaps were joined between syntenic anchors if the bases between two flanking regions agreed with 
synteny (direction and mouse location). Finally, we extended the syntenic block to include those 
areas.<p>
<H2>Credits</H2>
<P>
Contact <A HREF="mailto:baertsch@cse.ucsc.edu">Robert Baertsch</A> at UCSC for more information about this track.
<H2>Data Release Policy</H2>
<P>Mouse data is made available before scientific publication with the
following understanding:
<OL>
<LI>The data may be freely downloaded, used in analyses, and
repackaged in databases.
<LI>Users are free to use the data in scientific papers analyzing
particular genes and regions, provided that the providers of this data
(the Mouse Sequencing Consortium) are properly acknowledged.
<LI>The Centers producing the data reserve the right to publish the initial
large-scale analyses of the dataset-including large-scale identification
of regions of evolutionary conservation and large-scale
genomic assembly. Large-scale refers to regions with size on the order of
a chromosome (that is, 30 Mb or more).
<LI>Any redistribution of the data should carry this notice.</OL>
</P>
~

