<H2>Description</H3>
This track shows the best mouse/$organism chain for 
every part of the $organism genome.  It is useful for
finding orthologous regions and for studying genome
rearrangement.  In full mode the top level (level 1)
chains are the largest, highest scoring chains that
span this region.  In many cases there are gaps in the
top level chain.  When possible, these are filled in by
other chains that are displayed at level 2.  The gaps in 
level 2 chains may be filled by level 3 chains and so
forth.<BR>
<BR>
In the graphical display, the boxes represent ungapped 
alignments, while the lines represent gaps.  Clicking
on a box brings up detailed information about the chain
as a whole, while clicking on a line brings up information
on the gap.  The detailed information is useful in determining
the cause of the gap or, for lower level chains, the genomic
rearrangement.  Each item in the display is identified by the 
chromosome, strand, and location of the match (in thousands). <BR>

<H2>Methods</H2>
First, chains are derived from blastz alignments as
described in the details pages of the chain tracks
and sorted so that the highest scoring chain in the
genome is first. The program chainNet then places
chains one at a time, trimming them as necessary to
fit into section that is not already covered by 
a higher scoring chain.  During this process, a
natural hierarchy emerges where chains that fill gaps
in a previous chain are considered underneath the
previous chain.   The program netSyntenic fills in
information about the relationship between upper
and lower level chains, including whether a lower level
chain is syntenic with the higher level chain,  whether
it is inverted with respect to the higher level chain,
and so forth.  The program netClass then fills in 
how much of the gaps and chains are filled with N's
(sequencing gaps) in one or both species,  how much
is filled with transposons inserted before and after
mouse and $organism diverged, and so forth.

<H2>Credits</H2>
<P>The chainNet, netSyntenic, and netClass programs were
developed at the University of California
at Santa Cruz by Jim Kent.</P>

<P>Lineage-specific repeats were identified by Arian Smit and his
program <A HREF="http://repeatmasker.genome.washington.edu/cgi-bin/RepeatMasker">RepeatMasker</A>.</P>

<P>The browser display and database storage of the nets were made
by Robert Baertsch and Jim Kent.</P>

