<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
</head>
<body>
The <I>Archaeoglobus fulgidus DSM 4304</I> genome is 2.18 Million bp long and contains approximately 2486
predicted genes. <I>A. fulgidus</I> is a sulfur-metabolizing anaerobe that grows optimally around 83C. High-temperature sulfate reduction by Archaeoglobus species contributes to deep subsurface oil-well 'souring' by iron sulfide, which causes corrosion of iron and steel in oil-and gas-processing systems.<BR><BR>
The sequence was released Dec 1997 by the TIGR, and was described
in <I>Nature</I> <B>390</B>:364-70 (1997) Klenk HP, Clayton RA, Tomb JF, White O, Nelson KE, <I>et al.</I> &nbsp;"<a
 href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=9389475">The complete genome sequence of the hyperthermophilic, sulphate-reducing archaeon Archaeoglobus fulgidus.
</a>"
<blockquote><B>Abstract:</B> Archaeoglobus fulgidus is the first sulphur-metabolizing organism to have its genome sequence determined. Its genome of 2,178,400 base pairs contains 2,436 open reading frames (ORFs). The information processing systems and the biosynthetic pathways for essential components (nucleotides, amino acids and cofactors) have extensive correlation with their counterparts in the archaeon Methanococcus jannaschii. The genomes of these two Archaea indicate dramatic differences in the way these organisms sense their environment, perform regulatory and transport functions, and gain energy. In contrast to M. jannaschii, A. fulgidus has fewer restriction-modification systems, and none of its genes appears to contain inteins. A quarter (651 ORFs) of the A. fulgidus genome encodes functionally uncharacteri... [Click above reference link for full abstract]
</blockquote>
<P>
<h3>Sample position queries</h3>
<p>
A genome position can be specified by chromosomal coordinate range, COG
ID, or keywords from the GenBank or TIGR description of a gene. 
The available chromosome/plasmid names are:<BR><BR>
<table border=2 cellspacing=2 cellpadding=2>
<tr><th>Browser Chrom/Plasmid Name</th><th>Length (bp)</th><th>GC Content (%)</th><th>Gene Count</th><th>NCBI RefSeq Accession</th></tr>
<tr><td><a href="/cgi-bin/hgTracks?db=archFulg1&position=chr:1-02178400">chr</a></td><td align=right>2178400</td><td align=right>48.58</td><td align=right>2486</td><td align=center><a href="http://www.ncbi.nlm.nih.gov/entrez/viewer.fcgi?val=NC_000917">NC_000917</a></td></tr>
</table>

<p>
The following list shows examples of valid position queries for this
genome:&nbsp;
</p>
<p>
<table cellspacing="1" cellpadding="1" border="0">
<tr><th align=left>Request:</th><th align=left>Genome Browser Response:</th></tr>
<tr><td>chr</td><td>Displays the entire sequence "chr" in the browser window</td></tr>
<tr><td>chr:1-10000&nbsp;&nbsp;&nbsp;&nbsp;</td><td>Displays first ten thousand bases of the sequence "chr"</td></tr>
<tr><td>transporter&nbsp;&nbsp;&nbsp;&nbsp;</td><td>Lists all genes with "transporter" in the name or description</td></tr>
<tr><td>AF0010</td><td>Display genome at position of gene AF0010</td></tr>
</tbody>
</table>
<P>
<HR>
<H3>Credits</H3>
The Archaeal Genome Browsers at UCSC were developed by members of the
<a href="http://lowelab.ucsc.edu/">Lowe Lab</a> (Kevin Schneider,
Katherine Pollard, Andy Pohl, Todd Lowe) and Robert Baertsch, with significant support from
the <a href="http://genome.ucsc.edu/staff.html">UCSC Human Genome
Browser group</a>.
The Archaeal Browsers are run by a slightly modified
version of the <a href="http://genome.ucsc.edu/">UCSC Human Genome
Browser</a> system.  All queries, bug reports, content corrections, suggested improvements,
and new track data submissions should be sent to Todd Lowe (lowe
@soe.ucsc.edu).

<P>If you use the browser for your research and wish to cite it, a
paper describing the work has been accepted to the 2006 <I>Nucleic
Acids Research</I>
<a href="http://www3.oup.co.uk/nar/database/a/">Database Issue</a> (KL Schneider, KS Pollard, R Baertsch, A Pohl, TM Lowe). 
Citations will help us obtain funding to continue
development of this community resource.

</body>
</html>
