# gbRnaMapInfo.sql was originally generated by the autoSql program, which also 
# generated gbRnaMapInfo.c and gbRnaMapInfo.h.  This creates the database representation of
# an object which can be loaded and saved from RAM in a fairly 
# automatic way.

#Summary of how a genbank transcript maps.  Most fields show . if FALSE, a shortened version of field name if true, just for better readability.
CREATE TABLE gbRnaMapInfo (
    chrom varchar(255) not null,	# Chromosome of alignment
    chromStart int not null,	# Start of alignment
    chromEnd int not null,	# End of alignment
    accession varchar(255) not null,	# Genbank accession
    genbankCds varchar(255) not null,	# True if came with a genbank CDS record
    genbankFullCds varchar(255) not null,	# True if genbank CDS record showed full length
    intronNudged varchar(255) not null,	# True if moved alignment a little to get gt/ag ends
    directionFlipped varchar(255) not null,	# True if flipped strand based on splice sites
    mergedSmallGap varchar(255) not null,	# True if merged a small gap in alignment into exon
    splitLargeGap varchar(255) not null,	# True if split based on a large gap in the alignment
    splitLargGapInCds varchar(255) not null,	# True if split large gap in CDS region
    smallGapBustsFrame varchar(255) not null,	# True if small gap in alignment breaks frame
    largeGapBustsFrame varchar(255) not null,	# True if a large gap in alignment breaks frame
    intronNudgeBustsFrame varchar(255) not null,	# True if we broke frame nudging intron
    cdsOutside varchar(255) not null,	# True if cds outside of bed
    outputCds varchar(255) not null,	# True if output with CDS
              #Indices
    PRIMARY KEY(chrom)
);
