<H3>About the <em>Mus musculus</em> assembly</H3>
<P>
The February 2002 mouse genome assembly (UCSC version mm2) 
is based on the MGSCv3 Whole Genome Shotgun assembly 
produced at the Whitehead Institute using their Arachne software. 
<P>
The mm2 sequence and annotation data may be downloaded from the UCSC Genome 
Browser <A HREF="ftp://genome.ucsc.edu/goldenPath/mmFeb2002/">FTP server</A>
or <A HREF="/downloads.html#mouse">Downloads</A> page.
<P>
Many thanks to the Whitehead Institute and the Mouse Genome Sequencing 
Consortium for providing this assembly. 
The mm2 annotation tracks were generated by UCSC and 
collaborators worldwide. See the <A HREF="/goldenPath/credits.html#mouse_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a sequenced 
genomic clone, an mRNA or EST or STS marker, or 
a cytological band, a chromosomal coordinate range, or keywords from 
the GenBank description of an mRNA. The following list provides examples 
of various types of position queries for the mouse genome. See the 
<A HREF="http://genome.cse.ucsc.edu/goldenPath/help/hgTracksHelp.html" 
TARGET=_blank>User's Guide</A> for more information. 
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><BR></TD>
    </TR>
	
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>chr16</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays all of chromosome 16</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>chr16:1-5000000</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays first 5 million bases of chr 16</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>D16Mit120</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays region around STS marker DMit16120 from the MGI consensus genetic map.
 			Includes 100,000 bases on each side as well.</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>D16Mit203;D16Mit70</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays region between STS markers D16Mit203 and D16Mit70.
			Includes 100,000 bases on each side as well.</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>AW045217</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays region of EST with GenBank accession AW045217</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>Ncam2</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays region of genome with official MGI mouse genetic nomenclature Ncam2</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>pseudogene mRNA</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists transcribed pseudogenes but not cDNAs</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>kruppel zinc finger</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>huntington</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists candidate genes associated with Huntington's disease</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>Xu,C.S.</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs deposited by co-author C.S. Xu</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Xu CS</em>, internally it uses 
			the format <em>Xu,C.S.</em>.</TD>
    </TR>
</TABLE>


