<H2>Description</H2>
<P>
The Viral mRNA track shows alignments between viral mRNAs
in GenBank and the SARS genome. Aligning regions (usually exons)
are shown as black boxes connected by lines for gaps (spliced
out introns usually). In full display, arrows on the introns
indicate the direction of transcription.</P>

<H2>Method</H2>
<P>
GenBank viral mRNAs were aligned against the genome using the 
blastz program. When a single mRNA aligned multiple places, 
the alignment having the highest base identity was found. 
Only those alignments having at least 60% base identity were kept.
These alignments were converted into axt format and the 
resulting axts fed into axtChain. AxtChain organizes all the 
alignments between a single viral mrna and the SARS genome into a group, 
and makes a kd-tree out of all the gapless subsections (blocks) of the 
alignments.  Next, maximally scoring chains of these blocks were found by 
running a dynamic program over the kd-tree. Chains scoring below a 
certain threshold were discarded; the remaining chains are displayed in
this track.</P>

<H2>Credits</H2>
<P>
This track was produced at UCSC from mRNA sequence data
submitted to the international public sequence databases by 
scientists worldwide.</P>

<H2>References</H2>
<P>
Benson, D.A., Karsch-Mizrachi, I., Lipman, D.J., Ostell, J., and 
Wheeler, D.L. 
<A HREF="http://nar.oupjournals.org/cgi/content/full/32/suppl_1/D23" 
TARGET=_blank>GenBank: update</A>. <em>Nucleic Acids Res.</em> <B>32</B>,
D23-6 (2004).</P>
<P>
Schwartz, S., Kent, W.J., Smit, A., Zhang, Z., Baertsch, R., Hardison, R.,
Haussler, D., and Miller, W.
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103"
TARGET=_blank>Human-mouse alignments with BLASTZ</A>.
<em>Genome Res.</em> <B>13</B>(1), 103-107 (2003).</P>

