<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>Human Genome Working Draft - v29</TITLE>
	<LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 WIDTH="100%">
<TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="/images/title.jpg"></TH></TR>

</TR>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
	<table bgcolor="000000" cellpadding="1" cellspacing="1" width="100%" height="27"><tr bgcolor="2636D1"><td valign="middle">
	<table BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24">
		<TR><TD VALIGN="middle"><font color="#89A1DE">&nbsp; 
			<A HREF="/index.html" class="topbar">
				Home</A> &nbsp; - &nbsp;
			<A HREF="/cgi-bin/hgGateway" class="topbar">
				Genome Browser</A> &nbsp; - &nbsp; 
			<A HREF="/cgi-bin/hgNear" class="topbar">
				Family Browser</A> &nbsp; - &nbsp; 
			<A HREF="/cgi-bin/hgBlat?command=start" class="topbar">
				Blat </A> &nbsp; - &nbsp; 
			<A HREF="/cgi-bin/hgText?command=start" class="topbar">
				Table Browser</A> &nbsp; - &nbsp; 
			<A HREF="/FAQ/" class="topbar">
				FAQ</A> &nbsp; - &nbsp; 
			<A HREF="/goldenPath/help/hgTracksHelp.html" class="topbar">
				Help</A> &nbsp;</font></TD>
    	</TR></TABLE>
	</TD></TR></TABLE>
</TD></TR>

<!--Content Tables------------------------------------------------------->
<TR><TD CELLPADDING=10>	
	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><b>&nbsp; Procedure for Creating a Mirror Site for Human Genome Browser </b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>

The following procedure provides you step-by-step instructions 
to create a mirror browser incrementally. You may choose to set up either 
a full mirror browser or a partial one depending on your disk space and 
needs. You are also welcome to try our <a href="#script">script</a> which 
can help you to create a mirror site on a Linux system automatically.

<p><b> Prerequisites:</b>
<p> To set up a full mirror browser, we recommend at a MINIMUM the following
disk partitions and sizes:
<ul>
<li>MySQL Table spaces: ~ 128 GB
<li>Apache web pages area: ~ 1 GB
<li>"/gbdb" area for the browser: ~ 165 GB
</ul>

<H3>WARNING! This is a major change from the previous /cluster and
 /projects based data areas.<BR>We recommend existing mirror sites closely
 check steps marked NEW ITEMS!</H3>

<br><p><a NAME="script"></a><b>Sample Automated Script:</b>
<p> We have created a script, <a href="GENOME-MIRROR.sh">GENOME-MIRROR.sh</a>,
which should automate the following steps but remains a work in progress. 
It has been tested only on Redhat Linux 7.2 systems, but you may find it useful.
<br>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
<BR>

	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><b>&nbsp;  Full Details on Steps:   </b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>

</font>
<UL>
<LI><a href="#step1">Install Apache server and MySQL server</a>
<LI><a href="#step2">Get all the html files and most of the text files</a> (1 GB)
<LI><a href="#step3">Get the data for each individual genome assembly and install databases </a> (70 GB)&nbsp;<B>NEW ITEMS!</B>
<LI><a href="#step4">Obtain the /gbdb data file area</a> (165 GB)&nbsp;<B>NEW ITEMS!</B>
<LI><a href="#step5">Set up database</a>&nbsp;<B>NEW ITEMS!</B>
<LI><a href="#step6">Get executable files</a>
<li><a href="#step7">Set up Hgcentral session preference table</a>&nbsp;<B>NEW ITEMS!</B>
<li><a href="#step8">Create a $WEBROOT/trash apache user writable directory for temp files</a>
<li><a href="#step9">Set up HgFixed database</a>
<li><a href="#step10">Set up three protein databases</a>&nbsp;<B>NEW ITEMS!</B>
</UL>
<br><br>
<hr>
<p><a NAME="step1"></a><b>1. Install Apache server and MySQL server</b>
<ol type="a">
<li> We will not provide consulting on the operation or configuration 
	of Apache or MySQL. If you are not familiar with the setup of Apache or MySQL, 
	you will have to find someone at your site that is. We use a reference platform 
	of Redhat 7.2 for all the steps described here. <br>
<li> Find where your web pages are located. For example, on a Redhat Linux
	system using a stock apache RPM, the web pages are stored in /var/www/html. 
	We will refer to this directory as WEBROOT in this document and you should 
	substitute your real path for WEBROOT whenever you see that in our write-up. But, before
	doing this, you <b>MUST</b> be aware of the following: <br>
      <ul type="DISC"><b>
      <li> THE FOLLOWING ASSUMES YOU ARE NOT ALREADY SERVING UP DATA USING YOUR 
	   APACHE SERVER. BEAR IN MIND THE SELECTION OF THE WEBROOT WILL OVERWRITE ANY 
	   EXISTING DATA IN THAT AREA.
      </ul></b>

<li> Find where your cgi-bin directory is located. It should be under
	the parent directory of the WEBROOT directory. For example, on a Redhat Linux 
	system using a stock apache RPM, the path of the cgi-bin directory is 
	/var/www/cgi-bin. We will refer to this directory as CGIBIN in this document 
	and you should substitute your real path for CGIBIN whenever you see that.<br>
<li> Also find where your MySQL data is located. For example on a Redhat 
	Linux system using stock RPM's, this is located in /var/lib/mysql. We will 
	refer to this directory as MYSQLDATA in this document and you should substitute 
	your real path for MYSQLDATA whenever you see that.<br>
<br></ol>

<p><a NAME="step2"></a><b>2. Get all the html files and most of the
text files (1.0 GB)</b>
<ol type="a">
<li> Obtain the software package rsync. If it isn't already installed on 
	your system, obtain it from <a href="http://rsync.samba.org">http://rsync.samba.org</a>
<br>
<li> We are running a rsync server, but you must email us first with the hostname 
	from which you will be rsyncing so that we can allow it access. Email to 
<A HREF="mailto:&#103;&#101;n&#111;&#109;&#101;&#45;mi&#114;&#114;&#111;&#114;&#64;&#99;&#115;&#101;.&#117;&#99;&#115;&#99;.&#101;du">
&#103;&#101;n&#111;&#109;&#101;&#45;mi&#114;&#114;&#111;&#114;&#64;&#99;&#115;&#101;.&#117;&#99;&#115;&#99;.&#101;du</A>.

<br>
<li>After we respond to your request, do a test connection: <br> 
<PRE>
rsync -navz --progress rsync://genome-rsync.cse.ucsc.edu
</PRE>
This should respond with:
<PRE>
genome          UCSC Human Genome Web Site (85 GB)
gbdb            UCSC Human Genome Browser New Config Files (161 GB)
</PRE>
<br>
<li>Determine the destination of the copy ($WEBROOT) and fire off the
	production copy:<br>
	The trailing slash is important!<br>
<PRE>
rsync --progress -avz \
 --exclude='goldenPath/*/bigZips/*' \ 
 --exclude='goldenPath/*/chromosomes/*' \
 --exclude='goldenPath/*/database/*' \
 --exclude='goldenPath/*/contigs/*' \
 --exclude='goldenPath/*/pieces/*' \ 
 --exclude='goldenPath/*/waba/*' \ 
 --exclude='goldenPath/*/alignments/*' \ 
 --exclude='goldenPath/*/sanger/*' \ 
 --exclude='goldenPath/*/vsMm3/*' \ 
 --exclude='goldenPath/*/vsMm2/*' \ 
 --exclude='goldenPath/chromReports/' \ 
 --exclude='goldenPath/mapPlots/' \ 
 rsync://genome-rsync.cse.ucsc.edu/genome/ $WEBROOT/
</PRE>
</ol>

<p><a NAME="step3"></a><b>3. Get the data for each individual
genome assembly and installing databases ( 70 GB )</b>

<ol type="a">

<li>Get C. briggsae July, 2002 full data set (cb1: 315 Mb) by doing:
<p> 
<PRE>
rsync --progress -avz \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/cbJul2002/ \
  $WEBROOT/goldenPath/cbJul2002/
</PRE>

<li>Get C. elegans May, 2003 full data set (ce1: 2.0 GB) by doing:
<p> 
<PRE>
rsync --progress -avz \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/ceMay2003/ \
  $WEBROOT/goldenPath/ceMay2003/
</PRE>

<li>Get SARS coronavirus April, 2003 full data set (sc1: 106 Mb) by doing:
<p> 
<PRE>
rsync --progress -avz \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/scApr2003/ \
  $WEBROOT/goldenPath/scApr2003/
</PRE>

<li>Get Rat June, 2003 full data set (rn3: 5.5 GB) by doing:
<p> 
<PRE>
rsync --progress -avz \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/rnJun2003/ \
  $WEBROOT/goldenPath/rnJun2003/
</PRE>

<li>Get Rat January, 2003 full data set (rn2: 4.1 GB) by doing:
<p> 
<PRE>
rsync --progress -avz \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/rnJan2003/bigZips/ \
  $WEBROOT/goldenPath/rnJan2003/bigZips/
rsync --progress -avz \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/rnJan2003/chromosomes/ \
  $WEBROOT/goldenPath/rnJan2003/chromosomes/
rsync --progress -avz  \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/rnJan2003/database/ \
  $WEBROOT/goldenPath/rnJan2003/database/
</PRE>

<li>Get Mouse February, 2003 full data set (mm3: 12 GB) by doing:
<p> 
<PRE>
rsync --progress -avz \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/mmFeb2003/bigZips/ \
  $WEBROOT/goldenPath/mmFeb2003/bigZips/
rsync --progress -avz \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/mmFeb2003/chromosomes/ \
  $WEBROOT/goldenPath/mmFeb2003/chromosomes/
rsync --progress -avz  \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/mmFeb2003/database/ \
  $WEBROOT/goldenPath/mmFeb2003/database/
</PRE>

<li>Get Mouse February, 2002 full data set (mm2: 6.2 GB) by doing:
<p> 
<PRE>
rsync --progress -avz \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/mmFeb2002/bigZips/ \
  $WEBROOT/goldenPath/mmFeb2002/bigZips/
rsync --progress -avz \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/mmFeb2002/chromosomes/ \
  $WEBROOT/goldenPath/mmFeb2002/chromosomes/
rsync --progress -avz  \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/mmFeb2002/database/ \
  $WEBROOT/goldenPath/mmFeb2002/database/
</PRE>


<li>Get Human July, 2003 full data set (hg16: 8.6 GB) by doing:
<p> 
<PRE>
rsync --progress -avz \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/hg16/bigZips/ \
  $WEBROOT/goldenPath/hg16/bigZips/
rsync --progress -avz \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/hg16/chromosomes/ \
  $WEBROOT/goldenPath/hg16/chromosomes/
rsync --progress -avz  \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/hg16/database/ \
  $WEBROOT/goldenPath/hg16/database/
</PRE>

<li>Get Human April 10, 2003 full data set (hg15: 6.5 GB) by doing:
<p> 
<PRE>
rsync --progress -avz \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/10april2003/bigZips/ \
  $WEBROOT/goldenPath/10april2003/bigZips/
rsync --progress -avz \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/10april2003/chromosomes/ \
  $WEBROOT/goldenPath/10april2003/chromosomes/
rsync --progress -avz  \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/10april2003/database/ \
  $WEBROOT/goldenPath/10april2003/database/
</PRE>

<li>Get Human November 14, 2002 full data set (hg13: 12.0 GB) by doing:
<p> 
<PRE>
rsync --progress -avz \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/14nov2002/bigZips/ \
  $WEBROOT/goldenPath/14nov2002/bigZips/
rsync --progress -avz \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/14nov2002/chromosomes/ \
  $WEBROOT/goldenPath/14nov2002/chromosomes/
rsync --progress -avz  \
  rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/14nov2002/database/ \
  $WEBROOT/goldenPath/14nov2002/database/
</PRE>

</ol>

<h4><a NAME="step4"></a>4. Obtain the /gbdb data file area</h4>

<ol type="a">
<li>You will need the portions of /gbdb used by the browser (160 GB):
<br>
<PRE>
rsync --progress -avz \
   rsync://genome-rsync.cse.ucsc.edu/gbdb/ \
   /gbdb/
</PRE>
<br>
</ol>

<h4><a NAME="step5"></a>5. Set up database</h4>
<B>Existing Mirror Sites: There are numerous changes to tables
and data within these tables to support the /gbdb file storage method.
You are advised to consider dropping and rebuilding the databases.
If that presents a problem, contact us and we can provide detailed
updates to be made, but that is not directly documented.
</B>

<ul>
<p><a NAME="table"></a> Use the following table to identify the freeze 
date ($FREEZEDATE) and the database version ($DBVERSION) for each genome 
assembly:
<p>
<TABLE CELLPADDING=0 CELLSPACING=0 MARGINS=0>
<TR><TD> $FREEZEDATE </TD><TD WIDTH=10></TD><TD> $DBVERSION </TD></TR>
<TR><TD> --------------- </TD><TD></TD><TD> ---------------  </TD></TR>
<TR><TD> cbJul2002   </TD><TD></TD><TD> cb1         </TD></TR>
<TR><TD> ceMay2003   </TD><TD></TD><TD> ce1         </TD></TR>
<TR><TD> scApr2003   </TD><TD></TD><TD> sc1         </TD></TR>
<TR><TD> rnJun2003   </TD><TD></TD><TD> rn3         </TD></TR>
<TR><TD> rnJan2003   </TD><TD></TD><TD> rn2         </TD></TR>
<TR><TD> mmFeb2003   </TD><TD></TD><TD> mm3         </TD></TR>
<TR><TD> mmFeb2002   </TD><TD></TD><TD> mm2         </TD></TR>
<TR><TD> hg16 </TD><TD></TD><TD> hg16         </TD></TR>
<TR><TD> 10april2003 </TD><TD></TD><TD> hg15         </TD></TR>
<TR><TD> 14nov2002   </TD><TD></TD><TD> hg13         </TD></TR>
</TABLE>
</ul>

<ol type="a">
<li>After connecting to the MySQL server, create a database called "$DBVERSION"
corresponding to $FREEZEDATE in the <a href="#table">table</a> above.
<PRE>
mysql> create database $DBVERSION;
</PRE>
<li>Create tables for the "$DBVERSION" database.
<ul type="i">
<li> Move to $WEBROOT/goldenPath/$FREEZEDATE/database/ directory.<BR>
All the files for creating the "$DBVERSION" database are in this directory.
<li>Create tables: (You may have to use the following shell script for this step.)
<PRE>
for file in *.sql 
do
mysql -youraccountoptions $DBVERSION < $file
done 
</PRE>
<li>Load data into the tables: (The script here is an example
for this process.  This takes a long time, on the order of hours.)
<PRE>
for file in *.txt.gz
do
        FN=`echo $file | sed -e "s/.gz//"`
        echo "working on $file"
        gunzip $file
        mysqlimport -youraccountoptions $DBVERSION $FN
        echo "imported $FN"
        gzip $FN &amp;
done
</PRE>
</ul>

<li>Issue permission to the "$DBVERSION" database.<br>
Permission could be issued as follows: <br>
<PRE>
mysql> grant select on $DBVERSION.* to $USERNAME@$HOSTNAME identified by "$PASSWORD";
</PRE>
<br>
To make this command work, you need to set up $USERNAME,$HOSTNAME and 
$PASSWORD properly. In this example, public access to the "$DBVERSION" 
database is restricted to read-only.
</ol>

<h4><a NAME="step6"></a>6. Get executable files</h4>
<ol type="a">

<li>The cgi-bin binaries are available via:
<br>
<PRE>
rsync --progress -avz rsync://genome-rsync.cse.ucsc.edu/cgi-bin/ $CGIBIN/
</PRE>
<br>

<li>In the CGIBIN directory, make a <a href="sample.html">hg.conf</a> file, in which, $USERNAME,$HOSTNAME and $PASSWORD used for the "$DBVERSION" database should be specified. Remember, variables should not be used in this file. 
<br>

<li>Set up environment variables needed by the Makefile to install the binaries for the website in the correct place.
<PRE>
setenv GLOBAL_CONFIG_FILE $CGIBIN/hg.conf
setenv HGCGI $CGIBIN
</PRE>
<li>Download the released zipped version of the source files from <a href="/admin/jksrc.zip">here</A>.
<br>
Follow the <a href="jk-install.html">instructions</a> for how to compile
the files. <br>
The source tree can also be obtained via <A HREF="cvs.html"> CVS </A>
</ol>

<B>Existing Mirror Sites: Note the Schema for Hgcentral Has Changed.
You are advised to drop the hgcentral database and reload it.
This will result in the loss of user settings. If this is a problem
contact us for a incremental fix.
</B>
<P>

<h4><a NAME="step7"></a>7. Setup the "hgcentral" tables </h4>
<ol type="a">
<li>Download the schema for the hgcentral database <a href="hgcentral.sql"> here</a>.
<li>Create a hgcentral database <P>
<PRE>
   mysql> create database hgcentral
</PRE>
<P>
<li>Add the hgcentral tables <P>
<PRE>
   mysql -youraccountoptions hgcentral < hgcentral.sql
</PRE>
<li>Create a user/password with the ability to update and insert. Many folks
 start out with the same user that they use for reading browser databases
 and then later create a separate user with higher priviledge. However, to
 get started, consider trying the same user.
<li>Add that user to the hg.conf. A sample hg.conf is below:
<PRE>
# Config file for the UCSC Human Genome server
#
# the format is in the form of name/value pairs
# written 'name=value' (note that there is no space between
# the name and its value. Also, no blank lines should be in
# this file
###########################################################
# db.host is the name of the MySQL host to connect to
db.host=localhost
# db.user is the username is use when connecting to the host
db.user=myhguser
# this is the password to use with the above hostname
db.password=myhguserpassword
db.trackDb=trackDb

# central.host is the name of the host of the central MySQL
# database where stuff common to all versions of the genome
# and the user database is stored.
central.db=hgcentral
central.host=localhost
#
# Be sure this user has UPDATE AND INSERT privs for hgcentral
#
central.user=myhguser
central.password=myhguserpassword
central.domain=.mydomain.edu
</PRE>
</ol>

<h4><a NAME="step8"></a>8. Create a "trash" directory </h4>
The cgi programs use a temporary area to create and store images
used by the browser. This directory is by default looked for in
$WEBROOT/trash. You should make this directory and allow the user
that runs the web server write access to it.

<h4><a NAME="step9"></a>9. Create the hgFixed database</h4>
<ol type="a">
<li>Download a copy of the dumped hgFixed tables.</li>
<PRE>
rsync --progress -avz \
   rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/hgFixed/ \
   $WEBROOT/goldenPath/hgFixed/database/
<li>Create a hgFixed database</li>
<PRE>
mysql> create database hgFixed
</PRE>
<li>Import the data in a similar fashion as listed above.</li>
</ol>
<p>
<h4><a NAME="step10"></a>10. Create the five protein databases</h4>
<ol type="a"> 
<li>Download a copy of the dumped protein tables.
<PRE>
rsync --progress -avz  \
rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/proteins/ \
$WEBROOT/goldenPath/proteins/database/
rsync --progress -avz  \
rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/proteins0111/ \
$WEBROOT/goldenPath/proteins0111/database/
rsync --progress -avz  \
rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/proteins070403/ \
$WEBROOT/goldenPath/proteins070403/database/
rsync --progress -avz  \
rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/proteins072003/ \
$WEBROOT/goldenPath/proteins072003/database/
rsync --progress -avz  \
rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/proteins0305/ \
$WEBROOT/goldenPath/proteins0305/database/
rsync --progress -avz  \
rsync://genome-rsync.cse.ucsc.edu/genome/goldenPath/proteins0405/ \
$WEBROOT/goldenPath/proteins0405/database/
</PRE>
<li>Create the protein databases
<PRE>
mysql> create database proteins
mysql> create database proteins0111
mysql> create database proteins072003
mysql> create database proteins070403
mysql> create database proteins0305
mysql> create database proteins0405
</PRE>
<li>Import the data in a similar fashion as listed above.
</ol>
<hr WIDTH="100%">
<br><i>Should you have any comments, inquires or questions, please contact
<A HREF="mailto:&#103;&#101;n&#111;&#109;&#101;&#45;mi&#114;&#114;&#111;&#114;&#64;&#99;&#115;&#101;.&#117;&#99;&#115;&#99;.&#101;du">
&#103;&#101;n&#111;&#109;&#101;&#45;mi&#114;&#114;&#111;&#114;&#64;&#99;&#115;&#101;.&#117;&#99;&#115;&#99;.&#101;du</A>
</i>.
<br><i>Last Modified: 2003-11-03 - 03 November 2003</i>
<br>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
<!-- END SECOND SECTION ---------------------------------------------------------->

</TD></TR></TABLE>
</BODY></HTML>
