<HTML>
<HEAD>
    <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
    <META name="description" content="UCSC Genome Browser">
    <META name="keywords" content="genome, genome browser, genome assembly, Blat, UCSC, mirror, genome browser source, bioinformatics, gene prediction, SNP, EST, mRNA">
    <TITLE>UCSC Genome Browser: Mirror Instructions</TITLE>
    <LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="/images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<!--HOTLINKS BAR-->
    <TR><TD COLSPAN=2 HEIGHT=40>
	<TABLE bgcolor="000000" cellpadding="1" cellspacing="1" width="100%" height="27">
	    <TR bgcolor="2636D1"><td valign="middle">
		<TABLE BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24">
	     	    <TR><TD VALIGN="middle"><font color="#89A1DE">&nbsp;
			<A HREF="/index.html" class="topbar">
				Home</A> &nbsp; - &nbsp;
			<A HREF="/cgi-bin/hgGateway" class="topbar">
				Genome Browser</A> &nbsp; - &nbsp;
			<A HREF="/cgi-bin/hgNear" class="topbar">
				Gene Sorter</A> &nbsp; - &nbsp;
			<A HREF="/cgi-bin/hgBlat?command=start" class="topbar">
				Blat </A> &nbsp; - &nbsp;
			<A HREF="/cgi-bin/hgText?command=start" class="topbar">
				Table Browser</A> &nbsp; - &nbsp;
			<A HREF="/FAQ/" class="topbar">
				FAQ</A> &nbsp; - &nbsp;
			<A HREF="/goldenPath/help/hgTracksHelp.html" class="topbar">
				Help</A> &nbsp;</font></TD>
    		    </TR>
		</TABLE>
	    </TD></TR>
	</TABLE>
    </TD></TR>

<!--Content Tables------------------------------------------------------->
    <TR><TD CELLPADDING=10>	
	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1>
	    <TR><TD>
	    	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
		    <TR><TD>
			<FONT SIZE="4"><B>&nbsp; Procedure for Creating a Mirror Site for the UCSC Genome Browser</B></FONT>
		    </TD></TR>
		</TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
		    <TR><TH HEIGHT=10></TH></TR>
		    <TR><TD WIDTH=10>&nbsp;</TD>
			<TD>
			    <P>
			    The following procedure provides you with step-by-step
			    instructions to incrementally create a mirror of the
			    UCSC Genome Browser. You may choose to set up either a
			    full mirror browser or a partial one, depending on your disk
			    space and needs.
			    <P>
			    <B> Prerequisites:</B>
			    <P>
			    To set up a full mirror browser, we recommend at a
			    MINIMUM the following disk partitions and sizes:
			    <UL>
			        <LI>MySQL Table spaces: ~ 400 Gb
			        <LI>Apache web pages area: ~ 1 Gb
			        <LI>"/gbdb" area for the browser: ~ 400 Gb
			        <LI>Genome database text dumps: ~ 100 Gb
			    </UL></P><P>
				These data requirements will increase
				in the future as genomes are added to
				the system.</P>
			</TD><TD WIDTH=15></TD>
		    </TR>
		</TABLE>
		<BR>
	    </TD></TR>
	</TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
	<BR>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	    <TR><TD>
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
		    <TR><TD>
			<FONT SIZE="4"><B>&nbsp;  Step-by-Step Details</B></FONT>
		    </TD></TR>
		</TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
		    <TR><TH HEIGHT=10></TH></TR>
		    <TR><TD WIDTH=10>&nbsp;</TD>
			<TD>

<OL>
<LI><A HREF="#step1">Install Apache server and MySQL server</A>
<LI><A HREF="#step2">Get all the html files and most of the text files</A>&nbsp;(1 Gb)
<LI><A HREF="#step3">Get the data for each individual genome assembly and install
databases </A>&nbsp;(100 Gb)
<LI><A HREF="#step4">Obtain the /gbdb data file area</A>&nbsp;(400 Gb)
<LI><A HREF="#step5">Set up database</A>&nbsp;(400 Gb MySQL data)
<LI><A HREF="#step6">Get executable files</A>
<LI><A HREF="#step7">Set up Hgcentral session preference table</A>
<LI><A HREF="#step8">Create a $WEBROOT/trash apache user writable directory for temp files</A>
<LI><A HREF="#step9">Set up HgFixed database</A>&nbsp;(500 Mb)
<LI><A HREF="#step10">Set up the protein databases</A>&nbsp;(700 Mb)
<LI><A HREF="#step11">Set up the swissProt database</A>&nbsp;(400 Mb)
</OL>
<BR><BR>
<HR>
<P>
<A NAME="step1"></A>
<B>1. Install Apache server and MySQL server</B>
<OL type="a">
<LI>
We will not provide consulting on the operation or configuration
of Apache or MySQL. If you are not familiar with the setup of
Apache or MySQL, you will have to find someone at your site that
is. We use a reference platform of Redhat 9 for all the steps
described here.
<BR>
<LI>
Find the location of your web pages. For example, on a Redhat Linux
system using a stock apache RPM, the web pages are stored in
<em>/var/www/html</em>. We will refer to this directory as
WEBROOT in this document and you should substitute your real
path for WEBROOT whenever you see that in our write-up. But, before
doing this, you <B>MUST</B> be aware of the following:
<BR>
<UL type="DISC">
<LI>
<B>THE FOLLOWING ASSUMES YOU ARE NOT ALREADY SERVING UP
DATA USING YOUR APACHE SERVER. BEAR IN MIND THE SELECTION OF
THE WEBROOT WILL OVERWRITE ANY EXISTING DATA IN THAT AREA.</B>
</UL>
<LI>
Find the location of your <em>cgi-bin</em> directory. It should be under
the parent directory of the WEBROOT directory. For example, on
a Redhat Linux system using a stock apache RPM, the path of
the <em>cgi-bin</em> directory is <em>/var/www/cgi-bin</em>. We will refer to
this directory as CGIBIN in this document and you should substitute your
real path for CGIBIN whenever you see that.
<BR>
<LI>
Next, find the location of your MySQL data. For example, on a
Redhat Linux system using stock RPM's, this is located in
<em>/var/lib/mysql</em>. We will refer to this directory as
MYSQLDATA in this document and you should substitute
your real path for MYSQLDATA whenever you see that.
<BR><BR>
</OL>
<P>
<A NAME="step2"></A>
<B>2. Get all the html files and most of the text files
(1.0 Gb)</B>
<OL type="a">
<LI>
Obtain the software package rsync. If it isn't already
installed on your system, obtain it from
<A HREF="http://rsync.samba.org">http://rsync.samba.org</A>
<BR>
<LI> Test the rsync connection:
<BR>
<PRE>
rsync -navz --progress rsync://hgdownload.cse.ucsc.edu
</PRE>
This should respond with:
<PRE>
genome          UCSC Human Genome Web Site (111 GB)
goldenPath      UCSC Human Golden Path Downloads (170 GB)
cgi-bin         UCSC Human Genome Web Site CGI Binaries
gbdb            UCSC Human Genome Browser New Config Files (313 GB)
mysql           UCSC Human Genome Raw Mysql Tables (356 GB)
</PRE>
<BR>
<LI>Determine the destination of the copy ($WEBROOT) and fire off the production copy<BR>
<em>The trailing slash is important!</em><BR>
<PRE>
rsync --progress --partial -avz \
 --exclude='goldenPath/hg1*' \
 --exclude='goldenPath/10april2003/*' \
 --exclude='goldenPath/14nov2002/*' \
 --exclude='goldenPath/panTro1/*' \
 --exclude='goldenPath/canFam1/*' \
 --exclude='goldenPath/misc' \
 --exclude='goldenPath/mmFeb2003/*' \
 --exclude='goldenPath/mm4/*' \
 --exclude='goldenPath/mm5/*' \
 --exclude='goldenPath/rnJun2003/*' \
 --exclude='goldenPath/rnJan2003/*' \
 --exclude='goldenPath/galGal2/*' \
 --exclude='goldenPath/danRer1/*' \
 --exclude='goldenPath/fr1/*' \
 --exclude='goldenPath/ci1/*' \
 --exclude='goldenPath/dm1/*' \
 --exclude='goldenPath/dp1/*' \
 --exclude='goldenPath/dp2/*' \
 --exclude='goldenPath/droYak1/*' \
 --exclude='goldenPath/anoGam1/*' \
 --exclude='goldenPath/ceMay2003/*' \
 --exclude='goldenPath/ce2/*' \
 --exclude='goldenPath/cbJul2002/*' \
 --exclude='goldenPath/sacCer1/*' \
 --exclude='goldenPath/scApr2003/*' \
 --exclude='goldenPath/hgFixed/*' \
 --exclude='goldenPath/chromReports/*' \
 --exclude='goldenPath/protein*' \
 --exclude='goldenPath/mapPlots/*' \
 --exclude='ancestors/*' \
 --exclude='zoo/*' \
 --exclude='ENCODE/*' \
 rsync://hgdownload.cse.ucsc.edu/genome/ $WEBROOT/
</PRE>
</OL>
<P>
<A NAME="step3"></A>
<B>3. Get the data for each individual genome assembly and installing databases ( 100 Gb )</B>

<OL type="a">

<LI>Get Human May, 2004 full data set (hg17: 7.5 Gb) by doing:
<P>
<PRE>
rsync --progress --partial -avz \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/hg17/database/ \
  $WEBROOT/goldenPath/hg17/
</PRE>

<LI>Get Human July, 2003 full data set (hg16: 9.5 Gb) by doing:
<P>
<PRE>
rsync --progress --partial -avz  \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/hg16/database/ \
  $WEBROOT/goldenPath/hg16/
</PRE>

<LI>Get Human April 10, 2003 full data set (hg15: 4.4 Gb) by doing:
<P>
<PRE>
rsync --progress --partial -avz  \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/10april2003/database/ \
  $WEBROOT/goldenPath/10april2003/
</PRE>

<LI>Get Chimp November, 2003 full data set (panTro1: 3.0 Gb) by doing:
<P>
<PRE>
rsync --progress --partial -avz \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/panTro1/database/ \
  $WEBROOT/goldenPath/panTro1/
</PRE>

<LI>Get Dog July, 2004 full data set (canFam1: 2.3 Gb) by doing:
<P>
<PRE>
rsync --progress --partial -avz \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/canFam1/database/ \
  $WEBROOT/goldenPath/canFam1/
</PRE>

<LI>Get Mouse May, 2004 full data set (mm5: 8.6 Gb) by doing:
<P>
<PRE>
rsync --progress --partial -avz \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/mm5/database/ \
  $WEBROOT/goldenPath/mm5/
</PRE>

<LI>Get Mouse October, 2003 full data set (mm4: 8.2 Gb) by doing:
<P>
<PRE>
rsync --progress --partial -avz  \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/mm4/database/ \
  $WEBROOT/goldenPath/mm4/
</PRE>

<LI>Get Rat June, 2003 full data set (rn3: 3.7 Gb) by doing:
<P>
<PRE>
rsync --progress --partial -avz \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/rnJun2003/database/ \
  $WEBROOT/goldenPath/rnJun2003/
</PRE>

<LI>Get Rat January, 2003 full data set (rn2: 537 Mb) by doing:
<P>
<PRE>
rsync --progress --partial -avz  \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/rnJan2003/database/ \
  $WEBROOT/goldenPath/rnJan2003/
</PRE>

<LI>Get Chicken February, 2004 full data set (galGal2: 755 Mb) by doing:
<P>
<PRE>
rsync --progress --partial -avz \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/galGal2/database/ \
  $WEBROOT/goldenPath/galGal2/
</PRE>

<LI>Get Zebrafish November, 2003 full data set (danRer1: 1.2 Gb) by doing:
<P>
<PRE>
rsync --progress --partial -avz \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/danRer1/database/ \
  $WEBROOT/goldenPath/danRer1/
</PRE>

<LI>Get C. intestinalis December, 2002 full data set (ci1: 179 Mb) by doing:
<P>
<PRE>
rsync --progress --partial -avz \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/ci1/database/ \
  $WEBROOT/goldenPath/ci1/
</PRE>

<LI>Get Fugu August, 2003 full data set (fr1: 149 Mb) by doing:
<P>
<PRE>
rsync --progress --partial -avz \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/fr1/database/ \
  $WEBROOT/goldenPath/fr1/
</PRE>

<LI>Get Fruitfly January, 2003 full data set (dm1: 299 Mb) by doing:
<P>
<PRE>
rsync --progress --partial -avz \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/dm1/database/ \
  $WEBROOT/goldenPath/dm1/
</PRE>

<LI>Get D. pseudoobscura August 2003 full data set (dp1: 15 Mb) by doing:
<P>
<PRE>
rsync --progress --partial -avz \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/dp1/database/ \
  $WEBROOT/goldenPath/dp1/
</PRE>

<LI>Get D. yakuba April 2004 full data set (droYak1: 127 Mb) by doing:
<P>
<PRE>
rsync --progress --partial -avz \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/droYak1/database/ \
  $WEBROOT/goldenPath/droYak1/
</PRE>

<LI>Get A. gambiae Feb 2003 full data set (anoGam1: 127 Mb) by doing:
<P>
<PRE>
rsync --progress --partial -avz \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/anoGam1/database/ \
  $WEBROOT/goldenPath/anoGam1/
</PRE>

<LI>Get C. elegans March, 2004 full data set (ce2: 562 Mb) by doing:
<P>
<PRE>
rsync --progress --partial -avz \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/ce2/database/ \
  $WEBROOT/goldenPath/ce2/
</PRE>

<LI>Get C. elegans May, 2003 full data set (ce1: 1.2 Gb) by doing:
<P>
<PRE>
rsync --progress --partial -avz \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/ceMay2003/database/ \
  $WEBROOT/goldenPath/ceMay2003/
</PRE>

<LI>Get C. briggsae July, 2002 full data set (cb1: 185 Mb) by doing:
<P>
<PRE>
rsync --progress --partial -avz \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/cbJul2002/database/ \
  $WEBROOT/goldenPath/cbJul2002/
</PRE>

<LI>Get Yeast (Saccharomyces cerevisiae) October, 2003 full data set (sacCer1: 58 Mb) by doing:
<P>
<PRE>
rsync --progress --partial -avz \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/sacCer1/database/ \
  $WEBROOT/goldenPath/sacCer1/
</PRE>

<LI>Get SARS coronavirus April, 2003 full data set (sc1: 12 Mb) by doing:
<P>
<PRE>
rsync --progress --partial -avz \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/scApr2003/database/ \
  $WEBROOT/goldenPath/scApr2003/
</PRE>

</OL>

<H4><A NAME="step4"></A>
4. Obtain the /gbdb data file area</H4>

<OL type="a">
<LI>You will need the portions of /gbdb used by the browser (400 Gb):
<BR>
<PRE>
rsync --progress --partial -avz rsync://hgdownload.cse.ucsc.edu/gbdb/ /
</PRE>
<BR>
</OL>

<H4><A NAME="step5"></A>
5. Set up database</H4>
<B>Existing Mirror Sites: There are numerous changes to tables
and data within these tables to support the /gbdb file storage method.
You are advised to consider dropping and rebuilding the databases.
If that presents a problem, contact us and we can provide detailed
updates to be made, but that is not directly documented.
</B>

<UL>
<P><A NAME="table"></A> Use the following table to identify the freeze
date ($FREEZEDATE) and the database version ($DBVERSION) for each genome
assembly:
<P>
<TABLE CELLPADDING=0 CELLSPACING=0 MARGINS=0>
<TR><TD> $FREEZEDATE </TD><TD WIDTH=10></TD><TD> $DBVERSION </TD></TR>
<TR><TD> --------------- </TD><TD></TD><TD> ---------------  </TD></TR>
<TR><TD> hg17        </TD><TD></TD><TD> hg17        </TD></TR>
<TR><TD> hg16        </TD><TD></TD><TD> hg16        </TD></TR>
<TR><TD> 10april2003 </TD><TD></TD><TD> hg15        </TD></TR>
<TR><TD> panTro1     </TD><TD></TD><TD> panTro1     </TD></TR>
<TR><TD> canFam1     </TD><TD></TD><TD> canFam1     </TD></TR>
<TR><TD> mm5         </TD><TD></TD><TD> mm5         </TD></TR>
<TR><TD> mm4         </TD><TD></TD><TD> mm4         </TD></TR>
<TR><TD> rnJun2003   </TD><TD></TD><TD> rn3         </TD></TR>
<TR><TD> rnJan2003   </TD><TD></TD><TD> rn2         </TD></TR>
<TR><TD> galGal2     </TD><TD></TD><TD> galGal2     </TD></TR>
<TR><TD> danRer1     </TD><TD></TD><TD> danRer1     </TD></TR>
<TR><TD> fr1         </TD><TD></TD><TD> fr1         </TD></TR>
<TR><TD> ci1         </TD><TD></TD><TD> ci1         </TD></TR>
<TR><TD> dm1         </TD><TD></TD><TD> dm1         </TD></TR>
<TR><TD> dp1         </TD><TD></TD><TD> dp1         </TD></TR>
<TR><TD> droYak1     </TD><TD></TD><TD> droYak1     </TD></TR>
<TR><TD> anoGam1     </TD><TD></TD><TD> anoGam1     </TD></TR>
<TR><TD> ce2         </TD><TD></TD><TD> ce2         </TD></TR>
<TR><TD> ceMay2003   </TD><TD></TD><TD> ce1         </TD></TR>
<TR><TD> cbJul2002   </TD><TD></TD><TD> cb1         </TD></TR>
<TR><TD> sacCer1     </TD><TD></TD><TD> sacCer1         </TD></TR>
<TR><TD> scApr2003   </TD><TD></TD><TD> sc1         </TD></TR>
</TABLE>
</UL>

<OL type="a">
<LI>After connecting to the MySQL server, create a database called "$DBVERSION"
corresponding to $FREEZEDATE in the <A HREF="#table">table</A> above.
<PRE>
mysql&gt; create database $DBVERSION;
</PRE>
<LI>Create tables for the "$DBVERSION" database.
<UL type="i">
<LI> Move to $WEBROOT/goldenPath/$FREEZEDATE/database/ directory.<BR>
All the files for creating the "$DBVERSION" database are in this directory.

<LI>Create and load data into the tables: (The script here is an example
for this process.  This takes a long time, 10 hours or more for one database.)
<PRE>
#!/bin/sh
DB=$FREEZEDATE
cd $WEBROOT/goldenPath/${DB}/database
for SQL in *.sql
do
    T_NAME=${SQL%%.sql}
    echo "loading table ${T_NAME}"
    mysql -youraccountoptions -e "DROP TABLE ${T_NAME};" ${DB} > /dev/null 2> /dev/null
    mysql -youraccountoptions ${DB} < ${SQL}
    zcat "${T_NAME}.txt.gz" | mysql -youraccountoptions -e \
        "LOAD DATA LOCAL INFILE \"/dev/stdin\" INTO TABLE ${T_NAME};" ${DB}
done
</PRE>
</UL>

<LI>Issue permission to the "$DBVERSION" database.<BR>
Permission could be issued as follows: <BR>
<PRE>
mysql&gt; grant select on $DBVERSION.* to $USERNAME@$HOSTNAME identified by "$PASSWORD";
</PRE>
<BR>
To make this command work, you need to set up $USERNAME,$HOSTNAME and
$PASSWORD properly. In this example, public access to the "$DBVERSION"
database is restricted to read-only.
</OL>

<h4><a NAME="step6"></A>6. Get executable files</h4>
<OL type="a">

<LI>The cgi-bin binaries are available via:
<BR>
<PRE>
rsync --progress --partial -avz rsync://hgdownload.cse.ucsc.edu/cgi-bin/ $CGIBIN/
</PRE>
<BR>

<LI>In the CGIBIN directory, make a <A HREF="sample.html">hg.conf</A> file, in which, $USERNAME,$HOSTNAME and $PASSWORD used for the "$DBVERSION" database should be specified. Remember, variables should not be used in this file.
<BR><BR>

<LI>Set up environment variables needed by the Makefile to install the binaries for the website in the correct place.
<PRE>
setenv GLOBAL_CONFIG_FILE $CGIBIN/hg.conf
setenv HGCGI $CGIBIN
</PRE>
<LI>Download the released zipped version of the source files from <A HREF="/admin/jksrc.zip">here</A>.
<BR>
Follow the <A HREF="jk-install.html">instructions</A> for how to compile
the files. <BR>
The source tree can also be obtained via <A HREF="cvs.html"> CVS </A>
</OL>

<B>Existing Mirror Sites: Note the Schema for Hgcentral HAS CHANGED.
You are advised to drop the hgcentral database and reload it.
This will result in the loss of user settings. If this is a problem,
contact us for an incremental fix.
</B>
<P>

<H4><A NAME="step7"></A>
7. Setup the "hgcentral" tables </H4>

<OL type="a">
<LI>Download the schema for the hgcentral database <A HREF="hgcentral.sql"> here</A>.
<LI>Create a hgcentral database <P>
<PRE>
   mysql&gt; create database hgcentral
</PRE>
<P>
<LI>Add the hgcentral tables <P>
<PRE>
   mysql -youraccountoptions hgcentral < hgcentral.sql
</PRE>
<LI>Create a user/password with the ability to update and insert. Many folks
 start out with the same user that they use for reading browser databases
 and then later create a separate user with higher privilege. However, to
 get started, consider trying the same user.
<LI>Add that user to the hg.conf. A sample hg.conf is below:
<PRE>
# Config file for the UCSC Human Genome server
#
# the format is in the form of name/value pairs
# written 'name=value' (note that there is no space between
# the name and its value. Also, no blank lines should be in
# this file
###########################################################
# db.host is the name of the MySQL host to connect to
db.host=localhost
# db.user is the username is use when connecting to the host
db.user=myhguser
# this is the password to use with the above hostname
db.password=myhguserpassword
db.trackDb=trackDb

# central.host is the name of the host of the central MySQL
# database where stuff common to all versions of the genome
# and the user database is stored.
central.db=hgcentral
central.host=localhost
#
# Be sure this user has UPDATE AND INSERT privs for hgcentral
#
central.user=myhguser
central.password=myhguserpassword
central.domain=.mydomain.edu
</PRE>
</OL>

<H4><A NAME="step8"></A>
8. Create a "trash" directory </H4>
The cgi programs use a temporary area to create and store images
used by the browser. This directory is by default looked for in
<em>$WEBROOT/trash</em>. You should make this directory and allow the user
that runs the web server write access to it.

<H4><a NAME="step9"></A>
9. Create the hgFixed database</H4>

<OL type="a">
<LI>Download a copy of the dumped hgFixed tables.  (500 Mb)
<PRE>
rsync --progress --partial -avz \
   rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/hgFixed/ \
   $WEBROOT/goldenPath/
</PRE>
<LI>Create a hgFixed database</LI>
<PRE>
mysql&gt; create database hgFixed
</PRE>
<LI>Import the data in a similar fashion as listed above.</LI>
</OL>
<P>
<H4><a NAME="step10"></A>
10. Create the five protein databases</H4>

<OL type="a">
<LI>Download a copy of the dumped protein tables. (700 Mb)
<PRE>
rsync --progress --partial -avz  \
    rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/proteinDB/ \
    $WEBROOT/goldenPath/proteinDB/
</PRE>
<LI>Create the protein databases
<PRE>
mysql&gt; create database proteins092903
mysql&gt; create database proteins040115
mysql&gt; create database proteins040315
mysql&gt; create database proteins040515
</PRE>
<LI>Import the data from proteinDB/proteins*/database in a similar
	fashion as listed above.
</OL>

<H4><a NAME="step11"></A>
11. Create the swissProt database</H4>

<OL type="a">
<LI>Please note <A HREF="/goldenPath/swissProt/database/README.txt">
	Usage Rights </A> for the swissProt database.

<LI>Download a copy of the dumped swissProt tables. (400 Mb)
<PRE>
rsync --progress --partial -avz  \
    rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/swissProt/ \
    $WEBROOT/goldenPath/
</PRE>
<LI>Create the swissProt database
<PRE>
mysql&gt; create database swissProt
</PRE>
<LI>Import the data in a similar fashion as listed above.
</OL>

<hr WIDTH="100%">
<BR><em>Should you have any comments or questions, please contact
<A HREF="mailto:&#103;&#101;n&#111;&#109;&#101;&#45;mi&#114;&#114;&#111;&#114;&#64;&#99;&#115;&#101;.&#117;&#99;&#115;&#99;.&#101;du">
&#103;&#101;n&#111;&#109;&#101;&#45;mi&#114;&#114;&#111;&#114;&#64;&#99;&#115;&#101;.&#117;&#99;&#115;&#99;.&#101;du</A>
</em>.
<!-- above address is genome-mirror at cse.ucsc.edu -->
<BR><em>Last Modified: 2004-05-19 - 19 May 2004</em>
<BR>
			</TD>
		    <TD WIDTH=15></TD></TR>
		</TABLE>
	    <BR>
	    </TD></TR>
	</TABLE>
<!-- END SECOND SECTION ---------------------------------------------------------->

    </TD></TR>
</TABLE>
</BODY></HTML>
