<HTML>
  <HEAD>
    <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
    <META name="description" content="UCSC Genome Browser">
    <META name="keywords" content="genome, genome browser, genome assembly, Blat, UCSC, mirror, genome browser source, bioinformatics, gene prediction, SNP, EST, mRNA">
    <TITLE>UCSC Genome Browser: Mirror Instructions</TITLE>
    <LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
  </HEAD>

  <BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
  <A NAME="TOP"></A>

    <TABLE BORDER=0 WIDTH="100%">
      <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="/images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

      <TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/topbar.html"--></TD></TR>

<!--Content Tables------------------------------------------------------->
    <TR><TD CELLPADDING=10>	
	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1>
	    <TR><TD>
	    	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
		    <TR><TD>
			<FONT SIZE="4"><B>&nbsp; Procedure for Creating a Mirror Site for the UCSC Genome Browser</B></FONT>
		    </TD></TR>
		</TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
		    <TR><TH HEIGHT=10></TH></TR>
		    <TR><TD WIDTH=10>&nbsp;</TD>
			<TD>
			    <P>
			    The following procedure provides you with step-by-step
			    instructions to incrementally create a mirror of the
			    UCSC Genome Browser. You may choose to set up either a
			    full mirror browser or a partial one, depending on your disk
			    space and needs.
			    <P>
			    <B> Prerequisites:</B>
			    <P>
			    To set up a full mirror browser, we recommend at a
			    MINIMUM the following disk partitions and sizes:
			    <UL>
			        <LI>MySQL Table spaces: ~ 400 Gb
			        <LI>Apache web pages area: ~ 270 Mb
			        <LI>"/gbdb" area for the browser: ~ 400 Gb
			        <LI>Genome database text dumps: ~ 100 Gb
			    </UL></P><P>
				These data requirements will increase
				in the future as genomes are added to
				the system.</P>
			    <P>
			    Mirror site questions may be directed to the 
			    mailing list <A HREF="mailto:&#103;en&#111;&#109;e&#45;&#109;&#105;&#114;&#114;&#111;&#114;&#64;&#115;&#111;&#101;.u&#99;&#115;c.e&#100;&#117;">&#103;en&#111;&#109;e&#45;&#109;&#105;&#114;&#114;&#111;&#114;&#64;&#115;&#111;&#101;.u&#99;&#115;c.e&#100;&#117;</A>.
			    <!-- above address is genome-mirror at soe.ucsc.edu -->
			    Messages sent to this address will be posted to the 
			    moderated genome-mirror mailing list, which is 
			    archived on a public Web-accessible pipermail 
			    archive.  This archive may be indexed by non-UCSC 
			    sites such as Google. 
			    <P>
			    <A HREF="http://www.soe.ucsc.edu/mailman/listinfo/genome-mirror"
			    TARGET=_blank>Subscribe</A> to the genome-mirror 
		 	    mailing list. </P>
			    <P>
			    <FORM name="googleForm2" method="GET" action="http://www.google.com/search" onSubmit="document.googleForm2.q.value=document.googleForm2.qq.value+'   site:www.soe.ucsc.edu/pipermail/genome-mirror';">
		  	      Search the Genome-mirror mailing list archives: &nbsp;
		  	      <input type="hidden" name="q" value="">
		  	      <input type="hidden" name="num" value="10">
		  	      <input type="hidden" name="filter" value="0">
		  	      <INPUT TYPE=text name=qq size=30 maxlength=255 value="">
		  	      <input TYPE="submit" VALUE="Submit">
			    </FORM></P>
			</TD><TD WIDTH=15></TD>
		    </TR>
		</TABLE>
		<BR>
	    </TD></TR>
	</TABLE>
	
<!--START SECOND SECTION ------------------------------------------------------->
	<BR>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	    <TR><TD>
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
		    <TR><TD>
			<FONT SIZE="4"><B>&nbsp;  Step-by-Step Details</B></FONT>
		    </TD></TR>
		</TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
		    <TR><TH HEIGHT=10></TH></TR>
		    <TR><TD WIDTH=10>&nbsp;</TD>
			<TD>

<OL>
<LI><A HREF="#step1">Install Apache server and MySQL server</A>
<LI><A HREF="#step2">Get all the html files and most of the text files</A>&nbsp;(270 Mb)
<LI><A HREF="#step3">Get the data for each individual genome assembly and install
databases </A>&nbsp;(100 Gb)
<LI><A HREF="#step4">Obtain the /gbdb data file area</A>&nbsp;(400 Gb)
<LI><A HREF="#step5">Set up database</A>&nbsp;(400 Gb MySQL data)
<LI><A HREF="#step6">Get executable files</A>
<LI><A HREF="#step7">Set up Hgcentral session preference table</A>
<LI><A HREF="#step8">Create a $WEBROOT/trash apache user writable directory for temp files</A>
<LI><A HREF="#step9">Set up HgFixed database</A>&nbsp;(500 Mb)
<LI><A HREF="#step10">Set up the protein databases</A>&nbsp;(700 Mb)
<LI><A HREF="#step11">Set up the UniProt database</A>&nbsp;(400 Mb)
</OL>
<BR><BR>
<HR>
<P>
<A NAME="step1"></A>
<B>1. Install Apache server and MySQL server</B>
<OL type="a">
<LI>
We will not provide consulting on the operation or configuration
of Apache or MySQL. If you are not familiar with the setup of
Apache or MySQL, you will have to find someone at your site that
is. We use a reference platform of Redhat 9 for all the steps
described here.
<BR>
<LI>
Beginning early March 2005, the static html pages for the browser expect
Apache to be configured with the XbitHack option enabled.
<PRE>
#
# Executable files will be processed for SSI
#
XBitHack on
</PRE>
<LI>
Find the location of your web pages. For example, on a Redhat Linux
system using a stock apache RPM, the web pages are stored in
<em>/var/www/html</em>. We will refer to this directory as
WEBROOT in this document and you should substitute your real
path for WEBROOT whenever you see that in our write-up. But, before
doing this, you <B>MUST</B> be aware of the following:
<BR>
<UL type="DISC">
<LI>
<B>THE FOLLOWING ASSUMES YOU ARE NOT ALREADY SERVING UP
DATA USING YOUR APACHE SERVER. BEAR IN MIND THE SELECTION OF
THE WEBROOT WILL OVERWRITE ANY EXISTING DATA IN THAT AREA.</B>
</UL>
<LI>
Find the location of your <em>cgi-bin</em> directory. It should be under
the parent directory of the WEBROOT directory. For example, on
a Redhat Linux system using a stock apache RPM, the path of
the <em>cgi-bin</em> directory is <em>/var/www/cgi-bin</em>. We will refer to
this directory as CGI_BIN in this document and you should substitute your
real path for CGI_BIN whenever you see that.
<BR>
<LI>
Next, find the location of your MySQL data. For example, on a
Redhat Linux system using stock RPM's, this is located in
<em>/var/lib/mysql</em>. We will refer to this directory as
MYSQLDATA in this document and you should substitute
your real path for MYSQLDATA whenever you see that.
<BR><BR>
</OL>
<P>
<A NAME="step2"></A>
<B>2. Get all the html files and most of the text files
(1.0 Gb)</B>
<OL type="a">
<LI>
Obtain the software package rsync. If it isn't already
installed on your system, obtain it from
<A HREF="http://rsync.samba.org">http://rsync.samba.org</A>
<BR>
<LI> Test the rsync connection:
<BR>
<PRE>
rsync -navz --progress rsync://hgdownload.cse.ucsc.edu
</PRE>
This should respond with:
<PRE>
genome          UCSC Human Genome Downloads (111 GB)
sars            UCSC Human Genome SARS Downloads (215MB)
htdocs          UCSC Human Genome Web Site Htdocs (28MB)
goldenPath      UCSC Human Golden Path Downloads (170 GB)
cgi-bin         UCSC Human Genome Web Site CGI Binaries
gbdb            UCSC Human Genome Browser New Config Files (313 GB)
archives        UCSC Human Genome Browser Archived Config Files
mysql           UCSC Human Genome Raw Mysql Tables (356 GB)
</PRE>
<BR>
<LI>Determine the destination of the copy ($WEBROOT) and fire off the production copy (270 Mb)<BR>
<em>The trailing slash is important!</em><BR>
<PRE>
rsync -avzP rsync://hgdownload.cse.ucsc.edu/htdocs/ $WEBROOT/
</PRE>
</OL>
<P>
<A NAME="step3"></A>
<B>3. Get the data for each individual genome assembly and installing databases ( 100 Gb )</B>

<OL type="a">

<LI>Get Human March 2006 full data set (hg18: 13.0 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/hg18/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/hg18/database/ \
  $WEBROOT/goldenPath/hg18/database/
</PRE>

<LI>Get Human May, 2004 full data set (hg17: 7.5 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/hg17/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/hg17/database/ \
  $WEBROOT/goldenPath/hg17/database/
</PRE>

<LI>Get Human July, 2003 full data set (hg16: 9.5 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/hg16/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/hg16/database/ \
  $WEBROOT/goldenPath/hg16/database/
</PRE>

<LI>Get Human April 10, 2003 full data set (hg15: 4.4 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/10april2003/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/10april2003/database/ \
  $WEBROOT/goldenPath/10april2003/database/
</PRE>

<LI>Get Chimp March, 2006 full data set (panTro2: 6.0 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/panTro2/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/panTro2/database/ \
  $WEBROOT/goldenPath/panTro2/database/
</PRE>

<LI>Get Chimp November, 2003 full data set (panTro1: 3.0 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/panTro1/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/panTro1/database/ \
  $WEBROOT/goldenPath/panTro1/database/
</PRE>

<LI>Get Rhesus January, 2006 full data set (rheMac2: 3.2 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/rheMac2/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/rheMac2/database/ \
  $WEBROOT/goldenPath/rheMac2/database/
</PRE>

<LI>Get Dog May, 2005 full data set (canFam2: 3.4 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/canFam2/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/canFam2/database/ \
  $WEBROOT/goldenPath/canFam2/database/
</PRE>

<LI>Get Dog July, 2004 full data set (canFam1: 2.3 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/canFam1/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/canFam1/database/ \
  $WEBROOT/goldenPath/canFam1/database/
</PRE>

<LI>Get Cow March 2005 full data set (bosTau2: 2.1 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/bosTau2/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/bosTau2/database/ \
  $WEBROOT/goldenPath/bosTau2/database/
</PRE>

<LI>Get Cow September 2004 full data set (bosTau1: 1.8 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/bosTau1/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/bosTau1/database/ \
  $WEBROOT/goldenPath/bosTau1/database/
</PRE>

<LI>Get Mouse February, 2006 full data set (mm8: 8.5 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/mm8/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/mm8/database/ \
  $WEBROOT/goldenPath/mm8/database/
</PRE>

<LI>Get Mouse August, 2005 full data set (mm7: 7.5 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/mm7/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/mm7/database/ \
  $WEBROOT/goldenPath/mm7/database/
</PRE>

<LI>Get Mouse March, 2005 full data set (mm6: 15 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/mm6/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/mm6/database/ \
  $WEBROOT/goldenPath/mm6/database/
</PRE>

<LI>Get Mouse May, 2004 full data set (mm5: 8.6 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/mm5/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/mm5/database/ \
  $WEBROOT/goldenPath/mm5/database/
</PRE>

<LI>Get Rat November, 2004 full data set (rn4: 7.1 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/rn4/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/rn4/database/ \
  $WEBROOT/goldenPath/rn4/database/
</PRE>

<LI>Get Rat June, 2003 full data set (rn3: 8.4 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/rnJun2003/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/rnJun2003/database/ \
  $WEBROOT/goldenPath/rnJun2003/database/
</PRE>

<LI>Get Rat January, 2003 full data set (rn2: 537 Mb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/rnJan2003/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/rnJan2003/database/ \
  $WEBROOT/goldenPath/rnJan2003/database/
</PRE>

<LI>Get Chicken May, 2006 full data set (galGal3: 1.4 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/galGal3/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/galGal3/database/ \
  $WEBROOT/goldenPath/galGal3/database/
</PRE>

<LI>Get Chicken February, 2004 full data set (galGal2: 989 Mb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/galGal2/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/galGal2/database/ \
  $WEBROOT/goldenPath/galGal2/database/
</PRE>

<LI>Get Opossum January, 2006 full data set (monDom1: 5.4 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/monDom4/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/monDom4/database/ \
  $WEBROOT/goldenPath/monDom4/database/
</PRE>

<LI>Get Opossum October, 2004 full data set (monDom1: 1.9 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/monDom1/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/monDom1/database/ \
  $WEBROOT/goldenPath/monDom1/database/
</PRE>

<LI>Get X. tropicalis August, 2005 full data set (xenTro2: 3.2 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/xenTro2/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/xenTro2/database/ \
  $WEBROOT/goldenPath/xenTro2/database/
</PRE>

<LI>Get X. tropicalis October, 2004 full data set (xenTro1: 1.8 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/xenTro1/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/xenTro1/database/ \
  $WEBROOT/goldenPath/xenTro1/database/
</PRE>

<LI>Get Zebrafish March, 2006 full data set (danRer4: 1.9 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/danRer4/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/danRer4/database/ \
  $WEBROOT/goldenPath/danRer4/database/
</PRE>

<LI>Get Zebrafish May, 2005 full data set (danRer3: 3.1 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/danRer3/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/danRer3/database/ \
  $WEBROOT/goldenPath/danRer3/database/
</PRE>

<LI>Get Zebrafish June, 2004 full data set (danRer2: 1.2 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/danRer2/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/danRer2/database/ \
  $WEBROOT/goldenPath/danRer2/database/
</PRE>

<LI>Get Tetraodon February, 2004 full data set (tetNig1: 694 Mb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/tetNig1/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/tetNig1/database/ \
  $WEBROOT/goldenPath/tetNig1/database/
</PRE>

<LI>Get Fugu August, 2002 full data set (fr1: 449 Mb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/fr1/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/fr1/database/ \
  $WEBROOT/goldenPath/fr1/database/
</PRE>

<LI>Get C. intestinalis March, 2005 full data set (ci2: 217 Mb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/ci2/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/ci2/database/ \
  $WEBROOT/goldenPath/ci2/database/
</PRE>

<LI>Get C. intestinalis December, 2002 full data set (ci1: 179 Mb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/ci1/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/ci1/database/ \
  $WEBROOT/goldenPath/ci1/database/
</PRE>

<LI>Get S. purpuratus April, 2005 full data set (strPur1: 723 Mb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/strPur1/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/strPur1/database/ \
  $WEBROOT/goldenPath/strPur1/database/
</PRE>

<LI>Get Fruitfly January, 2003 full data set (dm2: 299 Mb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/dm2/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/dm2/database/ \
  $WEBROOT/goldenPath/dm2/database/
</PRE>

<LI>Get Fruitfly January, 2003 full data set (dm1: 299 Mb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/dm1/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/dm1/database/ \
  $WEBROOT/goldenPath/dm1/database/
</PRE>

<LI>Get D. yakuba April 2003 full data set (droYak1: 15 Mb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/droYak1/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/droYak1/database/ \
  $WEBROOT/goldenPath/droYak1/database/
</PRE>

<LI>Get D. ananassae July 2004 full data set (droAna1: x Mb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/droAna1/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/droAna1/database/ \
  $WEBROOT/goldenPath/droAna1/database/
</PRE>

<LI>Get D. pseudoobscura August 2003 full data set (dp2: 130 Mb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/dp2/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/dp2/database/ \
  $WEBROOT/goldenPath/dp2/database/
</PRE>

<LI>Get D. virilis July 2004 full data set (droVir1: x Mb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/droVir1/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/droVir1/database/ \
  $WEBROOT/goldenPath/droVir1/database/
</PRE>

<LI>Get D. mojavensis August 2004 full data set (droMoj1: x Mb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/droMoj1/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/droMoj1/database/ \
  $WEBROOT/goldenPath/droMoj1/database/
</PRE>

<LI>Get A. mellifera January 2005 full data set (apiMel2: x Mb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/apiMel2/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/apiMel2/database/ \
  $WEBROOT/goldenPath/apiMel2/database/
</PRE>

<LI>Get A. mellifera July 2004 full data set (apiMel1: x Mb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/apiMel1/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/apiMel1/database/ \
  $WEBROOT/goldenPath/apiMel1/database/
</PRE>

<LI>Get A. gambiae Feb 2003 full data set (anoGam1: 127 Mb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/anoGam1/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/anoGam1/database/ \
  $WEBROOT/goldenPath/anoGam1/database/
</PRE>

<LI>Get C. elegans March, 2004 full data set (ce2: 562 Mb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/ce2/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/ce2/database/ \
  $WEBROOT/goldenPath/ce2/database/
</PRE>

<LI>Get C. elegans May, 2003 full data set (ce1: 1.2 Gb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/ceMay2003/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/ceMay2003/database/ \
  $WEBROOT/goldenPath/ceMay2003/database/
</PRE>

<LI>Get C. briggsae July, 2002 full data set (cb1: 185 Mb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/cbJul2002/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/cbJul2002/database/ \
  $WEBROOT/goldenPath/cbJul2002/database/
</PRE>

<LI>Get Yeast (Saccharomyces cerevisiae) October, 2003 full data set (sacCer1: 58 Mb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/sacCer1/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/sacCer1/database/ \
  $WEBROOT/goldenPath/sacCer1/database/
</PRE>

<LI>Get SARS coronavirus April, 2003 full data set (sc1: 12 Mb) by doing:
<P>
<PRE>
mkdir -p $WEBROOT/goldenPath/scApr2003/database/
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/scApr2003/database/ \
  $WEBROOT/goldenPath/scApr2003/database/
</PRE>

</OL>

<H4><A NAME="step4"></A>
4. Obtain the /gbdb data file area</H4>

<OL type="a">
<LI>You will need the portions of /gbdb used by the browser (400 Gb):
<BR>
<PRE>
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/gbdb/ /gbdb/
</PRE>
<BR>
</OL>

<H4><A NAME="step5"></A>
5. Set up database</H4>

<UL>
<P><A NAME="table"></A> Use the following table to identify the freeze
date ($FREEZEDATE) and the database version ($DBVERSION) for each genome
assembly:
<P>
<TABLE CELLPADDING=0 CELLSPACING=0 MARGINS=0>
<TR><TD> $FREEZEDATE </TD><TD WIDTH=10></TD><TD> $DBVERSION </TD></TR>
<TR><TD> --------------- </TD><TD></TD><TD> ---------------  </TD></TR>
<TR><TD> hg18        </TD><TD></TD><TD> hg18        </TD></TR>
<TR><TD> hg17        </TD><TD></TD><TD> hg17        </TD></TR>
<TR><TD> hg16        </TD><TD></TD><TD> hg16        </TD></TR>
<TR><TD> 10april2003 </TD><TD></TD><TD> hg15        </TD></TR>
<TR><TD> panTro2     </TD><TD></TD><TD> panTro2     </TD></TR>
<TR><TD> panTro1     </TD><TD></TD><TD> panTro1     </TD></TR>
<TR><TD> canFam2     </TD><TD></TD><TD> canFam2     </TD></TR>
<TR><TD> canFam1     </TD><TD></TD><TD> canFam1     </TD></TR>
<TR><TD> bosTau2     </TD><TD></TD><TD> bosTau2     </TD></TR>
<TR><TD> bosTau1     </TD><TD></TD><TD> bosTau1     </TD></TR>
<TR><TD> mm8         </TD><TD></TD><TD> mm8         </TD></TR>
<TR><TD> mm7         </TD><TD></TD><TD> mm7         </TD></TR>
<TR><TD> mm6         </TD><TD></TD><TD> mm6         </TD></TR>
<TR><TD> mm5         </TD><TD></TD><TD> mm5         </TD></TR>
<!--
<TR><TD> mm4         </TD><TD></TD><TD> mm4         </TD></TR>
<TR><TD> mmFeb2003   </TD><TD></TD><TD> mm3         </TD></TR>
-->
<TR><TD> rn4         </TD><TD></TD><TD> rn4         </TD></TR>
<TR><TD> rnJun2003   </TD><TD></TD><TD> rn3         </TD></TR>
<TR><TD> rnJan2003   </TD><TD></TD><TD> rn2         </TD></TR>
<TR><TD> galGal3     </TD><TD></TD><TD> galGal3     </TD></TR>
<TR><TD> galGal2     </TD><TD></TD><TD> galGal2     </TD></TR>
<TR><TD> monDom4     </TD><TD></TD><TD> monDom4     </TD></TR>
<TR><TD> monDom1     </TD><TD></TD><TD> monDom1     </TD></TR>
<TR><TD> xenTro2     </TD><TD></TD><TD> xenTro2     </TD></TR>
<TR><TD> xenTro1     </TD><TD></TD><TD> xenTro1     </TD></TR>
<TR><TD> danRer4     </TD><TD></TD><TD> danRer4     </TD></TR>
<TR><TD> danRer3     </TD><TD></TD><TD> danRer3     </TD></TR>
<TR><TD> danRer2     </TD><TD></TD><TD> danRer2     </TD></TR>
<TR><TD> danRer1     </TD><TD></TD><TD> danRer1     </TD></TR>
<TR><TD> tetNig1     </TD><TD></TD><TD> tetNig1     </TD></TR>
<TR><TD> fr1         </TD><TD></TD><TD> fr1         </TD></TR>
<TR><TD> ci1         </TD><TD></TD><TD> ci1         </TD></TR>
<TR><TD> dm2         </TD><TD></TD><TD> dm2         </TD></TR>
<TR><TD> dm1         </TD><TD></TD><TD> dm1         </TD></TR>
<TR><TD> droYak1     </TD><TD></TD><TD> droYak1     </TD></TR>
<TR><TD> droAna1     </TD><TD></TD><TD> droAna1     </TD></TR>
<TR><TD> dp2         </TD><TD></TD><TD> dp2         </TD></TR>
<TR><TD> droVir1     </TD><TD></TD><TD> droVir1     </TD></TR>
<TR><TD> droMoj1     </TD><TD></TD><TD> droMoj1     </TD></TR>
<TR><TD> apiMel2     </TD><TD></TD><TD> apiMel2     </TD></TR>
<TR><TD> apiMel1     </TD><TD></TD><TD> apiMel1     </TD></TR>
<TR><TD> anoGam1     </TD><TD></TD><TD> anoGam1     </TD></TR>
<TR><TD> ce2         </TD><TD></TD><TD> ce2         </TD></TR>
<TR><TD> ceMay2003   </TD><TD></TD><TD> ce1         </TD></TR>
<TR><TD> cbJul2002   </TD><TD></TD><TD> cb1         </TD></TR>
<TR><TD> sacCer1     </TD><TD></TD><TD> sacCer1         </TD></TR>
<TR><TD> scApr2003   </TD><TD></TD><TD> sc1         </TD></TR>
</TABLE>
</UL>

<OL type="a">
<LI>After connecting to the MySQL server, create a database called "$DBVERSION"
corresponding to $FREEZEDATE in the <A HREF="#table">table</A> above.
<PRE>
mysql&gt; create database $DBVERSION;
</PRE>
<LI>Create tables for the "$DBVERSION" database.
<UL type="i">
<LI> Move to $WEBROOT/goldenPath/$FREEZEDATE/database/ directory.<BR>
All the files for creating the "$DBVERSION" database are in this directory.

<LI>Create and load data into the tables: (The script here is an example
for this process.<BR>
This takes a long time, 10 hours or more for one database.)<BR>
Optionally, you can download the MySQL tables via rsync directly into
your <em>/var/lib/mysql/</em> database hierarchy.<BR>
Note the rsync source location for these directories is
<em>rsync://hgdownload.cse.ucsc.edu/mysql/</em><BR>
You can see a list of directories to fetch with the command:
<em> rsync -v --dry-run rsync://hgdownload.cse.ucsc.edu/mysql/ </em>
<PRE>
#!/bin/sh
DB=$FREEZEDATE
cd $WEBROOT/goldenPath/${DB}/database
for SQL in *.sql
do
    T_NAME=${SQL%%.sql}
    echo "loading table ${T_NAME}"
    mysql -youraccountoptions -e "DROP TABLE ${T_NAME};" ${DB} > /dev/null 2> /dev/null
    mysql -youraccountoptions ${DB} < ${SQL}
    zcat "${T_NAME}.txt.gz" | mysql -youraccountoptions --local-infile=1 \
        -e "LOAD DATA LOCAL INFILE \"/dev/stdin\" INTO TABLE ${T_NAME};" ${DB}
done
</PRE>
</UL>

<LI>Issue permission to the "$DBVERSION" database.<BR>
Permission could be issued as follows: <BR>
<PRE>
mysql&gt; grant select on $DBVERSION.* to $USERNAME@$HOSTNAME identified by "$PASSWORD";
</PRE>
<BR>
To make this command work, you need to set up $USERNAME,$HOSTNAME and
$PASSWORD properly. In this example, public access to the "$DBVERSION"
database is restricted to read-only.
</OL>

<h4><a NAME="step6"></A>6. Get executable files</h4>
<OL type="a">

<LI>Pre-compiled Red-Hat (2.6.12-1.1381_FC3smp) AMD Opteron x86_64 64-bit
binaries can be fetched with the rsync command:<BR><BR>
<TT> rsync -avzP rsync://hgdownload.cse.ucsc.edu/cgi-bin/ $CGI_BIN/ </TT><BR><BR>
There are a number of data files that are also used in this directory.
This rsync will fetch them all.  If you need i386 (x86) 32-bit binaries, please
use the following rsync in addition to and after the above rsync,
to replace the 64 bit binaries:<BR><BR>
<TT> rsync -avzP rsync://hgdownload.cse.ucsc.edu/cgi-bin-i386/ $CGI_BIN/ </TT>
<BR><BR>

<LI>In the CGI_BIN directory, make a <A HREF="sample.html">hg.conf</A> file, in which, $USERNAME,$HOSTNAME and $PASSWORD used for the "$DBVERSION" database should be specified. Remember, variables should not be used in this file.
<BR><BR>

<LI>Set up environment variables needed by the Makefile to install the binaries for the website in the correct place.
<PRE>
setenv GLOBAL_CONFIG_FILE $CGI_BIN/hg.conf
setenv HGCGI $CGI_BIN
</PRE>
<LI>Download the released zipped version of the source files from 
<A HREF="http://hgdownload.cse.ucsc.edu/admin/jksrc.zip">here</A>.
<BR>
Follow the <A HREF="jk-install.html">instructions</A> for how to compile
the files. <BR>
The source tree can also be obtained via <A HREF="cvs.html"> CVS </A>
</OL>

<P>

<A NAME="step7"></A>
<B>Existing Mirror Sites: Please note the change history for the
hgcentral database.  Table structure changes have corresponding
structure changes in the source code.  The browser version can
be seen in the title window decoration of the genome browser display
page.  Or the source tree file <em> src/hg/inc/versionInfo.h </em>
since v62.
</B>
<H5> Change history for hgcentral database tables: </H5>
<UL>
<LI> 2005-03-28 table liftOverChain structure change, several fields added.
	(browser version v102)
<LI> 2004-12-21 added two new tables: clade and genomeClade to support
	new pulldown menus in the gateway page.  (browser version v93)
<LI> 2004-08-02 table blatServers has canPcr field added
	(browser version v74)
<LI> 2004-06-01 table dbDb has hgNearOk, hgPbOk and sourceName fields added
	(browser version v66)
<LI> 2004-04-16 table liftOverChain added (browser version v60)
</UL>

<H4>
7. Set up the "hgcentral" tables </H4>

<OL type="a">
<LI>Download the schema for the hgcentral database <A HREF="http://hgdownload.cse.ucsc.edu/admin/hgcentral.sql"> here</A>.
<LI>Create a hgcentral database <P>
<PRE>
   mysql&gt; create database hgcentral
</PRE>
<P>
<LI>Add the hgcentral tables <P>
<PRE>
   mysql -youraccountoptions hgcentral < hgcentral.sql
</PRE>
<LI>Create a user/password with the ability to update and insert. Many folks
 start out with the same user that they use for reading browser databases
 and then later create a separate user with higher privilege. However, to
 get started, consider trying the same user.
<LI>Add that user to the hg.conf. A sample hg.conf is below:
<PRE>
###########################################################
# Config file for the UCSC Human Genome server.
#	This file specifies the host and user for MySQL
#	database access.
#
# the format is in the form of name/value pairs
# written 'name=value' (note that there is no space between
# the name and its value.)
###########################################################
# db.host is the name of the MySQL host to connect to
db.host=localhost
# db.user is the username is use when connecting to the host
#	This user only needs SELECT permissions for read-only access
db.user=myhguser
# this is the password to use with the above hostname
db.password=myhguserpassword
db.trackDb=trackDb
###########################################################
# central.host is the name of the host of the central MySQL
# database where items common to all versions of the genome
# and the user database is stored.  central.db is the name of
# the database to access on that host for this information.
# Typical table names in the central.db:
#	blatServers
#	dbDb
#	defaultDb
#	gdbPdb
#	liftOverChain
#	sessionDb
#	userDb
central.host=localhost
central.db=hgcentral
#
# The central.user needs SELECT, INSERT, UPDATE, DELETE, CREATE, DROP and ALTER
#	permissions for hgcentral to allow maintainence of the session
#	and user Db's
#
central.user=myhguser
central.password=myhguserpassword
central.domain=.mydomain.edu
###########################################################
# backupcentral is used when the primary central DB fails.
#	This can be identical to the central entries as above.
backupcentral.host=localhost
backupcentral.db=hgcentral
backupcentral.user=myhguser
backupcentral.password=myhguserpassword
backupcentral.domain=.mydomain.edu
</PRE>
</OL>

<H4><A NAME="step8"></A>
8. Create a "trash" directory </H4>
The cgi programs use a temporary area to create and store images
used by the browser. This directory is by default looked for in
<em>$WEBROOT/trash</em>. You should make this directory and allow the user
that runs the web server write access to it.

<H4><a NAME="step9"></A>
9. Create the hgFixed database</H4>

<OL type="a">
<LI>Download a copy of the dumped hgFixed tables.  (500 Mb)
<PRE>
rsync -avzP --delete --max-delete=20 \
   rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/hgFixed/ \
   $WEBROOT/goldenPath/hgFixed/
</PRE>
<LI>Create a hgFixed database</LI>
<PRE>
mysql&gt; create database hgFixed
</PRE>
<LI>Import the data in a similar fashion as listed above.</LI>
</OL>
<P>
<H4><a NAME="step10"></A>
10. Create the protein databases</H4>

<OL type="a">
<LI>Download a copy of the dumped protein tables. (700 Mb)
<PRE>
rsync -avzP --delete --max-delete=20 \
    rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/proteinDB/ \
    $WEBROOT/goldenPath/proteinDB/
</PRE>
<LI>Create the protein databases
<PRE>
mysql&gt; create database proteins040315
mysql&gt; create database proteins040515
mysql&gt; create database proteins050415
mysql&gt; create database proteins051015
mysql&gt; create database proteins060115
</PRE>
<LI>Import the data from proteinDB/proteins*/database in a similar
	fashion as listed above.
<LI>Create symlinks for your MySQL database of:
<PRE>
	/var/lib/mysql/proteome -&gt; proteins060115
</PRE>
</OL>

<H4><a NAME="step11"></A>
11. Create the UniProt database</H4>

<OL type="a">
<LI>Please note <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/swissProt/database/README.txt">usage rights</A> for the UniProt database.

<LI>Download a copy of the dumped UniProt tables. (400 Mb)
<PRE>
rsync -avzP --delete --max-delete=20 \
  rsync://hgdownload.cse.ucsc.edu/genome/goldenPath/uniProt/ \
  $WEBROOT/goldenPath/uniProt/
</PRE>
<LI>Create the UniProt database
<PRE>
mysql&gt; create database uniProt
</PRE>
<LI>Import the data in a similar fashion as listed above.
</OL>

<hr WIDTH="100%">
<BR><em>Should you have any comments or questions, please contact
<A HREF="mailto:&#103;&#101;n&#111;&#109;&#101;&#45;mi&#114;&#114;&#111;&#114;&#64;&#99;&#115;&#101;.&#117;&#99;&#115;&#99;.&#101;du">
&#103;&#101;n&#111;&#109;&#101;&#45;mi&#114;&#114;&#111;&#114;&#64;&#99;&#115;&#101;.&#117;&#99;&#115;&#99;.&#101;du</A>
</em>.
<!-- above address is genome-mirror at cse.ucsc.edu -->
<BR><em>This page last modified: <!--#echo var="LAST_MODIFIED"-->.</em>
<BR>
			</TD>
		    <TD WIDTH=15></TD></TR>
		</TABLE>
	    <BR>
	    </TD></TR>
	</TABLE>
<!-- END SECOND SECTION ---------------------------------------------------------->

    </TD></TR>
</TABLE>
</BODY></HTML>
