<H2>Description</H2>
<P>Bacterial artificial chromosomes (BACs) are a key part of many large
scale sequencing projects.   A BAC typically consists of 50 - 300 kb of
DNA.  For $organism, the BAC clones are in the range of 2 - 650 kb. During the 
early phase of a sequencing project, it is common
to sequence a single read (approximately 500 bases) off each end of
a large number of BACs.  Later on in the project, these BAC end reads
can be mapped to the genome sequence.  
</P>
<P>This track shows these mappings
in cases where both ends could be mapped.  These BAC end pairs can
be useful for validating the assembly over relatively long ranges.  In some
cases, the BACs are useful biological reagents.  This track can also be
used for determining which BAC contains a given gene, useful information
for certain wet lab experiments.

<P>A valid pair of BAC end sequences must be
at least 2 kb but no more than 800 kb away from each other. 
The orientation of the first BAC end sequence must be &quot+&quot; and
the orientation of the second BAC end sequence must be &quot;-&quot;.</P>

<H2>Methods</H2>
<P>BAC end sequences are placed on the assembled sequence using
Jim Kent's 
<A HREF="http://www.genome.org/cgi/content/abstract/12/4/656" TARGET=_blank>blat</A> 
program followed by pslReps using parameters such that only alignments with 
least 85% identity, a minimum sequence coverage of 60% and a base identity level
within 2% of the bestwere kept and there was no penalty for not having introns (<em>-nearTop=0.02 -minCover=0.60 -minAli=0.85 -noIntrons</em>). </P>

<H2>Credits</H2>
<P> Information about the libraries may be found on the Sanger Institute 
<A HREF="http://www.sanger.ac.uk/Projects/D_rerio/library_details.shtml" 
TARGET=_blank>Zebrafish Library Details</A> page. Additional information about 
some of the clones, including how they can be obtained, may be found at the
<A HREF="http://www.ncbi.nlm.nih.gov/genome/clone/index.html" 
TARGET=_blank>NCBI Clone Registry</A>.
To view the registry entry for a specific clone, open the details page for the 
clone and click on its name at the top of the page. Not all $organism BAC 
clones have been submitted to NCBI Clone Registry as of July 2004; therefore, 
some of the clone links to NCBI may not yet be active.
</P>
<P>
The $Organism BAC End Pairs track was produced at UCSC from BAC End Pairs 
sequence data obtained from 
<A HREF="http://www.eb.tuebingen.mpg.de/dept3/geisler/home.html" 
TARGET=_blank>Robert Geisler</A> at the Max Planck Institute for 
Developmental Biology, Germany and the 
<A HREF="http://www.niob.knaw.nl" TARGET=_blank>Netherlands Institute for 
Developmental Biology</A> (Hubrecht Laboratory). The track was produced in 
collaboration with the <A HREF="http://zon.tchlab.org"
TARGET=_blank>Zebrafish Genome Initiative</A> at Childrens Hospital, Boston.
GenBank accessions for BAC clones were obtained from The Wellcome Trust Sanger
Institute, Cambridge, United Kingdom.</P>
