<H2>Description</H2>
<P>Bacterial artificial chromosomes (BACs) are a key part of many large
scale sequencing projects.  A BAC typically consists of 50 - 300 kb of DNA.  
For the $organism assembly, the BACs around 150-200kb and they are used for 
both the FPC and RH maps. During the early phase of a sequencing project, it is common to sequence a single read (approximately 500 bases) off each end of a 
large number of BACs.  Later on in the project, these BAC end reads can be 
mapped to the genome sequence. After alignment of $organism BAC end sequences 
to this genome, the distances between pairs of BAC ends are in the range of
2 - 650 kb with the average distance being approximately 170 kb. 
</P>
<P>This track shows these mappings in cases where both ends could be mapped.  
These BAC end pairs can be useful for validating the assembly over relatively 
long ranges.  In some cases, the BACs are useful biological reagents.  This 
track can also be used for determining which BAC contains a given gene, useful 
information for certain wet lab experiments.

<P>A valid pair of BAC end sequences must be
at least 2 kb but no more than 800 kb away from each other. 
The orientation of the first BAC end sequence must be &quot+&quot; and
the orientation of the second BAC end sequence must be &quot;-&quot;. There are replicate reads for some BAC end sequences - these have similar read
names and the BAC clone name is the same in each case. If reads for the same 
BAC clone have identical alignments, they are shown on the same description 
page. 
</P>
<P>The scoring scheme used allocates 1000 to an alignment when the BAC end 
pairs only aligned to one location in the genome (after filtering).  For 
multiple alignments to the genome, a BAC end pair or clone scores 1500/Number of
alignments.
</P>
<H2>Methods</H2>
<P>BAC end sequences are placed on the assembled sequence using
Jim Kent's 
<A HREF="http://www.genome.org/cgi/content/abstract/12/4/656" TARGET=_blank>blat</A> 
program followed by pslReps using parameters such that only alignments with 
least 85% identity, a minimum sequence coverage of 40% and a base identity level
within 2% of the best were kept and there was no penalty for not having introns
 (<em>-nearTop=0.02 -minCover=0.40 -minAli=0.85 -noIntrons</em>). Furthermore, 
a base identity of at least 91% was required of at least one BAC end of the 
pair.</P>

<H2>Credits</H2>
<P> Information about the libraries may be found on the Sanger Institute 
<A HREF="http://www.sanger.ac.uk/Projects/D_rerio/library_details.shtml" 
TARGET=_blank>Zebrafish Library Details</A> page. Additional information about 
some of the clones, including how they can be obtained, may be found at the
<A HREF="http://www.ncbi.nlm.nih.gov/genome/clone/index.html" 
TARGET=_blank>NCBI Clone Registry</A>.
To view the registry entry for a specific clone, open the details page for the 
clone and click on its name at the top of the page. Not all $organism BAC 
clones have been submitted to NCBI Clone Registry as of July 2004; therefore, 
some of the clone links to NCBI may not yet be active.
</P>
<P>
Information about STS markers associated with the BAC clone is displayed if it 
is available. Aliases include those for the BAC clone and those for associated 
STS markers and the BAC ends tracks may be searched using the clone name, the 
internal name or any of these aliases. The UniSTS ID(s) shown are those 
associated with each Sanger STS names. 
NOTE: The primer sequences shown may be different than those in UniSTS for the 
UniSTS IDs shown. Where there is more than one UniSTS ID, primer sequences may 
be the same as those in UniSTS for one of the UniSTS IDs.
The relationship field represents the method of finding the STS markers 
associated with this BAC clone.
</P>
<P><B>RElationship</B>
<UL>
<LI><B>1</B> - STS is designed from the end sequence of this BAC clone.
<LI><B>2</B> - STS is found by e-PCR in the genomic sequence of this BAC clone.
<LI><B>3</B> - STS is found in the WGS contig by e-PCR which has been aligned to this region (approximately +/- 150kb). There is no direct relationship with the clone and this data is covered by all the same caveats as the WGS contigs.
</UL>
If the relationship field shows 0, there is no information for this STS or 
there is no STS marker for this BAC clone.
</P>
<P>
The $Organism BAC End Pairs track was produced at UCSC from BAC End Pairs 
sequence data obtained from 
<A HREF="http://www.eb.tuebingen.mpg.de/dept3/geisler/home.html" 
TARGET=_blank>Robert Geisler</A> at the Max Planck Institute for 
Developmental Biology, Germany and the 
<A HREF="http://www.niob.knaw.nl" TARGET=_blank>Netherlands Institute for 
Developmental Biology</A> (Hubrecht Laboratory). The track was produced in 
collaboration with the <A HREF="http://zon.tchlab.org"
TARGET=_blank>Zebrafish Genome Initiative</A> at Childrens Hospital, Boston.
GenBank accessions for BAC clones and the STS marker data were obtained from 
The Wellcome Trust Sanger Institute, Cambridge, United Kingdom. UniSTS IDs were
obtained from NCBI UniSTS database.</P>
