<P>
The November 2003 zebrafish (<em>Danio rerio</em>) Zv3 assembly was produced by 
The Wellcome Trust Sanger Institute in collaboration with the Max Planck 
Institute for Developmental Biology in Tuebingen, Germany, the Netherlands 
Institute for Developmental Biology (Hubrecht Laboratory), Utrecht, The 
Netherlands, and the Zebrafish Genome Initiative at Children's Hospital in 
Boston, MA, USA.

<H3>Sample position queries</P></H3>
<P>
A genome position can be specified by the accession number of
an mRNA or EST, a scaffold range, an Ensembl transcript ID, or keywords from 
the GenBank description of an mRNA. The following list provides 
examples of various types of position queries for the zebrafish genome. Note 
that some position queries (e.g. "huntington") may return matches to the mRNA 
records of other species. In these cases, the mRNAs are mapped to their homologs 
in zebrafish.
See the 
<A HREF="http://genome.cse.ucsc.edu/goldenPath/help/hgTracksHelp.html" TARGET=_blank>User Guide</A> for more help. 
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
<TR><TD VALIGN=Top NOWRAP><B>Request:</B><br></TD>
	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><br></TD></TR>
	
<TR><TD VALIGN=Top><br></TD></TR>
	
<TR><TD VALIGN=Top NOWRAP>chr1</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays all of chromosome 1</TD></TR>
<TR><TD VALIGN=Top NOWRAP>chr1:1-200000</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays first two hundred thousand bases of chromosome 1</TD></TR>
<TR><TD VALIGN=Top><br></TD></TR>

<TR><TD VALIGN=Top NOWRAP>U30710</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays region containing zebrafish mRNA with GenBank accession number U30710</TD></TR>

<TR><TD VALIGN=Top NOWRAP>AA658622</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays region containing zebrafish EST with GenBank accession AA658622
</TD></TR>

<TR><TD VALIGN=Top NOWRAP>ENSDART00000021228</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays region containing Ensembl gene prediction transcript ENSDART00000021228
</TD></TR>

<TR><TD VALIGN=Top><br></TD></TR>

<TR><TD VALIGN=Top NOWRAP>p53</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs related to the p53 tumor suppressor</TD></TR> 
</TD></TR>


<TR><TD VALIGN=Top NOWRAP>pseudogene mRNA</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists transcribed pseudogenes but not cDNAs, in GenBank</TD></TR>
<TR><TD VALIGN=Top NOWRAP>homeobox caudal</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes in GenBank</TD></TR>
<TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD></TR>
<TR><TD VALIGN=Top NOWRAP>kruppel zinc finger</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD></TR>
<TR><TD VALIGN=Top NOWRAP>huntington</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs associated with Huntington's disease</TD></TR>

<TR><TD VALIGN=Top><br></TD></TR>

<TR><TD VALIGN=Top NOWRAP>porter</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Porter</TD></TR>
<TR><TD VALIGN=Top NOWRAP>Amsterdam,A.</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs deposited by co-author A. Amsterdam</TD></TR>

<TR><TD VALIGN=Top><br></TD></TR>
	
<TR><TD COLSPAN="3" > Use this last format for entry authors -- even though 
GenBank searches require Amsterdam A format, GenBank entries themselves use 
Amsterdam,A. internally.
</TABLE>
<P>
<HR>
<H3>Assembly details</H3>
This assembly was constructed by using two different strategies. In the first, 
a whole genome shotgun (WGS) approach was used.  DNA from 1000 five-day-old 
Tuebingen embryos was used to generate plasmid and fosmid libraries. The 
libraries were sequenced and the resulting traces were assembled with the 
Sanger Institute assembler, Phusion. The second approach used traditional 
clone mapping and sequencing techniques to produce a higher quality genome 
sequence. BAC libraries were selected and 
fingerprinted to generate an FPC map. From this map, a tiling path was calculated
that covers the genome sequence clone-by-clone. Clones from this path were 
selected for high quality sequencing and then 
pieced together to form the genome sequence. 
<P>
The Zv3 assembly consists of 1,459,115,486 bp in 58,339 supercontigs, with a 
sequence coverage of approximately 5.7X. This zebrafish assembly is the first 
to be tied to the FPC map: 1,083,447,588 bp (74%) of the sequence were mapped
in this way. 
<P>
The Sanger Institute notes that this is a preliminary assembly; a high 
level of misassembly is present due to polymorphisms in the DNA source.  Highly 
variable regions of the assembly are difficult to assemble,
most likely due to differing haplotypes. 
For more information about this assembly, see the Sanger Institute 
page for the <A HREF="http://www.sanger.ac.uk/Projects/D_rerio/" 
TARGET=_blank><em>Danio rerio</em> Sequencing Project</A>. 
<P>
Downloads of the Zebrafish data and annotations can be obtained from the UCSC
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/danRer1/" TARGET=_blank>FTP 
site</A> or 
<A HREF=http://hgdownload.cse.ucsc.edu/downloads.html#zebrafish>Downloads 
page</A>. 
The danRer1 annotation tracks were generated by UCSC and collaborators worldwide. 
See the <A HREF="/goldenPath/credits.html#zebrafish_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
