<H2>Description</H2>
<P>This track shows the Zebrafish Zv3 (Nov. 2003)  assembly 
provided by the <A HREF="http://www.sanger.ac.uk/Projects/D_rerio/Zv3_assembly_information.shtml" TARGET=_blank>The Wellcome Trust Sanger Institute</A>. The assembly was constructed with the assembler, Phusion, from paired end sequencing 
reads resulting in a sequence coverage of about 5.7X.  This assembly contains 
58,339 scaffolds (supercontigs) totaling 1.5 billion base pairs. 
</P>
<P>
For the clone based mapping and finishing, clones from different libraries were
fingerprinted by digestion with the HindIII restriction enzyme.  From the 
information produced in this way, overlapping clones were linked into 
FingerPrinted Contigs (FPCs). Next, clones from a tiling path through the FPC 
contigs are being selected for high quality sequencing and the resulting 
sequence is submitted to EMBL/Genbank. The supercontigs tied to the FPC map 
create the assembly shown in this track.  The finished clone sequence is then 
analyzed via a pipeline that includes repeatmasking, ab initio gene prediction
 and blast searches against all protein, EST and cDNA sequences that are 
available. Results from this analysis are used to manually annotate clones with
gene structures, descriptions and poly-A features. At this point, the clone is 
submitted to EMBL/Genbank again and can be browsed in <A HREF="http://vega.sanger.ac.uk/Danio_rerio/" TARGET=_blank>Vega</A>.
</P>
<P>In dense mode, this track depicts the path through the draft and
finished clones (aka the golden path) used to create the assembled sequence.
Clone boundaries are distinguished by the use of alternating gold and brown
coloration. Where gaps
exist in the path, spaces are shown between the gold and brown
blocks.  If the relative order and orientation of the contigs
between the two blocks is known, a line is drawn to bridge the
blocks.</P>
<P>
All components within this track are of fragment type &quot;W&quot;:
Whole Genome Shotgun contig except for those in chrFinished which are of type  
&quot;F&quot;: Finished.
</P>
<P>
The Genome Browser depicts the zebrafish genome as 25 chromsomes consisting of
WGS supercontigs that were mapped to an FPC contig and were from a known chromosome. There are also 3 unordered virtual chromosomes namely chrUn, chrFinished 
and chrNA. These 3 chromosomes have 1000 bp scaffold gaps, which are shown in 
the Gap track annotation.
</P>
<P>
Unordered Virtual Chromosomes:
<UL>
<LI><B>chrNA</B> - WGS contigs that could not be related to any FPC contig
<LI><B>chrFinished</B> - Finished clones that could not be place into WGS supercontigs
<LI><B>chrUn</B> - WGS supercontigs that mapped to FPC contigs and the chromosome is unknown
</UL>

