<H2>Description</H2>
<P>
This track displays human-centric multiple sequence alignments in the 
ENCODE regions for the 28 vertebrates included in the 
<A TARGET=_BLANK HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg17/encode/alignments/SEP-2005/">September 2005 ENCODE MSA freeze</A>,
based on comparative sequence data generated for the ENCODE project
as well as whole-genome assemblies residing at UCSC, as listed:
<UL>
<LI>human (May 2004, hg17)
    <LI>armadillo (NISC and May 2005 Broad Assisted Assembly v 1.0)
    <LI>baboon (NISC)
    <LI>chicken (Feb 2004, galGal2)
    <LI>chimp (Nov 2003, panTro1)
    <LI>colobus_monkey (NISC)
    <LI>cow (BCM)
    <LI>dog (July 2004, canFam1)
    <LI>dusky_titi (NISC)
    <LI>elephant (NISC and May 2005 Broad Assisted Assembly v 1.0)
    <LI>fugu (Aug 2002, fr1)
    <LI>galago (NISC)
    <LI>hedgehog (NISC)
    <LI>macaque (Jan 2005, rheMac1)
    <LI>marmoset (NISC)
    <LI>monodelphis (Oct 2004, monDom1)
    <LI>mouse (Mar 2005, mm6)
    <LI>mouse_lemur (NISC)
    <LI>owl_monkey (NISC)
    <LI>platypus (NISC and Aug 2005 Mullikin Phusion Assembly of WUGSC Traces)
    <LI>rabbit (NISC and May 2005 Broad Assisted Assembly v 1.0)
    <LI>rat (June 2003, rn3)
    <LI>rfbat (NISC)
    <LI>shrew (NISC and Sep 2005 Mullikin Phusion Assembly of Broad Traces)
    <LI>tenrec (Apr 2005 Mullikin Phusion Assembly of Broad Traces)
    <LI>tetraodon (Feb 2004, tetNig1)
    <LI>xenopus (Oct 2004, xenTro1)
    <LI>zebrafish (June 2004, danRer2)
</UL>
</P>
The alignments in this track were generated using the 
<A TARGET=_BLANK HREF="http://bio.math.berkeley.edu/mercator/">
Mercator</A> 
orthology mapping program and the
<A TARGET=_BLANK HREF="http://baboon.math.berkeley.edu/mavid/">
MAVID</A>
multiple global alignment program.
The Genome Browser companion tracks, MAVID Cons and MAVID Elements, 
display conservation scoring and conserved elements for these alignments based 
on various conservation methods.</P>

<H2>Display Conventions and Configuration</H2>
<P>
In full display mode, this track shows pairwise alignments
of each species aligned to the human genome.
The alignments are shown in dense display mode using a gray-scale
density gradient.  The checkboxes in the track configuration section allow
the exclusion of species from the pairwise display.
<P>
When zoomed-in to the base-display level, the track shows the base
composition of each alignment. The numbers and symbols on the &quot;human
gap&quot; line indicate the lengths of gaps in the human sequence at those
alignment positions relative to the longest non-human sequence. If the gap
size is greater than 9, a non-numeric symbol is shown; if the
gap size is a multiple of 3, a &quot;*&quot; is displayed,
otherwise &quot;+&quot; is shown.
To view detailed information about the
alignments at a specific position, zoom in the display to 30,000 or fewer
bases, then click on the alignment.  </P>

<H2>Methods</H2>

Mercator was first used to identify the colinear and orthologous  
segments in the sequences given for each ENCODE region.  
Input to Mercator was generated through the prediction of genes in all  
sequences with Genscan, comparison of predicted coding exons with  
BLAT, and identification of non-coding exact matches between all  
pairs of sequences with MUMmer.  The output of Mercator was a small-
scale one-to-one orthology map for each ENCODE region, as well as a
set of alignment constraints based on matched landmarks (e.g., exons
and long non-coding exact matches).  
<P>
MAVID then constructed a global
multiple alignment of each colinear orthologous segment set   
specified in the orthology map.  As part of its input, MAVID used a
phylogenetic tree determined from alignments of four-fold degenerate
sites in the ENCODE regions.

<H2>Credits</H2>
<P>
Generation of the MAVID alignments was engineered by Colin Dewey at the 
<A TARGET=_BLANK HREF="http://math.berkeley.edu/~lpachter/comp_genomics.html">
Pachter Lab Comparative Genomics Group</A> at UC Berkeley.
<P>
Mercator was written by Colin Dewey and Lior Pachter.</P>
<P>
MAVID was authored by Nicholas Bray and Lior Pachter.</P>

<H2>References</H2>
<P>
N. Bray and L. Pachter. 
<A TARGET=_BLANK HREF="http://www.genome.org/cgi/content/abstract/14/4/693">
MAVID: Constrained Ancestral Alignment of Multiple Sequences</A>
Genome Res., 14(4):693-699, 2004.
<P>
C. N. Dewey and L. Pachter. 
<A TARGET=_BLANK HREF="http://bio.math.berkeley.edu/mercator/">
Mercator: multiple whole-genome orthology  map construction</A>
In preparation.
<P>
C. Burge and S. Karlin. 
<A TARGET=_BLANK HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=9149143">
Prediction of complete gene structures in human genomic DNA</A>
J Mol Biol, 268(1):78-94, 1997.
<P>
W. J. Kent. 
<A TARGET=_BLANK HREF="http://www.genome.org/cgi/content/full/12/4/656">
BLAT-the BLAST-like alignment tool</A>
Genome Res., 12(4): 656-664, 2002.
<P>
S. Kurtz, A. Phillippy, A. L. Delcher, M. Smoot, 
M. Shumway, C. Antonescu, and S. L. Salzberg. 
<A TARGET=_BLANK HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?CMD=search&DB=pubmed">
Versatile and open software for comparing large genomes</A>
Genome Biol, 5(2):R12, 2004.
</P>
