<H2>Description</H2>
<P>
This track shows DNase I sensitivity measured across ENOCDE regions using
the Quantitative Chromatin Profiling (QCP) method (Dorschner <em>et al</em> 
(2004). QCP provides a quantitative high-throughout method for the 
identification of the continous distribution of DNase I sensitivity across 
gene loci. The moving baseline of mean DNaseI sensitivity is computed over 
each PCR amplicon using a locally-weighted least squares (LOWESS)-based 
algorithm.
DNaseI has long been used to map general chromatin accessibility and the 
DNaseI &quot;hyperaccessibility&quot; or &quot;hypersensitivity&quot; that is a 
universal feature of active <em>cis</em>-regulatory sequences. Use of this 
method has lead to the discovery of functional regulatory elements that 
include enhancers, insulators, promotors, locus control regions and novel 
elements.</P>

<H2>Display Conventions and Configuration</H2>
<P>
DNaseI sensitivity is expressed in standard units, where each
increment of 1 unit corresponds to an increase of 1SD from the baseline.
The displayed values are calculated as <TT>(copies in DNaseI-untreated / copies 
in DNaseI-treated)</TT>. Thus, increasing values represent increasing 
sensitivity.  Major DNaseI hypersensitive sites are readily identified as
peaks in the signal that exceed 2 SDs (corresponding to the ~95% confidence
bound on outliers).</P>
<P>
The subtracks within this composite annotation track correspond to data
from different tissues, and may be configured in a variety of ways to 
highlight different aspects of the displayed data. The graphical 
configuration options are shown at the top of the track description page, 
followed by a list of subtracks.  To display only selected subtracks, uncheck 
the boxes next to the tracks you wish to hide. For more information about the 
graphical configuration options, click the 
<A HREF="../goldenPath/help/hgWiggleTrackHelp.html" TARGET=_blank>Graph
configuration help</A> link.</P>
<P>
Color differences among the subtracks are arbitrary; they provide a
visual cue for distinguishing the different cell lines/phenotypes.</P>

<H2>Methods</H2>
<P>
QCP was performed as described in Dorschner <em>et al</em>. The article 
describes the methods of chromatin preparation and DNase I digestion. Briefly,
DNase I treated nuclei were prepared for each cell type. Primers were designed
such that they amplified tiled contiguous amplicons (approximately 250 bp
in size) across the study regions. An amplicon tiling path has been computed 
over all regions and is available
through <A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=unists"
TARGET=_blank>UniSTS</A>. The relative number of intact copies of the
genomic DNA sequence was determined using real-time PCR for both 
DNase-I treated and untreated samples.  For the replicates for each amplicon,
the mean value of copies remaining in DNaseI-treated <em>vs.</em> untreated
samples were compared. The results were analyzed with a statistical algorithm
to compute the moving baseline of mean DNase I sensitivity and to identify
outliers that correspond with DNase I hypersensitive sites. 
</P>

<H2>Verification</H2>
All data points represent the mean of four replicate measurements. Biological
replicate sampels were pooled. Validation of the results was carried out
by convential DNase I hypersensitivity assays using end-labeling/Southern
blotting.</P>

<H2>Credits</H2>
<P>
Data generation, analysis, and validation were performed jointly by groups at
Regulome Corporation and the University of Washington (UW) in Seattle.</P>
<P>
<A HREF="http://www.regulome.com/" TARGET=_blank>Regulome Corp.</A>: Michael O.
Dorschner, Richard Humbert, Peter J. Sabo, Anthony Shafer, Jeff Goldy,
Molly Weaver, Kristin Lee, Fidencio Neri, Brendan Henry, Mike Hawrylycz, Paul
Tittel, Jim Wallace, Josh Mack, Janelle Kawamoto,
and John A. Stamatoyannopoulos.
</P>
<P>
<A HREF="http://depts.washington.edu/medgen/" TARGET=_blank>UW Medical
Genetics</A>: Patrick Navas, Man Yu, Hua Cao, Brent Johnson, Ericka
Johnson, George Stamatoyannopoulos.</P>
<P>
<A HREF="http://www.gs.washington.edu/" TARGET=_blank>UW Genome Sciences</A>: 
Scott Kuehn, Robert Thurman, William S. Noble.</P>

<H2>References</H2>
<P>
Dorschner, M.O., Hawrylycz, M., Humbert, R., Wallace, J.C., Shafer, A., 
Kawamoto, J., Mack, J., Hall, R., Goldy, J., Sabo, P.J. <em>et al</em>.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=15782197&query_hl=1"
TARGET=_blank>High-throughput localization of functional elements by 
quantitative chromatin profiling.</A>
<em>Nat Methods</em> <B>1</B>(3), 219-25 (2004).</P>
