<H2>Description</H2>
<P>
This track shows putative binding loci of c-myc as determined 
by Sequence Tag Analysis of Genomic Enrichment (STAGE). 
The c-myc (cellular myelocytomatosis) protein is a transcription 
factor associated with cell proliferation, diffentiation, and
neoplastic disease.</P>
<P>
STAGE was performed in HeLa cells under normal growth conditions
(10% Fetal Bovine Serum) using anti-myc antibody. 
Cloned STAGE tags were sequenced and mapped to the human genome as described in 
Kim <em>et al.</em> (2005), referenced below. </P>
<P>
The Tags subtrack shows all STAGE tags within the ENCODE region and thus 
represents the raw data. 
The Peaks subtrack shows high confidence c-myc binding  
regions derived from the STAGE tags.</P>

<H2>Display Conventions and Configuration</H2>
<P>
This annotation follows the display conventions for composite 
tracks.  To display only one of the subtracks, uncheck the boxes next to the 
track you wish to hide.  </P>

<H2>Methods</H2>
<P>
RepeatMasker was applied at low stringency (score more than 1500) 
to exclude tags within repeat regions.
Each tag was assigned a probability of enrichment calculated 
from the frequency of occurrence of the tag in the 
STAGE sequencing pool and the number of times the tag is present in 
the genome, assuming a binomial distribution. 
Tags having a genomic frequency of more than 10 were discarded. 
In general, a low STAGE pool frequency with a high genomic
frequency generates a score close to 1, indicating
a very low probability of enrichment.  For purposes of 
visualization, these tags were assigned an artificially
inflated score of 300; however, for analysis, a score of 1 should be used.</P>
<P>
Peaks were determined by using a 500 bp window to scan across each chromosome. 
Each window was assigned a probability based on the tags mapped 
within that window. </P>

<H2>Verification</H2>
<P>
Scores generated from the real data were compared to
simulations where similar numbers of tags were randomly sampled from the genome.
Calculating probabilities as above, a probability cut-off of 0.8 gave a false 
positive rate of less than 0.05.</P>

<H2>Credits</H2>
<P>
These data were contributed by Jonghwan Kim, Akshay Bhinge, and Vishy Iyer 
from the <A HREF="http://www.iyerlab.org" TARGET=_blank>Iyer lab</A> at the 
University of Texas at Austin. </P>

<H2>Reference</H2>
<P>
Kim, J., Bhinge, A., Morgan, X.C. and Iyer, V.R. 
<A HREF="http://www.nature.com/nmeth/journal/v2/n1/full/nmeth726.html"
TARGET=_blank>Mapping DNA-protein interactions in large genomes by sequence tag 
analysis of genomic enrichment</A>. <em>Nature Methods</em> <B>2</B>, 47-53 
(2005).</P>
