<H2>Description</H2>
<P>
This track displays regions containing putative functional RNA secondary
structures as predicted by RNAz on on the basis of thermodynamic stability
and evolutionary conservation.
	  
<H2>Methods</H2>
<P>
RNAz evaluates multiple sequence alignments for unusually stable and
conserved RNA secondary structures, two typical characteristics for
functional RNA structures that can be found in noncoding RNAs or cis-acting
regulatory elements of mRNAs.
</P>
<P>
The RNAz algorithm works as follows: First a consensus secondary structure
is predicted using the RNAalifold approach (Hofacker et al., 2002), which is 
an extension of classical minimum free energy folding algorithms for aligned 
sequences. The significance of a predicted consensus structure is evaluated by 
calculating a <I>structure conservation index</I>, which is the ratio of 
unconstrained folding energies relative to the folding energies under the 
constraint that all aligned sequences are forced to fold into a common
structure. Thermodynamical stability is evaluated by calculating a
normalized <I>z-score</I> of the sequences in the alignment. The z-score
indicates whether the given sequences are more stable than random sequences
of the same length and base composition. Based on these two features,
structure conservation index and z-score, an alignment is classified as
structural RNA or "other" using a support vector machine classification
algorithm (Washietl et al., 2005; Washietl et al., 2007).
</P>

<P>
This track shows the result of a RNAz screen of 28-way TBA/MULTIZ
alignments.  Alignments were sliced in overlapping windows of 120 nt in size 
and with a step size of 40 nt. Sequences with more than 25% gaps with respect 
to the human sequence were discarded. Only alignments with more than four 
sequences, a minimum size of 50 columns and at most 1% repeat masked letters 
were considered. RNAz can only handle alignments with up to six sequences. From
alignments with more than six sequences we chose a subset of six. For subset
selection, we used a greedy algorithm and iteratively selected sequences 
optimizing the set for a mean pairwise identity of around 80%. In cases of 
alignments with more than 10 sequences we sampled three different of such 
subsets. The windows were finally scored with RNAz version 0.1.1 in the 
forward and reverse complement direction. Overlapping hits with at least one 
sampled alignment with RNAz score > 0.5 were combined to a single genomic 
regions. The track shows regions with at least one window in the cluster with 
an average RNAz score of all samples > 0.5 and at least one hit with RNAz 
score > 0.9. More details may be found in Washietl et al., 2007.
</P>

<H2>Credits</H2>

<P>The RNAz program and browser track were developed by Stefan Washietl,
Ivo Hofacker (Institute for Theoretical Chemistry, Univ. of Vienna) and
Peter F. Stadler (Bioinformatics group, Department of Computer Science,
Univ. of Leipzig).
</p>

	  
<H2>References</H2>

<P>
Washietl S et al. Structured RNAs in the ENCODE Selected Regions
of the Human Genome, <I>Genome Res.</I> 2007 (in press)
</P>

<P>
Washietl S, Hofacker IL, Stadler PF. Fast and reliable prediction of noncoding 
RNAs. <I>Proc. Natl. Acad. Sci. USA</I>. 2005 102:2454-2459.
</P>

<P>
Hofacker IL, Fekete M, Stadler PF. Secondary structure prediction for 
aligned RNA sequences. <I>J. Mol. Biol.</I> 2002 319:1059-1066.
</P>

