<H2>Description</H2>
<P>
This track displays DNaseI-hypersensitive sites identified using two 
methods (DNase-chip and MPSS sequencing) in four human cell 
types (primary CD4+ T cells, GM06990 lymphoblastoid, HeLa S3 cervical
 carcinoma, and HepG2 liver carcinoma cell lines). 
Experiments were performed in unactivated and activated CD4+ T cells.
CD4+ T cells, also known as helper or inducer T cells, are
involved in generating an immune response.  CD4+ T cells are also
one of the primary targets of the HIV virus.
<P>
DNaseI-hypersensitive sites are associated with all types of gene 
regulatory regions, including promoters, enhancers, silencers, insulators, and 
locus control regions

<H2>Display Conventions and Configuration</H2>
<P>
The subtracks within this track are grouped in 4 sections:
<UL>
<LI>
<B>New Chip Raw</B> subtracks display log2 ratio data averaged from
3 biological replicates and 3 DNase concentrations.
<LI>
<B>New Chip Pval</B> subtracks show significant regions that likely 
represent valid DNaseI-hypersensitive sites, based on the Raw data.
The higher the score for the region, the more likely the site
is to be hypersensitive.
<LI>
<B>Old Chip</B> subtracks also display likely hypersensitive sites
based on DNase-chip experiments.
Unlike the New Chip tracks, this track has identified significant
signal from one, two, or three DNase concentrations.
The items in these subtracks have unique identifiers, visible
when the track is displayed in full or packed mode. The first part of
the identifier describes the cell type from which the data was derived.
The last number of each identifier indicates the percent of
regions with this score that represent valid DNase hypersensitive sites.
Sites are displayed in the range light gray to black,  based on score.
<LI>
<B>MPSS</B> subtracks show hypersensitive sites determined by MPSS 
sequencing.  
Sites are displayed in the range light gray to black, 
based on score.
</UL>
To display only selected subtracks, uncheck the boxes next to the tracks 
you wish to hide.

<H2>Methods</H2>
<H3> DNase-Chip</H3>
<P>
DNaseI hypersensitive sites were isolated using a method called 
DNase-chip (Crawford et al., 2006). 
Briefly, DNaseI digested ends from intact chromatin were captured using
three different DNase concentrations as well as three biological replicates. 
This material was amplified, labeled, and hybridized to NimbleGen 
ENCODE tiled microarrays.
<P>
For the New Chip experiments, the raw data was averaged from 
9 hybridizations per cell type.  The Pval scores represent -log10 p values as 
determined by the ACME (Algorithm for Capturing Microarray Enrichment) 
program (Scacheri, Crawford, and Davis, 2006).  
Only regions that have p value &lt; 0.001 are included.
<P>
For the Old Chip experiments, significant signal was identified separately
from each of the 3 different DNase concentrations. The regions
were assigned scores based on the number of DNase concentrations for which
the region exhibited significant signal.


<H3> MPSS Sequencing </H3>
<P>
Primary human CD4+ T cells were activated by incubation with anti-CD3
and anti-CD28 antibodies for 24 hours. DNaseI-hypersensitive sites
were cloned from the cells before and after activation,
and sequenced using massively parallel signature sequencing 
(Brenner <em>et al</em>., 2000; Crawford <em>et al</em>., 2006). 
Only those clusters of multiple DNaseI library sequences that map within 500 
bases of each other are displayed.
Each cluster has a unique identifier, visible when the track is displayed
in full or packed mode. The last digit of each identifier represents the 
number of sequences that map within that particular cluster. The sequence number
is also reflected in the score, <em>e.g.</em> a cluster of two sequences scores 
500, three sequences scores 750 and four or more sequences scores 1000.  </P>

<H2>Verification</H2>
<H3> DNase-Chip </H3>
<P>
A real-time PCR assay (McArthur et al., 2001, Crawford et al., 2004) 
was used to validate a randomly selected subset
of DNase-chip regions.  
For the New DNase-chip, we determined the 
Sensitivity of DNase-chip to be  &gt; 86% and 
Specificity to be &gt; 97%.
For the Old DNase-chip, significant signal was detected in one,
two, or three DNase concentrations.
Approximately 20-30% of regions detected 
in only a single DNase concentration are valid. 
50-80% of regions detected in 2 out of 3 DNase concentrations are valid
(the exact percentage depends on which 2 DNase concentrations had
significant signal).
90% of regions detected in all 3 DNase concentrations are valid
<P>
This data set includes elements for all 44 ENCODE regions.</P>

<H3> MPSS Sequencing </H3>
<P>
Real-time PCR was used to verify valid
DNaseI-hypersensitive sites.  Approximately 50% of
clusters of two sequences are valid.  These clusters are shown
in light gray. 80% of clusters of three sequences are valid, indicated by
dark gray. 100% of clusters of four or more
sequences are valid, shown in black.</P>
<P>
This data set includes confirmed elements for 35 of the 44
ENCODE regions.
It is estimated that these data identify
10-20% of all hypersensitive sites within CD4+ T cells. Further
sequencing will be required to identify additional sites.
MPSS data from the whole genome can be found in the Expression and Regulation
track group (NHGRI DNaseI-HS track).</P>

<H2>Credits</H2>
<P>
These data were produced at the 
<A HREF="http://www.genome.duke.edu/people/faculty/crawford/" TARGET=_BLANK>
Crawford Lab</A> 
at Duke University, and at the
<A HREF="http://www.genome.gov/10000351" TARGET=_blank>Collins Lab</A> 
at NHGRI.  Thanks to Gregory E. Crawford and Francis S. Collins for supplying 
the information for this track.</P>

<H2> References </H2>
<P>
Brenner, S. <em>et al</em>.
<A HREF="http://www.nature.com/cgi-taf/DynaPage.taf?file=/nbt/journal/v18/n6/abs/nbt0600_630.html&dynoptions=doi1097883799" 
TARGET=_blank>Gene expression analysis by massively parallel signature 
sequencing (MPSS) on microbead arrays</A>.
<em>Nat. Biotechnol.</em> <B>18</B>(6), 630-4 (2000).</P>
<P>
Crawford, G.E., Davis, S., Scacheri, P.C., Renaud, G., Halawi, M.J., Erdos, M.R., Green, R., Meltzer, P.S., Wolfsberg, T.G., Collins, F.S.
<A HREF="http://www.nature.com/nmeth/journal/v3/n7/abs/nmeth888.html" TARGET=_BLANK>
DNase-chip: A High Resolution Method to Identify DNaseI Hypersensitive Sites using Tiled Microarrays</A>.
<em>Nature Methods</em> <B>3</B>(7), 503-509 (2006).
<P>
Crawford, G.E., Holt, I.E., Whittle, J., Webb, B.D., Tai, D., Davis, S., Margulies, E.H., Chen, Y., Bernat, J.A., Ginsburg, D., Zhou, D., Luo, S., Vasicek, T.J., Daly, M.J., Wolfsberg, T.G., Collins, F.S. 
<A HREF="http://www.genome.org/cgi/content/abstract/16/1/123?etoc" TARGET=_BLANK>
Genome-wide mapping of DNase hypersensitive sites using massively parallel signature sequencing (MPSS)</A>. 
<em>Genome Res.</em><B> 16</B>(1), 123-131 (2006).
(See also NHGRI's
<A HREF="http://research.nhgri.nih.gov/DNaseHS/May2005/"
TARGET=_blank>data site</A> for the project.)</P>
<P>
Scacheri, P.C., Crawford, G.E., Davis, S.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=pubmed&cmd=Retrieve&dopt=AbstractPlus&list_uids=16939795&query_hl=3&itool=pubmed_docsum" TARGET=_BLANK>
Statistics for ChIP-chip and DNase hypersensitivity experiments on NimbleGen arrays</A>.  
<em>Methods Enzymol.</em> <B>411</B> 270-82 (2006).
<P>
Crawford, G.E. <em>et al</em>.
<A HREF="http://www.pubmedcentral.nih.gov/articlerender.fcgi?tool=pubmed&pubmedid=14732688" 
TARGET=_blank>Identifying gene regulatory elements by genome-wide recovery of 
DNase hypersensitive sites</A>.
<em>Proc. Natl. Acad. Sci. USA</em> <B>101</B>(4), 992-7 (2004).</P>
<P>
McArthur, M., Gerum, S., and Stamatoyannopoulos, G.
<A HREF="http://www.sciencedirect.com/science?_ob=ArticleURL&_udi=B6WK7-457CXYS-4M&_coverDate=10%2F12%2F2001&_alid=210456595&_rdoc=1&_fmt=&_orig=search&_qd=1&_cdi=6899&_sort=d&view=c&_acct=C000050221&_version=1&_urlVersion=0&_userid=10&md5=5a2dd2b59e368e190942669df47d72c8"
TARGET=_blank>Quantification of DNaseI-sensitivity by real-time PCR: quantitative analysis of DNaseI-hypersensitivity of the mouse beta-globin LCR</A>.
<em>J. Mol. Biol.</em> <B>313</B>(1), 27-34 (2001).
