<H2>Description</H2>
<P>
This track shows the map of
-log10(P-value) for ChIP-chip using 5 antibodies in Human Hela 
S3 cells hybridized to maskless photolithographic arrays with 50-mer 
oligonucleotides tiled with
12-nt overlaps covering most of the non-repetitive DNA sequence of the
ENCODE regions.
<P>
This track shows the combined results of three multiple biological
replicates. For all arrays, the ChIP DNA was labeled with Cy5 and the
control DNA was labeled with Cy3.
These data are available at NCBI GEO, which also provides additional
information about the experimental protocols.
The antibodies are described in the following section.

<H4>BAF155 and BAF170</H4>
The Swi-Snf chromatin-remodeling complex was first described in yeast, and
similar proteins have been found in mammalian cells. The human Swi-Snf
complex is comprised of at least nine polypeptides, including two ATPase
subunits, Brm and Brg-1. Other members of the human Swi-Snf complex are
termed BAFs for Brg1-associated factors. BAF155 and BAF170 are conserved
(core) components that stimulate the chromatin remodeling activity of
Brg1.

<H4>c-Fos</H4>
The transcriptional activity of the proto-oncogene c-Fos has been
implicated in cell growth, differentiation, and development. Fos is induced
by many stimuli, ranging from mitogens to pharmacological agents. c-Fos has
been shown to be associated with another proto-oncogene, c-Jun, and 
together they bind to the AP-1 binding site to regulate gene transcription.
Like CREB, c-Fos is regulated by p90Rsk.

<H4>c-Jun</H4>
C-Jun, also known as AP-1 (activator protein 1), is the cellular homologue
of the avian sarcoma virus oncogene v-jun, and as such can be referred to
as a proto-oncogene. 

<H4>TAF1/TAF250 </H4>
TAF250 (TBP associated factors, with molecular weight 250 kD, also known
as TAF1), has histone acetyltransferase activity, 
which can relieve the binding between
DNA and histones in the nucleosome.

<H2>Methods</H2>
<P>
The data from replicates were quantile-normalized and median-scaled to each
other (both Cy3 and Cy5 channels). Using a 1000 bp sliding window centered
on each oligonucleotide probe, a signal map (estimating the fold enrichment
[log2 scale] of ChIP DNA) was generated by computing the pseudomedian
signal of all log2(Cy5/Cy3) ratios (median of pairwise averages) within the
window, including replicates. Using the same procedure, a -log10(P-value)
map (measuring significance of enrichment of oligonucleotide probes in the
window) for all sliding windows was made by computing P-values using the
Wilcoxon paired signed rank test comparing fluorensent intensity between
Cy5 and Cy3 for each oligonucleotide probe (Cy5 and Cy3 signals from the
same array). A binding site was determined by thresholding oligonucleotide
positions with -log10(P-value) (>= 4), extending qualified positions
upstream and downstream 250 bp, and requiring 1000 bp space between two
sites. Top 400 sites are retained.
<H2> Verification</H2>
<P>
ChIP-chip binding sites were verified by comparing "hit lists" generated
from combinations of different biological replicates. Only experiments that
yielded a significant overlap (greater than 50 percent) were accepted. As
an independent check (for maskless arrays), data on the microarray were
randomized with respect to position and re-scored; significantly fewer hits
(consistent with random noise) were generated this way.
<H2> Credits</H2>
<P>
These data were generated and analyzed by the labs of Michael Snyder, Mark
Gerstein and Sherman Weissman at Yale University.

<H2> References </H2>
<P>
Cawley, S., Bekiranov, S., Ng, H.H., Kapranov, P., Sekinger, E.A., Kampa,
D., Piccolboni, A., Sementchenko, V., Cheng, J. et al. Unbiased mapping of
transcription factor binding sites along human chromosomes 21 and 22 points
to widespread regulation of noncoding RNAs. Cell 116(4), 499-509 (2004).
Euskirchen, G., Royce, T.E., Bertone, P., Martone, R., Rinn, J.L., Nelson,
F.K., Sayward, F., Luscombe, N.M., Miller, P. et al. CREB binds to multiple
loci on human chromosome 22, Mol Cell Biol. 24(9), 3804-14 (2004).
Martone, R., Euskirchen, G., Bertone, P., Hartman, S., Royce, T.E.,
Luscombe, N.M., Rinn, J.L., Nelson, F.K., Miller, P. et al. Distribution of
NF-kappaB-binding sites across human chromosome 22. Proc Natl Acad Sci U S
A. 100(21), 12247-52 (2003).
Quackenbush, J.. Microarray data normalization and transformation, Nat
Genet. 32(Suppl), 496-501 (2002).

