<H2>Description</H2>
<P>
This track displays DNaseI-hypersensitive sites identified using two methods
(DNase-chip and MPSS sequencing) in four human cell types (primary CD4+ T cells,
GM06990 lymphoblastoid, HeLa S3 cervical carcinoma, 
and HepG2 liver carcinoma cell lines). 
Experiments were performed in T cells activated by 
antibody treament, and in unactivated T cells.
CD4+ T cells, also known as helper or inducer T cells, are
involved in generating an immune response.  CD4+ T cells are also
one of the primary targets of the HIV virus.
<P>
DNaseI-hypersensitive sites are associated with all types of gene 
regulatory regions, including promoters, enhancers, silencers, and 
locus control regions

<H2>Display Conventions and Configuration</H2>
<P>
The subtracks within this track are grouped in 4 sections:
<UL>
<LI>
<B> New Chip Raw</B> subtracks display averaged log2 ratio data.
<LI>
<B>New Chip Pval</B> subtracks show significant regions that likely 
represent valid DNase I hypersensitive sites, based on the Raw data.
The higher the score for the region, the more likely the site
is to be hypersensitive.
<LI>
<B>Old Chip</B> subtracks also display likely hypersensitive sites
based on Dnase-chip experiments.
The items in these subtracks have unique identifiers, visible
when the track is displayed in full or packed mode. The first part of
the identifier describes the cell type from which the data was derived.
The last number of each identifier indicates the percent of
regions that represent valid DNase hypersensitive sites.
<LI>
<B>MPSS</B> subtracks show hypersensitive sites determined by MPSS 
sequencing.
</UL>
To display only selected subtracks, uncheck the boxes next to the tracks 
you wish to hide.

<H2>Methods</H2>
<H3> DNase-Chip</H3>
<P>
DNaseI hypersensitive sites were isolated using a method called 
DNase-chip (Crawford et al., 2006). 
Briefly, DNaseI digested ends from intact chromatin were captured using
three different DNase concentrations as well as three biological replicates. 
This material was amplified, labeled, and hybridized to NimbleGen 
ENCODE tiled microarrays.
<P>
For the New Chip experiments, the Pval scores represent -log10 p values as 
determined by the ACME (Algorithm for Capturing Microarray Enrichment) 
program (Scacheri, Crawford, and Davis, 2006).  
Only regions that have p value &lt; 0.001 are included.

<H3> MPSS Sequencing </H3>
<P>
Primary human CD4+ T cells were activated by incubation with anti-CD3
and anti-CD28 antibodies for 24 hours. DNaseI-hypersensitive sites
were cloned from the cells before and after activation,
and sequenced using massively parallel signature sequencing 
(Brenner <em>et al</em>., 2000; Crawford <em>et al</em>., submitted). 
Only those clusters of multiple DNaseI library sequences that map within 500 
bases of each other are displayed.
Each cluster has a unique identifier, visible when the track is displayed
in full or packed mode. The last digit of each identifier represents the 
number of sequences that map within that particular cluster. The sequence number
is also reflected in the score, <em>e.g.</em> a cluster of two sequences scores 
500, three sequences scores 750 and four or more sequences scores 1000.  </P>

<H2>Verification</H2>
<H3> Dnase-Chip </H3>
<P>
Real-time PCR assay was used to validate a randomly selected subset
of DNase-chip regions. Approximately 20-30% of regions detected 
in only a single DNase concentration are valid. These clusters are shown
in light gray. 50-80% of regions detected in 2 out of 3 DNase
concentrations are valid, indicated by dark gray. 90% of regions
detected in all 3 DNase concentrations are valid, shown in black.</P>
<P>
This data set includes elements for all 44 ENCODE regions.</P>

<H3> MPSS Sequencing </H3>
<P>
Real-time PCR assay was used to verify valid
DNaseI-hypersensitive sites.  Approximately 50% of
clusters of two sequences are valid.  These clusters are shown
in light gray. 80% of clusters of three sequences are valid, indicated by
dark gray. 100% of clusters of four or more
sequences are valid, shown in black.</P>
<P>
This data set includes confirmed elements for 35 of the 44
ENCODE regions.
It is estimated that these data identify
10-20% of all hypersensitive sites within CD4+ T cells. Further
sequencing will be required to identify additional sites.</P>

<H2>Credits</H2>
<P>
These data were produced at the Crawford Lab at Duke University,
and at 
<A HREF="http://www.genome.gov/10000351" TARGET=_blank>Collins Lab</A> 
at NHGRI.  Thanks to Gregory E. Crawford and Francis S. Collins for supplying 
the information for this track.</P>

<H2> References </H2>

< P>
Crawford GE, Davis S, Scacheri PC, Renaud G, Halawi MJ, Erdos MR, Green R, Meltzer PS, Wolfsberg TG, Collins FS.
DNase-chip: A High Resolution Method to Identify DNaseI Hypersensitive Sites using Tiled Microarrays, 
<em>Nature Methods</em> <B>3</B>(7), 503-509 (2006).
<P>
Crawford GE, Holt IE, Whittle J, Webb BD, Tai D, Davis S, Margulies EH, Chen Y, Bernat JA, Ginsburg D, Zhou D, Luo S, Vasicek TJ, Daly MJ, Wolfsberg TG, Collins FS. Genome-wide mapping of DNase hypersensitive sites using massively parallel signature sequencing (MPSS). 
<em>Genome Res.</em><B> 16</B>(1), 123-131 (2006).
<P>
Scacheri PC, Crawford GE, Davis S.
Analysis of ChIP-chip and DNase-chip data obtained from tiled microarrays.  
<em>Methods Enzymol.</em><B>411</B> 270-82 (2006).
