<H2>Description</H2>
<P>
The exoniphy program identifies evolutionarily conserved protein-coding
exons in a multiple alignment using a phylogenetic hidden Markov
model (phylo-HMM), a kind of statistical model that simultaneously
describes exon structure and exon evolution.  This track shows exoniphy
predictions for the human May 2004 (hg17), mouse Aug. 2005 (mm7), rat
Jun. 2003 (rn3), and dog May 2005 (canFam2) genomes, as aligned by the
multiz program.  For this track, only alignments on the "syntenic net"
between human and each other species were considered. 
</P>
<H2>Methods</H2>
<P>
Exoniphy is described in Siepel A & Haussler D (2004), "<a
href="http://www.cse.ucsc.edu/~acs/recomb2004.pdf"
TARGET=_blank>Computational identification of evolutionarily conserved
exons</a>," <em>RECOMB '04.</em> Multiz is described in Blanchette M <em>et
al.</em> (2004), "<a
href="http://www.genome.org/cgi/content/abstract/14/4/708"
TARGET=_blank>Aligning multiple genomic sequences with the threaded
blockset aligner</a>," <em>Genome Res.</em> 14:708-175.  The alignment
chaining methods behind the "syntenic net" are described in Kent <em>et
al.</em> (2003), "<a
href="http://www.pnas.org/cgi/content/abstract/1932072100v1"
TARGET=_blank>Evolution's cauldron: duplication, deletion, and
rearrangement in the mouse and human genomes</a>,"
<em>P. Natl. Acad. Sci. USA</em> 100(20), 11484-11489.
