<H2>Description</H2>
<P>
Consolidates mutations from locus-specific databases and a genome wide
mutation database into a single track.
This prototype version includes entries from the few Locus Specific databases,
listed below under Variant Sources. 
</P>

<H2>Display Conventions and Configuration</H2>
<p>Color key: The track is color coded by the mutation type.
<UL>
<LI>
substitution = <font color="#CC00FF">purple</font>
<LI>
insertion = <font color="#009900">green</font>
<LI>
deletion = <font color="#0000FF">blue</font>
<LI>
duplication = <font color="#FF9900">orange</font>
<LI>
complex = <font color="#643200">brown</font>
<LI>
unknown = black
</UL>
</p>
<p>
Labels:  By default items are labeled by the HGVS name.  To change this
click the desired option or options.  None will produce no labels no 
matter what else is selected.
</p>

<a name="source">
<H2>Variant Sources</H2>
<UL>
<LI>
The HbVar database describes variants and thalassemia mutations in the
globin loci.
<A HREF="http://www.bx.psu.edu/" TARGET="_blank">http://www.bx.psu.edu/</A>
<LI>
The PAHdb has mutations in the phenylalanine hydroxylase (PAH) gene.
<A HREF="http://www.pahdb.mcgill.ca/" TARGET="_blank">http://www.pahdb.mcgill.ca/</A>
<LI>
The Blood Group Antigen Gene Mutation Database (BGMUT) has mutations in many genes.
<A HREF="http://www.ncbi.nlm.nih.gov/projects/mhc/xslcgi.fcgi?cmd=bgmut/home" TARGET="_blank">http://www.ncbi.nlm.nih.gov/projects/mhc/xslcgi.fcgi?cmd=bgmut/home</A>
<LI>
The Androgen Receptor Gene Mutations Database (ARdb) has mutations in the AR gene.
<A HREF="http://www.androgendb.mcgill.ca/" TARGET="_blank">http://www.androgendb.mcgill.ca/</A>
<LI>
The Cystic Fibrosis Mutation Database (CFMDB) has mutations in the CFTR gene.
<A HREF="http://www.genet.sickkids.on.ca/cftr/" TARGET="_blank">http://www.genet.sickkids.on.ca/cftr/</A>
<LI>
Swiss-Prot data on variants within coding regions is available genome wide.
<LI>
A possible connection with WayStation, the HGVS community-wide database.
<A HREF="http://www.centralmutations.org" TARGET="_blank">http://www.centralmutations.org/</A>
</UL>

<H2>Data Filtering</H2>
<UL>
<LI>
Filtering can be performed on the location of the mutations with respect to
genes, i.e. to exclude those in exons, introns, UTR regions, or those
not within any known transcription unit.
<LI>
One can also exclude mutations by type: substitution, insertion, deletion,
duplication, or complex.
<LI>
Filtering by source is also provided.  The locus specific databases are 
listed as one source.
</UL>

<H2>Credits</H2>
<H4>PhenCode developers</H4>
<DL>
<DT>Belinda Giardine, Ross Hardison, Webb Miller, Cathy Riemer
<DD>Center for Comparative Genomics and Bioinformatics, Penn State University, University Park, PA
<DT>Fan Hsu, Jim Kent, Andrew Kern, Robert Kuhn, Heather Trumbower
<DD>Center for Biomolecular Science and Engineering, University of California, Santa Cruz, CA
<DT>Richard Gibbons, Doug Higgs, Jim Hughes
<DD>Weatherall Institute of Molecular Medicine, Oxford UK
<DT>Garry Cutting, Andrew P. Feinberg
<DD>Johns Hopkins University School of Medicine, Baltimore, MD
</DL> 
<H4>Cooperating databases</H4>
<DL>
<DT>ARdb:
<DD>Dr. Bruce Gottlieb,
Lady Davis Institute for Medical Research,
Sir Mortimer B. Davis-Jewish General Hospital
Montreal, Quebec, Canada
<DT>BGMUT:
<DD>Olga O. Blumenfeld, Ph.D., Professor <br />
Santosh K. Patnaik<br />
Dept. of Biochemistry,
Albert Einstein College of Medicine,
Bronx, NY
<DT>CFMDB:
<DD>Dr. Julian Zielenski, Dr. Richard Sang,
The Hospital for Sick Children,
Genetics and Genomic Biology,
Ontario, Canada
<DT>HbVar:
<DD>George Patrinos, Erasmus University, Rotterdam, Netherlands<br />
Henri Wajcman, Hospital Henri Mondor, Creteil, France<br />
David H.K. Chui, Boston University, Boston, MA<br />
Nicholas Anagnou, U. Athens and IIBEAA, Athens, Greece<br />
Georgi D. Efremov, Macedonian Academy of Sciences and Arts, RCGEB, Skopje, Macedonia
<DT>PAH:
<DD>Charles R. Scriver MDCM FRS,
Alva Professor Emeritus of Human Genetics,
McGill University,
Montreal Children's Hospital Research Institute, Canada
</DL>
