<H2>Description</H2>
<P>
Consolidates mutations from locus-specific databases and a genome wide
mutation database into a single track.
This prototype version includes entries mostly from the globin loci, plus
a few sample entries from the CFTR region for the locus-specific sources.
</P>

<H2>Display Conventions and Configuration</H2>
<p>Color key: The track is color coded by the mutation type.
<UL>
<LI>
substitution = <font color="#CC00FF">purple</font>
<LI>
insertion = <font color="#009900">green</font>
<LI>
deletion = <font color="#0000FF">blue</font>
<LI>
duplication = <font color="#FF9900">orange</font>
<LI>
complex = <font color="#643200">brown</font>
<LI>
unknown = black
</UL>
</p>
<p>
Labels:  By default items are labeled by the HGVS name.  To change this
click the desired option or options.  None will produce no labels no 
matter what else is selected.
</p>

<a name="source">
<H2>Variant Sources</H2>
<UL>
<LI>
The HbVar database describes variants and thalassemia mutations in the
globin loci.
<A HREF="http://www.bx.psu.edu/" TARGET="_blank">http://www.bx.psu.edu/</A>
<LI>
The PAHdb has mutations in the phenylalanine hydroxylase (PAH) gene.
<A HREF="http://www.pahdb.mcgill.ca/" TARGET="_blank">http://www.pahdb.mcgill.ca/</A>
<LI>
Coming soon: CFTR data.
(A few sample variants from this region have already been added.)
<LI>
Swiss-Prot data on variants within coding regions is available genome wide.
<LI>
A possible connection with WayStation, the HGVS community-wide database.
<A HREF="http://www.centralmutations.org" TARGET="_blank">http://www.centralmutations.org/</A>
</UL>

<H2>Data Filtering</H2>
<UL>
<LI>
Filtering can be performed on the location of the mutations with respect to
genes, i.e. to exclude those in exons, introns, UTR regions, or those
not within any known transcription unit.
<LI>
One can also exclude mutations by type: substitution, insertion, deletion,
duplication, or complex.
<LI>
Filtering by source is also provided.  The locus specific databases are 
listed as one source.
</UL>

<H2>Credits</H2>
<DL>
<DT>Center for Comparative Genomics and Bioinformatics, Penn State University, University Park, PA
<DD>Belinda Giardine, Ross Hardison, Webb Miller, Cathy Riemer
<DT>Erasmus University, Rotterdam, Netherlands
<DD>George Patrinos
<DT>Hospital Henri Mondor, Creteil, France
<DD>Henri Wajcman
<DT>Boston University
<DD>David H.K. Chui
<DT>U. Athens and IIBEAA, Athens, Greece
<DD>Nicholas Anagnou
<DT>Macedonian Academy of Sciences and Arts, RCGEB, Skopje, Macedonia
<DD>Georgi D. Efremov
<DT>Weatherall Institute of Molecular Medicine, Oxford UK
<DD>Richard Gibbons, Doug Higgs, Jim Hughes
<DT>Johns Hopkins University School of Medicine, Baltimore, MD
<DD>Garry Cutting, Andrew P. Feinberg
<DT>Center for Biomolecular Science and Engineering, University of California, Santa Cruz, CA
<DD>Fan Hsu, Jim Kent, Andrew Kern, Robert Kuhn, Heather Trumbower
</DL> 

