<H2>Description</H2>
<P>
The potential pseudo fragment track shows genomic fragments that are likely to 
be pseudogenes. Compared with the homologous known gene in the same genome, 
the potential pseudo fragment has pseudogene-like features 
such as frameshift, premature stop codon, non conserved splicing sites, and so on. 
The structure of a potential pseudo fragment shown in the browser is based on 
the gene structure of its homologous known gene.
</P>

<H2>Methods</H2>
<P> 
A set of pre-aligned human known genes is mapped across the human 
genome through the Human Blastz Self Alignment. We call this method homologous
mapping method. The fragments identified by this method 
are homologues of known genes. The known genes we used are from 
Consensus CDS (CCDS) gene set, RefSeq gene set, or MGC gene set, 
which are currently the most reliable 
gene sets. These genes have been aligned to the human genome and 
their locations in the genome are available at the UCSC Genome Browser.  

</P>
<P> 
We compared each homologous fragment with its known gene and 
collected a set of features to determine whether the homologous 
fragment is a gene.
The features include sequence identity, dN/dS ratio, 
the splicing sites, number of premature stop codon, and so on. 
These homologous fragments either belong to real genes or belong to
pseudogenes. We took homologous fragments overlapping known genes as  
positive samples and  overlapping known pseudogenes as negative samples.
We used them to train Support Vector Machines (SVMs) to learn how to separate 
pseudo fragments from pseudo fragments. The trained SVMs are used to
classify homologous fragments into potential pseudo elements or potential pseudo
elements. Finally, a heuristic filter is used to correct
some misclassified fragments and to determine whether the homologous fragment is
likely to be a pseudogene. 
</P>

<H2>References</H2>
<P>
Kent, W.J., Baertsch, R., Hinrichs, A., Miller, W., and Haussler, D.
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1" 
TARGET=_blank>Evolution's cauldron: 
Duplication, deletion, and rearrangement in the mouse and human genomes</A>. 
<I>Proc Natl Acad Sci USA</I> <B>100</B>(20), 11484-11489 (2003).

<P>
Schwartz, S., Kent, W.J., Smit, A., Zhang, Z., Baertsch, R., Hardison, R., 
Haussler, D., and Miller, W.
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103" TARGET=_blank>Human-Mouse 
Alignments with BLASTZ</A>. <em>Genome Res.</em> <B>13</B>(1), 103-7 (2003).
</P>

<P>
Pruitt, K.D., Tatusova, T., and Maglott D.R.

<A HREF="http://nar.oupjournals.org/cgi/content/full/33/suppl_1/D501" 
TARGET=_blank>NCBI Reference Sequence (RefSeq): a curated non-redundant 
sequence database of genomes, transcripts and proteins</A>. 
<I>Nucl. Acids Res.</I> <B>33</B>(1), D501-D504 (2005). 
</P>


<H2>Credits</H2>
<P>
The homologous mapping method and this browser track were developed by 
<A HREF="mailto:&#121;&#116;&#108;&#117;@s&#111;e.&#117;&#99;sc.&#101;&#100;&#117;">Yontao Lu</A> at UCSC.</P>

