<H2>Description</H2>
<P>
N-SCAN and N-SCAN EST gene predictions.
</P>

<H2>Methods</H2>
<H3>N-SCAN</H3>
<P>
N-SCAN combines biological-signal modeling in the target genome sequence along
with information from a multiple-genome alignment to generate de novo gene
predictions. It extends the TWINSCAN target-informant genome pair to allow for
an arbitrary number of informant sequences as well as richer models of
sequence evolution. It models the phylogenetic relationships between the
aligned genome sequences, context-dependent substitution rates, insertions,
and deletions.
</P>
<P>
Human N-SCAN uses mouse (mm5) as the informant.
</P>

<H3>N-SCAN EST</H3>
<P>
N-SCAN EST combines EST alignments into N-SCAN. Similar to the 
conservation sequence models in TWINSCAN, separate probability models 
are developed for EST alignments to genomic sequence in exons, introns, 
splice sites and UTRs, reflecting the EST alignment patterns in these 
regions. N-SCAN EST is more accurate than N-SCAN while retaining the 
ability to discover novel genes to which no ESTs align.
</P>
<P>
Human N-SCAN EST useds uses mouse (mm5),
rat (rn3), and chicken (galGal2) as informants.
</P>

<H2>References</H2>
<P>
Gross, S.S. and M.R. Brent, 
<A HREF="http://www.stanford.edu/~ssgross/RECOMB-2005.pdf">
Using Multiple Alignments To Improve Gene Prediction</A>, in
<EM>9th Annual International Conference, RECOMB 2005</EM>,
S. Miyano, et al., Editors. 2005, Springer: Boston. p. 374-388.
</P>
<P>
Korf, I., Flicek, P., Duan, D. and Brent, M.R. 
<A HREF="http://bioinformatics.oupjournals.org/cgi/reprint/17/suppl_1/S140.pdf"
TARGET=_blank>Integrating genomic homology into gene structure prediction</A>.
<em>Bioinformatics</em> <B>17</B>, S140-148 (2001).</P>

<H2>Credits</H2>
<P>
Thanks to <A HREF="http://genes.cs.wustl.edu/BrentLab/MB-Lab-index.html" 
TARGET=_blank>Michael Brent's Computational Genomics Group</A> at Washington 
University St. Louis for providing this data.
</P>
