<H2>Description</H2>
<P>
This track shows predicted "novel" single exon genes (SEGs) from several sources
-- i.e., predicted SEGs that do not overlap known genes or spliced mRNAs/ESTs.</P>

<H2>Display Conventions and Configuration</H2>
<P>
This is a composite track, with sub-tracks for the source gene predictions.
The sub-tracks follow the ordinary display conventions for gene prediction
tracks.  At the bottom are two additional sub-tracks that summarize
information across sources.  The first indicates "loci" (maximal intervals
spanned by overlapping predictions on the same strand) supported by
predictions from two or more independent sources.  The second indicates
loci containing predictions that have passed all validation 
filters (see below).</P>

<H2>Methods</H2>
<P>
For each source, we extracted all predictions of genes having single
CDS exons, then discarded those overlapping genes from the RefSeq,
Vega, UCSC Known Genes, and MGC sets, and those overlapping spliced
mRNAs or ESTs from Genbank.  Predictions that fell in introns of known
genes on the same strand were also discarded (because they are more
likely to be alternative exons than separate genes), as were
predictions that overlapped predicted processed pseudogenes.  The
predictions labeled "best" also have: (1) at least 80% overlap with
the syntenic nets between human and mouse, rat, and dog; (2) at least
40% of their CDS bases in "most conserved" regions according to
phastCons; (3) "end-to-end" homology with a known vertebrate gene
(i.e., a BLASTP alignment covering >85% of both query and target); and
(4) no frame-shift indels or nonsense mutations in human/mouse/rat/dog
alignments.</P>

<H2>Credits</H2>
<P>
Thanks to <A
HREF="http://genes.cs.wustl.edu/BrentLab/MB-Lab-index.html"
TARGET=_blank>Michael Brent's Computational Genomics Group</A> at
Washington University St. Louis for providing the N-SCAN and Twinscan
predictions, the <A HREF="http://www.ensembl.org"
TARGET=_blank>ENSEMBL project</A> for providing their predictions, and
the <A HREF="http://www.imim.es/grib/eng/default.htm"
TARGET=_blank>Grup de Recerca en Informàtica Biomèdica (GRIB)</A> at
Institut Municipal d'Investigació Mèdica (IMIM) in Barcelona for
providing the SGP predictions.  TransMap was developed by Mark
Diekhans at UC Santa Cruz and ExoniPhy was developed by Adam Siepel at
UC Santa Cruz (now Cornell).</P>

<H2>References</H2>
<P>
Gross, S.S. and Brent, M.R.
<A HREF="http://www.stanford.edu/~ssgross/RECOMB-2005.pdf" TARGET=_blank>Using 
multiple alignments to improve gene prediction</A>. In
<EM>9th Annual International Conference, RECOMB 2005</EM>,
Miyano, S. <em>et al</em>., ed. Springer: Boston. p. 374-388 (2005).
</P>
<P>
Hubbard, T. <em>et al.</em>. 
<A HREF="http://nar.oupjournals.org/cgi/content/full/30/1/38" 
TARGET=_blank>The Ensembl genome database project</A>.
<em>Nucleic Acids Research</em> <B>30</B>(1), 38-41 (2002).</P>
<P>
Korf, I., Flicek, P., Duan, D. and Brent, M.R.
<A HREF="http://bioinformatics.oupjournals.org/cgi/reprint/17/suppl_1/S140.pdf"
TARGET=_blank>Integrating genomic homology into gene structure prediction</A>.
<em>Bioinformatics</em> <B>17</B>, S140-148 (2001).</P>
<P>
Parra, G. <em>et al.</em>.  <A
HREF="http://www.genome.org/cgi/content/abstract/13/1/108"
TARGET=_blank>Comparative gene prediction in human and mouse</A>.
<em>Genome Res</em>, 13:108-117 (2003).</P>
<P>
Siepel, A. and Haussler, D. 
<A HREF="http://www.cse.ucsc.edu/~acs/recomb2004.pdf"
TARGET=_blank>Computational identification of evolutionarily conserved
exons</A>. <em>RECOMB '04</em> (2004).</P>
