<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=185>
  <TR><TD ALIGN=RIGHT>
    <A HREF=""http://www.genome.gov/pressDisplay.cfm?photoID=42"
    TARGET=_blank>
    <IMG SRC="/images/Felis_catus.jpg" WIDTH=*** HEIGHT=*** ALT="Cat"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Felis catus</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      (NHGRI <A HREF="http://www.genome.gov/pressDisplay.cfm?photoID=42"
              TARGET=_blank>Press Photos</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The Dec. 2008 <em>Felis catus</em> draft assembly
(NHGRI/Genome Technology Branch (NCBI project <A HREF="http://www.ncbi.nlm.nih.gov/sites/entrez?Db=genomeprj&cmd=ShowDetailView&TermToSearch=10703"
TARGET=_blank>10703</A>, <A HREF="ftp://ftp.ncbi.nlm.nih.gov/genbank/genomes/Eukaryotes/vertebrates_mammals/Felis_catus/catChrV17e/"
TARGET=_blank>GCA_000001445.1</A>))
was produced by the <A HREF="http://www.beckmangenomics.com/"
TARGET=_blank>Agencourt Bioscience Corporation</A> and <A
HREF="http://www.hillspet.com/" TARGET=_blank>Hill's Pet Nutrition, Inc.</A>.
The mitochondrial genome was sequenced by <A HREF="http://home.ncifcrf.gov/ccr/lgd/comparative_genome/catgenome/index_n.asp"
Target=_blank>Laboratory of Genomic Diversity.</A>

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
Cat genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>chrA3</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all of chromosome A3</TD>
    </TR>
    <TR><TD VALIGN=Top>chrA3:1-1000000</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first million bases of chr A3/TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chrA3:1000000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chr A3 that spans 2000 bases, starting with position 1000000 </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>BX950328</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession
		number BX950328</TD>
    </TR>
    <TR><TD VALIGN=Top>BU335953</TD>
        <TD WIDTH=14>&nbsp;</TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession
		BU335953 on chr Z</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>pseudogene mRNA</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes, but not cDNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>homeobox caudal</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top>zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top>zhang</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Zhang</TD>
    </TR>
    <TR><TD VALIGN=Top>Taguchi,T.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author T. Taguchi</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Taguchi T</em>, internally it
	uses the format <em>Taguchi,T.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>

<P>
This assembly adds 0.9X coverage (from 6 other breeds and one wild cat) 
to the cinnamon assembly (1.9X, ANG00000000) to 2.8X coverage 
using paired-end fosmid ABI 3730 reads.
See also: <A HREF="http://www.ncbi.nlm.nih.gov/sites/entrez?db=genomeprj&cmd=Retrieve&dopt=Overview&list_uids=32759"
TARGET=_blank>Genome sequencing project by NHGRI/Genome Technology Branch </A> description.
</P>

<P>
There are 19 chromosomes, chrA1 - A3, chrB1 - B4, chrC1 - C2, 
chrD1 - D4, chrE1 - E3, chrF1 - F21, chrX plus the mitochondrial
sequence chrM from genbank accession NC_001700 for a total sequence
length of 2,872,644,707 bases. There are 1,165,972,091 'N' bases in gaps,
leaving 1,706,672,616 ACGT bases.
</P>


<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/felCatV17e/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#cat">Downloads</A> page. 


The Cat browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#cat_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
