<H2>Description</H2>
<P>
This track shows the RNA-seq data published by
<A HREF="http://genes.mit.edu/burgelab/" TARGET=_BLANK>Chris Burge's lab</A>
(Wang <EM>et al.</EM>) mapped to the genome using GEM Mapper by the 
<A HREF=http://pasteur.crg.es/portal/page/portal/Internet/02_Research/01_Programmes/01_Bioinformatics_Genomics/01_Bioinformatics_Genomics
TARGET=_BLANK>Guig&oacute; lab</A> at the Center for Genomic Regulation
(<A HREF=http://www.crg.es TARGET=_BLANK>CRG</A>). The subtracks display
RNA-seq data from different tissues/cell lines:
<ol>
<li>Brain</li>
<li>Liver</li>
<li>Heart</li>
<li>Muscle</li>
<li>Colon</li>
<li>Adipose</li>
<li>Testes</li>
<li>Lymph Node</li>
<li>Breast</li>
<li>BT474 - Breast Tumour Cell Line</li>
<li>HME - Human Mammary Epithelial Cell Line</li>
<li>MCF7 - Breast Adenocarcinoma Cell Line</li>
<li>MB-435 - Breast Ductal Adenocarcinoma Cell Line*</li>
<li>T-47D - Breast Ductal Carcinoma Cell Line</li>
</ol>
*NOTE: studies have shown that the MDA-MB-435 cell line appears to have been
contaminated with the M14 melanoma cell line. See this 
<A HREF="http://www.atcc.org/ATCCAdvancedCatalogSearch/ProductDetails/tabid/452/Default.aspx?ATCCNum=HTB-129&Template=cellBiology" TARGET=_BLANK>entry</A> on
the American Type Culture Collection (ATCC) website for more details.

<H2>Methods</H2>
<P>
The group at CRG obtained RNA-seq reads generated by Wang <EM>et al</EM>. 
(2008), from the Short Read Archive section of GEO at NCBI under accession 
number <A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE12946" 
TARGET=_BLANK>GSE12946</A>. Using their GEM mapper program, CRG mapped the 
RNA-seq reads to the genome and transcriptome (GENCODE Release 3, October 
2009 Freeze). GEM mapper was run using default parameters and allowing up 
to two mismatches in the read alignments. Since mapping to the transcriptome 
depends on length of the reads mapped, reads were only mapped for the 
14 tissues or cell lines where reads were of length 32 bp. This excluded
reads from MAQC human cell lines (mixed human brain) and MAQC UHR (mixed 
human cell lines). 

<H2>Credits</H2>
<P>
These data were generated by Chris Burge's lab at the Massachusetts Institute
of Technology and by Roderic Guig&oacute;'s lab at the Center for Genomic
Regulation (CRG) in Barcelona, Spain. GTF files of the mapped data were
provided by Thomas Derrien from CRG. GEM mapper sofware can be obtained
<A HREF="http://sourceforge.net/apps/mediawiki/gemlibrary/index.php?title=The_GEM_library"
TARGET=_BLANK>here</A>.

<H2>References</H2>
<P>
<BR>Wang ET, Sandberg R, Luo S, Khrebtukova I, Zhang L, Mayr C, Kingsmore SF, 
Schroth GP, Burge CB.
<A HREF="http://www.nature.com/nature/journal/v456/n7221/abs/nature07509.html"
TARGET=_BLANK>Alternative isoform regulation in human tissue transcriptomes</A>.
<I>Nature</I>. 2008 Nov 27;456(7221):470-6.
</P>
