<h2>Description</h2>

<p>
Retrotransposition is a process involving the copying of DNA by a group of
enzymes that have the ability to reverse transcribe spliced mRNAs, resulting
in single-exon copies of genes and sometime chimeric genes. RetroGenes can be
either functional genes that have acquired a promoter from a neighboring gene,
non-functional pseudogenes, or transcribed pseudogenes.
</p>

<h2>Methods</h2>

<p>
All mRNAs of a species from GenBank were aligned to the genome using
<a href="http://www.bx.psu.edu/~rsharris/lastz/" target="_blank">lastz</a>
(Miller lab, Pennsylvania State University). mRNAs that aligned twice in the
genome (once with introns and once without introns) were initially screened.
Next, a series of features were scored to determine candidates for retrotranspostion
events. These features include position and length of the polyA tail, degree of
synteny with mouse, coverage of repetitive elements, number of exons that can still
be aligned to the retrogene and degree of divergence from the parent gene. Retrogenes
are classified using a threshold score function that is a linear combination
of this set of features. Retrogenes in the final set are selected using
a score threshold based on a ROC plot against the
<a href="http://vega.sanger.ac.uk" target="_blank">Vega</a> annotated pseudogenes.
</p>

<h3>Retrogene Statistics table:</h3>

<ul>
  <li><b>Expression of retrogene</b>: The following values are possible where
  those that are not expressed are classed as <em>pseudogene</em> or
  <em>mrna</em>:</li>
  <ul>
    <li><em>pseudogene</em> indicates that the parent gene has been annotated
    by one of NCBI's RefSeq, UCSC Genes or Mammalian Gene Collection (MGC).</li>
    <li><em>mrna</em> indicates that the parent gene is a spliced mrna that
    has no annotation in NCBI's RefSeq, UCSC Genes or Mammalian Gene Collection
    (MGC). Therefore, the retrogene is a product of a potentially non-annotated
    parent gene and is a putative pseudogene of that putative parent gene.</li>
    <li><em>expressed weak</em> indicates that there is a mRNA overlapping
    the retrogene, indicating possible transcription. <em>noOrf</em> indicates
    that an ORF was not identified by BESTORF.</li>
    <li><em>expressed</em> indicates that there is a medium level of mRNAs/ESTs
    mapping to the retrogene locus, indicating possible transcription.</li>
    <li><EM>expressed strong</em> indicates that there is a mRNA overlapping
    the retrogene, and at least five spliced Ests indicating probable transcription.
    <em>noOrf</em> indicates that an ORF was not identified by BESTORF.</li>
    <li><em>expressed shuffle</em> indicates that the retrogene was inserted into
    a pre-existing annotated gene.</li>
  </ul>
  <li><b>Score</b>: Based on features of the potential retrogene.</li>
  <li><b>Percent Gene Alignment Coverage (Bases matching Parent)</b>: shows
  the percentage of the parent gene aligning to this region.</li>
  <li><b>Intron Count</b>: Number of introns is the number of gaps in
  the alignment between the parent mRNA and the genome where gaps are &gt;80 bp and
  the ratio of the mRNA alignment gap to the genome alignment gap is less than
  30% after removing repeats.</li>
  <li><b>Gap Count</b>: Numer of gaps in the alignment of between the parent
  mRNA and the genome after removing repeats. Gaps are not counted if the gap on
  the mRNA side of the alignment is a similar size to the gap in the genome
  alignment.</li>
  <li><b>BESTORF Score</b>:
  <a href="http://linux1.softberry.com/berry.phtml?topic=bestorf&group=help&subgroup=gfind"
  target="_blank">BESTORF</a> (written by Victor Solovyev) predicts potential open reading
  frames (ORFs) in mRNAs/ESTs with very high accuracy using a Markov chain model of coding
  regions and a probabilistic model of translation start codon potential. The score threshold
  for finding an ORF is 50 (Jim Kent, personal communication).</li>
</ul>

<h3>Break in Orthology table:</h3>

<p>
Retrogenes inserted into the genome since the human/mouse divergence show
a break in the mouse genome syntenic net alignments to the human genome.
The percentage break represents the portion of the genome that is missing in
each species relative to the reference genome (human hg19) at the retrogene
locus as defined by syntenic alignment nets. Breaks in orthology with mouse
and dog tend to be due to genomic insertions in the primate lineage. Relative
orthology of dog/human and Rhesus macque/human nets are used to avoid false
positives due to deletions in the mouse genome. Older retrogenes will not show
a break in orthology so this feature is weighted lower than other features when
scoring putative retrogenes.
</p>

<p>
These features can be downloaded from the table retroMrnaInfo in many
formats using the Table Browser option from the Tools menu on the top blue bar.
</p>

<h2>Credits</h2>

<p>
The RetroFinder program and browser track were developed by
<a href="mailto:&#98;a&#101;r&#116;&#115;&#99;&#104;&#64;&#115;&#111;&#101;.
&#117;&#99;sc.&#101;&#100;&#117;">Robert Baertsch</a> at UCSC.
</p>
<!-- above address is baertsch at soe.ucsc.edu -->

<h2>References</h2>




<P>
Pei B, Sisu C, Frankish A, Howald C, Habegger L, Mu XJ, Harte R, 
Balasubramanian S, Tanzer A, Diekhans M <I>et al.</I>
<A HREF="http://genomebiology.com/content/13/9/R51">The GENCODE pseudogene resource.</A>
<I>Genome Biology</I> 2012 Sep 26;13(9):R51.      
<P>
Baertsch R, Diekhans M, Kent J, Haussler D, Brosius J. 
<a href="http://www.biomedcentral.com/1471-2164/9/466">Retrocopy contributions to the evolution of the human genome.</a>
<I>BMC Genomics</I> 2008 Oct 8;9:466.
<P>
Deyou Zheng, Adam Frankish, Robert Baertsch, Philipp Kapranov, Alexandre Reymond , Siew Woh Choo , Yontao Lu , France Denoeud , Stylianos E Antonarakis , Michael Snyder , Yijun Ruan, Chia-Lin Wei , Thomas R. Gingeras , Roderic Guigo , Jennifer Harrow , and Mark B. Gerstein 
<a href="http://genome.cshlp.org/cgi/content/full/17/6/839">Pseudogenes in the ENCODE Regions: Consensus Annotation, Analysis of Transcription and Evolution.</a>
<I>Genome Res.</i> 2007 Jun;17(6):839-51.
<P>
Kent, W.J., Baertsch, R., Hinrichs, A., Miller, W., and Haussler, D.
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1" 
TARGET=_blank>Evolution's cauldron: 
Duplication, deletion, and rearrangement in the mouse and human genomes</A>. 
<I>Proc Natl Acad Sci USA</I> <B>100</B>(20), 11484-11489 (2003).
<P>
Schwartz, S., Kent, W.J., Smit, A., Zhang, Z., Baertsch, R., Hardison, R., 
Haussler, D., and Miller, W.
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103" TARGET=_blank>Human-Mouse 
Alignments with BLASTZ</A>. <em>Genome Res.</em> <B>13</B>(1), 103-7 (2003).
</P>

