<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <title>wgEncodeHistoneSuper.html</title>
</head>
<body>
<h2>Description</h2>
<p>These tracks display maps of chromatin state of <a
 href="http://www.genome.gov/10005107">ENCODE</a> cell types.
 Histones are a group of closely-related proteins that together form
 the nucleosome: H3K4, H3K9, H3K14, H3K27, H3K79, H4K20, and H2BK5.
 When histone proteins undergo pust-transcriptional modifications at
 specific amino acids (methylation, acyetylation), these modifications
 (referred to as <em>marks</em>) perform a role in regulating the
 winding of the DNA around the nucleosome.  This in turn regulates
 gene expression by controlling the accessibility of the chromatin.
 Histone marks vary in their effect. Promoters, enhancers, transcribed
 regions, and silenced regions are each associated with specific
 histone marks.  <br>

 The ChIP-seq method enables identifying regions of DNA that interact
 with specific histone marks.  This method involves cross-linking
 histones and other DNA associated proteins to genomic DNA within
 cells using formaldehyde. The cross-linked chromatin is subsequently
 extracted, mechanically sheared, and immunoprecipitated using
 specific antibodies. This has the effect of capturing segments of DNA
 that are bound to the protein selected by the antibody.  After
 reversal of cross-links, the immunoprecipitated DNA is sequenced and
 mapped to the human reference genome. The relative enrichment of each
 antibody-target (epitope) across the genome is inferred from the
 density of mapped fragments.  If the antibody is specific to a given
 histone mark, then this procedure identifies segments of the genome
 associated with that mark.<br>

</p>
<h2>Display Conventions</h2>
These tracks are multi-view composite tracks that contains multiple
data types (views). For each view, there are multiple subtracks that
display individually on the browser.&nbsp; Each view within each track
has separate display controls, as described <a
 href="http://hgwdev.cse.ucsc.edu/goldenPath/help/multiView.html">here</a>.&nbsp;
Specific
aspects
of the views vary by track.&nbsp;&nbsp; The views
include:<br>
<br>
<div style="margin-left: 40px;"><span style="font-style: italic;">Peaks</span><br>
<div style="margin-left: 40px;">Regions of significant ChIP-Seq
affinity.<br>
</div>
<span style="font-style: italic;">Signal</span><br>
<div style="margin-left: 40px;">The density of tags mapping within a
sliding window.<br>
</div>
<span style="font-style: italic;">Raw Signal</span><br>
<div style="margin-left: 40px;">Density graph (wiggle) of signal
enrichment based on aligned read density.<br>
</div>
</div>
<h2>Credits<br>
</h2>
These data were generated and analyzed as part of the <a
 href="http://www.genome.gov/10005107">ENCODE</a> project, a
genome-wide consortium project with the aim of cataloging all
functional elements in the human genome.&nbsp; This effort includes
collecting a variety of data over a specific set of <a
 href="http://genome.ucsc.edu/ENCODE/cellTypes.html">cell types</a>.&nbsp;
Consequently,
data
related
to these tracks be available under ENCODE
tracks. &nbsp;&nbsp;
<br>
<h2>References</h2>
Segal E, Widom J. <a href="http://www.ncbi.nlm.nih.gov/pubmed/19596482"><span
 style="font-style: italic;">What controls nucleosome positions?</span></a>&nbsp;
Trends
Genet.
2009 Aug;25(8):335-43.<br>
<br>
Geiman TM, Robertson KD.&nbsp; <a
 href="http://www.ncbi.nlm.nih.gov/pubmed/12244565"><span
 style="font-style: italic;">Chromatin remodeling, histone
modifications, and DNA methylation-how does it all fit together?</span></a>&nbsp;
J
Cell
Biochem. 2002;87(2):117-25.<br>
<br>
<h2>Data Release Policy</h2>
Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset
until nine months following the release of the dataset. This date is
listed in the Restricted Until column on the track configuration page
and the download page. The full data release policy for ENCODE is
available <a href="http://genome.ucsc.edu/ENCODE/terms.html">here</a>.<br>
<br>
</body>
</html>
