<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<HTML>
<body>

<h2>Description</h2>

<p>The ENCODE project has revealed the functional elements of segments
of the human genome in unprecedented detail. However, the ability to
distinguish between transcripts designated for translation into protein
and those that serve purely regulatory roles remains elusive. A
standard means to determine if translation is occuring is to measure 
protein produced by transcripts via mass spectrometry-based proteogenomic
mapping. In this process, proteins were digested to peptides using protease 
such as trypsin and these petides were chromatographically fractionated and
fed into a tandem mass spectrometer (MS/MS). This process creates a signature 
series of fragment masses which can be scanned against the theoretical 
translation and proteolytic digest of an entire genome to identify the genomic 
origins of sample proteins (Giddings <EM>et al.</EM> 2003).</p>

<p>The proteogenomic track displays mass spectrometry data that have been 
matched to genomic sequences for selected cell lines, using a workflow and 
software specifically designed for this purpose.  The tracks can be used to 
identify which parts of the genome are translated into proteins, to verify 
which transcripts discovered by other ENCODE experiments are protein-coding,
to reveal new genes and/or splice variants and proteins with post-translational 
modifications (PTM). Of particular interest is the possibility
of uncovering the translation of small open reading frames (ORFs), antisense transcripts, 
or protein encoding region that have been annotated as introns previously.
</p>

<h2>Display Conventions and Configuration</h2>

<p>The display for this track shows peptide mappings as contiguous,
rectangular items rendered in grayscale according to their
score, with darker items representing higher-confidence peptide mappings.  
The name of each item is the amino acid sequence of the peptide and a period (.) 
at the end of a name, it signifies a stop codon.  
<br>
</p>
<p>In addition to the common fields in genome browser display, these Proteogenomics tracks
have some unique fields and some of which are common to both regular and PTM peptides and some are 
specific to regular peptides or PTM peptides only <br>
</p>
<ul>

<p>
Fields common to both regular and PTM peptides:
</p>
  <li>The <span style="font-style: italic;">Item</span> shows the peptide sequence for 
regular pepitde mapping and peptide sequence appended with the integer fraction of the 
PTM mass for PTM peptide mapping. These appear in the left or right space besides the main 
Genome Browser display window depending on the configuration</li>
  <li>The <span style="font-style: italic;">Score</span> is derived from the 'Raw score 
for a peptide/spectrum match' (see below) given by the proteomics peptide mapping software 
Peppy (Giddings <EM>et al.</EM> 2003).  Score is computed as 
[RawScore - RawScore at 10% FDR cutoff) / (RawScore at near 0% FDR cutoff - RawScoreat 10% FDR cutoff)] * 1000, 
and is then converted to an integer.  Raw scores above the 0% FDR threshold have a score of 1000 (best), 
while those below the 10% FDR threshold have a score of 0 (worst).  The score is used by 
the Genome Browser to render shade to the displaying solid rectangular.</li>
  <li>The <span style="font-style: italic;">rawScore </span> is the original peptide mapping score given by Peppy, 
which reflects the strength of the peptide mapping; as an estimated <span style="font-style: italic;">p</span>-value, 
indicating the confidence of the peptide mapping.</li>
  <li>The <span style="font-style: italic;">spectrumId </span> is the original identifier of the spectrum 
that mapped to the peptide sequence, and it can be used to track the original spectrum.<br></li>
  <li>The <span style="font-style: italic;">peptideRank </span>indicates the rank of each peptide/spectrum 
mapping.  A spectrum can be chimeric, containing more than one peptide, and the spectrum can be
mapped to two or more distinct peptides.  Here, we are only reporting the top-scoring match.  
If more than one peptide 'tied' for the top score, then all peptides were included and all matches 
have a peptideRank of 1.<br></li>
<li>The <span style="font-style: italic;">peptideRepeatCount</span> Indicates the number of places 
in the genome where the peptide is encoded for a peptide/spectrum match.</li>
</ul>


<h2>Methods</h2>

<p>ENCODE cell lines K562 and GM12878 were used for large scale
proteomic analysis. Cell lines were cultured according to standard
<A HREF="/ENCODE/protocols/cell" target="_blank">ENCODE cell culture protocols</A>
and in-gel digestion was completed according to the standard
protocol (Shevchenko, et al. 2007). </p>
The proteolytic enzyme trypsin was used to digest the proteins
in order to produce short, MS/MS analyzable peptides. Trypsin is a
common protease that typically cleaves proteins after Arginine or Lysine. The
metadata parameter <i>enzyme</i> specifies the restriction enzyme used
for digestion. Tandem mass spectrometry (RPLC-MS/MS) analysis was then
performed on an Eksigent Ultra-LTQ Orbitrap system. However, due
to enzyme inefficiency, it does not always cleave at Arginine or Lysine, so there
may be peptides that include an uncleaved Arg/Lys site. The number of such
missed cleavages allowed in the search is described by the metadata
parameter <i>miscleavages</i>.
<p>We performed proteogenomic mapping (Jaffe, et al., 2004) with two missed cleavages
allowed and using the whole human genomic sequence (UCSC hg19) via the
genome fingerprint scanning (GFS) program (Giddings, et al. 2003)  and newly 
developed Peppy 
(<a href="http://www.peppyresearch.com/">http://www.peppyresearch.com</a>/). 
We used HMM_Score (Khatun, et al. 2008) to accurately match MS/MS
spectra to their corresponding genome sequences.  E-values are
calculated, which estimate the number of results at the given score
level which would be expected by random chance. We then empirically
derived the false discovery rate for a given E-Value using a decoy
database search and only those matches falling within the specified 5%
FDR rate (E-Value &lt;0.01) are included in the track. The results with
10% FDR (E-Value &lt;0.05) are available under the Downloads page as Raw Signal.</a> </p>
<h2>Credits</h2>
<p>Proteogenomic mapping: Dr. Jainab Khatun, Brian Risk, Mustaque
Ahamed, Christopher Maier, Dr. John Wrobel and Dennis Crenshaw (<A HREF=http://giddingslab.org/>Giddings Lab</A>). </p>
<p>Proteomic analysis: Drs. Yanbao Yu and Ling Xie (<A HREF=http://www.med.unc.edu/biochem/chenlab/chen-research-lab>Chen Lab</A>).</p>
Main Contact:
<A HREF="mailto:&#107;ha&#116;&#117;&#110;&#64;&#101;&#109;&#97;&#105;&#108;.
&#117;n&#99;.
&#101;&#100;&#117;">
Jainab Khatun</A>
<!-- above address is khatun at email.unc.edu -->

<h2>References</h2>

<p>
Giddings MC, Shah AA, Gesteland R, Moore B.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/12518051" target="_blank">
Genome-based peptide fingerprint scanning</a>.
<em>Proc Natl Acad Sci U S A</em>. 2003 Jan 7;100(1):20-5.
</p>

<p>
Jaffe JD, Berg HC, Church GM.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/14730672" target="_blank">
Proteogenomic mapping as a complementary method to perform genome annotation</a>.
<em>Proteomics</em>. 2004 Jan;4(1):59-77.
</p>

<p>
Khatun J, Hamlett E, Giddings MC.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/18187442" target="_blank">
Incorporating sequence information into the scoring function: a hidden Markov model for improved peptide identification</a>.
<em>Bioinformatics</em>. 2008 Mar 1;24(5):674-81.
</p>

<p>
Shevchenko A, Tomas H, Havlis J, Olsen JV, Mann M.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/17406544" target="_blank">
In-gel digestion for mass spectrometric characterization of proteins and proteomes</a>.
<em>Nat Protoc</em>. 2006;1(6):2856-60.
</p>
 
<h2>Data Release Policy</h2>

<p>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset
until
nine months following the release of the dataset. This date is listed
in
the <em>Restricted Until</em> column on the track configuration page
and
the download page. The full data release policy for ENCODE is available
<a href="../ENCODE/terms.html" target="_BLANK">here</a>.</p>
</body>
</html>

<BR>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</TD></TR></TABLE>

</BODY></HTML>
