<H2>Description</H2>

<P>This track is produced as part of the ENCODE Project. 
RNA-Seq is a method for mapping and quantifying the transcriptome of any organism that has a genomic DNA sequence assembly. 
RNA-Seq is performed by reverse-transcribing an RNA sample into cDNA, followed by high throughput DNA sequencing, 
which was done here on an Illumina Genome Analyzer (GA2) (Mortazavi et al., 2008). The transcriptome measurements shown on these tracks were performed on 
<A target="_BLANK" href="http://genome-test.cse.ucsc.edu/cgi-bin/hgEncodeVocab?term=longPolyA&type=rnaExtract">polyA selected RNA</A> from 
<A target="_BLANK" href="http://genome-test.cse.ucsc.edu/cgi-bin/hgEncodeVocab?term=cell&type=localization">total cellular RNA</A> using two different protocols û
 one that preserves information about the strand the read is coming from and one that does not. Due to the specifics of the enzymology of library construction gene 
 and transcript quantification is more accurate based on the non-strand-specific protocol, while the strand-specific protocol is useful for assigning strandedness but 
 in general less reliable for quantification.</P>
 
<H3>Non-strand-specific protocol:</H3>

<P>PolyA-selected RNA was fragmented by magnesium-catalyzed hydrolysis and then converted into cDNA by random priming and amplified. Data have been produced in two formats: 
single reads, each of which comes from one end of a cDNA molecule, and paired-end reads, which are obtained as pairs from both ends of cDNAs. This RNA-Seq protocol does not 
specify the coding strand. As a result, there will be ambiguity at loci where both strands are transcribed. The "randomly primed" reverse transcription is, apparently, not 
fully random. This is inferred from a sequence bias in the first residues of the read population, and this likely contributes to observed unevenness in sequence coverage 
across transcripts.</P>
<H3>Strand specific protocol:</H3>
<P>PolyA-selected RNA was fragmented by magnesium-catalyzed hydrolysis. 3' adapters were ligated to the 3' end of fragments, then  5' adapters were ligated to the 5' end. 
The resulting RNA molecules were converted to cDNA and amplified. This RNA-Seq protocol does specify the coding strand as each read is in the same 5'-3' orientation as the 
original RNA strand. As a result, loci where both strands are transcribed can be disambiguated. However, RNA ligation is an inherently biased process and as a result greater 
unevenness in sequence coverage across transcripts is observed compared to the non-strand-specific data, and quantification is less accurate.
</P>
<P>Sequence reads produced by both protocols are then informatically mapped onto the genome sequence (Alignments). Those that don't map to the genome are mapped to known RNA splice 
junctions (Splice Sites). These mapped reads are then counted to determine their frequency of occurrence at known gene models. Sequence reads that cluster at genome locations that 
lack an existing transcript model are also identified informatically and they are quantified. RNA-Seq is especially suited for giving information about RNA splicing patterns and for 
determining unequivocally the presence or absence of lower abundance class RNAs. As performed here, internal RNA standards are used to assist in quantification and to provide internal process controls. 
</P>

<H2>Display Conventions and Configuration</H2>
<P>
This track is a multi-view composite track that contains multiple data types
(<EM>views</EM>).  For each view, there are multiple subtracks that
display individually on the browser.  Instructions for configuring multi-view
tracks are <A href="http://genome-test.cse.ucsc.edu/goldenPath/help/multiView.html" target="_BLANK">here</A>.
The following views are in this track:
</P><DL>
<DT><I>Raw Signal</I>
</DT><DD>Density graph (wiggle) of signal enrichment based on a normalized aligned 
read density (Read Per Million, RPM). The RPM measure assists in visualizing the relative amount of a given transcript across multiple samples. 
A separate tracks for the forward (plus) and reverse (minuse) strands are provided for strand-sepcific data.
</DD><DT><I>Splice Sites</I>
</DT><DD>RNA-seq tags aligning to mRNA splice sites.
</DD><DT><I>Alignments</I>
</DT><DD>The Alignments view shows reads mapped to the genome. 
Alignments are colored by cell type.
</DD></DL>

<H2>Methods</H2>

<H3>Experimental Procedures</H3>

<P> 
Cells were grown according to the approved
<A href="http://genome-test.cse.ucsc.edu/ENCODE/protocols/cell" target="_BLANK">ENCODE cell culture protocols</A>.
The cells (either 2 X 10<SUP>7</SUP> or 4 X 10<SUP>7</SUP> cells ù 
GM12878 and K562, 
and 8 X 10<SUP>7</SUP> cells HepG2) 
were lysed in either 4mls (GM12878 and K562) or 12 mls (HepG2) of RLT buffer 
(Qiagen RNEasy kit),
and processed on either 2 (GM12878 and K562) or 3 (HepG2) RNEasy midi columns 
according to the 
manufacturer's protocol, with the inclusion of the "on-column" DNAse digestion 
step to remove 
residual genomic DNA.</P>  

<P>75 ╡gs of total RNA was selected twice with oligodT beads (Dynal) 
according to the manufacturer's 
protocol to isolate mRNA from each of the preparations.  For 2x75bp non-stranded RNA-Seq 100 ngs of mRNA was 
then processed according 
to the protocol in Mortazavi et al (2008), and prepared for sequencing on the 
Genome Analyzer flow cell
 according to the protocol for the ChIPSeq DNA genomic DNA kit (Illumina).  Strand-specific RNA-Seq libraries were prepared from 100ng
 of mRNA from the same preparation following <A href="http://www.illumina.com/applications/sequencing/rna.ilmn" target="_BLANK">Illumina's Strand-Specific RNA-Seq protocol </A></P>

 <H3>Data Processing and Analysis</H3>
 
<P>Reads were mapped to the genome using <A href="http://tophat.cbcb.umd.edu/"" target="_BLANK">TopHat</A>. TopHat is 
specifically designed to align RNA-Seq reads and identify splice junctions de novo. It was used with defaul settings with the 
exception of specifying an empirically determined mean insert length (size selection is performed with the goal of the resulting distribution 
being centered around 200bp, but as it is practically impossible to have precisely the same cut size for each sample, the actual distribution differs slightly between 
libraries, and it is desirable to use a parameter as close as possible to the actual value).</P>

<P>After mapping reads to the genome and identifying splice junctions, the data was further analyzed using the transcript assembly and quantification software 
<A href="http://cufflinks.cbcb.umd.edu/"" target="_BLANK">Cufflinks</A>. Cufflinks was used in two modes. First, expression for genes 
and invidiual transcripts was quantified based on the GENCODE annotation. Second, Cufflinks was run in de novo transcript discovery mode; the resulting 
gene models were compared to the GENCODE annotation, and novel, previously unannotated transcripts identified and added to the annotation, then 
gene and transcript-level expression levels were determined again with Cufflinks against the new annotation. </P>

<P>The following files can be found on the <A target="_BLANK" href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeCaltechRnaSeq/">downloads page</A>.
</P>

<P>
<b>.wiggle</b> read density files (<I>RawSignal</I> view, wiggle format)<br>
<b>.BAM</b> files of all alignments (<I>Alignments</I> view)<br>
<b>.spliced.BAM</b> files for all spliced reads (<I>Spliced Alignments</I> view)<br>
<b>.Gencode.genes.expr</b> files containing FPKM expression level estimates at the gene level. FPKM (Fragments Per Kilobase per Million reads, where a fragment is defined
as the nucleic acid fragment from which reads originate, and a pair of reads is counted as one fragment) is a 
metric analogous to the widley used RPKM (Reads Per Kilobase per Million reads), which normalizes against both transcript length and sequencing depth<br>
<b>.Gencode.transcript.expr</b> files containing FPKM expression level estimatesat the transcript level in FPKMs<br>
<b>.Gencode.denovo.gene.expr</b> files containing FPKM expression level measurements at the gene level including novel transcripts<br>
<b>.Gencode.denovo.transcript.expr</b> files containing FPKM expression level measurements at the transcript level including novel transcripts<br>
<b>transcripts.gtf</b> files containing all Cufflinks de novo assembly models in gtf format<br></P>

<H2>Verification</H2>

<UL>
<LI>Known exon maps as displayed on the genome browser are confirmed by the 
alignment of sequence reads. </LI>
<LI>Known spliced exons are detected at the expected frequency for transcripts 
of given abundance.   </LI>
<LI>Linear range detection of spiked in RNA transcripts from Arabidopsis and 
phage lambda over 5 orders of magnitude.   </LI>
<LI>Endpoint RTPCR confirms presence of selected RNAFAR 3&#8242;UTR extensions.
</LI>
<LI>Correlation to published microarray data r = 0.62 </LI>
</UL>

<H2>Credits</H2>

<P>Wold Group: Ali Mortazavi, Brian Williams, Georgi Marinov Diane Trout, Brandon King, Ken McCue, Lorian Schaeffer.</P>
<P>Myers Group: Norma Neff, Florencia Pauli, Fan Zhang, Tim Reddy, Rami Rauch, Chris Partridge</P> 
<P>Illumina gene expression group: Gary Schroth, Shujun Luo, Eric Vermaas.</P>

<P>
Contacts: 
<A href="mailto:georgi@caltech.edu">
Georgi Marinov</A> (data coordination/informatics/experimental).
<!-- above address is georgi at caltech.edu -->
<A href="mailto:diane@caltech.edu">
Diane Trout</A> (informatics) and
<!-- above address is diane at caltech.edu -->
<A href="mailto:bawilli_91125@yahoo.com">
Brian Williams
</A> (experimental).
<!-- above address is bawilli_91125 at yahoo.com -->

</P>

<H2>References</H2>

<P>Mortazavi A, Williams BA, McCue K, Schaeffer L, and Wold BJ. 
<A target="_BLANK" href="http://www.ncbi.nlm.nih.gov/pubmed/18516045">Mapping and quantifying mammalian transcriptomes by RNA-Seq</A>
<EM>Nature Methods</EM>. 2008 Jul; 5(7):621-628.</P> 

<P>Langmead B, Trapnell C, Pop M, Salzberg SL.
<A target="_BLANK" href="http://genomebiology.com/2009/10/3/R25"><U>Ultrafast and memory-efficient alignment of short DNA sequences to the human genome</U></A>
<EM>Genome Biology</EM>. 2009 Mar; 10:R25.</P>

<P>Trapnell C, Pachter L, Salzberg SL..
<A target="_BLANK" href="http://bioinformatics.oxfordjournals.org/cgi/content/full/25/9/1105"><U>TopHat: discovering splice junctions with RNA-Seq.</U></A>
<EM>Bioinformatics</EM>. 2009 May; 25(9):1105-11.</P>

<P>Trapnell C, Williams BA, Pertea G, Mortazavi A, Kwan G, van Baren MJ, Salzberg SL, Wold BJ, Pachter L.
<A target="_BLANK" href="http://www.nature.com/nbt/journal/v28/n5/abs/nbt.1621.html"><U>Transcript assembly and quantification by RNA-Seq reveals unannotated transcripts and isoform switching during cell differentiation.</U></A>
<EM>Nat Biotechnology</EM>. 2010 May; 28(5):511-5.</P>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column, above.  The full data release policy
for ENCODE is available
<A href="http://genome-test.cse.ucsc.edu/ENCODE/terms.html" target="_BLANK">here</A>.</P>


