<H2>Description</H2>
<P>
These tracks are high throughput RNA sequencingi from <A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=cellType">ENCODE cell
lines</A> generated by the ENCODE consortia.  They contain information
about human RNAs > 200 nucleotides obtained as short reads off the Illumina GAIIx platform. Data
is available from biological replicates of several cell lines. In addition to profiling <A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=rnaExtract">
Poly-A+ RNA samples</A> from
whole cells there is also data from various <A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=localization">
subcellular compartments</A> as well and profiles from
Poly-A depleted (-) RNA. In many cases, there are also <A TARGET=_BLANK HREF="/cgi-bin/hgTrackUi?hgsid=1961355&c=chr21&g=wgEncodeRikenCage"> CAGE</A>, Small RNA-Seq and PET-diTAG datasets
available from the same biological replicates.
</P>

<H2>Display Conventions and Configuration</H2>
This track is a multi-view composite track that contains the following
<EM>views</EM>:
<DL>

<DT><i>Alignments</i>
<DD>The Alignments view shows reads mapped to the genome.  Sequences determined 
to be transcribed on the positive strand are shown in <span style="color:blue">blue</span>.  
Sequences determined to be transcribed on the negative strand are shown in 
<span style="color:red">red</span>.  Sequences for which the direction of 
transcription was not able to be determined are shown in <B>black</B>.
Please see the
<a href="http://samtools.sourceforge.net/SAM1.pdf#page=4" target=_blank>SAM Format Specification</a>
for more information on the SAM/BAM file format.

<DT><i>Raw Signals</i>
<DD> Plus Raw Signal (bigWig)  shows the density of aligned tags that map to the genome between 1-10 times on the plus strand. 
<DD> Minus Raw Signal (bigWig)  shows the density of aligned tags that map to the genome between 1-10 times on the minus strand. 
<DD> All Raw Signal (bigWig)  shows the density of aligned tags that map to the genome between 1-10 times on both strands. 

<DT><i> Downloadable files</i>
<DD> The raw data in fastq format.  
<DD> Splice Junction files ?????
<DD> Collapsed Contig files ?????
<DD> Production Documents ?????
</DL>

<H2>Methods</H2>

<DT><b>Cell culture</b>
Cells were grown according to the approved 
<A HREF="../../ENCODE/protocols/cell" TARGET=_BLANK>ENCODE cell culture protocols</A>.
<P>
<DT><b>Biological Replicates</b>  Whenever possible we grow up and assay RNA from 2 biological replicates.  
Each replicate is issued a unique ID# (e.g. 001).  If RNA from multiple cellular compartments is desired, 
the batch will be partitioned just prior to RNA isolation and the ID# appended with its compartment label 
(e.g. 001WC = Whole Cell RNA from replicate 001). These bio-replicate ID's are constant across experiments
allowing the improved data comparison. (WC) Whole Cell RNA, (N) Nuclear RNA, (C) Cytoplasmic RNA,
 (NP) Nucleoplasm, (NL) Nucleolus, (CH) Chromatin, (+) Poly-A+ RNA, (-) Poly-A- RNA.
</P>

<P>
<DT><b>Library Preparation</b> We are using the protocol published by...??? This protocol allows one to
determine the strand of transcript origin. Exogenous RNA spike-ins, still in development at NIST were
added to each bioreplicate and carried through library construction. Library insert sizes are ~ 200
bases. Accompanying each RNA-Seq dataset we submit is a <B>Production Document</B>. This document
contains details about the RNA isolations and treatments, library construction, spike-ins as well as
quality control figures relevant for that library.
</P>

<P>
<DT><b>Sequencing and Mapping</b> The data are generated on the Illumina GAIIx platform in pair-end 76
base format to a current depth of ~200 million reads. The data are mapped using Spliced Transcript
Alignment and Reconstruction (STAR) written by Alex Dobin, CSHL (in preparation). STAR is able
to map split reads in an annotation agnostic fashion allowing for identification of novel elements like
splice junctions and also takes into account the quality of the sequence itself trimming of mismatched
bases at the read ends.
???? How about number of mismatches?  How about mapping to single locus? Any filtering?? What about the C-tail chopping from before?
</P>

</P>
<H2>Verification</H2>
<P>
We take several steps to ensure that the data being generated is high quality and
reproducible. Each lane is mapped independently and FPKMs are computed for Gencode.v3c
protein-coding transcripts. Lanes of data from the same library, i.e. technical replicates are only
merged together if their FPKMs have a Spearman R value > 0.98 and show similar mapping stats.
We check to see that we recover a broad range of spike-ins in their indicated concentrations.
Additionally, we assess the strandedness of the library by verifying that >98.5% of the reads that map
to spike-ins map to the sense strand. To assess biological reproducibility we only submit biological
replicates where the FPKMs of Genocde transcripts have a Spearman R value >0.95.
</P>

<H2>Release Notes</H2>
This is Release 1 (Jan 2011) of this track for hg19. It includes data from the
hg18 releases that has been remapped to hg19 (GM12878 and K562 Total RNA whole 
cell and K562 PolyA+ RNA cytosol) as well as novel data from 13 additional cell
lines and 4 additional cell localizations, as well as some PolyA- samples.  The
singnals are now in bigWig format.  The alignments are in BAM format.  The newer
data all have Bio Replicate numbers and Protocol ID numbers that can be used to
link the samples with other experiments.


<H2>Credits</H2>
<P>These data were generated and analyzed by the transcriptome group at
Cold Spring Harbor Laboratories, and the Center for Genomic
Regulation (Barcelona), who are participants in the ENCODE Transcriptome Group.
</P>
Carrie A. Davis, Alex Dobin, Felix Schlesinger, Wei Lin, Chris Zaleski, Sonali Jha, Jorg
Drenkow, Huaien Wang, Kim Bell, Lei-Hoon See, Megan Fastuca and Tom Gingeras
<P>
<H2>Contacts</H2>
<A HREF="mailto:&#100;a&#118;&#105;s&#99;&#64;&#99;sh&#108;.&#101;&#100;&#117;">Carrie Davis</A> (experimental)
<!-- above address is davisc at cshl.edu -->
<A HREF="mailto:&#100;a&#118;&#105;s&#99;&#64;&#99;sh&#108;.&#101;&#100;&#117;">Alex Dobin</A> (computational)
 <A HREF="mailto:&#100;a&#118;&#105;s&#99;&#64;&#99;sh&#108;.&#101;&#100;&#117;">Felix Schlesinger</A> (computational) 
<A HREF="mailto:&#100;a&#118;&#105;s&#99;&#64;&#99;sh&#108;.&#101;&#100;&#117;">Tom Gingeras</A> (primary investigator)
</P>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior 
consent, submit publications that use an unpublished ENCODE dataset until 
nine months following the release of the dataset.  This date is listed in 
the <EM>Restricted Until</EM> column, above.  The full data release policy 
for ENCODE is available 
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P> 
