<H2>Description</H2>
<P>This track is produced as part of the ENCODE Project.
This track displays transcriptional fragments associated with RNA binding proteins in different
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=cellType">cell lines</A>,
using RIP-Seq(Ribonomic) profiling via Illumina SBS.
The <i>RBP Assoc RNA</i> view shows the genomic location of sequencing peaks.
Data for this track was produced as part of the
<A TARGET=_BLANK HREF="/ENCODE/index.html">
Encyclopedia of DNA Elements (ENCODE) Project</A>,

</P>


<P>In eukaryotic organisms gene regulatory networks require an additional
level of coordination that links transcriptional and post-transcriptional
processes. Messenger RNAs have traditionally been viewed as passive
molecules in the pathway from transcription to translation. However,
it is now clear that RNA-binding proteins play a major role in
regulating multiple mRNAs in order to facilitate gene expression patterns.
These tracks show the associated mRNAs that co-precipitate with the
targeted RNA-binding proteins using RIP-Seq profiling.
</P>


<H2>Display Conventions and Configuration</H2>
<P>This track is a multi-view composite track. For each view there
are multiple subtracks that display individually in the browser.
The subtracks within this track correspond to different antibodies/target
proteins tested in different cell lines.
This track is initially released with a single view:
</P>
<DL>
<DT><i>Peaks</i>
<DD>The Peaks view shows the genomic extent of the sequencing read peaks.

</DL>
Instructions for configuring multi-view tracks are

<A HREF="../../goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.

<H2>Methods</H2>
<P>RBP-mRNA complexes were purified from cells grown according to the approved
<A HREF="../../ENCODE/protocols/cell">ENCODE cell culture protocols </A>.
RNA samples were amplified and converted to cDNA with the <A HREF="http://www.nugeninc.com">Nugen</A> Ovation© RNA-Seq System and prepped for sequencing with the
<A HREF="http://www.illumina.com">Illumina</A> mRNA-Seq protocol (beginning at the end repair step).  Sequencing was done using 36 cycle Illumina single end reads.
Resulting libraries were ~23 million and ~35 million reads for K562 and GM12878 respectively.
<br>
Libraries were analyzed using <A HREF="http://http://tophat.cbcb.umd.edu/">tophat</A> version 1.0.14 with <A HREF="http://http://bowtie-bio.sourceforge.net/index.shtml">bowtie</A> version 0.12.5
<br>
Peak tracks were developed as follows:
<br>A) .wig files from tophat were normalized by "average mapped reads per base".
<br>B) Peaks were called from these normalized files using thresholds of
<br>&nbsp;&nbsp;&nbsp;i) at or above 60th percentile signal
<br>&nbsp;&nbsp;&nbsp;ii) max gap = 24(read length-1/3)
<br>&nbsp;&nbsp;&nbsp;iii) min run = 48(read length + 1/3)
<br>C) Unions of overlapping peak regions from total RNA replicates are presented with pVal from a one tailed t-test for average signal from replicates versus 0, no cut-off was used for totals.
<br>D) Intersections of overlapping (across replicates) positive RIP treatment peaks are presented with pVal from one tailed t-test versus signal for same region in negative control replicates. RIP peaks must meet two criteria:
<br>&nbsp;&nbsp;&nbsp;i) They must be equal or longer than the minimum fragment expected based on the Illumina mRNA-Seq protocol size selection (in this case, expected to be 120b)
<br>&nbsp;&nbsp;&nbsp;ii) Their pVal must be less than .05
<br>E) For both totals and RIPs, the peak scores are calculated as:
<br>&nbsp;&nbsp;&nbsp;i)80% weighting given to pVal, scaled as 0.05(or higher) to 0.0 = 0 to 800
<br>&nbsp;&nbsp;&nbsp;ii)20% weighting given to (average treatment signal minus average control signal)/(maximum treatment vs control difference amongst all peaks)

<br>Note: while not used for cut-off thresholding or score, qVals based on Benjamini Hochberg are also reported.
<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Also, an f-test was performed prior to all t-tests and used to determine if t-test should be applied for equal or unequal variance.


</P>
<P>

</P>




<P>For additional RIP methods detail, see
Tenenbaum <EM>et al.</EM> 2002; Baroni <EM>et al.</EM> 2008; Penalva
<EM>et al.</EM> 2004, below.



<H2>Verification</H2>

<P>
All experiments (including controls) performed in and analyzed as duplicates.
</P>

<H2>Credits</H2>

<P>These data were produced and analyzed by a collaboration between the
<A HREF="http://cnse.albany.edu/StaffDirectory/index.cfm?InstanceID=576&step=staffdetail&StaffDirectoryID=129"
TARGET=_BLANK>
Tenenbaum lab</A>
at the University at Albany-SUNY, College of Nanoscale
Science and Engineering, the
<A HREF="http://www.uthscsa.edu/csb/faculty/penalva.asp" TARGET=_BLANK>
Luiz Penalva group</A>
at the Greehey Children's Cancer Research Institute,
University of Texas Health Science Center and the
<A HREF="http://www.cfgbiotech.com/microarray/index.htm" TARGET=_BLANK>
Microarray Core Facility</A> at the Center for Functional Genomics, Rensselaer, NY .
</P>

<H2>References</H2>

<P>Tenenbaum SA, Lager PJ, Carson CC, Keene JD.
<A TARGET=_BLANK" HREF="http://www.ncbi.nlm.nih.gov/pubmed/12054896">
Ribonomics: identifying mRNA subsets in mRNP complexes using
antibodies to RNA-binding proteins and genomic arrays.
</A>
<EM>Methods</EM>. 2002 Feb;26(2):191-8.

</P>

<P>Baroni TE, Chittur SV, George AD, Tenenbaum SA.
<A TARGET=_BLANK" HREF="http://www.ncbi.nlm.nih.gov/pubmed/18369977">
Advances in RIP-chip analysis : RNA-binding protein
immunoprecipitation-microarray profiling.
</A>
<EM>Methods Mol Biol</EM>. 2008;419:93-108.
</P>

<P>Penalva LO, Tenenbaum SA, Keene JD.
<A TARGET=_BLANK" HREF="http://www.ncbi.nlm.nih.gov/pubmed/14770002">
Gene expression analysis of messenger RNP complexes.
</A>
<EM>Methods Mol Biol</EM>. 2004;257:125-34.
</P>

<P>Keene JD, Tenenbaum SA.
<A TARGET=_BLANK" HREF="http://www.ncbi.nlm.nih.gov/pubmed/12086614">
Eukaryotic mRNPs may represent posttranscriptional operons.
</A>
<EM>Mol Cell</EM>. 2002;9(6):1161-7.
</P>

<P>George AD, Tenenbaum SA.
<A TARGET=_BLANK" HREF="http://www.ncbi.nlm.nih.gov/pubmed/17114943">
MicroRNA modulation of RNA-binding protein regulatory elements.
</A>
<EM>RNA Biol</EM>. 2006;3(2):57-9. Epub 2006 Apr 1.
</P>



<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column, above.  The full data release policy
for ENCODE is available
<A TARGET=_BLANK HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P>

