<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <title>wgEncodeRnaSeqSuper.html</title>
</head>
<body>
<h2>Description</h2>
<P>In eukaryotic organisms gene regulatory networks require an additional
level of coordination that links transcriptional and post-transcriptional
processes. Messenger RNAs have traditionally been viewed as passive
molecules in the pathway from transcription to translation. However,
it is now clear that RNA-binding proteins play a major role in
regulating multiple mRNAs in order to facilitate gene expression patterns.
These tracks show the associated mRNAs that co-precipitate with the
targeted RNA-binding proteins using RIP-Chip profiling.
</P>

<h2>Display Conventions</h2>
These tracks are multi-view composite tracks that contain multiple
data types (views). Each view within a track
has separate display controls, as described <a
 href="http://hgwdev.cse.ucsc.edu/goldenPath/help/multiView.html">here</a>.
Most ENCODE tracks contain multiple subtracks, corresponding to
multiple experimental conditions. If a track contains a large
number of subtracks, only some subtracks will be displayed by default.
The user can select which subtracks are displayed via the display controls
on the track details pages.
<h2>Credits<br>
</h2>
These data were generated and analyzed as part of the <a
 href="http://www.genome.gov/10005107">ENCODE</a> project, a
genome-wide consortium project with the aim of cataloging all
functional elements in the human genome. This effort includes
collecting a variety of data across related experimental conditions, to
facilitate integrative analysis. Consequently, additional ENCODE tracks may 
contain data that is relevant to the data in these tracks. 
<h2>References</h2>
<P>Tenenbaum SA, Lager PJ, Carson CC, Keene JD.
<A TARGET=_BLANK" HREF="http://www.ncbi.nlm.nih.gov/pubmed/12054896">
Ribonomics: identifying mRNA subsets in mRNP complexes using
antibodies to RNA-binding proteins and genomic arrays.
</A>
<EM>Methods</EM>. 2002 Feb;26(2):191-8.

</P>

<P>Baroni TE, Chittur SV, George AD, Tenenbaum SA.
<A TARGET=_BLANK" HREF="http://www.ncbi.nlm.nih.gov/pubmed/18369977">
Advances in RIP-chip analysis : RNA-binding protein
immunoprecipitation-microarray profiling.
</A>
<EM>Methods Mol Biol</EM>. 2008;419:93-108.
</P>

<P>Penalva LO, Tenenbaum SA, Keene JD.
<A TARGET=_BLANK" HREF="http://www.ncbi.nlm.nih.gov/pubmed/14770002">
Gene expression analysis of messenger RNP complexes.
</A>
<EM>Methods Mol Biol</EM>. 2004;257:125-34.
</P>

<P>Keene JD, Tenenbaum SA.
<A TARGET=_BLANK" HREF="http://www.ncbi.nlm.nih.gov/pubmed/12086614">
Eukaryotic mRNPs may represent posttranscriptional operons.
</A>
<EM>Mol Cell</EM>. 2002;9(6):1161-7.
</P>

<P>George AD, Tenenbaum SA.
<A TARGET=_BLANK" HREF="http://www.ncbi.nlm.nih.gov/pubmed/17114943">
MicroRNA modulation of RNA-binding protein regulatory elements.
</A>
<EM>RNA Biol</EM>. 2006;3(2):57-9. Epub 2006 Apr 1.
</P>


<h2>Data Release Policy</h2>
Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset
until nine months following the release of the dataset. This date is
listed in the Restricted Until column on the track configuration page
and the download page. The full data release policy for ENCODE is
available <a href="http://genome.ucsc.edu/ENCODE/terms.html">here</a>.<br>
<br>
</body>
</html>
