<H2>Description</H2>
<P>
This track is produced as part of the ENCODE Chromatin Interaction Analysis with Paired-End
Tag (ChIA-PET) Project. It shows the locations of oestrogen receptor .-bound (ER-.-bound)
chromatin interactions determined by Paired-End Tag (PET) sequencing using proximity-ligated
chromatin extracts from the MCF-7 human breast cancer cell line. A chromatin interaction is
defined as two regions of the genome that are far apart in terms of genomic distance, but are
spatially proximate to each other in the 3-dimensional cellular nucleus.
<P>
Additionally, the ER-. ChIA-PET experiment generates oestrogen receptor . binding sites. A
binding site is defined as a region of the genome that is highly enriched by specific Chromatin
ImmunoPrecipitation (ChIP) against a transcription factor, which indicates that the transcription
factor binds specifically to this region. Generally, a chromatin interaction is more likely to
connect two oestrogen receptor . binding sites together.
<P>
<h2>Display Conventions and Configuration</h2>
In the graphical display, the tags (ends) are represented by blocks, and connected by a horizontal
line if both ends are in the same chromosome and in a viewable window. PET sequences that
overlap at both ends form PET clusters and the total count of PETs in a PET cluster reflects
the strength of a chromatin interaction. Singleton PETs (PET count = 1 in a PET cluster) are
most likely noise, whereas PET clusters with PET count 3 and above could indicate genuine
chromatin interactions. A density track of the tags shows the ChIP enrichment at different points
of genome, and high peaks indicate oestrogen receptor . binding sites.
</P>


<dl>
<dt><i>IntraInteractions</i>
</dt><dd>Description from lab needed here
</dd><dt><i>Interactions</i>
</dt><dd>Description from lab needed here
</dd><dt><i>Alignments</i>
</dt><dd>The Alignments view shows reads mapped to the genome. The lab should put information about any bam tags used here.
 For more information about the SAM output and the <a href="http://samtools.sourceforge.net/SAM1.pdf#page=4" target=_blank>SAM Format Specification</a> 
for more information on the SAM/BAM file format.
</dd></dl>

<p>To show only selected subtracks, uncheck the boxes next to the tracks that
you wish to hide.</p>


<H2>Methods</H2>
<P>
Recently, we developed Chromatin Interaction Analysis with Paired-End Tag (ChIA-PET)
Sequencing, a global, de novo, high-throughput method for characterizing the 3-dimensional
structure of chromatins in the nucleus. In the ChIA-PET protocol, cells were cross-linked,
fragmented, and chromatin immunoprecipitated. The DNA fragments which are brought close
together by the chromatin interactions were then proximity-ligated. During this proximity
ligation step, the half-linkers containing flanking MmeI sites (type IIS restriction enzymes) were
first ligated to the DNA fragments; then, the half-linkers were ligated to each other to form full
linkers. Full linkers bridge either two ends of a self-circularized fragment, or two ends of two
different chromatin fragments. The material was then reverse cross-linked, purified, and digested
with MmeI. As MmeI cuts 20 base pairs away from its recognition site, tag-linker-tag (Paired-
End Tag, .PET.) constructs were obtained, which was then sequenced by ultra-high-throughput
methods (454 pyrosequencing and Illumina paired end sequencing).
<P>
<H2>Verification</H2>
<P>
For the validation, 14 experimental sites were tested, with 3 negative controls. Of the 14
experimental sites 12 were validated and 2 experimental sites had
unreproducible interchromosomal interactions. All 3 negative control sites were validated.
We used a variety of methods, including FISH, ChIP-3C, 3C, and 4C, to do the validation.
An additional 3 sites were also shown to have chromatin interactions by other papers. The
successful experimental sites we tested were: TFF1 (3C, ChIP-3C), GREB1 (3C, ChIP-3C),
P2RY2 (3C, ChIP-3C), SIAH2 (3C, ChIP-3C), Keratin cluster (4C), NR2F2 (FISH), GATA3
(FISH), BX647900 (ChIP-3C), CAP2 (ChIP-3C), ELOVL2 (ChIP-3C), SLC9A3R1 (ChIP-3C),
and DB554846 (ChIP-3C). The unsuccessful experimental sites were: chr7:2692293-2698011 to
chr12:120670670-120673138, and chr9:113812545-113817679 to chr14:90811137-90814459.
The negative control sites were: NOTCH2, GREB1 (chr2) to TFF1 (chr21),
and P2RY2 (chr11) to GREB1 (chr2). Validation was performed by FISH, 3C, ChIP-3C and 4C
analyses on selected sites.
<P>
<H2>Credits</H2>
<P>
The ChIA-PET libraries were produced at the Genome Institute of Singapore. The data were
mapped and analyzed by scientists from the Genome Institute of Singapore.
<P>
<H2>References</H2>
<P> The pubmed link with the pubmed format needs to go here. </P>
<P>

Li G, Fullwood MJ, Xu H, Mulawadi FH, Velkov S, Vega V, Ariyaratne PN, Mohamed
YB, Ooi H-S, Tennakoon C et al: ChIA-PET tool for comprehensive chromatin
interaction analysis with paired-end tag sequencing. Genome Biology 2010, 11(2):R22.
<P>

Fullwood, M.J., Han, Y.Y., Wei, C.L., Ruan, X., Ruan, Y. Chromatin Interaction
Analysis using Paired-End Tag Sequencing, Current Protocols in Molecular Biology,
2010.
<P>

Fullwood MJ, Liu MH, Pan YF, Liu J, Xu H, Mohamed YB, Orlov YL, Velkov S, Ho
A, Mei PH et al: An oestrogen-receptor-alpha-bound human chromatin interactome.
Nature 2009, 462(7269):58-64.

<H2>Credits</H2>
<P>Hannon lab members: Katalin Fejes-Toth, Vihra Sotirova, Gordon  Assaf, Jon Preall</P>
<P>And members of the Gingeras and Guigo labs. </P>

Contact:
<A HREF="http://www.gis.a-star.edu.sg/internet/site/investigators.php?f=cv&user_id=19"
TARGET=_BLANK>Yijun Ruan</A>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column, above.  The full data release policy
for ENCODE is available
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P>
