<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
</head>
<body>
The <I>Natronomonas pharaonis DSM 2160</I> genome is 2.75 Million bp long and contains approximately 2894
predicted genes. <I>N. pharaonis</I>  grows at high salt and pH and has multiple systems for nitrogen assimilation. <BR><BR>
The sequence was released Oct 2005 by the Max Planck Institute, and was described
in <I>Genome Res</I> <B>15</B>:1336-43 (2005) Falb M, Pfeiffer F, Palm P, Rodewald K, Hickmann V, <I>et al.</I> &nbsp;"<a
 href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=16169924">Living with two extremes: conclusions from the genome sequence of Natronomonas pharaonis.
</a>"
<blockquote><B>Abstract:</B> Natronomonas pharaonis is an extremely haloalkaliphilic archaeon that was isolated from salt-saturated lakes of pH 11. We sequenced its 2.6-Mb GC-rich chromosome and two plasmids (131 and 23 kb). Genome analysis suggests that it is adapted to cope with severe ammonia and heavy metal deficiencies that arise at high pH values. A high degree of nutritional self-sufficiency was predicted and confirmed by growth in a minimal medium containing leucine but no other amino acids or vitamins. Genes for a complex III analog of the respiratory chain could not be identified in the N. pharaonis genome, but respiration and oxidative phosphorylation were experimentally proven. These studies identified protons as coupling ion between respiratory chain and ATP synthase, in contrast to other alkaliphiles usi... [Click above reference link for full abstract]
</blockquote>
<P>
<h3>Sample position queries</h3>
<p>
A genome position can be specified by chromosomal coordinate range, COG
ID, or keywords from the GenBank or TIGR description of a gene. 
The available chromosome/plasmid names are:<BR><BR>
<table border=2 cellspacing=2 cellpadding=2>
<tr><th>Browser Chrom/Plasmid Name</th><th>Length (bp)</th><th>GC Content (%)</th><th>Gene Count</th><th>NCBI RefSeq Accession</th></tr>
<tr><td><a href="/cgi-bin/hgTracks?db=natrPhar1&position=chr:1-02595221">chr</a></td><td align=right>2595221</td><td align=right>63.44</td><td align=right>2726</td><td align=center><a href="http://www.ncbi.nlm.nih.gov/entrez/viewer.fcgi?val=NC_007426">NC_007426</a></td></tr>
<tr><td><a href="/cgi-bin/hgTracks?db=natrPhar1&position=plasmid_PL131:1-130989">plasmid_PL131</a></td><td align=right>130989</td><td align=right>57.22</td><td align=right>132</td><td align=center><a href="http://www.ncbi.nlm.nih.gov/entrez/viewer.fcgi?val=NC_007427">NC_007427</a></td></tr>
<tr><td><a href="/cgi-bin/hgTracks?db=natrPhar1&position=plasmid_PL23:1-23486">plasmid_PL23</a></td><td align=right>23486</td><td align=right>60.59</td><td align=right>36</td><td align=center><a href="http://www.ncbi.nlm.nih.gov/entrez/viewer.fcgi?val=NC_007428">NC_007428</a></td></tr>
</table>

<p>
The following list shows examples of valid position queries for this
genome:&nbsp;
</p>
<p>
<table cellspacing="1" cellpadding="1" border="0">
<tr><th align=left>Request:</th><th align=left>Genome Browser Response:</th></tr>
<tr><td>chr</td><td>Displays the entire sequence "chr" in the browser window</td></tr>
<tr><td>chr:1-10000&nbsp;&nbsp;&nbsp;&nbsp;</td><td>Displays first ten thousand bases of the sequence "chr"</td></tr>
<tr><td>transporter&nbsp;&nbsp;&nbsp;&nbsp;</td><td>Lists all genes with "transporter" in the name or description</td></tr>
<tr><td>NP0020A</td><td>Display genome at position of gene NP0020A</td></tr>
</tbody>
</table>
<P>
<HR>
<H3>Credits</H3>
The Archaeal Genome Browsers at UCSC were developed by members of the
<a href="http://lowelab.ucsc.edu/">Lowe Lab</a> (Kevin Schneider,
Katherine Pollard, Andy Pohl, Todd Lowe) and Robert Baertsch, with significant support from
the <a href="http://genome.ucsc.edu/staff.html">UCSC Human Genome
Browser group</a>.
The Archaeal Browsers are run by a slightly modified
version of the <a href="http://genome.ucsc.edu/">UCSC Human Genome
Browser</a> system.  All queries, bug reports, content corrections, suggested improvements,
and new track data submissions should be sent to Todd Lowe (lowe
@soe.ucsc.edu).

<P>If you use the browser for your research and wish to cite it, a
paper describing the work has been accepted to the 2006 <I>Nucleic
Acids Research</I>
<a href="http://www3.oup.co.uk/nar/database/a/">Database Issue</a> (KL Schneider, KS Pollard, R Baertsch, A Pohl, TM Lowe). 
Citations will help us obtain funding to continue
development of this community resource.

</body>
</html>
