<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
</head>
<body>
The <I>Methanothermobacter thermautotrophicus str. Delta H</I> genome is 1.75 Million bp long and contains approximately 1920
predicted genes. <I>M. thermoautotrophicus</I> is anaerobic and thermophilic. It grows at pH 7.2-7.6, and produces methane by reducing carbon dioxide. <BR><BR>
The sequence was released Aug 1997 by the Oscient Pharmaceuticals Corporation, and was described
in <I>J Bacteriol</I> <B>179</B>:7135-55 (1997) Smith DR, Doucette-Stamm LA, Deloughery C, Lee H, Dubois J, <I>et al.</I> &nbsp;"<a
 href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=9371463">Complete genome sequence of Methanobacterium thermoautotrophicum deltaH: functional analysis and comparative genomics.
</a>"
<blockquote><B>Abstract:</B> The complete 1,751,377-bp sequence of the genome of the thermophilic archaeon Methanobacterium thermoautotrophicum deltaH has been determined by a whole-genome shotgun sequencing approach. A total of 1,855 open reading frames (ORFs) have been identified that appear to encode polypeptides, 844 (46%) of which have been assigned putative functions based on their similarities to database sequences with assigned functions. A total of 514 (28%) of the ORF-encoded polypeptides are related to sequences with unknown functions, and 496 (27%) have little or no homology to sequences in public databases. Comparisons with Eucarya-, Bacteria-, and Archaea-specific databases reveal that 1,013 of the putative gene products (54%) are most similar to polypeptide sequences described previously for other organ... [Click above reference link for full abstract]
</blockquote>
<P>
<h3>Sample position queries</h3>
<p>
A genome position can be specified by chromosomal coordinate range, COG
ID, or keywords from the GenBank or TIGR description of a gene. 
The available chromosome/plasmid names are:<BR><BR>
<table border=2 cellspacing=2 cellpadding=2>
<tr><th>Browser Chrom/Plasmid Name</th><th>Length (bp)</th><th>GC Content (%)</th><th>Gene Count</th><th>NCBI RefSeq Accession</th></tr>
<tr><td><a href="/cgi-bin/hgTracks?db=methTher1&position=chr:1-01751377">chr</a></td><td align=right>1751377</td><td align=right>49.54</td><td align=right>1920</td><td align=center><a href="http://www.ncbi.nlm.nih.gov/entrez/viewer.fcgi?val=NC_000916">NC_000916</a></td></tr>
</table>

<p>
The following list shows examples of valid position queries for this
genome:&nbsp;
</p>
<p>
<table cellspacing="1" cellpadding="1" border="0">
<tr><th align=left>Request:</th><th align=left>Genome Browser Response:</th></tr>
<tr><td>chr</td><td>Displays the entire sequence "chr" in the browser window</td></tr>
<tr><td>chr:1-10000&nbsp;&nbsp;&nbsp;&nbsp;</td><td>Displays first ten thousand bases of the sequence "chr"</td></tr>
<tr><td>transporter&nbsp;&nbsp;&nbsp;&nbsp;</td><td>Lists all genes with "transporter" in the name or description</td></tr>
<tr><td>MTH10</td><td>Display genome at position of gene MTH10</td></tr>
</tbody>
</table>
<P>
<HR>
<H3>Credits</H3>
The Archaeal Genome Browsers at UCSC were developed by members of the
<a href="http://lowelab.ucsc.edu/">Lowe Lab</a> (Kevin Schneider,
Katherine Pollard, Andy Pohl, Todd Lowe) and Robert Baertsch, with significant support from
the <a href="http://genome.ucsc.edu/staff.html">UCSC Human Genome
Browser group</a>.
The Archaeal Browsers are run by a slightly modified
version of the <a href="http://genome.ucsc.edu/">UCSC Human Genome
Browser</a> system.  All queries, bug reports, content corrections, suggested improvements,
and new track data submissions should be sent to Todd Lowe (lowe
@soe.ucsc.edu).

<P>If you use the browser for your research and wish to cite it, a
paper describing the work has been accepted to the 2006 <I>Nucleic
Acids Research</I>
<a href="http://www3.oup.co.uk/nar/database/a/">Database Issue</a> (KL Schneider, KS Pollard, R Baertsch, A Pohl, TM Lowe). 
Citations will help us obtain funding to continue
development of this community resource.

</body>
</html>
