<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
</head>
<body>
The <span style="font-style: italic;">Methanosarcina Acetivoans C2A </span>
genome sequence is 5,751,492 bp long and contains approximately 4,500
protein-coding genes.&nbsp; <span style="font-style: italic;">M.
Acetivorans</span> is the largest known archeal genome.  Only Methanosarcina species possess all three known pathways for methanogenesis and are capable of utilizing no less than nine methanogenic substrates, including acetate. In contrast, all other orders of methanogens possess a single pathway for methanogenesis, and many utilize no more than two substrates.&nbsp; 

<h3>Sample position queries</h3>
<p>
A genome position can be specified chromosomal coordinate range, COG
ID, or keywords from the GenBank or TIGR description of an gene. The
following list shows examples of valid position queries for the human
genome.&nbsp;
</p>
<p>
<table cellspacing="0" cellpadding="0" border="0">
  <tbody>
    <tr>
      <td nowrap="nowrap" valign="top"><b>Request:</b><br>
      </td>
      <td colspan="2" valign="top"><b>&nbsp;&nbsp; Genome Browser
Response:</b><br>
      </td>
    </tr>
    <tr>
      <td valign="top"><br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">chr</td>
      <td width="14"><br>
      </td>
      <td valign="top">Displays all of chromosome (the entire genome).<br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">chr:1-10000</td>
      <td width="14"><br>
      </td>
      <td valign="top">Displays first ten thousand bases on the
chromosome.<br>
      </td>
    </tr>
<!-- 
    <TR><TD VALIGN=Top NOWRAP>ctgchr7_ctg</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of the clone contig ctgchr7_ctg 
                       (set "Map contigs" track to "dense" and refresh to see contigs)</TD>
    </TR> 
-->
    <tr>
      <td nowrap="nowrap" valign="top">dehydratase</td>
      <td width="14"><br>
      </td>
      <td valign="top">Lists genes with "dehydratase" in the description.<br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">MA2536</td>
      <td width="14"><br>
      </td>
      <td valign="top">Display chromosome at position of gene <a
 href="http://www.ncbi.nlm.nih.gov/entrez/viewer.fcgi?val=AAM05919">MA2536</a>.<br>
      </td>
    </tr>
    <tr>
      <td nowrap="nowrap" valign="top">cam</td>
      <td width="14"><br>
      </td>
      <td valign="top">Lists genes with names beginning with "cam".<br>
      </td>
    </tr>
    <tr>
    </tr>
  </tbody>
</table>
<p><h3>Sequencing and Assembly</h3></b><br>
The sequence was completed in April 2002 by the Whitehead Institute for Genome Research, Cambridge Mass., and was published
in <a href="http://www.genome.org/cgi/doi/10.1101/gr.223902">Galagan et al (2002), Genome Research 12(4):532-542</a>.</p>

The <i>Methanosarcina acetivorans</i> sequencing project reflects a close collaboration with <a href="http://www.life.uiuc.edu/micro/metcalf.html">Dr. William Metcalf</a> from the University of Illinois. Principal investigators from the Broad Institute include <a href="mailto:jgalag@mit.edu">James Galagan</a>, <a href="mailto:bwb@broad.mit.edu">Bruce Birren</a> and <a href="mailto:chad@broad.mit.edu">Chad Nusbaum</a>.
<p><a name="sequencing"></a><b>
<p>
</b>The genome was sequenced by the whole genome shotgun method.
<p>
 The Methanosarcina acetivorans strain C2A was grown in single cell morphology47 at 35 degrees C in HS broth medium containing 125 mM methanol plus 40 mM sodium acetate (HS-MA medium)48.
<p>
 Genomic DNA was isolated from M. acetivorans and was used to construct m13 (1.5kb inserts), plasmid (4kb inserts), and fosmid (40kb inserts) libraries. Plasmid and Fosmid inserts were sequenced from both ends to generate paired-reads. We generated sequence coverage of 7X from plasmids, 1X from M13 and 0.076X from Fosmids and assembled it with Phrap.Initial analysis of the assembly was done with the Mapper software (M.C. Zody, personal communication) to select gap-spanning clones for finishing. 200 gaps spanned by plasmid clones were closed by transposon-based sequencing using the EZ::TN <KAN-2> (tm) from Epicentre. 48 gaps spanned only by Fosmids were closed by sequencing Fosmid-derived PCR products. Sequence from 28 unspanned gaps was obtained from fragments generated by combinatorial PCR using genomic DNA as template and pooled primers50. One unspanned gap was closed by sequencing a small-insert library51 produced from an 8.5 kb PCR product . Regions of low sequence quality were resolved by:
<p>
<li> use of ABI dGTP Big Dye Terminator sequencing mix</li>
<li> transposon-primed sequencing of plasmid clones, or</li>
<li> sequencing PCR products obtained from plasmid or genomic template </li>
<p>             
Paired-reads within the assembly were visualized with the Mapper software and used in assembly validation. Regions of the assembly spanned by paired-reads occurring with the appropriate orientation and spacing were considered valid. 99.99% of the genome was validated in this way while only 6270 bases of the finished assembly were spanned soley by sequenced PCR products. Regions of the assembly containing nonsensical paired-reads were analyzed further. Eighteen of these regions proved to have been misassembled by Phrap, and were resolved manually. The finished genome sequence was manually inspected for quality and edited using the Staden package viewer Gap4. During annotation 480 possible sequence errors (based on breaks in open reading frames) were identified. These were manually reviewed and one was shown to be an editing error and corrected. Five of the possible errors were not unambiguous by quality, but after re-sequencing were shown to have been correct.
                                
</p>
</body>
</html>
