<H2>Description</H2>
<P>This track displays <EM>Blastz</EM> alignments of the Feb. 2002 mouse draft
assembly to the human genome filtered to display only the best alignment for any
given region of the human genome. The track has an optional
feature that color codes alignments to indicate the mouse chromosomes from which 
they are derived. To activate the color feature,
click the "on" radio button next to "Color track based on chromosome".
<br>
<H2>Methods</H2>
For <em>Blastz</em>, we use 12 of 19 seeds and then score using: 
<br>
<table>
<tr><td>&nbsp;</td><td>  A</td><td>  C</td><td>  G</td><td>  T</td></tr>
<tr><td>A</td><td>&nbsp;&nbsp;91</td><td>-114</td><td>&nbsp;-31</td><td>-123</td></tr>
<tr><td>C</td><td>-114</td><td>&nbsp;100</td><td>-125</td><td>&nbsp;-31</td></tr>
<tr><td>G</td><td>&nbsp;-31</td><td>-125</td><td>&nbsp;100</td><td>-114</td></tr>
<tr><td>T</td><td>-123</td><td>&nbsp;-31</td><td>-114</td><td>&nbsp;91</td></tr>
</table>
<br>
  O = 400, E = 30, K = 3000, L = 3000, M = 50
<br>
However, we now do a second pass at reduced stringency (7mer seeds and
MSP threshold of K=2200) to attempt to fill in gaps (up to about 10Kbp).
Lineage specific repeats are abridged during this alignment.
<br>
<H2>Credits</H2>
These alignments are contributed by Scott Schwartz, <A HREF="http://bio.cse.psu.edu/"
TARGET=_blank>Penn State Bioinformatics Group</A>. The best in genome filtering
is done by UCSC's axtBest program.

<H2>Data Release Policy</H2>
<P>Mouse data is made available before scientific publication with the 
following understanding:
<OL>
<LI>The data may be freely downloaded, used in analyses, and 
repackaged in databases.
<LI>Users are free to use the data in scientific papers analyzing 
particular genes and regions, provided that the providers of this data 
(the Mouse Sequencing Consortium) are properly acknowledged.
<LI>The Centers producing the data reserve the right to publish the initial 
large-scale analyses of the dataset-including large-scale identification 
of regions of evolutionary conservation and large-scale 
genomic assembly. Large-scale refers to regions with size on the order of 
a chromosome (that is, 30 Mb or more). 
<LI>Any redistribution of the data should carry this notice.</OL>
</P>
