<H2>Description</H2>
<P>
This track contains tBLASTn alignments of the peptides
from the predicted and known genes indentified in the hg16 knownGene track as of 2004-05-27.

<H2>Methods</H2>
First the predicted proteins from the human knownGene track are aligned with 
the human genome using the BLAT program to discover exon boundaries.  
Next the amino acid sequences that make
up each exon are aligned with the sequence of $Organism using the tBLASTn
program.
Finally the putative $Organism exons are chained together using an organism specific maximum gap size
but no gap penalty.  The single best exon chains that extend over more than 60% of the query
protein are included.  Exon chains that extend over 60% of the query and
match at least 20% of the protein's amino acids are also included.

<H2>Credits</H2> 
tBLASTn is part of the NCBI Blast tool set.
<P>
Altschul, S.F., Gish, W., Miller, W., Myers, E.W. & Lipman, D.J. (1990) "Basic local alignment search tool." J. Mol. Biol. 215:403-410
<P>
BLAT was written by Jim Kent. 
<P>
The rest of the required utilities were written by Jim Kent or Brian Raney.
