<H2>Description</H2>
This track shows predicted transcription factor binding sites.
In dense mode the gold areas indicate the extents of the area
searched for binding sites and the black boxes indicate the actual
binding sites.  In other modes the gold areas disappear and you only
see the binding sites. Clicking on a particular binding site brings up
a page which indicates the sequence motif associated with the predicted
transcription factor, and the sequence at the predicted binding site.
<H2>Methods</H2>
This analysis was performed according to 
<A HREF="http://cs.stanford.edu/~eran/publications/segal_promoters.pdf">
Genome-wide discovery of transcriptional modules from DNA sequence and gene
expression</A> on various pre-existing microarray datasets.
A regulatory module is comprised of a set of genes predicted to be regulated by the same
combination of DNA sequence motifs. The predictions are based on the
co-expression of the set of genes in the module and on the appearance of
common combinations of motifs in the upstream regions of genes assigned to
the same module. For each module, you can view the set of genes that belong
to it, the combination of motifs that play a role in controlling the
expression of the module genes, the profiles of each of the individual
motifs, and the specific locations of the motif appearances within the
promoter of each of the genes in the module.
<H2>Credits</H2>
Thanks to <A HREF="http://robotics.stanford.edu/~erans/">Eran Segal</A> for
providing the data analysis that forms the basis for this track.  The display
was programmed by <A HREF="http://www.soe.ucsc.edu/~kent">Jim Kent</A>.
