<H2>Description</H2>
<P>
This track shows predictions from the 
<A HREF="http://genes.mit.edu/GENSCAN.html" TARGET=_blank>Genscan</A> program 
written by <A HREF="http://genes.mit.edu/chris/" TARGET=_blank>Chris Burge</A>.
The predictions are based on transcriptional, 
translational, and donor/acceptor splicing signals, as well as the length 
and compositional distributions of exons, introns and intergenic regions.</P>

<H2>Display Conventions and Configuration</H2>
<P>
This track follows the display conventions for 
<A HREF="../goldenPath/help/hgTracksHelp.html#GeneDisplay">gene prediction 
tracks</A>. 
<P>
The track description page offers the following filter and configuration
options:
<UL>
<LI><B>Color track by codons:</B> Select the <em>genomic codons</em> option
to color and label each codon in a zoomed-in display to facilitate validation 
and comparison of gene predictions. Click 
<A HREF="../goldenPath/help/hgCodonColoring.html">here</A> for more 
information about this feature.
</UL></P>

<H2>Methods</H2>
<P>
For a description of the Genscan program and the model that underlies it, 
refer to Burge, C. and Karlin, S. (1997) in the References section below. 
The splice site models used are described in more detail in Burge, C. (1998)
below. </P>

<H2>Credits</H2>
Thanks to Chris Burge for providing these data.

<H2>References</H2>
<P>
Burge, C. 
Modeling Dependencies in Pre-mRNA Splicing Signals. 
In Salzberg, S., Searls, D., and Kasif, S., eds. 
<A HREF="http://www.cs.jhu.edu/~salzberg/compbio-book.html" 
TARGET=_blank><em>Computational Methods in Molecular Biology</em></A>, 
Elsevier Science, Amsterdam, 127-163 (1998). </P>
<P>
Burge, C. and Karlin, S. 
<A HREF="http://www.sciencedirect.com/science?_ob=ArticleURL&_udi=B6WK7-45VGF7T-9&_coverDate=04%2F25%2F1997&_alid=195877955&_rdoc=1&_fmt=&_orig=search&_qd=1&_cdi=6899&_sort=d&view=c&_acct=C000050221&_version=1&_urlVersion=0&_userid=10&md5=569f44875aa5243d28129109de86f569" 
TARGET=_blank>Prediction of Complete Gene Structures in Human Genomic DNA</A>.
<em>J. Mol. Biol.</em> <B>268</B>(1), 78-94 (1998). </P>
