<H2>Description</H2>
<P>
Description: Results of RT-PCR testing of <EM>novel</EM> genes in $organism
done in collaboration with Phil Green's lab
(<A HREF="http://www.phrap.org/">U. Washington and HHMI</A>).  
This track shows predictions that are
currently being tested.
</P>

<H2>Methods</H2>
<P>
Predicted gene fragments from the ExoniphyGenes track were
tested.  To be selected, predictions were required not to overlap
RefSeq or Vega genes, UCSC <EM>Known Genes</EM> linked to SwissProt entries,
or MGC A- or B-list genes.  They also had to contain at least one
intron, could not fall within introns of, and on the same strand as,
known genes, and could not be in recent segmental duplications in the
human genome (with >95% identity over >1000bp to other regions of the
genome); moreover, they could not overlap C-list predictions tested or
in the pipeline to be tested by the Brent lab or the Green lab.
Predicted human genes and orthologous predictions in mouse were tested
in parallel (only the human results are shown here).  When choosing
predictions to be tested, priority was given to those with little or
no cDNA support in human or mouse; however, some predictions with
significant cDNA evidence in one or both species have been tested.
</P>
<P>
Sequence reads were assembled using Phrap (if possible) and then
aligned to the genome using BLAT.  Currently, a test is considered a
<EM>success</EM> if the sequenced product (either assembled or the raw reads, 
if
assembly is not possible) matches the genome with at least
95% identity and 30% coverage in the same location as the original
prediction, and the alignment confirms that at least one GT-AG intron
has been spliced out. (A low coverage threshold is used because of a
tendency for poor sequence quality near the ends of the reads.)
If any of these criteria are not met, the test is considered a 
<EM>failure</EM>.

<H2>Credits</H2>
<P>
Phil Green, Colleen Davis, and Brent Ewing at the Green Lab
(<A HREF="http://www.phrap.org">http://www.phrap.org</A>).</P>
