<H2>Description</H2>
<P>
This track shows alignments between $organism 
<A HREF="http://mgc.nci.nih.gov/" TARGET=_blank>Mammalian Gene Collection</A> (MGC) mRNAs with 
full-length open reading frames (ORFs) and the genome. 
Coding exons are represented by 
blocks connected by horizontal lines representing introns. The 5' and 3' 
untranslated regions (UTRs) are displayed as shorter blocks on the leading 
and trailing ends of the aligning regions. In full display mode, arrowheads 
on the connecting intron lines indicate the direction of transcription.
</P>
<H2>Method</H2>
<P>
Genbank $organism MGC mRNAs identified as having full-length ORFs are
aligned against the genome using the 
<A HREF="http://genome.cse.ucsc.edu/cgi-bin/hgBlat" TARGET=_blank>BLAT</A> 
program. When a single mRNA aligns
in multiple places, the alignment having the highest base identity is found. 
Only alignments that have a base identity level within 1% of
the best and also have at least 95% base identity are kept.

<H2>Credits</H2>
The $Organism MGC full-length mRNA track is produced at UCSC from mRNA sequence data
submitted to 
<A HREF="http://www.ncbi.nlm.nih.gov/Genbank/index.html" TARGET=_blank>Genbank</A>
by the 
<A HREF="http://mgc.nci.nih.gov/" TARGET=_blank>Mammalian Gene Collection</A> project.
