<h2>Description</h2>
<P>
Failed RT-PCR clones for
<A HREF="http://mgc.nci.nih.gov/">Mammalian Gene Collection</A>
$organism C-list from Washington University St. Louis.
</P>
<P>
Updated 2005-02-28
</P>

<h2>Methods</h2>

<H4>Twinscan Set 1</H4>
<p>
All twinscan predictions are categorized as following: All ESTs 
were aligned to overall Twinscan predictions on NCBI34. EST alignments 
are picked if they met the criterion ( at least 98 match in a 100 bp 
window and the best alignment is at least 30 better than the second best 
alignment). From all the Twinscan predictions,  filter out all 
predictions overlap with MGC annotation by at least 1 bp. The 
predictions that not overlapping with MGC annotation are then compared 
to the EST alignment result. Those predictions that not overlapping with 
MGC annotation and not supported by any EST are picked as the target 
set. Twinscan prediction have been done in two ways:  one is all the 
known genes are cut out first, the other is no known genes is cut out. 
The target set 1 is the common set of the result sets of above procedure 
on those two twinscan prediction sets.
</p>

<H4>Twinscan set 2</H4>
<p>
All twinscan predictions are categorized as following (see the 
diagram below for more details): All ESTs were aligned to overall 
Twinscan predictions on NCBI34. EST alignments are picked if they met 
the criterion ( at least 98 match in a 100 bp window and the best 
alignment is at least 30 better than the second best alignment). From 
all the Twinscan predictions,  filter out all those overlapping with MGC 
annotation by at least 1 bp. The predictions not overlapping with MGC 
annotation are then compared to the EST alignment result. Those 
predictions are selected as target set 2 if they do not overlap MGC 
annotation and are aligned by some ESTs but not all their introns are 
confirmed.
</p>

<H4>Twinscan set 3</H4>
<p>
All N-SCAN prediction are categorized as following (see the 
diagram below for more details): All ESTs were aligned to overall 
predictions on NCBI35. EST alignments are picked if they met the 
criterion ( at least 98 match in a 100 bp window and the best alignment 
is at least 30 better than the second best alignment). From all the 
NSCAN predictions,  filter out all predictions overlap with MGC 
annotation by at least 1 bp. The predictions that not overlapping with 
MGC annotation are then compared to the EST alignment result. Those 
predictions are selected as target set 3 if they do not overlap MGC 
annotation and are not aligned by any EST.
</p>

<H4>Twinscan set 4</H4>
<p>
All Twinscan prediction are categorized as following (see the 
diagram below for more details): All ESTs (ESTs, mRNAs, Refseqs, and the 
experimental sequences) were aligned to overall predictions on NCBI35. 
EST alignments are picked if they met the criterion ( at least 98 match 
in a 100 bp window ). From all the Twinscan predictions,  filter out all 
predictions overlapping with MGC annotation by at least 1 bp. Those 
predictions not overlapping with MGC annotations are then compared to 
the EST alignment results. Those predictions are selected as target set 
4 if they do not overlap MGC annotation and are not aligned by any EST.
</p>


<H2>Credits</H2>
<P>
<A HREF="http://genes.cs.wustl.edu/BrentLab/MB-Lab-index.html" 
TARGET=_blank>Michael Brent's Computational Genomics Group</A> at Washington 
University St. Louis.
</p>
