<H2>Description</H2>
<P>
The miRNA track shows microRNAs from the
<A HREF="http://microrna.sanger.ac.uk/" TARGET=_blank>
miRBase</A> at <A HREF="http://www.sanger.ac.uk/" TARGET=_blank>The 
Wellcome Trust Sanger Institute</A>. </P>

<H2>Display Conventions and Configuration</H2>
<P>
Mature miRNAs (miRs) are represented by 
thick blocks. The predicted stem-loop portions of the primary transcripts
are indicated by thinner blocks. miRNAs in the sense orientation are shown in
black; those in the reverse orientation are colored grey. When a single 
precursor produces two mature miRs from its 5' and 3' parts, it is displayed 
twice with the two different positions of the mature miR. </P>
<P>
To display only those items that exceed a specific unnormalized score, enter
a minimum score between 0 and 1000 in the text box at the top of the track 
description page.
</P>

<H2>Methods</H2>
<P>
Mature and precursor miRNAs from the miRNA Registry were
aligned against the genome using blat.
The extents of the precursor sequences were not generally known, and were
predicted based on base-paired hairpin structure.  
miRBase is described in Griffiths-Jones, S. et al. (2006).
The miRNA Registry is
described in Griffiths-Jones, S. (2004) and Weber, M.J. (2005) in the 
References section below.</P>

<H2>Credits</H2>
<P>

This track was created by Michel Weber of <a
href="http://www-lbme.biotoul.fr/" target="_blank">
Laboratoire de Biologie Moléculaire Eucaryote,
CNRS Université Paul Sabatier</a>
(Toulouse, France), Yves Quentin of <a
href="http://www-lmgm.biotoul.fr/" target="_blank">Laboratoire de Microbiologie et Génétique
Moléculaires</a> (Toulouse, France) and Sam Griffiths-Jones of
<a href="http://www.sanger.ac.uk" target="_blank">
The Wellcome Trust Sanger Institute</a>
(Cambridge, UK).
</P>
<H2>References</H2>
<P>
When making use of these data, please cite:</P>
<P>
Griffiths-Jones S, Grocock RJ, van Dongen S, Bateman A, Enright AJ.
<A HREF="http://nar.oxfordjournals.org/cgi/content/full/34/suppl_1/D140"
TARGET=_BLANK>miRBase: microRNA sequences, targets and gene nomenclature</A>.
<em>Nucl. Acids Res.</em> <B>34</B>, Database Issue, D140-D144 (2006).</P>
<P>
Griffiths-Jones, S. 
<A HREF="http://nar.oupjournals.org/cgi/content/full/32/suppl_1/D109"
TARGET=_blank>The microRNA Registry</A>,
<em>Nucl. Acids Res.</em> <B>32</B>, D109-D111 (2004).</P>
<P>
Weber, M. J. 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Citation&list_uids=15634332"
TARGET=_blank>New human and mouse microRNA genes found by homology search</A>.
<em>Febs J</em> <B>272</B>, 59-73 (2005).</P>
<P>
You may also want to cite The Wellcome Trust Sanger Institute 
<A HREF="http://www.sanger.ac.uk/Software/Rfam/mirna/"
TARGET=_blank>miRNA Registry</A>.</P>
<P>
The following publication provides guidelines on miRNA annotation:
Ambros, V. <em>et al</em>., 
<A HREF="http://www.rnajournal.org/cgi/content/full/9/3/277" 
TARGET=_blank>A uniform system for microRNA annotation</A>.  
<em>RNA</em> <B>9</B>(3), 277-279 (2003).</P>
<P>
For more information on blat, see 
Kent, W.J.
<A HREF="http://www.genome.org/cgi/content/abstract/12/4/656" 
TARGET=_blank>BLAT - the BLAST-like alignment tool</A>.
<em>Genome Res.</em> <B>12</B>(4), 656-664 (2002).</P>

