<H2>Description</H2>
<P>
The miRNA track shows microRNAs from the
<A HREF="http://www.sanger.ac.uk/Software/Rfam/mirna/" TARGET=_blank>
miRNA Registry</A> at <A HREF="http://www.sanger.ac.uk/" TARGET=_blank>The 
Wellcome Trust Sanger Institute</A>. Mature miRNAs (miRs) are represented by 
thick lines; the predicted stem-loop portions of the primary transcripts
are indicated by thinner lines. miRNAs in the sense orientation are shown in
black; those in the reverse orientation are colored grey. When a single 
precursor produces two mature miRs from its 5' and 3' parts, it is displayed 
twice with the two different positions of the mature miR.
</P>

<H2>Methods</H2>
<P>
Mature and precursor miRNAs from the miRNA Registry are
aligned against the genome using blat.
The extents of the precursor sequences are not generally known, and are
predicted based on base-paired hairpin structure.  The miRNA Registry is
described in Griffiths-Jones, S. (2004) in the References section below. </P>

<H2>Credits</H2>
<P>
This track was created by Michel Weber of 
<A HREF="http://www-lbme.biotoul.fr/" TARGET=_blank>Laboratoire de Biologie 
Moléculaire Eucaryote, CNRS Université Paul Sabatier</A> (Toulouse, France) 
and Sam Griffiths-Jones of <A HREF="http://www.sanger.ac.uk" 
TARGET=_blank>The Wellcome Trust Sanger Institute</A> (Cambridge, UK).</P> 

<H2>References</H2>
<P>
When making use of these data, please cite: 
Griffiths-Jones, S. 
<A HREF="http://nar.oupjournals.org/cgi/content/full/32/suppl_1/D109"
TARGET=_blank>The microRNA Registry</A>,
<em>Nucl. Acids Res.</em> <B>32</B>, D109-D111 (2004).</P>
<P>
You may also want to cite The Wellcome Trust Sanger Institute 
<A HREF="http://www.sanger.ac.uk/Software/Rfam/mirna/"
TARGET=_blank>miRNA Registry</A>.</P>
<P>
The following publication provides guidelines on miRNA annotation:
Ambros, V. <em>et al</em>., 
<A HREF="http://www.rnajournal.org/cgi/content/full/9/3/277" 
TARGET=_blank>A uniform system for microRNA annotation</A>.  
<em>RNA</em> <B>9</B>(3), 277-279 (2003).</P>
<P>
For more information on blat, see this publication:
Kent, W.J.
<A HREF="http://www.genome.org/cgi/content/abstract/12/4/656" 
TARGET=_blank>BLAT - the BLAST-like alignment tool</A>.
<em>Genome Res.</em> <B>12</B>(4), 656-664 (2002).</P>

