<H2>Description</H2>
<P>
This track shows the best <em>$o_organism</em>/$organism chain for 
every part of the $organism genome. It is useful for
finding orthologous regions and for studying genome
rearrangement.  The <em>$o_organism</em> sequence used in this annotation is 
from the $o_date ($o_db) assembly.</P>

<H2>Display Conventions and Configuration</H2>
<P>
In full display mode, the top-level (level 1)
chains are the largest, highest-scoring chains that
span this region.  In many cases gaps exist in the
top-level chain.  When possible, these are filled in by
other chains that are displayed at level 2.  The gaps in 
level 2 chains may be filled by level 3 chains and so
forth. </P>
<P>
In the graphical display, the boxes represent ungapped 
alignments; the lines represent gaps.  Click
on a box to view detailed information about the chain
as a whole; click on a line to display information
about the gap.  The detailed information is useful in determining
the cause of the gap or, for lower level chains, the genomic
rearrangement. </P> 
<P> 
Individual items in the display are categorized as one of four types
(other than gap):</P>
<P><UL>
<LI><B>Top</B> - the best, longest match. Displayed on level 1.
<LI><B>Syn</B> - line-ups on the same chromosome as the gap in the level above
it.
<LI><B>Inv</B> - a line-up on the same chromosome as the gap above it, but in 
the opposite orientation.
<LI><B>NonSyn</B> - a match to a chromosome different from the gap in the 
level above.
</UL></P>

<H2>Methods</H2>
<P>
Chains were derived from blastz alignments using the methods
described on the chain tracks description pages. The chains were
sorted with the highest-scoring chains in the genome first. The program 
chainNet was used to place the chains one at a time, trimming them as 
necessary to fit into sections not already covered by a higher-scoring chain. 
During this process, a natural hierarchy emerged in which a chain that filled 
a gap in a higher-scoring chain was placed underneath that chain. The program 
netSyntenic was used to fill in
information about the relationship between upper
and lower level chains, including whether a lower level

