<H2>Disclaimer</H2>
<P>
PhenCode is intended for research purposes only. Although the data are freely
available to all, users should treat the reported mutations with extreme
caution in clinical settings or for any diagnostic or population screening
purpose. This information requires expertise to interpret properly; clinical
diagnosis and/or treatment recommendations should be made only by medical 
professionals. 
<a href="http://globin.bx.psu.edu/mutation/disclaimer.html" target="_blank">
<H4>Terms and Conditions of Use</H4></a>
</P>

<H2>Description</H2>
<P>
This track displays literature-curated regulatory regions, transcription
factor binding sites, and regulatory polymorphisms from
<a href="http://www.oreganno.org/oregano/Index.jsp"
target="_blank">ORegAnno</a> (Open Regulatory Annotation). For more detailed
information on a particular regulatory element, follow the link to ORegAnno
from the details page. This track was implemented in the Genome Browser as
part of the <a href="http://globin.bx.psu.edu/mutation/"
target="_blank">PhenCode project</a>. 
<!--
<b><font color="666600">O</font><font color="CC0033">Reg</font><font color="BBBB55">Anno</font></b> (Open Regulatory Annotation).
-->
</P>

<H2>Display Conventions and Configuration</H2>
<P>
The display may be filtered to show only selected region types, such as
regulatory regions, regulatory polymorphisms, or transcription factor
binding sites. To exclude a region type, check the appropriate box
in the &quot;Exclude region type&quot; list at the top of the Track Settings
page. 
</P>

<H2>Methods</H2>
<P>
An ORegAnno record describes an experimentally proven and published regulatory
region (promoter, enhancer, etc.), transcription factor binding site, or
regulatory polymorphism.  Each annotation must have the following attributes:
<ol>
<li>A stable ORegAnno identifier.
<li>A valid taxonomy ID from the NCBI taxonomy database.
<li>A valid PubMed reference.  
<li>A target gene which is either user-defined, in Entrez Gene or in EnsEMBL.
<li>A sequence with at least 40 flanking bases (preferably more) to allow the
site to be mapped to any release of an associated genome.
<li>At least one piece of specific experimental evidence, including the
biological technique used to discover the regulatory sequence. (Currently
only the evidence subtypes are supplied with the UCSC track.)
<li>A positive, neutral or negative outcome based on the experimental results
from the primary reference. (Only records with a positive outcome are currently
included in the UCSC track.)
</ol>
The following attributes are optionally included:
<ol>
<li>A transcription factor which is either user-defined, in Entrez Gene
or in EnsEMBL.
<li>A specific cell type for each piece of experimental evidence, using the
eVOC cell type ontology.
<li>A specific dataset identifier (e.g. the REDfly dataset) that allows
external curators to manage particular annotation sets using ORegAnno's
curation tools.
<li>A &quot;search space&quot; sequence that specifies the region that was
assayed, not just the regulatory sequence.  
<li>A dbSNP identifier and type of variant (germline, somatic or artificial)
for regulatory polymorphisms.
</ol>
Mapping to genome coordinates is performed periodically to current genome
builds by BLAST sequence alignment.  
The information provided in this track represents an abbreviated summary of the 
details for each ORegAnno record.  Please visit the official ORegAnno entry 
(by clicking on the ORegAnno link on the details page of a specific regulatory
element) for complete details such as evidence descriptions, comments,
validation score history, etc.
</P>

<H2>Credits</H2>
<P>
<UL>
<LI>The ORegAnno core team and principal contacts:
<UL>Stephen Montgomery, Obi Griffith, Steven Jones<BR>
  Canada's Michael Smith Genome Sciences Centre, Vancouver, British Columbia,
  Canada</UL>
<LI>The ORegAnno community (please see individual citations for various
  features): 
<UL><a href="http://www.oreganno.org/oregano/Cite.jsp"
  target="_blank">ORegAnno Citation</a></UL>
<LI>PhenCode developers:
<UL><LI>Belinda Giardine, Ross Hardison, Webb Miller, Cathy Riemer<br>
  Center for Comparative Genomics and Bioinformatics, Penn State University,
  University Park, Pennsylvania
<LI>Fan Hsu, Jim Kent, Andrew Kern, Robert Kuhn, Heather Trumbower<br>
  Center for Biomolecular Science and Engineering, University of California,
  Santa Cruz, California</UL>
</UL>
<!--
<DT>Laura Elnitski
<DD>National Human Genome Research Institute, Rockville, MD
<DT>Richard Gibbons, Doug Higgs, Jim Hughes
<DD>Weatherall Institute of Molecular Medicine, Oxford UK
<DT>Garry Cutting, Andrew P. Feinberg
<DD>Johns Hopkins University School of Medicine, Baltimore, MD
-->
</DL> 

<H2>References</H2>
<!--
<P>
Elnitski, L., Hardison, R.C., Li, J., Yang, S., Kolbe, D., Eswara, P., 
O'Connor, M.J., Schwartz, S., Miller, W. and Chiaromonte, F. 
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/64" 
TARGET=_blank>Distinguishing regulatory DNA from neutral sites</A>.
<em>Genome Res.</em> <B>13</B>(1):64-72 (2003).</P>
-->
<P>
Montgomery, S.B., Griffith, O.L., Sleumer, M.C., Bergman, C.M., Bilenky, M., 
Pleasance, E.D., Prychyna, Y., Zhang, X., Jones, S.J.M. 
<A HREF="http://bioinformatics.oxfordjournals.org/cgi/content/full/22/5/637" TARGET=_blank>ORegAnno: an open access database and curation system for literature-derived promoters, transcription factor binding sites and regulatory variation</A>.
<em>Bioinformatics.</em> <B>22</B>(5), 637-40 (2006).
</P>
