<H2>Description</H2> 
<P> 
This track shows predictions of conserved elements from the phastCons
program.  PhastCons bases its predictions on a phylogenetic hidden Markov model
(phylo-HMM), a type of probabilistic model that describes both the process
of DNA substitution at each site in a genome and the way this process
changes from one site to the next. </P>

<H2>Methods</H2>
<P> Best-in-genome pairwise alignments were generated for each species using
blastz, followed by chaining and netting.  A multiple alignment was then
constructed from these pairwise alignments using multiz.  A phylogenetic
model was fitted to the multiz alignments by maximum likelihood,
conditioning on the accepted tree topology.  This was accomplished with the
program phyloFit, using the general reversible (REV) substitution model and
discrete gamma model for rate variation (Yang, 1994).  (With some sets of
species, special allowances are made for distant species, which can be
aligned only in conserved regions.  Branch lengths to these species are
estimated approximately, using branch length proportions inferred from
sites in 3rd codon position.)</P>

<P> The multiz alignments were split into fragments of approximately 1Mb in
size, such that cut-points occurred between alignment blocks where
possible.  The phastCons program was run on each of these fragments with
the --viterbi option and the globally estimated phylogenetic model.
PhastCons constructs a two-state phylo-HMM with a state for conserved
regions and a state for non-conserved regions.  It "cuts" the estimated
distribution of evolutionary rates at a designated quantile and creates a
phylogenetic model for the conserved state based on the slower evolutionary
rates and a phylogenetic model for the non-conserved state based on the
faster evolutionary rates.  When run with the --viterbi option, the program
identifies segments of the alignment that are likely to have been
"generated" by the conserved state--i.e., maximal segments in which the
maximum-likelihood (Viterbi) path through the phylo-HMM completely resides
in the conserved state.  These predicted segments are combined across
alignment fragments and used to construct the track.  The transition
probabilities between the states in the phylo-HMM are approximately
estimated by maximum likelihood.</P>

<H2>Credits</H2>
<P>
This track was created at UCSC using the following programs:
<UL>
<LI>
Blastz and multiz from Scott Schartz and Webb Miller of the 
<A HREF="http://www.bx.psu.edu/" TARGET=_blank>Penn State Bioinformatics 
Group</A>. 
<LI>
AxtBest, axtChain, chainNet, netSyntenic, and netClass 
developed by Jim Kent at UCSC. 
<LI>Tree estimation (phyloFit) and conservation-scoring (phastCons)
software by Adam Siepel at UCSC. 
</UL>
</P>

<H2>References</H2>

<H3>Phylo-HMMs</H3>
<P>
Felsenstein J and Churchill GA (1996).  A hidden Markov model approach to
variation among sites in rate of evolution.
<em>Mol Biol Evol</em> 13:93-104. </P>
<P>
Siepel A and Haussler D (2004). Phylogenetic hidden Markov models.
In R. Nielsen, ed.,  Statistical Methods in Molecular Evolution, Springer  
(in press). </P>
<P>
Yang Z (1994).  Maximum likelihood phylogenetic estimation from DNA sequences
with variable rates over sites: approximate methods,
<em>J Mol Evol</em> 39:306-314.</P>
<P>
Yang Z (1995).  A space-time process model for the evolution of DNA
sequences.  <em>Genetics</em>, 139:993-1005. </P>

<H3>Chain/Net</H3>
<P>
Kent WJ, Baertsch R, Hinrichs A, Miller W, and Haussler D (2003). 
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1" 
TARGET=_blank>Evolution's cauldron: 
Duplication, deletion, and rearrangement in the mouse and human genomes</A>. 
<I>Proc Natl Acad Sci USA</I> 100(20):11484-11489 Sep 30 2003.

<H3>Multiz</H3>
<P>
Blanchette M, Kent WJ, Riemer C, Elnitski L, Smit AFA, Roskin KM, Baertsch R,
Rosenbloom K, Clawson H, Green ED, Haussler D, Miller W. (2004). 
<A HREF="http://www.genome.org/cgi/content/abstract/14/4/708" TARGET=_blank>Aligning Multiple Genomic Sequences with the Threaded Blockset Aligner</A>.
<em>Genome Res.</em> 14(4):708-15.

<H3>Blastz</H3>
<P>
Chiaromonte F, Yap VB, and Miller W (2002). 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" 
TARGET=_blank>Scoring pairwise genomic sequence alignments</A>. 
<em>Pac Symp Biocomput</em> 2002;:115-26.</P>
<P>
Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison R, Haussler D, and 
Miller W. (2003). 
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103" 
TARGET=_blank>Human-Mouse Alignments with BLASTZ</A>. 
<em>Genome Res.</em> 13(1):103-7.</P>
