<H2>Description</H2>
<P>
The RefSeq Genes track shows known protein-coding genes taken from mRNA 
reference sequences compiled at LocusLink. On assemblies in which incremental
GenBank downloads are supported, the data underlying this track are updated 
nightly.
Coding exons are represented by 
blocks connected by horizontal lines representing introns. The 5' and 3' 
untranslated regions (UTRs) are displayed as thinner blocks on the leading 
and trailing ends of the aligning regions. In full display mode, arrowheads 
on the connecting intron lines indicate the direction of transcription.
The color shading indicates the level of review the RefSeq record has 
undergone: predicted (light), provisional (medium), reviewed (dark). 
</P>
<P>
Non-coding RNA genes have their own track in some assemblies.
</P>
<H2>Method</H2>
<P>
Refseq mRNAs are aligned against the genome using the 
<A HREF="../cgi-bin/hgBlat?db=$db" TARGET=_blank>blat</A>
program. When a single mRNA aligns in multiple places, only
the best alignments which also have at least 98% sequence identity are kept.
</P>
<H2>Using the Filter</H2>
<P>The track filter can be used to configure the labeling of the features within
the track. By default, items are labeled by gene name. Click the 
appropriate Label option to display the accession name instead of the gene
name, show both the gene and accession names, or turn off the label completely.
After you have made your selection, click Submit to return to the tracks display
page.
<H2>Credits</H2>
<P>
The RefSeq Genes track is produced at UCSC from mRNA sequence data
generated by scientists worldwide and curated by the 
NCBI <A HREF="http://www.ncbi.nih.gov/RefSeq/" TARGET=_blank>RefSeq project</A>. 
</P>
