<H2>Description</H2>
<P>
This track shows the boundaries of genes and the direction of
transcription as deduced from clustering spliced ESTs and mRNAs
against the genome.  When there are many spliced variants
of the same gene, this track shows the variant that
spans the greatest distance in the genome.</P>

<H2>Method</H2>
<P>
ESTs and mRNAs from 
<A HREF="http://www.ncbi.nlm.nih.gov/Genbank/index.html" 
TARGET=_blank>GenBank</A> are aligned against the genome with the 
<A HREF="../cgi-bin/hgBlat?db=$db" TARGET=_blank>blat</A>
program, and filtered to keep only those alignments
that have at least 97.5% base identity within the 
aligning blocks.  When multiple alignments occur, only the
alignments with a percentage identity within 0.2% of the
best alignment are kept.  ESTs that align without any
introns are discarded.  Blocks that are less than 130 bases
and are not next to an intron are discarded.   Blocks smaller
than 10 bases are discarded.   The orientations of the 
ESTs and mRNAs are deduced from the GT/AG splice sites
at the introns,  and ESTs and mRNAs with overlapping blocks
on the same strand are merged into clusters.  Only the
extent and orientation of the clusters are shown here.
</P>
<P>
Scores for individual gene boundaries are assigned based on the number of 
cDNA alignments used:<br>
300 if based on a single cDNA alignment<br> 
600 if based on 2<br> 
900 if based on 3<br>
1000 if based on 4 or more<br>
</P>
<H2>Credits</H2>
<P>
This track -- which was originally developed by Jim Kent --
was generated at UCSC and uses data submitted to GenBank by 
scientists worldwide.</P>

<H2>References</H2>
<P>
Benson, D.A., Karsch-Mizrachi, I., Lipman, D.J., Ostell, J., and Wheeler, D.L.
<A HREF="http://nar.oupjournals.org/cgi/content/full/32/suppl_1/D23" 
TARGET=_blank>GenBank: update</A>. 
<em>Nucleic Acids Res.</em> <B>32</B>, D23-6 (2004).
