<H2>Description</H2>

<P>
This track contains
<A HREF=" http://www.ncbi.nlm.nih.gov/SNP/" target=_blank>dbSNP</A>
build 125, available from
<A HREF="ftp://ftp.ncbi.nih.gov/snp/organisms" target=_blank>ftp.ncbi.nih.gov/snp</A>.
</P>

<H2>Interpreting and Configuring the Graphical Display</H2>
<P>
  Variants are shown as single tick marks at most zoom levels.
  When viewing the track at or near base-level resolution, the displayed
  width of the SNP corresponds to the width of the variant in the reference
  sequence. Insertions are indicated by a single tick mark displayed between
  two nucleotides, single nucleotide polymorphisms are displayed as the width 
  of a single base, and multiple nucleotide variants are represented by a 
  block that spans two or more bases.
</P>
<P>
  When the start coordinate for a SNP is shown as chromStart = chromEnd+1 on 
  the SNP's details page, this is generally not an 
  error; rather, it indicates that the variant is an insertion at this genomic
  position.  
  Note that insertions are represented as chromStart = 
  chromEnd in the <em>snp</em> table accessible from the Table Browser 
  or downloads server, due to the half-open zero-based representation of
  data in the underlying database. 
</P>

<P>
  The configuration categories reflect the following definitions:
  <UL>

    <LI>
      <A name="LocType"></A>
      <B>Location Type</B>: Describes the alignment of the flanking sequence<BR>
      <UL>
        <LI><B>Range</B> - The flank alignments leave a gap of 2 or more bases in the reference assembly.
        <LI><B>Exact</B> - The flank alignments leave exactly one base between them.
        <LI><B>Between</B> - The flank alignments are contiguous; the variation is an insertion. 
        <LI><B>RangeInsertion</B> - The flank alignments surround a distinct polymorphism between
	                            the submitted sequence and reference assembly.    
				    The submitted sequence is shorter.
        <LI><B>RangeSubstitution</B> - The flank alignments surround a distinct polymorphism between
	                            the submitted sequence and reference assembly.    
				    The submitted sequence and the reference assembly sequence are of equal length.
        <LI><B>RangeDeletion</B> - The flank alignments surround a distinct polymorphism between
	                           the submitted sequence and reference assembly.    
				    The submitted sequence is longer.
      </UL>
    </LI>
    <LI>
      <A name="Class"></A>
      <B>Variant Class</B>: Describes the observed alleles<BR>
      <UL>
        <LI><B>Single</B> - single nucleotide variation: all observed alleles are single nucleotides
	    (can have 2, 3 or 4 alleles)
        <LI><B>Insertion</B> - the polymorphism is an insertion relative to the reference assembly
        <LI><B>Deletion</B> - the polymorphism is a deletion relative to the reference assembly
        <LI><B>In-del</B> - insertion/deletion (applies to RangeInsertion, RangeSubstitution, RangeDeletion)
        <LI><B>Mixed</B> - the cluster contains submissions from multiple classes
        <LI><B>Named</B> - the observed allele from dbSNP is given as a text name
        <LI><B>Microsatellite</B> - the observed allele from dbSNP is variation in counts of short tandem repeats
        <LI><B>Unknown</B> - no classification provided by data contributor
      </UL>
    </LI>


    <LI>
      <A name="Valid"></A>
      <B><A HREF="http://www.ncbi.nlm.nih.gov/SNP/snp_legend.cgi?legend=validation" 
	target="_blank">Validation Status</A></B>: Method used to validate
	the variant (<I>each variant may be validated by more than one method</I>)<BR>
        <UL>
        <LI><B>By Frequency</B> - at least one submitted SNP in cluster has frequency data submitted
        <LI><B>By Cluster</B> - cluster has at least 2 submissions, with at least one submission assayed with a non-computational method
        <LI><B>By Submitter</B> - at least one submitter SNP in cluster was validated by independent assay
        <LI><B>By 2 Hit/2 Allele</B> - all alleles have been observed in at least 2 chromosomes
        <LI><B>By HapMap</B> - validated by HapMap project
        <LI><B>Unknown</B> - no validation has been reported for this variant
      </UL>
    </LI>
    <LI>
      <A name="Func"></A>
      <B>Function</B>: Predicted functional role 
	(<I>each variant may have more than one functional role</I>)<BR>
      <UL>
        <LI><B>Locus Region</B> - variation within 2000 bases of gene, but not in transcript
        <LI><B>Coding - Synonymous</B> - no change in peptide for allele with respect to reference assembly
        <LI><B>Coding - Non-Synonymous</B> - change in peptide for allele with respect to reference assembly
        <LI><B>Untranslated</B> - variation in transcript, but not in coding region interval
        <LI><B>Intron</B> - variation in intron, but not in first two or last two bases of intron
        <LI><B>Splice Site</B> - variation in first two or last two bases of intron
        <LI><B>Reference</B> - allele observed in a coding region of the reference sequence
        <LI><B>Unknown</B> - no known functional classification
      </UL>
    </LI>
    <LI>
      <A name="MolType"></A>
      <B>Molecule Type</B>: Sample used to find this variant<BR>
      <UL>
        <LI><B>Genomic</B> - variant discovered using a genomic template
        <LI><B>cDNA</B> - variant discovered using a cDNA template
        <LI><B>Unknown</B> - sample type not known
      </UL>
    </LI>


  </UL>
</P>
<H2>UCSC Annotations</H2>
<P>
  In addition to presenting the dbSNP data, we provide the following annotations:
  <UL>
  <LI>dbSNP uses a class called 'in-del'.   We split this into
      'insertion' and 'deletion', based on location type.
      Location type 'range' and 'exact' are deletions relative 
      to the reference assembly.
      Location type 'between' are insertions relative to the
      reference assembly.</LI>
  <LI>We check that the size of the dbSNP reference allele matches the coordinate span.</LI>

  <LI>We compare the dbSNP reference allele to our reference allele,
      and note differences.   We also note if the difference is specifically
      that the dbSNP reference allele is the reverse complement.</LI>
      
  <LI>For single-base substitutions, we note where the alignments of the
      flanking sequences are adjacent or have a gap of more than one base.</LI>
  <LI>We note if the observed alleles are not available from the rs_fasta files.</LI>
  <LI>We note if the observed alleles have an unexpected format.</LI>
  <LI>We check that the length of the observed alleles match the coordinate span.</LI>
  <LI>For simple deletions, we check that the observed allele matches the reference allele.</LI>
  <LI>For single-base substitutions, we check that one of the observed alleles matches the reference allele.</LI>
  <LI>We note tri-allelic and quad-allelic single-base substitutions.</LI>
  <LI>We note variants that have multiple mappings.</LI>
  </UL>
</P>

<H2>Data Sources</H2>
<P>
  <UL>
  <LI>We used b125_SNPContigLoc.bcp.gz for coordinates, orientation, location type
      and the dbSNP reference allele.  
  <LI>We used b125_SNPMapInfo.bcp.gz for alignment weights, filtering out 
      alignments with weight = 3 or 10.
  <LI>We obtained functional classification from b125_SNPContigLocusId.bcp.gz.
  <LI>We obtained validation status and heterozygosity from SNP.bcp.gz.
  <LI>We used the header line in the rs_fasta files for class, 
      observed polymorphism and molecule type.
  </UL>
</P>



