<H2>Description</H2>

<P>
This track contains
<A HREF=" http://www.ncbi.nlm.nih.gov/SNP/" target=_blank>dbSNP</A>
build 126, available from
<A HREF="ftp://ftp.ncbi.nih.gov/snp/organisms" target=_blank>ftp.ncbi.nih.gov/snp</A>.
</P>

<H2>Interpreting and Configuring the Graphical Display</H2>
<P>
  Variants are shown as single tick marks at most zoom levels.
  When viewing the track at or near base-level resolution, the displayed
  width of the SNP corresponds to the width of the variant in the reference
  sequence. Insertions are indicated by a single tick mark displayed between
  two nucleotides, single nucleotide polymorphisms are displayed as the width 
  of a single base, and multiple nucleotide variants are represented by a 
  block that spans two or more bases.
</P>

<P>
  The configuration categories reflect the following definitions:
  <UL>

    <LI>
      <A name="LocType"></A>
      <B>Location Type</B>: Describes the alignment of the flanking sequence<BR>
      <UL>
        <LI><B>Range</B> - the flank alignments leave a gap of 2 or more bases in the reference assembly
        <LI><B>Exact</B> - the flank alignments leave exactly one base between them
        <LI><B>Between</B> - the flank alignments are contiguous; the variation is an insertion
        <LI><B>RangeInsertion</B> - the flank alignments surround a distinct polymorphism between
	                            the submitted sequence and reference assembly;    
				    the submitted sequence is shorter
        <LI><B>RangeSubstitution</B> - the flank alignments surround a distinct polymorphism between
	                            the submitted sequence and reference assembly;
				    the submitted sequence and the reference assembly sequence are of equal length
        <LI><B>RangeDeletion</B> - the flank alignments surround a distinct polymorphism between
	                           the submitted sequence and reference assembly;
				    the submitted sequence is longer
      </UL>
    </LI>
    <LI>
      <A name="Class"></A>
      <B>Class</B>: Describes the observed alleles<BR>
      <UL>
        <LI><B>Single</B> - single nucleotide variation: all observed alleles are single nucleotides
	    (can have 2, 3 or 4 alleles)
        <LI><B>In-del</B> - insertion/deletion (applies to RangeInsertion, RangeSubstitution, RangeDeletion)
        <LI><B>Heterozygous</B> - heterozygous (undetermined) variation: allele contains string '(heterozygous)'
        <LI><B>Microsatellite</B> - the observed allele from dbSNP is variation in counts of short tandem repeats
        <LI><B>Named</B> - the observed allele from dbSNP is given as a text name
        <LI><B>No Variation</B> - no variation asserted for sequence
        <LI><B>Mixed</B> - the cluster contains submissions from multiple classes
        <LI><B>Multiple Nucleotide Polymorphism</B> - alleles of the same length, length > 1, and from set of {A,T,C,G}
        <LI><B>Insertion</B> - the polymorphism is an insertion relative to the reference assembly
        <LI><B>Deletion</B> - the polymorphism is a deletion relative to the reference assembly
        <LI><B>Unknown</B> - no classification provided by data contributor
      </UL>
    </LI>


    <LI>
      <A name="Valid"></A>
      <B><A HREF="http://www.ncbi.nlm.nih.gov/SNP/snp_legend.cgi?legend=validation" 
	target="_blank">Validation</A></B>: Method used to validate
	the variant (<I>each variant may be validated by more than one method</I>)<BR>
        <UL>
        <LI><B>By Frequency</B> - at least one submitted SNP in cluster has frequency data submitted
        <LI><B>By Cluster</B> - cluster has at least 2 submissions, with at least one submission assayed with a non-computational method
        <LI><B>By Submitter</B> - at least one submitter SNP in cluster was validated by independent assay
        <LI><B>By 2 Hit/2 Allele</B> - all alleles have been observed in at least 2 chromosomes
        <LI><B>By HapMap</B> - validated by HapMap project
        <LI><B>Unknown</B> - no validation has been reported for this variant
      </UL>
    </LI>
    <LI>
      <A name="Func"></A>
      <B>Function</B>: Predicted functional role 
	(<I>each variant may have more than one functional role</I>)<BR>
      <UL>
        <LI><B>Locus Region</B> - variation within 2000 bases of gene, but not in transcript
        <LI><B>Coding - Synonymous</B> - no change in peptide for allele with respect to reference assembly
        <LI><B>Coding - Non-Synonymous</B> - change in peptide for allele with respect to reference assembly
        <LI><B>Untranslated</B> - variation in transcript, but not in coding region interval
        <LI><B>Intron</B> - variation in intron, but not in first two or last two bases of intron
        <LI><B>Splice Site</B> - variation in first two or last two bases of intron
        <LI><B>Reference</B> - allele observed in a coding region of the reference sequence
        <LI><B>Unknown</B> - no known functional classification
      </UL>
    </LI>
    <LI>
      <A name="MolType"></A>
      <B>Molecule Type</B>: Sample used to find this variant<BR>
      <UL>
        <LI><B>Genomic</B> - variant discovered using a genomic template
        <LI><B>cDNA</B> - variant discovered using a cDNA template
        <LI><B>Unknown</B> - sample type not known
      </UL>
    </LI>
    <LI>
      <A name="Weight"></A>
      <B>Weight</B>: Alignment quality assigned by dbSNP<BR>
      <UL>
      <LI>Weight can be 0, 1, 2, 3 or 10.   
      <LI>We excluded weight = 0 and weight = 10 from the data set.
      <LI>We support a filter on maximum weight value, which defaults to 3.
      </UL>
    </LI>
    <LI>
      <A name="AvHet"></A>
      <B>Average heterozygosity</B>: Calculated by dbSNP as described 
      <A HREF="http://www.ncbi.nlm.nih.gov/SNP/Hetfreq.html" target=_blank>here</A>
      <UL>
      <LI> Average heterozygosity should not exceed .5 for bi-allelic single-base substitutions.
      </UL>
    </LI>
  </UL>
</P>
<H2>Insertions/Deletions</H2>
dbSNP uses a class called 'in-del'.   We split this into 'insertion' and 'deletion', based on location type.
Location type 'range' and 'exact' are deletions relative to the reference assembly.
Location type 'between' are insertions relative to the reference assembly.
For the new location types, we preserve the class 'in-del'.

<H2>UCSC Annotations</H2>
<P>
  In addition to presenting the dbSNP data, we provide the following annotations:
  <UL>
  <LI>We compare the dbSNP reference allele to our reference allele and note if the 
      dbSNP reference allele is the reverse complement of our reference allele.</LI>
  <LI>For single-base substitutions, we note where the alignments of the
      flanking sequences are adjacent or have a gap of more than one base.</LI>
  <LI>We note if the observed alleles have an unexpected format.</LI>
  <LI>We check that the length of the observed alleles is consistent with the location type.</LI>
  <LI>For single-base substitutions, we check that one of the observed alleles matches the reference allele.</LI>
  <LI>For simple deletions, we check that the observed allele matches the reference allele.</LI>
  <LI>We note tri-allelic and quad-allelic single-base substitutions.</LI>
  <LI>We note variants that have multiple mappings.</LI>
  </UL>
</P>

<H2>Data Sources</H2>
<P>
  <UL>
  <LI>We used b126_SNPContigLoc_36_1.bcp.gz for coordinates, orientation, location type
      and the dbSNP reference allele.  
  <LI>We used b126_SNPMapInfo_36_1.bcp.gz for alignment weights, filtering out 
      alignments with weight = 0 or weight = 10.
  <LI>We obtained class and observed polymorphism from the shared UniVariation.bcp.gz,
      using the univar_id from SNP.bcp.gz as an index.
  <LI>We obtained functional classification from b126_SNPContigLocusId_36_1.bcp.gz.
  <LI>We obtained validation status and heterozygosity from SNP.bcp.gz.
  <LI>We used the header line in the rs_fasta files for molecule type.
  </UL>
</P>



