<H2>Description</H2>

<P>This track shows locations of Single Nucleotide Polymorphisms
detected by aligning reads from random genomic clones from
a diverse pool of human DNA against the genome.</P>

<P>The SNPs in this track include all of the polymorphisms that can be
mapped against the current assembly.  These include known point
mutations (Single Nucleotide Polymorphisms), insertions, deletions,
and segmental mutations from the current build of 
<A HREF="ftp://ftp.ncbi.nih.gov/snp/" TARGET=_blank>dbSNP</A>.  
</P>
<P>
There are three major cases that are not mapped and/or annotated:
<UL>
<LI> 
Submissions that are completely masked as repetitive
elements. These are dropped from any further computations. This set of
refSNPs is dumped in chromosome &quot;rs_chMasked&quot; on the dbSNP ftp site.
<LI>
Submissions that are defined in a cDNA context with extensive
splicing. These SNPs are typically annotated on refseq mRNAs through a
separate annotation process. Effort is being made to reverse map these
variations back to contig coordinates, but that has not been
implemented. For now, you can find this set of variations in
&quot;rs_chNotOn&quot; on the dbSNP ftp site.
<LI>
Submissions with excessive hits to the genome. Variations with
3+ hits to the genome are not included in the tracks, but are available in
&quot;rs_chMulti&quot; on the dbSNP ftp site.
</UL>
</P>
<P>The heuristics for the non-SNP variations (i.e. named elements and
STRs) are quite conservative; therefore, some of these are probably lost. This
approach was chosen to avoid false annotation of variation in
inappropriate locations.</P>

<H2>Credits</H2>

<P>Thanks to the <A HREF="http://snp.cshl.org/" TARGET=_blank>SNP
Consortium</A> and NIH's <A HREF="http://www.ncbi.nlm.nih.gov/SNP/"
TARGET=_blank>dbSNP</A> for providing these data.</P>

