<H2>Description</H2>
The 
<A HREF="http://supfam.org/SUPERFAMILY/index.html">Superfamily</A> 
track shows proteins having homologs with known structures or functions.
<P>
Each entry on this track shows the coding region of a gene (based on Ensembl gene prediction).
The label,
shown in Full mode at the left hand side of each entry, 
consists of the names of 
all known protein domains coded by this gene.  This 
usually contains structural and/or function descriptions.  They provide valuable information for 
our users to get a quick grasp of the biological significance for each gene.
<H2>Method</H2>
Data are downloaded from the Superfamily server.
Using the cross-reference between Superfamily entries and Ensembl gene prediction entries
and their alignment to the appropriate genome, the associated data are processed to generate 
a simple bed format UCSC Genome Browser track.
<H2>Credits</H2>
Superfamily is developed by
<a href="http://www.mrc-lmb.cam.ac.uk/genomes/julian/julian.html">Julian
Gough</a> at the <a href="http://www2.mrc-lmb.cam.ac.uk/">MRC Laboratory
of Molecular Biology</a>, Cambridge.
<P>
<a href="http://www.mrc-lmb.cam.ac.uk/genomes/julian/gough_jmb_supfam_2001.pdf">
<font face="Arial,Helvetica">Gough, J., Karplus, K., Hughey, R. and
Chothia, C. (2001). "Assignment of Homology to Genome Sequences using a
Library of Hidden Markov Models that Represent all Proteins of Known Structure." 
<i>J. Mol. Biol.</i>, <b>313(4)</b>, 903-919.</font>
</a>

