<H2>Description</H2>
<P>
These tracks contain cDNA and gene alignments produced by
the <em>TransMap</em> cross-species alignment algorithm
from other vertebrate species in the UCSC Genome Browser.
For closer evolutionary distances, the alignments are created using
syntenically filtered BLASTZ alignment chains, resulting a prediction of the
orthologous genes in $organism.
</P>

<!-- everything below here common to all transMap*.html pages -->
<H2>Display Conventions and Configuration</H2>
<P>
This track follows the display conventions for 
<A HREF="../goldenPath/help/hgTracksHelp.html#PSLDisplay" 
TARGET=_blank>PSL alignment tracks</A>. </P>
<P>
This track may also be configured to display codon coloring, a feature that
allows the user to quickly compare mRNAs against the genomic sequence. For more 
information about this option, click 
<A HREF="../goldenPath/help/hgCodonColoringMrna.html" TARGET=_blank>here</A>.
Several types of alignment gap may also be colored; 
for more information, click 
<A HREF="../goldenPath/help/hgIndelDisplay.html" TARGET=_blank>here</A>.


<H2>Methods</H2>
<P>
  <ol>
    <li> Source transcript alignments are obtained from vertebrate organisms
    in the UCSC Genome Browser Database. BLAT alignments of RefSeq Genes,
    GenBank mRNAs, and GenBanks Spliced ESTs to their cognate genome, along
    with UCSC Genes, were used as available.
    <li> For all vertebrate assemblies that have BLASTZ alignment chains and
      nets to the $organism $db genome, a subset of the alignment chains are
      selected as follows:
      <li> For organisms who's branch distance is no more than 0.5
        (as computed by <tt>phyloFit</tt>, see Conservation track description for details)
        use syntenic filtering.  If reciprocal best nets are available, use
        them, otherwise select nets with the <tt>netfilter -syn</tt> command.
        The chains that correspond to the nets are used for mapping.
      <li> More distance species, where the determination of synteny is difficult,
        use the full set of chains for mapping.2 This allows for more genes to
        map at the expense of some mapping to paralogus regions.  The post
        alignment filtering step will remove some of the duplications.
    <li> The <tt>pslMap</tt> program is used to do a base-level projection of
      the source transcript alignments via the syntenic chains to this genome.
      to this genome, resulting in pairwise alignments of the source transcript to
      the $organism genome.
    <li> Filter the resulting alignments with <tt>pslCDnaFilter</tt>
      with a global near-best criteria of 0.5% in finished genomes
      (human and mouse) and 1.0% in other genomes.  Alignments
      were less than 20% of the transcript mapped are discarded.
  </ol>
</P>

<P>
To ensure unique identifiers for each alignment, cDNA and gene accessions are
made unique by appending a suffix for each location in the source genome and
again for each mapped location in the destination genome.  The format is:
<pre>
   accession.version-srcUniq.destUniq
</pre>

Were <tt>srcUniq</tt> is a number added to make the each source alignment unique, and
<tt>destUniq</tt> is added to give the subsequent TransMap alignments unique
identifiers.
</P>
<P>
For example, in the cow bosTau4 genome, there are two alignments of mRNA <tt>BC149621.1</tt>.
These are assigned the identifiers <tt>BC149621.1-1</tt> and <tt>BC149621.1-2</tt>.
When these are mapped to the human hg18 genome, <tt>BC149621.1-1</tt> maps to a single
location and is given the identifier <tt>BC149621.1-1.1</tt>.  However, <tt>BC149621.1-2</tt>
maps to two locations, resulting in <tt>BC149621.1-2.1</tt> and <tt>BC149621.1-2.2</tt>.  Note
that multiple TransMap mappings are usually the result of tandem duplications, where both
chains are identified as syntenic.
</P>


<H2>Credits</H2>
<P>
This track was produced at UCSC from cDNA sequence data
submitted to the international public sequence databases by 
scientists worldwide.</P>

<H2>References</H2>
<P>
Zhu J, Sanborn JZ, Diekhans M, Lowe CB, Pringle T, Haussler D. 
<A HREF="http://compbiol.plosjournals.org/perlserv/?request=get-document&doi=10.1371%2Fjournal.pcbi.0030247.eor"
TARGET=_blank>Comparative genomics search for losses of long-established genes 
on the human lineage</A>. 
<em>PLoS Comput Biol</em>. 2007 Dec;3(12):e247.</P>
<P>
Stanke M, Diekhans M, Baertsch R, Haussler D.
<A HREF="http://bioinformatics.oxfordjournals.org/cgi/content/full/24/5/637"
TARGET=_blank>Using native and syntenically mapped cDNA alignments to improve 
de novo gene finding</A>.
<em>Bioinformatics</em>. 2008 Mar 1;24(5):637-44.</P>
<P>
Siepel A, Diekhans M, Brejová B, Langton L, Stevens M, Comstock CL, Davis C, 
Ewing B, Oommen S, Lau C <em>et al.</em>
<A HREF="http://www.genome.org/cgi/content/abstract/gr.7128207v1"
TARGET=_blank>Targeted discovery of novel human exons by comparative 
genomics</A>.
<em>Genome Res.</em> 2007 Dec;17(12):1763-73.</P>

