<H2>Description</H2>
<P>
These tracks contain cDNA and gene alignments produced by
the <em>TransMap</em> cross-species alignment algorithm
from other vertebrate species in the UCSC Genome Browser.
The alignments are creating using syntenicly filtered 
BLASTZ alignment chains, resulting a prediction of the
orthologous genes in $organism.
</P>

<H2>Display Conventions and Configuration</H2>
<P>
This track follows the display conventions for 
<A HREF="../goldenPath/help/hgTracksHelp.html#PSLDisplay" 
TARGET=_blank>PSL alignment tracks</A>. </P>
<P>
This track may also be configured to display codon coloring, a feature that
allows the user to quickly validate and compare mRNA. For more 
information about this option, click 
<A HREF="../goldenPath/help/hgCodonColoringMrna.html" TARGET=_blank>here</A>.
Several types of alignment gap may also be colored; 
for more information, click 
<A HREF="../goldenPath/help/hgIndelDisplay.html" TARGET=_blank>here</A>.


<H2>Methods</H2>
<P>
  <ol>
    <li> BLAT alignments of native RefSeq Genes, GenBank mRNAs, and GenBanks Spliced ESTs,
      along with UCSC Gene Annotations, were obtained from vertebrate organisms in the
      UCSC Genome Browser Database, when available.
    <li> For all assemblies that have BLASTZ alignment chains and nets to the $organism $db
      genome, alignment chains identified as syntenic using <tt>netfilter -syn</tt> are selected.
    <li> The <tt>pslMap</tt> program is used to do a base-level projection of the cDNA alignments
      to this genome.
  </ol>
</P>

<P>
To ensure unique identifiers for each alignment, cDNA and gene accessions are
made unique by appending a suffix for each location in the source genome and
again for each mapped location in the destination genome.  The format is:
<pre>
   accession.version-srcUniq.destUniq
</pre>

Were <tt>srcUniq</tt> is a number added to make the each source alignment unique, and
<tt>destUniq</tt> is added to give the subsequent TransMap alignments unique
identifiers.
</P>
<P>
For example, in the cow bosTau4 genome, there are two alignments of mRNA <tt>BC149621.1</tt>.
These are assigned the identifiers <tt>BC149621.1-1</tt> and <tt>BC149621.1-2</tt>.
When these are mapped to the human hg18 genome, <tt>BC149621.1-1</tt> maps to a single
location and is given the identifier <tt>BC149621.1-1.1</tt>.  However, <tt>BC149621.1-2</tt>
maps to two locations, resulting in <tt>BC149621.1-2.1</tt> and <tt>BC149621.1-2.2</tt>.  Note
that multiple TransMap mappings are usually the result of tandem duplications, where both
chains are identified as syntenic.
</P>


<H2>Credits</H2>
<P>
This track was produced at UCSC from cDNA sequence data
submitted to the international public sequence databases by 
scientists worldwide.</P>

<H2>References</H2>
<P>
Zhu J, Sanborn JZ, Diekhans M, Lowe CB, Pringle T, Haussler D. 
<A HREF="http://compbiol.plosjournals.org/perlserv/?request=get-document&doi=10.1371%2Fjournal.pcbi.0030247.eor"
TARGET=_blank>Comparative genomics search for losses of long-established genes 
on the human lineage</A>. 
<em>PLoS Comput Biol</em>. 2007 Dec;3(12):e247.</P>
<P>
Stanke M, Diekhans M, Baertsch R, Haussler D.
<A HREF="http://bioinformatics.oxfordjournals.org/cgi/content/full/24/5/637"
TARGET=_blank>Using native and syntenically mapped cDNA alignments to improve 
de novo gene finding</A>.
<em>Bioinformatics</em>. 2008 Mar 1;24(5):637-44.</P>
<P>
Siepel A, Diekhans M, Brejová B, Langton L, Stevens M, Comstock CL, Davis C, 
Ewing B, Oommen S, Lau C <em>et al.</em>
<A HREF="http://www.genome.org/cgi/content/abstract/gr.7128207v1"
TARGET=_blank>Targeted discovery of novel human exons by comparative 
genomics</A>.
<em>Genome Res.</em> 2007 Dec;17(12):1763-73.</P>

