<H2>Description</H2>
<P>
The UCSC Known Genes track shows known protein-coding genes based on
protein data from UniProt (SWISS-PROT and TrEMBL) and mRNA data from
the NCBI reference sequences collection (RefSeq) and GenBank. 

<H2>Display Conventions and Configuration</H2>
<P>
This track follows the display conventions for 
<A HREF="../goldenPath/help/hgTracksHelp.html#GeneDisplay">gene prediction 
tracks</A>. Black coloring indicates features that have corresponding entries 
in the Protein Databank (PDB). Blue indicates features associated with
mRNAs from NCBI RefSeq or (dark blue) items having associated proteins in 
the SWISS-PROT database. The variation in blue shading of RefSeq items
corresponds to the level of review the RefSeq record has undergone: 
predicted (light), provisional (medium), or reviewed (dark). </P>
<P>
This track contains an optional codon coloring 
feature that allows users to quickly validate and compare gene predictions.
To display codon colors, select the <em>genomic codons</em> option from the
<em>Color track by codons</em> pull-down menu. Click 
<A HREF="../goldenPath/help/hgCodonColoring.html">here</A> for more 
information about this feature. </P>

<H2>Methods</H2>
<P>
mRNA sequences from RefSeq and GenBank were aligned
against the $organism genome using blat. When a single mRNA aligned in
multiple places, only alignments having at least 98% base identity with the 
genomic sequence were kept. The mouse protein sequences were also aligned.</P>
<P>
The genomic mRNA and protein alignments were compared, and
protein-mRNA pairings were determined from alignments with significant 
overlap. CDS data for the mRNAs were obtained from RefSeq, 
GenBank and UCSC protein-mRNA blat alignments.
The initial set of UCSC Known Genes candidates, comprised of the mRNA 
alignments with their CDS data, was then analyzed by a gene 
check program to remove questionable candidates, such as those with in-frame 
stop codons, missing start or stop codons, <em>etc</em>.</P>
<P>
Among multiple mRNAs referenced by a single protein, the best mRNA
was chosen, based on a quality score derived from its length, the level
of the match between its translation and the protein sequence, and its release
date. The resulting mRNA and protein pairs were further consolidated by 
removing duplicated entries with identical CDS regions to produce the final
data set shown in this track.</P>

<H2>Credits</H2>
<P>
The UCSC Known Genes track was produced using protein data from UniProt and
mRNA data from NCBI RefSeq and GenBank.</P>

<H2>Data Use Restrictions</H2>
<P>
The UniProt entries in this annotation track are copyrighted. They
are produced through a collaboration between the Swiss Institute of
Bioinformatics and the EMBL Outstation - the European Bioinformatics
Institute. There are no restrictions on their use by non-profit
institutions as long as their content is in no way modified and this
statement is not removed. Usage by and for commercial entities requires
a license agreement (see <A HREF="http://www.isb-sib.ch/announce/"
TARGET="_blank">http://www.isb-sib.ch/announce/</A> or send an email to
<A HREF="mailto:license@isb-sib.ch">license@isb-sib.ch</A>).  </P>

<H2>References</H2>
<P>
Benson, D.A., Karsch-Mizrachi, I., Lipman, D.J., Ostell, J., and 
Wheeler, D.L. 
<A HREF="http://nar.oupjournals.org/cgi/content/full/32/suppl_1/D23" 
TARGET=_blank>GenBank: update</A>. <em>Nucleic Acids Res.</em> <B>32</B>,
D23-6 (2004).</P>
<P>
Kent, W.J.
<A HREF="http://www.genome.org/cgi/content/abstract/12/4/656" 
TARGET=_blank>BLAT - the BLAST-like alignment tool</A>.
<em>Genome Res.</em> <B>12</B>(4), 656-664 (2002).</P>

