<H2>Description</H2>
<P>
This track displays aligned clones from the Stem Cell Database (SCDb) to the 
mouse genome.  SCDb is the work of the Lemischka and Moore labs (Princeton), 
and the Stoeckert lab (Penn).  Multipotent hematopoietic stem cells are 
responsible for the production of at least eight distinct blood cell lineages.
Regulatory mechanisms that are involved in determining the fate of stem cells 
are derived using molecular components of both stem cells and cells making up 
their environment. Elucidation of the pathways and networks which are involved 
requires the identification of these molecular components. This was achieved 
using high throughput DNA sequence analysis of both human and mouse cell lines. Clones were derived from subtracted cDNA libraries from hematopoietic stem 
cells (HSC) and an HSC supportive stromal cell line for each species.  

<H2>Methods</H2>
<P>
Briefly, for the investigation for HSCs, cDNA libraries from a highly purified
HSC population and from a stem cell-depleted population were utilized 
(see  
<A HREF="http://stemcell.princeton.edu/v1/mat_met.html" TARGET=_blank>Methods
</A> and Phillips <em>et al</em>., 2000). The HSC library was converted into 
single stranded molecules and DNA from the stem cell-depleted library was used 
as a template to synthesize biotinylated RNA. Two successive rounds of 
hybridization followed by the synthesis of the second DNA strand and several 
rounds of purification of the resulting double stranded DNA. Sequences 
present in a stem cell-depleted cell population were subtracted, in this 
manner, from the HSC libraries in order to remove "housekeeping" gene 
products and to enrich for transcripts produced by the HSCs. Randomly 
selected sequences were sequenced using a single-pass and the sequences were 
subjected to bioinformatics analyses including queries of seven public 
databases: SwissProt, Genbank proteins and nucleotides, ESTs, murine and 
human EST contigs and SCDb itself. Additionally, ORF analysis included 
searches for protein motifs (Pfam, Prodom, SMART, Prosite, eMatrix), 
Transmembrane helices (Tmpred) and Signal Peptides (SignalP). This information 
along with that obtained from virtual Northerns, nearest neighbor analysis and 
PubMed queries was used in order to provide annotation for the sequences. 
Northern blots, semi-quantitative PCR and microarray experiments were also 
performed. A similar method was used to investigate the transcripts expressed 
in the HSC supportive stromal cell line (Hackney <em> et al</em>., 2002).   
This track was assembled from BLAT alignments of SCDb as of July 15 to the 
mouse genome (mm6/March. 2005). Default parameters were used for BLAT. </P>
<P> 
For more information about the construction SCDb clone library and to access
the freely available SCDb, see the 
<A HREF="http://stemcell.princeton.edu/" TARGET=_blank>SCDb home page</A>. 
Computational methods have been employed to extensivly analyze these
sequence sets. The preliminary results of the computational analyses and the 
raw sequences were previously available in the original Stem Cell Database 
(Phillips <em>et al</em>., 2000) and the Stromal Cell Database (StroCDB)
(Hackney <em> et al</em>., 2002). 
</P>

<H2>References</H2>
<P>
Phillips, R.L., Ernst, R.E., Brunk, B., Ivanova, N., Mahan, M.A., 
Deanehan, J.K., Moore, K.A., Overton, G.C., and Lemischka I.R.  
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=10834841&dopt=Abstract" TARGET=_blank>The Genetic Program of 
Hematopoietic Stem Cells</A>. <em>Science</em> <B>288</B> (5471), 1635-1640 
(2000).</P>
<P>
Hackney, J.A., Charbord, P., Brunk, B.P., Stoeckert, C.J., Lemischka, I.R., 
and Moore, K.A. <A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=12226475" TARGET=_blank>A molecular 
profile of a hematopoietic stem cell niche.</A><em>Proc. Natl. Acad. Sci. USA
</em> <B>99</B> (20), 13061-13066 (2002). </P>
<P>Please see the
<A HREF="http://stemcell.princeton.edu/publications.shtml" TARGET=_blank>
SCDb Publications page</A> for additional references.</P>
