<H2>Description</H2>
<P>
This track shows a measure of evolutionary conservation in $organism, 
Tetraodon, Fugu, mouse and human based on a phylogenetic hidden 
Markov model (phastCons). Multiz alignments of the following assemblies 
were used to generate this annotation: 
<UL>
<LI>$organism ($date, $db)
<LI>Tetraodon (Feb. 2004, tetNig1)
<LI>Fugu (Aug. 2002, fr1)
<LI>human (Feb. 2006, hg18)
<LI>mouse (Aug. 2005, mm7)
</UL></P>

<H2>Display Conventions and Configuration</H2> <P>
<P>
There are four different display modes for the information in this 
track: dense, squish, pack, and full. In dense mode, a grayscale 
gradient is shown in which the darker values represent higher levels of 
overall conservation as scored by phastCons. In squish, pack, and full 
modes, the phastCons scores are shown as a wiggle graph where higher 
conservation scores are shown as taller bars.
<P>
In full or pack display mode, the pairwise aliginments of human, mouse, 
Fugu and Tetraodon to zebrafish are shown. In pack mode, the pairwise 
alignments are shown in dense display mode using a grayscale density 
gradient. In full mode, the aligning blocks are shown as wiggles where 
the height of the bar represents the level of conservation found between 
zebrafish and the aligned species.
<P>
Whenever the pairwise alignments are being displayed, the gaps between 
the aligned blocks are annotated with information about what is 
happening in the aligned species in this region. (1) If in the aligned 
species there are no bases between the aligned blocks, this will be 
shown as a single line between the aligned blocks. (2) When there are 
one or more unalignable bases in the aligned species between the 
bounding aligned regions, two lines are drawn between the aligned 
blocks. (3) If there are N's in the aligned species between the bounding 
aligned regions, then the space between the alignment blocks is filled 
with a yellowish-white color. Evolutionarily, situation (1) suggests 
either a lineage-specific insertion between the aligned blocks in the 
zebrafish genome or a lineage-specific deletion between the aligned 
blocks in the aligned species, situation (2) suggests either excessive 
evolutionary distance or independent indels in the region between the 
aligned blocks in both the zebrafish and the aligning species, and 
situation (3) reflects uncertaintly in the relationship between the 
zebrafish DNA and the DNA of the aligned species because relevant 
portions of the aligned species are not yet sequenced.
<P>
In addition, the track shows places in the alignment where there is a 
discontinuity in the location (genomic context) of the aligned DNA in 
the aligned species. The genomic context can be defined as a chromosome, 
scaffold or region of the aligned species. Discontinuities are shown 
either by a vertical blue bar, representing a discontinuity that 
persists indefinitely on either side (e.g. when DNA for a very long 
region on either side of the bar comes from a different chromosome in 
the aligned species due to a large scale rearrangement), or green square 
brackets, which enclose shorter alignments consisting of DNA from one 
genomic context in the aligned species that sit inside a larger chain of 
alignments from a different genomic context.  The alignment between the 
green square brackets might represent a short mis-alignment, a 
lineage-specific insertion of a transposon in the zebrafish genome that 
is aligning to a paralogous copy somewhere else in the aligning species, 
or any number of other similar things.
<P>
When zoomed-in to the base-display level, the track shows the base 
composition of each alignment. The numbers and symbols on the Gaps line 
indicate the lengths of gaps in the zebrafish sequence at those 
alignment positions relative to the longest non-zebrafish sequence. If 
the gap size is greater than 9, a non-numeric symbol is shown: a "*" if 
the gap size is a multiple of 3; otherwise "+".
<P>
Codon translation is also available in base mode if there exists 
annotation identifying the displayed region as being in a coding 
segment. There are four different modes controlling how this annotation 
is displayed. In "No codon translation" mode, the gene annotation is not 
used so the bases are displayed without translation. In "Use default 
species for translation" mode, the annotations from the genome displayed 
in the "Default species for translation" pull-down are used to translate 
all the aligned species present in the alignment. In "Use frames for 
species if available, otherwise no translation" mode, codon translation 
will only be done on those species that are annotated as being coding 
over the aligned region. In "Use frames for species if available, 
otherwise use default species" mode, codon translation will be done on 
those species that are annotated as being coding over the aligned region 
using their particular species specific annotation while all the rest of 
the species will be translated using the default species annotation.
<P>
To view detailed information about the alignments at a specific 
position, zoom in the display to 30,000 or fewer bases, then click on 
the alignment.
<P>
The checkboxes in the track configuration section allow the exclusion of 
species from the pairwise display; however, this does not remove them 
from the conservation score display.
<P>
This track may be configured in a variety of ways to highlight different aspects
of the displayed information. Click the &quot;Graph configuration help&quot; 
link for an explanation of the configuration options.</P>

<H2>Methods</H2>
<P> 
Best-in-genome blastz pairwise alignments were multiply aligned using 
multiz, beginning with $organism-human alignments and subsequently adding 
in mouse. This was added to an alignment of $organism-Tetraodon 
aligned to Fugu. The resulting multiple alignments were then assigned 
conservation scores by phastCons.</P>
<P>
The phastCons program computes conservation scores based on a phylo-HMM, a
type of probabilistic model that describes both the process of DNA
substitution at each site in a genome and the way this process changes from
one site to the next (Felsenstein and Churchill 1996, Yang 1995, Siepel and
Haussler 2005).  PhastCons uses a two-state phylo-HMM, with a state for
conserved regions and a state for non-conserved regions.  The value plotted
at each site is the posterior probability that the corresponding alignment
column was "generated" by the conserved state of the phylo-HMM.  These
scores reflect the phylogeny (including branch lengths) of the species in
question, a continuous-time Markov model of the nucleotide substitution
process, and a tendency for conservation levels to be autocorrelated along
the genome (i.e., to be similar at adjacent sites).  The general reversible
(REV) substitution model was used.  Note that, unlike many
conservation-scoring programs, phastCons does not rely on a sliding window
of fixed size, so short highly-conserved regions and long moderately
conserved regions can both obtain high scores.  More information about
phastCons can be found in Siepel et al. (2005).</P> 
<P> 
PhastCons currently treats alignment gaps as missing data, which
sometimes has the effect of producing undesirably high conservation scores
in gappy regions of the alignment.  We are looking at several possible ways
of improving the handling of alignment gaps.</P>

<H2>Credits</H2>
<P>
This track was created at UCSC using the following programs:
<UL>
<LI>
Blastz and multiz by Minmei Hou, Scott Schwartz and Webb Miller of the 
<A HREF="http://bio.cse.psu.edu/" TARGET=_blank>Penn State Bioinformatics 
Group</A>. 
<LI>
AxtBest, axtChain, chainNet, netSyntenic, and netClass 
by Jim Kent at UCSC. 
<LI> PhastCons by Adam Siepel at UCSC. 
<LI>"Wiggle track" plotting software by Hiram Clawson at UCSC.
<LI>Gap annotation software by Brian Raney at UCSC.
</UL>
</P>

<H2>References</H2>

<H3>Phylo-HMMs and phastCons</H3>
<P>
Felsenstein J and Churchill GA (1996).  
<A HREF="http://mbe.oxfordjournals.org/cgi/content/abstract/13/1/93"
TARGET=_blank>A hidden Markov model approach to 
variation among sites in rate of evolution</A>.
<em>Mol Biol Evol</em> 13:93-104. </P>
<P>
Siepel A and Haussler D (2005). Phylogenetic hidden Markov models.
In R. Nielsen, ed., <em>Statistical Methods in Molecular Evolution</em>,
pp. 325-351, Springer, New York. </P>
<P>
Siepel, A., Bejerano, G., Pedersen, J.S., Hinrichs, A., Hou, M., Rosenbloom, 
K., Clawson, H., Spieth, J., Hillier, L.W., Richards, S., Weinstock, G.M., 
Wilson, R. K., Gibbs, R.A., Kent, W.J., Miller, W., and Haussler, D. 
<A HREF="http://www.genome.org/cgi/doi/10.1101/gr.3715005"
TARGET=_blank>Evolutionarily conserved elements in vertebrate, insect, worm, 
and yeast genomes</A>.
<em>Genome Res.</em> <B>15</B>, 1034-1050 (2005).</P>
<P>
Yang Z (1995).  
<A HREF="http://www.genetics.org/cgi/content/abstract/139/2/993" 
TARGET=_blank>A space-time process model for the evolution of DNA
sequences</A>.  <em>Genetics</em>, 139:993-1005. </P>

<H3>Chain/Net</H3>
<P>
Kent, W.J., Baertsch, R., Hinrichs, A., Miller, W., and Haussler, D.
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1"
TARGET=_blank>Evolution's cauldron: Duplication, deletion, and rearrangement
in the mouse and human genomes</A>.
<em>Proc Natl Acad Sci USA</em> <B>100</B>(20), 11484-11489 (2003).</P>

<H3>Multiz</H3>
<P>
Blanchette, M., Kent, W.J., Riemer, C., Elnitski, L., Smit, A.F.A.,
Roskin, K.M., Baertsch, R., Rosenbloom, K., Clawson, H., Green, E.D.,
Haussler, D., Miller, W.
<A HREF="http://www.genome.org/cgi/content/abstract/14/4/708"
TARGET=_blank>Aligning multiple genomic sequences with the threaded blockset
aligner</A>.
<em>Genome Res.</em> <B>14</B>(4), 708-15 (2004).</P>

<H3>Blastz</H3>
<P>
Chiaromonte, F., Yap, V.B., Miller, W.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract"
TARGET=_blank>Scoring pairwise genomic sequence alignments</A>.
<em>Pac Symp Biocomput 2002</em>, 115-26 (2002).</P>
<P>
Schwartz, S., Kent, W.J., Smit, A., Zhang, Z., Baertsch, R., Hardison, R.,
Haussler, D., and Miller, W.
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103"
TARGET=_blank>Human-Mouse Alignments with BLASTZ</A>.
<em>Genome Res.</em> <B>13</B>(1), 103-7 (2003).</P>

