<H2>Description</H2>
<P>
This track shows alignments between $organism radiation hybrid (RH) map 
sequences (consisting of ESTs and other genetic markers) and the genome. Many 
of these markers are not genetically polymorphic and therefore cannot be 
placed on genetic maps. The markers with RH map positions are useful for 
locating genes lying in genomic regions with mutations and in the 
positional cloning of zebrafish genetic mutants that exhibit interesting 
phenotypes. </P>
<P>
11514 RH map sequences (February 23, 2006 update) were aligned, consisting of:
<UL> 
<LI>5265 markers were mapped by Dr. Robert Geisler at the 
<A HREF="http://www.eb.tuebingen.mpg.de/" TARGET=_blank>Max Planck Institute 
(MPI) for Developmental Biology</A>, Germany (Max-Planck-Institut fur 
Entwicklungsbiologie or MPIEB).
<LI>6249 markers were mapped by the
<A HREF="http://zon.tchlab.org" 
TARGET=_blank>Zebrafish Genome Initiative</A> at Childrens Hospital, Boston
(CHBG)
</UL></P>
<P>

Linkage group number is equivalent to chromosome number for zebrafish. 
Position is relative to the top of the linkage group and it is the marker 
number used by the SAMapper software. Marker types can be BAC_END, EST, GENE, 
SSLP or STS. The markers are from the following sources:
<P>
<UL> 
<LI>BACENDS - end sequences of fingerprinted BAC clones.
<LI>CUSTOM - various genes sent from outside labs to CHBG for mapping
<LI>INSERTMUT - genomic sites where retroviral insertions occurred in 
insertional mutants  
<LI>MGH - 
<LI>NCBI - primer sequences for the microsatellite (CA repeats) submitted to 
NCBI by Mark Fishman's lab at Harvard University
<LI>OTHER_ZFEST - zebrafish ESTs generated elsewhere other than Washington 
University
<LI>SEQPROJ - contig or scaffold from the zebrafish genome assembly
<LI>THISSE - cDNAs from the Thisse lab with specific expression patterns found 
by in situ hybridization 
<LI>WU_ZFEST - EST assemblies from Washington University
</UL></P>
<P>

The majority of these sequences are single-read sequences 
in the range of about 40 - 1300 bp. The ESTs usually
represent fragments of transcribed genes.  

<H2>Display Conventions and Configuration</H2>
<P>
This track follows the display conventions for 
<A HREF="../goldenPath/help/hgTracksHelp.html#PSLDisplay" 
TARGET=_blank>PSL alignment tracks</A>. 
The strand information (+/-) indicates the
direction of the match between the EST and the matching
genomic sequence.  It bears no relationship to the direction
of transcription of the RNA with which it might be associated.</P>

<H2>Methods</H2>
<P>
To generate this track, $organism RH map sequences were aligned against the 
genome using blat. Only alignments with a base identity level within 0.1% of 
the best and at least 80% base identity with the genomic sequence were kept. 
The following pslReps parameters were used: <em>-nearTop=0.0001 -minAli=0.80
-minCover=0.20</em>. </P>

<H2>Credits</H2>
<P>
The $Organism RH Map track was produced by UCSC in collaboration with the 
Zebrafish Genome Initiative at Childrens Hospital, Boston, from sequence data
obtained from the individuals and institutions mentioned in the Description 
section. </P>
