<H2>Description</H2>
<P>This track shows AceView gene models constructed from
mRNA, EST and genomic evidence by Danielle and Jean Thierry-Mieg
and Vahan Simonyan using the <A HREF="http://www.ncbi.nih.gov/IEB/Research/Acembly/index.html"
TARGET=_blank>Acembly</A> program.</P> 

<H2>Methods</H2>

<P>AceView attempts to find the best alignment of each mRNA/EST against the
genome, and clusters the alignments into the least possible number of
alternatively spliced transcripts. The reconstructed transcripts are then
clustered into genes by simple transitive contact. To see the evidence that 
supports each transcript, use the &quot;<b>Outside Link:</b>&quot; on an individual 
transcripts's details page to access the NCBI AceView web site. <\P>

<P>Each AceView transcript model belongs to a gene cluster that is categorized 
into a <b>Prediction Class</b> of main or putative.</P>

<P><b>Prediction Class:</b> main 
<BR> Class of genes that includes the protein coding genes (defined
here by CDS &gt; 100 amino acids) and all genes with at least one
well-defined standard intron, i.e. an intron with a gt-ag or gc-ag
boundary, supported by at least one clone matching exactly, and with
no uncalled base, the 16 bp bordering the intron (identical to the
genome over the last and first 8 bp of the successive exons). Also
included are some genes for which the CDS is smaller than 100
aminoacid, provided they have a NCBI RefSeq sequence (NM_#) or an
OMIM, or they encode a protein with BlastP homology (&lt; 10e-3) to
a cDNA-supported nematode AceView protein.</P>

<P><b>Prediction Class:</b> putative
<BR>Class of genes that have no standard intron and do not
encode CDS of more than 100 aminoacids, yet they belong to a category
that may be useful not to disregard completely. They may be of two
types: either they are supported by more than 6 cDNA clones, or they
encode a putative protein with an interesting annotation, for example
a PFAM motif, a BlastP hit to another species than itself (&lt; 10e-3), 
a transmembrane domain or other rare and meaningful domains
identified by Psort2, or a highly probable localization in a cell
compartment (excluding cytoplasm and nucleus).</P>


<H2>Credits</H2>
<P>Thanks to <A HREF="mailto:mieg@ncbi.nlm.nih.gov">Danielle and Jean Thierry-Mieg</A> at NIH for 
providing this track.</P>


