<H2>Description</H2>
<P>
This track shows blastz alignments of the $o_organism assembly
($o_db, $o_date) to the $organism genome. The track has
an optional feature that color codes alignments to indicate the chromosomes
from which they are derived in the aligning assembly. To activate the color
feature, click the <em>on</em> button next to &quot;Color track
based on chromosome&quot; on the track description page.</P>

<H2>Methods</H2>
<P>
For blastz, 12 of 19 seeds were used and then scored using:
<PRE>
      A     C     G     T
A    91  -114   -31  -123
C  -114   100  -125   -31
G   -31  -125   100  -114
T  -123   -31  -114    91

O = 400, E = 30, K = 3000, L = 3000, M = 50
</PRE></P>
<P>
A second pass was made at reduced stringency (7mer seeds and
MSP threshold of K=2200) to attempt to fill in gaps of up to about 10K bp.
Lineage-specific repeats were abridged during this alignment.</P>

<H2>Credits</H2>
<P>
These alignments were contributed by Scott Schwartz of the
<A HREF="http://bio.cse.psu.edu/" TARGET=_blank>Penn State Bioinformatics
Group</A>. The best-in-genome filtering was done using UCSC's
<A HREF="http://www.genome.org/cgi/content/full/13/1/103"
TARGET=_blank>axtBest</A> program.</P> Mouse sequence data were provided by 
the Mouse Genome Sequencing Consortium. </P>

<H2>References</H2>
<P>
Chiaromonte, F., Yap, V.B., Miller, W.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract"
TARGET=_blank>Scoring pairwise genomic sequence alignments</A>.
<em>Pac Symp Biocomput 2002</em>, 115-26 (2002).</P>
<P>
Schwartz, S., Kent, W.J., Smit, A., Zhang, Z., Baertsch, R., Hardison, R.,
Haussler, D., and Miller, W.
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103"
TARGET=_blank>Human-mouse alignments with BLASTZ</A>.
<em>Genome Res.</em> <B>13</B>(1), 103-107 (2003).</P>

