<H2>Description</H2>
<P>This track displays <A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>BLASTZ</A> alignments of the Feb. 2003 mouse draft
assembly to the human genome filtered to display only the best alignment for any
given region of the human genome.  The track shows a graph of the score of the 
alignment.

<H2>Methods</H2>
For BLASTZ, we use 12 of 19 seeds and then score using: 
<br>
<table>
<tr><td>&nbsp;</td><td>  A</td><td>  C</td><td>  G</td><td>  T</td></tr>
<tr><td>A</td><td>&nbsp;&nbsp;91</td><td>-114</td><td>&nbsp;-31</td><td>-123</td></tr>
<tr><td>C</td><td>-114</td><td>&nbsp;100</td><td>-125</td><td>&nbsp;-31</td></tr>
<tr><td>G</td><td>&nbsp;-31</td><td>-125</td><td>&nbsp;100</td><td>-114</td></tr>
<tr><td>T</td><td>-123</td><td>&nbsp;-31</td><td>-114</td><td>&nbsp;91</td></tr>
</table>
<br>
  O = 400, E = 30, K = 3000, L = 3000, M = 50
<br><br>
We then do a second pass at reduced stringency (7mer seeds and
MSP threshold of K=2200) to attempt to fill in gaps of up to about 10Kbp.
Lineage specific repeats are abridged during this alignment.
<br>

<H2>Using the Filter</H2>
<P>The track filter can be used to turn on the chromosome color track or to 
filter the display output by chromosome.
<UL>
<LI><B>Color track</B>: To display the chromosome color track, click the <I>on</I>
button next to &quot;Color track based on chromosome&quot;. When the color track is 
activated, each of the items within the annotation track will be colored to 
show the chromosome in the mouse genome assembly from which the alignment 
originated.
<LI><B>Chromosome filter</B>: To display only alignments from a specific 
chromomsome in the mouse assembly, enter the chromosome number (in the form
<I>chrN</I>) in the text box to the right of &quot;Filter by chromosome&quot;. For example, to
display mouse alignments from mouse chromosome 6, type &quot;chr6&quot;.
</OL></P>
When you have finished configuring the filter, click the <I>Submit</I> button.

<H2>Credits</H2> These alignments are contributed by Scott Schwartz from the <A HREF="http://bio.cse.psu.edu/"
TARGET=_blank>Penn State Bioinformatics Group</A>. The best in genome filtering
is done by UCSC's axtBest program. Mouse sequence data is provided by the 
Mouse Genome Sequencing Consortium. 

<H2>References</H2>

<P><A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>Human-Mouse Alignments with BLASTZ</A>. Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison R, Haussler H, and Miller W.  <I>Genome Research</I> 2003 Jan;13(1):103-7.</P>

<P><A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" TARGET=_blank>Scoring pairwise genomic sequence alignments</A>. Chiaromonte F, Yap VB, Miller W. <I>Pac Symp Biocomput</I> 2002;:115-26.

