<H2>Description</H2>
<P>
This track displays <EM>multiz</EM> alignments of the Feb. 2003 mouse draft
assembly and the rat Jan. 2003 assembly to the human genome.

<H2>Methods</H2>
<em>Multiz</em> is a multiple alignment program that takes
blastz/axtBest alignments as input.  It uses the same scoring
matrix as blastz between pairs of sequences:  

The overall score is the sum of the score between all pairs.  

<H2>Credits</H2>
<em>Multiz</em> was created by Webb Miller from the
<A HREF="http://bio.cse.psu.edu/" TARGET=_blank>
Penn State Bioinformatics Group</A>. Jim Kent wrote
scripts to run multiz genome-wide and to display
the alignments in this browser.  Mouse sequence data 
is provided by the Mouse Genome Sequencing Consortium. 
Rat sequence data is provided by the Rat Sequencing 
Consortium.

<H2>References</H2>

<P><A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>Human-Mouse Alignments with BLASTZ</A> Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison R, Haussler H, and Miller W.  <I>Genome Research</I> 2003 Jan;13(1):103-7.</P>

<P><A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" TARGET=_blank>Scoring pairwise genomic sequence alignments</A> Chiaromonte F, Yap VB, Miller W. <I>Pac Symp Biocomput</I> 2002;:115-26

