<P>
The Dec. 2002 <em>Ciona intestinalis</em> draft assembly was obtained from 
the US DOE Joint Genome Institute (JGI) v.1.0. The assembly was generated
using a whole genome shotgun strategy constructed with the JGI assembler,
JAZZ, paired-end sequencing reads at a coverage of 8.2X. The assembly contains
116.7 million bp of nonrepetitive sequence in 2,501 scaffolds greater than 3 kb. 
60 Mbp of this has been assembled into 117 scaffolds longer than 190 Kbp, and
85% of the assembly (104.1 Mbp) is found in 905 scaffolds longer than 20 kb. 
The assembly, gene modeling and analysis were performed at the JGI.
<P>
To read more information about this assembly and an analysis of the euchromatic
regions of the genome, see Dehal et al., 
<A HREF="http://www.sciencemag.org/cgi/content/full/298/5601/2157" 
TARGET=_blank>The Draft Genome of Ciona intestinalis: Insights into Chordate and 
Vertebrate Origins</A>. <em>Science</em>. 2002 Dec 13;298(5601):2157-67.
<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/ci1/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#ciona">Downloads</A> page. 
The hg16 annotation tracks were generated by UCSC and 
collaborators worldwide. See the <A HREF="/goldenPath/credits.html#ciona_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic clone, an mRNA or EST or STS marker, or
a cytological band, a chromosomal coordinate range, or keywords from 
the GenBank description of an mRNA. The following list shows 
examples of valid position queries for the ciona genome.
See the
<A HREF="http://genome.cse.ucsc.edu/goldenPath/help/hgTracksHelp.html" TARGET=_blank>
User's Guide</A> for more information. FIXME - generate valid examples here.
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
       	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><br></TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>chr7</TD>
        <TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays all of chromosome 7</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>20p13</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region for band p13 on chr 20</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr3:1-1000000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays first million bases of chr 3, counting from p arm telomere</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>scf1:1-1000000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays first million bases of scaffold 1 of an unmapped genome assembly</TD>
    </TR>
    <TR><TD VALIGN=Top><br></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>D16S3046</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region around STS marker D16S3046 from the Genethon/Marshfield maps.
                      Includes 100,000 bases on each side as well.</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>RH18061;RH80175</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region between STS markers RH18061;RH80175.
                      Includes 100,000 bases on each side as well.</TD>
    </TR>
    <TR><TD VALIGN=Top><br></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>AA205474</TD>
       	<TD WIDTH=14></TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession AA205474 in BRCA1 cancer gene on chr 17</TD>
    </TR>
<!-- 
    <TR><TD VALIGN=Top NOWRAP>ctgchr7_ctg</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of the clone contig ctgchr7_ctg 
                       (set "Map contigs" track to "dense" and refresh to see contigs)</TD>
    </TR> 
-->
    <TR><TD VALIGN=Top NOWRAP>AC008101</TD>
    	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of clone with GenBank accession AC008101</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>AF083811</TD>
       	<TD WIDTH=14></TD>
   	<TD VALIGN=Top>Displays region of mRNA with GenBank accession number AF083811</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>PRNP</TD>
       	<TD WIDTH=14></TD>
       	<TD>Displays region of genome with HUGO identifier PRNP</TD>
    </TR>
    <TR>
	<TD VALIGN=Top NOWRAP>NM_017414</TD>
	<TD><BR></TD>
	<TD VALIGN=Top>Displays the region of genome with RefSeq identifier NM_017414</TD>
    </TR>
    <TR>
	<TD VALIGN=Top NOWRAP>NP_059110</TD> 
	<TD><BR></TD>
 	<TD VALIGN=Top NOWRAP> Displays the region of genome with protein accession number NP_059110</TD>
    </TR>
    <TR>
	<TD VALIGN=Top NOWRAP>11274</TD>
	<TD><BR></TD>
	<TD VALIGN=Top NOWRAP>Displays the region of genome with LocusLink identifier 11274</TD>
    </TR>
    <TR><TD VALIGN=Top><br></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>pseudogene mRNA</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes, but not cDNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>homeobox caudal</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>kruppel zinc finger</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>huntington</TD>
       	<TD WIDTH=14></TD>
 	<TD VALIGN=Top>Lists candidate genes associated with Huntington's disease</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>zahler</TD>
       	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Zahler</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Evans,J.E.</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author J.E.  Evans</TD>
    </TR>
    <TR><TD VALIGN=Top><br></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Evans JE</em>, internally it uses 
			the format <em>Evans,J.E.</em>.</TD>
    </TR>
</TABLE>
