
<H2>Description</H2>
This track represents the super-contigs as assembled from the fragment contigs.

<H2>Methods</H2>
The locations were extracted from the WUSTL delivered file supercontigs.agp.gz

<H2>See also</H2>
<A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Caenorhabditis+japonica"
TARGET=_blank>WUSTL</A> Genome Sequencing Center

<H3>Assembly details</H3>
<P>
The genome has been sequenced to 6.3X coverage.  See also:
<A HREF="http://genome.wustl.edu/pub/organism/Invertebrates/Caenorhabditis_japonica/assembly/Caenorhabditis_japonica-3.0.2/ASSEMBLY"
TARGET=_blank>WUSTL assembly description</A>
</P>
<P>
There are 35,273 contigs in the assembly containing 129,347,181 nucleotides.
The mean size of these contigs is 3,667 bases, with an N50 measurement of
6,365 contigs of size 5,781 bases or larger containing half of the sequence.
</P>
<P>
The smaller contigs have been assembled into 4,657 supercontigs
named Contig0 through Contig10115 containing 151,722,573 nucleotides.
The mean size of these supercontigs is 32,579 bases, with an N50 measurement
of 596 supercontigs of size 68,505 bases or larger containing half of the
sequence.
</P>
<P>
The UCSC genome browser combines the supercontigs together as an
artifical chromosome chrUn with 1,000 base gaps between these supercontigs.
This chrUn sequence is 156,378,573 bases, with the artifical 1,000 base
gaps covering 4,656,000 bases.
</P>
