<H2>Description</H2>
<P>
This track shows Human/$organism genomic
alignments using
a gap scoring system that allows longer gaps than traditional
affine gap scoring systems.  It can also tolerate gaps
in both human and $organism simultaneously.  These &quot;double-sided&quot;
gaps can be caused by local inversions and overlapping deletions
in both species. The human sequence is from the July 2003 (hg16) 
assembly.</P>
<P>
The chain track displays boxes joined together by either single or 
double lines. The boxes represent aligning regions. Single lines 
indicate gaps that are largely due to missing sequence in the non-$organism
species.  Double lines represent gaps in the chain where 
sequence is present in both species but they do not align.
In cases where there are multiple 
chains over a particular portion of the $organism genome, chains with 
single-lined gaps are often due to processed pseudogenes, while chains 
with double-lined gaps are more often due to paralogs and unprocessed 
pseudogenes. In the fuller display
modes, the individual feature names indicate the chromosome, strand, and 
location (in thousands) of the match for each matching alignment.

<H2>Methods</H2>
<P>
Transposons that have been inserted since the human/$organism
split are removed,  and the resulting abbreviated genomes are
aligned with blastz.  The transposons are then put back into the
alignments.  The resulting alignments are converted into axt format
and the resulting axts are fed into axtChain.  AxtChain organizes all 
the alignments between a single human and a single $organism chromosome,
into a group,  and makes a kd-tree out of all the gapless subsections
(blocks) of the alignments.  Next, maximally scoring chains of these
blocks are found by running a dynamic program over the kd-tree.  Chains
scoring below a threshold are discarded, and the remaining chains are
displayed here.</P>

<H2>Credits</H2>
<P>Blastz was developed at <A HREF="http://bio.cse.psu.edu">Pennsylvania State University</A> by 
Scott Schwartz, Zheng Zhang, and Webb Miller with advice from
Ross Hardison.</P>

<P>Lineage-specific repeats were identified by Arian Smit and his
program <A HREF="http://repeatmasker.org/">RepeatMasker</A>.</P>

<P>The axtChain program was developed at the University of California
at Santa Cruz by Jim Kent with advice from Webb Miller and David Haussler.</P>

<P>The browser display and database storage of the chains were made
by Robert Baertsch and Jim Kent.</P>

<H2>References</H2>
<P>
Chiaromonte F, Yap VB, Miller W (2002). 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" TARGET=_blank>Scoring pairwise genomic sequence alignments</A>. 
<em>Pac Symp Biocomput</em> 2002;:115-26.</P>
<P>
Kent WJ, Baertsch R, Hinrichs A, Miller W, and Haussler D (2003).
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1" 
TARGET=_blank>Evolution's cauldron: 
Duplication, deletion, and rearrangement in the mouse and human genomes</A>. 
<I>Proc Natl Acad Sci USA</I> 100(20):11484-11489 Sep 30 2003.
<P>
Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison R, Haussler D, and 
Miller W (2003). 
<A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>Human-Mouse 
Alignments with BLASTZ</A>. <em>Genome Res.</em> 13(1):103-7.</P>
