<P>
The October 2004 frog (<em>Xenopus tropicalis</em>) whole genome shotgun (WGS) assembly release version 3.0
was sequenced and assembled by the DOE Joint Genome Institute (JGI).
</P>

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a sequenced 
genomic clone, an mRNA or EST,
a chromosomal coordinate range, or keywords from 
the GenBank description of an mRNA. The following list provides examples 
of various types of position queries for the frog genome. See the 
<A HREF="http://genome.cse.ucsc.edu/goldenPath/help/hgTracksHelp.html" 
TARGET=_blank>User's Guide</A> for more information. 
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><BR></TD>
    </TR>
	
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>scaffold_3</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays all of scaffold 3</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>scaffold_3:1-2000000</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays first 2 million bases of scaffold_3</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>AY572227</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays location of mRNA with GenBank accession AY572227</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>BM538765</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays location of EST with GenBank accession BM538765</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>scaffold_1001.8</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays location of Genscan gene prediction with identifier scaffold_1001.8</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>pseudogene mRNA</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists transcribed pseudogenes but not cDNAs</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists zinc finger mRNAs</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>breast cancer</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs associated with breast cancer susceptibility proteins</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>evans</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Evans</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Murphy,K.E.</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs deposited by co-author K.E. Murphy</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Murphy KE</em>, internally it uses 
			the format <em>Murphy,K.E.</em>.</TD>
    </TR>
</TABLE>

<H3>Assembly Details</H3>
<P>
The assembly release version 3.0 of whole genome shotgun reads was constructed with the JGI assembler, Jazz, using paired end sequencing reads at a coverage of 7.4X. After trimming for vector and quality, 19.1 Million reads assembled into 27,064 scaffolds totaling 1.63 Gbp. Roughly half of the genome is contained in 392 scaffolds all at least 1.2 Mb in length.
For more information about this assembly, see the 
<A HREF="http://genome.jgi-psf.org/frog4x1/frog4x1.home.html" TARGET=_blank>JGI Xenopus tropicalis site</A>.
<P>
The xenTro1 sequence and annotation data can be downloaded from the 
UCSC Genome Browser 
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/xenTro1/">FTP server</A>
or 
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#frog">downloads</A> 
page. These data have <A HREF="/goldenPath/credits.html#frog_use">specific 
conditions for use</A>.
<P>
Many thanks to JGI, NHGRI, and the many
institutions who contributed to the sequencing and mapping effort for this
release. The xenTro1 annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#frog_credits">credits</A> page
for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
