<H2>Description</H2> <P>
This track shows a measure of evolutionary conservation in zebrafish, 
Tetraodon, Fugu, mouse, human and opossum based on a phylogenetic hidden 
Markov model (phastCons). Multiz alignments of the following assemblies 
were used to generate thisannotation: 
<UL>
<LI>Human (May 2004, hg17)
<LI>Mouse (Mar. 2005, mm6)
<LI>Opossum (Oct. 2004, monDom1)
<LI>Tetraodon (Feb. 2004, tetNig1)
<LI>Fugu (Aug. 2002, fr1)
<LI>Tetraodon (Feb. 2004, tetNig1)
</UL></P>

<H2>Display Conventions and Configuration</H2> <P>
<P>
In full display mode, this track shows the overall conservation score across 
all species, as well as pairwise alignments of human, mouse, opossum, 
Fugu and Tetraodon each aligned to the zebrafish genome.  The pairwise 
alignments are shown in dense display mode using a grayscale 
density gradient.  The checkboxes in the track configuration section allow
the exclusion of species from the pairwise display; however, this does not
remove them from the conservation score display.  </P>
<P>
When zoomed-in to the base-display level, the track shows the base 
composition of each alignment. The numbers and symbols on the &quot;Gaps&quot;
 line indicate the lengths of gaps in the zebrafish sequence at those 
alignment positions. If the gap size is greater than 9, the &quot;+&quot; 
symbol is displayed. To view detailed information about the 
alignments at a specific position, zoom in the display to 30,000 or fewer 
bases, then click on the alignment.  </P>
<P>
This track may be configured in a variety of ways to highlight different 
aspects of the displayed information. Click 
<A HREF="../goldenPath/help/hgWiggleTrackHelp.html" TARGET=_blank>here</A>
for an explanation of the configuration options.</P>

<H2>Methods</H2>
<P> 
Best-in-genome blastz pairwise alignments were multiply aligned using 
multiz, beginning with zebrafish-human alignments and subsequently adding 
in mouse and opossum. This was added to an alignment of zebrafish-Tetraodon 
aligned to Fugu. The resulting multiple alignments were then assigned 
conservation scores by phastCons.</P>
<P>
The phastCons program computes conservation scores based on a phylo-HMM, a
type of probabilistic model that describes both the process of DNA substitution at each site in a genome and the way this process changes from one site to the
next (Felsenstein and Churchill 1996, Yang 1995, Siepel and Haussler 2003,
Siepel and Haussler 2004). In this case, a two-state phylo-HMM was used,
with a state for conserved regions and a state for non-conserved regions.
The value plotted at each site is the posterior probability that the
corresponding alignment column was "generated" by the conserved state of
the phylo-HMM.  These scores reflect the phylogeny (including branch
lengths) of the species in question, a continuous-time Markov model of the
nucleotide substitution process, and a tendency for conservation levels to
be autocorrelated along the genome (i.e., to be similar at adjacent sites).  The
general reversible (REV) substitution model was used.
Note that, unlike many conservation-scoring programs, phastCons does not
rely on a sliding window of fixed size, so short highly-conserved regions
and long moderately conserved regions can both obtain high scores.</P>
<P> 
PhastCons currently treats alignment gaps as missing data, which
sometimes has the effect of producing undesirably high conservation scores
in gappy regions of the alignment.  We are looking at several possible ways
of improving the handling of alignment gaps.</P>

<H2>Credits</H2>
<P>
This track was created at UCSC using the following programs:
<UL>
<LI>
Blastz and multiz from Scott Schartz and Webb Miller of the 
<A HREF="http://bio.cse.psu.edu/" TARGET=_blank>Penn State Bioinformatics 
Group</A>. 
<LI>
AxtBest, axtChain, chainNet, netSyntenic, and netClass 
developed by Jim Kent at UCSC. 
<LI>Tree estimation (phyloFit) and conservation-scoring (phastCons)
software by Adam Siepel at UCSC. 
<LI>"Wiggle track" plotting software by Hiram Clawson at UCSC.
</UL>
</P>

<H2>References</H2>

<H3>Phylo-HMM</H3>
<P>
Felsenstein, J. and Churchill, G.A.  
<A HREF="http://mbe.oupjournals.org/cgi/content/abstract/13/1/93"
TARGET=_blank>A hidden Markov model approach to variation among sites in
rate of evolution</A>.
<em>Mol Biol Evol</em> <B>13</B>, 93-104 (1996). </P>
<P>
Siepel, A. and Haussler, D.
<A HREF="http://www.cse.ucsc.edu/~acs/phyhmm.pdf" 
TARGET=_blank>Combining phylogenetic and hidden Markov models in biosequence
analysis</A>.
In <em> Proceedings of the Seventh Annual International Conference on
Computational Molecular Biology (RECOMB 2003)</em>, pp. 277-286 (2003).</P>
<P>
Siepel, A. and Haussler, D. Phylogenetic hidden Markov models.
In <em>Statistical Methods in Molecular Evolution</em>, (ed. Nielsen, R.)
(in press) (Springer, 2004). </P>
<P>
Yang, Z. <A HREF="http://www.genetics.org/cgi/content/abstract/139/2/993"
TARGET=_blank>A space-time process model for the evolution of DNA
sequences</A>.  <em>Genetics</em>, <B>139</B>, 993-1005 (1995). </P>

<H3>Chain/Net</H3>
<P>
Kent, W.J., Baertsch, R., Hinrichs, A., Miller, W., and Haussler, D.
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1"
TARGET=_blank>Evolution's cauldron: Duplication, deletion, and rearrangement
in the mouse and human genomes</A>.
<em>Proc Natl Acad Sci USA</em> <B>100</B>(20), 11484-11489 (2003).</P>

<H3>Multiz</H3>
<P>
Blanchette, M., Kent, W.J., Riemer, C., Elnitski, L., Smit, A.F.A.,
Roskin, K.M., Baertsch, R., Rosenbloom, K., Clawson, H., Green, E.D.,
Haussler, D., Miller, W.
<A HREF="http://www.genome.org/cgi/content/abstract/14/4/708"
TARGET=_blank>Aligning multiple genomic sequences with the threaded blockset
aligner</A>.
<em>Genome Res.</em> <B>14</B>(4), 708-15 (2004).</P>

<H3>Blastz</H3>
<P>
Chiaromonte, F., Yap, V.B., Miller, W.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract"
TARGET=_blank>Scoring pairwise genomic sequence alignments</A>.
<em>Pac Symp Biocomput 2002</em>, 115-26 (2002).</P>
<P>
Schwartz, S., Kent, W.J., Smit, A., Zhang, Z., Baertsch, R., Hardison, R.,
Haussler, D., and Miller, W.
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103"
TARGET=_blank>Human-Mouse Alignments with BLASTZ</A>.
<em>Genome Res.</em> <B>13</B>(1), 103-7 (2003).</P>

