<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=185>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/seaurchin/"
    TARGET=_blank>
    <IMG SRC="/images/Strongylocentrotus_purpuratus.jpg" WIDTH=*** HEIGHT=*** ALT="S. purpuratus"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Strongylocentrotus purpuratus</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of Steven Murray, CSU Fullerton
      (<A HREF="http://biology.fullerton.edu/people/faculty/steve-murray/" 
      TARGET=_blank>***source</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The Sep. 2006 <em>Strongylocentrotus purpuratus</em> draft assembly 
(Baylor release 3 Spur 2.1) was produced by the Baylor College of Medicine's Human Genome Sequencing Center (BCM HGSC) and corresponds to their Spur_2.1 whole genome shotgun assembly..

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
S. purpuratus genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>chrZ</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all of chromosome Z</TD>
    </TR>
    <TR><TD VALIGN=Top>chr3:1-1000000</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first million bases of chr 3</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr3:1000000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chr 3 that spans 2000 bases, starting with position 1000000 </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>BX950328</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession
		number BX950328</TD>
    </TR>
    <TR><TD VALIGN=Top>BU335953</TD>
        <TD WIDTH=14>&nbsp;</TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession
		BU335953 on chr Z</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>pseudogene mRNA</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes, but not cDNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>homeobox caudal</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top>zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top>zhang</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Zhang</TD>
    </TR>
    <TR><TD VALIGN=Top>Taguchi,T.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author T. Taguchi</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Taguchi T</em>, internally it
	uses the format <em>Taguchi,T.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
The Spur_2.1 release was assembled from 2-fold average coverage in 
sequence reads from Bacterial Artificial Chromosomes (BAC) and 6-fold 
coverage in WGS with the HGSC Atlas-2.0 genome 
assembly system at Baylor College of Medicine. The BAC reads were 
produced by the Clone-Array Pooled Shotgun Sequencing method (CAPSS)
from BAC clones selected based on a minimal FingerPrinted 
Contigs (FPC) tiling path.  In CAPSS pooled BAC reads are 
assigned to individual BACs by deconvolution.  Each BAC assembly was 
enriched with WGS reads that overlap with the individual BAC reads. 
The mixed reads sets were assembled locally with Atlas.  
Sets of overlapping BAC clones were identified based on shared WGS reads 
and sequence overlaps.  The overlapping enriched BACs were then merged 
together to form the backbone of genome assembly.  The merged BAC 
assemblies were further scaffolded using information from mate pairs, BAC 
clone vector locations, and BAC tiling path information. Finally contigs 
from the WGS assembly Spur_0.5 were used to fill gaps in BAC assembly to 
produce Spur_2.0 release.  Extensive contamination analysis was done on 
Spur_2.0 release.  Spur_2.1 release was produced by removing contaminated 
sequences from Spur_v2.0 release. 
</P>
<P>
The total length of all contigs greater than 1kb is 804 Mbps.  When the 
gaps between contigs in scaffolds are included, the total span of the 
assembly is 907 Mbps.  The estimated size of the genome based on the 
assembly is 814 Mbps.
</P>
<P>
The draft assembly may contain errors so users should exercise caution. 
Typical errors in draft genome sequences include misassemblies of repeat 
sequences, collapses of repeat regions, and artificial duplications in 
polymorphic regions. However base accuracy in contigs is usually 
very high with most errors near the ends of contigs. 
</P>
<P>
More assembly details are in the <a href="ftp://ftp.hgsc.bcm.tmc.edu/pub/data/Spurpuratus/fasta/Spur_v2.1/README_Spur_v2.1.txt">Spur_2.1 README</a> file and on the <a href="http://www.hgsc.bcm.tmc.edu/projects/seaurchin/">BCM HGSC Sea Urchin Genome Project</a> web page.
</P>


<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/strPur2/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#s_purpuratus">Downloads</A> page. 

These data have
<A HREF="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html">specific 
conditions for use</A>. 

The S. purpuratus browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#s_purpuratus_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
