<h2>Description</h2>

<p>
This track shows alternate sequence indicated by the
<a href="http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/"
target="_blank">GRC</a> (Genome Reference Consortium) to the GRCh38 (hg38)
reference sequence in an effort to provide the best possible reference
assembly for the human genome. The GRC generates multiple
representations (haplotypes/alternate loci) for regions that are too complex
to be represented by a single path.
</p>

<h3><font color="#0000FF">Haplotypes</font></h3>

<p>
Items in the <em>Haplotype</em> track show sequence variation present in the
population for certain sections of the genome.
Data points in this track are always displayed in
<font color="#0000FF"><b>blue</b></font>.
</p>

<h2>Methods</h2>

<h3>Haplotype/Patch Tracks</h3>

<p>
The locations of the Patches and Haplotypes were established by NCBI and specified in the files at:<br>
<a href="ftp://ftp.ncbi.nih.gov/genbank/genomes/Eukaryotes/vertebrates_mammals/Homo_sapiens/GRCh38/"
target="_blank">ftp://ftp.ncbi.nih.gov/genbank/genomes/Eukaryotes/vertebrates_mammals/Homo_sapiens/GRCh38/</a>ALT_REF_LOCI_*/placed_scaffolds/alt_locus_scaf*primary.pos
</p>

<p>
and the patch release site:<br>
<a href="ftp://ftp.ncbi.nih.gov/genbank/genomes/Eukaryotes/vertebrates_mammals/Homo_sapiens/"
target="_blank">ftp://ftp.ncbi.nih.gov/genbank/genomes/Eukaryotes/vertebrates_mammals/Homo_sapiens/</a>GRCh38.p*/PATCHES/alt_scaffolds/alt_scaffold_placement.txt
</p>

<h2>Credits</h2>

<p>
Lastz (previously known as blastz) was developed at
<a href="http://www.bx.psu.edu/miller_lab/" target="_blank">Pennsylvania State University</a> by
Minmei Hou, Scott Schwartz, Zheng Zhang, and Webb Miller with advice from Ross Hardison.
</p>

<p>
Lineage-specific repeats were identified by Arian Smit and his
<a href="http://www.repeatmasker.org" target="_blank">RepeatMasker</a> program.
</p>

<p>
The axtChain program was developed at the University of California Santa Cruz
by Jim Kent with advice from Webb Miller and David Haussler.
</p>

<p>
The browser display and database storage of the chains and nets were created
by Robert Baertsch and Jim Kent.
</p>

<p>
The chainNet, netSyntenic, and netClass programs were developed at the
University of California Santa Cruz by Jim Kent.
</p>

<h2>References</h2>

<p>
Chiaromonte F, Yap VB, Miller W.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/11928468" target="_blank">
Scoring pairwise genomic sequence alignments</a>.
<em>Pac Symp Biocomput</em>. 2002:115-26.
</p>
<p>
Kent WJ, Baertsch R, Hinrichs A, Miller W, Haussler D.
<a href="http://www.pnas.org/content/100/20/11484.abstract" target="_blank">
Evolution's cauldron: duplication, deletion, and rearrangement in the mouse and human genomes</a>.
<em>Proc Natl Acad Sci U S A</em>. 2003 Sep 30;100(20):11484-9.
</p>
<p>
Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison RC, Haussler D, Miller W.
<a href="http://genome.cshlp.org/content/13/1/103.abstract" target="_blank">
Human-mouse alignments with BLASTZ</a>.
<em>Genome Res</em>. 2003 Jan;13(1):103-7.
</p>

