<H2>Description</H2>
<P>This track shows the Zebrafish Zv7 (July 2007) assembly 
provided by 
<A HREF="http://www.sanger.ac.uk/Projects/D_rerio/Zv7_assembly_information.shtml" 
TARGET=_blank>The Wellcome Trust Sanger Institute</A>. 
The assembly contains scaffolds (supercontigs) totaling 1.44 billion base 
pairs. A tiling path of sequenced clones is provided by the Fingerprinted 
Contig (FPC) map (data freeze April 11, 2007). The Whole Genome Shotgun (WGS) 
assembly (coverage 5.5x) that was used to fill the gaps in the tiling path 
is comprised of reads only from a library created from a single Tuebingen 
doubled haploid zebrafish. 1,277,075,233 bp (89%) of the total 
1,440,582,308 bp of the assembly were placed on chromosomes 1-25 (including 
estimated gap sizes and 100 bp gaps betweeen scaffolds). The assembly 
consists of 5036 fragments. 
</P>
<P>In dense mode, this track depicts the path through the draft and
finished clones (aka the golden path) used to create the assembled sequence.
Clone boundaries are distinguished by the use of alternating gold and brown
coloration. Where gaps exist in the path, spaces are shown between the gold 
and brown blocks.  If the relative order and orientation of the scaffolds
between the two blocks is known, a line is drawn to bridge the
blocks.</P>
<P>
The Genome Browser depicts the zebrafish genome as 25 chromosomes consisting of
1.02 Gb of sequences from 7,823 sequenced clones (7,139 finished and 684 
unfinished) that are tied to an fingerprinted contig (FPC) map with contigs 
from whole genome shotgun (WGS) filling in the gaps. The assembly scaffolds 
including 5010 unplaced scaffolds (supercontigs) fall into 2 groups:
<UL>
<LI><B>Zv7_NAXX</B> - WGS contigs that could not be related to any FPC conitg 
and could not be be placed on chromosomes (4,844 unmapped scaffolds)
<LI><B>Zv7_scaffoldXXXX</B> - sequences based on FPC contigs or are linked to 
chromosomes via a marker (166 unmapped scaffolds)
</UL>
The virtual chromosomes contain 100 bp 
scaffold gaps that are shown in the Gap track annotation. All the unplaced
scaffolds from chrFinished (in assembly Zv3, danRer1) have been mapped to the
chromosomes since the Zv4 assembly (danRer2).
<P>
All components within this track are of fragment type &quot;W&quot;
(WGS contig) except for chrM which is type &quot;F&quot; (Finished).
</P>
<H2>Methods</H2>
<P>
This assembly was constructed using the assembler, Phusion, to cluster reads. 
Phrap was then used for cluster assembly and consensus generation. Supercontigs or scaffolds were created from contigs joined together based on read-pair
information where the sizes of gaps were estimated by using insert sizes of
different lengths. For the clone-based mapping and finishing, clones from 
different libraries were fingerprinted by digestion with the HindIII 
restriction enzyme.  From the information produced in this way, overlapping 
clones were linked into FPCs. Next, clones from a tiling path through the FPC 
contigs were selected for high quality sequencing. The resulting 
sequence was submitted to EMBL/GenBank. Clone sequences and WGS contigs were
integrated by considering sequence alignments, BAC end placements and 
zebrafish cDNAs and markers. Improvements to the integration algorithm allowed 
the placement of the WGS contigs that contained markers but could not be linked to the FPC contigs. In cases where markers from different chromosomes appear 
on the same contig, priority has been given to the Heat Shock Diploid Cross
Panel (HS) and the Boston MGH Cross Map (MGH). Some of these discrepancies
are due to misassemblies but there may also be inconsistencies between the 
zebrafish marker panels.  
<P>
The supercontigs tied to the FPC map create the assembly shown in this track. 
The finished clone sequence was then 
analyzed via a pipeline that included repeatmasking, <em>ab initio</em> gene 
prediction and blast searches against all protein, EST and cDNA sequences 
that were available. Results from this analysis were used to manually 
annotate clones with gene structures, descriptions and poly-A features 
(see Vega annotation tracks). At this point, the clone was submitted to 
EMBL/GenBank again and can be browsed in 
<A HREF="http://vega.sanger.ac.uk/Danio_rerio/" TARGET=_blank>Vega</A>.
</P>
<H2>Credits</H2>
<P>
The Zv7 Zebrafish assembly was produced by The Wellcome Trust Sanger Institute,
in collaboration with the Max Planck Institute for Developmental Biology, 
the Netherlands Institute for Developmental Biology (Hubrecht Laboratory),
and Yi Zhou, Anthony DiBiase and Leonard Zon from the Boston Children's 
Hospital. 
