<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html>
    <head>
        <meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
        <title>ENCODE Integrative Analysis</title>
        <link rel="stylesheet" href="/style/HGStyle.css" type="text/css">
        <link rel="stylesheet" href="/style/encodeProject.css" type="text/css">
    </head>

    <body>
        <!--#include virtual="/inc/encodeProjectSoftware.topbar.html"-->

        <div class="encodeHeader">
            <a href="http://www.genome.gov/10005107"><img src="/images/ENCODE_scaleup_logo.png" alt="ENCODE Project at NHGRI" title="ENCODE Project at NHGRI"></a>
            <span class="txt">Integrative Analysis of ENCODE Data</span>
        </div>

       <div class="wrapper">
                <div class="bar"><h4 class="title">Contents</h4></div>
                <div class="content">

                <p>
                <ul>
                  <li><a href="#intro">Introduction</a></li>
                  <li><a href="#tools">Analysis Tools and Resources</a></li>
                  <li><a href="#data">Data</a></li>
                </ul>
                </p>
                <p class="date">Updated 5 September 2012</p>

               </div><!--end content-->
       </div><!--end wrapper-->

       <div class="wrapper">
                <div class="bar"><h4 class="title"><a name="intro">Introduction</a></h4></div>
                <div class="content">
                <p>
The major goal of the ENCODE project is to identify all functional elements in the human genome sequence, where <em>functional element</em> is defined as a discrete region of the genome that encodes a reproducible biochemical signature.  ENCODE data production groups generate data and submit the data to the ENCODE Data Coordinating Center (DCC) for quality control and release.  A cross-consortium effort to perform integrated analysis of all the data types to generate useful integrative data interpretations for the community has come to completion.  The results of these analyses have been published as the <a target=_blank href="www.nature.com/encode/">ENCODE integrative analysis publication package</a>. This page describes a series of resources associated with the integrative analysis of ENCODE data.</p>
               </div><!--end content-->
       </div><!--end wrapper-->

       <div class="wrapper">
                <div class="bar"><h4 class="title"><a name="tools">Analysis Tools and Other Resources</a></h4></div>
                <div class="content">
                        <h4>ENCODE Analysis Package Publications and Website</h4>
                        <p>
                        A description of the ENCODE project, data production, data display, and data download has been published previously in the article, 
                        <a target=_blank href="http://www.plosbiology.org/article/info%3Adoi%2F10.1371%2Fjournal.pbio.1001046">A User's Guide to the Encyclopedia of DNA</a>.
                        The resources in this article, and follow-on analyses, are described in detail in an extensive package of <a target=_blank href="https://docs.google.com/document/pub?id=1Ny3oWp13YsxNX7rl8sqkQ66XLDRZy5O8W88IVs_HJFA">ENCODE integrative analysis publications</a> and on a <a target=_blank href="www.nature.com/ENCODE">Nature microsite</a>.
                        Questions regarding the package should be directed to Ian Dunham<dunham@ebi.ac.uk>.
                        </p>

                        <h4>ENCODE Analysis Virtual Machine</h4>
                        <p>
                        The <a target=_blank href="ftp://ftp.ebi.ac.uk/pub/databases/ensembl/encode/supplementary/">supplementary information</a> for the ENCODE integrative analysis Nature publication includes a set of code bundles that provide the scripts and processing steps corresponding to the methodology used in the analyses associated with the paper.  The analysis group has established an
                        <a target=_blank href="http://encodeproject.org/ENCODE/integrativeAnalysis/VM"</a>
                        ENCODE analysis virtual machine</a>
                        instance of the software, using the code bundles, where each analysis program has been tested and run. 
                        The virtual machines are freely available for interested parties to use to work with the data and tools used in the integrative analysis.</p>
                        Questions regarding the ENCODE analysis virtual machines should be directed to James Taylor<james.taylor@emory.edu> or Dannon Baker<dannon.baker@emory.edu>.
                        <h4>Software Tools</h4>
                        A page describing the 
                        <a target=_blank href="http://encodeproject.org/ENCODE/softwareTools.html">software tools</a> used in the ENCODE project is provided at ENCODE portal.</p>

                        <h4>Data Standards and Quality Metrics</h4>
                        As part of the integrative analysis, the ENCODE project has established a number of standards, viewable from 
                        the ENCODE portal at the 
                        <a target=_blank href="http://encodeproject.org/ENCODE/dataStandards.html">Data Standards</a> page.
                        A detailed description of the ChIP-seq standards is provided in the publication:
                        Landt, et al.  ChIP-seq guidelines and practices used by the ENCODE and modENCODE consortia. Genome Res. 22(9) doi: 10.1101/gr.136184.111 (2012)). 
                        Metrics that have been developed to enable comparison and standardized processing of the data are described on the portal
                        <a target=_blank href="http://encodeproject.org/ENCODE/qualityMetrics.html">Quality Metrics</a> page.</p>

                        <h4>Other ENCODE Publications</h4>
                        <p>
                        A list of publications funded by the ENCODE project and publications from groups outside the consortium using ENCODE data can be found on the
                        <a target=_blank href="http://encodeproject.org/ENCODE/pubs.html">ENCODE Publications</a> page.</p>
               </div><!--end content-->
       </div><!--end wrapper-->

       <div class="wrapper">
                <div class="bar"><h4 class="title"><a name="data">Data</a></h4></div>
                <div class="content">

                       <h4>Data Coordination Center Resources</h4>
                       <p>
                       All ENCODE production data is submitted to the DCC at the University of California, Santa Cruz (UCSC). 
                       Data is reviewed for quality and released for display as tracks in the <a target=_blank href=
                       "http://encodeproject.org/cgi-bin/hgTracks?db=hg19">UCSC Genome Browser</a> and for download at the 
                       <a target=_blank href="http://encodeproject.org/ENCODE/downloads.html"> UCSC downloads site</a>.
                       There are a number of useful tools including 
                       <a target=_blank" href="http://encodeproject.org/ENCODE/search.html">track and file search</a>
                       to assist with locating data of interest.  
                       The DCC maintains the <a target=_blank href="http://encodeproject.org">ENCODE portal</a> providing access to this data and tools for interpreting and accessing the data. </p>

                        <h4>Analysis Data Hub</h4>
                        <p>
                       The integrative analysis process has been a distributed effort by many groups.  Individual analysts downloaded and processed files from the ENCODE download site, and created intermediate and final analysis products in various forms. Now that the analysis has been completed, the analysis data is being made available for viewing through a UCSC public data hub.  The ENCODE Integrative Analysis hub is loaded from the Public Hubs tab of the
                       <a target=_blank href="http://encodeproject.org/cgi-bin/hgHubConnect">UCSC Track Data Hubs</a> page.
                       This data hub includes descriptions of ENCODE data in uniformly processed signal and element representations, as well as genome segmentations. </p>

                        <h4>Analysis FTP Site</h4>
                        <p>
                        Access to the analysis products are also provided via anonymous FTP from the 
                        <a target=_blank href="ftp://ftp.ebi.ac.uk/pub/databases/ensembl/encode/supplementary/">
                        EBI ENCODE analysis FTP server</a>.
                        This site contains an organized file structure with the ENCODE analysis datasets located in subdirectories within the <em>byDataType</em> directory.</p>

                        <h4>Other Locations</h4>
                        <p>
                        ENCODE data is also available through other genomics portals, including 
                        <a target=_blank href="http://www.ensembl.org">Ensembl</a> and the
                        <a target=_blank href="http://www.ncbi.nlm.nih.gov/geo/info/ENCODE.html">NCBI Gene Expression Omnibus (GEO)</a>.
                        Raw sequence data are deposited in the sequence read archives, 
                        <a target=_blank href="http://trace.ncbi.nlm.nih.gov/Traces/sra">SRA</a> and 
                        <a target=_blank href="http://www.ebi.ac.uk/ena/">ENA</a>.</p>

               </div><!--end content-->
       </div><!--end wrapper-->


    </body>
</html>
