<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="description" content="ENCODE Project at UCSC - Contributors">
	<META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
	<TITLE>ENCODE Project at UCSC: Contributors</TITLE>
	<LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<!-- Header Images and links -->
<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
  <TR>
    <TH COLSPAN=2 ALIGN="left">&nbsp;&nbsp;
      <A HREF="http://www.genome.gov/10506161" TARGET=_BLANK><IMG SRC="/images/encodelogo.gif" height=50 ALT="ENCODE Project at NHGRI"></A>
      &nbsp;&nbsp;
      <A HREF="/encode/"><IMG SRC="/images/encode.jpg" ALT="ENCODE Project at UCSC"></A>
    </TH>
  </TR>

<!--HOTLINKS BAR-->
    <TR><TD COLSPAN=2 HEIGHT=40>
    <TABLE BGCOLOR="000000" CELLPADDING="1" CELLSPACING="1" WIDTH="99%" HEIGHT="27">
        <TR BGCOLOR="2636D1"><TD VALIGN="middle">
        <TABLE BORDER=0 CELLSPACING=0 CELLPADDING=0 BGCOLOR="2636D1" HEIGHT="24">
    	    <TR><TD VALIGN="middle"><FONT color="#89A1DE">&nbsp;
 	    <A class="topbar" HREF="encode.html">
	       Regions        </a> &nbsp; - &nbsp;
 	    <A class="topbar" HREF="submission.html">
	       Data Submission</a> &nbsp; - &nbsp;
 	    <A class="topbar" HREF="http://genome-test.cse.ucsc.edu/encode/downloads">
	       Downloads      </a> &nbsp; - &nbsp;
	    <A class="topbar" HREF="/cgi-bin/hgLiftOver">
	       Tools          </a> &nbsp; - &nbsp;
	    <A class="topbar" HREF="terms.html">
	       Terms          </a> &nbsp; - &nbsp;
	    <A class="topbar" HREF="/goldenPath/help/hgTracksHelp.html">
	       Help           </a> &nbsp;
	    </FONT></TD>
        </TR></TABLE>
    </TD></TR></TABLE>
    </TD></TR>


    <TR HEIGHT="100%" VALIGN="top">
<!--Content Tables-->  
    <TD VALIGN="top">
<FONT size=5><B>&nbsp;&nbsp;ENCODE Contributors</B></FONT>
    <TABLE CELLPADDING=5 BORDER=0 WIDTH="100%"><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	

	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="Existing"></A><FONT SIZE="4"><B>&nbsp; 
		    Large-scale application of existing technologies
		    </B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
		    These researchers received NHGRI funds for the
		    large-scale application of existing technologies
		    to determining functional elements across all
		    ENCODE target regions.<P>
		    <UL>
		    <LI>
		    Richard Myers, Ph.D., Stanford University, Palo Alto, California 
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    <A HREF="http://www-shgc.stanford.edu/genetics/encode.html">The Stanford ENCODE Project</A>
		    </LI><P><LI>
		    George Stamatoyannopoulos, M.D., Dr. Sci., University of Washington, Seattle 
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Identification of functional DNA elements by HSqPCR
		    </LI><P><LI>
		    Michael Snyder, Ph.D., Yale University, New Haven, Connecticut
		    <BR>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Transcription and regulatory elements in ENCODE regions
		    </LI><P><LI>
		    Bing Ren, Ph.D., Ludwig Institute for Cancer Research, University of California, San Diego
		    <BR>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Mapping transcriptional regulatory elements in human DNA
		    </LI><P><LI>
		    Thomas Gingeras, Ph.D., Affymetrix, Inc., Santa Clara, California
		    <BR>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Mapping sites of transcription and regulation
		    </LI><P><LI>
		    Roderic Guigo, Ph.D., Municipal Institute of Medical Research, Barcelona, Spain
		    <BR>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Encyclopedia of genes and gene variants
		    </LI><P><LI>
		    Anindya Dutta, Ph.D., University of Virginia, Charlottesville
		    <BR>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Mapping replication elements on human chromosomes
		    </LI><P><LI>
		    Ian Dunham, Ph.D., The Wellcome Trust Sanger Institute, Hinxton, U.K.
		    <BR>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Detecting human functional sequences with microarrays
		    </LI>
		    </UL>
		    </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>

                </TD></TR></TABLE>
    <TABLE CELLPADDING=5 BORDER=0 WIDTH="100%"><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	

	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="Existing"></A><FONT SIZE="4"><B>&nbsp; 
		    New or improved technologies for finding functional elements in genomic DNA
		    </B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
		    These researchers received NHGRI funds to improve
		    or develop new technologies for finding functional
		    elements in genomic DNA, and will employ them
		    across all ENCODE target regions.<P>
		    <UL>
		    <LI> 
		    Zhiping Weng, Ph.D., Boston University
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Alternative promoter usage in tissue-specific gene expression
		    </LI><P><LI>
		    Xiang-Dong Fu, Ph.D., University of California, San Diego
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    A novel chIP-chip technology for ENCODE
		    </LI><P><LI>
		    Robert Kingston, Ph.D., Massachusetts General Hospital, Boston
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Long-range, high-resolution mapping of chromatin
		    </LI><P><LI>
		    Roland Green, Ph.D., Nimblegen Systems, Inc., Madison, Wisconsin
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Discovery of binding sites for transcription factors
		    </LI><P><LI>
		    Mark McCormick, Ph.D., Nimblegen Systems, Inc., Madison, Wisconsin
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    DNA array-based exon detection and linkage mapping
		    </LI><P><LI>
		    Job Dekker, Ph.D., University of Massachusetts Medical School, Worcester
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Structural annotation of the human genome
		    </LI>
		    </UL>
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>

                </TD></TR></TABLE>
    <TABLE CELLPADDING=5 BORDER=0 WIDTH="100%"><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	

	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="Other"></A><FONT SIZE="4"><B>&nbsp; 
		    Other Groups
		    </B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>

		    These researchers received NHGRI funds for
		    infrastructure development to support
		    ENCODE-related research.<P>

		    <UL>
		    <LI> 
		    Eric Green, M.D, Ph.D., NHGRI
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Comparative sequencing
		    </LI><P><LI>
		    David Haussler, Ph.D., UCSC Genome Bioinformatics
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Coordination of the database for all sequence-related data
		    </LI><P><LI>
		    Andreas D. Baxevanis, Ph.D., NHGRI
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Coordination of the database for other data types
		    </LI><P><LI>
		    Pieter de Jong, Ph.D., Children's Hospital Oakland Research Institute
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Clone resources needed to support the comparative sequencing
		    </LI>
		    </UL>
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
    </TD></TR></TABLE>
</TD></TR></TABLE>

</BODY></HTML>

