<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="description" content="ENCODE Project at UCSC - Contributors">
	<META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
	<TITLE>ENCODE Project at UCSC: Contributors</TITLE>
	<LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<!-- Header Images and links -->
<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
  <TR>
    <TH COLSPAN=2 ALIGN="left">&nbsp;&nbsp;
      <A HREF="http://www.genome.gov/10005107" TARGET=_BLANK><IMG SRC="/images/encodelogo.gif" height=50 ALT="ENCODE Project at NHGRI"></A>
      &nbsp;&nbsp;
      <A HREF="/encode/"><IMG SRC="/images/encode.jpg" ALT="ENCODE Project at UCSC"></A>
    </TH>
  </TR>

<!--HOTLINKS BAR-->
    <TR><TD COLSPAN=2 HEIGHT=40>
    <TABLE BGCOLOR="000000" CELLPADDING="1" CELLSPACING="1" WIDTH="99%" HEIGHT="27">
        <TR BGCOLOR="2636D1"><TD VALIGN="middle">
        <TABLE BORDER=0 CELLSPACING=0 CELLPADDING=0 BGCOLOR="2636D1" HEIGHT="24">
    	    <TR><TD VALIGN="middle"><FONT color="#89A1DE">&nbsp;
 	    <A class="topbar" HREF="encode.html">
	       Regions        </a> &nbsp; - &nbsp;
 	    <A class="topbar" HREF="submission.html">
	       Data Submission</a> &nbsp; - &nbsp;
 	    <A class="topbar" HREF="http://genome-test.cse.ucsc.edu/encode/downloads">
	       Downloads      </a> &nbsp; - &nbsp;
	    <A class="topbar" HREF="/cgi-bin/hgLiftOver">
	       Tools          </a> &nbsp; - &nbsp;
	    <A class="topbar" HREF="terms.html">
	       Terms          </a> &nbsp; - &nbsp;
	    <A class="topbar" HREF="/goldenPath/help/hgTracksHelp.html">
	       Help           </a> &nbsp;
	    </FONT></TD>
        </TR></TABLE>
    </TD></TR></TABLE>
    </TD></TR>


    <TR HEIGHT="100%" VALIGN="top">
<!--Content Tables-->  
    <TD VALIGN="top">
<FONT size=5><B>&nbsp;&nbsp;ENCODE Contributors</B></FONT>
    <TABLE CELLPADDING=5 BORDER=0 WIDTH="100%"><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	

	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="Existing"></A><FONT SIZE="4"><B>&nbsp; 
		    Large-scale application of existing technologies
		    </B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
		    These researchers received NHGRI funds for the
		    large-scale application of existing technologies
		    to determining functional elements across all
		    ENCODE target regions.<P>
		    <UL>
                    <P><LI>
		    Ian Dunham,
                    <A TARGET=_BLANK HREF="http://www.sanger.ac.uk/PostGenomics/encode/">
                    The Wellcome Trust Sanger Institute</A>, Hinxton, U.K.
		    <BR>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Detecting human functional sequences with microarrays
		    </LI>
                    <P><LI>
                    <A TARGET=_BLANK HREF= "http://mexico.bioch.virginia.edu/">
		    Anindya Dutta</A> ,  University of Virginia, Charlottesville
		    <BR>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Mapping replication elements on human chromosomes
		    </LI>
                    <P><LI>
		    Thomas Gingeras,  Affymetrix, Inc., Santa Clara, California
		    <BR>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    <A HREF="http://www.affymetrix.com/transcriptome/index.affx" target=_blank>Mapping sites of transcription and regulation</A>
		    </LI>
                    <P><LI>
		    Roderic Guigo,  
                    Municipal Institute of Medical Research, Barcelona, Spain
		    <BR>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    <A HREF="http://genome.imim.es/gencode/" target=_blank>
                    Encyclopedia of genes and gene variants</A>
		    </LI>
                    <P><LI>
		    <A HREF="http://www-shgc.stanford.edu/myerslab/" TARGET=_BLANK> 
                        Richard Myers</A>,  Stanford University, Palo Alto, California 
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    <A HREF="http://www-shgc.stanford.edu/genetics/encode.html" target=_blank>The Stanford ENCODE Project</A>
                    <P><LI>
                    <A TARGET=_BLANK HREF="http://licr-renlab.ucsd.edu">
		    Bing Ren</A> ,  
                    Ludwig Institute for Cancer Research, University of California, San Diego
		    <BR>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Mapping transcriptional regulatory elements in human DNA
		    </LI>
                    <P><LI>
                    <A TARGET=_BLANK HREF="http://www.yale.edu/snyder/">
		    Michael Snyder</A>,  Yale University, New Haven, Connecticut
		    <BR>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Transcription and regulatory elements in ENCODE regions
		    </LI>
		    <P><LI>
                    <A TARGET=_BLANK HREF="http://www.gs.washington.edu/faculty/stam.htm">
		    George Stamatoyannopoulos</A>, 
                    University of Washington, Seattle 
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Identification of functional DNA elements by HSqPCR
		    </LI>
		    </UL>
		    </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>

                </TD></TR></TABLE>
    <TABLE CELLPADDING=5 BORDER=0 WIDTH="100%"><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	

	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="Existing"></A><FONT SIZE="4"><B>&nbsp; 
		    New or improved technologies for finding functional elements in genomic DNA
		    </B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
		    These researchers received NHGRI funds to improve or develop new
                    technologies for finding functional elements in genomic DNA, 
                    and will employ them across all ENCODE target regions.<P>
		    <UL>
                    <LI>
                    <A TARGET=_BLANK HREF="http://www.umassmed.edu/pgfe/faculty/Dekker.cfm?start=0&">
		    Job Dekker</A>,  University of Massachusetts Medical School, Worcester
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Structural annotation of the human genome
                    </LI>
                    <P><LI>
                    <A TARGET=_BLANK HREF="http://cmm.ucsd.edu/fu/">
		    Xiang-Dong Fu</A>,  University of California, San Diego
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    A novel chIP-chip technology for ENCODE
		    </LI>
                    <P><LI>
		    Roland Green,  
                    <A TARGET=_BLANK HREF="http://www.nimblegen.com/technology/index.html">
                    Nimblegen Systems, Inc.</A>, Madison, Wisconsin
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Discovery of binding sites for transcription factors
		    </LI>
                    <P><LI>
		    Robert Kingston,  Massachusetts General Hospital, Boston
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Long-range, high-resolution mapping of chromatin
		    </LI>
                    <P><LI>
		    Mark McCormick,  Nimblegen Systems, Inc., Madison, Wisconsin
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    DNA array-based exon detection and linkage mapping
		    </LI>
		    <P><LI> 
                    <A TARGET=_BLANK HREF="http://zlab.bu.edu/zhiping/">
		    Zhiping Weng</A>,  Boston University
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Alternative promoter usage in tissue-specific gene expression
		    </LI>
		    </UL>
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>

	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
                    Second round of technology development grantees (September 2004):
		    <UL>
                    <P><LI>
                    <A HREF="http://www-biology.ucsd.edu/faculty/ecker.html" TARGET=_BLANK>
                    Joseph Ecker</A>,  Salk Institute,
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
                    Genome Wide Analysis of DNA Methylation
		    </LI>
                    <P><LI>
                    <A HREF="http://www.biosci.utexas.edu/mgm/people/faculty/profiles/iyer.htm" TARGET=_BLANK>
                    Vishwanath Iyer</A>,  University of Texas, Austin
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
                    STAGE and FAIRE for Regulatory Element Identification
		    </LI>
                    <P><LI>
                    Madaiah Puttaraju,  
                    <A HREF="http://www.intronn.com/pages/smart.html" TARGET=_BLANK>
                    Intronn, Inc. </A>,
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
                    Use of RNA trans-splicing to identify splice sites
		    </LI>
                    <P><LI>
                    <A HREF="http://www.gis.a-star.edu.sg/homepage/cvdetailview.jsp?invid=33&alpha=R" TARGET=_BLANK>
                    Yijun Ruan</A>,  
                    <A HREF="http://www.gis.a-star.edu.sg/homepage/default.jsp" TARGET=_BLANK>
                    Genome Institute of Singapore</A>,
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
                    Ditag technologies for complete transcriptome annotation
		    </LI>
                    <P><LI>
                    <A HREF="http://www.albany.edu/cancergenomics/faculty/stenenbaum/stenenbaum.html" TARGET=_BLANK>
                    Scott Tenenbaum</A>,  SUNY Albany,
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
                    Identifying functional regulatory elements in RNA
		    </LI>
                    <P><LI>
                    <A HREF="http://www.bu.edu/chemistry/faculty/profile/tullius.html" TARGET=_BLANK>
                    Tom Tullius</A>,  
                    Boston University,
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
                    Structure of genomic DNA at single-nucleotide resolution
		    </LI>
		    </UL>
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>

                </TD></TR></TABLE>
    <TABLE CELLPADDING=5 BORDER=0 WIDTH="100%"><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	

	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="Other"></A><FONT SIZE="4"><B>&nbsp; 
		    Other Groups
		    </B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>

		    These researchers received NHGRI funds for
		    infrastructure development to support
		    ENCODE-related research.
		    <UL>
                    <P><LI>
		    <A HREF="http://www.genome.gov/10000736" TARGET=_BLANK>
                    Andreas D. Baxevanis</A>,  NHGRI
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Coordination of the database for other data types
		    </LI>
                    <P><LI>
		    Pieter De Jong,  Children's Hospital Oakland Research Institute
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
                    <A TARGET=_BLANK HREF="http://bacpac.chori.org/">
		    Clone resources</A> to support comparative sequencing.
		    </LI>
		    <P><LI> 
                    <A TARGET=_BLANK HREF="http://www.genome.gov/10000452">
		    Eric Green</A>,  NHGRI
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Comparative sequencing
		    </LI>
                    <P><LI>
                    <A TARGET=_BLANK HREF="http://www.soe.ucsc.edu/~haussler/">
		    David Haussler</A>,  
                    <A TARGET=_BLANK HREF="http://genome.ucsc.edu/staff.html">
                    UCSC Genome Bioinformatics</A>
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
		    Coordination of the database for all sequence-related data
		    </LI>
		    </UL>
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>

	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
                    Additional ENCODE pilot phase participants:
		    <UL>
                    <P><LI>
                    Greg Crawford, Ph.D, 
                    <A HREF="http://www.genome.gov/10000351" TARGET=_BLANK>
                    Francis Collins</A>, 
                    NHGRI Intramural Program
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
                     Identification of DNAse hypersensitive sites
                     </LI>
                    <P><LI>
                    <A HREF="http://www.bmb.psu.edu/faculty/hardison/hardison.html" TARGET=_BLANK>
                    Ross Hardison</A>,  
                    <A HREF="http://www.bx.psu.edu/miller_lab/" TARGET=_BLANK>
                    Webb Miller</A>,  Pennsylvania State University
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
                    Computational tools for comparative genomics; collaboration with
                    UCSC
                     </LI>
                    <P><LI>
                    <A HREF="http://www.bcgsc.ca/about/faculty/person?pid=sjones" TARGET=_BLANK>
                    Steven Jones</A>, 
                    <A HREF="http://www.bcgsc.ca/about/" TARGET=_BLANK>
                     British Columbia Cancer Agency Genome Sciences Center </A>
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
                    Computational tools for comparative genomics,
                    visualization of genome,
                    features, and prediction of cis-regulatory elements
                     </LI>
                    <P><LI>
                    Kerstin Lindblad-Toh,  Michele Clamp, Ph.D, 
                    <A HREF="http://www.broad.mit.edu/" TARGET=_BLANK>
                    Broad Institute</A>
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
                    Computational tools for comparative genomics and prediction 
                    of cis-regulatory elements
                     </LI>
                    <P><LI>
                    <A HREF="http://www.bcgsc.ca/about/faculty/person?pid=mmarra" TARGET=_BLANK>
                    Marco Marra</A>,  
                    British Columbia Cancer Agency Genome Sciences Center
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
                    Generation of maps and contigs for ENCODE BACs
                     </LI>
                    <P><LI>
                    <A HREF="http://www.tigr.org/~salzberg/" TARGET=_BLANK>
                    Steven Salzberg</A>, 
                    <A HREF="http://www.tigr.org/" TARGET=_BLANK>
                    The Institute for Genomic Research (TIGR) </A>
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
                    Computational tools for comparative genomics
                     </LI>
                    <P><LI>
                    Greg Schuler,  
                    National Center for Biotechnology
                    Information (NCBI):
		    <BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
                    Coordination of ENCODE data with NCBI
                    </LI>
		    </UL>
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>

	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
    </TD></TR></TABLE>
</TD></TR></TABLE>

</BODY></HTML>

