<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="description" content="ENCODE Project at UCSC - Contributors">
	<META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
	<TITLE>ENCODE Project at UCSC - Contributors</TITLE>
	<LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../images/encode.jpg" ALT="ENCODE Project at UCSC"></TH></TR>

<!--HOTLINKS BAR-->
    <TR><TD COLSPAN=2 HEIGHT=40>
    <TABLE BGCOLOR="000000" CELLPADDING="1" CELLSPACING="1" WIDTH="99%" HEIGHT="27">
        <TR BGCOLOR="2636D1"><TD VALIGN="middle">
        <TABLE BORDER=0 CELLSPACING=0 CELLPADDING=0 BGCOLOR="2636D1" HEIGHT="24">
    	    <TR><TD VALIGN="middle"><FONT color="#89A1DE">&nbsp;
 	    <A class="topbar" HREF="encode.html">
	       Regions        </a> &nbsp; - &nbsp;
 	    <A class="topbar" HREF="submission.html">
	       Data Submission</a> &nbsp; - &nbsp;
 	    <A class="topbar" HREF="downloads/">
	       Downloads      </a> &nbsp; - &nbsp;
	    <A class="topbar" HREF="http://hgwdev-kate.cse.ucsc.edu/cgi-bin/hgLiftOver">
	       Tools          </a> &nbsp; - &nbsp;
	    <A class="topbar" HREF="terms.html">
	       Terms          </a> &nbsp; - &nbsp;
	    <A class="topbar" HREF="../goldenPath/help/hgTracksHelp.html">
	       Help           </a> &nbsp;
	    </FONT></TD>
        </TR></TABLE>
    </TD></TR></TABLE>
    </TD></TR>

    <TR HEIGHT="100%" VALIGN="top">
<!--Content Tables-->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0 WIDTH="100%"><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	

	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="Existing"></A><FONT SIZE="4"><B>&nbsp; 
		    Large-scale application of existing technologies
		    for determining functional elements.
		    </B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>

<LIST>
<LI> 
Richard Myers, Ph.D., Stanford University, Palo Alto, California 
<BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
The Stanford ENCODE Project
</LI><LI>
George Stamatoyannopoulos, M.D., Dr. Sci., University of Washington, Seattle 
<BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
Identification of Functional DNA Elements by HSqPCR
</LI><LI>
Michael Snyder, Ph.D., Yale University, New Haven, Connecticut
<BR>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
Transcription and Regulatory Elements in ENCODE Regions
</LI><LI>
Bing Ren, Ph.D., Ludwig Institute for Cancer Research, University of California, San Diego
<BR>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
Mapping Transcriptional Regulatory Elements in Human DNA
</LI><LI>
Thomas Gingeras, Ph.D., Affymetrix, Inc., Santa Clara, California
<BR>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
Mapping Sites of Transcription and Regulation
</LI><LI>
Roderic Guigo, Ph.D., Municipal Institute of Medical Research, Barcelona, Spain
<BR>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
Encyclopedia of Genes and Gene Variants
</LI><LI>
Anindya Dutta, Ph.D., University of Virginia, Charlottesville
<BR>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
Mapping Replication Elements on Human Chromosomes
</LI><LI>
Ian Dunham, Ph.D., The Wellcome Trust Sanger Institute, Hinxton, U.K.
<BR>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
Detecting Human Functional Sequences with Microarrays
</LI>
</LIST>
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>

                </TD></TR></TABLE>
    <TABLE CELLPADDING=5 BORDER=0 WIDTH="100%"><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	

	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="Existing"></A><FONT SIZE="4"><B>&nbsp; 
		    New or improved technologies for finding functional elements in genomic DNA.
		    </B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>

<LIST>
<LI> 
Zhiping Weng, Ph.D., Boston University
<BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
Alternative Promoter Usage in Tissue-Specific Gene Expression
</LI><LI>
Xiang-Dong Fu, Ph.D., University of California, San Diego
<BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
A Novel chIP-Chip Technology for ENCODE
</LI><LI>
Robert Kingston, Ph.D., Massachusetts General Hospital, Boston
<BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
Long-Range, High-Resolution Mapping of Chromatin
</LI><LI>
Roland Green, Ph.D., Nimblegen Systems, Inc., Madison, Wisconsin
<BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
Discovery of Binding Sites for Transcription Factors
</LI><LI>
Mark McCormick, Ph.D., Nimblegen Systems, Inc., Madison, Wisconsin
<BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
DNA Array-based Exon Detection and Linkage Mapping
</LI><LI>
Job Dekker, Ph.D., University of Massachusetts Medical School, Worcester
<BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
Structural Annotation of the Human Genome
</LI>
</LIST>
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>

                </TD></TR></TABLE>
    <TABLE CELLPADDING=5 BORDER=0 WIDTH="100%"><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	

	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="Existing"></A><FONT SIZE="4"><B>&nbsp; 
		    Other Groups.
		    </B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>

<LIST>
<LI> 
Eric Green, M.D, Ph.D., NHGRI
<BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
Comparative sequencing
</LI><LI>
David Haussler, Ph.D., University of California, Santa Cruz
<BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
Coordination of the database for all sequence-related data
</LI><LI>
Andreas D. Baxevanis, Ph.D., NHGRI
<BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
Coordinate of the database for other data types
</LI><LI>
Pieter de Jong, Ph.D., Children's Hospital Oakland Research Institute
<BR> &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp
Clone resources needed to support the comparative sequencing
</LI>
</LIST>
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
    </TD></TR></TABLE>
</TD></TR></TABLE>

</BODY></HTML>

