<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
  <HEAD>
    <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
    <META name="description" content="ENCODE Project at UCSC">
    <META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
    <TITLE>ENCODE Project at UCSC</TITLE>
    <LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
  </HEAD>

  <BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
  <A NAME="TOP"></A>

    <!-- Header Images and links -->
    <TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
      <TR>
        <TH COLSPAN=2 ALIGN="left">&nbsp;&nbsp;
          <A HREF="http://www.genome.gov/10005107" TARGET=_blank><IMG SRC="/images/encodelogo.gif" height=50 ALT="ENCODE Project at NHGRI"></A>
          &nbsp;&nbsp;
          <IMG SRC="/images/encode.jpg" ALT="ENCODE Project at UCSC">
        </TH>
      </TR>

      <TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/encode.home.topbar.html"--></TD></TR>

    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="/images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="encode.html">
		   Regions</A><HR>
		<A class="leftbar" HREF="contributors.html">
		   Contributors</a><HR>
	        <A CLASS="leftbar" HREF="submission.html">
		   Submission</A><HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/goldenPath/encode/">
		   Downloads</A><HR>
                <A CLASS="leftbar" HREF="releaseLog.html">
                    Release Log</a> <HR>
                <A CLASS="leftbar" HREF="trackStatus.html">
                    Data Status</a> <HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgLiftOver">
		   Tools</A><HR>
	        <A CLASS="leftbar" HREF="terms.html">
		   Terms of Use</a><HR>
<!--                <A CLASS="leftbar" HREF="releaseLog.html">  Release Log</A><HR>-->
		<BR><BR>&nbsp<HR>
                <A CLASS="leftbar" HREF="/cgi-bin/hgGateway">
		   Genome Browser</A><HR>
 	        <BR>
	        <IMG SRC="/images/DOT.gif" ALT="" HEIGHT=970 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Content Tables-------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	

	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; 
		    About the ENCODE Project</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                   <TR><TH HEIGHT=3></TH></TR>
                   <TR><TD WIDTH=10></TD><TD>
	             <P> This site contains information related to the 
		     <A HREF="http://www.genome.gov/10005107" 
		     TARGET=_blank>ENCODE project</A> at 
		     <A HREF="http://www.genome.gov/" 
		     TARGET=_blank>NHGRI</A>. The UCSC Genome Bioinformatics 
		     Group manages the official repository of 
		     sequence-related data for the ENCODE consortium and 
		     supports the coordination of data submission, storage, 
		     retrieval, and visualization.  
                     A summary of the status of datasets submitted to UCSC 
                     by ENCODE contributors is available on the 
                     <A TARGET=_BLANK HREF="trackStatus.html">
                     ENCODE data status page</A>.
                     <P>
                     UCSC also has a special 
		     interest in comparative genomics, and we provide 
		     resources for the ENCODE multiple sequence alignment 
		     interest group. <A HREF="http://www.ensembl.org" 
		     TARGET=_blank> Ensembl</A> also provides an 
		     <A HREF="http://www.ensembl.org/Homo_sapiens/encode.html"
		     TARGET=_blank> ENCODE resource page</A>.</P>
		     <P>
		     We'd like to thank NHGRI and the contributors of 
		     annotations and analyses to this project. The team at 
		     UCSC that develops and maintains this ENCODE site is 
		     made up of <A HREF="http://www.soe.ucsc.edu/~daryl" 
		     TARGET=_blank>Daryl Thomas</A>, 
		     <A HREF="mailto:&#107;&#97;t&#101;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;u">Kate Rosenbloom</A>, 
		     <A HREF="http://www.soe.ucsc.edu/~kent/" 
		     TARGET=_blank>Jim Kent</A>, and the 
		     <A HREF="../staff.html">UCSC Genome Bioinformatics 
		     staff</A>. <A HREF="about.html">Read more.</A></P>
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%" BORDER=0>
                  <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                      <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT>
 		    	<TD ALIGN="right"><A HREF="newsarch.html"><IMG SRC="/images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>

		      </TD><TD ALIGN="right">&nbsp;</TD></TR>
		    </TABLE>
	          </TD></TR>
		</TABLE>

		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
    		  <TR><TH HEIGHT=3></TH></TR>
    		  <TR><TD WIDTH=10></TD><TD>
<!-- start news item -->
		    <FONT FACE="courier" SIZE=3><B>24 May 2005 - New MSA sequence data freeze available</B></FONT><BR>
		    <P>
		    A new ENCODE MSA sequence data freeze is available on the 
		    UCSC downloads server. The latest freeze contains 
		    sequences from 23 vertebrates provided by NISC, Baylor, 
		    the Broad Institute (2X) and the whole genome shotgun 
		    (WGS) assemblies.  The data may be downloaded as 
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg16/encode/alignments/MAY-2005/">individual data files</A> or a
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg16/encode/alignments/MAY-2005.tgz">directory tarball</A>. 
		    Aligners are encouraged to upload alignments and related 
		    data (such as conservation scores and elements) to the 
		    UCSC ENCODE ftp site as soon as possible and then notify 
		    <A HREF="mailto:kate@soe.ucsc.edu">Kate Rosenbloom</A>. 
		    Other data, (conservation, trees, etc.) 
		    will be generated based on this dataset.</P>
		    <P>
		    The following is a summary of data updates from the
		    previous release:
		    <UL>
		    <LI>
		    The human assembly version remains at hg16 (Jul. 2003).
		    <LI>
		    The mouse assembly has been updated from mm5 (May 2004) to
		    mm6 (Mar. 2005).
		    <LI>
		    The multiple rat sequences have been replaced by a single
		    sequence: rn3 (Jun. 2003).
		    <LI>
		    The cow sequences have been updated using an assembly of 
		    BAC-based sequences provided by Baylor College of 
		    Medicine.
		    <LI>
		    Fugu (fr1), macaque (rheMac1), opossum (monDom1), 
		    Tetraodon (tetNig1), Xenopus (xenTro1), and zebrafish 
		    (danRer2) have been added.  The macaque sequence was 
		    obtained from a Baylor 
		    College of Medicine assembly that has not yet been 
		    officially released. 
		    <LI>
		    A new NISC species, rfbat 
		    (<em>Rhinolophus ferrumequinum</em>), is now available.
		    <LI>
		    Platypus data from regions 
		    where NISC has not yet generated data were provided by 
		    Jim Mullikin from a preliminary assembly of Washington 
		    University WGS reads. 
		    <LI>
		    This freeze includes low-redundancy sequence data from 
		    tenrec, elephant, armadillo, and rabbit. Only one set of 
		    sequences are provided per species/target combination; 
		    where available, NISC data is provided instead of 
		    the 2X assemblies. These data are not yet accessioned at 
		    NCBI, but were made available by the Broad Institute 
		    (rabbit, elephant, armadillo) and Jim Mullikin (tenrec).
		    <LI>
		    Orthology predictions for Fugu were made only by 
		    MAVID/Mercator; predictions for all other assemblies 
		    supported by the UCSC Genome Browser represent a union 
		    with UCSC predictions as well. Because no additional 
		    post-processing was done on the Fugu predictions, they 
		    contain a few very small contigs.
		    </UL>
		    <P>
		    Thanks to the many people, particularly Elliott Margulies
		    and Daryl Thomas, who made this release possible.</P>
		    <P>
		    <HR>

<!-- start news item -->
		    <FONT FACE="courier" SIZE=3><B>24 May 2005 - ChIP-PET/GIS annotation track (Genome Institute of Singapore) released</B></FONT><BR>
		    <P>
		    The ENCODE Genome Browser now features the ChIP-PET/GIS 
		    annotation track, which shows paired-end 
		    ditag (PET) sequences derived from 65,572 individual p53 
		    ChIP fragments of 5-fluorouracil (5FU) stimulated HCT116 
		    (colon) cells. Only PETs with a single specific mapping 
		    to the genome are included in this track.
		    <P>
		    Thanks to Atif Shahab, Chia-Lin Wei, and Yijun Ruan at the 
		    <A HREF="http://www.gis.a-star.edu.sg/homepage/default.jsp"
		    TARGET=_blank>Genome Institute of Singapore</A> for 
		    providing the p53 ChIP-PET library and sequence data. The 
		    data were mapped and analyzed by scientists from the 
		    Genome Institute of Singapore, the Bioinformatics 
		    Institute, Singapore, and Boston University. For more
		    information about this annotation, see the ChIP-PET/GIS
		    track description page.</P>
		    <HR>
<!-- end news item -->
 
<!-- start archived item section-->
		    <FONT FACE="courier" SIZE=3><B>23 May 2005 - Boston University First Exon annotation track released: </B></FONT>
		    The First Exon/BU annotation track, contributed by the
		    <A HREF="http://zlab.bu.edu/zlab/index.shtml"
		    TARGET=_blank>ZLAB</A> at Boston University, is now 
		    available in the UCSC Genome Browser. This track displays 
		    expression levels of computationally identified first 
		    exons and a constitutive exon of 20 genes in the ENCODE 
		    regions. <A HREF="newsarch.html#052305">Read more</A>.
		    <P>

		    <FONT FACE="courier" SIZE=3><B>7 May 2005 - ENCODE status page now available: </B></FONT>
		    A simple summary page has been added to the UCSC ENCODE
		    portal to show the status of datasets submitted to UCSC by
		    ENCODE contributors. The page may be found at
		    <A HREF="http://genome.ucsc.edu/ENCODE/trackStatus.html">http://genome.ucsc.edu/ENCODE/trackStatus.html</A>.
		    <A HREF="newsarch.html#050705">Read more</A>.
		    <P>

                    <FONT FACE="courier" SIZE=3><B>6 May 2005 - UCSD/Ludwig Institute histone modification ChIp/Chip data released: </B></FONT>
                    Two additional sets of ChIp/Chip data from UCSD/Ludwig
                    Institute are now available: ChIp/LI AcH3 and ChIp/LI MeH3K4.   
		    <A HREF="newsarch.html#050605">Read more</A>.
<!-- end archived item section--> 



    	          </TD><TD WIDTH=15></TD></TR>
	        </TABLE>
	      <BR></TD></TR>
	    </TABLE>
          </TD></TR>
        </TABLE>
        <BR>
	    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%">
		  <TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	          </TD></TR>
		</TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome 
		    Browser are freely available for academic, nonprofit, and 
		    personal use with the following conditions:
		    <UL>
		      <LI>
		      The general 
		      <A HREF="http://genome.ucsc.edu/#Conditions">Conditions of Use</A> 
		      for the UCSC Genome Browser apply.
		      <LI>
		      The ENCODE-specific conditions of use are
		      still being developed and will be displayed here
		      in the future.  Until then, <A
		      HREF="mailto:genome@soe.ucsc.edu">please contact
		      us</A> for the conditions on the use of this data.
		    </UL>
	          </TD><TD WIDTH=15></TD></TR>
		</TABLE>
	      <BR></TD></TR>
	    </TABLE>
          </TD></TR>
	</TABLE>
      <BR></TD></TR>
    </TABLE>
  </TD></TR>
</TABLE>
</BODY></HTML>

