<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
  <HEAD>
    <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
    <META name="description" content="ENCODE Project at UCSC">
    <META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
    <TITLE>ENCODE Project at UCSC</TITLE>
    <LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
  </HEAD>

  <BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
  <A NAME="TOP"></A>

    <!-- Header Images and links -->
    <TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
      <TR>
        <TH COLSPAN=2 ALIGN="left">&nbsp;&nbsp;
          <A HREF="http://www.genome.gov/10005107" TARGET=_blank><IMG SRC="/images/encodelogo.gif" height=50 ALT="ENCODE Project at NHGRI"></A>
          &nbsp;&nbsp;
          <IMG SRC="/images/encode.jpg" ALT="ENCODE Project at UCSC">
        </TH>
      </TR>

      <TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/encode.home.topbar.html"--></TD></TR>

    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="/images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="encode.html">
		   Regions</A><HR>
		<A class="leftbar" HREF="contributors.html">
		   Contributors</a><HR>
	        <A CLASS="leftbar" HREF="submission.html">
		   Submission</A><HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/goldenPath/encode/">
		   Downloads</A><HR>
                <A CLASS="leftbar" HREF="releaseLog.html">
                    Release Log</a> <HR>
<!-- Commented out until functional
                <A CLASS="leftbar" HREF="customTracks/index.html">
                    Custom Tracks</a> <HR>
-->
                <A CLASS="leftbar" HREF="trackStatus.html">
                    Data Status</a> <HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgLiftOver">
		   Tools</A><HR>
	        <A CLASS="leftbar" HREF="terms.html">
		   Terms of Use</a><HR>
<!--                <A CLASS="leftbar" HREF="releaseLog.html">  Release Log</A><HR>-->
		<BR><BR>&nbsp<HR>
                <A CLASS="leftbar" HREF="/cgi-bin/hgGateway">
		   Genome Browser</A><HR>
 	        <BR>
	        <IMG SRC="/images/DOT.gif" ALT="" HEIGHT=970 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Content Tables-------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	

	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; 
		    About the ENCODE Project</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                   <TR><TH HEIGHT=3></TH></TR>
                   <TR><TD WIDTH=10></TD><TD>
	             <P> This site contains information related to the 
		     <A HREF="http://www.genome.gov/10005107" 
		     TARGET=_blank>ENCODE project</A> at 
		     <A HREF="http://www.genome.gov/" 
		     TARGET=_blank>NHGRI</A>. The UCSC Genome Bioinformatics 
		     Group manages the official repository of 
		     sequence-related data for the ENCODE consortium and 
		     supports the coordination of data submission, storage, 
		     retrieval, and visualization.  
                     A summary of the status of datasets submitted to UCSC 
                     by ENCODE contributors is available on the 
                     <A TARGET=_BLANK HREF="trackStatus.html">
                     ENCODE data status page</A>.
                     <P>
                     UCSC also has a special 
		     interest in comparative genomics, and we provide 
		     resources for the ENCODE multiple sequence alignment 
		     interest group. <A HREF="http://www.ensembl.org" 
		     TARGET=_blank> Ensembl</A> also provides an 
		     <A HREF="http://www.ensembl.org/Homo_sapiens/encode.html"
		     TARGET=_blank> ENCODE resource page</A>.</P>
		     <P>
		     We'd like to thank NHGRI and the contributors of 
		     annotations and analyses to this project. The team at 
		     UCSC that develops and maintains this ENCODE site is 
		     made up of <A HREF="http://www.soe.ucsc.edu/~daryl" 
		     TARGET=_blank>Daryl Thomas</A>, 
		     <A HREF="mailto:&#107;&#97;t&#101;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;u">Kate Rosenbloom</A>, 
		     <A HREF="http://www.soe.ucsc.edu/~kent/" 
		     TARGET=_blank>Jim Kent</A>, and the 
		     <A HREF="../staff.html">UCSC Genome Bioinformatics 
		     staff</A>. <A HREF="about.html">Read more.</A></P>
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%" BORDER=0>
                  <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                      <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT>
 		    	<TD ALIGN="right"><A HREF="newsarch.html"><IMG SRC="/images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>

		      </TD><TD ALIGN="right">&nbsp;</TD></TR>
		    </TABLE>
	          </TD></TR>
		</TABLE>

		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
    		  <TR><TH HEIGHT=3></TH></TR>
    		  <TR><TD WIDTH=10></TD><TD>
<!-- start news item -->
		    <FONT FACE="courier" SIZE=3><B>12 July 2005 - EGASP genes, RIKEN CAGE tags, and Stanford RTPCR, ChIP/Chip and DNA Methylation data released</B></FONT><BR>
		    <P>
		    Several new ENCODE data sets have been released in the UCSC
		    Genome Browser. </P>
		    <P>
		    <B>EGASP Full, Partial, and Update: </B> These three 
		    gene prediction tracks are from the ENCODE Gene Annotation 
		    Assessment  Project (EGASP) Prediction Workshop 2005. 
		    The EGASP Full track shows 20 sets of gene predictions 
		    originally submitted for the workshop, covering all 44 
		    ENCODE regions.  THE EGASP Partial track shows eight sets of
		    gene precdictions that were submitted for the workshop, but 		    do not cover all ENCODE regions. The EGASP Update track 
		    shows updated versions of some of the submitted predictions.
		    Thanks to Julien Legarde at IMIM for providing the EGASP 
		    Full and EGASP Partial data sets.  Thanks to Tyler Alioto of
		    IMIM (GeneID-U12 and SGP2-U12), Deyou Zheng of Yale (Yale 
		    Pseudogenes), Sarah Djebali of Ecole Normale Superior 
		    (Exogean), Jonathan Allen of TIGR/Univ. Maryland (Jigsaw) 
		    and Mario Stanke of the University of Gottingen (Augustus) 
		    for providing their EGASP Update gene sets.</P>
		    <P>
		    <B>RIKEN CAGE Predicted Gene Start Sites: </B> This track 
		    shows the numbers of 5' cap analysis gene expression (CAGE) 
		    tags that map to the genome at specific locations. Areas in 
		    which many tags map to the same region may indicate a 
		    significant transcription start site. Thanks to Albin 
		    Sandelin at RIKEN and the FANTOM (Functional Annotation
		    of Mouse) Consortium for providing these data.</P>
		    <P>
		    <B>Stanford RTPCR: </B> This track displays absolute 
		    transcript copy numbers for 136 genes and 12 negative 
		    control intergenic regions, determined by real-time PCR in
		    HCT116 cells.</P>
		    <P>
		    <B>Stanford ChIP/Chip, Stanford ChIP/Chip Smoothed 
		    Score: </B> These tracks display raw scores and 
		    sliding-window mean scores for regions bound by the Sp1 
		    and Sp3 transcription factors in three cell lines (HCT116, 
		    Jurkat, K562), assayed by chromatin immunoprecipitation and 
		    hybridization to Nimblegen oligo tiling arrays. The 
		    Smoothed Score track is provided solely for visualization 
		    purposes -- the raw scores should be used for data analysis.
		    </P>
		    <P>
		    <B>Stanford Methylation Digest: </B> This track displays 
		    experimentally determined regions of unmethylated 
		    CpGs in eight cell lines (BE2C, CRL1690, HCT116, HT1080, 
		    HepG2, JEG3, SNU182, and U87MG). Genomic DNA was digested 
		    with a cocktail of six methyl-sensitive restriction enzymes,
		    size-selected, amplified, labeled, and hybridized to 
		    Nimblegen oligo tiling arrays. 
		    <P>
		    Thanks to Nathan Trinklein for providing the Stanford 
		    datasets.</P>
		    <P>
		    We'd like to acknowledge the work of the UCSC Genome
		    Bioinformatics team members who produced these tracks:
		    Kate Rosenbloom, Angie Hinrichs, and Hiram Clawson 
		    (development), Ann Zweig, Bob Kuhn, Galt Barber, Rachel
		    Hart, and Ali Sultan-Quarrie (quality review), and Donna 
		    Karolchik (documentation).</P>
		    <P>
		    <HR>
<!-- end news item -->
<!-- start archived item section-->
		    <FONT FACE="courier" SIZE=3><B>6 July 2005 - NHGRI DIPs, HapMap SNPs, Sanger Assoc annotation tracks released: </B></FONT>
		    The first datasets in the ENCODE Variation group are now
		    Available in the UCSC browser.
		    <A HREF="newsarch.html#070605">Read more</A>.</P>
		    <P>
		    <FONT FACE="courier" SIZE=3><B>23 June 2005 - Yale and Affymetrix ChIP/chip and transcription data release; New ENCODE track groupings: </B></FONT>
		    To aid ENCODE analysis and reduce visual clutter, we have 
		    split the Genome Browser ENCODE track group into six new 
		    groups.  We have also released a set of new Yale 
		    data and an extensive update of Affymetrix data.  
		    <A HREF="newsarch.html#062305">Read more</A>.</P>
		    <P>
<!-- end archived item section--> 

    	          </TD><TD WIDTH=15></TD></TR>
	        </TABLE>
	      <BR></TD></TR>
	    </TABLE>
          </TD></TR>
        </TABLE>
        <BR>
	    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%">
		  <TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	          </TD></TR>
		</TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome 
		    Browser are freely available for academic, nonprofit, and 
		    personal use with the following conditions:
		    <UL>
		      <LI>
		      The general 
		      <A HREF="http://genome.ucsc.edu/#Conditions">Conditions of Use</A> 
		      for the UCSC Genome Browser apply.
		      <LI>
		      The ENCODE-specific conditions of use are
		      still being developed and will be displayed here
		      in the future.  Until then, <A
		      HREF="mailto:genome@soe.ucsc.edu">please contact
		      us</A> for the conditions on the use of this data.
		    </UL>
	          </TD><TD WIDTH=15></TD></TR>
		</TABLE>
	      <BR></TD></TR>
	    </TABLE>
          </TD></TR>
	</TABLE>
      <BR></TD></TR>
    </TABLE>
  </TD></TR>
</TABLE>
</BODY></HTML>

