<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
  <HEAD>
    <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
    <META name="description" content="ENCODE Project at UCSC">
    <META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
    <TITLE>ENCODE Project at UCSC</TITLE>
    <LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
  </HEAD>

  <BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
  <A NAME="TOP"></A>

    <!-- Header Images and links -->
    <TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
      <TR>
        <TH COLSPAN=2 ALIGN="left">&nbsp;&nbsp;
          <A HREF="http://www.genome.gov/10005107" TARGET=_BLANK><IMG SRC="../images/ENCODE_scaleup_logo.png" height=50 ALT="ENCODE Project at NHGRI"></A>
          &nbsp;&nbsp;
          <A HREF="/encode/"><IMG SRC="../images/encodeDcc.jpg" ALT="ENCODE Data Coordination Center at UCSC"></A>
        </TH>
      </TR>

      <TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/encode.home.topbar.html"--></TD></TR>
      </TR>

    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="../images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
                <A CLASS="leftbar" HREF="/cgi-bin/hgTracks?db=hg18">
		   Genome Browser</A>
	        <HR>
	        <A CLASS="leftbar" HREF="cellTypes.html">
		   Cell Types</A>
	        <HR>
	        <A CLASS="leftbar" HREF="antibodies.html">
		   Antibodies</A>
	        <HR>
                <A CLASS="leftbar" HREF="releaseLog.html">
                    Release Log</A>
	        <HR>
	        <A CLASS="leftbar" HREF="downloads.html">
		   Downloads</A>
	        <HR>
		<A class="leftbar" HREF="contributors.html">
		   Contributors</A>
		<HR>
	        <A CLASS="leftbar" HREF="pubs.html">
		   Publications</A>
	        <HR>
	        <A CLASS="leftbar" HREF="terms.html">
		   Data Policy</A>
		<HR>
	        <A CLASS="leftbar" HREF="pilot.html">
		   Pilot Project</A>
		<HR>
                <A CLASS="leftbar" HREF="contacts.html">
		   Contact Us</A>
		<HR>
 	        <BR>
	        <IMG SRC="../images/DOT.gif" ALT="" HEIGHT=970 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Content Tables-------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>
                <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
                  <TR><TD>
                    <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; About the ENCODE Data Coordination Center (DCC)</B></FONT>
                  </TD></TR>
                </TABLE>
                <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
<P>
The <A HREF="http://www.genome.gov/10005107" TARGET=_blank>Encyclopedia of DNA 
Elements</A> (ENCODE) Consortium is an international collaboration of research 
groups funded by the National Human Genome Research Institute 
(<A HREF="http://www.genome.gov/" TARGET=_blank>NHGRI</A>).
The goal of the consortium is to build a comprehensive parts list of 
the functional elements of the human genome, including elements 
that act at the protein level (coding genes) and RNA level (non-coding 
genes), and regulatory elements that control the cells and circumstances 
in which a gene is active. The discovery and annotation of gene elements is
accomplished primarily by sequencing RNA from a diverse range of sources, 
comparative genomics, integrative bioinformatic methods, and human curation. 
Regulatory elements are typically investigated through DNA hypersensitivity 
assays, assays of DNA methylation, and chromatin immunoprecipitation (ChIP) of 
proteins that interact with DNA, including modified histones and transcription 
factors, followed by sequencing (ChIP-Seq). The results of ENCODE experiments,
collected in the ENCODE DCC database, are displayed on the 
UCSC Genome Browser. The data can also be downloaded from the ENCODE DCC website
in text format.
</P>
<P>
To access ENCODE data, open the 
<A HREF="../cgi-bin/hgGateway?org=human">Genome Browser</A>, select the March 
2006 assembly of the human genome, and go to 
your region of interest. ENCODE tracks will be marked with the NHGRI logo
<IMG SRC="../images/nhgri.gif" ALT="" HEIGHT=16 WIDTH=16>. 
The bulk of the ENCODE data 
can be found in the <em>Expression</em> and <em>Regulation</em> track groups, with a few 
in the <em>Mapping</em>, <em>Genes</em>, and <em>Variation</em> groups. 
Although most participating research groups have provided several tracks, 
generally only selected data from each research group are displayed by 
default. Click the hyperlinked name of a particular track to display a 
page containing configuration options and details about the methods used to 
generate the data. See the Genome Browser 
<A HREF="../goldenPath/help/hgTracksHelp.html">User's Guide</A> for further 
information about displaying tracks and navigating in the Genome Browser.
To receive notifications of ENCODE data releases and related news by email,
subscribe to the 
<A HREF="https://lists.soe.ucsc.edu/mailman/listinfo/encode-announce" 
TARGET=_blank>encode-announce</A> mailing list.</P>
<P>
Data from the earlier pilot phase of the ENCODE project are available on the 
May 2004 and March 2006 human assemblies. These datasets are generally 
available only in the initial ENCODE-targeted regions that covered 
approximately 1% of the genome. The 
<A HREF="pilot.html">ENCODE 
Pilot Project</A> web pages provide convenient browser access to these regions.
</P>
<P>
Before publishing research that uses ENCODE data, please read the
<A HREF="terms.html">data release policy</A>, which places some restrictions 
on publication use of data for nine months following the data release.
</P>
	          </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%" BORDER=0>
                  <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                      <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT>
 		    	<TD ALIGN="right"><A HREF="newsarch.html"><IMG SRC="../images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>

		      </TD><TD ALIGN="right">&nbsp;</TD></TR>
		    </TABLE>
	          </TD></TR>
		</TABLE>

		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
    		  <TR><TH HEIGHT=3></TH></TR>
    		  <TR><TD WIDTH=10></TD><TD>
		    <P></P>
<!-- start news item -->
<FONT FACE="courier" SIZE=3><B>23 September 2009 - ENCODE data releases since July 1
</B></FONT></BR>
<P>
During this period a total of 9 new ENCODE tracks were released to the UCSC public server.
Functional elements and region characterization in these tracks include:

<UL>
<LI>DNaseI hypersensitive sites and hotspots (University of Washington)
<LI>Regions of DNA methylation, by Methyl-seq and Illumina 27 array (HudsonAlpha Institute)
<LI>Bi-directional promoters (NHGRI Elnitski lab)
<LI>Transcriptome profiling by RNA-seq, including single-molecule sequencing (CSHL, Helicos)
<LI>Expression levels by exon array (University of Washington)
<LI>Regions of copy number variation (HudsonAlpha Institute)
<LI>Mappability/uniqueness of nmers (University of Massachusetts, Duke, Broad, Rosetta)
</UL>
<P> For links to these tracks, see the <EM>Release Log</EM> and <em>Downloads</em> links listed in the left menu bar.
<P>
We would like to thank the contributing ENCODE labs and the the DCC team at 
UCSC for their efforts completing these tracks.
</P>
<HR>
<!-- end news item --> 

<!-- start archived item section-->
<FONT FACE="courier" SIZE=3><B>1 July  2009 - ENCODE data releases for the period April - June 2009
</B></FONT></BR>
<P>
The following ENCODE tracks were released to the ENCODE DCC/UCSC public server during this period: 
<A HREF="newsarch.html#070109">Read more</A>.</P>
<HR>
<FONT FACE="courier" SIZE=3><B>13 March 2009 - ENCODE Data Release: Transcription Factor Binding Sites from Yale/UC-Davis/Harvard 
</B></FONT></BR>
<P>
We are pleased to announce the release to the ENCODE DCC/UCSC public server
of the 
<A TARGET=_BLANK HREF="/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeYaleChIPseq">
ENCODE Transcription Factor Binding Sites by ChIP-seq from Yale/UC-Davis/Harvard
</A>
(Yale TFBS, in the <I>Regulation</I> group).
<A HREF="newsarch.html#031309">Read more</A>.</P>

<!-- end archived item section--> 
		    <P></P>
    	          </TD><TD WIDTH=15></TD></TR>
	        </TABLE>
	      </TD></TR>
	    </TABLE>
          </TD></TR>
        </TABLE>
	<BR>
	    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
		  <TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	          </TD></TR>
		</TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome 
		    Browser are freely available for academic, nonprofit, and 
		    personal use with the following conditions:
		    <UL>
		      <LI>
                      ENCODE data is covered by the
		      <A HREF="terms.html">
                      ENCODE Consortium Data Release Policy</A>.
		      <LI>
		      The general 
		      <A HREF="http://genome.ucsc.edu/#Conditions">Conditions of Use</A> 
		      for the UCSC Genome Browser apply.
		    </UL>
	          </TD><TD WIDTH=15></TD></TR>
		</TABLE>
	      </TD></TR>
	    </TABLE>
          </TD></TR>
	</TABLE>
      <BR></TD></TR>
    </TABLE>
  </TD></TR>
</TABLE>
</BODY></HTML>

