<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
  <HEAD>
    <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
    <META name="description" content="ENCODE Project at UCSC">
    <META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
    <TITLE>ENCODE Project at UCSC</TITLE>
    <LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
  </HEAD>

  <BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
  <A NAME="TOP"></A>

    <!-- Header Images and links -->
    <TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
      <TR>
        <TH COLSPAN=2 ALIGN="left">&nbsp;&nbsp;
          <A HREF="http://www.genome.gov/10005107" TARGET=_blank><IMG SRC="/images/encodelogo.gif" height=50 ALT="ENCODE Project at NHGRI"></A>
          &nbsp;&nbsp;
          <IMG SRC="/images/encode.jpg" ALT="ENCODE Project at UCSC">
        </TH>
      </TR>

      <TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/encode.home.topbar.html"--></TD></TR>

    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="/images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="encode.html">
		   Regions</A><HR>
		<A class="leftbar" HREF="contributors.html">
		   Contributors</a><HR>
	        <A CLASS="leftbar" HREF="submission.html">
		   Submission</A><HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/goldenPath/encode/">
		   Downloads</A><HR>
                <A CLASS="leftbar" HREF="releaseLog.html">
                    Release Log</a> <HR>
                <A CLASS="leftbar" HREF="trackStatus.html">
                    Data Status</a> <HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgLiftOver">
		   Tools</A><HR>
	        <A CLASS="leftbar" HREF="terms.html">
		   Terms of Use</a><HR>
<!--                <A CLASS="leftbar" HREF="releaseLog.html">  Release Log</A><HR>-->
		<BR><BR>&nbsp<HR>
                <A CLASS="leftbar" HREF="/cgi-bin/hgGateway">
		   Genome Browser</A><HR>
 	        <BR>
	        <IMG SRC="/images/DOT.gif" ALT="" HEIGHT=970 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Content Tables-------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	

	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; 
		    About the ENCODE Project</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                   <TR><TH HEIGHT=3></TH></TR>
                   <TR><TD WIDTH=10></TD><TD>
	             <P> This site contains information related to the 
		     <A HREF="http://www.genome.gov/10005107" 
		     TARGET=_blank>ENCODE project</A> at 
		     <A HREF="http://www.genome.gov/" 
		     TARGET=_blank>NHGRI</A>. The UCSC Genome Bioinformatics 
		     Group manages the official repository of 
		     sequence-related data for the ENCODE consortium and 
		     supports the coordination of data submission, storage, 
		     retrieval, and visualization.  
                     A summary of the status of datasets submitted to UCSC 
                     by ENCODE contributors is available on the 
                     <A TARGET=_BLANK HREF="trackStatus.html">
                     ENCODE data status page</A>.
                     <P>
                     UCSC also has a special 
		     interest in comparative genomics, and we provide 
		     resources for the ENCODE multiple sequence alignment 
		     interest group. <A HREF="http://www.ensembl.org" 
		     TARGET=_blank> Ensembl</A> also provides an 
		     <A HREF="http://www.ensembl.org/Homo_sapiens/encode.html"
		     TARGET=_blank> ENCODE resource page</A>.</P>
		     <P>
		     We'd like to thank NHGRI and the contributors of 
		     annotations and analyses to this project. The team at 
		     UCSC that develops and maintains this ENCODE site is 
		     made up of <A HREF="http://www.soe.ucsc.edu/~daryl" 
		     TARGET=_blank>Daryl Thomas</A>, 
		     <A HREF="mailto:&#107;&#97;t&#101;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;u">Kate Rosenbloom</A>, 
		     <A HREF="http://www.soe.ucsc.edu/~kent/" 
		     TARGET=_blank>Jim Kent</A>, and the 
		     <A HREF="../staff.html">UCSC Genome Bioinformatics 
		     staff</A>. <A HREF="about.html">Read more.</A></P>
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%" BORDER=0>
                  <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                      <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT>
 		    	<TD ALIGN="right"><A HREF="newsarch.html"><IMG SRC="/images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>

		      </TD><TD ALIGN="right">&nbsp;</TD></TR>
		    </TABLE>
	          </TD></TR>
		</TABLE>

		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
    		  <TR><TH HEIGHT=3></TH></TR>
    		  <TR><TD WIDTH=10></TD><TD>
<!-- start news item -->
		    <FONT FACE="courier" SIZE=3><B>6 July 2005 - NHGRI DIPs, HapMap SNPs, Sanger Assoc annotation tracks released</B></FONT><BR>
		    <P>
		    The first datasets in the ENCODE Variation group are now
		    available in the UCSC browser.</P>
		    <P>
		    <B>NHGRI Deletion/Insertion Polymorphisms: </B> All human 
		    trace data from NCBI's trace archive were aligned to the 
		    genome and processed using the programs ssahaSNP and
		    ssahaDIP to detect deletion and insertion polymorphisms.
		    Thanks to Jim Mullikin at NHGRI for performing the analyses 
		    and providing these data.</P>
		    <P>
		    <B>HapMap Allele Frequencies: </B> This track shows allele 
		    frequencies for the four HapMap populations in the ten 
		    ENCODE regions that have been resequenced for variation 
		    (manually selected regions m010, m013, and m014 and 
		    randomly selected regions r112, r113, r123, r131, r213, 
		    r232, and r321).  These data were obtained from HapMap 
		    public release #16c.1. Thanks to the International HapMap 
		    Project for making this information available.</P>
		    <P>
		    <B>Sanger Genotype-Expression Association: </B> This track 
		    displays associations among gene expression data from the
		    60 unrelated Centre d'Etude du Polymorphisme Humain (CEPH) 
		    individuals of the International HapMap Project with SNPs 
		    genotyped by HapMap, 
		    in eight ENCODE regions (m010, m013, m014 and r123, r131,
		    r213, r232, and r321). The CEPH population is composed 
		    of Utah residents with ancestry from northern and western 
		    Europe.  The expression data were generated with the 
		    Illumina platform at the Wellcome Trust Sanger Institute.
		    Thanks to Manolis Dermitzakis at the Sanger Institute for 
		    providing these data.</P>
		    <P>
		    We'd also like to acknowledge the UCSC ENCODE team members 
		    who worked on these tracks: Heather Trumbower, Daryl Thomas,
		    and Angie Hinrichs (development), Galt Barber and Ali 
		    Sultan-Qurraie (quality assurance), and Donna Karolchik
		    (documentation). </P>
		    <P>
		    <HR>
<!-- end news item -->
<!-- start archived item section-->
		    <FONT FACE="courier" SIZE=3><B>23 June 2005 - Yale and Affymetrix ChIP/chip and transcription data release; New ENCODE track groupings: </B></FONT>
		    To aid ENCODE analysis and reduce visual clutter, we have 
		    split the Genome Browser ENCODE track group into six new 
		    groups.  We have also released a set of new Yale 
		    data and an extensive update of Affymetrix data.  
		    <A HREF="newsarch.html#062305">Read more</A>.</P>
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>15 June 2005 - Six data sets released in Genome Browser: </B></FONT>
		    Six more ENCODE annotation tracks have been added to the
		    Genome Browser this week.
		    <A HREF="newsarch.html#061505">Read more</A>.</P>
		    <P>

		    <FONT FACE="courier" SIZE=3><B>9 June 2005 - Sanger Histone Modification ChIP/chip track updated: </B></FONT>
		    The Sanger Institute has submitted ChIP/chip data for 
		    additional antibodies and cell lines, which we have 
		    incorporated into the existing Sanger ChIP browser track. 
		    <A HREF="newsarch.html#060905">Read more</A>. </P>

<!-- end archived item section--> 

    	          </TD><TD WIDTH=15></TD></TR>
	        </TABLE>
	      <BR></TD></TR>
	    </TABLE>
          </TD></TR>
        </TABLE>
        <BR>
	    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%">
		  <TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	          </TD></TR>
		</TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome 
		    Browser are freely available for academic, nonprofit, and 
		    personal use with the following conditions:
		    <UL>
		      <LI>
		      The general 
		      <A HREF="http://genome.ucsc.edu/#Conditions">Conditions of Use</A> 
		      for the UCSC Genome Browser apply.
		      <LI>
		      The ENCODE-specific conditions of use are
		      still being developed and will be displayed here
		      in the future.  Until then, <A
		      HREF="mailto:genome@soe.ucsc.edu">please contact
		      us</A> for the conditions on the use of this data.
		    </UL>
	          </TD><TD WIDTH=15></TD></TR>
		</TABLE>
	      <BR></TD></TR>
	    </TABLE>
          </TD></TR>
	</TABLE>
      <BR></TD></TR>
    </TABLE>
  </TD></TR>
</TABLE>
</BODY></HTML>

