<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
  <HEAD>
    <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
    <META name="description" content="ENCODE Project at UCSC">
    <META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
    <TITLE>ENCODE Project at UCSC</TITLE>
    <LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
  </HEAD>

  <BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
  <A NAME="TOP"></A>

    <!-- Header Images and links -->
    <TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
      <TR>
        <TH COLSPAN=2 ALIGN="left">&nbsp;&nbsp;
          <A HREF="http://www.genome.gov/10005107" TARGET=_blank><IMG SRC="/images/encodelogo.gif" height=50 ALT="ENCODE Project at NHGRI"></A>
          &nbsp;&nbsp;
          <IMG SRC="/images/encode.jpg" ALT="ENCODE Project at UCSC">
        </TH>
      </TR>

      <TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/encode.home.topbar.html"--></TD></TR>

    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="/images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="encode.hg17.html">
		   Regions (hg17)</A><HR>
	        <A CLASS="leftbar" HREF="encode.hg16.html">
		   Regions (hg16)</A><HR>
                <A CLASS="leftbar" HREF="trackStatus.html">
                    Data Status</a> <HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/goldenPath/encode/index.html">
		   Downloads</A><HR>
	        <A CLASS="leftbar" HREF="submission.html">
		   Submission</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgLiftOver">
		   Tools</A><HR>
                <A CLASS="leftbar" HREF="releaseLog.html">
                    Release Log</a> <HR>
<!-- Commented out until functional
                <A CLASS="leftbar" HREF="customTracks/index.html">
                    Custom Tracks</a> <HR>
-->
		<A class="leftbar" HREF="contributors.html">
		   Contributors</a><HR>
	        <A CLASS="leftbar" HREF="terms.html">
		   Terms of Use</a><HR>
<!--                <A CLASS="leftbar" HREF="releaseLog.html">  Release Log</A><HR>-->
		<BR><BR>&nbsp<HR>
                <A CLASS="leftbar" HREF="/cgi-bin/hgGateway">
		   Genome Browser</A><HR>
 	        <BR>
	        <IMG SRC="/images/DOT.gif" ALT="" HEIGHT=970 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Content Tables-------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	

	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; 
		    About the ENCODE Project</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                   <TR><TH HEIGHT=3></TH></TR>
                   <TR><TD WIDTH=10></TD><TD>
	             <P> This site contains information related to the 
		     <A HREF="http://www.genome.gov/10005107" 
		     TARGET=_blank>ENCODE project</A> at 
		     <A HREF="http://www.genome.gov/" 
		     TARGET=_blank>NHGRI</A>. The UCSC Genome Bioinformatics 
		     Group manages the official repository of 
		     sequence-related data for the ENCODE consortium and 
		     supports the coordination of data submission, storage, 
		     retrieval, and visualization.  
                     A summary of the status of datasets submitted to UCSC 
                     by ENCODE contributors is available on the 
                     <A TARGET=_BLANK HREF="trackStatus.html">
                     ENCODE data status page</A>.
                     <P>
                     UCSC also has a special 
		     interest in comparative genomics, and we provide 
		     resources for the ENCODE multiple sequence alignment 
		     interest group. <A HREF="http://www.ensembl.org" 
		     TARGET=_blank> Ensembl</A> also provides an 
		     <A HREF="http://www.ensembl.org/Homo_sapiens/encode.html"
		     TARGET=_blank> ENCODE resource page</A>.</P>
		     <P>
		     We'd like to thank NHGRI and the contributors of 
		     annotations and analyses to this project. The team at 
		     UCSC that develops and maintains this ENCODE site is 
		     made up of <A HREF="http://www.soe.ucsc.edu/~daryl" 
		     TARGET=_blank>Daryl Thomas</A>, 
		     <A HREF="mailto:&#107;&#97;t&#101;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;u">Kate Rosenbloom</A>, 
		     <A HREF="http://www.soe.ucsc.edu/~kent/" 
		     TARGET=_blank>Jim Kent</A>, and the 
		     <A HREF="../staff.html">UCSC Genome Bioinformatics 
		     staff</A>. <A HREF="about.html">Read more.</A></P>
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%" BORDER=0>
                  <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                      <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT>
 		    	<TD ALIGN="right"><A HREF="newsarch.html"><IMG SRC="/images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>

		      </TD><TD ALIGN="right">&nbsp;</TD></TR>
		    </TABLE>
	          </TD></TR>
		</TABLE>

		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
    		  <TR><TH HEIGHT=3></TH></TR>
    		  <TR><TD WIDTH=10></TD><TD>

<!-- start news item -->
		    <P>
		    <FONT FACE="courier" SIZE=3><B>21 June 2006 - New ENCODE data at UCSC</B></FONT><BR>
		    <P>
		    During the build-up to the analysis paper submissions,
		    UCSC received a flurry of ENCODE data submissions
		    (6 during the month of May).  We have recently
		    released three data sets to our public server; 
		    the remaining tracks are in progress, as indicated below.
		    <P>
		    <B>Released data:</B>
		    <P>
		    <UL>
		    <LI>
		    Sanger ChIP/Chip (HFL-1 cells) - 
		    New data added to the Sanger ChIP track show the
		    location of modified histones in HFL-1 (embryonic lung
		    fibroblast) cells.   Thanks to Rob Andrews and
		    Christopher Koch for providing this data.
		    <LI>
		    DLESS (Detection of LinEage Specific Selection) - 
		    This track shows elements predicted by the DLESS program
		    to be under lineage-specific selection, based on
		    alignments of 17 mammalian species from NHGRI/PSU 
		    TBA ENCODE alignments. DLESS is based on a phylo-HMM 
		    with states for neutrally evolving and conserved regions, 
		    and for gains and losses on each branch of the tree.  
		    Thanks to Adam Siepel of Cornell University, who 
		    developed the DLESS program, generated the data, and 
		    loaded the annotation track.
		    <LI>
		    UW/Regulome Dnase/Array - 
		    This track displays DNaseI sensitivity in GM06990 cells,
		    using the DNase/Array methodology.  Dnase/Array is a novel
 		    method for isolating DNA segments corresponding to 
		    specific DNaseI cleavage events on individual nuclear 
		    chromatin templates. Thanks to Scott Kuehn at the
 		    University of Washington for providing these data.
		    </UL></P>
		    <P>
		    Thanks to our UCSC staff who worked on these tracks: 
		    Rachel Harte (development), Ann Zweig, Kayla Smith, 
		    and Archana Thakkapallayil (quality review).
		    <P>
		    <B>In quality review:</B>
		    <UL>
		    <LI>UNC FAIRE Peaks (Update, 2091 fibroblasts)
		    </UL></P>
		    <P>
		    <B>In progress:</B>
		    <UL>
		    <LI>University of Uppsala Chip/Chip (butyrate treatment in HepG2)
		    <LI>UW/Regulome QCP Dnase HS (update and new cell lines -- 16 tissues)
		    <LI>PECAN Alignments from EBI
		    </UL></P>
		    <P>
		    The ENCODE data status page has been updated to reflect the 
		    recent activity.</P>
		    <P>
<!-- end news item --> 

<!-- start archived item section-->
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>13 January 2006 - Release of October freeze data complete</B></FONT><BR>
		    <P>
		    All datasets submitted for the October 2005 ENCODE freeze
		    have now been released.  
		    <A HREF="newsarch.html#011306">Read more</A>.</P>
		    <P>
		    <FONT FACE="courier" SIZE=3><B>17 Oct. 2005 - ENCODE hg17 Genome Browser released: </B></FONT>
		    The ENCODE browser for UCSC human genome assembly hg17 
		    (NCBI Build 35) is now available at
    		    <A TARGET=_BLANK HREF="http://genome.ucsc.edu/ENCODE/encode.hg17.html">
                    http://genome.ucsc.edu/ENCODE/encode.hg17.html</A>. To 
		    access the browser from the ENCODE portal page, click the 
    		    <A HREF="http://genome.ucsc.edu/ENCODE/encode.hg17.html"
		    TARGET=_blank>Regions (hg17)</A> link on the left sidebar 
		    menu.
		    <A HREF="newsarch.html#101705">Read more</A>.</P>
		    <P>
		    <HR>
<!-- end archived item section--> 

    	          </TD><TD WIDTH=15></TD></TR>
	        </TABLE>
	      <BR></TD></TR>
	    </TABLE>
          </TD></TR>
        </TABLE>
        <BR>
	    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%">
		  <TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	          </TD></TR>
		</TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome 
		    Browser are freely available for academic, nonprofit, and 
		    personal use with the following conditions:
		    <UL>
		      <LI>
		      The general 
		      <A HREF="http://genome.ucsc.edu/#Conditions">Conditions of Use</A> 
		      for the UCSC Genome Browser apply.
		      <LI>
		      The ENCODE-specific conditions of use are
		      still being developed and will be displayed here
		      in the future.  Until then, <A
		      HREF="mailto:genome@soe.ucsc.edu">please contact
		      us</A> for the conditions on the use of this data.
		    </UL>
	          </TD><TD WIDTH=15></TD></TR>
		</TABLE>
	      <BR></TD></TR>
	    </TABLE>
          </TD></TR>
	</TABLE>
      <BR></TD></TR>
    </TABLE>
  </TD></TR>
</TABLE>
</BODY></HTML>

