<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
  <HEAD>
    <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
    <META name="description" content="ENCODE Project at UCSC">
    <META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
    <TITLE>ENCODE Project at UCSC</TITLE>
    <LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
  </HEAD>

  <BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
  <A NAME="TOP"></A>

    <!-- Header Images and links -->
    <TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
      <TR>
        <TH COLSPAN=2 ALIGN="left">&nbsp;&nbsp;
          <A HREF="http://www.genome.gov/10005107" TARGET=_blank><IMG SRC="../images/encodelogo.gif" height=50 ALT="ENCODE Project at NHGRI"></A>
          &nbsp;&nbsp;
          <IMG SRC="../images/encode.jpg" ALT="ENCODE Project at UCSC">
        </TH>
      </TR>

      <TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/encode.home.topbar.html"--></TD></TR>

    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="../images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="encode.hg18.html">
		   Regions (hg18)</A><HR>
	        <A CLASS="leftbar" HREF="encode.hg17.html">
		   Regions (hg17)</A><HR>
                <A CLASS="leftbar" HREF="trackStatus.html">
                    Data Status</a> <HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/goldenPath/encode/index.html">
		   Downloads</A><HR>
	        <A CLASS="leftbar" HREF="submission.html">
		   Submission</A><HR>
	        <A CLASS="leftbar" HREF="../cgi-bin/hgLiftOver">
		   Tools</A><HR>
                <A CLASS="leftbar" HREF="releaseLog.html">
                    Release Log</a> <HR>
		<A class="leftbar" HREF="contributors.html">
		   Contributors</a><HR>
	        <A CLASS="leftbar" HREF="terms.html">
		   Terms of Use</a><HR>
		<BR>&nbsp<HR>
                <A CLASS="leftbar" HREF="../cgi-bin/hgGateway">
		   Genome Browser</A><HR>
                <A CLASS="leftbar" HREF="../contacts.html">
		   Contact Us</A><HR>
 	        <BR>
	        <IMG SRC="../images/DOT.gif" ALT="" HEIGHT=970 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Content Tables-------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	

	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; 
		    About the ENCODE Project</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
	            <P>
		    Welcome to the UCSC ENCODE Project portal. 
		    This site contains information related to the 
		    <A HREF="http://www.genome.gov/10005107" 
		    TARGET=_blank>ENCODE</A> project at 
		    <A HREF="http://www.genome.gov/" TARGET=_blank>NHGRI</A>. 
		    The University of California Santa Cruz 
		    (<A HREF="http://www.ucsc.edu/public/" 
		    TARGET=_blank>UCSC</A>) manages the official repository of 
		    sequence-related data for the ENCODE Consortium and 
		    supports the coordination of data submission, storage, 
		    retrieval, and visualization. 
		    <A HREF="http://www.ensembl.org/Homo_sapiens/encode.html" 
		    TARGET=_blank>Ensembl</A>, a joint project of
		    EBI and the Wellcome Trust Sanger Institute, coordinates
		    much of the analysis effort. 
		    <!--
		    <A HREF="about.html">Read more</A>. </P> 
		    -->
		    <P>
		    To view the ENCODE data on the NCBI Build 36 human assembly
		    (March 2006), click the <A HREF="encode.hg18.html">Regions 
		    (hg18)</A> link in the left sidebar menu. To view data on
                    the  previous ENCODE reference assembly (May 2004, hg17, Build 35)
                    that was used as the basis of the June 2007 Nature and Genome
                    Research publications, 
                    click the <A HREF="encode.hg17.html">Regions (hg17)</A> 
		    link.
                    <P>
                    The data from ENCODE contributors
                    that has been released by UCSC is listed on the 
                    <A HREF="releaseLog.html">release log 
		    page</A>. Primary data for ENCODE is available from the 
                    <A HREF="http://www.ncbi.nlm.nih.gov/projects/geo/info/ENCODE.html"
		    TARGET=_blank>NCBI GEO</A> and 
		    <A HREF="http://www.ebi.ac.uk/arrayexpress/"
		    TARGET=_blank>EBI ArrayExpress</A> public array data 
		    repositories.  Ensembl provides an 
		    <A HREF="http://www.ensembl.org/Homo_sapiens/encode.html"
                    TARGET=_blank>ENCODE resource page</A>, and NHGRI provides 
		    the <A HREF="http://research.nhgri.nih.gov/ENCODEdb"
                    TARGET=_blank>ENCODEdb</A> portal.</P>
		    <P>
		    We'd like to thank NHGRI and the contributors of 
		    annotations and analyses to this project. 
		    This portal is maintained by the 
		    UCSC Genome Bioinformatics Group, a cross-departmental 
		    team within the Center for Biomolecular Science and 
		    Engineering (<A HREF="http://www.cbse.ucsc.edu/" 
		    TARGET=_blank>CBSE</A>) at UCSC. The individuals who 
		    develop and maintain this website include
                    <A HREF="mailto:&#107;&#97;t&#101;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;u"
		    TARGET=_blank>Kate Rosenbloom</A>, 
                    <A HREF="mailto:&#104;&#97;rt&#101;r&#97;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;u"
		    TARGET=_blank>Rachel Harte</A>, 
                    <A HREF="mailto:t&#105;n&#103;w&#97;n&#103;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;u"
                    TARGET=_blank>Ting Wang</A>, 
                    <A HREF="http://www.soe.ucsc.edu/~kent/"
		    TARGET=_blank>Jim Kent</A>, and the 
		    <A HREF="../staff.html">UCSC Genome Bioinformatics 
		    staff</A>, with input from
                    <A HREF="http://www.soe.ucsc.edu/~daryl" 
		    TARGET=_blank>Daryl Thomas</A>.</P>
		    <P>
 		    To go to the main UCSC Genome Browser site
		    that provides access to sequence and annotation data
		    for a large collection of genome assemblies, click 
		    <A HREF="../index.html">here</A>. </P>
	          </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%" BORDER=0>
                  <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                      <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT>
 		    	<TD ALIGN="right"><A HREF="newsarch.html"><IMG SRC="../images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>

		      </TD><TD ALIGN="right">&nbsp;</TD></TR>
		    </TABLE>
	          </TD></TR>
		</TABLE>

		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
    		  <TR><TH HEIGHT=3></TH></TR>
    		  <TR><TD WIDTH=10></TD><TD>
		    <P>
<!-- start news item -->
		    <P>
		    <FONT FACE="courier" SIZE=3><B>4 Oct. 2007 - ENCODE Genome Browser Released for hg18 Assembly</B></FONT><BR>
		    <P>
		    The ENCODE browser for UCSC human genome assembly hg18 
		    (NCBI Build 36) is now available. You can access the browser
		    directly at 
		    <A HREF="encode.hg18.html">http://genome.ucsc.edu/ENCODE/encode.hg18.html</A> 
		    or by clicking the ENCODE link on the Genome Browser 
		    <A HREF="../index.html">home page</A>, then
		    selecting the Regions (hg18) item in the sidebar menu on 
		    the ENCODE portal page.</P>
		    <P>
		    The hg18 ENCODE browser includes 540 data tables in 59 
		    browser tracks that were migrated from the hg17 browser. 
		    The hg17 data coordinates were converted to hg18 coordinates
		    using the UCSC liftOver process.</P>
		    <P>
		    To improve the accessibility of the data, related ENCODE 
		    tracks have been gathered into new configuration groupings 
		    (&quot;super-tracks&quot;) that can be displayed or hidden
		    using a single visiblity control.  We have also reduced the
 		    number of track groups and have modified some of the group 
		    names for clarity. </P>
		    <P>
		    The following table
		    summarizes the data currently present in the hg18 ENCODE 
		    browser:
		    <BR><BR>
            	    <TABLE BORDER=1 BORDERCOLOR="#aaaaaa" WIDTH="100%">        
		      <TR>
		 	<TD nowrap><FONT color="#006666"><B>Group</B></FONT></TD>
                	<TD nowrap><FONT color="#006666"><B>Super-tracks</B></FONT></TD>
                	<TD nowrap><FONT color="#006666"><B>Tracks</B></FONT></TD>
                	<TD nowrap><FONT color="#006666"><B>Tables</B></FONT></TD>
            	      </TR>
  			<TR><TD>Regions and Genes</TD><TD>2</TD><TD>12</TD><TD>73</TD></TR>
  			<TR><TD>Transcription</TD><TD>2</TD><TD>11</TD><TD>67</TD></TR>
  			<TR><TD>Chromatin Immunoprecipitation</TD><TD>8</TD><TD>28</TD><TD>349</TD></TR>
  			<TR><TD>Chromatin Structure</TD><TD>2</TD><TD>8</TD><TD>51</TD></TR>
		    </TABLE></P>
		    <P>
		    Note that the Variation and Comparative Genomics data were 
		    not lifted during this migration; instead, they will be 
		    replaced by new data. The first ENCODE MSA alignment for 
		    hg18 (TBA) is currently in progress on the UCSC 
		    <A HREF="http://hgwdev.cse.ucsc.edu/ENCODE/">development 
		    server</A>.
		    <P>
		    During the migration, ENCODE tracks with whole-genome 
		    data were moved into the standard browser track 
		    groups. These include the GIS PET and UCSD/LI
		    TAG1 tracks. Future submissions of whole-genome ENCODE data
 		    will be loaded directly into the standard track groups.
		    </P>
		    <P>
		    We have expanded the ENCODE downloads site to include 
		    original data for all &quot;wiggle&quot; datasets. These 
		    data files now have filename extensions indicating the 
		    wiggle input format (fixed step, variable step, or 
		    bedGraph).</P>
		    <P>
		    You can find a description of the migration project and 
		    full details of the tables, tracks, and super-tracks 
		    available at the UCSC ENCODE portal on the UCSC
		    <A HREF="http://genomewiki.cse.ucsc.edu/index.php/ENCODE_Hg18_MigrationUCSC" TARGET=_blank>genomeWiki</A>.</P>
		    <P>
		    The UCSC team members who contributed to this effort were:
		    Andy Pohl (data conversion and database loading), Ting Wang
		    and Donna Karolchik (super-track documentation), Bob Kuhn 
		    (portal updates), Brooke Rhead, Kayla Smith, and Ann Zweig (quality 
		    assurance), and Kate Rosenbloom (super-track development, 
		    project management).
		    <P>

<!-- end news item --> 
<!-- start archived item section-->
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>13 Jun. 2007 - ENCODE Findings Published in <em>Nature</em> and <em>Genome Research</em>: </B></FONT>
		    The findings of the ENCODE project have been
		    released to the public today, the culmination of a 
		    four-year effort to catalog the biologically functional 
		    elements in 1 percent of the human genome. 
		    <A HREF="newsarch.html#061307">Read more</A>.</P>
		    <P>
		    <FONT FACE="courier" SIZE=3><B>12 Jun. 2007 - Spring 2007 ENCODE News: </B></FONT>
		    Between January and May of 2007, several new or upgraded 
		    data tracks were released by UCSC.
		    <A HREF="newsarch.html#061207">Read more</A>.</P>
<!-- end archived item section--> 
		    <P></P>
    	          </TD><TD WIDTH=15></TD></TR>
	        </TABLE>
	      </TD></TR>
	    </TABLE>
          </TD></TR>
        </TABLE>
	<BR>
	    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
		  <TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	          </TD></TR>
		</TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome 
		    Browser are freely available for academic, nonprofit, and 
		    personal use with the following conditions:
		    <UL>
		      <LI>
		      The general 
		      <A HREF="http://genome.ucsc.edu/#Conditions">Conditions of Use</A> 
		      for the UCSC Genome Browser apply.
		      <LI>
		      The ENCODE-specific conditions of use are
		      still being developed and will be displayed here
		      in the future.  Until then, <A
		      HREF="../contacts.html">please contact
		      us</A> for the conditions on the use of this data.
		    </UL>
	          </TD><TD WIDTH=15></TD></TR>
		</TABLE>
	      </TD></TR>
	    </TABLE>
          </TD></TR>
	</TABLE>
      <BR></TD></TR>
    </TABLE>
  </TD></TR>
</TABLE>
</BODY></HTML>

