<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
  <HEAD>
    <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
    <META name="description" content="ENCODE Project at UCSC">
    <META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
    <TITLE>ENCODE Project at UCSC</TITLE>
    <LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
  </HEAD>

  <BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
  <A NAME="TOP"></A>

    <!-- Header Images and links -->
    <TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
      <TR>
        <TH COLSPAN=2 ALIGN="left">&nbsp;&nbsp;
          <A HREF="http://www.genome.gov/10005107" TARGET=_BLANK><IMG SRC="../images/ENCODE_scaleup_logo.png" height=50 ALT="ENCODE Project at NHGRI"></A>
          &nbsp;&nbsp;
          <A HREF="/encode/"><IMG SRC="../images/encodeDcc.jpg" ALT="ENCODE Data Coordination Center at UCSC"></A>
        </TH>
      </TR>

      <TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/encode.home.topbar.html"--></TD></TR>
      </TR>

    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="../images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
                <A CLASS="leftbar" HREF="/cgi-bin/hgTracks?db=hg18">
		   Genome Browser</A>
	        <HR>
	        <A CLASS="leftbar" HREF="cellTypes.html">
		   Cell Types</A>
	        <HR>
	        <A CLASS="leftbar" HREF="antibodies.html">
		   Antibodies</A>
	        <HR>
                <A CLASS="leftbar" HREF="releaseLog.html">
                    Release Log</A>
	        <HR>
	        <A CLASS="leftbar" HREF="downloads.html">
		   Downloads</A>
	        <HR>
		<A class="leftbar" HREF="contributors.html">
		   Contributors</A>
		<HR>
	        <A CLASS="leftbar" HREF="pubs.html">
		   Publications</A>
	        <HR>
	        <A CLASS="leftbar" HREF="terms.html">
		   Data Policy</A>
		<HR>
	        <A CLASS="leftbar" HREF="pilot.html">
		   Pilot Project</A>
		<HR>
                <A CLASS="leftbar" HREF="contacts.html">
		   Contact Us</A>
		<HR>
 	        <BR>
	        <IMG SRC="../images/DOT.gif" ALT="" HEIGHT=970 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Content Tables-------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>
                <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
                  <TR><TD>
                    <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; About the ENCODE Data Coordination Center (DCC)</B></FONT>
                  </TD></TR>
                </TABLE>
                <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
<P>
The <A HREF="http://www.genome.gov/10005107" TARGET=_blank>Encyclopedia of DNA 
Elements</A> (ENCODE) Consortium is an international collaboration of research 
groups funded by the National Human Genome Research Institute 
(<A HREF="http://www.genome.gov/" TARGET=_blank>NHGRI</A>).
The goal of the consortium is to build a comprehensive parts list of 
the functional elements of the human genome, including elements 
that act at the protein level (coding genes) and RNA level (non-coding 
genes), and regulatory elements that control the cells and circumstances 
in which a gene is active. The discovery and annotation of gene elements is
accomplished primarily by sequencing RNA from a diverse range of sources, 
comparative genomics, integrative bioinformatic methods, and human curation. 
Regulatory elements are typically investigated through DNA hypersensitivity 
assays, assays of DNA methylation, and chromatin immunoprecipitation (ChIP) of 
proteins that interact with DNA, including modified histones and transcription 
factors, followed by sequencing (ChIP-Seq). The results of ENCODE experiments,
collected in the ENCODE DCC database, are displayed on the 
UCSC Genome Browser. The data can also be downloaded from the ENCODE DCC website
in text format.
</P>
<P>
To access ENCODE data, open the 
<A HREF="../cgi-bin/hgGateway?org=human">Genome Browser</A>, select the March 
2006 assembly of the human genome, and go to 
your region of interest. ENCODE tracks will be marked with the NHGRI logo
<IMG SRC="../images/nhgri.gif" ALT="" HEIGHT=16 WIDTH=16>). 
The bulk of the ENCODE data 
can be found in the <em>Expression</em> and <em>Regulation</em> track groups, with a few 
in the <em>Mapping</em>, <em>Genes</em>, and <em>Variation</em> groups. 
Although most participating research groups have provided several tracks, 
generally only selected data from each research group are displayed by 
default. Click the hyperlinked name of a particular track to display a 
page containing configuration options and details about the methods used to 
generate the data. See the Genome Browser 
<A HREF="../goldenPath/help/hgTracksHelp.html">User's Guide</A> for further 
information about displaying tracks and navigating in the Genome Browser.
</P>
<P>
Data from the earlier pilot phase of the ENCODE project are available on the 
May 2004 and March 2006 human assemblies. These datasets are generally 
available only in the initial ENCODE-targeted regions that covered 
approximately 1% of the genome. The 
<A HREF="pilot.html">ENCODE 
Pilot Project</A> web pages provide convenient browser access to these regions.
</P>
<P>
Before publishing research that uses ENCODE data, please read the
<A HREF="terms.html">data release policy</A>, which places some restrictions 
on publication use of data for nine months following the data release.
</P>
	          </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%" BORDER=0>
                  <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                      <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT>
 		    	<TD ALIGN="right"><A HREF="newsarch.html"><IMG SRC="../images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>

		      </TD><TD ALIGN="right">&nbsp;</TD></TR>
		    </TABLE>
	          </TD></TR>
		</TABLE>

		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
    		  <TR><TH HEIGHT=3></TH></TR>
    		  <TR><TD WIDTH=10></TD><TD>
		    <P></P>
<!-- start news item -->
<FONT FACE="courier" SIZE=3><B>23 September 2009 - ENCODE data releases since July 1
</B></FONT></BR>
<P>
During this period a total of 9 new ENCODE tracks were released to the UCSC public server.
Functional elements and region characterization in these tracks include:

<UL>
<LI>DNaseI hypersensitive sites and hotspots (University of Washington)
<LI>Regions of DNA methylation, by Methyl-seq and Illumina 27 array (HudsonAlpha Institute)
<LI>Bi-directional promoters (NHGRI Elnitski lab)
<LI>Transcriptome profiling by RNA-seq, including single-molecule sequencing (CSHL, Helicos)
<LI>Expression levels by exon array (University of Washington)
<LI>Regions of copy number variation (HudsonAlpha Institute)
<LI>Mappability/uniqueness of nmers (University of Massachusetts, Duke, Broad, Rosetta)
</UL>
<P> For links to these tracks, see the <EM>Release Log</EM> and <em>Downloads</em> links listed in the left menu bar.
<P>
We would like to thank the contributing ENCODE labs and the the DCC team at 
UCSC for their efforts completing these tracks.
</P>
<HR>
<!-- end news item --> 

<!-- start news item -->
<FONT FACE="courier" SIZE=3><B>1 July  2009 - ENCODE data releases for the period April - June 2009
</B></FONT></BR>
<P>
The following ENCODE tracks were released to the ENCODE DCC/UCSC public server during this period:
<UL>
<LI>
<A TARGET=_BLANK HREF="/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeAffyRnaChip">
Affymetrix/CSHL Subcellular RNA Localization by Tiling Array
</A>
<LI>
<A TARGET=_BLANK HREF="/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeChromatinMap">
Open Chromatin, Duke/UNC/UT
</A>
<LI>
<A TARGET=_BLANK HREF="/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeRikenCage">
RIKEN RNA Subcellular Localization by CAGE Tags
</A>
<LI>
<A TARGET=_BLANK HREF="/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaChipSeq">
HudsonAlpha Transcription Factor Binding Sites by ChIP-seq
</A>
<LI>
<A TARGET=_BLANK HREF="/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeSangerGencode">
Gencode Gene Annotations
</A>
<LI>
<A TARGET=_BLANK HREF="/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeSunyalbanyRnaGeneChip">
SUNY Albany RNA Binding Proteins by RIP-chip
</A>
<LI>
<A TARGET=_BLANK HREF="/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeBroadChipSeq">
Histone Modifications by Broad Institute ChIP-seq
</A>
</UL>
<P> The <EM>Release Log</EM> listed in the left menu bar shows all released ENCODE tracks.
<P>
We would like to thank the HudsonAlpha, Transcriptome, Open Chromatin, Broad, and SUNY
ENCODE groups and the DCC team at UCSC for their efforts completing these tracks.
</P>
<HR>
<!-- end news item --> 

<!-- start news item -->
<FONT FACE="courier" SIZE=3><B>13 March 2009 - ENCODE Data Release: Transcription Factor Binding Sites from Yale/UC-Davis/Harvard 
</B></FONT></BR>
<P>
We are pleased to announce the release to the ENCODE DCC/UCSC public server
of the 
<A TARGET=_BLANK HREF="/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeYaleChIPseq">
ENCODE Transcription Factor Binding Sites by ChIP-seq from Yale/UC-Davis/Harvard
</A>
(Yale TFBS, in the <I>Regulation</I> group).
This track shows probable binding sites of 12 transcription factors
and RNA polymerase II in 7 cell types, as determined by chromatin
immunoprecipitation followed by high-throughput sequencing.
The Genome Browser displays discrete <I>Peaks</I>
of enrichment and <I>Signal</I> graphs of enrichment density for these
experiments.
The sequence reads, quality scores, and sequence alignment coordinates from
these experiments are available for 
<A TARGET=_BLANK HREF="/ENCODE/downloads.html">download</A>.
<P>
<P>
We would like to acknowledge the efforts of the Yale/UC-Davis/Harvard
ENCODE group and the work of the UCSC data wrangler for this group,
Tim Dreszer, for completing this track.  We also thank
the entire UCSC ENCODE team and the UCSC Quality Assurance
group for contributing to this first ENCODE track release.
</P>
<!-- end news item --> 

<HR>
<!-- start news item -->
		    <FONT FACE="courier" SIZE=3><B>27 Feb 2009 - ENCODE February 2009 data freeze</B></FONT><BR>
		    <P>
                    The February 2009 ENCODE data freeze supplements the data contributed for the
                    November 2008 freeze, and will be used together with the earlier freeze
                    for the initial analysis effort of the ENCODE Consortium. Data from
                    this freeze is being incorporated into tracks created from the first
                    data freeze, and is being reviewed by the UCSC Quality Assurance team.
		    </P>
<!-- end news item -->
<HR>
<!-- start news item -->
		    <FONT FACE="courier" SIZE=3><B>9 Dec. 2008 - First ENCODE whole-genome data freeze completed</B></FONT><BR>
		    <P>
		    The ENCODE Consortium has just completed the first freeze
		    (November 2008) of whole-genome experimental data 
		    produced for the ENCODE production phase.  Data submitted
		    to the DCC for this freeze include:
		    <UL>
		      <LI>transcription factor binding sites
		      <LI>histone modifications
		      <LI>DNaseI hypersensitive sites
		      <LI>DNA methylation
		      <LI>transcription maps and tags, localized to subcellular 
		          compartments
		      <LI>GENCODE gene annotations
		    </UL>
		    <P>
		    Experiments during this freeze focused on the ENCODE 
		    Tier1 cell lines -- K562 leukemia, and GM12878 lymphoblastoid
                    (which is also a 
                    <A HREF="http://www.1000genomes.org/page.php" TARGET=_BLANK>1000 Genomes project</A>
                    sample designated for in-depth analysis of genetic variation).
		    The freeze also includes data 
                    from some ENCODE Tier2 and Tier3 cell lines (see
                    <A HREF="cellTypes.html" TARGET=_BLANK>Cell Types</A>). The majority 
		    of these experiments were assayed by high-throughput 
		    sequencing (ChIP-seq, DNase-seq, and RNA-seq).  
		    </P>
		    <P>
		    The UCSC quality team is currently reviewing these data. 
		    When the review is complete, the browser tracks and 
		    associated downloads will be released to the UCSC public 
		    Genome Browser.
		    </P>
		    <P>
		    Thanks to the many labs who contributed data for the
		    initial phase of this project. We'd also like to acknowledge
		    the UCSC ENCODE team for data wrangling during the freeze, 
		    and for the development and maintenance of 
		    the ENCODE automated data submission pipeline and 
		    associated tools: Kate Rosenbloom, Tim Dreszer, Larry Meyer,
		    Michael Pheasant, Ting Wang, Galt Barber, and Andy Pohl. 
		    </P>
<!-- end news item --> 

<!-- start archived item section-->
<!-- end archived item section--> 
		    <P></P>
    	          </TD><TD WIDTH=15></TD></TR>
	        </TABLE>
	      </TD></TR>
	    </TABLE>
          </TD></TR>
        </TABLE>
	<BR>
	    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
		  <TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	          </TD></TR>
		</TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome 
		    Browser are freely available for academic, nonprofit, and 
		    personal use with the following conditions:
		    <UL>
		      <LI>
                      ENCODE data is covered by the
		      <A HREF="terms.html">
                      ENCODE Consortium Data Release Policy</A>.
		      <LI>
		      The general 
		      <A HREF="http://genome.ucsc.edu/#Conditions">Conditions of Use</A> 
		      for the UCSC Genome Browser apply.
		    </UL>
	          </TD><TD WIDTH=15></TD></TR>
		</TABLE>
	      </TD></TR>
	    </TABLE>
          </TD></TR>
	</TABLE>
      <BR></TD></TR>
    </TABLE>
  </TD></TR>
</TABLE>
</BODY></HTML>

