<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
  <meta http-equiv="Content-Type"
 content="text/html; charset=ISO-8859-1">
  <meta name="description" content="ENCODE Project at UCSC">
  <meta name="keywords"
 content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
  <title>ENCODE Project at UCSC</title>
  <link rel="STYLESHEET" href="../style/HGStyle.css">
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
</head>
<body style="background-color: rgb(255, 249, 210);" link="#0000cc"
 vlink="#330066" alink="#6600ff">
<a name="TOP"></a><!-- Header Images and links -->
<table width="100%" border="0" cellpadding="0" cellspacing="0">
  <tbody>
    <tr>
      <th colspan="2" align="left">&nbsp;&nbsp; <a
 href="http://www.genome.gov/10005107" target="_BLANK"><img
 src="../images/ENCODE_scaleup_logo.png" alt="ENCODE Project at NHGRI"
 height="50"></a> &nbsp;&nbsp; <a href="/encode/"><img
 src="../images/encodeDcc.jpg"
 alt="ENCODE Data Coordination Center at UCSC"></a> </th>
    </tr>
    <tr>
      <td colspan="2" height="40"><!--#include virtual="../inc/encode.home.topbar.html"-->
      <br>
      </td>
    </tr>
    <tr height="100%" valign="top">
      <td rowspan="1" height="100%" valign="top"><!-- outer table is to align leftbar & content boxes at top of page -->
      <table width="100%" border="0" cellpadding="0" cellspacing="0">
        <tbody>
          <tr>
            <td><img src="../images/DOT.gif" alt="" height="7"></td>
          </tr>
          <tr height="100%">
            <td height="100%"><!--outer table is for black border-->
            <table width="100%" bgcolor="#000000" border="0"
 cellpadding="1" cellspacing="0">
              <tbody>
                <tr>
                  <td>
                  <table width="100%" border="0" cellpadding="10"
 cellspacing="0">
                    <tbody>
                      <tr height="100%">
                        <td height="100%" valign="top" bgcolor="#738ed3"><a
 class="leftbar" href="/cgi-bin/hgTracks?db=hg18">Genome Browser</a>
                        <hr> <a class="leftbar" href="cellTypes.html">Cell
Types</a>
                        <hr> <a class="leftbar" href="antibodies.html">Antibodies</a>
                        <hr> <a class="leftbar" href="releaseLog.html">Release
Log</a>
                        <hr> <a class="leftbar" href="downloads.html">Downloads</a>
                        <hr> <a class="leftbar"
 href="contributors.html">Contributors</a>
                        <hr> <a class="leftbar" href="pubs.html">Publications</a>
                        <hr> <a class="leftbar" href="terms.html">Data
Policy</a>
                        <hr> <a class="leftbar" href="pilot.html">Pilot
Project</a>
                        <hr> <a class="leftbar"
 href="http://www.cbse.ucsc.edu/jobs/" target="_blank">Jobs</a>
                        <hr> <a class="leftbar" href="contacts.html">Contact
Us</a>
                        <hr> <br>
                        <img src="../images/DOT.gif" alt="" height="970"
 width="1"> </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            </td>
          </tr>
        </tbody>
      </table>
      </td>
<!--Content Tables--------------------> <td valign="top">
      <table border="0" cellpadding="5">
        <tbody>
          <tr>
            <td><!--outer table is for border purposes-->
            <table width="100%" bgcolor="#888888" border="0"
 cellpadding="1" cellspacing="0">
              <tbody>
                <tr>
                  <td>
                  <table width="100%" bgcolor="#fffee8" border="0"
 cellpadding="0" cellspacing="0">
                    <tbody>
                      <tr>
                        <td>
                        <table width="100%"
 background="../images/hr.gif" bgcolor="#d9e4f8" border="0">
                          <tbody>
                            <tr>
                              <td><a name="About"></a><font size="4"><b>&nbsp;
About
the ENCODE Data Coordination Center (DCC)</b></font> </td>
                            </tr>
                          </tbody>
                        </table>
                        <table width="100%" bgcolor="#fffee8" border="0"
 cellpadding="0">
                          <tbody>
                            <tr>
                              <th height="3"><br>
                              </th>
                              <td><br>
                              </td>
                              <td><br>
                              </td>
                            </tr>
                            <tr>
                              <td width="10"><br>
                              </td>
                              <td>
                              <p>The <a
 href="http://www.genome.gov/10005107" target="_blank">Encyclopedia of
DNA Elements</a> (ENCODE) Consortium is an international collaboration
of research groups funded by the National Human Genome Research
Institute (<a href="http://www.genome.gov/" target="_blank">NHGRI</a>).
The goal of the consortium is to build a comprehensive parts list of
the functional elements of the human genome, including elements that
act at the protein level (coding genes) and RNA level (non-coding
genes), and regulatory elements that control the cells and
circumstances in which a gene is active. The discovery and annotation
of gene elements is accomplished primarily by sequencing RNA from a
diverse range of sources, comparative genomics, integrative
bioinformatic methods, and human curation. Regulatory elements are
typically investigated through DNA hypersensitivity assays, assays of
DNA methylation, and chromatin immunoprecipitation (ChIP) of proteins
that interact with DNA, including modified histones and transcription
factors, followed by sequencing (ChIP-Seq). The results of ENCODE
experiments, collected in the ENCODE DCC database, are displayed on the
UCSC Genome Browser. The data can also be downloaded from the ENCODE
DCC website in text format. </p>
                              <p>To access ENCODE data, open the <a
 href="../cgi-bin/hgGateway?org=human">Genome Browser</a>, select the
March 2006 assembly of the human genome, and go to your region of
interest. ENCODE tracks will be marked with the NHGRI logo <img
 src="../images/nhgri.gif" alt="" height="16" width="16">. The bulk of
the ENCODE data can be found in the <em>Expression</em> and <em>Regulation</em>
track groups, with a few in the <em>Mapping</em>, <em>Genes</em>, and
                              <em>Variation</em> groups. Although most
participating research groups have provided several tracks, generally
only selected data from each research group are displayed by default.
Click the hyperlinked name of a particular track to display a page
containing configuration options and details about the methods used to
generate the data. See the Genome Browser <a
 href="../goldenPath/help/hgTracksHelp.html">User's Guide</a> for
further information about displaying tracks and navigating in the
Genome Browser. To receive notifications of ENCODE data releases and
related news by email, subscribe to the <a
 href="https://lists.soe.ucsc.edu/mailman/listinfo/encode-announce"
 target="_blank">encode-announce</a> mailing list.</p>
                              <p>Data from the earlier pilot phase of
the ENCODE project are available on the May 2004 and March 2006 human
assemblies. These datasets are generally available only in the initial
ENCODE-targeted regions that covered approximately 1% of the genome.
The <a href="pilot.html">ENCODE Pilot Project</a> web pages provide
convenient browser access to these regions. </p>
                              <p>Before publishing research that uses
ENCODE data, please read the <a href="terms.html">data release policy</a>,
which
places some restrictions on publication use of data for nine
months following the data release. </p>
                              </td>
                              <td width="15"><br>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        <br>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <br>
<!--outer table is for border purposes-->
            <table width="100%" bgcolor="#888888" border="0"
 cellpadding="1" cellspacing="0">
              <tbody>
                <tr>
                  <td>
                  <table width="100%" bgcolor="#fffee8" border="0"
 cellpadding="0" cellspacing="0">
                    <tbody>
                      <tr>
                        <td>
                        <table width="100%"
 background="../images/hr.gif" bgcolor="#d9e4f8" border="0">
                          <tbody>
                            <tr>
                              <td><!--cheap trick to get background on row to be continuous-->
                              <table width="100%" border="0"
 cellpadding="0" cellspacing="0">
                                <tbody>
                                  <tr>
                                    <td align="left"><font size="4"><a
 name="TOC"></a><b>&nbsp; News</b></font> </td>
                                    <td align="right"><a
 href="newsarch.html"><img src="../images/news.jpg" alt="" align="right"
 border="0"></a> </td>
                                    <td align="right">&nbsp;</td>
                                  </tr>
                                </tbody>
                              </table>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        <table width="100%" bgcolor="#fffee8" border="0"
 cellpadding="0">
                          <tbody>
                            <tr>
                              <th height="3"><br>
                              </th>
                              <td><br>
                              </td>
                              <td><br>
                              </td>
                            </tr>
                            <tr>
                              <td width="10"><br>
                              </td>
                              <td><!-- start news item --> <font
 face="courier" size="3"><b>6 January 2010 - December 2009 ENCODE data
releases<br>
                              <br>
                              </b></font><span
 style="font-weight: bold;">Initial release of the <a
 href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeCaltechRnaSeq">Caltech
RNA-seq</a> track: </span>This track contains sequence reads and RPKM
transcript abundance measures for sequences that map to either the
genome or to known RNA splice sites.&nbsp; The results of four
different mapping algorithms are provided, enabling comparison between
different mapping algorithms.&nbsp; Results are available for polyA+
and total RNA for the two ENCODE Tier 1 cell lines.<br>
                              <br>
                              <span style="font-weight: bold;">Release
2 of the <a
 href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeBroadChipSeq">Broad
Histone</a> track: </span>This track displays maps of chromatin state
generated using CHIP-seq.&nbsp; Release 2 adds data for the ENCODE Tier
2 cell lines H1-hESC and HepG2, plus NHLF (normal human lung
fibroblasts) and HMEC (human mammary epithelial) cells.&nbsp; This
expands the track data to 9 cell lines, and 11 antibodies plus
an input control. <br>
                              <br>
                              <span style="font-weight: bold;">Release
2 of the <a
 href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeCshlLongRnaSeq">CSHL

Long
RNA-seq</a> track: </span>This track depicts sequencing of long
RNAs of more than 200 nucleotides in length.&nbsp; Release 2 adds data
from strand-specific assays of total RNA for the two ENCODE Tier 1 cell
lines.<br>
                              <br>
                              <span style="font-weight: bold;">Release
2 of the <a
 href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeChromatinMap">ENCODE
Open Chromatin</a> track: </span>This track displays evidence of open
chromatin as identified by two complementary methods, DNaseI
hypersensitivity and FAIRE, combined with ChIP identification
methods.&nbsp; Release 2 adds data from eight additional cell types,
expanding the track to 41 experiments in 13 cell lines.<br>
                              <hr><font face="courier" size="3"><b>7
November 2009 - October ENCODE News </b></font><br>
                              <p><b>Sep 2009 data freeze complete:</b>&nbsp;&nbsp;The
ENCODE
Consortium has just completed data submissions for the fourth
production data freeze (Sep 09). The first set of data from this freeze
to complete quality review is now available on the UCSC public server,
in Release 2 of the <a
 href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeYaleChIPseq">ENCODE
Transcription Factor Binding Sites from Yale/UC-Davis/Harvard</a>
track. Release 2 adds 59 ChIP-seq experiments to this track. </p>
                              <p><b>Other October track releases:</b>&nbsp;&nbsp;The

                              <a
 href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeAffyRnaChip">Affymetrix/CSHL

Subcellular
RNA Localization by Tiling Array</a> track was expanded to
include 4 additional experiments. </p>
                              <p><b>encodeproject.org:</b>&nbsp;&nbsp;By

request
of the ENCODE Consortium, the domain <i>encodeproject.org</i>
has been registered by the ENCODE Data Coordination Center, and is
redirected to the ENCODE portal at UCSC. </p>
                              <p><b>New grants funded:</b>&nbsp;&nbsp;NHGRI
has
funded 5 new ENCODE grants, as part of the American Investment and
Recovery Act. The new grants include expansion of ENCODE to the mouse
genome and proteogenomics. </p>
                              <p><b>Job openings at UCSC:</b>&nbsp;&nbsp;The

UCSC
Genome Browser and ENCODE projects are currently accepting
applications for <a href="http://www.cbse.ucsc.edu/jobs/staff#sftwr">Software
Developer</a> and <a href="http://www.cbse.ucsc.edu/jobs/staff#bdtust">Biological

Database
Testing/User Support Technician</a> positions. We are looking
for talented individuals who would like to use their skills in computer
science, biology, and bioinformatics on fast-paced projects featuring
the work of top genomics scientists worldwide. </p>
                              <hr><!-- end news item --><!-- start news item -->
                              <font face="courier" size="3"><b>24
September 2009 - ENCODE data releases
since July 1 </b></font><br>
                              <p>During this period a total of 10 new
ENCODE tracks were released to the UCSC public server. Functional
elements and region characterization in these tracks include: </p>
                              <ul>
                                <li><a
 href="../../cgi-bin/hgTrackUi?g=wgEncodeUwDnaseSeq">DNaseI</a>
hypersensitive sites and hotspots (University of Washington) </li>
                                <li>Regions of DNA methylation, by <a
 href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeHudsonalphaMethylSeq">Methyl-seq</a>
and <a
 href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeHudsonalphaMethyl27">Illumina
27</a> array (HudsonAlpha Institute) </li>
                                <li><a
 href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeNhgriBip">Bi-directional
promoters</a> and <a
 href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeNhgriNre">negative
regulatory elements</a> (NHGRI Elnitski lab) </li>
                                <li>Transcriptome profiling by RNA-seq,
including single-molecule sequencing (<a
 href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeCshlLoRnaSeq">CSHL</a>,


                                  <a
 href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeHelicosRnaSeq">Helicos</a>)


                                </li>
                                <li>Expression levels by <a
 href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeUwAffyExonArray">exon
array</a> (University of Washington) </li>
                                <li>Regions of <a
 href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeHudsonalphaCnv">copy
number variation</a> (HudsonAlpha Institute) </li>
                                <li><a
 href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeMapability">Mappability/uniqueness</a>
of nmers (University of Massachusetts, Duke, Broad, Rosetta) </li>
                              </ul>
                              <p>For track names and file access, see
the <em>Release Log</em> and <em>Downloads</em> links listed in the
left menu bar. </p>
                              <p>We would like to thank the
contributing ENCODE labs and the the DCC team at UCSC for their efforts
completing these tracks. </p>
                              <hr><!-- end news item --><!-- start archived item section-->
                              <font face="courier" size="3"><b>1 July
2009 - ENCODE data releases for the
period April - June 2009 </b></font><br>
                              <p>During this period, a total of seven
new ENCODE tracks were released to the UCSC public server. These tracks
include high-quality gene annotations, maps of transcription factor
binding, histone modifications, and open chromatin, RNA subcellular
localization, and RNA/protein binding sites. <a
 href="newsarch.html#070109">Read more</a>.</p>
<!-- end archived item section--> </td>
                              <td width="15"><br>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <br>
<!--outer table is for border purposes-->
            <table width="100%" bgcolor="#888888" border="0"
 cellpadding="1" cellspacing="0">
              <tbody>
                <tr>
                  <td>
                  <table width="100%" bgcolor="#fffee8" border="0"
 cellpadding="0" cellspacing="0">
                    <tbody>
                      <tr>
                        <td>
                        <table width="100%"
 background="../images/hr.gif" bgcolor="#d9e4f8" border="0">
                          <tbody>
                            <tr>
                              <td><a name="Conditions"></a><font
 size="4"><b>&nbsp; Conditions of Use</b></font> </td>
                            </tr>
                          </tbody>
                        </table>
                        <table width="100%" bgcolor="#fffee8" border="0"
 cellpadding="0">
                          <tbody>
                            <tr>
                              <th height="3"><br>
                              </th>
                              <td><br>
                              </td>
                              <td><br>
                              </td>
                            </tr>
                            <tr>
                              <td width="10"><br>
                              </td>
                              <td>
                              <p>The sequence and annotation data
displayed in the Genome Browser are freely available for academic,
nonprofit, and personal use with the following conditions: </p>
                              <ul>
                                <li>ENCODE data is covered by the <a
 href="terms.html">ENCODE Consortium Data Release Policy</a>. </li>
                                <li>The general <a
 href="http://genome.ucsc.edu/#Conditions">Conditions of Use</a> for
the UCSC Genome Browser apply. </li>
                              </ul>
                              </td>
                              <td width="15"><br>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <br>
            </td>
          </tr>
        </tbody>
      </table>
      </td>
    </tr>
  </tbody>
</table>
</body>
</html>
