<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
  <HEAD>
    <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
    <META name="description" content="ENCODE Project at UCSC">
    <META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
    <TITLE>ENCODE Project at UCSC</TITLE>
    <LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
  </HEAD>

  <BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
  <A NAME="TOP"></A>

    <!-- Header Images and links -->
    <TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
      <TR>
        <TH COLSPAN=2 ALIGN="left">&nbsp;&nbsp;
          <A HREF="http://www.genome.gov/10005107" TARGET=_BLANK><IMG SRC="../images/ENCODE_scaleup_logo.png" height=50 ALT="ENCODE Project at NHGRI"></A>
          &nbsp;&nbsp;
          <A HREF="/encode/"><IMG SRC="../images/encodeDcc.jpg" ALT="ENCODE Data Coordination Center at UCSC"></A>
        </TH>
      </TR>

      <TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/encode.home.topbar.html"--></TD></TR>
      </TR>

    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="../images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
                <A CLASS="leftbar" HREF="/cgi-bin/hgTracks?db=hg18">
		   Genome Browser</A>
	        <HR>
	        <A CLASS="leftbar" HREF="cellTypes.html">
		   Cell Types</A>
                <!-- TODO: Replace when first tracks are released 
	        <HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/goldenPath/wgEncode/index.html">
		   Downloads</A>
                   -->
	        <HR>
                <A CLASS="leftbar" HREF="releaseLog.html">
                    Release Log</A>
	        <HR>
	        <A CLASS="leftbar" HREF="downloads.html">
		   Downloads</A>
	        <HR>
		<A class="leftbar" HREF="contributors.html">
		   Contributors</A>
		<HR>
	        <A CLASS="leftbar" HREF="pubs.html">
		   Publications</A>
	        <HR>
	        <A CLASS="leftbar" HREF="terms.html">
		   Data Policy</A>
		<HR>
	        <A CLASS="leftbar" HREF="pilot.html">
		   Pilot Project</A>
		<HR>
                <A CLASS="leftbar" HREF="contacts.html">
		   Contact Us</A>
		<HR>
 	        <BR>
	        <IMG SRC="../images/DOT.gif" ALT="" HEIGHT=970 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Content Tables-------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	

	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%"><TR>
	          <TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; 
		    About the ENCODE Data Coordination Center (DCC)</B></FONT></TD>
	        </TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
                  <P>
		    The UCSC Genome Browser displays data produced by the
		    <A HREF="http://www.genome.gov/10005107" 
		    TARGET=_blank>
		    Encyclopedia of DNA Elements</A>
                    (ENCODE) Consortium, an international
                    collaboration of research groups funded by the 
                    National Human Genome Research Institute
		    (<A HREF="http://www.genome.gov/" TARGET=_blank>NHGRI</A>).
                    The goal of ENCODE is to
		    build a comprehensive parts list of the 
		    functional elements in the human genome.
                    </P>
                    <P>
                    In September 2007, the ENCODE project
                    was scaled up from  a pilot phase to cover the entire human genome.
		    To access genome-wide ENCODE data in the 
		    <A HREF="../cgi-bin/hgTracks?org=human">Genome Browser</A>,
		    go to your region of interest and select
		    ENCODE tracks (marked with the NHGRI logo 
		    <IMG SRC="../images/nhgri.gif" ALT="" HEIGHT=16 WIDTH=16>). 
                    </P>
                    <P>
                    During the pilot phase of the ENCODE project (2003-2007), 
                    experiments focused on a limited set of genomic regions comprising
                    roughly 1% of the human genome.  
		    Data limited to the pilot regions 
                    is located in ENCODE-specific track groups in the
                    browser. The <A HREF="/encode/pilot.html">ENCODE Pilot 
		    Project</A> web pages provide
                    convenient browser access to these regions.  
                    </P>
                    Other key differences from the pilot project are:
                    <UL>
                    <LI> 
		    Identification of 
                    <A HREF="cellTypes.html">common cell types</A> to
                    facilitate integrative analysis
                    <LI> 
		    New experimental technologies based on high-throughput 
		    sequencing
                    <LI>
		    A  <A HREF="terms.html">data release policy</A> restricting 
		    use of data for nine months following release
                    </UL>
                    </P>
		    <P>
 		    Click <A HREF="../index.html">here</A> to go to 
		    the main UCSC Genome Browser site, which provides access to 
		    sequence and annotation data for a large collection of 
		    genome assemblies.
                    See the Genome Browser 
		    <A HREF="../goldenPath/help/hgTracksHelp.html">User's 
		    Guide</A> for information about displaying tracks and
		    navigating in the Genome Browser.</A>
	          </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%" BORDER=0>
                  <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                      <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT>
 		    	<TD ALIGN="right"><A HREF="newsarch.html"><IMG SRC="../images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>

		      </TD><TD ALIGN="right">&nbsp;</TD></TR>
		    </TABLE>
	          </TD></TR>
		</TABLE>

		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
    		  <TR><TH HEIGHT=3></TH></TR>
    		  <TR><TD WIDTH=10></TD><TD>
		    <P></P>
<!-- start news item -->
<FONT FACE="courier" SIZE=3><B>13 March 2008 - First ENCODE data release from November 2008 and February 2009 data freezes 
</B></FONT></BR>
<P>
We are pleased to announce the release to the UCSC public server
of the 
<A TARGET=_BLANK HREF="/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeYaleChIPseq">
ENCODE Transcription Factor Binding Sites by ChIP-seq from Yale/UC-Davis/Harvard
</A>
(Yale TFBS).
This track shows probable binding sites of 12 transcription factors
and RNA polymerase II in 7 cell types, as determined by chromatin
immunoprecipitation followed by high throughput sequencing.
The Genome Browser displays discrete <I>Peaks</I>
of enrichment and <I>Signal</I> graphs of enrichment density for these
experiments.
The sequence reads, quality scores, and sequence alignment coordinates from
these experiments are available for 
<A TARGET=_BLANK HREF="/ENCODE/downloads.html">download</A>.
<P>
<P>
We would like to acknowledge the efforts of the Yale/UC-Davis/Harvard
ENCODE group and the work of the UCSC data wrangler for this group,
Tim Dreszer, for completing this track.  We also thank
the entire UCSC ENCODE team and the UCSC Quality Assurance
group for contributing to this first ENCODE track release.
</P>
<!-- end news item --> 
<HR>
<!-- start news item -->
		    <FONT FACE="courier" SIZE=3><B>9 Dec. 2008 - First ENCODE whole-genome data freeze completed</B></FONT><BR>
		    <P>
		    The ENCODE Consortium has just completed the first freeze
		    (November 2008) of whole-genome experimental data 
		    produced for the ENCODE production phase.  Data submitted
		    to the DCC for this freeze include:
		    <UL>
		      <LI>transcription factor binding sites
		      <LI>histone modifications
		      <LI>DNaseI hypersensitive sites
		      <LI>DNA methylation
		      <LI>transcription maps and tags, localized to subcellular 
		          compartments
		      <LI>GENCODE gene annotations
		    </UL>
		    <P>
		    Experiments during this freeze focused on the ENCODE 
		    Tier1 cell lines -- K562 leukemia, and GM12878 lymphoblastoid
                    (which is also a 
                    <A HREF="http://www.1000genomes.org/page.php" TARGET=_BLANK>1000 Genomes project</A>
                    sample designated for in-depth analysis of genetic variation).
		    The freeze also includes data 
                    from some ENCODE Tier2 and Tier3 cell lines (see
                    <A HREF="cellTypes.html" TARGET=_BLANK>Cell Types</A>). The majority 
		    of these experiments were assayed by high-throughput 
		    sequencing (ChIP-seq, DNase-seq, and RNA-seq).  
		    </P>
		    <P>
		    The UCSC quality team is currently reviewing these data. 
		    When the review is complete, the browser tracks and 
		    associated downloads will be released to the UCSC public 
		    Genome Browser.
		    </P>
		    <P>
		    Thanks to the many labs who contributed data for the
		    initial phase of this project. We'd also like to acknowledge
		    the UCSC ENCODE team for data wrangling during the freeze, 
		    and for the development and maintenance of 
		    the ENCODE automated data submission pipeline and 
		    associated tools: Kate Rosenbloom, Tim Dreszer, Larry Meyer,
		    Michael Pheasant, Ting Wang, Galt Barber, and Andy Pohl. 
		    </P>
<!-- end news item --> 

<!-- start archived item section-->
<!-- end archived item section--> 
		    <P></P>
    	          </TD><TD WIDTH=15></TD></TR>
	        </TABLE>
	      </TD></TR>
	    </TABLE>
          </TD></TR>
        </TABLE>
	<BR>
	    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
		  <TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	          </TD></TR>
		</TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome 
		    Browser are freely available for academic, nonprofit, and 
		    personal use with the following conditions:
		    <UL>
		      <LI>
                      ENCODE data is covered by the
		      <A HREF="terms.html">
                      ENCODE Consortium Data Release Policy</A>.
		      <LI>
		      The general 
		      <A HREF="http://genome.ucsc.edu/#Conditions">Conditions of Use</A> 
		      for the UCSC Genome Browser apply.
		    </UL>
	          </TD><TD WIDTH=15></TD></TR>
		</TABLE>
	      </TD></TR>
	    </TABLE>
          </TD></TR>
	</TABLE>
      <BR></TD></TR>
    </TABLE>
  </TD></TR>
</TABLE>
</BODY></HTML>

