
<html>
<head>
  <meta http-equiv="Content-Type"
 content="text/html; charset=ISO-8859-1">
  <meta name="description" content="ENCODE Project at UCSC">
  <meta name="keywords"
 content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
  <title>ENCODE Project at UCSC</title>
  <link rel="STYLESHEET" href="../style/HGStyle.css">
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
 <script type="text/javascript" src="../js/jquery.js"></script>
 <script type="text/javascript" src="../js/staticDoc.js"></script>
</head>
<body style="font-family: arial, Geneva, Helvetica, sans-serif;" alink="#6600ff"
 link="#0000cc" vlink="#330066">
<a name="TOP"></a><!-- Header Images and links -->
<table border="0" cellpadding="0" cellspacing="0" width="100%">
  <tbody>
    <tr>
      <th colspan="2" align="left">
<img src="../images/ENCODE_scaleup_logo.png" alt="ENCODE Project at NHGRI" height="75"> &nbsp;&nbsp;
 <font size="6" face="arial, Geneva, Helvetica, sans-serif"><font color="004D99">
 Encyclopedia of DNA Elements</font>
       </th>
    </tr>
    <tr height="100%" valign="top">
      <td rowspan="1" height="100%" valign="top"><!-- outer table is to align leftbar & content boxes at top of page -->
      <table border="0" cellpadding="0" cellspacing="0" width="100%">
        <tbody>
          <tr>
            <td><img src="../images/DOT.gif" alt="" height="7"></td>
          </tr>
          <tr height="100%">
            <td height="100%"><!--outer table is for black border-->
            <table bgcolor="#004D99" border="0" cellpadding="1" cellspacing="0" width="100%">
              <tbody>
                <tr>
                  <td>
                  <table border="0" cellpadding="10" cellspacing="0" width="100%">
                    <tbody>
                      <tr height="100%">
                        <td bgcolor="#0004D99" height="100%" valign="top">
                        <font color="#FFCC00"><b><em>Human</em></b></font><br>
                        <img src="../images/DOT.gif" alt="" height="8" width="0"><br>
                        <a class="leftbar" href="/ENCODE/dataMatrix/encodeDataMatrixHuman.html">Experiment Matrix</a>
                        <hr><a class="leftbar" href="/ENCODE/dataSummary.html" title="formerly Data Summary">Experiment List</a>
                        <hr><a class="leftbar" href="search.html">Search</a>
                        <hr><a class="leftbar" href="downloads.html">Downloads</a>
                        <hr><a class="leftbar" href="/cgi-bin/hgTracks?db=hg19">Genome Browser (hg19)</a>
                        <hr><a class="leftbar" href="http://genome-preview.ucsc.edu/cgi-bin/hgTracks?db=hg19">Preview Browser (hg19)</a>
                        <hr><a class="leftbar" href="http://genomewiki.cse.ucsc.edu/index.php/Genome_Browser_Session_Gallery"
                        TARGET=_blank>Session Gallery</a>
                        <hr><a class="leftbar" href="cellTypes.html">Cell Types</a>
                        <hr>
                        <img src="../images/DOT.gif" alt="" height="5" width="0"><br>
                        <font color="#FFCC00"><b><em>Mouse</em></b></font><br>
                        <img src="../images/DOT.gif" alt="" height="8" width="0"><br>
                        <a class="leftbar" href="/ENCODE/dataMatrix/encodeDataMatrixMouse.html">Experiment Matrix</a>
                        <hr><a class="leftbar" href="/ENCODE/dataSummaryMouse.html"title="formerly Data Summary">Experiment List</a>
                        <hr><a class="leftbar" href="search.html">Search</a>
                        <hr><a class="leftbar" href="downloadsMouse.html">Downloads</a>
                        <hr><a class="leftbar" href="/cgi-bin/hgTracks?db=mm9">Genome Browser (mm9)</a>
                        <hr><a class="leftbar" href="http://genome-preview.ucsc.edu/cgi-bin/hgTracks?db=mm9">Preview Browser (mm9)</a>
                        <hr><a class="leftbar" href="cellTypesMouse.html">Cell Types</a>
                        <hr>
                        <img src="../images/DOT.gif" alt="" height="5" width="0"><br>
                        <font color="FFCC00"><b><em>General</em></b></font><br>
                        <img src="../images/DOT.gif" alt="" height="8"><br>
                        <a class="leftbar" href="otherTerms.html">Registered Variables</a>
                        <hr><a class="leftbar" href="antibodies.html">Antibodies</a>
                        <hr><a class="leftbar" href="usageResources.html">Education and Outreach</a>
                        <hr><a class="leftbar" href="releaseLog.html">Release Log</a>
                        <hr><a class="leftbar" href="terms.html">Data Policy</a>
                        <hr><a class="leftbar" href="dataStandards.html">Data Standards</a><font color="#FF0000"><sup><strong><em>Updated</strong></em></sup></font>
                        <hr><a class="leftbar" href="/FAQ/FAQformat.html#ENCODE">File Formats</a>
                        <hr><a class="leftbar" href="softwareTools.html">Software Tools</a>
                        <hr><a class="leftbar" href="pubs.html">Publications</a>
                        <hr><a class="leftbar" href="contributors.html">Contributors</a>
                        <hr><a class="leftbar" href="pilot.html">Pilot Project</a>
                        <hr><a class="leftbar" href="http://www.cbse.ucsc.edu/jobs/" target="_blank">Jobs</a>
                        <hr><a class="leftbar" href="contacts.html">Contact Us</a>
                        <hr><br>
                        <img src="../images/DOT.gif" alt="" height="970" width="1"> </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            </td>
          </tr>
        </tbody>
      </table>
      </td>
<!-- Content Tables --> <td valign="top">
      <table border="0" cellpadding="5">
        <tbody>
          <tr>
            <td><!--outer table is for border purposes-->
            <table bgcolor="#888888" border="0" cellpadding="1"
 cellspacing="0" width="100%">
              <tbody>
                <tr>
                  <td>
                  <table bgcolor="#FFFFFF" border="0" cellpadding="0"
 cellspacing="0" width="100%">
                    <tbody>
                      <tr>
                        <td>
                        <table background="../images/hr.gif"
 bgcolor="#FFCC00" border="0" width="100%">
                          <tbody>
                            <tr>
                              <td><a name="About"></a><font size="4"><b>&nbsp;
About ENCODE Data</b></font> </td>
                            </tr>
                          </tbody>
                        </table>
                        <table bgcolor="#FFFFFF" border="0"
 cellpadding="0" width="100%">
                          <tbody>
                            <tr>
                              <th height="3"><br>
                              </th>
                              <td><br>
                              </td>
                              <td><br>
                              </td>
                            </tr>
                            <tr>
                              <td width="15"><br>
                              </td>
                              <td>
<p>The <a href="http://www.genome.gov/10005107" target="_blank">Encyclopedia of DNA Elements</a>
(ENCODE) Consortium is an international collaboration
of research groups funded by the National Human Genome Research
Institute (<a href="http://www.genome.gov/" target="_blank">NHGRI</a>).
The goal of ENCODE is to build a comprehensive parts list of
functional elements in the human genome, including elements that
act at the protein and RNA levels,
and regulatory elements that control cells and
circumstances in which a gene is active.
</p>
<!--#if expr='"$SERVER_NAME" = /hgwdev.*.cse.ucsc.edu/ || "$SERVER_NAME" = "genome-test.cse.ucsc.edu"'-->
<p>
<b>WARNING: This is a test site.</b>
Data and tools here are under 
construction, have not been quality reviewed, and are subject to change 
at any time.  
For 
high-quality reviewed annotations on our production server, visit
<a HREF="http://encodeproject.org">http://encodeproject.org</a>.</p>
<!--#endif -->

<!--#if expr='"$SERVER_NAME" = /genome-preview.*.ucsc.edu/' -->
<p>
<b>WARNING: This is our preview site.</b>
This website is a weekly mirror of our internal 
development server for public access.  
Data and tools here are under 
construction, have not been quality reviewed, and are subject to change 
at any time.  We provide this site for early access, with the warning that it is
less available and stable than our public site. For 
high-quality reviewed annotations on our production server, visit
<a HREF="http://encodeproject.org">http://encodeproject.org</a>.</p>
<!--#endif -->

<table><tr>
<td><a href="/ENCODE/aboutScaleup.html" target="_blank"><img src="../images/EncodeNatureGraphic.png" alt="" height="200">
<em><font size=-2>Click to enlarge</em></font></td>

<td>
ENCODE data are now available for the entire human genome.
<b><em>All ENCODE data are free and available for immediate use via </em></b>:
<ul class="simple">
<li><a href="/ENCODE/search.html" target="_blank">Search</a> for displayable tracks and downloadable files
</li>
<li><a href="/ENCODE/downloads.html" target="_blank">Download</a> of data files
</li>
<li><a href="http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOtherUser=submit&hgS_otherUserName=Kate&hgS_otherUserSessionName=encodePortalSession"
target="_blank">Visualization</a> in the UCSC Genome Browser (ENCODE data marked with
the <img src="../images/nhgri.gif" alt="" height="16" width="16"> NHGRI logo)
</li>
<li><a href="/cgi-bin/hgTables?db=hg18&hgta_group=regulation&hgta_track=wgEncodeHudsonalphaChipSeq"
target="_blank">Data mining</a> with the UCSC Table Browser and other
<a href="../">UCSC Genome Bioinformatics tools</a>
</li>
</ul>
</td>

</tr></table>
</p>
<p>
To search for ENCODE data related to your area of interest and set up a browser view, use the UCSC
<a href="dataMatrix/encodeDataMatrixHuman.html" target="_blank">Experiment Matrix</a> or
<a href="/cgi-bin/hgTracks?hgt_tSearch=1" target="_blank">Track Search tool</a> (<em>Advanced</em> features).
The <a href="dataSummary.html" target="_blank">Experiment List (Human)</a> and 
 <a href="dataSummaryMouse.html" target="_blank">Experiment List (Mouse)</a> links provide
comprehensive listings of ENCODE data that is released or in preparation.
Early access to pre-release ENCODE data is provided at
<a href="http://genome-preview.ucsc.edu" target="_blank">http://genome-preview.ucsc.edu</a>.
If you would like to receive notifications of ENCODE data releases and related news by email,
subscribe to the
<a href="https://lists.soe.ucsc.edu/mailman/listinfo/encode-announce" target="_blank">encode-announce mailing list</a>.
For more information about how to access this data, see the free online
<a target=="_blank" href="http://www.openhelix.com/ENCODE/" target="_blank">OpenHelix ENCODE tutorial</a>.
</p>
<p>
All ENCODE data is freely available for download and analysis.
However, before publishing research that uses ENCODE data,
please read the <a href="terms.html" target="_blank">ENCODE Data Release Policy</a>,
which places some restrictions on publication use of data for nine
months following data release. &nbsp;&nbsp;
<a href="/ENCODE/aboutScaleup.html">Read more</a> about ENCODE data at UCSC.</p>

                              </td>
                              <td width="15"><br>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        <br>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <br>
<!--outer table is for border purposes-->
            <table bgcolor="#888888" border="0" cellpadding="1"
 cellspacing="0" width="100%">
              <tbody>
                <tr>
                  <td>
                  <table bgcolor="#FFFFFF" border="0" cellpadding="0"
 cellspacing="0" width="100%">
                    <tbody>
                      <tr>
                        <td>
                        <table background="../images/hr.gif"
 bgcolor="#FFCC00" border="0" width="100%">
                          <tbody>
                            <tr>
                              <td><!--cheap trick to get background on row to be continuous-->
                              <table border="0" cellpadding="0" cellspacing="0" width="100%">
                                <tbody>
                                  <tr>
                                    <td align="left"><font size="4"><a name="TOC"></a><b>&nbsp; News</b></font> </td>
                                    <td align="right"><a href="newsarch.html"><img src="../images/news.jpg" alt="" align="right" border="0"></a> </td>
                                    <td align="right">&nbsp;</td>
                                  </tr>
                                </tbody>
                              </table>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        <table bgcolor="#FFFFFF" border="0" cellpadding="0" width="100%">
                          <tbody>
                            <tr>
                              <th height="3"><br>
                              </th>
                              <td><br>
                              </td>
                              <td><br>
                              </td>
                            </tr>
                            <tr>
                              <td width="15"><br>
                              </td>
                              <td style="font-family: arial, Geneva, Helvetica, sans-serif;">

<!-- start news item -->
<p>
<font size="3"><b>02 July 2012 -  Quality metrics for human ENCODE data
</b></font>
<p>
The ENCODE Consortium analysis working group has analyzed the quality of the data produced using a variety of metrics.  The ENCODE portal
<a class="euro" href="qualityMetrics.html">
Quality Metrics</a> page provides
spreadsheets showing quality metrics for human ENCODE datasets, along with descriptions of what the metrics are and what they appear to measure.  Spreadsheets are provided for <em>DNase-seq</em>, <em>FAIRE-seq</em>, <em>Transcription Factor ChIP-seq</em>, <em>Histone ChIP-seq</em>, and <em>ChIP Controls</em>. This page is linked from the related
<a class="euro" href="dataStandards.html">
Data Standards</a> page.
Many of the software tools used for quality metrics, along with their citations, can be found on the portal 
<a class="euro" href="softwareTools.html">
Software Tools</a> page.</p>
<hr>
<!-- end news item -->

<!-- start news item -->
<p>
<font size="3"><b> 26 Jun 2012 -  Human ENCODE data releases: GENCODE Genes V12, CSHL Long RNA-seq (Release 2), SYDH Histone (Rel 3), UW DNaseI HS (Rel 5)
</b></font>
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV12">
Gene Annotations from ENCODE/GENCODE Version 12</a>:
This version is based on the Ensembl 67 (May 2012) and Vega 47 (April 2012) gene predictions and curation.</p>
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCshlLongRnaSeq">
Long RNA-seq from ENCODE/Cold Spring Harbor Lab (Release 2)</a>:
includes 25 new cell lines: CD20+, CD34+_Mobilized, HAoAF, HAoEC, HCH, HFDPC, HMEpC, hMNC-CB, hMNC-PB, hMSC-AT, hMSC-BM, hMSC-UC, HOB, HPC-PL, HPIEpC, HSaVEC, HVMF, HWP, IMR90, Monocytes-CD14+, NHDF, NHEM.f_M2, NHEM_M2, SkMC, and SK-N-SH. The downloadable elements files (Transcripts, Genes and Exons) in this release were generated using GENCODE V10, while the older datasets were generated using GENCODE V7.</p>
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSydhHistone">
Histone Modifications by ChIP-seq from ENCODE/Stanford/Yale/USC/Harvard (Release 3)</a>:
adds 9 new experiments for the MCF-7, HCT-116 and PANC-1 cell lines.</p>
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDnase">
DNaseI Hypersensitivity by Digital DNaseI from ENCODE/University of Washington (Release 5)</a>:
adds 11 experiments across 9 new cell lines.</p>

<hr>
<!-- end news item -->

<!-- start news item -->
<p>
<font size="3"><b> 25 Jun 2012 -  Mouse ENCODE data releases: Stan/Yale RNA-seq, Stan/Yale TFBS (Release 3)
: 
</b></font>
<p>
One new track and one track update were released on the mm9 genome browser:</p>
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=mm9&g=wgEncodeSydhRnaSeq">
RNA-seq from ENCODE/Stanford</a>:
This track shows enrichment of RNA sequence tags mapped to the mouse genome generated by high  throughput sequencing in CH12, MEL, and ES cells.</p>
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=mm9&g=wgEncodeSydhTfbs">
Transcription Factor Binding Sites by ChIP-seq from ENCODE/Stanford/Yale (Release 3)</a>:
adds 17 new experiments, for a total of 54 experiments, studying 26 factors in CH12, MEL and ES cells.</p>
<hr>
<!-- end news item -->

<!-- start news item -->
<p>
<font size="3"><b> 15 Jun 2012 - Mouse Experiment Matrix web tool
</b></font>
<p>
The ENCODE DCC at UCSC has released an 'Experiment Matrix' for Mouse ENCODE data on the portal.
The three web pages in this application
(<a class="euro" target=_blank href="dataMatrix/encodeDataMatrixMouse.html">Experiment Matrix</a>,
<a class="euro" target=_blank href="dataMatrix/encodeDataSummaryMouse.html">Experiment Summary</a>, and
<a class="euro" target=_blank href="dataMatrix/encodeChipMatrixMouse.html">ChIP-seq Experiment Matrix</a>)
provide an up-to-date view of the publicly 
available Mouse ENCODE data, along with an interface for selecting 
experiments for viewing in the browser or downloading as files for 
analysis. Each page has a file/track selector.  Clicking an experiment 
item produces a search window with resulting files or tracks listed.</p>
<p>
The pages are best viewed in Firefox, Chrome, and Safari browsers, and 
zoomed out as far as readability on your screen allows.</p>
<hr>
<!-- end news item -->

<!-- start news item -->
<p>
<font size="3"><b> 31 May 2012 -  Human ENCODE data releases: GIS ChIA-PET, SYDH TFBS (Release 2), RIKEN CAGE Loc (Release 3)

</b></font>
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGisChiaPet">
Chromatin Interaction Analysis Paired-End Tags (ChIA-PET) from ENCODE/GIS-Ruan</a>:
<p>
This track shows the locations of protein factor mediated chromatin interactions determined
by ChIA-PET data in five different human cancer cell lines  (K562 (chronic myeloid leukemia),
HCT116 (colorectal cancer), HeLa-S3 (cervical cancer), MCF-7 (breast cancer),
and NB4 (promyelocytic)).  The protein factors displayed in the track include
estrogen receptor alpha, RNA polII, and CTCF.</p>
<p>
Chromatin interaction analysis with paired-end tag sequencing (ChIA-PET) is a global 
de novo high-throughput method for characterizing the 3-dimensional structure of chromatin 
in the nucleus.  A chromatin interaction is defined as the association of two regions 
of the genome that are far apart in terms of genomic distance, but are spatially 
proximate to each other in the 3-dimensional cellular nucleus.</p> 

<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSydhTfb">  
Transcription Factor Binding Sites by ChIP-seq from ENCODE/Stanford/Yale/USC/Harvard (Release 2)</a>:
contains 79 new experiments.</p>

<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRikenCage">  
RNA Subcellular CAGE Localization from ENCODE/RIKEN (Release 3)</a>:
includes viewable Transcription Start Sites determined by Hidden Markov Models. It contains the following new cell lines and tissues: CD20+, CD34+ Mobilized, HAoAF, HAoEC, HCH, HFDPC, HMEpC, hMSC-AT, hMSC-BM, hMSC-UC, HOB, HPC-PL, HPIEpC, HSaVEC, HVMF, HWP, IMR90, Monocytes-CD14+, NHDF, NHEM.f_M2, NHEM_M2, SkMC, and SK-N-SH.</p>

<hr>
<!-- end news item -->

<!-- start news item -->
<p>
<font size="3"><b> 25 May 2012 -  Mouse ENCODE data releases: Caltech RNA-seq, PSU Histone, LICR RNA-seq (Release 2)
</b></font>
<p>
Two new tracks and one track update were released on the mm9 genome browser:
<a href="newsarch.html#052512">Read more</a>.</p>
<br>
<!-- end news item -->


<!-- end archived item section--> </td>
                              <td width="15"><br>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <br>
<!--outer table is for border purposes-->
            <table bgcolor="#888888" border="0" cellpadding="1"
 cellspacing="0" width="100%">
              <tbody>
                <tr>
                  <td>
                  <table bgcolor="#FFFFFF" border="0" cellpadding="0"
 cellspacing="0" width="100%">
                    <tbody>
                      <tr>
                        <td>
                        <table background="../images/hr.gif"
 bgcolor="#FFCC00" border="0" width="100%">
                          <tbody>
                            <tr>
                              <td><a name="Conditions"></a><font
 size="4"><b>&nbsp; Conditions of Use</b></font> </td>
                            </tr>
                          </tbody>
                        </table>
                        <table bgcolor="#FFFFFF" border="0"
 cellpadding="0" width="100%">
                          <tbody>
                            <tr>
                              <th height="3"><br>
                              </th>
                              <td><br>
                              </td>
                              <td><br>
                              </td>
                            </tr>
                            <tr>
                              <td width="10"><br>
                              </td>
                              <td>
                              <p>The sequence and annotation data
displayed in the Genome Browser are freely available for academic,
nonprofit, and personal use with the following conditions: </p>
                              <ul>
                                <li>ENCODE data are covered by the <a
 href="terms.html">ENCODE Consortium Data Release Policy</a>. </li>
                                <li>The general <a
 href="../#Conditions">Conditions of Use</a> for
the UCSC Genome Browser apply. </li>
                              </ul>
                              </td>
                              <td width="15"><br>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <br>
            </td>
          </tr>
        </tbody>
      </table>
      </td>
    </tr>
  </tbody>
</table>
</body>
</html>
