<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
  <HEAD>
    <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
    <META name="description" content="ENCODE Project at UCSC">
    <META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
    <TITLE>ENCODE Project at UCSC</TITLE>
    <LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
  </HEAD>

  <BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
  <A NAME="TOP"></A>

    <!-- Header Images and links -->
    <TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
      <TR>
        <TH COLSPAN=2 ALIGN="left">&nbsp;&nbsp;
          <A HREF="http://www.genome.gov/10005107" TARGET=_blank><IMG SRC="/images/encodelogo.gif" height=50 ALT="ENCODE Project at NHGRI"></A>
          &nbsp;&nbsp;
          <IMG SRC="/images/encode.jpg" ALT="ENCODE Project at UCSC">
        </TH>
      </TR>

      <TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/encode.home.topbar.html"--></TD></TR>

    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="/images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="encode.html">
		   Regions</A><HR>
		<A class="leftbar" HREF="contributors.html">
		   Contributors</a><HR>
	        <A CLASS="leftbar" HREF="submission.html">
		   Submission</A><HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/goldenPath/encode/">
		   Downloads</A><HR>
                <A CLASS="leftbar" HREF="releaseLog.html">
                    Release Log</a> <HR>
                <A CLASS="leftbar" HREF="trackStatus.html">
                    Data Status</a> <HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgLiftOver">
		   Tools</A><HR>
	        <A CLASS="leftbar" HREF="terms.html">
		   Terms of Use</a><HR>
<!--                <A CLASS="leftbar" HREF="releaseLog.html">  Release Log</A><HR>-->
		<BR><BR>&nbsp<HR>
                <A CLASS="leftbar" HREF="/cgi-bin/hgGateway">
		   Genome Browser</A><HR>
 	        <BR>
	        <IMG SRC="/images/DOT.gif" ALT="" HEIGHT=970 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Content Tables-------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	

	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; 
		    About the ENCODE Project</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                   <TR><TH HEIGHT=3></TH></TR>
                   <TR><TD WIDTH=10></TD><TD>
	             <P> This site contains information related to the 
		     <A HREF="http://www.genome.gov/10005107" 
		     TARGET=_blank>ENCODE project</A> at 
		     <A HREF="http://www.genome.gov/" 
		     TARGET=_blank>NHGRI</A>. The UCSC Genome Bioinformatics 
		     Group manages the official repository of 
		     sequence-related data for the ENCODE consortium and 
		     supports the coordination of data submission, storage, 
		     retrieval, and visualization.  
                     A summary of the status of datasets submitted to UCSC 
                     by ENCODE contributors is available on the 
                     <A TARGET=_BLANK HREF="trackStatus.html">
                     ENCODE data status page</A>.
                     <P>
                     UCSC also has a special 
		     interest in comparative genomics, and we provide 
		     resources for the ENCODE multiple sequence alignment 
		     interest group. <A HREF="http://www.ensembl.org" 
		     TARGET=_blank> Ensembl</A> also provides an 
		     <A HREF="http://www.ensembl.org/Homo_sapiens/encode.html"
		     TARGET=_blank> ENCODE resource page</A>.</P>
		     <P>
		     We'd like to thank NHGRI and the contributors of 
		     annotations and analyses to this project. The team at 
		     UCSC that develops and maintains this ENCODE site is 
		     made up of <A HREF="http://www.soe.ucsc.edu/~daryl" 
		     TARGET=_blank>Daryl Thomas</A>, 
		     <A HREF="mailto:&#107;&#97;t&#101;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;u">Kate Rosenbloom</A>, 
		     <A HREF="http://www.soe.ucsc.edu/~kent/" 
		     TARGET=_blank>Jim Kent</A>, and the 
		     <A HREF="../staff.html">UCSC Genome Bioinformatics 
		     staff</A>. <A HREF="about.html">Read more.</A></P>
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%" BORDER=0>
                  <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                      <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT>
 		    	<TD ALIGN="right"><A HREF="newsarch.html"><IMG SRC="/images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>

		      </TD><TD ALIGN="right">&nbsp;</TD></TR>
		    </TABLE>
	          </TD></TR>
		</TABLE>

		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
    		  <TR><TH HEIGHT=3></TH></TR>
    		  <TR><TD WIDTH=10></TD><TD>
<!-- start news item -->
		    <FONT FACE="courier" SIZE=3><B>15 June 2005 - Six data sets released in Genome Browser</B></FONT><BR>
		    <P>
		    Six more ENCODE annotation tracks have been added to the
		    Genome Browser this week:</P>
		    <P>
		    <B>NHGRI DNaseI-Hypersensitive Sites (update): </B>The 
		    NHGRI DNaseI-HS track has been updated with new data. The
		    track now includes DNaseI-hypersensitive sites in CD4+ 
		    T-cells before and after activation by anti-CD3 and 
		    anti-CD28 antibodies. Thanks to Greg Crawford at the 
		    Collins lab (NHGRI) for providing these data.</P>
		    <P>
		    <B>Genome Institute of Singapore PET of PolyA+ RNA: </B>
		    The GIS PET RNA track displays starts and ends of mRNA 
		    transcripts determined by paired-end ditag sequencing in 
		    two cell lines, MCF7 and HCT116 treated with 5 
		    fluoro-uracil. A total of 584,624 PETs were generated for 
		    MCF7 and 280,340 were generated for HCT116. More than
		    80% of the PETs in each group were mapped to the genome.
		    Thanks to Atif Shahab, Yijun Ruan, the GIS, and the 
		    Bioinformatics Institute of Singapore for providing these 
		    data.</P>
		    <P>
		    <B>Gencode Gene Annotations and Intron Validation: </B>
		    The Gencode Genes track displays high-quality manual 
		    annotations in the ENCODE regions generated by the 
		    GENCODE project.  A companion track, Gencode Introns, 
		    shows experimental gene structure validations for these
		    annotations.  Thanks to the HAVANA team at the Wellcome
		    Trust Sanger Institute; France Denoeud, Julien Lagarde,
		    and Roderic Guigo at the IMIM; and Alexandre Reymond
		    at the University of Geneva for providing the
		    annotations and experimental confirmation, as well as  
		    working with UCSC to develop the track display.</P>
		    <P>
		    <B>Boston University Hydroxyl Radical Cleavage: </B>
		    The BU ORChID track displays predicted hydroxyl radical 
		    cleavage intensity on naked DNA for each nucleotide in 
		    the ENCODE regions.  The prediction algorithm draws data 
		    from a database of 150 experimentally-determined cleavage 
		    patterns. Thanks to Jay Greenbaum at the Tullius lab for 
		    providing these data.</P>
		    <P>
		    <B>UC Davis ChIP/Chip: </B> The UCD Ng E2F1 track shows 
		    ChIP analysis of HeLa cells using E2F1 antibody as 
		    assayed on Nimblegen arrays. The E2F1 factor is associated
		    with cell cycle control, transcriptional regulation, and 
		    apoptosis. Thanks to the Farnham lab and Kyle Munn, Todd 
		    Richmond and Roland Green of Nimblegen for providing 
		    these data.</P>
		    <P>
		    <B>Ludwig Institute/UCSD ChIP/Chip: </B>
		    Three tracks of ChIP/chip data using Nimblegen arrays 
		    have been released: LI Ng TAF1 IMR90, LI Ng Validation,
     		    and LI Ng gIF ChIP. The TAF1 track shows genome-wide 
		    TAF1 binding sites as determined by ChIP in IMR90 
		    (fibroblast) cells assayed on Nimblegen high-density 
		    oligo arrays. The peaks from the genome scan experiments 
		    were verified using four antibodies associated with 
		    transcription start (Pol2, H3ac, H3K4me3, and TAF1) on 
		    condensed arrays covering the putative TAF1 binding sites.
		    The gIF track shows the results of ChIP/chip experiments 
		    using gamma interferon-treated HeLa cells, with Pol2 and 
		    H3K4me3 antibodies.</P>
		    <P>
		    Two additional tracks, LI ChIP Various and LI gIF ChIP, 
 		    display ChIP/chip data assayed on Ren lab ENCODE PCR
		    tiling arrays for five antibodies (Pol2, TAF1, H3K4me2, 
		    SUZ12, H3K27me3) and four cell lines (HeLa, THP1, IMR90, 
		    HCT116), as well as gamma interferon experiments using 
		    seven antibodies (Pol2, TAF1, H3K4me2, H3K4me3, H3ac, 
		    H4ac, and STAT1) in HeLa cells.
		    Thanks to Chunxu Qu and Bing Ren, at the Ren lab for 
		    providing these data and assisting with track display 
		    issues.</P>
		    <P>
		    We'd also like to acknowledge the UCSC team members who 
		    developed these tracks: Kate Rosenbloom, Hiram Clawson, 
		    and Angie Hinrichs for track development; Bob Kuhn, Ali
		    Sultan-Qurraie and Galt Barber for quality assurance; 
		    and Donna Karolchik and Jim Kent for documentation.</P>
		    <P>
		    <HR>

<!-- start archived item section-->
		    <FONT FACE="courier" SIZE=3><B>9 June 2005 - Sanger Histone Modification ChIP/chip track updated: </B></FONT>
		    The Sanger Institute has submitted ChIP/chip data for 
		    additional antibodies and cell lines, which we have 
		    incorporated into the existing Sanger ChIP browser track. 
		    <A HREF="newsarch.html#060905">Read more</A>. </P>

		    <FONT FACE="courier" SIZE=3><B>6 June 2005 - Changes to ENCODE track labels in Genome Browser: </B></FONT>
		    We have modified the Genome Browser labels for the 
		    existing ENCODE data tracks to trim overly-long labels
		    that were truncated in the display and to facilitate 
		    cross-track analysis. 
		    <A HREF="newsarch.html#060605b">Read more</A>. </P>
		    <P>
		    <FONT FACE="courier" SIZE=3><B>6 June 2005 - Stanford Promoters and UVa DNA Replication tracks released: </B></FONT>
		    We are pleased to announce the first UCSC Genome Browser 
		    tracks released for the June 2005 ENCODE data freeze:
		    Stanford Promoters and UVa DNA Replication Temporal 
		    Profiling. 
		    <A HREF="newsarch.html#060605a">Read more</A>.</P>

<!-- end archived item section--> 



    	          </TD><TD WIDTH=15></TD></TR>
	        </TABLE>
	      <BR></TD></TR>
	    </TABLE>
          </TD></TR>
        </TABLE>
        <BR>
	    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%">
		  <TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	          </TD></TR>
		</TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome 
		    Browser are freely available for academic, nonprofit, and 
		    personal use with the following conditions:
		    <UL>
		      <LI>
		      The general 
		      <A HREF="http://genome.ucsc.edu/#Conditions">Conditions of Use</A> 
		      for the UCSC Genome Browser apply.
		      <LI>
		      The ENCODE-specific conditions of use are
		      still being developed and will be displayed here
		      in the future.  Until then, <A
		      HREF="mailto:genome@soe.ucsc.edu">please contact
		      us</A> for the conditions on the use of this data.
		    </UL>
	          </TD><TD WIDTH=15></TD></TR>
		</TABLE>
	      <BR></TD></TR>
	    </TABLE>
          </TD></TR>
	</TABLE>
      <BR></TD></TR>
    </TABLE>
  </TD></TR>
</TABLE>
</BODY></HTML>

