<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
  <meta http-equiv="Content-Type"
 content="text/html; charset=ISO-8859-1">
  <meta name="description" content="ENCODE Project at UCSC">
  <meta name="keywords"
 content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
  <title>ENCODE Project at UCSC</title>
  <link rel="STYLESHEET" href="../style/HGStyle.css">
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
</head>
<body style="background-color: rgb(255, 249, 210);" alink="#6600ff"
 link="#0000cc" vlink="#330066">
<a name="TOP"></a><!-- Header Images and links -->
<table border="0" cellpadding="0" cellspacing="0" width="100%">
  <tbody>
    <tr>
      <th colspan="2" align="left">&nbsp;&nbsp; <a
 href="http://www.genome.gov/10005107" target="_BLANK"><img
 src="../images/ENCODE_scaleup_logo.png" alt="ENCODE Project at NHGRI"
 height="50"></a> &nbsp;&nbsp; <a href="/encode/"><img
 src="../images/encodeDcc.jpg"
 alt="ENCODE Data Coordination Center at UCSC"></a> </th>
    </tr>
    <tr>
      <td colspan="2" height="40"><!--#include virtual="../inc/encode.home.topbar.html"-->
      <br>
      </td>
    </tr>
    <tr height="100%" valign="top">
      <td rowspan="1" height="100%" valign="top"><!-- outer table is to align leftbar & content boxes at top of page -->
      <table border="0" cellpadding="0" cellspacing="0" width="100%">
        <tbody>
          <tr>
            <td><img src="../images/DOT.gif" alt="" height="7"></td>
          </tr>
          <tr height="100%">
            <td height="100%"><!--outer table is for black border-->
            <table bgcolor="#000000" border="0" cellpadding="1"
 cellspacing="0" width="100%">
              <tbody>
                <tr>
                  <td>
                  <table border="0" cellpadding="10" cellspacing="0"
 width="100%">
                    <tbody>
                      <tr height="100%">
                        <td bgcolor="#738ed3" height="100%" valign="top">
			<font color="#FFFFAA"><b><em>Human</em></b></font><br>
                        <img src="../images/DOT.gif" alt="" height="8" width="0"><br>
			<a class="leftbar" href="/cgi-bin/hgTracks?db=hg18">Genome Browser</a>
                        <hr> <a class="leftbar" href="cellTypes.html">Cell Types</a>
                        <hr> <a class="leftbar" href="downloads.html">Downloads</a>
			<hr> <a class="leftbar" href="http://genomewiki.cse.ucsc.edu/index.php/Genome_Browser_Session_Gallery"
TARGET=_blank>Session Gallery</a>
                        <hr>
			<img src="../images/DOT.gif" alt="" height="5" width="0"><br>
			<font color="#FFFFAA"><b><em>Mouse</em></b></font><br>
			<img src="../images/DOT.gif" alt="" height="8" width="0"><br>
			<!-- <a class="leftbar" href="/cgi-bin/hgTracks?db=mm9">Genome Browser</a> -->
			<!-- <hr> --><a class="leftbar" href="cellTypesMouse.html">Cell Types</a>
                        <!-- <hr> <a class="leftbar" href="downloadsMouse.html">Downloads</a> -->
			<hr>
                        <img src="../images/DOT.gif" alt="" height="5" width="0"><br>
			<font color="FFFFAA"><b><em>General</em></b></font><br>
                        <img src="../images/DOT.gif" alt="" height="8"><br>
			<a class="leftbar" href="antibodies.html">Antibodies</a>
			<hr> <a class="leftbar" href="otherTerms.html">Registered Variables</a>
			<hr> <a class="leftbar" href="releaseLog.html">Release Log</a>
                        <hr> <a class="leftbar" href="terms.html">Data Policy</a>
			<!-- <hr> <a class="leftbar" href="dataStandards.html">Data Standards</a> -->
                        <hr> <a class="leftbar" href="contributors.html">Contributors</a>
		    	<hr> <a class="leftbar" href="pubs.html">Publications</a>
                        <hr> <a class="leftbar" href="pilot.html">Pilot Project</a>
                        <hr> <a class="leftbar" href="http://www.cbse.ucsc.edu/jobs/" target="_blank">Jobs</a>
                        <hr> <a class="leftbar" href="contacts.html">Contact Us</a>
                        <hr> <br>
                        <img src="../images/DOT.gif" alt="" height="970" width="1"> </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            </td>
          </tr>
        </tbody>
      </table>
      </td>
<!-- Content Tables --> <td valign="top">
      <table border="0" cellpadding="5">
        <tbody>
          <tr>
            <td><!--outer table is for border purposes-->
            <table bgcolor="#888888" border="0" cellpadding="1"
 cellspacing="0" width="100%">
              <tbody>
                <tr>
                  <td>
                  <table bgcolor="#fffee8" border="0" cellpadding="0"
 cellspacing="0" width="100%">
                    <tbody>
                      <tr>
                        <td>
                        <table background="../images/hr.gif"
 bgcolor="#d9e4f8" border="0" width="100%">
                          <tbody>
                            <tr>
                              <td><a name="About"></a><font size="4"><b>&nbsp;
About
the
ENCODE
Data Coordination Center (DCC)</b></font> </td>
                            </tr>
                          </tbody>
                        </table>
                        <table bgcolor="#fffee8" border="0"
 cellpadding="0" width="100%">
                          <tbody>
                            <tr>
                              <th height="3"><br>
                              </th>
                              <td><br>
                              </td>
                              <td><br>
                              </td>
                            </tr>
                            <tr>
                              <td width="10"><br>
                              </td>
                              <td>
                              <p>The <a
 href="http://www.genome.gov/10005107" target="_blank">Encyclopedia of
DNA Elements</a> (ENCODE) Consortium is an international collaboration
of research groups funded by the National Human Genome Research
Institute (<a href="http://www.genome.gov/" target="_blank">NHGRI</a>).
The goal of the consortium is to build a comprehensive parts list of
the functional elements of the human genome, including elements that
act at the protein level (coding genes) and RNA level (non-coding
genes), and regulatory elements that control the cells and
circumstances in which a gene is active. The discovery and annotation
of gene elements is accomplished primarily by sequencing RNA from a
diverse range of sources, comparative genomics, integrative
bioinformatic methods, and human curation. Regulatory elements are
typically investigated through DNA hypersensitivity assays, assays of
DNA methylation, and chromatin immunoprecipitation (ChIP) of proteins
that interact with DNA, including modified histones and transcription
factors, followed by sequencing (ChIP-Seq). The results of ENCODE
experiments, collected in the ENCODE DCC database, are displayed on the
UCSC Genome Browser. The data can also be downloaded from the ENCODE
DCC website in text format. </p>
                              <p>ENCODE data is now available for the
entire human genome.&nbsp; <b>Mouse ENCODE experiments are currently
underway, and data on the mouse genome from such experiments will be
made available as soon as possible.</b> To access the human ENCODE data, open the
<a href="../cgi-bin/hgGateway?org=human">Genome Browser</a>, select the
March 2006 assembly of the human genome, and go to your region of
interest. ENCODE tracks will be marked with the NHGRI logo <img
 src="../images/nhgri.gif" alt="" height="16" width="16">. The bulk of
the ENCODE data can be found in the <em>Expression</em> and <em>Regulation</em>
track groups, with a few in the <em>Mapping</em>, <em>Genes</em>, and
                              <em>Variation</em> groups. Although most
participating research groups have provided several tracks, generally
only selected data from each research group are displayed by default.
Click the hyperlinked name of a particular track to display a page
containing configuration options and details about the methods used to
generate the data. See the Genome Browser <a
 href="../goldenPath/help/hgTracksHelp.html">User's Guide</a> for
further information about displaying tracks and navigating in the
Genome Browser. To receive notifications of ENCODE data releases and
related news by email, subscribe to the <a
 href="https://lists.soe.ucsc.edu/mailman/listinfo/encode-announce"
 target="_blank">encode-announce</a> mailing list.</p>
                              <p>Data from the earlier ENCODE project
pilot phase, which covered approximately 1% of the genome,&nbsp; are
available on the May 2004 and March 2006 human
assemblies. The <a href="pilot.html">ENCODE Pilot Project</a> web
pages provide convenient browser access to these regions. </p>
                              <p>Before publishing research that uses
ENCODE data, please read the <a href="terms.html">data release policy</a>,
which
places
some
restrictions on publication use of data for nine
months following the data release.<br>
                              </p>
                              </td>
                              <td width="15"><br>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        <br>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <br>
<!--outer table is for border purposes-->
            <table bgcolor="#888888" border="0" cellpadding="1"
 cellspacing="0" width="100%">
              <tbody>
                <tr>
                  <td>
                  <table bgcolor="#fffee8" border="0" cellpadding="0"
 cellspacing="0" width="100%">
                    <tbody>
                      <tr>
                        <td>
                        <table background="../images/hr.gif"
 bgcolor="#d9e4f8" border="0" width="100%">
                          <tbody>
                            <tr>
                              <td><!--cheap trick to get background on row to be continuous-->
                              <table border="0" cellpadding="0"
 cellspacing="0" width="100%">
                                <tbody>
                                  <tr>
                                    <td align="left"><font size="4"><a
 name="TOC"></a><b>&nbsp; News</b></font> </td>
                                    <td align="right"><a
 href="newsarch.html"><img src="../images/news.jpg" alt="" align="right"
 border="0"></a> </td>
                                    <td align="right">&nbsp;</td>
                                  </tr>
                                </tbody>
                              </table>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        <table bgcolor="#fffee8" border="0"
 cellpadding="0" width="100%">
                          <tbody>
                            <tr>
                              <th height="3"><br>
                              </th>
                              <td><br>
                              </td>
                              <td><br>
                              </td>
                            </tr>
                            <tr>
                              <td width="10"><br>
                              </td>
                              <td
 style="font-family: Times New Roman,Times,serif;"><!-- start news item -->
                              <font size="3"><b>6 August 2010 - June and July ENCODE news</b></font><br>
                              <br>
				<b>Initial Release of the
				<a href="">HudsonAlpha RNA-seq</a> track:</b> This track shows short
				tag sequencing of cDNA obtained from biological replicate samples
				(different culture plates) of the ENCODE cell lines. The sequences
				were aligned to the human genome (hg18) and UCSC known-gene splice
				junctions.<br>
				<br>
				<b>Release 2 of the 
				<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeCaltechRnaSeq">Caltech RNA-seq</a>
				track: </b></font> This track shows alignments, signal density, and
				splice sites based on 75 bp paired reads and 32 bp strand-specific
				single reads of polyA+ RNA aligned to the human genome (hg18) and
				UCSC known-gene splice junctions. Also included with the track as
				downloadable files are RPKM expression level measurements at the
				gene-level and exon-level, and candidate novel exons.  Release 2
				of this track adds five new cell types: H1-hESC, HeLa-S3, HepG2,
				HUVEC, and NHEK.<br>
                              	<br>
				<b>Initial Release of the 
				<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeGisPet">GIS PET Loc</a>
				track: </b></font> This track shows starts and ends of full length
				mRNA transcripts determined by PET sequencing of polyA+ and total
				RNA from 6 subcellular compartments and whole cell, in 8 cell lines.<br>
				<br>
				<b>Release 2 of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaChipSeq">HAIB TFBS</a>
				track: </b></font> This track shows signal density and binding sites
				of selected transcription factors in a variety of cell types.
				Release 2 of this track adds 73 new experiments covering 13 new cell
				lines and 27 antibodies. Additionally, DEX and EtOH treatments have
				been included in the A549 cell line.<br>
				<br>
				<b>Initial Release of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeBuOrchid">BU ORChID</a>
				track: </b></font> This track displays the predicted hydroxyl
				radical cleavage intensity on naked DNA for each nucleotide in the
				genome.<br>
				<br>
				<b>Update of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeMapability">Mapability</a>
				track: </b></font> This track displays the level of sequence
				uniqueness of the reference hg18 genome. The update adds CRG
				Alignability data, which displays how uniquely k-mer sequences align
				to a region of the genome.<br>
				<br>
				<hr>
                              <font size="3"><b>19 July 2010 - April and May ENCODE news</b></font><br>
                              <br>
				<b>Initial Release of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeGisRnaSeq">GIS RNA-seq</a>
				track: </b></font> This track shows RNA-seq of high-quality PolyA+
				RNA in ENCODE Tier1 cell lines and H1 ESC sequenced on the ABI Solid
				platform.<br>
                              	<br>
				<b>Release 3 of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeYaleChIPseq">Yale TFBS</a>
				track: </b></font> This track shows probable binding sites of the
				specified transcription factors (TFs) in the given cell types as
				determined by chromatin immunoprecipitation followed by high
				throughput sequencing (ChIP-Seq). Release 3 adds 64 experiments
				and 26 input/control datasets for a total of 54 factors in 21 cell lines.<br>
                              	<br>
				<b>Release 2 of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeRikenCage">RIKEN CAGE Loc</a>
				track: </b></font> This track shows 5' cap analysis gene expression
				(CAGE) tags and clusters in RNA extracts from different sub-cellular
				localizations. Release 2 of this track adds data for eight new
				cell-type/compartment combinations (GM12878 Nucleus, H1-hESC whole
				cell, HepG2 cytosol/nucleus/nucleolus, HUVEC cytosol, and NHEK
				cytosol/nucleus).<br>
                              	<br>
				<b>Initial Release of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeDukeAffyExonArray">Duke Affy Exon</a>
				track: </b></font> This track shows gene expression by microarray of
				RNA extracted from 28 cell lines that were also analyzed by DNaseI
				hypersensitivity, FAIRE, and ChIP assays (Open Chromatin track).<br>
                              	<br>
				<b>Initial Release of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeUwDGF">UW DNase DGF</a>
				track: </b></font> This track shows high-resolution DNase
				annotations from samples sequenced to depths of 300-fold or greater,
				in 5 cell lines.<br>
                              	<br>
				<b>Initial Release of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeGisDnaPet">GIS DNA PET</a>
				track: </b></font> This track shows the starts and ends of DNA
				fragments from different cell lines determined by paired-end ditag
				(PET) sequencing using different DNA fragment sizes for analysis of
				genome structural variation.<br>
			      	<br>
                              <hr>
                              <font size="3"><b>18 March 2010 - February and March 2010 ENCODE news</b></font><br>
                              <br>
				<b>Release 3 of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeChromatinMap">Open Chromatin</a>
				track: </b></font> This track displays evidence of open chromatin in
				multiple cell types from the Duke/UNC/UT-Austin/EBI ENCODE group.
				Release 3 of this track includes 18 new cell line or cell/treatment
				experiments. In addition, a number of new experiments were added to
				existing cell lines. Almost all Peaks have been called anew using
				improved cut-offs and p-Values. Finally, a second type of peak
				called using a ZINBA algorithm has been provided for several of the
				FAIRE-seq experiments.<br>
                              	<br>
				<b>Release 3 of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeBroadChipSeq">Broad Histone</a>
				track: </b></font> This track shows maps of chromatin state
				generated using CHIP-seq. Release 3 of this track adds the HSMM cell
				line and includes new experiments for H1-hESC and NHLF.<br>
			      <br>
                              <font size="3"><b>Release 2
of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeUwAffyExonArray">UW Affy Exon</a> track: </b></font>
                              <meta charset="utf-8"
 id="webkit-interchange-charset">
                              This track displays
human tissue microarray data using the 
<a href="http://www.affymetrix.com/estore/browse/products.jsp?productId=131452">Affymetrix Human Exon 1.0</a> GeneChip.  This release includes 28 new cell types, and replaces the data for four existing tables (replicate 1 for K562, NB4, and SKMC; replicate 2 for HeLa-S3).
<br>
                              <br>
                              <font size="3"><b>Initial release
of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeUwChIPSeq">UW Histone</a> track: </b></font>
                              This track displays
maps of histone modifications genome-wide in different cell lines, using ChIP-seq high-throughput sequencing.<br>
                              <br>
                              <font size="3"><b>Release 2 of the <a
 href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeHudsonalphaMethylSeq">HudsonAlpha
Methyl-seq</a> track: </b>
                         Release 2 adds data for five new cell
types.<br>
                              <font size="3"><b><br>
Release 3 of the <a
 href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeSangerGencode">Gencode
Genes</a> track:</b> </font>shows high-quality
manual annotations in the ENCODE regions generated by the 
<a href="http://www.sanger.ac.uk/PostGenomics/encode/" target="_blank">GENCODE
project</a>.<font size="3"><b> </b></font>
                              Version 3 of the
Gencode gene set presents a full merge between HAVANA and ENSEMBL,
giving priority to the manually curated Havana objects and using
ENSEMBL objects where they are different or fall into un-annotated
regions.<br>
                              <font size="3"><br>
                              <b>Initial release of the <a
 href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeCshlShortRnaSeq">CSHL
                              </a></b></font>
                              <a
 href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeCshlShortRnaSeq">Small
RNA-seq</a> track: This track depicts
NextGen sequencing information for RNAs between the sizes of 20-200 nt
isolated from RNA samples from tissues of sub cellular compartments
from ENCODE cell lines. <br>
                              <br>
<b>Release 3 of the <a
 href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeUwDnaseSeq">UW
DNaseI HS</a> track: </b></font>
                              This track shows
DNaseI sensitivity measured genome-wide in different using the Digital DNaseI
methodology, and DNaseI hypersensitive sites. This
release includes 19 new cell lines as
well as new version of NB4 replicate 1.<br>
<br>
                              <hr><!-- end news item --><!-- start archived item section-->
<!-- start of archived news item -->
<font size="3"><b>6 January 2010 - December 2009 ENCODE news</b>: During this period the
<b>&quot;ENCODE whole-genome data in the UCSC Genome Browser&quot;</b> paper was announced and four
ENCODE tracks were released (one initial release and three updates).
<a href="newsarch.html#010610">Read more</a>.</p>
<!-- end of archived news item -->
<!-- start of archived news item -->
<font size="3"><b>7 November 2009 - October ENCODE news</b>: During this period, the September 2009
data freeze was completed, the Affy RNA Loc track was expanded, the <em>encodeproject.org</em>
domain was registered, new grants were funded, and two new job openings were annouced.
<a href="newsarch.html#110709">Read more</a>.</p>
<!-- end of archived news item -->
<!-- end archived item section--> </td>
                              <td width="15"><br>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <br>
<!--outer table is for border purposes-->
            <table bgcolor="#888888" border="0" cellpadding="1"
 cellspacing="0" width="100%">
              <tbody>
                <tr>
                  <td>
                  <table bgcolor="#fffee8" border="0" cellpadding="0"
 cellspacing="0" width="100%">
                    <tbody>
                      <tr>
                        <td>
                        <table background="../images/hr.gif"
 bgcolor="#d9e4f8" border="0" width="100%">
                          <tbody>
                            <tr>
                              <td><a name="Conditions"></a><font
 size="4"><b>&nbsp; Conditions of Use</b></font> </td>
                            </tr>
                          </tbody>
                        </table>
                        <table bgcolor="#fffee8" border="0"
 cellpadding="0" width="100%">
                          <tbody>
                            <tr>
                              <th height="3"><br>
                              </th>
                              <td><br>
                              </td>
                              <td><br>
                              </td>
                            </tr>
                            <tr>
                              <td width="10"><br>
                              </td>
                              <td>
                              <p>The sequence and annotation data
displayed in the Genome Browser are freely available for academic,
nonprofit, and personal use with the following conditions: </p>
                              <ul>
                                <li>ENCODE data is covered by the <a
 href="terms.html">ENCODE Consortium Data Release Policy</a>. </li>
                                <li>The general <a
 href="http://genome.ucsc.edu/#Conditions">Conditions of Use</a> for
the UCSC Genome Browser apply. </li>
                              </ul>
                              </td>
                              <td width="15"><br>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <br>
            </td>
          </tr>
        </tbody>
      </table>
      </td>
    </tr>
  </tbody>
</table>
</body>
</html>
