<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
  <HEAD>
    <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
    <META name="description" content="ENCODE Project at UCSC">
    <META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
    <TITLE>ENCODE Project at UCSC</TITLE>
    <LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
  </HEAD>

  <BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
  <A NAME="TOP"></A>

    <!-- Header Images and links -->
    <TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
      <TR>
        <TH COLSPAN=2 ALIGN="left">&nbsp;&nbsp;
          <A HREF="http://www.genome.gov/10005107" TARGET=_blank><IMG SRC="/images/encodelogo.gif" height=50 ALT="ENCODE Project at NHGRI"></A>
          &nbsp;&nbsp;
          <IMG SRC="/images/encode.jpg" ALT="ENCODE Project at UCSC">
        </TH>
      </TR>

      <TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/encode.topbar.html"--></TD></TR>

    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="/images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="encode.html">
		   Regions</A><HR>
		<A class="leftbar" HREF="contributors.html">
		   Contributors</a><HR>
	        <A CLASS="leftbar" HREF="submission.html">
		   Submission</A><HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/goldenPath/encode/">
		   Downloads</A><HR>
                <A CLASS="leftbar" HREF="releaseLog.html">
                    Release Log </a> <HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgLiftOver">
		   Tools</A><HR>
	        <A CLASS="leftbar" HREF="terms.html">
		   Terms of Use</a><HR>
<!--                <A CLASS="leftbar" HREF="releaseLog.html">  Release Log</A><HR>-->
		<BR><BR>&nbsp<HR>
                <A CLASS="leftbar" HREF="/cgi-bin/hgGateway">
		   Genome Browser</A><HR>
 	        <BR>
	        <IMG SRC="/images/DOT.gif" ALT="" HEIGHT=970 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Content Tables-------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	

	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; 
		    About the ENCODE Project</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                   <TR><TH HEIGHT=3></TH></TR>
                   <TR><TD WIDTH=10></TD><TD>
	             <P> This site contains information related to the 
		     <A HREF="http://www.genome.gov/10005107" 
		     TARGET=_blank>ENCODE project</A> at 
		     <A HREF="http://www.genome.gov/" 
		     TARGET=_blank>NHGRI</A>. The UCSC Genome Bioinformatics 
		     Group manages the official repository of 
		     sequence-related data for the ENCODE consortium and 
		     supports the coordination of data submission, storage, 
		     retrieval, and visualization.  UCSC also has a special 
		     interest in comparative genomics, and we provide 
		     resources for the ENCODE multiple sequence alignment 
		     interest group. <A HREF="http://www.ensembl.org" 
		     TARGET=_blank> Ensembl</A> also provides an 
		     <A HREF="http://www.ensembl.org/Homo_sapiens/encode.html"
		     TARGET=_blank> ENCODE resource page</A>.</P>
		     <P>
		     We'd like to thank NHGRI and the contributors of 
		     annotations and analyses to this project. The team at 
		     UCSC that develops and maintains this ENCODE site is 
		     made up of <A HREF="http://www.soe.ucsc.edu/~daryl" 
		     TARGET=_blank>Daryl Thomas</A>, 
		     <A HREF="mailto:&#107;&#97;t&#101;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;u">Kate Rosenbloom</A>, 
		     <A HREF="http://www.soe.ucsc.edu/~kent/" 
		     TARGET=_blank>Jim Kent<A>, and the 
		     <A HREF="../staff.html">UCSC Genome Bioinformatics 
		     staff</A>. <A HREF="about.html">Read more.</A></P>
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%" BORDER=0>
                  <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                      <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT>
		      </TD><TD ALIGN="right">&nbsp;</TD></TR>
		    </TABLE>
	          </TD></TR>
		</TABLE>

<!-- start news item -->
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
    		  <TR><TH HEIGHT=3></TH></TR>
    		  <TR><TD WIDTH=10></TD><TD>
		    <FONT FACE="courier" SIZE=3><B>17 February 2005 -
ENCODE downloads released</B></FONT><BR>
		    <P>
		    The ENCODE download area has been reorganized and updated 
		    on our public download server.  The downloads access page 
		    is now:
		    <P>
		    &nbsp; &nbsp; &nbsp; &nbsp;
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/encode/"
		    TARGET=_blank>http://hgdownload.cse.ucsc.edu/goldenPath/encode/</A>
		    <P>
		    and the annotations are now located in the 
		    assembly-specific download directory, currently:
		    <P>
		    &nbsp; &nbsp; &nbsp; &nbsp;
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg16/encode/"
		    TARGET=_blank>http://hgdownload.cse.ucsc.edu/goldenPath/hg16/encode/</A>
		    <P>
		    Any web pages referencing the previous UCSC ENCODE 
		    downloads will need to be updated.  Please contact us if 
		    you have any difficulties.
		    <P>
		    <HR>
<!-- end news item -->

<!-- start news item -->
    		    <FONT FACE="courier" SIZE=3><B>2 February 2005 -
Affymetrix ChIp/Chip and PolyA RNA data released</B></FONT><BR>
		    <P>
		    A second set of ENCODE ChIp/Chip data is now available on 
		    the July 2003 human genome assembly:
<PRE>
    ChIp/Affy Pol2 Pval
    ChIp/Affy Pol2 Sites
    RNA/Affy Signal
    RNA/Affy Sites
</PRE>
		    These tracks show RNA Polymerase II precipitation and 
		    RNA abundance in retinoic acid-stimulated HL-60 cells at 
		    0, 2, 8, and 32 hours, as measured by Affymetrix tiling 
		    arrays in the non-repetitive ENCODE regions.  The 
		    <em>Pval</em> and <em>Signal</em> tracks show values 
		    for each tiled probe; the <em>Sites</em> tracks show 
		    contiguous regions of enrichment.
		    <P>
		    A new <em>composite</em> track display was developed to 
		    concisely display multiple data sets of similar types, a 
		    common feature of ENCODE data.  Each of these new tracks 
		    contains 4 subtracks, one for each time interval.  The 
		    subtracks share a single description page and set of 
		    visibility controls.  Checkboxes on the track 
		    configuration page allow selected subtracks to be hidden 
		    in the display.
		    <P>
		    These data were generated and analyzed by Tom Gingeras' 
		    group at 
		    <A HREF="http://www.affymetrix.com"
		    TARGET=_blank>Affymetrix</A> and  
		    <A HREF="http://www.hms.harvard.edu/dms/bbs/fac/struhl.html" 
		    TARGET=_blank>Kevin Struhl's group at Harvard Medical 
		    School</A>. We would like to thank Stefan Bekiranov at 
		    Affymetrix for submitting the data and working closely 
		    with us to clarify the experimental methods and 
		    verification descriptions.
		    <P>
		    <HR>

<!-- end news item -->
<!-- start news item -->

          	    <FONT FACE="courier" SIZE=3><B>5 November 2004 -
First ENCODE data tracks in the UCSC Browser</B></FONT><BR>
		    <P>
		    The first datasets submitted for the ENCODE project are 
		    now publicly available:
		    <UL>
		      <LI>
 		      ChIP/chip and transcription (Ludwig Institute/UCSD)
		      <LI>
		      Temporal profiling of DNA replication (University of 
		      Virginia)
		      <LI>
		      Promoter activity (Stanford)
		      <LI>
		      DNaseI hypersensitive sites (NHGRI)
		    </UL>
		    These tracks are visible in the ENCODE track group of the 
		    July 2003 (hg16) human genome assembly.  We would like to 
		    thank the labs of Bing Ren (LI/UCSD), Anindya Dutta (UVA),
		    Rick Myers (Stanford), and Francis Collins (NHGRI) for 
		    contributing the initial ENCODE data sets.  
		    <P>
                    <HR>

<!-- end news item -->
<!-- start news item -->

	            <FONT FACE="courier" SIZE=3><B>26 October 2004 -
		    Sequence Freeze For Multiple Alignments</B></FONT><BR>
		    <P>
		    We are pleased to release the first "official" sequence 
		    data freeze for the ENCODE multiple sequence alignment 
		    projects. The data formats are described in the 
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg16/encode/alignments/OCT-2004/README.txt" 
		    TARGET=_blank>README</A> file, and the sequences and 
		    supporting information is collected in the 
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg16/encode/alignments/OCT-2004" 
		    TARGET=_blank> data directory</A>.
		    <P>
		    The species included in this freeze are as follows:
<PRE>
       _SPECIES_      _SOURCE_
        Human           hg16
        Chimpanzee      panTro1
        Dog             canFam1
        Rat             rn3
        RatB	        BCM
        Mouse           mm5
        Chicken		galGal2
        Galago		NISC
        Baboon		NISC
        Marmoset        NISC
        Armadillo       NISC
        Platypus        NISC
</PRE>
		    You will notice that we have worked hard to include a 
		    number of species for which genome-wide sequence data was 
		    already available. The process by which these orthologous 
		    regions were identified is still an area of active 
		    research development, the details of which will be 
		    presented at the upcoming ENCODE meeting at CSHL.
		    <P>
		    Please note that we have also included a *second* rat 
		    sequence (ratB) in this freeze.  RatB represents an 
		    initiative to standardize the quality level of sequences 
		    in the ENCODE regions for species with genome-wide 
		    sequence data. The data were made available just before 
		    our freeze date, so we decided to include both versions of
		    the rat sequence for now. Eventually these sequences will 
		    likely be rolled into future genome assemblies.
		    <P>
		    Remember, this is a work in progress, so not all targets 
		    have sequence from all species. And some species/target 
		    combinations may not be complete yet. Progress on the 
		    NISC-generated sequences can be found at the 
		    <A HREF="http://www.nisc.nih.gov/open_page.html?/projects/encode/index.cgi" 
		    TARGET=_blank> NISC ENCODE Project: Comparative 
		    Sequencing</A>.
		    <P>
		    Many people have worked very hard to make these data 
		    available. Special thanks to Daryl Thomas, Kate 
		    Rosenbloom and the entire UCSC Team; Greg Schuler and the 
		    NCBI team; Colin Dewey and Lior Pachter; Pam Thomas and 
		    the NISC team; and David Wheeler and the BCM team.
		    <P>
		    Please send any comments to the 
		    <A HREF="mailto: <ENCODE_MSA-L@LIST.NIH.GOV>" 
		    TARGET=_blank>ENCODE MSA Mailing List</A>.
		    <P>
		    &nbsp; &nbsp; Elliott
		    <BR>
		    -- 
		    <BR>
		    Elliott H. Margulies, Ph.D.
		    Genome Technology Branch, NHGRI
		    <P>
                    <HR>

<!-- end news item -->
<!-- start news item -->

	            <FONT FACE="courier" SIZE=3><B>24 June 2004 -
		    ENCODE Project Portal Released</B></FONT><BR>
		    <P>
		    We are proud to announce the release of features
		    in the UCSC Genome Browser that are tailored to
		    the ENCODE project community, including this home
		    page to consolidate these resources.
		    <P>
		    The initial resources include sequences for the current 
		    human assemblies (hg16, hg15, hg13, and hg12), sequence of
		    the comparative species from 
		    <A HREF="http://www.nisc.nih.gov/" TARGET=_blank>NISC</A>,
		    tools for coordinate conversion between human
		    assemblies, format descriptions for data
		    submission, and contact information for help with
		    submitting annotation data and analyses.
		    <P>
		    Bulk downloads of the sequence and annotations may
		    be obtained from the ENCODE Project 
		    <!--<A HREF="ftp://genome-test.cse.ucsc.edu/encode/downloads/">FTP</A> or -->
		    <A HREF="http://hgdownload.cse.ucsc.edu/encode/">Downloads</A> 
		    page.  The sequences available here are repeat-masked 
		    versions of the GenBank records.
		    <P>
    	          </TD><TD WIDTH=15></TD></TR>
	        </TABLE>
	      <BR></TD></TR>
	    </TABLE>
          </TD></TR>
        </TABLE>
        <BR>
	    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%">
		  <TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	          </TD></TR>
		</TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome 
		    Browser are freely available for academic, nonprofit, and 
		    personal use with the following conditions:
		    <UL>
		      <LI>
		      The general 
		      <A HREF="http://genome.ucsc.edu/#Conditions">Conditions of Use</A> 
		      for the UCSC Genome Browser apply.
		      <LI>
		      The ENCODE-specific conditions of use are
		      still being developed and will be displayed here
		      in the future.  Until then, <A
		      HREF="mailto:genome@soe.ucsc.edu">please contact
		      us</A> for the conditions on the use of this data.
		    </UL>
	          </TD><TD WIDTH=15></TD></TR>
		</TABLE>
	      <BR></TD></TR>
	    </TABLE>
          </TD></TR>
	</TABLE>
      <BR></TD></TR>
    </TABLE>
  </TD></TR>
</TABLE>
</BODY></HTML>

