<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
  <HEAD>
    <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
    <META name="description" content="ENCODE Project at UCSC">
    <META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
    <TITLE>ENCODE Project at UCSC</TITLE>
    <LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
  </HEAD>

  <BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
  <A NAME="TOP"></A>

    <!-- Header Images and links -->
    <TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
      <TR>
        <TH COLSPAN=2 ALIGN="left">&nbsp;&nbsp;
          <A HREF="http://www.genome.gov/10005107" TARGET=_blank><IMG SRC="../images/encodelogo.gif" height=50 ALT="ENCODE Project at NHGRI"></A>
          &nbsp;&nbsp;
          <IMG SRC="../images/encode.jpg" ALT="ENCODE Project at UCSC">
        </TH>
      </TR>

      <TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/encode.home.topbar.html"--></TD></TR>

    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="../images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="encode.hg17.html">
		   Regions (hg17)</A><HR>
	        <A CLASS="leftbar" HREF="encode.hg16.html">
		   Regions (hg16)</A><HR>
                <A CLASS="leftbar" HREF="trackStatus.html">
                    Data Status</a> <HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/goldenPath/encode/index.html">
		   Downloads</A><HR>
	        <A CLASS="leftbar" HREF="submission.html">
		   Submission</A><HR>
	        <A CLASS="leftbar" HREF="../cgi-bin/hgLiftOver">
		   Tools</A><HR>
                <A CLASS="leftbar" HREF="releaseLog.html">
                    Release Log</a> <HR>
		<A class="leftbar" HREF="contributors.html">
		   Contributors</a><HR>
	        <A CLASS="leftbar" HREF="terms.html">
		   Terms of Use</a><HR>
		<BR>&nbsp<HR>
                <A CLASS="leftbar" HREF="../cgi-bin/hgGateway">
		   Genome Browser</A><HR>
                <A CLASS="leftbar" HREF="../contacts.html">
		   Contact Us</A><HR>
 	        <BR>
	        <IMG SRC="../images/DOT.gif" ALT="" HEIGHT=970 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Content Tables-------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	

	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; 
		    About the ENCODE Project</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
	            <P>
		    Welcome to the ENCODE Project portal sponsored by the
		    UCSC Genome Bioinformatics Group, a cross-departmental 
		    team within the Center for Biomolecular Science and 
		    Engineering (<A HREF="http://www.cbse.ucsc.edu/" 
		    TARGET=_blank>CBSE</A>) at the University of California 
		    Santa Cruz (<A HREF="http://www.ucsc.edu/public/" 
		    TARGET=_blank>UCSC</A>). 
		    This site contains information related to the 
		    <A HREF="http://www.genome.gov/10005107" 
		    TARGET=_blank>ENCODE</A> project at 
		    <A HREF="http://www.genome.gov/" TARGET=_blank>NHGRI</A>. 
		    UCSC manages the official repository of 
		    sequence-related data for the ENCODE Consortium and 
		    supports the coordination of data submission, storage, 
		    retrieval, and visualization. 
		    <A HREF="http://www.ensembl.org/Homo_sapiens/encode.html" 
		    TARGET=_blank>Ensembl</A>, a joint project of
		    EBI and the Wellcome Trust Sanger Institute, coordinates
		    much of the analysis effort. 
		    <!--
		    <A HREF="about.html">Read more</A>. </P> 
		    -->
		    <P>
		    To view the ENCODE data on the NCBI Build 35 human assembly
		    (May 2004), click the <A HREF="encode.hg17.html">Regions 
		    (hg17)</A> link in the left sidebar menu. For earlier data, 
		    click the <A HREF="encode.hg16.html">Regions (hg16)</A> 
		    link. Data will be available on the most recent NCBI Build
		    36 assembly (hg18, March 2006) in early summer 2007. </P>
                    <P>
                    The data from ENCODE contributors
                    that has been released by UCSC is listed on the 
                    <A HREF="releaseLog.html">release log 
		    page</A>. Primary data for ENCODE is available from the 
                    <A HREF="http://www.ncbi.nlm.nih.gov/projects/geo/info/ENCODE.html"
		    TARGET=_blank>NCBI GEO</A> and 
		    <A HREF="http://www.ebi.ac.uk/arrayexpress/"
		    TARGET=_blank>EBI ArrayExpress</A> public array data 
		    repositories.  Ensembl provides an 
		    <A HREF="http://www.ensembl.org/Homo_sapiens/encode.html"
                    TARGET=_blank>ENCODE resource page</A>, and NHGRI provides 
		    the <A HREF="http://research.nhgri.nih.gov/ENCODEdb"
                    TARGET=_blank>ENCODEdb</A> portal.</P>
		    <P>
		    We'd like to thank NHGRI and the contributors of 
		    annotations and analyses to this project. The team at 
		    UCSC that develops and maintains this website is 
		    made up of 
                    <A HREF="mailto:&#107;&#97;t&#101;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;u"
		    TARGET=_blank>Kate Rosenbloom</A>, 
                    <A HREF="mailto:&#104;&#97;rt&#101;r&#97;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;u"
		    TARGET=_blank>Rachel Harte</A>, 
                    <A HREF="mailto:t&#105;n&#103;w&#97;n&#103;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;u"
                    TARGET=_blank>Ting Wang</A>, 
                    <A HREF="http://www.soe.ucsc.edu/~kent/"
		    TARGET=_blank>Jim Kent</A>, and the 
		    <A HREF="../staff.html">UCSC Genome Bioinformatics 
		    staff</A>, with input from
                    <A HREF="http://www.soe.ucsc.edu/~daryl" 
		    TARGET=_blank>Daryl Thomas</A>.</P>
		    <P>
 		    To go to the main UCSC Genome Browser site
		    that provides access to sequence and annotation data
		    for a large collection of genome assemblies, click 
		    <A HREF="../index.html">here</A>. </P>
	          </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%" BORDER=0>
                  <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                      <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT>
 		    	<TD ALIGN="right"><A HREF="newsarch.html"><IMG SRC="../images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>

		      </TD><TD ALIGN="right">&nbsp;</TD></TR>
		    </TABLE>
	          </TD></TR>
		</TABLE>

		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
    		  <TR><TH HEIGHT=3></TH></TR>
    		  <TR><TD WIDTH=10></TD><TD>
		    <P>
<!-- start news item -->
		    <P>
		    <FONT FACE="courier" SIZE=3><B>13 Jun. 2007 - ENCODE Data Published in <em>Nature</em> and <em>Genome Research</em></B></FONT><BR>
		    <P>
		    The data and findings of the ENCODE project have been 
		    released to the public today, the culmination of a 
		    four-year effort to catalog the biologically functional 
		    elements in 1 percent of the human genome. The publications,
		    which include a group paper in the 14 June 2007 issue of 
		    <em>Nature</em> and 28 companion papers in the June 2007
		    issue of <em>Genome Research</em>, were authored by 
		    researchers from academic, governmental, and industry
		    organizations located in 11 countries. The <em>Nature</em> 
		    issue includes a pull-out poster featuring a screenshot of 
		    the UCSC Genome Browser displaying a broad range of the 
		    ENCODE data.</P>
		    <P>
		    In the press release accompanying the publication rollout,
		    NHGRI Director Francis S. Collins was quoted as saying
		    &quot;This impressive effort has uncovered many exciting 
		    surprises and blazed the way for future efforts to explore 
		    the functional landscape of the entire human genome. Because
		    of the hard work and keen insights of the ENCODE consortium,
		    the scientific community will need to rethink some long-held
		    views about what genes are and what they do, as well as how 
	   	    the genome's functional elements have evolved. This could 
		    have significant implications for efforts to identify the 
		    DNA sequences involved in many human diseases.&quot;</P>
		    <P>
		    The main portal for the ENCODE data is the UCSC ENCODE
		    Genome Browser. The analysis effort has been coordinated
		    from 
		    <A HREF="http://wwww.ensembl.org/Homo_sapiens?encode.html" 
		    TARGET=_blank>Ensembl</A>. Much of the primary data have 
		    been deposited in the 
		    <A HREF="http://www.ncbi.nlm.nih.gov/projects/geo/info/ENCODE.html"
		    TARGET=_blank>NCBI GEO</A> and 
		    <A HREF="http://www.ebi.ac.uk/arrayexpress/"
		    TARGET=_blank>EBI ArrayExpress</A> databases. 
		    For more detailed information on the ENCODE project, 
		    including the consortium's data release and accessibility 
		    policies and a list of NHGRI-funded participants, see the
		    <A HREF="http://www.genome.gov/ENCODE"
		    TARGET=_blank>NHGRI ENCODE website</A>.
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>12 Jun. 2007 - Spring 2007 ENCODE News</B></FONT><BR>
		    <P>
		    Between January and May of 2007, several new or upgraded 
		    data tracks were released by UCSC:
		    <UL>
		    <LI><B>Gencode March 2007 Genes</B> and <B>Gencode 
		    RACEfrags</B> -- 
		    Reannotation of 69 loci consisting of 132 transcripts
		    based on RACE, array, and sequencing analyses.  New features
		    include the addition of PolyA features, polymorphic
		    gene type, and integration of experimental intron 
		    validation.
		    <LI><B>Yale ChIP-chip RFBR</B> -- 
		    Analysis of ChIP-chip data to identify regulatory factor 
		    binding regions, resulting from over 105 experiments 
		    representing 29 transcription factors in 9 cells lines, 
		    performed in 7 labs using 3 microarray platforms. Data 
		    provided by the ENCODE Transcriptional Regulation analysis 
		    group.
		    <LI><B>University of Virginia DNA Replication Origins</B> --
		    Three new datasets representing two new experimental
		    methods for origin mapping (bubble trapping and nascent 
		    strand) in HeLa and GM06990 cells.
		    <LI><B>University of Vienna RNAz</B> --
		    RNA secondary structure prediction as predicted by RNAz,
		    based on evolutionary conservation and thermodynamic 
		    stability.
		    <LI><B>Duke DNaseI Hypersensitivity</B> -- 
		    DNase-chip results in IMR90 (fibroblast), K562 (leukemia) 
		    and H9 (stem) cells.
		    <LI><B>UCSC EvoFold</B> -- Replaces TBA23 EvoFold.
		    <LI><B>UC Davis ChIP Hits</B> -- Updated cMyc and E2F1.
		    </UL></P>
		    <P>
		    Thanks to the UCSC staff who worked on these tracks:
		    Rachel Harte and Kate Rosenbloom (development), and Ann 
		    Zweig, Archana Thakkapallayil, and Kayla Smith (quality 
		    review).</P>
<!-- end news item --> 
<!-- start archived item section-->
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>9 Jan. 2007 - Winter 2006 ENCODE News: </B></FONT>
		    To improve communication, we have posted instructions for 
		    our ENCODE ftp site on the ENCODE Wiki and have set up an 
		    email alias for notifying UCSC about your ENCODE data 
		    submissions: <A HREF="mailto:encode@soe.ucsc.edu">encode@soe.ucsc.edu</A>.
		    The current UCSC recipients are Kate Rosenbloom, Daryl 
		    Thomas, Ting Wang, and Rachel Harte.
		    During November and December, four new/improved data tracks 
		    were released. 
		    <A HREF="newsarch.html#010907">Read more</A>.</P>
<!-- end archived item section--> 
    	          </TD><TD WIDTH=15></TD></TR>
	        </TABLE>
	      </TD></TR>
	    </TABLE>
          </TD></TR>
        </TABLE>
	<BR>
	    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
		  <TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	          </TD></TR>
		</TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome 
		    Browser are freely available for academic, nonprofit, and 
		    personal use with the following conditions:
		    <UL>
		      <LI>
		      The general 
		      <A HREF="http://genome.ucsc.edu/#Conditions">Conditions of Use</A> 
		      for the UCSC Genome Browser apply.
		      <LI>
		      The ENCODE-specific conditions of use are
		      still being developed and will be displayed here
		      in the future.  Until then, <A
		      HREF="mailto:genome@soe.ucsc.edu">please contact
		      us</A> for the conditions on the use of this data.
		    </UL>
	          </TD><TD WIDTH=15></TD></TR>
		</TABLE>
	      </TD></TR>
	    </TABLE>
          </TD></TR>
	</TABLE>
      <BR></TD></TR>
    </TABLE>
  </TD></TR>
</TABLE>
</BODY></HTML>

