<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
  <HEAD>
    <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
    <META name="description" content="ENCODE Project at UCSC">
    <META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
    <TITLE>ENCODE Project at UCSC</TITLE>
    <LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
  </HEAD>

  <BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
  <A NAME="TOP"></A>

    <!-- Header Images and links -->
    <TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
      <TR>
        <TH COLSPAN=2 ALIGN="left">&nbsp;&nbsp;
          <A HREF="http://www.genome.gov/10005107" TARGET=_blank><IMG SRC="/images/encodelogo.gif" height=50 ALT="ENCODE Project at NHGRI"></A>
          &nbsp;&nbsp;
          <IMG SRC="/images/encode.jpg" ALT="ENCODE Project at UCSC">
        </TH>
      </TR>

      <TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/encode.home.topbar.html"--></TD></TR>

    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="/images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="encode.html">
		   Regions</A><HR>
		<A class="leftbar" HREF="contributors.html">
		   Contributors</a><HR>
	        <A CLASS="leftbar" HREF="submission.html">
		   Submission</A><HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/goldenPath/encode/">
		   Downloads</A><HR>
                <A CLASS="leftbar" HREF="releaseLog.html">
                    Release Log</a> <HR>
                <A CLASS="leftbar" HREF="trackStatus.html">
                    Data Status</a> <HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgLiftOver">
		   Tools</A><HR>
	        <A CLASS="leftbar" HREF="terms.html">
		   Terms of Use</a><HR>
<!--                <A CLASS="leftbar" HREF="releaseLog.html">  Release Log</A><HR>-->
		<BR><BR>&nbsp<HR>
                <A CLASS="leftbar" HREF="/cgi-bin/hgGateway">
		   Genome Browser</A><HR>
 	        <BR>
	        <IMG SRC="/images/DOT.gif" ALT="" HEIGHT=970 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Content Tables-------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	

	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; 
		    About the ENCODE Project</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                   <TR><TH HEIGHT=3></TH></TR>
                   <TR><TD WIDTH=10></TD><TD>
	             <P> This site contains information related to the 
		     <A HREF="http://www.genome.gov/10005107" 
		     TARGET=_blank>ENCODE project</A> at 
		     <A HREF="http://www.genome.gov/" 
		     TARGET=_blank>NHGRI</A>. The UCSC Genome Bioinformatics 
		     Group manages the official repository of 
		     sequence-related data for the ENCODE consortium and 
		     supports the coordination of data submission, storage, 
		     retrieval, and visualization.  
                     A summary of the status of datasets submitted to UCSC 
                     by ENCODE contributors is available on the 
                     <A TARGET=_BLANK HREF="trackStatus.html">
                     ENCODE data status page</A>.
                     <P>
                     UCSC also has a special 
		     interest in comparative genomics, and we provide 
		     resources for the ENCODE multiple sequence alignment 
		     interest group. <A HREF="http://www.ensembl.org" 
		     TARGET=_blank> Ensembl</A> also provides an 
		     <A HREF="http://www.ensembl.org/Homo_sapiens/encode.html"
		     TARGET=_blank> ENCODE resource page</A>.</P>
		     <P>
		     We'd like to thank NHGRI and the contributors of 
		     annotations and analyses to this project. The team at 
		     UCSC that develops and maintains this ENCODE site is 
		     made up of <A HREF="http://www.soe.ucsc.edu/~daryl" 
		     TARGET=_blank>Daryl Thomas</A>, 
		     <A HREF="mailto:&#107;&#97;t&#101;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;u">Kate Rosenbloom</A>, 
		     <A HREF="http://www.soe.ucsc.edu/~kent/" 
		     TARGET=_blank>Jim Kent</A>, and the 
		     <A HREF="../staff.html">UCSC Genome Bioinformatics 
		     staff</A>. <A HREF="about.html">Read more.</A></P>
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%" BORDER=0>
                  <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                      <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT>
 		    	<TD ALIGN="right"><A HREF="newsarch.html"><IMG SRC="/images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>

		      </TD><TD ALIGN="right">&nbsp;</TD></TR>
		    </TABLE>
	          </TD></TR>
		</TABLE>

		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
    		  <TR><TH HEIGHT=3></TH></TR>
    		  <TR><TD WIDTH=10></TD><TD>
<!-- start news item -->
		    <FONT FACE="courier" SIZE=3><B>23 June 2005 - Yale and Affymetrix ChIP/chip and transcription data release; New ENCODE track groupings</B></FONT><BR>
		    <P>
		    To aid ENCODE analysis and reduce visual clutter, we have 
		    split the Genome Browser ENCODE track group into six new 
		    groups:
		    <UL>
		    <LI>ENCODE Regions and Genes
     		    <LI>ENCODE Transcript Levels
     		    <LI>ENCODE Chromatin Immunoprecipitation
     		    <LI>ENCODE Chromosome, Chromatin and DNA Structure
     		    <LI>ENCODE Variation
     		    <LI>ENCODE Comparative Genomics
		    </UL>
		    All of these track groups are visible on the UCSC test 
		    browser. The last two groups, Variation and Comparative 
		    Genomics, do not yet have published tracks on the public 
		    server and therefore are not visible on that server.
		    <P>
		    We have also released a set of new Yale 
		    data and an extensive update of Affymetrix data.  The track 
		    controls for these datasets can be found in the
		    track groups ENCODE Transcript Levels and ENCODE Chromatin 
		    Immunoprecipitation.</P>
		    <P>
		    <B>Yale ChIP/Chip and RNA:</B> Three tracks of ChIP/chip 
		    data from Yale, evaluating microarray platforms, have been 
		    released: Yale ChIP pVal, Yale ChIP Sig, and Yale ChIP 
		    Sites.  These tracks show results of ChIP experiments using
		    STAT1 antibody in HeLa cells on four different microarrays 
		    -- three custom maskless photolithographic oligo arrays, 
		    designed at different resolutions, and the PCR amplicon 
		    array developed by the Ren lab at the Ludwig Institute/UCSD.
		    </P>
		    <P>
		    Two tracks of RNA transcript data from Yale have been 
		    released: Yale RNA and Yale TAR. These tracks show 
		    transcriptionally active regions and transcribed
		    fragments for three cell types (neutrophil, placenta, and 
		    NB4 variously treated for differentiation).</P>
		    <P>
		    Thanks to Joel Rozowsky at Yale for providing this data.
		    Additional Yale ChIP/chip data is currently under review by 
		    our quality assurance group.</P>
		    <P>
		    <B>Affymetrix ChIP/Chip and RNA: </B>
		    The Affymetrix ChIP/chip dataset now contains experimental 
		    results for ten factors (Brg1, CEBPe, CTCF, H3K27me3, 
		    H4Kac4, P300, PU1, Pol2, RARA, and SIRT1) in HeLa cells at 
		    four timepoints after retinoic acid treatment, plus TFIIB 
		    for the final timepoint only. The data is displayed in 
		    eight tracks: Affy PVal 0h, 2h, 8h, 32h and Affy Sites 0h, 
		    2h, 8h, 32h. We acknowledge that this track grouping 
		    is a bit awkward and are working composite track 
		    enhancements to provide more flexibility.</P>
		    <P>
		    The Affy RNA tracks show RNA abundance and transfrags in 
		    retinoic acid-stimulated HL-60 cells at four timepoints,
		    and in GM06990 and HeLa cells: Affy RNA Signal and Affy 
		    Transfrags. </P>
		    <P> 
		    Thanks to Stefan Bekiranov and Srinka Ghosh at Affymetrix 
		    for providing these data. </P>
		    <P>
		    Thanks also to the UCSC ENCODE 
		    team members who developed and reviewed these tracks and 
		    to Rachel Harte in the UCSC browser group
		    for her assistance with track review.</P>
<!-- end news item -->
		    <P>
		    <HR>
<!-- start archived item section-->
		    <FONT FACE="courier" SIZE=3><B>15 June 2005 - Six data sets released in Genome Browser: </B></FONT>
		    Six more ENCODE annotation tracks have been added to the
		    Genome Browser this week.
		    <A HREF="newsarch.html#061505">Read more</A>.</P>
		    <P>

		    <FONT FACE="courier" SIZE=3><B>9 June 2005 - Sanger Histone Modification ChIP/chip track updated: </B></FONT>
		    The Sanger Institute has submitted ChIP/chip data for 
		    additional antibodies and cell lines, which we have 
		    incorporated into the existing Sanger ChIP browser track. 
		    <A HREF="newsarch.html#060905">Read more</A>. </P>

<!-- end archived item section--> 

    	          </TD><TD WIDTH=15></TD></TR>
	        </TABLE>
	      <BR></TD></TR>
	    </TABLE>
          </TD></TR>
        </TABLE>
        <BR>
	    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%">
		  <TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	          </TD></TR>
		</TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome 
		    Browser are freely available for academic, nonprofit, and 
		    personal use with the following conditions:
		    <UL>
		      <LI>
		      The general 
		      <A HREF="http://genome.ucsc.edu/#Conditions">Conditions of Use</A> 
		      for the UCSC Genome Browser apply.
		      <LI>
		      The ENCODE-specific conditions of use are
		      still being developed and will be displayed here
		      in the future.  Until then, <A
		      HREF="mailto:genome@soe.ucsc.edu">please contact
		      us</A> for the conditions on the use of this data.
		    </UL>
	          </TD><TD WIDTH=15></TD></TR>
		</TABLE>
	      <BR></TD></TR>
	    </TABLE>
          </TD></TR>
	</TABLE>
      <BR></TD></TR>
    </TABLE>
  </TD></TR>
</TABLE>
</BODY></HTML>

