<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
  <HEAD>
    <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
    <META name="description" content="ENCODE Project at UCSC">
    <META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
    <TITLE>ENCODE Project at UCSC</TITLE>
    <LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
    <link href="HGStyle.css" rel="stylesheet" type="text/css">
  </HEAD>

  <BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
  <A NAME="TOP"></A>

    <!-- Header Images and links -->
    <TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
      <TR>
        <TH COLSPAN=2 ALIGN="left">&nbsp;&nbsp;
          <A HREF="http://www.genome.gov/10005107" TARGET=_BLANK><IMG SRC="../images/ENCODE_scaleup_logo.png" height=50 ALT="ENCODE Project at NHGRI"></A>
          &nbsp;&nbsp;
          <A HREF="/encode/"><IMG SRC="../images/encodeDcc.jpg" ALT="ENCODE Data Coordination Center at UCSC"></A>
        </TH>
      </TR>

      <TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/encode.home.topbar.html"--></TD></TR>
      </TR>

    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="../images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
                <A CLASS="leftbar" HREF="/cgi-bin/hgTracks?db=hg18">
		   Genome Browser</A>
	        <HR>
	        <A CLASS="leftbar" HREF="cellTypes.html">
		   Cell Types</A>
                <!-- TODO: Replace when first tracks are released 
	        <HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/goldenPath/wgEncode/index.html">
		   Downloads</A>
	        <HR>
                <A CLASS="leftbar" HREF="releaseLog.html">
                    Release Log</A>
                   -->
	        <HR>
	        <A CLASS="leftbar" HREF="http://www.ncbi.nlm.nih.gov/projects/geo/info/ENCODE.html" TARGET=_blank>
		   GEO</A>
	        <HR>
	        <A CLASS="leftbar" HREF="terms.html">
		   Terms of Use</A>
		<HR>
		<A class="leftbar" HREF="http://www.genome.gov/26525220">
		   Contributors</A>
		<HR>
	        <A CLASS="leftbar" HREF="pilot.html">
		   ENCODE Pilot Project</A>
		<HR>
	        <A CLASS="leftbar" HREF="pubs.html">
		   Publications</A>
	        <HR>
                <A CLASS="leftbar" HREF="contacts.html">
		   Contact Us</A>
		<HR>
 	        <BR>
	        <IMG SRC="../images/DOT.gif" ALT="" HEIGHT=970 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Content Tables-------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	

	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%"><TR>
	          <TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; 
		    About the ENCODE Data Coordination Center (DCC)</B></FONT></TD>
	        </TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
	            <P>
		    The UCSC Genome Browser displays data produced by the
		    <A HREF="http://www.genome.gov/10005107" 
		    TARGET=_blank>
		    Encyclopedia of DNA Elements</A>
                    (ENCODE) Consortium, an international
                    collaboration of research groups funded by the 
		    <A HREF="http://www.genome.gov/" TARGET=_blank>
                    National Human 
		    Genome Research Institute</A>
                    (NHGRI).
                    The goal of ENCODE is to
		    build a comprehensive parts list of the 
		    functional elements in the human genome
                   <P> 
                    The initial phase of the ENCODE project (2003-2007) was a pilot 
                    where experiments focused on a limited set of genomic regions comprising
                    roughly 1% of the human genome.  
                    In September 2007, the project
                    was scaled up to production phase.
                    Key differences from the pilot project are:
                    <UL>
                    <LI> 
		    Whole-genome data (with a few remaining pilot region 
		    projects)
                    <LI> 
		    Identification of 
                    <A HREF="cellTypes.html">common cell types</A> to
                    facilitate integrative analysis
                    <LI> 
		    New experimental technologies based on high-throughput 
		    sequencing
                    <LI>
		    A  <A HREF="terms.html">data release policy</A> restricting 
		    use of data for nine months following release
                    </UL>
                    <P>
                    The
                    <A HREF="http://genome.ucsc.edu">Genome Bioinformatics</A> 
		    group at the <A HREF="http://www.ucsc.edu/public/" 
		    TARGET=_blank>University of California Santa Cruz</A> (UCSC)
                    was the official repository for
                    sequence-related data during the pilot project, and now
                    serves as the data coordination 
		    center for the production phase of ENCODE, supporting data 
		    submission, 
		    organization, storage, retrieval, and visualization. 
                    The DCC at UCSC works closely with the ENCODE Data Analysis 
		    Center (DAC) at the 
		    <A TARGET=_BLANK HREF="http://www.ebi.ac.uk/">European 
		    Bioinformatics Institute</A>. Primary data for ENCODE is 
		    available from the NCBI Gene Expression Omnibus 
		    (<A HREF="http://www.ncbi.nlm.nih.gov/projects/geo/info/ENCODE.html" 
		    TARGET=_blank>GEO</A>) public data repository, which
		    may be accessed through the GEO link on the left sidebar.
		    </P>
	            <P>
		    To access genome-wide ENCODE data in the 
		    <A HREF="../cgi-bin/hgTracks?org=human">Genome Browser</A>,
		    go to your region of interest and select
		    ENCODE tracks (marked with the NHGRI logo 
		    <IMG SRC="../images/nhgri.gif" ALT="" HEIGHT=16 WIDTH=16>). 
		    Data from the pilot phase of
                    ENCODE, which is limited to the defined ENCODE regions, 
                    is located in ENCODE-specific track groups in the
                    browser. The <A HREF="/encode/pilot.html">ENCODE Pilot 
		    Project</A> web pages provide
                    convenient browser access to these regions.  
		    </P>
		    <P>
 		    Click <A HREF="../index.html">here</A> to go to 
		    the main UCSC Genome Browser site, which provides access to 
		    sequence and annotation data for a large collection of 
		    genome assemblies.
                    See the Genome Browser 
		    <A HREF="../goldenPath/help/hgTracksHelp.html">User's 
		    Guide</A> for information about displaying tracks and
		    navigating in the Genome Browser.</A>
<!--
	            <P>UCSC figured actively in the <a href="pilot.html">pilot 
		    phase</a> of the ENCODE project, which targeted regions 
		    amounting to about 1% of the genome, and continues to 
		    support the project, which  entered the full-genome phase 
		    in the fall of 2007. In addition to serving as the official 
		    repository of the sequence-related data for the ENCODE 
		    consortium, the UCSC group has a special interest in 
		    comparative genomics and provides additional 
		    resources for the ENCODE groups working in this area.
		    <P>
		    We thank NHGRI and those who have contributed 
		    annotations and analyses to this project. 
		    This portal is maintained by the 
		    UCSC Genome Bioinformatics Group, a cross-departmental 
		    team within the Center for Biomolecular Science and 
		    Engineering (<A HREF="http://www.cbse.ucsc.edu/" 
		    TARGET=_blank>CBSE</A>) at UCSC. Click the 
		    <A HREF="contributors.html">Contributors</A> link 
		    for a complete list of the ENCODE data contributors
		    and the UCSC staff who develop and maintain this website.
		    </P>
<!--
		    The individuals who 
		    develop and maintain this website include
                    <A HREF="mailto:&#107;&#97;t&#101;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;u"
		    TARGET=_blank>Kate Rosenbloom</A>, 
                    <A HREF="mailto:&#104;&#97;rt&#101;r&#97;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;u"
		    TARGET=_blank>Rachel Harte</A>, 
                    <A HREF="mailto:&#116;&#105;&#110;&#103;&#119;&#97;&#110;&#103;&#64;&#115;&#111;&#101;&#46;&#117;&#99;&#115;&#99;&#46;&#101;&#100;&#117;"
                    TARGET=_blank>Ting Wang</A>, 
                    <A HREF="http://www.soe.ucsc.edu/~kent/"
		    TARGET=_blank>Jim Kent</A>, <A HREF="mailto:&#102;&#97;&#110;&#104;&#115;&#117;&#64;&#115;&#111;&#101;&#46;&#117;&#99;&#115;&#99;&#46;&#101;&#100;&#117;"
		    TARGET=_blank>Fan Hsu</A>, <A HREF="mailto:&#107;&#97;&#121;&#108;&#97;&#64;&#115;&#111;&#101;&#46;&#117;&#99;&#115;&#99;&#46;&#101;&#100;&#117;" TARGET=_blank>Kayla Smith</A>, and the 
		    <A HREF="../staff.html">UCSC Genome Bioinformatics 
		    staff</A>.</P>
-->

	          </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%" BORDER=0>
                  <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                      <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT>
 		    	<TD ALIGN="right"><A HREF="newsarch.html"><IMG SRC="../images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>

		      </TD><TD ALIGN="right">&nbsp;</TD></TR>
		    </TABLE>
	          </TD></TR>
		</TABLE>

		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
    		  <TR><TH HEIGHT=3></TH></TR>
    		  <TR><TD WIDTH=10></TD><TD>
		    <P></P>
<!-- start news item -->
<!--
		    <FONT FACE="courier" SIZE=3><B>4 Oct. 2007 - ENCODE Genome Browser Released for hg18 Assembly</B></FONT><BR>
		    <P>
		    The ENCODE browser for UCSC human genome assembly hg18 
		    (NCBI Build 36) is now available. You can access the browser
		    directly by clicking <A HREF="encode.hg18.html">here</A> 
		    or via the Regions (hg18) link on the 
		    <A HREF="pilot.html">ENCODE Pilot Project</A> page.</P>
		    <P>
		    The hg18 ENCODE browser includes 540 data tables in 59 
		    browser tracks that were migrated from the hg17 browser. 
		    The hg17 data coordinates were converted to hg18 coordinates
		    using the UCSC liftOver process.</P>
		    <P>
		    To improve the accessibility of the data, related ENCODE 
		    tracks have been gathered into new configuration groupings 
		    (&quot;super-tracks&quot;) that can be displayed or hidden
		    using a single visiblity control.  We have also reduced the
 		    number of track groups and have modified some of the group 
		    names for clarity. </P>
		    <P>
		    The following table
		    summarizes the data currently present in the hg18 ENCODE 
		    browser:
		    <BR><BR>
            	    <TABLE BORDER=1 BORDERCOLOR="#aaaaaa" WIDTH="100%">        
		      <TR>
		 	<TD nowrap><FONT color="#006666"><B>Group</B></FONT></TD>
                	<TD nowrap><FONT color="#006666"><B>Super-tracks</B></FONT></TD>
                	<TD nowrap><FONT color="#006666"><B>Tracks</B></FONT></TD>
                	<TD nowrap><FONT color="#006666"><B>Tables</B></FONT></TD>
            	      </TR>
  			<TR><TD>Regions and Genes</TD><TD>2</TD><TD>12</TD><TD>73</TD></TR>
  			<TR><TD>Transcription</TD><TD>2</TD><TD>11</TD><TD>67</TD></TR>
  			<TR><TD>Chromatin Immunoprecipitation</TD><TD>8</TD><TD>28</TD><TD>349</TD></TR>
  			<TR><TD>Chromatin Structure</TD><TD>2</TD><TD>8</TD><TD>51</TD></TR>
		    </TABLE></P>
		    <P>
		    Note that the Variation and Comparative Genomics data were 
		    not lifted during this migration; instead, they will be 
		    replaced by new data. The first ENCODE MSA alignment for 
		    hg18 (TBA) is currently in progress on the UCSC 
		    <A HREF="http://hgwdev.cse.ucsc.edu/ENCODE/">development 
		    server</A>.
		    <P>
		    During the migration, ENCODE tracks with whole-genome 
		    data were moved into the standard browser track 
		    groups. These include the GIS PET and UCSD/LI
		    TAF1 tracks. Future submissions of whole-genome ENCODE data
 		    will be loaded directly into the standard track groups.
		    </P>
		    <P>
		    We have expanded the ENCODE downloads site to include 
		    original data for all &quot;wiggle&quot; datasets. These 
		    data files now have filename extensions indicating the 
		    wiggle input format (fixed step, variable step, or 
		    bedGraph).</P>
		    <P>
		    You can find a description of the migration project and 
		    full details of the tables, tracks, and super-tracks 
		    available at the UCSC ENCODE portal on the UCSC
                    <A HREF="http://genomewiki.ucsc.edu/index.php/ENCODE_Hg18_Migration" TARGET=_BLANK>
                    Genome Browser wiki</A>.</P>
		    <P>
		    The UCSC team members who contributed to this effort were:
		    Andy Pohl (data conversion and database loading), Ting Wang
		    and Donna Karolchik (super-track documentation), Bob Kuhn 
		    (portal updates), Brooke Rhead, Kayla Smith, and Ann Zweig (quality 
		    assurance), and Kate Rosenbloom (super-track development, 
		    project management).
		    <P>
                    -->

<!-- end news item --> 
<!-- start archived item section-->
<!-- end archived item section--> 
		    <P></P>
    	          </TD><TD WIDTH=15></TD></TR>
	        </TABLE>
	      </TD></TR>
	    </TABLE>
          </TD></TR>
        </TABLE>
	<BR>
	    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
		  <TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	          </TD></TR>
		</TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome 
		    Browser are freely available for academic, nonprofit, and 
		    personal use with the following conditions:
		    <UL>
		      <LI>
                      The 
		      <A HREF="terms.html">
                      ENCODE Consortium Data Release Policy (September 2008 Draft)</A>
                      is currently under review by the research community.  The final policy will
                      apply to all ENCODE data generated after 2007.
		      <LI>
		      The general 
		      <A HREF="http://genome.ucsc.edu/#Conditions">Conditions of Use</A> 
		      for the UCSC Genome Browser apply.
		    </UL>
	          </TD><TD WIDTH=15></TD></TR>
		</TABLE>
	      </TD></TR>
	    </TABLE>
          </TD></TR>
	</TABLE>
      <BR></TD></TR>
    </TABLE>
  </TD></TR>
</TABLE>
</BODY></HTML>

