<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
 	<META name="description" content="ENCODE Project at UCSC - News Archives">
        <META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
        <TITLE>ENCODE Project at UCSC: News Archives</TITLE>
        <LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<!-- Header Images and links -->
<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
  <TR>
    <TH COLSPAN=2 ALIGN="left">&nbsp;&nbsp;
      <A HREF="http://www.genome.gov/10005107" TARGET=_BLANK><IMG SRC="/images/encodelogo.gif" height=50 ALT="ENCODE Project at NHGRI"></A>
      &nbsp;&nbsp;
      <A HREF="/encode/"><IMG SRC="/images/encode.jpg" ALT="ENCODE Project at UCSC"></A>
    </TH>
  </TR>

  <TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/encode.topbar.html"--></TD></TR>

<!--Content Tables------------------------------------------------------->
  <TR><TD COLSPAN=2 CELLPADDING=10>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1>
      <TR><TD>
      	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="../images/hr.gif" WIDTH="100%">
	  <TR><TD>
	    <FONT SIZE="4"><B> &nbsp;News Archives </B></FONT>
	  </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0>
	  <TH HEIGHT=10></TH>
	  <TR><TD WIDTH=10></TD><TD>
	    <UL>
	      <LI><A CLASS="toc" HREF="#2005">2005 News</A></LI>
	      <LI><A CLASS="toc" HREF="#2004">2004 News</A></LI>
	    </UL>
	    <HR>
	    <A NAME="2005"></A>
	    <A NAME="070605"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 6 July 2005 - NHGRI DIPs, HapMap SNPs, Sanger Assoc annotation tracks released</FONT><BR>
		    <P>
		    The first datasets in the ENCODE Variation group are now
		    available in the UCSC browser.</P>
		    <P>
		    <B>NHGRI Deletion/Insertion Polymorphisms: </B> All human 
		    trace data from NCBI's trace archive were aligned to the 
		    genome and processed using the programs ssahaSNP and
		    ssahaDIP to detect deletion and insertion polymorphisms.
		    Thanks to Jim Mullikin at NHGRI for performing the analyses 
		    and providing these data.</P>
		    <P>
		    <B>HapMap Allele Frequencies: </B> This track shows allele 
		    frequencies for the four HapMap populations in the ten 
		    ENCODE regions that have been resequenced for variation 
		    (manually selected regions m010, m013, and m014 and 
		    randomly selected regions r112, r113, r123, r131, r213, 
		    r232, and r321).  These data were obtained from HapMap 
		    public release #16c.1. Thanks to the International HapMap 
		    Project for making this information available.</P>
		    <P>
		    <B>Sanger Genotype-Expression Association: </B> This track 
		    displays associations among gene expression data from the
		    60 unrelated Centre d'Etude du Polymorphisme Humain (CEPH) 
		    individuals of the International HapMap Project with SNPs 
		    genotyped by HapMap, 
		    in eight ENCODE regions (m010, m013, m014 and r123, r131,
		    r213, r232, and r321). The CEPH population is composed 
		    of Utah residents with ancestry from northern and western 
		    Europe.  The expression data were generated with the 
		    Illumina platform at the Wellcome Trust Sanger Institute.
		    Thanks to Manolis Dermitzakis at the Sanger Institute for 
		    providing these data.</P>
		    <P>
		    We'd also like to acknowledge the UCSC ENCODE team members 
		    who worked on these tracks: Heather Trumbower, Daryl Thomas,
		    and Angie Hinrichs (development), Galt Barber and Ali 
		    Sultan-Qurraie (quality assurance), and Donna Karolchik
		    (documentation). </P>
		    <P>
		    <HR>

	    <A NAME="062305"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 23 June 2005 - Yale and Affymetrix ChIP/chip and transcription data release; New ENCODE track groupings</FONT><BR>
		    <P>
		    To aid ENCODE analysis and reduce visual clutter, we have 
		    split the Genome Browser ENCODE track group into six new 
		    groups:
		    <UL>
		    <LI>ENCODE Regions and Genes
     		    <LI>ENCODE Transcript Levels
     		    <LI>ENCODE Chromatin Immunoprecipitation
     		    <LI>ENCODE Chromosome, Chromatin and DNA Structure
     		    <LI>ENCODE Variation
     		    <LI>ENCODE Comparative Genomics
		    </UL>
		    All of these track groups are visible on the UCSC test 
		    browser. The last two groups, Variation and Comparative 
		    Genomics, do not yet have published tracks on the public 
		    server and therefore are not visible on that server.
		    <P>
		    We have also released a set of new Yale 
		    data and an extensive update of Affymetrix data.  The track 
		    controls for these datasets can be found in the
		    track groups ENCODE Transcript Levels and ENCODE Chromatin 
		    Immunoprecipitation.</P>
		    <P>
		    <B>Yale ChIP/Chip and RNA:</B> Three tracks of ChIP/chip 
		    data from Yale, evaluating microarray platforms, have been 
		    released: Yale ChIP pVal, Yale ChIP Sig, and Yale ChIP 
		    Sites.  These tracks show results of ChIP experiments using
		    STAT1 antibody in HeLa cells on four different microarrays 
		    -- three custom maskless photolithographic oligo arrays, 
		    designed at different resolutions, and the PCR amplicon 
		    array developed by the Ren lab at the Ludwig Institute/UCSD.
		    </P>
		    <P>
		    Two tracks of RNA transcript data from Yale have been 
		    released: Yale RNA and Yale TAR. These tracks show 
		    transcriptionally active regions and transcribed
		    fragments for three cell types (neutrophil, placenta, and 
		    NB4 variously treated for differentiation).</P>
		    <P>
		    Thanks to Joel Rozowsky at Yale for providing this data.
		    Additional Yale ChIP/chip data is currently under review by 
		    our quality assurance group.</P>
		    <P>
		    <B>Affymetrix ChIP/Chip and RNA: </B>
		    The Affymetrix ChIP/chip dataset now contains experimental 
		    results for ten factors (Brg1, CEBPe, CTCF, H3K27me3, 
		    H4Kac4, P300, PU1, Pol2, RARA, and SIRT1) in HeLa cells at 
		    four timepoints after retinoic acid treatment, plus TFIIB 
		    for the final timepoint only. The data is displayed in 
		    eight tracks: Affy PVal 0h, 2h, 8h, 32h and Affy Sites 0h, 
		    2h, 8h, 32h. We acknowledge that this track grouping 
		    is a bit awkward and are working composite track 
		    enhancements to provide more flexibility.</P>
		    <P>
		    The Affy RNA tracks show RNA abundance and transfrags in 
		    retinoic acid-stimulated HL-60 cells at four timepoints,
		    and in GM06990 and HeLa cells: Affy RNA Signal and Affy 
		    Transfrags. </P>
		    <P> 
		    Thanks to Stefan Bekiranov and Srinka Ghosh at Affymetrix 
		    for providing these data. </P>
		    <P>
		    Thanks also to the UCSC ENCODE 
		    team members who developed and reviewed these tracks and 
		    to Rachel Harte in the UCSC browser group
		    for her assistance with track review.</P>
		    <P>
		    <HR>

	    <A NAME="061505"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 15 June 2005 - Six data sets released in Genome Browser</FONT><BR>
		    <P>
		    Six more ENCODE annotation tracks have been added to the
		    Genome Browser this week:</P>
		    <P>
		    <B>NHGRI DNaseI-Hypersensitive Sites (update): </B>The 
		    NHGRI DNaseI-HS track has been updated with new data. The
		    track now includes DNaseI-hypersensitive sites in CD4+ 
		    T-cells before and after activation by anti-CD3 and 
		    anti-CD28 antibodies. Thanks to Greg Crawford at the 
		    Collins lab (NHGRI) for providing these data.</P>
		    <P>
		    <B>Genome Institute of Singapore PET of PolyA+ RNA: </B>
		    The GIS PET RNA track displays starts and ends of mRNA 
		    transcripts determined by paired-end ditag sequencing in 
		    two cell lines, MCF7 and HCT116 treated with 5 
		    fluoro-uracil. A total of 584,624 PETs were generated for 
		    MCF7 and 280,340 were generated for HCT116. More than
		    80% of the PETs in each group were mapped to the genome.
		    Thanks to Atif Shahab, Yijun Ruan, the GIS, and the 
		    Bioinformatics Institute of Singapore for providing these 
		    data.</P>
		    <P>
		    <B>Gencode Gene Annotations and Intron Validation: </B>
		    The Gencode Genes track displays high-quality manual 
		    annotations in the ENCODE regions generated by the 
		    GENCODE project.  A companion track, Gencode Introns, 
		    shows experimental gene structure validations for these
		    annotations.  Thanks to the HAVANA team at the Wellcome
		    Trust Sanger Institute; France Denoeud, Julien Lagarde,
		    and Roderic Guigo at the IMIM; and Alexandre Reymond
		    at the University of Geneva for providing the
		    annotations and experimental confirmation, as well as  
		    working with UCSC to develop the track display.</P>
		    <P>
		    <B>Boston University Hydroxyl Radical Cleavage: </B>
		    The BU ORChID track displays predicted hydroxyl radical 
		    cleavage intensity on naked DNA for each nucleotide in 
		    the ENCODE regions.  The prediction algorithm draws data 
		    from a database of 150 experimentally-determined cleavage 
		    patterns. Thanks to Jay Greenbaum at the Tullius lab for 
		    providing these data.</P>
		    <P>
		    <B>UC Davis ChIP/Chip: </B> The UCD Ng E2F1 track shows 
		    ChIP analysis of HeLa cells using E2F1 antibody as 
		    assayed on Nimblegen arrays. The E2F1 factor is associated
		    with cell cycle control, transcriptional regulation, and 
		    apoptosis. Thanks to the Farnham lab and Kyle Munn, Todd 
		    Richmond and Roland Green of Nimblegen for providing 
		    these data.</P>
		    <P>
		    <B>Ludwig Institute/UCSD ChIP/Chip: </B>
		    Three tracks of ChIP/chip data using Nimblegen arrays 
		    have been released: LI Ng TAF1 IMR90, LI Ng Validation,
     		    and LI Ng gIF ChIP. The TAF1 track shows genome-wide 
		    TAF1 binding sites as determined by ChIP in IMR90 
		    (fibroblast) cells assayed on Nimblegen high-density 
		    oligo arrays. The peaks from the genome scan experiments 
		    were verified using four antibodies associated with 
		    transcription start (Pol2, H3ac, H3K4me2, and TAF1) on 
		    condensed arrays covering the putative TAF1 binding sites.
		    The gIF track shows the results of ChIP/chip experiments 
		    using gamma interferon-treated HeLa cells, with Pol2 and 
		    H3K4me3 antibodies.</P>
		    <P>
		    Two additional tracks, LI ChIP Various and LI gIF ChIP, 
 		    display ChIP/chip data assayed on Ren lab ENCODE PCR
		    tiling arrays for five antibodies (Pol2, TAF1, H3K4me2, 
		    SUZ12, H3K27me3) and four cell lines (HeLa, THP1, IMR90, 
		    HCT116), as well as gamma interferon experiments using 
		    seven antibodies (Pol2, TAF1, H3K4me2, H3K4me3, H3ac, 
		    H4ac, and STAT1) in HeLa cells.
		    Thanks to Chunxu Qu and Bing Ren, at the Ren lab for 
		    providing these data and assisting with track display 
		    issues.</P>
		    <P>
		    We'd also like to acknowledge the UCSC team members who 
		    developed these tracks: Kate Rosenbloom, Hiram Clawson, 
		    and Angie Hinrichs for track development; Bob Kuhn, Ali
		    Sultan-Qurraie and Galt Barber for quality assurance; 
		    and Donna Karolchik and Jim Kent for documentation.</P>
		    <P>
		    <HR>

	    <A NAME="060905"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 9 June 2005 - Sanger Histone Modification ChIP/chip track updated</FONT><BR>
		    <P>
		    The Sanger Institute has submitted ChIP/chip data for 
		    additional antibodies and cell lines, which we have 
		    incorporated into the existing Sanger ChIP browser track. 
		    The expanded track now contains data for five antibodies
		    (H3K4me1, H3K4me2, H3K4me3, H3ac, H4ac) and two cell 
		    lines (GM06990, K562 (leukemia)).</P>
		    <P>
		    Thanks to Rob Andrews and Chris Koch at the Dunham lab 
		    for providing these data.  UCSC team members who developed
		    this track include Hiram Clawson (track development), Ali 
		    Sultan-Qurraie (quality assurance), and Donna Karolchik 
		    and Jim Kent (documentation).</P>
		    <P>
		    <HR>

	    <A NAME="060605b"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 6 June 2005 - Changes to ENCODE track labels in Genome Browser</FONT><BR>
		    <P>
		    We have modified the Genome Browser labels for the 
		    existing ENCODE data tracks to trim overly-long labels
		    that were truncated in the display and to facilitate 
		    cross-track analysis. The new label format shows the 
		    submitter and the experiment, followed by the cell line 
		    (in tracks where the data includes only one cell line).
		    </P>
		    <P>
		    <HR>

	    <A NAME="060605a"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 6 June 2005 - Stanford Promoters and UVa DNA Replication tracks released</FONT><BR>
		    <P>
		    We are pleased to announce the first UCSC Genome Browser 
		    tracks released for the June 2005 ENCODE data freeze:
		    Stanford Promoters and UVa DNA Replication Temporal 
		    Profiling.
		    <P>
		    Stanford has provided an update of their promoter activity
		    data based on transient transfection luciferase reporter 
		    assays of 643 putative promoter fragments in the ENCODE 
		    regions.  The update includes two additional cell lines 
		    and activity averaged across all cell lines.  The data 
		    tables now contain additional experimental detail to 
		    facilitate analysis.  This track, containing 17 subtracks 
		    (16 cell lines and the average), is labeled 				  &quot;Stanf. Promoter&quot;. Thanks to Sara Hartman at 
		    the Myers lab for providing these data.</P>
		    <P>
		    The Dutta lab at Univerity of Virginia (UVa) has completed
		    the second biological replicate of their temporal 
		    profiling of HeLa cell replication products and has
		    provided a dataset containing merged data from the two 
		    replicates. The track, containing five subtracks 
		    representing two-hour intervals, is labeled
		    &quot;UVa DNA Rep&quot;. Thanks to Christopher Taylor at 
		    the Dutta lab for providing these data.</P>
		    <P>
		    We'd also like to acknowledge the UCSC team members who
		    worked on these annotation tracks: Angie Hinrichs (track
		    development), Galt Barber and Ali Sultan-Qurraie (QA), and
		    Jim Kent and Donna Karolchik (track documentation).</P>
		    <P>
		    <HR>

	    <A NAME="052405b"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 24 May 2005 - New MSA sequence data freeze available</FONT><BR>
		    <P>
		    A new ENCODE MSA sequence data freeze is available on the 
		    UCSC downloads server. The latest freeze contains 
		    sequences from 23 vertebrates provided by NISC, Baylor, 
		    the Broad Institute (2X) and the whole genome shotgun 
		    (WGS) assemblies.  The data may be downloaded as 
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg16/encode/alignments/MAY-2005/">individual data files</A> or a
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg16/encode/alignments/MAY-2005.tgz">directory tarball</A>. 
		    Aligners are encouraged to upload alignments and related 
		    data (such as conservation scores and elements) to the 
		    UCSC ENCODE ftp site as soon as possible and then notify 
		    <A HREF="mailto:kate@soe.ucsc.edu">Kate Rosenbloom</A>. 
		    Other data, (conservation, trees, etc.) 
		    will be generated based on this dataset.</P>
		    <P>
		    The following is a summary of data updates from the
		    previous release:
		    <UL>
		    <LI>
		    The human assembly version remains at hg16 (Jul. 2003).
		    <LI>
		    The mouse assembly has been updated from mm5 (May 2004) to
		    mm6 (Mar. 2005).
		    <LI>
		    The multiple rat sequences have been replaced by a single
		    sequence: rn3 (Jun. 2003).
		    <LI>
		    The cow sequences have been updated using an assembly of 
		    BAC-based sequences provided by Baylor College of 
		    Medicine.
		    <LI>
		    Fugu (fr1), macaque (rheMac1), opossum (monDom1), 
		    Tetraodon (tetNig1), Xenopus (xenTro1), and zebrafish 
		    (danRer2) have been added.  The macaque sequence was 
		    obtained from a Baylor 
		    College of Medicine assembly that has not yet been 
		    officially released. 
		    <LI>
		    A new NISC species, rfbat 
		    (<em>Rhinolophus ferrumequinum</em>), is now available.
		    <LI>
		    Platypus data from regions 
		    where NISC has not yet generated data were provided by 
		    Jim Mullikin from a preliminary assembly of Washington 
		    University WGS reads. 
		    <LI>
		    This freeze includes low-redundancy sequence data from 
		    tenrec, elephant, armadillo, and rabbit. Only one set of 
		    sequences are provided per species/target combination; 
		    where available, NISC data is provided instead of 
		    the 2X assemblies. These data are not yet accessioned at 
		    NCBI, but were made available by the Broad Institute 
		    (rabbit, elephant, armadillo) and Jim Mullikin (tenrec).
		    <LI>
		    Orthology predictions for Fugu were made only by 
		    MAVID/Mercator; predictions for all other assemblies 
		    supported by the UCSC Genome Browser represent a union 
		    with UCSC predictions as well. Because no additional 
		    post-processing was done on the Fugu predictions, they 
		    contain a few very small contigs.
		    </UL>
		    <P>
		    Thanks to the many people, particularly Elliott Margulies
		    and Daryl Thomas, who made this release possible.</P>
		    <P>
		    <HR>

	    <A NAME="052405a"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 24 May 2005 - ChIP-PET/GIS annotation track (Genome Institute of Singapore) released</FONT><BR>
		    <P>
		    The ENCODE Genome Browser now features the ChIP-PET/GIS 
		    annotation track, which shows paired-end 
		    ditag (PET) sequences derived from 65,572 individual p53 
		    ChIP fragments of 5-fluorouracil (5FU) stimulated HCT116 
		    (colon) cells. Only PETs with a single specific mapping 
		    to the genome are included in this track.
		    <P>
		    Thanks to Atif Shahab, Chia-Lin Wei, and Yijun Ruan at the
		    <A HREF="http://www.gis.a-star.edu.sg/homepage/default.jsp"
		    TARGET=_blank>Genome Institute of Singapore</A> for 
		    providing the p53 ChIP-PET library and sequence data. The 
		    data were mapped and analyzed by scientists from the 
		    Genome Institute of Singapore, the Bioinformatics 
		    Institute, Singapore, and Boston University. For more
		    information about this annotation, see the ChIP-PET/GIS
		    track description page.</P>
		    <P>
		    <HR>

	    <A NAME="052305"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 23 May 2005 - Boston University First Exon annotation track released</FONT><BR>
		    <P>
		    The First Exon/BU annotation track, contributed by the
		    <A HREF="http://zlab.bu.edu/zlab/index.shtml"
		    TARGET=_blank>ZLAB</A> at Boston University, is now 
		    available in the UCSC Genome Browser. This track displays 
		    expression levels of computationally identified first 
		    exons and a constitutive exon of 20 genes in the ENCODE 
		    regions.</P>
		    <P>
		    For each gene, all alternative first exons were identified
		    based on manual selection of predictions from the 
		    <A HREF="http://biowulf.bu.edu/zlab/Promoser/"
		    TARGET=_blank>PromoSer</A>
		    program. The expression levels of exons were then 
		    quantified using rcPCR in ten normal human tissues.</P>
		    <P>
		    Thanks to Ulas Karaoz and the Zhiping Weng lab at 
		    Boston University for providing these data. For more
		    information about this annotation, as well as a complete
		    list of the individuals who contributed to this track,
		    see the First Exon/BU track description page.</P>
		    <P>
		    <HR>

	    <A NAME="050705"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 7 May 2005 - ENCODE status page now available</FONT><BR>
		    <P>
		    A simple summary page has been added to the UCSC ENCODE
		    portal to show the status of datasets submitted to UCSC by
		    ENCODE contributors. The page may be found at
		    <A HREF="http://genome.ucsc.edu/ENCODE/trackStatus.html">http://genome.ucsc.edu/ENCODE/trackStatus.html</A> and can be accessed via
		    links on the ENCODE home page and the ENCODE data
		    submission page. The status page will be updated 
		    approximately once a week.
		    <P>
		    To review the latest ENCODE data releases, see the
		    <A HREF="releaseLog.html">release log</A>.
		    <P>
		    <HR>

	    <A NAME="050605"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 6 May 2005 - UCSD/Ludwig Institute histone modification ChIp/Chip data released</FONT><BR>
                    <P>  
                    Two additional sets of ChIp/Chip data from UCSD/Ludwig
                    Institute are now available:
<PRE>
    ChIp/LI AcH3     
    ChIp/LI MeH3K4   
</PRE>               
                    These tracks display locations of acetylated H3 and
                    dimethylated K4H3 binding in IMR90 (lung fibroblast) 
		    cells.
                    <P>
                    To consolidate viewing in the browser, the previously 
		    released eight datasets from UCSD/LI have been reformatted
                    as two composite tracks (one track per antibody) with 
		    each track containing four subtracks (one per cell line). 
		    These tracks are:
<PRE>
    ChIp/LI Pol2
    ChIp/LI TAF1
</PRE>
                    To facilitate data analysis, the data were also reloaded
                    in a format that allows extraction of the original data 
		    values via the UCSC table browser.
                    <P>
                    Thanks to Chunxu Qu and the
                    <A HREF="http://licr-renlab.ucsd.edu/" TARGET=_blank>Ren 
		    lab</A> for providing these data.
                    <P>
                    <HR>

	    <A NAME="021705"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 17 Feb. 2005 - ENCODE downloads released</FONT><BR>
   		<P>
		    The ENCODE download area has been reorganized and updated 
		    on our public download server.  The downloads access page 
		    is now:
		    <P>
		    &nbsp; &nbsp; &nbsp; &nbsp;
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/encode/"
		    TARGET=_blank>http://hgdownload.cse.ucsc.edu/goldenPath/encode/</A>
		    <P>
		    and the annotations are now located in the 
		    assembly-specific download directory, currently:
		    <P>
		    &nbsp; &nbsp; &nbsp; &nbsp;
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg16/encode/"
		    TARGET=_blank>http://hgdownload.cse.ucsc.edu/goldenPath/hg16/encode/</A>
		    <P>
		    Any web pages referencing the previous UCSC ENCODE 
		    downloads will need to be updated.  Please contact us if 
		    you have any difficulties.
		    <P>
		    <HR>

	    <A NAME="020205"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 2 Feb. 2005 - Affymetrix ChIp/Chip and PolyA RNA data released</FONT><BR>
		    <P>
		    A second set of ENCODE ChIp/Chip data is now available on 
		    the July 2003 human genome assembly:
<PRE>
    ChIp/Affy Pol2 Pval
    ChIp/Affy Pol2 Sites
    RNA/Affy Signal
    RNA/Affy Sites
</PRE>
		    These tracks show RNA Polymerase II precipitation and 
		    RNA abundance in retinoic acid-stimulated HL-60 cells at 
		    0, 2, 8, and 32 hours, as measured by Affymetrix tiling 
		    arrays in the non-repetitive ENCODE regions.  The 
		    <em>Pval</em> and <em>Signal</em> tracks show values 
		    for each tiled probe; the <em>Sites</em> tracks show 
		    contiguous regions of enrichment.
		    <P>
		    A new <em>composite</em> track display was developed to 
		    concisely display multiple data sets of similar types, a 
		    common feature of ENCODE data.  Each of these new tracks 
		    contains 4 subtracks, one for each time interval.  The 
		    subtracks share a single description page and set of 
		    visibility controls.  Checkboxes on the track 
		    configuration page allow selected subtracks to be hidden 
		    in the display.
		    <P>
		    These data were generated and analyzed by Tom Gingeras' 
		    group at 
		    <A HREF="http://www.affymetrix.com"
		    TARGET=_blank>Affymetrix</A> and  
		    <A HREF="http://www.hms.harvard.edu/dms/bbs/fac/struhl.html" 
		    TARGET=_blank>Kevin Struhl's group at Harvard Medical 
		    School</A>. We would like to thank Stefan Bekiranov at 
		    Affymetrix for submitting the data and working closely 
		    with us to clarify the experimental methods and 
		    verification descriptions.
		    <P>
		    <HR>

	    <A NAME="2004"></A>
	    <A NAME="110504"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 5 Nov. 2004 - First ENCODE data tracks in the UCSC Browser</FONT><BR>
		    <P>
		    The first datasets submitted for the ENCODE project are 
		    now publicly available:
		    <UL>
		      <LI>
 		      ChIP/chip and transcription (Ludwig Institute/UCSD)
		      <LI>
		      Temporal profiling of DNA replication (University of 
		      Virginia)
		      <LI>
		      Promoter activity (Stanford)
		      <LI>
		      DNaseI hypersensitive sites (NHGRI)
		    </UL>
		    These tracks are visible in the ENCODE track group of the 
		    July 2003 (hg16) human genome assembly.  We would like to 
		    thank the labs of Bing Ren (LI/UCSD), Anindya Dutta (UVA),
		    Rick Myers (Stanford), and Francis Collins (NHGRI) for 
		    contributing the initial ENCODE data sets.  
		    <P>
                    <HR>

	    <A NAME="102604"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 26 Oct. 2004 - Sequence Freeze For Multiple Alignments</FONT><BR>
		    <P>
		    We are pleased to release the first "official" sequence 
		    data freeze for the ENCODE multiple sequence alignment 
		    projects. The data formats are described in the 
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg16/encode/alignments/OCT-2004/README.txt" 
		    TARGET=_blank>README</A> file, and the sequences and 
		    supporting information is collected in the 
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg16/encode/alignments/OCT-2004" 
		    TARGET=_blank> data directory</A>.
		    <P>
		    The species included in this freeze are as follows:
<PRE>
       _SPECIES_      _SOURCE_
        Human           hg16
        Chimpanzee      panTro1
        Dog             canFam1
        Rat             rn3
        RatB	        BCM
        Mouse           mm5
        Chicken		galGal2
        Galago		NISC
        Baboon		NISC
        Marmoset        NISC
        Armadillo       NISC
        Platypus        NISC
</PRE>
		    You will notice that we have worked hard to include a 
		    number of species for which genome-wide sequence data was 
		    already available. The process by which these orthologous 
		    regions were identified is still an area of active 
		    research development, the details of which will be 
		    presented at the upcoming ENCODE meeting at CSHL.
		    <P>
		    Please note that we have also included a *second* rat 
		    sequence (ratB) in this freeze.  RatB represents an 
		    initiative to standardize the quality level of sequences 
		    in the ENCODE regions for species with genome-wide 
		    sequence data. The data were made available just before 
		    our freeze date, so we decided to include both versions of
		    the rat sequence for now. Eventually these sequences will 
		    likely be rolled into future genome assemblies.
		    <P>
		    Remember, this is a work in progress, so not all targets 
		    have sequence from all species. And some species/target 
		    combinations may not be complete yet. Progress on the 
		    NISC-generated sequences can be found at the 
		    <A HREF="http://www.nisc.nih.gov/open_page.html?/projects/encode/index.cgi" 
		    TARGET=_blank> NISC ENCODE Project: Comparative 
		    Sequencing</A>.
		    <P>
		    Many people have worked very hard to make these data 
		    available. Special thanks to Daryl Thomas, Kate 
		    Rosenbloom and the entire UCSC Team; Greg Schuler and the 
		    NCBI team; Colin Dewey and Lior Pachter; Pam Thomas and 
		    the NISC team; and David Wheeler and the BCM team.
		    <P>
		    Please send any comments to the 
		    <A HREF="mailto: <ENCODE_MSA-L@LIST.NIH.GOV>" 
		    TARGET=_blank>ENCODE MSA Mailing List</A>.
		    <P>
		    &nbsp; &nbsp; Elliott
		    <BR>
		    -- 
		    <BR>
		    Elliott H. Margulies, Ph.D.
		    Genome Technology Branch, NHGRI
		    <P>
                    <HR>

	    <A NAME="062404"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 24 Jun. 2004 - ENCODE Project Portal Released</FONT><BR>
		    <P>
		    We are proud to announce the release of features
		    in the UCSC Genome Browser that are tailored to
		    the ENCODE project community, including this home
		    page to consolidate these resources.
		    <P>
		    The initial resources include sequences for the current 
		    human assemblies (hg16, hg15, hg13, and hg12), sequence of
		    the comparative species from 
		    <A HREF="http://www.nisc.nih.gov/" TARGET=_blank>NISC</A>,
		    tools for coordinate conversion between human
		    assemblies, format descriptions for data
		    submission, and contact information for help with
		    submitting annotation data and analyses.
		    <P>
		    Bulk downloads of the sequence and annotations may
		    be obtained from the ENCODE Project 
		    <!--<A HREF="ftp://genome-test.cse.ucsc.edu/encode/downloads/">FTP</A> or -->
		    <A HREF="http://hgdownload.cse.ucsc.edu/encode/">Downloads</A> 
		    page.  The sequences available here are repeat-masked 
		    versions of the GenBank records.
	    <BR><BR>
	  </TD><TD WIDTH=15></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
  </TD></TR>
</TABLE>
</BODY></HTML>
