<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
  <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
  <META name="description" content="ENCODE Project at UCSC: Data Submission">
  <META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
  <TITLE>ENCODE Project at UCSC: Data Submission</TITLE>
  <LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<!-- Header Images and links -->
<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
  <TR>
    <TH COLSPAN=2 ALIGN="left">&nbsp;&nbsp;
      <A HREF="http://www.genome.gov/10506161" TARGET=_BLANK><IMG SRC="/images/encodelogo.gif" height=50 ALT="ENCODE Project at NHGRI"></A>
      &nbsp;&nbsp;
      <A HREF="/encode/"><IMG SRC="/images/encode.jpg" ALT="ENCODE Project at UCSC"></A>
    </TH>
  </TR>

<!--Hotlinks bar------------------------------------------------------->
<TABLE BORDER=0 WIDTH="100%">
  <TR>
    <TH COLSPAN=2 ALIGN="left">
    </TH>
  </TR>
  <TR>
    <TD COLSPAN=2 HEIGHT=40>
      <TABLE bgcolor="000000" cellpadding="1" cellspacing="1" width="100%" height="27">
        <TR bgcolor="2636D1">
          <TD valign="middle">
	    <TABLE BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24">
	      <TR>
	        <TD VALIGN="middle">
	          <font color="#89A1DE">    &nbsp;
		    <A class="topbar" HREF="/encode/encode.html"> 
		     Regions         </a> &nbsp; - &nbsp;
		    <A class="topbar" HREF="/encode/submission.html">
		     Data Submission </a> &nbsp; - &nbsp;
		    <A class="topbar" HREF="http://genome-test.cse.ucsc.edu/encode/downloads/"> 
		     Downloads       </a> &nbsp; - &nbsp;
		    <A class="topbar" HREF="/cgi-bin/hgLiftOver"> 
		     Tools           </a> &nbsp; - &nbsp;
		    <A class="topbar" HREF="/encode/terms.html">
		     Terms           </a> &nbsp; - &nbsp;
		    <A class="topbar" HREF="/goldenPath/help/hgTracksHelp.html"> 
		     Help            </a> &nbsp;
		  </font>
	        </TD>
	      </TR>
	    </TABLE>
	  </TD>
        </TR>
      </TABLE>
    </TD>
  </TR>
</TABLE>



<!--Downloads menu------------------------------------------------------->
<TABLE BORDER=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"></TH></TR>
<TR><TD CELLPADDING=10>	
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>

		            <TR>
			      <TD>
			        <A NAME="Existing"></A><FONT SIZE="4"><B>&nbsp; Submission of ENCODE data </B></FONT>
			      </TD>
			    </TR> 
		          </TABLE>
		          <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
			    <TR><TH HEIGHT=3></TH></TR> 
			    <TR>
			      <TD WIDTH=10>
			        </TD>
			        <TD> We are looking forward to the
                    submission of ENCODE-related data, and are ready
                    to help throughout the process.  We recommend the
                    following steps: <UL>
		    <LI>Choose an appropriate format (<A
		    HREF="#BED">BED</A>, <A HREF="#MAF">MAF</A>, or <A
		    HREF="#WIG">WIG</A>) for your data from the
		    descriptions below and create a tab-delimited file
		    in that format.</LI>

		    <LI>If necessary, data from previous assemblies in
		    BED format may be transformed to the Human July
		    2003 (hg16) assembly with our <a
		    href="/cgi-bin/hgLiftOver" >liftOver</a>
		    tools.</LI>

		    <LI>Test your file by loading it into the browser
                    as a <a
                    href="http://genome.ucsc.edu/goldenPath/help/customTrack.html"
                    >custom track</a>.</LI>

		    <LI>When your data is displayed as you desire,
		    please <A HREF="mailto:genome@soe.ucsc.edu">
		    contact us</A> to load the tracks into the
		    database.  This makes the data available to the rest
		    of the ENCODE community and to the public.</LI>
		    </UL>
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>

                </TD></TR></TABLE>
    <TABLE CELLPADDING=5 BORDER=0 WIDTH="100%"><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><A name="BED"></A><B>&nbsp; 
		Browser Extensible Data Format (BED) 
		</B></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>

	<P>The BED format was defined for the efficient storage and
	retrieval of genomic annotations.  It provides a flexible way
	to define the data lines that are displayed in an annotation
	track. 

	<!--BED lines have three required fields and nine
	additional optional fields. The number of fields per line must
	be consistent throughout any single set of data in an
	annotation track.-->

	Please see the <A HREF="http://genome.ucsc.edu/goldenPath/help/customTrack.html#BED">BED format description</A> on the <A HREF="http://genome.ucsc.edu/goldenPath/help/customTrack.html">custom tracks</A> page for more details.

<!--
	<P>The first three required BED fields are:</P>
	<OL>
	<LI><B>chrom</B> -- The name of the chromosome (e.g. chr3, chrY,
	chr2_random) or contig (e.g. ctgY1).
	<LI><B>chromStart</B> -- The starting position of the feature in the
	chromosome or contig. The first base in a chromosome is numbered 0.
	<LI><B>chromEnd</B> -- The ending position of the feature in the
	chromosome or contig. The <I>chromEnd</I> base is not included in the
	display of the feature. For example, the first 100 bases of a
	chromosome are defined as <I>chromStart=0, chromEnd=100</I>, and span
	the bases numbered 0-99.
	</OL>

	<P>The 9 additional optional BED fields are:</P>
	<OL START=4>
	<LI><B>name</B> -- Defines the name of the BED line.  This label is
	displayed to the left of the BED line in the Genome Browser
	window when the track is open to full display mode or directly to the
	left of the item in pack mode. 
	<LI><B>score</B> -- A score between 0 and 1000. If the track line 
	<I>useScore</I> attribute is set to 1 for this annotation data set, the
	<I>score</I> value will determine the level of gray in which
	this feature is displayed (higher numbers = darker gray).
	<LI><B>strand</B> -- Defines the strand -- either '+' or '-'.
	<LI><B>thickStart</B> -- The starting position at which the feature
	is drawn thickly (for example, the start codon in gene
	displays).
	<LI><B>thickEnd</B> -- The ending position at which the feature is
	drawn thickly (for example, the stop codon in gene displays).
	<LI><B>reserved</B> -- This should always be set to zero.
	<LI><B>blockCount</B> -- The number of blocks (exons) in the
	BED line.
	<LI><B>blockSizes</B> -- A comma-separated list of the block
	sizes. The number of items in this list should correspond to
	<I>blockCount</I>.
	<LI><B>blockStarts</B> -- A comma-separated list of block starts.
	All of the <I>blockStart</I> positions should be calculated relative to
	<I>chromStart</I>. The number of items in
	this list should correspond to <I>blockCount</I>.
	</OL>

	<P><B><I>Example:</I></B><BR>
	Here's an example of an annotation track that uses a complete BED definition:</P>
	<FONT color="#0000FF"><PRE><TT>track name=pairedReads description=&quot;Clone Paired Reads&quot; useScore=1
chr22 1000 5000 cloneA 960 + 1000 5000 0 2 567,488, 0,3512
chr22 2000 6000 cloneB 900 - 2000 6000 0 2 433,399, 0,3601</TT></PRE></FONT>
		    </TD><TD WIDTH=15>
-->
			</TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
		    </TD></TR></TABLE>
		    </TD></TR></TABLE>

    <TABLE CELLPADDING=5 BORDER=0 WIDTH="100%"><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="MAF"></A><FONT SIZE="4"><B>&nbsp; 
		    Multiple Alignment Format (MAF) 
		    </B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>

		    The multiple alignment format stores a series of
		    multiple alignments in a format that is easy to
		    parse and relatively easy to read.  This format
		    stores multiple alignments at the DNA level
		    between entire genomes. The previously existing
		    formats are suitable for multiple alignments of
		    single proteins or regions of DNA without
		    rearrangements, and would require considerable
		    extension to cope with genomic issues such as
		    forward and reverse strand directions, multiple
		    pieces to the alignment, and so forth.  

		    <P><B>General Structure</B><P> The .maf format is
		    line-oriented. Each multiple alignment ends with a
		    blank line.  Each sequence in an alignment is on a
		    single line, which can get quite long, but there is not length limit. Words in a
		    line are delimited by any white space. Lines
		    starting with # are considered to be comments.
		    Lines starting with ## can be ignored by most
		    programs, but contain meta-data of one form or
		    another.
		    <P>
		    The file is broken into paragraphs that terminate
		    in a blank line.  Within a paragraph, the first
		    word of a line indicates its type.  Each multiple
		    alignment is in a separate paragraph that begins
		    with an 'a' line and contains an 's' line for each
		    sequence in the multiple alignment.  For now,
		    parsers should ignore other types of paragraphs
		    and other types of lines within an alignment
		    paragraph.

		    <P><B>The First Line -- Header</B><P> 
		    <FONT color="#0000FF"><PRE><TT> ##maf version=1 scoring=tba.v8</TT></PRE></FONT>
		    The first line of a .maf file begins with ##maf.
		    This word is followed by white-space-separated
		    variable=value pairs.  There must *not* be white
		    space surrounding the =.  The currently defined
		    variables are: 
		    <UL>
		    <LI>version -- Required.  Currently set
		                   to one.  </LI>
		    <LI>scoring -- Optional. A name for the scoring
		                   scheme used for the alignments.
		                   The current scoring schemes are:
		        <UL><LI>bit -- roughly corresponds to blast bit
		    		   values -- roughly 2 points per
		    		   aligning base minus penalties for
		    		   mismatches and inserts.  </LI>
		            <LI>blastz -- blastz scoring scheme -- roughly
		                   100 points per aligning base.</LI>
		            <LI>probability -- some score normalized
		                   between 0 and 1.  </LI> 
		        </UL>

		    <LI>program -- Optional. Name of the program
		                   generating the alignment.</LI>
                    </UL>
                    Parsers ignore variables they do not understand.
		    <P><B>The Second Line -- Parameters for creating the alignment</B><P> 
		    <FONT color="#0000FF"><PRE><TT> # tba.v8 (((human chimp) baboon) (mouse rat))</TT></PRE></FONT>
		    The second line displays the parameters that were
		    used to run the alignment program.

		    <P><B>Lines starting with 'a' -- parameters for a new alignment block</B><P> 
		    <FONT color="#0000FF"><PRE><TT> a score=23262.0</TT></PRE></FONT>
		    Each alignment begins with an 'a' line.  The 'a'
		    is followed by name=value pairs.  There are no
		    required name=value pairs.  The currently defined
		    variables are: <UL>

		    <LI> score -- Optional. Floating point score. If
		                 this is present, it is good practice
		                 to also define scoring in the first
		                 line.  </LI>

		    <LI> pass --  Optional.  Positive integer value.
		                 For programs that do multiple pass
		                 alignments such as blastz, this shows
		                 which pass this alignment came from.
		                 Typically, pass 1 will find the
		                 strongest alignments genome-wide, and
		                 pass 2 will find weaker alignments
		                 between two first-pass alignments.</LI> 
		     </UL>

		    <P><B>Lines starting with 's' -- a sequence within an alignment block</B><P> 
<FONT color="#0000FF"><PRE><TT> s hg16.chr7    27707221 13 + 158545518 gcagctgaaaaca
 s panTro1.chr6 28869787 13 + 161576975 gcagctgaaaaca
 s baboon         249182 13 +   4622798 gcagctgaaaaca
 s mm4.chr6     53310102 13 + 151104725 ACAGCTGAAAATA</TT></PRE></FONT>
		    The 's' lines together with the 'a' lines define a
		    multiple alignment.  The 's' lines have the
		    following fields which are defined by position
		    rather than name=value pairs.<UL>

		    <LI> src -- The name of one of the source sequences
		               for the alignment.  For sequences that
		               are resident in a browser assembly, the
		               form 'database.chromosome' allows
		               automatic creation of links to other
		               assemblies.  Non-browser sequences are
		               typically reference by the species name
		               alone.</LI>

		    <LI> start -- The start of the aligning region in
		               the source sequence.  This is a
		               zero-based number. If the strand field
		               is '-' then this is the start
		               relative to the reverse-complemented
		               source sequence.  </LI>
		    <LI> size -- The size of the aligning region in the
		               source sequence.  This number is equal
		               to the number of non-dash characters in
		               the alignment text field below.  </LI>
		    <LI> strand -- Either '+' or '-'.  If '-', then the
		               alignment is to the
		               reverse-complemented source.  </LI>
		    <LI> srcSize -- The size of the entire source
		               sequence, not just the parts involved
		               in the alignment.  </LI>
		    <LI> text -- The nucleotides (or amino acids) in
		               the alignment and any insertions
		               (dashes) as well.  </LI>
		    </UL>

		    <P><B>A Simple Example</B><P> 
		    Here is a simple example of a three alignment
		    blocks derived from five starting sequences.
		    Repeats are shown as lowercase, and each block may
		    have a subset of the input sequences.  All
		    sequence columns and rows must contain at least one nucleotide
		    (no full columns or rows of insertions).
<FONT color="#0000FF"><PRE><TT> ##maf version=1 scoring=tba.v8
 # tba.v8 (((human chimp) baboon) (mouse rat))
 # multiz.v7
 # maf_project.v5 _tba_right.maf3 mouse _tba_C
 # single_cov2.v4 single_cov2 /dev/stdin

 a score=23262.0
 s hg16.chr7    27578828 38 + 158545518 AAA-GGGAATGTTAACCAAATGA---ATTGTCTCTTACGGTG
 s panTro1.chr6 28741140 38 + 161576975 AAA-GGGAATGTTAACCAAATGA---ATTGTCTCTTACGGTG
 s baboon         116834 38 +   4622798 AAA-GGGAATGTTAACCAAATGA---GTTGTCTCTTATGGTG
 s mm4.chr6     53215344 38 + 151104725 -AATGGGAATGTTAAGCAAACGA---ATTGTCTCTCAGTGTG
 s rn3.chr4     81344243 40 + 187371129 -AA-GGGGATGCTAAGCCAATGAGTTGTTGTCTCTCAATGTG

 a score=5062.0
 s hg16.chr7    27699739 6 + 158545518 TAAAGA
 s panTro1.chr6 28862317 6 + 161576975 TAAAGA
 s baboon         241163 6 +   4622798 TAAAGA
 s mm4.chr6     53303881 6 + 151104725 TAAAGA
 s rn3.chr4     81444246 6 + 187371129 taagga

 a score=6636.0
 s hg16.chr7    27707221 13 + 158545518 gcagctgaaaaca
 s panTro1.chr6 28869787 13 + 161576975 gcagctgaaaaca
 s baboon         249182 13 +   4622798 gcagctgaaaaca
 s mm4.chr6     53310102 13 + 151104725 ACAGCTGAAAATA</TT></PRE></FONT>


	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>

    <TABLE CELLPADDING=5 BORDER=0 WIDTH="100%"><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="WIG"></A><FONT SIZE="4"><B>&nbsp; 
		    Wiggle Format (WIG) 
		    </B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>

		    The wiggle (WIG) format allows display of
		    continuous-valued data in track format.  This is
		    useful for GC percent, probability scores, and
		    transcriptome data.

		    <P><B>General Structure</B><P> The .wig format is
		    line-oriented.  <!-- The first section begins with <A
		    HREF="http://genome.ucsc.edu/goldenPath/help/customTrack.html#BROWSER">browser
		    lines</A> that configure the overall display of
		    the Genome Browser window when your annotation
		    file is uploaded. Each line defines one display
		    attribute. 

		    <P> The browser lines are followed by a <A
		    <P> The browser lines are followed -->
		    The wiggle data are preceeded by a <A
		    HREF="http://genome.ucsc.edu/goldenPath/help/customTrack.html#TRACK">
		    track definition line</A>, which
		    adds a number of options for
		    controlling the default display of this track.

		    <P> Following the track definition line are the
		    track data, which can be entered in several
		    different formats described below.  Three
		    formats were developed to allow the file to be
		    written most compactly, depending on the density of
		    the data:
		    <UL>

		    <LI> <B>BED</B> format with no declaration line and four columns of data:
<FONT color="#0000FF"><PRE>  <I>chromA</I>  <I>chromStartA</I>  <I>chromEndA</I>  <I>dataValueA</I>
  <I>chromB</I>  <I>chromStartB</I>  <I>chromEndB</I>  <I>dataValueB</I></PRE></FONT>

		    <LI> <B>variableStep</B> two-column data; started by a
		    declaration line and followed with chromosome positions and data values:
<FONT color="#0000FF"><PRE>  <B>variableStep</B>  <B>chrom=</B><I>chrN</I>  <B>[span=</B><I>windowSize</I><B>}</B>
  <I>chromStartA</I>  <I>dataValueA</I>
  <I>chromStartB</I>  <I>dataValueB</I></PRE></FONT>

		    <LI> <B>fixedStep</B> single-column data; started by a
		    declaration line and followed with data values:
<FONT color="#0000FF"><PRE>  <B>fixedStep</B>  <B>chrom=</B><I>chrN</I>  <B>start=</B><I>position</I>  <B>step=</B><I>stepInterval</I>  <B>[span=</B><I>windowSize</I><B>]</B>
  <I>dataValue1</I>
  <I>dataValue2</I></PRE></FONT>
		    </UL>

		    <P><B>Parameters for wiggle track definition lines</B></P>
		    All options are placed in a single line separated by spaces:
<FONT color="#0000FF"><PRE>  <B>track type=wiggle_0 name=</B><I>track_label</I> <B>description=</B><I>center_label</I>
<B>        visibility=</B><I>display_mode</I> <B>color=</B><I>r,g,b</I> <B>altColor=</B><I>r,g,b</I>
<B>        priority=</B><I>priority</I> <B>autoScale=</B><I>on|off</I>
<B>        gridDefault=</B><I>on|off</I> <B>maxHeightPixels=</B><I>max:default:min</I>
<B>        graphType=</B><I>bar|points</I> <B>viewLimits=</B><I>lower:upper</I>
<B>        yLineMark=</B><I>real-value</I> <B>yLineOnOff=</B><I>on|off</I>
<B>        windowingFunction=</B><I>maximum|mean|minimum</I> <B>smoothingWindow=</B>off|2-16
</I></PRE></FONT>
		    The track type with version is REQUIRED, and it currently must be <I>wiggle_0</I>:
		    <FONT color="#0000FF"><PRE>  <B>type wiggle_0</B></PRE></FONT>
		    The remaining values are OPTIONAL:
		    <FONT color="#0000FF"><PRE>  <B>name</B>              <I>trackLabel</I>           # default is "User Track"
  <B>description       </B><I>centerlabel         </I> # default is "User Supplied Track"
  <B>visibility        </B><I>full|dense|hide     </I> # default is hide
  <B>color             </B><I>RRR,GGG,BBB         </I> # default is 255,255,255
  <B>altColor          </B><I>RRR,GGG,BBB         </I> # default is 128,128,128
  <B>priority          </B><I>N                   </I> # default is 100
  <B>autoScale         </B><I>on|off              </I> # default is off
  <B>gridDefault       </B><I>on|off              </I> # default is off
  <B>maxHeightPixels   </B><I>max:default:min     </I> # default is 128:128:11
  <B>graphType         </B><I>bar|points          </I> # default is bar
  <B>viewLimits        </B><I>lower:upper         </I> # default is range found in data
  <B>yLineMark         </B><I>real-value          </I> # default is 0.0
  <B>yLineOnOff        </B><I>on|off              </I> # default is off
  <B>windowingFunction </B><I>maximum|mean|minimum</I> # default is maximum
  <B>smoothingWindow   </B><I>off|[2-16]          </I> # default is off</PRE></FONT>

		    A functional description of these options can be
		    seen in the <A target=_blank
		    HREF="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=panTro1&org=Chimp&c=chr6&g=quality">Quality
		    Scores track</A>.

		    <P><B>Data Values</B></P>

		    Wiggle track data values can be integer or real,
		    positive or negative values. Chromosome positions
		    are specified as 1-relative.  The first chromosome
		    position is 1.  The last position in a chromosome
		    of length N would be N.  Only positions specified
		    have data.  Positions not specified do not have
		    data and will not be graphed.  All positions specified
		    in the input data must be in numerical order.

		    <UL><LI>The BED format is a subset of the previous description, with four columns of data:
<FONT color="#0000FF"><PRE> <I>chrom chromStart chromEnd dataValue</I></PRE></FONT> 
		    In many cases this format will be too verbose to
		    conveniently specify a large number of data
		    values.  Two other two methods of specifying
		    data were developed to be more space efficient.</LI>
<P>
		    <LI>The <B>variableStep</B> two-column data format is the most commonly used
		    data format.  It begins with a declaration line:
<FONT color="#0000FF"><PRE>  <B>variableStep chrom=</B><I>chrN</I> <B>[span=</B><I>windowSize</I><B>]</B></PRE></FONT>
		    and is followed with two-column data representing
		    chromosome position and data value.  The declaration line starts with the word
		    <B>variableStep</B> and is followed by a
		    specification for chromosome.  The optional
		    <B>span</B> allows more efficient encoding of a
		    single value for a contiguous run of bases
		    beginning at the specified position.
<P>
		    <LI>The <B>fixedStep</B> format takes a single
		    column data, and is used when there is a regular 
		    interval between new data values.  This is used for
		    data values that apply to a constant size run of bases.
		    The fixedStep single-column data format begins with the
		    declaration line: 
<FONT color="#0000FF"><PRE>  <B>fixedStep chrom=</B><I>chrN</I> <B>start=</B>chromStart<I></I> <B>step=</B><I>stepInterval</I> <B>[span=</B><I></I>]</PRE></FONT>
		    and is followed by a single-column of data values.
		    The declaration line starts with the word
		    fixedStep, and it must include specifications for
		    chrom, start, and step.
		    </UL>
		    <P><B>Example</B></P>
		    Specifying 28 separate data points in three tracks on chr19 in the region 59,302,001 to 59,311,000<BR>
		    (view this example as a custom track in the
<A HREF="/cgi-bin/hgTracks?db=hg16&hgt.customText=http://genome.ucsc.edu/encode/wiggleExample.txt"
TARGET=_blank>browser</A>)
		    <FONT color="#0000FF"><PRE>
browser position chr19:59302001-59311000
browser hide all
browser pack refGene encodeRegions
browser full altGraph
#	300 base wide bar graph, autoScale is on by default == graphing
#	limits will dynamically change to always show full range of data
#	in viewing window, priority = 20 positions this as the second graph
track type=wiggle_0 name="Bed Format" description="BED format" visibility=full color=200,100,0 altColor=0,100,200 priority=20
chr19 59302001 59302300 -1.0
chr19 59302301 59302600 -0.75
chr19 59302601 59302900 -0.50
chr19 59302901 59303200 -0.25
chr19 59303201 59303500 0.0
chr19 59303501 59303800 0.25
chr19 59303801 59304100 0.50
chr19 59304101 59304400 0.75
chr19 59304401 59304700 1.00
#	150 base wide bar graph at arbitrarily spaced positions,
#	threshold line drawn at y=11.76
#	autoScale off viewing range set to [0:25]
#	priority = 10 positions this as the first graph
track type=wiggle_0 name="variableStep" description="variableStep format" visibility=full autoScale=off viewLimits=0.0:25.0 color=255,200,0 yLineMark=11.76 yLineOnOff=on priority=10
variableStep chrom=chr19 span=150
59304701 10.0
59304901 12.5
59305401 15.0
59305601 17.5
59305901 20.0
59306081 17.5
59306301 15.0
59306691 12.5
59307871 10.0
#	200 base wide points graph at every 300 bases, 50 pixel high graph
#	autoScale off and viewing range set to [0:1000]
#	priority = 30 positions this as the third graph
track type=wiggle_0 name="fixedStep" description="fixed step" visibility=full autoScale=off viewLimits=0:1000 color=0,200,100 maxHeightPixels=100:50:20 graphType=points priority=30
fixedStep chrom=chr19 start=59307401 step=300 span=200
1000
 900
 800
 700
 600
 500
 400
 300
 200
 100
                    </PRE></FONT>
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
</BODY></HTML>
