<table width="80%"><tr><td>
<h2>Description.</h2>

<p>This track represents regions (transfrags) of chromosomes 6, 7, 13,
14, 19, 20, 21, 22, X, and Y that are thought to be transcribed based
on the transcriptome data from <a
href="http://www.affymetrix.com">Affymetrix</a>. This track is a
preview of the phase two Affymetrix transcriptome project and is based
on hybridzation of mRNA from the SK-N-AS cell line as seen in the <a
href="../cgi-bin/hgTrackUi?g=affyTranscription">Transcription</a>
track.. Data from seven additional cell lines will be available when
the project is completed.  While the coverage of the genome is much
larger and the probe density greater, the data generation is similar
to the phase one project carried out on chromosomes 21 and 22
(Kapranov, P., Cawley, S. E., Drenkow, J., Bekiranov, S, Strausberg,
R. L., Fodor, S.P.A. and Gingeras, T.R. (2002) <a
href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=11988577">"Large-Scale
Transcriptional Activity in Chromosomes 21 and 22."</A>
<i>Science</i>, 296(5569):916-9).


<p>Transfrags that overlap a putative pseudogene or that have a strong
BLAT hit elsewhere in the genome are colored in a lighter blue. By
default these transfrags are hidden from display, but can be viewed by
changing the options on the track configuration page. For users wishing
to use the table browser to access the data the filtering status of a
transfrag is stored by its score field in the database:

<pre>
   0 - Passes all filters
   1 - Overlaps a pseudogene
   2 - Overlaps a blat hit
   3 - Overlaps both a pseudogene and a blat hit.
</pre>

<h2>Methods</h2>

<p>mRNA was isolated and hybridized to tiling GeneChips that contain a
probe ever 5bp over chromosomes 6, 7, 13, 14, 19, 20, 21, 22, X, and Y
(over 74 million probes total). The resulting data was normalized and
smoothed using the Hodges-Lehmann estimator associated with the
Wilcoxon signed rank statistic of the pm - mm values that lie within
+/- 30bp of a sliding window centered at every genomic coordinate. 
This is very similar to the method used previously on the phase one data
(Kampa, D.,
Cheng, J., Kapranov, P., Yamanaka, M., Brubaker, S., Cawley, S. Drenkow, J.,
Piccolboni, A., Bekiranov, S., Helt, G., Tammana, H., and T. R. Gingeras
(2004) <a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=14993201">"Novel RNAs identified from a comprehensive analysis of the
transcriptome of human chromosomes 21 and 22."</a> Genome Res. 14: 331-342).


<p>To determine the locations of the transfrags a threshold of 13.93
(which gives an average FP rate of ~ 0.05 in negative bacterial
controls) was applied to all the signal graphs and defined coordinates
as positive.  Positive coordinates separated by gaps less than or
equal to 30bp were merged.  Resulting regions that were less than 50bp
in length were filtered out.  Finally, 12bp were added to all start
coordinates and 13bp to all stop coordinates (i.e. interbase
coordinates) which shifts the coordinates to be centered over probes
as opposed to being left justified with respect to probes as is the
case in the raw and graph data. Transfrags were filtered to remove Repeat
Masker Repeats and low complexity repeats, but not pseudogenes.


</td></tr></table>

