
<h2>Description</h2>

<p>The <a
href="http://www.affymetrix.com/products/arrays/specific/exon.affx">
Human Exon 1.0 ST </a>GeneChip contains over 1.4 million probe sets
designed to interrogate individual exons rather than 3' end of transcripts
as in traditional GeneChips. Exons were derived from a variety of
annotations which have been divided into the classes Core, Extended
and Full.  <ul> <li> <b>Core:</b> <a
href="http://www.ncbi.nlm.nih.gov/RefSeq/">RefSeq Transcripts</a>,
Full-Length <a
href="http://www.ncbi.nlm.nih.gov/Genbank/index.html">GenBank</a>
mRNAs.</li> <li> <b>Extended:</b> <a
href="http://www.ncbi.nlm.nih.gov/dbEST/">dbEst</a> Alignments, <a
href="#ensembl">Ensembl </a>annotations, syntenic mRNA from rat and
mouse, microRNA annotations, Mitomap annotations, <a
href="#vega">Vegagene</a>,<a href="#vega">VegaPseudogene</a> </li>

<li> <b>Full:</b>
<a href="#geneid">Geneid</a>,
<a href="#genscan">Genscan</a>,
<a href="#genscan">GenscanSubopt</a>,
<a href="#exoniphy">Exoniphy</a>,
<a href="#rnagene">RNAgene</a>,
<a href="#sgpgene">SgpGene</a>,
<a href="#twinscan">Twinscan</a>,
</li>
</ul>

<p>Probe sets are colored by their class with the Core probe sets being
the darkest and the Full being the lightest color. Additionally probe
sets that do not overlap the exons of a transcript cluster, but fall
inside of its introns are considered bounded by that transcript
cluster and are colored slightly lighter. Probe sets that overlap the
coding portion of the Core class are colored slightly darker.

<h2> Credits </h2>

The exons interrogated by the probe sets displayed in this track are
from the Affymetrix Human Exon 1.0 GeneChip and were derived from a
number of sources. In addition to the millions of cDNA sequences
contributed to the <a
href="http://www.ncbi.nlm.nih.gov/Genbank/index.html">GenBank</a>, <a
href="http://www.ncbi.nlm.nih.gov/dbEST/">dbEst</a> and <a
href="http://www.ncbi.nlm.nih.gov/RefSeq/">RefSeq</a> databases by
individual labs and scientists the following annotations were used:

<p><a name="ensembl"> <b>Ensembl:</b> Hubbard T, <i>et al.</i>. "The Ensembl genome database project." Nucleic Acids Research 30(1), 38-41 (2002).</a>
<p><a name="exoniphy"><b>Exoniphy:</b> Siepel A and Haussler D. "Computational identification of evolutionarily conserved exons." Proc. 8th Annual Int'l Conf. on Research in Computational Biology, pp. 177-186 (2004). </a>
<p><a name="geneid"> <b>Geneid:</b> Parra G, Blanco E, and Guigo R. "Geneid in Drosophila", Genome Research 10(4):511-515 (2000).</a>
<p><a name="genscan"><b>Genscan:</b> Burge C and Karlin S., "Prediction of Complete Gene Structures in Human Genomic DNA." J. Mol. Biol. 268(1), 78-94 (1997).</a>
<p><a name="rnagene"> <b>Rna Gene:</b> Lowe TM , Eddy SR. "tRNAscan-SE: A Program for Improved Detection of Transfer RNA Genes in Genomic Sequence." Nucleic Acids Research, 25:955-964 (1997). Other Sean Eddy Lab annotations</a>
<p><a name="sgpgene"> <b>SGP Gene:</b> Wiehe T, Gebauer-Jung S, Mitchell-Olds T, Guigo R. "SGP-1: prediction and validation of homologous genes based on sequence alignments." Genome Research. 11(9) 1574-83, (2001)</a>
<p><a name="twinscan"> <b>Twinscan:</b> Korf I, Flicek P, Duan D and Brent MR. "Integrating genomic homology into gene structure prediction." Bioinformatics 17, S140-148 (2001).</a>
<p><a name="vega"><b>Vega:</b> The HAVANA group, Wellcome Trust Sanger Institute.</a>
