
<table width="80%"><tr><td>
<h2>Description</h2>

<p>Transcriptome data from tiling GeneChips produced by <a
href="http://www.affymetrix.com">Affymetrix</a>.  For the ten
Chromosomes 6, 7, 13, 14, 19, 20, 21, 22, X, and Y probes were tiled
every 5bp in areas that were not repeat masked (over 74 million
probes) and hybridized to mRNA from the SK-N-AS cell line. This data
is a preview of the phase two of the transcriptome project which will
include data from 7 additional cell lines when completed. While the
coverage of the genome is much larger and the probe density greater,
the general method is similar to the phase one project carried out on
chromosomes 21 and 22  (Kapranov, P.,
Cawley, S. E., Drenkow, J., Bekiranov, S, Strausberg, R. L., Fodor,
S.P.A. and Gingeras, T.R. (2002) <a
href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=11988577">"Large-Scale
Transcriptional Activity in Chromosomes 21 and 22."</A> <i>Science</i>, 296(5569):916-9).

<p>The track is colored blue in areas that are thought to be
transcribed as a statistically significant level as described in the
accompanying <a
href="../cgi-bin/hgTrackUi?g=affyTransfrags">Transfrags</a>
(transcribed fragments) track. Transfrags that have significant BLAT
hit elsewhere in the genome are colored a lighter shade of
blue. Transfrags that overlap putative pseudogenes are colored an even
lighter shade of blue. While the raw data is based on PM - MM values
and may contain negative values the track has a minimum value of 0 for
visualization purposes.

<h2>Analysis</h2>

<p>For each data point probes within 30bp on either side were used to
help better estimate the expression level of a particular probe. This
helps to smooth the data and produce a more robust estimate of the
transcription level at a particular genomic location. Specifically the
analysis method used was as follows:

<ul>

    <li>Replicate arrays were quantile-normalized and the median
    intensity (using both pm and mm intensities) of each array was
    scaled to a target value of 44.

    <li>The expression level was estimated for each mapped probe position, i, by 
    <ul>
       <li>collecting all the probe pairs that fell within a window of +\- 30bp
    
       <li>calculating all non-redundant pairwise averages of pm - mm
       values of all probe pairs in the window

       <li>taking the median of all resulting pairwise averages.
   </ul>

   <li>The resulting signal value is the Hodges-Lehmann estimator
   associated with the Wilcoxon signed rank statistic of the pm - mm
   values that lie within &#177 30bp of the sliding window centered at
   every genomic coordinate

</ul>
</td></tr></table>
