<table><tr><td>
<h2>Description</h2>

<p>This track represents regions of chromosomes 6, 7, 13,
14, 19, 20, 21, 22, X, and Y that are thought to be transcribed (transfrags),
based on the transcriptome data from <a
href="http://www.affymetrix.com" TARGET = _BLANK>Affymetrix</a>. This track is a
preview of the Phase Two Affymetrix transcriptome project and is based
on hybridzation of mRNA from the SK-N-AS cell line as seen in the <a
href="../cgi-bin/hgTrackUi?g=affyTranscription">Transcription</a>
track. Data from seven additional cell lines will be available when
the project is completed.  While the coverage of the genome is much
larger and the probe density greater, the data generation is similar
to the Phase One project carried out on chromosomes 21 and 22
(Kapranov, P., Cawley, S. E., Drenkow, J., Bekiranov, S, Strausberg,
R. L., Fodor, S.P.A. and Gingeras, T.R. (2002) "<a
href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=11988577" TARGET = _BLANK>Large-Scale
Transcriptional Activity in Chromosomes 21 and 22.</A>"
<EM>Science</EM>, 296(5569):916-9).


<p>Transfrags that overlap a putative pseudogene or that have a strong
BLAT hit elsewhere in the genome are colored in a lighter blue. By
default these transfrags are hidden from display, but can be viewed by
changing the options on the track configuration page.  

<P>The Score value is interpreted as follows:

<pre>
   0 - Passes all filters
   1 - Overlaps a pseudogene
   2 - Overlaps a blat hit
   3 - Overlaps both a pseudogene and a blat hit
</pre>

<h2>Methods</h2>

<p>mRNA was isolated and hybridized to tiling GeneChips that contain a
probe every 5 bp over chromosomes 6, 7, 13, 14, 19, 20, 21, 22, X, and Y
(over 74 million probes total). The resulting data were normalized and
smoothed using the Hodges-Lehmann estimator associated with the
Wilcoxon signed-rank statistic of the pm - mm values that lie within
&#177 30 bp of a sliding window centered at every genomic coordinate. 
This is very similar to the method used previously on the phase one data
(Kampa, D.,
Cheng, J., Kapranov, P., Yamanaka, M., Brubaker, S., Cawley, S. Drenkow, J.,
Piccolboni, A., Bekiranov, S., Helt, G., Tammana, H., and T. R. Gingeras
(2004) "<a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=14993201" TARGET = _BLANK>Novel RNAs identified from a comprehensive analysis of the
transcriptome of human chromosomes 21 and 22.</a>" <EM>Genome Res.</EM> 14: 331-342).


<p>To determine the locations of the transfrags, a threshold of 13.93
(which gives an average FP rate of ~ 0.05 in bacterial negative 
controls) was applied to all the signal graphs and defined coordinates
as positive.  Positive coordinates separated by gaps 30 bp or less 
were merged.  Resulting regions that were smaller than 50 bp
in length were filtered out.  Transfrags were filtered to remove Repeat 
Masker Repeats and low complexity repeats, but not pseudogenes.
Finally, to center the coordinates over the probes for the display, 12 bp 
were added to all starts and 13 bp to all stop coordinates.  (The raw and 
graph data are left-justified with respect to probes).


</td></tr></table>

