<html>

<head>
<meta http-equiv=Content-Type content="text/html; charset=windows-1252">
<meta name=Generator content="Microsoft Word 11 (filtered)">
<title>Description: Affy Txn sRNA signals</title>

<style>
<!--
 /* Font Definitions */
 @font-face
	{font-family:Tahoma;
	panose-1:2 11 6 4 3 5 4 4 2 4;}
 /* Style Definitions */
 p.MsoNormal, li.MsoNormal, div.MsoNormal
	{margin:0in;
	margin-bottom:.0001pt;
	font-size:12.0pt;
	font-family:"Times New Roman";}
h2
	{margin-right:0in;
	margin-left:0in;
	font-size:18.0pt;
	font-family:"Times New Roman";
	font-weight:bold;}
a:link, span.MsoHyperlink
	{color:blue;
	text-decoration:underline;}
a:visited, span.MsoHyperlinkFollowed
	{color:purple;
	text-decoration:underline;}
p
	{margin-right:0in;
	margin-left:0in;
	font-size:12.0pt;
	font-family:"Times New Roman";}
p.MsoAcetate, li.MsoAcetate, div.MsoAcetate
	{margin:0in;
	margin-bottom:.0001pt;
	font-size:8.0pt;
	font-family:Tahoma;}
span.msoIns
	{text-decoration:underline;
	color:teal;}
span.msoDel
	{text-decoration:line-through;
	color:red;}
@page Section1
	{size:8.5in 11.0in;
	margin:.5in .5in .5in .5in;}
div.Section1
	{page:Section1;}
 /* List Definitions */
 ol
	{margin-bottom:0in;}
ul
	{margin-bottom:0in;}
-->
</style>

</head>

<body lang=EN-US link=blue vlink=purple>

<div class=Section1>

<h2>Description (change track name to “Affy Txn sRNA signals”)</h2>

<p class=MsoNormal style='margin-bottom:12.0pt'>This track displays
transcriptome data from tiling GeneChips produced by <a
href="http://www.affymetrix.com" target="_blank">Affymetrix</a>. For the complete
non-repetitive portion of the human genome, more than 256 million probe pairs
were tiled every 5 bp in non-repeat-masked areas and hybridized to whole-cell
short RNA (&lt;200 nucleotides) from 2 different cell lines (HeLa cell lines
are female and contain no data for chrY). Data was collected using a
strand-specific methodology which allows separate measurements for top strand
(sense) and bottom strand (antisense) RNA signals. For experimental details and
results, see Kapranov <em>et al</em>. in the References section below. This
track contains the signal level estimated for each mapped probe position after
hybridization with short RNA samples.</p>

<p class=MsoNormal align=center style='text-align:center'><img border=0
width=32 height=32 src="affyTxnPhase3S_files/image001.gif"></p>

<h2>Display Conventions and Configuration</h2>

<p>This annotation follows the display conventions for composite tracks. The
subtracks within this annotation may be configured in a variety of ways to
highlight different aspects of the displayed data. The graphical configuration
options are shown at the top of the track description page, followed by a list
of subtracks. For more information about the graphical configuration options,
click the <a
href="Documents%20and%20Settings/pkapra/Local%20Settings/Temporary%20Internet%20Files/Local%20Settings/Temporary%20Internet%20Files/goldenPath/help/hgWiggleTrackHelp.html">Graph
configuration help</a> link. To display only selected subtracks, uncheck the
boxes next to the tracks you wish to hide. </p>

<p>A separate track named “Affy Txn sRNA regions” contains transcribed
fragments (transfrags) representing short RNAs (see Kapranov <em>et al</em>. in
the References section below). While the raw data are based on perfect match
minus mismatch (PM - MM) probe values and may contain negative values, the
track has a minimum value of zero for visualization purposes. Likewise, the
probes with high signal values will be cut off at the intensity of 150 and will
appear to have similar magnitude.</p>

<h2>Methods</h2>

<p>Genomic regions correspond to the sRNA transcripts are mapped for the plus
or the minus strands of the genome and for HepG2 and HeLa cell lines
respectively. The following analysis method was used: </p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.5in'>1.<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Replicate
arrays were quantile-normalized and the median intensity (using both PM and MM
intensities) of each array was scaled to a target value of 20. </p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.5in'>2.<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>The
expression level was estimated for each mapped probe position without
application of a smoothing window. This was followed by generation of
transfrags (transcribed fragments) in the track labeled “Affy Txn sRNA regions”
as below: </p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:-1.0in'><span
style='font-size:10.0pt;font-family:"Courier New"'>o<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span>probes with intensities corresponding to the 98th percentile of
all the PM-MM probe intensities were identified.</p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:-1.0in'><span
style='font-size:10.0pt;font-family:"Courier New"'>o<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span>transcribed regions (transfrags) were determined by connecting
adjacent positive probes with a maximum allowable gap of 4 nucleotides and a
minimum run of 7 nucleotides (at lease 2 consecutive positive probes with no
negative probes in-between), and must be called positive reproducibly in both biological
replicas.</p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:-1.0in'><span
style='font-size:10.0pt;font-family:"Courier New"'>o<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span>to further prioritize the transfrag maps, the intensity of all the
transfrag was calculated based on the composite signals of the two biological
replicas, and transfrags corresponding to the top 25 quartile were taken and used
for further analysis.</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.5in'>3.<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>In
addition, several successive steps to avoid potentially cross-hybridizing sequences
and/or those that mapped to potentially non-unique regions: transfrags which
overlap pseudogenes, or harboring low complexity or repetitive sequences were
discarded. We thus believe that our maps represent a very conservative view of
the sRNA population in the cell.</p>

<h2>Credits</h2>

<p>Data generation and analysis was performed by the transcriptome group at
Affymetrix together with assistance from colleagues at the University of Leipzig, Fraunhofer Institute, and University of Vienna: Kapranov P., Cheng J.,
Dike S., Nix D.A., Duttagupta R., Willingham A.T., Stadler P.F., Hertel J.,
Hackermüller J., Hofacker I.L., Bell I., Cheung E., Drenkow J., Dumais E.,
Patel S., Helt G., Ganesh M., Ghosh S., Piccolboni A., Sementchenko V., Tammana
H., Gingeras T.R.</p>

<p>Questions or comments about this annotation? Email. </p>

<h2>References</h2>

<p>Kapranov <em>et al</em>. <a
href="http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=17510325&amp;ordinalpos=1&amp;itool=EntrezSystem2.PEntrez.Pubmed.Pubmed_ResultsPanel.Pubmed_RVDocSum"
target="_blank">RNA Maps Reveal New RNA Classes and a Possible Function for
Pervasive Transcription</a>. <em>Science</em> <b>316</b>(5830), 1484-8 (2007).</p>

<p class=MsoNormal>&nbsp;</p>

</div>

</body>

</html>
