<h2>Description</h2>

<p>This track summarizes the alternative splicing seen in the mRNA and
EST tracks. The blocks represent exons and the lines are the possible
splice junctions. The track will draw itself such that no exons
overlap each other making alternative events easier to see if the track
is in full mode and the resolution is approximately gene level.

<p> To help filter out the noise present in the EST libraries both
exons and splice junctions are filtered using both orthologous mouse
transcripts and number of times an exon or intron is seen in human
transcript libraries. Only those exons and splice junctions which have
an orthologous exon or splice junction present in the mouse
transcriptome, or are present three or more times, are kept. Transcripts
marked in GenBank as mRNA contribute an extra count to reflect the
fact that they are usually of higher quality. This process is similar
to that presented in: <a
href="http://www-smi.stanford.edu/projects/helix/psb04/sugnet.pdf">
"Transcriptome and Genome Conservation of Alternative Splicing Events
in Humans and Mice" C.W. Sugnet, W.J. Kent, M. Ares Jr., and
D. Haussler.</a>

<h2>Methods</h2>
This track was generated in the following manner:

<ol>
<li> The splicing graphs for each genome were generated separately
from their native EST and mRNA transcripts using the following
process: <ul>

    <li> The mRNAs and ESTs were aligned to the genomic sequence using <a
    href="../cgi-bin/hgBlat?db=$db">BLAT</a>. A near-best-in-genome filter was
    applied such that only alignments with 97% identity over 90% of the
    transcript and with a score no more than 0.5% lower than the best
    score were kept.</li>

    <li> The ESTs and mRNAs were oriented by examining the splice
    sites used in the genomic sequence. Only consensus splice sites,
    GT->AG and the less common GC->AG were used to orient the
    transcripts.</li>

    <li> Alignments were clustered together by sequence overlap in
    exons. As new splice sites were discovered, they were entered into
    the graphs as vertexes, and the exons, introns, and splice
    junctions connecting them were recorded as edges. Each graph is
    considered to be a single locus, although they may be fragments of
    an actual gene structure. The supporting mRNA and EST accessions
    for each edge were also stored.</li>

    <li> Truncated transcripts were extended by overlap with other
    transcripts to the next consensus splice site. This avoids
    keeping vertexes in the graph that are not true splice sites.</li>
    </ul>

<li> Once the splicing graphs were constructed independently for both
human and mouse they were mapped to each other using the the whole
genome <A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1"
TARGET=_blank> Mouse Net</a> alignments (viewable on the browser as
the Mouse Net track). Exons and splice junctions that were common to
both or that occur three or more times in human are kept in the
splicing graph and displayed. When counting the number of times an
exon or splice junction were included in human transcripts those
designated as mRNA are weighted more than those designated as EST.</li>
</ol>


