<h2>Description</h2>

<p>This track summarizes the alternative splicing seen in the mRNA and
EST tracks. The blocks represent exons and the lines are the possible
splice junctions.  When the track is in full mode
and the resolution is approximately gene level the exons are laid out
such that alternative exons don't overlap each other and are easier to
see.  The more ESTs and mRNAs that contained that exon or splice
junction the darker they are drawn.

<p> To help filter out the noise present in the EST libraries both
exons and splice junctions have been filtered. Only those exons and
splice junctions which have an orthologous exon or splice junction
present in the mouse transcriptome, or are present three or more
times, are kept. This process is similar to that presented in: <a
href="http://www-smi.stanford.edu/projects/helix/psb04/sugnet.pdf">
"Transcriptome and Genome Conservation of Alternative Splicing Events
in Humans and Mice" C.W. Sugnet, W.J. Kent, M. Ares Jr., and
D. Haussler.</a>

<h2>Methods</h2>
The outline for generating this track is:

<ol>
<li> For both the mouse and human genomes separately.
    <ul>

    <li> Align mRNAs and ESTs to the genomic sequence using <a
    href="../cgi-bin/hgBlat">BLAT</a>. A near best in genome filter is
    applied where only alignments with 97% identity over 90% of the
    transcript and with a score no more than .5% lower than the best
    score are kept.</li>

    <li> Retrieve genomic sequence and use it to orient ESTs and mRNAs
    using consensus splice sites, GT->AG, and the less common
    GC->AG.</li>

    <li> Cluster alignments together by sequence overlap in exons. As
    new splice sites are discovered, they are entered into the graphs
    as vertexes, and the exons, introns, and splice junctions
    connecting them are recorded as edges. Each graph is considered to
    be a single locus, although they may be fragments of an actual
    gene structure. The supporting mRNA and EST accessions for each
    edge are also stored.</li>

    <li> Extend truncated transcripts by overlap with other
    transcripts to the next consensus splice site.  This avoids
    keeping vertexes in the graph that are not true splice sites.</li>

    </ul>

<li> Using the graphs generated independently for both the human
and mouse genomes find exons and splice junctions that are common to
both or that occur three or more times in human. Ortholgous splicing graphs
are identified using the Mouse Net track.</li>

</ol>
