<H2>Description</H2>
<P>
The CCDS Genes track shows the $organism genome high-confidence gene annotations
from the
<A HREF="http://www.ncbi.nlm.nih.gov/CCDS/" TARGET=_blank>Consensus CDS (CCDS) project</A>.
This project is a collaborative effort to identify a core set of human protein
coding regions that are consistently annotated and of high quality. The long term goal is to support 
convergence towards a standard set of gene annotations on the human genome.
</P>
<H4>Collaborators</H4>
   <a href="http://www.ebi.ac.uk/" TARGET=_blank>European Bioinformatics Institute (EBI)</a><br>
   <a href="http://www.ncbi.nlm.nih.gov" TARGET=_blank>National Center for Biotechnology Information (NCBI)</a><br>
   <a href="http://www.cbse.ucsc.edu/" TARGET=_blank>University of California, Santa Cruz (UCSC)</a><br>
   <a href="http://www.sanger.ac.uk/" TARGET=_blank>Wellcome Trust Sanger Institute (WTSI)</a><br>

<H2>Methods</H2>
<OL>
    <LI> CDS annotations of the $organism genome are obtained from two sources:
        <UL>
        <LI> <A HREF="http://www.ncbi.nlm.nih.gov/RefSeq/index.html" TARGET=_blank>NCBI RefSeq</A>
        <LI> A union of the gene annotations from  <A HREF="http://www.ensembl.org/" TARGET=_blank>Ensembl</A>
             and Vega <A HREF="http://vega.sanger.ac.uk/" TARGET=_blank>Vega</A>, collectively known as <EM>Hinxton</EM>.
        </UL>
    <LI> genes with identical CDS genomic coordinates in both sets become CCDS candidates
    <LI> genes undergo quality evaluation, which currently uses the following criteria:
         <UL>
         <LI> have an initiating ATG, a valid stop-codon, and no in-frame stop-codons.
         <LI> can be translated from the genome without frameshifts
         <LI> have consensus splice-sites
         <LI> don't intersect putative pseudogenes predictions
         <LI> have supporting transcripts and protein homology
         <LI> have conservation evidence with other species
         </UL>
    <LI> all collaborators must agree to pass the quality evaluation
    <LI> a unique CCDS id is assigned to the CCDS, which links together
    all of the gene annotations with the same CDS.
</OL>

<H2>Credits</H2>
<P>
This track was produced at UCSC with data download from the
<A HREF="http://www.ncbi.nlm.nih.gov/CCDS/" TARGET=_blank>Consensus CDS (CCDS) project</A>
web site.
</P>

<H2>References</H2>
<P>
Pruitt K.D., Tatusova, T., and Maglott D.R.,
<A HREF="http://nar.oupjournals.org/cgi/content/full/33/suppl_1/D501" TARGET=_blank>
NCBI Reference Sequence (RefSeq): a curated non-redundant sequence database of genomes, transcripts and proteins</A>, 
<I>Nucl. Acids Res.</I>,33(1):D501-D504  2005 Jan 1;
</P>

<P>
Hubbard, T. <EM>et al</EM>., 
<A HREF="http://nar.oupjournals.org/cgi/content/abstract/30/1/38" 
TARGET=_blank>The Ensembl genome database project</A>, 
<I>Nucl. Acids Res.</I>, <B>30</B>(1), 38-41 (2002).</P>
