<H2>Description</H2>
<P>
This track shows alignments of the $organism genome with itself, using
a gap scoring system that allows longer gaps than traditional
affine gap scoring systems. The system can also tolerate gaps
in both sets of sequence simultaneously.  After 
filtering out the &quot;trivial&quot; alignments produced when
identical locations of the genome map to one another (e.g. chr22 maps to 
chr22), the remaining alignments point out areas of duplication within the 
$organism genome.
<P>
The chain track displays boxes joined together by either single or
double lines. The boxes represent aligning regions. Single lines
indicate gaps that are largely due to missing sequence in either the query
or target genome. Double lines represent gaps in the chain where
sequence is present in both but they do not align.
In cases where there are multiple
chains over a particular portion of the $organism genome, chains with
single-lined gaps are often due to processed pseudogenes, while chains
with double-lined gaps are more often due to paralogs and unprocessed
pseudogenes.
<P>
The display indicates the chromosome, strand, and location of the match
for each matching alignment (in thousands).

<H2>Methods</H2>
<P>
The genome is aligned to itself using 
<A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>blastz</A>.  
Trivial alignments are filtered out. The remaining alignments 
are converted into axt format
and the resulting axts are fed into axtChain.  AxtChain organizes all 
the alignments between a single target and a single query chromosome
into a group and makes a kd-tree out of all the gapless subsections
(blocks) of the alignments.  Next, maximally scoring chains of these
blocks are found by running a dynamic program over the kd-tree.  Chains
scoring below a threshold are discarded and the remaining chains are
displayed.

<H2>Credits</H2>
<P>Blastz was developed at
<A HREF="http://bio.cse.psu.edu" TARGET=_blank>Pennsylvania State University</A> by 
Scott Schwartz, Zheng Zhang, and Webb Miller with advice from
Ross Hardison.</P> 

<P>Lineage-specific repeats were identified by Arian Smit and his
program
<A HREF="http://repeatmasker.genome.washington.edu/cgi-bin/RepeatMasker" TARGET=_blank>
RepeatMasker</A>.</P>

<P>The axtChain program was developed at the University of California
at Santa Cruz by Jim Kent with advice from Webb Miller and David Haussler.</P>

<P>The browser display and database storage of the chains were made
by Hiram Clawson (<A
HREF="mailto:&#104;&#105;&#114;&#97;&#109;&#64;&#115;&#111;&#101;.&#117;&#99;sc.
e&#100;u">&#104;&#105;&#114;&#97;&#109;&#64;&#115;&#111;&#101;.&#117;&#99;sc.e&#100;u</A>)
and <A HREF="http://www.soe.ucsc.edu/~kent/">Jim Kent</A>.</P>

<H2>References</H2>

<P><A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>
Human-Mouse Alignments with BLASTZ</A>. Schwartz S, Kent WJ, Smit A,
Zhang Z, Baertsch R, Hardison R, Haussler H, and Miller W. (2003).
<em>Genome Res</em> 13(1):103-7.</P>

<P><A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" TARGET=_blank>
Scoring pairwise genomic sequence alignments</A>. Chiaromonte F, Yap VB,
Miller W. (2002). <em>Pac Symp Biocomput</em> 2002:115-26.
