<h2>Description</h2>
<p>
The Gencode Gene track (v3.1, March 2007) shows <b>high-quality manual
annotations</b> in the
ENCODE regions generated by the
<a href="http://genome.imim.es/gencode/" target="_blank">GENCODE project</a>.</p>
<p>
The gene annotations are colored based on the Havana annotation type.
"Known" and "Novel CDS" transcripts are colored <em>dark green</em>,
putative and unconfirmed are <em>light green</em>, polymorphic loci
are <em>purple</em>, pseudogenes are <em>blue</em>, and artifacts are 
<em>grey</em>. PolyA signals/sites are <em>brown/orange</em>,
respectively and Psuedo Poly A signals are <em>pink</em>.
The transcript and feature types are defined in more detail in the
accompanying table.
</p>
<p>The Gencode project recommends that the annotations
with known and validated transcripts; i.e., the types <em
 style="font-weight: bold;">Known</em>
and <em style="font-weight: bold;">Novel_CDS</em> (which are colored
dark green in the track
display) be used as the reference gene annotation.
</p>
<p><b><u>Release notes</u></b> (compared to v2.2, October 2005): </p>
<div style="margin-left: 40px;">
<ul>
  <li>Apart from usual additions/corrections, 69 loci (consisting of 132
transcripts) have been re-annotated based on RACE/array/sequencing
analyses performed within the Affymetrix/GENCODE collaboration
(see <span style="font-style: italic;">Methods</span> below, Denoeud 
<em>et al</em>., 2007 and The ENCODE Project Consortium, 2007).</li>
  <li><em>Polymorphic</em> gene type added</li>
  <li><em>polyA</em> features added</li>
  <li>Bug fix: The bug affecting frames of CDSs with missing
start or stop codons has been fixed.</li>
  <li>The experimental validation data contained in the "Gencode
Introns" track from the previous release has been integrated into the
Gencode Genes track by HAVANA annotators.<br>
  </li>
</ul>
</div>
<br>
<p>
<table border="1" bordercolor="#aaaaaa" cellpadding="4">
  <tbody>
    <tr>
      <th align="left">Type</th>
      <th align="left">Color</th>
      <th align="left">Description</th>
    </tr>
    <tr>
      <td>Known</td>
      <td><font color="#215b30">dark green</font></td>
      <td><b>Known protein coding genes</b> (<em>i.e.</em>, referenced
in <a style="font-weight: bold;" target="_blank"
 href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=gene">Entrez
Gene</a>)</td>
    </tr>
    <tr>
      <td>Novel_CDS</td>
      <td><font color="#215b30">dark green</font></td>
      <td><b>Have an open reading frame</b> (ORF) and are identical, or
have homology, to cDNAs or proteins but do not fall in the above
category; these can be known in the sense that they are represented by
mRNA sequences in the public databases, but they are <b>not yet represented
in <a target="_blank"
 href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=gene">Entrez
Gene</a></b> or have not received an official gene name. They can also
be novel in that they are not yet represented by an mRNA sequence in
human.</td>
    </tr>
    <tr>
      <td>Novel_transcript</td>
      <td><font color="#54bc00">light green</font></td>
      <td>Similar to Novel_CDS, however <b>cannot be assigned an
unambigous ORF</b>.</td>
    </tr>
    <tr>
      <td>Putative</td>
      <td><font color="#54bc00">light green</font></td>
      <td>Have identical, or have homology to <b>spliced ESTs</b>, but
are <b>devoid of a significant ORF and polyA features</b>; these are
generally short (two or three exon) genes or gene fragments.</td>
    </tr>
    <tr>
      <td>TEC</td>
      <td><font color="#54bc00">light green</font></td>
      <td>(<b><u>T</u>o <u>E</u>xperimentally <u>C</u>onfirm</b>)<br>
Single-exon objects (supported by multiple unspliced ESTs with polyA
sites and signals). </td>
    </tr>
    <tr>
      <td>Polymorphic</td>
      <td><font color="#a020f0">purple</font></td>
      <td> Have <b>functional transcripts in one haplotype</b>, and <b>"pseudo"
(non-functional) transcripts in another</b>.</td>
    </tr>
    <tr>
      <td>Processed_pseudogene</td>
      <td><font color="#005bbf"> blue</font></td>
      <td><b>Pseudogenes that lack introns</b> and are thought to arise
from <b>reverse transcription</b> of mRNA followed by reinsertion of
DNA into the genome. </td>
    </tr>
    <tr>
      <td>Unprocessed_pseudogene</td>
      <td><font color="#005bbf"> blue</font></td>
      <td><b>Pseudogenes that can contain introns</b>, as they are
produced by <b>gene duplication</b>.</td>
    </tr>
    <tr>
      <td>Artifact</td>
      <td><font color="#636863">grey</font></td>
      <td><b>Transcript evidence and/or its translation equivocal</b>.
Usually these arise from high-throughput cDNA sequencing projects which
submit automatic annotation, sometimes resulting in erroneous CDSs in
what turns out to be, for example, 3' UTRs. In addition HAVANA has
extended this category to include cDNAs with non canonical splice sites
due to deletion/sequencing errors.</td>
    </tr>
    <tr>
      <td>PolyA_signal</td>
      <td><font color="#5e2605">brown</font></td>
      <td> <b>Polyadenylation signal</b> </td>
    </tr>
    <tr>
      <td>PolyA_site</td>
      <td style="color: rgb(255, 102, 0);">orange</td>
      <td> <b>Polyadenylation site</b> </td>
    </tr>
    <tr>
      <td>pseudo_polyA</td>
      <td style="color: rgb(255, 153, 255);">pink</td>
      <td> <b>"Pseudo"-polyadenylation signal</b> detected in the
sequence
of a <b>processed pseudogene</b>.<br>
      <span style="font-style: italic; text-decoration: underline;">Warning</span>:
      <span style="font-style: italic;">pseudo_polyA </span>features
and <span style="font-style: italic;">processed_pseudogenes</span>
generally don't overlap. The reason is that pseudogene annotations are
based solely on protein evidence, whereas pseudo_polyA signals are
identified from transcript evidence - as they are found at the end of
the 3' UTR they can lie several kb downstream of the 3' end of the
pseudogene. </td>
    </tr>
  </tbody>
</table>
</p>
<p><br>
The current full set of GENCODE annotations is available for download <a
 target="_blank"
 href="ftp://genome.imim.es/pub/projects/gencode/data/havana-encode/current/">here</a>.<br>
</p>
<h2>Methods</h2>
(See Harrow <em>et al</em>., 2006 and Denoeud <em>et al</em>., 2007 for full 
details and references)
<h3 style="margin-left: 40px;">5'RACE/array experiments<br>
</h3>
<div style="margin-left: 40px;">A combination of 5&#8217;RACE (<span
 style="text-decoration: underline;">R</span>apid <span
 style="text-decoration: underline;">A</span>mplification of 5&#8217; <span
 style="text-decoration: underline;">c</span>DNA <span
 style="text-decoration: underline;">E</span>nds) and high-density
tiling microarrays were used to empirically annotate 5&#8217; transcription
start sites (TSSs) and internal exons of all 410 annotated
protein-coding loci across the 44 ENCODE regions (Oct 2005 GENCODE
freeze). The 5&#8217;RACE reactions were performed with oligonucleotides
mapping to a coding exon common to most of the transcripts of a protein
coding gene locus annotated by GENCODE (Oct 2005 freeze) on polyA+ RNA
from twelve adult human tissues (brain, heart, kidney, spleen, liver,
colon, small intestine, muscle, lung, stomach, testis, placenta) and
three cell lines (GM06990, HL60 and HeLaS3). <br>
The RACE reactions were then hybridized to 20 nucleotide-resolution
Affymetrix tiling arrays covering the non-repeated regions of the 44
ENCODE regions. The resulting "<span style="font-style: italic;">RACEfrags</span>"
-- array-detected fragments of RACE products -- were assessed for
novelty by comparing their genome coordinates to those of
GENCODE-annotated exons. Connectivity between novel RACEfrags and their
respective index exon were further investigated by RT-PCR, cloning and
sequencing. The resulting cDNA sequences (deposited in GenBank under
accession numbers DQ655905-DQ656069 and
EF070113-EF070122) were then fed into the HAVANA annotation pipeline as
mRNA evidence (see "Havana manual annotations" below).<br>
</div>
<h3 style="margin-left: 40px;">HAVANA manual annotations<br>
</h3>
<p style="margin-left: 40px;">The Human and Vertebrate Analysis and
Annotation manual curation
process (<a href="http://www.sanger.ac.uk/HGP/havana/" target="_blank">HAVANA</a>)
was
used to produce these annotations.
</p>
<div style="margin-left: 40px;">Before the process of manual annotation
begins, an automated
analysis pipeline
for similarity searches and <em>ab initio</em> predictions is run. <br>
The searches are run on a computer farm and stored in an Ensembl MySQL
database using a modified Ensembl analysis pipeline system. All
searches and prediction algorithms, except CpG island prediction (see
cpgreport in the EMBOSS application suite) are run on repeat masked
sequence. <br>
RepeatMasker is used to mask interspersed repeats, followed by Tandem
repeats finder to mask tandem repeats.
Nucleotide sequence databases are searched with wuBLASTN, and
significant hits are re-aligned to the unmasked genomic sequence using
est2genome.
The Uniprot protein database is searched with wuBLASTX, and the
accession numbers of significant hits are looked up in the Pfam
database. The hidden Markov models for Pfam protein domains are aligned
against the genomic sequence using Genewise to provide annotation of
protein domains. <br>
We also run a number of <em>ab initio</em> prediction algorithms:
Genescan and Fgenesh for genes, tRNAscan to find tRNAgenes and Eponine
TSS, to predict transcription start sites.<br>
Once the automated analysis is complete, the annotator uses a Perl/Tk
based graphical interface, called 'otterlace', developed in-house to
edit annotation data held in a separate MySQL database system. The
interface displays a rich,
interactive graphical view of the genomic region, showing features like
database matches, gene predictions, and transcripts created by the
annotators. Gapped alignments of nucleotide and protein blast hits to
the genomic sequence are viewed and explored using the 'Blixem'
alignment viewer. <br>
Additionally, the 'Dotter' dot plot tool is used for showing the
pair-wise alignments of unmasked sequence, thus revealing the location
of exons that are occasionally missed by the automated blast searches
because of their small size and/or match to repeat-masked sequence. <br>
The interface provides a number of tools that the annotator uses to
build genes and edit annotations: adding transcripts, exon coordinates,
translation regions, gene names and descriptions, remarks and
polyadenlyation signals and sites.<br>
</div>
<h2>Verification</h2>
<p>
See Harrow et al., 2006.
</p>
<h2>Credits</h2>
<p>This GENCODE release is the result of a collaborative effort between
the following
laboratories:<span style="font-weight: bold;"><br>
</span></p>
<p><span style="font-weight: bold;"></span></p>
<table style="text-align: left; width: 100%;" border="1" cellpadding="2"
 cellspacing="2">
  <tbody>
    <tr>
      <td style="vertical-align: top; text-align: center;"><span
 style="font-weight: bold;">Lab/Institution</span><br>
      </td>
      <td style="vertical-align: top; text-align: left;"><span
 style="font-weight: bold;">Contributors</span><br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: center;"><span
 style="font-weight: bold;">HAVANA annotation group</span>,
Wellcome Trust Sanger Insitute, Hinxton, UK</td>
      <td style="vertical-align: top;">Adam Frankish, Jonathan
Mudge, James Gilbert, Tim Hubbard, Jennifer Harrow<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: center;"><span
 style="font-weight: bold;">Genome Bioinformatics Lab</span>,
CRG, Barcelona, Spain</td>
      <td style="vertical-align: top;">France Denoeud, Julien Lagarde,
Sylvain Foissac, Robert Castelo, Roderic
Guig&oacute; (GENCODE Principal Investigator)</td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: center;"><span
 style="font-weight: bold;">Department of Genetic Medicine and
Development</span>, University of Geneva, Switzerland</td>
      <td style="vertical-align: top;">Catherine Ucla, Carine Wyss,
Caroline Manzano, Colette Rossier,
Stylianos E. Antonorakis</td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: center;"><span
 style="font-weight: bold;">Center for Integrative Genomics</span>,
University of Lausanne, Switzerland</td>
      <td style="vertical-align: top;">Jacqueline Chrast, Charlotte N.
Henrichsen, Alexandre Reymond</td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: center;"><span
 style="font-weight: bold;">Affymetrix, Inc.</span>, Santa Clara, CA,
USA</td>
      <td style="vertical-align: top;">Philipp Kapranov, Thomas R.
Gingeras</td>
    </tr>
  </tbody>
</table>
<p><span style="font-weight: bold;"></span><span
 style="font-weight: bold;"><br>
</span></p>
<ul>
</ul>
<h2>References</h2>
<p>
Denoeud F, Kapranov P, Ucla C, Frankish A, Castelo R, Drenkow J,
Lagarde J, Alioto TS, Manzano C, Chrast J <em>et al</em>.
Prominent use of distal 5' transcription start sites and discovery of a 
large number of additional exons in ENCODE regions. <em>Genome Res</em>. 
2007 (in press).<br>
<br>
Harrow J, Denoeud F, Frankish A, Reymond A, Chen CK, Chrast J,
Lagarde J, Gilbert JG, Storey R, Swarbreck D <em>et al</em>. GENCODE: 
producing a reference annotation for ENCODE. <em>Genome Biol</em>. 2006;7 
Suppl 1:S4.1-9.</br>
<br>
The ENCODE Project Consortium. Identification and analysis of
functional elements in 1% of the human genome by the ENCODE pilot
project. <em>Nature</em>. 2007 (accepted for publication).<br>
<br>
</p>
