<H2>Description</H2>
<P>
This track shows pseudogenes in the ENCODE regions derived from a
consensus of five different methods (Yale Pipeline, GenCode manual annotation, 
two different UCSC methods, and Genome Institute of Singapore (GIS)). 201 
consensus pseudogenes were found in 
the 6 Jan. 2006 freeze. In this analysis,
pseudogenes are defined as genomic sequences that are similar to known 
proteins with various inactivating disablements (<em>e.g.</em> premature stop 
codons or frameshifts) in their putative protein-coding regions and are
are flagged as either recently-processed or non-processed.</P>

<H2>Methods</H2>
<P>
Sets of pseudogenes were obtained from four research groups&mdash;GIS, 
Havana/Gencode, UCSC, and Yale&mdash;and were processed as follows:
<UL>
<LI><B>Step I:</B> The four data sets were filtered to remove pseudogenes 
that overlap with current Gencode coding exons/loci. Pseudogenes overlapping 
with introns or noncoding genes were kept. Subsequent filtering of pseudogene
sets, excluding the Havana set, removed pseudogenes overlapping with exons of 
UCSC Known Genes.

<LI><B>Step II:</B> A union of the pseudogenes from each filtered set was 
created. If a pseudogenic region was annotated by more than one group, 
the lowest starting coordinate and highest ending coordinate were used as the
boundaries.

<LI><B>Step III:</B> A parent protein for each pseudogene in the union was
assigned using a protein set from UniProt. Pseudogenes without a matching 
protein were excluded.

<LI><B>Step IV:</B> Each pseudogene was realigned to its parent protein.

<LI><B>Step V:</B> The consensus list of pseudogenes was updated with 
boundaries derived from the alignment in Step IV.

<LI><B>Step VI:</B> The consensus list of pseudogenes was updated with the
assigned parent proteins and new classifications (processed or non-processed).
</UL></P>
     
<H2>Verification</H2>
<P>
All pseudogenes in the list have been extensively curated by Adam Frankish and
Jennifer Harrow at the The Wellcome Trust Sanger Institute.</P>

<H2>Credits</H2>
<P>
These data were generated by the ENCODE pseudogene annotation group.</P>

<H2>References</H2>
<P>
More information is available from <A
HREF= http://www.pseudogene.org/ENCODE TARGET=_blank>pseudogene.org/ENCODE</A>.
</P>
