<H2>Disclaimer</H2>
<P>
PhenCode is intended for research purposes only. Although the data are freely
available to all, users should treat the reported mutations with extreme
caution in clinical settings or for any diagnostic or population screening
purpose. This information requires expertise to interpret properly; clinical
diagnosis and/or treatment recommendations should be made only by medical 
professionals. 
<A HREF="http://globin.bx.psu.edu/mutation/disclaimer.html" target="_blank">
<H4>Terms and Conditions of Use</H4></A>
</P>

<H2>Description</H2>
<P>
This track is a result of the <A HREF="http://globin.bx.psu.edu/mutation/"
target="_blank">PhenCode project</A>.  It consolidates variants
from several curated locus-specific databases and one genome-wide database.
Rich genotype and phenotype information is provided. This version includes
entries from the databases listed in the Methods section below.  Work is in
progress to add more locus-specific databases.
</P>

<H2>Display Conventions and Configuration</H2>
<P>
This track is color-coded by the mutation type or phenotype association.<BR><BR>
The default colors for mutation type are:
<UL>
<LI>
substitution = <font color="#CC00FF">purple</font>
<LI>
insertion = <font color="#009900">green</font>
<LI>
duplication = <font color="#FF9900">orange</font>
<LI>
deletion = <font color="#0000FF">blue</font>
<LI>
complex = <font color="#643200">brown</font>
<LI>
unknown = black
</UL>
The default colors for phenotype association are:
<UL>
<LI>
phenotype-associated = <font color="#DD0000">red</font>
<LI>
not phenotype-associated = <font color="#009900">green</font>
<LI>
phenotype association unknown = <font color="GRAY">gray</font>
</UL>
The color choices are configurable using the drop-down menus on the Track
Settings page.
</P>
<P>
By default, items are labeled with the Human Genome Variation Society (HGVS) 
name.  Additionally, the Browser can be configured to display the common name
(if available) or both labels using the &quot;Label&quot; checkboxes at the 
top of the Track Settings page.</P>
<P>
All mutation types, locations, and data sources, except for the UniProt
(Swiss-Prot/TrEMBL) data source, are included in the default
display. To filter the display based on one or more of these characteristics,
check the appropriate boxes in the exclusion lists on the 
Track Settings page.
</P>

<A NAME="source"></A>
<H2>Methods</H2>
<P>
The data shown in this track were obtained from the following variant sources:
<UL>
<LI>
<A HREF="http://www.androgendb.mcgill.ca/" TARGET=_blank>
Androgen Receptor Gene Mutations Database</A> (ARdb) - mutations in the AR gene.
<!--
<LI>
<A HREF="http://research.nhgri.nih.gov/projects/bic/" TARGET=_blank>BIC</A>
Breast Cancer Information Core.
-->
<LI>
<A HREF="http://www.ncbi.nlm.nih.gov/projects/mhc/xslcgi.fcgi?cmd=bgmut/home"
TARGET=_blank>Blood Group Antigen Gene Mutation Database</A> (BGMUT) -
mutations in many genes.
<LI>
<A HREF="http://www.genet.sickkids.on.ca/cftr/" TARGET=_blank>Cystic Fibrosis
Mutation Database</A> (CFMDB) - mutations in the CFTR gene.
<LI>
<A HREF="http://globin.bx.psu.edu/hbvar/menu.html" TARGET=_blank>HbVar 
database</A> - variants and thalassemia mutations in the globin loci.
<LI>
<A HREF="http://bioinf.uta.fi/base_root/" TARGET=_blank>Databases for immunodeficiency-causing mutations</A> (IDbases) - mutations in many genes.
<LI>
<A HREF="http://www.dmd.nl/" TARGET=_blank>Leiden Muscular Dystrophy pages</A>
(LMDp) - mutations in many genes
<LI>
<A HREF="http://www.pahdb.mcgill.ca/" TARGET=_blank>PAHdb</A> - mutations 
in the phenylalanine hydroxylase (PAH) gene.
<LI>
SRD5A2 - mutations in the steroid-5-alpha-reductase, alpha polypeptide 2
(3-oxo-5 alpha-steroid delta 4-dehydrogenase alpha 2) gene.
<LI>
<A HREF="http://www.expasy.org/sprot/" TARGET=_blank>UniProt 
(Swiss-Prot/TrEMBL)</A> - 
genome-wide variants within coding regions of UCSC Known Genes.
<!--
<LI>
A possible connection with <A HREF="http://www.centralmutations.org" 
TARGET=_blank>WayStation</A>, the HGVS community-wide database.
-->
</UL>
<p>The <A HREF="http://www.hgvs.org/mutnomen/" TARGET=_blank>HGVS-style name</A>
was pulled directly or indirectly from the source
data.  The information in this name, in combination with alignments of the
reference sequence against
the genome sequence, was used to position the variants.  The source may have
additional variants that could not be mapped to the genome.  Additional 
attributes displayed for each variant depend on the information available
from the different source databases.  If the source has web-accessible entries
for the variants, links are provided back to the source.
</p>
<p><A name="phenoassoc"></A>
Variants with phenotype associations that have been determined by a database
curator are listed as &quot;phenotype-associated&quot; or &quot;not 
phenotype-associated&quot;. Variants with phenotype associations that have not
been determined by a curator are listed as unknown.
</p>

<H2>Credits</H2>

<H4>PhenCode developers</H4>
<DL>
<DT>Belinda Giardine, Ross Hardison, Webb Miller, Cathy Riemer
<DD>Center for Comparative Genomics and Bioinformatics, Penn State University,
University Park, Pennsylvania
<DT>Fan Hsu, Jim Kent, Andrew Kern, Robert Kuhn, Heather Trumbower
<DD>Center for Biomolecular Science and Engineering, University of California,
Santa Cruz, California
<DT>Richard Gibbons, Doug Higgs, Jim Hughes
<DD>Weatherall Institute of Molecular Medicine, Oxford, United Kingdom
<DT>Garry Cutting, Andrew P. Feinberg
<DD>Johns Hopkins University School of Medicine, Baltimore, Maryland
</DL> 
<H4>Cooperating databases</H4>
<DL>
<DT>ARdb:
<DD>Bruce Gottlieb,
Lady Davis Institute for Medical Research, Sir Mortimer B. Davis Jewish
General Hospital, Montreal, Quebec, Canada
<!--
<DT>BIC:
<DD>Breast Cancer Information Core, An International Collaborative Effort hosted by NHGRI
-->
<BR><BR>
<DT>BGMUT:
<DD>Olga O. Blumenfeld and Santosh K. Patnaik,
Department of Biochemistry, Albert Einstein College of Medicine, Bronx, New
York
<BR><BR>
<DT>
CFMDB:
<A HREF ="http://www.genet.sickkids.on.ca/cftr/" BORDER = 0 TARGET=_blank><IMG 
 SRC="/images/logos/cftrlogo.png" ALIGN = TOP ALT = ""></A>
<DD>Julian Zielenski and Richard Sang,
The Hospital for Sick Children, Genetics and Genomic Biology, Toronto,
Ontario, Canada

<BR><BR>
<DT> HbVar:
<A HREF ="http://globin.bx.psu.edu/hbvar/menu.html" BORDER = 0 
  TARGET=_blank><IMG 
  SRC="/images/logos/HbVarLogo2.png" ALIGN = TOP ALT = ""></A>
<DD>George Patrinos, Erasmus University, Rotterdam, Netherlands<br>
 Henri Wajcman, Hospital Henri Mondor, Creteil, France<br>
 David H.K. Chui, Boston University, Boston, Massachusetts<br>
 Nicholas Anagnou, University of Athens and IIBEAA, Athens, Greece<br>
 Georgi D. Efremov, Macedonian Academy of Sciences and Arts, RCGEB, Skopje,
  Macedonia

<BR><BR>
<DT>IDbases:
<A HREF ="http://bioinf.uta.fi/base_root/" BORDER = 0 TARGET=_blank><IMG 
  SRC="/images/logos/ID-bases-logo.jpg" ALIGN = TOP ALT = ""></A>
<DD>Anne Durandy,
  Michael Hershfield,
  Laszlo Marodi,
  Luigi D. Notarangelo,
  Claudio Pignata,
  Jose R. Regueiro,
  Dirk Roos,
  C.I.Edvard Smith,
  Jouni Valiaho,
  Mauno Vihinen,
  Anna Villa,
  Institute of Medical Technology, University of Tampere, Tampere, Finland

<BR><BR>
<DT>LMDp
<DD>Johan T. den Dunnen,
  Leiden University Medical Center, Leiden, Nederland<br>
  See also the database tool <A HREF="http://www.LOVD.nl">LOVD</A>

<BR><BR>
<DT>PAHdb:
<A HREF ="http://www.pahdb.mcgill.ca/" BORDER = 0 TARGET=_blank><IMG 
 SRC="/images/logos/PAHLogoSmall.png" ALIGN = TOP IMG = ""></A>
<DD>Charles R. Scriver, McGill University, Montreal Children's Hospital
  Research Institute, Montreal, Quebec, Canada
<BR><BR>
<DT>Gene srd5a2:
<DD>Juergen Reichardt, University of Sydney, Sydney, Australia<br>
  Nick Makridakis, University of Southern California, Los Angeles, California

<BR><BR>
<DT>UniProt (Swiss-Prot/TrEMBL):
<DD>Swiss Institute of Bioinformatics, Geneva, Switzerland

<BR><BR>
</DL>
