<H2>Description</H2> 
<P> 
This track shows three different measures (D', r<sup>2</sup>, and LOD)
of linkage disequilibrium (LD) between pairs of SNPs as genotyped by
the HapMap consortium.  Linkage disequilibrium is a measure of the
association between two markers.  This is useful to understand which
regions of the genome are commonly inherited in blocks.  See the
Recombination Rates and Recombination Hotspots tracks for related
analyses. 
</P>
<P>
By default, LOD values are displayed in full mode.  Each diagonal
represents a different SNP with each square representing a pairwise
comparison between two SNPs.  Shades are used to indicate linkage
disequilibrium between the pair of SNPs, with darker shades indicating
stronger LD.  White squares indicate pairwise D' values less than 1
with no statistically significant evidence of LD (LOD < 2).  Blue squares
indicate pairwise D' values of 1 but without statistical significance.
</P>

<H2>Methods</H2>
<P>
Genotypes from HapMap Phase II release 19 were used with HaploView to
infer phasing and calculate LD values for all SNP pairs within 250kb.
</P>

<H2>Display Conventions and Configuration</H2>
<P>
<UL>
 <LI>
  Display Mode<BR>
  <UL>
   <LI>
    <i>Full</i> mode shows the pairwise LD values in a HaploView-style
    mountain plot.<BR>
   </LI>
   <LI>
    <i>Dense</i> mode shows the pairwise LD values in a single line for
    each population, where the intensity at each position is the average
    of all of the LD values for all SNPs within 250 kb.
   </LI>
  </UL>
 </LI>
 <LI>
  LD Values: measures of linkage disequilibrium<BR>
  <UL>
   <LI>
    <i>r<sup>2</sup></i><BR>
This displays the raw r<sup>2</sup> value for a given marker pair.  SNPs that
have not been separated by recombination have r<sup>2</sup> = 1; in this case, these
two markers are said to be redundant.  Lower r<sup>2</sup> values show a lower
degree of LD, indicating that some recombination has occurred in this
population.  Please see Pritchard and Przeowrski (2001) for details.
   </LI>
   <LI>
    <i>D'</i><BR>
This displays the raw D' value for a given marker pair.  A D' value of
1 (complete LD) indicates that two SNPs have not been separated by
recombination, while lower values indicate that the ancestral allele
has undergone some recombination during the history of the population.
Only D' values near 1 are a reliable measure of LD; lower values are
difficult to interpret as the magnitude of D' depends strongly on
sample size.  Please see Lewontin (1988) for more details.
   </LI>
   <LI>
    <i>LOD</i><BR>
This displays the log odds score for linkage disequilibrium between a given marker pair, and is shown by default.
   </LI>
  </UL>
 </LI>
 <LI>
  Track Geometry<BR>
  <UL>
   <LI>
    <i>Trim to triangle</i> shows the standard mountain plot (default); turning this option off will show LD values with SNPs outside the window.
   </LI>
   <LI>
    <i>Invert</i>ing makes it easier to visually compare two adjacent populations.
   </LI>
  </UL>
 </LI>
 <LI>
  Colors<BR>
  <UL>
   <LI>
    <i>LD Values</i> can be drawn in a variety of colors, with red as default.  The intensity of the color is proportional to the strength of the LD measure chosen above.
   </LI>
   <LI>
    <i>Outline</i>s can be drawn in contrasting colors or turned off.
   </LI>
  </UL>
 </LI>
 <LI>
  Population Selection<BR>
  The four HapMap populations can be displayed or hidden individually.
 </LI>
</UL>
</P>

<H2>Credits</H2>
<P>
This track was created by 
<A HREF="mailto:daryl@soe.ucsc.edu">Daryl Thomas</A> 
at UCSC using 
<A HREF="http://www.hapmap.org/downloads/ld_data/2005-10/" 
TARGET=_blank>data</A> from the 
<A HREF="http://www.hapmap.org/">International HapMap Project</A>, 
following the display style from 
<A HREF="http://www.broad.mit.edu/mpg/haploview/" TARGET=_blank>HaploView</A>.
</P>

<H2>References</H2>

<H3>HapMap Project</H3>
<P>
The International HapMap Consortium.
<A HREF="http://www.hapmap.org/downloads/presentations/Nature_HapMap_phaseI.pdf"TARGET=_blank>A haplotype map of the human genome</A>.
<em>Nature</em> <B>437</B>, 1299-1320 (2005).</P> 
<P>
The International HapMap Consortium.
<A HREF="http://www.hapmap.org/downloads/nature02168.pdf" TARGET=_blank>The 
International HapMap Project</A>.
<em>Nature</em> <B>426</B>, 789-96 (2003).</P>

<H3>HapMap Data Coordination Center</H3>
<P>
Thorisson, G.A., Smith, A.V., Krishnan, L. and Stein, L.D. 
<A HREF="http://www.hapmap.org/downloads/presentations/Genome_research_art10.pdf" 
TARGET=_blank>The International HapMap Project Web site</A>.
<em>Genome Res</em> <B>15</B>, 1591-3 (2005).</P> 

<H3>HaploView</H3>
<P>
Barrett, J.C., Fry, B., Maller, J. and Daly, M.J. 
<A HREF="http://bioinformatics.oxfordjournals.org/cgi/content/full/bth457?ijkey=54SiAdNbKzbNg&keytype=ref" 
TARGET=_blank>Haploview: analysis and visualization of LD and haplotype 
maps</A>. 
<em>Bioinformatics</em> <B>21</B>(2), 263-5 (2005).</P>

<H3>General references on Linkage disequilibrium</H3>
<P>
Lewontin, R.C.
<A HREF="http://www.genetics.org/cgi/reprint/120/3/849"
TARGET=_blank>On measures of gametic disequilibrium</A>.
<em>Genetics</em> <B>120</B>, 849-52 (1988).</P>
<P>
Pritchard, J.K. and Przeworski, M.
<A HREF="http://www.journals.uchicago.edu/AJHG/journal/issues/v69n1/012882/012882.html"
TARGET=_blank>Linkage disequilibrium in humans: models and data</A>.
<em>Am. J. Hum. Genet.</em>, <B>69</B>, 1-14 (2001).</P>
