<H2>Description</H2> 
<P> 

The HapMap project identified and genotyped a set of approximately 4 million
common SNPs from dbSNP build 125 to be used as a reference for future studies
of human disease. This track displays the genotype counts and allele
frequencies of those SNPs. The data displayed is release 21a (HapMap Phase II).
<P>
There are 4 populations: CEU, CHB, JPT and YRI. Each of the 4 populations is
displayed in a separate subtrack.
<P>
The CEU and YRI data are comprised of 90 individuals in trios. The UCSC display
removes the data for the children, leaving 60 individuals in each population.
The CHB and JPT data are comprised of 45 individuals. Over 12% of HapMap SNPs
are available for only a subset (1-3) of the populations. When data is
available, the CHB and JPT SNPs were assayed in a minimum of 18 individuals,
with over 97% of SNPs assayed in 45 or more individuals. The minimums for CEU
and YRI are 26 and 24 respectively with over 94% of SNPs assayed in 55 or more
individuals.
</P>
The HapMap assays provided biallelic results. Over 99.9% of HapMap SNPs are
known to dbSNP build125 as biallelic; approximately 3000 are more complex.
Two-thirds of the HapMap SNPs are transitions; one-third are A/G, one-third
are C/T.
<P>
Orthologous alleles in chimp (panTro2) and macaque (rheMac2) are also provided.
These were derived using liftOver. Chimp alleles are available for over 96% of
the human HapMap SNPs. Macaque alleles are available for 88% of the human SNPs.
<P>
15% of HapMap SNPs were monomorphic in all individuals in all populations.
Within single populations, monomorphism ranged from 21.5% of the YRI
individuals to 38% of the JPT individuals.
<P>
Approximately 20% of HapMap SNPs have a different major allele in different
populations.
<P>
No two HapMap SNPs occupy the same position. Aside from 7 SNPs from the pseudo
autosomal region of chrX, no rsIds are included more than once. No HapMap SNPs
occur on chrM or on "random" chroms.


<H2>Display Conventions and Configuration</H2>
<P>
The human SNPs are displayed using a color gradient based on minor allele
frequency. The higher the minor allele frequency, the darker the display. The
maximum minor allele frequency is of course 50%. When zoomed to base level,
the major allele is displayed for the human SNPs.
<P>
Reversing the base position track will cause the HapMap display to reverse as
well. This is the recommended configuration for SNPs on the negative strand.
<P>
The ortho alleles are displayed using a color gradient based on quality score.
Quality scores range from 0 to 100 representing low to high quality. For ortho
alleles, the higher the quality, the darker the display.   Quality scores are
not available for chimp chroms chr21 and chrY.
<P>
Filters are provided for the data attributes described above. Additionally,
a filter is provided for heterozgosity over all populations. The measure of
heterozygosity used is 2pq. Filters are applied to all 6 subtracks.

<H2>Credits</H2>
<P>
This track is based on <A HREF="http://www.hapmap.org/" TARGET=_BLANK>
International HapMap Project</A> release 21a data, provided by the <A
HREF="http://www.hapmap.org/downloads/" TARGET=_BLANK>HapMap Data Coordination
Center</A>.

<H2>References</H2>

<H3>HapMap Project</H3>
<P>
The International HapMap Consortium.
<A HREF="http://www.nature.com/nature/journal/v437/n7063/abs/nature04226.html"
TARGET=_blank>A haplotype map of the human genome</A>.
<em>Nature</em>. 2005 Oct 27;437(7063):1299-320.</P> 
<P>
The International HapMap Consortium.
<A HREF="http://www.nature.com/nature/journal/v426/n6968/abs/nature02168.html"
TARGET=_blank>The International HapMap Project</A>.
<em>Nature</em>. 2003 Dec 18;426(6968):789-96.</P>

<H3>HapMap Data Coordination Center</H3>
<P>
Thorisson GA, Smith AV, Krishnan L, Stein LD.
<A HREF="http://www.genome.org/cgi/content/abstract/15/11/1592"
TARGET=_blank>The International HapMap Project Web site</A>.
<em>Genome Res</em>. 2005 Nov;15(11):1592-3.</P>

<H3>Data Source</H3>
<P>
sftp://www.hapmap.org/downloads/genotypes/2007-01/rs_strand/non-redundant/genotypes_chr*_*_r21a_nr.txt.gz
</P>
