<H2>Description</H2> 
<P> 

The HapMap project identified and genotyped a set of SNPs to be used
as a reference for future studies of human disease.  These SNPs re
shown here, along with their Chimp (panTro1) and Rhesus (rheMac1)
alleles.  The human allele that does not match the Chimp allele is
taken to be the derived allele, and the derived allele frequency is
reported in all HapMap populations.</P>

<H2>Methods</H2>
<P>
</P>

Strand issues can be rather when discussing comparisons of species, so
this is a real situation from hg16 to explan the details.  In these
table, the strand has already been adjusted to match the human and can
be compared to the chimp/rhesus nucleotide directly.  Human alleles
are always capitalized; Chimp and Rhesus bases are shown in lower case
when they are in a RepeatMasked region in their assembly.

<PRE>
  # This example shows the two human bases ('T' and 'A') and the chimp allele ('a')
  name      humanChrom humanPos  hStrand hRef hOth chimpChrom chimpPos  cStrand cAllele
  rs4970776 chr1       109570184 +       T    A    chr1       112470519 -       a
</PRE>
From this point, the SNP is marked the SNP with a caret ('^'), and the
flanking sequences are included.  Also note the revComp state, which
is an indicator of the strand.  (The 'DNA' button at the top of the
Genome Browser browser will provide the DNA directly.)
<PRE>
  Assembly  chr   pos         strand   range (+/- 5 bp)
  hg16      chr1  109570184   +        chr1:109570179-109570189
  panTro1   chr1  112470519   -        chr1:112470514-112470524

  >hg16_dna range=chr1:109570179-109570189 revComp=FALSE
  TTCAGTGGCTT                                      ^^^^^  ==> '+' strand
       ^   ==> 'T' is the reference allele

  >panTro1_dna range=chr1:112470514-112470524 revComp=FALSE
  AAGCCTCTGAA
  (This needs to be reverse complemented to see the '-' strand)

  >panTro1_dna range=chr1:112470514-112470524 revComp=TRUE
  TTCAGAGGCTT                                         ^^^^  ==> '-' strand
       ^   ==> 'A' is the orthologous base in chimp
</PRE>
This means that the 'A' allele is shared between chimp and humans.
Without additional outgroup information, the term 'ancestral' is a bit
premature here.

<H2>Credits</H2>
<P>
This track is based on release 16b from the HapMap Data Coordination
Center.  The liftOver tool was used to identify the orthologous
nucleotides in the Chimp (panTro1) and Rhesus (rheMac1) assemblies.

<H2>References</H2>

<H3>HapMap Project</H3>
<P>
The International HapMap Consortium.
<A HREF="http://www.hapmap.org/downloads/presentations/Nature_HapMap_phaseI.pdf" 
TARGET=_blank> A haplotype map of the human genome</A>.
<em>Nature</em> <B>437</B>, 1299-1320 (2005).</P> 
<P>
The International HapMap Consortium.
<A HREF="http://www.hapmap.org/downloads/nature02168.pdf" 
TARGET=_blank> The International HapMap Project</A>.
<em>Nature</em> <B>426</B>, 789-96 (2003).</P>

<H3>HapMap Data Coordination Center</H3>
<P>
Thorisson, G.A., Smith, A.V., Krishnan, L. and Stein, L.D. 
<A HREF="http://www.hapmap.org/downloads/presentations/Genome_research_art10.pdf" 
TARGET=_blank> The International HapMap Project Web site</A>.
<em>Genome Res</em> <B>15</B>, 1591-3 (2005).</P> 

<H3>Data Source</H3>
<P>
sftp://www.hapmap.org/hapmap-data/ftp/private/common/data_freezes/2005-04_rel16b_patched/full/frequencies/non-redundant/allele_freqs_chr*_*.txt.gz
</P>
