<H2>Description</H2> 
<P> 

This track shows three different measures (D', r^2, and LOD) of
linkage disequilibrium (LD) between pairs of SNPs as genotyped by the
HapMap consortium.  Linkage disequilibrium is a measure of the
association between two markers.  This is useful to understand which
regions of the genome are commonly inherited in blocks without
recombination.  See the Recombination Rates and Recombination Hotspots
tracks for related analyses.

</P>

In full mode, each diagonal represents a different SNP, with each
square representing a pairwise comparison between two SNPs.  Shades of
red are used to indicate linkage disequilibrium between the pair of
SNPs, with darker shades indicating stronger LD.

<!--
<PRE>
sub encodeLod($$)
{
    my $lod    = shift @_;
    my $dPrime = shift @_;
    $ret = ord('a');
    if ($lod>=2) # high LOD
        {
        if (abs($dPrime)<0.5) { $ret = ord('y'); } # high LOD, low D'  -> pink
        else { $ret += min((int($lod-abs($dPrime)-1.5)), 9) ;}        
	}
    elsif (abs($dPrime)>0.99) { $ret = ord('z'); } # high D', low LOD  -> blue
    return chr($ret);
}
</PRE>           
-->

<H2>Methods</H2>
<P>
Genotypes from HapMap Phase II release 19 were used with HaploView to infer phasing and calculate LD values for all SNP pairs within 250kb.
</P>

<H2>Display Conventions and Configuration</H2>
<P>
<UL>
 <LI>
  Display Mode<BR>
  <UL>
   <LI>
    <i>Full</i> mode shows the pairwise LD values in a HaploView-style
    mountain plot.<BR>
   </LI>
   <LI>
    <i>Dense</i> mode shows the pairwise LD values in a single line for
    each population, where the intensity at each position is the average
    of all of the LD values for all SNPs within 250 kb.
   </LI>
  </UL>
 </LI>
 <LI>
  LD Values: measures of linkage disequilibrium<BR>
  <UL>
   <LI>
    <i>r^2</i><BR>
This displays the raw r^2 value for a given marker pair.  SNPs that
have not been separated by recombination, r^2 = 1; in this case, these
two markers are said to be redundant.  Lower r^2 values show a lower
degree of LD, indicating that some recombination has occurred in this
population.  <BR>
One useful property of r^2 is that its inverse, 1/r^2,
provides a practical estimate of the magnitude by which the sample
size must be increased in a study design to detect association between
the phenotype and a marker locus, as compared to the size required for
detecting association with the susceptibility locus itself.  Please
see Pritchard and Prezowrski (2001) for details.
   </LI>
   <LI>
    <i>D'</i><BR>
This displays the raw D' value for a given marker pair.  A D' value of
1 (complete LD) indicates that two SNPs have not been separated by
recombination, while lower values indicate that the ancestral allele
has undergone some recombination during the history of the population.
Only D' values near 1 are a reliable measure of LD; lower values are
difficult to interpret as the magnitude of D' depends stongly on
sample size.  Please see Lewontin (1988) for more details. <BR>
White squares indicate pairwise D' values less than 1 with no
statistically significant evidence of LD.  Blue squares indicate
pairwise D' values of 1 but without statistical significance (LOD <
2).
   </LI>
   <LI>
    <i>LOD</i><BR>
This displays the log odds score for linkage disequilibrium between a given marker pair.
   </LI>
  </UL>
 </LI>
 <LI>
  Track Geometry<BR>
  <UL>
   <LI>
    <i>Trim to triangle</i> shows the standard mountain plot (default); turning this option off will show LD values with SNPs outside the window.
   </LI>
   <LI>
    <i>Invert</i>ing makes it easier to visually compare two adjacent populations.
   </LI>
  </UL>
 </LI>
 <LI>
  Colors<BR>
  <UL>
   <LI>
    <i>LD Values</i> can be drawn in a variety of colors, with red as default.  The intensity of the color is proportional to the strength of the LD measure chosen above.
   </LI>
   <LI>
    <i>Outline</i>s can be drawn in contrasting colors or turned off.
   </LI>
  </UL>
 </LI>
 <LI>
  Population Selection<BR>
  The four HapMap populations can be displayed or hidden individually.
 </LI>
</UL>
</P>

<H2>Credits</H2>
<P>
This track was created by 
<a href="mailto:daryl@soe.ucsc.edu">Daryl Thomas</a> 
at UCSC using 
<a href="http://www.hapmap.org/downloads/ld_data/2005-10/" target=_blank>data</a> 
from the 
<a href="http://www.hapmap.org/">International HapMap Project</a>, 
following the display style from 
<a href="http://www.broad.mit.edu/mpg/haploview/" target=_blank>HaploView</a>.
</P>

<H2>References</H2>

<H3>HapMap Project</H3>
<UL>
 <LI>
The International HapMap Consortium. <br>
<a href="http://www.hapmap.org/downloads/presentations/Nature_HapMap_phaseI.pdf">
A haplotype map of the human genome</a>.<br>
<i>Nature</i> <B>437</B>:1299-1320. 2005. 
 <LI>
The International HapMap Consortium.<br>
<a href="http://www.hapmap.org/downloads/nature02168.pdf" target=_blank>
The International HapMap Project</a>.<br>
<i>Nature</i> <B>426</B>:789-796. 2003.
</UL>

<H3>HapMap Data Coordination Center</H3>
<UL>
 <LI>
Thorisson, G.A., Smith, A.V., Krishnan, L., and Stein, L.D. <br>
<a href="http://www.hapmap.org/downloads/presentations/Genome_research_art10.pdf" target=_blank>
The International HapMap Project Web site</a>.<br>
<i>Genome Research</i> <B>15</B>:1591-1593. 2005. 
</UL>

<H3>HaploView</H3>
<UL>
 <LI>
  Barrett JC, Fry B, Maller J, Daly MJ. <BR>
  <a href="http://bioinformatics.oxfordjournals.org/cgi/content/full/bth457?ijkey=54SiAdNbKzbNg&keytype=ref" target=_blank>
  Haploview: analysis and visualization of LD and haplotype maps</a>. <BR>
  <i>Bioinformatics</i> <B>21</B>(2):263-265. 2005 Jan 15 [PubMed ID: 15297300]
 </LI>
 <LI>
  Jonathan K. Pritchard and Molly Przeworski.<BR>
  <a href="http://www.journals.uchicago.edu/AJHG/journal/issues/v69n1/012882/012882.html">
  Linkage Disequilibrium in Humans: Models and Data </a>. <BR>
  <i>American Journal of Humuman Genetics</i>, <B>69</B>:1-14. 2001.
 </LI>
 <LI>
  Richard C. Lewontin<BR>
  <a href="http://www.genetics.org/cgi/reprint/120/3/849">On Measures of Gametic Disequilibrium</a>.
  <i>Genetics</i> <B>120</B>:849-852. 1988
 </LI>
</UL>

<H3>Tagger</H3>
<UL>
 <LI>
de Bakker, P.I.W., Yelensky, R., Pe'er, I., Gabriel, S.B., Daly, M.J., and Altshuler, D. <br>
<a href="http://www.nature.com/ng/journal/v37/n11/pdf/ng1669.pdf" target=_blank>
Efficiency and power in genetic association studies</a>.<br>
<i>Nature Genetics</i> <B>37</B>(11):1217-1223. 2005. 
</UL>
