<H2>Description</H2>
<P>
This track shows per-continent integrated haplotype score (iHS, 
Voight <em>et al.</em>)
calculated using SNPs genotyped in 53 populations worldwide by the 
<A HREF="http://www.stanford.edu/group/morrinst/hgdp.html" TARGET=_BLANK>Human 
Genome Diversity Project</A> in collaboration with the
<A HREF="http://www.cephb.fr/en/cephdb/" TARGET=_BLANK>Centre d'Etude 
du Polymorphisme Humain</A> (HGDP-CEPH).
This track and several others are available from the 
<A HREF="http://hgdp.uchicago.edu/" TARGET=_BLANK>HGDP Selection Browser</A>.
</P>
<P>
The score shown for each SNP is -log<sub>10</sub> of the 

<em>??? SNP's position in the distribution of windowed iHS score ???</em>.
</P>

<H2>Methods</H2>
<P>
Samples collected by the HGDP-CEPH were genotyped for 657,000 SNPs at
<A HREF="http://hagsc.org/hgdp/" TARGET=_BLANK>Stanford</A>,
using the Illumina BeadStation technology.  
The 53 populations were divided into seven continental groups: Africa,
Middle East, Europe, South Asia, East Asia, Oceania and the Americas.
iHS was then calculated as described in (Voight <em>et al.</em>).
<em>???
Per-SNP iHS scores were smoothed in windows of 31 SNPs, centered on 
each SNP.  The final score is -log<sub>10</sub> of the proportion of 
smoothed scores higher than each SNP's smoothed score. ???</em>
</P>

<H2>Credits</H2>
<P>
Thanks to the HGDP-CEPH and Joe Pickrell in the 
<A HREF="http://pritch.bsd.uchicago.edu/"TARGET=_BLANK>Pritchard
lab</A> at the University of Chicago for providing these data.
</P>


<H2>References</H2>
<P>Voight BF, Kudaravalli S, Wen X, Pritchard JK.
<A HREF="http://www.ncbi.nlm.nih.gov/pubmed/16494531?dopt=Abstract" 
TARGET=_BLANK>A map of recent positive selection in the human genome.</A>
<em>PLoS Biol</em>. 2006 Mar;4(3):e72.</P>

<P>
Pickrell JK, Coop G, Novembre J, Kudaravalli S, Li J, Absher D,
Srinivasan BS, Barsh GS, Myers RM, Feldman MW, Pritchard JK.
Signals of recent positive selection in a worldwide sample of human
populations. <em>Genome Research</em> (2009). In Press.</P>

<P>
Li JZ*, Absher DM*, Tang H, Southwick AM, Casto AM, Ramachandran S,
Cann HM, Barsh GS, Feldman M, Cavalli-Sforza LL, Myers RM.
<A HREF="http://www.ncbi.nlm.nih.gov/pubmed/18292342?dopt=Abstract"
TARGET=_BLANK>Worldwide human relationships inferred from genome-wide 
patterns of variation.</A>
<em>Science</em>. 2008 Feb 22;319(5866):1100-4.</P>

<P>
Cann HM, de Toma C, Cazes L, Legrand MF, Morel V, Piouffre L, Bodmer
J, Bodmer WF, Bonne-Tamir B, Cambon-Thomsen A, Chen Z, Chu J, Carcassi
C, Contu L, Du R, Excoffier L, Ferrara GB, Friedlaender JS, Groot H,
Gurwitz D, Jenkins T, Herrera RJ, Huang X, Kidd J, Kidd KK, Langaney
A, Lin AA, Mehdi SQ, Parham P, Piazza A, Pistillo MP, Qian Y, Shu Q,
Xu J, Zhu S, Weber JL, Greely HT, Feldman MW, Thomas G, Dausset J,
Cavalli-Sforza LL.
<A HREF="http://www.ncbi.nlm.nih.gov/pubmed/11954565?dopt=Abstract" 
TARGET=_BLANK>A human genome diversity cell line panel.</A>
<em>Science</em>. 2002 Apr 12;296(5566):261-2.</P>

