<H2>Description</H2>

<!--#insert file="omimDescriptionText.html"-->

<P>
This track shows the genomic positions of all gene enries
in the Online Mendelian Inheritance in Man
(<A HREF="http://www.omim.org/"
TARGET=_blank>OMIM</A>) database.
</P>

<H2>Display Conventions and Configuration</H2>

<P>Genomic locations of OMIM gene entries are displayed as solid
blocks.  The entries are colored according to the OMIM phenotype class of associated disorders:
<UL>
<LI><B><FONT COLOR="#AACBAA">Lighter Green</FONT></B> for Class 1 OMIM records 
-  the disorder is placed on the map based on its association with
a gene, but the underlying defect is not known.
<LI><B><FONT COLOR="#66A266">Light Green</FONT></B> for Class 2 OMIM records 
- the disorder has been placed on the map by linkage; no mutation has
been found.
<LI><B><FONT COLOR="#006400">Dark Green</FONT></B> for Class 3 OMIM records 
- the molecular basis for the disorder is known; a mutation has been
found in the gene.
<LI><B><FONT COLOR="#9900CC">Purple</FONT></B> for Class 4 OMIM records 
- a contiguous gene deletion or duplication syndrome, multiple genes
are deleted or duplicated causing the phenotype.
<LI><B><FONT COLOR="#C8C8C8">Light Gray</FONT></B> for Others
- no associated OMIM phenotype class info available.
</UL>
<P> Gene symbol and disease information, when available, are displayed on the details page for an
item.
</P>

<P>On the main browser display, the description of the OMIM entry are shown when Full display mode
is chosen.  In Pack mode, the entry description is shown when mousing over each entry.
<H2>Methods</H2>
<P>
The mappings displayed in this track are based on the mapping between OMIM gene entries and ENTREZ
gene IDs:
<UL>
<LI>The data file, <TT>genemap.txt</TT>, from OMIM is loaded into an MySQL table, <TT>omimGeneMap</TT>.
<LI>The data file, <TT>mim2gene.txt</TT>, from OMIM is loaded into an MySQL table, <TT>mim2gene</TT>.
<LI>Entries in <TT>genemap.txt</TT> having disorder info are parsed and loaded into the 
<TT>omimPhenotyp</TT> table.
<LI>From the <TT>mim2gene</TT> table, for each OMIM gene, its ENTREZ gene ID is used to get the
corresponding RefSeq gene ID via the <TT>refLink</TT> table.  The RefSeq gene genomic chrom, start,
and end data are obtained from the <TT>refGene</TT> table.  Together with the OMIM gene ID, a bed 4
table <TT>omimGene2</tt> is constructed as the primary table for this track.
<LI>Links to related RefSeq Genes and UCSC Genes are listed on the details page (using Entrez gene
ID as cross-reference and other info from mim2gene, refLink, and kgXref tables).
</UL>
<H2>Credits</H2>
<P>
Thanks to OMIM and NCBI for the use of their data. This track was
constructed by Fan Hsu, Robert Kuhn, and Brooke Rhead of the UCSC Genome Bioinformatics Group.</P> 

<H2>References</H2>
<P>
Hamosh A, Scott AF, Amberger JS, Bocchini CA, McKusick VA. 
<A HREF="http://nar.oxfordjournals.org/cgi/content/full/33/suppl_1/D514"
TARGET=_blank>Online Mendelian Inheritance in Man (OMIM), a knowledgebase of 
human genes and genetic disorders</A>. 
<em>Nucleic Acids Res</em>. 2005 Jan 1;33(Database issue):D514-7.
</P>

<P>Amberger J, Bocchini CA, Scott AF, Hamosh A.  
<A HREF="http://nar.oxfordjournals.org/cgi/content/full/37/suppl_1/D793"
TARGET=_blank>McKusick's Online Mendelian Inheritance in Man (OMIM&#174;)</A>.  
<EM>Nucleic Acids Res.</EM> 2009 Jan;37(Database issue):D793-6. Epub 2008 Oct 8.
</P>
