<H2>Description</H2>

<TABLE BORDER = 1 CELLPADDING = 4><TR><TD>
<P><FONT COLOR = RED><B>NOTE</FONT>: OMIM is intended for use
 primarily by physicians and other
professionals concerned with genetic disorders, by genetics researchers, and
by advanced students in science and medicine. While the OMIM database is
open to the public, users seeking information about a personal medical or
genetic condition are urged to consult with a qualified physician for
diagnosis and for answers to personal questions.
</P>
<P>Because the UCSC Genes mappings are based on associations from RefSeq and
UniProt, they are dependent on any interpretations from those sources.
Furthermore, because many OMIM records refer to multiple gene names,
or syndromes not tightly mapped to individual genes, the associations
in this track should be treated with skepticism and any conclusions
based on them should be carefully scrutinized using independent
resources. </B> 
</TD></TR>
</TABLE>
</P> 

<P>
This track shows the positions of canonical UCSC genes that have been
associated with identifiers in the Online Mendelian Inheritance in Man
(<A HREF="http://www.ncbi.nlm.nih.gov/sites/entrez?db=omim"
TARGET=_blank>OMIM</A>) database.
The associations were obtained from
<A HREF="http://www.ncbi.nlm.nih.gov/RefSeq/" TARGET=_blank>RefSeq</A>
and <A HREF="http://www.uniprot.org/" TARGET=_blank>UniProt</A>
annotations. To simplify the track, annotations on splice variants have been
collapsed so that only canonical UCSC splice variants are displayed.
</P>

<P>OMIM is a compendium of human genes and genetic phenotypes. The full-text, 
referenced overviews in OMIM contain information on all known Mendelian 
disorders and over 12,000 genes. OMIM is authored and edited at the 
McKusick-Nathans Institute of Genetic Medicine, Johns Hopkins University 
School of Medicine, under the direction of Dr. Ada Hamosh.  This database 
was initiated in the early 1960s by Dr. Victor A. McKusick as a catalog 
of Mendelian traits and disorders, entitled Mendelian Inheritance
in Man (MIM). 
</P> 
<H2>Display Conventions and Configuration</H2>

<P>Positions of OMIM-associated canonical UCSC genes are displayed as solid
blocks. The color of the block depends on the classification assigned
by OMIM.   

<UL>
  <LI>If the OMIM identifier is listed in the
    <A HREF="http://www.ncbi.nlm.nih.gov/Omim/omimfaq.html#morbid_map"
    TARGET=_blank>OMIM</A>
    <B><FONT COLOR = "C00000"> Morbid Map</FONT></B>, 
    the item is colored red.
  </LI>
  <LI>
    If there is 
    <B><FONT COLOR = "0000C0">no Morbid Map annotation </FONT></B>, 
    the item is colored blue. 
  </LI>
</UL>
</P>

<P> Where relevant, disease information from the Morbid Map 
&quot;disorder&quot; column is displayed on the details page for an item. 
If the canonical UCSC gene represents a cluster of more than one gene, the
other genes are listed on the track details page.
</P>

<P>By default, items are labeled with the
<A HREF="http://www.ncbi.nlm.nih.gov/Omim/omimfaq.html#numbering_system"
TARGET=_blank>OMIM identifier</A>. The label can be changed to the OMIM
gene/syndrome or the UCSC gene symbol on the track configuration page.
</P>

<H2>Methods</H2>
<P>
This annotation was constructed as follows: 
<UL>
<LI>A set of UCSC genes  was compiled that have cross-references to 
RefSeq genes with associated OMIM IDs, as defined in the Genome Browser 
database <TT>refLink</TT> table.
<LI>A second set of UCSC genes was compiled in which the corresponding
UniProt proteins have external links to OMIM, based on data in the 
<TT>spXref2</TT> table in the UCSC proteome database.
<LI>The two gene sets were combined and duplicates removed, resulting in 
a subset of
UCSC genes that have associated OMIM IDs.
<LI>Each of the genes in the combined set was replaced with its associated
canonical gene, based on the UCSC canonical gene set defined by
the <TT>knownIsoforms</TT> and <TT>knownCanonical</TT> tables.
<LI>Duplicates were removed from the canonical set, and the data were
sorted and loaded into the <TT>omimGene</TT> table.
</UL>
<P>
In addition, data from the OMIM <TT>genemap</TT> and <TT>morbidmap</TT> 
 <A HREF = 
"http://www.ncbi.nlm.nih.gov/Omim/omimfaq.html#download" TARGET = 
_BLANK>files</A> were loaded into the <TT>omimGeneMap</TT> and 
<TT>omimMorbidMap</TT> tables in the Genome Browser database. 
</P>

<H2>Credits</H2>
<P>
Thanks to OMIM, NCBI, and UniProt for the use of their data. This track was
constructed by Fan Hsu in the UCSC Genome Bioinformatics Group.</P> 

<H2>References</H2>
<P>
Hamosh A, Scott AF, Amberger JS, Bocchini CA, McKusick VA. 
<A HREF="http://nar.oxfordjournals.org/cgi/content/full/33/suppl_1/D514"
TARGET=_blank>Online Mendelian Inheritance in Man (OMIM), a knowledgebase of 
human genes and genetic disorders</A>. 
<em>Nucleic Acids Res</em>. 2005 Jan 1;33(Database issue):D514-7.
</P>

<P>Amberger J, Bocchini CA, Scott AF, Hamosh A.  
<A HREF="http://nar.oxfordjournals.org/cgi/content/full/37/suppl_1/D793"
TARGET=_blank>McKusick's Online Mendelian Inheritance in Man (OMIM&#174;)</A>.  
<EM>Nucleic Acids Res.</EM> 2009 Jan;37(Database issue):D793-6. Epub 2008 Oct 8.
</P>
