<H2>Description</H2>
The retroGene track shows processed mRNAs that have been inserted back into the genome since the mouse/human split. RetroGenes can be either functional genes that have acquired a promoter from a neighboring gene, non-functional pseudogenes, or transcribed pseudogenes.
<P>
</P>
<H2>Methods</H2>

<P>
All mRNAs of a species from GenBank are aligned to the genome using <A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>Blastz</A>. mRNAs that aligned twice in the genome (once with introns and once without introns) are initially screened. Next, a series of features are scored to determine candidates for retrotranspostion events. These features include, position and length of the polyA tail, degree of synteny with mouse, coverage of repetative elements, number of exons that can still be aligned to the retroGene and degree of divergence from the parent gene.  These features are combined and scored against a training set of known pseudogenes using AdaBoost. RetroGenes in the final set have an AdaBoost confidence above 50% and less that 50% overlap with simple repeats, greater than 65% identity and an blastz alignment axtScore greater 10000. <p>
The type field has four possible values, singleExon indicates that the parent gene is a single exon gene. Mrna indicates that the parent gene is a spliced mrna that has no annotation in NCBI refSeq, UCSC knownGene or Mammalian Gene Collection (MGC). Better indicates that there is a spliced mrna that overlaps the retrogene that indicates transcription. Good indicates that this is a highly confident prediction. <p> 
These features can be downloaded from the table pseudoGeneLink in many formats using the Table Browser option on the blue bar. 
</P>
<H2>Credits</H2>
<P>
The pseudoMasker program and browser track was developed by <A HREF="email:baertsch@soe.ucsc.edu" TARGET=_blank>Robert Baertsch</A> for providing this track.

