<H2>Description</H2>
The retroGene track shows processed mRNAs that have been inserted back 
into the genome since the mouse/human split. RetroGenes can be either 
functional genes that have acquired a promoter from a neighboring gene, 
non-functional pseudogenes, or transcribed pseudogenes.
<P>
</P>
<H2>Methods</H2>

<P>
All mRNAs of a species from GenBank were aligned to the genome using 
<A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>Blastz</A>. 
mRNAs that aligned twice in the genome (once with introns and once without 
introns) were initially screened. Next, a series of features were scored to 
determine candidates for retrotranspostion events. These features include
position and length of the polyA tail, degree of synteny with mouse, coverage 
of repetitive elements, number of exons that can still be aligned to the 
retroGene and degree of divergence from the parent gene.  These features are 
combined and scored against a training set of known pseudogenes using AdaBoost. 
RetroGenes in the final set have an AdaBoost confidence above 50%, less 
than 50% overlap with simple repeats, greater than 65% identity and a blastz 
alignment axtScore greater than 10000. <p>

The "type" field has four possible values: <EM>singleExon</EM> indicates that 
the parent gene is a single exon gene; <EM>Mrna</EM> indicates that the parent 
gene is a spliced mrna that has no annotation in NCBI refSeq, UCSC knownGene or 
Mammalian Gene Collection (MGC); <EM>Better</EM> indicates that there is a 
spliced mrna that overlaps the retrogene, indicating transcription; 
<EM>Good</EM> indicates that this is a highly confident prediction. 
<p> 
These features can be downloaded from the table pseudoGeneLink in many formats 
using the Table Browser option on the blue bar. 
</P>
<H2>References</H2>
<P>
Kent WJ, Baertsch R, Hinrichs A, Miller W, and Haussler D (2003).
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1" 
TARGET=_blank>Evolution's cauldron: 
Duplication, deletion, and rearrangement in the mouse and human genomes</A>. 

<I>Proc Natl Acad Sci USA</I> 100(20):11484-11489 Sep 30 2003.
<P>
Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison R, Haussler D, and 
Miller W (2003). 
<A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>Human-Mouse 
Alignments with BLASTZ</A>. <em>Genome Res.</em> 13(1):103-7.</P>
Robert E. Schapire.
<A HREF="http://www.research.att.com/~schapire/cgi-bin/uncompress-papers/Schapire99d.ps" TARGET=_blank>
Theoretical views of boosting and applications</A>.
<em>In Tenth International Conference on Algorithmic Learning Theory, 1999. </em>
<p>
<H2>Credits</H2>
<P>
The pseudoMasker program and browser track were developed by 
<A HREF="mailto:&#98;a&#101;r&#116;&#115;&#99;&#104;&#64;&#115;&#111;&#101;.
&#117;&#99;sc.
&#101;&#100;&#117;">
Robert Baertsch</A> at UCSC.
<!-- above address is baertsch at soe.ucsc.edu -->


