<H2>Description</H2>
<p>This track displays regulatory potential (RP) score, computed from
alignments of human, chimpanzee (panTro2), macaque (rheMac2), mouse (mm8), rat
(rn4), dog (canFam2), and cow (bosTau2).</p>

<p>RP scores compare frequencies of short alignment patterns between known
regulatory elements and neutral DNA. Results from a calibration study
investigating sensitivity and specificity RP scores on the hemoglobin
beta gene cluster suggest the use of a threshold ~0.0 for identifying new
putative regulatory elements. </p>

<p>The default viewing range for this track is from 0 to 0.01
(score values below the 0 default indicate resemblance to alignment patterns
typical of neutral DNA, while score values above the 0.01 default indicate very
marked resemblance to alignment patterns typical of regulatory elements in the
training set). The range of RP scores from 0 to 0.01 contains the prediction
threshold suggested by calibration studies, and provides an effective
visualization of the score for most genomic loci. However, the user can specify
different viewing ranges if desired. Note: Absence of a score value at a given
location indicates lack of sufficient alignment -- scores are computed
for all regions of the reference genome in which no region of more than
100 bases lacks alignment in at least three non-human species.</p>

<p>This track may be configured in a variety of ways to highlight different
aspects of the displayed information. Click the &quot;Graph configuration
help&quot; link for an explanation of the configuration options. </p>

<H2>Methods</H2>

<p>The comparison employs log-ratios of transitions probabilities from two
variable order Markov models. Training the score entails selecting
appropriate alphabet (alignment column symbols) and maximal order (length of
the longest patterns = order + 1) for the Markov models, and estimating their
transition probabilities, based on alignment data from known
regulatory elements and ancestral repeats. The scores in this track 
are computed using a maximal order of 2. </p>

<p>In the track, score values are displayed using a system of overlapping
windows of size 100 bp along sufficiently alignable portions of the
human sequence. Log-ratios are added over positions in a window, and
the sum is normalized for length. </p>

<H2>Credits</H2>

<p>Work on RP scores is performed by members of the <a
href="http://www.bx.psu.edu/" target="_blank">Comparative Genomics and
Bioinformatics Center at Penn State University</a>. More information on this
research and the collection of known regulatory elements used in training the
score can be found at this site. </p>

<H2>References</H2>

<p>King DC, Taylor J, Elnitski L, Chiaromonte F, Miller W, Hardison
RC. (2005). <a href="http://www.genome.org/cgi/content/abstract/15/8/1051">
Evaluation of regulatory potential and conservation scores for detecting cis-regulatory modules in
aligned mammalian genome sequences.</a>. <em>Genome Res.</em> 15(8):1051-60. </p>

<p>Kolbe D, Taylor J, Elnitski L, Eswara P, Li J, Miller W, Hardison RC and
Chiaromonte F. (2004). <a href="http://www.genome.org/cgi/content/full/14/4/700"
target="_blank">Regulatory potential scores from genome-wide three-way
alignments of human, mouse, and rat</a>. <em>Genome Res.</em> 14(4):700-707. </p>

