<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
</head>
<body>
The <I>Nanoarchaeum equitans Kin4-M</I> genome is 0.49 Million bp long and contains approximately 582
predicted genes. <I>N. equitans</I> is a hyperthermophilic archaeon found as a parasite to Ignicoccus.<BR><BR>
The sequence was released Oct. 2003 by the DIVERSA, and was described
in <I>Proc Natl Acad Sci U S A</I> <B>100</B>:12984-8 (2003) Waters E, Hohn MJ, Ahel I, Graham DE, Adams MD, <I>et al.</I> &nbsp;"<a
 href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=14566062">The genome of Nanoarchaeum equitans: insights into early archaeal evolution and derived parasitism.
</a>"
<blockquote><B>Abstract:</B> The hyperthermophile Nanoarchaeum equitans is an obligate symbiont growing in coculture with the crenarchaeon Ignicoccus. Ribosomal protein and rRNA-based phylogenies place its branching point early in the archaeal lineage, representing the new archaeal kingdom Nanoarchaeota. The N. equitans genome (490,885 base pairs) encodes the machinery for information processing and repair, but lacks genes for lipid, cofactor, amino acid, or nucleotide biosyntheses. It is the smallest microbial genome sequenced to date, and also one of the most compact, with 95% of the DNA predicted to encode proteins or stable RNAs. Its limited biosynthetic and catabolic capacity indicates that N. equitans' symbiotic relationship to Ignicoccus is parasitic, making it the only known archaeal parasite. Unlike the small... [Click above reference link for full abstract]
</blockquote>
<P>
<h3>Sample position queries</h3>
<p>
A genome position can be specified by chromosomal coordinate range, COG
ID, or keywords from the GenBank or TIGR description of a gene. 
The available chromosome/plasmid names are:<BR><BR>
<table border=2 cellspacing=2 cellpadding=2>
<tr><th>Browser Chrom/Plasmid Name</th><th>Length (bp)</th><th>GC Content (%)</th><th>Gene Count</th><th>NCBI RefSeq Accession</th></tr>
<tr><td><a href="/cgi-bin/hgTracks?db=nanEqu1&position=chr:1-0490885">chr</a></td><td align=right>490885</td><td align=right>31.56</td><td align=right>582</td><td align=center><a href="http://www.ncbi.nlm.nih.gov/entrez/viewer.fcgi?val=NC_005213">NC_005213</a></td></tr>
</table>

<p>
The following list shows examples of valid position queries for this
genome:&nbsp;
</p>
<p>
<table cellspacing="1" cellpadding="1" border="0">
<tr><th align=left>Request:</th><th align=left>Genome Browser Response:</th></tr>
<tr><td>chr</td><td>Displays the entire sequence "chr" in the browser window</td></tr>
<tr><td>chr:1-10000&nbsp;&nbsp;&nbsp;&nbsp;</td><td>Displays first ten thousand bases of the sequence "chr"</td></tr>
<tr><td>transporter&nbsp;&nbsp;&nbsp;&nbsp;</td><td>Lists all genes with "transporter" in the name or description</td></tr>
<tr><td>NEQ009</td><td>Display genome at position of gene NEQ009</td></tr>
</tbody>
</table>
<P>
<HR>
<H3>Credits</H3>
The Archaeal Genome Browsers at UCSC were developed by members of the
<a href="http://lowelab.ucsc.edu/">Lowe Lab</a> (Kevin Schneider,
Katherine Pollard, Andy Pohl, Todd Lowe) and Robert Baertsch, with significant support from
the <a href="http://genome.ucsc.edu/staff.html">UCSC Human Genome
Browser group</a>.
The Archaeal Browsers are run by a slightly modified
version of the <a href="http://genome.ucsc.edu/">UCSC Human Genome
Browser</a> system.  All queries, bug reports, content corrections, suggested improvements,
and new track data submissions should be sent to Todd Lowe (lowe
@soe.ucsc.edu).

<P>If you use the browser for your research and wish to cite it, a
paper describing the work has been accepted to the 2006 <I>Nucleic
Acids Research</I>
<a href="http://www3.oup.co.uk/nar/database/a/">Database Issue</a> (KL Schneider, KS Pollard, R Baertsch, A Pohl, TM Lowe). 
Citations will help us obtain funding to continue
development of this community resource.

</body>
</html>
