<H2>Description</H2>
<P>This track displays <A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>BLASTZ</A> alignments of the Feb. 2003 mouse draft
assembly to the human genome filtered to display only the best alignment for any
given region of the human genome. The track has an optional
feature that color codes alignments to indicate the mouse chromosomes from which 
they are derived. To activate the color feature,
click the <I>on</I> radio button next to &quot;Color track based on chromosome&quot;.
<br>
<H2>Methods</H2>
For BLASTZ, we use 12 of 19 seeds and then score using: 
<br>
<table>
<tr><td>&nbsp;</td><td>  A</td><td>  C</td><td>  G</td><td>  T</td></tr>
<tr><td>A</td><td>&nbsp;&nbsp;91</td><td>-114</td><td>&nbsp;-31</td><td>-123</td></tr>
<tr><td>C</td><td>-114</td><td>&nbsp;100</td><td>-125</td><td>&nbsp;-31</td></tr>
<tr><td>G</td><td>&nbsp;-31</td><td>-125</td><td>&nbsp;100</td><td>-114</td></tr>
<tr><td>T</td><td>-123</td><td>&nbsp;-31</td><td>-114</td><td>&nbsp;91</td></tr>
</table>
<br>
  O = 400, E = 30, K = 3000, L = 3000, M = 50
<br><br>
We then do a second pass at reduced stringency (7mer seeds and
MSP threshold of K=2200) to attempt to fill in gaps of up to about 10Kbp.
Lineage specific repeats are abridged during this alignment.
<br>


<H2>Credits</H2>
These alignments are contributed by Scott Schwartz from the <A HREF="http://bio.cse.psu.edu/"
TARGET=_blank>Penn State Bioinformatics Group</A>. The best in genome filtering
is done by UCSC's axtBest program. Mouse sequence data is provided by the 
Mouse Genome Sequencing Consortium.  The web display is by Jim Kent.
