<H2>Description and Credits</H2>

<P>
Slam predicts coding exons and conserved noncoding regions in a pair of homologous DNA sequences, incorporating both statistical sequence properties and degree of conservation in making the predictions.  This particular annotation uses the Nov. 2002 (rn1) assembly of the rat genome. The model is symmetric and the same gene structure (with possibly different exon lengths) is predicted in both sequences.


<P>
The symmetry of the model gives it a higher degree of accuracy for regions where the true underlying gene structures contain the same number of coding exons, in cases where this is not true, or when one of the sequences is of lower quality and contains in-frame stop codons, the resulting predictions tend to have lower accuracy.

<P>
More information on the accuracy of the predictions can be found at <A HREF="http://bio.math.berkeley.edu/slam/rat">http://bio.math.berkeley.edu/slam/rat</A>.  A web server for individual requests is available at <A HREF="http://bio.math.berkeley.edu/slam">http://bio.math.berkeley.edu/slam</A>.

<P>
References
<br>M. Alexandersson, S. Cawley, L. Pachter (2003) SLAM - Cross-species Gene Finding and Alignment with a Generalized Pair Hidden Markov Model. <I>Genome Research 13(3):496-502.</I>
<br>L. Pachter , M. Alexandersson, S. Cawley (2002) Applications of Generalized Pair Hidden Markov Models to Alignment and Gene Finding Problems, <I>Journal of Computational Biology</I> 9(2):389-400.
<br>L. Pachter, M. Alexandersson, S. Cawley (2001) Applications of Generalized Pair Hidden Markov Models to Alignment and Gene Finding Problems, <I>Proceedings of the Fifth Annual International Conference on Computational Molecular Biology</I> (RECOMB 2001)

