<H2>Description</H2>
<P>
The
<A HREF="http://www.affymetrix.com/products/arrays/specific/mouse_exon.affx"
TARGET=_blank>GeneChip Mouse Exon 1.0 ST Array</A> 
contains approximately 1.2 million
probe sets that interrogate about 1 million exons from the mouse
transcriptome.  This track shows the alignments of the 25-mer
oligonucleotide probes and transcript clusters(groups of putative
alternatively spliced transcripts from the same gene).  The intron/exon
structure of each transcript cluster is also included in the display.
</P>

Exons were derived from a variety of annotations that have been divided
into the classes Core, Extended and Full, Free and Ambiguous.
<UL> 
  <LI><B>Core:
    </B>RefSeq transcripts, full-length GenBank mRNAs</LI>
  <LI><B>Extended:</B> 
    dbEst alignments, Ensembl annotations, syntenic mRNA from rat and mouse, 
    microRNA annotations, MITOMAP annotations, Vega genes, Vega pseudogenes
  <LI><B>Full:</B>
    Geneid genes, Genscan genes, Genscan Subopt, Exoniphy, RNA genes, SGP genes,
    Twinscan genes
  </LI>
  <LI><B>Free:</B>
    Designed against annotations which were merged such that no
    single annotation (or evidence) contains the probe set.
  </LI>
  <LI><B>Ambiguous:</B>
    Can not be unambiguously assigned to a particular transcript cluster
  </LI>
</UL></P> 

<P>
Probe sets are colored by class with the Core probe sets being the
darkest and the Ambiguous being the lightest color.  Score vs. class:
<UL>
<LI> 1000 - Core
<LI>  700 - Extended
<LI>  400 - Full
<LI>  200 - Free
<LI>   50 - Ambiguous
</UL>
</P>

<H2>Credits and References</H2>
<P>
The probes were designed by 
<A HREF = "http://affymetrix.com" TARGET = _BLANK>Affymetrix</A>
from sequences obtained from various sources
such as mouse RefSeq, GenBank mRNAs and dbEST.  Gene predictions from
published algorithms such as GENSCAN, GENSCAN suboptimal, Ensembl,
geneid, sgp and TWINSCAN were also included.  Other miscellaneous
content on the array includes the microRNA Registry.
</P>
