<H2>Description</H2>
<P>
The whole genome shotgun assembly was constructed with the JGI assembler, JAZZ, paired end sequencing reads at a coverage of 5.7X produced at JGI, Myriad Genetics, and Celera Genomics. All reads are plasmid, cosmid, or BAC end-sequences, with the predominant coverage coming from 2 kb insert plasmids. The assembly contains 8,597 scaffolds totaling 319 million base pairs; the largest 679 scaffolds total 160 million base pairs. Gene modeling and analysis were performed at JGI, Singapore, Cambridge, and the Institute for Systems Biology.

<H2>Methods</H2>
<P>The alignments were done 
with 
<A HREF="http://www.genome.org/cgi/content/abstract/12/4/656" TARGET=_blank>blat</A> 
in translated protein mode requiring two nearby 4-mer matches
to trigger a detailed alignment.  The human
genome was masked with 
<A HREF="http://repeatmasker.genome.washington.edu/cgi-bin/RepeatMasker" TARGET=_blank>RepeatMasker</A> and Tandem Repeat Finder before 
running blat. </P>

<H2>Credits</H2>
<P>
The 3.0 draft from <A HREF="http://genome.jgi-psf.org/fugu6/fugu6.info.html" TARGET=_blank>http://genome.jgi-psf.org/fugu6/fugu6.info.html</A> was used in the UCSC fugu BLAT alignments.
These data have been provided freely by the DOE Joint Genome Institute for use in this publication/correspondence only. </P>
