<html>
<body>

<h2>Description</h2>
<p>
This dataset represents 31 independent samples of different regions of developing craniofacial
structures obtained from embryonic day 10.5 and 11.5 mouse embryos. RNA was prepared separately from
each sample and subjected to a custom mRNA microlabeling protocol so as to allow for hybridization
to individual Affymetrix <a href="http://www.affymetrix.com/estore/catalog/131476/AFFY/Mouse-Gene-ST-Arrays#1_1"
target="_blank">Mouse Genes 1.0 ST Arrays</a>. The various samples correspond to independent
replicates from the following cell types:
<ul>
<li>e10.5 Rathke's Pouch cells
<li>e10.5 Nasal Pit cells
<li>e10.5 Medial Neuroepithelial region cells
<li>e10.5 Medial Nasal Prominence cells
<li>e10.5 Maxillary Arch cells
<li>e10.5 Lateral Nasal Prominence cells
<li>e11.5 medial Nasal Prominence cells
</ul>
</p>

<h2>Display Conventions</h2>
<p>
In dense mode, the track color denotes the average signal over all experiments on a log base 2 scale.
Lighter colors correspond to lower signals and darker colors correspond to higher signals. In full
mode, the color of each item represents the log base 2 ratio of the signal of that particular
experiment to the median signal of all experiments for that probe.
</p>

<h2>Methods</h2>
<p>
Microarry data was subjected to <a href="https://www.ncbi.nlm.nih.gov/pubmed/12925520" target="_blank"
>RMA normalization</a> and individual gene expression levels per probeset per sample were expressed
as a ratio relative to the level of that probeset's expression using whole Postnatal day 1 mouse RNA
as a universal reference. Colored blocks extend over the length of the corresponding gene and depict
expression in the craniofacial sample relative to that in the reference RNA sample.
</p>

<h2>Credits</h2>
<p>
This track was created with the help of the following people:
<ul>
<li>Dissections, RNA purification, and labeling were carried out by Eric Brunskill and Steven Potter.
<li>Hybridization and processing by Sean Smith and Hung-Chi Liang in the Cincinnati Children's Hospital
gene expression analysis core.
<li>Data was analyzed and transformed into UCSC genome browser array
format by Andrew Plassard and Bruce Aronow.
</ul>

<p>
This work was part of the <a href="http://www.facebase.org/node/154" target="_blank"
>Global Gene Expression Atlas of Craniofacial Development</a> project being
carried out by the <a href="http://www.facebase.org/" target="_blank">Facebase Consortium</a>.
</p>

Contact: <A HREF="mailto:&#66;&#114;&#117;&#99;e.
&#65;&#114;&#111;n&#111;&#119;&#64;&#99;&#99;&#104;m&#99;.
&#111;&#114;g">
&#66;&#114;&#117;&#99;e.
&#65;&#114;&#111;n&#111;&#119;&#64;&#99;&#99;&#104;m&#99;.
&#111;&#114;g</A>
<!-- above address is Bruce.Aronow at cchmc.org -->


<h2>References</h2>
<p>
Brunskill EW, Potter AS, Distasio A, Dexheimer P, Plassard A, Aronow BJ, Potter SS.
<a href="http://www.sciencedirect.com/science/article/pii/S0012160614002231" target="_blank">
A gene expression atlas of early craniofacial development</a>.
<em>Dev Biol</em>. 2014 Jul 15;391(2):133-46.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/24780627" target="_blank">24780627</a>
</p>

</body>
</html>
