<H2>Description &amp; Credits</H2>

<P>
Slam predicts coding exons and conserved noncoding regions in a pair of
homologous DNA sequences, incorporating both statistical sequence properties
and degree of conservation into predictions.  The model is symmetric and the
same structure (with possibly different lengths) is predicted in both
sequences.

<P>
The CNS (conserved non-coding sequence) predictions are ab-initio
predictions of conserved regions that do not fit in with a gene structure.
Thus, SLAM is not simply trying to predict conserved regions to be coding,
but is classifying such regions according to an overall probabilistic model
of gene structure. The set of SLAM CNS predictions is therefore highly
enriched for conserved <b>non-coding</b> regions.

<P>
More information and a web server can be found at <A
HREF="http://baboon.math.berkeley.edu/~syntenic/slam.html">http://baboon.math.berkeley.edu/~syntenic/slam.html</A>.

<P>
References
<br>L. Pachter , M. Alexandersson, S. Cawley (2002) Applications of
Generalized Pair Hidden Markov Models to Alignment and Gene Finding
Problems, <I>Journal of Computational Biology</I> 9(2):389-400.
<br>M. Alexandersson, S. Cawley, L. Pachter (2002) SLAM - Cross-species Gene
Finding and Alignment with a Generalized Pair Hidden Markov Model. <I>Genome
Research 13(3):496-502.</I>

