<H2>Description</H2> <P>
This track shows a measure of evolutionary conservation in Sulfulobus species based on a phylogenetic hidden Markov model (phylo-HMM).
A multiz alignment of the Sulfolobus solfataricus (June 2001), Sulfolobus 
tokodaii (October 2001), and Sulfolobus acidocaldarius (July 2005) genomes 
was used to generate the annotation. The phylogenetic tree for these 
organisms is shown below, with branch lengths drawn to scale and given 
in substitutions per site. The tree is based on fitting a
time-reversible (REV) model for nucleotide substitutions to the whole 
genome multiple alignment of these species.</P>

<P>
<img src="/images/sul-tree.jpg" border="1" align="left">
</P>

<P>
In &quot;full&quot; visibility mode, this track displays pairwise alignments 
of S. acideocaldarius and S. solfactaricus each aligned to the S. tokodaii
genome. The pairwise alignments are displayed in the standard UCSC browser 
&quot;dense&quot; mode using a greyscale density gradient.  The checkboxes 
in the track configuration section allow the exclusion of species from the 
pairwise display; however, this does not remove them from the conservation 
score display.  </P>

<P>
When zoomed-in to the base display level, the track shows the base 
composition of each alignment. The numbers and symbols on the &quot;
S. tokodaii gap&quot; line indicate the lengths of gaps in the 
S. tokodaii sequence at those alignment positions. If the gap size 
is greater than 9, the &quot;+&quot; symbol is displayed. To view detailed 
information about the alignments at a specific position, zoom the display in 
to 30,000 or fewer bases, then click on the alignment.  </P>

<P>
This track may be configured in a variety of ways to highlight different 
aspects of the displayed information. Click the &quot;Graph configuration 
help&quot; link for an explanation of the configuration options.</P>

<H2>Methods</H2>
<P> 
Best-in-genome blastz pairwise alignments of S. tokodaii to each
of the other Sulfolobus species were multiply aligned using a program 
called <em>Multiz</em>. The resulting multiple alignments were then 
assigned conservation scores by phylo-HMM.</P>

<P>
A phylo-HMM is a probabilistic model that describes both the process
of DNA substitution at each site in a genome, and the way this process
changes from one site to the next (Felsenstein and Churchill 1996,
Yang 1995, Siepel and Haussler 2003, Siepel and Haussler 2004).  
A phylo-HMM can be thought
of as a machine that generates a multiple alignment, in the same way
that an ordinary hidden Markov model (HMM) generates an individual
sequence.  While the states of an ordinary HMM are associated with
simple multinomial probability distributions, the states of a
phylo-HMM are associated with more complex distributions defined by
probabilistic phylogenetic models.  These distributions can capture
differences in the rates and patterns of nucleotide
substitution observed in different types of genomic regions (e.g., coding
or noncoding regions, conserved or nonconserved regions).

<P> 
<img src="/images/fels-church.jpg" border="1"
align="right"> To compute a conservation score, we use a
<em>k</em>-state phylo-HMM, whose <em>k</em> associated phylogenetic
models differ only in overall evolutionary rate (Felsenstein and
Churchill 1996, Yang 1995). In the image at right, there are three
<EM>k</EM> states, 
S<SUB>1</SUB>, S<SUB>2</SUB>, and S<SUB>3</SUB>, but in practice we 
use <EM>k</EM> = 10.  
A phylogenetic model is estimated globally, using the discrete gamma model
for rate variation (Yang 1994), then a scaled version of the estimated model
is associated with each state in a phylo-HMM. There is a
separate &quot;rate constant&quot;, <em>r<sub>i</sub></em>, for each state i, 
which is multiplied by all branch lengths in the globally estimated model.
The transition probabilities between states allow for autocorrelation of
substitution rates, i.e., for adjacent sites to tend to exhibit similar
overall substitution rates.  A single parameter, lambda, describes the
degree of autocorrelation and defines all transition probabilities.  
Here, we have estimated the rate constants from the data,
similarly to Yang (1995) (Siepel and Haussler 2003), but have
allowed lambda to be treated as a tuning parameter.  For the
conservation score, we use the posterior probability that each site was
"generated" by the state having the smallest rate constant.  Because of
the way the rate categories are defined, the plotted values can be
thought of as approximately representing the posterior probability that
each site is among the 10% most conserved sites in the data set
(allowing for autocorrelation of substitution rates).  </P>

<P> 
In this case, the general reversible (REV) substitution model was
used in parameter estimation, and lambda was set to 0.9.  Alignment
gaps were treated as missing data, which sometimes has the effect of
producing undesirably high posterior probabilities in gappy regions of
the alignment.  We are looking at several possible ways of improving
the handling of alignment gaps.</P>

<H2>Credits</H2>
<P>
This track was created at UCSC using the following programs:
<UL>
<LI>
Blastz and multiz from Scott Schwartz and Webb Miller of the 
<A HREF="http://bio.cse.psu.edu/" TARGET=_blank>Penn State Bioinformatics 
Group</A>. 
<LI>
AxtBest, axtChain, chainNet, netSyntenic, and netClass 
developed by Jim Kent at UCSC. 
<LI>Tree estimation and phylo-HMM software by Adam Siepel at UCSC.
<LI>"Wiggle track" plotting software by Hiram Clawson at UCSC.
</UL>
</P>

<H2>References</H2>

<H3>Phylo-HMM</H3>
<P>
Felsenstein J and Churchill GA (1996).  A hidden Markov model approach to
variation among sites in rate of evolution.
<em>Mol Biol Evol</em> 13:93-104. </P>
<P>
Siepel A and Haussler D (2003).  
<a href="http://www.cse.ucsc.edu/~acs/phyhmm.pdf">Combining
phylogenetic and hidden Markov models in biosequence analysis</a>.
In <em> Proceedings of the Seventh Annual 
International Conference on Computational Molecular Biology (RECOMB
2003)</em>, pp. 277-286. </P>
<P>
Siepel A and Haussler D (2004). Phylogenetic hidden Markov models.
In R. Nielsen, ed.,  Statistical Methods in Molecular Evolution, Springer  
(in press). </P>
<P>
Yang Z (1994).  Maximum likelihood phylogenetic estimation from DNA sequences
with variable rates over sites: approximate methods,
<em>J Mol Evol</em> 39:306-314.</P>
<P>
Yang Z (1995).  A space-time process model for the evolution of DNA
sequences.  <em>Genetics</em>, 139:993-1005. </P>

<H3>Chain/Net</H3>
<P>
Kent WJ, Baertsch R, Hinrichs A, Miller W, and Haussler D (2003). 
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1" 
TARGET=_blank>Evolution's cauldron: 
Duplication, deletion, and rearrangement in the mouse and human genomes</A>. 
<I>Proc Natl Acad Sci USA</I> 100(20):11484-11489 Sep 30 2003.

<H3>Multiz</H3>
<P>
<P>
Blanchette M, Kent WJ, Riemer C, Elnitski L, Smit AFA, Roskin KM, Baertsch R,
Rosenbloom K, Clawson H, Green ED, Haussler D, Miller W. (2004). 
<A HREF="http://www.genome.org/cgi/content/abstract/14/4/708" TARGET=_blank>Aligning Multiple Genomic Sequences with the Threaded Blockset Aligner</A>.
<em>Genome Res.</em> 14(4):708-15.
</P>

<H3>Blastz</H3>
<P>
Chiaromonte F, Yap VB, and Miller W (2002). 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" 
TARGET=_blank>Scoring pairwise genomic sequence alignments</A>. 
<em>Pac Symp Biocomput</em> 2002;:115-26.</P>
<P>
Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison R, Haussler D, and 
Miller W. (2003). 
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103" 
TARGET=_blank>Human-Mouse Alignments with BLASTZ</A>. 
<em>Genome Res.</em> 13(1):103-7.</P>
