<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=215>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://www.broad.mit.edu/mammals/"
    TARGET=_blank>
    <IMG SRC="/images/Cavia_porcellus.jpg" WIDTH=200 ALT="Guinea Pig"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Cavia porcellus</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of .
      <A HREF="http://www.genome.gov/pressDisplay.cfm?photoID=75" 
      TARGET=_blank>Bill Branson, NHGRI</A>
    </FONT>
  </TD></TR>
</TABLE>

<P>
The Feb. 2008 <em>Cavia porcellus</em> draft assembly 
(Broad Institute cavPor3 ) was produced by the <A HREF="http://www.broad.mit.edu/" 
      TARGET=_blank>Broad Institute</A> at MIT and Harvard.

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
Guinea Pig genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>scaffold_1</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all of scaffold 1</TD>
    </TR>
    <TR><TD VALIGN=Top>scaffold_2:1-1000000</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first million bases of scaffold 2</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>scaffold_123:1000000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of scaffold 123 that spans 2000 bases, starting with position 1000000 </TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chrM</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays all if the mitochondial chromosome </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>BX950328</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession
		number BX950328</TD>
    </TR>
    <TR><TD VALIGN=Top>EB366004</TD>
        <TD WIDTH=14>&nbsp;</TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession
		BU335953 on chr Z</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
<!--
    <TR><TD VALIGN=Top>pseudogene mRNA</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes, but not cDNAs</TD>
    </TR>
-->
    <TR><TD VALIGN=Top>homeobox caudal</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top>zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top>zhang</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Zhang</TD>
    </TR>
    <TR><TD VALIGN=Top>Taguchi,T.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author T. Taguchi</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Taguchi T</em>, internally it
	uses the format <em>Taguchi,T.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
The guinea pig is one of 24 mammals whose genomes are being sequenced as part 
of the <A HREF="http://www.broad.mit.edu/mammals/">Mammalian Genome Project</A>, 
funded by NIH. While most of these are slated for low-coverage (2X), a limited 
subset (including the guinea pig genome) are being sequenced to a higher quality (6-7X).
</P><P> 
The genome has been sequenced to 6.76X coverage with 95.55% of bases assembled.
A total of 3143 scaffolds cover 2,722,377,657 bases (2.17% in gaps),
with 50% of the scaffolds having a (N50) length of at least 27,408,292 bases 
(not including gaps).  Approximately 97.52% of contigs have been assembled 
into scaffolds of 200,000 bases or more. This genome assembly has a Q20 coverage
of 5.99X with 99.57% q>=20 and 97.72% q >= 50.
<!--
the following percentages of base quality:

<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
<TR><TH>Bases </TD><TD WIDTH=10>&nbsp;</TH><TH>Quality</TR>
<TR><TD>100%  </TD><TD WIDTH=10>&nbsp;</TD><TD>q >= 1 </TD></TR>
<TR><TD>99.57%</TD><TD WIDTH=10>&nbsp;</TD><TD>q >= 20</TD></TR>
<TR><TD>99.02%</TD><TD WIDTH=10>&nbsp;</TD><TD>q >= 30</TD></TR>
<TR><TD>98.4% </TD><TD WIDTH=10>&nbsp;</TD><TD>q >= 40</TD></TR>
<TR><TD>97.72%</TD><TD WIDTH=10>&nbsp;</TD><TD>q >= 50</TD></TR>
</TABLE>
-->
</P><P> 
In this assembly scaffolds have not been anchored to chromosomes.  However,
the guinea pig mitochondrial sequence is available as the virtual chromosome 
&quot;chrM&quot;. 
</P>


<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/cavPor3/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#guinea_pig">Downloads</A> page. 

<!--  *** Developer -- remove this if no conditions for use:-->
These data have
<A HREF="../goldenPath/credits.html#guinea_pig_use">specific 
conditions for use</A>. 
<!-- -->
The Guinea Pig browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#guinea_pig_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
