<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
</head>
<body>
The <I>Methanosarcina mazei Go1</I> genome is 4.10 Million bp long and contains approximately 3438
predicted genes. <BR><BR>
The sequence was released July 2002 by University of Goettingen, and was described
in <I>J Mol Microbiol Biotechnol</I> <B>4</B>:453-61 (2002) Deppenmeier U, Johann A, Hartsch T, Merkl R, Schmitz RA, <I>et al.</I> &nbsp;"<a
 href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=12125824">The genome of Methanosarcina mazei: evidence for lateral gene transfer between bacteria and archaea.
</a>"
<blockquote><B>Abstract:</B> The Archaeon Methanosarcina mazei and related species are of great ecological importance as they are the only organisms fermenting acetate, methylamines and methanol to methane, carbon dioxide and ammonia (in case of methylamines). Since acetate is the precursor of 60% of the methane produced on earth these organisms contribute significantly to the production of this greenhouse gas, e.g. in rice paddies. The 4,096,345 base pairs circular chromosome of M. mazei is more than twice as large as the genomes of the methanogenic Archaea currently completely sequenced (Bult et al., 1996; Smith et al., 1997). 3,371 open reading frames (ORFs) were identified. Based on currently available sequence data 376 of these ORFs are Methanosarcina-specific and 1,043 ORFs find their closest homologue in the ba... [Click above reference link for full abstract]
</blockquote>
<P>
<h3>Sample position queries</h3>
<p>
A genome position can be specified by chromosomal coordinate range, COG
ID, or keywords from the GenBank or TIGR description of a gene. 
The available chromosome/plasmid names are:<BR><BR>
<table border="2" cellspacing=2 cellpadding=2>
<tr><th>Browser Chrom/Plasmid Name</th><th>Length (bp)</th><th>Gene Count</ht><th>NCBI RefSeq Accession</th></tr>
<tr><td>chr</td><td align=right>4096345</td><td align=right>3438</td><td align=center>NC_003901</td></tr>
</table>

<p>
The following list shows examples of valid position queries for this
genome:&nbsp;
</p>
<p>
<table cellspacing="1" cellpadding="1" border="0">
<tr><th align=left>Request:</th><th align=left>Genome Browser Response:</th></tr>
<tr><td>chr</td><td>Displays the entire sequence "chr" in the browser window</td></tr>
<tr><td>chr:1-10000</td><td>Displays first ten thousand bases of the sequence "chr"</td></tr>
<tr><td>transporter&nbsp;&nbsp;&nbsp;&nbsp;</td><td>Lists all genes with "transporter" in the name or description</td></tr>
<tr><td>MM0010</td><td>Display genome at position of gene MM0010</td></tr>
</tbody>
</table>
<P>
<HR>
<H3>Credits</H3>
The Archaeal Genome Browsers at UCSC were developed by members of the
<a href="http://lowelab.ucsc.edu/">Lowe Lab</a> (Kevin Schneider,
Katherine Pollard, Andy Pohl, Todd Lowe) and Robert Baertsch, with significant support from
the <a href="http://genome.ucsc.edu/staff.html">UCSC Human Genome
Browser group</a>.
The Archaeal Browsers are run by a slightly modified
version of the <a href="http://genome.ucsc.edu/">UCSC Human Genome
Browser</a> system.  All queries, bug reports, content corrections, suggested improvements,
and new track data submissions should be sent to Todd Lowe (lowe
@soe.ucsc.edu).

<P>If you use the browser for your research and wish to cite it, a
paper describing the work has been accepted to the 2006 <I>Nucleic
Acids Research</I>
<a href="http://www3.oup.co.uk/nar/database/a/">Database Issue</a> (KL Schneider, KS Pollard, R Baertsch, A Pohl, TM Lowe). 
Citations will help us obtain funding to continue
development of this community resource.

</body>
</html>
