<H2>Description</H2>
<P>
These tracks display the level of sequence uniqueness of the reference hg18
genome.  They were generated using different window sizes and high signal
will be found in areas where the sequence is unique.

<H2>Methods</H2>
<P>
The <em>CRG Alignability</em> tracks shows how uniquely k-mer sequences align
to a region of the genome. To generate the data, sliding windows of k-mers
(where k is 36, 40, 50, 75 or 100 nts) of the human genome sequence were
mapped back to the genome using the GEM mapper aligner allowing 0, 1 or 2 
mismatches. For each window, a mapability score was computed 
(S = 1/(number of bases of match found)).

<H2>Credits</H2>
<P>
The <em>CRG Alignability</em> track wsa created by Thomas Derrien and Paolo
Ribeca in <A HREF="http://pasteur.crg.es/portal/page/portal/Internet/02_Research/01_Programmes/01_Bioinformatics_Genomics" TARGET=_BLANK>Roderic Guigo's lab</A> at the Centre for Genomic Regulation (CRG), Barcelona, Spain. 
</P>
