
<H2>Description</H2>
<P>
This track contains genome-wide maps of Repressor Element 1-Silencing Transcription Factor (REST) binding sites in mouse stem cells, generated by the Stem Cell and Developmental Biology Group at the Genome Institute of Singapore (GIS). REST is a zinc finger transcriptional repressor that regulates a large cohort of neural genes throughout development. Aberrant REST activity has been implicated in various disease states, including cancer, epilepsy, Huntington's disease and cardiovascular disease.</P>

<H2>Display Conventions and Configuration</H2>
<P>
The entries with names that start with "ESC_only" are colored red.  The others are colored blue.
</P>

<H2>Methods</H2>
<P>
The ChIP-PET (chromatin immunoprecipitation coupled to paired-end di-tagging) method was used to identify genomic loci occupied by REST. 36bp PETs generated from ChIP DNA fragments enriched for REST binding regions were sequenced using the high-throughput 454 sequencing technology. REST binding sites were defined as those having at least five overlapping PETs (PET5+). Mapping was carried out in two isogenic cell lines: the mouse embyronic stem cell E14, and the neural stem cell line NS5. The REST binding sites are divided into two categories: (1) ES and NS shared sites, "ES_NS" (PET5+ in ES, PET2+ in NS); (2) ES-specific sites, "ES_only" (PET5+ in ES, PET<2 in NS). Negligible numbers of NS-only sites were identified. The coordinates displayed represent the PET minimal overlap region, defined as the region that is overlapped in common by all PETs constituting a cluster.
</P>
<H2>Credits</H2>
<P>
These data were generated by the Stanton Group at the Genome Institute of Singapore, and have been published [see below]. 
</P>

<H2>References</H2>
<P>
Johnson R, Teh CH, Kunarso G, Wong KY, Srinivasan G, Cooper ML, Volta M, Chan SS, Lipovich L, Pollard SM, Karuturi RK, Wei CL, Buckley NJ, Stanton LW. REST regulates distinct transcriptional networks in embryonic and neural stem cells. PLoS Biol. 2008 Oct 28;6(10):e256.
</P>
