<H2>Description</H2>
<P>
MicroRNA target sites in 3' UTRs as predicted by PicTar,
based on the 3' UTRs annoatated by WormBase and based on our own
"guessed" 3' UTRs (see below).
</P>

<H2>Methods</H2>
<P>
The original PicTar algorithm was published in Krek et al, Nature
Genetics, 2005. Fly predictions were published in Grun et al 2006.
Displayed are updated predictions as published in Lall et
al, Current Biology 2006. Briefly, PicTar is a Hidden Markov Model that
assigns probabilities to 3' UTR subsequences to be a binding site for a
microRNA, considers all possible ways the 3' UTR could be bound by
microRNAs, and then uses a maximum likelihood method to compute the
optimal Likelihood under which the 3' UTR could be explained by microRNAs
and background. The Score is this likelihood divided by backround, ie the
local basecomposition of each 3' UTR is taken into account. To fit the
Score track conventions of the UCSC browser (integers), we scaled all
scores by the maximum score of all microRNA:3' UTR scores observed. Note
that PicTar scores any 3' UTR that has at least 1 aligned, conserved
predicted binding site for a microRNA, but then incorporates all possible
binding sites into the score, even if they appear to be non-conserved.
Since the Score for a 3' UTR is a "phylo" average over all orthologous 3'
UTRs used, "scattered" sites that appear in many species may boost the
score, and individual sites shown in the display may not be aligned and
conserved in all species under consideration.
</P>
<P>
For the nematode target predictions, three nematode species were used
(elegans, briggsae, remanei). Since 3' UTRs were only relatively rarely
annotated in WormBase or RefSeq, we guessed for all transcripts without
annotated 3' UTRs the 3' UTR sequences by extending the CDS by 500 bases
downstream of the STOP codon, but stopped when hitting another CDS.
</P>
<P>
For a detailed analysis regarding costruction of 3' UTRs and signal to noise
ratios and and sensitivity please refer to Lall et al Current biology 2006.
</P>

<H2>References</H2>
<P>
Krek, A, Grün, D, Poy, MN, Wolf R, Rosenberg L, Epstein EJ, MacMenamin P,
da Piedade I, Gunsalus KC, Stoffel M, &amp; Rajewsky N
<A HREF="http://www.nature.com/ng/journal/v37/n5/abs/ng1536.html" TARGET=_blank>
Combinatorial microRNA target predictions,</A>
Nature Genetics  37, 495 - 500 (2005) 
</P>
<P>
S. Lall, D. Grün, A. Krek, K. Chen, Y-L Wang, CN Dewey, P. Sood,
T. Colombo, N. Bray, P. MacMenamin, H.-L. Kao, K. C. Gunsalus, L. Pachter,
F. Piano &amp; N. Rajewsky
<A HREF="http://www.current-biology.com/content/article/abstract?uid=PIIS0960982206010591" TARGET=_blank>
A Genome-Wide Map of Conserved MicroRNA Targets in C. elegans,</A>
Current Biology, Vol 16, 460-471, 7 March 2006
</P>

<H2>Credits</H2>
<P>
Thanks to the Dominic Grün, Yi-Lu Wang, and Nikolaus Rajewsky for
providing this annotation. PicTar is a project of the Rajewsky lab at
NYU's Center for Comaprative Functional Genomics. Much more detailed
information about individual predictions can be found at
<A HREF="http://pictar.bio.nyu.edu/" TARGET=_blank>The Rajewsky Lab,</A>
NYU Department of Biology, including links to other databases. 
</P>
