<P>
The August 2003 <em>Drosophila pseudoobscura</em> genome assembly 
is based on the Freeze 1 assembly produced at the 
<A HREF="http://hgsc.bcm.tmc.edu/projects/drosophila/" TARGET=_blank> 
Baylor Human Genome Sequencing Center</A> (HGSC).  

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a 
sequenced genomic clone, an mRNA or EST or STS marker, 
a scaffold coordinate range, or keywords from 
the GenBank description of an mRNA. The following list shows 
examples of valid position queries for the <em>D. pseudoobscura</em> genome. 
See the
<A HREF="/goldenPath/help/hgTracksHelp.html" TARGET=_blank>User's Guide</A>
for more information.
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
       	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><BR></TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Contig3128_Contig3153</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays all of scaffold Contig3128_Contig3153</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Contig53_Contig54:1-50,000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays first 50,000 bases of scaffold Contig53_Contig54</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>AI501130</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of EST with GenBank accession AI501130</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>AF199335</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession AF199335</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>apoe</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of genome with gene identifier apoE</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>NM_145881</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of genome with RefSeq identifier NM_145881</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>pseudogene mRNA</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes but not cDNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>kruppel zinc finger</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>huntington</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists candidate genes associated with Huntington's disease</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Jones,R.</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author R. Jones</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Jones R</em>, internally it uses 
			the format <em>Jones,R.</em>.</TD>
    </TR>
</TABLE>
<P>
<HR>
<H3>Assembly details</H3>
The Freeze 1 assembly is a whole-genome shotgun assembly, 
using Baylor HGSC's assembly engine Atlas, 
with approximately 7x coverage of the euchromatic portion of the genome.  
There are 759 scaffolds in the assembly.  
Scaffold N50 is 1,018,646 bp; total size is 139.3 Mbp; average 184,465 bp.  
Four large scaffolds were split (due to size) into "A" and "B" parts 
in the downloadable assembly files:
<UL>
<LI>Contig3286_Contig7811 was split into Contig3286_Contig7811A and Contig3286_Contig7811B.</LI>
<LI>Contig4971_Contig7717 was split into Contig4971_Contig7717A and Contig4971_Contig7717B.</LI>
<LI>Contig6811_Contig7852 was split into Contig6811_Contig7852A and Contig6811_Contig7852B.</LI>
<LI>Contig7891_Contig7492 was split into Contig7891_Contig7492A and Contig7891_Contig7492B.</LI>
</UL>
Baylor HGSC has made available a 
<A HREF="ftp://ftp.hgsc.bcm.tmc.edu/pub/data/Dpseudoobscura/fasta/D_pseudo_freeze1_scaffold_to_chromosome_assignments_8_28_03.txt" TARGET=BLANK>
putative chromosome assignment</A> for the majority of larger scaffolds 
(&gt; 90% of unique sequence), based on conservation between the Muller 
elements.
<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser 
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/dp2/">FTP server</A> or the 
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#dp">Downloads</A> page. 
These data are made available with 
<A HREF="http://genome.ucsc.edu/goldenPath/credits.html#dp_use">specific conditions for use</A> (see also 
<A HREF="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" TARGET=_BLANK>here</A>).
The dp2 annotation tracks were generated by UCSC and collaborators worldwide. 
See the <A HREF="/goldenPath/credits.html#dp_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
