<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=210 HEIGHT=250>
  <TR><TD ALIGN=RIGHT><A HREF="http://www.hgsc.bcm.tmc.edu/projects/seaurchin/" 
    TARGET=_blank>
    <IMG SRC="/images/S_purpuratus.jpg" WIDTH=200 HEIGHT=188 ALT="Purple Sea Urchin"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1> <em>Strongylocentrotus purpuratus</em><BR>
    </FONT>
    <FONT SIZE=-2> (Photo courtesy of
      <A HREF="http://biology.fullerton.edu/people/faculty/steve-murray/" TARGET=_blank>Steven Murray, CSU Fullerton</A>)
    </FONT>
  </TD></TR>
</TABLE>

The April 2005 release of the Purple Sea Urchin genome (<I>Strongylocentrotus purpuratus</I>) was produced by Baylor College of Medicine's Human Genome Sequencing Center, and corresponds to their Spur_0.5 whole genome shotgun assembly.  
<H3>Sample position queries</H3>
<P>
A genome position can be specified by a linkage group coordinate range,
the accession number of an mRNA or EST, a gene name, or
keywords from the GenBank description of an mRNA. The following
list shows examples of valid position queries for the
<em>S. purpuratus</em> genome.  Note that some position queries
(e.g. "brachyury") may return matches to the mRNA records of
other species. In these cases, the mRNAs are mapped to their 
homologs in <em>S. purpuratus</em>. 
See the <A HREF="../goldenPath/help/hgTracksHelp.html" 
TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top><B>Genome Browser Response:</B><BR></TD>
    </TR>
    <TR><TD COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top NOWRAP>Group1</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all 24,891 bases of Scaffold5728</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Scaffold49105:1-10,000</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first 10,000 bases of Scaffold49105</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>AY044638</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Searches for regions of genome aligning to mRNA with GenBank accession AY044638</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>cdk4</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Searches for regions of genome aligning to cdk4 gene.</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>cyclin dependent</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists cyclin-dependentprotein mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>davidson</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists accessions deposited by authors named Davidson</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Davidson,E.H.</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author E.H. Davidson</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Davidson EH</em>, internally it uses 
			the format <em>Davidson,E.H.</em></TD>
    </TR>
</TABLE>

<P>
<HR>
<H3>Assembly Details</H3>
<P>
Excluding gaps, the size of the preliminary assembled genome is 835 Mb, and is comprised of 187,943 linearized scaffolds.  The N50 size of the scaffolds is 55 Kb.  The Atlas assembler was used.
<P>
More assembly details are in the <A HREF="ftp://ftp.hgsc.bcm.tmc.edu/pub/data/Spurpuratus/fasta/Spur20050415/READMEApril2005.txt">Spur_0.5 README file</A>, and on the <A HREF="http://www.hgsc.bcm.tmc.edu/projects/seaurchin/">Baylor HGSC Sea Urchin Genome Project Web Page</A>.
