<H3>About the <em>Anopheles gambiae</em> (Mosquito) assembly</H3>
<P>
The 
<em>A. gambiae</em> Release MOZ2 (Sep. 2003?)
draft genome sequence (UCSC version anoGam1) 
was produced by the 
International Anopheles Genome Executive Committee, and 
downloaded from 
<A HREF="http://www.ensembl.org/Anopheles_gambiae/" TARGET=_BLANK>
Ensembl</A>.
The MOZ2 assembly is a 10x whole genome shotgun assembly.  The
assembled portion of the genome is about 278 Mbp in length with a
total of 8,987 unique scaffolds, the largest scaffold being 23.1
Mbp. Approximately 85% of the sequence has been assigned to
chromosomal locations; chromosome arms chr2L, chr2R, chr3L, chr3R, and
chrX are represented by 13, 49, 42, 28, and 10 large scaffolds
respectively. No scaffolds have yet been assigned to the Y chromosome.
The unassigned scaffolds, concatenated together in arbitrary order,
are in the artificial unknown 'chromosome' chrUn.

<P>
Downloads of the anoGam1 data and annotations may be obtained from the 
UCSC Genome Browser 
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/anoGam1/" TARGET=_blank>FTP server</A>
or <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#gambiae">Downloads</A> page. 
The anoGam1 annotation tracks were generated by UCSC and 
collaborators worldwide. Thanks to the WUSTL GSC for 
providing the sequence of this genome.  See the 
<A HREF="/goldenPath/credits.html#gambiae_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of an
mRNA or EST, a chromosomal range, an Ensembl transcript ID, or keywords
from the Genbank description of an mRNA. The following list provides
examples of valid position queries for the Anopheles
genome. Note that some position queries (e.g. &quot;huntington&quot;) may return
matches to the mRNA records of other species. In these cases, the
mRNAs are mapped to their homologs in Anopheles.  See the 
<A HREF="http://genome.cse.ucsc.edu/goldenPath/help/hgTracksHelp.html" TARGET=_blank>
User's Guide</A> for more information.
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><BR></TD>
    </TR>
	
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>chr2L</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays all of chromosome 2L</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>chrYh:1-200000</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays first two hundred thousand bases of chromosome Yh (h = heterochromatin random reads)</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>ENSANGT00000020705.2</TD>
       	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays the region of genome with Ensembl identifier 
	ENSANGT00000020705.2. NOTE: Due to space constraints, the Genome Browser 
	displays abridged Ensembl identifiers, but the full ID name must be used in 
	searches. For example, the ENSANGT00000020705.2 identifier displays as 
	ENSANG20705.2 in the Genome Browser.</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>AY122244</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays region containing <em>A. gambiae</em> mRNA with GenBank accession number AY122244</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>BQ103726</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays region containing <em>A. gambiae</em> EST with GenBank accession BQ103726</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>p53</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs related to the p53 tumor suppressor</TD>
    </TR> 

    <TR><TD VALIGN=Top NOWRAP>PDZ</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs for PDZ domain proteins</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>huntington</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs associated with Huntington's disease</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>zahler</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Zahler</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>Celniker,S.</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs deposited by co-author S. Celniker</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Celniker S</em>, internally it uses 
			the format <em>Celniker,S.</em>.</TD>
    </TR>
</TABLE>
