<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
</head>
<body>
The <I>Pyrococcus horikoshii OT3</I> genome is 1.74 Million bp long and contains approximately 2005
predicted genes. <I>P. horikoshii</I> is a hyperthermophilic archaeon isolated from a hydrothermal vent at a depth of 1395 meters in the Okinawa Trough in the Pacific Ocean. It grows optimally at 98C.<BR><BR>
The sequence was released April 1998 by NITE, and was described
in <I>DNA Res</I> <B>5</B>:55-76 (1998) Kawarabayasi Y, Sawada M, Horikawa H, Haikawa Y, Hino Y, <I>et al.</I> &nbsp;"<a
 href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=9679194">Complete sequence and gene organization of the genome of a hyper-thermophilic archaebacterium, Pyrococcus horikoshii OT3.
</a>"
<blockquote><B>Abstract:</B> The complete sequence of the genome of a hyper-thermophilic archaebacterium, Pyrococcus horikoshii OT3, has been determined by assembling the sequences of the physical map-based contigs of fosmid clones and of long polymerase chain reaction (PCR) products which were used for gap-filling. The entire length of the genome was 1,738,505 bp. The authenticity of the entire genome sequence was supported by restriction analysis of long PCR products, which were directly amplified from the genomic DNA. As the potential protein-coding regions, a total of 2061 open reading frames (ORFs) were assigned, and by similarity search against public databases, 406 (19.7%) were related to genes with putative function and 453 (22.0%) to the sequences registered but with unknown function. The remaining 1202 ORFs ... [Click above reference link for full abstract]
</blockquote>
<P>
<h3>Sample position queries</h3>
<p>
A genome position can be specified by chromosomal coordinate range, COG
ID, or keywords from the GenBank or TIGR description of a gene. 
The available chromosome/plasmid names are:<BR><BR>
<table border="2" cellspacing=2 cellpadding=2>
<tr><th>Browser Chrom/Plasmid Name</th><th>Length (bp)</th><th>Gene Count</ht><th>NCBI RefSeq Accession</th></tr>
<tr><td>chr</td><td align=right>1738505</td><td align=right>2005</td><td align=center>NC_000961</td></tr>
</table>

<p>
The following list shows examples of valid position queries for this
genome:&nbsp;
</p>
<p>
<table cellspacing="1" cellpadding="1" border="0">
<tr><th align=left>Request:</th><th align=left>Genome Browser Response:</th></tr>
<tr><td>chr</td><td>Displays the entire sequence "chr" in the browser window</td></tr>
<tr><td>chr:1-10000</td><td>Displays first ten thousand bases of the sequence "chr"</td></tr>
<tr><td>transporter&nbsp;&nbsp;&nbsp;&nbsp;</td><td>Lists all genes with "transporter" in the name or description</td></tr>
<tr><td>PH0008</td><td>Display genome at position of gene PH0008</td></tr>
</tbody>
</table>
<P>
<HR>
<H3>Credits</H3>
The Archaeal Genome Browsers at UCSC were developed by members of the
<a href="http://lowelab.ucsc.edu/">Lowe Lab</a> (Kevin Schneider,
Katherine Pollard, Andy Pohl, Todd Lowe) and Robert Baertsch, with significant support from
the <a href="http://genome.ucsc.edu/staff.html">UCSC Human Genome
Browser group</a>.
The Archaeal Browsers are run by a slightly modified
version of the <a href="http://genome.ucsc.edu/">UCSC Human Genome
Browser</a> system.  All queries, bug reports, content corrections, suggested improvements,
and new track data submissions should be sent to Todd Lowe (lowe
@soe.ucsc.edu).

<P>If you use the browser for your research and wish to cite it, a
paper describing the work has been accepted to the 2006 <I>Nucleic
Acids Research</I>
<a href="http://www3.oup.co.uk/nar/database/a/">Database Issue</a> (KL Schneider, KS Pollard, R Baertsch, A Pohl, TM Lowe). 
Citations will help us obtain funding to continue
development of this community resource.

</body>
</html>
