<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=185>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://www.ncbi.nlm.nih.gov/genomeprj/13974"
    TARGET=_blank>
    <IMG SRC="/images/Nomascus_leucogenys.jpg" WIDTH=193 HEIGHT=300 ALT="Gibbon"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Nomascus leucogenys</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of NHGRI
      (<A HREF="http://www.genome.gov/pressDisplay.cfm?photoID=74"
      TARGET=_blank>press photos</A>)
    </FONT>
  </TD></TR>
</TABLE>
<P>
The Jan. 2010 <em>Nomascus leucogenys</em> (Northern White-Cheeked Gibbon) draft assembly 
NCBI project 
<A HREF="http://www.ncbi.nlm.nih.gov/nuccore/ADFV00000000.1"
TARGET=_blank>13975</A> (GCA_000146795.1) 
was produced by Gibbon Genome Sequencing Consortium 
(the <A HREF="http://genome.wustl.edu/genomes/"
TARGET=_blank>Genome Sequencing Center</A> 
at Washington University
and Baylor College of Medicine  
<A HREF="http://www.hgsc.bcm.tmc.edu/project-species-p-Gibbon.hgsc?pageLocation=Gibbon"
TARGET=_blank>Human Genome Sequencing Center</A>).


<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
Marmoset genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
</P>
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>GL397261</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all of contig GL397261</TD>
    </TR>
    <TR><TD VALIGN=Top>Contig3759:130,210-140,209</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays ten thousand bases of contig 3759</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Contig2815:171328+1500</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of Contig 2815 that spans 1500 bases, starting with position 171328 </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>EF215092</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession
		number EF215092</TD>
    </TR>
    <TR><TD VALIGN=Top>EH380513</TD>
        <TD WIDTH=14>&nbsp;</TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession
		EH380513</TD>
    </TR>
    <TR><TD VALIGN=Top>homeobox caudal</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top>zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top>datson</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Datson</TD>
    </TR>
    <TR><TD VALIGN=Top>Fuchs,E.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author E. Fuchs</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Fuchs E</em>, internally it
	uses the format <em>Fuchs,E.</em>.</TD>
    </TR>
</TABLE>


<HR>
<H3>Assembly details</H3>
<P>

The genome has been sequenced to 5.6x coverage using Sanger methods.
Included in this assembly are 7,963 scaffolds (GL397261:GL405224) with total length of 2,936,035,333 bases. 

 
For more information about this assembly, see also:
<A HREF="http://www.ncbi.nlm.nih.gov/Traces/wgs/?val=ADFV00"
TARGET=_blank>WGS record at NCBI.</A>
</P>


<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/nomLeu1/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#gibbon">Downloads</A> page. 

The Gibbon browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#gibbon_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
