<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
        <META name="description" content="UCSC Genome Browser FAQ">
        <META name="keywords" content="genome, genome browser, genome assembly, UCSC, bioinformatics, sequence, gene prediction, SNP, EST, mRNA">
	<TITLE>UCSC Genome Bioinformatics: FAQ</TITLE>
	<LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="/images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<!--Hotlinks bar---------------------------------------------------->
<TR><TD COLSPAN=2 HEIGHT=40>
    <TABLE bgcolor="000000" cellpadding="1" cellspacing="1" width="100%" height="27"><TR bgcolor="2636D1"><TD valign="middle">
        <TABLE BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24">
            <TR><TD VALIGN="middle"><font color="#89A1DE">&nbsp; &nbsp;
		<A HREF="/index.html" class="topbar">
			Home</A> &nbsp; - &nbsp;
		<A HREF="/cgi-bin/hgGateway" class="topbar">
			Genome Browser</A> &nbsp; - &nbsp; 
		<A HREF="/cgi-bin/hgNear" class="topbar">
			Family Browser</A> &nbsp; - &nbsp; 
		<A HREF="/cgi-bin/hgBlat?command=start" class="topbar">
			Blat </A> &nbsp; - &nbsp; 
		<A HREF="/cgi-bin/hgText?command=start" class="topbar">
			Table Browser</A> &nbsp; - &nbsp; 
		<A HREF="/goldenPath/help/hgTracksHelp.html" class="topbar">
			Help</A> &nbsp;</font></TD>
        </TR></TABLE>
    </TD></TR></TABLE>
</TD></TR></TABLE>

<!--Content Tables------------------------------------------------------->
<TR><TD CELLPADDING=10>
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>
            <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
                <TR><TD> <FONT SIZE="4"><b>&nbsp;   Frequently Asked Questions: Blat</b></FONT>
                </TD></TR></TABLE>
            <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
                <TR><TD>&nbsp;</TD><TD>
		<UL>
                <LI><A HREF="#blat1">Blat vs. Blast</A>
		</UL>
		<P>
		<A HREF="index.html">Return to FAQ Table of Contents</A>
		<BR>
            </TD><TD WIDTH=15></TD></TR></TABLE><BR>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="blat1"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Blat vs. Blast</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"What are the differences between Blat and Blast?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Blat is an alignment tool like BLAST, but it is structured differently.  On 
			DNA, Blat works by keeping an index of the entire human genome in memory. 
			Thus, the target database of BLAT is not a set of GenBank sequences, but instead an 
			index derived from the current assembly of the entire draft human genome. The 
			index consists of all non-overlapping 11-mers except for those heavily 
			involved in repeats. This smaller size means that Blat is far more easily 
			<A HREF="http://genome.ucsc.edu/mirror.html">mirrored</A>.  Blat of DNA 
			is designed to quickly find sequences of 95% and greater similarity of length 
			40 bases or more. It may miss more divergent or short sequence alignments. 
			<P>
			On protein, Blat uses 4-mers rather than 11-mers, finding protein sequences 
			of 80% and greater similarity to the query of length 20+ amino acids. In 
			practice -- due to sequence divergence rates over evolutionary time -- DNA
			Blat works well within humans and primates, while protein Blat 
			continues to find good matches within terrestrial vertebrates and even earlier 
			organisms for conserved proteins. Within humans, protein Blat gives a much better 
			picture of gene families (paralogs) than DNA Blat. However, BLAST and 
			psi-BLAST at NCBI can find much more remote matches.
			<P>
			From a practical standpoint, Blat has several advantages over BLAST: 
			<UL>
			<LI>speed (no queues, response in seconds) at the price of lesser homology depth
			<LI>the ability to submit a long list of simultaneous queries in fasta format
			<LI>five convenient output sort options
			<LI>a direct link into the UCSC browser
			<LI>alignment block details in natural genomic order
			<LI>an option to launch the alignment later as part of a custom track
			</UL>
			Blat is commonly used to look up the location of a 
			sequence in the genome or determine the exon structure of an mRNA, but expert 
			users can run large batch jobs and make internal parameter sensitivity 
			changes by installing command line Blat on their own Linux server.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
    </TD><TD WIDTH=15></TD></TR>
</TABLE>
</BODY></HTML>

