<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
        <META name="description" content="UCSC Genome Browser FAQ">
        <META name="keywords" content="genome, genome browser, genome assembly, UCSC, bioinformatics, sequence, gene prediction, SNP, EST, mRNA">
	<TITLE>UCSC Genome Bioinformatics: FAQ</TITLE>
	<LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/FAQ.topbar.html"--></TD></TR>
<!--Content Tables------------------------------------------------------->
    <TR><TD COLSPAN=2 CELLPADDING=10>
    <!--outer table is for border purposes-->
   	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
	    <TR><TD>
           	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
		    <TR><TD>
            		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
                	    <TR><TD> <FONT SIZE="4"><b>&nbsp;   Frequently Asked Questions: Data File Formats</b></FONT> 
			    </TD></TR>
			</TABLE>
            		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
			    <TR><TH HEIGHT=10></TH></TR>
                	    <TR><TD>&nbsp;</TD><TD>
				<UL>
                		<LI><A HREF="#format1">BED format</A>
                		<LI><A HREF="#format2">PSL format</A>
                		<LI><A HREF="#format3">GFF format</A>
                		<LI><A HREF="#format4">GTF format</A>
                		<LI><A HREF="#format5">MAF format</A>
                		<LI><A HREF="#format6">WIG format</A>
				</UL>
				<P>
				<A HREF="index.html">Return to FAQ Table of Contents</A>
				<BR>
            	     	     </TD><TD WIDTH=15></TD></TR>
			</TABLE><BR>
        	     </TD></TR>
		</TABLE>
    	     </TD></TR>
	</TABLE>
	<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="format1"></A>
    <!--outer table is for border purposes-->        
    	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
             <TR><TD>            
        	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    	     <TR><TD>                
  	    		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        	     <TR><TD>                
	        	     <!--cheap trick to get background on row to be continuous-->
	   			<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    		     <TR><TD ALIGN="left"><FONT SIZE="4"><b>&nbsp; BED format</b></FONT></TD> 
		    		     <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            		     </TD></TR>
  	        		</TABLE>                
	        	     </TD></TR>
  	    		</TABLE>                
   	    		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        	    <TR><TH HEIGHT=10></TH></TR>
                	    <TR><TD WIDTH=10>&nbsp;</TD>
		    	    <TD>
	            	    <P>
<P>BED format provides a flexible way to define the data lines that
are displayed in an annotation track. BED lines have three required fields and
nine additional optional fields. The number of fields per line must be consistent
throughout any single set of data in an annotation track.

<P>The first three required BED fields are:</P>
<OL>
        <LI><B>chrom</B> - The name of the chromosome (e.g. chr3, chrY,
        chr2_random) or contig (e.g. ctgY1).
        <LI><B>chromStart</B> - The starting position of the feature in the
        chromosome or contig. The first base in a chromosome is numbered 0.
        <LI><B>chromEnd</B> - The ending position of the feature in the
        chromosome or contig. The <I>chromEnd</I> base is not included in the
        display of the feature. For example, the first 100 bases of a
        chromosome are defined as <I>chromStart=0, chromEnd=100</I>, and span
        the bases numbered 0-99.
</OL>

<P>The 9 additional optional BED fields are:</P>
<OL START=4>
        <LI><B>name</B> - Defines the name of the BED line.  This label is
        displayed to the left of the BED line in the Genome Browser
        window when the track is open to full display mode or directly to the
        left of the item in pack mode.
        <LI><B>score</B> - A score between 0 and 1000. If the track line
        <I>useScore</I> attribute is set to 1 for this annotation data set, the
        <I>score</I> value will determine the level of gray in which
        this feature is displayed (higher numbers = darker gray).
        <LI><B>strand</B> - Defines the strand - either '+' or '-'.
        <LI><B>thickStart</B> - The starting position at which the feature
        is drawn thickly (for example, the start codon in gene
        displays).
        <LI><B>thickEnd</B> - The ending position at which the feature is
        drawn thickly (for example, the stop codon in gene displays).
        <LI><B>reserved</B> - This should always be set to zero.
        <LI><B>blockCount</B> - The number of blocks (exons) in the
        BED line.
        <LI><B>blockSizes</B> - A comma-separated list of the block
        sizes. The number of items in this list should correspond to
        <I>blockCount</I>.
        <LI><B>blockStarts</B> - A comma-separated list of block starts.
        All of the <I>blockStart</I> positions should be calculated relative to
        <I>chromStart</I>. The number of items in
        this list should correspond to <I>blockCount</I>.
</OL>
<P><B><I>Example:</I></B><BR>
Here's an example of an annotation track that uses a complete BED definition:</P>
<PRE><TT>track name=pairedReads description=&quot;Clone Paired Reads&quot; useScore=1
chr22 1000 5000 cloneA 960 + 1000 5000 0 2 567,488, 0,3512
chr22 2000 6000 cloneB 900 - 2000 6000 0 2 433,399, 0,3601</TT></PRE>
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="format2"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; PSL format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
<P>PSL lines represent alignments, and are typically taken from files generated by BLAT
or psLayout. See the
<a
href="http://genome.ucsc.edu/goldenPath/help/hgTracksHelp.html#BLATAlign"
TARGET=_blank>BLAT
documentation</a> for more details. All of the following fields are
required on each data line within a PSL file:
<OL>
<LI><B>matches</B> - Number of bases that match that aren't repeats
<LI><B>misMatches</B> - Number of bases that don't match
<LI><B>repMatches</B> - Number of bases that match but are part of repeats
<LI><B>nCount</B> - Number of 'N' bases
<LI><B>qNumInsert</B> - Number of inserts in query
<LI><B>qBaseInsert</B> - Number of bases inserted in query
<LI><B>tNumInsert</B> - Number of inserts in target
<LI><B>tBaseInsert</B> - Number of bases inserted in target
<LI><B>strand</B> - '+' or '-' for query strand. For translated alignments, second '+'or '-' is for genomic strand
<LI><B>qName</B> - Query sequence name
<LI><B>qSize</B> - Query sequence size
<LI><B>qStart</B> - Alignment start position in query
<LI><B>qEnd</B> - Alignment end position in query
<LI><B>tName</B> - Target sequence name
<LI><B>tSize</B> - Target sequence size
<LI><B>tStart</B> - Alignment start position in target
<LI><B>tEnd</B> - Alignment end position in target
<LI><B>blockCount</B> - Number of blocks in the alignment
<LI><B>blockSizes</B> - Comma-separated list of sizes of each block
<LI><B>qStarts</B> - Comma-separated list of starting positions of each block in query
<LI><B>tStarts</B> - Comma-separated list of starting positions of each block in target
</OL>
<P><B><I>Example:</I></B><BR>
Here is an example of an annotation track in PSL format. Note that line
breaks have been inserted into the PSL lines in this example for
documentation display purposes.
 Click <A HREF="../goldenPath/help/fishBlats.txt">
here</A> for a copy of this example that can be pasted into the browser without editing.
<PRE><TT>track name=fishBlats description=&quot;Fish BLAT&quot; useScore=1
59 9 0 0 1 823 1 96 +- FS_CONTIG_48080_1 1955 171 1062 chr22
    47748585 13073589 13073753 2 48,20,  171,1042,  34674832,34674976,
59 7 0 0 1 55 1 55 +- FS_CONTIG_26780_1 2825 2456 2577 chr22
    47748585 13073626 13073747 2 21,45,  2456,2532,  34674838,34674914,
59 7 0 0 1 55 1 55 -+ FS_CONTIG_26780_1 2825 2455 2676 chr22
    47748585 13073727 13073848 2 45,21,  249,349,  13073727,13073827,
</TT></PRE>

<P>Be aware that the coordinates for a negative strand in a PSL line are handled in a special way. In the
<I>qStart</I> and <I>qEnd</I> fields, the coordinates indicate the position where the query matches from
the point of view of the forward strand, even when the match is on the reverse strand.
However, in the <I>qStarts</I> list, the coordinates are reversed.</P>
<P>
<B><I>Example:</I></B><BR>
Here is a 30-mer containing 2 blocks that align on the minus strand and
2 blocks that align on the plus strand (this sometimes can happen in response to
assembly errors): <BR><BR>
<TABLE CELLPADDING=0 CELLSPACING=0 MARGINS=0 BORDER=0>
<TR><TD><PRE>0         1         2         3 tens position in query   </TD></TR>
<TR><TD><PRE>0123456789012345678901234567890 ones position in query   </TD></TR>
<TR><TD><PRE>            ++++          +++++ plus strand alignment on query   </TD></TR>
<TR><TD><PRE>    --------    ----------      minus strand alignment on query   </TD></TR>
<TR><TD></TD></TD>
<TR><TD><PRE></TD></TD>
<TR><TD><PRE>Plus strand:   </TD></TR>
<TR><TD><PRE>     qStart=12 </TD></TR>
<TR><TD><PRE>     qEnd=31 </TD></TR>
<TR><TD><PRE>     blockSizes=4,5 </TD></TR>
<TR><TD><PRE>     qStarts=12,26   </TD></TR>
<TR><TD><PRE>                      </TD></TR>
<TR><TD><PRE>Minus strand:   </TD></TR>
<TR><TD><PRE>     qStart=4 </TD></TR>
<TR><TD><PRE>     qEnd=26 </TD></TR>
<TR><TD><PRE>     blockSizes=10,8 </TD></TR>
<TR><TD><PRE>     qStarts=5,19    </TD></TR>
</TABLE><BR>
Essentially, the minus strand <I>blockSizes</I> and <I>qStarts</I> are
what you would get if you reverse-complemented the query.
However, the <I>qStart</I> and <I>qEnd</I> are not reversed. To convert one to the other:
<PRE>
     qStart = qSize - revQEnd
     qEnd = qSize - revQStart
</PRE>
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="format3"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; GFF format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
<P>GFF (General Feature Format) lines are based on the GFF standard file format.  GFF
lines have nine required fields that <I>must</I> be tab-separated. If the fields are
separated by spaces instead of tabs, the track will not display correctly. For more
information on GFF format, refer to <A
HREF="http://www.sanger.ac.uk/Software/formats/GFF"
TARGET=_blank>http://www.sanger.ac.uk/Software/formats/GFF</A>. </P>
<P>
Here is a brief description of the GFF fields:</P>

<OL>
        <LI><B>seqname</B> - The name of the sequence.  Must be a chromosome or a contig.
        <LI><B>source</B> - The program that generated this feature.
        <LI><B>feature</B> - The name of this type of feature. Some examples of
        standard feature types are "CDS", "start_codon", "stop_codon", and
        "exon".
        <LI><B>start</B> - The starting position of the feature in the sequence.  The first base is numbered 1.
        <LI><B>end</B> - The ending position of the feature (inclusive).
        <LI><B>score</B> - A score between 0 and 1000. If the track line
        <I>useScore</I> attribute is set to 1 for this annotation data set, the
        <I>score</I> value will determine the level of gray in which
        this feature is displayed (higher numbers = darker gray). If there is no
        score value, enter ".".        <LI><B>strand</B> - Valid entries include '+', '-', or '.' (for don't know/don't care).        <LI><B>frame</B> - If the feature is a coding exon, <I>frame</I> should be a number between 0-2 that represents the reading frame of the
        first base.  If the feature is not a coding exon, the value should be '.'.        <LI><B>group</B> - All lines with the same group are linked together into a single item.
</OL>
<P><B><I>Example:</I></B><BR>
Here's an example of a GFF-based track.
Click <A HREF ="../goldenPath/help/regulatory.txt">here</A> for a copy of this example that can be 
pasted into the browser without editing. NOTE: Paste operations on some 
operating systems will replace tabs with spaces, which will result in an error 
when the GFF track is uploaded. You can circumvent this problem by pasting the 
URL of the above example
(http://genome.ucsc.edu/goldenPath/help/regulatory.txt) instead of the text 
itselfinto the custom annotation track text box.

<PRE><TT>track name=regulatory description=&quot;TeleGene(tm) Regulatory Regions&quot;
chr22  TeleGene enhancer  1000000  1001000  500 +  .  touch1
chr22  TeleGene promoter  1010000  1010100  900 +  .  touch1
chr22  TeleGene promoter  1020000  1020000  800 -  .  touch2
</TT></PRE>

			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="format4"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; GTF format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
<P>GTF (Gene Transfer Format) is a refinement to GFF that tightens the specification.The first eight GTF fields are the same as GFF. The <I>group</I> field has been
expanded into an <I>attribute</I> field that includes a list ofsemicolon-separated attribute/value pairs. For more information on this format,see <A HREF="http://genes.cs.wustl.edu/GTF2.html" TARGET=_blank>http://genes.cs.wustl.edu/GTF2.html</A>.

<P>Some examples of entries for the <I>attribute</I> field include:
<UL>
<LI><B>gene_id <I>value</I></B> - A globally unique identifier for the genomic
source of the sequence.
<LI><B>transcript_id <I>value</I></B> - A globally unique identifier for the
predicted transcript.</UL>
<P><B><I>Example:</I></B><BR>Here is an example of the ninth field in a GTF data line:
<P>
<tt>gene_id Em:U62317.C22.6.mRNA; transcript_id Em:U62317.C22.6.mRNA;
exon_number 1</tt>

<P>The Genome Browser groups together GTF lines that have the same
<I>transcript_id</I> value.  It only looks at <I>EXON</I> and <I>CDS</I>type features.</P>

			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="format5"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; MAF format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
                    The multiple alignment format stores a series of
                    multiple alignments in a format that is easy to
                    parse and relatively easy to read.  This format
                    stores multiple alignments at the DNA level
                    between entire genomes. The previously existing formats 
		    are suitable for multiple alignments of single proteins or 
		    regions of DNA without rearrangements, and would require 
		    considerable extension to cope with genomic issues such as
                    forward and reverse strand directions, multiple pieces 
		    to the alignment, and so forth.
                    <P>
		    <B>General Structure</B>
		    <P> The <em>.maf</em> format is
                    line-oriented. Each multiple alignment ends with a
                    blank line.  Each sequence in an alignment is on a single 
		    line, which can get quite long, but there is no length limit.
		    Words in a line are delimited by any white space. Lines
                    starting with # are considered to be comments.
                    Lines starting with ## can be ignored by most
                    programs, but contain meta-data of one form or
                    another.
                    <P>
                    The file is divided into paragraphs that terminate
                    in a blank line.  Within a paragraph, the first
                    word of a line indicates its type.  Each multiple
                    alignment is in a separate paragraph that begins
                    with an &quot;a&quot; line and contains an &quot;s&quot; line 
		    for each sequence in the multiple alignment.  For now,
                    parsers should ignore other types of paragraphs
                    and other types of lines within an alignment
                    paragraph.
                    <P>
		    <B>The Header Line</B>
		    <P>
                    The first line of a <em>.maf</em> file begins with ##maf. 
		    This word is followed by white-space-separated
                    variable=value pairs.  There should be <em>no</em> white
                    space surrounding the &quot;=&quot;.  
		    <FONT color="#0000FF"><PRE><TT> ##maf version=1 scoring=tba.v8 </TT></PRE></FONT>
		    The currently defined variables are:
                    <UL>
                    <LI><B>version</B> - Required.  Currently set to one.  </LI>
                    <LI><B>scoring</B> - Optional. A name for the scoring
                                   scheme used for the alignments.
                                   The current scoring schemes are:
                        <UL>
			<LI><em>bit</em> - roughly corresponds to blast bit
                                   values (roughly 2 points per
                                   aligning base minus penalties for
                                   mismatches and inserts).  </LI>
                        <LI><em>blastz</em> - blastz scoring scheme -- roughly
                                   100 points per aligning base.</LI>
                        <LI><em>probability</em> - some score normalized
                                   between 0 and 1.  </LI>
                        </UL>
                    <LI><B>program</B> - Optional. Name of the program
                                   generating the alignment.</LI>
                    </UL>
                    Undefined variables are ignored by the parser.
                    <P>
		    <B>The Alignments Parameter Line</B>
		    <P>
                    The second line displays the parameters that were
                    used to run the alignment program.
		    <FONT color="#0000FF"><PRE><TT> # tba.v8 (((human chimp) baboon) (mouse rat))</TT></PRE></FONT>
                    <P>
		    <B>Alignment Block Lines</B> (lines starting with 'a' -- parameters for a new alignment block</B>
		    <P>                    
		    <FONT color="#0000FF"><PRE><TT> a score=23262.0</TT></PRE></FONT>
                    Each alignment begins with an 'a' line that set 
		    variables for the entire alignment block.  The 'a'
                    is followed by name=value pairs.  There are no
                    required name=value pairs.  The currently defined
                    variables are: 
		    <UL>
                    <LI> score -- Optional. Floating point score. If
                                 this is present, it is good practice
                                 to also define scoring in the first line.  </LI>

                    <LI> pass --  Optional.  Positive integer value.
                                 For programs that do multiple pass
                                 alignments such as blastz, this shows
                                 which pass this alignment came from.
                                 Typically, pass 1 will find the
                                 strongest alignments genome-wide, and
                                 pass 2 will find weaker alignments
                                 between two first-pass alignments.</LI>
                    </UL>
                    <P>
		    <B>Lines starting with 's' -- a sequence within an alignment block</B>
		    <P>
		    <FONT color="#0000FF">
<PRE><TT> 
 s hg16.chr7    27707221 13 + 158545518 gcagctgaaaaca
 s panTro1.chr6 28869787 13 + 161576975 gcagctgaaaaca
 s baboon         249182 13 +   4622798 gcagctgaaaaca
 s mm4.chr6     53310102 13 + 151104725 ACAGCTGAAAATA
</TT></PRE></FONT>
                    The 's' lines together with the 'a' lines define a
                    multiple alignment.  The 's' lines have the
                    following fields which are defined by position
                    rather than name=value pairs.
		    <UL>
                    <LI> src -- The name of one of the source sequences
                               	for the alignment.  For sequences that
                               	are resident in a browser assembly, the
			 	form 'database.chromosome' allows
                               	automatic creation of links to other 
				assemblies.  Non-browser sequences are
                               	typically reference by the species name
                               	alone.</LI>
                    <LI> start -- The start of the aligning region in the 
				source sequence. This is a
                               	zero-based number. If the strand field 
				is '-' then this is the start
                               	relative to the reverse-complemented
                               	source sequence.  </LI>
                    <LI> size -- The size of the aligning region in the
                               	source sequence.  This number is equal
                                to the number of non-dash characters in
                               	the alignment text field below.  </LI>          
		    <LI> strand -- Either '+' or '-'.  If '-', then the           
                               	alignment is to the
                               	reverse-complemented source.  </LI>
                    <LI> srcSize -- The size of the entire source sequence, 
				not just the parts involved in the alignment.</LI>
		    <LI> text -- The nucleotides (or amino acids) in
				the alignment and any insertions
                               (dashes) as well.  </LI> 
                    </UL>          
                                   
                    <P>
		    <B>A Simple Example</B>
		    <P>                     
		    Here is a simple example of a three alignment
                    blocks derived from five starting sequences.
		    Repeats are shown as lowercase, and each block may
                    have a subset of the input sequences.  All
		    sequence columns and rows must contain at least one nucleotide
                    (no full columns or rows of insertions).

		    <FONT color="#0000FF">
<PRE><TT> 
 ##maf version=1 scoring=tba.v8 
 # tba.v8 (((human chimp) baboon) (mouse rat)) 
 # multiz.v7
 # maf_project.v5 _tba_right.maf3 mouse _tba_C
 # single_cov2.v4 single_cov2 /dev/stdin
                    
 a score=23262.0     
 s hg16.chr7    27578828 38 + 158545518 AAA-GGGAATGTTAACCAAATGA---ATTGTCTCTTACGGTG
 s panTro1.chr6 28741140 38 + 161576975 AAA-GGGAATGTTAACCAAATGA---ATTGTCTCTTACGGTG
 s baboon         116834 38 +   4622798 AAA-GGGAATGTTAACCAAATGA---GTTGTCTCTTATGGTG
 s mm4.chr6     53215344 38 + 151104725 -AATGGGAATGTTAAGCAAACGA---ATTGTCTCTCAGTGTG
 s rn3.chr4     81344243 40 + 187371129 -AA-GGGGATGCTAAGCCAATGAGTTGTTGTCTCTCAATGTG
                    
 a score=5062.0                    
 s hg16.chr7    27699739 6 + 158545518 TAAAGA
 s panTro1.chr6 28862317 6 + 161576975 TAAAGA
 s baboon         241163 6 +   4622798 TAAAGA 
 s mm4.chr6     53303881 6 + 151104725 TAAAGA
 s rn3.chr4     81444246 6 + 187371129 taagga

 a score=6636.0
 s hg16.chr7    27707221 13 + 158545518 gcagctgaaaaca
 s panTro1.chr6 28869787 13 + 161576975 gcagctgaaaaca
 s baboon         249182 13 +   4622798 gcagctgaaaaca
 s mm4.chr6     53310102 13 + 151104725 ACAGCTGAAAATA
</TT></PRE></FONT>
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="format6"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; WIG format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
<P>
Wiggle format (WIG) allows the display of continuous-valued data in a track
format. Click <A HREF="../goldenPath/help/wiggle.html">here</A> for more 
information.</P>
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>
    </TD><TD WIDTH=15></TD></TR>
</TABLE>
</BODY></HTML>
