<P>
The October 2004 <em>Drosophila erecta</em> genome assembly 
was produced by 
<A HREF="http://www.agencourt.org/" TARGET=_blank>Agencourt</A> and can be 
downloaded 
<A HREF="http://rana.lbl.gov/drosophila/assemblies/dere_agencourt_arachne_28oct04.tar.gz" TARGET=_BLANK>here</A>.  
To see a list of the assembly's 18,915 contigs ranked by size, click 
<A HREF="/cgi-bin/hgTracks?db=droEre1&chromInfoPage=">here</A>.
</P>

<H3>Sample position queries</H3>
<P>
A genome position can be specified by a contig coordinate range, the accession 
number of an mRNA or RefSeq entry, a gene name, or keywords from 
the GenBank description of an mRNA. The following list shows 
examples of valid position queries for the <em>D. erecta</em> genome. 
Note that some position queries (e.g. "huntington") may return matches to the 
mRNA records of other species. In these cases, the mRNAs are mapped to their 
homologs in <em>D. erecta</em>. 
See the
<A HREF="/goldenPath/help/hgTracksHelp.html" TARGET=_blank>User's Guide</A>
for more information.
</P>
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
       	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><BR></TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>contig_985</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays all of contig_985</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>contig_985:1-50,000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays first 50,000 bases of contig_985</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>AF025805</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of genome aligning to mRNA with GenBank accession AF025805</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>dock</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of genome aligning to gene with identifier dock (a <em>D. melanogaster</em> gene)</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>NM_057677</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of genome aligning to RefSeq accession with identifier NM_057677 (a <em>D. melanogaster</em> RefSeq entry)</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>kruppel zinc finger</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>huntington</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists candidate genes associated with Huntington's disease</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>zahler</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Zahler</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Zeng,L.W.</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author L.W. Zeng</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Zeng LW</em>, internally it uses 
			the format <em>Zeng,L.W.</em></TD>
    </TR>
</TABLE>
</P>
<P>
<HR>
<H3>Assembly details</H3>
Agencourt produced the 28 October 2004 assembly using the 
<A HREF="" TARGET=_BLANK>Arachne</A> assembler.  
The assembly contains 18,915 contigs ranging in size from 335 bases to 
196,364 bases, with a mean size of mean 7061.9 and median of 1698.  
</P>
<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser 
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/droEre1/">FTP server</A> or the 
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#droEre">Downloads</A> page. 
The droEre1 annotation tracks were generated by UCSC and collaborators worldwide. 
See the <A HREF="/goldenPath/credits.html#droEre_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
