<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>Human Genome Working Draft: Best Links</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">

<TABLE BORDER=0 WIDTH=100%><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg"><A NAME="TOP"></A></TH></TR>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
   <CENTER>
	<TABLE BGCOLOR="2636D1" WIDTH=100% CELLSPACING=0 CELLPADDING=0 BORDERCOLOR="000000" BORDER=1 HEIGHT="22"><TR><TD>
		<TABLE WIDTH=431 CELLSPACING=4 CELLPADDING=0 BORDER=0><TR>
		<TD WIDTH=1></TD>
		<TD WIDTH=75>
			<A HREF="index.html" onMouseOver="HOME.src='images/hl_home.jpg'" onMouseOut="HOME.src='images/h_home.jpg'">
			<IMG SRC="images/h_home.jpg" ALIGN="absmiddle" BORDER="0" NAME="HOME" ALT="Home" ></A></TD>

		<TD WIDTH=155>
			<A HREF="goldenPath/hgTracks.html" onMouseOver="BROWSER.src='images/hl_browser.jpg'" onMouseOut="BROWSER.src='images/h_browser.jpg'">
			<IMG SRC="images/h_browser.jpg" ALIGN="absmiddle" BORDER="0" NAME="BROWSER" ALT="&nbsp;&nbsp; Genome Browser"></A></TD>
			
		<TD WIDTH=100>
			<A HREF="../cgi-bin/hgBlat?command=start" onMouseOver="SEARCH.src='images/hl_blat.jpg'" onMouseOut="SEARCH.src='images/h_blat.jpg'">
			<IMG SRC="images/h_blat.jpg" ALIGN="absmiddle" BORDER="0" NAME="SEARCH" ALT="&nbsp;&nbsp; BLAT Search"></A></TD>
			
		<TD WIDTH=100>
			<A HREF="FAQ.html" onMouseOver="GUIDE.src='images/hl_faq.jpg'" onMouseOut="GUIDE.src='images/h_faq.jpg'">
			<IMG SRC="images/h_faq.jpg" ALIGN="absmiddle" BORDER="0" NAME="GUIDE" ALT="&nbsp;&nbsp; Help"></A></TD>
		</TR></TABLE>
	 </TR></TABLE>
	 </CENTER>
    </TD>
</TR>

<a name="000">&nbsp;</a>

<TR>
 <!--Content Tables-->  
  <TD WIDTH="100%" CELLPADDING=10 ROWSPAN=2 VALIGN="top">
  	<IMG SRC="images/DOT.gif" HEIGHT=7 WIDTH=1 ALT=""><BR>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><B> &nbsp;UCSC Genome Browser Best Links </B></FONT> 
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
	Links validated on 23 Aug 01; please <A HREF="mailto:learithe@cats.ucsc.edu">report</A> problems and suggestions
	</TD></TR><TR><TD COLSPAN=2>
	<TABLE CELLPADDING=0><TH HEIGHT=5></TH><TR><TD><UL><B>
  		<LI><A CLASS="toc" HREF="#166"> Blast, Blat, and Align Sequences </A>
  		<LI><A CLASS="toc" HREF="#133"> Gene Prediction and Feature Finding</A>
  		<LI><A CLASS="toc" HREF="#122"> Genome Browsers and Complete Genomes </A>
  		<LI><A CLASS="toc" HREF="#188"> Journals, Alerts, and General Internet </A>
	</TD><TD VALIGN="top"><UL><B>
  		<LI><A CLASS="toc" HREF="#111"> Main Databases </A>
  		<LI><A CLASS="toc" HREF="#155"> Family Specific </A>
  		<LI><A CLASS="toc" HREF="#177"> Tool Collections </A>
  		<LI><A CLASS="toc" HREF="#144"> Protein Tools and Browsers </A>
	</TD><TD VALIGN="top"><UL><B>
 	</TD></TR></TABLE>
</TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	
	
<a name="111"><BR></a>
<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4" ><b>&nbsp; Main Databases</b></FONT></TD>
			<TD ALIGN="right"><A HREF="#000"><IMG SRC="images/index.gif" ALT="" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD><UL>
<li><a href="http://www.ncbi.nlm.nih.gov:80/entrez/query.fcgi?CMD=Limits&DB=PubMed">PubMed</a> -- abstracts for 11 million biomedical articles
<li><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=Nucleotide">Entrez</a> -- gateway to nucleotide and protein sequences
<li><a href="http://www.ncbi.nlm.nih.gov/LocusLink/">LocusLink</A> -- curated sequence and gene descriptions
<li><a href="http://www.rcsb.org/">PDB</A> -- 3D protein database at RCSB
<li><a href="http://www.expasy.ch/cgi-bin/sprot-search-ful">SwissProt</A> -- keyword search annotated protein collection
<li><A HREF="http://www.expasy.org/srs5/">SRS</A> -- query Swiss sequence retrieval system
<li><A HREF="http://srs6.ebi.ac.uk/">EBI SRS</A> -- query session at EBI sequence retrieval center 
<li><a href="http://www.ebi.ac.uk/Databases/index.html">EMBL</A> -- gateway to EMBL sequence and structure databases
<li><a href="http://pbil.univ-lyon1.fr/databases/hovergen.html">Hovergen</A> -- homologous vertebrate genes database
<li><A HREF="http://zearth.kazusa.or.jp/huge/">HUGE</A> -- unidentified human proteins in KIAA series
<li><A HREF="http://www.mips.biochem.mpg.de/">MIPS</A> -- gateway to protein information center in Munich
<li><A HREF="http://www.gdb.org/Dan/proteins/owl.html">OWL</A> -- non-redundant protein sequence database
<li><A HREF="http://www-nbrf.georgetown.edu/pir/searchdb.html">PIR</A> -- gateway to PIR protein database and query tools
<li><A HREF="http://www.ncbi.nlm.nih.gov/UniGene/Hs_DATA/ChromLists/Chr12.html">ChromList</A> -- lists UniGene clusters by chromosome
<li><A HREF="http://www.expasy.org/cgi-bin/lists?humchr12.txt">ChromProt</A> -- lists human proteins by chromosome
<li><A HREF="http://www.ncbi.nlm.nih.gov/Taxonomy/taxonomyhome.html">Taxonomy</A> -- taxonomic classes at NCBI
<li><A HREF="http://phylogeny.arizona.edu/tree/phylogeny.html">TreeLife</A> -- tree of life taxonomy project
<br><br>
<li><a href="http://www.gene.ucl.ac.uk/nomenclature/">Hugo</A> -- recommended gene symbol nomenclature
<li><a href="http://www3.ncbi.nlm.nih.gov/omim/">OMIM 1</A>+
	<A HREF="http://www.ncbi.nlm.nih.gov:80/entrez/query.fcgi?CMD=Limits&DB=OMIM">2</A> -- simple and complex searches at OMIM
<li><a href="http://research.marshfieldclinic.org/genetics/">Marshfield</A> -- genetic and physical maps at Marshfield
<li><a href="http://archive.uwcm.ac.uk/uwcm/mg/hgmd0.html">HGMD Mut</A> -- human gene mutation database at Cardiff
<li><A HREF="http://www.ncbi.nlm.nih.gov:80/entrez/Omim/allresources.html#LocusSpecific">LocMutations</A> -- locus-specific mutation databases
<li><A HREF="http://www.gen.emory.edu/mitomap.html">Mitomap</A> -- human mitochondrial genome database
<li><A HREF="http://www.ncbi.nlm.nih.gov/cgi-bin/Entrez/mzorder?tax=40674&name=Mammalia">Mito</A> -- mitochondrial genomes and alignment
<li><A HREF="http://www.neuro.wustl.edu/neuromuscular/mitosyn.html">MitoWustl</A> -- mitochondrial genes, mutations and disorders
<li><A HREF="http://www.mips.biochem.mpg.de/proj/medgen/mitop">MitoP</A> -- mitochondrial genes, proteins and diseases at Munich
<li><A HREF="http://www-lecb.ncifcrf.gov/mitoDat/">MitoDat</A> -- nuclear genes specifying mitochondrial biogenesis
<br><br>
<li><a href="http://bioinfo.weizmann.ac.il/udb">UDB</A> -- integrated human genome map
<li><A HREF="http://www.ncbi.nlm.nih.gov/UniGene/query.cgi?TEXT=&ORG=Hs&OR">UniGene</A> -- query UniGene
<li><A HREF="http://www.mips.biochem.mpg.de/proj/human/">UniGene HIB</A> -- annotated Unigene clusters and blast service
<li><a href="http://www.ncbi.nlm.nih.gov/genome/seq/chr.cgi?CHR=20&SRT=ppos&MIN=0&ORG=Hs">Contigs</A> -- search NT contigs, dbSTS, GenMap99
<li><a href="http://www-genome.wi.mit.edu/cgi-bin/contig/phys_map">Marker</A> -- search for marker, YACs, contigs
<li><A HREF="http://snp.cshl.org">SNP</A> -- single nucleotide polymorphisms at SNP Consortium
<li><A HREF="http://www.ncbi.nlm.nih.gov/SNP">dbSNP</A> -- search for single nucleotide polymorphisms at NCBI
<li><a href="http://bodymap.ims.u-tokyo.ac.jp/">BodyMap</A> -- database of human and mouse gene expression
<br><br>
<li><a href="http://uther.otago.ac.nz/Transterm.html">TransTerm</A> -- sequence context of stop and start codons
<li><a href="http://www.ch.embnet.org/software/ESTScan.html">ESTScan</A> -- detects coding regions despite frameshifts and low quality
<li><a href="http://www.cifn.unam.mx/regulondb/">RegulonDB</A> -- transcription regulation database for various organisms
<li><A HREF="http://rebase.neb.com/rebase/rebase.html">Rebase</A> -- restriction enzyme cleavage sites
<li><A HREF="http://genomic.sanger.ac.uk/spldb/SpliceDB.html">SpliceDB</A> -- canonical and non-canonical splice sites in mammals
<li><a href="http://www.biology.ucsd.edu/~ipaulsen/transport/">MemTrans</A> -- genomic comparisons of membrane transport systems
<li><a href="http://www.Geneimprint.com ">Geneimprint</A> -- imprinted gene database
<li><a href="http://www.kazusa.or.jp/codon/">Codon</A> -- codon usage database
</UL></TD></TR></TABLE>
	</TD></TR></TABLE>
	
	
<a name="122"><BR></a>
<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4" ><b>&nbsp; Genome Browsers & "Complete" Genomes </b></FONT></TD>
			<TD ALIGN="right"><A HREF="#000"><IMG SRC="images/index.gif" ALT="" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD><UL>
<li><a href="http://genome.cse.ucsc.edu/">Genome Browser</A> -- human genome assembly and track browser at UCSC
<li><a href="http://www.ensembl.org/perl/mapview?chr=20">Ensembl</A> -- human genome and disease gene browser at EBI/Sanger	
<li><A HREF="http://www.ncbi.nlm.nih.gov/cgi-bin/Entrez/hum_srch?chr=hum_chr.inf&query">Map Viewer</A> -- human genome assembly and map viewer at NCBI
<li><A HREF="http://genome.cse.ucsc.edu/goldenPath/hgText.html">Genome</A> -- query human genome and annotation database tables
<li><A HREF="http://genome.cse.ucsc.edu/goldenPath/01apr2001/bigZips">Download</A> -- download human genome and install mirror
<li><A HREF="http://iubio.bio.indiana.edu:8089/man/">euGenes</A> -- five species map browser and 37,049 human gene reports
<li><a href="http://sichuan.lbl.gov/ids-bin/VistaInput">Vista</A> -- visualization tool for long genomic alignments
<li><A HREF="http://www.tigr.org/tigr-scripts/CMR2/CMRHomePage.spl">TIGR CMR</A> -- comprehensive microbial resource and genome browser
<br><br>
<li><a href="http://www.ncbi.nlm.nih.gov/Genomes/index.html">NCBI</A> -- human and animal genome projects at NCBI
<li><a href="http://www.sanger.ac.uk/HGP/links.shtml">Sanger</A> -- human genome project at Sanger Centre
<li><a href="http://www.informatics.jax.org/">Jax</A> -- mouse genomics at Jackson Laboratory
<li><A HREF="http://ratmap.gen.gu.se/">RatMap</A> -- rat genome database at Goteborg			
<li><a href="http://zfish.uoregon.edu/">Zfin</A> -- zebrafish information network and database project
<li><a href="http://www.genoscope.cns.fr/externe/tetraodon/">Exofish</A> -- genome analysis of Tetraodon nigroviridis 
<li><a href="http://flybase.bio.indiana.edu/">Flybase</A> -- database of drosophila genomics at Indiana 
<li><A HREF="http://www.fruitfly.org">BDGP</A> -- fruit fly genomics center at Berkeley
<li><A HREF="http://www.wormbase.org/">WormBase</A> -- nematode mapping, sequencing and phenotypic repository
<li><a href="http://elegans.swmed.edu/genome.shtml">ACeDB</A> -- genetics and genomics resource launcher
<li><a href="http://genome-www.stanford.edu/Saccharomyces/">SGD</A> -- yeast genome database	
<li><A HREF="http://arabidopsis.org">TAIR</A> -- the arabidopsis information resource and tool center		
<li><a href="http://www.tigr.org/tdb/mdb/mdbcomplete.html">TIGR db</A> -- bacterial genomes at TIGR
<li><A HREF="http://igweb.integratedgenomics.com/GOLD/">GOLD</A> -- monitors genome projects world-wide
<li><a href="http://www.ebi.ac.uk/genomes/mot/">GenTotals</A> -- monitoring tables for genome progress
</UL></TD></TR></TABLE>
	</TD></TR></TABLE>

	
<a name="133"><BR></a>
<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4" ><b>&nbsp; Gene Prediction and Feature Finding </b></FONT></TD>
			<TD ALIGN="right"><A HREF="#000"><IMG SRC="images/index.gif" ALT="" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD>
<UL>
<li><a href="http://genome.dkfz-heidelberg.de/cgi-bin/GENSCAN/genscan.cgi">GenscanD</A>+
	<a href="http://genes.mit.edu/GENSCAN.html">M</A> -- identification of complete gene structures in genomic DNA
<li><A HREF="http://rgd.mcw.edu/">RGD</A> -- comparative mapping tool for rat-mouse-humany synteny
<li><a href="http://genomic.sanger.ac.uk/gf/gf.shtml">FGENEH</a> -- splice sites, coding exons, gene models, promotor and polyA
<li><A HREF="http://genes.cs.wustl.edu/">TwinScan</A> -- gene prediction for eukaryotic genomic sequences
<li><A HREF="http://www.itba.mi.cnr.it/webgene/">WebGene</A> -- ten tools for analysis of protein-coding gene structure 
<li><A HREF="http://plover.lcs.mit.edu/cgi-bin/rosetta.cgi">Rosetta</A> -- predicts human coding exons based on homology 
<li><A HREF="http://grail.lsd.ornl.gov/grailexp/">GrailEXP</A> -- exons, genes, promoters, polyA, CpG islands at ORNL 
<li><A HREF="http://www.sanger.ac.uk/Software/Wise2/genewiseform.shtml">GeneWise2</A> -- compares protein to genomic with introns and frameshifts 
<li><A HREF="http://www.fruitfly.org/seq_tools/genie.html">Genie</A> -- finds multi-exon genes only, trained on human 
<li><A HREF="http://genome.cs.mtu.edu/aat.html">AAT</A> -- analysis and annotation tool for finding genes 
<li><a href="http://www.cbs.dtu.dk/services/HMMgene/">HMMgene</a> -- prediction of vertebrate and nematode genes 
<li><a href="http://www.cbs.dtu.dk/services/NetGene2/">NetGene2</a> -- neural network predictions of splice sites 
<br><br>
<li><A HREF="http://www.ebi.ac.uk/genemark/">GeneMark</A> -- gene prediction server at EBI 
<li><a href="http://www1.imim.es/geneid.html">GeneID</a> -- gene prediction of selected signals and exons 
<li><A HREF="http://gobi.lbl.gov/~steveh/rnagene/">RNAGenie</A> -- locates RNA genes from secondary structures 
<li><a href="http://www.tbi.univie.ac.at/~ivo/RNA/ ">Vienna RNA</A> -- seven RNA analysis tools 
<br><br>
<li> <a href="http://dot.imgen.bcm.tmc.edu:9331/seq-util/seq-util.html">Reverse Comp</A> -- reverse complement of DNA sequence 
<li> <A HREF="http://promoter.informatik.uni-erlangen.de/">McPromoter</A> -- finds eukaryotic RNA polymerase II transcription starts 
<li> <A HREF="http://bimas.dcrt.nih.gov/molbio/proscan/index.html">PromoterScan</A> -- predicts promoters using homologies with Pol II promoters	 	
<li> <A HREF="http://www.ncbi.nlm.nih.gov/gorf/gorf.html">ORF Finder</A> -- open reading frame finder for ORFs of cutoff size 
<li> <a href="http://alces.med.umn.edu/cuse.html">SeqTrans</A> -- identifies open reading frames in finished genomic 
<li> <A HREF="http://www.ebi.ac.uk/cpg/">Emboss</A> -- predict CpG islands and isochores 
<li> <A HREF="http://141.80.80.48/splice/">AltSplice</A> -- database of alternate splicing in Ests and disease genes 
<li> <a href="http://www.firstmarket.com/firstmarket/cutter/cut2.html">Webcutter</A> -- finds restriction endonuclease sites 
<li> <A HREF="http://ir2lcb.cnrs-mrs.fr/d_fsed/fsed.html">FSED</A> -- frameshift error detection in new sequences  
<li> <A HREF="http://www.scms.rgu.ac.uk/staff/jb/submit.html">CheckSeq</A> -- uses homology to find coding regions and sequence errors  
<li> <A HREF="http://www.bork.embl-heidelberg.de/ERR_WISE/">ERR_WISE</A> -- detection of frameshift sequencing errors  
<li> <A HREF="http://www.bork.embl-heidelberg.de/SEQERR/">SEQERR</A> -- detection of frameshift error in coding regions  
<li> <A HREF="http://www.ebi.ac.uk/cgi-bin/mutations/check.cgi">MutCheck</A> -- transcription and translation effect of mutations  
<li> <a href="http://blocks.fhcrc.org/~pauline/SIFT.html">Sift</A> -- predicts tolerated and untolerated protein substitutions  
<br><br>
<li> <a href="http://transfac.gbf.de/TRANSFAC/index.html">TransFac</A> -- transcription factor database and tool collection 
<li> <A HREF="http://transfac.gbf.de/cgi-bin/matSearch/matsearch.pl">MatInspector</A> -- searches for transcription factor binding sites  
<li> <A HREF="http://www.expasy.org/ftp/databases/sp_tr_nrdb/">VarSplice</A> -- collection of annotated splice variants  
<li> <A HREF="http://193.175.244.148/">TransPath</A> -- pathways in regulation of transcription factors  
<li> <A HREF="http://bioweb.pasteur.fr/seqanal/interfaces/tfscan.html">TfScan</A> -- scans DNA sequences for transcription factors  
<br><br>
<li> <a href="http://ftp.genome.washington.edu/cgi-bin/RepeatMasker">RepeatMasker</A> -- finds retroposons and repeats  	
<li> <a href="http://www.girinst.org/Censor_Server-Data_Entry_Forms.html">Censor</A> -- finds repeated elements, Repbase updates, maps  
<li> <a href="http://c3.biomath.mssm.edu/trf.html">Tandem</A> -- tandem repeat finder in DNA  
<li> <a href="http://alces.med.umn.edu/rawdot.html">DotPlot</A> -- dot plot for internal DNA sequence repeats  
<li> <a href="http://www.biophys.uni-duesseldorf.de/local/TINOCO/tinoco.html">Tinoco</A> -- suggests secondary structure in RNA or DNA  
<li> <A HREF="http://www.embl-heidelberg.de/~andrade/papers/rep/search.html">REP</A> -- searches proteins for internal repeats  
<li> <A HREF="http://www.ebi.ac.uk/Radar/">Radar</A> -- detects and aligns protein repeats  
<li> <A HREF="http://mathbio.nimr.mrc.ac.uk/~rgeorge/repro/">Repro</A> -- recognises repeats in a single protein query 
</UL></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<a name="144"><BR></a>
<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4" ><b>&nbsp; Protein Tools and Browserse </b></FONT></TD>
			<TD ALIGN="right"><A HREF="#000"><IMG SRC="images/index.gif" ALT="" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD><UL>
<li><a href="http://www.expasy.ch/cgi-bin/sprot-search-ful">SwissTools</A> -- large collection of protein tools and databases
<li><a href="http://www.expasy.ch/tools/dna.html">Translate</A> -- translates DNA into 6 frames 
<li><a href="http://www.ebi.ac.uk/cgi-bin/mutations/revtransl.cgi">ReverseTrans 1</A>+
	<a href="http://www.entelechon.com/eng/backtranslation.html">2</A> -- reverse-translates protein into DNA
<li><a href="http://www.ebi.ac.uk/saps/">SAPS</A> -- statistical analysis of protein sequences
<li><a href="http://www.ebi.ac.uk/emboss/pepinfo/">Emboss</A> -- peptide information and statistics
<br><br>
<li><a href="http://www.cbs.dtu.dk/services/SignalP/">Signalp</A> -- predicts signal peptide cleavage sites
<li><a href="http://psort.nibb.ac.jp/form.html">Psort</A> -- protein sorting signals
<li><a href="http://www.ch.embnet.org/software/TMPRED_form.html">TMpred</A> -- predicts transmembrane segments and orientation
<li><a href="http://www.cbs.dtu.dk/services/TMHMM/">TMHMM</A> -- predicts transmembrane helices
<li><a href="http://www.sbc.su.se/~erikw/toppred2/">TopPred2</A> -- topological prediction of membrane proteins
<li><a href="http://www.mbb.ki.se/tmap/index.html">	TMAP</A> -- predicts membrane proteins from multiple alignment		
<li><a href="http://www.ch.embnet.org/software/COILS_form.html">COILS 1</A>+
	<a href="http://nightingale.lcs.mit.edu/cgi-bin/score">2</A> -- prediction of coiled-coil regions in proteins
<li><a href="http://www.embl-heidelberg.de/argos/predator/predator_info.html">PREDATOR</A> -- secondary structure from multiple sequences
<li><a href="http://www.bork.embl-heidelberg.de/SSCP/">SSCP</A> -- predicts helix, strand and coil from composition
<li><a href="http://www.embl-heidelberg.de/argos/stride/stride_info.html">STRIDE</A> -- secondary structure from atomic coordinates
<li><a href="http://dot.imgen.bcm.tmc.edu:9331/pssprediction/pssp.html">BCM SS</A> -- secondary structure by four methods
<li><a href="http://jpred.ebi.ac.uk/">Jpred</A> -- consensus method for secondary structure prediction
<li><a href="http://www3.ebi.ac.uk/tops">TOPS</A> -- protein topology atlas and tool collection
<br><br>
<li><a href="http://www.ebi.ac.uk/interpro/">Interpro</A> -- integrated resource of protein families, domains and sites
<li><a href="http://www.sanger.ac.uk/Pfam/">Pfam</A> -- alignments and hidden Markov models of protein domains
<li><a href="http://smart.embl-heidelberg.de/">SMART</A> -- simple modular architecture research tool
<li><a href="http://protein.toulouse.inra.fr/prodom.html">ProDom</A> -- protein domain database
<li><a href="http://www.expasy.org/tools/scnpsit2.html">ScanProsite</A> -- scans a sequence against PROSITE
<li><a href="http://www.isrec.isb-sib.ch/software/PFSCAN_form.html">ProfileScan</A> -- scans sequence against profile databases
<li><a href="http://www.ebi.ac.uk/printsscan/">FingerPRINTScan</A> -- search tool for Prints motifs
<li><a href="http://www.ebi.ac.uk/ppsearch/">ppsearch</A> -- pattern search tool for DNA or protein motifs
<li><a href="http://www.blocks.fhcrc.org/">BLOCKS</A> -- aligns ungapped segments of conserved protein regions 
<li><a href="http://blocks.fhcrc.org/blocks/impala.html">IMPALA</A> -- blast search of BLOCKS conserved protein regions
<li><a href="http://bioweb.pasteur.fr/seqanal/motif/blimps-uk.html">Blimps</A> -- improved BLOCKS search tool
<li><a href="http://motif.genome.ad.jp/">Motif</A> -- searches for motifs in query sequence
<li><a href="http://fps.sdsc.edu/">FPS</A> -- family pairwise search library of protein families
<li><a href="http://www.ncbi.nlm.nih.gov/COG/cognitor.html">Cognitor</A> -- compare sequence to clusters of orthologous proteins
<li><a href="http://scop.mrc-lmb.cam.ac.uk/scop/">Scop</A> -- structural classification of proteins
<li><a href="http://www.ebi.ac.uk/pratt/">Pratt</A> -- search for conserved patterns in protein sequences
<li><a href="http://www.biochem.ucl.ac.uk/bsm/cath/">Cath</A> -- hierarchical classification of protein domain structures
<li><a href="http://meme.sdsc.edu/meme/website/mast.html">MAST</A> -- motif alignment search tool
<li><a href="http://meme.sdsc.edu/meme/website/meme.html">MEME</A> -- analyzes protein and DNA for motifs	
<li><a href="http://mips.gsf.de/proj/protfam/">Profam</A> -- curated protein classification and homology domains
<li><a href="http://www.tigr.org/TIGRFAMs/">TIGRFAMs</A> -- protein families based on hidden Markov models
<li><a href="http://vms.mips.biochem.mpg.de/mips/programs/classification.html">SuperFam</A> -- classification browser
<li><a href="http://www.embl-heidelberg.de/Services/argos/services.html">Argos</A> -- protein surface and secondary structure tools				
<br><br>
<li><a href="http://www.ebi.ac.uk/dali">Dali</A> -- coordinates of query compared to PDB database
<li><a href="http://genomic.sanger.ac.uk/123D/run123D.shtml">123D</A> -- predicts tertiary structure
<li><a href="http://jura.ebi.ac.uk:8080/3Dee/search/domains_server.html">3Dee</A> -- database of protein domain definitions
<li><a href="http://www.ebi.ac.uk/dali/fssp">FSSP</A> -- fold classification based on structure
<li><a href="http://www.embl-heidelberg.de/predictprotein/predictprotein.html">PredProt</A> -- predicts protein structure
<li><a href="http://dodo.cpmc.columbia.edu/pp/submit_adv.html">PredictProt</A> -- threading, secondary, solvent and transmembrane
<li><a href="http://dove.embl-heidelberg.de/3D/">Fold</A> -- 3D structures based on homologous sequences
<li><a href="http://www.ebi.ac.uk/dali/maxsprout/">MaxSprout</A> -- protein backbone and side chains from alpha coordinates
<li><a href="http://cl.sdsc.edu/ce/all-to-all/all-to-all-r.html">CE</A> -- finds structural alignments from PDB
<br><br>
<li><a href="http://pqs.ebi.ac.uk/">PQS</A> -- protein quaternary structure query form at EBI
<li><a href="http://dip.doe-mbi.ucla.edu/main.html">DIP</A>	 -- database of interacting proteins
<br><br>
<li><a href="http://www-camis.stanford.edu/projects/helix/LPFC/">ProtFam</A> -- average core structures for families from structure
<li><a href="http://www.fccc.edu/research/labs/dunbrack/scwrl/">SCWRL</A> -- sidechain placement using a rotamer library	
<li><a href="http://www.bork.embl-heidelberg.de/ASC/">ASC</A> -- analytic surface calculation package
<li><a href="http://promoter.ics.uci.edu/BRNN-PRED/">AA Contacts</A> -- residue contacts, solvent accessibility
<li><a href="http://prostar.carb.nist.gov/">ProStar</A> -- protein potential site
<br><br>
<li><a href="http://www.dkfz-heidelberg.de/spec/sweet2/doc/">Sweet</A> -- constructs 3D models of oligosaccharides from formula
<li><a href="http://www.glycosuite.com">GlycoSuiteDB</A> -- curated relational database of glycan structures	
<br><br>
<li><a href="http://www.expasy.ch/spdbv/">SwissPdbView</A> -- deep view for manipulation and analysis of PDB structures
<li><a href="http://www.rcsb.org/pdb/cgi/explore.cgi?job=graphics&pdbId=4HHB&page=&pid=11682995474077">PBD Viewers</A> -- choices of plug-ins and 3D structure viewers
<li><a href="http://www.faseb.org/protein/kinemages/kinpage.html">Kinemages</A> -- interactive 3D protein web display
<li><a href="http://www.umass.edu/microbio/rasmol/index2.htm">RasMol</A> -- molecular visualization resource
<li><a href="http://www.umass.edu/microbio/rasmol/">ProtExplorer</A> -- macromolecular viewer related to RasMol
<li><a href="http://www.umass.edu/microbio/chime">Chime</A> -- interactive web plug-in for 3D molecular structures
<li><a href="http://www.umass.edu/microbio/chime/morpher/">Morph</A> -- 	downloadable protein shape morpher
<li><a href="http://www.ch.embnet.org/software/BOX_form.html">Boxshade</A> -- pretty shading of multiple alignments
<li><a href="http://www.ebi.ac.uk/cgi-bin/translate/visprot.pl">ProtCol</A> -- tool for coloring proteins, default is by hydrophobicity
<li><a href="http://proteinexplorer.org/">ProtExpl</A> -- protein explorer interactive software

 </UL></TD></TR></TABLE>
	</TD></TR></TABLE>
	
	
<a name="155"><BR></a>
<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4" ><b>&nbsp; Family Specific </b></FONT></TD>
			<TD ALIGN="right"><A HREF="#000"><IMG SRC="images/index.gif" ALT="" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD><UL>
<li><a href="http://www.gpcr.org/7tm/">GPCRDB</A> -- G-protein coupled receptor database
<li><a href="http://www.receptors.org/NR/">NucleaRDB</A> -- nuclear receptors at UCSF
<li><a href="http://biochem1.basic-sci.georgetown.edu/nrr/NRR1.html">NRR1</A> -- nuclear receptor resource at Georgetown
<li><a href="http://bioinformatics.weizmann.ac.il/HORDE/">HORDE</A> -- human olfactory receptor genes
<li><a href="http://www.cse.ucsc.edu/research/compbio/gpcr-subclass">GPCR</A> -- classifies G-protein coupled recepters
<li><a href="http://www-bio3d-igbmc.u-strasbg.fr/~wicker/Secator/secator.html">Secator</A> -- classifies any protein family (receptors, tRNA synthetases)
<li><a href="http://www.merops.co.uk/">Merops</A> -- curated database on proteases
<li><a href="http://www.mad-cow.org/00/sci_archive_frame.html">Prion</A> -- annotated prion-doppel gene family
</UL></TD></TR></TABLE>
	</TD></TR></TABLE>
	
	
<a name="166"><BR></a>
<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4" ><b>&nbsp; Blast, Blat, and Align Sequences </FONT></b></TD>
			<TD ALIGN="right"><A HREF="#000"><IMG SRC="images/index.gif" ALT="" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD><UL>
<li><a href="http://www.ncbi.nlm.nih.gov/blast/index.nojs.cgi">Blast</A>  --  java-free NCBI blast launch
<li><a href="http://www.expasy.org/cgi-bin/BLASTEMBnet-CH.pl">Swiss</A>  --  blastp and tBlastn HTGS, UniGene at EMBnet
<li><a href="http://www.ncbi.nlm.nih.gov/gorf/bl2.html">Blast2</A>  --  restricts blast to user query and target
<li><a href="http://www.proweb.org/proweb/Tools/WU-blast.html">BlastU</A>  --  blast query against user fasta database
<li><a href="http://searchlauncher.bcm.tmc.edu/downloads/software/batch_client.html">Batch</A>  --  batch blast from Baylor
<br><br>
<li><a href="http://genome.cse.ucsc.edu/cgi-bin/hgBlat?db=hg7">Blat</A>  --  quick alignment of query to human genome assembly
<li><a href="http://www.ncbi.nlm.nih.gov/genome/seq/page.cgi?F=HsBlast.html&&ORG=Hs">HumFin</A>  --  blast finished human genome
<li><a href="http://www.ncbi.nlm.nih.gov/genome/seq/page.cgi?F=MmBlast.html&&ORG=Mm">Mus</A>  --  blast finished mouse genome by chromosome
<li><a href="http://www.ensembl.org/perl/blastview">Ensembl</A>  --  blast human genome project
<li><a href="http://www.sanger.ac.uk/HGP/Chr20/Chr20_blast_server.shtml">Sanger</A>  --  blast sequencing center chromosomes
<li><a href="http://www.ncbi.nlm.nih.gov/blast/mmtrace.html">Traces</A>  --  blast against raw new genomic trace reads
<li><a href="http://trace.ensembl.org/perl/ssahaview">SSAHA</A>  --  trace server at Ensembl
<li><a href="http://www.hri.co.jp/HUNT">Hunt</A>  --  search human novel transcript cDNAs
<br><br>
<li><a href="http://www.cse.ucsc.edu/~kent/xenoAli/xenAliTwo.html">WABA</A>  --  genomic to genomic comparisons
<li><a href="http://bio.cse.psu.edu/pipmaker/">Pipmaker</A>  --  percent identity plot alignment of two genomic sequences
<li><a href="http://www.sanger.ac.uk/Software/Wise2/dbaform.shtml">DBA</A>  --  aligns two sequences sharing conserved blocks
<li><a href="http://www.sanger.ac.uk/Software/Alfresco/">Alfresco</A>  --  comparative genome sequence analysis
<br><br>
<li><a href="http://www2.ebi.ac.uk/clustalw/">ClustalW</A>  --  aligns multiple sequences			
<li><a href="http://www.toulouse.inra.fr/multalin.html">MAlign</A>  --  aligns multiple sequences with flexible formatting
<li><a href="http://www.ch.embnet.org/software/LALIGN_form.html">Lalign</A>  --  find multiple matching subsegments in two sequences
<li><a href="http://www.bork.embl-heidelberg.de/Alignment/consensus.html">Consensus</A>  --  calculates consensus to ClustalW alginment
<li><a href="http://www.ebi.ac.uk/fasta33/">Fasta3</A>  --  fasta3 alignment of protein sequence
<li><a href="http://www.ebi.ac.uk/snpfasta3/">SnpFasta3</A>  --  alignment to HgBase SNP consortium
<li><a href="http://www.ebi.ac.uk/MPsrch/">MPsrch</A>  --  alignment tool with many parameter controls
<li><a href="http://www.ebi.ac.uk/scanps/">Scanps</A>  --  multi-alignment of query protein sequence
</UL></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<a name="177"><BR></a>
<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4" ><b>&nbsp; Tool Collections </FONT></b></TD>
			<TD ALIGN="right"><A HREF="#000"><IMG SRC="images/index.gif" ALT="" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD><UL>
<li><a href="http://www.nhgri.nih.gov/genome_hub.html">NHGRI</A>,
	<a href="http://www.ensembl.org/genome/central/">EBI</A>,
	<a href="http://www.ncbi.nlm.nih.gov/genome/guide/central.html">NCBI</A>
	-- "Genome Hub" pages
<li><a href="http://www.ncbi.nlm.nih.gov/Sitemap/index.html">NCBI</A>  --  site map to extensive services
<li><a href="http://searchlauncher.bcm.tmc.edu/">BCM</A>  --  launches numerous nucleotide and protein tools
<li><a href="http://www.unl.edu/stc-95/ResTools/cmshp.html">CMS</A>  --  extensive directory of tools by category
<li><a href="http://www.up.univ-mrs.fr/~wabim/english/logligne.html">ABI</A>  --  very large French tool compilation
<li><a href="http://www.hum-molgen.de/BioLinks/Quick.html">Biolinks</A>  --  multiple search launcher and tools
<li><a href="http://expasy.hcuge.ch/www/amos_www_links.html">Amos list</A>  --  favorite biolinks at SwissProt
<li><a href="http://workbench.sdsc.edu">Workbench</A  --  integrated genomics work environment
<li><a href="http://genomic.sanger.ac.uk/">Sanger</A>  --  computational genetics group
<li><a href="http://www.ebi.ac.uk/Tools/index.html">EMBL1</A>  --  collection of database searching and tools
<li><a href="http://www.embl-heidelberg.de/Services/index.html">EMBL2</A>  --  various EMBL alignment tools
<li><a href="http://picasso.ucsf.edu/mol_str_links.html">UCSF1</A>+<a href="http://www.cmpharm.ucsf.edu/cohen/links/">2</A>			collections of tool links at UCSF
<li><a href="http://www.bioinf.mdc-berlin.de/">MDC</A>  --  bioinformatics tools at Delbruck Center
<li><a href="http://www.ebi.ac.uk/Tools/index.html">EBI</A>  --  tool and utility collection at EBI
<li><a href="http://www.sdsc.edu/Software/molbio.html">SDSC</A>  --  tools at San Diego Supercomputer Center
<li><a href="http://www.bork.embl-heidelberg.de/">Bork</A>  --  ten tool server of Bork lab
<li><a href="http://www.ebi.ac.uk/biocat/biocat.html">Biocatalog</A>  --  bioinformatic software collection
<li><a href="http://hmmer.wustl.edu/">HMMER</A>  --  profile hidden Markov models for sequence analysis
<li><a href="http://www.cse.ucsc.edu/research/compbio/sam.html">SAM</A>  --  sequence alignment and modelling system by HMM
<li><a href="http://mia.sdsc.edu">MIA</A>  --  search agent finds macromolecular information
<li><a href="http://evolution.genetics.washington.edu/phylip/software.html">Phylip</A>  --  tools and software for phylogenies
<li><a href="http://www.hgmp.mrc.ac.uk/Registered/Menu/phylo.html">Phylo</A>  --  provides 14 utilties for phylogenetic analysis
<li><a href="http://abacus.gene.ucl.ac.uk/software/paml.html">PAML</A>  --  tool for phylogenies
</UL></TD></TR></TABLE>
	</TD></TR></TABLE>
	
	
<a name="188"><BR></a>
<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4" ><b>&nbsp; Journals, Alerts, and General Internet</FONT></b></TD>
			<TD ALIGN="right"><A HREF="#000"><IMG SRC="images/index.gif" ALT="" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD><UL>
<li><a href="http://www.ncbi.nlm.nih.gov:80/entrez/jrbrowser.cgi">PubMedJB</A>  --  journal browser allows access to current issues
<li><a href="http://www.gen.tcd.ie/pubcrawler/">Pubcrawler</A>  --  provides alerts to new abstracts at PubMed
<li><a href="http://highwire.stanford.edu/">HiWire</A>  --  publishes online journals and provides alerts
<li><a href="http://www.bork.embl-heidelberg.de/Alerting/">SeqAlert</A>  --  provides daily homology sequence alerting
<br><br>				
<li><a href="http://www.hum-molgen.de/journals/">New Issues</A>  --  provides entry to 35 genomics jounals
<li><a href="http://genomebiology.com/start.asp">GenomeBio</A>  --  free access to post-genomics journal, pre-print server	
<li><a href="http://www.pnas.org/">PNAS</A>  --  free access to month-old articles in PNAS
<li><a href="http://www.jbc.org">JBC</A>  --  nearly free access to most cited journal
<li><a href="http://www.sciencemag.org/">Science</A>  --  high circulation general science journal
<li><a href="http://www.nature.com/">Nature</A>  --  gateway to restricted spin-off journals
<li><a href="http://www.ajhg.org">AmJHumGen</A>  --  leading disease gene annotation journal
<li><a href="http://www.academicpress.com/genomics">Genomics</A>  --  genomics articles sometimes stale
<li><a href="http://www.genome.org/">GenomeRes</A>  --  genomic research and methods articles
<li><a href="http://www.public.iastate.edu/~pedro/rt_journals.html">Pedro</A>  --  large journal (and tool) collection
<br><br>
<li><a href="http://www.graylab.ac.uk/omd/index.html">OMD</A>  --  online medical dictionary
<li><a href="http://www.google.com/">Google </A>  --  best internet search site for web tools
<li><a href="http://www.lycos.com/ ">Lycos</A>  --  widely used general search tool
<li><a href="http://dir.yahoo.com/Health/News_and_Media/">Yahoo</A>  --  internet categories and news
<li><a href="http://www.nandotimes.com/noframes/healthscience">Nando</A>  --  science and health news feed
<li><a href="http://www.altavista.com/r?F09">LangTrans</A>  --  useful for rough translations between languages
</UL></TD></TR></TABLE>
	</TD></TR></TABLE>
	
</TD></TR></TABLE>
</BODY></HTML>