<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="description" content="UCSC Genome Browser">
	<META name="keywords" content="genome, genome browser, human genome, genome assembly, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, mouse genome, rat genome, chicken genome, chimp genome, dog genome fruitfly, drosophila, fugu, yeast, ciona,  comparative genomics">
	<TITLE>UCSC Genome Browser Home</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">

</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<!--Top Menu Bar---------------------------------------------->  
    <TR><TD COLSPAN=2 HEIGHT=40>
    <TABLE BGCOLOR="000000" CELLPADDING="1" CELLSPACING="1" WIDTH="99%" HEIGHT="27">
        <TR BGCOLOR="2636D1"><TD VALIGN="middle">
        <TABLE BORDER=0 CELLSPACING=0 CELLPADDING=0 BGCOLOR="2636D1" HEIGHT="24">
    	    <TR><TD VALIGN="middle"><FONT color="#89A1DE">&nbsp; 
 	    <A HREF="/cgi-bin/hgGateway" class="topbar">
			Genomes</a> &nbsp; - &nbsp; 
 	    <A HREF="/cgi-bin/hgNear" class="topbar">
			Gene Sorter</A> &nbsp; - &nbsp; 
	    <A HREF="/cgi-bin/hgBlat" class="topbar">
			Blat</a> &nbsp; - &nbsp; 
	    <A HREF="/cgi-bin/hgPcr" class="topbar">
			PCR</a> &nbsp; - &nbsp; 
	    <A HREF="/cgi-bin/hgText" class="topbar">
			Tables</a> &nbsp; - &nbsp; 
	    <A HREF="/FAQ/" class="topbar">
			FAQ</a> &nbsp; - &nbsp; 
	    <A HREF="goldenPath/help/hgTracksHelp.html" class="topbar">
			Help</a> &nbsp; </FONT></TD>
        </TR></TABLE>
    </TD></TR></TABLE>
    </TD></TR>

<!--Left Sidebar Menu----------------------------------------->  
    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="/cgi-bin/hgGateway">
			Genome Browser</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgNear">
			Gene Sorter</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgBlat">
			Blat</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgPcr">
			In Silico PCR</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgText">
			Table Browser</A><HR>
	        <A CLASS="leftbar" HREF="util.html">
			Utilities</A><HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/downloads.html">
			Downloads</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/releaseLog.html">
			Release Log</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/customTracks/custTracks.html">
			Custom Tracks</A><HR>
	        <A CLASS="leftbar" HREF="/encode/">
			ENCODE</A><HR>
	        <A CLASS="leftbar" HREF="mirror.html">
			Mirrors</A><HR>
  	        <A CLASS="leftbar" HREF="http://genome-archive.cse.ucsc.edu">
			Archives</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/credits.html">
			Credits</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/pubs.html">
			Publications</A><HR>
	        <A CLASS="leftbar" HREF="http://openhelix.com/ucscmaterials.shtml"
			TARGET=_blank>
			Training</A><HR>
	        <A CLASS="leftbar" HREF="cite.html">
			Cite Us</A><HR>
	        <A CLASS="leftbar" HREF="license/">
			Licenses</A><HR>
	        <A CLASS="leftbar" HREF="jobs.html">
			Jobs</A><HR>
  	        <A CLASS="leftbar" HREF="staff.html">
			Staff</A><HR>
  	        <A CLASS="leftbar" HREF="contacts.html">
			Contact Us</A><HR>
	        <BR>
	        <IMG SRC="images/DOT.gif" ALT="" HEIGHT=950 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Introduction---------------------------------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; About the UCSC Genome Bioinformatics Site</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
		    This site contains the reference sequence and working draft
		    assemblies for a large collection of genomes. It also shows 
		    the CFTR (cystic fibrosis) region in 13 species and provides
		    a portal to the ENCODE project.  
		    <P>
		    We encourage you to explore these sequences with our tools. The Genome 
		    Browser zooms and scrolls over chromosomes, showing the work of annotators 
		    worldwide.  The Gene Sorter shows expression, homology and other 
		    information on groups of genes that can be related in many ways.  
		    Blat quickly maps your sequence to the genome.
		    The Table Browser 
		    provides convenient access to the underlying database.  
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
		        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		    </TD></TR></TABLE>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
		    <FONT FACE="courier" SIZE=3><B>1 September 2004  - <em>Drosophila pseudoobscura</em> Genome Browser</B></FONT><BR>
		    <P>
		    The UCSC Genome Bioinformatics Group has released a Genome 
		    Browser and Blat server for the <em>Drosophila 
		    pseudoobscura</em> Freeze 1 draft assembly (Aug. 2003).
		    This assembly, UCSC version dp2, was 
		    produced by the Human Genome Sequencing Center (HGSC) at 
		    Baylor College of Medicine. 
		    <P>
		    Freeze 1 is a whole genome shotgun assembly produced using 
		    Baylor HGSC's assembly engine, Atlas. The assembly, which 
		    provides approximately 7x coverage of the euchromatic portion 
		    of the genome, contains 759 scaffolds. The scaffold N50 size 
		    is 1,018,646 bp. The total scaffold size for this assembly 
		    is 139.3 Mbp, with an average size of 184,465 bp. Due to an 
		    assembly error, four large scaffolds "jumped" chromosomes. 
		    These have been split into "A" and "B" parts in the 
		    downloadable assembly files. See the gateway page for more
		    information.
		    <P>
		    Baylor HGSC has provided a 
		    <A HREF="ftp://ftp.hgsc.bcm.tmc.edu/pub/data/Dpseudoobscura/fasta/D_pseudo_freeze1_scaffold_to_chromosome_assignments_8_28_03.txt" 
		    TARGET=_blank>putative chromosome assignment</A> 
		    for the majority of larger scaffolds (> 90% of unique 
		    sequence), based on conservation between the Muller elements. 
		    <P>     
		    Downloads of the dp2 data and annotations may be obtained 
		    from the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/dp2/" 
		    TARGET=_blank>FTP server</A>  
		    or <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#dp">Downloads</A> 
		    page. The initial set of annotation tracks were generated by 
		    UCSC. 
		    <P>
		    Many thanks to the Baylor HGSC for providing the genome
		    assembly data. The UCSC team who produced this browser are 
		    Angie Hinrichs, Heather Trumbower, Robert Kuhn,
		    Donna Karolchik, and the Genome Browser sysadmin team. 
		    See the
		    <A HREF="/goldenPath/credits.html#dp_credits">Credits</A>
		    page for a detailed list of the organizations and individuals 
		    who contributed to this release. 
		    <P>
		    <FONT FACE="courier" SIZE=3><B>30 August 2004  - October West 
		    Coast Genome Browser Seminars</B></FONT><BR>
		    <P>
		    The UCSC Bioinformatics Group announces 2 seminars and 
		    hands-on workshops on the UCSC Genome Browser, presented by 
		    <A HREF="http://www.openhelix.com/" TARGET=_blank>OpenHelix</A>, 
		    a bioinformatics training, software testing and consulting 
		    company.
		    <P>
		    These introductory sessions are geared towards anyone with a 
		    basic knowledge of genomic and biological concepts who is 
		    interested in learning how to use the UCSC Genome Browser. 
		    No programming experience is required. The seminars will
		    cover the topics necessary to learn how to effectively use 
		    the browser tool set, including basic Genome Browser 
		    functionality, searching and BLAT use, Table Browser use, 
		    creating and using custom annotation tracks, and an 
		    introduction to the Gene Sorter. Lectures will be 
		    accompanied by hands-on computer exercises conducted directly 
		    on the Genome Browser web site.
		    <P>
		    The first three-hour course will be held on Monday 4 October
		    in Kirkland, WA (Seattle area).  Two sessions will be 
		    offered: 9:00 a.m. - noon or 1 - 4 p.m. 
		    <P>
		    The second seminar will be held in Cupertino (San 
		    Francisco Bay Area) on Tuesday, 5 October.  Two sessions 
		    will be offered: 1 - 4 p.m. or 6 - 9 p.m. 
		    <P>
		    For registration information, visit the 
		    <A HREF="http://www.openhelix.com/noteworthy.shtml" 
		    TARGET=_blank>OpenHelix</A> website or call 1-888-861-5051. 
		    Academic, student, and early registration discounts are 
		    available.
		    <P>
		    <HR>
		    <P>
		    <FONT FACE="courier" SIZE=3><B>10 August 2004  - <em>Anopheles gambiae</em> Genome Browser Released: </B></FONT>
		    The UCSC Genome Bioinformatics Group has released a Genome 
		    Browser and Blat server for the <em>Anopheles gambiae</em> 
		    v. MOZ2 draft genome sequence (Feb. 2003). This assembly, 
		    UCSC version anoGam1, was produced by the International 
		    Anopheles Genome Project and downloaded from Ensembl.
		    <A HREF="/goldenPath/newsarch.html#081004">Read more</A>.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>29 July 2004  - Zv3 Zebrafish Genome Browser Available:</B></FONT>
		    The Zv3 Zebrafish genome assembly (UCSC version danRer1) is 
		    now available on the UCSC Genome Browser and Blat server. 
		    <A HREF="/goldenPath/newsarch.html#072904">Read more</A>.
		    <P>
    		    <FONT FACE="courier" SIZE=3><B>23 July 2004  - NCBI Human Build 35 Released on Genome Browser:</B></FONT>
		    The latest human genome reference sequence (NCBI Build 35, 
		    May 2004) is now available as database hg17 in the UCSC 
		    Genome Browser and Blat server. 
		    <A HREF="/goldenPath/newsarch.html#072304b">Read more</A>.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>23 July 2004  - <em>Drosophila yakuba</em> Genome Browser Released:</B></FONT>
		    The UCSC Genome Bioinformatics Group has released a Genome
		    Browser and Blat server on a second species of fruitfly,
		    <em>D. yakuba</em>. 
		    <A HREF="/goldenPath/newsarch.html#072304a">Read more</A>.
    	            </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
	    	
<!--Conditions of Use Section--------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome Browser 
                    are freely available for academic, nonprofit, and personal use 
		    with the following conditions:
		    <UL>
		    <LI>Genome sequence data use restrictions are noted within the
		    species sections on the <A HREF="/goldenPath/credits.html">Credits</A> 
		    page.
		    <LI>Some annotation tracks contributed by external collaborators
		    contain proprietary data that have specific use restrictions. To
		    check for restrictions associated with a particular genome assembly, 
		    review the <em>database/README.txt</em> file in the assembly's 
		    downloads directory. 
		    </UL>
		    <P>
		    The Genome Browser and Blat software are free for academic,
		    nonprofit, and personal use. A license is required for commercial use.
		    See the <A HREF="/license/">Licenses</A> page for more information.
		    <P>
                    Program-driven use of this software is limited to a maximum of one 
                    hit every 15 seconds and no more than 5,000 hits per day.  
		    <P>
		    Questions or problems regarding the UCSC Genome Browser 
		    software, database, genome assemblies, or release cycles 
		    should be sent to 
		    <A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A>.
		    Messages sent to this address will be posted automatically 
		    on the genome mailing list, which is also archived on a 
		    public Web-accessible pipermail archive. NOTE: This archive may be 
		    indexed by non-UCSC sites such as Google. 
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>

        <BR>

<!--Technical Information Section----------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	        
                    <FONT SIZE="4" ><A name="Tech"></A><B>&nbsp; Technical Information About the Assembled Sequence</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
	            <LI><A HREF="goldenPath/gbdDescriptions.html"><FONT SIZE="4">Description of the annotation database</FONT></A>
	            <LI><A HREF="goldenPath/datorg.html"><FONT SIZE="4">Data organization and format</FONT></A>
	            <LI><A HREF="goldenPath/term.html"><FONT SIZE="4">Terminology</FONT></A>
                    </UL>
                    </TD>
                    <TD WIDTH=15>&nbsp;</TD>
                 </TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
    	
<!--Related Links Section--------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	            <FONT SIZE="4" ><A NAME="Other"></A><B>&nbsp; Related Links</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
<!-- Commented out til this page is updated - very out of date
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/"><FONT SIZE="4">Human Genome Project at UCSC</FONT></A><FONT SIZE="4"></FONT>
-->
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/"><FONT SIZE="4">Center for Biomolecular Science and Engineering Home Page</FONT></A>
                    </UL>
                </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
</TD></TR></TABLE>

</BODY></HTML>

