<!-- News Section ============================================- -->  
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
    <TR><TD>        
    <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
        <TR><TD>        
        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list.</P>

    <!-- start news -->
        <hr>
        <p>
        <font face="courier" size="3"><b> 30 May 2014 -
	New Human Variation Track Now Available</b>
        </font>
        </p>

	<p>
	We are pleased to announce the release of the new Exome Variant Server (EVS) Variants track. The
	<a href="http://evs.gs.washington.edu/EVS/" target="_blank">EVS</a>, maintained by the
	<a href="http://chum.gs.washington.edu/" target="_blank">Nickerson lab</a> at the University of
	Washington, contains variant annotations for over 6,500 exomes that were sequenced as part of the
	<a href="https://esp.gs.washington.edu/drupal/" target="_blank">NHLBI GO Exome Sequencing Project</a>
	(ESP). This annotation includes information such as function, clinical association, dbSNP rsID,
	average sample read depth, and much more. You can find more information on how the data for this
	track was generated on the <a href="../cgi-bin/hgTrackUi?db=hg19&g=evsEsp6500"
	>track description page</a>.

	<p>
	We would like to acknowledge Tim Dreszer and Matthew Speir for their hard work in creating this track.
	</p>

	<hr>
        <p>
        <font face="courier" size="3"><b> 27 May 2014 -
        New Roadmap Epigenomics Integrative Analysis Hub now available</b>
        </font>
	</p>

	<p>
	We are pleased to announce the addition of the
	<a href="../cgi-bin/hgTracks?db=hg19&hubUrl=http://vizhub.wustl.edu/VizHub/RoadmapIntegrative.txt"
	>Roadmap Epigenomics Integrative Analysis Hub</a> to our list of publicly available track hubs.
	This new hub, produced by <a href="http://vizhub.wustl.edu/" target="blank">VizHub</a> at Washington
	University in St. Louis (WUSTL), contains hundreds of tracks that cover the wide range of epigenomic data
	available from the <a href="http://www.roadmapepigenomics.org/" target='blank'>Roadmap Epigenomics
	Project</a>. This hub contains data from over 40 different assays carried out on over 250 different
	cell and sample types. These assays include RNA-seq, MeDIP-seq, MRE-seq, methylC-seq, DNase
	hypersensitivity, chromHMM segmentation, ChIP-seq, and more than 30 histone modifications. The data
	contained in this hub is complementary to the ENCODE data available at UCSC and in the ENCODE
	Integrative Analysis public hub.
	</p>

	<p>
	You can view more data produced by the Roadmap Epigenomics Project in their other publicly available
	track hub, the Roadmap Epigenomics Data Complete Collection. You can use these two hubs together
	to access all the available Roadmap Epigenomics Project data using the UCSC Genome Browser. To access
	and view these hubs, navigate to the <a href="../cgi-bin/hgHubConnect">Track Hub gateway page</a>
	and select both the Roadmap Epigenomics Integrative Analysis Hub and Roadmap Epigenomics Data
	Complete Collection Hub from the Public Hubs list.
	</p>

	<p>
	Any queries about the data should be directed to
	<A HREF="mailto:&#116;w&#97;&#110;&#103;&#64;&#103;&#101;&#110;&#101;t&#105;&#99;&#115;.
&#119;&#117;&#115;&#116;&#108;.
&#101;d&#117;">Ting Wang</A> from VizHub.
<!-- above address is twang at genetics.wustl.edu -->
	</p>

        <hr>

        <!-- staged for archive use.  NOTE: condense header line from .... </b></font><p> to </b>:</font>

	<p>
	<font face="courier" size="3"><b> 30 May 2014 -
	New Human Variation Track Now Available</b>:</font>
        We are pleased to announce the release of the new Exome Variant Server (EVS) Variants track.
        <a href="goldenPath/newsarch.html#053014">Read more</a>
	</p>

        <p>
        <font face="courier" size="3"><b> 22 May 2014 -
        New Roadmap Epigenomics Integrative Analysis Hub now available</b>:</font>
        We are pleased to announce the addition of the
        <a href="../cgi-bin/hgTracks?db=hg19&hubUrl=http://vizhub.wustl.edu/VizHub/RoadmapIntegrative.txt"
        >Roadmap Epigenomics Integrative Analysis Hub</a> to our list of publicly available track hubs.
	<a href="goldenPath/newsarch.html#052714">Read more</a>
	</p>
                -->

        <!-- start archives -->

	<p>
	<font face="courier" size="3"><b> 15 May 2014 -
	We're happy to announce the release of a new UCSC Genes track for the
        GRCh38/hg38 human Genome Browser.</b></font>
	<a href="goldenPath/newsarch.html#051514">Read more</a>
	</p>

	<p>
        <font face="courier" size="3"><b> 09 May 2014 -
        Three New Assemblies Now Available in the Genome Browser</b>:</font>
        A Genome Browser is now available for the Chinese hamster (<em>Cricetulus griseus</em>),
        hedgehog (<em>Erinaceus europaeus</em>), and zebra finch (<em>Taeniopygia guttata</em>).
        <a href="goldenPath/newsarch.html#050914">Read more</a>
	</p>

            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
