<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="description" content="UCSC Genome Browser">
	<META name="keywords" content="genome, genome browser, human genome, genome assembly, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, fruitfly, fugu, yeast comparative genomics">
	<TITLE>UCSC Genome Browser Home</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">

</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<!--HOTLINKS BAR-->
    <TR><TD COLSPAN=2 HEIGHT=40>
    <TABLE BGCOLOR="000000" CELLPADDING="1" CELLSPACING="1" WIDTH="99%" HEIGHT="27">
        <TR BGCOLOR="2636D1"><TD VALIGN="middle">
        <TABLE BORDER=0 CELLSPACING=0 CELLPADDING=0 BGCOLOR="2636D1" HEIGHT="24">
    	    <TR><TD VALIGN="middle"><FONT color="#89A1DE">&nbsp; 
 	    <A HREF="../cgi-bin/hgGateway" class="topbar">
			Genome Browser</a> &nbsp; - &nbsp; 
 	    <A HREF="../cgi-bin/hgNear" class="topbar">
			Family Browser</a> &nbsp; - &nbsp; 
	    <A HREF="../cgi-bin/hgBlat" class="topbar">
			Blat</a> &nbsp; - &nbsp; 
	    <A HREF="../cgi-bin/hgText" class="topbar">
			Table Browser</a> &nbsp; - &nbsp; 
	    <A HREF="/FAQ/" class="topbar">
			FAQ</a> &nbsp; - &nbsp; 
	    <A HREF="goldenPath/help/hgTracksHelp.html" class="topbar">
			Help</a> &nbsp; </FONT></TD>
        </TR></TABLE>
    </TD></TR></TABLE>
    </TD></TR>

    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="../cgi-bin/hgGateway">
			Genome Browser</A><HR>
	        <A CLASS="leftbar" HREF="../cgi-bin/hgNear">
			Family Browser</A><HR>
	        <A CLASS="leftbar" HREF="../cgi-bin/hgBlat">
			Blat</A><HR>
	        <A CLASS="leftbar" HREF="../cgi-bin/hgText">
			Tables</A><HR>
	        <A CLASS="leftbar" HREF="downloads.html">
			Downloads</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/releaseLog.html">
			Release Log</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/customTracks/custTracks.html">
			Custom Tracks</A><HR>
	        <A CLASS="leftbar" HREF="mirror.html">
			Mirrors</A><HR>
  	        <A CLASS="leftbar" HREF="http://genome-archive.cse.ucsc.edu">
			Archives</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/credits.html">
			Credits</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/pubs.html">
			Pubs</A><HR>
	        <A CLASS="leftbar" HREF="cite.html">
			Cite Us</A><HR>
	        <A CLASS="leftbar" HREF="license/">
			Licenses</A><HR>
	        <A CLASS="leftbar" HREF="jobs.html">
			Jobs</A><HR>
  	        <A CLASS="leftbar" HREF="staff.html">
			Staff</A><HR>
  	        <A CLASS="leftbar" HREF="contacts.html">
			Contact Us</A><HR>
	        <BR>
	        <IMG SRC="images/DOT.gif" ALT="" HEIGHT=970 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Content Tables-->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; About the UCSC Genome Bioinformatics Site</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
		    This site contains the reference sequence for the human and <em>C. elegans</em> 
		    genomes and working drafts for the chimpanzee, mouse, rat, <em>Fugu</em>, <EM>Drosophila</EM>, 
		    <em>C. briggsae</em>, yeast, and SARS genomes.  It also shows the
		    CFTR (cystic fibrosis) region in 13 species.  
		    <P>
		    We encourage you to explore these sequences with our tools. The Genome 
		    Browser zooms and scrolls over chromosomes, showing the work of annotators 
		    worldwide.  The Family Browser shows expression, homology and other 
		    information on groups of genes that can be related in many ways.  
		    The Table Browser provides convenient access to the underlying database.  
		    Blat quickly maps your sequence to the genome.
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
		        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		    </TD></TR></TABLE>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
	            <P> 
		    <FONT FACE="courier" SIZE=3><B>23 Feb. 2004  - Chimpanzee Draft Assembly Released</B></FONT><BR>
		    <P>
		    We are happy to announce the release of a Genome Browser and Blat server for the
		    chimpanzee (<EM>Pan troglodytes</EM>). 
		    The 13 Nov. 2003 Arachne assembly -- labeled Chimp Build 1
		    Version 1 (UCSC version panTro1) -- was produced by the 
		    Chimpanzee Genome Sequencing 
		    Consortium. 
		    <P>
		    This assembly covers 
		    about 95 percent of the genome and is based on 4X sequence coverage.
	  	    It is composed of 361,782 contigs with an N50 length of 15.7 kb, and 		    37,849 supercontigs having an N50 length of 8.6 Mb (not including 
		    gaps). The total contig length is 2.73 Gb, spanning 3.02 Gb.
		    The DNA donor for this genome assembly, &quot;Clint&quot;, 
		    is a captive-born West African chimpanzee (<em>Pan troglodytes 
		    verus</em>). 
		    <A HREF="http://www.genome.gov/Pages/Research/Sequencing/SeqProposals/ChimpGenome2.pdf" 
		    TARGET=_blank>Background information</A> on the chimp genome
		    sequencing project and the <A HREF="http://www.genome.gov/11509418"
		    TARGET=_blank>initial news release</A> about the chimp 
		    assembly can be found on the NHGRI website.
		    <P>
		    Because of the close relationship between chimpanzees and 
		    humans, this assembly should facilitate comparative analyses 
		    of the two genomes that have not been possible with other 
		    species that have been sequenced to date. Research has indicated that
		    the human and chimp genomes probably differ by approximately one
		    percent. As comparative studies proceed, scientists are observing 
		    some of the same rapidly-rearranging regions in chimp/human 
		    comparisons that had been previously detected between
	 	    the mouse and human genomes. Sequence differences between the human
		    and chimp may shed light on the genetic underpinnings of 
		    speech and language development, among other things. 
		    <P>
		    The initial release of the Chimp Browser provides several 
		    annotation tracks comparing the chimp and human genomes.
		    More comparative annotations will be added in upcoming
		    weeks.
		    <P>
		    Bulk downloads of the chimp sequence and annotations may be obtained from the Genome 
		    Browser <A HREF="ftp://genome.cse.ucsc.edu/goldenPath/panTro1/">FTP server</A> or 
		    <A HREF="/downloads.html#chimp">Downloads</A> page. 
		    The complete set of sequence reads is available at the 
		    <A HREF="http://www.ncbi.nlm.nih.gov/Traces" 
		    TARGET=_blank>NCBI trace archive</A>.
		    Blat searches on chrUn_random are not supported in the initial
		    release, but will be available soon.
		    <P>
                    We'd like to thank NHGRI, the Eli & Edythe L. Broad Institute at 
		    MIT/Harvard, and Washington University at St. Louis School of 
		    Medicine for providing this sequence, and LaDeana Hillier, 
		    Washington University School of Medicine, and the
                    Whitehead Institute for their work on the alignments. The chimpanzee
		    browser annotation tracks were generated by UCSC and collaborators 
		    worldwide. 
                    <P>
                    The UCSC team who worked on this release consisted of
                    Kate Rosenbloom, Jim Kent, Hiram Clawson, Heather Trumbower, Robert
                    Kuhn, Donna Karolchik, and the Genome Browser sysadmin team.
		    <P></P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>12 Feb. 2004  - New UCSC-supported mirror site:</B></FONT>
		    The Genome Browser project now has a UCSC-supported mirror site 
		    that may be used during power or network outages on the UCSC 
		    campus. The mirror -- which can be found at 
		    <A HREF="http://genome.brc.mcw.edu/" TARGET=_blank>http://genome.brc.mcw.edu/</A>
		    -- is located at the Medical College of Wisconsin in Milwaukee. 
		    <A HREF="/goldenPath/newsarch.html#021204">Read more</A>.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>16 Jan. 2004  - Hg16 table updates:</B></FONT>
		    We've discovered a handful of hg16 chrN_random_gap 
		    and chrN_random_gold tables on our public server that are 
		    out of date. <A HREF="/goldenPath/newsarch.html#011604">Read more</A>.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>14 Jan. 2004 - Yeast Genome Released:</B></FONT>
		    We are proud to add yeast
		    (<em>S. cerevisiae</em>) to our growing list
		    of genome assemblies. <A HREF="/goldenPath/newsarch.html#011404">Read more</A>.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>10 Dec. 2003 - Human/Chimp Alignment 
		    Tracks Released:</B></FONT>
                    UCSC has released alignments of the Nov. 2003 
		    chimpanzee
		    draft assembly to the July 2003 human assembly in the Genome Browser. <A HREF="/goldenPath/newsarch.html#121003">Read more</A>.
    	            </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
	    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome Browser 
                    are freely available for academic, nonprofit, and personal use 
		    with the following conditions:
		    <UL>
		    <LI>Genome sequence data use restrictions are noted within the
		    species sections on the <A HREF="/goldenPath/credits.html">Credits</A> 
		    page.
		    <LI>Some annotation tracks contributed by external collaborators
		    contain proprietary data that have specific use restrictions. To
		    check for restrictions associated with a particular genome assembly, 
		    review the <em>database/README.txt</em> file in the assembly's 
		    downloads directory. 
		    </UL>
		    <P>
		    The Genome Browser and Blat software are free for academic,
		    nonprofit, and personal use. A license is required for commercial use.
		    See the <A HREF="/license/">Licenses</A> page for more information.
		    <P>
                    Program-driven use of this software is limited to a maximum of one 
                    hit every 15 seconds and no more than 5,000 hits per day.  
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>

        <BR>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	        
                    <FONT SIZE="4" ><A name="Tech"></A><B>&nbsp; Technical Information About the Assembled Sequence</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
	            <LI><A HREF="goldenPath/gbdDescriptions.html"><FONT SIZE="4">Description of the annotation database</FONT></A>
	            <LI><A HREF="goldenPath/datorg.html"><FONT SIZE="4">Data organization and format</FONT></A>
	            <LI><A HREF="goldenPath/term.html"><FONT SIZE="4">Terminology</FONT></A>
                    </UL>
                    </TD>
                    <TD WIDTH=15>&nbsp;</TD>
                 </TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	            <FONT SIZE="4" ><A NAME="Other"></A><B>&nbsp; Other Links</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/"><FONT SIZE="4">Human Genome Project at UCSC</FONT></A><FONT SIZE="4"></FONT>
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/"><FONT SIZE="4">Center for Biomolecular Science and Engineering Home Page</FONT></A>
                    </UL>
                </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
</TD></TR></TABLE>

</BODY></HTML>

