<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="description" content="UCSC Genome Browser">
	<META name="keywords" content="genome, genome browser, genome assembly, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA">
	<TITLE>UCSC Genome Bioinformatics Home</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">

<SCRIPT LANGUAGE="JavaScript">
<!--

var theDb = 'human';

function setDb()
{
  theDb = document.dbForm.genome.options[document.dbForm.genome.selectedIndex].value;
}

function initDb()
{
  theDb = document.dbForm.genome.options[document.dbForm.genome.selectedIndex].value;
  var queryString = document.location.search.substring(1, document.location.search.length);
  var dbIndex = queryString.indexOf('org=');
  var ampIndex = queryString.indexOf('&');
  
  if (dbIndex >= 0)
  {
    if (ampIndex > 0) {
      theDb = queryString.substring(dbIndex + 4, ampIndex);
  } else {
      theDb = queryString.substring(dbIndex + 4);
  }   

    for (var items = 0; items < document.dbForm.genome.options.length; items++)
    {
      if(sameOrganism(theDb, document.dbForm.genome.options[items].value))
      {
        document.dbForm.genome.options[items].selected = true;
      }
    }     
  }
}

function sameOrganism(db1, db2)
{
  if (db1.toLowerCase() == db2.toLowerCase())
  {
    return true;
  }
  else
  {
    return false;
  }
}

function dbGo(theUrl)
{
document.location = theUrl + 'org=' + theDb;
}

//-->
</SCRIPT>


</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF" onLoad="initDb()">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg"></TH></TR>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
	<table bgcolor="000000" cellpadding="1" cellspacing="1" width="99%" height="27"><tr bgcolor="2636D1"><td valign="middle">
	<table BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24">
		<TR><TD VALIGN="middle"><font color="#89A1DE">&nbsp; 
			<a href="javascript:dbGo('/cgi-bin/hgGateway?');" class="topbar">
				Genome Browser</a> &nbsp; - &nbsp; 
			<A HREF="javascript:dbGo('/cgi-bin/hgBlat?command=start&');" class="topbar">
				Blat Search</a> &nbsp; - &nbsp; 
			<A HREF="javascript:dbGo('/cgi-bin/hgText?command=start&');" class="topbar">
				Table Browser</a> &nbsp; - &nbsp; 
			<A HREF="FAQ.html" class="topbar">
				FAQ</a> &nbsp; - &nbsp; 
			<A HREF="goldenPath/help/hgTracksHelp.html" class="topbar">
				User Guide</a> &nbsp; </font></TD>
    	</TR></TABLE>
	</TD></TR></TABLE>
</TD></TR>

<TR HEIGHT="100%" VALIGN="top">
  <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
  	<!-- outer table is to align leftbar & content boxes at top of page -->
  	<TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0 MARGIN=0>
	<TR><TD><IMG SRC="images/DOT.gif" HEIGHT="7"></TD></TR>
	<TR HEIGHT="100%"><TD HEIGHT="100%">
		<!--outer table is for black border-->
		<TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
   	 	<TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
		<TR HEIGHT="100%">
			<TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%"> 
				<FORM NAME="dbForm">
			<table cellpadding="" cellspacing="0" margins="0" border="0"><tr><td valign="top">
				<FONT CLASS="leftbar">Genome</FONT>
			</td></tr><tr valign="bottom"><td valign="bottom">
					<SELECT NAME="genome" onChange="setDb();">
						<OPTION VALUE="human">Human
						<OPTION VALUE="mouse">Mouse 
						<OPTION VALUE="rat">Rat 
					</SELECT>
			</td></tr></table><hr>
			<A CLASS="leftbar" HREF="javascript:dbGo('/cgi-bin/hgGateway?');">
				Browser</A><hr>
			<A CLASS="leftbar" HREF="javascript:dbGo('/cgi-bin/hgBlat?command=start&');">
				Blat</A><hr>
			<A CLASS="leftbar" HREF="javascript:dbGo('/cgi-bin/hgText?');">
				Tables</A><hr>
			<A CLASS="leftbar" HREF="downloads.html">
				Downloads</A><hr>
		  	<A CLASS="leftbar" HREF="mirror.html">
				Mirrors</A><hr>
  			<A CLASS="leftbar" HREF="http://genome-archive.cse.ucsc.edu">
				Archives</A><hr>
			<A CLASS="leftbar" HREF="goldenPath/credits.html">
				Credits</A><hr>
			<A CLASS="leftbar" HREF="cite.html">
				Cite Us</A><hr>
			<A CLASS="leftbar" HREF="jobs.html">
				Jobs</A><hr>
  			<A CLASS="leftbar" HREF="bestlinks.html">
				Links</A><hr>
  			<A CLASS="leftbar" HREF="contacts.html">
				Contact Us</A><hr>
                                </FORM>
			<BR>
			<IMG SRC="images/DOT.gif" HEIGHT=1100 WIDTH=1>
		</TD></TR></TABLE>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
</TD>

<!--Content Tables-->  
  <TD VALIGN="top"><TABLE CELLPADDING=5 BORDER=0><TR><TD>
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<A NAME="About"></A><FONT SIZE="4"><b>&nbsp; About the UCSC Genome
	Bioinformatics Site</FONT>
	</TD></TR></TABLE>
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD WIDTH=10></TD><TD>
		<P>This site contains working drafts for the human, mouse, and
		rat genomes. 
		<P>The human genome is over 95% complete. Approximately 
		90% of the sequence is in a highly accurate 'finished' state.  The
		other 10% is merely 'draft' quality.  Some care must be taken
		when interpreting draft regions, but these are still very 
		useful to the working scientist.  
		<P>The mouse genome is
		largely the result of a full genome shotgun assembly. While there are 
		many gaps in the sequence, the order and orientation of the contigs on 
		either side of a gap are known in almost all cases. We estimate
		that 90-96% of the mouse genome is present in the
		assembly. We encourage you to explore the
		working drafts with the genome browser, which displays
		the work of many annotators worldwide.
		<P>The Genome Browser and the data it displays are freely available for 
		academic, nonprofit, and personal use with a few exceptions. Check the
		README.txt file in the download directory of a particular
		assembly to view the use restrictions
		specific to that release. Interactive use is limited to a 
		maximum of 10,000 hits/day or 10 hits/minute. 
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100% BORDER=0><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; News </FONT></TD>
			<TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD WIDTH=10></TD><TD>
	<P><FONT FACE="courier" SIZE=3><B>Mar. 26, 2003</B></FONT><BR>
	We've added an updated rat assembly to our site: Rat Jan 2003 (rn2). This
	corresponds to the Version 2.1 Jan 2003 Update of the rat genome 
	assembly, produced by the Atlas group at Baylor HGSC as part of the Rat
	Genome Sequencing Consortium.
	<P>
	This update corrects duplications that were assembly artifacts in the 
	previous version and improves the linking of bactigs to create larger
	"ultrabactigs". Compared with the previous rat assembly, sequence mapped 
	to specific chromosomal coordinates is reduced by about 1.6 percent. 
	Loosely mapped and unmapped sequence is reduced by 17 percent. For more
	details and statistics on the Jan 2003 assembly, see the Baylor HGSC
	<A HREF="ftp://rat-ftp.hgsc.bcm.tmc.edu/pub/analysis/rat/README">
	README</A> for this release.
	<P>
	UCSC has released a Genome Browser and BLAT server for this assembly 
	update. The initial <A HREF="http://genome.ucsc.edu/cgi-bin/hgTracks?db=rn2">
	browser</A> contains 16 annotation tracks, with more to follow in coming 
	weeks. Sequence downloads are currently available at 
	<A HREF="ftp://genome.ucsc.edu/goldenPath/rnJan2003/" TARGET=_blank>
	ftp://genome.ucsc.edu/goldenPath/rnJan2003/</A> or via the Downloads 
	link on this page. A complete set of database downloads will be available at the beginning 
	of next week. This data contains 
	<A HREF="http://genome.ucsc.edu/goldenPath/credits.html#rat_use">specific 
	conditions for use</A>.
	<P>
	Thanks to the Atlas group at Baylor HGSC, the Rat Genome Sequencing Consortium,
	the UCSC Genome Bioinformatics group, and contributors worldwide for 
	making this release available.
	
	<P><FONT FACE="courier" SIZE=3><B>Mar. 13, 2003</B></FONT><BR>
	We're happy to announce an update to the mouse genome sequence. This new 
	version (Mouse Feb.  2003) includes 705 megabases of finished sequence, 
	compared to 96 megabases of finished sequence in the previous assembly. 
	Many people in the Mouse Genome Sequencing Consortium contributed to 
	this update. The Sanger Institute in particular contributed a large 
	amount of finished sequence. Richa Agarwala, Deanna Church, and 
	coworkers at NCBI layered the finished clones on top of the Arachne whole genome shotgun assembly. Arian Smit constructed a new RepeatMasker 
	library.  
	<P>
	UCSC has released a Genome Browser and BLAT server for the Feb. 2003 
	Mouse genome. The initial 
	<A HREF="http://genome.ucsc.edu/cgi-bin/hgTracks?db=mm3">
	browser</A> contains 14 annotation tracks, with more to follow in coming 
	weeks. Sequence downloads are currently available at 
	<A HREF="ftp://genome.ucsc.edu/goldenPath/mmFeb2003/" TARGET=_blank>
	ftp://genome.ucsc.edu/goldenPath/mmFeb2003/</A> or via the Downloads 
	link on this page. Database downloads will be available at the beginning 
	of next week. 
	<P>
	Thanks to everybody at UCSC and around the world that contributed to the 
	success of this release! 

	<P><FONT FACE="courier" SIZE=3><B>Feb. 5, 2003</B></FONT><BR>
	We're proud to announce the release of version 17 of the UCSC Genome 
	Browser. 
	This version contains powerful new features, numerous improvements to 
	the annotation track display, additional annotation 
	tracks, and a number of bug fixes. In this release cycle, we've also introduced
	an enhanced QA process that formalizes our testing and verification
	of the Genome Browser software and the data displayed in the browser.
	<P>
	<HR>
	<B>New functionality in v.17:</B><BR> 
	-- Numerous enhancements to the table browser that allow the user to 
	conduct more complex and specific searches. New features include support 
	for intersections of tracks, a new summary statistics output format, and 
	the ability to output query results as a custom annotation track that 
	can be viewed in the Genome Browser.
	 The new 
	<A HREF="goldenPath/help/hgTextHelp.html" 
	TARGET=_blank> Table Browser User's Guide</A> contains a detailed 
	description of the new features and provides a wealth of information and
	examples for conducting various types of searches on the database tables. 
	<P>
	-- Two new display modes available for most annotation tracks: pack and
	squish modes. In <I>pack mode</I> display, annotation track features are fully
	displayed, but more than one feature may be displayed on the same line. 
	This greatly reduces the amount of display space needed by a track 
	when a user wishes to view a large number of individual features at one 
	time. <I>Squish mode</I> is similar to pack mode, but displays features 
	at 50% height and without labels. This mode is particularly 
	useful for viewing tracks in which a large number of features align to 
	the same section of a chromosome, e.g. EST tracks.
	<P>
	-- Functional groupings of annotation track controls. This 
	makes it much easier to find a particular item in the track control list 
	and gives a better visual overview 
	of the annotations available in a particular category, e.g. comparative
	genomics tracks or gene prediction tracks.
	<P>
	-- A mechanism for saving the annotation tracks image in postscript or 
	PDF format. This much requested feature enables Genome Browser users to print an image at
	high resolution, edit it with a drawing progam, or display it in a 
	postscript or PDF viewer.
	<P>
	-- A <A HREF="goldenPath/customTracks/custTracks.html"
	TARGET=_blank>collection of custom annotation tracks</A> supplied by 
	Genome Browser users and members of the UCSC Genome Bioinformatics lab.
	Additional contributions to this collection are welcome! Contact 
	<A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A> if you have 
	an annotation you'd like to share.
	<P>
	<HR>
	<B>New annotation tracks in v.17:</B><BR> 
	-- a new Known Genes track (Human Builds 30 & 31, Feb 2002 Mouse, 
	Nov 2002 Rat) that shows known 
	protein coding genes based on proteins from SWISS-PROT, TrEMBL, and
   	TrEMBL-NEW and their corresponding mRNAs from Genbank. Features within
	the track are color-coded according to origin and review status.  
	<P>
	-- a new Superfamily track (Human Build 30, Feb 2002 Mouse) that shows proteins having 
	homologs with known structures or functions. Each entry in the track 
	shows the coding region of a gene (based on Ensembl gene prediction). 
	The feature label consists of the names of all known protein domains coded 
	by the gene, and usually contains structural and/or function 
	descriptions that provide valuable information for getting a quick grasp 
	of the biological significance for the gene.
	<P>
	We have also released several additional annotation 
	tracks on the latest human and rat assemblies in the past
	month.
	<P>
	<HR>
	<B>Bug fixes in v.17:</B><BR>
	-- Approximately 40 bugs (mostly minor problems) have been fixed in 
	this version. 
	<P>
	<HR>
	To take full advantage of the new display features in this release, we 
	recommend that you reset your browser to the new default settings. NOTE:
	you may not want to reset your browser if you have customized settings
	that you wish to preserve. You can reset your browser by clicking 
	the "Click here to reset" link on the Browser Gateway page.
	<P>
	We hope this new release facilitates your work with the UCSC Genome Browser.
	If you have any questions or comments about the new release, send email
	to <A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A>.

    </TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<BR>
		
<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4" ><a name="Tech"></a><b>&nbsp; Technical Information About the Assembled Sequence</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD>
<UL>
	<LI><A HREF="goldenPath/term.html"><FONT SIZE="4">Terminology</FONT></A>
	<LI><A HREF="goldenPath/datorg.html"><FONT SIZE="4">Data organization and format</FONT></A>
	<LI><A HREF="goldenPath/gbdDescriptions.html"><FONT SIZE="4">Description of the annotation database</FONT></A>
	<LI><A HREF="goldenPath/stats.html"><FONT SIZE="4">Summary statistics</FONT></A>
	<LI><A HREF="goldenPath/chromReports"><FONT SIZE="4">Chromosome Reports</FONT></A>
         <LI><A HREF="goldenPath/mapPlots/"><FONT SIZE="4">Genome Map Plots</FONT></A>
</UL>
</TD><TD WIDTH=15>&nbsp;</TD></TR></TABLE>
	<BR></TD></TR></TABLE>
</TD></TR></TABLE>
	
<BR>
	
<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4" ><a name="Other"></a><b>&nbsp; Other Links</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD>
<UL>
	<LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/"><FONT SIZE="4">Human Genome Project at UCSC</FONT></A><FONT SIZE="4"></FONT>
	<LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/"><FONT SIZE="4">Center for Biomolecular Science and Engineering Home Page</FONT></A>
</UL>
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
</TD></TR></TABLE>

</TD></TR></TABLE>
</TD></TR></TABLE>
</BODY></HTML>

