<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                          <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="https://lists.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>

                    <!-- start news -->

		    <FONT FACE="courier" SIZE="3"><B>25 Jan. 2010 - New Custom Track type: Binary Alignment/Map (BAM)</B></FONT>
		    <P>
		    We are pleased to announce the availability of another new track type for 
		    Custom Tracks: the Binary Alignment/Map 
		    (<A HREF="goldenPath/help/bam.html">BAM</A>).
		    </P>
		    <P>
		    BAM is the compressed binary version of the 
		    <A HREF="http://samtools.sourceforge.net/" TARGET=_blank>Sequence Alignment/Map (SAM)</A> 
		    format, a compact and index-able representation of nucleotide sequence 
		    alignments. Many next-generation sequencing and analysis tools work with 
		    SAM/BAM. For custom track display, the main advantage of indexed BAM over 
		    PSL and other human-readable alignment formats is that only the portions 
		    of the files needed to display a particular region are transferred to UCSC. 
		    This makes it possible to display alignments from files that are so large 
		    that the connection to UCSC would time out when attempting to upload the 
		    whole file to UCSC. Both the BAM file and its associated index file remain 
		    on your web-accessible server (http or ftp), not on the UCSC server. 
		    UCSC temporarily caches the accessed portions of the files to speed up 
		    interactive display.
		    </P>

		    <HR>

		    <FONT FACE="courier" SIZE="3"><B>13 Jan. 2010 - Lifespan of custom tracks within sessions:</B></FONT>
		    Due to the popularity of UCSC 
		    <A HREF="goldenPath/help/hgTracksHelp.html#CustomTracks">custom tracks</A>
		    and 
		    <A HREF="goldenPath/help/hgSessionHelp.html">sessions</A>,
		    we are running out of disk space for storing custom 
		    tracks accessed within sessions. Although sessions 
		    themselves are saved for one year, user-generated tracks 
		    within sessions (including custom tracks) are saved for 48 
		    hours, the typical period for other custom tracks. Until 
		    now, because we have had space available on the server, we 
		    have attempted to keep custom tracks alive for the lifetime 
		    of the session. 
		    <A HREF="goldenPath/newsarch.html#011310">Read more</A>.
		    </P>

		    <P>
		    <FONT FACE="courier" SIZE="3"><B>11 Jan. 2010 - Zebrafish release zv8 now available in Genome Browser:</B></FONT>
		    The most recent zebrafish assembly -- zv8 (UCSC version 
		    danRer6, Dec. 2008) -- is now available in the UCSC
		    Genome Browser. This assembly was produced by 
		    the Wellcome Trust Sanger Institute, UK. 
		    <A HREF="goldenPath/newsarch.html#011110">Read more</A>.
		    </P>

		    <P>
<FONT FACE="courier" SIZE="3"><B>14 Dec. 2009 - New job posting: Biological Data Technician: </B></FONT>
		    The UCSC Genome Browser project is looking for a 
		    bioinformatician, biologist, or software engineer with a 
		    strong biology background to collect and import data  
		    into the UCSC Genome Browser database and 
		    website. 
		    <A HREF="goldenPath/newsarch.html#121409">Read more</A>.
		    </P>

		    <P>
<FONT FACE="courier" SIZE="3"><B>7 Dec. 2009 - Human Genome Browser default 
		    changing to hg19: </B></FONT>
		    In conjunction with the release of the UCSC Genes and
		    Conservation tracks on the hg19 (GRCh37) human assembly,
		    we have changed the default human browser on our website 
		    from hg18 to hg19. 
		    <A HREF="goldenPath/newsarch.html#120709">Read more</A>.
		    </P>
                        </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
