<!--News Section----------------------------------------------->  
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
    <TR><TD>        
    <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
        <TR><TD>        
        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list.</P>

    <!-- start news -->
 	<p>
 	<font face="courier" size="3"><b>28 June 2013
	- New UCSC Genes Track Released for GRCh37/hg19</b></font>
	</p>

	<p>
	We're happy to announce the release of an updated UCSC Genes
	track for the GRCh37/hg19 human Genome Browser. 
	The new release has 82,960 total transcripts, compared with 
	80,922 in the previous version. The total number of canonical 
	genes has increased from 31,227 to 31,848. Comparing the new 
	gene set with the previous version:
	<ul>
	<li>74,657 transcripts did not change between versions.
	<li>88 transcripts were not carried forward to the new version.
	<li>5,556 transcripts are "compatible" with those in the previous set, 
	meaning that the two transcripts show consistent splicing. 
	In most cases, the old and new transcripts differ in the 
	lengths of their UTRs.
	<li>621 transcripts overlap with those in the previous set but 
	do not show consistent splicing, i.e., they contain overlapping 
	introns with differing splice sites.
	</ul>
	</p>
	<p>
	More details about the new UCSC Genes track can be found on
	the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=knownGene"
	>UCSC Genes track description page</a>.
	</p>
	<p>
	Many thanks to Brian Raney, Jim Kent, and Luvina Guruvadoo 
	for their work on this track!
	</p> 
 
        <p>
        <font face="courier" size="3"><b>27 June 2013
        - New Official European Mirror Released</b></font>
        </p>

        <p>
        The UCSC Genome Browser is pleased to announce the
        introduction of a new mirror site to serve our users
        in Europe.  Genome-euro is an official European mirror
        site of the UCSC Genome Browser, at
        <A HREF = "http://genome-euro.ucsc.edu" TARGET = _BLANK>
        http://genome-euro.ucsc.edu</A>. The server is physically
        located at the Universit&auml;t Bielefeld Center for
        Biotechnology in Bielefeld, Germany, and is administered
        by UCSC. Genome-euro is meant to be an alternate, faster
        access point for those Browser users who are geographically
        closer to central Europe than to the western United States.
        </p>
        <p>
        All functionality will be the same as on the US server,
        although Custom Tracks will not be transfered. Saved Sessions
        containing Custom Tracks will require some manual <A HREF =
        "http://genome.ucsc.edu/goldenPath/help/genomeEuro.html#sessions"
        TARGET = _BLANK>intervention</A>.
        </p>
        <p>
        When European users navigate to the US server home page
        and click the "Genomes" menu item, they will receive a
        notification that they have been
        redirected to the more geographically-appropriate server.
        They will have the option to remain on the US server, as
        described in our
        <A HREF = "http://genome.ucsc.edu/goldenPath/help/genomeEuro.html"
        TARGET = _BLANK>documentation</A>.
        </p>
        <p>
        The backup mirror in <A HREF = "http://genome-mirror.moma.ki.au.dk/"
        TARGET = _BLANK>Aarhus, Denmark</A> will continue to serve as an
        emergency site in the event of the official sites in California
        and Germany malfunctioning.
        </p>
        <p>
        We wish to express our special thanks to our colleagues at the
        Universit&auml;t Bielefeld Bioinformatics, especially Jens Stoye,
        for making this possible.
        </p>

	<p>
	<font face="courier" size="3"><b>26 June 2013
	- Suggestion Box Released</b></font
	</p>
	<p>
	Due to popular demand, we have created a suggestion box for track, assembly,
	tool, and other requests. All suggestions will be assigned a reference
	number so that you may follow up on their progress with our staff. Please
	limit the use of this form to requests for new features; questions and reports
	of problems about existing features and data should continue to be directed to
	our mailing lists. You may reach the suggestion box from the
	<a href="contacts.html">Contact Us</a> page or by
	<a href="cgi-bin/hgUserSuggestion">clicking here</a>.
	</p>
	<p>
 
        <!-- start archives -->
	<hr>

    <!-- staged for archive use

        <p>
        <font face="courier" size="3"><b>27 June 2013
        - New Official European Mirror Released</b></font> Announcing
        an official mirror for European users, with automatic
        redirection.
        <a href="goldenPath/newsarch.html#062713">Read more</a>.</p>
        </p>

	<p>
	<font face="courier" size="3"><b>26 June 2013
	- Suggestion Box Released:</b></font>
	Due to popular demand, we have created a suggestion box for track, assembly,
	tool, and other feature requests.
	<a href="goldenPath/newsarch.html#062613">Read more</a>.</p>
    -->

        <p>
        <font face="courier" size="3"><b>12 June 2013
        - Ferret Genome Browser Released:</b></font>
        A Genome Browser is now available for the Ferret
        (<em>Mustela putorius furo</em>) assembly released by the Ferret Genome Sequencing Consortium
        April  2011 (version MusPutFur1.0, UCSC version musFur1).
	<a href="goldenPath/newsarch.html#061213">Read more</a>.</p>


 	<p>
        <font face="courier" size="3"><b>22 May 2013
        - Gibbon Genome Browser Released:</b></font>
        A Genome Browser is now available for the Gibbon
        (<em>Nomascus leucogenys</em>) assembly released by the Gibbon Genome Sequencing Consortium
	October 2012 (version Nleu_3.0, UCSC version nomLeu3).
        <a href="goldenPath/newsarch.html#052213">Read more</a>.</p>

==> <a href="goldenPath/newsarch.html">News Archives</a>

            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
