<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                          <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="http://www.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>

		    <P>
		    <FONT FACE="courier" SIZE=3><B>4 Oct. 2007 - ENCODE Genome Browser Released for hg18 Assembly</B></FONT><BR>
		    <P>
		    The UCSC ENCODE browser for the human genome assembly hg18 
		    (NCBI Build 36) is now available. You can access the browser
		    directly at 
		    <A HREF="encode.hg18.html">http://genome.ucsc.edu/ENCODE/encode.hg18.html</A> 
		    or by clicking the ENCODE link in the sidebar menu on this 
		    page, then clicking the Regions (hg18) link in the sidebar 
		    menu on the ENCODE portal page.</P>
		    <P>
		    The hg18 ENCODE browser includes 540 data tables in 59 
		    browser tracks that were migrated from the hg17 browser. 
		    The hg17 data coordinates were converted to hg18 coordinates
		    using the UCSC liftOver process.</P>
		    <P>
		    To improve the accessibility of the data, related ENCODE 
		    tracks have been gathered into new configuration groupings 
		    (&quot;super-tracks&quot;) that can be displayed or hidden
		    using a single visiblity control.  We have also reduced the
 		    number of track groups and have modified some of the group 
		    names for clarity. ENCODE tracks with whole-genome data 
		    have been moved into the standard browser track groups.</P>
		    <P>
		    For more information about the hg18 ENCODE data migration, 
		    see the News section on the UCSC 
		    <A HREF="ENCODE/">ENCODE portal page</A> and the UCSC 
		    <A HREF="http://genomewiki.cse.ucsc.edu/index.php/ENCODE_Hg18_Migration" TARGET=_blank>genomeWiki</A>.</P>
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>22 Aug. 2007 - New Mouse Browser Released: </B></FONT>
		    The latest mouse genome assembly from the Mouse Genome
		    Sequencing Consortium, NCBI Build 37.1, is now available in 
		    the UCSC Genome Browser. 
                    <A HREF="goldenPath/newsarch.html#082207b">Read more</A>.</P>
		    <P>
		    <FONT FACE="courier" SIZE=3><B>22 Aug. 2007 - Latest Bovine Assembly Available: </B></FONT>
		    The UCSC Genome Bioinformatics Group has released a 
		    Genome Browser and Blat server for the Aug. 2006 Btau_3.1
		    draft assembly of the cow genome. 
                    <A HREF="goldenPath/newsarch.html#082207a">Read more</A>.</P>
		    <P>
		    <FONT FACE="courier" SIZE=3><B>2 Aug. 2007 - Five New Worm Assemblies: </B></FONT>
		    We've updated our existing nematode browsers--for <em>C. 
		    elegans</em> and <em>C. briggsae</em>--to the latest 
		    publicly available assemblies. In addition, we've added
		    three new worms to our collection: <em>C. brenneri</em> 
		    (Caenorhabditis n. sp. PB2801), <em>C. remanei</em>,
		    and <em>Pristionchus pacificus</em>. 
                    <A HREF="goldenPath/newsarch.html#080207">Read more</A>.</P>
                        </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
