<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                          <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="http://www.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>
<!--PLATYPUS COMMENTED OUT UNTIL RELEASE
		    <P>
                    <FONT FACE="courier" SIZE=3><B>21 May 2007 - Platypus
		    Assembly Available </B></FONT>
		    <P>
		    We have released a Genome Browser and Blat server for 
		    the Jan. 2007 v5.0.1 draft assembly of <em>Ornithorhynchus 
		    anatinus</em> (UCSC version ornAna1) produced by the 
		    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Ornithorhynchus%20anatinus"
		    TARGET=_blank>Genome Sequencing Center at Washington 
		    University</A>, St. Louis, MO (WUSTL).</P>
		    <P>
		    This assembly, which was sequenced using a combination of
		    whole genome shotgun plasmid, fosmid and BAC end sequences, 
		    has a coverage of approximately 6X. It is comprised of
		    about 1.84 Gb of actual sequence (excluding gap estimates),
		    with 437 Mb anchored and ordered on chromosomes.</P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/ornAna1/">FTP server</A> 
		    or the
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#platypus">Downloads</A>
		    page.  These data have 
		    <A HREF="goldenPath/credits.html#platypus_use">specific
		    conditions for use</A>.</P>
		    <P>
		    We'd like to thank WUSTL for providing this assembly. The
		    platypus Genome Browser was produced by Angie Hinrichs,
		    Kayla Smith, Robert Kuhn, Brian Raney, and Donna Karolchik.
		    The platypus browser annotation tracks were generated by 
		    UCSC and collaborators worldwide. See the
		    <A HREF="goldenPath/credits.html#platypus_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release.  </P>
END PLATYPUS COMMENT-->
<!--LIZARD COMMENTED OUT 
		    <P>
                    <FONT FACE="courier" SIZE=3><B>17 May 2007 - Lizard
		    Assembly Available in Genome Browser</B></FONT>
                    <P>
		    A Genome Browser and Blat server are now available for
		    the Jan. 2007 v1.0 draft assembly (UCSC version anoCar1) of 
		    <em>Anolis carolinensis</em> produced by the 
		    <A HREF="http://www.broad.mit.edu/" TARGET=_blank>Broad 
		    Institute</A>.
		    This assembly has been sequenced to 6.8X coverage. The draft
		    sequence contains 7,233 scaffolds comprised of nearly 1.74
		    Gb. </P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/anoCar1/">FTP server</A> 
		    or the
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#lizard">Downloads</A>
		    page.  These data have 
		    <A HREF="goldenPath/credits.html#lizard_use">specific
		    conditions for use</A>.
		    <P>
		    We'd like to thank the Broad Institute for providing this
		    assembly. The lizard Genome Browser was produced by Hiram
		    Clawson, Archana Thakkapallayil, Robert Kuhn, and Donna 
		    Karolchik. The lizard browser annotation tracks were 
		    generated by UCSC and collaborators worldwide. See the
		    <A HREF="goldenPath/credits.html#lizard_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release.
		    </P>
END LIZARD COMMENT-->
		    <P>
                    <FONT FACE="courier" SIZE=3><B>16 May 2007 - New Dates for
		    OpenHelix Seminars in L.A., D.C., Phila.</B></FONT>
		    <P>
		    The dates for the upcoming OpenHelix Genome Browser seminars
		    in Washington, D.C. and Los Angeles, CA have been changed.
		    The following updated announcement includes the new dates:
		    <P>
		    The UCSC Bioinformatics Group announces three regional 
		    seminars and hands-on computer workshops on the UCSC 
		    Genome Browser presented by 
		    <A HREF="http://www.openhelix.com/" 
		    TARGET=_blank>OpenHelix:</A>
		    <UL>
		    <LI>
		    Philadelphia -- Wednesday, 13 June
		    <LI>
		    Washington. D.C./Baltimore -- Wednesday, 20 June
		    <LI>
		    Los Angeles -- Wednesday, 27 June
		    </UL>
		    All sessions will be held 1 p.m. to 4 p.m.
		    </P>
		    <P>
		    These introductory sessions are geared towards anyone with a
		    basic knowledge of genomic and biological concepts who is 
		    interested in learning how to use the UCSC Genome Browser. 
		    No programming experience is required. The seminars will
		    cover the topics necessary to learn how to effectively use 
		    the browser tool set, including basic Genome Browser 
		    functionality, searching and BLAT use, Table Browser use, 
		    creating and using custom annotation tracks, and an 
		    introduction to the Gene Sorter. Lectures will be 
		    accompanied by hands-on computer exercises conducted 
		    directly on the Genome Browser web site. Participants 
		    receive a complete set of slide and exercise handouts and 
		    printed Quick Reference Cards. </P>
		    <P>
		    For further information or to make a reservation, visit the 
		    <A HREF="http://www.openhelix.com/" 
		    TARGET=_blank>OpenHelix</A> website or call 1-888-861-5051. 
		    Register early; seating is limited. Academic and student 
		    discounts are available.</P>
                    <P>
		    <HR>
                    <FONT FACE="courier" SIZE=3><B>14 May 2007 - Horse
		    Genome Browser Now Available: </B></FONT>
		    The Jan. 2007 EquCab1 release of the horse genome 
		    (<em>Equus caballus</em>) is now available
		    in the UCSC Genome Browser. 
                    <A HREF="goldenPath/newsarch.html#051407">Read more</A>.
                    <P>
                    <FONT FACE="courier" SIZE=3><B>27 April 2007 - Medaka
		    Genome Browser Released: </B></FONT>
		    We're happy to announce the release of a Genome Browser and
		    Blat server for the Medaka Version 1.0 draft assembly (Apr. 
		    2006, UCSC version oryLat1). 
                    <A HREF="goldenPath/newsarch.html#042707">Read more</A>.
		    <P>
                    <FONT FACE="courier" SIZE=3><B>11 April 2007 - Latest
		    Fugu Assembly Available in Genome Browser: </B></FONT>
		    The UCSC Genome Browser now includes the latest 
		    release of the Fugu genome. 
                    <A HREF="goldenPath/newsarch.html#041107">Read more</A>.
		    </P>
                        </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
