<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
		        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		    </TD></TR></TABLE>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
		    <P>
		    To receive announcements of new genome 
		    assembly releases, new software features, updates and 
		    training seminars by email, subscribe to the
		    <A HREF="http://www.soe.ucsc.edu/mailman/listinfo/genome-announce" TARGET=_blank>genome-announce</A> mailing list.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>18 July 2006 - New Chimpanzee Genome Browser Released</B></FONT>
		    <P>
		    We are happy to announce the release of a Genome Browser for
		    the latest release of the chimpanzee (Pan troglodytes) 
		    genome. The Jan. 2006 assembly -- labeled Chimp Build 2 
		    Version 1 (UCSC version panTro2) -- was produced by the 
		    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Pan%20troglodytes&SECTION=collaborators"
		    TARGET=_BLANK>Chimpanzee Sequencing and Analysis 
		    Consortium</A>.</P>
		    <P>
		    This 6X whole genome assembly contains sequence from the 
		    initial 4X chimpanzee assembly described and 
		    analyzed in <em>Nature</em> 
		    (<A HREF="http://www.nature.com/nature/journal/v437/n7055/full/nature04072.html" 
		    TARGET=_blank>The Chimpanzee Sequencing and Analysis 
		    Consortium, 2005</A>), with additional 2X 
		    sequence generated, assembled, and assigned to chromosomes 
		    by the 
		    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Pan%20troglodytes"
		    TARGET=_BLANK>Genome Sequencing Center of Washington 
		    University School of Medicine</A>, St. Louis, MO, USA.
		    <P> 
		    The whole genome shotgun data were derived primarily from 
		    the donor Clint, a captive-born male chimpanzee from the 
		    Yerkes Primate Research Center in Atlanta, GA, USA. The 
		    reads were assembled using the whole-genome assembly
		    program PCAP. For information about the assembly process,
		    see the panTro2 
		    <A HREF="../cgi-bin/hgGateway?db=panTro2">Gateway</A> page.
		    </P>
		    <P>
		    This assembly covers about 97 percent of the genome and is 
		    based on 6X sequence coverage. It is composed of 265,882 
		    contigs with an N50 length of 29 kb and 44,460 supercontigs 
		    with an N50 length of 9.7 Mb. The total contig length, not 
		    including estimated gap sizes, is 2.97 Gb. Of that total, 
		    2.82 Gb of sequence have been ordered and oriented along 
		    specific chimpanzee chromosomes, 107 Mb have been placed in 
		    chr*_random, and 50 Mb remain in chrUn. </P>
		    <P>
		    A major difference between this assembly and the previous 
		    Nov. 2003 version is the chromosomal numbering scheme, 
		    which  has been changed to reflect a new standard that 
		    preserves orthology with human chromomes. Proposed by 
		    <A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=pubmed&cmd=Retrieve&dopt=Abstract&list_uids=15218271&query_hl=1&itool=pubmed_docsum"
		    TARGET=_blank>E.H. McConkey</A> in 2004, the new numbering 
		    convention was subsequently endorsed by the International 
		    Chimpanzee Sequencing and Analysis Consortium. This 
		    standard assigns the 
		    identifiers &quot;2a&quot; and &quot;2b&quot; to the
		    two chimp chromosomes that fused in the human genome to form
		    chromosome 2. Note that the genome assembly shown in the 
		    Nov. 2003 panTro1 Genome Browser retains the older 
		    numbering scheme.</P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/panTro2/">FTP
		    server</A> or the 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#chimp">Downloads</A> 
		    page. The complete set of sequence reads is available at the

		    <A HREF="http://www.ncbi.nlm.nih.gov/Traces" 
		    TARGET=_blank>NCBI trace archive</A>. These data have 
		    <A HREF="../goldenPath/credits.html#chimp_use">specific
		    conditions for use</A>.</P>
		    <P>
		    We'd like to thank the International Chimpanzee 
		    Sequencing and Analysis Consortium, Washington 
		    University at St. Louis School of Medicine Genome Sequencing
		    Center, and the Broad Institute for providing this sequence.
		    We'd also like to acknowledge the UCSC team who worked on 
		    this release: Kate Rosenbloom, Brian Raney, Hiram Clawson, 
		    Ann Zweig, Archana Thakkapallayil, and Donna Karolchik. 
		    The chimpanzee browser annotation tracks were generated by 
		    UCSC and collaborators worldwide. </P>
		    <HR>
		    <P>
		    <FONT FACE="courier" SIZE=3><B>20 June 2006 - Genome Browser Released for Baylor v3.4 Rat Assembly: </B></FONT>
		    The UCSC Genome Bioinformatics group has released a Genome
		    Browser for the v3.4 rat (<em>Rattus 
		    norvegicus</em>) genome. 
		    <A HREF="goldenPath/newsarch.html#062006">Read more</A>.</P>
		    <P>
		    <FONT FACE="courier" SIZE=3><B>12 June 2006 - Latest <em>X. tropicalis</em> Assembly Available in Browser: </B></FONT>
		    The v.4.1 <em>Xenopus tropicalis</em> assembly is now
		    available on the UCSC Genome Browser. 
		    <A HREF="goldenPath/newsarch.html#061206">Read more</A>.</P>
		    <P>
    	            </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
