<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                          <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="https://lists.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>

                    <!-- start news -->

		    <FONT FACE="courier" SIZE="3"><B>12 Mar. 2010 - Updated Tetraodon Genome Browser Available</B></FONT>
		    <P>
                    <P>
		    We have released a Genome Browser for the latest assembly
		    of <em>Tetraodon nigroviridis</em>. The V8 assembly 
		    (March 2007, UCSC version tetNig2) was provided by
		    <A HREF="http://www.genoscope.cns.fr/" 
		    TARGET=_blank>Genoscope</A>, Evry, France, in collaboration 
		    with the 
		    <A HREF="http://www.broad.mit.edu/annotation/tetraodon/background.html"
		    TARGET=_blank>Broad Institute</A>, Cambridge, MA, USA.  
		    For detailed sequencing and assembly information, see the 
		    Genoscope <em>Tetraodon nigroviridis</em>
		    <A HREF="http://www.genoscope.cns.fr/spip/Tetraodon-nigroviridis-whole.html"
		    TARGET=_blank>assembly</A> page.
		    </P>
		    <P>
		    Downloads of the Tetraodon data and annotations may be 
		    obtained from the UCSC
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/tetNig2/" 
		    TARGET=_blank>FTP site</A> or 
		    <A HREF=http://hgdownload.cse.ucsc.edu/downloads.html#tetraodon>
		    Downloads</A> page.
		    These data have been freely provided by Genoscope for use 
		    in the UCSC Genome Browser with
		    <A HREF="goldenPath/credits#tetraodon_use"
		    TARGET=_blank>specific conditions for use</A>. For more 
		    information about the <em>Tetraodon nigroviridis</em> 
		    project, see the Genoscope
		    <A HREF="http://www.genoscope.cns.fr/spip/-Tetraodon-nigroviridis,387-.html"
		    TARGET=_blank>website</A>.
		    </P>
		    <P>
		    Many thanks to Genoscope and the Broad Institute for
		    providing these data. The tetNig2 annotation tracks were 
		    generated by UCSC and its collaborators. We'd also like to 
		    acknowledge the UCSC Tetraodon browser team, which includes 
		    several new faces: Hiram Clawson, Mark Diekhans, Brian 
		    Raney, Ann Zweig, Mary Goldman, Vanessa Swing, Antonio 
		    Coelho, Robert Kuhn, and Donna Karolchik.  
		    See the
		    <A HREF="../goldenPath/credits.html#tetraodon_credits">credits</A> 
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release.
		    </P>
<!--
		    <P>
		    <FONT FACE="courier" SIZE="3"><B>12 Mar. 2010 - California Sea Hare Genome Browser Released</B></FONT>
		    <P>
                    <P>
		    We have released a Genome Browser for the California
		    sea hare, <em>Aplysia californica</em>. This assembly
		    (UCSC version aplCal1, Broad version Aplcal2.0) was produced
		    by the <A HREF="http://www.broadinstitute.org/"
		    TARGET=_blank>Broad Institute</A>, Cambridge, MA, USA. The 
		    sea hare is the first mollusc to be sequenced, and is 
		    particularly noted for its nervous system, which is quite 
		    useful for neurobiological experimentation. 
		    </P><P>
		    This draft of the sea hare genome has a size of 
		    approximately 716 Mb with about 10X coverage. The assembly 
		    comprises 8767 scaffolds and chrM (mitochondrial DNA).
		    For more information on the assembly, see the Broad 
		    Institute
		    <A HREF="http://www.broadinstitute.org/science/projects/mammals-models/vertebrates-invertebrates/aplysia/aplysia-genome-sequencing-project"
		    TARGET=_blank>Aplysia Genome Project</A> page.
		    </P><P>
		    Bulk downloads of the sequence and annotation data are
		    available via the Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/aplCal1/">FTP server</A>
		    or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#seahare">Downloads</A>
		    page. These data have <A HREF="goldenPath/credits#seahare_use">specific
		    conditions for use</A>.
		    </P><P>
		    Many thanks to the Broad Institute for the sea hare assembly
		    data. The aplcal1 annotation tracks were generated by UCSC
		    and collaborators. See the
		    <A HREF="../goldenPath/credits.html#seahare_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release.
		    </P>
-->
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE="3"><B>24 Feb. 2010 - New Gene Name Search Functionality Released: </B></FONT>
		    UCSC has introduced a new gene search feature in the
		    Genome Browser.  
		    <A HREF="goldenPath/newsarch.html#022410">Read more</A>.
		    </P>
		    <P>
		    <FONT FACE="courier" SIZE="3"><B>25 Jan. 2010 - New Custom Track type - Binary Alignment/Map (BAM): </B></FONT>
		    We have added support for a new track type in our 
		    custom tracks feature: Binary Alignment/Map 
		    (<A HREF="goldenPath/help/bam.html">BAM</A>).
		    <A HREF="goldenPath/newsarch.html#012510">Read more</A>.
		    </P>

                        </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
