<!-- News Section ============================================- -->  
<TABLE style="background-color:#888888; width: 100%; border-width: 0px; border-spacing: 0px; padding: 0px">
    <TR><TD>        
    <TABLE style="background-color:#fffee8; width: 100%; border-width: 0px; border-spacing: 0px; padding: 0px">
        <TR><TD>        
        <TABLE style="background-color:#D9E4F8; background-image: url('images/hr.gif'); width: 100%; border: 0px">
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" title="Follow &#64;GenomeBrowser on Twitter" alt="Follow &#64;GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a>
                 <a title="Genome Browser Facebook page"
                 href="http://www.facebook.com/ucscGenomeBrowser"
                 target="_blank"><img
                 src="/images/FB-f-Logo__blue_22.jpg"
                 style="vertical-align:text-bottom; margin-left: 15px;"
                 alt="Genome Browser Facebook page"></a>
                 <a href="https://plus.google.com/107643636314131661344" rel="publisher"></a>
               </TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE style="background-color:#fffee8; width: 100%; padding: 0px">
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list. Please see our
          <a href="http://genome.ucsc.edu/blog"
          target=_blank>blog</a> for posts about Genome Browser tools, features, projects and more.
          </p>

    <!-- start news -->
<!-- Staged for later release
  <hr>
  <font color="#006666" size="4"><b>20 Apr 2016
  - Data from the Lens PatSeq Database Now Available</b></font>
  <p>
  We are pleased to announce the release of a set of tracks containing data from the
  <a href="https://www.lens.org/" target="_blank">Lens</a>
  PatSeq database in the UCSC Genome Browser. These tracks contain data from the
  <a href="https://www.lens.org/lens/bio" target="_blank">PatSeq</a>
  database mapped to the genome and then colored based on where the sequence appears in
  the patent document. Details pages contain the patent titles with links to the patents in the
  Lens database. The data from PatSeq is divided into two tracks, one includes those sequences
  that were submitted in bulk patents containing 100 or more sequences and the other containing
  all other sequences. These tracks are currently present on the human (hg19), mouse (mm10),
  and Ebola virus (eboVir3) genomes.
  </p>

  <p>
  Thanks to our collaborators at the Lens, Osmat Jefferson and Deniz Koellhofer, for providing the
  data and feedback on the visualization. Thank you to Max Haeussler and Matthew Speir, members of
  the UCSC Genome Browser team, for their efforts in creating these tracks.
  </p>
-->
  <!-- Staging for Tuesday Announcement of GTEX track
  <hr>
  <font color="#006666" size="4"><b>19 April 2016
  - New GTEx Gene Tissue Expression Track for hg19 and hg38</b></font>
  <p>
  We are excited to announce the release of a new gene expression track based on data from the NIH
  <a href="http://commonfund.nih.gov/GTEx/index" TARGET = _BLANK> Genotype-Tissue Expression (GTEx)</a> project.  
  This track displays tissue-specific expression based on RNA-Seq data of 53 tissues from 570 donors, obtained from
  the GTEx 'midpoint' data release (October 2015, V6).
  The original data for this 
  track can be found at the <a href="http://www.gtexportal.org/home" target = _blank>GTEX Portal</a> 
  hosted by the Broad Institute.
  <p>
  This track also features a new display method that extends the traditional Genome Browser 
  display &mdash; a bar graph!  Each gene is represented by a graph with colored bars depicting the median expression 
  level (in RPKM) of each tissue assayed by the GTEX project.
  Note that the graph is a fixed width, and starts at the gene transcription start.
  <p>
  <center><img src="images/gtexAnnounceExample.png"></center>
  <p>
  Notice how the blue gene-model underneath the graph shown here extends past the end of the graph? This gene
  model represents the target of the RNA-Seq experiment, and the graph displays the RPKM levels
  across many different tissues for the <em>entire</em> gene. Traditionally the 3 purple bars 
  would be associated with the intron of TCAP gene, in this case those three bars indicate that 
  the entire gene is highly expressed in cardiac and skeletal muscle tissue. Clicking on the graph 
  brings up a description page which displays a more detailed box and whisker plot of the RPKM 
  data, as well as a description of the methods used to generate the data. 
  <p>
  Credit goes to Kate Rosenbloom for implementing and Christopher Lee for quality reviewing this feature.
  <-->
  <hr>
  <p>
  <font color="#006666" size="4"><b> 15 April 2016 -
  dbSNP 146 Available for hg19 and hg38</b></font>
  <p>
  We are pleased to announce the release of four tracks derived
  from NCBI <a href="http://www.ncbi.nlm.nih.gov/SNP/">
  dbSNP</a> Build 146 data, available on the two most
  recent human assemblies GRCh37/hg19 and GRCh38/hg38.
  NCBI's dbSNP database is a collection of &quot;simple nucleotide
  polymorphisms&quot; (SNPs), which are a class of genetic variations
  that include single nucleotide polymorphisms and small
  insertions/deletions (indels). This immense database contains over
  150 million such SNPs that cover the human genome.
  </p>

  <p>
  There are four SNP tracks available as part of this release.
  One is a track containing all mappings of reference SNPs
  to the human assembly, labeled &quot;All SNPs (146)&quot;.
  The other three tracks are subsets of this track
  and show interesting and easily defined subsets of dbSNP:
  </p>

  <ul>
  <li>Common SNPs (146): uniquely mapped variants that appear
  in at least 1% of the population or are 100% non-reference.</li>
  <li>Flagged SNPs (146): uniquely mapped variants, excluding
  Common SNPs, that have been flagged by dbSNP as
  &quot;clinically associated&quot;.</li>
  <li>Mult. SNPs (146): variants that have been mapped to more
  than one genomic location. This track has been shrinking over
  the course of the last few releases as dbSNP now excludes most
  SNPs whose flanking sequences map to multiple locations in
  the genome.</li>
  </ul>

  <p>
  By default, only the Common SNPs (146) are visible;
  other tracks must be made visible using the track controls.
  You will find the other SNPs (146) tracks on both
  of GRCh37/hg19 and GRCh38/hg38 browsers in the &quot;Variation&quot; group.
  </p>
  <p>
  The tracks were produced at UCSC by Angie Hinrichs and
  Matthew Speir. We'd like to thank the dbSNP group at
  NCBI for providing access to these data.
  </p>

  <hr>
  <a name="041516"></a>
  <p>
  <font color="#006666" size="4"><b> 15 April 2016 -
  Three New Assemblies Now Available in the Genome Browser</b></font>
  <p>
  A Genome Browser is now available for the gorilla (<em>Gorilla gorilla gorilla</em>)
  assembly released December 2014 by the Wellcome Trust Sanger Institute. 
  For more information and statistics about this assembly, see the NCBI assembly record
  for <a href="http://www.ncbi.nlm.nih.gov/assembly/503571" target="_blank">gorGor4</a>.  
  There are 25 complete chromosomes alongside 40,667 scaffolds with a total size of
  3,063,362,754 bases.
  </p>

  <p>
  Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
  <a href="ftp://hgdownload.soe.ucsc.edu/goldenPath/gorGor4/">FTP server</a> or the
  <a href="http://hgdownload.soe.ucsc.edu/downloads.html#gorilla">Downloads</a> page.
  Please observe the <a href="goldenPath/credits.html#gorilla_use">conditions for use</a> when
  accessing and using these data sets.  The annotation tracks for this browser were
  generated by UCSC and collaborators worldwide.  See the <a
  href="goldenPath/credits.html#gorilla_credits">Credits</a> page for a detailed list of the
  organizations and individuals who contributed to this release.
  </p>
  - - -
  <p>
  A Genome Browser is now available for the rhesus monkey (<em>Macaca mulatta</em>)
  assembly released November 2015 by the 
  <A HREF="https://www.hgsc.bcm.edu/non-human-primates/rhesus-monkey-genome-project"
  TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center</A>. 
  For more information and statistics about this assembly, see the NCBI assembly record
  for <a href="http://www.ncbi.nlm.nih.gov/assembly/2715158" target="_blank"> Mmul_8.0.1</a>.  
  There are 23 complete chromosomes alongside 284,705 scaffolds with a total size of
  3,236,224,332 bases.
  </p>

  <p>
  Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
  <a href="ftp://hgdownload.soe.ucsc.edu/goldenPath/rheMac8/">FTP server</a> or the
  <a href="http://hgdownload.soe.ucsc.edu/downloads.html#rhesus">Downloads</a> page.
  Please observe the <a href="goldenPath/credits.html#rhesus_use">conditions for use</a> when
  accessing and using these data sets.  The annotation tracks for this browser were
  generated by UCSC and collaborators worldwide.  See the <a
  href="goldenPath/credits.html#rhesus_credits">Credits</a> page for a detailed list of the
  organizations and individuals who contributed to this release.
  </p>
  - - -
  <p>
  A Genome Browser is now available for the crab-eating macaque (<em>Macaca fascicularis</em>)
  assembly released June 2013 by the 
  <A HREF="http://genome.wustl.edu/genomes/detail/macaca-fascicularis/"
  TARGET=_blank>Genome Sequencing Center at Washington University, St. Louis</A>.
  For more information about this assembly, see
  <A HREF="http://www.ncbi.nlm.nih.gov/assembly/704988"
  TARGET="_blank">Macaca_fascicularis_5.0</A> in the NCBI Assembly database. There are 
  22 complete chromosomes alongside 7,579 scaffolds with a total size of 2,946,843,737 bases.
  </p>

  <p>
  Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
  <a href="ftp://hgdownload.soe.ucsc.edu/goldenPath/macFas5/">FTP server</a> or the
  <a href="http://hgdownload.soe.ucsc.edu/downloads.html#crab-eating_macaque">Downloads</a> page.
  Please observe the <a href="goldenPath/credits.html#crab-eating_macaque_use">conditions for use</a> when
  accessing and using these data sets.  The annotation tracks for this browser were
  generated by UCSC and collaborators worldwide.  See the <a
  href="goldenPath/credits.html#crab-eating_macaque_credits">Credits</a> page for a detailed list of the
  organizations and individuals who contributed to this release.
  </p>

  <hr>
  <font color="#006666" size="4"><b>4 Apr 2016
  - New video shows how to obtain sequence or coordinates for exons. </b></font>
  <p>
  We have released a new <A HREF =
  "http://genome.ucsc.edu/training/vids/index.html#vid08" TARGET =
  _BLANK>video</A> to our YouTube channel.  This newest installment in our effort
  to share the capabilities of the Genome Browser shows how to
  use the Table Browser to obtain either the sequences or the coordinates for
  exons in the genes from a chosen region of the genome.
  </p>

  <p>
  We also want to take this opportuinty to remind you that we offer
  on-site <A HREF = "http://genome.ucsc.edu/training/index.html" TARGET =
  _BLANK>workshops</A> to help you and your colleagues make the most out
  of the Genome Browser.
  </p>
  <hr>

<!-- start archives -->
<!--staged for future release
        <p>
	<font color="#006666" size="4"><b>20 Apr 2016
	- Data from the Lens PatSeq Database Now Available</b>: </font>
	We are pleased to announce the release of a set of tracks containing data from the
	<a href="https://www.lens.org/" target="_blank">Lens</a>
	PatSeq database in the UCSC Genome Browser.
	<a href="goldenPath/newsarch.html#042016">Read more</a>.

        <p>
        <font color="#006666" size="4"><b>19 April 2016
        - New Genotyping Tissue Expression Track for hg19 and hg38!</b></font>
        <p>
        We are excited to announce the release of a new track based on data from the 
        <a href="http://commonfund.nih.gov/GTEx/index" TARGET = _BLANK>
        Genotyping Tissue Expression Consortium</a> V6 data freeze. This track displays tissue-specific
        gene expression for a single gene based on RNA-Seq data, as well as a new display 
        method that extends the traditional Genome Browser display &mdash; a bar graph! Each bar graph 
        depicts the expression level (in RPKM) of the various tissues assayed by the GTEX Consortium, 
        and is specific to each gene, not the graph's vertical location in the browser! 
	<a href="goldenPath/newsarch.html#04192016">Read more</a>.

        <p>
	<font color="#006666" size="4"><b> 15 April 2016 -
	dbSNP 146 Available for hg19 and hg38</b>: </font>
	We are pleased to announce the release of four tracks derived
	from NCBI <a href="http://www.ncbi.nlm.nih.gov/SNP/">
	dbSNP</a> Build 146 data.
	<a href="goldenPath/newsarch.html#041516_2">Read more</a>.
	</p>

	<p>
	<font color="#006666" size="4"><b>15 April 2016
	- Genome Browsers for Gorilla, Rhesus, and Crab-Eating Macacque Now Available.</b></font>
	<a href="goldenPath/newsarch.html#041516_1">Read more</a>.
	</p>
	
	<p>
	<font color="#006666" size="4"><b>4 Apr 2016
        - New video shows how to obtain sequence or coordinates for exons. </b></font>
	<a href="goldenPath/newsarch.html#040416">Read more</a>.
	</p>

-->
	<p>
	<font color="#006666" size="4"><b>1 Apr 2016
	- Going Back to Our Roots!</b></font>
	<a href="goldenPath/newsarch.html#040116">Read more</a>.
	</p>

	<p>
	<font color="#006666" size="4"><b>21 Mar 2016
	- Genome Browsers for <i>C. elegans</i> and Brown Kiwi Now Available.</b></font>
	<a href="goldenPath/newsarch.html#032116">Read more</a>.
	</p>

            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
