<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="description" content="UCSC Genome Browser">
	<META name="keywords" content="genome, genome browser, human genome, genome assembly, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, mouse genome, rat genome, chicken genome, chimp genome, dog genome fruitfly, drosophila, fugu, yeast, ciona,  comparative genomics">
	<TITLE>UCSC Genome Browser Home</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">

</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="inc/home.topbar.html"--></TD></TR>
<!--Left Sidebar Menu----------------------------------------->  
    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="../cgi-bin/hgGateway">
			Genome Browser</A><HR>
	        <A CLASS="leftbar" HREF="../cgi-bin/hgNear">
			Gene Sorter</A><HR>
	        <A CLASS="leftbar" HREF="../cgi-bin/hgBlat">
			Blat</A><HR>
	        <A CLASS="leftbar" HREF="../cgi-bin/hgPcr">
			In Silico PCR</A><HR>
	        <A CLASS="leftbar" HREF="../cgi-bin/hgTables">
			Table Browser</A><HR>
	        <A CLASS="leftbar" HREF="../cgi-bin/pbGateway">
			Proteome Browser</A><HR>
	        <A CLASS="leftbar" HREF="util.html">
			Utilities</A><HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/downloads.html">
			Downloads</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/releaseLog.html">
			Release Log</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/customTracks/custTracks.html">
			Custom Tracks</A><HR>
	        <A CLASS="leftbar" HREF="ENCODE/">
			ENCODE</A><HR>
	        <A CLASS="leftbar" HREF="mirror.html">
			Mirrors</A><HR>
  	        <A CLASS="leftbar" HREF="http://genome-archive.cse.ucsc.edu">
			Archives</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/credits.html">
			Credits</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/pubs.html">
			Publications</A><HR>
	        <A CLASS="leftbar" HREF="http://openhelix.com/ucscmaterials.shtml"
			TARGET=_blank>
			Training</A><HR>
	        <A CLASS="leftbar" HREF="cite.html">
			Cite Us</A><HR>
	        <A CLASS="leftbar" HREF="license/">
			Licenses</A><HR>
	        <A CLASS="leftbar" HREF="jobs.html">
			Jobs</A><HR>
  	        <A CLASS="leftbar" HREF="staff.html">
			Staff</A><HR>
  	        <A CLASS="leftbar" HREF="contacts.html">
			Contact Us</A><HR>
	        <BR>
	        <IMG SRC="images/DOT.gif" ALT="" HEIGHT=700 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Introduction---------------------------------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; About the UCSC Genome Bioinformatics Site</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
		    This site contains the reference sequence and working draft
		    assemblies for a large collection of genomes. It also shows 
		    the CFTR (cystic fibrosis) region in 13 species and provides
		    a portal to the ENCODE project.  
		    <P>
		    We encourage you to explore these sequences with our tools. The Genome 
		    Browser zooms and scrolls over chromosomes, showing the work of annotators 
		    worldwide.  The Gene Sorter shows expression, homology and other 
		    information on groups of genes that can be related in many ways.  
		    Blat quickly maps your sequence to the genome.
		    The Table Browser 
		    provides convenient access to the underlying database.  
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
		        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		    </TD></TR></TABLE>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
		    <P>
		    To receive announcements of new genome 
		    assembly releases, new software features, updates and 
		    training seminars by email, subscribe to the
		    <A HREF="http://www.soe.ucsc.edu/mailman/listinfo/genome-announce" TARGET=_blank>genome-announce</A> mailing list.
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>2 March 2005  - CCDS Data Set Released</B></FONT>
		    <P>
		    The initial results from the Consensus Coding Sequence 
		    project (CCDS) were released to the public today. 
		    CCDS is a collaborative effort to identify a core set of 
		    human protein-coding regions that are consistently 
		    annotated and of high quality. </P>
		    <P>
		    The initial CCDS data set, 
		    containing nearly 15,000 transcripts, has been posted on three 
		    Internet sites: the 
		    <A HREF="http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg17">UCSC Genome Browser</A>, 
		    the <A HREF="http://www.ensembl.org/" 
		    TARGET=_blank>Ensembl Browser</A> and the 
		    <A HREF="http://www.ncbi.nlm.nih.gov/CCDS/"
		    TARGET=_blank>NCBI CCDS Database</A> website. 
		    The genes may be viewed on the UCSC hg17 (May 2004) 
		    Human Genome Browser in the CCDS annotation track 
		    located in the Genes and Gene Prediction Tracks section.
		    </P>
		    <P>
		    The CCDS gene set is built by consensus among the 
		    following collaborating organizations:
		    <UL>
		    <LI>
		    <A HREF="http://www.ebi.ac.uk/" TARGET=_blank>European 
		    Bioinformatics Institute</A> (EBI)
		    <LI>
		    <A HREF ="http://www.ncbi.nlm.nih.gov/" 
		    TARGET=_blank>National Center for Biotechnology 
		    Information</A> (NCBI)
		    <LI>
		    <A HREF="http://www.cbse.ucsc.edu/" 
		    TARGET=_blank>University of California Santa Cruz</A>
		    (UCSC)
		    <LI>
		    <A HREF="http://www.sanger.ac.uk/"
		    TARGET=_blank>Wellcome Trust Sanger Institute</A> (WTSI)
		    </UL></P>
		    <P>
		    Each gene in the CCDS set is assigned a unique identifier
		    and version number, e.g. CCDS234.1. The version number 
		    will be updated if either the CDS structure or the 
		    underlying genome sequence at that location changes. The 
		    CCDS set will be mapped forward, with identifiers 
		    preserved, when the annotations or sequence update. 
		    Changes to existing CCDS genes will be done by 
		    collaboration agreement; no single group will change the 
		    set unilaterally. CCDS identifiers are included on 
		    relevant NCBI RefSeq and Entrez Gene records.</P>
		    <P>
		    Gene structure information is obtained from both curated
		    and automated genes sources. The primary curation groups 
		    are the <A HREF="http://www.sanger.ac.uk/HGP/havana/" 
		    TARGET=_blank>Havana</A> team at the WTSI and the 
		    <A HREF="http://www.ncbi.nlm.nih.gov/RefSeq/" 
		    TARGET=_blank>RefSeq</A> annotation group at NCBI. In 
		    addition, manually curated information on chr14
		    (Genoscope) and Chr7 (WUSTL) has been included from the 
		    <A HREF="http://vega.sanger.ac.uk/" TARGET=_blank>Vega</A>
		    database. The <A HREF="http://www.ensembl.org/" 
		    TARGET=_blank>Ensembl</A> group and the RefSeq 
		    computational pipeline provide the automated data. Curated
		    information is favored over automated information; the 
		    information must be consistent between the Hinxton 
		    (Vega/Ensembl) and NCBI groups and also pass UCSC's 
		    stringent QC requirements. </P>
		    <P>
		    At a minimum, a gene must meet three quality criteria
		    to be included in the CCDS set:
		    <UL>
		    <LI>
		    The coding region has a full-length annotation with
		    an initiating ATG, a valid stop codon, and no in-frame
		    stop codons. 
		    <LI>
		    Protein can be translated from the genome without 
		    frameshifts. 
		    <LI>
		    Consensus splice-sites are used. 
		    </UL></P>
		    <P>
		    For more information on the process used to identify and 
		    evaluate CCDS genes, refer to the description page for 
		    the Genome Browser CCDS track. The number and type of 
		    quality tests performed may be expanded in the future, 
		    but includes analysis to identify putative pseudogenes, 
		    retrotransposed genes, consensus splice sites, supporting 
		    transcripts, and protein homology.
		    The completeness of the CCDS gene set should improve as 
		    the curation and automated annotation processes mature
		    and additional experimental validation is obtain for
		    weakly-supported genes.</P>
		    <P>
		    UCSC would like to thank EBI, NCBI and WTSI for their
		    collaboration in the release of these data. 
		    We would also like to acknowledge the efforts of the 
		    UCSC staff who worked on this project: Mark Diekhans, 
		    Robert Baertsch, Adam Siepel, Fan Hsu, and Robert Kuhn.
		    </P>
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>24 February 2005  - Genome Browser Training - St. Louis and New York City: </B></FONT>
		    We'd like to announce upcoming UCSC Genome Browser March
		    seminars and hands-on computer workshops in St. Louis, MO,
		    and New York, NY,  presented by OpenHelix, the 
		    bioinformatics training, software testing and consulting 
		    company. 
		    <A HREF="goldenPath/newsarch.html#022405">Read more</A>.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>23 February 2005  - New Honeybee Assembly in Browser: </B></FONT>
		    The UCSC Genome Bioinformatics group has released a
		    Genome Browser for a second honeybee assembly, Amel_2.0
		    (UCSC version apiMel2), produced by the Baylor College of 
		    Medicine Human Genome Sequencing Center.
		    <A HREF="goldenPath/newsarch.html#022305">Read more</A>.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>17 February 2005  - Cow Genome Browser Released: </B></FONT>
		    The UCSC Genome Bioinformatics Group has released a 
		    Genome Browser and Blat server for the Sep. 2004 Btau_1.0
		    draft assembly of the cow genome. This assembly (UCSC
		    version bosTau1) was provided by Baylor
		    College of Medicine Human Genome Sequencing Center in
		    Houston, TX.
		    <A HREF="goldenPath/newsarch.html#021705">Read more</A>.
    	            </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
	    	
<!--Conditions of Use Section--------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome Browser 
                    are freely available for academic, nonprofit, and personal use 
		    with the following conditions:
		    <UL>
		    <LI>Genome sequence data use restrictions are noted within the
		    species sections on the <A HREF="goldenPath/credits.html">Credits</A> 
		    page.
		    <LI>Some annotation tracks contributed by external collaborators
		    contain proprietary data that have specific use restrictions. To
		    check for restrictions associated with a particular genome assembly, 
		    review the <em>database/README.txt</em> file in the assembly's 
		    downloads directory. 
		    </UL>
		    <P>
		    The Genome Browser and Blat software are free for academic,
		    nonprofit, and personal use. A license is required for commercial use.
		    See the <A HREF="license/">Licenses</A> page for more information.
		    <P>
                    Program-driven use of this software is limited to a maximum of one 
                    hit every 15 seconds and no more than 5,000 hits per day.  
		    <P>
		    Questions or problems regarding the UCSC Genome Browser 
		    software, database, genome assemblies, or release cycles 
		    should be sent to 
		    <A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A>.
		    Messages sent to this address will be posted automatically 
		    on the genome mailing list, which is also archived on a 
		    public Web-accessible pipermail archive. NOTE: This archive may be 
		    indexed by non-UCSC sites such as Google. 
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>

        <BR>

<!--Technical Information Section----------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	        
                    <FONT SIZE="4" ><A name="Tech"></A><B>&nbsp; Technical Information About the Assembled Sequence</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
	            <LI><A HREF="goldenPath/gbdDescriptions.html"><FONT SIZE="4">Description of the annotation database</FONT></A>
	            <LI><A HREF="goldenPath/datorg.html"><FONT SIZE="4">Data organization and format</FONT></A>
	            <LI><A HREF="goldenPath/term.html"><FONT SIZE="4">Terminology</FONT></A>
                    </UL>
                    </TD>
                    <TD WIDTH=15>&nbsp;</TD>
                 </TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
    	
<!--Related Links Section--------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	            <FONT SIZE="4" ><A NAME="Other"></A><B>&nbsp; Related Links</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
<!-- Commented out til this page is updated - very out of date
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/"><FONT SIZE="4">Human Genome Project at UCSC</FONT></A><FONT SIZE="4"></FONT>
-->
	            <LI><A HREF="http://www.cbse.ucsc.edu/" TARGET=_blank><FONT SIZE="4">Center for Biomolecular Science and Engineering Home Page</FONT></A>
                    </UL>
                </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
</TD></TR></TABLE>

<!-- Web visibility------------------------------------------------>
<A HREF="goldenPath/knownGeneLists.html"><IMG SRC="/images/DOT.gif" WIDTH="1" HEIGHT="1" ALIGN="BOTTOM" BORDER="0" ALT=""></A>
</BODY></HTML>

