<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="description" content="UCSC Genome Browser">
	<META name="keywords" content="genome, genome browser, human genome, genome assembly, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, mouse genome, rat genome, chicken genome, chimp genome, dog genome fruitfly, drosophila, fugu, yeast, ciona,  comparative genomics">
	<TITLE>UCSC Genome Browser Home</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">

</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<!--Top Menu Bar---------------------------------------------->  
    <TR><TD COLSPAN=2 HEIGHT=40>
    <TABLE BGCOLOR="000000" CELLPADDING="1" CELLSPACING="1" WIDTH="99%" HEIGHT="27">
        <TR BGCOLOR="2636D1"><TD VALIGN="middle">
        <TABLE BORDER=0 CELLSPACING=0 CELLPADDING=0 BGCOLOR="2636D1" HEIGHT="24">
    	    <TR><TD VALIGN="middle"><FONT color="#89A1DE">&nbsp; 
 	    <A HREF="/cgi-bin/hgGateway" class="topbar">
			Genomes</a> &nbsp; - &nbsp; 
 	    <A HREF="/cgi-bin/hgNear" class="topbar">
			Gene Sorter</A> &nbsp; - &nbsp; 
	    <A HREF="/cgi-bin/hgBlat" class="topbar">
			Blat</a> &nbsp; - &nbsp; 
	    <A HREF="/cgi-bin/hgPcr" class="topbar">
			PCR</a> &nbsp; - &nbsp; 
	    <A HREF="/cgi-bin/hgTables" class="topbar">
			Tables</a> &nbsp; - &nbsp; 
	    <A HREF="/cgi-bin/pbGateway" class="topbar">
			Proteome</a> &nbsp; - &nbsp; 
	    <A HREF="/FAQ/" class="topbar">
			FAQ</a> &nbsp; - &nbsp; 
	    <A HREF="goldenPath/help/hgTracksHelp.html" class="topbar">
			Help</a> &nbsp; </FONT></TD>
        </TR></TABLE>
    </TD></TR></TABLE>
    </TD></TR>

<!--Left Sidebar Menu----------------------------------------->  
    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="/cgi-bin/hgGateway">
			Genome Browser</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgNear">
			Gene Sorter</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgBlat">
			Blat</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgPcr">
			In Silico PCR</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgTables">
			Table Browser</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/pbGateway">
			Proteome Browser</A><HR>
	        <A CLASS="leftbar" HREF="util.html">
			Utilities</A><HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/downloads.html">
			Downloads</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/releaseLog.html">
			Release Log</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/customTracks/custTracks.html">
			Custom Tracks</A><HR>
	        <A CLASS="leftbar" HREF="/encode/">
			ENCODE</A><HR>
	        <A CLASS="leftbar" HREF="mirror.html">
			Mirrors</A><HR>
  	        <A CLASS="leftbar" HREF="http://genome-archive.cse.ucsc.edu">
			Archives</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/credits.html">
			Credits</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/pubs.html">
			Publications</A><HR>
	        <A CLASS="leftbar" HREF="http://openhelix.com/ucscmaterials.shtml"
			TARGET=_blank>
			Training</A><HR>
	        <A CLASS="leftbar" HREF="cite.html">
			Cite Us</A><HR>
	        <A CLASS="leftbar" HREF="license/">
			Licenses</A><HR>
	        <A CLASS="leftbar" HREF="jobs.html">
			Jobs</A><HR>
  	        <A CLASS="leftbar" HREF="staff.html">
			Staff</A><HR>
  	        <A CLASS="leftbar" HREF="contacts.html">
			Contact Us</A><HR>
	        <BR>
	        <IMG SRC="images/DOT.gif" ALT="" HEIGHT=700 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Introduction---------------------------------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; About the UCSC Genome Bioinformatics Site</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
		    This site contains the reference sequence and working draft
		    assemblies for a large collection of genomes. It also shows 
		    the CFTR (cystic fibrosis) region in 13 species and provides
		    a portal to the ENCODE project.  
		    <P>
		    We encourage you to explore these sequences with our tools. The Genome 
		    Browser zooms and scrolls over chromosomes, showing the work of annotators 
		    worldwide.  The Gene Sorter shows expression, homology and other 
		    information on groups of genes that can be related in many ways.  
		    Blat quickly maps your sequence to the genome.
		    The Table Browser 
		    provides convenient access to the underlying database.  
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
		        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		    </TD></TR></TABLE>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
		    <P>
		    To receive announcements of new genome 
		    assembly releases, new software features, updates and 
		    training seminars by email, subscribe to the
		    <A HREF="http://www.soe.ucsc.edu/mailman/listinfo/genome-announce" TARGET=_blank>genome-announce</A> mailing list.
		    <P>
<HR>
		    <FONT FACE="courier" SIZE=3><B>18 January 2005  - Opossum Assembly in Genome Browser</B></FONT>
		    <P>
		    The opossum (<em>Monodelphis domestica</em>) is the latest
		    vertebrate addition to the UCSC Genome Browser genome
		    collection. The Oct. 2004 preliminary assembly -- UCSC 
		    version monDom1 -- was sequenced and assembled by 
		    <A HREF="http://www.broad.mit.edu/seq/"
		    TARGET=_blank>The Broad Institute</A>, Cambridge, MA, USA.
		    <P>
		    This preliminary draft was assembled from 33,507,069 
		    placed reads, and consists of 109,065 contigs containing 
		    a total of 3,492,108,230 bases.
		    There are 3,563,247,205 total bases in the 19,348 
		    scaffolds in the assembly, with 71,138,975 bases in
		    the gaps between contigs within the scaffolds. The
		    scaffolds range in size from 1,000 bases to 22,286,839 
		    bases.
		    <P>
		    The monDom1 sequence and annotation data can be 
		    downloaded from the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/monDom1/">FTP server</A>
		    or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#opossum">downloads</A>
		    page. Please review the 
		    <A HREF="/goldenPath/credits.html#opossum_use"
		    TARGET=_blank>guidelines</A> for using these data.
		    <P>
		    Many thanks to The Broad Institute for providing these
		    data. The UCSC opossum Genome Browser
		    was produced by Hiram Clawson, Brian Raney, Robert Kuhn,
		    Ali Sultan-Qurraie and Donna Karolchik. The 
		    initial set of annotation tracks was generated by the 
		    UCSC Genome Bioinformatics Group. See the
		    <A HREF="/goldenPath/credits.html#opossum_credits">credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to the release of
		    this browser.
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>23 December 2004  - Updated Zebrafish Assembly Now Available:</B></FONT>
		    The latest zebrafish assembly -- Zv4 (UCSC version 
		    danRer2, June 2004) -- is now available in the UCSC
		    Genome Browser and Blat server. The Zv4 assembly was 
		    produced by 
		    The Wellcome Trust Sanger Institute in collaboration with 
		    the Max Planck Institute for Developmental Biology in 
		    Tuebingen, Germany, and the Netherlands Institute for 
		    Developmental Biology (Hubrecht Laboratory), Utrecht, 
		    The Netherlands.
		    <A HREF="/goldenPath/newsarch.html#122304">Read more</A>.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>22 December 2004  - Browser Released for Honey Bee (<em>A. mellifera</em>):</B></FONT>
		    The UCSC Genome Bioinformatics group has released a
		    Genome Browser for the Amel_1.2 honey bee assembly
		    produced by the Baylor College of Medicine Human Genome
		    Sequencing Center.
		    <A HREF="/goldenPath/newsarch.html#122204">Read more</A>.
    	            </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
	    	
<!--Conditions of Use Section--------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome Browser 
                    are freely available for academic, nonprofit, and personal use 
		    with the following conditions:
		    <UL>
		    <LI>Genome sequence data use restrictions are noted within the
		    species sections on the <A HREF="/goldenPath/credits.html">Credits</A> 
		    page.
		    <LI>Some annotation tracks contributed by external collaborators
		    contain proprietary data that have specific use restrictions. To
		    check for restrictions associated with a particular genome assembly, 
		    review the <em>database/README.txt</em> file in the assembly's 
		    downloads directory. 
		    </UL>
		    <P>
		    The Genome Browser and Blat software are free for academic,
		    nonprofit, and personal use. A license is required for commercial use.
		    See the <A HREF="/license/">Licenses</A> page for more information.
		    <P>
                    Program-driven use of this software is limited to a maximum of one 
                    hit every 15 seconds and no more than 5,000 hits per day.  
		    <P>
		    Questions or problems regarding the UCSC Genome Browser 
		    software, database, genome assemblies, or release cycles 
		    should be sent to 
		    <A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A>.
		    Messages sent to this address will be posted automatically 
		    on the genome mailing list, which is also archived on a 
		    public Web-accessible pipermail archive. NOTE: This archive may be 
		    indexed by non-UCSC sites such as Google. 
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>

        <BR>

<!--Technical Information Section----------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	        
                    <FONT SIZE="4" ><A name="Tech"></A><B>&nbsp; Technical Information About the Assembled Sequence</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
	            <LI><A HREF="goldenPath/gbdDescriptions.html"><FONT SIZE="4">Description of the annotation database</FONT></A>
	            <LI><A HREF="goldenPath/datorg.html"><FONT SIZE="4">Data organization and format</FONT></A>
	            <LI><A HREF="goldenPath/term.html"><FONT SIZE="4">Terminology</FONT></A>
                    </UL>
                    </TD>
                    <TD WIDTH=15>&nbsp;</TD>
                 </TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
    	
<!--Related Links Section--------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	            <FONT SIZE="4" ><A NAME="Other"></A><B>&nbsp; Related Links</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
<!-- Commented out til this page is updated - very out of date
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/"><FONT SIZE="4">Human Genome Project at UCSC</FONT></A><FONT SIZE="4"></FONT>
-->
	            <LI><A HREF="http://www.cbse.ucsc.edu/" TARGET=_blank><FONT SIZE="4">Center for Biomolecular Science and Engineering Home Page</FONT></A>
                    </UL>
                </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
</TD></TR></TABLE>

<!-- Web visibility------------------------------------------------>
<A HREF="/goldenPath/knownGeneLists.html"><IMG SRC="/images/DOT.gif" WIDTH="1" HEIGHT="1" ALIGN="BOTTOM" BORDER="0" ALT=""></A>
</BODY></HTML>

