<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="description" content="UCSC Genome Browser">
	<META name="keywords" content="genome, genome browser, genome assembly, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA">
	<TITLE>UCSC Genome Bioinformatics Home</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">

<SCRIPT LANGUAGE="JavaScript">
<!--

var theDb = 'human';

function setDb()
{
  theDb = document.dbForm.genome.options[document.dbForm.genome.selectedIndex].value;
}

function initDb()
{
  theDb = document.dbForm.genome.options[document.dbForm.genome.selectedIndex].value;
  var queryString = document.location.search.substring(1, document.location.search.length);
  var dbIndex = queryString.indexOf('org=');
  var ampIndex = queryString.indexOf('&');
  
  if (dbIndex >= 0)
  {
    if (ampIndex > 0) {
      theDb = queryString.substring(dbIndex + 4, ampIndex);
  } else {
      theDb = queryString.substring(dbIndex + 4);
  }   

    for (var items = 0; items < document.dbForm.genome.options.length; items++)
    {
      if(sameOrganism(theDb, document.dbForm.genome.options[items].value))
      {
        document.dbForm.genome.options[items].selected = true;
      }
    }     
  }
}

function sameOrganism(db1, db2)
{
  if (db1.toLowerCase() == db2.toLowerCase())
  {
    return true;
  }
  else
  {
    return false;
  }
}

function dbGo(theUrl)
{
document.location = theUrl + 'org=' + theDb;
}

//-->
</SCRIPT>


</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF" onLoad="initDb()">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg"></TH></TR>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
	<table bgcolor="000000" cellpadding="1" cellspacing="1" width="99%" height="27"><tr bgcolor="2636D1"><td valign="middle">
	<table BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24">
		<TR><TD VALIGN="middle"><font color="#89A1DE">&nbsp; 
			<a href="javascript:dbGo('/cgi-bin/hgGateway?');" class="topbar">
				Genome Browser</a> &nbsp; - &nbsp; 
			<A HREF="javascript:dbGo('/cgi-bin/hgBlat?command=start&');" class="topbar">
				Blat Search</a> &nbsp; - &nbsp; 
			<A HREF="javascript:dbGo('/cgi-bin/hgText?command=start&');" class="topbar">
				Table Browser</a> &nbsp; - &nbsp; 
			<A HREF="FAQ.html" class="topbar">
				FAQ</a> &nbsp; - &nbsp; 
			<A HREF="goldenPath/help/hgTracksHelp.html" class="topbar">
				User Guide</a> &nbsp; </font></TD>
    	</TR></TABLE>
	</TD></TR></TABLE>
</TD></TR>

<TR HEIGHT="100%" VALIGN="top">
  <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
  	<!-- outer table is to align leftbar & content boxes at top of page -->
  	<TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0 MARGIN=0>
	<TR><TD><IMG SRC="images/DOT.gif" HEIGHT="7"></TD></TR>
	<TR HEIGHT="100%"><TD HEIGHT="100%">
		<!--outer table is for black border-->
		<TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
   	 	<TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
		<TR HEIGHT="100%">
			<TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%"> 
				<FORM NAME="dbForm">
			<table cellpadding="" cellspacing="0" margins="0" border="0"><tr><td valign="top">
				<FONT CLASS="leftbar">Genome</FONT>
			</td></tr><tr valign="bottom"><td valign="bottom">
					<SELECT NAME="genome" onChange="setDb();">
						<OPTION VALUE="human">Human
						<OPTION VALUE="mouse">Mouse 
						<OPTION VALUE="rat">Rat 
						<OPTION VALUE="sars">SARS 
					</SELECT>
			</td></tr></table><hr>
			<A CLASS="leftbar" HREF="javascript:dbGo('/cgi-bin/hgGateway?');">
				Browser</A><hr>
			<A CLASS="leftbar" HREF="javascript:dbGo('/cgi-bin/hgBlat?command=start&');">
				Blat</A><hr>
			<A CLASS="leftbar" HREF="javascript:dbGo('/cgi-bin/hgText?');">
				Tables</A><hr>
			<A CLASS="leftbar" HREF="downloads.html">
				Downloads</A><hr>
			<A CLASS="leftbar" HREF="goldenPath/customTracks/custTracks.html">
				Custom Tracks</A><hr>
		  	<A CLASS="leftbar" HREF="mirror.html">
				Mirrors</A><hr>
  			<A CLASS="leftbar" HREF="http://genome-archive.cse.ucsc.edu">
				Archives</A><hr>
			<A CLASS="leftbar" HREF="goldenPath/credits.html">
				Credits</A><hr>
			<A CLASS="leftbar" HREF="cite.html">
				Cite Us</A><hr>
			<A CLASS="leftbar" HREF="jobs.html">
				Jobs</A><hr>
  			<A CLASS="leftbar" HREF="bestlinks.html">
				Links</A><hr>
  			<A CLASS="leftbar" HREF="contacts.html">
				Contact Us</A><hr>
                                </FORM>
			<BR>
			<IMG SRC="images/DOT.gif" HEIGHT=1100 WIDTH=1>
		</TD></TR></TABLE>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
</TD>

<!--Content Tables-->  
  <TD VALIGN="top"><TABLE CELLPADDING=5 BORDER=0><TR><TD>
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<A NAME="About"></A><FONT SIZE="4"><b>&nbsp; About the UCSC Genome
	Bioinformatics Site</FONT>
	</TD></TR></TABLE>
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD WIDTH=10></TD><TD>
	<P>This site contains the reference sequence for the human
	genome and the working drafts for the mouse and rat genomes. 
	<P>
	The human reference sequence - produced by the International Human
	Genome Sequencing Consortium - covers about 99 percent of 
	the gene-containing regions in the genome, and has been sequenced to an accuracy of 
	99.99 percent. The missing portions are essentially contained in approximately 
	400 defined gaps in the euchromatic region representing DNA regions with unusual structures 
	that can't be reliably sequenced using current technology. The reference
	sequence is considered to be "finished", a technical term indicating that 
	the sequence is highly accurate (with fewer than one error per 10,000 
	bases) and highly contiguous (with the only remaining gaps corresponding 
	to regions whose sequence cannot be reliably resolved with current 
	technology). Future work on the reference sequence will focus on improving
	accuracy and reducing gaps in the sequence. 

	<P>The mouse genome (Mus musculus) was produced by the Mouse Genome Sequencing Consortium and is
	largely the result of a full genome shotgun assembly. The current assembly
	includes 705 megabases of finished sequence. While there are 
	many gaps in the sequence, the order and orientation of the contigs on 
	either side of a gap are known in almost all cases. We estimate
	that 90-96% of the mouse genome is present in the
	assembly. 

	<P>
	The rat genome (Rattus norvegicus) has been provided by the Rat Genome Sequencing Consortium.
	The sequence was assembled using a hybrid approach that combines the 
	clone by clone and whole genome shotgun methods. The current assembly
	provides approximately 6.5-fold coverage of the rat genome.
	
	<P>
	We encourage you to explore the reference sequence and working drafts 
	with the UCSC Genome Browser, which displays the work of many annotators 
	worldwide. The Genome Browser and the data it displays are freely 
	available for 
	academic, nonprofit, and personal use with a few exceptions. Check the
	README.txt file in the download directory of a particular
	assembly to view the use restrictions
	specific to that release. Program-driven use is limited to a 
	maximum of one hit every 10 seconds and no more than 5,000 hits per day.  
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100% BORDER=0><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; News </FONT></TD>
			<TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD WIDTH=10></TD><TD>

	<P><FONT FACE="courier" SIZE=3><B>Apr. 14, 2003</B></FONT><BR>
	The International Human Genome Sequencing Consortium today announced the
	successful completion of the Human Genome Project. The most
	significant outcome of this project is the reference sequence of
	the human genome. The sequencing of
	the 3 billion letters of DNA in the human genome - which many consider to 
	be one of the most ambitious scientific undertakings in history - was
	completed 2 years ahead of schedule and at substantially less cost than
	original estimates. The reference sequence will serve as a new foundation
	for research in the fields of medicine and human biology.
	<P>
	In conjunction with this announcement, the UCSC Genome Bioinformatics 
	group is proud to release a genome browser and BLAT server on the 
	reference sequence (NCBI Build 33), along with bulk downloads of the 
	sequence and annotation data. The initial browser provides a preliminary 
	set of annotations that will be expanded in coming weeks. Bulk downloads
	of the data are available via FTP at 
	<A HREF="ftp://genome.ucsc.edu/goldenPath/10april2003">ftp://genome.ucsc.edu/goldenPath/10april2003</A> 
	or through the <A HREF="downloads.html">Downloads</A> link on this page. 
	We
	recommend that FTP be used rather than HTML for the download
	of large or multiple files.
	<P>
	The reference sequence covers about 99 percent of the human genome's gene-containing
	regions, and has been sequenced to an accuracy of 99.99 percent. The
	missing portions are essentially contained in less than 400 defined gaps
	that represent DNA regions with unusual structures that can't be reliably
	sequenced using current technology. The average DNA letter now lies within
	a stretch of approximately 27,332,000 base pairs of uninterrupted 
	sequence! 
	<P>
	Chromosomal sequences for this release were assembled by the 
	International Human Genome Sequencing Consortium sequencing centers and verified by 
	NCBI and UCSC. In some cases, sequence joins between adjacent clones 
	could not be computationally validated, e.g. due to polymorphisms in the 
	overlapping sequence. In situations like this, supporting evidence for 
	the join has been provided by the sequencing center responsible for that 
	particular chromosome. The <A HREF="goldenPath/certTable.html">
	Non-standard Join Certificates</A> table displays this information. The 
	annotations on the UCSC website have been provided
	by UCSC and collaborators worldwide. See the 
	<A HREF="goldenPath/credits.html">Credits</A> page for a detailed
	list of organizations and individuals who contributed to the success of 
	this release.

	<P>
	We'd like to congratulate the many people worldwide who have worked on 
	the Human Genome Project for this landmark achievement. We'd also like to acknowledge the UCSC
	Genome Browser project team who worked many long hours to ensure that
	the genome browser and sequence data were released on time for this
	announcement: David Haussler, Jim Kent, Terry Furey, Matt Schwartz, Heather Trumbower, 
	Angie Hinrichs, Fan Hsu, 
	Donna Karolchik, Jorge Garcia, Patrick Gavin, Chuck Sugnet, Yontao Lu, Mark Diekhans, Ryan Weber, Robert Baertsch, Krishna Roskin, and the many other students in the UCSC Genome Bioinformatics group.

    </TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<BR>
		
<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4" ><a name="Tech"></a><b>&nbsp; Technical Information About the Assembled Sequence</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD>
<UL>
	<LI><A HREF="goldenPath/certTable.html"><FONT SIZE="4">Non-standard Join Certificates (April 2003 Human Genome)</FONT></A>
	<LI><A HREF="goldenPath/gbdDescriptions.html"><FONT SIZE="4">Description of the annotation database</FONT></A>
	<LI><A HREF="goldenPath/datorg.html"><FONT SIZE="4">Data organization and format</FONT></A>
	<LI><A HREF="goldenPath/term.html"><FONT SIZE="4">Terminology</FONT></A>
	<LI><A HREF="goldenPath/stats.html"><FONT SIZE="4">Summary statistics</FONT></A>
	<LI><A HREF="goldenPath/chromReports"><FONT SIZE="4">Chromosome Reports</FONT></A>
         <LI><A HREF="goldenPath/mapPlots/"><FONT SIZE="4">Genome Map Plots</FONT></A>
</UL>
</TD><TD WIDTH=15>&nbsp;</TD></TR></TABLE>
	<BR></TD></TR></TABLE>
</TD></TR></TABLE>
	
<BR>
	
<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4" ><a name="Other"></a><b>&nbsp; Other Links</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD>
<UL>
	<LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/"><FONT SIZE="4">Human Genome Project at UCSC</FONT></A><FONT SIZE="4"></FONT>
	<LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/"><FONT SIZE="4">Center for Biomolecular Science and Engineering Home Page</FONT></A>
</UL>
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
</TD></TR></TABLE>

</TD></TR></TABLE>
</TD></TR></TABLE>
</BODY></HTML>

