<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                          <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="http://www.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>

		    <P>
		    <FONT FACE="courier" SIZE="3"><B>1 December 2008 - Updated Horse Genome Browser Available</B></FONT><BR>
		    <P>
		    The Sep. 2007 EquCab2 release of the horse genome (Equus 
		    caballus) is now available in the UCSC Genome Browser. This 
		    assembly, UCSC version equCab2, was produced by the 
		    <A HREF="http://www.broad.mit.edu/" TARGET=_blank>Broad 
		    Institute</A>. </P>
		    <P>
		    The horse draft genome has been sequenced to 6.8X coverage. 
		    Approximately 84% of the sequence has been anchored to 
		    chromosomes, which include autosomes 1-31 and sex 
		    chromosome X. Unanchored contigs that could not be 
		    localized to a chromosome have been concatenated into the 
		    virtual chromosome &quot;chrUn&quot;, separated by gaps of 
		    1,000 bp. The mitochondrial sequence is also available in
		    the Genome Browser as the virtual chromosome 
		    &quot;chrM&quot;. 
		    <P>
		    For more details about the current assembly, see 
		    the Broad Institute 
		    <A HREF="http://www.broad.mit.edu/mammals/horse/"
		    TARGET=_blank>Horse Genome Project</A> page. </P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available from the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/equCab2/">FTP server</A> 
		    or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#horse">Downloads</A> 
		    page. These data have 
		    <A HREF="goldenPath/credits.html#horse_use">specific 
		    conditions for use</A>. </P>
		    <P>
                    The UCSC Horse Genome Browser was produced by Larry Meyer,
		    Hiram Clawson, Pauline Fujita, Brooke Rhead, and Donna 
		    Karolchik. See the 
		    <A HREF="goldenPath/credits.html#horse_credits">Credits</A> 
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release. </P>
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE="3"><B> 7 November 2008 - Medaka Genome Browser Updated</B></FONT><BR>
		    <P>
		    We have updated the initial UCSC Medaka Genome Browser 
		    (oryLat1) to correct an error with chrUn in which the gap 
		    relationships between the contigs within their 
		    ultracontigs were incorrect. 
                    <A HREF="goldenPath/newsarch.html#110708">Read more</A>.</P>
		    <P>
		    <FONT FACE="courier" SIZE="3"><B>17 October 2008 - Quality Assurance Engineer Job Opening:</B></FONT>
		    The UCSC Genome Browser group has reopened a search for a 
		    Quality Assurance engineer: Biological Database Testing and
		    User Support Technician. Applicants should have a strong 
		    background in biology and computation, and excellent skills 
		    in scientific writing and editing. For details see the 
		    <A HREF="http://www.cbse.ucsc.edu/jobs/jobs_staff.shtml#bdt"
		    TARGET=_blank>job listing</A>.
                        </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
