<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Bioinformatics Home</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">

<SCRIPT LANGUAGE="JavaScript">
<!--

var theDb = 'hg12';

function setDb()
{
  theDb = document.dbForm.genome.options[document.dbForm.genome.selectedIndex].value;
}

function initDb()
{
  theDb = document.dbForm.genome.options[document.dbForm.genome.selectedIndex].value;
  var queryString = document.location.search.substring(1, document.location.search.length);
  var dbIndex = queryString.indexOf('db=');
  if (dbIndex >= 0)
  {
    theDb = queryString.substring(dbIndex + 3);
    for (var items = 0; items < document.dbForm.genome.options.length; items++)
    {
      if(sameOrganism(theDb, document.dbForm.genome.options[items].value))
      {
        document.dbForm.genome.options[items].selected = true;
      }
    }     
  }
}

function sameOrganism(db1, db2)
{
  if (db1.substring(0, 2) == db2.substring(0, 2))
  {
    return true;
  }
  else
  {
    return false;
  }
}

function dbGo(theUrl)
{
document.location = theUrl + 'db=' + theDb;
}

//-->
</SCRIPT>


</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF" onLoad="initDb()">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg"></TH></TR>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
	<table bgcolor="000000" cellpadding="1" cellspacing="1" width="99%" height="27"><tr bgcolor="2636D1"><td valign="middle">
	<table BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24">
		<TR><TD VALIGN="middle"><font color="#89A1DE">&nbsp; 
			<a href="javascript:dbGo('/cgi-bin/hgGateway?');" class="topbar">
				Genome Browser</a> &nbsp; - &nbsp; 
			<A HREF="javascript:dbGo('/cgi-bin/hgBlat?command=start&');" class="topbar">
				Blat Search</a> &nbsp; - &nbsp; 
			<A HREF="FAQ.html" class="topbar">
				FAQ</a> &nbsp; - &nbsp; 
			<A HREF="goldenPath/help/hgTracksHelp.html" class="topbar">
				User Guide</a> &nbsp; </font></TD>
    	</TR></TABLE>
	</TD></TR></TABLE>
</TD></TR>

<TR HEIGHT="100%" VALIGN="top">
  <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
  	<!-- outer table is to align leftbar & content boxes at top of page -->
  	<TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0 MARGIN=0>
	<TR><TD><IMG SRC="images/DOT.gif" HEIGHT="7"></TD></TR>
	<TR HEIGHT="100%"><TD HEIGHT="100%">
		<!--outer table is for black border-->
		<TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
   	 	<TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
		<TR HEIGHT="100%">
			<TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%"> 
				<FORM NAME="dbForm">
			<table cellpadding="" cellspacing="0" margins="0" border="0"><tr><td valign="top">
				<FONT CLASS="leftbar">Organism</FONT>
			</td></tr><tr valign="bottom"><td valign="bottom">
					<SELECT NAME="genome" onChange="setDb();">
						<OPTION VALUE="hg12">Human
						<OPTION VALUE="mm2">Mouse 
					</SELECT>
			</td></tr></table><hr>
			<A CLASS="leftbar" HREF="javascript:dbGo('/cgi-bin/hgGateway?');">
				Browser</A><hr>
			<A CLASS="leftbar" HREF="javascript:dbGo('/cgi-bin/hgBlat?command=start&');">
				Blat</A><hr>
			<A CLASS="leftbar" HREF="javascript:dbGo('/cgi-bin/hgText?');">
				Tables</A><hr>
			<A CLASS="leftbar" HREF="downloads.html">
				Downloads</A><hr>
		  	<A CLASS="leftbar" HREF="mirror.html">
				Mirrors</A><hr>
  			<A CLASS="leftbar" HREF="http://genome-archive.cse.ucsc.edu">
				Archives</A><hr>
			<A CLASS="leftbar" HREF="goldenPath/credits.html">
				Credits</A><hr>
			<A CLASS="leftbar" HREF="cite.html">
				Cite Us</A><hr>
			<A CLASS="leftbar" HREF="goldenPath/pubs.html">
				Pubs</A><hr>
			<A CLASS="leftbar" HREF="jobs.html">
				Jobs</A><hr>
  			<A CLASS="leftbar" HREF="bestlinks.html">
				Links</A><hr>
                                </FORM>
			<BR>
			<IMG SRC="images/DOT.gif" HEIGHT=1100 WIDTH=1>
		</TD></TR></TABLE>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
</TD>

<!--Content Tables-->  
  <TD VALIGN="top"><TABLE CELLPADDING=5 BORDER=0><TR><TD>
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<A NAME="About"></A><FONT SIZE="4"><b>&nbsp; About the UCSC Genome
	Bioinformatics Site</FONT>
	</TD></TR></TABLE>
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD WIDTH=10></TD><TD>
		<P>This site contains working drafts for the human genome and
		the mouse genome. The human genome is over 90% complete. Approximately 
		85% of the sequence is in a highly accurate 'finished' state.  The
		other 15% is merely 'draft' quality.  Some care must be taken
		when interpreting draft regions, but these are still very 
		useful to the working scientist.  
		<P>The mouse genome is
		largely the result of a full genome shotgun assembly. While there are 
		many gaps in the sequence, the order and orientation of the contigs on 
		either side of a gap are known in almost all cases. We estimate
		that 90-96% of the mouse genome is present in the
		assembly. We encourage you to explore the
		working drafts with the genome browser, which displays
		the work of many annotators worldwide.
		<P>The Genome Browser and the data it displays are freely available for 
		academic, nonprofit, and personal use with a few exceptions. Check the
		README.txt file in the download directory of a particular
		assembly to view the use restrictions
		specific to that release. Interactive use is limited to a 
		maximum of 10,000 hits/day or 10 hits/minute. 
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100% BORDER=0><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; News </FONT></TD>
			<TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD WIDTH=10></TD><TD>

	<P><FONT FACE="courier" SIZE=2>Sep. 12, 2002</FONT><BR>
	We've corrected a problem with the Human June 2002 (hg12) cytoBand annotation 
	track that affected chromosome Y. The clones on this chromosome were
	erroneously pushed from the q-arm onto the p-arm, creating some confusion. The
	currently available version of the cytoBand data on this website contains this 
	correction.

	<P><FONT FACE="courier" SIZE=2>Sep. 12, 2002</FONT><BR>
	<I>Nature Genetics</I> has just published <A
	HREF="http://www.nature.com/cgi-taf/DynaPage.taf?file=/ng/journal/v32/n1s/index.html"
	TARGET=_blank>User's Guide to the Human Genome</A>, a
	hands-on tutorial for using genome browsers as web tools for browsing and
	analyzing data from the Human Genome Project and other sequencing efforts. The 
	3 browsers featured in the tutorial include the UCSC Genome Browser, NCBI's 
	Map Viewer, and the Ensembl Genome Browser. The guide is organized around a 
	collection of step-by-step solutions to 13 typical research questions,
	and serves as a nice supplement to the documentation materials
	available on the UCSC Genome Browser web site. The guide is accessible as a 
	link off the <I>Nature Genetics</I> home page at 
	<A HREF="http://www.nature.com/ng/" TARGET=_blank>http://www.nature.com.ng</A>. 
	
	<P><FONT FACE="courier" SIZE=2>Aug. 15, 2002</FONT><BR>
	We've just released several new annotation tracks/tables for the human genome. 
	On the June 2002 assembly, we've added Gene Bounds, 
	UniGene, CpG Islands, Nonhuman mRNA & EST, SNPs, NC160, and GNF Ratio.
	On the April 2002 assembly we've added a Fgenesh++ Genes track, and have also
	updated the RepeatMasker track. The Dec. 2001 release now includes a Sanger 22 
	track. 

	<P>On the mouse genome, we have 2 new tracks for the Feb. 2002 assembly:
	TIGR Gene Index and RNA Genes.
	
	<P><FONT FACE="courier" SIZE=2>Aug. 6, 2002</FONT><BR>
	We have fixed an error with six of the chr3 contigs in the bigZips/contigAgp.zip 
	file. The following .agp files were corrected: NT_005684.agp, NT_005663.agp, 
	NT_022554.agp, NT_022459.agp, NT_006031.agp, and NT_022419.agp. The
	chr3.agp file in bigZips/chromAgp.zip was also modified. This change does not 
	affect the .gl files, the .fa files, the 
	lift files, or the annotations. Alignments made on the previous version
	of chr3 are still good. Updated versions of the contigAgp.zip and
	chromAgp.zip files were
	uploaded to our site today. You can download the new versions via ftp at <A 
	HREF="ftp://genome.cse.ucsc.edu/goldenPath/28jun2002" TARGET=_blank>
        ftp://genome.cse.ucsc.edu/goldenPath/28jun2002</A> or via the Downloads
	link in the left sidebar.

	<P><FONT FACE="courier" SIZE=2>Aug. 2, 2002</FONT><BR>
	The problems with the June 2002 Build 30 (hg12) RepeatMasker track have been 
	resolved. The new RepeatMasker track, along with regenerated Fish Blat
	and Genscan tracks, are now available in the Browser and 
	through our
	Downloads link. We've also added a few new annotation tracks for the
	June 2002 release, and will be adding more over the next 2 weeks.

	<P>The latest Genome Browser has 2 new features. We've added filter
	functionality to the Table Browser, accessible via the Filter Fields
	button on the Table Browser main page. Also, some of the Dec. 2001 human genome tracks (eg. RefSeq Genes) now have a
	Comparative Sequence link from the details page that shows annotated
	codons and translated protein with alignment to the mouse genome.
	
	<P><FONT FACE="courier" SIZE=2>July 26, 2002</FONT><BR>
	Bulk downloads of the June 2002 Build 30 human genome assembly (hg12) are now
	available from
	<A HREF="ftp://genome.cse.ucsc.edu/goldenPath/28jun2002" TARGET=_blank>
	ftp://genome.cse.ucsc.edu/goldenPath/28jun2002</A>. You can also 
	access the data via the Downloads link in the left sidebar. This initial
	release of the annotation database download contains a limited set of
	tables. Additional files will be available for download next week.

	<P><FONT FACE="courier" SIZE=2>July 24, 2002</FONT><BR>
	The BLAT server and the coordinates conversion feature for human genome 
	assembly Build 30 (hg12) are now functional.

	<P><FONT FACE="courier" SIZE=2>July 23, 2002</FONT><BR>
	We're pleased to announce the pre-release of a browser for 
	human genome assembly Build 30 from NCBI (UCSC version hg12). This assembly was 
	produced at NCBI based on sequence information submitted into GenBank as of
	June 28, 2002. Build 30 release notes and statistics will soon be
	available from the <A
	HREF="http://www.ncbi.nlm.nih.gov/genome/guide/human/" TARGET=_blank> NCBI web 
	site</A>.  
	<P>
	Build 30 is an excellent high-quality assembly. It contains nearly 87%
 	finished sequence, and 94%-97% coverage. The sequence coverage of this build
	is much higher than in previous releases, and there is a high level of
	correspondence between the sequence and the map. Currently, the human genome
	project appears to be on track to achieve the goal of finishing at least 95%
	of the human genome (using Bermuda standards) by April 2003.
	<P>
	UCSC has generated a set of high-level comparisons of the Build 30 draft
	sequence against various types of information (STS maps, BAC end pairs, and
	clone overlaps), accessible from the <A 
	HREF="http://hgwdev.cse.ucsc.edu/goldenPath/chromReports/24Jun2002/"
	TARGET=_blank>Chromosome Reports</A> and <A 
	HREF="http://hgwdev.cse.ucsc.edu/goldenPath/mapPlots/" TARGET=_blank>Genome 
	Map Plots</A> links in the "Technical Information about the Assembled 
	Sequence" section below.
	<P>
	A Blat server for Build 30 is not yet available, but should be
	accessible from this site later this week. Data for the mitochondrial genome 
	and several more annotation tracks will be posted for this release as they 
	become available.  Bulk downloads of the hg12 data should be available from 
	this site in a few days. 

    </TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<BR>
		
<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4" ><a name="Tech"></a><b>&nbsp; Technical Information About the Assembled Sequence</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD>
<UL>
	<LI><A HREF="goldenPath/term.html"><FONT SIZE="4">Terminology</FONT></A>
	<LI><A HREF="goldenPath/datorg.html"><FONT SIZE="4">Data organization and format</FONT></A>
	<LI><A HREF="goldenPath/gbdDescriptions.html"><FONT SIZE="4">Description of the annotation database</FONT></A>
	<LI><A HREF="goldenPath/stats.html"><FONT SIZE="4">Summary statistics</FONT></A>
	<LI><A HREF="goldenPath/chromReports"><FONT SIZE="4">Chromosome Reports</FONT></A>
         <LI><A HREF="goldenPath/mapPlots/"><FONT SIZE="4">Genome Map Plots</FONT></A>
<!-- CHEAT: not closing <UL> tag tightens bottom spacing reliably in unix & windows! -->
</TD><TD WIDTH=15>&nbsp;</TD></TR></TABLE>
	<BR></TD></TR></TABLE>
</TD></TR></TABLE>
	
<BR>
	
<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4" ><a name="Other"></a><b>&nbsp; Other Links</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD>
<UL>
	<LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/"><FONT SIZE="4">Human Genome Project at UCSC</FONT></A><FONT SIZE="4"></FONT>
	<LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/"><FONT SIZE="4">Center for Biomolecular Science and Engineering Home Page</FONT></A>
<!-- CHEAT: not closing <UL> tag tightens bottom spacing reliably in unix & windows! -->
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
</TD></TR></TABLE>

</TD></TR></TABLE>
</TD></TR></TABLE>
</BODY></HTML>

