<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                         <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD>
                  <TD>
                  <P>
                  To receive announcements of new genome 
                  assembly releases, new software features, updates and 
                  training seminars by email, subscribe to the
                  <A HREF="https://lists.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                  TARGET=_blank>genome-announce</A> mailing list.</P>

            
            <!-- start news -->
            	<FONT FACE="courier" SIZE="3"><B>05 March 2012 - New Mouse 
                Browser Available</B></FONT>
                <P>
                We have released the latest Genome Browser for the December 2011
		Mouse genome assembly produced by the 
		<a href="http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/mouse/" TARGET=_blank>
		Mouse Genome Reference Consortium</a> (NCBI Build 38, UCSC version mm10).	
		This version, which includes approximately 2.6 Gb of sequence, 
		is considered to be "essentially complete". The assembly includes 
		chromosomes 1-19, X, Y, M (mitochondrial DNA) and chr*_random 
		(unlocalized) and chrUn_* (unplaced clone contigs).
		Note that the UCSC mm10 database contains only the 
		reference strain C57BL/6J.
		<P>					
		Bulk downloads of the sequence and annotation data are available
		via the Genome Browser <a href="ftp://hgdownload.cse.ucsc.edu/goldenPath/mm10/" 
		TARGET=_blank>FTP server</a> or the 
		<a href="http://hgdownload.cse.ucsc.edu/downloads.html#mouse" TARGET=_blank>Downloads</a>
		page. The Mouse browser annotation tracks were generated by UCSC
		and collaborators worldwide. See the 
		<a href="http://hgwdev.cse.ucsc.edu/goldenPath/credits.html#mouse_credits" 
		TARGET=_blank>Credits</a> page for a detailed list of the 
		organizations and individuals who contributed to the success of this release.				                   <P>
                <HR>

            <!-- more news -->
		 <FONT FACE="courier" SIZE="3"><B>10 February 2012 
                 - dbSNP 135 Available for hg19</B></FONT>
                 <P>We are pleased to announce the release of four tracks derived
                 from dbSNP build 135, available on the human assembly (GRCh37/hg19).
                 dbSNP build 135 is available at NCBI. The new tracks contain
                 additional annotation data not included in previous dbSNP tracks,
                 with corresponding coloring and filtering options in the Genome
                 Browser.
                 </P>
                 <P>
                 As for dbSNP build 132, there are four tracks in this release.
                 One is a track containing all mappings of reference SNPs to the
                 human assembly, labeled "All SNPs (135)". The other three tracks
                 are subsets of this track and show interesting and easily defined
                 subsets of dbSNP:
                 </P>

                 <UL>
                 <LI>Common SNPs (135): uniquely mapped variants that appear in
                 at least 1% of the population
                 <LI>Flagged SNPs (135): uniquely mapped variants, excluding
                 Common SNPs, that have been flagged by dbSNP as "clinically
                 associated"
                 <LI>Mult. SNPs (135): variants that have been mapped to more
                 than one genomic location
                 </UL>

                 <P>By default, only the Common SNPs (135) are visible; other
                 tracks must be made visible using the track controls.
                 </P>

                 <P>You will find the four SNPs (135) tracks on the Human Feb.
                 2009 (GRCh37/hg19) browser in the "Variation and Repeats" group.
                 </P>

                 </P> Many thanks to dbSNP at NCBI for the data, and to
                 Mary-Claire King at the University of Washington for helpful
                 comments that motivated us to enhance our dbSNP annotations. The
                 tracks were produced at UCSC by Angie Hinrichs and Brooke Rhead.
                 </P>   
		 <HR>
		  
            <!-- start archives -->
		   <FONT FACE="courier" SIZE="3"><B>2 February 2012 - New UCSC 
                   Genes Track Released for GRCh37/hg19: </B></FONT>
		   We're happy to announce the release of an updated UCSC Genes
                   track for the GRCh37/hg19 human Genome Browser. 
		   <A HREF="goldenPath/newsarch.html#020212">Read more</A>.
		   <P>
                   <FONT FACE="courier" SIZE="3"><B>23 January 2012 - Wallaby 
		   Browser Available: </B></FONT>
                   We have released a Genome Browser for the September 2009
                   genome assembly of the Tammar Wallaby, <em>Macropus 
                   eugenii</em> (TWGS version Meug_1.1, UCSC version macEug2).
                   <A HREF="goldenPath/newsarch.html#230112">Read more</A>.
		   <P>
		   <FONT FACE="courier" SIZE="3"><B>3 January 2012 - Roadmap 
		   Epigenomics Now Available through Data Hub at Washington 
		   University: </B></FONT>
		   We are pleased to announce the release of the Roadmap 
		   Epigenomics data on the UCSC Genome Browser through our 
		   Data Hub function. 
                   <A HREF="goldenPath/newsarch.html#030112">Read more</A>.
		   <P>

==> <a href="goldenPath/newsarch.html">News Archives</a>

                    </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
