<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                          <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="http://www.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>

		    <P>
		    <FONT FACE="courier" SIZE="3"><B>17 September 2008 - Seven Assemblies Archived </B></FONT><BR>
		    <P> We have archived the following seven genome assemblies:
        	    <P> &nbsp;  bosTau1
        	    &nbsp; ce1
        	    &nbsp; danRer2
        	    &nbsp; hg15
        	    &nbsp; mm5
        	    &nbsp; mm6
        	    &nbsp; rn2
		    </P>
        	    <P>	They are being moved from the main Genome Browser website to our
        		<A HREF ="http://genome-archive.cse.ucsc.edu">archive server</A> 
		        where they still be available for browsing.  Each organism (cow, 
		        <EM>C. elegans</EM>, human,
		        zebrafish, mouse and rat) has at least two later assemblies remaining 
		        on the main site.  We will no longer be supporting blat 
        		servers for these assemblies and we will turn off the daily GenBank
		        updates.  It will still be possible to lift coordinates
        		to and from these assemblies on the main site.  
		    </P>

                    <P>
		    <HR>
		    <FONT FACE="courier" SIZE="3"><B>15 September 2008 - Request for Input on ENCODE Data Release Plan</B></FONT><BR>
		    <P>
		    The National Human Genome Research Institute (NHGRI) 
		    has updated the data release plan for the 
		    <A HREF="http://www.genome.gov/ENCODE"
		    TARGET=_blank>ENCODE</A> and model organism ENCODE 
		    (<A HREF="http://www.genome.gov/modENCODE"
		    TARGET=_blank>modENCODE</A>) projects. 
		    Because the plan is intended to take 
		    into account the needs and responsibilities of the resource 
		    users, NHGRI is soliciting feedback on the proposal from the
		    research community before the plan is finalized. To 
		    facilitate this review, UCSC has made the
		    plan available to the community 
		    <A HREF="http://genome.ucsc.edu/encodeDCC/ENCODE_Consortia_Data_Release_Policy_2008-09-11.doc">here</A> on our ENCODE website.
		    Please send comments directly to 
		    <A HREF="mailto:&#69;&#110;&#99;o&#100;e&#64;&#109;&#97;i&#108;.&#110;&#105;&#104;.g&#111;&#118;">&#69;&#110;&#99;o&#100;e&#64;&#109;&#97;i&#108;.&#110;&#105;&#104;.g&#111;&#118;</A>
		    <!-- above address is Encode at mail.nih.gov -->
		    by Wednesday, October 15, 2008.  </P>

		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE="3"><B>3 September 2008 - Updated UCSC Genes Set Released for hg18</B></FONT><BR>
		    <P>
		    We have released an updated UCSC Genes data set on the 
		    latest human assembly (hg18, March 2006). The new annotation
		    can be found on the hg18 Genome Browser in the "Genes and 
		    Gene Predictions" track group. The previous set of UCSC 
		    Genes on the hg18 assembly has been renamed as Old UCSC 
		    Genes in the same track group.  </P>
		    <P>
		    The UCSC Genes track shows gene predictions based on data 
		    from RefSeq, Genbank, CCDS and UniProt. This is a moderately
		    conservative set of predictions, requiring the support of 
		    one GenBank RNA sequence plus at least one additional line 
		    of evidence. The RefSeq RNAs are an exception to this, 
		    requiring no additional evidence. The track includes both 
		    protein-coding and putative non-coding transcripts. Some of 
		    these non-coding transcripts may actually code for protein, 
		    but the evidence for the associated protein is weak. 
		    Compared to RefSeq, this gene set has generally about 10% 
		    more protein-coding genes, approximately five times as many 		    putative non-coding genes, and about twice as many splice 
		    variants. </P>
		    <P>
		    The latest version of UCSC Genes explicitly incorporates 
		    CCDS proteins in addition to the evidence included in 
		    previous versions, as well as additional data from GenBank, 
		    RefSeq, and UniProt. Although we have made every effort to 
		    preserve UCSC Gene IDs across versions, some gene IDs have 
		    changed in the new data set. In general, the ID number has 
		    been incremented if the new version of a gene is larger 
		    than the previous version, and a completely new ID has been 
		    assigned in those instances where a splice site changed or 
		    the protein-coding region shrank. However, searching on an 
		    old discontinued ID from the previous gene set will take you
		    to the corresponding gene in the updated track.  
		    Alternatively, you can check the kg3ToKg4 table directly to 
		    find corresponding gene IDs.</P>
		    <P>
		    The UCSC Genes set was produced by Jim Kent. Thanks to Ann 
		    Zweig for heading up the testing of the new data set, and 
		    to the entire Genome Browser QA team for coordinating its 
		    release on our public site.  </P>
<!--
		    <P>
		    <FONT FACE="courier" SIZE="3"><B>18 August 2008 - Updated Horse Genome Browser Available</B></FONT><BR>
		    <P>
		    The Sep. 2007 EquCab2 release of the horse genome (Equus 
		    caballus) is now available in the UCSC Genome Browser. This 
		    assembly, UCSC version equCab2, was produced by the 
		    <A HREF="http://www.broad.mit.edu/" TARGET=_blank>Broad 
		    Institute</A>. </P>
		    <P>
		    The horse draft genome has been sequenced to 6.8X coverage. 
		    Approximately 84% of the sequence has been anchored to 
		    chromosomes, which include autosomes 1-31 and sex 
		    chromosome X. Unanchored contigs that could not be 
		    localized to a chromosome have been concatenated into the 
		    virtual chromosome &quot;chrUn&quot;, separated by gaps of 
		    1,000 bp. The mitochondrial sequence is also available in
		    the Genome Browser as the virtual chromosome 
		    &quot;chrM&quot;. 
		    <P>
		    The UCSC Genome Browser team has identified an apparent
		    assembly error in the Broad EquCab2 release: a duplicated
		    25.9 Mb region from chr6 is appended to the distal end of 
		    chr27 (visible as scaffold_34 in the Scaffolds track). We 
		    have notified the Broad Institute of this problem, which 
		    likely will be corrected in their next assembly release. 
		    For more details about the current assembly, see 
		    the Broad Institute 
		    <A HREF="http://www.broad.mit.edu/mammals/horse/"
		    TARGET=_blank>Horse Genome Project</A> page. </P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available from the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/equCab2/">FTP server</A> 
		    or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#horse">Downloads</A> 
		    page. These data have 
		    <A HREF="goldenPath/credits.html#horse_use">specific 
		    conditions for use</A>. </P>
		    <P>
                    The UCSC Horse Genome Browser was produced by Larry Meyer,
		    Hiram Clawson, Pauline Fujita, Brooke Rhead, and Donna 
		    Karolchik. See the 
		    <A HREF="goldenPath/credits.html#horse_credits">Credits</A> 
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release. </P>
-->
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE="3"><B>27 August 2008 - Zebra Finch Genome Browser Released: </B></FONT>
		    We have added the Jul. 2008 release of the zebra finch genome 
		    (<em>Taeniopygia guttata</em>) to our collection of 
		    vertebrate genome browsers. 
                    <A HREF="goldenPath/newsarch.html#082708">Read more</A>.</P>
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE="3"><B>22 July 2008 - Updated <em>C. elegans</em> Browser Released: </B></FONT>
		    Culminating our recent updates of the nematode browsers on
		    our site, we've released a Genome Browser for the latest
		    <em>C. elegans</em> assembly based on sequence version WS190
		    deposited into WormBase in May 2008. 
                    <A HREF="goldenPath/newsarch.html#072208">Read more</A>.</P>
                        </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
