<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                          <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="https://lists.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>

                    <!-- start news -->

		    <FONT FACE="courier" SIZE="3"><B>19 Mar. 2010 - California Sea Hare Genome Browser Released</B></FONT>
		    <P>
                    <P>
		    We have released a Genome Browser for the California
		    sea hare, <em>Aplysia californica</em>. This assembly
		    (UCSC version aplCal1, Broad version Aplcal2.0) was produced
		    by the <A HREF="http://www.broadinstitute.org/"
		    TARGET=_blank>Broad Institute</A>, Cambridge, MA, USA. The 
		    sea hare is the first mollusc to be sequenced, and is 
		    particularly noted for its nervous system, which is quite 
		    useful for neurobiological experimentation. 
		    </P><P>
		    This draft of the sea hare genome has a size of 
		    approximately 712 Mb with nearly 10X coverage. The assembly 
		    comprises 8766 scaffolds and chrM (mitochondrial DNA).
		    For more information on the assembly, see the Broad 
		    Institute
		    <A HREF="http://www.broadinstitute.org/science/projects/mammals-models/vertebrates-invertebrates/aplysia/aplysia-genome-sequencing-project"
		    TARGET=_blank>Aplysia Genome Project</A> page.
		    </P><P>
		    Bulk downloads of the sequence and annotation data are
		    available via the Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/aplCal1/">FTP server</A>
		    or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#seahare">Downloads</A>
		    page. These data have <A HREF="goldenPath/credits#seahare_use">specific
		    conditions for use</A>.
		    </P><P>
		    Many thanks to the Broad Institute for the sea hare assembly
		    data. The aplcal1 annotation tracks were generated by UCSC
		    and collaborators. We'd also like to acknowledge the UCSC Sea Hare browser
		    team: Galt Barber, Brian Raney, Hiram Clawson, Katrina Learned, Antonio Coelho, and Donna Karolchik. See the
		    <A HREF="../goldenPath/credits.html#seahare_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release.
		    </P>
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE="3"><B>12 Mar. 2010 - Updated Tetraodon Genome Browser Available: </B></FONT>
		    We have released the newest version of the 
		    <em>Tetraodon nigroviridis</em> Genome Browser.
		    <A HREF="goldenPath/newsarch.html#031210">Read more</A>.
		    </P>

		    <P>
		    
		    <FONT FACE="courier" SIZE="3"><B>24 Feb. 2010 - New Gene Name Search Functionality Released: </B></FONT>
		    UCSC has introduced a new gene search feature in the
		    Genome Browser.  
		    <A HREF="goldenPath/newsarch.html#022410">Read more</A>.
		    </P>
		    <P>
		    <FONT FACE="courier" SIZE="3"><B>25 Jan. 2010 - New Custom Track type - Binary Alignment/Map (BAM): </B></FONT>
		    We have added support for a new track type in our 
		    custom tracks feature: Binary Alignment/Map 
		    (<A HREF="goldenPath/help/bam.html">BAM</A>).
		    <A HREF="goldenPath/newsarch.html#012510">Read more</A>.
		    </P>

                        </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
