<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="description" content="UCSC Genome Browser">
	<META name="keywords" content="genome, genome browser, human genome, genome assembly, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, mouse genome, rat genome, chicken genome, chimp genome, dog genome fruitfly, drosophila, fugu, yeast, ciona,  comparative genomics">
	<TITLE>UCSC Genome Browser Home</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">

</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="inc/home.topbar.html"--></TD></TR>
<!--Left Sidebar Menu----------------------------------------->  
    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="cgi-bin/hgGateway">
			Genome Browser</A><HR>
	        <A CLASS="leftbar" HREF="ENCODE/">
			ENCODE</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/hgBlat">
			Blat</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/hgTables">
			Table Browser</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/hgNear">
			Gene Sorter</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/hgPcr">
			In Silico PCR</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/hgVisiGene">
			VisiGene</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/pbGateway">
			Proteome Browser</A><HR>
	        <A CLASS="leftbar" HREF="util.html">
			Utilities</A><HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/downloads.html">
			Downloads</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/releaseLog.html">
			Release Log</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/customTracks/custTracks.html">
			Custom Tracks</A><HR>
	        <A CLASS="leftbar" HREF="mirror.html">
			Mirrors</A><HR>
  	        <A CLASS="leftbar" HREF="http://genome-archive.cse.ucsc.edu">
			Archives</A><HR>
	        <A CLASS="leftbar" HREF="http://openhelix.com/ucscmaterials.shtml"
			TARGET=_blank>
			Training</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/credits.html">
			Credits</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/pubs.html">
			Publications</A><HR>
	        <A CLASS="leftbar" HREF="cite.html">
			Cite Us</A><HR>
	        <A CLASS="leftbar" HREF="license/">
			Licenses</A><HR>
	        <A CLASS="leftbar" HREF="http://www.cbse.ucsc.edu/jobs/" TARGET=_blank>
			Jobs</A><HR>
  	        <A CLASS="leftbar" HREF="staff.html">
			Staff</A><HR>
  	        <A CLASS="leftbar" HREF="contacts.html">
			Contact Us</A><HR>
	        <BR>
	        <IMG SRC="images/DOT.gif" ALT="" HEIGHT=700 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Introduction---------------------------------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; About the UCSC Genome Bioinformatics Site</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
		    This site contains the reference sequence and working draft
		    assemblies for a large collection of genomes. It also 
		    provides a portal to the ENCODE project.  
		    <P>
		    We encourage you to explore these sequences with our tools.
		    The Genome Browser zooms and scrolls over chromosomes, 
		    showing the work of annotators worldwide. The Gene Sorter 
		    shows expression, homology and other information on groups 
		    of genes that can be related in many ways.  
		    Blat quickly maps your sequence to the genome.  The Table 
		    Browser provides convenient access to the underlying 
		    database. VisiGene lets you browse through a large 
		    collection of <em>in situ</em> mouse and frog images to
		    examine expression patterns.</P>
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
		        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		    </TD></TR></TABLE>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
		    <P>
		    To receive announcements of new genome 
		    assembly releases, new software features, updates and 
		    training seminars by email, subscribe to the
		    <A HREF="http://www.soe.ucsc.edu/mailman/listinfo/genome-announce" TARGET=_blank>genome-announce</A> mailing list.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>6 March 2006 - Purple Sea Urchin Genome Assembly Available in Genome Browser</B></FONT>
		    <P>
		    The April 2005 release of the Purple Sea Urchin genome 
		    (<em>Strongylocentrotus purpuratus</em>) is now available
		    in the UCSC Genome Browser. This assembly, UCSC version
		    strPur1, was produced by the Baylor College of Medicine 
		    Human Genome Sequencing Center (BCM HGSC) and corresponds 
		    to their Spur_0.5 whole genome shotgun assembly. </P>
		    <P>
		    This release was assembled from whole genome shotgun reads 
		    using the Atlas genome assembly system at the BCM HGSC. 
		    Several whole genome shotgun libraries, with inserts of 2-6 
		    kb, were used to produce the data. About 7 million reads 
		    were assembled, representing about 800 Mb of sequence and 
		    about 6x coverage of the (clonable) sea urchin genome. 
		    Highly repeated sequences were assembled separately into 
		    reptigs and merged into the genome assembly. Sequences from 
		    BAC clones were omitted from this assembly and will be 
		    placed in a subsequent version of the draft sequence. </P>
		    <P>
		    This is a draft sequence and may contain errors; therefore, 
		    users should exercise caution. Typical errors in draft 
		    genome sequences include misassemblies of repeated 
		    sequences, collapses of repeated regions, and unmerged 
		    overlaps (<em>e.g.</em> due to polymorphisms) creating 
		    artificial duplications. However, base accuracy in contigs 
		    (contiguous blocks of sequence) is usually very high with 
		    most errors near the ends of contigs.</P>
		    <P>
		    More assembly details can be found in the 
		    <A HREF="ftp://ftp.hgsc.bcm.tmc.edu/pub/data/Spurpuratus/fasta/Spur20050415/READMEApril2005.txt" 
		    TARGET=_blank>Spur_0.5 README file</A> and on the BCM HGSC 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/seaurchin/"
		    TARGET=_blank>Sea Urchin Genome Project</A> web page.</P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/strPur1">FTP server</A>
		    or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#urchin">Downloads</A> 
		    page. These data have 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" 
		    TARGET=_blank>specific conditions for use</A>. The strPur1 
		    annotation tracks were generated by UCSC and collaborators 
		    worldwide. See the 
		    <A HREF="goldenPath/credits.html#urchin_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release. </P>
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>27 February 2006 - Upcoming Genome Browser Seminars in Texas, Florida and Washington, DC:</B></FONT>
		    The UCSC Bioinformatics Group announces four regional 
		    seminars and hands-on computer workshops on the UCSC 
		    Genome Browser, presented by 
		    <A HREF="http://www.openhelix.com/" 
		    TARGET=_blank>OpenHelix.</A>
		    <A HREF="goldenPath/newsarch.html#022706">Read more</A>.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>17 February 2006 - Zebrafish Browser Updated: </B></FONT>
		    The latest zebrafish assembly -- Zv5 (UCSC version 
		    danRer3, May 2005) -- is now available in the UCSC
		    Genome Browser and Blat server. 
		    <A HREF="goldenPath/newsarch.html#021706">Read more</A>.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>9 February 2006 - New Rhesus Macaque Browser Released: </B></FONT>
		    The latest rhesus macaque (<em>Macaca mulatta</em>) draft
		    assembly--v.1.0, Mmul_051212--is now available 
		    in the UCSC Genome Browser. 
		    <A HREF="goldenPath/newsarch.html#020906">Read more</A>.
		    </P>
    	            </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
	    	
<!--Conditions of Use Section--------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome Browser 
                    are freely available for academic, nonprofit, and personal use 
		    with the following conditions:
		    <UL>
		    <LI>Genome sequence data use restrictions are noted within the
		    species sections on the <A HREF="goldenPath/credits.html">Credits</A> 
		    page.
		    <LI>Some annotation tracks contributed by external collaborators
		    contain proprietary data that have specific use restrictions. To
		    check for restrictions associated with a particular genome assembly, 
		    review the <em>database/README.txt</em> file in the assembly's 
		    downloads directory. 
		    </UL>
		    <P>
		    The Genome Browser and Blat software are free for academic,
		    nonprofit, and personal use. A license is required for commercial use.
		    See the <A HREF="license/">Licenses</A> page for more information.
		    <P>
                    Program-driven use of this software is limited to a maximum of one 
                    hit every 15 seconds and no more than 5,000 hits per day.  
		    <P>
		    For assistance with questions or problems regarding the UCSC Genome 
                    Browser software, database, genome assemblies, or release cycles, 
		    see the <A HREF="http://genome.ucsc.edu/FAQ/FAQmaillist">FAQ</A>.
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>

        <BR>

<!--Technical Information Section----------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	        
                    <FONT SIZE="4" ><A name="Tech"></A><B>&nbsp; Technical Information About the Assembled Sequence</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
	            <LI><A HREF="goldenPath/gbdDescriptions.html"><FONT SIZE="4">Description of the annotation database</FONT></A>
	            <LI><A HREF="goldenPath/datorg.html"><FONT SIZE="4">Data organization and format</FONT></A>
	            <LI><A HREF="goldenPath/term.html"><FONT SIZE="4">Terminology</FONT></A>
                    </UL>
                    </TD>
                    <TD WIDTH=15>&nbsp;</TD>
                 </TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
    	
<!--Related Links Section--------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	            <FONT SIZE="4" ><A NAME="Other"></A><B>&nbsp; Related Links</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
<!-- Commented out til this page is updated - very out of date
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/"><FONT SIZE="4">Human Genome Project at UCSC</FONT></A><FONT SIZE="4"></FONT>
-->
	            <LI><A HREF="http://www.cbse.ucsc.edu/" TARGET=_blank><FONT SIZE="4">Center for Biomolecular Science and Engineering Home Page</FONT></A>
                    </UL>
                </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
</TD></TR></TABLE>

<!-- Web visibility------------------------------------------------>
<A HREF="knownGeneLists.html"><IMG SRC="images/DOT.gif" WIDTH="1" HEIGHT="1" ALIGN="BOTTOM" BORDER="0" ALT=""></A>
</BODY></HTML>

