<!-- News Section ============================================- -->  
<TABLE style="background-color:#888888; width: 100%; border-width: 0px; border-spacing: 0px; padding: 0px">
    <TR><TD>        
    <TABLE style="background-color:#fffee8; width: 100%; border-width: 0px; border-spacing: 0px; padding: 0px">
        <TR><TD>        
        <TABLE style="background-color:#D9E4F8; background-image: url('images/hr.gif'); width: 100%; border: 0px">
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" title="Follow &#64;GenomeBrowser on Twitter" alt="Follow &#64;GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a>
                 <a title="Genome Browser Facebook page"
                 href="http://www.facebook.com/ucscGenomeBrowser"
                 target="_blank"><img
                 src="/images/FB-f-Logo__blue_22.jpg"
                 style="vertical-align:text-bottom; margin-left: 15px;"
                 alt="Genome Browser Facebook page"></a>
                 <a href="https://plus.google.com/107643636314131661344" rel="publisher"></a>
               </TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE style="background-color:#fffee8; width: 100%; padding: 0px">
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list. Please see our
          <a href="http://genome.ucsc.edu/blog"
          target=_blank>blog</a> for posts about Genome Browser tools, features, projects and more.
          </p>
   
     <hr>      
    <!-- start news -->

  <font color="#006666" size="4"><b>6 June 2016 - New video:  Exon-only mode</b></font>
  </P>
  We are pleased to announce the release to our YouTube channel of a new
  training <A HREF = "https://www.youtube.com/watch?v=mNEJBOkf_Gk" TARGET = _BLANK>video</A>,
  in which we demonstrate the recently released Multi-Region feature in
  exon-only mode.  This feature is useful for those studying gene expression or
  undertaking whole-exome sequencing.  By removing intronic and intergenic regions,
  exon-only mode shows the portion (sometimes as little as a few percent) of the
  genome most useful for these analyses.
  </P>
  <P>
  This video is
  <A HREF ="http://hgwdev-kuhn.cse.ucsc.edu/training/vids/#vid09" TARGET = _BLANK>indexed</A> 
  to allow easy access to relevant sections and comes with a transcript for the
  hard-of-hearing.
  </P>
  <P>
  Thanks to Robert Kuhn and David Gibson for production.
  </P>
  <hr>

  <font color="#006666" size="4"><b>1 June 2016 - New Density Graph Feature!</b></font>
  <p>
  The Genome Browser is proud to announce a new &quot;density graph&quot; feature for
  a variety of tracks, including BAM custom tracks, that converts data into a summary display.</p>
  <p>
  The density graph feature enables certain track types to be displayed as a bar graph where
  the height is proportional to the number of reads mapped to each genomic position.
  Through dynamic calculation of items in the current window, this feature plots a line
  similar to a wiggle graph that can be customized with a number of
  <a href="goldenPath/help/hgWiggleTrackHelp.html"
  target="_blank">graph-based configuration options</a> such as drawing indicator lines,
  smoothing plots, adjusting graph height and vertical range, and switching from bars to
  points. Please note that the feature is best displayed with Display mode set to full
  and that the default is auto-scale data to view. Please note that using this
  feature on a <a href="goldenPath/help/gbib.html#GbibMirroring"
  target="_blank">virtual machine Genome Browser in a Box (GBiB)</a> may benefit from
  increasing memory.</p> 
  <p>
  To turn on the feature click into the track controls for certain track types,
  such as a BAM custom track, and check the box next to &quot;Display data as a
  density graph&quot; to see graph configurations and then click &quot;Submit&quot;.
  To see the new density graph feature in use, please see this example
  <a href="cgi-bin/hgTracks?hgS_doLoadUrl=submit&hgS_loadUrlName=http://genome.ucsc.edu/goldenPath/help/examples/sessions/session.hg19.densityGraph&hgct_customText=track%20type=bam%20name=bamExample%20visibility=full%20bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bamExample.bam%20doWiggle=1"
  target="_blank">session</a>.
  <p>
  Thanks to Brian Raney for the implementation of this feature.</p>
  <hr>

  <font color="#006666" size="4"><b>10 May 2016 - New Gateway Page!</b></font>
  <p>
  The UCSC Genome Browser team is proud to announce a newly redesigned 
  <a href="../cgi-bin/hgGateway" target="_blank">Genome Browser Gateway</a> page.
  The Gateway retains its original functionality as a central access point for all genome assemblies
  available on our site, while sporting several helpful new features and updates:</p>
  <ul>
  <li>Autocomplete searching for any genome browser, genome version, or public hub</li>
  <li>Species browsing through a scrollable visual
  &quot;tree&quot; menu based on phylogenetic order</li>
  <li>Quick access shortcuts to popular browsers</li>
  <li>New style and color scheme</li>
  </ul>
  <p>
  Please see our <a href="https://youtu.be/5zc9sqd4SD4"
  target="_blank">Gateway video</a> for an introduction to these new features. 
  </p>
  <p>
  The new Gateway addresses the need for streamlined access to the rapidly 
  growing number of genome assemblies available on our public site. 
  As an alternative to scrolling through long drop-down menus of genome assemblies, you can
  now simply type in assembly search terms for autocomplete recognition, visually scroll
  through our collection, or click a button to access our most popular species.
  </p>
  <p>
  Note that a few browser utilities that were previously accessed through links and buttons on the 
  Gateway page have been moved to the top menu bar:
  </p>
  <ul>
  <li>Browser reset: Genome Browser > Reset All User Settings</li>
  <li>Track search: Genome Browser > Track Search</li>
  <li>Add custom tracks: My Data > Custom Tracks</li>
  <li>Track hubs: My Data > Track Hubs</li>
  <li>Configure tracks and display: Genome Browser > Configure</li>
  </ul>
  <p>
  We would like to acknowledge the following people for their contributions
  to the new Gateway: original art by Jeff West Design, concept by Robert Kuhn,
  software design by Angie Hinrichs, and quality assurance by Cath Tyner.</p>
  <hr>

<!-- start archives -->
<!--staged for future release
       <p>
        <font color="#006666" size="4"><b>6 June 2016 - New video:  Exon-only mode</b></font>
        We are proud to announce a new <A HREF = "">video</A> demonstrating the
        exon-only feature.  This feature is useful in gene-expresson studies and
        for those doing whole-exome sequencing.
        <a href="goldenPath/newsarch.html#060616">Read more</a>.
       </p>

       <p>
        <font color="#006666" size="4"><b>1 June 2016 - New Density Graph Feature!</b></font>
        The Genome Browser is proud to announce a new &quot;density graph&quot; feature for
        a variety of tracks, including BAM custom tracks, that converts data into a summary display.
        <a href="goldenPath/newsarch.html#060116">Read more</a>.
       </p>

        <p>
        <font color="#006666" size="4"><b>10 May 2016 - New Gateway Page!</b>: </font>
        The UCSC Genome Browser team is proud to announce a newly redesigned
        <a href="../../cgi-bin/hgGateway" target="_blank">Genome Browser Gateway page</a>.
        Please see our <a href="https://youtu.be/5zc9sqd4SD4"
        target="_blank">Gateway video</a> for an introduction to these new features.
        <a href="goldenPath/newsarch.html#051016">Read more</a>.
       </p>
-->
        <p>
       <font color="#006666" size="4"><b>22 Apr 2016
       - Data from the Lens PatSeq Database Now Available</b>: </font>
       We are pleased to announce the release of a set of tracks containing data from the
       <a href="https://www.lens.org/" target="_blank">Lens</a>
       PatSeq database in the UCSC Genome Browser.
       <a href="goldenPath/newsarch.html#042216">Read more</a>.
       </p>
            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
