<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="description" content="UCSC Genome Browser">
	<META name="keywords" content="genome, genome browser, human genome, genome assembly, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, mouse genome, rat genome, chicken genome, chimp genome, dog genome fruitfly, drosophila, fugu, yeast, ciona,  comparative genomics">
	<TITLE>UCSC Genome Browser Home</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">

</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<!--HOTLINKS BAR-->
    <TR><TD COLSPAN=2 HEIGHT=40>
    <TABLE BGCOLOR="000000" CELLPADDING="1" CELLSPACING="1" WIDTH="99%" HEIGHT="27">
        <TR BGCOLOR="2636D1"><TD VALIGN="middle">
        <TABLE BORDER=0 CELLSPACING=0 CELLPADDING=0 BGCOLOR="2636D1" HEIGHT="24">
    	    <TR><TD VALIGN="middle"><FONT color="#89A1DE">&nbsp; 
 	    <A HREF="/cgi-bin/hgGateway" class="topbar">
			Genomes</a> &nbsp; - &nbsp; 
 	    <A HREF="/cgi-bin/hgNear" class="topbar">
			Gene Sorter</A> &nbsp; - &nbsp; 
	    <A HREF="/cgi-bin/hgBlat" class="topbar">
			Blat</a> &nbsp; - &nbsp; 
	    <A HREF="/cgi-bin/hgPcr" class="topbar">
			PCR</a> &nbsp; - &nbsp; 
	    <A HREF="/cgi-bin/hgText" class="topbar">
			Tables</a> &nbsp; - &nbsp; 
	    <A HREF="/FAQ/" class="topbar">
			FAQ</a> &nbsp; - &nbsp; 
	    <A HREF="goldenPath/help/hgTracksHelp.html" class="topbar">
			Help</a> &nbsp; </FONT></TD>
        </TR></TABLE>
    </TD></TR></TABLE>
    </TD></TR>

    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="/cgi-bin/hgGateway">
			Genome Browser</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgNear">
			Gene Sorter</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgBlat">
			Blat</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgPcr">
			In Silico PCR</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgText">
			Table Browser</A><HR>
	        <A CLASS="leftbar" HREF="util.html">
			Utilities</A><HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/downloads.html">
			Downloads</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/releaseLog.html">
			Release Log</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/customTracks/custTracks.html">
			Custom Tracks</A><HR>
	        <A CLASS="leftbar" HREF="/encode/">
			ENCODE</A><HR>
	        <A CLASS="leftbar" HREF="mirror.html">
			Mirrors</A><HR>
  	        <A CLASS="leftbar" HREF="http://genome-archive.cse.ucsc.edu">
			Archives</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/credits.html">
			Credits</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/pubs.html">
			Pubs</A><HR>
	        <A CLASS="leftbar" HREF="http://openhelix.com/ucscmaterials.shtml"
			TARGET=_blank>
			Training</A><HR>
	        <A CLASS="leftbar" HREF="cite.html">
			Cite Us</A><HR>
	        <A CLASS="leftbar" HREF="license/">
			Licenses</A><HR>
	        <A CLASS="leftbar" HREF="jobs.html">
			Jobs</A><HR>
  	        <A CLASS="leftbar" HREF="staff.html">
			Staff</A><HR>
  	        <A CLASS="leftbar" HREF="contacts.html">
			Contact Us</A><HR>
	        <BR>
	        <IMG SRC="images/DOT.gif" ALT="" HEIGHT=950 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Content Tables-->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; About the UCSC Genome Bioinformatics Site</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
		    This site contains the reference sequence and working draft
		    assemblies for a large collection of genomes. It also shows 
		    the CFTR (cystic fibrosis) region in 13 species and provides
		    a portal to the ENCODE project.  
		    <P>
		    We encourage you to explore these sequences with our tools. The Genome 
		    Browser zooms and scrolls over chromosomes, showing the work of annotators 
		    worldwide.  The Gene Sorter shows expression, homology and other 
		    information on groups of genes that can be related in many ways.  
		    Blat quickly maps your sequence to the genome.
		    The Table Browser 
		    provides convenient access to the underlying database.  
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
		        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		    </TD></TR></TABLE>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
		    <FONT FACE="courier" SIZE=3><B>26 July 2004  - Zv3 Zebrafish Genome Browser Available</B></FONT><BR>
		    <P>
		    The Zv3 Zebrafish genome assembly (UCSC version danRer1) is 
		    now available on the
		    UCSC Genome Browser and Blat server. This assembly was
		    produced by The Wellcome Trust Sanger Institute, Hinxton, 
		    UK, in 
		    collaboration with the Max Planck Institute for 
		    Developmental Biology in Tuebingen, Germany, the Netherlands
		    Institute for Developmental Biology (Hubrecht Laboratory), 
		    Utrecht, The Netherlands, and Yi Zhou and Leonard Zon from 
		    the Children's Hospital in Boston, MA, USA.
		    <P>
		    The Zv3 assembly consists of 1,459,115,486 bp in 58,339 
		    supercontigs, with a sequence coverage of approximately 
		    5.7X. This zebrafish assembly is the first to be tied to 
		    the FPC map: 1,083,447,588 bp (74%) of the sequence were 
		    mapped in this way. Please note that this is a preliminary 
		    assembly; a high level of misassembly is present due to 
		    polymorphisms in the DNA source.  
		    For more information about this assembly, see the Sanger 
		    Institute's  
		    <A HREF="http://www.sanger.ac.uk/Projects/D_rerio/"
		    TARGET=_blank><em>Danio rerio</em> Sequencing Project</A> 
		    web page.
		    <P>
		    Downloads of the Zebrafish data and annotations can be 
		    obtained from the UCSC 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/danRer1/" 
		    TARGET=_blank>FTP site</A> or
		    <A HREF=http://hgdownload.cse.ucsc.edu/downloads.html#zebrafish>Downloads
		    page</A>.
		    <P>
		    The danRer1 annotation tracks were generated by UCSC and 
		    collaborators worldwide. See the 
		    <A HREF="/goldenPath/credits.html#zebrafish_credits">Credits</A>
		    page for a detailed list of the organizations and individuals 
		    who contributed to the success of this release.
		    <P>
		    We'd like to thank The Wellcome Trust Sanger Institute
		    and their collaborators for providing this assembly.
		    The UCSC Zebrafish
		    Genome Browser team is Rachel Harte, Heather Trumbower, and
		    Donna Karolchik.
		    <P>
		    <HR>
    		    <FONT FACE="courier" SIZE=3><B>23 July 2004  - NCBI Human Build 35 released on Genome Browser</B></FONT><BR>
		    The latest human genome reference sequence (NCBI Build 35, 
		    May 2004) is now available as database hg17 in the UCSC 
		    Genome Browser and Blat server. This sequence was obtained
		    from NCBI and was produced by the International Human
		    Genome Sequencing Consortium.
		    <P>
		    Bulk downloads of the data are available via FTP at 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/hg17">ftp://hgdownload.cse.ucsc.edu/goldenPath/hg17</A> 
		    or through the Downloads link on this page. We recommend 
		    that you use FTP rather than HTML for the download of large 
		    or multiple files. 
		    <P>
		    We'd like to thank NCBI and the International Human Genome 
		    Sequencing Consortium for furnishing the data, and the UCSC 
		    team members who contributed to this release: Hiram Clawson, 
		    Terry Furey, Heather Trumbower, Robert Kuhn, Donna Karolchik, 
		    Kate Rosenbloom, Angie Hinrichs, Rachel Harte, Jim Kent and 
		    our sysadmin team Patrick Gavin, Jorge Garcia, 
		    and Paul Tatarsky. 
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>23 July 2004  - <em>Drosophila yakuba</em> Genome Browser Released</B></FONT><BR>
		    The UCSC Genome Bioinformatics Group has released a Genome
		    Browser and Blat server on a second species of fruitfly,
		    <em>D. yakuba</em>. The April 2004 Release 1.0 of this genome
		    (UCSC version droYak1) was sequenced and assembled by the 
		    Genome Sequencing Center, Washington University (WUSTL) 
		    School of Medicine in St. Louis. 
		    <P>
		    <em>D. yakuba</em> is closely related to the model organism,
		    <em>D. melanogaster</em>, with which it shared a common 
		    ancestor approximately 10 million years ago. 
		    The <em>D. yakuba</em> genome is largely alignable to the
		    <em>D. melanogaster</em> genome, but differs sufficiently to
		    offer an interesting study of sequence divergence between
		    the two species. <em>D. yakuba</em> occupies a critical 
		    intermediate position among several Drosophila species that 
		    will facilitate evolutionary studies among the fruitflies.
		    For information about the <em>D. yakuba</em> assembly 
		    and statistics, see the WUSTL Genome Sequencing Center
		    <A HREF="http://genome.wustl.edu/projects/yakuba/"
		    TARGET=_blank><em>Drosophila yakuba</em></A> web page.
		    <P>
	  	    Downloads of the droYak1 data and annotations can be obtained
		    from the UCSC Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/droYak1/">FTP server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#droYak">Downloads</A> page.
		    The droYak1 annotation tracks were generated by UCSC and
		    collaborators worldwide. 
		    <P>Thanks to the Genome Sequencing Center at WUSTL School of 
		    Medicine for providing the sequence and assembly of this 
		    genome.  The 
		    UCSC <em>D. yakuba</em> Genome Browser was produced by Angie 
		    Hinrichs, Michael Chalup, and Donna Karolchik. See the
		    <A HREF="/goldenPath/credits.html#droYak_credits">Credits</A>
		    page for a detailed list of the organizations and individuals 
		    who contributed to the success of this release.
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>16 July 2004  - Genome Browser for Build 33 Mouse Assembly:</B></FONT>
		    The latest mouse assembly -- Build 33 from NCBI (UCSC version
		    mm5) -- is now available via the UCSC Genome Browser and 
		    Blat server.  
		    <A HREF="/goldenPath/newsarch.html#071604">Read more</A>.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>15 July 2004  - <em>C. intestinalis</em> Genome Browser Released:</B></FONT>
		    The v1.0 <em>C. intestinalis</em> draft assembly from the
		    US DOE Joint Genome Institute is now available for study 
		    using the UCSC Genome Browser and Blat server (UCSC database
		    ci1). 
		    <A HREF="/goldenPath/newsarch.html#071504">Read more</A>.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>14 July 2004  - Dog Genome Browser Released:</B></FONT>
		    UCSC has released a Genome Browser and Blat server on the
		    July 2004  v1.0 dog genome sequenced and assembled by the 
		    Broad Institute of MIT and Harvard and Agencourt Bioscience. 
		    <A HREF="/goldenPath/newsarch.html#071404">Read more</A>.
    	            </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
	    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome Browser 
                    are freely available for academic, nonprofit, and personal use 
		    with the following conditions:
		    <UL>
		    <LI>Genome sequence data use restrictions are noted within the
		    species sections on the <A HREF="/goldenPath/credits.html">Credits</A> 
		    page.
		    <LI>Some annotation tracks contributed by external collaborators
		    contain proprietary data that have specific use restrictions. To
		    check for restrictions associated with a particular genome assembly, 
		    review the <em>database/README.txt</em> file in the assembly's 
		    downloads directory. 
		    </UL>
		    <P>
		    The Genome Browser and Blat software are free for academic,
		    nonprofit, and personal use. A license is required for commercial use.
		    See the <A HREF="/license/">Licenses</A> page for more information.
		    <P>
                    Program-driven use of this software is limited to a maximum of one 
                    hit every 15 seconds and no more than 5,000 hits per day.  
		    <P>
		    Questions or problems regarding the UCSC Genome Browser 
		    software, database, genome assemblies, or release cycles 
		    should be sent to 
		    <A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A>.
		    Messages sent to this address will be posted automatically 
		    on the genome mailing list, which is also archived on a 
		    public Web-accessible pipermail archive. NOTE: This archive may be 
		    indexed by non-UCSC sites such as Google. 
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>

        <BR>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	        
                    <FONT SIZE="4" ><A name="Tech"></A><B>&nbsp; Technical Information About the Assembled Sequence</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
	            <LI><A HREF="goldenPath/gbdDescriptions.html"><FONT SIZE="4">Description of the annotation database</FONT></A>
	            <LI><A HREF="goldenPath/datorg.html"><FONT SIZE="4">Data organization and format</FONT></A>
	            <LI><A HREF="goldenPath/term.html"><FONT SIZE="4">Terminology</FONT></A>
                    </UL>
                    </TD>
                    <TD WIDTH=15>&nbsp;</TD>
                 </TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	            <FONT SIZE="4" ><A NAME="Other"></A><B>&nbsp; Other Links</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
<!-- Commented out til this page is updated - very out of date
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/"><FONT SIZE="4">Human Genome Project at UCSC</FONT></A><FONT SIZE="4"></FONT>
-->
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/"><FONT SIZE="4">Center for Biomolecular Science and Engineering Home Page</FONT></A>
                    </UL>
                </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
</TD></TR></TABLE>

</BODY></HTML>

