<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                          <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="https://lists.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>

                    <!-- start news -->
<FONT FACE="courier" SIZE="3"><B>11 Jan. 2010 - Zebrafish release zv8 now available in Genome Browser</B></FONT><BR>
		    <P>
		    The most recent zebrafish assembly -- zv8 (UCSC version 
		    danRer6, Dec. 2008) -- is now available in the UCSC
		    Genome Browser. This assembly was produced by 
		    The Wellcome Trust Sanger Institute in collaboration with 
		    the Max Planck Institute for Developmental Biology in 
		    Tuebingen, Germany, and the Netherlands Institute for 
		    Developmental Biology (Hubrecht Laboratory), Utrecht, 
		    The Netherlands.</P>
		    <P>
		    The danRer6 assembly consists of nearly 1.5 billion bp in 
		    11,623 scaffolds with a coverage of 6.5-7x. It includes 
		    sequence anchored to chromosomes 
		    1-25 and chrM (mitochondrial), as well as 11,418 unplaced 
		    scaffolds.  Significant problems existing in previous 
		    assemblies have been rectified in this assembly through the
		    use of a whole-genome shotgun assembly with more coverage 
		    and a reorganization of the fingerprint contig order and
		    orientation through more careful use of maps. 
		    For more information about this assembly, see the Sanger 
		    Institute web page for the 
		    <A HREF="http://www.sanger.ac.uk/Projects/D_rerio/Zv8_assembly_information.shtml"
		    TARGET=_blank><em>Danio rerio</em> Sequencing Project</A> 
		    and the track description page for the danRer6 Assembly 
		    track.</P>
		    <P>
		    The danRer6 sequence and annotation data can be 
		    downloaded from the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/danRer6/">FTP server</A>
		    or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#zebrafish">downloads</A>
		    page. Please review the 
		    <A HREF="http://www.sanger.ac.uk/notices/use-policy.shtml"
		    TARGET=_blank>guidelines</A> for using these data.
		    <P>
		    We'd like to thank the Wellcome Trust Sanger Institute, 
		    the Max Planck Institute for Developmental Biology,
		    Hubrecht Laboratory and the other
		    institutions who contributed to the sequencing and 
		    mapping effort of this release. 
		    The UCSC zebrafish Genome Browser
		    was produced by Galt Barber, Robert Kuhn, Katrina Learned,
		    and Donna Karolchik. The initial set of annotation tracks 
		    was generated by the UCSC Genome Bioinformatics Group. See 
		    the <A HREF="goldenPath/credits.html#zebrafish_credits">credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to the release of
		    this browser.</P>
		    <P>

<HR>
<FONT FACE="courier" SIZE="3"><B>14 Dec. 2009 - New job posting: Biological Data Technician: </B></FONT>
		    The UCSC Genome Browser project is looking for a 
		    bioinformatician, biologist, or software engineer with a 
		    strong biology background to collect and import data  
		    into the UCSC Genome Browser database and 
		    website. 
		    <A HREF="goldenPath/newsarch.html#121409">Read more</A>.
		    </P>
		    <P>
<FONT FACE="courier" SIZE="3"><B>7 Dec. 2009 - Human Genome Browser default 
		    changing to hg19: </B></FONT>
		    In conjunction with the release of the UCSC Genes and
		    Conservation tracks on the hg19 (GRCh37) human assembly,
		    we have changed the default human browser on our website 
		    from hg18 to hg19. 
		    <A HREF="goldenPath/newsarch.html#120709">Read more</A>.
		    </P>
<FONT FACE="courier" SIZE="3"><B>1 Dec. 2009 - New UCSC Genes and Conservation tracks released on hg19 browser: </B></FONT>
		    We're happy to announce the release of two of our most 
		    popular data sets on the hg19/GRCh37 human Genome Browser. 
		    <A HREF="goldenPath/newsarch.html#120109">Read more</A>.
                    </P>                                                                                                       

                        </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
