<!-- News Section ============================================- -->  
<TABLE style="background-color:#888888; width: 100%; border-width: 0px; border-spacing: 0px; padding: 0px">
    <TR><TD>        
    <TABLE style="background-color:#fffee8; width: 100%; border-width: 0px; border-spacing: 0px; padding: 0px">
        <TR><TD>        
        <TABLE style="background-color:#D9E4F8; background-image: url('images/hr.gif'); width: 100%; border: 0px">
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" title="Follow &#64;GenomeBrowser on Twitter" alt="Follow &#64;GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a>
                 <a title="Genome Browser Facebook page"
                 href="http://www.facebook.com/ucscGenomeBrowser"
                 target="_blank"><img
                 src="/images/FB-f-Logo__blue_22.jpg"
                 style="vertical-align:text-bottom; margin-left: 15px;"
                 alt="Genome Browser Facebook page"></a>
                 <a href="https://plus.google.com/107643636314131661344" rel="publisher"></a>
               </TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE style="background-color:#fffee8; width: 100%; padding: 0px">
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list. Please see our
          <a href="http://genome.ucsc.edu/blog"
          target=_blank>blog</a> for posts about Genome Browser tools, features, projects and more.
          </p>

    <!-- start news -->
  <hr>
  <p>
  <font color="#006666" size="4"><b> 08 January 2016 -
  dbSNP 144 Available for hg19 and hg38</b></font>
  <p>
  We are pleased to announce the release of four tracks derived 
  from NCBI <a href="http://www.ncbi.nlm.nih.gov/SNP/">
  dbSNP</a> Build 144 data, available on the two most 
  recent human assemblies GRCh37/hg19 and GRCh38/hg38. 
  </p>
  <p>
  There are four SNP tracks available as part of this release. 
  One is a track containing all mappings of reference SNPs 
  to the human assembly, labeled "All SNPs (144)". 
  The other three tracks are subsets of this track 
  and show interesting and easily defined subsets of dbSNP:
  </p>

  <ul>
  <li>Common SNPs (144): uniquely mapped variants that appear 
  in at least 1% of the population or are 100% non-reference</li>
  <li>Flagged SNPs (144): uniquely mapped variants, excluding 
  Common SNPs, that have been flagged by dbSNP as 
  "clinically associated"</li>
  <li>Mult. SNPs (144): variants that have been mapped to more 
  than one genomic location</li> 
  </ul>
  <p>
  By default, only the Common SNPs (144) are visible; 
  other tracks must be made visible using the track controls. 
  You will find the other SNPs (144) tracks on both 
  of GRCh37/hg19 and GRCh38/hg38 browsers in the "Variation" group.
  </p>
  <p>
  The tracks were produced at UCSC by Angie Hinrichs and 
  Luvina Guruvadoo. We'd like to thank the dbSNP group at 
  NCBI for providing access to these data. 
  </p>
  <hr>
  
  <p>
  <font color="#006666" size="4"><b> 21 December 2015 -
  Two New Assemblies Now Available in the Genome Browser</b></font>
  <p>
  A Genome Browser is now available for the frog (<em>Xenopus tropicalis</em>)
  assembly released September 2012 by the <a href="http://genome.jgi.doe.gov/"
  target="_blank">US DOE Joint Genome Institute (JGI-PGF)</a>
  (JGI v7.0, UCSC version xenTro7).  For more
  information and statistics about this assembly, see the NCBI assembly record
  for <a href="http://www.ncbi.nlm.nih.gov/assembly/515038" target="_blank">
  Xtropicalis_v7</a>.  There are 7,728 scaffolds with a total size of 1,437,530,879
  bases.
  </p>

  <p>
  Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
  <a href="ftp://hgdownload.soe.ucsc.edu/goldenPath/xenTro7/">FTP server</a> or the
  <a href="http://hgdownload.soe.ucsc.edu/downloads.html#xentro">Downloads</a> page.
  Please observe the <a href="goldenPath/credits.html#xentro_use">conditions for use</a> when
  accessing and using these data sets.  The annotation tracks for this browser were
  generated by UCSC and collaborators woldwide.  See the <a
  href="goldenPath/credits.html#xentro_credits">Credits</a> page for a detailed list of the
  organizations and individuals who contributed to this release.
  </p>
  ---      
  <p>
  A Genome Browser is now available for the cat (<em>Felis catus</em>)
  assembly released November 2014 by the International Cat Genome Sequencing Consortium
 (ICGSC version Felis_catus_8.0, UCSC version felCat8).  For more
  information and statistics about this assembly, see the NCBI assembly record
  for <a href="http://www.ncbi.nlm.nih.gov/assembly/1373248" target="_blank">
  Felis_catus_8.0</a>.  There are 267,625 scaffolds with a total size of 2,641,342,258
  bases.
  </p>

  <p>
  Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
  <a href="ftp://hgdownload.soe.ucsc.edu/goldenPath/felCat8/">FTP server</a> or the
  <a href="http://hgdownload.soe.ucsc.edu/downloads.html#cat">Downloads</a> page.
  Please observe the <a href="goldenPath/credits.html#cat_use">conditions for use</a> when
  accessing and using these data sets.  The annotation tracks for this browser were
  generated by UCSC and collaborators woldwide.  See the <a
  href="goldenPath/credits.html#cat_credits">Credits</a> page for a detailed list of the
  organizations and individuals who contributed to this release.
  </p>
  
  <hr>
        <!-- start archives -->
<!--staged for future release

        <p>
        <font color="#006666" size="4"><b>08 January 2016 &mdash; 
        dbSNP 144 Available for hg19 and hg38
        <a href="goldenPath/newsarch.html#010816">Read more</a>.
        </p>

        <p>
        <font color="#006666" size="4"><b> 21 December 2015 -
        Two New Assemblies Now Available in the Genome Browser</b></font>
        <p>
        Genome Browsers are now available for the frog (<em>Xenopus tropicalis</em>)
        and cat (<i>Felis catus</i>) assemblies.
	<a href="goldenPath/newsarch.html#122115">Read more</a>. 
        </p>
-->

        <p>
	<font color="#006666" size="4"><b>16 December 2015 &mdash;
	Output from Related Tables Now Available in the Data Integrator</b>: </font>
	You can now obtain data from these related tables in your output from the
	<a href="cgi-bin/hgIntegaror">Data Integrator</a>.
	<a href="goldenPath/newsarch.html#121615">Read more</a>.
        </p>
	
            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
