<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="description" content="UCSC Genome Browser">
	<META name="keywords" content="genome, genome browser, human genome, genome assembly, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, mouse genome, rat genome, chicken genome, chimp genome, dog genome fruitfly, drosophila, fugu, yeast, ciona,  comparative genomics">
	<TITLE>UCSC Genome Browser Home</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">

</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="inc/home.topbar.html"--></TD></TR>
<!--Left Sidebar Menu----------------------------------------->  
    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="cgi-bin/hgGateway">
			Genome Browser</A><HR>
	        <A CLASS="leftbar" HREF="ENCODE/">
			ENCODE</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/hgBlat">
			Blat</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/hgTables">
			Table Browser</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/hgNear">
			Gene Sorter</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/hgPcr">
			In Silico PCR</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/pbGateway">
			Proteome Browser</A><HR>
	        <A CLASS="leftbar" HREF="util.html">
			Utilities</A><HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/downloads.html">
			Downloads</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/releaseLog.html">
			Release Log</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/customTracks/custTracks.html">
			Custom Tracks</A><HR>
	        <A CLASS="leftbar" HREF="mirror.html">
			Mirrors</A><HR>
  	        <A CLASS="leftbar" HREF="http://genome-archive.cse.ucsc.edu">
			Archives</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/credits.html">
			Credits</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/pubs.html">
			Publications</A><HR>
	        <A CLASS="leftbar" HREF="http://openhelix.com/ucscmaterials.shtml"
			TARGET=_blank>
			Training</A><HR>
	        <A CLASS="leftbar" HREF="cite.html">
			Cite Us</A><HR>
	        <A CLASS="leftbar" HREF="license/">
			Licenses</A><HR>
	        <A CLASS="leftbar" HREF="jobs.html">
			Jobs</A><HR>
  	        <A CLASS="leftbar" HREF="staff.html">
			Staff</A><HR>
  	        <A CLASS="leftbar" HREF="contacts.html">
			Contact Us</A><HR>
	        <BR>
	        <IMG SRC="images/DOT.gif" ALT="" HEIGHT=700 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Introduction---------------------------------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; About the UCSC Genome Bioinformatics Site</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
		    This site contains the reference sequence and working draft
		    assemblies for a large collection of genomes. It also 
		    provides a portal to the ENCODE project.  
		    <P>
		    We encourage you to explore these sequences with our tools. The Genome 
		    Browser zooms and scrolls over chromosomes, showing the work of annotators 
		    worldwide.  The Gene Sorter shows expression, homology and other 
		    information on groups of genes that can be related in many ways.  
		    Blat quickly maps your sequence to the genome.
		    The Table Browser 
		    provides convenient access to the underlying database.  
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
		        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		    </TD></TR></TABLE>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
		    <P>
		    To receive announcements of new genome 
		    assembly releases, new software features, updates and 
		    training seminars by email, subscribe to the
		    <A HREF="http://www.soe.ucsc.edu/mailman/listinfo/genome-announce" TARGET=_blank>genome-announce</A> mailing list.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>20 September 2005 - Three Drosophila Assemblies Released in Genome Browser</B></FONT>
		    <P>
		    <P>
		    The UCSC Genome Bioinformatics Group has added three more 
		    Drosophila genomes to the collection of fly genomes in 
		    the UCSC Genome Browser. The new 
		    genomes, which include the initial assembly of <em>D.
		    erecta</em> (droEre1, Aug. 2005) and the updated assemblies
		    of <em>D. virilis</em> (droVir2, Aug. 2005) and <em>D.  
		    mojavensis</em> (droMoj2, Aug. 2005), were
		    produced by <A HREF ="http://www.agencourt.com/" 
		    TARGET=_blank>Agencourt Bioscience Corporation</A> in
		    Beverly, MA, USA. All three genomes were assembled using
		    the Arachne assembler.</P>
		    <P>
		    The droEre1 assembly contains 5,124 scaffolds ranging in 
		    size from 154 bases to 26,647,023 bases, with a mean size 
		    of 29832.7 and median of 1740. 
		    The droVir2 assembly consists of 13,562 scaffolds ranging in
		    size from 57 bases to 25,269,527 bases, with a mean size of 
		    15263.1 and median of 1249.  
		    The droMoj2 assembly is comprised of 6,843 scaffolds ranging
		    in size from 101 bases to 34,172,700 bases, with a mean size
		    of 28389.6 and median of 1671.  </P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available from the UCSC downloads server:
		    <UL>
		    <LI>droEre1 -  <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/droEre1">FTP</A>, <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/droEre1">HTTP</A>
		    <LI>droVir2 - <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/droVir2">FTP</A>, <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/droVir2">HTTP</A>
		    <LI>droMoj2 - <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/droMoj2">FTP</A>, <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/droMoj2">HTTP</A>
		    </UL>
		    <P>
		    Please review the data use guidelines outlined in
		    the README.txt files that accompany the downloads. The
		    data use restrictions are also available on the Genome 
		    Browser <A HREF="/goldenPath/credits.html">credits</A> 
		    page. 
		    <P>
		    We'd like to thank Agencourt Bioscience Corporation
		    for providing these assemblies. The 
		    UCSC Drosophila browsers were produced by 
		    Angie Hinrichs, Brian Raney, Jennifer Jackson and Donna
		    Karolchik. The UCSC 
		    Genome Bioinformatics Group generated the initial set 
		    of annotation tracks. See the
		    <A HREF="/goldenPath/credits.html">credits</A> page for 
		    a detailed list of the organizations and individuals who 
		    contributed to this release. 
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>9 September 2005 - <em>C. intestinalis</em> v2.0 Genome Browser Released</B></FONT>
		    <P>
		    The v2.0 <em>C. intestinalis</em> draft assembly from the
		    US DOE Joint Genome Institute (JGI) is now available 
		    in the UCSC Genome Browser and Blat server (UCSC database
		    ci2). </P>
		    <P>
		    This whole genome shotgun assembly was constructed with 
		    the JGI assembler, JAZZ, using paired-end sequencing reads. 
		    Starting with a coverage of 11x, additional data -- 
		    including BAC and FISH markers -- were used to map scaffolds
		    to chromosome arms. The size of this assembly, including 
		    unmapped scaffolds, is 173 Mb, with 94 Mb of the sequence 
		    mapped to chromosome arms. </P>
		    <P>
		    For more information about the ci2 assembly, see the JGI
		    <A HREF="http://genome.jgi-psf.org/ciona4/ciona4.info.html"
		    TARGET=_blank><em>C. intestinalis</em> project page</A>.
		    Additional information and an analysis of the euchromatic 
		    regions of this genome may be found in Dehal, P. 
		    <em>et al.</em>
		    <A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=12481130&query_hl=1" 
		    TARGET=_blank>The draft genome of Ciona intestinalis: 
		    insights into chordate and vertebrate origins</A>. 
		    <em>Science</em> <B>298</B>(5601), 2157-67 (2002). 
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/ci2/" 
		    TARGET=_blank>FTP server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#ciona"
		    TARGET=_blank>Downloads page</A>. The ci2 annotation tracks 
		    were generated by UCSC and collaborators worldwide. See the 
		    <A HREF="goldenPath/credits.html#ciona_credits">Credits page</A> for a 
		    detailed list of the organizations and 
		    individuals who contributed to this release. 
		    <P>
		    Many thanks to the JGI and their collaborators for providing
		    the v2.0 sequence and annotations.  The ci2 Genome Browser 
		    was produced by Brian Raney, Mark Diekhans, Ann Zweig, 
		    Kayla Smith, Robert Kuhn and Donna Karolchik.</P>
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>11 August 2005 - Upcoming Genome Browser Seminars - San Francisco, Seattle, Chicago, Boston, Philadelphia, Atlanta:</B></FONT>
		    The UCSC Bioinformatics Group announces six regional 
		    seminars and hands-on computer workshops on the UCSC 
		    Genome Browser, presented by 
		    <A HREF="http://www.openhelix.com/" 
		    TARGET=_blank>OpenHelix:</A>
		    <A HREF="goldenPath/newsarch.html#081105">Read more</A>.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>13 June 2005 - <em>Drosophila simulans</em> Genome Browser Released: </B></FONT>
		    The UCSC Genome Bioinformatics Group has released a
		    Genome Browser for the <em>Drosophila simulans</em> draft
		    genome sequence. The Release 1.0 assembly (UCSC version
		    droSim1, April 2005) was produced by the 
		    <A HREF="http://genome.wustl.edu/" TARGET=_blank>Genome
		    Sequencing Center</A> at the Washington University in St. 
		    Louis (WUSTL) School of Medicine. 
		    <A HREF="goldenPath/newsarch.html#061305">Read more</A>.
		    </P>
    	            </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
	    	
<!--Conditions of Use Section--------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome Browser 
                    are freely available for academic, nonprofit, and personal use 
		    with the following conditions:
		    <UL>
		    <LI>Genome sequence data use restrictions are noted within the
		    species sections on the <A HREF="goldenPath/credits.html">Credits</A> 
		    page.
		    <LI>Some annotation tracks contributed by external collaborators
		    contain proprietary data that have specific use restrictions. To
		    check for restrictions associated with a particular genome assembly, 
		    review the <em>database/README.txt</em> file in the assembly's 
		    downloads directory. 
		    </UL>
		    <P>
		    The Genome Browser and Blat software are free for academic,
		    nonprofit, and personal use. A license is required for commercial use.
		    See the <A HREF="license/">Licenses</A> page for more information.
		    <P>
                    Program-driven use of this software is limited to a maximum of one 
                    hit every 15 seconds and no more than 5,000 hits per day.  
		    <P>
		    Questions or problems regarding the UCSC Genome Browser 
		    software, database, genome assemblies, or release cycles 
		    should be sent to 
		    <A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A>.
		    Messages sent to this address will be posted automatically 
		    on the genome mailing list, which is also archived on a 
		    public Web-accessible pipermail archive. NOTE: This archive may be 
		    indexed by non-UCSC sites such as Google. 
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>

        <BR>

<!--Technical Information Section----------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	        
                    <FONT SIZE="4" ><A name="Tech"></A><B>&nbsp; Technical Information About the Assembled Sequence</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
	            <LI><A HREF="goldenPath/gbdDescriptions.html"><FONT SIZE="4">Description of the annotation database</FONT></A>
	            <LI><A HREF="goldenPath/datorg.html"><FONT SIZE="4">Data organization and format</FONT></A>
	            <LI><A HREF="goldenPath/term.html"><FONT SIZE="4">Terminology</FONT></A>
                    </UL>
                    </TD>
                    <TD WIDTH=15>&nbsp;</TD>
                 </TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
    	
<!--Related Links Section--------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	            <FONT SIZE="4" ><A NAME="Other"></A><B>&nbsp; Related Links</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
<!-- Commented out til this page is updated - very out of date
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/"><FONT SIZE="4">Human Genome Project at UCSC</FONT></A><FONT SIZE="4"></FONT>
-->
	            <LI><A HREF="http://www.cbse.ucsc.edu/" TARGET=_blank><FONT SIZE="4">Center for Biomolecular Science and Engineering Home Page</FONT></A>
                    </UL>
                </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
</TD></TR></TABLE>

<!-- Web visibility------------------------------------------------>
<A HREF="knownGeneLists.html"><IMG SRC="images/DOT.gif" WIDTH="1" HEIGHT="1" ALIGN="BOTTOM" BORDER="0" ALT=""></A>
</BODY></HTML>

