<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="description" content="UCSC Genome Browser">
	<META name="keywords" content="genome, genome browser, genome assembly, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA">
	<TITLE>UCSC Genome Bioinformatics Home</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">

<SCRIPT LANGUAGE="JavaScript">
<!--

var theDb = 'human';

function setDb()
{
  theDb = document.dbForm.genome.options[document.dbForm.genome.selectedIndex].value;
}

function initDb()
{
  theDb = document.dbForm.genome.options[document.dbForm.genome.selectedIndex].value;
  var queryString = document.location.search.substring(1, document.location.search.length);
  var dbIndex = queryString.indexOf('db=');

  if (dbIndex >= 0)
  {
    theDb = queryString.substring(dbIndex + 3);
    for (var items = 0; items < document.dbForm.genome.options.length; items++)
    {
      if(sameOrganism(theDb, document.dbForm.genome.options[items].value))
      {
        document.dbForm.genome.options[items].selected = true;
      }
    }     
  }
}

function sameOrganism(db1, db2)
{
  if (db1.substring(0, 2) == db2.substring(0, 2))
  {
    return true;
  }
  else
  {
    return false;
  }
}

function dbGo(theUrl)
{
document.location = theUrl + 'org=' + theDb;
}

//-->
</SCRIPT>


</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF" onLoad="initDb()">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg"></TH></TR>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
	<table bgcolor="000000" cellpadding="1" cellspacing="1" width="99%" height="27"><tr bgcolor="2636D1"><td valign="middle">
	<table BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24">
		<TR><TD VALIGN="middle"><font color="#89A1DE">&nbsp; 
			<a href="javascript:dbGo('/cgi-bin/hgGateway?');" class="topbar">
				Genome Browser</a> &nbsp; - &nbsp; 
			<A HREF="javascript:dbGo('/cgi-bin/hgBlat?command=start&');" class="topbar">
				Blat Search</a> &nbsp; - &nbsp; 
			<A HREF="FAQ.html" class="topbar">
				FAQ</a> &nbsp; - &nbsp; 
			<A HREF="goldenPath/help/hgTracksHelp.html" class="topbar">
				User Guide</a> &nbsp; </font></TD>
    	</TR></TABLE>
	</TD></TR></TABLE>
</TD></TR>

<TR HEIGHT="100%" VALIGN="top">
  <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
  	<!-- outer table is to align leftbar & content boxes at top of page -->
  	<TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0 MARGIN=0>
	<TR><TD><IMG SRC="images/DOT.gif" HEIGHT="7"></TD></TR>
	<TR HEIGHT="100%"><TD HEIGHT="100%">
		<!--outer table is for black border-->
		<TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
   	 	<TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
		<TR HEIGHT="100%">
			<TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%"> 
				<FORM NAME="dbForm">
			<table cellpadding="" cellspacing="0" margins="0" border="0"><tr><td valign="top">
				<FONT CLASS="leftbar">Genome</FONT>
			</td></tr><tr valign="bottom"><td valign="bottom">
					<SELECT NAME="genome" onChange="setDb();">
						<OPTION VALUE="human">Human
						<OPTION VALUE="mouse">Mouse 
						<OPTION VALUE="rat">Rat 
					</SELECT>
			</td></tr></table><hr>
			<A CLASS="leftbar" HREF="javascript:dbGo('/cgi-bin/hgGateway?');">
				Browser</A><hr>
			<A CLASS="leftbar" HREF="javascript:dbGo('/cgi-bin/hgBlat?command=start&');">
				Blat</A><hr>
			<A CLASS="leftbar" HREF="javascript:dbGo('/cgi-bin/hgText?');">
				Tables</A><hr>
			<A CLASS="leftbar" HREF="downloads.html">
				Downloads</A><hr>
		  	<A CLASS="leftbar" HREF="mirror.html">
				Mirrors</A><hr>
  			<A CLASS="leftbar" HREF="http://genome-archive.ucsc.edu">
				Archives</A><hr>
			<A CLASS="leftbar" HREF="goldenPath/credits.html">
				Credits</A><hr>
			<A CLASS="leftbar" HREF="cite.html">
				Cite Us</A><hr>
			<A CLASS="leftbar" HREF="jobs.html">
				Jobs</A><hr>
  			<A CLASS="leftbar" HREF="bestlinks.html">
				Links</A><hr>
  			<A CLASS="leftbar" HREF="contacts.html">
				Contact Us</A><hr>
                                </FORM>
			<BR>
			<IMG SRC="images/DOT.gif" HEIGHT=1100 WIDTH=1>
		</TD></TR></TABLE>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
</TD>

<!--Content Tables-->  
  <TD VALIGN="top"><TABLE CELLPADDING=5 BORDER=0><TR><TD>
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<A NAME="About"></A><FONT SIZE="4"><b>&nbsp; About the UCSC Genome
	Bioinformatics Site</FONT>
	</TD></TR></TABLE>
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD WIDTH=10></TD><TD>
		<P>This site contains working drafts for the human, mouse, and
		rat genomes. 
		<P>The human genome is over 90% complete. Approximately 
		85% of the sequence is in a highly accurate 'finished' state.  The
		other 15% is merely 'draft' quality.  Some care must be taken
		when interpreting draft regions, but these are still very 
		useful to the working scientist.  
		<P>The mouse genome is
		largely the result of a full genome shotgun assembly. While there are 
		many gaps in the sequence, the order and orientation of the contigs on 
		either side of a gap are known in almost all cases. We estimate
		that 90-96% of the mouse genome is present in the
		assembly. We encourage you to explore the
		working drafts with the genome browser, which displays
		the work of many annotators worldwide.
		<P>The Genome Browser and the data it displays are freely available for 
		academic, nonprofit, and personal use with a few exceptions. Check the
		README.txt file in the download directory of a particular
		assembly to view the use restrictions
		specific to that release. Interactive use is limited to a 
		maximum of 10,000 hits/day or 10 hits/minute. 
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100% BORDER=0><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; News </FONT></TD>
			<TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD WIDTH=10></TD><TD>

	<P><FONT FACE="courier" SIZE=2>Dec. 6, 2002 </FONT><BR>
	We're pleased to announce the release of a UCSC Genome Browser on the 
	Nov. 2002 rat assembly produced by the Baylor College of Medicine Rat 
	Genome Sequencing Center and the Rat Genome Sequencing Consortium. 
	The sequence was assembled using a hybrid approach that combines the 
	clone by clone and whole genome shotgun methods. A new software program -
	ATLAS - was developed for this effort. The assembly process resulted in 
	a 6.5-fold coverage of the rat genome, which is estimated to be 
	approximately 2.8 Gigabases in size.
	<P>
	Downloads of the rat data and annotations are available through our 
	ftp site at <A HREF="ftp://genome.ucsc.edu/goldenPath/rnNov2002" TARGET=_blank>ftp://genome.ucsc.edu/goldenPath/rnNov2002</A> 
	or via the <A HREF="downloads.html">Downloads</A> 
	link on this page. This data contains 
	<A HREF="/goldenPath/credits.html#rat_use">specific 
	conditions for use</A>. The sequence is also available from the 
	<A HREF="http://www.hgsc.bcm.tmc.edu/projects/rat" TARGET=_blank> Rat 
	Genome Project website</A> for the Human Genome Sequencing Center at 
	Baylor College of Medicine or from GenBank. 
	<P>
	We'd like to thank the Baylor team and the Rat Genome Sequencing 
	Consortium for their collaboration on this project. See the 
	<A HREF="/goldenPath/credits.html#rat_credits">Credits page</A> 
	for a complete list of acknowledgements.. For more 
	information on the rat genome, the assembly process, and the
	Rat Genome Sequencing Consortium, refer to the website for the 
	<A HREF="http://www.hgsc.bcm.tmc.edu/" TARGET=_blank> 
	Human Genome Sequencing Center at Baylor College of Medicine</A>. 

	<P><FONT FACE="courier" SIZE=2>Dec. 5, 2002 </FONT><BR>
	The International Mouse Genome Sequencing Consortium has announced 
	the publication of a high-quality draft sequence of the mouse genome, 
	together with a comparative analysis of the mouse and human genomes. 
	The results from this analysis can be found in the Mouse Genome 
	Browser on this website. The paper appears in the Dec. 5 issue of the 
	journal Nature at 
	<A HREF="http://www.nature.com/nature/mousegenome/" TARGET=_blank>http://www.nature.com/nature/mousegenome/</A>.
	The co-author list includes several members of the UCSC Genome 
	Bioinformatics Group: CBSE Director David Haussler, Research 
	Scientist Jim Kent and research team members 
	Robert Baertsch, Mark Diekhans, Terrence Furey, Angie Hinrichs, Fan Hsu,
	Donna Karolchik, Krishna Roskin, Matt Schwartz, Charles Sugnet and Ryan 
	Weber. 

	<P><FONT FACE="courier" SIZE=2>Oct. 29, 2002 </FONT><BR>
	We've added several new directories of downloadable data to the 28 June 
	2002 human genome assembly. These directories contain mouse/human 
	alignments of the June 2002 human assembly vs. the Feb. 2002 mouse assembly. You can access these 
	directories from our <A HREF="http://genome.ucsc.edu/downloads.html" TARGET=_blank>Downloads</A> 
	link or or via our ftp site at <A HREF="ftp://genome.ucsc.edu/goldenPath/28jun2002" TARGET=_blank>ftp://genome.ucsc.edu/goldenPath/28jun2002/vsMm2/</A>. 
	<P>Within the main directory <I>vsMm2</I> are 3 subdirectories that contain all 
	the alignments (<I>axtAll</I>), alignments filtered to provide only the best alignment 
	for any given region of the human genome (<I>axtBest</I>), and a relatively stringent 
	subset of the axtBest alignments (<I>axtTight</I>). For more information 
	about the format of the alignment files and the methods used to generate 
	the alignments, consult the <I>README.txt</I> file in the vsMm2 directory.


    </TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<BR>
		
<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4" ><a name="Tech"></a><b>&nbsp; Technical Information About the Assembled Sequence</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD>
<UL>
	<LI><A HREF="goldenPath/term.html"><FONT SIZE="4">Terminology</FONT></A>
	<LI><A HREF="goldenPath/datorg.html"><FONT SIZE="4">Data organization and format</FONT></A>
	<LI><A HREF="goldenPath/gbdDescriptions.html"><FONT SIZE="4">Description of the annotation database</FONT></A>
	<LI><A HREF="goldenPath/stats.html"><FONT SIZE="4">Summary statistics</FONT></A>
	<LI><A HREF="goldenPath/chromReports"><FONT SIZE="4">Chromosome Reports</FONT></A>
         <LI><A HREF="goldenPath/mapPlots/"><FONT SIZE="4">Genome Map Plots</FONT></A>
<!-- CHEAT: not closing <UL> tag tightens bottom spacing reliably in unix & windows! -->
</TD><TD WIDTH=15>&nbsp;</TD></TR></TABLE>
	<BR></TD></TR></TABLE>
</TD></TR></TABLE>
	
<BR>
	
<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4" ><a name="Other"></a><b>&nbsp; Other Links</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD>
<UL>
	<LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/"><FONT SIZE="4">Human Genome Project at UCSC</FONT></A><FONT SIZE="4"></FONT>
	<LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/"><FONT SIZE="4">Center for Biomolecular Science and Engineering Home Page</FONT></A>
<!-- CHEAT: not closing <UL> tag tightens bottom spacing reliably in unix & windows! -->
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
</TD></TR></TABLE>

</TD></TR></TABLE>
</TD></TR></TABLE>
</BODY></HTML>

