<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="description" content="UCSC Genome Browser">
	<META name="keywords" content="genome, genome browser, human genome, genome assembly, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, mouse genome, rat genome, chicken genome, chimp genome, dog genome fruitfly, drosophila, fugu, yeast, ciona,  comparative genomics">
	<TITLE>UCSC Genome Browser Home</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">

</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="inc/home.topbar.html"--></TD></TR>
<!--Left Sidebar Menu----------------------------------------->  
    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="cgi-bin/hgGateway">
			Genome Browser</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/hgNear">
			Gene Sorter</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/hgBlat">
			Blat</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/hgPcr">
			In Silico PCR</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/hgTables">
			Table Browser</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/pbGateway">
			Proteome Browser</A><HR>
	        <A CLASS="leftbar" HREF="util.html">
			Utilities</A><HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/downloads.html">
			Downloads</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/releaseLog.html">
			Release Log</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/customTracks/custTracks.html">
			Custom Tracks</A><HR>
	        <A CLASS="leftbar" HREF="ENCODE/">
			ENCODE</A><HR>
	        <A CLASS="leftbar" HREF="mirror.html">
			Mirrors</A><HR>
  	        <A CLASS="leftbar" HREF="http://genome-archive.cse.ucsc.edu">
			Archives</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/credits.html">
			Credits</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/pubs.html">
			Publications</A><HR>
	        <A CLASS="leftbar" HREF="http://openhelix.com/ucscmaterials.shtml"
			TARGET=_blank>
			Training</A><HR>
	        <A CLASS="leftbar" HREF="cite.html">
			Cite Us</A><HR>
	        <A CLASS="leftbar" HREF="license/">
			Licenses</A><HR>
	        <A CLASS="leftbar" HREF="jobs.html">
			Jobs</A><HR>
  	        <A CLASS="leftbar" HREF="staff.html">
			Staff</A><HR>
  	        <A CLASS="leftbar" HREF="contacts.html">
			Contact Us</A><HR>
	        <BR>
	        <IMG SRC="images/DOT.gif" ALT="" HEIGHT=700 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Introduction---------------------------------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; About the UCSC Genome Bioinformatics Site</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
		    This site contains the reference sequence and working draft
		    assemblies for a large collection of genomes. It also shows 
		    the CFTR (cystic fibrosis) region in 13 species and provides
		    a portal to the ENCODE project.  
		    <P>
		    We encourage you to explore these sequences with our tools. The Genome 
		    Browser zooms and scrolls over chromosomes, showing the work of annotators 
		    worldwide.  The Gene Sorter shows expression, homology and other 
		    information on groups of genes that can be related in many ways.  
		    Blat quickly maps your sequence to the genome.
		    The Table Browser 
		    provides convenient access to the underlying database.  
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
		        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		    </TD></TR></TABLE>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
		    <P>
		    To receive announcements of new genome 
		    assembly releases, new software features, updates and 
		    training seminars by email, subscribe to the
		    <A HREF="http://www.soe.ucsc.edu/mailman/listinfo/genome-announce" TARGET=_blank>genome-announce</A> mailing list.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>22 April 2005 - New Mouse Assembly Available in Genome Browser</B></FONT>
		    <P>
		    The latest mouse genome assembly from the Mouse Genome
		    Sequencing Consortium, NCBI Build 34 (UCSC version mm6),
		    is now available in the UCSC Genome Browser. In 
		    conjunction with this release, we are archiving mouse
		    assemblies mm3 and mm4. All archived assemblies are 
		    accessible on our 
		    <A HREF="http://genome-archive.cse.ucsc.edu/"
		    TARGET=_blank>archive server</A>.</P>
		    <P>
		    The Build 34 assembly includes approximately 2.6 Gb, of 
		    which about 2.48 Gb is finished sequence. Chromosomes 2, 
		    4, 11 and X are finished in Build 34. To review in-depth 
		    statistics on this assembly, see the 
		    NCBI
                    <A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/mouse/contig/Build34.html"
                    TARGET=_blank>Build 34 Data</A> web page. </P>
		    <P>
		    NOTE: In the Genome Browser mm6 assembly,
		    chrY_random erroneously contains a region duplicated from 
		    chrY. For more information about this issue, see the 
		    Genome Browser
		    <A HREF="FAQ/FAQdownloads#download30">FAQ</A>. Please also
		    note that the UCSC mm6 database contains only the 
		    reference strain C57BL/6J.</P>
                    <P>
                    The mm6 sequence and annotation data may be downloaded 
		    from the Genome Browser
                    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/mm6/">FTP
                    server</A> or
                    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/mm6/">Downloads</A>
                    web page. The mm6 annotation tracks were generated by UCSC
                    and collaborators worldwide.
		    <P>
                    We'd like to thank Deanna Church and the Mouse Genome 
		    Sequencing Consortium for this assembly. We'd also like to
                    acknowledge the work of the UCSC mm6 team: Hiram Clawson, 
		    Fan Hsu, Jennifer Jackson, Robert Kuhn, Ali 
		    Sultan-Qurraie, and Donna Karolchik. For a complete list
		    of the individuals and organizations who participated 
		    in this assembly, see the 
		    <A HREF="goldenPath/credits#mouse_credits">Credits</A> 
		    page.
                    <P>
<!--Commented out until release

		    <FONT FACE="courier" SIZE=3><B>8 April 2005 - Rhesus Monkey Genome Browser Released</B></FONT>
		    <P>
		    The UCSC Genome Bioinformatics group has released a 
		    Genome Browser for the rhesus monkey draft assembly,
		    Mmul_0.1 (UCSC version rheMac1), produced by the 
                    Baylor College of Medicine Human Genome Sequencing Center 
		    in collaboration with J. Craig Venter Science Foundation 
		    Joint Technology Center and the Genome
                    Sequencing Center at Washington University School of
                    Medicine in St. Louis.
		    This Old World monkey, which is a primate model organism,
		    is important for the study of human biology and disease
		    due to its genetic, physiologic and metabolic 
		    similarities to humans.
		    <P>
		    Mmul_0.1 is a preliminary assembly using whole genome 
		    shotgun (WGS) reads from small and medium insert clones. 
		    Approximately 14.6 million reads were used in the 
		    assembly, representing about 12.6 Gb of sequence and 
		    about 4.6x coverage of the (clonable) genome. The N50 of 
		    the contigs in this assembly is 10.5 kb; the N50 of the 
		    scaffolds is 79.7 kb. The total length of all contigs is 
		    approximately 2.7 Gb, or 2.8 Gb including gaps between 
		    contigs.
		    Highly-repeated sequences and sequences from BAC clones 
		    were omitted from this assembly and will be placed in a 
		    subsequent version of the draft sequence. 
		    <P>
		    For more information about the rheMac1 assembly, see the 
		    Baylor 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/rmacaque/"
		    TARGET=_blank>Rhesus Monkey Project</A> web page.</P>
		    <P>
		    The rheMac1 sequence and annotation data can be downloaded
		    from the UCSC Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/rheMac1/">FTP
		    server</A> or the
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#rhesus">Downloads</A>
		    page. These data have specific
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" 
		    TARGET=_blank>conditions for use</A>.</P> 
		    <P>
		    Many thanks to Baylor College of Medicine, the Rhesus
		    Monkey Genome Sequencing Consortium, and the other 
		    institutions who contributed to the sequencing and mapping
		    effort of the Mmul_0.1 release. The UCSC Rhesus Genome 
		    Browser was produced by Robert Baertsch, Galt Barber, 
		    and Donna Karolchik. The initial set of rheMac1 annotation
	            tracks was generated by the <A HREF="staff.html">UCSC 
		    Genome Bioinformatics Group</A>. See the 
		    <A HREF="goldenPath/credits.html#rhesus_credits">credits</A> 
		    page for a detailed list of the organizations and 
		    individuals who contributed to the release of this 
		    browser. </P>
End comment-->
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>24 March 2005 - Genome Browser Tutorial at Experimental Biology 2005 on 5 April: </B></FONT>
		    OpenHelix will be presenting a one-hour introductory 
		    seminar on the UCSC Genome Browser at the 
		    Experimental Biology 2005 meeting on 5 April, 5:00-6:00. 
		    <A HREF="goldenPath/newsarch.html#032405">Read more</A>.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>2 March 2005  - CCDS Data Set Released:</B></FONT>
		    The initial results from the Consensus Coding Sequence 
		    project (CCDS) were released to the public today. 
		    CCDS is a collaborative effort to identify a core set of 
		    human protein-coding regions that are consistently 
		    annotated and of high quality. 
		    <A HREF="goldenPath/newsarch.html#030205">Read more</A>.
    	            </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
	    	
<!--Conditions of Use Section--------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome Browser 
                    are freely available for academic, nonprofit, and personal use 
		    with the following conditions:
		    <UL>
		    <LI>Genome sequence data use restrictions are noted within the
		    species sections on the <A HREF="goldenPath/credits.html">Credits</A> 
		    page.
		    <LI>Some annotation tracks contributed by external collaborators
		    contain proprietary data that have specific use restrictions. To
		    check for restrictions associated with a particular genome assembly, 
		    review the <em>database/README.txt</em> file in the assembly's 
		    downloads directory. 
		    </UL>
		    <P>
		    The Genome Browser and Blat software are free for academic,
		    nonprofit, and personal use. A license is required for commercial use.
		    See the <A HREF="license/">Licenses</A> page for more information.
		    <P>
                    Program-driven use of this software is limited to a maximum of one 
                    hit every 15 seconds and no more than 5,000 hits per day.  
		    <P>
		    Questions or problems regarding the UCSC Genome Browser 
		    software, database, genome assemblies, or release cycles 
		    should be sent to 
		    <A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A>.
		    Messages sent to this address will be posted automatically 
		    on the genome mailing list, which is also archived on a 
		    public Web-accessible pipermail archive. NOTE: This archive may be 
		    indexed by non-UCSC sites such as Google. 
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>

        <BR>

<!--Technical Information Section----------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	        
                    <FONT SIZE="4" ><A name="Tech"></A><B>&nbsp; Technical Information About the Assembled Sequence</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
	            <LI><A HREF="goldenPath/gbdDescriptions.html"><FONT SIZE="4">Description of the annotation database</FONT></A>
	            <LI><A HREF="goldenPath/datorg.html"><FONT SIZE="4">Data organization and format</FONT></A>
	            <LI><A HREF="goldenPath/term.html"><FONT SIZE="4">Terminology</FONT></A>
                    </UL>
                    </TD>
                    <TD WIDTH=15>&nbsp;</TD>
                 </TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
    	
<!--Related Links Section--------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	            <FONT SIZE="4" ><A NAME="Other"></A><B>&nbsp; Related Links</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
<!-- Commented out til this page is updated - very out of date
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/"><FONT SIZE="4">Human Genome Project at UCSC</FONT></A><FONT SIZE="4"></FONT>
-->
	            <LI><A HREF="http://www.cbse.ucsc.edu/" TARGET=_blank><FONT SIZE="4">Center for Biomolecular Science and Engineering Home Page</FONT></A>
                    </UL>
                </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
</TD></TR></TABLE>

<!-- Web visibility------------------------------------------------>
<A HREF="knownGeneLists.html"><IMG SRC="images/DOT.gif" WIDTH="1" HEIGHT="1" ALIGN="BOTTOM" BORDER="0" ALT=""></A>
</BODY></HTML>

