<!-- News Section ============================================- -->  
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
    <TR><TD>        
    <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
        <TR><TD>        
        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" title="Follow &#64;GenomeBrowser on Twitter" alt="Follow &#64;GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a>
                 <a title="Genome Browser Facebook page"
                 href="http://www.facebook.com/ucscGenomeBrowser"
                 target="_blank"><img
                 src="/images/FB-f-Logo__blue_22.jpg"
                 style="vertical-align:text-bottom; margin-left: 15px;"
                 alt="Genome Browser Facebook page"></a>
                 <a href="https://plus.google.com/107643636314131661344" rel="publisher"></a>
               </TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list.</P>

    <!-- start news -->
	<hr>
	<font face="courier" size="3"><b>03 September 2014 -
	Data from the FaceBase Consortium Now Available in the Genome Browser</b>
	</font>
	</p>
	<p>
	We are pleased to announce that much of the data from the <a href="https://www.facebase.org/"
	target"_blank">FaceBase Consortium</a> is now available in the UCSC Genome Browser as a
	public track hub. This track hub contains data from numerous experiments carried out by
	the Potter and Visel labs to study craniofacial development in mice and humans, with
	tracks for the mm9, mm10, hg18, and hg19 assemblies. You can check out the hub on the
	mm9 assembly
	<a href="/cgi-bin/hgTracks?db=mm9&position=chr12%3A57795963-57815592&hubUrl=http://trackhub.facebase.org/hub.txt"
	target="_blank">here</a>. In addition to this public track hub, we have a set of
	microarray expression data hosted as a native track for both the mm9 and mm10 assemblies.
	This expression data compares expression across the mouse genome for a number of embryonic
	days and craniofacial cell types. For more information on this track, please see the track
	<a href="/cgi-bin/hgTrackUi?org=mouse&db=mm9&g=FaceBase24SampleTypesAvg"
	target="_blank">description page</a>.
	</p>
	<p>
	We would like to thank Robert Kuhn, Pauline Fujita, and Matthew Speir from UCSC and Thomas
	Maher from the FaceBase Consortium for their efforts in bringing these resources to the
	UCSC Genome Browser.
	</p>

        <hr>
	<font face="courier" size="3"><b>30 July 2014 -
	New Rat (rn6) Assembly Now Available in the Genome Browser</b>
        </font>
        </p>
        <p>
	We are excited to announce the release of a Genome Browser for the July 2014 assembly
	of rat, <em>Rattus norvegicus</em> (RGSC Rnor_6.0, UCSC version rn6)! This
	assembly is provided by the Rat Genome Sequencing Consortium, which is comprised of
	eight research organizations across the United States and Canada and led by the
	<a href="https://www.hgsc.bcm.edu/" target="_blank">Baylor College of Medicine</a>.
	The new RGSC Rnor_6.0 assembly contains a new, partially assembled Y chromosome
	as well as improvements to other regions of the genome. You can find more information
	on the RGSC's efforts to sequence rat genome on the Baylor College of Medicine's
	<a href="https://www.hgsc.bcm.edu/other-mammals/rat-genome-project"
	target="_blank">project page</a>.
	</p>
	<p>
        Bulk downloads of the sequence and annotation data are available via the Genome
        Browser
        <a href="ftp://hgdownload.cse.ucsc.edu/goldenPath/rn6/">FTP server</a>
        or the
        <a href="http://hgdownload.cse.ucsc.edu/downloads.html#rat">Downloads</a>
        page. These data have
        <a href="../goldenPath/credits.html#rat_use">specific conditions for use</a>.
        The rat (rn6) browser annotation tracks were generated by UCSC and
        collaborators worldwide. See the
        <a href="../goldenPath/credits.html#rat_credits">Credits</a> page for a detailed
        list of the organizations and individuals who contributed to this release.
        </p>

        <hr>

        <!-- staged for archive use.  NOTE: condense header line from .... </b></font><p> to </b>: </font>

	<p>
	<font face="courier" size="3"><b>01 September 2014 -
	Data from the FaceBase Consortium Now Available in the Genome Browser</b>: </font>
	We are pleased to announce that much of the data from the <a href="https://www.facebase.org/"
	target"_blank">FaceBase Consortium</a> is now available in the UCSC Genome Browser as a public track hub.
	</p>

	<p>
	<font face="courier" size="3"><b>30 July 2014 -
        New Rat (rn6) Assembly Now Available in the Genome Browser</b>:</font>
        We are excited to announce the release of a Genome Browser for the July 2014 assembly
        of rat, <em>Rattus norvegicus</em> (RGSC Rnor_6.0, UCSC version rn6)!
	</p>
                -->

        <!-- start archives -->
        <p>
        <font face="courier" size="3"><b> 15 July 2014 -
        New Shrew (sorAra2) Assembly Now Available in the Genome Browser</b>: </font>
        We are pleased to announce the release of a Genome Browser for the August 2008
        assembly of shrew, <em>Sorex araneus</em> (Broad SorAra2.0, UCSC version
        sorAra2). <a href="goldenPath/newsarch.html#071514">Read more</a>.
        </p>

        <p>
        <font face="courier" size="3"><b> 15 July 2014 -
        New Pika (ochPri3) Assembly Now Available in the Genome Browser</b>: </font>
        We are pleased to announce the release of a Genome Browser for the May 2012
        assembly of pika, <em>Ochotona princeps</em> (Broad OchPri3.0, UCSC version
        ochPri3). <a href="goldenPath/newsarch.html#071514">Read more</a>.
        </p>

            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
