<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                         <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="https://lists.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>

            
            <!-- start news -->


            <!-- OMIM release  -->
		    <P>
                    <FONT FACE="courier" SIZE="3"><B>16 June 2011 -
Re-engineered OMIM Tracks Released </B></FONT>
		    <P>
                    We announce today the release of our newly re-engineered OMIM 
                    (Online Mendelian Inheritance in Man) tracks for both hg18 and 
                    hg19. With the kind assistance of Ada Hamosh (director), 
                    Joanna Amberger and Francois Schiettecatte of the OMIM project, 
                    we have separated the OMIM records into three separate tracks:
                    </P>
                    <P>
                    <B>OMIM Allelic Variant SNPs</B>
                    <BR> &nbsp; &nbsp; &nbsp; &nbsp;
                      Variants in the OMIM database that have associated dbSNP identifiers.
                    </P>

                    <P>
                    <B>OMIM Genes</B>
                    <BR> &nbsp; &nbsp; &nbsp; &nbsp;
                      The genomic positions of gene entries in the OMIM database. 
                      The coloring indicates the associated OMIM phenotype class.
                    </P>

                    <P>
                    <B>OMIM Phenotypes - Gene Unknown</B>
                    <BR> &nbsp; &nbsp; &nbsp; &nbsp;
                      Regions known to be associated with a phenotype, but for which no 
                      specific gene is known to be causative. This track also includes 
                      known multi-gene syndromes.
                    </P>

                    <P>
                    The OMIM tracks are searchable by OMIM number. In most cases, 
                    simply typing the 6-digit MIM number into the postition/search box 
                    on the Browser will take you to the record.
                    </P>

                    <P>
                    The OMIM data are the property of Johns Hopkins University and 
                    will not be available for download from UCSC. Please contact the 
                    OMIM project at <http://omim.org/download>omim.org</A> for download 
                    information.
                    </P>

                    <P>
                    UCSC thanks engineers Fan Hsu, Brooke Rhead and Robert Kuhn for 
                    this release.
                    </P>
                    <HR>

            <!-- Preview Browser -->
		    <P>
                    <FONT FACE="courier" SIZE="3"><B>9 June 2011 -
UCSC Preview Browser Available</B></FONT>
		    <P>
                    Early access to ENCODE and other UCSC browser data tracks under 
                    construction is now available from the new UCSC Preview Browser site:
                    <A HREF="http://genome-preview.ucsc.edu">http://genome-preview.ucsc.edu</A></P>

		    This site is a weekly mirror of our internal development server for 
		    public access. Data and tools there are under construction, have not 
		    been quality reviewed, and are subject to change at any time. We provide 
		    this site for early access, with the warning that it is less available 
		    and stable than our public site. For high-quality reviewed annotations 
		    on our production server, visit our public website: 
                    <A HREF="http://genome.ucsc.edu">http://genome.ucsc.edu</A>.
                    </P>

                    <HR>

            <!-- anoCar2 Browser Release -->
		    <P>
                    <FONT FACE="courier" SIZE="3"><B>7 June 2011 -
Updated Lizard Browser Available: </B></FONT>
		    We have released a Genome Browser for the May 2010 
		    genome assembly of the green anole lizard, <em>Anolis 
		    carolinensis</em> (Broad 
		    version AnoCar2.0, UCSC version anoCar2). 
		    <A HREF="goldenPath/newsarch.html#060711">Read more</A>.
		    </P>

            <!-- mm9 UCSC Genes -->
		    <P>
                    <FONT FACE="courier" SIZE="3"><B>26 May 2011 - New
Release of UCSC Genes for Mouse:</B></FONT>
		    We've released an updated set of UCSC Genes for the mm9 
		    (NCBI Build 37) mouse Genome Browser. 
		    <A HREF="goldenPath/newsarch.html#052611">Read more</A>.
		    </P>

            <!-- ENCODE usability survey -->
		    <P>
                    <FONT FACE="courier" SIZE="3"><B>1 May 2011 - ENCODE
User's Guide: </B></FONT>
                  The ENCODE project has just published an overview of their ongoing large-scale
                  efforts to interpret the human genome sequence in the journal PLoS Biology.
                  <a target=_blank href="http://www.plosbiology.org/article/info:doi/10.1371/journal.pbio.1001046 ">
                  A Users Guide to the Encyclopedia of DNA Elements</a>,
                  highlights the scope of data production, and provides guidance for locating 
                  and using the data.  
		  <A HREF="goldenPath/newsarch.html#050111">Read more</A>.
		    </P>

            <!-- hg19 snp132 release -->
		    <P>
                    <FONT FACE="courier" SIZE="3"><B>18 April 2011 -
dbSNP 132 Available for hg19: </B></FONT>
                    We are pleased to announce the release of four tracks derived from
                    <a href="http://www.ncbi.nlm.nih.gov/projects/SNP/" target=_blank>dbSNP</a>
                    build 132, available on the human assembly (GRCh37/hg19).
                    dbSNP build 132 is available at NCBI. 
		    <A HREF="goldenPath/newsarch.html#041811.2">Read more</A>.
		    </P>
                   
==> <a href="goldenPath/newsarch.html">News Archives</a>

                    </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
