<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                          <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="http://www.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>
		    <P>
<!--
		    <FONT FACE="courier" SIZE=3><B>16 Oct. 2007 - Genome Browser for Zebrafish Zv7 Released</B></FONT><BR>
		    <P>
		    The latest zebrafish assembly -- Zv7 (UCSC version 
		    danRer5, July 2007) -- is now available in the UCSC
		    Genome Browser. The Zv7 assembly was produced by 
		    The Wellcome Trust Sanger Institute in collaboration with 
		    the Max Planck Institute for Developmental Biology in 
		    Tuebingen, Germany, and the Netherlands Institute for 
		    Developmental Biology (Hubrecht Laboratory), Utrecht, 
		    The Netherlands.</P>
		    <P>
		    This assembly consists of 1,440,582,308 bp in 5,036 
		    fragments. It includes sequence anchored to chromosomes 
		    1-25 and chrM (mitochondrial), as well as 5010 unplaced 
		    scaffolds.  The assembly was produced by integrating 
		    finished clone sequence from the physical map with whole 
		    genome shotgun assembly sequence. The N50 size is 
		    1,153,933, n = 277 (i.e. the length such that 50% of the 
		    assembled genome lies in blocks of the N50 size or longer). 
		    For more information about this assembly, see the Sanger 
		    Institute web page for the 
		    <A HREF="http://www.sanger.ac.uk/Projects/D_rerio/Zv7_assembly_information.shtml"
		    TARGET=_blank><em>Danio rerio</em> Sequencing Project</A> 
		    and the track description page for the Assembly track.</P>
		    <P>
		    The danRer5 sequence and annotation data can be 
		    downloaded from the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/danRer5/">FTP server</A>
		    or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#zebrafish">downloads</A>
		    page. Please review the 
		    <A HREF="http://www.sanger.ac.uk/notices/use-policy.shtml"
		    TARGET=_blank>guidelines</A> for using these data.
		    <P>
		    We'd like to thank the Wellcome Trust Sanger Institute, 
		    the Max Planck Institute for Developmental Biology,
		    Hubrecht Laboratory and the other
		    institutions who contributed to the sequencing and 
		    mapping effort of this release. Special thanks to the 
		    Zebrafish Genome Initiative at Children's Hospital in 
		    Boston for their collaboration on this release. 
		    The UCSC zebrafish Genome Browser
		    was produced by Rachel Harte, Ann Zweig, and Donna 
		    Karolchik. The 
		    initial set of annotation tracks was generated by the 
		    UCSC Genome Bioinformatics Group.  See the
		    <A HREF="goldenPath/credits.html#zebrafish_credits">credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to the release of
		    this browser.</P>
		    <P>
		    <HR>
		    <P>
-->
		    <FONT FACE="courier" SIZE=3><B>8 Oct. 2007 - New Browser for Purple Sea Urchin</B></FONT><BR>
		    <P>
		    The Sep. 2006 release of the purple sea urchin genome 
		    (<em>Strongylocentrotus purpuratus</em>) is now available
		    in the UCSC Genome Browser. This assembly, UCSC version
		    strPur2, was produced by the Baylor College of Medicine 
		    Human Genome Sequencing Center (BCM HGSC) and corresponds 
		    to their Spur_2.1 assembly. </P>
		    <P>
		    The Spur_2.1 release was assembled from BAC sequence reads 
		    (approximately 2x coverage) and whole genome shotgun 
		    reads (6x coverage), and utilizes BAC tiling path 
		    information.  The BCM HGSC Atlas-2.0 genome 
		    assembly system was used to generate this assembly.
		    The total length of all contigs greater than 1kb is 804 
		    Mbp.  When the gaps between contigs in scaffolds are 
		    included, the total span of the assembly is 907 Mbp.  The 
		    estimated size of the genome based on the assembly is 814 
		    Mbp.</P>
		    <P>
		    Compared to previous sea urchin releases, the Spur_2.1 
		    assembly is more continuous and has fewer false 
		    duplications; contaminations identified in the previous
		    Spur_2.0 assembly have been removed.
		    This draft assembly may contain errors; therefore, users 
		    should exercise caution. Typical errors may include 
		    misassemblies of repeat sequences, 
		    collapses of repeat regions, and artificial duplications in 
		    polymorphic regions. However, base accuracy in contigs is 
		    usually very high with most errors near the ends of contigs.
		    </P>
		    <P>
		    More assembly details can be found in the 
		    <A HREF="ftp://ftp.hgsc.bcm.tmc.edu/pub/data/Spurpuratus/fasta/Spur_v2.1/README_Spur_v2.1.txt"
		    TARGET=_blank>Spur_2.1 README file</A> and on the BCM HGSC 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/seaurchin/"
		    TARGET=_blank>Sea Urchin Genome Project</A> web page.</P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/strPur2">FTP server</A>
		    or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#urchin">Downloads</A> 
		    page. These data have 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" 
		    TARGET=_blank>specific conditions for use</A>. The initial
		    set of strPur2 annotation tracks was generated by UCSC. 
		    See the <A HREF="goldenPath/credits.html#urchin_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release. </P>
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>4 Oct. 2007 - ENCODE Genome Browser Released for hg18 Assembly</B></FONT><BR>
		    <P>
		    The UCSC ENCODE browser for the human genome assembly hg18 
		    (NCBI Build 36) is now available. You can access the browser
		    directly at 
		    <A HREF="encode.hg18.html">http://genome.ucsc.edu/ENCODE/encode.hg18.html</A> 
		    or by clicking the ENCODE link in the sidebar menu on this 
		    page, then clicking the Regions (hg18) link in the sidebar 
		    menu on the ENCODE portal page.</P>
		    <P>
		    The hg18 ENCODE browser includes 540 data tables in 59 
		    browser tracks that were migrated from the hg17 browser. 
		    The hg17 data coordinates were converted to hg18 coordinates
		    using the UCSC liftOver process.</P>
		    <P>
		    To improve the accessibility of the data, related ENCODE 
		    tracks have been gathered into new configuration groupings 
		    (&quot;super-tracks&quot;) that can be displayed or hidden
		    using a single visiblity control.  We have also reduced the
 		    number of track groups and have modified some of the group 
		    names for clarity. ENCODE tracks with whole-genome data 
		    have been moved into the standard browser track groups.</P>
		    <P>
		    For more information about the hg18 ENCODE data migration, 
		    see the News section on the UCSC 
		    <A HREF="ENCODE/">ENCODE portal page</A> and the UCSC 
		    <A HREF="http://genomewiki.cse.ucsc.edu/index.php/ENCODE_Hg18_Migration" TARGET=_blank>genomeWiki</A>.</P>
<!--
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>22 Aug. 2007 - New Mouse Browser Released: </B></FONT>
		    The latest mouse genome assembly from the Mouse Genome
		    Sequencing Consortium, NCBI Build 37.1, is now available in 
		    the UCSC Genome Browser. 
                    <A HREF="goldenPath/newsarch.html#082207b">Read more</A>.</P>
-->
                        </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
