<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
		        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		    </TD></TR></TABLE>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
		    <P>
		    To receive announcements of new genome 
		    assembly releases, new software features, updates and 
		    training seminars by email, subscribe to the
		    <A HREF="http://www.soe.ucsc.edu/mailman/listinfo/genome-announce" 
		    TARGET=_blank>genome-announce</A> mailing list.</P>
		    <P>
                    <FONT FACE="courier" SIZE=3><B>22 January 2007 - Cat Assembly Available in Genome Browser</B></FONT>
		    <P>
		    The Mar. 2006 release of <em>Felis catus</em> (UCSC version
		    felCat3) is now available in the Genome Browser.  This
		    assembly was produced by <A HREF="http://www.broad.mit.edu/"
		    TARGET=_blank>The Broad Institute</A> of MIT/Harvard and
		    <A HREF="http://www.agencourt.com/"
		    TARGET=_blank>Agencourt Bioscience</A>.</P>
		    <P>
		    The felCat3 genome has been sequenced to 2X coverage and
		    consists of 217,790 scaffolds. The total contig length for
		    this assembly is approximately 1.6 Gb spanning nearly 4.0
		    Gb (with 60.1&#37; in gaps). There are 749,376 contigs, with an
		    N50 length of 2,506 bases. There are 149,283 supercontigs,
		    with an N50 length of 49,769 bases (not including gaps).
		    The N50 size is the length such that 50&#37; of the assembled
		    genome lies in blocks of the N50 size or longer.</P>
		    <P>
		    The felCat3 sequence and annotation data can be downloaded 
		    from the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/felCat3/">
		    FTP server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/felCat3/">Downloads</A> 
		    page. Please review the 
		    <A HREF="goldenPath/credits.html#cat_use">guidelines</A>
		    for using the cat assembly data. </P>
		    <P>
		    Many thanks to The Broad Institute for providing these data.
		    The UCSC cat Genome Browser was produced by Heather Trumbower,
		    Angie Hinrichs, Brooke Rhead, and Archana Thakkapallayil.
		    The 
		    initial set of annotation tracks was generated by the UCSC 
		    Genome Bioinformatics Group. See the Genome Browser
		    <A HREF="goldenPath/credits.html#cat_credits">Credits</A> 
		    page for a 
		    detailed list of the organizations and individuals who 
		    contributed to the release of this browser. </P>
		    <P>
		    <HR>
                    <FONT FACE="courier" SIZE=3><B>1 January 2007 - Upcoming Genome Browser Seminars:  SF, Seattle, NYC, Cleveland</B></FONT>
		    <P>
		    The UCSC Bioinformatics Group announces four regional 
		    seminars and hands-on computer workshops on the UCSC 
		    Genome Browser, presented by 
		    <A HREF="http://www.openhelix.com/" 
		    TARGET=_blank>OpenHelix</A>:
		    <UL>
		    <LI>
		    San Francisco, CA -- Wednesday, 31 January
		    <LI>
		    Seattle, WA -- Thursday, 1 February
		    <LI>
		    New York City, NY -- Tuesday, 13 February
		    <LI>
		    Cleveland, OH -- Wednesday, 14 February
		    </UL>
		    <A HREF="goldenPath/newsarch.html#010107">Read more</A>.</P>
		    <P>
    	            </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
