<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                          <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="https://lists.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>

                    <!-- start news -->
		    <P>
<FONT FACE="courier" SIZE="3"><B>7 Dec. 2009 - Human Genome Browser default 
		    changing to hg19</B></FONT>
		    <P>
		    In conjunction with the release of the UCSC Genes and
		    Conservation tracks on the hg19 (GRCh37) human assembly,
		    we have changed the default human browser on our website 
		    from hg18 to hg19. In general, if you are currently using 
		    the hg18 (or older) browser, the 
		    Genome Browser will continue to display that assembly for 
		    you when you start it up. However, there are circumstances 
		    in which the assembly may switch to the newer version (for 
		    instance, if you reset your browser defaults). If you find 
		    yourself in a situation where some of your favorite browser 
		    tracks have "disappeared", you may want to check that 
		    you're viewing the right assembly.
		    </P>
		    <P>
<FONT FACE="courier" SIZE="3"><B>1 Dec. 2009 - New UCSC Genes and Conservation tracks released on hg19 browser</B></FONT><BR>
		    <P>
		    We're happy to announce the release of two of our most 
		    popular data sets on the hg19/GRCh37 human Genome Browser. 
		    The <B>UCSC Genes</B> track is a moderately conservative 
		    set of gene predictions based on data from RefSeq, Genbank, 
		    CCDS and UniProt. The <B>Conservation</B> track shows 
		    multiple alignments of 46 vertebrate species and 
		    measurements of evolutionary conservation using two methods 
		    (phastCons and phyloP) from the PHAST package for all 
		    vertebrate species as well as primate and placental mammal 
		    subsets. For more information about these annotations, read 
		    the description pages that accompany these tracks.</P>
		    <P>
		    <HR>
<FONT FACE="courier" SIZE="3"><B>20 Oct. 2009 - Job opening with UCSC Genome Browser project</B></FONT>
                    The UCSC Genome Browser project is currently accepting
                    applications for a Biological Database Testing/User Support Technician
                    (Programmer/Analyst I).
		    <A HREF="goldenPath/newsarch.html#102009">Read more</A>.</P>
                    </P>                                                                                                       
                    <HR>                                       

                    <FONT FACE="courier" SIZE="3"><B>9 September 2009 - Changes
 		    to the bigBed/bigWig data formats:</B></FONT>
		    If you have been taking advantage of the new bigBed format 
		    (for very large data sets), you'll be happy to hear that we 
		    have considerably slimmed down the memory footprint of the 
		    program that converts BED files into bigBed files: 
		    bedToBigBed. 
		    <A HREF="goldenPath/newsarch.html#090909">Read more</A>.</P>
                        </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
