<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                          <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="https://lists.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>

            
            <!-- start news -->

            <!-- hg19 snp132 release -->
		    <P>
                    <FONT FACE="courier" SIZE="3"><B>18 April 2011 - dbSNP 132 Available for hg19</B></FONT>
		    </P>
                    <P>
                    We are pleased to announce the release of four tracks derived from
                    <a href="http://www.ncbi.nlm.nih.gov/projects/SNP/" target=_blank>dbSNP</a>
                    build 132, available on the human assembly (GRCh37/hg19).
                    dbSNP build 132 is available at NCBI. The new tracks contain
                    additional annotation data not included in previous dbSNP tracks, 
                    with corresponding coloring and filtering options in the Genome
                    Browser.
                    </P>
                    <P>
                    As for prior builds of dbSNP, there is a track that contains all
                    mappings of reference SNPs to the human assembly; it is now labeled
                    &quot;All SNPs (132)&quot;.  Three new tracks have been added to show
                    interesting and easily defined subsets of dbSNP:
                    </P>
                    <ul>
                    <li> Common SNPs (132): uniquely mapped variants that appear in at
                    least 1% of the population</li>

                    <li>Flagged SNPs (132): uniquely mapped variants, excluding Common
                    SNPs, that have been flagged by dbSNP as &quot;clinically associated&quot;</li>

                    <li>Mult. SNPs (132): variants that have been mapped to more than one
                    genomic location</li>
                    </ul>
                    <P>
                    By default, only the Common SNPs (132) are visible; other tracks must
                    be made visible using the track controls.
                    </P>
                    <P>
                    The counts of distinct reference SNPs and their mappings to hg19
                    differ greatly among the tracks:
                    </P>
                    <table border="1" cellpadding="3">
                      <tr>
                        <th>Track</th>
                        <th>Reference SNPs</th>
                        <th>Mappings to hg19</th>
                      </tr>
                      <tr>
                        <td>Common SNPs (132)</td>
                        <td>13,842,381</td>
                        <td>14,024,295 *</td>
                      </tr>
                      <tr>
                        <td>Flagged SNPs (132)</td>
                        <td>17,899</td>
                        <td>18,084 *</td>
                      </tr>
                      <tr>
                        <td>Mult. SNPs (132)</td>
                        <td>1,114,599</td>
                        <td>3,568,988</td>
                      </tr>
                      <tr>
                        <td>All SNPs (132)</td>
                        <td>30,030,754</td>
                        <td>33,026,121</td>
                      </tr>
                    </table>
                    <P>
                    * There are more mappings than SNPs for the Common and Flagged
                    subsets because multiple mappings to alternate haplotype sequences
                    or the pseudoautosomal regions on X and Y don't disqualify a SNP
                    from consideration as uniquely mapped.
		    </P>
                    <P>
                    You will find the four SNPs (132) tracks on the Human Feb. 2009
                    (GRCh37/hg19) browser in the &quot;Variation and Repeats&quot; group.
		    </P>
                    <P>
                    Many thanks to dbSNP at NCBI for the data, and to 
                    <a href="http://www.gs.washington.edu/faculty/king.htm"
                       target=_blank>Mary-Claire King</a>                    
                    at the University of Washington for helpful comments that motivated
                    us to enhance our dbSNP annotations. The tracks were produced at UCSC
                    by Angie Hinrichs, Brooke Rhead, Bob Kuhn, and Jim Kent.
		    </P>

                    <HR>  
            
            <!-- SHEEP BROWSER RELEASE -->
		    <p>
             <font face="courier" size="3"><b>18 April 2011 - Sheep Browser Released</b></font>
		    </p>
		    
		    <p>We are happy to announce a Genome Browser for
		    the initial release of sheep (<em>Ovis aries</em>) genome. The Feb. 2010 <em>Ovis
			aries</em> draft assembly (NCBI project <a
			href="http://www.ncbi.nlm.nih.gov/sites/entrez?Db=genomeprj&cmd=ShowDetailView&TermToSearch=10709"
			target=_blank>10709</a>, <a
			href="ftp://ftp.ncbi.nlm.nih.gov/genbank/genomes/Eukaryotes/vertebrates_mammals/Ovis_aries/
			Ovis_aries_1.0/" target=_blank>GCA_000005525.1</a>) was produced by the <a
			href="http://www.sheephapmap.org/" target=_blank>International Sheep Genomics Consortium (ISGC)
			sequencing center</a>. The mitochondrial genome was sequenced by <a href="http://www.uni-giessen.de/cms/"
			target=_blank>Justus-Liebig-University</a>.</p>
		    
		    <p>There are 27 chromosomes: chr1 - chr26, chrX, plus the mitochondrial sequence chrM from
			genbank accession NC_001941, for a total sequence length of 2,860,512,983 bases. There are
			1,659,241,706 'N' bases in gaps, leaving 1,201,271,277 ACGT bases.</p>
			
			<p>The genome has been sequenced using 454 FLX sequencing to 3X coverage (DNA derived from
			six female sheep each sequenced with 0.5X coverage). Contigs were assembled based on alignment to
			the bovine genome and ordered into ovine chromosomes using BAC-end sequences and the virtual sheep
			genome.</p>
			
			<p>We'd like to thank the International Sheep Genomics Consortium (ISGC)
			sequencing center and Justus-Liebig-University for providing this assembly.
		    We'd also like to acknowledge the UCSC team who worked on 
		    this release: Chin Li, Greg Roe, and Luvina Guruvadoo. 
		    The sheep browser annotation tracks were generated by 
		    UCSC and collaborators worldwide.</p>

		    <hr>

            <p><FONT FACE="courier" SIZE="3"><B>04 March 2011 - The Meaning of Red: Consensus on color scheme for CNVs: </B></FONT>
                    Consensus was reached on color standards to represent CNV loss (red) 
		    and gain (blue) in genomics databases by DGV, NCBI, DECIPHER, UCSC and ISCA 
		    during the 1st annual ISCA Scientific Conference.
		    <A HREF="goldenPath/newsarch.html#030411">Read more</A>
		    </p>
		    
		    <p><FONT FACE="courier" SIZE="3"><B>01 March 2011 - Updated Chimpanzee Browser Released: </B></FONT>
            We are happy to announce the release of a Genome Browser for
		    the latest release of the chimpanzee (<em>Pan 
		    troglodytes</em>) genome.
		    <A HREF="goldenPath/newsarch.html#030111">Read more</A>
		    </p>
                   
                   <p><FONT FACE="courier" SIZE="3"><B>19 January 2011 -
Browser Released for Zebrafish Zv9 Assembly: </B></FONT>
                    The most recent zebrafish assembly -- Zv9 (UCSC version
                    danRer7, Jul. 2010) -- is now available in the UCSC
                    Genome Browser. 
		    <A HREF="goldenPath/newsarch.html#011911">Read more</A>
		    </p>

==> <a href="goldenPath/newsarch.html">News Archives</a>

                    </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
