<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                          <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="http://www.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>
		   
                    <P>
                    <FONT FACE="courier" SIZE=3><B>27 March 2007 - Stickleback
		    Assembly Released in Genome Browser</B></FONT>
                    <P>
		    We have released a Genome Browser and Blat server for 
		    the Feb. 2006 v1.0 draft assembly of <em>Gasterosteus 
		    aculeatus</em> produced by the 
		    <A HREF="http://www.broad.mit.edu/" TARGET=_blank>Broad 
		    Institute</A>.
		    <P>
		    This assembly has been sequenced to approximately 6X 
		    coverage. An estimated  87% of the sequence has been 
		    anchored to chromosomes (chrI - chrXXI). Of the remaining 
		    unanchored scaffolds, those that could be localized to a 
		    chromosome have been concatenated into the virtual 
		    chromosome &quot;chrUn&quot; with 1000bp gaps between 
		    scaffolds. The stickleback mitochondrial sequence is also 
		    available as the virtual chromosome &quot;chrM&quot;.
		    </P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/gasAcu1/">FTP server</A> 
		    or the
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#stickleback">Downloads</A>
		    page.  These data have 
		    <A HREF="../goldenPath/credits.html#stickleback_use">specific
		    conditions for use</A>.
		    </P>
		    <P>
		    The stickleback browser annotation tracks were generated by 
		    UCSC and collaborators worldwide. See the
		    <A HREF="/goldenPath/credits.html#stickleback_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release.
		    </P>
		    <P>
                    <HR>
                    <FONT FACE="courier" SIZE=3><B>26 February 2007 - Zebrafish
		    danRer1 Assembly Archived: </B></FONT>
		    We have archived the initial zebrafish assembly, danRer1 
		    (Nov. 2003). This assembly has been removed from our 
		    primary website, but is still available for browsing and 
		    downloading on our 
		    <A HREF="http://genome-archive.cse.ucsc.edu/">archive server</A>. 
		    Because of this move, blat searches on this assembly are no 
		    longer supported.
		    </P>
		    <P>
                    <FONT FACE="courier" SIZE=3><B>16 February 2007 - New 
                    Browser Session-Sharing Function Available: </B></FONT>
                    We are pleased to announce the release of a new session 
                    management functionality 
                    in the Genome Browser, which allows users to save and
                    share browser sessions.
                    <A HREF="goldenPath/newsarch.html#021607">Read more</A>.</P>
		    <P>
                    <FONT FACE="courier" SIZE=3><B>2 February 2007 - New Genome 
                    Graphs Tool Available in Genome Browser: </B></FONT>
                    We are pleased to announce the release of a new software 
                    tool in the Genome Browser collection, the 
                    <A HREF="cgi-bin/hgGenome">Genome Graphs</A> 
                    tool. Genome Graphs offers the ability to 
                    upload and display genome-wide data sets such as the 
                    results of genome-wide SNP association studies, linkage 
                    studies and homozygosity mapping. 
                    <A HREF="goldenPath/newsarch.html#020207">Read more</A>.</P>
                        </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
