<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                         <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="https://lists.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>

            
            <!-- start news -->

		<FONT FACE="courier" SIZE="3"><B>3 January 2012 - Roadmap 
		Epigenomics Now Available through Data Hub at Washington 
		University</B></FONT>
		   <P>
		   We are pleased to announce the release of the Roadmap 
		   Epigenomics data on the UCSC Genome Browser through our 
		   Data Hub function. The Roadmap Epigenomics Project is part 
		   of The NIH Common Fund's <a href="https://commonfund.nih.gov/epigenomics/" 
		   target="_blank">Epigenomics Program</a>. 
		   It was launched with the goal of producing a public resource 
		   of human epigenomic data to catalyze basic biology
		   and disease-oriented research. The Consortium leverages 
		   experimental pipelines built around next-generation sequencing 
		   technologies to map DNA methylation, histone modifications, 
		   chromatin accessibility and small RNA transcripts in stem
		   cells and primary ex vivo tissues selected to represent the 
		   normal counterparts of tissues and organ systems frequently 
		   involved in human disease. The Consortium expects to deliver 
		   a collection of normal epigenomes that will provide a framework 
		   or reference for comparison and integration within a broad
		   array of future studies.
		   <P>
		   All data were produced and processed by the 
		   <a href="http://www.roadmapepigenomics.org/" target="_blank">
		   Roadmap Epigenomics Mapping Consortium</a>, and will be periodically
		   updated. Genome Browser tracks were constructed and hosted by 
		   <a href="http://vizhub.wustl.edu/" 
		   target="_blank">VizHub</a> at Washington University in St. Louis. 
		   Tracks are available at UCSC Genome Browser via the Data Hub 
		   function, or follow this 
		   <a href="http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&hubUrl=
			http://vizhub.wustl.edu/VizHub/RoadmapRelease4.txt" target="_blank">link</a>. 
		   The Roadmap Epigenomics Mapping Consortium is responsible for 
		   the quality of the data.
		<P>
                <FONT FACE="courier" SIZE="3"><B>19 December 2011 - Variant
                Call Format (VCF) Now Supported in Genome Browser</B></FONT>
                   <P>
                   We are pleased to announce that the UCSC Genome Browser
                   now supports Variant Call Format (VCF).  VCF is a flexible
                   and extendable line-oriented text format developed by the
                   1000 Genomes Project for releases of single nucleotide variants,
                   indels, copy number variants and structural variants discovered
                   by the project.  Similar to bigBed, bigWig and BAM, the Browser
                   transfers only the portions of VCF files necessary to display
                   viewed regions, making VCF a fast and attractive option for
                   large data sets.  VCF files will need to be compressed and
                   indexed using the tabix package available from
                   <a href="http://samtools.sourceforge.net/"
                   target="_blank">SAMtools</a>.  This new format is available
                   for use in custom tracks and data hubs.  For more information
                   about VCF and tabix, please see our <a href="goldenPath/help/vcf.html"
                   target="_blank">VCF Track Format help page</a>.</P>
                <P>
        	<FONT FACE="courier" SIZE="3"><B>7 November 2011 - UMD Cow
		Assembly Now Available in Genome Browser</B></FONT>
		    <P>
		    In response to the many requests from the bovine research
		    community, we now offer the University of Maryland (UMD)
		    <em>Bos taurus</em> assembly in the Genome Browser, in 
		    addition to the assemblies produced by the Baylor College 
		    of Medicine Human Genome Sequencing Center. The UMD v3.1 
		    assembly (Nov. 2009, UCSC bosTau6) was produced by the 
		    <A HREF="http://www.cbcb.umd.edu/" TARGET=_blank>UMD Center 
		    for Bioinformatics and Computational 
		    Biology</A> (CBCB) in College Park, MD. </P>
		    <P>
		    The UMD 3.1 assembly is based on reads downloaded from 
		    the NCBI Trace Archive, most of which were sequenced by
		    the Baylor College of Medicine. UMD reports that they were
		    able to apply several assembly improvement techniques
		    to produce an assembly that is substantially more accurate 
		    and complete than previous alternative versions, as 
		    evaluated using independent metrics. Based on UMD 
		    statistics, this assembly provides more genome coverage, 
		    closes thousands of gaps, corrects many erroneous 
		    inversions, deletions, and translocations, and fixes 
		    thousands of single-nucleotide errors. For more information,
		    see Zimin AV <em>et al</em>, 
		    <A HREF="http://genomebiology.com/2009/10/4/R42"
		    TARGET=_blank>A whole-genome assembly of the domestic
		    cow, <em>Bos taurus</em></A>, <em>Genome Biol.</em> 2009;10(4) and 
		    the UMD CBCB 
		    <A HREF="http://www.cbcb.umd.edu/research/bos_taurus_assembly.shtml"
		    TARGET=_blank><em>Bos taurus</em> assembly</A> web page.</P>
		    <P>
		    Downloads of the UMD bovine data and annotations may be 
		    obtained from the UCSC Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/bosTau6/">FTP server</A> 
		    or <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#cow">Downloads</A> page.
		    </P>
		    <P>
              	    We'd like to thank the UMD CBCB for providing this genome 
		    assembly. The UMD Cow Genome Browser and annotation tracks 
		    were produced by Hiram Clawson, Chin Li, Brooke Rhead, Greg 
		    Roe, Steve Heitner, and Donna Karolchik.  See the 
		    <A HREF="goldenPath/credits.html#cow_credits">Credits</A> 
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release.  </P>
		    <P>
		    <HR>

            <!-- start archives -->
            <!-- sacCer3 Assembly Release -->
        	<FONT FACE="courier" SIZE="3"><B>10 October 2011 - Updated Yeast Browser Released: </B></FONT>
		    We are happy to announce the release of an updated browser 
		    for the <em>Saccharomyces cerevisiae</em> yeast genome. 
		    <A HREF="goldenPath/newsarch.html#101011">Read more</A>.
		    </P>
            
            <!-- New javascript features -->
		    <P>
                    <FONT FACE="courier" SIZE="3"><B>8 September 2011 -
New Navigation and Display Features: </B></FONT>
		    We've added several new features to the Genome Browser 
		    that make it easier to quickly configure and navigate 
		    around in the browser's annotation tracks window.
		    <A HREF="goldenPath/newsarch.html#090811">Read more</A>.

            <!-- data hubs release  -->
		    <P>
                    <FONT FACE="courier" SIZE="3"><B>18 August 2011 -
New Feature: Track Data Hubs:  </B></FONT>
                    We are pleased to announce a new feature in the UCSC Genome 
   		    Browser: Track Data Hubs. Track hubs are web-accessible 
		    directories of genomic data that can be viewed on the UCSC 
		    Genome Browser alongside native annotation tracks. 
		    <A HREF="goldenPath/newsarch.html#081811">Read more</A>.
		    </P>

==> <a href="goldenPath/newsarch.html">News Archives</a>

                    </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
