<!-- News Section ============================================- -->  
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
    <TR><TD>        
    <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
        <TR><TD>        
        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" title="Follow &#64;GenomeBrowser on Twitter" alt="Follow &#64;GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a>
                 <a title="Genome Browser Facebook page"
                 href="http://www.facebook.com/ucscGenomeBrowser"
                 target="_blank"><img
                 src="/images/FB-f-Logo__blue_22.jpg"
                 style="vertical-align:text-bottom; margin-left: 15px;"
                 alt="Genome Browser Facebook page"></a>
                 <a href="https://plus.google.com/107643636314131661344" rel="publisher"></a>
               </TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list.</P>

    <!-- start news -->
	<hr>
	<font face="courier" size="3"><b>30 September 2014 -
	Genome Browser and Portal Released for the Ebola Virus</b>
	</font>
	<p>
        We are excited to announce the release of a Genome Browser and information portal for the 
	Aug. 2014 assembly of the Ebola virus (UCSC version eboVir3, GenBank accession 
	<a href="http://www.ncbi.nlm.nih.gov/nuccore/KM034562" target="_blank">KM034562</a>) 
	submitted by the Broad Institute. We have worked closely with the Pardis Sabeti lab at 
	the Broad Institute and other Ebola experts throughout the world to incorporate
	annotations that will be useful to those studying Ebola. Annotation tracks included in this
	initial release include genes from <a href="http://www.ncbi.nlm.nih.gov/"
	target="_blank">NCBI</a>, B- and T-cell epitopes from the <a href="http://www.iedb.org/"
	target="_blank">IEDB</a>, structural annotiations from <a href="http://www.uniprot.org/"
	target="_blank">UniProt</a> and a wealth of SNP data from the
	<a href="http://www.ncbi.nlm.nih.gov/pubmed/25214632" target="_blank">2014 publication</a>
	by the Sabeti lab. This initial release also contains a 160-way alignment
	comprising 158 Ebola virus sequences from various African outbreaks and 2 Marburg virus 
	sequences.
	You can find links to the Ebola virus Genome Browser and more information on the Ebola virus
	itself on our <a href="/ebolaPortal/">Ebola Portal</a> page.
        </p>
        <p>
        Bulk downloads of the sequence and annotation data are available via the Genome
        Browser
        <a href="ftp://hgdownload.cse.ucsc.edu/goldenPath/eboVir3/">FTP server</a>
        or the
        <a href="http://hgdownload.cse.ucsc.edu/downloads.html#ebola_virus">Downloads</a>
        page. 
        The Ebola virus (eboVir3) browser annotation tracks were generated by UCSC and
        collaborators worldwide. See the
        <a href="goldenPath/credits.html#ebola_credits">Credits</a> page for a detailed
        list of the organizations and individuals who contributed to this release and 
	the conditions for use of these data.
	</p>

	<hr>
	<font face="courier" size="3"><b>23 September 2014 -
	Ensembl Regulatory Build Now Available in the Genome Browser</b>
	</font>
	<p>
	We are pleased to announce that data from the
	<a href="http://www.ensembl.org/info/genome/funcgen/regulatory_build.html"
	target="_blank">Ensembl Regulatory Build</a> is now available in the UCSC Genome Browser
	as a public track hub for both hg19 and hg38. This track hub contains promoters and their
	flanking regions, enhancers, and many other regulatory features predicted across a number
	of cell lines using annotated segmentation states. Click
	<a href="/cgi-bin/hgTracks?db=hg19&position=chr21%3A33031597-33041570&hubUrl=http://ngs.sanger.ac.uk/production/ensembl/regulation/hub.txt">here</a>
	to view this hub on the hg19 assembly, or click
	<a href="/cgi-bin/hgTracks?db=hg38&position=chr9%3A133252000-133280861&hubUrl=http://ngs.sanger.ac.uk/production/ensembl/regulation/hub.txt">here</a>
	to view it on the hg38 assembly.
	</p>
	<p>
	We would like to thank our collaborators Daniel Zerbino, Steve Wilder, and Nathan Johnson
	from Ensembl and Kate Rosenbloom and Matthew Speir from UCSC for their efforts in
	bringing these resources to the UCSC Genome Browser.
	</p>
        <hr>

        <!-- staged for archive use.  NOTE: condense header line from .... </b></font><p> to </b>: </font>

        <p>
	<font face="courier" size="3"><b>30 September 2014 -
        Genome Browser for the Ebola Virus Now Available</b>: </font>
        We are excited to announce the release of a Genome Browser for the June 2014 assembly
        of Ebola Virus (Sierra Leone 2014 (G3683/KM034562.1), UCSC version eboVir3).
	</p>

	<p>
        <font face="courier" size="3"><b>23 September 2014 -
        Ensembl Regulatory Build Now Available in the Genome Browser</b>: </font>
        We are pleased to announce that data from the
        <a href="http://www.ensembl.org/info/genome/funcgen/regulatory_build.html"
        target="_blank">Ensembl Regulatory Build</a> is now available in the UCSC Genome Browser
        as a public track hub for both hg19 and hg38.
	<a href="goldenPath/newsarch.html#092314">Read more</a>.
	</p>
	-->

        <!-- start archives -->

        <p>
        <font face="courier" size="3"><b>04 September 2014 -
        New ENCODE portal launched at
        <a href="https://www.encodeproject.org">www.encodeproject.org</a></b>: </font>
        The website for up-to-date information about the ENCODE project is no
        longer hosted at <a href="ENCODE/index.html">genome.ucsc.edu/ENCODE</a>.
        <a href="goldenPath/newsarch.html#090414">Read more</a>.
        </p>
	
	<p>
	<font face="courier" size="3"><b>03 September 2014 -
	Data from the FaceBase Consortium Now Available in the Genome Browser</b>: </font>
	We are pleased to announce that much of the data from the <a href="https://www.facebase.org/"
	target"_blank">FaceBase Consortium</a> is now available in the UCSC Genome Browser as a public track hub.
        <a href="goldenPath/newsarch.html#090314">Read more</a>.
	</p>

            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
