<!-- News Section ============================================- -->  
<TABLE style="background-color:#888888; width: 100%; border-width: 0px; border-spacing: 0px; padding: 0px">
    <TR><TD>        
    <TABLE style="background-color:#fffee8; width: 100%; border-width: 0px; border-spacing: 0px; padding: 0px">
        <TR><TD>        
        <TABLE style="background-color:#D9E4F8; background-image: url('images/hr.gif'); width: 100%; border: 0px">
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" title="Follow &#64;GenomeBrowser on Twitter" alt="Follow &#64;GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a>
                 <a title="Genome Browser Facebook page"
                 href="http://www.facebook.com/ucscGenomeBrowser"
                 target="_blank"><img
                 src="/images/FB-f-Logo__blue_22.jpg"
                 style="vertical-align:text-bottom; margin-left: 15px;"
                 alt="Genome Browser Facebook page"></a>
                 <a href="https://plus.google.com/107643636314131661344" rel="publisher"></a>
               </TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE style="background-color:#fffee8; width: 100%; padding: 0px">
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list. Please see our
          <a href="http://genome.ucsc.edu/blog"
          target=_blank>blog</a> for posts about Genome Browser tools, features, projects and more.
          </p>

    <!-- start news -->
<!-- Staged for later release
  <hr>
  <font color="#006666" size="4"><b>7 Apr 2016
  - Data from the Lens PatSeq Database Now Available</b></font>
  <p>
  We are pleased to announce the release of a set of tracks containing data from the
  <a href="https://www.lens.org/" target="_blank">Lens</a>
  PatSeq database in the UCSC Genome Browser. These tracks contain data from the
  <a href="https://www.lens.org/lens/bio" target="_blank">PatSeq</a>
  database mapped to the genome and then colored based on where the sequence appears in
  the patent document. Details pages contain the patent titles with links to the patents in the
  Lens database. The data from PatSeq is divided into two tracks, one includes those sequences
  that were submitted in bulk patents containing 100 or more sequences and the other containing
  all other sequences. These tracks are currently present on the human (hg19), mouse (mm10),
  and Ebola virus (eboVir3) genomes.
  </p>

  <p>
  Thanks to our collaborators at the Lens, Osmat Jefferson and Deniz Koellhofer, for providing the
  data and feedback on the visualization. Thank you to Max Haeussler and Matthew Speir, members of
  the UCSC Genome Browser team, for their efforts in creating these tracks.
  </p>
-->

<!-- Staged for upcoming release
  <hr>
  <a name="041516"></a>
  <p>
  <font color="#006666" size="4"><b> 15 April 2016 -
  Three New Assemblies Now Available in the Genome Browser</b></font>
  <p>
  A Genome Browser is now available for the gorilla (<em>Gorilla gorilla gorilla</em>)
  assembly released Dec 2014 by the Wellcome Trust Sanger Institute. 
  For more information and statistics about this assembly, see the NCBI assembly record
  for <a href="http://www.ncbi.nlm.nih.gov/assembly/503571" target="_blank">gorGor4</a>.  
  There are 40,730 scaffolds with a total size of 3,063,362,754 bases.
  </p>

  <p>
  Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
  <a href="ftp://hgdownload.soe.ucsc.edu/goldenPath/gorGor4/">FTP server</a> or the
  <a href="http://hgdownload.soe.ucsc.edu/downloads.html#gorilla">Downloads</a> page.
  Please observe the <a href="credits.html#gorilla_use">conditions for use</a> when
  accessing and using these data sets.  The annotation tracks for this browser were
  generated by UCSC and collaborators worldwide.  See the <a
  href="credits.html#gorilla_credits">Credits</a> page for a detailed list of the
  organizations and individuals who contributed to this release.
  </p>
  - - -
  <p>
  A Genome Browser is now available for the rhesus monkey (<em>Macaca mulatta</em>)
  assembly released Nov 2015 by the 
<A HREF="https://www.hgsc.bcm.edu/non-human-primates/rhesus-monkey-genome-project"
TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center</A>. 
  For more information and statistics about this assembly, see the NCBI assembly record
  for <a href="http://www.ncbi.nlm.nih.gov/assembly/2715158" target="_blank"> Mmul_8.0.1</a>.  
There are 286,263 scaffolds with a total size of 3,236,224,332 bases.
  </p>

  <p>
  Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
  <a href="ftp://hgdownload.soe.ucsc.edu/goldenPath/rheMac8/">FTP server</a> or the
  <a href="http://hgdownload.soe.ucsc.edu/downloads.html#rhesus">Downloads</a> page.
  Please observe the <a href="credits.html#rhesus_use">conditions for use</a> when
  accessing and using these data sets.  The annotation tracks for this browser were
  generated by UCSC and collaborators worldwide.  See the <a
  href="credits.html#rhesus_credits">Credits</a> page for a detailed list of the
  organizations and individuals who contributed to this release.
  </p>
  - - -
  <p>
  A Genome Browser is now available for the crab-eating macaque (<em>Ornithorhynchus anatinus</em>)
  assembly released February 2007 by the 
<A HREF="http://genome.wustl.edu/genomes/detail/macaca-fascicularis/"
TARGET=_blank>Genome Sequencing Center at Washington University, St. Louis</A>.
For more information about this assembly, see
<A HREF="http://www.ncbi.nlm.nih.gov/assembly/704988" TARGET="_blank">
Macaca_fascicularis_5.0</A> in the NCBI Assembly database. There are 
7,625 scaffolds with a total size of 2,946,843,737 bases.
 </p>

  <p>
  Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
  <a href="ftp://hgdownload.soe.ucsc.edu/goldenPath/macFas5/">FTP server</a> or the
  <a href="http://hgdownload.soe.ucsc.edu/downloads.html#crab-eating_macaque">Downloads</a> page.
  Please observe the <a href="credits.html#crab-eating_macaque_use">conditions for use</a> when
  accessing and using these data sets.  The annotation tracks for this browser were
  generated by UCSC and collaborators worldwide.  See the <a
  href="credits.html#crab-eating_macaque_credits">Credits</a> page for a detailed list of the
  organizations and individuals who contributed to this release.
  </p>
  -->

  <hr>
  <font color="#006666" size="4"><b>4 Apr 2016
  - New video shows how to obtain sequence or coordinates for exons. </b></font>
  <p>
  We have released a new <A HREF =
  "http://genome.ucsc.edu/training/vids/index.html#vid08" TARGET =
  _BLANK>video</A> to our YouTube channel.  This newest installment in our effort
  to share the capabilities of the Genome Browser shows how to
  use the Table Browser to obtain either the sequences or the coordinates for
  exons in the genes from a chosen region of the genome.
  </p>

  <p>
  We also want to take this opportuinty to remind you that we offer
  on-site <A HREF = "http://genome.ucsc.edu/training/index.html" TARGET =
  _BLANK>workshops</A> to help you and your colleagues make the most out
  of the Genome Browser.
  </p>
  <hr>

  <font color="#006666" size="4"><b>1 Apr 2016
  - Going Back to Our Roots</b></font>
  <p>
  For many years, the UCSC Genome Browser has provided a dynamic, cutting-edge interface for
  viewing genomic data.  As time has gone by, however, it has become clear to us that we
  cannot keep this up.  It is not fair to you, our users, that our interface keeps changing,
  and it has increasingly been a burden on us to keep up with emerging web technologies.  Today
  that all changes, as we return you to the roots of the web: ASCII.
  <p>
  <center><img src="images/joke04012016.png" width="65%"></center>
  <p>
  We apologize again for subjecting you to so many changes over the years, and hope that this
  new, more consistent interface will find your approval.
  <p>
  In related news, we have also revised the interface for uploading custom tracks.  Rather than
  forcing you to wrestle with inconvenient file formats for your data or attempt to set up a web server,
  we now provide the option of loading your data from a deck of ISO 6586:1980 punch cards.  Simply
  punch your data onto a deck and mail the deck to us.  Our staff of highly trained specialists
  will take care of loading track, and will inform you by mail when the process is complete.  Please
  note that our current policy of expiring unused custom tracks remains in effect; you must visit
  your track within 72 hours of the postmark date.  Regrettably, decks cannot be returned.
  <p>
  Card decks may be addressed to:<br>
  UCSC Genome Informatics Group<br>
  Center for Biomolecular Science & Engineering<br>
  CBSE, 501D Engineering II Building<br>
  University of California, Santa Cruz<br>
  1156 High Street<br>
  Santa Cruz, CA 95064
  <p>
  This new file format is also an option for data output from the <a href="cgi-bin/hgTables">UCSC Table Browser</a>.
  Simply select "mail card deck" from the "output format" menu, and then enter your name and address on the subsequent
  page.  Please allow 4-6 weeks for delivery.
  <hr>

<!-- start archives -->
<!--staged for future release
	<p>
	<font color="#006666" size="4"><b>15 April 2016
	- Genome Browsers for Gorilla, Rhesus, and Crab-Eating Macacque Now Available.</b></font>
	<a href="goldenPath/newsarch.html#041516">Read more</a>.
	</p>
	

        <p>
	<font color="#006666" size="4"><b>7 Apr 2016
	- Data from the Lens PatSeq Database Now Available</b>: </font>
	We are pleased to announce the release of a set of tracks containing data from the
	<a href="https://www.lens.org/" target="_blank">Lens</a>
	PatSeq database in the UCSC Genome Browser.
	<a href="goldenPath/newsarch.html#040716">Read more</a>.

  <p>
	<font color="#006666" size="4"><b>4 Apr 2016
        - New video shows how to obtain sequence or coordinates for exons. </b></font>
	<a href="goldenPath/newsarch.html#040416">Read more</a>.
	</p>

  <p>
  <font color="#006666" size="4"><b>1 Apr 2016
  - Going Back to Our Roots!</b></font>
  <a href="goldenPath/newsarch.html#040116">Read more</a>.
  </p>
-->
	<p>
	<font color="#006666" size="4"><b>21 Mar 2016
	- Genome Browsers for <i>C. elegans</i> and Brown Kiwi Now Available.</b></font>
	<a href="goldenPath/newsarch.html#030816">Read more</a>.
	</p>

	<p>
	<font color="#006666" size="4"><b>8 Mar 2016
	- Combine Multiple Regions of the Genome Browser into a Single Visualization!</b></font>
	<a href="goldenPath/newsarch.html#030816">Read more</a>.
	</p>

            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
