<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                          <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="https://lists.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>

            
            <!-- start news -->

            <!-- OpenHelix Webinar  -->
		    <P>
                    <FONT FACE="courier" SIZE="3"><B>13 May 2011 - Free Browser Training 
                     Webinar from OpenHelix </B></FONT>
		    </P>
                    <P>
                  The UCSC Bioinformatics Group announces two free webinars 
                  on the UCSC Genome Browser. The 
                  webinars will be conducted by OpenHelix, a provider of 
                  training on many free, publicly accessible bioinformatics 
                  and genomics resources. 
                  </P>
                  <P>
                  The webinars (one hour and 15 minutes long) will cover the 
                  topics needed to effectively use the Genome Browser.  The 
                  first webinar, to be held Tuesday, May 24, 11:00-12:15 PM PDT,  
                  <B>Introduction to the UCSC Genome Browser</B>, is designed 
                  for new users of the UCSC Genome Browser, and those who want 
                  to improve their skills at basic navigation and display.
                  </P>
                  <P>
                  The second webinar, <B>UCSC Genome Browser: Custom Tracks 
                  and Table Browser</B> will be held Thursday, May 26, 1:00-2:15 PM PDT.   
                  </P>
                  <P>Signup required.  For details see 
                   <A HREF = "http://blog.openhelix.eu/?p=8359" TARGET = _BLANK>Open Helix</A>.

		    </P>
                    <HR>  

            <!-- ENCODE usability survey -->
		    <P>
                    <FONT FACE="courier" SIZE="3"><B>1 May 2011 - ENCODE User's Guide and Usability Survey</B></FONT>
		    </P>
                    <P>
                  The ENCODE project has just published an overview of their ongoing large-scale
                  efforts to interpret the human genome sequence in the journal PLoS Biology.
                  <a target=_blank href="http://www.plosbiology.org/article/info:doi/10.1371/journal.pbio.1001046 ">
                  A Users Guide to the Encyclopedia of DNA Elements</a>,
                  highlights the scope of data production, and provides guidance for locating 
                  and using the data.  With the vast amount of data now available, the ENCODE Data Coordination Center
                  at UCSC continues to focus on improving the accessiblity and usability of this valuable resource.
                  We invite all current and prospective users of ENCODE data to participate in the 
                    <a target=_blank href="http://www.surveymonkey.com/s/5YYTRWQ">2011 ENCODE Usability Survey</a>. 
                    Your input will help us to make this data more accessible to the scientific community. Thank you!
		    </P>
                    <HR>  

            <!-- hg19 snp132 release -->
		    <P>
                    <FONT FACE="courier" SIZE="3"><B>18 April 2011 - dbSNP 132 Available for hg19</B></FONT>
		    </P>
                    <P>
                    We are pleased to announce the release of four tracks derived from
                    <a href="http://www.ncbi.nlm.nih.gov/projects/SNP/" target=_blank>dbSNP</a>
                    build 132, available on the human assembly (GRCh37/hg19).
                    dbSNP build 132 is available at NCBI. The new tracks contain
                    additional annotation data not included in previous dbSNP tracks, 
                    with corresponding coloring and filtering options in the Genome
                    Browser.
                    </P>
                    <P>
                    As for prior builds of dbSNP, there is a track that contains all
                    mappings of reference SNPs to the human assembly; it is now labeled
                    &quot;All SNPs (132)&quot;.  Three new tracks have been added to show
                    interesting and easily defined subsets of dbSNP:
                    </P>
                    <ul>
                    <li> Common SNPs (132): uniquely mapped variants that appear in at
                    least 1% of the population</li>

                    <li>Flagged SNPs (132): uniquely mapped variants, excluding Common
                    SNPs, that have been flagged by dbSNP as &quot;clinically associated&quot;</li>

                    <li>Mult. SNPs (132): variants that have been mapped to more than one
                    genomic location</li>
                    </ul>
                    <P>
                    By default, only the Common SNPs (132) are visible; other tracks must
                    be made visible using the track controls.
                    </P>
                    <P>
                    The counts of distinct reference SNPs and their mappings to hg19
                    differ greatly among the tracks:
                    </P>
                    <table border="1" cellpadding="3">
                      <tr>
                        <th>Track</th>
                        <th>Reference SNPs</th>
                        <th>Mappings to hg19</th>
                      </tr>
                      <tr>
                        <td>Common SNPs (132)</td>
                        <td>13,842,381</td>
                        <td>14,024,295 *</td>
                      </tr>
                      <tr>
                        <td>Flagged SNPs (132)</td>
                        <td>17,899</td>
                        <td>18,084 *</td>
                      </tr>
                      <tr>
                        <td>Mult. SNPs (132)</td>
                        <td>1,114,599</td>
                        <td>3,568,988</td>
                      </tr>
                      <tr>
                        <td>All SNPs (132)</td>
                        <td>30,030,754</td>
                        <td>33,026,121</td>
                      </tr>
                    </table>
                    <P>

                    * The Common and Flagged tracks have more mappings than SNPs due to the
                    alternate haplotype sequences and the pseudoautosomal regions on X and Y.
                    SNPs are considered uniquely mapped if they map only once to a haploid
                    reference genome.  These regions add non-haploid sequence to the reference
                    genome; therefore, multiple mappings involving these regions are still 
                    considered unique.
		    </P>
                    <P>
                    You will find the four SNPs (132) tracks on the Human Feb. 2009
                    (GRCh37/hg19) browser in the &quot;Variation and Repeats&quot; group.
		    </P>
                    <P>
                    Many thanks to dbSNP at NCBI for the data, and to 
                    <a href="http://www.gs.washington.edu/faculty/king.htm"
                       target=_blank>Mary-Claire King</a>                    
                    at the University of Washington for helpful comments that motivated
                    us to enhance our dbSNP annotations. The tracks were produced at UCSC
                    by Angie Hinrichs, Brooke Rhead, Bob Kuhn, and Jim Kent.
		    </P>

		    <hr>
		    <p><FONT FACE="courier" SIZE="3"><B>18 April 2011 - Sheep Browser Released:</B></FONT>
		    We are happy to announce a Genome Browser for
		    the initial release of the sheep (<em>Ovis aries</em>) genome. 
		    <A HREF="goldenPath/newsarch.html#041811.1">Read more</A>
		    </p>
                   
            <p><FONT FACE="courier" SIZE="3"><B>04 March 2011 - The Meaning of Red: Consensus on color scheme for CNVs: </B></FONT>
                    Consensus was reached on color standards to represent CNV loss (red) 
		    and gain (blue) in genomics databases by DGV, NCBI, DECIPHER, UCSC and ISCA 
		    during the 1st annual ISCA Scientific Conference.
		    <A HREF="goldenPath/newsarch.html#030411">Read more</A>
		    </p>
		    
==> <a href="goldenPath/newsarch.html">News Archives</a>

                    </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
