<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Bioinformatics Home</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">

<SCRIPT LANGUAGE="JavaScript">
<!--

var theDb = 'hg12';

function setDb()
{
  theDb = document.dbForm.genome.options[document.dbForm.genome.selectedIndex].value;
}

function initDb()
{
  theDb = document.dbForm.genome.options[document.dbForm.genome.selectedIndex].value;
  var queryString = document.location.search.substring(1, document.location.search.length);
  var dbIndex = queryString.indexOf('db=');
  if (dbIndex >= 0)
  {
    theDb = queryString.substring(dbIndex + 3);
    for (var items = 0; items < document.dbForm.genome.options.length; items++)
    {
      if(sameOrganism(theDb, document.dbForm.genome.options[items].value))
      {
        document.dbForm.genome.options[items].selected = true;
      }
    }     
  }
}

function sameOrganism(db1, db2)
{
  if (db1.substring(0, 2) == db2.substring(0, 2))
  {
    return true;
  }
  else
  {
    return false;
  }
}

function dbGo(theUrl)
{
document.location = theUrl + 'db=' + theDb;
}

//-->
</SCRIPT>


</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF" onLoad="initDb()">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg"></TH></TR>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
	<table bgcolor="000000" cellpadding="1" cellspacing="1" width="99%" height="27"><tr bgcolor="2636D1"><td valign="middle">
	<table BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24">
		<TR><TD VALIGN="middle"><font color="#89A1DE">&nbsp; 
			<a href="javascript:dbGo('/cgi-bin/hgGateway?');" class="topbar">
				Genome Browser</a> &nbsp; - &nbsp; 
			<A HREF="javascript:dbGo('/cgi-bin/hgBlat?command=start&');" class="topbar">
				Blat Search</a> &nbsp; - &nbsp; 
			<A HREF="FAQ.html" class="topbar">
				FAQ</a> &nbsp; - &nbsp; 
			<A HREF="goldenPath/help/hgTracksHelp.html" class="topbar">
				User Guide</a> &nbsp; </font></TD>
    	</TR></TABLE>
	</TD></TR></TABLE>
</TD></TR>

<TR HEIGHT="100%" VALIGN="top">
  <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
  	<!-- outer table is to align leftbar & content boxes at top of page -->
  	<TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0 MARGIN=0>
	<TR><TD><IMG SRC="images/DOT.gif" HEIGHT="7"></TD></TR>
	<TR HEIGHT="100%"><TD HEIGHT="100%">
		<!--outer table is for black border-->
		<TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
   	 	<TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
		<TR HEIGHT="100%">
			<TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%"> 
				<FORM NAME="dbForm">
			<table cellpadding="" cellspacing="0" margins="0" border="0"><tr><td valign="top">
				<FONT CLASS="leftbar">Organism</FONT>
			</td></tr><tr valign="bottom"><td valign="bottom">
					<SELECT NAME="genome" onChange="setDb();">
						<OPTION VALUE="hg12">Human
						<OPTION VALUE="mm2">Mouse 
					</SELECT>
			</td></tr></table><hr>
			<A CLASS="leftbar" HREF="javascript:dbGo('/cgi-bin/hgGateway?');">
				Browser</A><hr>
			<A CLASS="leftbar" HREF="javascript:dbGo('/cgi-bin/hgBlat?command=start&');">
				Blat</A><hr>
			<A CLASS="leftbar" HREF="javascript:dbGo('/cgi-bin/hgText?');">
				Tables</A><hr>
			<A CLASS="leftbar" HREF="downloads.html">
				Downloads</A><hr>
		  	<A CLASS="leftbar" HREF="mirror.html">
				Mirrors</A><hr>
  			<A CLASS="leftbar" HREF="http://genome-archive.cse.ucsc.edu">
				Archives</A><hr>
			<A CLASS="leftbar" HREF="goldenPath/credits.html">
				Credits</A><hr>
			<A CLASS="leftbar" HREF="cite.html">
				Cite Us</A><hr>
			<A CLASS="leftbar" HREF="goldenPath/pubs.html">
				Pubs</A><hr>
			<A CLASS="leftbar" HREF="jobs.html">
				Jobs</A><hr>
  			<A CLASS="leftbar" HREF="bestlinks.html">
				Links</A><hr>
                                </FORM>
			<BR>
			<IMG SRC="images/DOT.gif" HEIGHT=1100 WIDTH=1>
		</TD></TR></TABLE>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
</TD>

<!--Content Tables-->  
  <TD VALIGN="top"><TABLE CELLPADDING=5 BORDER=0><TR><TD>
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<A NAME="About"></A><FONT SIZE="4"><b>&nbsp; About the UCSC Genome
	Bioinformatics Site</FONT>
	</TD></TR></TABLE>
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD WIDTH=10></TD><TD>
		<P>This site contains working drafts for the human genome and
		the mouse genome. The human genome is over 90% complete. Approximately 
		85% of the sequence is in a highly accurate 'finished' state.  The
		other 15% is merely 'draft' quality.  Some care must be taken
		when interpreting draft regions, but these are still very 
		useful to the working scientist.  
		<P>The mouse genome is
		largely the result of a full genome shotgun assembly. While there are 
		many gaps in the sequence, the order and orientation of the contigs on 
		either side of a gap are known in almost all cases. We estimate
		that 90-96% of the mouse genome is present in the
		assembly. We encourage you to explore the
		working drafts with the genome browser, which displays
		the work of many annotators worldwide.
		<P>The Genome Browser and the data it displays are freely available for 
		academic, nonprofit, and personal use with a few exceptions. Check the
		README.txt file in the download directory of a particular
		assembly to view the use restrictions
		specific to that release. Interactive use is limited to a 
		maximum of 10,000 hits/day or 10 hits/minute. 
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100% BORDER=0><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; News </FONT></TD>
			<TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD WIDTH=10></TD><TD>

	<P><FONT FACE="courier" SIZE=2>Sep. 19, 2002</FONT><BR>
	We're pleased to announce the release of the Mouse Cons (Human/Mouse
	Evolutionary Conservation Score) annotation track for the <A
	HREF="http://genome.cse.ucsc.edu/cgi-bin/hgTracks?db=hg12">June 2002
	human genome assembly</A>. This track allows a user to interactively explore 
	conservation between the human and mouse genomes and identify highly conserved 
	regions. Highest levels of conservation are typically seen over coding exons. 
	High levels of conservation are also frequently associated with noncoding RNA,
	promoters, other regulatory elements, and pseudogenes.

	<P>The Mouse Cons annotation is displayed using a new type of Genome
	Browser graphical track that plots a continuous function along a chromosome. The
	conservation levels are calculated over 50bp  windows in the human genome
	that have at least 15 bp aligned to mouse. The score for a window reflects
	the probability that the level of observed conservation in that 50bp region
	would occur by chance under neutral evolution. This information is given on
	a logarithmic scale and displays in the track as "mountain ranges". Details
	pages associated with the individual peaks in the track provide access to
	the base level alignments for the whole region and for the individual 50bp
	windows.

	<P>The team that produced this track includes Ryan Weber, Krishna Roskin,
	Mark Diekhans, Jim Kent, Scott Schwartz, and Webb Miller. 

	<P><FONT FACE="courier" SIZE=2>Sep. 12, 2002</FONT><BR>
	<I>Nature Genetics</I> has just published <A
	HREF="http://www.nature.com/cgi-taf/DynaPage.taf?file=/ng/journal/v32/n1s/index.html"
	TARGET=_blank>User's Guide to the Human Genome</A>, a
	hands-on tutorial for using genome browsers as web tools for browsing and
	analyzing data from the Human Genome Project and other sequencing efforts. The 
	3 browsers featured in the tutorial include the UCSC Genome Browser, NCBI's 
	Map Viewer, and the Ensembl Genome Browser. The guide is organized around a 
	collection of step-by-step solutions to 13 typical research questions,
	and serves as a nice supplement to the documentation materials
	available on the UCSC Genome Browser web site. The guide is accessible as a 
	link off the <I>Nature Genetics</I> home page at 
	<A HREF="http://www.nature.com/ng/" TARGET=_blank>http://www.nature.com.ng</A>. 
	
	<P><FONT FACE="courier" SIZE=2>Sep. 12, 2002</FONT><BR>
	We've corrected a problem with the Human June 2002 (hg12) cytoBand annotation 
	track that affected chromosome Y. The clones on this chromosome were
	erroneously pushed from the q-arm onto the p-arm, creating some confusion. The
	currently available version of the cytoBand data on this website contains this 
	correction.

    </TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<BR>
		
<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4" ><a name="Tech"></a><b>&nbsp; Technical Information About the Assembled Sequence</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD>
<UL>
	<LI><A HREF="goldenPath/term.html"><FONT SIZE="4">Terminology</FONT></A>
	<LI><A HREF="goldenPath/datorg.html"><FONT SIZE="4">Data organization and format</FONT></A>
	<LI><A HREF="goldenPath/gbdDescriptions.html"><FONT SIZE="4">Description of the annotation database</FONT></A>
	<LI><A HREF="goldenPath/stats.html"><FONT SIZE="4">Summary statistics</FONT></A>
	<LI><A HREF="goldenPath/chromReports"><FONT SIZE="4">Chromosome Reports</FONT></A>
         <LI><A HREF="goldenPath/mapPlots/"><FONT SIZE="4">Genome Map Plots</FONT></A>
<!-- CHEAT: not closing <UL> tag tightens bottom spacing reliably in unix & windows! -->
</TD><TD WIDTH=15>&nbsp;</TD></TR></TABLE>
	<BR></TD></TR></TABLE>
</TD></TR></TABLE>
	
<BR>
	
<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4" ><a name="Other"></a><b>&nbsp; Other Links</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD>
<UL>
	<LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/"><FONT SIZE="4">Human Genome Project at UCSC</FONT></A><FONT SIZE="4"></FONT>
	<LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/"><FONT SIZE="4">Center for Biomolecular Science and Engineering Home Page</FONT></A>
<!-- CHEAT: not closing <UL> tag tightens bottom spacing reliably in unix & windows! -->
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
</TD></TR></TABLE>

</TD></TR></TABLE>
</TD></TR></TABLE>
</BODY></HTML>

