<!-- News Section ============================================- -->  
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
    <TR><TD>        
    <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
        <TR><TD>        
        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" title="Follow &#64;GenomeBrowser on Twitter" alt="Follow &#64;GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a>
                 <a title="Genome Browser Facebook page"
                 href="http://www.facebook.com/ucscGenomeBrowser"
                 target="_blank"><img
                 src="/images/FB-f-Logo__blue_22.jpg"
                 style="vertical-align:text-bottom; margin-left: 15px;"
                 alt="Genome Browser Facebook page"></a>
                 <a href="https://plus.google.com/107643636314131661344" rel="publisher"></a>
               </TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list. Please see our
          <a href="http://genome.ucsc.edu/blog"
          target=_blank>blog</a> for posts about Genome Browser tools, features, projects and more.
          </p>

    <!-- start news -->

	<hr>
        <p>
        <font face="courier" size="3"><b>23 January 2015 - Genome Browser YouTube Channel</b></font>

        <p>We are pleased to announce the release of the UCSC Genome Browser
        <a href="http://bit.ly/genomebrowserYoutube" target=_blank">YouTube channel</a>.  
        The channel contains short videos showing how to use the Genome Browser and associated 
        tools to solve selected problems.  The videos were produced by Robert Kuhn and Pauline Fujita, 
        with assistance from Luvina Guruvadoo, Jonathan Casper, Matt Speir and the rest of the Browser staff.
        </p>

        <p>See also the other materials on our <a href="http://bit.ly/genomebrowserTraining">training page</a>.

        </p>
  
	<hr>
	<p>
	<font face="courier" size="3"><b>15 January 2015 - 
	Proteomics Data Now Available in Genome Browser</b></font>
	<p>
	We are pleased to announce the release of proteomics data for the 
	human hg19 assembly. Data from the National Cancer Institute's (NCI) 
	<a href="http://proteomics.cancer.gov/programs/cptacnetwork" target="_blank">
	Clinical Proteomic Tumor Analysis Consortium (CPTAC)</a> is now available 
	in the UCSC Genome Browser as a public track hub. This track hub 
	contains peptides that were identified by CPTAC in their deep mass 
	spectrometry based characterization of the proteome content of 
	breast, colorectal and ovarian cancer biospecimens that were initially 
	sequenced by <a href="http://cancergenome.nih.gov/" target="_blank">
	The Cancer Genome Atlas</a>. This effort extends the 
	accessibility of CPTAC data to more researchers and provides an 
	additional level of analysis to assist the cancer biology community.
	<p>
	To access and view this hub, navigate to the 
	<a href="http://genome.ucsc.edu/cgi-bin/hgHubConnect" target="_blank">
	Track Hub gateway page</a> and select &quot;CPTAC Hub v1&quot; from the Public Hubs list. 
	Please direct any queries to the Fenyo Lab at <a href="mailto:info@fenyolab.org">info@fenyolab.org</a>.
	<p>
	In addition, we have also released a <a href="http://www.peptideatlas.org/" target="_blank">
	PeptideAtlas</a> track which displays peptide identifications from the 
	<a href="https://db.systemsbiology.net/sbeams/cgi/PeptideAtlas/buildDetails?atlas_build_id=433" target="_blank">
	PeptideAtlas August 2014 (Build 433) Human build</a>. This build, based on 971 samples containing more than 420 
	million spectra, identified over a million distinct peptides covering 
	more than 15,000 canonical proteins. To read more about this track, 
	please see the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=peptideAtlas" target="_blank">
	track description page</a>. Many thanks to Eric Deutsch, 
	Zhi Sun, and the PeptideAtlas team at the Institute for Systems Biology, 
	Seattle for providing this data.
	</p>        



	<hr>
        <p>
        <font face="courier" size="3"><b>19 December 2014 -
        New Cow (bosTau8) Assembly Now Available in the Genome Browser</b></font>
        <p>
        We are pleased to announce the release of a Genome Browser for the June 2014 assembly
        of cow, <i>Bos taurus</i> (Bos_taurus_UMD 3.1.1, UCSC version bosTau8). This updated cow assembly was 
	provided by the  <a href="http://www.cbcb.umd.edu/"
        target="_blank">UMD Center for Bioinformatics and Computational Biology</a> (CBCB). This
	assembly is an update to the previous UMD 3.1 (bosTau6) assembly.  UMD 3.1 contained 
	138 unlocalized contigs that were found to be contaminants. 
	These have been suppressed in UMD 3.1.1.
        <p>
        Bulk downloads of the sequence and annotation data are available via the Genome
        Browser
        <a href="ftp://hgdownload.cse.ucsc.edu/goldenPath/bosTau8/">FTP server</a>
        or the
        <a href="http://hgdownload.cse.ucsc.edu/downloads.html#cow">Downloads</a>
        page. These data have
        <a href="/goldenPath/credits.html#cow_credits">specific conditions for use</a>.
        The cow browser annotation tracks were generated by UCSC and
        collaborators worldwide. See the
        <a href="/goldenPath/credits.html#cow_credits">Credits</a> page for a detailed
        list of the organizations and individuals who contributed to this release.
        </p>
	<hr>

        <!-- start archives -->

	<p>
	<font face="courier" size="3"><b>12 December 2014 -
        New <i>D. melanogaster</i> (dm6) Assembly Now Available in the Genome Browser: </b></font>
        We are pleased to announce the release of a Genome Browser for the August 2014 assembly
        of <i>Drosophila melanogaster</i> (BDGP Release 6, UCSC version dm6). 
	<a href="goldenPath/newsarch.html#121214">Read more</a>. 

 	<p>
        <font face="courier" size="3"><b>29 October 2014 -
        Genome Browser in a Box (GBiB)</b>:</font>
        Sometimes you just want to keep your genomics data to yourself. Have you ever
        hesitated when uploading your data set into the UCSC Genome Browser? If so,
        you'll be happy to know that we have created a stand-alone personal version:
        Genome Browser in a Box (GBiB). <a href="goldenPath/newsarch.html#102914">Read more</a>.
	</p>

            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
