<!-- News Section ============================================- -->  
<TABLE style="background-color:#888888; width: 100%; border-width: 0px; border-spacing: 0px; padding: 0px">
    <TR><TD>        
    <TABLE style="background-color:#fffee8; width: 100%; border-width: 0px; border-spacing: 0px; padding: 0px">
        <TR><TD>        
        <TABLE style="background-color:#D9E4F8; background-image: url('images/hr.gif'); width: 100%; border: 0px">
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" title="Follow &#64;GenomeBrowser on Twitter" alt="Follow &#64;GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a>
                 <a title="Genome Browser Facebook page"
                 href="http://www.facebook.com/ucscGenomeBrowser"
                 target="_blank"><img
                 src="/images/FB-f-Logo__blue_22.jpg"
                 style="vertical-align:text-bottom; margin-left: 15px;"
                 alt="Genome Browser Facebook page"></a>
                 <a href="https://plus.google.com/107643636314131661344" rel="publisher"></a>
               </TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE style="background-color:#fffee8; width: 100%; padding: 0px">
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list. Please see our
          <a href="http://genome.ucsc.edu/blog"
          target=_blank>blog</a> for posts about Genome Browser tools, features, projects and more.
          </p>

    <!-- start news -->
  <hr>
  <p>
  <font color="#006666" size="4"><b>
  Happy Holidays from the UCSC Genome Browser! &mdash;</b></font>
  Here's a <A
  HREF="../cgi-bin/hgTracks?db=hg38&hgct_customText=http://hgwdev.cse.ucsc.edu/~brianlee/customTracks/testBigGenePred/testFun/customTrackLines">card</A> 
  for you!
  <hr>
  <p>
  <font color="#006666" size="4"><b> 21 December 2015 -
  Two New Assemblies Now Available in the Genome Browser</b></font>
  <p>
  A Genome Browser is now available for the frog (<em>Xenopus tropicalis</em>)
  assembly released September 2012 by the <a href="http://genome.jgi.doe.gov/"
  target="_blank">US DOE Joint Genome Institute (JGI-PGF)</a>
  (JGI v7.0, UCSC version xenTro7).  For more
  information and statistics about this assembly, see the NCBI assembly record
  for <a href="http://www.ncbi.nlm.nih.gov/assembly/515038" target="_blank">
  Xtropicalis_v7</a>.  There are 7,728 scaffolds with a total size of 1,437,530,879
  bases.
  </p>

  <p>
  Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
  <a href="ftp://hgdownload.soe.ucsc.edu/goldenPath/xenTro7/">FTP server</a> or the
  <a href="http://hgdownload.soe.ucsc.edu/downloads.html#xentro">Downloads</a> page.
  Please observe the <a href="goldenPath/credits.html#xentro_credits">conditions for use</a> when
  accessing and using these data sets.  The annotation tracks for this browser were
  generated by UCSC and collaborators woldwide.  See the <a
  href="goldenPath/credits.html#xentro_credits">Credits</a> page for a detailed list of the
  organizations and individuals who contributed to this release.
  </p>
  ---      
  <p>
  A Genome Browser is now available for the cat (<em>Felis catus</em>)
  assembly released November 2014 by the International Cat Genome Sequencing Consortium
 (ICGSC version Felis_catus_8.0, UCSC version felCat8).  For more
  information and statistics about this assembly, see the NCBI assembly record
  for <a href="http://www.ncbi.nlm.nih.gov/assembly/1373248" target="_blank">
  Felis_catus_8.0</a>.  There are 267,625 scaffolds with a total size of 2,641,342,258
  bases.
  </p>

  <p>
  Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
  <a href="ftp://hgdownload.soe.ucsc.edu/goldenPath/felCat8/">FTP server</a> or the
  <a href="http://hgdownload.soe.ucsc.edu/downloads.html#cat">Downloads</a> page.
  Please observe the <a href="goldenPath/credits.html#cat_use">conditions for use</a> when
  accessing and using these data sets.  The annotation tracks for this browser were
  generated by UCSC and collaborators woldwide.  See the <a
  href="goldenPath/credits.html#cat_credits">Credits</a> page for a detailed list of the
  organizations and individuals who contributed to this release.
  </p>
  
  <hr>
  <p>
  <font color="#006666" size="4"><b>16 December 2015 &mdash;
  Output from Related Tables Now Available in the Data Integrator</b></font>
  <p>
  In the UCSC Genome Browser database, detailed information on the annotations for many tracks
  is stored in extra tables. This information can include things such as identifiers
  in other databases, transcript status, or other descriptive information. You can now obtain
  data from these related tables in your output from the
  <a href="cgi-bin/hgIntegrator">Data Integrator</a>.
  </p>

  <p>
  To add related tables to your Data Integrator output, first, select and add at least one
  track to your data sources. Then, scroll down to the "Output Options" section and click the
  <button type="button">Choose fields</button> button.
  Lastly, at the bottom the section for each selected track, you will see a drop-down menu
  labeled "Related tables" which allows you to select and add various tables to your output.
  <p><img style="border: 1px solid black; display: block;"
                src="images/hgIntegratorRelatedTablesAnnouncement.png">
  <p>
  For more information about using the Data Integrator, please see the
  <a href="goldenPath/help/hgIntegratorHelp.html">User Guide</a>.
  </p>

  <p>
  Thank you to Angie Hinrichs and Matthew Speir for their work on this feature.
  </p>


  
        <!-- start archives -->
<!--staged for future release
        <p>
        <font color="#006666" size="4"><b> 21 December 2015 -
        Two New Assemblies Now Available in the Genome Browser</b></font>
        <p>
        Genome Browsers are now available for the frog (<em>Xenopus tropicalis</em>)
        and cat (<i>Felis catus</i>) assemblies.
	<a href="goldenPath/newsarch.html#122115">Read more</a>. 
        </p>

        <p>
	<font color="#006666" size="4"><b>16 December 2015 &mdash;
	Output from Related Tables Now Available in the Data Integrator</b>: </font>
	You can now obtain data from these related tables in your output from the
	<a href="cgi-bin/hgIntegaror">Data Integrator</a>.
	<a href="goldenPath/newsarch.html#121615">Read more</a>.
        </p>
-->
	<p>
	<font color="#006666" size="4"><b>9 December 2015 &mdash;
	Data from 1000 Genomes Project Phase 3 Now Available</b>: </font>
	We have made data from Phase 3 of the
	<a href="http://www.1000genomes.org/" target="_blank">1000 Genomes Project</a>
	available for the hg19 version of the human assembly.
        <a href="goldenPath/newsarch.html#120915">Read more</a>.
	</p>

        <p>
        <font color="#006666" size="4"><b>25 November 2015 &mdash;
        New &quot;View - In External Tools&quot; Menu Sends Data to Prediction Websites!</b></font>
        Do you often get the current sequence with &quot;View - DNA&quot; just to copy
        and paste it into another website? With the new menu &quot;View - In External Tools&quot;
        you can now send the current region to remote tools with two clicks.
        <a href="goldenPath/newsarch.html#112515">Read more</a>.
        </p>
            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
