<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="description" content="UCSC Genome Browser">
	<META name="keywords" content="genome, genome browser, human genome, genome assembly, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, chicken genome, chimp genome, fruitfly, fugu, yeast comparative genomics">
	<TITLE>UCSC Genome Browser Home</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">

</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<!--HOTLINKS BAR-->
    <TR><TD COLSPAN=2 HEIGHT=40>
    <TABLE BGCOLOR="000000" CELLPADDING="1" CELLSPACING="1" WIDTH="99%" HEIGHT="27">
        <TR BGCOLOR="2636D1"><TD VALIGN="middle">
        <TABLE BORDER=0 CELLSPACING=0 CELLPADDING=0 BGCOLOR="2636D1" HEIGHT="24">
    	    <TR><TD VALIGN="middle"><FONT color="#89A1DE">&nbsp; 
 	    <A HREF="/cgi-bin/hgGateway" class="topbar">
			Genome Browser</a> &nbsp; - &nbsp; 
 	    <A HREF="/cgi-bin/hgNear" class="topbar">
			Family Browser</a> &nbsp; - &nbsp; 
	    <A HREF="/cgi-bin/hgBlat" class="topbar">
			Blat</a> &nbsp; - &nbsp; 
	    <A HREF="/cgi-bin/hgText" class="topbar">
			Table Browser</a> &nbsp; - &nbsp; 
	    <A HREF="/FAQ/" class="topbar">
			FAQ</a> &nbsp; - &nbsp; 
	    <A HREF="goldenPath/help/hgTracksHelp.html" class="topbar">
			Help</a> &nbsp; </FONT></TD>
        </TR></TABLE>
    </TD></TR></TABLE>
    </TD></TR>

    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="/cgi-bin/hgGateway">
			Genome Browser</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgNear">
			Family Browser</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgBlat">
			Blat</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgText">
			Tables</A><HR>
	        <A CLASS="leftbar" HREF="downloads.html">
			Downloads</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/releaseLog.html">
			Release Log</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/customTracks/custTracks.html">
			Custom Tracks</A><HR>
	        <A CLASS="leftbar" HREF="mirror.html">
			Mirrors</A><HR>
  	        <A CLASS="leftbar" HREF="http://genome-archive.cse.ucsc.edu">
			Archives</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/credits.html">
			Credits</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/pubs.html">
			Pubs</A><HR>
	        <A CLASS="leftbar" HREF="cite.html">
			Cite Us</A><HR>
	        <A CLASS="leftbar" HREF="license/">
			Licenses</A><HR>
	        <A CLASS="leftbar" HREF="jobs.html">
			Jobs</A><HR>
  	        <A CLASS="leftbar" HREF="staff.html">
			Staff</A><HR>
  	        <A CLASS="leftbar" HREF="contacts.html">
			Contact Us</A><HR>
	        <BR>
	        <IMG SRC="images/DOT.gif" ALT="" HEIGHT=1150 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Content Tables-->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; About the UCSC Genome Bioinformatics Site</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
		    This site contains the reference sequence for the human and <em>C. elegans</em> 
		    genomes and working drafts for the chimpanzee, mouse, rat, chicken, <em>Fugu</em>, <EM>Drosophila</EM>, 
		    <em>C. briggsae</em>, yeast, and SARS genomes.  It also shows the
		    CFTR (cystic fibrosis) region in 13 species.  
		    <P>
		    We encourage you to explore these sequences with our tools. The Genome 
		    Browser zooms and scrolls over chromosomes, showing the work of annotators 
		    worldwide.  The Family Browser shows expression, homology and other 
		    information on groups of genes that can be related in many ways.  
		    The Proteome Browser displays protein property data and
		    links to a wealth of related information. The Table Browser 
		    provides convenient access to the underlying database.  
		    Blat quickly maps your sequence to the genome.
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
		        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		    </TD></TR></TABLE>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
	            <P> 
		    <FONT FACE="courier" SIZE=3><B>22 Apr. 2004  - Proteome Browser Available for Mouse Genome</B></FONT><BR>
		    <P>
		    Proteome Browser functionality is now available on the 
		    Oct. 2003 mouse genome assembly (mm4). Protein
		    information may be viewed for most genes in the Known Genes 
		    track by clicking the Proteome Browser link on the gene's 
		    details page.
		    For more information on the UCSC Proteome Browser, see the
		    news release dated <A HREF="http://genome.ucsc.edu/goldenPath/newsarch.html#031004">10 March 2004</A>.
		    <P>
		    In conjunction with this release, the Known Genes and Gene
		    Family Browser protein data have been updated to the
		    Swiss-Prot version dated 15 March 2004.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>16 Apr. 2004  - Genome Browser Training Class 6 May</B></FONT><BR>
		    <P>
		    The UCSC Bioinformatics Group announces a seminar and 
		    hands-on workshop on the UCSC Genome Browser, presented by 
		    <A HREF="http://www.openhelix.com/" TARGET=_blank>OpenHelix</A>, 
		    a bioinformatics training, 
		    software testing and consulting company.  
		    <P>
		    This introductory session is geared towards industry and
		    academic biologists engaged in genomics research. No
		    programming experience is required. The seminar will cover 
		    the topics necessary 
		    to learn how to effectively use the browser tool set, 
		    including basic Genome Browser functionality, searching and 
		    BLAT use, Table Browser use, creating and using Custom Tracks,
		    and an introduction to the Family Browser. The lecture will
		    be accompanied by hands-on computer exercises conducted
		    directly on the Genome Browser web site. 
		    <P>
		    The three-hour course will be held at the UCSC 
		    Extension Campus computer lab in Cupertino, CA on Thursday, 
		    May 6th 6-9 p.m. For pricing information or to reserve a
		    seat in the class, visit the 
		    <A HREF="http://www.openhelix.com/noteworthy.shtml"
		    TARGET=_blank>OpenHelix</A> web site or call 1-888-861-5051.
		    Pre-registration is required. Academic, student, and early 
		    registration discounts are available. 
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>12 Apr. 2004  - GNF Gene Expression Atlas 2 Data Available in Browser: </B></FONT>
		    Expression data from the GNF Gene Expression Atlas 2 are now 
		    available on the July 2003 human genome assembly on the UCSC
		    web site. The data may be viewed graphically in the 
		    <A HREF="http://genome.ucsc.edu/cgi-bin/hgNear?db=hg16&near.order=expGnfAtlas2&near.col.gnfHumanAtlas2Id.vis=on">Family Browser</A> or via
		    the GNF Atlas 2 track in the Genome Browser. 
		    <A HREF="/goldenPath/newsarch.html#041204">Read more</A>.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>10 Mar. 2004  - Introducing the UCSC Proteome Browser:</B></FONT>
		    We are proud to announce a new addition to the
		    UCSC family of genome browsing and analysis tools.  The UCSC Proteome 
		    Browser presents a rich set of useful protein properties as well as 
		    links to several protein and genomic data sources 
		    on the Web. 
		    <A HREF="/goldenPath/newsarch.html#031004">Read more</A>.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>1 Mar. 2004  - Chicken Draft Assembly Available:</B></FONT>
	   	    We've added the chicken genome to the collection of assemblies available
		    in the UCSC Genome Browser and Blat Server. 
		    The Feb. 2004 assembly (UCSC version galGal2) was produced by 
	 	    the Genome Sequencing Center at the Washington University 
		    School of Medicine in St. Louis. 
		    <A HREF="/goldenPath/newsarch.html#030104">Read more</A>.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>23 Feb. 2004  - Chimpanzee Draft Assembly Released: </B></FONT>
		    We are happy to announce the release of a Genome Browser and Blat server for the
		    chimpanzee (<EM>Pan troglodytes</EM>). 
		    The 13 Nov. 2003 Arachne assembly -- labeled Chimp Build 1
		    Version 1 (UCSC version panTro1) -- was produced by the 
		    Chimpanzee Genome Sequencing 
		    Consortium. <A HREF="/goldenPath/newsarch.html#022304">Read more</A>.
    	            </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
	    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome Browser 
                    are freely available for academic, nonprofit, and personal use 
		    with the following conditions:
		    <UL>
		    <LI>Genome sequence data use restrictions are noted within the
		    species sections on the <A HREF="/goldenPath/credits.html">Credits</A> 
		    page.
		    <LI>Some annotation tracks contributed by external collaborators
		    contain proprietary data that have specific use restrictions. To
		    check for restrictions associated with a particular genome assembly, 
		    review the <em>database/README.txt</em> file in the assembly's 
		    downloads directory. 
		    </UL>
		    <P>
		    The Genome Browser and Blat software are free for academic,
		    nonprofit, and personal use. A license is required for commercial use.
		    See the <A HREF="/license/">Licenses</A> page for more information.
		    <P>
                    Program-driven use of this software is limited to a maximum of one 
                    hit every 15 seconds and no more than 5,000 hits per day.  
		    <P>
		    Questions or problems regarding the UCSC Genome Browser 
		    software, database, genome assemblies, or release cycles 
		    should be sent to 
		    <A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A>.
		    Messages sent to this address will be posted automatically 
		    on the genome mailing list, which is also archived on a 
		    public Web-accessible pipermail archive. NOTE: This archive may be 
		    indexed by non-UCSC sites such as Google. 
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>

        <BR>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	        
                    <FONT SIZE="4" ><A name="Tech"></A><B>&nbsp; Technical Information About the Assembled Sequence</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
	            <LI><A HREF="goldenPath/gbdDescriptions.html"><FONT SIZE="4">Description of the annotation database</FONT></A>
	            <LI><A HREF="goldenPath/datorg.html"><FONT SIZE="4">Data organization and format</FONT></A>
	            <LI><A HREF="goldenPath/term.html"><FONT SIZE="4">Terminology</FONT></A>
                    </UL>
                    </TD>
                    <TD WIDTH=15>&nbsp;</TD>
                 </TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	            <FONT SIZE="4" ><A NAME="Other"></A><B>&nbsp; Other Links</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/"><FONT SIZE="4">Human Genome Project at UCSC</FONT></A><FONT SIZE="4"></FONT>
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/"><FONT SIZE="4">Center for Biomolecular Science and Engineering Home Page</FONT></A>
                    </UL>
                </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
</TD></TR></TABLE>

</BODY></HTML>

