<!-- News Section ============================================- -->  
<TABLE style="background-color:#888888; width: 100%; border-width: 0px; border-spacing: 0px; padding: 0px">
    <TR><TD>        
    <TABLE style="background-color:#fffee8; width: 100%; border-width: 0px; border-spacing: 0px; padding: 0px">
        <TR><TD>        
        <TABLE style="background-color:#D9E4F8; background-image: url('images/hr.gif'); width: 100%; border: 0px">
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" title="Follow &#64;GenomeBrowser on Twitter" alt="Follow &#64;GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a>
                 <a title="Genome Browser Facebook page"
                 href="http://www.facebook.com/ucscGenomeBrowser"
                 target="_blank"><img
                 src="/images/FB-f-Logo__blue_22.jpg"
                 style="vertical-align:text-bottom; margin-left: 15px;"
                 alt="Genome Browser Facebook page"></a>
                 <a href="https://plus.google.com/107643636314131661344" rel="publisher"></a>
               </TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE style="background-color:#fffee8; width: 100%; padding: 0px">
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list. Please see our
          <a href="http://genome.ucsc.edu/blog"
          target=_blank>blog</a> for posts about Genome Browser tools, features, projects and more.
          </p>

    <!-- start news -->
  <hr>
  <font color="#006666" size="4"><b>6 May 2016
  - New Gateway Coming 5/10/16!</b></font>
  <p>
  The UCSC Genome Browser team is proud to announce that we will soon release
  a newly designed Gateway, which will be live on the public site Tuesday afternoon
  (PDT), May 10th, 2016. Another announcement will be shared Tuesday with further
  details and a demonstration video.</p>
  <p>
  While the main functionality remains unchanged as a gateway to all available genome
  browsers, several helpful new features have been added:</p>
  <li>Autocomplete search for any genome browser, genome version, or public assembly hub
  <li>Browse available genome browser species by scrolling through a  menu based on phylogenetic order 
  <li>Quick access shortcuts to popular browsers
  <li>A new style and color scheme
  <p>
  Here is <a href="images/sneakPeekGateway.png"
  target="_blank">sneak picture of the new Gateway web page</a>.</p>
  <p>
  We hope that you will enjoy the new Gateway page next week! As always, we invite your
  questions, comments, and suggestions as directed through the details on our Contact Us web page.
  <p>
  <font color="#006666" size="4"><b>22 Apr 2016
  - Data from the Lens PatSeq Database Now Available</b></font>
  <p>
  We are pleased to announce the release of a set of tracks showing the genomic mapping of
  biomedical sequences submitted as part of patent application documents worldwide. The sequence 
  data, mappings and associated patent information were obtained from the 
  <a href="https://www.lens.org/lens/bio" target="_blank">PatSeq</a> database provided by 
  <a href="https://www.lens.org/" target="_blank">The Lens</a>. The PatSeq data are 
  mapped to the genome, and the individual sequence features within the tracks are then
  color-coded to indicate their status within the associated patent documents. 
  Track details pages show information about the patent documents in which the sequences
  are referenced, and provide a link to The Lens PatSeq Analyzer tool for the given chromosome 
  range. The PatSeq data are divided into two tracks: a bulk patent track for sequences affiliated 
  with patents in which more than 100 sequences were submitted, and a non-bulk patent track
  for all other sequences. These tracks are currently present on the human (hg19), mouse (mm10),
  and Ebola virus (eboVir3) genomes.
  </p>
  <p>
  Thanks to our collaborators at The Lens, Osmat Jefferson and Deniz Koellhofer, for providing the
  data and feedback on the visualization. Thank you to Max Haeussler and Matthew Speir, members of
  the UCSC Genome Browser team, for their efforts in creating these tracks.
  </p>

  <hr>
  <font color="#006666" size="4"><b>20 April 2016
  - New GTEx Gene Expression Track for hg19 and hg38</b></font>
  <p>
  We are excited to announce the release of a new gene expression track based on data from the NIH
  <a href="http://commonfund.nih.gov/GTEx/index" TARGET = _BLANK> Genotype-Tissue Expression</a> 
  (GTEx) project.  
  This track displays tissue-specific gene expression based on RNA-seq in 53 tissues from 570 
  donors obtained from the GTEx 'midpoint' data release (V6, October 2015).
  The original data for this 
  track can be found at the <a href="http://www.gtexportal.org/home" target = _blank>GTEX Portal</a>
  hosted by the Broad Institute.
  <p>
  This track also features a new gene expression display method that extends the traditional Genome 
  Browser display &mdash; a horizontal bar graph. Every gene is annotated by a graph with colored 
  bars, each of which corresponds to a specific tissue assayed by the GTEx project. Within
  a graph, the bar color indicates the tissue type, using GTEx conventions, and the bar height 
  depicts the median expression level (in RPKM).  
  To quickly view the tissue and expression level represented by a bar in the tracks display, 
  mouse over the bar in the graph.
  The complete tissue color legend is shown on the 
  <a href="../../cgi-bin/hgTrackUi?db=hg19&g=gtexGene" target=_blank>track configuration page</a>, 
  and can also be popped up for viewing alongside the track using the right-click menu.
  Below the bar graph, a line is shown indicating the gene extent that was used to generate the 
  annotation, colored by gene class using GENCODE conventions 
  (e.g. <span style='color: #0c0c78'>blue</span> for protein-coding, 
  <span style='color: #006400'>green</span> for non-coding).
  <p>
  <a href="../../cgi-bin/hgTracks?hgS_doLoadUrl=submit&hgS_loadUrlName=http://genome.ucsc.edu/goldenPath/help/examples/sessions/gtexAnnouncementSession.txt" 
  target=_blank>
  <center><img src="images/gtexAnnounceExample.png" width=75%></center></a>
  <p>
  User alert: In the figure above, do you notice how the bar graph sometimes extends past the 
  righthand end of the associated gene annotation? This is because all bar graphs in the
  display are the same width (and have the same tissue ordering) to facilitate comparison.
  For example, in this figure (which may also be viewed 
  <a href="../../cgi-bin/hgTracks?hgS_doLoadUrl=submit&hgS_loadUrlName=http://genome.ucsc.edu/goldenPath/help/examples/sessions/gtexAnnouncementSession.txt" 
  target=_blank>here</a>),
  the three tall bars in the annotation of the TCAP gene indicate that this <em>entire</em> gene is 
  highly expressed in cardiac and skeletal muscle. It does <em>not</em> indicate that there are 
  expression peaks in the intergenic region beyond the gene!
  <p>
  Similar to other Genome Browser track displays, clicking on the graph will bring up a description 
  page that shows a detailed box-and-whiskers plot of
  the RPKM data, as well as a description of the methods used to generate the data. 
  <p>
  Credit goes to Kate Rosenbloom and Christopher Lee for the implementation and testing of this 
  feature.
  </p>
  <hr>

<!-- start archives -->
<!--staged for future release
        <p>
	<font color="#006666" size="4"><b>22 Apr 2016
	- Data from the Lens PatSeq Database Now Available</b>: </font>
	We are pleased to announce the release of a set of tracks containing data from the
	<a href="https://www.lens.org/" target="_blank">Lens</a>
	PatSeq database in the UCSC Genome Browser.
	<a href="goldenPath/newsarch.html#042216">Read more</a>.

        <p>
        <font color="#006666" size="4"><b>19 April 2016
        - New Genotyping Tissue Expression Track for hg19 and hg38!</b></font>
        <p>
        We are excited to announce the release of a new track based on data from the 
        <a href="http://commonfund.nih.gov/GTEx/index" 
	TARGET = _BLANK>Genotyping Tissue Expression Consortium</a> V6 data freeze. 
	<a href="goldenPath/newsarch.html#04202016">Read more</a>.
-->

        <p>
	<font color="#006666" size="4"><b> 15 April 2016 -
	dbSNP 146 Available for hg19 and hg38</b>: </font>
	We are pleased to announce the release of four tracks derived
	from NCBI <a href="http://www.ncbi.nlm.nih.gov/SNP/">dbSNP</a> Build 146 data.
	<a href="goldenPath/newsarch.html#041516_2">Read more</a>.
	</p>

	<p>
	<font color="#006666" size="4"><b>15 April 2016
	- Genome Browsers for Gorilla, Rhesus, and Crab-Eating Macacque Now Available.</b></font>
	<a href="goldenPath/newsarch.html#041516_1">Read more</a>.
	</p>


            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
