<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="description" content="UCSC Genome Browser">
	<META name="keywords" content="genome, genome browser, human genome, genome assembly, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, mouse genome, rat genome, chicken genome, chimp genome, dog genome fruitfly, drosophila, fugu, yeast, ciona,  comparative genomics">
	<TITLE>UCSC Genome Browser Home</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">

</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<!--Top Menu Bar---------------------------------------------->  
    <TR><TD COLSPAN=2 HEIGHT=40>
    <TABLE BGCOLOR="000000" CELLPADDING="1" CELLSPACING="1" WIDTH="99%" HEIGHT="27">
        <TR BGCOLOR="2636D1"><TD VALIGN="middle">
        <TABLE BORDER=0 CELLSPACING=0 CELLPADDING=0 BGCOLOR="2636D1" HEIGHT="24">
    	    <TR><TD VALIGN="middle"><FONT color="#89A1DE">&nbsp; 
 	    <A HREF="/cgi-bin/hgGateway" class="topbar">
			Genomes</a> &nbsp; - &nbsp; 
 	    <A HREF="/cgi-bin/hgNear" class="topbar">
			Gene Sorter</A> &nbsp; - &nbsp; 
	    <A HREF="/cgi-bin/hgBlat" class="topbar">
			Blat</a> &nbsp; - &nbsp; 
	    <A HREF="/cgi-bin/hgPcr" class="topbar">
			PCR</a> &nbsp; - &nbsp; 
	    <A HREF="/cgi-bin/hgTables" class="topbar">
			Tables</a> &nbsp; - &nbsp; 
	    <A HREF="/cgi-bin/pbGateway" class="topbar">
			Proteome</a> &nbsp; - &nbsp; 
	    <A HREF="/FAQ/" class="topbar">
			FAQ</a> &nbsp; - &nbsp; 
	    <A HREF="goldenPath/help/hgTracksHelp.html" class="topbar">
			Help</a> &nbsp; </FONT></TD>
        </TR></TABLE>
    </TD></TR></TABLE>
    </TD></TR>

<!--Left Sidebar Menu----------------------------------------->  
    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="/cgi-bin/hgGateway">
			Genome Browser</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgNear">
			Gene Sorter</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgBlat">
			Blat</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgPcr">
			In Silico PCR</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgTables">
			Table Browser</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/pbGateway">
			Proteome Browser</A><HR>
	        <A CLASS="leftbar" HREF="util.html">
			Utilities</A><HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/downloads.html">
			Downloads</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/releaseLog.html">
			Release Log</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/customTracks/custTracks.html">
			Custom Tracks</A><HR>
	        <A CLASS="leftbar" HREF="/encode/">
			ENCODE</A><HR>
	        <A CLASS="leftbar" HREF="mirror.html">
			Mirrors</A><HR>
  	        <A CLASS="leftbar" HREF="http://genome-archive.cse.ucsc.edu">
			Archives</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/credits.html">
			Credits</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/pubs.html">
			Publications</A><HR>
	        <A CLASS="leftbar" HREF="http://openhelix.com/ucscmaterials.shtml"
			TARGET=_blank>
			Training</A><HR>
	        <A CLASS="leftbar" HREF="cite.html">
			Cite Us</A><HR>
	        <A CLASS="leftbar" HREF="license/">
			Licenses</A><HR>
	        <A CLASS="leftbar" HREF="jobs.html">
			Jobs</A><HR>
  	        <A CLASS="leftbar" HREF="staff.html">
			Staff</A><HR>
  	        <A CLASS="leftbar" HREF="contacts.html">
			Contact Us</A><HR>
	        <BR>
	        <IMG SRC="images/DOT.gif" ALT="" HEIGHT=950 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Introduction---------------------------------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; About the UCSC Genome Bioinformatics Site</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
		    This site contains the reference sequence and working draft
		    assemblies for a large collection of genomes. It also shows 
		    the CFTR (cystic fibrosis) region in 13 species and provides
		    a portal to the ENCODE project.  
		    <P>
		    We encourage you to explore these sequences with our tools. The Genome 
		    Browser zooms and scrolls over chromosomes, showing the work of annotators 
		    worldwide.  The Gene Sorter shows expression, homology and other 
		    information on groups of genes that can be related in many ways.  
		    Blat quickly maps your sequence to the genome.
		    The Table Browser 
		    provides convenient access to the underlying database.  
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
		        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		    </TD></TR></TABLE>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
		    <P>
		    To receive announcements of new genome 
		    assembly releases, new software features, updates and 
		    training seminars by email, subscribe to the
		    <A HREF="http://www.soe.ucsc.edu/mailman/listinfo/genome-announce" TARGET=_blank>genome-announce</A> mailing list.
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>3 December 2004  - SNP Track for Build 35 (hg17) Now Available</B></FONT>
		    <P>
		    UCSC has released the much-awaited SNP annotation track for the 
		    Build 35 (hg17) Human Genome Browser. The SNP data were
	 	    obtained from dbSnp Build 123, the first dbSnp
		    release based on the latest human assembly.
		    <P>
		    In conjunction with the data release, we have made 
		    several improvements to the graphical display of the 
		    track. 
		    Our users now have the ability to filter the SNP data
		    based on validation status,
	 	    functional class, variant class, average heterozygosity
		    and other features. Variants may be drawn in 
		    user-selected colors to highlight additional details
		    in the graphical image. Also, the browser now displays all
		    SNP variants at the appropriate base width, rather than
		    as generic ticks.
		
<!--
		    <FONT FACE="courier" SIZE=3><B>3 December 2004  - Browser Released for Honey Bee (<em>A. mellifera</em>)</B></FONT>
		    <P>
		    The UCSC Genome Bioinformatics group has released a
		    Genome Browser for the Amel_1.2 honey bee assembly
		    produced by the Baylor College of Medicine Human Genome
		    Sequencing Center.
		    <P>
		    The assembly -- which is approximately 213 Mb in size 
		    including contig gaps -- 
		    was sequenced using a combined whole genome shotgun (WGS)
		    and BAC clone approach. The BAC clone sequence was
		    produced following a pooled-array strategy. Overall 
		    sequence coverage is estimated at 6X.
		    For more information about this assembly, see the Baylor
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/honeybee/"
		    TARGET=_blank>honey bee project</A> website.
		    <P>
		    The honey bee genome is of interest to the agricultural
		    community, in addition to its role as a model organism in the
		    study of several human health issues, including immunity,
		    allergic reaction, antibiotic resistance, development,
		    mental health, longevity, and diseases of the X chromosome.
		    The honey bee is also interesting for its social and
		    behavioral traits.
		    <P>
		    The apiMel1 sequence and annotation data can be 
		    downloaded from the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/apiMel1/">FTP server</A>
		    or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#apiMel1">downloads</A>
		    page. Please see the Baylor
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html"
		    TARGET=_blank>conditions of use</A> page for guidelines 
		    regarding the use of these data.
		    <P>
		    Many thanks to Baylor, CHORI,  and the other
		    institutions who contributed to the sequencing and 
		    mapping effort of the Amel_1.2 release. The UCSC 
		    <em>A. mellifera</em> Genome Browser
		    was produced by Angie Hinrichs, Robert Kuhn, and Donna
		    Karolchik. The initial set of apiMel1 annotation tracks 
		    was generated by the UCSC Genome Bioinformatics Group. 
		    See the
		    <A HREF="/goldenPath/credits.html#apiMel_credits">credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to the release of
		    this browser.
-->
<!--
		    <FONT FACE="courier" SIZE=3><B>17 November 2004  - <em>X. tropicalis</em> Released</B></FONT>
		    <P>
		    The v.3.0 <em>Xenopus tropicalis</em> assembly is now
		    available on the UCSC Genome Browser. This whole genome
		    shotgun assembly 
		    (xenTro1, October 2004) was provided by the U.S. DOE 
		    Joint Genome Institute (JGI).
		    <P>
		    Version 3.0 was constructed with the JGI assembler, 
		    Jazz, using paired end sequencing reads at a coverage of 
		    7.4X.  After trimming for vector and quality, 19.1 
		    million reads assembled into 27,064 scaffolds totaling 
		    1.63 Gbp. Roughly half the genome is contained in 392 
		    scaffolds, each of which is at least 1.2 Mb in length.
		    For more information about this assembly, see the JGI
		    <A HREF="http://genome.jgi-psf.org/Xentr3/Xentr3.home.html"
		    TARGET=_blank><em>X. tropicalis</em> website</A>.
		    <P>
		    The xenTro1 sequence and annotation data can be 
		    downloaded from the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/xenTro1/">FTP server</A>
		    or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#xenTro">downloads</A>
		    page. Please see the JGI
		    <A HREF="http://genome.jgi-psf.org/Xentr3/Xentr3.download.html"
		    TARGET=_blank>data release policy</A> for data use 	
		    guidelines.
		    <P>
		    Many thanks to JGI and the other
		    institutions who contributed to the sequencing and 
		    mapping effort of the v. 3.0 release. The UCSC <em>X. 
		    tropicalis</em> Genome Browser
		    was produced by Fan Hsu, Jim Kent, Heather Trumbower, 
		    Hiram Clawson, Brian Raney, Galt Barber, Mark Diekhans, 
		    Angie Hinrichs, and Donna Karolchik. The initial set of 
		    xenTro1 annotation tracks was generated by the
		    UCSC Genome Bioinformatics Group. See the
		    <A HREF="/goldenPath/credits.html#xenTro_credits">credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to the release of
		    this browser.
-->
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>17 November 2004  - <em>X. tropicalis</em> Released:</B></FONT>
		    The v.3.0 <em>Xenopus tropicalis</em> assembly is now
		    available on the UCSC Genome Browser. This whole genome
		    shotgun assembly 
		    (xenTro1, October 2004) was provided by the U.S. DOE 
		    Joint Genome Institute (JGI).
		    <A HREF="/goldenPath/newsarch.html#111704">Read more</A>.
		    <P>
		   
		    <FONT FACE="courier" SIZE=3><B>8 October 2004  - Proteome Browser
		    Update:</B></FONT>
		    UCSC has released v1.1 of the 
		    <A HREF="/cgi-bin/pbGateway">UCSC 
		    Proteome Browser</A>. 
		    <A HREF="/goldenPath/newsarch.html#100804">Read more</A>.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>1 October 2004  - New Table Browser Released:</B></FONT>
		    We are pleased to announce the release of an enhanced version
		    of the <A HREF="cgi-bin/hgTables">UCSC Table Browser</A>. 
		    <A HREF="/goldenPath/newsarch.html#100104">Read more</A>.
    	            </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
	    	
<!--Conditions of Use Section--------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome Browser 
                    are freely available for academic, nonprofit, and personal use 
		    with the following conditions:
		    <UL>
		    <LI>Genome sequence data use restrictions are noted within the
		    species sections on the <A HREF="/goldenPath/credits.html">Credits</A> 
		    page.
		    <LI>Some annotation tracks contributed by external collaborators
		    contain proprietary data that have specific use restrictions. To
		    check for restrictions associated with a particular genome assembly, 
		    review the <em>database/README.txt</em> file in the assembly's 
		    downloads directory. 
		    </UL>
		    <P>
		    The Genome Browser and Blat software are free for academic,
		    nonprofit, and personal use. A license is required for commercial use.
		    See the <A HREF="/license/">Licenses</A> page for more information.
		    <P>
                    Program-driven use of this software is limited to a maximum of one 
                    hit every 15 seconds and no more than 5,000 hits per day.  
		    <P>
		    Questions or problems regarding the UCSC Genome Browser 
		    software, database, genome assemblies, or release cycles 
		    should be sent to 
		    <A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A>.
		    Messages sent to this address will be posted automatically 
		    on the genome mailing list, which is also archived on a 
		    public Web-accessible pipermail archive. NOTE: This archive may be 
		    indexed by non-UCSC sites such as Google. 
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>

        <BR>

<!--Technical Information Section----------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	        
                    <FONT SIZE="4" ><A name="Tech"></A><B>&nbsp; Technical Information About the Assembled Sequence</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
	            <LI><A HREF="goldenPath/gbdDescriptions.html"><FONT SIZE="4">Description of the annotation database</FONT></A>
	            <LI><A HREF="goldenPath/datorg.html"><FONT SIZE="4">Data organization and format</FONT></A>
	            <LI><A HREF="goldenPath/term.html"><FONT SIZE="4">Terminology</FONT></A>
                    </UL>
                    </TD>
                    <TD WIDTH=15>&nbsp;</TD>
                 </TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
    	
<!--Related Links Section--------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	            <FONT SIZE="4" ><A NAME="Other"></A><B>&nbsp; Related Links</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
<!-- Commented out til this page is updated - very out of date
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/"><FONT SIZE="4">Human Genome Project at UCSC</FONT></A><FONT SIZE="4"></FONT>
-->
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/"><FONT SIZE="4">Center for Biomolecular Science and Engineering Home Page</FONT></A>
                    </UL>
                </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
</TD></TR></TABLE>

</BODY></HTML>

