<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                          <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="http://www.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>
		   
                    <P>
                    <FONT FACE="courier" SIZE=3><B>6 April 2007 - New UCSC
		    Gene Prediction Set Released</B></FONT>
                    <P>
		    We are pleased to announce the release of a new gene 
		    prediction set, UCSC Genes, on the latest human 
		    Genome Browser (hg18, NCBI Build 36). This annotation,
		    which includes putative noncoding genes as well as 
		    protein-coding genes and 99.9% of RefSeq genes,
		    is the next generation of the Known Genes set that UCSC 
		    has been providing for several years and supersedes the
		    existing Known Genes annotation on the hg18 assembly.
		    <P>
		    The UCSC Genes is a moderately conservative prediction set 
		    based on data from RefSeq, GenBank, and UniProt. Each
		    entry requires the support of one GenBank RNA sequence plus 
		    at least one additional line of evidence, with the exception
		    of RefSeq RNAs, which require no additional evidence. 
		    Some of the noncoding transcripts in the set may actually 
		    code for protein, but the evidence for the associated 
		    protein is weak at best. Compared to RefSeq, this gene set 
		    generally has about 10% more protein-coding genes, 
		    approximately five times as many putative noncoding genes, 
		    and about twice as many splice variants.
		    </P>
		    <P>
		    The UCSC Genes set is produced using a computational 
		    pipeline developed at UCSC by Jim Kent, Chuck Sugnet and 
		    Mark Diekhans. For detailed information about the process 
		    used to construct the genes set, see the 
		    <A HREF="../cgi-bin/hgTrackUi?db=hg18&g=knownGene">track 
		    description page</A>. In upcoming months, we plan to 
		    release UCSC Genes sets on several organisms in addition
		    to human.
		    </P>
		    <P>
		    As part of this change, we are now using our own UCSC 
		    Genes accession numbers as the primary key into the 
		    underlying knownGene table, rather than the GenBank mRNA 
		    accessions 
		    we used in the previous Known Genes prediction set.
		    Note that this may affect external sites with URLs that
		    link into our genes track using the older-style accessions.
		    </P>
		    <P>
		    We will continue to provide the older Known Genes track on 
		    hg18 under the name &quot;Old Known Genes&quot;. You may
		    find the following tables useful in referencing the older 
		    gene set and converting between the two sets:
		    <UL>
		    <LI>
		    knownGeneOld2: new name for table underlying the old Known 
		    Genes (previously called knownGene)
		    <LI>
		    kgXrefOld2: new name for table that contains data for 
		    converting old Known Genes IDs to other IDs (previously 
		    called kgXref)
		    <LI>
		    kg2ToKg3: data for converting old Known Genes IDs 
		    to the newer UCSC Genes IDs
		    </UL></P>
		    <P>
		    We'd like to acknowledge the many people affiliated with 
		    the UCSC Genome
		    Bioinformatics group who worked hard to release this new 
		    annotation: developers Jim Kent,
		    Mark Diekhans, and Fan Hsu (with technical support from 
		    several
		    other engineers in the group); David Haussler; our splendid 
		    QA team -- 
		    Archana Thakkapallayil, Ann Zweig, Robert Kuhn, Kayla Smith,
		    and Brooke Rhead; our build engineer -- Andy
		    Pohl; and our sysadmin group. We'd also like to thank Chuck
		    Sugnet for his input, the people and organizations 
		    maintaining the RefSeq, UniProt, and GenBank databases, and 
		    the scientists worldwide who have contributed to them. If 
		    you have any questions about this new release,
		    feel free to contact us at 
		    <A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A>
		    (general questions) or 
		    <A HREF="mailto:genome-mirror@soe.ucsc.edu">genome-mirror@soe.ucsc.edu</A> 
		    (mirror-specific questions).
		    </P>
		    <P>
		    <HR>
                    <FONT FACE="courier" SIZE=3><B>27 March 2007 - Stickleback
		    Assembly Released in Genome Browser: </B></FONT>
		    We have released a Genome Browser and Blat server for 
		    the Feb. 2006 v1.0 draft assembly of <em>Gasterosteus 
		    aculeatus</em> produced by the 
		    <A HREF="http://www.broad.mit.edu/" TARGET=_blank>Broad 
		    Institute</A>.
                    <A HREF="goldenPath/newsarch.html#032707">Read more</A>.
		    </P>
                        </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
