<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                         <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD>
                  <TD>
                  <P>
                  To receive announcements of new genome 
                  assembly releases, new software features, updates and 
                  training seminars by email, subscribe to the
                  <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
                  TARGET=_blank>genome-announce</A> mailing list.</P>

            <!-- start news -->
	    	<font face="courier" size="3"><b>27 November 2012 -
		New UCSC Genes Track Released for Mouse</b></font>
		<p>
		We're happy to announce the release of a new UCSC Genes 
		track for the GRCm38/mm10 mouse Genome Browser.
		</p>
		<p>
		The new release has 59,121 transcripts, compared with 
		55,419 in the previous version on mm9. The total number 
		of canonical genes has increased from 28,661 to 31,227. 
		Comparing the new gene set with the previous version:
		<p>
		<ul>
		<li>48,739 transcripts did not change between versions.
		<li>423 transcripts were not carried forward to the new version.
		<li>5,108 transcripts are "compatible" with those in the previous 
		set, meaning that the two transcripts show consistent splicing.
		<li>1,037 transcripts overlap with those in the previous set.
		</ul>
		<p>
		More details about the new UCSC Genes track can be found on 
		the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=mm10&g=knownGene"
		>UCSC Genes track description page</a>. Also, please note that GRCm38/mm10 
		is now the default mouse assembly on the Genome Browser Gateway.
		<p>
		Many thanks to Brian Raney, Jim Kent, and Luvina Guruvadoo 
		for their work on this track!
		</p>
		<hr>
	    <!-- start archives -->
                <font face="courier" size="3"><b>14 November 2012 -
		ENCODE Tutorials Available at Open Helix</b></font>
                <p>
                OpenHelix and the UCSC Bioinformatics group have worked
                together to give everyone free access to the newly
                updated ENCODE tutorial until the end of February 2013.
                </p>
                <p>
                The online tutorial suite teaches users how to access
                the ENCODE data in the UCSC Genome Browser.  The online
                tutorial, created by OpenHelix in conjunction with the
                UCSC Bioinformatics Group can be viewed for free at
                <A HREF = "http://www.openhelix.com/ENCODE2" TARGET =
                _BLANK>http://www.openhelix.com/ENCODE2</A>.  The
                tutorial has been updated to reflect the recent changes
                to the UCSC Genome Browser.
                The original tutorial is still available at
                <A HREF = "http://www.openhelix.com/ENCODE" TARGET =
                _BLANK>http://www.openhelix.com/ENCODE</A>.
                </p>
                <p>
                The online narrated tutorial, which runs in just about
                any browser, can be viewed from beginning to end or
                navigated using chapters and forward and backward sliders.
                The approximately 60-minute tutorial explores ways
                to access the ENCODE data available within the UCSC Genome
                Browser.
                </p>
                <p>
                In addition to the tutorial, users can also access useful
                training materials including the animated PowerPoint slides
                used as a basis for the tutorial, suggested script for the
                slides, slide handouts, and exercises.
                </p>
                

==> <a href="goldenPath/newsarch.html">News Archives</a>

                    </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
