<!-- News Section ============================================- -->  
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
    <TR><TD>        
    <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
        <TR><TD>        
        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list.</P>

    <!-- start news -->

        <hr>
        <p>
        <font face="courier" size="3"><b> 09 May 2014 -
        Three New Assemblies Now Available in the Genome Browser</b>
        </font>
        <p>
        A Genome Browser is now available for the Chinese hamster (<em>Cricetulus griseus</em>)
        assembly released July 2013 by the <a href="http://www.genomics.cn/en/index"
        target="_blank">Beijing Genomics Institution-Shenzhen</a>
        (BGI version C_griseus_v1.0, UCSC version criGri1).  For more
        information and statistics about this assembly, see the NCBI assembly record
        for <a href="http://www.ncbi.nlm.nih.gov/assembly/40911" target="_blank">
        C_griseus_v1.0</a>.  There are 52,711 scaffolds with a total size of 2,360,146,428
        bases.
        </p>

        <p>
        Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
        <a href="ftp://hgdownload.soe.ucsc.edu/goldenPath/criGri1/">FTP server</a> or the
        <a href="http://hgdownload.soe.ucsc.edu/downloads.html#chinese_hamster">Downloads</a> page.
        Please observe the <a href="goldenPath/credits.html#chinese_hamster_use">conditions for use</a> when
        accessing and using these data sets.  The annotation tracks for this browser were
        generated by UCSC and collaborators woldwide.  See the <a
        href="goldenPath/credits.html#chinese_hamster_credits">Credits</a> page for a detailed list of the
        organizations and individuals who contributed to this release.
        </p>
        ---
        <p>
        A Genome Browser is now available for the hedgehog (<em>Erinaceus europaeus</em>)
        assembly released May 2012 by the <a href="http://www.broadinstitute.org/"
        target="_blank">Broad Institute</a>
        (Broad version EriEur2.0, UCSC version eriEur2).  For more
        information and statistics about this assembly, see the NCBI assembly record
        for <a href="http://www.ncbi.nlm.nih.gov/assembly/426148" target="_blank">
        EriEur2.0</a>.  There are 5,802 scaffolds with a total size of 2,715,720,925
        bases.
        </p>

        <p>
        Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
        <a href="ftp://hgdownload.soe.ucsc.edu/goldenPath/eriEur2/">FTP server</a> or the
        <a href="http://hgdownload.soe.ucsc.edu/downloads.html#hedgehog">Downloads</a> page.
        Please observe the <a href="goldenPath/credits.html#hedgehog_use">conditions for use</a> when
        accessing and using these data sets.  The annotation tracks for this browser were
        generated by UCSC and collaborators woldwide.  See the <a
        href="goldenPath/credits.html#hedgehog_credits">Credits</a> page for a detailed list of the
        organizations and individuals who contributed to this release.
        </p>
        ---
        <p>
        A Genome Browser is now available for the zebra finch (<em>Taeniopygia guttata</em>)
        assembly released February 2013 by <a href="http://genome.wustl.edu/genomes/detail/taeniopygia-guttata/"
        target="_blank">The Genome Institute at Washington University</a>
        (WUSTL version Taeniopygia_guttata-3.2.4, UCSC version taeGut2).  For more
        information and statistics about this assembly, see the NCBI assembly record
        for <a href="http://www.ncbi.nlm.nih.gov/assembly/524908" target="_blank">
        Taeniopygia_guttata-3.2.4</a>.  There are 37,096 scaffolds with a total size of 1,232,135,591
        bases.
        </p>

        <p>
        Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
        <a href="ftp://hgdownload.soe.ucsc.edu/goldenPath/taeGut2/">FTP server</a> or the
        <a href="http://hgdownload.soe.ucsc.edu/downloads.html#zebra_finch">Downloads</a> page.
        Please observe the <a href="goldenPath/credits.html#zebra_finch_use">conditions for use</a> when
        accessing and using these data sets.  The annotation tracks for this browser were
        generated by UCSC and collaborators woldwide.  See the <a
        href="goldenPath/credits.html#zebra_finch_credits">Credits</a> page for a detailed list of the
        organizations and individuals who contributed to this release.
        </p>

        <hr>
        <p>
        <font face="courier" size="3"><b> 08 May 2014 -
        New Elephant Shark (calMil1) Assembly Now Available in the Genome Browser</b>
        </font>
        <p>
        A Genome Browser is now available for the elephant shark (<em>Callorhinchus milii</em>)
        assembly released December 2013 by the <a href="http://www.imcb.a-star.edu.sg/"
        target="_blank">Singapore Institute of Molecular and Cell Biology</a>
        (IMCB version Callorhinchus_milii-6.1.3, UCSC version calMil1).  For more
        information and statistics about this assembly, see the NCBI assembly record
        for <a href="http://www.ncbi.nlm.nih.gov/assembly/85971" target="_blank">
        Callorhinchus_milii-6.1.3</a>.  There are 21,204 scaffolds with a total size of 974,498,586
        bases.
        </p>
        <p>
        Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
        <a href="ftp://hgdownload.soe.ucsc.edu/goldenPath/calMil1/">FTP server</a> or the
        <a href="http://hgdownload.soe.ucsc.edu/downloads.html#elephant_shark">Downloads</a> page.
        Please observe the <a href="goldenPath/credits.html#elephant_shark_use">conditions for use</a> when
        accessing and using these data sets.  The annotation tracks for this browser were
        generated by UCSC and collaborators woldwide.  See the <a
        href="goldenPath/credits.html#elephant_shark_credits">Credits</a> page for a detailed list of the
        organizations and individuals who contributed to this release.
        </p>

        <hr>
        <p>
        <font face="courier" size="3"><b> 05 May 2014 -
        Introducing the new stacked overlay for multi-wiggle tracks</b></font>
        <p>
        We are excited to announce the release of the new stacked
	overlay method for multi-wiggle tracks. This new stacked overlay
	method displays each graph stacked on top of each other.
	</p>
	<img style="border: 2px solid black; margin-left:auto;
                    margin-right:auto; display: block;"
	     src="images/stackedMultiWig_screenshot.png"
	     alt="Stacked screenshot">
        <p>
	Navigate to the track settings for <A HREF="cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRegMarkH3k27ac"
	>H3K27Ac Mark</A> on the human hg19 assembly to view this new stacked
	overlay method on a multi-wiggle track. On the track settings page,
	from the "Overlay method" drop-down menu select "stacked". Make sure
	that the display mode is not set to hide, and click submit.
        </p>
	<p>
	Credit for the implementation and testing of this new feature goes to
	Brian Raney and Matthew Speir.
	</p>
        <hr>

        <!-- staged for archive use.  NOTE: condense header line from .... </b></font><p> to </b></font>: 

        <font face="courier" size="3"><b> 09 May 2014 -
        Three New Assemblies Now Available in the Genome Browser</b>:</font>
        A Genome Browser is now available for the Chinese hamster (<em>Cricetulus griseus</em>),
        hedgehog (<em>Erinaceus europaeus</em>), and zebra finch (<em>Taeniopygia guttata</em>).
        <a href="goldenPath/newsarch.html#050914">Read more</a>

	<font face="courier" size="3"><b> 08 May 2014 -
        New Elephant Shark (calMil1) Assembly Now Available in the Genome
	Browser</b>:</font>
        A Genome Browser is now available for the elephant shark (<em>Callorhinchus milii</em>)
        assembly released December 2013 by the Singapore Institute of Molecular
        and Cell Biology.
	<a href="goldenPath/newsarch.html#050814">Read more</a>

	<font face="courier" size="3"><b> 05 May 2014 -
        Introducing the new stacked overlay for multi-wiggle tracks</b>:</font>
        We are excited to announce the release of the new &quot;stacked&quot;
        overlay method for multi-wiggle tracks.
	<a href="goldenPath/newsarch.html#050614">Read more</a>
                -->
        <!-- start archives -->

        <p>
        <font face="courier" size="3"><b> 29 Apr 2014 -
        100-species Conservation track re-release</b>:</font>
        We released the initial version of the 100-species Conservation track
        for the hg19 human assembly in November 2013. Over the past few months,
        we found a couple of inconsistencies and decided, for the integrity of
        the data, that we should rerun the computation pipeline and re-release
        the data. <a href="goldenPath/newsarch.html#042914">Read more</a>
        </p>

	<p>
        <font face="courier" size="3"><b>18 April 2014 - 
        New Motif Displays and New Genome Segmentations from ENCODE</b></font>:
        The Browser's default displayed 
        <a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRegTfbsClusteredV3">Transcription Factor ChIP-seq track</a>
        updated to the latest release, enhanced with the display of Factorbook motifs. And the Browser added
        <a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeAwgSegmentation">Genome
        Segmentations from ENCODE</a> tracks, which segment the genome into twenty-five states that were then grouped
        and colored to highlight predicted functional elements.
        <a href="goldenPath/newsarch.html#041814">Read more</a>.
        </p>
	
            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
