<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                         <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="https://lists.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>

            
            <!-- start news -->

		    <P>
                    <FONT FACE="courier" SIZE="3"><B>7 June 2011 -
Updated Lizard Browser Available</B></FONT>
		    <P>
		    We have released a Genome Browser for the May 2010 
		    genome assembly of the green anole lizard, <em>Anolis 
		    carolinensis</em> (Broad 
		    version AnoCar2.0, UCSC version anoCar2). This assembly,
		    which has been sequenced to 7.1X coverage, was produced by 
		    the <A HREF="http://www.broad.mit.edu/" 
		    TARGET=_blank>Broad Institute</A>, Cambridge, MA. 
		    </P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/anoCar2/">FTP server</A> 
		    or the
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#lizard">Downloads</A>
		    page.  These data have 
		    <A HREF="goldenPath/credits.html#lizard_use">specific
		    conditions for use</A>.
		    <P>
		    We'd like to thank the Broad Institute for providing this
		    assembly. The lizard Genome Browser and annotation
		    tracks were produced by Hiram Clawson, Brian Raney, and 
		    Luvina Guruvadoo. See the
		    <A HREF="goldenPath/credits.html#lizard_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release.
		    </P>
		    <HR>
                    <FONT FACE="courier" SIZE="3"><B>26 May 2011 - New
Release of UCSC Genes for Mouse:</B></FONT>
		    We've released an updated set of UCSC Genes for the mm9 
		    (NCBI Build 37) mouse Genome Browser. 
		    <A HREF="goldenPath/newsarch.html#052611">Read more</A>.
		    </P>

            <!-- ENCODE usability survey -->
		    <P>
                    <FONT FACE="courier" SIZE="3"><B>1 May 2011 - ENCODE
User's Guide: </B></FONT>
                  The ENCODE project has just published an overview of their ongoing large-scale
                  efforts to interpret the human genome sequence in the journal PLoS Biology.
                  <a target=_blank href="http://www.plosbiology.org/article/info:doi/10.1371/journal.pbio.1001046 ">
                  A Users Guide to the Encyclopedia of DNA Elements</a>,
                  highlights the scope of data production, and provides guidance for locating 
                  and using the data.  
		  <A HREF="goldenPath/newsarch.html#050111">Read more</A>.
		    </P>

            <!-- hg19 snp132 release -->
		    <P>
                    <FONT FACE="courier" SIZE="3"><B>18 April 2011 -
dbSNP 132 Available for hg19: </B></FONT>
                    We are pleased to announce the release of four tracks derived from
                    <a href="http://www.ncbi.nlm.nih.gov/projects/SNP/" target=_blank>dbSNP</a>
                    build 132, available on the human assembly (GRCh37/hg19).
                    dbSNP build 132 is available at NCBI. 
		    <A HREF="goldenPath/newsarch.html#041811.2">Read more</A>.
		    </P>
                   
==> <a href="goldenPath/newsarch.html">News Archives</a>

                    </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
