<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
		        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		    </TD></TR></TABLE>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
		    <P>
		    To receive announcements of new genome 
		    assembly releases, new software features, updates and 
		    training seminars by email, subscribe to the
		    <A HREF="http://www.soe.ucsc.edu/mailman/listinfo/genome-announce" TARGET=_blank>genome-announce</A> mailing list.</P>
                    <P>
                    <FONT FACE="courier" SIZE=3><B>6 October 2006 - Announcing Upgraded
                    Custom Tracks Tool: </B></FONT><BR><BR>

                    We are pleased to announce the release of an upgraded software 
                    tool in the Genome Browser collection &mdash; the Custom Tracks 
                    <A HREF="/cgi-bin/hgCustom">tool</A>.<BR><BR>

                    The new Custom Tracks Tool provides more flexibility and a 
                    more user-friendly interface for creating and managing your 
                    custom tracks than the tool it replaces.<BR><BR>

                    The initial release of this new tool includes the following 
                    features:
                     <LI> Add several custom tracks at one time (by means of URL, 
                          file or text).
                     <LI> Manage custom tracks: add, delete and modify.
                     <LI> Create a description page for each of your custom tracks.
                     <LI> Custom tracks will now persist on our server for 
                          48 hours (instead of 8 hours).

                    For more information about the new Custom Tracks tool, visit 
                    the <A HREF="goldenPath/help/customTrack.html">Users's Guide</A>.<BR><BR>

                    The hgCustom CGI was written by Kate Rosenbloom with the 
                    assistance of Archana Thakkapallayil, Ann Zweig and other 
                    members of the UCSC Genome Bioinformatics Group.
                    </P>

		    <P>
                    <FONT FACE="courier" SIZE=3><B>2 October 2006 - October Genome
                    Browser Seminar at ASHG: </B></FONT><BR><BR>
                    The UCSC Bioinformatics Group announces the sponsorship of a 
                    one hour seminar at the American Society of Human Genetics 2006 
                    meeting on October 11th, 1:00-2:00 p.m.  The training will be 
                    an introduction to the UCSC Genome Browser presented by OpenHelix.<BR><BR>

                    The introductory tutorial will cover the topics needed to 
                    effectively use the Genome Browser, including: basic 
                    functionality of Genome Browser searching and BLAT use, 
                    Table Browser use, creating and using Custom Tracks, and 
                    an introduction to the Gene Sorter.<BR><BR>

                    The free one hour tutorial will be conducted at the Hilton 
                    New Orleans Riverside, Compass Room, 2nd Floor, Riverside 
                    Building, Two Poydras Street. For further information visit 
                    <A HREF="http://www.openhelix.com" TARGET=_blank>OpenHelix</A>
                    or call 1-888-861-5051.<BR><BR>

                    <B>The Genome Browser seminars in Seattle and San Francisco, 
                    originally scheduled for October are postponed until January, 
                    2007.  Further details will be provided as they become available.</B>
                    </P>
                    <P>
                    <FONT FACE="courier" SIZE=3><B>7 September 2006 - Old Rhesus
		    Assembly Archived: </B></FONT>
		    The Jan. 2005 rheMac1 draft assembly has been moved from
		    the main Genome Browser website to our 
		    <A HREF="http://genome-rheMac1.cse.ucsc.edu">archive
		    server</A>. 
		    <A HREF="goldenPath/newsarch.html#090706">Read more</A>.</P>
		    <P>
		    <FONT FACE="courier" SIZE=3><B>8 August 2006 - New Opossum
		    Assembly Available in Genome Browser: </B></FONT>
		    The UCSC Genome Browser now includes the latest draft 
		    assembly of the opossum genome. The Jan. 2006 release of 
		    <em>Monodelphis domestica</em> (UCSC version monDom4) was 
		    sequenced and assembled by 
		    <A HREF="http://www.broad.mit.edu/mammals/opossum/"
		    TARGET=_blank>The Broad Institute</A>, Cambridge, MA, USA. 
		    <A HREF="goldenPath/newsarch.html#080806">Read more</A>.</P>
		    <P>
		    <FONT FACE="courier" SIZE=3><B>1 August 2006 - v2.1 Chicken Assembly Available in Genome Browser: </B></FONT>
	   	    We have updated the Chicken Genome Browser to include the 
		    May 2006 v2.1 assembly (UCSC version galGal3) produced by 
	 	    the Genome Sequencing Center at the Washington University 
		    School of Medicine in St. Louis, MO, USA (WUSTL). 
		    <A HREF="goldenPath/newsarch.html#080106">Read more</A>.</P>
    	            </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
