<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>Human Genome Working Draft</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg"></TH></TR>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
	<table bgcolor="000000" cellpadding="1" cellspacing="1" width="99%" height="27"><tr bgcolor="2636D1"><td valign="middle">
	<table BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24">
		<TR><TD VALIGN="middle"><font color="#89A1DE">&nbsp; 
			<a href="/cgi-bin/hgGateway" class="topbar">
				Genome Browser</a> &nbsp; - &nbsp; 
			<A HREF="/cgi-bin/hgBlat?command=start" class="topbar">
				Blat Search</a> &nbsp; - &nbsp; 
			<A HREF="FAQ.html" class="topbar">
				FAQ</a> &nbsp; - &nbsp; 
			<A HREF="goldenPath/help/hgTracksHelp.html" class="topbar">
				User Guide</a> &nbsp; </font></TD>
    	</TR></TABLE>
	</TD></TR></TABLE>
</TD></TR>

<TR HEIGHT="100%" VALIGN="top">
  <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
  	<!-- outer table is to align leftbar & content boxes at top of page -->
  	<TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0 MARGIN=0>
	<TR><TD><IMG SRC="images/DOT.gif" HEIGHT="7"></TD></TR>
	<TR HEIGHT="100%"><TD HEIGHT="100%">
		<!--outer table is for black border-->
		<TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
   	 	<TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
		<TR HEIGHT="100%">
			<TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%"> 
				<FORM ACTION="" METHOD="">
			<table cellpadding="" cellspacing="0" margins="0" border="0"><tr><td valign="top">
				<FONT CLASS="leftbar">Organism</FONT>
			</td></tr><tr valign="bottom"><td valign="bottom">
					<SELECT NAME="" >
						<OPTION VALUE="">Human
						<OPTION VALUE="">Mouse 
					</SELECT>
			</td></tr></table><hr>
			<A CLASS="leftbar" HREF="../cgi-bin/hgGateway">
				Browser</A><hr>
			<A CLASS="leftbar" HREF="../cgi-bin/hgBlat?command=start">
				Blat</A><hr>
			<A CLASS="leftbar" HREF="goldenPath/hgText.html">
				Tables</A><hr>
			<A CLASS="leftbar" HREF="downloads.html">
				Download</A><hr>
		  	<A CLASS="leftbar" HREF="mirror.html">
				Mirrors</A><hr>
  			<A CLASS="leftbar" HREF="goldenPath/hgText.html">
				Tables</A><hr>
  			<A CLASS="leftbar" HREF="http://genome-archive.cse.ucsc.edu">
				Archive</A><hr>
			<A CLASS="leftbar" HREF="goldenPath/credits.html">
				Credits</A><hr>
			<A CLASS="leftbar" HREF="cite.html">
				Cite Us</A><hr>
			<A CLASS="leftbar"
			HREF="http://www2.ucsc.edu/staff_hr/employment/listings/020452.htm" TARGET=_blank>
				Jobs</A><hr>
  			<A CLASS="leftbar" HREF="bestlinks.html">
				Links</A><hr>
				</FORM>
			<BR>
			<IMG SRC="images/DOT.gif" HEIGHT=1800 WIDTH=1>
		</TD></TR></TABLE>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
</TD>

<!--Content Tables-->  
  <TD VALIGN="top"><TABLE CELLPADDING=5 BORDER=0><TR><TD>
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<A NAME="About"></A><FONT SIZE="4"><b>&nbsp; About UCSC Genome Bioinformatics</FONT>
	</TD></TR></TABLE>
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD WIDTH=10></TD><TD>
		<P>This site contains working drafts for the human genome and
		the mouse genome. The human genome is over 90% complete. Approximately 
		75% of the sequence is in a highly accurate 'finished' state.  The
		other 25% is merely 'draft' quality.  Some care must be taken
		when interpreting draft regions, but these are still very 
		useful to the working scientist.  
		<P>The mouse genome is
		largely the result of a full genome shotgun assembly. While there are 
		many gaps in the sequence, the order and orientation of the contigs on 
		either side of a gap are known in almost all cases. We estimate
		that 90-96% of the mouse genome is present in the
		assembly. We encourage you to explore the
		working drafts with the genome browser, which displays
		the work of many annotators worldwide.
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100% BORDER=0><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; News </FONT></TD>
			<TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD WIDTH=10></TD><TD>

        <P><FONT FACE="courier" SIZE=2>May 24, 2002</FONT><BR>
	Bulk downloads of the April 2002 hg11 human genome assembly (NCBI Build 29) 
	are now available from <A
	HREF="ftp://genome.cse.ucsc.edu/goldenPath/05apr2002"
	TARGET=_blank>ftp://genome.cse.ucsc.edu/goldenPath/05apr2002</A>. You
	can also access the data from the Downloading the Human Genome and
	Annotation of the Human Genome sections below.

        <P><FONT FACE="courier" SIZE=2>May 22, 2002</FONT><BR>
	We've just released a browser and BLAT server on 
	the latest Build 29 human genome assembly from NCBI (UCSC 
	version hg11). This assembly is based 
	on sequence information submitted into GenBank as of Apr. 5 2002. As with the 
	Dec. 2001 (hg10) release, this assembly was produced at 
	NCBI rather than at UCSC.  Consult NCBI's 
	<A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/human/release_notes.html"
	TARGET=_blank>Build 29 release notes</A> and 
	<A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/human/HsStats.html"
	TARGET=_blank>statistics</A> for more information about this release.
	This assembly contains nearly 75% finished sequence. 
	Currently, the human genome project appears to be on track to achieve the goal 
	of finishing at least 95% of the human genome (using Bermuda standards) by April 2003.
	
	<P>Although the NCBI human genome assembly has been steadily
	improving over the past year, mapping problems still exist in
	the current release. Most are small, relatively local rearrangements. 
	Larger scale problems include a rearrangement in the p-arm of Chr16 and
	several discrepancies in Chr17.  Researchers - especially positional 
	cloners - are strongly encouraged to use the tools provided 
	(<A HREF="http://genome.ucsc.edu/goldenPath/mapPlots/"
	TARGET=_blank>comparison plots</A>, <A
	HREF="http://genome.ucsc.edu/goldenPath/chromReports/"
	TARGET=_blank>chromosome reports</A>) to evaluate the
	accuracy of the assembly in specific regions of interest.

	<P>Bulk downloads of the hg11 data should be available from this site
	in approximately one week. New annotation tracks will be posted as soon
	as they become available.


        <P><FONT FACE="courier" SIZE=2>Apr 24, 2002</FONT><BR>
	Bulk downloads of the February 2002 mouse genome assembly are now
	available from <A HREF="ftp://genome.cse.ucsc.edu/goldenPath/mmFeb2002"
	TARGET=_blank>ftp://genome.cse.ucsc.edu/goldenPath/mmFeb2002</A>. You
	can also access the data from the Downloading the Mouse Genome section
	below.

        <P><FONT FACE="courier" SIZE=2>Apr 19, 2002</FONT><BR>
	The February 2002 mouse genome assembly is now available
	in the browser and for BLAT searching. This assembly
	was produced at the Whitehead Institute using their Arachne software.
	We'd like to thank them and the Mouse Genome Sequencing Consortium for 
	providing this assembly, which has 
	<A HREF="goldenPath/credits.html#mouserestrict">specific conditions for
	use</A>. Bulk downloads of the data
	should be available in approximately one week. Coordination with
	mouse genome data access at 
	<A HREF="http://www.ensembl.org/Mus_musculus/">Ensembl</A> and 
	<A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/M_musculus.html">NCBI</A> 
	is in progress.  We'd also like to acknowledge the UCSC team that produced 
	this release: Jim Kent, Terry Furey, Matt Schwartz, Fan Hsu, Yontao Lu, 
	Donna Karolchik, Chuck Sugnet, and Ryan Weber.


    </TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<BR>
		
<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4" ><a name="Downloading"></a><b>&nbsp; Downloading the Human Genome</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD>
<UL>
	<LI><A HREF="goldenPath/05apr2002/bigZips/"><FONT SIZE="4"
	TARGET=_blank>Apr. 5, 2002 full data set</FONT></A>
	<LI><A HREF="goldenPath/05apr2002/chromosomes/"><FONT SIZE="4"
	TARGET=_blank>Apr. 5, 2002 data set by chromosome</FONT></A>
	<LI><A HREF="goldenPath/22dec2001/bigZips/"><FONT SIZE="4" TARGET=_blank>Dec. 22, 2001 full data set</FONT></A>
	<LI><A HREF="goldenPath/22dec2001/chromosomes/"><FONT SIZE="4" TARGET=_blank>Dec. 22, 2001 data set by chromosome</FONT></A>
	<LI><A HREF="goldenPath/06aug2001/bigZips/"><FONT SIZE="4" TARGET=_blank>August 6, 2001 full data set</FONT></A>
	<LI><A HREF="goldenPath/06aug2001/chromosomes/"><FONT SIZE="4" TARGET=_blank>August 6, 2001 data set by chromosome</FONT></A>
	<LI><A HREF="http://genome-archive.cse.ucsc.edu/"><FONT SIZE="4" TARGET=_blank>Archive of older assemblies</FONT></A>
<!-- CHEAT: not closing <UL> tag tightens bottom spacing reliably in unix & windows! -->
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TH></TD></TR></TABLE>
</TD></TR></TABLE>
	
<BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4" ><a name="Annotation"></a><b>&nbsp; Annotation of the
	Human Genome</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD>
<UL>
	<LI><A HREF="goldenPath/gbdDescriptions.html"><FONT SIZE="4">Description of annotation database</FONT></A><FONT
	SIZE="4"></FONT>
	<LI><A HREF="goldenPath/05apr2002/database/"><FONT SIZE="4">Apr. 5,
	2002 freeze annotation database</FONT></A>
	<LI><A HREF="goldenPath/22dec2001/database/"><FONT SIZE="4">Dec. 22,
	2001 freeze annotation database</FONT></A>
	<LI><A HREF="goldenPath/06aug2001/database/"><FONT SIZE="4">August 6,
	2001 freeze annotation database</FONT></A>
	<LI><A HREF="goldenPath/mapPlots/"><FONT SIZE="4">Genome Map Plots</FONT></A><FONT
	SIZE="4"></FONT>
<!-- CHEAT: not closing <UL> tag tightens bottom spacing reliably in unix & windows! -->
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TH></TD></TR></TABLE>
</TD></TR></TABLE>
	
<BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4" ><a name="DownloadingMouse"></a><b>&nbsp; Downloading the Mouse Genome</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD>
<UL>
	<LI><A HREF="goldenPath/mmFeb2002/bigZips/"><FONT SIZE="4"
	TARGET=_blank>Feb. 2002 full data set </FONT></A>
	<LI><A HREF="goldenPath/mmFeb2002/chromosomes/"><FONT SIZE="4"
	TARGET=_blank>Feb. 2002 data set by chromosome</FONT></A>
	<LI><A HREF="goldenPath/mmNov2001/bigZips/"><FONT SIZE="4" TARGET=_blank>Nov. 2001 full data set </FONT></A>
	<LI><A HREF="goldenPath/mmNov2001/chromosomes/"><FONT SIZE="4"
	TARGET=_blank>Nov. 2001 data set by chromosome</FONT></A>
<!-- CHEAT: not closing <UL> tag tightens bottom spacing reliably in unix & windows! -->
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TH></TD></TR></TABLE>
</TD></TR></TABLE>
	
<BR>
	
<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4" ><a name="AnnotationMouse"></a><b>&nbsp; Annotation of
	the Mouse Genome</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD>
<UL>
	<LI><A HREF="goldenPath/mmFeb2002/database/"><FONT SIZE="4"
	TARGET=_blank>Feb. 2002 freeze annotation database</FONT></A>
	<LI><A HREF="goldenPath/mmNov2001/database/"><FONT SIZE="4" TARGET=_blank>Nov. 2001
	freeze annotation database</FONT></A>
<!-- CHEAT: not closing <UL> tag tightens bottom spacing reliably in unix & windows! -->
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TH></TD></TR></TABLE>
</TD></TR></TABLE>
	
<BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4" ><a name="Tech"></a><b>&nbsp; Technical Information About the Assembled Sequence</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD>
<UL>
	<LI><A HREF="goldenPath/term.html"><FONT SIZE="4">Terminology</FONT></A><FONT SIZE="4"></FONT>
	<LI><A HREF="goldenPath/datorg.html"><FONT SIZE="4">Data organization and format</FONT></A><FONT SIZE="4"></FONT>
	<LI><A HREF="goldenPath/algo.html"><FONT SIZE="4">Algorithms</FONT></A><FONT SIZE="4"></FONT>
	<LI><A HREF="goldenPath/limits.html"><FONT SIZE="4">Known limitations</FONT></A><FONT SIZE="4"></FONT>
	<LI><A HREF="goldenPath/stats.html"><FONT SIZE="4">Summary statistics</FONT></A><FONT SIZE="4"></FONT>
	<LI><A HREF="goldenPath/chromReports"><FONT SIZE="4">Chromosome Reports</FONT></A><FONT SIZE="4"></FONT>
<!-- CHEAT: not closing <UL> tag tightens bottom spacing reliably in unix & windows! -->
</TD><TD WIDTH=15>&nbsp;</TD></TR></TABLE>
	<BR></TD></TR></TABLE>
</TD></TR></TABLE>
	
<BR>
	
<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4" ><a name="Other"></a><b>&nbsp; Other Links</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=3></TH></TR><TR><TD>
<UL>
	<LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/"><FONT SIZE="4">Human Genome Project at UCSC</FONT></A><FONT SIZE="4"></FONT>
	<LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/"><FONT SIZE="4">Center for Biomolecular Science and Engineering Home Page</FONT></A>
<!-- CHEAT: not closing <UL> tag tightens bottom spacing reliably in unix & windows! -->
</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
</TD></TR></TABLE>

</TD></TR></TABLE>
</TD></TR></TABLE>
</BODY></HTML>

