<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                          <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="http://www.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>

		    <P>
		    <FONT FACE="courier" SIZE=3><B>10 Mar. 2008 - Orangutan Genome Browser Now Available</B></FONT><BR>
		    <P>
		    The UCSC Genome Bioinformatics Group has released a Genome 
		    Browser and Blat server for the Jul. 2007 draft assembly of
 		    the Sumatran orangutan genome, <em>Pongo pygmaeus 
		    abelii</em>. This assembly (UCSC version 
		    ponAbe2, WUSTL version Pongo_albelii-2.0.2) was provided by 
		    the Genome Sequencing Center at Washington University School
		    of Medicine in St. Louis (WUSTL), MO, USA. </P>
		    <P>
		    The orangutan genome was sequenced to 6X coverage 
		    using a female orangutan known as &quot;Susie&quot; from 
		    the Gladys Park Zoo (Brownsville, TX, USA). The combined 
		    sequence reads were assembled using 
		    <A HREF="http://www.genome.org/cgi/content/abstract/13/9/2164"
		    TARGET=_blank>PCAP</A> and filtered for all 
		    known non-orangutan sequence contaminants. For more details
		    about the assembly, see the orangutan browser
		    <A HREF="../cgi-bin/hgGateway?db=ponAbe2">gateway</A>
		    page and the WUSTL 
		    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Pongo%20abelii"
		    TARGET=_blank><em>Pongo abelii</em></A> web page.</P>
		    <P>
		    Of the 3.09 Gb of total sequence, 3.08 Gb are ordered and 
		    oriented along the chromosomes. Gap sizes between 
		    supercontigs were estimated based on their size in human, 
		    with a maximum gap size of 30 kb allowed. For a list of the
		    chromosomes in this assembly, click the 
		    "Sequences" link on the orangutan browser 
		    <A HREF="../cgi-bin/hgGateway?db=ponAbe2">gateway</A> page.
		    The mitochondrial sequence is also available as the virtual 
		    chromosome &quot;chrM&quot;.</P> 
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/ponAbe2">FTP 
		    server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#orangutan">Downloads</A> 
		    page. These data have 
		    <A HREF="goldenPath/credits.html#orangutan_use">specific 
		    conditions for use</A>. The orangutan browser annotation 
		    tracks were generated by UCSC and collaborators worldwide. 
		    <P>
		    We'd like to thank WUSTL for providing this
		    assembly. The UCSC ponAbe2 browser and documentation were
		    produced by Hiram Clawson, Kayla Smith, Robert Kuhn, Ann
		    Zweig and Donna Karolchik.
		    See the <A HREF="goldenPath/credits.html#orangutan_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release. 
		    </P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>7 Mar. 2008 - New Reverse Functionality Released </B></FONT><BR>
		    <P>
		    The Genome Browser team is happy to announce new functionality
		    in the main genome browser track display.  A new configuration
                    button, "Reverse," now allows users to view the entire browser 
                    image flipped right-to-left.
                    This is especially useful when a user's gene of interest aligns 
                    on the opposite strand from the reference assembly. Those
                    genes then appear in the 5' to 3' direction.
                     </P>
                    <P>When the reverse function has been activated, all of the track
                    labels usually displayed on the left side of the track (including
                    the mini-button providing access to configuration options) are
                    displayed on the right side.  This allows users to see at a
                    glance which direction is being displayed.  All navigation options
                    operate as expected.</P>
                    <P>Thanks to Mark Diekhans, Ann Zweig, Robert Kuhn and the
                    rest of the engineering team for implementing this feature.
		    </P>

<!--
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>15 Oct. 2007 - Genome Browser for Zebrafish Zv7 Released: </B></FONT>
		    The latest zebrafish assembly -- Zv7 (UCSC version 
		    danRer5, July 2007) -- is now available in the UCSC
		    Genome Browser. 
                    <A HREF="goldenPath/newsarch.html#101507">Read more</A>.</P>
-->
                        </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
