<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="description" content="UCSC Genome Browser">
	<META name="keywords" content="genome, genome browser, genome assembly, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA">
	<TITLE>UCSC Genome Bioinformatics Home</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">

</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<!--HOTLINKS BAR-->
    <TR><TD COLSPAN=2 HEIGHT=40>
    <TABLE BGCOLOR="000000" CELLPADDING="1" CELLSPACING="1" WIDTH="99%" HEIGHT="27">
        <TR BGCOLOR="2636D1"><TD VALIGN="middle">
        <TABLE BORDER=0 CELLSPACING=0 CELLPADDING=0 BGCOLOR="2636D1" HEIGHT="24">
    	    <TR><TD VALIGN="middle"><FONT color="#89A1DE">&nbsp; 
 	    <A HREF="../cgi-bin/hgGateway" class="topbar">
			Genome Browser</a> &nbsp; - &nbsp; 
	    <A HREF="../cgi-bin/hgBlat" class="topbar">
			Blat Search</a> &nbsp; - &nbsp; 
	    <A HREF="../cgi-bin/hgText" class="topbar">
			Table Browser</a> &nbsp; - &nbsp; 
	    <A HREF="FAQ.html" class="topbar">
			FAQ</a> &nbsp; - &nbsp; 
	    <A HREF="goldenPath/help/hgTracksHelp.html" class="topbar">
			User Guide</a> &nbsp; </FONT></TD>
        </TR></TABLE>
    </TD></TR></TABLE>
    </TD></TR>

    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0 MARGIN=0>
        <TR><TD><IMG SRC="images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="../cgi-bin/hgGateway">
			Browser</A><HR>
	        <A CLASS="leftbar" HREF="../cgi-bin/hgBlat">
			Blat</A><HR>
	        <A CLASS="leftbar" HREF="../cgi-bin/hgText">
			Tables</A><HR>
	        <A CLASS="leftbar" HREF="downloads.html">
			Downloads</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/releaseLog.html">
			Release Log</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/customTracks/custTracks.html">
			Custom Tracks</A><HR>
	        <A CLASS="leftbar" HREF="mirror.html">
			Mirrors</A><HR>
  	        <A CLASS="leftbar" HREF="http://genome-archive.cse.ucsc.edu">
			Archives</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/credits.html">
			Credits</A><HR>
	        <A CLASS="leftbar" HREF="cite.html">
			Cite Us</A><HR>
	        <A CLASS="leftbar" HREF="jobs.html">
			Jobs</A><HR>
  	        <A CLASS="leftbar" HREF="contacts.html">
			Contact Us</A><HR>
	        <BR>
	        <IMG SRC="images/DOT.gif" ALT="" HEIGHT=1100 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Content Tables-->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; About the UCSC Genome Bioinformatics Site</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
		    This site contains the reference sequence for the human
	            and C. elegans genomes and working drafts for the mouse, rat, 
		    C. briggsae, and SARS genomes. To access a genome, click the 
		    Browser, Blat, or Tables link in the left sidebar. 
	            <P>
	            We encourage you to explore the reference sequence and 
		    working drafts with the UCSC Genome Browser, which displays 
                    the work of many annotators worldwide. The Genome Browser and 
                    the data it displays are freely available for academic, 
                    nonprofit, and personal use with a few exceptions. Check the 
                    README.txt file in the download directory of a particular 
                    assembly to view the use restrictions specific to that 
                    release. Program-driven use is limited to a maximum of one 
                    hit every 10 seconds and no more than 5,000 hits per day.  
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
		        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		    </TD></TR></TABLE>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
	            <P>
                    <FONT FACE="courier" SIZE=3><B>Aug 8, 2003</B></FONT><BR>
		    The latest human genome reference sequence (NCBI Build 34,
		    July 2003) is now available as database hg16 in the UCSC
		    Genome Browser and blat server.
		    <P>
		    There are 
		    2,843,433,602 finished sequenced bases in the ordered and 
		    oriented portion of the assembly, which is an increase of 
		    0.4 percent, or approximately 11 Mb, over the Build 33 
		    assembly. The missing portions are essentially contained in 
		    approximately 301 defined gaps in the euchromatic region 
		    representing DNA regions with unusual structures that can't 
		    be reliably sequenced using current technology. 
		    Of particular note in this release is the addition of the 
		    pseudoautosomal regions of the Y chromosome. 
		    <P>
		    Some sequence joins between adjacent clones in this assembly
		    could not be computationally 
		    validated because the clones originated from different 
		    haplotypes and contained polymorphisms in the overlapping 
		    sequence, or the overlap was too small to be to be reliable. 
		    In these instances, the sequencing center responsible for 
		    the particular chromosome has provided data to support 
		    the join in the form of an electronic certificate. The 
		    Build 34 certificates may be reviewed 
		    <A HREF="http://genome.cse.ucsc.edu/goldenPath/certificates/certificate_hg16.html">here</A>.
		    <P>
		    Bulk downloads of the data are available via FTP at 
		    <A HREF="ftp://genome.ucsc.edu/goldenPath/hg16">ftp://genome.ucsc.edu/goldenPath/hg16</A> 
		    or through the <A HREF="../downloads.html">Downloads</A> 
		    link on this page.  We recommend that FTP be used rather 
		    than HTML for the download of large or multiple files.
		    <P>
		    We'd like to thank NCBI and the International Human Genome 
		    Sequencing Consortium for
		    furnishing the data, and the UCSC team members who 
		    contributed to this release: Terry Furey, Hiram Clawson, 
		    Heather Trumbower, Mark Diekhans, Donna Karolchik, Jim
		    Kent and our sysadmin team Patrick Gavin, Jorge Garcia, 
		    and Paul Tatarsky.

	            <P>
                    <FONT FACE="courier" SIZE=3><B>Jul 14, 2003</B></FONT><BR>
                    The UCSC Genome Bioinformatics Group has released a browser
		    and BLAT server on the v. 3.1 rat genome assembly from the 
		    Rat Genome Sequencing Consortium. This
		    assembly (UCSC version rn3, June 2003) was produced by the 
		    Atlas group at <A HREF="http://hgsc.bcm.tmc.edu/projects/rat/" TARGET=_blank>Baylor Human Genome Sequencing Center</A> (HGSC). 
 		    This assembly is a minor update to the 3.0 release.
  		    Sequence changes affect only chromosomes 7 and X. No 
		    additional assembly releases are planned prior to the 
		    publication of the rat genome analysis papers. 
		    <P>
		    The 3.x assemblies reflect several sequence additions and 
		    software improvements over the previous 2.x assemblies, 
		    including the sequencing of over 1100 new BACs to cover gaps, 
 		    an improved marker set from the Medical College of Wisconsin, 
		    a new FPC map from the BC Genome Sciences Centre, and 
		    improved linking of bactigs. For detailed information and 
		    statistics about the 3.x assemblies, see the Baylor HGSC 
		    <A HREF="ftp://ftp.hgsc.bcm.tmc.edu/pub/analysis/rat/README"
		    TARGET=_blank>README</A>. 
		    <P>
		    Downloads of the rat sequence and annotation data are 
		    available at 
		    <A HREF="ftp://genome.ucsc.edu/goldenPath/rnJun2003/" TARGET=_blank>
		    ftp://genome.ucsc.edu/goldenPath/rnJun2003/</A> or via 
		    the Downloads link on this page. These data are made 
		    available with 
		    <A HREF="http://genome.ucsc.edu/goldenPath/credits.html#rat_use">specific conditions for use</A>.
		    <P>
		    We'd like to thank the Rat Genome Sequencing Consortium and
		    Baylor HGSC for providing this assembly, collaborators
		    from other institutions who have contributed annotations, 
 		    and Arian Smit for updating RepeatMasker for this release.
		    We'd also like
		    to acknowledge the contributions of several individuals at
		    UCSC, including Hiram Clawson, Heather Trumbower, Robert Kuhn,
		    Yontao Lu, Terry Furey, Mark Diekhans, Robert Baertsch, 
		    Donna Karolchik, Jim Kent, and our sysadmin team Jorge 
		    Garcia, Patrick Gavin, and Paul Tatarsky.
		    <P>

    	            </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
	    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	        
                    <FONT SIZE="4" ><A name="Tech"></A><B>&nbsp; Technical Information About the Assembled Sequence</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
	            <LI><A HREF="goldenPath/gbdDescriptions.html"><FONT SIZE="4">Description of the annotation database</FONT></A>
	            <LI><A HREF="goldenPath/datorg.html"><FONT SIZE="4">Data organization and format</FONT></A>
	            <LI><A HREF="goldenPath/term.html"><FONT SIZE="4">Terminology</FONT></A>
                    </UL>
                    </TD>
                    <TD WIDTH=15>&nbsp;</TD>
                 </TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	            <FONT SIZE="4" ><A NAME="Other"></A><B>&nbsp; Other Links</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/"><FONT SIZE="4">Human Genome Project at UCSC</FONT></A><FONT SIZE="4"></FONT>
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/"><FONT SIZE="4">Center for Biomolecular Science and Engineering Home Page</FONT></A>
                    </UL>
                </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
</TD></TR></TABLE>

</BODY></HTML>

