<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="description" content="UCSC Genome Browser">
	<META name="keywords" content="genome, genome browser, human genome, genome assembly, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, mouse genome, rat genome, chicken genome, chimp genome, dog genome fruitfly, drosophila, fugu, yeast, ciona,  comparative genomics">
	<TITLE>UCSC Genome Browser Home</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">

</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="inc/home.topbar.html"--></TD></TR>
<!--Left Sidebar Menu----------------------------------------->  
    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="cgi-bin/hgGateway">
			Genome Browser</A><HR>
	        <A CLASS="leftbar" HREF="ENCODE/">
			ENCODE</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/hgBlat">
			Blat</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/hgTables">
			Table Browser</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/hgNear">
			Gene Sorter</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/hgPcr">
			In Silico PCR</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/hgVisiGene">
			VisiGene</A><HR>
	        <A CLASS="leftbar" HREF="cgi-bin/pbGateway">
			Proteome Browser</A><HR>
	        <A CLASS="leftbar" HREF="util.html">
			Utilities</A><HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/downloads.html">
			Downloads</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/releaseLog.html">
			Release Log</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/customTracks/custTracks.html">
			Custom Tracks</A><HR>
	        <A CLASS="leftbar" HREF="mirror.html">
			Mirrors</A><HR>
  	        <A CLASS="leftbar" HREF="http://genome-archive.cse.ucsc.edu">
			Archives</A><HR>
	        <A CLASS="leftbar" HREF="http://openhelix.com/ucscmaterials.shtml"
			TARGET=_blank>
			Training</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/credits.html">
			Credits</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/pubs.html">
			Publications</A><HR>
	        <A CLASS="leftbar" HREF="cite.html">
			Cite Us</A><HR>
	        <A CLASS="leftbar" HREF="license/">
			Licenses</A><HR>
	        <A CLASS="leftbar" HREF="jobs.html">
			Jobs</A><HR>
  	        <A CLASS="leftbar" HREF="staff.html">
			Staff</A><HR>
  	        <A CLASS="leftbar" HREF="contacts.html">
			Contact Us</A><HR>
	        <BR>
	        <IMG SRC="images/DOT.gif" ALT="" HEIGHT=700 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Introduction---------------------------------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; About the UCSC Genome Bioinformatics Site</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
		    This site contains the reference sequence and working draft
		    assemblies for a large collection of genomes. It also 
		    provides a portal to the ENCODE project.  
		    <P>
		    We encourage you to explore these sequences with our tools.
		    The Genome Browser zooms and scrolls over chromosomes, 
		    showing the work of annotators worldwide. The Gene Sorter 
		    shows expression, homology and other information on groups 
		    of genes that can be related in many ways.  
		    Blat quickly maps your sequence to the genome.  The Table 
		    Browser provides convenient access to the underlying 
		    database. VisiGene lets you browse through a large 
		    collection of <em>in situ</em> mouse and frog images to
		    examine expression patterns.</P>
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
		        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		    </TD></TR></TABLE>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
		    <P>
		    To receive announcements of new genome 
		    assembly releases, new software features, updates and 
		    training seminars by email, subscribe to the
		    <A HREF="http://www.soe.ucsc.edu/mailman/listinfo/genome-announce" TARGET=_blank>genome-announce</A> mailing list.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>20 December 2005 - Dog Genome Browser Update Released</B></FONT>
		    <P>
		    UCSC has updated the dog Genome Browser to include the 
		    May 2005 v2.0 assembly (UCSC version canFam2) sequenced and
		    assembled by the Broad Institute of MIT and Harvard and 
		    Agencourt Bioscience. 
		    The whole genome shotgun (WGS) sequence is 
		    based on 7.6X coverage of the dog genome which includes
		    more than 98% of the euchromatic genome. 
		    <P>
		    The dog genome, which contains approximately 2.5 billion 
		    base pairs, is similar in size to the genomes of humans and
		    other mammals. The boxer breed was selected for the initial 
		    sequencing effort, based on the lower variation rate in its 
		    genome relative to other breeds. In addition to the boxer,
		    samples from several other dog breeds were used to generate
		    a set of single nucleotide polymorphisms (SNPs) to 
		    facilitate disease studies.
		    The SNPs are available from the Broad Institute
		    <A HREF="http://www.broad.mit.edu/ftp/pub/papers/dog_genome/snps_canfam2/"
		    TARGET =_blank>dog SNP</A> web page.
		    For more information about the dog draft assembly, see the 
		    Broad Institute 
		    <A HREF="http://www.broad.mit.edu/mammals/dog/"
		    TARGET=_blank>Dog Genome Sequencing Project</A> web page.
		    <P>
		    The dog sequence and annotation data can be downloaded from
		    the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/canFam1/">FTP server</A>
		    or <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#dog">downloads</A>
		    page. These data have 
		    <A HREF="/goldenPath/credits.html#dog_use">specific 
		    conditions for use</A>.
		    <P>
		    Many thanks to the Broad Institute of MIT and Harvard, 
		    Agencourt Bioscience, and the other institutions who 
		    contributed to the sequencing, assembly, and mapping
		    efforts.  The canFam2 Genome Browser team included
		    Angie Hinrichs, Jennifer Jackson, and Donna Karolchik.
		    See the 
		    <A HREF="/goldenPath/credits.html#dog_credits">credits</A> 
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release.
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>15 December 2005 - New Mouse Assembly Available in Genome Browser</B></FONT>
		    <P>
		    The latest mouse genome assembly from the Mouse Genome
		    Sequencing Consortium, NCBI Build 35 (UCSC version mm7),
		    is now available in the UCSC Genome Browser. 
		    <P>
		    The Build 35 assembly includes approximately 2.6 Gb of
		    sequence, of which about 2.2 Gb is finished 
		    sequence. Chromosomes 2, 
		    4, 11 and X are finished in this build. To review in-depth
		    statistics on the assembly, see the NCBI
                    <A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/mouse/contig/Build35.html"
                    TARGET=_blank>Build 35 Data</A> web page.
<!--
		    NOTE: In the Genome Browser mm6 assembly,
		    CHRY_random erroneously contains a region duplicated from 
		    chrY. For more information about this issue, see the 
		    Genome Browser
		    <A HREF="FAQ/FAQdownloads#download30">FAQ</A>. 
-->
		    Please note that the UCSC mm7 database contains only the 
		    reference strain C57BL/6J.</P>
                    <P>
                    The mm7 sequence and annotation data may be downloaded 
		    from the Genome Browser
                    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/mm7/">FTP
                    server</A> or
                    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/mm7/">Downloads</A>
                    web page. The mm7 annotation tracks were generated by UCSC
                    and collaborators worldwide. </P>
<!--
		    <P>
		    Mirror sites, please note: due to the size of the 
		    comparative genomics annotations in this assembly, the 
		    data set requires an unusually large amount of disk space.
		    If you regularly download updates of the mouse data to 
		    your site, you may want to examine your available disk 
		    space before adding the annotation database from this 
		    assembly and/or reschedule your download for a period of 
		    low activity on your server. The size of the mm6 
		    annotation database directory 
		    ($WEBROOT/goldenPath/mm6/database/) is 
		    approximately 93 GB. Including the related net and chains,
		    the total size is 124 GB.</P>
-->
		    <P>
                    We'd like to thank Deanna Church and the Mouse Genome 
		    Sequencing Consortium for this assembly. We'd also like to
                    acknowledge the work of the UCSC mm7 team: Hiram Clawson, 
		    Fan Hsu, Ann Zweig, Kayla Smith, Robert Kuhn
		    and Donna Karolchik. 
		    For a complete list
		    of the individuals and organizations who participated 
		    in this assembly, see the 
		    <A HREF="goldenPath/credits#mouse_credits">Credits</A> 
		    page.
                    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>2 December 2005 - Announcing the VisiGene Image Browser:</B></FONT>
		    We are pleased to announce the release of a new software 
		    tool in the Genome Browser collection, the 
		    <A HREF="../cgi-bin/hgVisiGene">VisiGene Image Browser</A>.
		    <A HREF="goldenPath/newsarch.html#1202b05">Read more</A>.
		    </P>
		    <P>
		    <FONT FACE="courier" SIZE=3><B>2 December 2005 - Allen 
		    Brain Atlas Annotation Added to Genome Browser:</B></FONT>
	   	    We have released a new annotation track &mdash; Allen Brain
		    Atlas Probes &mdash; that may be found in the Expression and
		    Regulation section of the latest mouse and human assemblies
		    (mm6 and hg17). 
		    <A HREF="goldenPath/newsarch.html#1202a05">Read more</A>.
		    </P>
    	            </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
	    	
<!--Conditions of Use Section--------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome Browser 
                    are freely available for academic, nonprofit, and personal use 
		    with the following conditions:
		    <UL>
		    <LI>Genome sequence data use restrictions are noted within the
		    species sections on the <A HREF="goldenPath/credits.html">Credits</A> 
		    page.
		    <LI>Some annotation tracks contributed by external collaborators
		    contain proprietary data that have specific use restrictions. To
		    check for restrictions associated with a particular genome assembly, 
		    review the <em>database/README.txt</em> file in the assembly's 
		    downloads directory. 
		    </UL>
		    <P>
		    The Genome Browser and Blat software are free for academic,
		    nonprofit, and personal use. A license is required for commercial use.
		    See the <A HREF="license/">Licenses</A> page for more information.
		    <P>
                    Program-driven use of this software is limited to a maximum of one 
                    hit every 15 seconds and no more than 5,000 hits per day.  
		    <P>
		    Questions or problems regarding the UCSC Genome Browser 
		    software, database, genome assemblies, or release cycles 
		    should be sent to 
		    <A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A>.
		    Messages sent to this address will be posted automatically 
		    on the genome mailing list, which is also archived on a 
		    public Web-accessible pipermail archive. NOTE: This archive may be 
		    indexed by non-UCSC sites such as Google. 
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>

        <BR>

<!--Technical Information Section----------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	        
                    <FONT SIZE="4" ><A name="Tech"></A><B>&nbsp; Technical Information About the Assembled Sequence</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
	            <LI><A HREF="goldenPath/gbdDescriptions.html"><FONT SIZE="4">Description of the annotation database</FONT></A>
	            <LI><A HREF="goldenPath/datorg.html"><FONT SIZE="4">Data organization and format</FONT></A>
	            <LI><A HREF="goldenPath/term.html"><FONT SIZE="4">Terminology</FONT></A>
                    </UL>
                    </TD>
                    <TD WIDTH=15>&nbsp;</TD>
                 </TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
    	
<!--Related Links Section--------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	            <FONT SIZE="4" ><A NAME="Other"></A><B>&nbsp; Related Links</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
<!-- Commented out til this page is updated - very out of date
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/"><FONT SIZE="4">Human Genome Project at UCSC</FONT></A><FONT SIZE="4"></FONT>
-->
	            <LI><A HREF="http://www.cbse.ucsc.edu/" TARGET=_blank><FONT SIZE="4">Center for Biomolecular Science and Engineering Home Page</FONT></A>
                    </UL>
                </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
</TD></TR></TABLE>

<!-- Web visibility------------------------------------------------>
<A HREF="knownGeneLists.html"><IMG SRC="images/DOT.gif" WIDTH="1" HEIGHT="1" ALIGN="BOTTOM" BORDER="0" ALT=""></A>
</BODY></HTML>

