<!-- News Section ============================================- -->  
<TABLE style="background-color:#888888; width: 100%; border-width: 0px; border-spacing: 0px; padding: 0px">
    <TR><TD>        
    <TABLE style="background-color:#fffee8; width: 100%; border-width: 0px; border-spacing: 0px; padding: 0px">
        <TR><TD>        
        <TABLE style="background-color:#D9E4F8; background-image: url('images/hr.gif'); width: 100%; border: 0px">
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" title="Follow &#64;GenomeBrowser on Twitter" alt="Follow &#64;GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a>
                 <a title="Genome Browser Facebook page"
                 href="http://www.facebook.com/ucscGenomeBrowser"
                 target="_blank"><img
                 src="/images/FB-f-Logo__blue_22.jpg"
                 style="vertical-align:text-bottom; margin-left: 15px;"
                 alt="Genome Browser Facebook page"></a>
                 <a href="https://plus.google.com/107643636314131661344" rel="publisher"></a>
               </TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE style="background-color:#fffee8; width: 100%; padding: 0px">
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list. Please see our
          <a href="http://genome.ucsc.edu/blog"
          target=_blank>blog</a> for posts about Genome Browser tools, features, projects and more.
          </p>

    <!-- start news -->
<p>
  <font color="#006666" size="4"><b> 10 February 2016 -
  Two New Assemblies Now Available in the Genome Browser</b></font>
  <p>
  A Genome Browser is now available for the mouse lemur (<em>Microcebus murinus</em>)
  assembly released May 2015 by the
<A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
TARGET="_blank">The Broad Institute</A> and the
<A HREF="https://www.hgsc.bcm.edu/non-human-primates/mouse-lemur-genome-project"
TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center</A>.
  For more information and statistics about this assembly, see the NCBI assembly record
  for <a href="http://www.ncbi.nlm.nih.gov/assembly/329621" target="_blank"> Mmur_2.0</a>.
There are 10,311 scaffolds with a total size of 2,438,804,424 bases.
  </p>

  <p>
  Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
  <a href="ftp://hgdownload.soe.ucsc.edu/goldenPath/micMur2/">FTP server</a> or the
  <a href="http://hgdownload.soe.ucsc.edu/downloads.html#mouse_lemur">Downloads</a> page.
  Please observe the <a href="./goldenPath/credits.html#mouse_lemur_use">conditions for use</a> when
  accessing and using these data sets.  The annotation tracks for this browser were
  generated by UCSC and collaborators worldwide.  See the <a
  href="./goldenPath/credits.html#mouse_lemur_credits">Credits</a> page for a detailed list of the
  organizations and individuals who contributed to this release.
  </p>
  ---
  <p>
  A Genome Browser is now available for the platypus (<em>Ornithorhynchus anatinus</em>)
  assembly released February 2007 by the
<A HREF="http://genome.wustl.edu/genomes/detail/ornithorhynchus-anatinus/"
TARGET=_blank>Genome Sequencing Center at Washington University, St. Louis</A>.
For more information about this assembly, see
<A HREF="http://www.ncbi.nlm.nih.gov/assembly/237598/" TARGET="_blank">
Ornithorhynchus_anatinus-5.0.1</A> in the NCBI Assembly database. There are
201,525 scaffolds with a total size of 1,996,826,513 bases.
  </p>

  <p>
  Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
  <a href="ftp://hgdownload.soe.ucsc.edu/goldenPath/ornAna2/">FTP server</a> or the
  <a href="http://hgdownload.soe.ucsc.edu/downloads.html#platypus">Downloads</a> page.
  Please observe the <a href="./goldenPath/credits.html#platypus_use">conditions for use</a> when
  accessing and using these data sets.  The annotation tracks for this browser were
  generated by UCSC and collaborators worldwide.  See the <a
  href="./goldenPath/credits.html#platypus_credits">Credits</a> page for a detailed list of the
  organizations and individuals who contributed to this release.
  </p>

  <hr>
  <font color="#006666" size="4"><b>20 Jan 2016
  - dbSNP 142 Available for mm10</b></font>
  <p>
  Data from dbSNP build 142 is now available for the most recent mouse
  assembly (mm10/GRCm38). As was the case for previous annotations
  based on dbSNP data, there are three tracks in this release.
  One is a track containing all mappings of reference SNPs to the
  mouse assembly, labeled "All SNPs (142)". The other two tracks
  are subsets of this track and show different interesting and
  easily defined subsets of dbSNP:
  </p>
  <ul>
  <li>Common SNPs (142): uniquely mapped variants that appear in
  at least 1% of the population
  <li>Mult. SNPs (142): variants that have been mapped to more
  than one genomic location
  </ul>
  <p>
  By default, only the Common SNPs (142) are visible. The other
  tracks can be made visible using the track controls.
  These three SNPs (142) tracks can be found on the Mouse Dec.
  2011 (mm10/GRCm38) browser in the "Variation and Repeats" group.
  </p>
  Thank you to the <a href="http://www.ncbi.nlm.nih.gov/SNP/">
  dbSNP</a> group at NCBI for making these data publicly available.
  The tracks were produced at UCSC by Brian Raney, Angie Hinrichs
  and Matthew Speir.
  </p>

<hr>
<!-- start archives -->
<!--staged for future release

        <p>
        <font color="#006666" size="4"><b> 10 February 2016 -
        Two New Assemblies Now Available in the Genome Browser</b></font>
        <p>
        Genome Browsers are now available for the mouse lemur 
	(<em>Microcebus murinus</em>)
        and platypus (<i>Ornithorhynchus anatinus</i>) assemblies.
	<a href="goldenPath/newsarch.html#021016">Read more</a>. 
        </p>

	<font color="#006666" size="4"><b>20 Jan 2016 &mdash;
	dbSNP 142 Available for mm10</b></font>
        <a href="goldenPath/newsarch.html#012016">Read more</a>.
        </p>
-->
        <p>
        <font color="#006666" size="4"><b>08 January 2016 &mdash; 
        dbSNP 144 Available for hg19 and hg38 </font></b>
        <a href="goldenPath/newsarch.html#010816">Read more</a>.
        </p>

        <p>
        <font color="#006666" size="4"><b> 21 December 2015 -
        Two New Assemblies Now Available in the Genome Browser</b></font>
        <p>
        Genome Browsers are now available for the frog (<em>Xenopus tropicalis</em>)
        and cat (<i>Felis catus</i>) assemblies.
	<a href="goldenPath/newsarch.html#122115">Read more</a>. 
        </p>

            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
