<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                          <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="https://lists.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>

                    <!-- start news -->
		    <P>
                    <FONT FACE="courier" SIZE="3"><B>04 March 2011 - The Meaning of Red: Consensus on color scheme for CNVs</B></FONT>
		    </P>
		    <P>
		    Consensus was reached on color standards to represent CNV loss (red) 
		    and gain (blue) in genomics databases by DGV, NCBI, DECIPHER, UCSC and ISCA 
		    during the 1st annual ISCA Scientific Conference.
		    </P>
		    <P>
		    At the recent ISCA Scientific Conference in Atlanta, GA, Nigel Carter, 
		    representing DECIPHER, made a plea for a standardized color scheme for 
		    representation of CNV loss (deletion) or gain (duplication). Discussions 
		    in person and by email with leaders of major CNV/genomic databases (including 
		    DGV, dbVar, ISCA, and the UCSC Genome Browser) resulted in the recommendation 
		    to utilize red to represent loss/deletion (consistent with the original 
		    convention for CGH on metaphase chromosomes) and blue to represent 
		    gain/duplication (avoiding green which is difficult for color-blind 
		    individuals to discriminate from red).
		    </P>
		    <P>
		    Announcememnt reprinted from the website for the International Standards for
		    Cytogenomic Arrays (<A HREF = "http://iscaconsortium.org" TARGET = _BLANK>ISCA</A>) 
		    Consortium.  The UCSC Genome Browser is now conforming to this standard for the 
		    DECIPHER and DGV (which was released today) tracks on human assemblies and on 
		    any future tracks derived from dbVar.
		    </P>
		    <P>
		    Thanks to Nigel Carter, Steve Scherer, Deanna Church, Angie Hinrichs, Fan Hsu,
		    Pauline Fujita and Robert Kuhn.
		    </P>

		    <HR>
		    <P>
                    <FONT FACE="courier" SIZE="3"><B>01 March 2011 - Updated Chimpanzee Browser Released</B></FONT></P>
		    <P>
		    We are happy to announce the release of a Genome Browser for
		    the latest release of the chimpanzee (<em>Pan 
		    troglodytes</em>) genome. The Oct. 2010 assembly -- CGSC 
		    v2.1.3, UCSC version panTro3 -- was produced by the 
		    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Pan%20troglodytes&SECTION=collaborators"
		    TARGET=_BLANK>Chimpanzee Sequencing and Analysis 
		    Consortium</A>.</P>
		    <P>
		    This assembly covers about 97 percent of the genome and is 
		    based on 6X sequence coverage. It comprises 192,898 contigs 
		    with an N50 length of 44 kb and 33,990 supercontigs with an
		    N50 length of 8.4 Mb.  
		    Improvements introduced with the 2.1.3 assembly include the 
		    addition of over 300,000 finishing reads and the merging in 
		    of 640 finished BACS. Approximately 49,000 additional merges
		    were made in this assembly as compared with the 2.1 
		    assembly.
		    <P>
		    The whole genome shotgun data were derived primarily from 
		    the donor Clint, a captive-born male chimpanzee from the 
		    Yerkes Primate Research Center in Atlanta, GA, USA. 
		    The sequence data were assembled and organized by the 
		    Washington University Genome Center. The underlying whole 
		    genome shotgun data were generated at the Washington 
		    University School of Medicine and the Broad Institute. 
		    For detailed information about the assembly 
		    process, see the sequencing and analysis discussion in the 
		    <A HREF="http://www.ncbi.nlm.nih.gov/nuccore/AACZ00000000"
		    TARGET=_blank>GenBank accession record</A>.
		    </P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/panTro3/">FTP
		    server</A> or the 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#chimp">Downloads</A> 
		    page. The complete set of sequence reads is available at the

		    <A HREF="http://www.ncbi.nlm.nih.gov/Traces" 
		    TARGET=_blank>NCBI trace archive</A>. These data have 
		    <A HREF="goldenPath/credits.html#chimp_use">specific
		    conditions for use</A>.</P>
		    <P>
		    We'd like to thank the International Chimpanzee 
		    Sequencing and Analysis Consortium, Washington 
		    University at St. Louis School of Medicine Genome Sequencing
		    Center, and the Broad Institute for providing this assembly.
		    We'd also like to acknowledge the UCSC team who worked on 
		    this release: Hiram Clawson, Greg Roe, and Donna Karolchik. 
		    The chimpanzee browser annotation tracks were generated by 
		    UCSC and collaborators worldwide. </P>
		    <P>
		    <HR>

                    <FONT FACE="courier" SIZE="3"><B>19 January 2011 -
Browser Released for Zebrafish Zv9 Assembly: </B></FONT>
                    The most recent zebrafish assembly -- Zv9 (UCSC version
                    danRer7, Jul. 2010) -- is now available in the UCSC
                    Genome Browser. 
		    <A HREF="goldenPath/newsarch.html#011911">Read more</A>.
		    <P>
                    <FONT FACE="courier" SIZE="3"><B>10 January 2011 - Introducing Genome Browser Right-click Navigation: </B></FONT>
		    Several of the common display and navigation operations 
		    offered on the Genome Browser tracks page may now be quickly
		    accessed by right-clicking on a feature on the tracks image 
		    and selecting an option from the displayed menu. 
		    <A HREF="goldenPath/newsarch.html#011011">Read more</A>.
		    <P>
                    <FONT FACE="courier" SIZE="3"><B>2 December 2010 -
Searching for Tracks Just Got Easier! </B></FONT> </P>
		    With our latest Genome Browser release, we've introduced a 
		    new track search feature that will help you to quickly and 
		    easily find and display specific annotation data sets. The 
		    search feature is available via the "track search" button 
		    on the browser gateway and tracks display pages.
		    <A HREF="goldenPath/newsarch.html#120210">Read more</A>.
		    </P>
                    </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
