<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                          <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="https://lists.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>

                    <!-- start news -->

		    <FONT FACE="courier" SIZE="3"><B>13 Jan. 2010 - Lifespan of custom tracks within sessions</B></FONT>
		    <P>
		    Due to the popularity of UCSC 
		    <A HREF="goldenPath/help/hgTracksHelp.html#CustomTracks">custom tracks</A>
		    and 
		    <A HREF="goldenPath/help/hgSessionHelp.html">sessions</A>,
		    we are running out of disk space for storing custom 
		    tracks accessed within sessions. Although sessions 
		    themselves are saved for one year, user-generated tracks 
		    within sessions (including custom tracks) are saved for 48 
		    hours, the typical period for other custom tracks. Until 
		    now, because we have had space available on the server, we 
		    have attempted to keep custom tracks alive for the lifetime 
		    of the session. </P>
		    <P>
		    While we provide short-term custom track storage as a 
		    service to users who want to view their own data in the 
		    UCSC Genome Browser, we unfortunately do not have the
		    resources to guarantee long-term data storage.
		    Over the next few days, we will begin to actively delete 
		    custom tracks within sessions that have not been accessed 
		    within the past four months.  If you actively use your 
		    custom tracks within saved sessions, they will not be 
		    deleted.</P>
		    <P>
		    Instead of uploading your data to our servers, please 
		    consider using alternate data types such as 
		    <A HREF="http://genome.ucsc.edu/goldenPath/help/bigBed.html">bigBed</A>, 
		    <A HREF="http://genome.ucsc.edu/goldenPath/help/bigWig.html">bigWig</A>, 
		    or other URL-based tracks that enable you to store your 
		    data locally.  </P>
		    <HR>
		    <FONT FACE="courier" SIZE="3"><B>11 Jan. 2010 - Zebrafish release zv8 now available in Genome Browser </B></FONT><BR>
		    <P>
		    The most recent zebrafish assembly -- zv8 (UCSC version 
		    danRer6, Dec. 2008) -- is now available in the UCSC
		    Genome Browser. This assembly was produced by 
		    the Wellcome Trust Sanger Institute, UK. 
		    </P>
                    <P>
                    The danRer6 assembly consists of nearly 1.5 billion bp in
                    11,623 scaffolds with a coverage of 6.5-7x. It includes
                    sequence anchored to chromosomes
                    1-25 and chrM (mitochondrial), as well as 11,418 unplaced
                    scaffolds.  Significant problems existing in previous
                    assemblies have been rectified in this assembly through the
                    use of a whole-genome shotgun assembly with more coverage
                    and a reorganization of the fingerprint contig order and
                    orientation through more careful use of maps.
                    For more information about this assembly, see the Sanger
                    Institute web page for the
                    <A HREF="http://www.sanger.ac.uk/Projects/D_rerio/Zv8_assembly_information.shtml"
                    TARGET=_blank><em>Danio rerio</em> Sequencing Project</A>
                    and the track description page for the danRer6 Assembly
                    track.</P>
                    <P>
                    The danRer6 sequence and annotation data can be
                    downloaded from the UCSC Genome Browser
                    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/danRer6/">FTP server</A>
                    or
                    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#zebrafish">downloads</A>
                    page. Please review the
                    <A HREF="http://www.sanger.ac.uk/notices/use-policy.shtml"
                    TARGET=_blank>guidelines</A> for using these data.
                    <P>
                    We'd like to thank the Wellcome Trust Sanger Institute
                    for providing this assembly.
                    The UCSC zebrafish Genome Browser
                    was produced by Galt Barber, Robert Kuhn, Katrina Learned,
                    and Donna Karolchik. The initial set of annotation tracks
                    was generated by the UCSC Genome Bioinformatics Group. See
                    the
                    <A HREF="goldenPath/credits.html#zebrafish_credits">credits</A>
                    page for a detailed list of the organizations and
                    individuals who contributed to the release of
                    this browser.</P>
		    <P>
		    <HR>
<FONT FACE="courier" SIZE="3"><B>14 Dec. 2009 - New job posting: Biological Data Technician: </B></FONT>
		    The UCSC Genome Browser project is looking for a 
		    bioinformatician, biologist, or software engineer with a 
		    strong biology background to collect and import data  
		    into the UCSC Genome Browser database and 
		    website. 
		    <A HREF="goldenPath/newsarch.html#121409">Read more</A>.
		    </P>
		    <P>
<FONT FACE="courier" SIZE="3"><B>7 Dec. 2009 - Human Genome Browser default 
		    changing to hg19: </B></FONT>
		    In conjunction with the release of the UCSC Genes and
		    Conservation tracks on the hg19 (GRCh37) human assembly,
		    we have changed the default human browser on our website 
		    from hg18 to hg19. 
		    <A HREF="goldenPath/newsarch.html#120709">Read more</A>.
		    </P>
                        </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
