<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="description" content="UCSC Genome Browser">
	<META name="keywords" content="genome, genome browser, human genome, genome assembly, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, chicken genome, chimp genome, fruitfly, fugu, yeast comparative genomics">
	<TITLE>UCSC Genome Browser Home</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">

</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<!--HOTLINKS BAR-->
    <TR><TD COLSPAN=2 HEIGHT=40>
    <TABLE BGCOLOR="000000" CELLPADDING="1" CELLSPACING="1" WIDTH="99%" HEIGHT="27">
        <TR BGCOLOR="2636D1"><TD VALIGN="middle">
        <TABLE BORDER=0 CELLSPACING=0 CELLPADDING=0 BGCOLOR="2636D1" HEIGHT="24">
    	    <TR><TD VALIGN="middle"><FONT color="#89A1DE">&nbsp; 
 	    <A HREF="/cgi-bin/hgGateway" class="topbar">
			Genome Browser</a> &nbsp; - &nbsp; 
 	    <A HREF="/cgi-bin/hgNear" class="topbar">
			Gene Sorter</A> &nbsp; - &nbsp; 
	    <A HREF="/cgi-bin/hgBlat" class="topbar">
			Blat</a> &nbsp; - &nbsp; 
	    <A HREF="/cgi-bin/hgText" class="topbar">
			Table Browser</a> &nbsp; - &nbsp; 
	    <A HREF="/FAQ/" class="topbar">
			FAQ</a> &nbsp; - &nbsp; 
	    <A HREF="goldenPath/help/hgTracksHelp.html" class="topbar">
			Help</a> &nbsp; </FONT></TD>
        </TR></TABLE>
    </TD></TR></TABLE>
    </TD></TR>

    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="/cgi-bin/hgGateway">
			Genome Browser</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgNear">
			Gene Sorter</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgBlat">
			Blat</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgText">
			Tables</A><HR>
	        <A CLASS="leftbar" HREF="util.html">
			Utilities</A><HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/downloads.html">
			Downloads</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/releaseLog.html">
			Release Log</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/customTracks/custTracks.html">
			Custom Tracks</A><HR>
	        <A CLASS="leftbar" HREF="/encode/">
			ENCODE</A><HR>
	        <A CLASS="leftbar" HREF="mirror.html">
			Mirrors</A><HR>
  	        <A CLASS="leftbar" HREF="http://genome-archive.cse.ucsc.edu">
			Archives</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/credits.html">
			Credits</A><HR>
	        <A CLASS="leftbar" HREF="/goldenPath/pubs.html">
			Pubs</A><HR>
	        <A CLASS="leftbar" HREF="training.html">
			Training</A><HR>
	        <A CLASS="leftbar" HREF="cite.html">
			Cite Us</A><HR>
	        <A CLASS="leftbar" HREF="license/">
			Licenses</A><HR>
	        <A CLASS="leftbar" HREF="jobs.html">
			Jobs</A><HR>
  	        <A CLASS="leftbar" HREF="staff.html">
			Staff</A><HR>
  	        <A CLASS="leftbar" HREF="contacts.html">
			Contact Us</A><HR>
	        <BR>
	        <IMG SRC="images/DOT.gif" ALT="" HEIGHT=950 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Content Tables-->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; About the UCSC Genome Bioinformatics Site</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
		    This site contains the reference sequence for the human and <em>C. elegans</em> 
		    genomes and working drafts for the chimpanzee, mouse, rat, chicken, <em>Fugu</em>, <EM>Drosophila</EM>, 
		    <em>C. briggsae</em>, yeast, and SARS genomes.  It also shows the
		    CFTR (cystic fibrosis) region in 13 species.  
		    <P>
		    We encourage you to explore these sequences with our tools. The Genome 
		    Browser zooms and scrolls over chromosomes, showing the work of annotators 
		    worldwide.  The Gene Sorter shows expression, homology and other 
		    information on groups of genes that can be related in many ways.  
		    Blat quickly maps your sequence to the genome.
		    The Table Browser 
		    provides convenient access to the underlying database.  
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
		        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		    </TD></TR></TABLE>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
<!--		    <FONT FACE="courier" SIZE=3><B>25 June 2004  - NCBI Human Build 35 released on Genome Browser</B></FONT><BR>
		    The latest human genome reference sequence (NCBI Build 35, 
		    May 2004) is now available as database hg17 in the UCSC 
		    Genome Browser and Blat server. 
		    There are ?? finished 
		    sequenced bases in the ordered and oriented portion of the
		    assembly, which is an increase of X percent, or approximately 
		    ? Mb, over the Build 34 assembly. 
		    <P>
		    Bulk downloads of the data are available via FTP at 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/hg17">ftp://hgdownload.cse.ucsc.edu/goldenPath/hg17</A> 
		    or through the Downloads link on this page. We recommend 
		    that you use FTP rather than HTML for the download of large 
		    or multiple files. 
		    <P>
		    We'd like to thank NCBI and the International Human Genome 
		    Sequencing Consortium for furnishing the data, and the UCSC 
		    team members who contributed to this release: Hiram Clawson, 
		    Terry Furey, Heather Trumbower, Donna Karolchik, 
		    Jim Kent and our sysadmin team Patrick Gavin, Jorge Garcia, 
		    and Paul Tatarsky. 
		    <P>
-->
		    <FONT FACE="courier" SIZE=3><B>16 July 2004  - Genome Browser for Build 33 Mouse Assembly</B></FONT><BR>
		    <P>
		    [NOTE: This news item is still under construction - waiting
		    on information from Deanna.]
		    <P>
		    The latest mouse assembly -- Build 33 from NCBI (UCSC version
		    mm5) -- is now
		    available via the UCSC Genome Browser and Blat server. This
		    assembly is a composite version in which phase 3 high 
		    throughput genome sequence (HTGS) was merged with the
		    Mouse Genome Sequencing Consortium v3 whole genome shotgun
		    assembly (MGSCv3). The assembly was driven by a combined
		    TPF created by Richa Agarwala that put the finished sequence
		    into the context of the MGSCv3 assembly. The assembly was
		    manually curated in a few places.
		    The UCSC mm5 assembly contains only the reference 
		    strain C57BL/6J.
		    <P>
		    The mm5 assembly includes ??? gigabases of sequence -- ??? 
		    Gb of which is finished -- contained in chromsomes 1-19, X, 
		    Y, M (mitochondrial DNA) and Un (unmapped clone contigs). We 
		    estimate that ????? percent of the mouse genome is present in 
		    the assembly. For more information about this version, see 
		    the NCBI assembly notes and Build 33 statistics. 
		    <P>
		    The mm5 sequence and annotation data may be downloaded from 
		    the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/mm5/">FTP 
		    server or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/mm5/">Downloads 
		    web page</A>. The mm5 annotation tracks were generated by UCSC
		    and collaborators worldwide. See the credits page for a 
		    detailed list of the organizations and individuals who 
		    contributed to the success of this release
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>14 July 2004  - Dog Genome Browser Released</B></FONT><BR>
		    <P>
		    UCSC has released a Genome Browser and Blat server on the 
		    July 2004  v1.0 dog genome sequenced and assembled by the 
		    Broad Institute of MIT and Harvard and Agencourt Bioscience. 
		    The whole genome shotgun (WGS) sequence is 
		    based on 7.6X coverage of the dog genome, assuming a WGS
		    assembly size of 2.4 Gb. The assembly has an N50 contig 
		    length of 123 kb and an N50 supercontig length of 41.6 Mb.
		    <P>
		    The dog genome, which contains approximately 2.5 billion 
		    base pairs, is similar in size to the genomes of humans and
		    other mammals. The boxer breed was selected for the initial 
		    sequencing effort, based on the lower variation rate in its 
		    genome relative to other breeds. In addition to the boxer,
		    samples from nine other dog breeds, four wolves and a coyote 
		    are being used to generate an initial set of single 
		    nucleotide polymorphisms (SNPs) to facilitate disease studies.
		    The SNPs should be available soon from
		    <A HREF="http://www.ncbi.nlm.nih.gov/SNP/" 
		    TARGET =_blank>dbSNP</A>.
		    <P>
<!--
		    For more information about the dog draft assembly, see the 
		    <A HREF="http://www.genome.gov/Media/" TARGET=_blank>NHGRI press 
		    release</A>.
-->
		    For more information about the dog draft assembly, see the 
		    <A HREF="http://www.genome.gov/12511476" TARGET=_blank>NHGRI 
		    press release</A>.
		    <P>
		    The dog sequence and annotation data can be downloaded from 
		    the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/canFam1/">FTP server</A>
		    or <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#dog">downloads</A>
		    page. These data have 
		    <A HREF="/goldenPath/credits.html#dog_use">specific 
		    conditions for use</A>.
		    <P>
		    Many thanks to the Broad Institute of MIT and Harvard, NHGRI,
		    Agencourt Bioscience, Children's Hospital Oakland Research 
		    Institute, Centre National de la Recherche Scientifique,
		    North Carolina State University, and Fred Hutchinson Cancer 
		    Research Center for their contributions to the sequencing, 
		    assembly, and mapping efforts. The initial canFam1 
		    annotation track set, generated by the UCSC Genome
		    Bioinformatics Group, will soon be
		    supplemented by annotations from collaborators 
		    worldwide. See the 
		    <A HREF="/goldenPath/credits.html#dog_credits">credits</A> page
		    for a detailed list of the organizations and individuals who 
		    contributed to the success of this release.
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>24 June 2004  - ENCODE Project Portal Released:</B></FONT>
		    We'd like to announce the release of UCSC Genome Browser 
		    features tailored to the ENCODE project community, including 
		    an 
		    <A HREF="/encode/">ENCODE-specific page</A> to 
		    highlight 
		    the ENCODE contributors and their work, guidelines for data 
		    submission, and a list of specific links to ENCODE regions 
		    in the Genome Browser. 
		    <A HREF="/goldenPath/newsarch.html#062404">Read more</A>.
		    <P>
		    <FONT FACE="courier" SIZE=3><B>10 June 2004  - Genome Browser Seminar in Boston Area on Aug. 10: </B></FONT>
		    The UCSC Bioinformatics Group announces a seminar and hands-on
		    workshop on the UCSC Genome Browser, presented by 
		    <A HREF="http://www.openhelix.com/" TARGET=_blank>OpenHelix</A>, 
		    a bioinformatics training, software testing and consulting 
		    company.
		    <A HREF="/goldenPath/newsarch.html#061004">Read more</A>.
    	            </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
	    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome Browser 
                    are freely available for academic, nonprofit, and personal use 
		    with the following conditions:
		    <UL>
		    <LI>Genome sequence data use restrictions are noted within the
		    species sections on the <A HREF="/goldenPath/credits.html">Credits</A> 
		    page.
		    <LI>Some annotation tracks contributed by external collaborators
		    contain proprietary data that have specific use restrictions. To
		    check for restrictions associated with a particular genome assembly, 
		    review the <em>database/README.txt</em> file in the assembly's 
		    downloads directory. 
		    </UL>
		    <P>
		    The Genome Browser and Blat software are free for academic,
		    nonprofit, and personal use. A license is required for commercial use.
		    See the <A HREF="/license/">Licenses</A> page for more information.
		    <P>
                    Program-driven use of this software is limited to a maximum of one 
                    hit every 15 seconds and no more than 5,000 hits per day.  
		    <P>
		    Questions or problems regarding the UCSC Genome Browser 
		    software, database, genome assemblies, or release cycles 
		    should be sent to 
		    <A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A>.
		    Messages sent to this address will be posted automatically 
		    on the genome mailing list, which is also archived on a 
		    public Web-accessible pipermail archive. NOTE: This archive may be 
		    indexed by non-UCSC sites such as Google. 
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>

        <BR>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	        
                    <FONT SIZE="4" ><A name="Tech"></A><B>&nbsp; Technical Information About the Assembled Sequence</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
	            <LI><A HREF="goldenPath/gbdDescriptions.html"><FONT SIZE="4">Description of the annotation database</FONT></A>
	            <LI><A HREF="goldenPath/datorg.html"><FONT SIZE="4">Data organization and format</FONT></A>
	            <LI><A HREF="goldenPath/term.html"><FONT SIZE="4">Terminology</FONT></A>
                    </UL>
                    </TD>
                    <TD WIDTH=15>&nbsp;</TD>
                 </TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	            <FONT SIZE="4" ><A NAME="Other"></A><B>&nbsp; Other Links</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
<!-- Commented out til this page is updated - very out of date
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/"><FONT SIZE="4">Human Genome Project at UCSC</FONT></A><FONT SIZE="4"></FONT>
-->
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/"><FONT SIZE="4">Center for Biomolecular Science and Engineering Home Page</FONT></A>
                    </UL>
                </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
</TD></TR></TABLE>

</BODY></HTML>

