<H2>Description</H2>
<P>
The Fugu whole genome shotgun assembly was constructed with the DOE
Joint Genome Institute (JGI)
assembler, JAZZ, paired end sequencing reads at a coverage of 5.7X
produced at JGI, Myriad Genetics, and Celera Genomics. All reads are
plasmid, cosmid, or BAC end-sequences, with the predominant coverage
coming from 2 kb insert plasmids. The assembly contains 8,597
scaffolds totaling 319 million base pairs; the largest 679 scaffolds
total 160 million base pairs. Gene modeling and analysis were
performed at JGI, Singapore, Cambridge, and the Institute for Systems
Biology.</P>

<H2>Methods</H2>
<P>The alignments were done 
with 
<A HREF="http://genome-test.cse.ucsc.edu/cgi-bin/hgBlat" TARGET=_blank>BLAT</A> 
in translated protein mode requiring 2 nearby 4-mer matches
to trigger a detailed alignment.  The $organism
genome was masked with 
<A HREF="http://repeatmasker.genome.washington.edu/cgi-bin/RepeatMasker"
TARGET=_blank>RepeatMasker</A> and 
<A HREF="http://tandem.biomath.mssm.edu/trf/trf.html TARGET=_blank">Tandem Repeats
Finder</A> 
before running BLAT. </P>

<H2>Credits</H2> 
<P>The 3.0 draft from 
<A HREF="http://genome.jgi-psf.org/fugu6/fugu6.info.html" TARGET=_blank>
http://genome.jgi-psf.org/fugu6/fugu6.info.html</A> was used in the
UCSC Fugu BLAT alignments.</P>
<P>
This data has been provided freely by the <A HREF="http://www.jgi.doe.gov/index.html" TARGET=_blank>DOE Joint Genome Institute</A>
for use in this publication/correspondence only.</P>
